diff -Nur orig/Components/Overlay/src/OgreFont.cpp mod/Components/Overlay/src/OgreFont.cpp
--- orig/Components/Overlay/src/OgreFont.cpp	2013-12-01 18:28:12.000000000 +0100
+++ mod/Components/Overlay/src/OgreFont.cpp	2016-12-19 22:15:48.009808091 +0100
@@ -297,8 +297,8 @@
 			{
 				FT_Load_Char( face, cp, FT_LOAD_RENDER );
 
-				if( ( 2 * ( face->glyph->bitmap.rows << 6 ) - face->glyph->metrics.horiBearingY ) > max_height )
-					max_height = ( 2 * ( face->glyph->bitmap.rows << 6 ) - face->glyph->metrics.horiBearingY );
+				if( ( 2 * ( ((int)face->glyph->bitmap.rows) << 6 ) - face->glyph->metrics.horiBearingY ) > max_height )
+					max_height = ( 2 * ( ((int)face->glyph->bitmap.rows) << 6 ) - face->glyph->metrics.horiBearingY );
 				if( face->glyph->metrics.horiBearingY > mTtfMaxBearingY )
 					mTtfMaxBearingY = static_cast<int>(face->glyph->metrics.horiBearingY);
 
