diff -Nur orig/masterpdfeditor5.desktop mod/masterpdfeditor5.desktop
--- orig/masterpdfeditor5.desktop	2021-08-02 09:19:24.000000000 +0200
+++ mod/masterpdfeditor5.desktop	2021-08-03 18:11:22.429383040 +0200
@@ -1,10 +1,10 @@
 [Desktop Entry]
 Name=Master PDF Editor 5
 Comment=Edit PDF files
-Exec=/opt/master-pdf-editor-5/masterpdfeditor5 %f
-Path=/opt/master-pdf-editor-5
+Exec=/usr/lib/master-pdf-editor-5/masterpdfeditor5 %f
+Path=/usr/lib/master-pdf-editor-5
 Terminal=false
-Icon=/opt/master-pdf-editor-5/masterpdfeditor5.png
+Icon=/usr/lib/master-pdf-editor-5/masterpdfeditor5.png
 Type=Application
 Categories=Office;Graphics;
-MimeType=application/pdf;application/x-bzpdf;application/x-gzpdf;
\ Nincs új sor a fájl végén
+MimeType=application/pdf;application/x-bzpdf;application/x-gzpdf;
