diff -Nur orig/net.code-industry.masterpdfeditor5.desktop mod/net.code-industry.masterpdfeditor5.desktop
--- orig/net.code-industry.masterpdfeditor5.desktop	2021-04-06 10:54:59.000000000 +0200
+++ mod/net.code-industry.masterpdfeditor5.desktop	2021-04-12 13:41:27.442384129 +0200
@@ -1,10 +1,10 @@
 [Desktop Entry]
 Name=Master PDF Editor 5
 Comment=Edit PDF files
-Exec=/opt/master-pdf-editor-5/masterpdfeditor5 %f
-Path=/opt/master-pdf-editor-5
+Exec=/usr/lib/master-pdf-editor-5/masterpdfeditor5 %f
+Path=/usr/lib/master-pdf-editor-5
 Terminal=false
-Icon=/opt/master-pdf-editor-5/masterpdfeditor5.png
+Icon=/usr/lib//master-pdf-editor-5/masterpdfeditor5.png
 Type=Application
 Categories=Office;Graphics;
-MimeType=application/pdf;application/x-bzpdf;application/x-gzpdf;
\ Nincs új sor a fájl végén
+MimeType=application/pdf;application/x-bzpdf;application/x-gzpdf;
