diff -Naur orig/promoe.desktop mod/promoe.desktop
--- orig/promoe.desktop	2010-02-21 20:31:06.000000000 +0100
+++ mod/promoe.desktop	2015-02-25 20:03:01.510782497 +0100
@@ -1,11 +1,8 @@
 [Desktop Entry]
-# Spec version, NOT application Version
-Version=1.0
 Name=Promoe
 GenericName=XMMS2 Client
-Categories=Qt;Application;AudioVideo;Audio;
+Categories=Qt;Player;AudioVideo;Audio;
 Exec=promoe
-Icon=xmms2
+Icon=promoe
 Terminal=false
 Type=Application
-
