diff -Naurd a/packaging/shared/efte.desktop b/packaging/shared/efte.desktop
--- a/packaging/shared/efte.desktop	2009-09-25 20:25:15.000000000 +0200
+++ b/packaging/shared/efte.desktop	2014-05-06 17:30:16.000000000 +0200
@@ -1,6 +1,5 @@
 [Desktop Entry]
 Name=eFTE
-Version=1.1
 GenericName=Text Editor
 GenericName[cz]=Textový editor
 GenericName[de]=Texteditor
@@ -21,6 +20,7 @@
 Comment[eo]=Rapida, etendabla tekstredaktilo por programado
 Comment[et]=Kiire, paindlik programmeerijate tekstiredaktor
 Comment[fi]=Nopea, laajennettava ohjelmoijien tekstieditori
+Comment[hu]=Programkód szerkesztő
 Comment[it]=Veloce, espandibile editor di programmi
 Comment[ja]=高速で、拡張可能なプログラマー向けテキストエディタ
 Comment[fr]=Editeur rapide et extensible, pour les programmeurs
