diff -Nur orig/pyproject.toml mod/pyproject.toml
--- orig/pyproject.toml	2025-10-15 19:14:39.000000000 +0200
+++ mod/pyproject.toml	2025-11-10 12:34:17.095516312 +0100
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools>=77.0.0", "setuptools-rust>=1.5.0"]
+requires = ["setuptools>=73.0.0", "setuptools-rust>=1.5.0"]
 build-backend = "setuptools.build_meta"
 
 [project]
@@ -7,7 +7,6 @@
 description = "Python implementation of the patiencediff algorithm"
 readme = "README.rst"
 maintainers = [{name = "Breezy Developers", email = "team@breezy-vcs.org"}]
-license = "GPL-2.0-or-later"
 classifiers = [
     "Development Status :: 6 - Mature",
     "Programming Language :: Python :: 3.9",
