diff -Naur a/settings/src/Makefile b/settings/src/Makefile
--- a/settings/src/Makefile	2023-02-24 04:51:35.000000000 +0100
+++ b/settings/src/Makefile	2023-04-05 21:06:49.329756825 +0200
@@ -255,18 +255,14 @@
 
 .PHONY: all install NVIDIA_SETTINGS_install clean clobber
 
-all: $(NVIDIA_SETTINGS) $(GTK2LIB) $(GTK3LIB)
+all: $(NVIDIA_SETTINGS) $(GTK3LIB)
 
 install: NVIDIA_SETTINGS_install NVIDIA_GTKLIB_install
 
-NVIDIA_GTKLIB_install: $(GTK2LIB) $(GTK3LIB)
+NVIDIA_GTKLIB_install: $(GTK3LIB)
 	$(MKDIR) $(LIBDIR)
-	$(INSTALL) $(INSTALL_LIB_ARGS) $(GTK2LIB) \
-	    $(LIBDIR)/$(GTK2LIB_SONAME)
-ifdef BUILD_GTK3LIB
 	$(INSTALL) $(INSTALL_LIB_ARGS) $(GTK3LIB) \
 	    $(LIBDIR)/$(GTK3LIB_SONAME)
-endif
 
 NVIDIA_SETTINGS_install: $(NVIDIA_SETTINGS)
 	$(MKDIR) $(BINDIR)
