diff -Nur orig/gnome-help/Makefile.am mod/gnome-help/Makefile.am
--- orig/gnome-help/Makefile.am	2018-10-28 22:10:13.000000000 +0100
+++ mod/gnome-help/Makefile.am	2019-03-14 09:06:07.236529176 +0100
@@ -405,4 +405,4 @@
 	wacom-stylus.page \
 	wacom.page
 
-HELP_LINGUAS = as ca cs da de el fi fr es gl gu he hi hr hu id it ja kn lv mr nl pa pl pt pt_BR ro ru sl sr sr@latin sv te vi ta zh_CN
+HELP_LINGUAS = cs de fi fr es hr hu it pl pt ro ru sl sr
diff -Nur orig/Makefile.am mod/Makefile.am
--- orig/Makefile.am	2012-08-14 16:52:48.000000000 +0200
+++ mod/Makefile.am	2019-03-14 09:03:15.976733030 +0100
@@ -1 +1 @@
-SUBDIRS = gnome-help system-admin-guide
+SUBDIRS = gnome-help
