diff -Naur a/Makefile b/Makefile
--- a/Makefile	2022-02-18 14:03:36.000000000 +0100
+++ b/Makefile	2022-05-09 00:28:56.157210787 +0200
@@ -2,7 +2,7 @@
 
 # Where to install
 #
-prefix = /usr/local
+prefix = /usr
 
 .PHONY: all check install clean dist
 
@@ -28,8 +28,8 @@
 cppp.o    : cppp.c gen.h types.h unixisms.h error.h symset.h ppproc.h
 
 install:
-	cp ./cppp $(prefix)/bin/.
-	cp ./cppp.1 $(prefix)/share/man/man1/.
+	cp ./cppp $(DESTDIR)$(prefix)/bin/.
+	cp ./cppp.1 $(DESTDIR)$(prefix)/share/man/man1/.
 
 test: cppp
 	./tests/testall
