diff -Naur xkeyboard-config-0.8.orig/symbols/hu xkeyboard-config-0.8/symbols/hu
--- xkeyboard-config-0.8.orig/symbols/hu	2005-12-21 01:40:06.000000000 +0100
+++ xkeyboard-config-0.8/symbols/hu	2006-06-09 16:16:55.000000000 +0200
@@ -373,18 +373,18 @@
 // Without dead keys support
 hidden partial alphanumeric_keys
 xkb_symbols "def_nodead" {
-    key <AE01>  { [            1,   apostrophe,        asciitilde                  ] };
-    key <AE02>  { [            2,     quotedbl,             caron                  ] };
-    key <AE03>  { [            3,         plus,       asciicircum                  ] };
-    key <AE04>  { [            4,       exclam,             breve                  ] };
-    key <AE05>  { [            5,      percent,            degree                  ] };
-    key <AE06>  { [            6,        slash,            ogonek                  ] };
-    key <AE07>  { [            7,        equal,             grave                  ] };
-    key <AE08>  { [            8,    parenleft,          abovedot                  ] };
-    key <AE09>  { [            9,   parenright,             acute                  ] };
-    key <AE10>  { [   odiaeresis,   Odiaeresis,       doubleacute                  ] };
-    key <AE11>  { [   udiaeresis,   Udiaeresis,         diaeresis                  ] };
-    key <AE12>  { [       oacute,       Oacute,           cedilla                  ] };
+    key <AE01>  { [            1,   apostrophe,        asciitilde,      dead_tilde ] };
+    key <AE02>  { [            2,     quotedbl,             caron,      dead_caron ] };
+    key <AE03>  { [            3,         plus,       asciicircum, dead_circumflex ] };
+    key <AE04>  { [            4,       exclam,             breve,      dead_breve ] };
+    key <AE05>  { [            5,      percent,            degree,  dead_abovering ] };
+    key <AE06>  { [            6,        slash,            ogonek,     dead_ogonek ] };
+    key <AE07>  { [            7,        equal,             grave,      dead_grave ] };
+    key <AE08>  { [            8,    parenleft,          abovedot,   dead_abovedot ] };
+    key <AE09>  { [            9,   parenright,             acute,      dead_acute ] };
+    key <AE10>  { [   odiaeresis,   Odiaeresis,       doubleacute,dead_doubleacute ] };
+    key <AE11>  { [   udiaeresis,   Udiaeresis,         diaeresis,  dead_diaeresis ] };
+    key <AE12>  { [       oacute,       Oacute,           cedilla,    dead_cedilla ] };
 };
 
 
