diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2021-03-19 13:59:31.000000000 +0100
+++ patched/configure.ac	2021-04-22 11:29:08.896732495 +0200
@@ -95,7 +95,7 @@
 	DO_CHECK=$enableval, DO_CHECK=yes)
 
 if test	"$DO_CHECK" = "yes"; then
-	accepted_versions="1.2 1.4 2.0 2.1 2.2"
+	accepted_versions="1.2 1.4 2.0 2.1 2.2 2.3 2.4"
 	AC_PATH_PROGS(GNUPG, [gpg gpg2], no)
 	AC_DEFINE_UNQUOTED(GNUPG, "$GNUPG", [Path to gpg executable.])
 	ok="no"
