diff -Nur orig/configure.ac mod/configure.ac
--- orig/configure.ac	2018-01-07 12:50:19.000000000 +0100
+++ mod/configure.ac	2018-01-07 13:05:50.004137464 +0100
@@ -31,7 +31,6 @@
 AC_PROG_CXX
 AC_PROG_INTLTOOL([0.35.0])
 AM_PROG_LIBTOOL
-GTK_DOC_CHECK([1.9])
 
 dnl Get the off_t size right
 AC_SYS_LARGEFILE
@@ -348,9 +347,6 @@
 pixmaps/48x48/Makefile
 pixmaps/scalable/Makefile
 daemon/Makefile
-reference/Makefile
-reference/libcryptui/Makefile
-reference/libcryptui/version.xml
 ])
 
 dnl ****************************************************************************
diff -Nur orig/Makefile.am mod/Makefile.am
--- orig/Makefile.am	2018-01-07 12:50:19.000000000 +0100
+++ mod/Makefile.am	2018-01-07 12:58:02.176942298 +0100
@@ -7,8 +7,7 @@
 	po \
 	data \
 	pixmaps \
-	daemon \
-	reference
+	daemon
 
 EXTRA_DIST = \
 	AUTHORS \
