--- p.orig/Makefile	2003-07-17 17:05:04.000000000 +0200
+++ p/Makefile	2009-05-16 11:03:49.000000000 +0200
@@ -17,8 +17,8 @@
 	rm -f $(DATADIR)/graphics/*.xcf
 	rm -f $(DATADIR)/sounds/*.orig
 
-	mkdir -p $(DESTDIR)/usr/games
-	cp pathological $(DESTDIR)/usr/games/
+	mkdir -p $(DESTDIR)/usr/bin
+	cp pathological $(DESTDIR)/usr/bin/
 
 	mkdir -p $(DESTDIR)/usr/lib/pathological/bin
 	cp write-highscores $(DESTDIR)/usr/lib/pathological/bin
