diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2009-10-23 20:16:44.000000000 +0200
+++ patched/configure.ac	2017-01-28 04:15:39.904533008 +0100
@@ -3,7 +3,7 @@
 AC_INIT([zbar], [0.10], [spadix@users.sourceforge.net])
 AC_CONFIG_AUX_DIR(config)
 AC_CONFIG_MACRO_DIR(config)
-AM_INIT_AUTOMAKE([1.10 -Wall -Werror foreign subdir-objects std-options dist-bzip2])
+AM_INIT_AUTOMAKE([1.10 -Wall foreign subdir-objects std-options dist-bzip2])
 AC_CONFIG_HEADERS([include/config.h])
 AC_CONFIG_SRCDIR(zbar/scanner.c)
 LT_PREREQ([2.2])
