diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2017-02-15 01:07:28.000000000 +0100
+++ patched/configure.ac	2017-02-15 01:18:41.759211201 +0100
@@ -324,7 +324,7 @@
 dnl double check ImageMagick install (bug #2582232)
       CPPFLAGS_save="$CPPFLAGS"
       CPPFLAGS="$CPPFLAGS $MAGICK_CFLAGS"
-      AC_CHECK_HEADER([wand/MagickWand.h],
+      AC_CHECK_HEADER([MagickWand/MagickWand.h],
         [have_IM="yes"],
         [have_IM="broken"])
       CPPFLAGS="$CPPFLAGS_save"])])
diff -Naur orig/zbarimg/zbarimg.c patched/zbarimg/zbarimg.c
--- orig/zbarimg/zbarimg.c	2017-02-15 01:07:28.000000000 +0100
+++ patched/zbarimg/zbarimg.c	2017-02-15 01:19:35.839843666 +0100
@@ -44,7 +44,7 @@
 #endif
 
 #ifdef HAVE_IMAGEMAGICK
-# include <wand/MagickWand.h>
+#include <MagickWand/MagickWand.h>
 
 /* ImageMagick frequently changes API names - just use the original
  * (more stable?) names to match GraphicsMagick
