diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2017-02-15 01:07:28.000000000 +0100
+++ patched/configure.ac	2017-02-15 01:25:37.616110292 +0100
@@ -182,7 +182,7 @@
 AS_IF([test "x$enable_video" = "xno"],
   [],
   [test "x$win32" = "xno"],
-  [AC_CHECK_HEADERS([linux/videodev.h], [have_v4l1="yes"])
+  [AC_CHECK_HEADERS([libv4l1-videodev.h], [have_v4l1="yes"])
    AC_CHECK_HEADERS([linux/videodev2.h], [have_v4l2="yes"])
    AS_IF([test "x$have_v4l2" = "xno" && test "x$have_v4l1" = "xno"],
      [AC_MSG_FAILURE([test for video support failed!
diff -Naur orig/zbar/video/v4l1.c patched/zbar/video/v4l1.c
--- orig/zbar/video/v4l1.c	2017-02-15 01:07:28.000000000 +0100
+++ patched/zbar/video/v4l1.c	2017-02-15 01:26:25.488673644 +0100
@@ -41,7 +41,7 @@
 #ifdef HAVE_SYS_MMAN_H
 # include <sys/mman.h>
 #endif
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 
 #include "video.h"
 #include "image.h"
