diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2017-02-15 01:07:28.000000000 +0100
+++ patched/configure.ac	2017-02-15 01:37:48.420732476 +0100
@@ -47,7 +47,7 @@
   [Library revision])
 
 AM_CPPFLAGS="-I$srcdir/include"
-AM_CFLAGS="-Wall -Wno-parentheses -Werror"
+AM_CFLAGS="-Wall -Wno-parentheses -fPIC"
 AM_CXXFLAGS="$AM_CFLAGS"
 AC_SUBST([AM_CPPFLAGS])
 AC_SUBST([AM_CFLAGS])
@@ -467,17 +467,18 @@
 dnl Qt
 AC_ARG_WITH([qt],
   [AS_HELP_STRING([--without-qt],
-    [disable support for Qt4 widget])],
+    [disable support for Qt5 widget])],
   [],
   [with_qt="yes"])
 
 AC_ARG_VAR([MOC], [full path to Qt moc program])
 
 AS_IF([test "x$with_qt" != "xno"],
-  [PKG_CHECK_MODULES([QT], [QtCore >= 4 QtGui >= 4])
-   MOC=`$PKG_CONFIG QtGui --variable=moc_location`
+  [PKG_CHECK_MODULES([QT], [Qt5Gui Qt5Widgets Qt5X11Extras])
+   MOCPATH=`$PKG_CONFIG Qt5Core --variable=host_bins`
+   MOC=$MOCPATH/moc
    AC_MSG_NOTICE([using moc from $MOC])
-   QT_VERSION=`$PKG_CONFIG QtGui --modversion`
+   QT_VERSION=`$PKG_CONFIG Qt5Gui --modversion`
    AC_MSG_NOTICE([using Qt version $QT_VERSION])])
 
 AM_CONDITIONAL([HAVE_QT], [test "x$with_qt" = "xyes"])
@@ -585,9 +586,9 @@
   [echo "        => the GTK+ widget will *NOT* be built"],
   [AS_IF([test "x$with_python" != "xyes"],
      [echo "        => the PyGTK widget wrapper will *NOT* be built"])])
-echo "Qt4               --with-qt=$with_qt"
+echo "Qt5               --with-qt=$with_qt"
 AS_IF([test "x$with_qt" != "xyes"],
-  [echo "        => the Qt4 widget will *NOT* be built"])
+  [echo "        => the Qt5 widget will *NOT* be built"])
 echo "Java              --with-java=$with_java"
 AS_IF([test "x$with_java" != "xyes"],
   [echo "        => the Java interface will *NOT* be built"])
diff -Naur orig/include/zbar/QZBar.h patched/include/zbar/QZBar.h
--- orig/include/zbar/QZBar.h	2017-02-15 01:07:28.000000000 +0100
+++ patched/include/zbar/QZBar.h	2017-02-15 01:33:41.361788121 +0100
@@ -26,7 +26,7 @@
 /// @file
 /// Barcode Reader Qt4 Widget
 
-#include <qwidget.h>
+#include <QtWidgets>
 
 namespace zbar {
 
diff -Naur orig/include/zbar/QZBarImage.h patched/include/zbar/QZBarImage.h
--- orig/include/zbar/QZBarImage.h	2017-02-15 01:07:28.000000000 +0100
+++ patched/include/zbar/QZBarImage.h	2017-02-15 01:39:33.774014839 +0100
@@ -26,7 +26,7 @@
 /// @file
 /// QImage to Image type conversion wrapper
 
-#include <qimage.h>
+#include <QImage>
 #include <zbar.h>
 
 namespace zbar {
@@ -54,7 +54,7 @@
         unsigned height = qimg.height();
         set_size(width, height);
         set_format(zbar_fourcc('B','G','R','4'));
-        unsigned long datalen = qimg.numBytes();
+        unsigned long datalen = qimg.byteCount();
         set_data(qimg.bits(), datalen);
 
         if((width * 4 != bpl) ||
diff -Naur orig/qt/QZBar.cpp patched/qt/QZBar.cpp
--- orig/qt/QZBar.cpp	2017-02-15 01:07:28.000000000 +0100
+++ patched/qt/QZBar.cpp	2017-02-15 01:33:41.365788168 +0100
@@ -21,9 +21,9 @@
 //  http://sourceforge.net/projects/zbar
 //------------------------------------------------------------------------
 
-#include <qevent.h>
-#include <qurl.h>
-#include <qx11info_x11.h>
+#include <QEvent>
+#include <QUrl>
+#include <QX11Info>
 #include <zbar/QZBar.h>
 #include "QZBarThread.h"
 
@@ -49,7 +49,7 @@
 
     thread = new QZBarThread;
     if(testAttribute(Qt::WA_WState_Created)) {
-        thread->window.attach(x11Info().display(), winId());
+        thread->window.attach(QX11Info::display(), winId());
         _attached = 1;
     }
     connect(thread, SIGNAL(videoOpened(bool)),
@@ -204,7 +204,7 @@
     try {
         QMutexLocker locker(&thread->mutex);
         if(event->type() == QEvent::ParentChange)
-            thread->window.attach(x11Info().display(), winId());
+            thread->window.attach(QX11Info::display(), winId());
     }
     catch(Exception) { /* ignore (FIXME do something w/error) */ }
 }
@@ -215,7 +215,7 @@
         return;
 
     try {
-        thread->window.attach(x11Info().display(), winId());
+        thread->window.attach(QX11Info::display(), winId());
         _attached = 1;
 
         _videoEnabled = !_videoDevice.isEmpty();
diff -Naur orig/zbar-qt.pc.in patched/zbar-qt.pc.in
--- orig/zbar-qt.pc.in	2017-02-15 01:07:28.000000000 +0100
+++ patched/zbar-qt.pc.in	2017-02-15 01:33:41.365788168 +0100
@@ -4,9 +4,9 @@
 includedir=@includedir@
 
 Name: zbar-qt
-Description: bar code scanning and decoding Qt4 widget
+Description: bar code scanning and decoding Qt5 widget
 URL: http://zbar.sourceforge.net
 Version: @VERSION@
-Requires: zbar, QtCore >= 4, QtGui >= 4
+Requires: zbar, Qt5Gui, Qt5Widgets, Qt5X11Extras
 Libs: -L${libdir} -lzbarqt
 Cflags: -I${includedir}
