diff -Naur bzflag-2.4.18-orig/src/platform/BzfDisplay.cxx bzflag-2.4.18/src/platform/BzfDisplay.cxx
--- bzflag-2.4.18-orig/src/platform/BzfDisplay.cxx	2018-10-18 12:39:15.000000000 +0200
+++ bzflag-2.4.18/src/platform/BzfDisplay.cxx	2018-11-01 11:13:02.412276474 +0100
@@ -139,9 +139,20 @@
 
 int         BzfDisplay::findResolution(const char* name) const
 {
-    for (int i = 0; i < numResolutions; i++)
-        if (strcmp(name, resolutions[i]->name) == 0)
+  for (int i = 0; i < numResolutions; i++) {
+    // only compare up to first space character
+    const char *cp1, *cp2;
+    int i1, i2;
+
+    cp1 = strstr (name, " ");
+    cp2 = strstr (resolutions[i]->name, " ");
+
+    i1 = cp1 ? (cp1 - name) : strlen (name);
+    i2 = cp2 ? (cp2 - resolutions[i]->name) : strlen (resolutions[i]->name);
+
+    if ((i1 == i2) && (strncmp(name, resolutions[i]->name, i1 + 1) == 0))
             return i;
+    }
     return -1;
 }
 
