diff -Naur orig/data/freshwrapper.conf.example patched/data/freshwrapper.conf.example
--- orig/data/freshwrapper.conf.example	2017-05-31 00:16:14.134739579 +0200
+++ patched/data/freshwrapper.conf.example	2017-06-25 23:58:52.280226027 +0200
@@ -37,7 +37,7 @@
 # a number of locations where it could be. Usually that's enough, but if
 # not, you should manually enter the right path. Multiple paths could
 # be specified, separated by colon.
-#pepperflash_path = "/opt/google/chrome/PepperFlash/libpepflashplayer.so"
+pepperflash_path = "/usr/lib/adobe-flashplugin/libpepflashplayer.so"
 
 # "Command-line" arguments for Flash
 flash_command_line = "enable_hw_video_decode=1,enable_stagevideo_auto=1"
