diff -Naur orig/src/modules/gtk2/producer_pango.c mod/src/modules/gtk2/producer_pango.c
--- orig/src/modules/gtk2/producer_pango.c	2015-07-30 01:04:39.000000000 +0200
+++ mod/src/modules/gtk2/producer_pango.c	2015-11-30 17:46:16.769773000 +0100
@@ -25,7 +25,7 @@
 #include <string.h>
 #include <gdk-pixbuf/gdk-pixbuf.h>
 #include <pango/pangoft2.h>
-#include <freetype.h>
+#include <freetype2/freetype/freetype.h>
 #include <iconv.h>
 #include <pthread.h>
 #include <ctype.h>
