diff -Naurd a/Makefile b/Makefile
--- a/Makefile	2012-11-24 23:45:42.000000000 +0100
+++ b/Makefile	2012-11-25 18:06:20.000000000 +0100
@@ -85,7 +85,7 @@
 	install -m 755 -o 0 -g 0 filterator $(ROOT)/usr/bin/filterator
 	install -m 755 -o 0 -g 0 -d $(ROOT)/usr/share/man/man1/
 	install -m 755 -o 0 -g 0 comparator.1 $(ROOT)/usr/share/man/man1/comparator.1
-	python setup.py install
+#	python setup.py install
 
 uninstall:
 	rm -f ${ROOT}/usr/bin/comparator 
