diff -Nur orig/pcmanfm/pcmanfm.conf.in mod/pcmanfm/pcmanfm.conf.in
--- orig/pcmanfm/pcmanfm.conf.in	2016-11-21 23:47:05.000000000 +0100
+++ mod/pcmanfm/pcmanfm.conf.in	2016-12-10 11:36:44.550875281 +0100
@@ -4,7 +4,7 @@
 
 [desktop]
 wallpaper_mode=crop
-wallpaper=@prefix@/share/lxde/wallpapers/lxde_blue.jpg
+wallpaper=@prefix@/share/backgrounds/UHU/uhu_linux_ubk.jpg
 desktop_bg=#000000
 desktop_fg=#ffffff
 desktop_shadow=#000000
