diff -Naur orig/src/3ds/SoStream.cpp patched/src/3ds/SoStream.cpp
--- orig/src/3ds/SoStream.cpp	2010-03-02 14:20:09.134739579 +0100
+++ patched/src/3ds/SoStream.cpp	2017-10-03 02:28:19.112694154 +0200
@@ -176,7 +176,7 @@
   if (!gotNum) { setBadBit(); return FALSE; } \
  \
   char *ce; \
-  s = '\0'; \
+  s = NULL; \
   _convertType_ tempVal = _convertFunc_(buf, &ce, 0); \
  \
   if (ce != s) \
@@ -273,7 +273,7 @@
 gotAll: \
   \
   char *ce; \
-  s = '\0'; \
+  s = NULL; \
   double tempVal = _convertFunc_(buf, &ce); \
  \
   if (ce != s) \
 
