diff -Naur orig/misc/syntax/Syntax.in patched/misc/syntax/Syntax.in
--- orig/misc/syntax/Syntax.in	2019-06-16 19:49:31.000000000 +0200
+++ patched/misc/syntax/Syntax.in	2019-06-30 21:22:26.499875682 +0200
@@ -31,7 +31,7 @@
 file ..\*\\.e$ Eiffel\sSource\sFile
 include eiffel.syntax
 
-file ..\*\\.(diff|rej|patch)$ Diff\sOutput ^(diff|Index:)\s
+file ..\*\\.(diff|rej|patch|skip)$ Diff\sOutput ^(diff|Index:)\s
 include diff.syntax
 
 file ..\*\\.lsm$ LSM\sFile
