diff -Naur a/misc/syntax/Syntax.in b/misc/syntax/Syntax.in
--- a/misc/syntax/Syntax.in	2023-08-06 15:26:05.000000000 +0200
+++ b/misc/syntax/Syntax.in	2023-08-18 17:27:36.976728940 +0200
@@ -45,7 +45,7 @@
 file ..\*\\.e$ Eiffel\sSource\sFile
 include eiffel.syntax
 
-file ..\*\\.(diff|rej|patch)|COMMIT_EDITMSG$ Diff\sOutput ^(diff|Index:)\s
+file ..\*\\.(diff|rej|patch|skip)|COMMIT_EDITMSG$ Diff\sOutput ^(diff|Index:)\s
 include diff.syntax
 
 file ..\*\\.lsm$ LSM\sFile
