diff -Naur orig/misc/syntax/Syntax.in patched/misc/syntax/Syntax.in
--- orig/misc/syntax/Syntax.in	2012-12-10 22:04:32.000000000 +0100
+++ patched/misc/syntax/Syntax.in	2012-12-29 11:26:17.699567005 +0100
@@ -31,7 +31,7 @@
 file ..\*\\.(e)$ Eiffel\sSource\sFile
 include eiffel.syntax
 
-file ..\*\\.(diff|rej|patch)$ Diff\sOutput ^(diff|Index:)\s
+file ..\*\\.(diff|rej|patch|skip)$ Diff\sOutput ^(diff|Index:)\s
 include diff.syntax
 
 file ..\*\\.lsm$ LSM\sFile
