diff -Naur orig/misc/syntax/php.syntax patched/misc/syntax/php.syntax
--- orig/misc/syntax/php.syntax	2019-06-16 19:49:31.000000000 +0200
+++ patched/misc/syntax/php.syntax	2019-06-30 21:23:38.000749106 +0200
@@ -3082,6 +3082,8 @@
 
     keyword wholeright $+ brightgreen
 
+    keyword " green
+    keyword '*' brightgreen
     keyword /\* brown
     keyword \*/ brown
     keyword // brown
@@ -3118,7 +3120,7 @@
     keyword : brightcyan
     keyword ; brightmagenta
 
-context " " green
+context exclusive " " green
     spellcheck
     keyword \\" brightgreen
     keyword \\\n brightgreen
@@ -3134,10 +3136,6 @@
     keyword \\' brightgreen
     keyword \\\{abtnvfr\} brightgreen
 
-context ' ' brightgreen
-    spellcheck
-    keyword \\' brightcyan
-
 context exclusive <? ?> cyan
     keyword whole &*; brightgreen
 context <html > cyan
