diff -Naur flac-1.2.1-orig/examples/cpp/encode/file/main.cpp flac-1.2.1/examples/cpp/encode/file/main.cpp
--- flac-1.2.1-orig/examples/cpp/encode/file/main.cpp	2007-09-13 17:58:03.000000000 +0200
+++ flac-1.2.1/examples/cpp/encode/file/main.cpp	2009-11-30 09:44:09.000000000 +0100
@@ -28,6 +28,7 @@
 #  include <config.h>
 #endif
 
+#include <string.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include "FLAC++/metadata.h"
