diff -urNad python-uniconvertor-1.1.1~/src/uniconv python-uniconvertor-1.1.1/src/uniconv
--- python-uniconvertor-1.1.1~/src/uniconv	2008-03-13 00:10:44.000000000 +0100
+++ python-uniconvertor-1.1.1/src/uniconv	2008-03-15 15:11:01.000000000 +0100
@@ -19,5 +19,4 @@
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
 #
 
-
-python -c "from uniconvertor import uniconv; uniconv();" "$@"
\ No newline at end of file
+exec python -c "from uniconvertor import uniconv; uniconv();" "$@"
