diff -urNad uniconvertor-1.1.1~/src/__init__.py uniconvertor-1.1.1/src/__init__.py
--- uniconvertor-1.1.1~/src/__init__.py	2008-03-13 00:10:44.000000000 +0100
+++ uniconvertor-1.1.1/src/__init__.py	2008-03-15 11:38:34.000000000 +0100
@@ -7,7 +7,7 @@
 # For more info see COPYRIGHTS file in root directory.
 
 '''
-USAGE: uniconv [OPTIONS] [INPUT FILE] [OUTPUT FILE]
+USAGE: uniconvertor [OPTIONS] [INPUT FILE] [OUTPUT FILE]
 
 Converts one vector graphics format to another using sK1 engine.
 sK1 Team (http://sk1project.org), copyright (C) 2007-2009 by Igor E. Novikov
@@ -38,7 +38,7 @@
      PS  - PostScript
 	 PLT - HPGL for cutting plotter files
 
-Example: uniconv drawing.cdr drawing.svg\n
+Example: uniconvertor drawing.cdr drawing.svg\n
 '''
 
 import sys, os, string
