--- UniConvertor-1.1.0/src/app/__init__.py.simplify	2007-12-20 14:33:27.000000000 +0200
+++ UniConvertor-1.1.0/src/app/__init__.py	2008-01-11 21:34:18.000000000 +0200
@@ -9,28 +9,20 @@
 
 import os, sys, string
 
-path_separator='/'
-if os.name=='nt':
-	path_separator='\\'
-
 _pkgdir = __path__[0]
 
-temp=string.split(_pkgdir,path_separator)
-temp.remove(temp[-1])
-_parentdir=string.join(temp,path_separator)
+_parentdir, dummy = os.path.split(_pkgdir)
 
 sKVersion = string.strip(open(os.path.join(_pkgdir, 'VERSION')).read())
 
 for _dir in ('modules', 'Base'):
 	__path__.insert(0, os.path.join(_pkgdir, _dir))
 
-dir = os.path.join(_parentdir, 'app')
-dir = os.path.join(dir, 'modules')
+dir = os.path.join(_parentdir, 'app', 'modules')
 if os.path.isdir(dir):
 	sys.path.insert(1, dir)
 
-message_dir = os.path.join(sys.path[0], 'Resources')
-message_dir = os.path.join(message_dir, 'Messages')
+message_dir = os.path.join(sys.path[0], 'Resources', 'Messages')
 try:
 	from intl import gettext, dgettext, bindtextdomain
 	import intl, locale
