diff -Naur a/documentation/Makefile b/documentation/Makefile
--- a/documentation/Makefile	2021-11-20 19:10:34.000000000 +0100
+++ b/documentation/Makefile	2022-05-09 01:52:39.499296746 +0200
@@ -86,11 +86,6 @@
 	  $(INSTALL_DATA) fltk.pdf $(DESTDIR)$(docdir); \
 	fi
 	echo "Installing man pages in $(DESTDIR)$(mandir) ..."
-	-$(INSTALL_DIR) $(DESTDIR)$(mandir)/cat1
-	$(INSTALL_MAN) $(SRC_DOCDIR)/fluid.$(CAT1EXT) $(DESTDIR)$(mandir)/cat1
-	$(INSTALL_MAN) $(SRC_DOCDIR)/fltk-config.$(CAT1EXT) $(DESTDIR)$(mandir)/cat1
-	-$(INSTALL_DIR) $(DESTDIR)$(mandir)/cat3
-	$(INSTALL_MAN) $(SRC_DOCDIR)/fltk.$(CAT3EXT) $(DESTDIR)$(mandir)/cat3
 	-$(INSTALL_DIR) $(DESTDIR)$(mandir)/man1
 	$(INSTALL_MAN) $(SRC_DOCDIR)/fluid.man $(DESTDIR)$(mandir)/man1/fluid.1
 	$(INSTALL_MAN) $(SRC_DOCDIR)/fltk-config.man $(DESTDIR)$(mandir)/man1/fltk-config.1
@@ -98,10 +93,6 @@
 	$(INSTALL_MAN) $(SRC_DOCDIR)/fltk.man $(DESTDIR)$(mandir)/man3/fltk.3
 
 install-linux install-osx:
-	-$(INSTALL_DIR) $(DESTDIR)$(mandir)/cat6
-	$(INSTALL_MAN) $(SRC_DOCDIR)/blocks.$(CAT6EXT) $(DESTDIR)$(mandir)/cat6
-	$(INSTALL_MAN) $(SRC_DOCDIR)/checkers.$(CAT6EXT) $(DESTDIR)$(mandir)/cat6
-	$(INSTALL_MAN) $(SRC_DOCDIR)/sudoku.$(CAT6EXT) $(DESTDIR)$(mandir)/cat6
 	-$(INSTALL_DIR) $(DESTDIR)$(mandir)/man6
 	$(INSTALL_MAN) $(SRC_DOCDIR)/blocks.man $(DESTDIR)$(mandir)/man6/blocks.6
 	$(INSTALL_MAN) $(SRC_DOCDIR)/checkers.man $(DESTDIR)$(mandir)/man6/checkers.6
diff -Naur a/fluid/fluid.desktop b/fluid/fluid.desktop
--- a/fluid/fluid.desktop	2021-11-20 19:10:34.000000000 +0100
+++ b/fluid/fluid.desktop	2022-05-09 01:52:39.499296746 +0200
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Name=FLUID
 Comment=FLTK GUI Designer
+Comment[hu]=FLTK felhasználóifelület-szerkesztő
 TryExec=fluid
 Exec=fluid %F
 Icon=fluid
