diff -Naur a/fltk-config.in b/fltk-config.in
--- a/fltk-config.in	2021-11-20 19:10:34.000000000 +0100
+++ b/fltk-config.in	2022-05-09 02:01:21.600734124 +0200
@@ -46,6 +46,7 @@
 CXXFLAGS="@CXXFLAGS@ @LARGEFILE@ @PTHREAD_FLAGS@"
 LDFLAGS="@LDFLAGS@"
 LDLIBS="@LIBS@"
+LDSTATIC=
 OPTIM="@OPTIM@"
 CAIROFLAGS="@CAIROFLAGS@"
 
@@ -53,21 +54,6 @@
 if test -f "$selfdir/FL/Fl_Window.H"; then
 	includedir="$selfdir"
 	libdir="$selfdir/lib"
-
-	if test -f "$libdir/libfltk_jpeg.a"; then
-		CFLAGS="-I$includedir/jpeg $CFLAGS"
-		CXXFLAGS="-I$includedir/jpeg $CXXFLAGS"
-	fi
-
-	if test -f "$libdir/libfltk_z.a"; then
-		CFLAGS="-I$includedir/zlib $CFLAGS"
-		CXXFLAGS="-I$includedir/zlib $CXXFLAGS"
-	fi
-
-	if test -f "$libdir/libfltk_png.a"; then
-		CFLAGS="-I$includedir/png $CFLAGS"
-		CXXFLAGS="-I$includedir/png $CXXFLAGS"
-	fi
 fi
 
 if test -d $includedir/FL/images; then
@@ -75,11 +61,6 @@
 	CXXFLAGS="-I$includedir/FL/images $CXXFLAGS"
 fi
 
-if test -f "$libdir/libfltk_cairo.a"; then
-	CFLAGS="$CAIROFLAGS $CFLAGS"
-	CXXFLAGS="$CAIROFLAGS $CXXFLAGS"
-fi
-
 # libraries to link with:
 LIBNAME="@LIBNAME@"
 DSONAME="@DSONAME@"
@@ -233,29 +214,23 @@
 fi
 
 # Calculate needed libraries
-LDSTATIC="$libdir/libfltk.a $LDLIBS"
 LDLIBS="-lfltk$SHAREDSUFFIX $LDLIBS"
 
 if test x$use_forms = xyes; then
     LDLIBS="-lfltk_forms$SHAREDSUFFIX $LDLIBS"
-    LDSTATIC="$libdir/libfltk_forms.a $LDSTATIC"
 fi
 if test x$use_gl = xyes; then
     LDLIBS="-lfltk_gl$SHAREDSUFFIX @GLLIBS@ $LDLIBS"
-    LDSTATIC="$libdir/libfltk_gl.a @GLLIBS@ $LDSTATIC"
 fi
 if test x$use_images = xyes; then
     LDLIBS="-lfltk_images$SHAREDSUFFIX $IMAGELIBS $LDLIBS"
-    LDSTATIC="$libdir/libfltk_images.a $STATICIMAGELIBS $LDSTATIC"
 fi
 
 if test x$use_cairo = xyes; then
     LDLIBS="-lfltk_cairo$SHAREDSUFFIX $CAIROLIBS $LDLIBS"
-    LDSTATIC="$libdir/libfltk_cairo.a $CAIROLIBS $LDSTATIC"
 fi
 
 LDLIBS="$DSOLINK $LDFLAGS $libs $LDLIBS"
-LDSTATIC="$LDFLAGS $LDSTATIC"
 
 # Answer to user requests
 if test -n "$echo_help"; then
@@ -284,8 +259,8 @@
 
     post="$prog"
 
-    echo $CXX $ARCHFLAGS $includes $CXXFLAGS $debug -o "'$prog'" "'$compile'" $LDSTATIC
-    $CXX $ARCHFLAGS $includes $CXXFLAGS $debug -o "$prog" "$compile" $LDSTATIC || exit 1
+    echo $CXX $ARCHFLAGS $includes $CXXFLAGS $debug -o "'$prog'" "'$compile'"
+    $CXX $ARCHFLAGS $includes $CXXFLAGS $debug -o "$prog" "$compile" || exit 1
 fi
 
 if test -n "$post"; then
@@ -369,26 +344,26 @@
 fi
 
 if test "$echo_libs" = "yes"; then
-    USELIBS="$libdir/libfltk.a"
+    USELIBS="$libdir/libfltk.so"
 
     if test x$use_forms = xyes; then
-        USELIBS="$libdir/libfltk_forms.a $USELIBS"
+        USELIBS="$libdir/libfltk_forms.so $USELIBS"
     fi
 
     if test x$use_gl = xyes; then
-        USELIBS="$libdir/libfltk_gl.a $USELIBS"
+        USELIBS="$libdir/libfltk_gl.so $USELIBS"
     fi
 
     if test x$use_cairo = xyes; then
-        USELIBS="$libdir/libfltk_cairo.a $USELIBS"
+        USELIBS="$libdir/libfltk_cairo.so $USELIBS"
     fi
 
     if test x$use_images = xyes; then
-        USELIBS="$libdir/libfltk_images.a $USELIBS"
+        USELIBS="$libdir/libfltk_images.so $USELIBS"
 
         for lib in fltk_jpeg fltk_png fltk_z; do
-            if test -f $libdir/lib$lib.a; then
-                USELIBS="$libdir/lib$lib.a $USELIBS"
+            if test -f $libdir/lib$lib.so; then
+                USELIBS="$libdir/lib$lib.so $USELIBS"
             fi
 	done
     fi
