diff -Naur orig/setup.py patched/setup.py
--- orig/setup.py	2014-09-02 02:17:40.000000000 +0200
+++ patched/setup.py	2014-11-16 23:56:05.909562050 +0100
@@ -100,8 +100,8 @@
 def update_desktop_file(filename, script_path):
     """Update the desktop file with prefixed paths."""
     try:
-        fin = open(filename, 'r')
-        fout = open(filename + '.new', 'w')
+        fin = open(filename, encoding='utf-8', mode='r')
+        fout = open(filename + '.new', encoding='utf-8', mode='w')
 
         for line in fin:
             if 'Exec=' in line:
