diff -Naur orig/qtwebengine/configure.pri patched/qtwebengine/configure.pri
--- orig/qtwebengine/configure.pri	2020-04-07 02:02:19.562050787 +0200
+++ patched/qtwebengine/configure.pri	2020-04-07 02:08:15.063106632 +0200
@@ -163,7 +163,7 @@ defineTest(qtConfTest_detectNinja) {
     !isEmpty(ninja) {
         qtLog("Found ninja from path: $$ninja")
         qtRunLoggedCommand("$$ninja --version", version)|return(false)
-        contains(version, "1\.([7-9]|1[0-9])\..*"): return(true)
+        contains(version, "1\.([7-9]|[1-9][0-9])\..*"): return(true)
         qtLog("Ninja version too old")
     }
     qtLog("Building own ninja")
