diff -Naur a/qtwebengine/src/3rdparty/chromium/build/linux/unbundle/icu.gn b/qtwebengine/src/3rdparty/chromium/build/linux/unbundle/icu.gn
--- a/qtwebengine/src/3rdparty/chromium/build/linux/unbundle/icu.gn	2021-12-15 17:12:54.000000000 +0100
+++ b/qtwebengine/src/3rdparty/chromium/build/linux/unbundle/icu.gn	2023-07-07 01:39:07.512845550 +0200
@@ -18,6 +18,15 @@
     "ICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC",
     "UCHAR_TYPE=uint16_t",
 
+    # As of icu 73 C++ nullptr, char16_t, override, final and noexcept are used
+    # instead of defines. These definitions can be removed when the bundled icu
+    # gets updated to 73. For more details, see:
+    # https://unicode-org.atlassian.net/browse/ICU-21833 and
+    # https://github.com/unicode-org/icu/commit/28643799377ecf654564f6f31854b02788cebe33
+    "U_FINAL=final",
+    "U_NOEXCEPT=noexcept",
+    "U_OVERRIDE=override",
+
     # U_EXPORT (defined in unicode/platform.h) is used to set public visibility
     # on classes through the U_COMMON_API and U_I18N_API macros (among others).
     # When linking against the system ICU library, we want its symbols to have
