diff -Naur orig/qtwebengine/src/3rdparty/chromium/base/trace_event/trace_event_memory_overhead.h patched/qtwebengine/src/3rdparty/chromium/base/trace_event/trace_event_memory_overhead.h
--- orig/qtwebengine/src/3rdparty/chromium/base/trace_event/trace_event_memory_overhead.h	2020-05-06 16:21:29.000000000 +0200
+++ patched/qtwebengine/src/3rdparty/chromium/base/trace_event/trace_event_memory_overhead.h	2020-08-19 12:29:20.485735239 +0200
@@ -8,6 +8,7 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#include <string>
 #include <unordered_map>
 
 #include "base/base_export.h"
