diff -Nur orig/usr/share/applications/fedora-dangerdeep.desktop mod/usr/share/applications/fedora-dangerdeep.desktop
--- orig/usr/share/applications/fedora-dangerdeep.desktop	2007-06-17 00:56:14.000000000 +0200
+++ mod/usr/share/applications/fedora-dangerdeep.desktop	2020-04-02 20:58:52.369548158 +0200
@@ -3,7 +3,7 @@
 Name=Danger from the deep
 Comment=Submarine simulation game
 Exec=dangerdeep
-Icon=dftd_icon.xpm
+Icon=/usr/share/pixmaps/dftd_icon.xpm
 Terminal=false
 Type=Application
 Categories=Game;Application;X-Fedora;
