diff -Naur orig/configure patched/configure
--- orig/configure	2016-11-30 19:12:35.000000000 +0100
+++ patched/configure	2017-01-27 01:31:33.542952912 +0100
@@ -39342,7 +39342,7 @@
        mint*)           os='MiNT';;
        mingw*)          os='MinGW';;
        lynxos*)         os='LynxOS';;
-       linux*)          os='GNU/Linux';;
+       linux*)          os='UHU-Linux';;
        hpux*)           os='HP-UX';;
        hiux*)           os='HI-UX';;
        gnu*)            os='GNU';;
