diff -Naur canorus-0.7.R1002-orig/canorus.desktop canorus-0.7.R1002/canorus.desktop
--- canorus-0.7.R1002-orig/canorus.desktop	2009-03-27 11:27:02.000000000 +0100
+++ canorus-0.7.R1002/canorus.desktop	2009-04-15 23:06:54.000000000 +0200
@@ -2,13 +2,15 @@
 Encoding=UTF-8
 Name=Canorus
 GenericName=Music score editor
+GenericName[hu]=Kottaszerkesztő
 GenericName[sl]=Program za pisanje not
 Comment=A free next-generation cross-platform music score editor!
+Comment[hu]=Kottaszerkesztő program
 Comment[sl]=Prost program za pisanje not naslednje generacije!
 Exec=canorus
-Icon=canorus
+Icon=/usr/share/canorus/images/clogosm.png
 Terminal=false
 Type=Application
-Categories=Application;AudioVideo;
+Categories=Application;AudioVideo;
 StartupNotify=true
 MimeType=application/x-notes
