diff -Naur canorus-0.7.3rc3-orig/src/layout/drawabletuplet.cpp canorus-0.7.3rc3/src/layout/drawabletuplet.cpp
--- canorus-0.7.3rc3-orig/src/layout/drawabletuplet.cpp	2018-06-18 19:48:04.000000000 +0200
+++ canorus-0.7.3rc3/src/layout/drawabletuplet.cpp	2018-07-19 15:32:37.298162608 +0200
@@ -10,13 +10,14 @@
 #include <QPen>
 #include <QPainter>
 #include <QFont>
+#include <math.h>
 
 CADrawableTuplet::CADrawableTuplet( CATuplet *tuplet, CADrawableContext *c, double x1, double y1, double x2, double y2 )
  : CADrawableMusElement( tuplet, c, x1, 0 ) {
 	setDrawableMusElementType( DrawableTuplet );
 
 	setWidth( x2-x1 );
-	setHeight( (abs(y2-y1)>5)?abs(y2-y1):8 );
+	setHeight( (abs(y2-y1)>5)?fabs(y2-y1):8 );
 	setYPos( (c && qMin(y1, y2)>c->yPos())?qMin(y1, y2):(qMin(y1,y2)-height()) );
 }
 
