diff -Nur orig/script/scons mod/script/scons
--- orig/script/scons	2019-08-08 16:50:09.000000000 +0200
+++ mod/script/scons	2020-02-10 09:13:59.659000050 +0100
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2
 #
 # SCons - a Software Constructor
 #
diff -Nur orig/script/scons-configure-cache mod/script/scons-configure-cache
--- orig/script/scons-configure-cache	2019-08-08 16:50:09.000000000 +0200
+++ mod/script/scons-configure-cache	2020-02-10 09:14:08.746988259 +0100
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2
 #
 # SCons - a Software Constructor
 #
diff -Nur orig/script/sconsign mod/script/sconsign
--- orig/script/sconsign	2019-08-08 16:50:09.000000000 +0200
+++ mod/script/sconsign	2020-02-10 09:14:23.958968523 +0100
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2
 #
 # SCons - a Software Constructor
 #
diff -Nur orig/script/scons-time mod/script/scons-time
--- orig/script/scons-time	2019-08-08 16:50:09.000000000 +0200
+++ mod/script/scons-time	2020-02-10 09:14:15.938978928 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # scons-time - run SCons timings and collect statistics
 #
