diff -Naur usermode-1.111-orig/sample/wrapped.pamd usermode-1.111/sample/wrapped.pamd
--- usermode-1.111-orig/sample/wrapped.pamd	2012-09-22 17:38:43.000000000 +0200
+++ usermode-1.111/sample/wrapped.pamd	2013-11-23 15:15:16.015394309 +0100
@@ -1,5 +1,4 @@
 #%PAM-1.0
 auth       sufficient	pam_rootok.so
-auth       required	pam_console.so
-#auth       include	system-auth
+auth       required	pam_locallogin.so
 account    required	pam_permit.so
