diff -Naur a/mateweather/docs/Makefile.am b/mateweather/docs/Makefile.am
--- a/mateweather/docs/Makefile.am	2021-08-07 13:25:17.000000000 +0200
+++ b/mateweather/docs/Makefile.am	2022-04-11 04:06:50.305331979 +0200
@@ -19,7 +19,7 @@
 		figures/stock_weather-sunny.png
 
 # Add linguas to be ignored, e.g. IGNORE_HELP_LINGUAS = ca de es fr
-IGNORE_HELP_LINGUAS =
+IGNORE_HELP_LINGUAS = sr
 HELP_LINGUAS = $(if $(IGNORE_HELP_LINGUAS), \
 	$(filter-out $(IGNORE_HELP_LINGUAS),$(subst /,,$(dir $(wildcard */*.po)))), \
 	$(subst /,,$(dir $(wildcard */*.po))) )
diff -Naur a/mateweather/docs/Makefile.in b/mateweather/docs/Makefile.in
--- a/mateweather/docs/Makefile.in	2021-08-07 13:28:47.000000000 +0200
+++ b/mateweather/docs/Makefile.in	2022-04-11 04:07:08.109379479 +0200
@@ -348,7 +348,7 @@
 
 
 # Add linguas to be ignored, e.g. IGNORE_HELP_LINGUAS = ca de es fr
-IGNORE_HELP_LINGUAS = 
+IGNORE_HELP_LINGUAS = sr
 HELP_LINGUAS = $(if $(IGNORE_HELP_LINGUAS), \
 	$(filter-out $(IGNORE_HELP_LINGUAS),$(subst /,,$(dir $(wildcard */*.po)))), \
 	$(subst /,,$(dir $(wildcard */*.po))) )
