diff -Naur a/drivemount/help/Makefile.am b/drivemount/help/Makefile.am
--- a/drivemount/help/Makefile.am	2022-07-09 17:01:49.000000000 +0200
+++ b/drivemount/help/Makefile.am	2022-07-09 20:30:43.485467887 +0200
@@ -10,7 +10,7 @@
 		figures/drivemount-applet_status.png
 
 # Add linguas to be ignored, e.g. IGNORE_HELP_LINGUAS = ca de es fr
-IGNORE_HELP_LINGUAS =
+IGNORE_HELP_LINGUAS = tr
 HELP_LINGUAS = $(if $(IGNORE_HELP_LINGUAS), \
 	$(filter-out $(IGNORE_HELP_LINGUAS),$(subst /,,$(dir $(wildcard */*.po)))), \
 	$(subst /,,$(dir $(wildcard */*.po))) )
diff -Naur a/drivemount/help/Makefile.in b/drivemount/help/Makefile.in
--- a/drivemount/help/Makefile.in	2022-07-09 17:05:58.000000000 +0200
+++ b/drivemount/help/Makefile.in	2022-07-09 20:31:03.761520422 +0200
@@ -343,7 +343,7 @@
 
 
 # Add linguas to be ignored, e.g. IGNORE_HELP_LINGUAS = ca de es fr
-IGNORE_HELP_LINGUAS = 
+IGNORE_HELP_LINGUAS = tr
 HELP_LINGUAS = $(if $(IGNORE_HELP_LINGUAS), \
 	$(filter-out $(IGNORE_HELP_LINGUAS),$(subst /,,$(dir $(wildcard */*.po)))), \
 	$(subst /,,$(dir $(wildcard */*.po))) )
diff -Naur a/mateweather/docs/Makefile.am b/mateweather/docs/Makefile.am
--- a/mateweather/docs/Makefile.am	2022-07-09 17:01:49.000000000 +0200
+++ b/mateweather/docs/Makefile.am	2022-07-09 20:31:37.901608827 +0200
@@ -19,7 +19,7 @@
 		figures/stock_weather-sunny.png
 
 # Add linguas to be ignored, e.g. IGNORE_HELP_LINGUAS = ca de es fr
-IGNORE_HELP_LINGUAS =
+IGNORE_HELP_LINGUAS = sr
 HELP_LINGUAS = $(if $(IGNORE_HELP_LINGUAS), \
 	$(filter-out $(IGNORE_HELP_LINGUAS),$(subst /,,$(dir $(wildcard */*.po)))), \
 	$(subst /,,$(dir $(wildcard */*.po))) )
diff -Naur a/mateweather/docs/Makefile.in b/mateweather/docs/Makefile.in
--- a/mateweather/docs/Makefile.in	2022-07-09 17:05:58.000000000 +0200
+++ b/mateweather/docs/Makefile.in	2022-07-09 20:31:46.817631903 +0200
@@ -352,7 +352,7 @@
 
 
 # Add linguas to be ignored, e.g. IGNORE_HELP_LINGUAS = ca de es fr
-IGNORE_HELP_LINGUAS = 
+IGNORE_HELP_LINGUAS = sr
 HELP_LINGUAS = $(if $(IGNORE_HELP_LINGUAS), \
 	$(filter-out $(IGNORE_HELP_LINGUAS),$(subst /,,$(dir $(wildcard */*.po)))), \
 	$(subst /,,$(dir $(wildcard */*.po))) )
