diff -Naur orig/config_unix.py patched/config_unix.py
--- orig/config_unix.py	2018-07-17 15:28:16.000000000 +0200
+++ patched/config_unix.py	2018-10-28 23:38:44.817772052 +0100
@@ -202,8 +202,8 @@
         Dependency('PNG', 'png.h', 'libpng', ['png']),
         Dependency('JPEG', 'jpeglib.h', 'libjpeg', ['jpeg']),
         Dependency('SCRAP', '', 'libX11', ['X11']),
-        Dependency('PORTMIDI', 'portmidi.h', 'libportmidi.so', ['portmidi']),
-        porttime_dep,
+        #Dependency('PORTMIDI', 'portmidi.h', 'libportmidi.so', ['portmidi']),
+        #porttime_dep,
         find_freetype(),
         #Dependency('GFX', 'SDL_gfxPrimitives.h', 'libSDL_gfx.so', ['SDL_gfx']),
     ]
diff -Naur orig/Setup.in patched/Setup.in
--- orig/Setup.in	2018-07-15 08:51:55.000000000 +0200
+++ patched/Setup.in	2018-10-28 23:36:53.064605225 +0100
@@ -14,8 +14,8 @@
 PNG = -lpng
 JPEG = -ljpeg
 SCRAP = -lX11
-PORTMIDI = -lportmidi
-PORTTIME = -lporttime
+#PORTMIDI = -lportmidi
+#PORTTIME = -lporttime
 FREETYPE = -lfreetype
 #--EndConfig
 
@@ -30,7 +30,7 @@
 mixer src/mixer.c $(SDL) $(MIXER) $(DEBUG)
 mixer_music src/music.c $(SDL) $(MIXER) $(DEBUG)
 scrap src/scrap.c $(SDL) $(SCRAP) $(DEBUG)
-pypm src/pypm.c $(SDL) $(PORTMIDI) $(PORTTIME) $(DEBUG)
+#pypm src/pypm.c $(SDL) $(PORTMIDI) $(PORTTIME) $(DEBUG)
 
 GFX = src/SDL_gfx/SDL_gfxPrimitives.c
 #GFX = src/SDL_gfx/SDL_gfxBlitFunc.c src/SDL_gfx/SDL_gfxPrimitives.c
