diff -Naur orig/untracked/pango/pangomm/attrlist.cc patched/untracked/pango/pangomm/attrlist.cc
--- orig/untracked/pango/pangomm/attrlist.cc	1970-01-01 01:00:00.000000000 +0100
+++ patched/untracked/pango/pangomm/attrlist.cc	2021-09-22 10:41:26.223600854 +0200
@@ -2,7 +2,7 @@
 
 
 #include <glibmm.h>
-
+#include <pango/pango-markup.h>
 #include <pangomm/attrlist.h>
 #include <pangomm/private/attrlist_p.h>
 
