diff -Naur orig/src/loaders/SILLYPNGImageLoader.cpp patched/src/loaders/SILLYPNGImageLoader.cpp
--- orig/src/loaders/SILLYPNGImageLoader.cpp	2006-06-19 00:42:12.000000000 +0200
+++ patched/src/loaders/SILLYPNGImageLoader.cpp	2014-08-27 02:34:48.052071577 +0200
@@ -41,6 +41,8 @@
 
 #include "loaders/SILLYPNGImageContext.h" 
 #include <png.h>
+#include <string.h>
+
 // Start section of namespace SILLY
 namespace SILLY
 {
@@ -66,7 +68,7 @@
     //  printf("PNG Error: %s\n", error);
     // copied from libpng's pngerror.cpp
     jmp_buf buf;
-    memcpy(buf, png_ptr->jmpbuf, sizeof(jmp_buf));
+    memcpy(buf, png_jmpbuf(png_ptr), sizeof(jmp_buf));
     longjmp(buf, 1);
 }
 
