diff -Naur a/src/images_list.cpp b/src/images_list.cpp
--- a/src/images_list.cpp	2022-02-06 13:45:11.000000000 +0100
+++ b/src/images_list.cpp	2022-03-27 18:42:44.626527642 +0200
@@ -33,7 +33,7 @@
 #include <wx/fs_mem.h>
 #include <wx/mstream.h>
 #include <wx/tokenzr.h>
-#include "../3rd/lunasvg/include/document.h"
+#include "../3rd/lunasvg/include/lunasvg.h"
 
 // SVG filename in Zip, the PNG enum to which it relates, whether to recolor background
 static const std::map<std::string, std::pair<int, bool>> iconName2enum = {
