diff -Nur orig/CMakeLists.txt mod/CMakeLists.txt
--- orig/CMakeLists.txt	2015-02-19 17:13:47.000000000 +0100
+++ mod/CMakeLists.txt	2020-04-21 13:03:21.958147423 +0200
@@ -497,8 +497,8 @@
 	# find python executables
 	find_program (PYTHON2_EXECUTABLE python2)  # if both python2&3 are installed, usually we will have 'python2' and 'python3' available (one of them being a link to 'python')
 	if (PYTHON2_EXECUTABLE AND EXISTS ${PYTHON2_EXECUTABLE})
-		set (PYTHON2_FOUND TRUE)
-		set (with_python2 yes)
+		set (PYTHON2_FOUND FALSE)
+		set (with_python2 no)
 	endif()
 	find_program (PYTHON3_EXECUTABLE python3)
 	if (PYTHON3_EXECUTABLE AND EXISTS ${PYTHON3_EXECUTABLE})
@@ -515,8 +515,8 @@
 			message (STATUS "   Python Version: ${PYTHON_VERSION}")
 			if ("${PYTHON_VERSION}" STREQUAL "2")
 				set (PYTHON2_EXECUTABLE ${PYTHON_EXECUTABLE})
-				set (PYTHON2_FOUND TRUE)
-				set (with_python2 yes)
+				set (PYTHON2_FOUND FALSE)
+				set (with_python2 no)
 			elseif ("${PYTHON_VERSION}" STREQUAL "3")
 				set (PYTHON3_EXECUTABLE ${PYTHON_EXECUTABLE})
 				set (PYTHON3_FOUND TRUE)
