diff -Nur orig/Dbus/data/cairo-dock-launcher-API-daemon.py mod/Dbus/data/cairo-dock-launcher-API-daemon.py
--- orig/Dbus/data/cairo-dock-launcher-API-daemon.py	2015-02-19 17:13:47.000000000 +0100
+++ mod/Dbus/data/cairo-dock-launcher-API-daemon.py	2020-04-21 13:17:50.311589604 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 #
 # This is a part of the Cairo-Dock plug-ins.
 # Copyright : (C) 2012 by Fabrice Rey
diff -Nur orig/Dbus/misc/launcher-test.py mod/Dbus/misc/launcher-test.py
--- orig/Dbus/misc/launcher-test.py	2015-02-19 17:13:47.000000000 +0100
+++ mod/Dbus/misc/launcher-test.py	2020-04-21 13:18:10.651533927 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # A simple test case.
 
 from gi.repository import Unity, Gio, GObject, Dbusmenu
diff -Nur orig/Global-Menu/registrar/appmenu-registrar.py mod/Global-Menu/registrar/appmenu-registrar.py
--- orig/Global-Menu/registrar/appmenu-registrar.py	2015-02-19 17:13:47.000000000 +0100
+++ mod/Global-Menu/registrar/appmenu-registrar.py	2020-04-21 13:18:31.743483486 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 #
 # This is a part of the Cairo-Dock plug-ins.
 # Copyright : (C) 2011 by Fabrice Rey
diff -Nur orig/Status-Notifier/watcher/status-notifier-watcher.py mod/Status-Notifier/watcher/status-notifier-watcher.py
--- orig/Status-Notifier/watcher/status-notifier-watcher.py	2015-02-19 17:13:47.000000000 +0100
+++ mod/Status-Notifier/watcher/status-notifier-watcher.py	2020-04-21 13:18:51.199443401 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 #
 # This is a part of the Cairo-Dock plug-ins.
 # Copyright : (C) 2011 by Fabrice Rey
