diff -Naur schismtracker-20180810-orig/sys/fd.org/schism.desktop schismtracker-20180810/sys/fd.org/schism.desktop
--- schismtracker-20180810-orig/sys/fd.org/schism.desktop	2018-08-10 07:04:54.000000000 +0200
+++ schismtracker-20180810/sys/fd.org/schism.desktop	2018-09-01 16:56:19.809059225 +0200
@@ -1,6 +1,5 @@
 [Desktop Entry]
 Actions=Play;FontEditor
-Version=1.0
 Name=Schism Tracker
 Comment=Impulse Tracker clone
 Comment[fr]=Clone d'Impulse Tracker
@@ -9,7 +8,7 @@
 Exec=schismtracker %f
 Type=Application
 Icon=schism-icon-128
-Categories=Audio;Midi;Sequencer;Player;Music
+Categories=AudioVideo;Audio;Midi;Sequencer;Player;Music
 MimeType=audio/x-it;audio/x-s3m;audio/x-xm;audio/x-mod
 Keywords=impulse;module;midi;music
 
