diff -Nur orig/sys/fd.org/itf.desktop mod/sys/fd.org/itf.desktop
--- orig/sys/fd.org/itf.desktop	2016-09-13 21:21:08.000000000 +0200
+++ mod/sys/fd.org/itf.desktop	2017-02-01 17:38:54.866515178 +0100
@@ -1,11 +1,10 @@
 [Desktop Entry]
-Version=1.0
 Name=Schism Tracker Font Editor
 Comment=ITF Clone
 Encoding=UTF-8
 Terminal=false
 Exec=schismtracker --font-editor
 Type=Application
-Icon=schism-itf-icon-128.png
-Categories=GNOME;Application;AudioVideo;Audio;Video;
+Icon=schism-itf-icon-128
+Categories=GNOME;AudioVideo;Audio;Video;
 X-Desktop-File-Install-Version=0.10
diff -Nur orig/sys/fd.org/schism.desktop mod/sys/fd.org/schism.desktop
--- orig/sys/fd.org/schism.desktop	2016-09-13 21:21:08.000000000 +0200
+++ mod/sys/fd.org/schism.desktop	2017-02-01 17:39:57.806417401 +0100
@@ -1,15 +1,13 @@
 [Desktop Entry]
 Actions=Play;
-Version=1.0
 Name=Schism Tracker
 Comment=Impulse Tracker Clone
-Encoding=UTF-8
 Terminal=false
 TryExec=schismtracker
 Exec=schismtracker %f
 Type=Application
-Icon=schism-icon-128.png
-Categories=GNOME;Application;AudioVideo;Audio;Video;
+Icon=schism-icon-128
+Categories=GNOME;AudioVideo;Audio;Video;
 MimeType=audio/x-mod
 X-Desktop-File-Install-Version=0.10
 
