diff -Naur orig/src/gmdb2/Makefile.am mod/src/gmdb2/Makefile.am
--- orig/src/gmdb2/Makefile.am	2016-03-25 07:33:19.000000000 +0100
+++ mod/src/gmdb2/Makefile.am	2016-03-25 09:16:32.385829500 +0100
@@ -3,7 +3,7 @@
 include_HEADERS = gmdb.h
 
 gmdb2_SOURCES = main2.c file.c table.c info.c table_def.c table_data.c table_export.c debug.c sql.c schema.c prefs.c
-LIBS =	-rdynamic $(GNOME_LIBS) @LIBS@ @LEXLIB@
+LIBS =	-rdynamic $(GNOME_LIBS) @LIBS@ 
 AM_CFLAGS	=	-I$(top_srcdir)/include $(GNOME_CFLAGS) \
         -DDATADIR="\"$(datadir)\""                      \
         -DGMDB_ICONDIR="\"$(datadir)/gmdb/glade/\""        \
diff -Naur orig/src/odbc/Makefile.am mod/src/odbc/Makefile.am
--- orig/src/odbc/Makefile.am	2016-03-25 07:33:19.000000000 +0100
+++ mod/src/odbc/Makefile.am	2016-03-25 09:16:32.385829500 +0100
@@ -11,5 +11,5 @@
 libmdbodbcW_la_LDFLAGS	= $(libmdbodbc_la_LDFLAGS)
 libmdbodbcW_la_CFLAGS	= $(AM_CFLAGS) -D ENABLE_ODBC_W=1
 endif
-LIBS	=	@LEXLIB@ $(GLIB_LIBS) $(ODBC_LIBS)
+LIBS	=	 $(GLIB_LIBS) $(ODBC_LIBS)
 unittest_LDADD	=	libmdbodbc.la ../libmdb/libmdb.la ../sql/libmdbsql.la
diff -Naur orig/src/util/Makefile.am mod/src/util/Makefile.am
--- orig/src/util/Makefile.am	2016-03-25 07:33:19.000000000 +0100
+++ mod/src/util/Makefile.am	2016-03-25 09:16:32.385829500 +0100
@@ -1,7 +1,7 @@
 SUBDIRS = bash-completion
 bin_PROGRAMS	=	mdb-export mdb-array mdb-schema mdb-tables mdb-parsecsv mdb-header mdb-sql mdb-ver mdb-prop 
 noinst_PROGRAMS = mdb-import prtable prcat prdata prkkd prdump prole updrow prindex
-LIBS	=	$(GLIB_LIBS) @LIBS@ @LEXLIB@ 
+LIBS	=	$(GLIB_LIBS) @LIBS@  
 DEFS = @DEFS@ -DLOCALEDIR=\"$(localedir)\"
 AM_CFLAGS	=	-I$(top_srcdir)/include $(GLIB_CFLAGS)
 LDADD	=	../libmdb/libmdb.la 
