diff -Naur orig/js/src/jsapi-tests/moz.build patched/js/src/jsapi-tests/moz.build
--- orig/js/src/jsapi-tests/moz.build	2017-04-11 04:13:16.000000000 +0200
+++ patched/js/src/jsapi-tests/moz.build	2019-05-22 19:56:26.766401213 +0200
@@ -147,7 +147,7 @@
 OS_LIBS += CONFIG['MOZ_ZLIB_LIBS']
 
 if CONFIG['GNU_CXX']:
-    CXXFLAGS += ['-Wno-shadow', '-Werror=format']
+    CXXFLAGS += ['-Wno-shadow']
 
 # This is intended as a temporary workaround to enable VS2015.
 if CONFIG['_MSC_VER']:
diff -Naur orig/js/src/old-configure.in patched/js/src/old-configure.in
--- orig/js/src/old-configure.in	2019-05-22 19:18:06.931264829 +0200
+++ patched/js/src/old-configure.in	2019-05-22 19:24:51.788957457 +0200
@@ -484,7 +484,7 @@
             DSO_LDOPTS="$DSO_LDOPTS -Wl,-z,defs"
         fi
     fi
-    WARNINGS_AS_ERRORS='-Werror'
+    WARNINGS_AS_ERRORS=''
     DSO_CFLAGS=''
     DSO_PIC_CFLAGS='-fPIC'
     ASFLAGS="$ASFLAGS -fPIC"
