diff -Naur lesspipe-1.71-orig/lesspipe.sh.in lesspipe-1.71/lesspipe.sh.in
--- lesspipe-1.71-orig/lesspipe.sh.in	2010-05-17 15:27:01.000000000 +0200
+++ lesspipe-1.71/lesspipe.sh.in	2010-05-17 15:28:26.000000000 +0200
@@ -31,9 +31,7 @@
 exec 2>/dev/null
 setopt KSH_ARRAYS SH_WORD_SPLIT
 set +o noclobber
-#ifdef tar
 tarcmd='tar'
-#endif
 
 cmd_exist () {
   command -v "$1" > /dev/null 2>&1 && return 0 || return 1
