diff -Naur orig/lesspipe.sh patched/lesspipe.sh
--- orig/lesspipe.sh	2020-08-25 09:24:11.000000000 +0200
+++ patched/lesspipe.sh	2020-09-01 14:14:54.157120349 +0200
@@ -964,6 +964,9 @@
     elif cmd_exist mp3info; then
       msg "append $sep to filename to view the raw data"
       mp3info "$2"
+    elif cmd_exist id3info; then
+      msg "append $sep to filename to view the raw data"
+      id3info "$2"
     fi
   elif [[ "$1" = "image" || "$1" = "mp3" || "$1" = "audio" || "$1" = "video" ]] && cmd_exist mediainfo; then
     msg "append $sep to filename to view the raw data"
diff -Naur orig/lesspipe.sh.in patched/lesspipe.sh.in
--- orig/lesspipe.sh.in	2020-09-01 14:11:48.208547783 +0200
+++ patched/lesspipe.sh.in	2020-09-01 14:17:08.817467231 +0200
@@ -1207,6 +1207,9 @@
     elif cmd_exist mp3info; then
       msg "append $sep to filename to view the raw data"
       mp3info "$2"
+    elif cmd_exist id3info; then
+      msg "append $sep to filename to view the raw data"
+      id3info "$2"
     fi
 #elif mp3info2
     elif [[ "$1" = "mp3" ]] && cmd_exist mp3info2; then
