diff -Nur orig/lesspipe.sh.in mod/lesspipe.sh.in
--- orig/lesspipe.sh.in	2021-08-25 16:35:57.000000000 +0200
+++ mod/lesspipe.sh.in	2021-09-04 20:32:09.792908110 +0200
@@ -28,11 +28,10 @@
 # Author:  Wolfgang Friebel, DESY (Wolfgang.Friebel AT desy.de)
 #
 #===============================================================================
-setopt KSH_ARRAYS SH_WORD_SPLIT 2>/dev/null
+exec 2>/dev/null
+setopt KSH_ARRAYS SH_WORD_SPLIT
 set +o noclobber
-#ifdef tar
 tarcmd='tar'
-#endif
 
 dir=${LESSOPEN#\|}
 dir=${dir%%lesspipe.sh*\%s}
@@ -1212,6 +1211,9 @@
   elif [[ "$1" = *mp3* ]] && cmd_exist mp3info; then
     msg "append $sep to filename to view the raw data"
     mp3info "$2"
+    elif cmd_exist id3info; then
+      msg "append $sep to filename to view the raw data"
+      id3info "$2"
 #endif
 #ifdef mediainfo
   elif [[ "$1" = *image* || "$1" = *mp3* || "$1" = *MP4* || "$1" = *audio* || "$1" = *video* ]] && cmd_exist mediainfo; then
