diff -Naur btmgr-3.7-1.orig/Makefile btmgr-3.7-1/Makefile
--- btmgr-3.7-1.orig/Makefile	2001-02-20 17:27:04.000000000 +0100
+++ btmgr-3.7-1/Makefile	2003-11-29 01:15:20.000000000 +0100
@@ -67,8 +67,7 @@
 ASM=		nasm $(CONFIG)
 CC=		gcc $(COMMON_FLAGS)
 
-SUBDIRS=	manager manager/themes installer docs
-#SUBDIRS=	manager manager/themes installer
+SUBDIRS=	manager manager/themes installer
 
 LOADER=		$(DEST_DIR)/loader.bin
 MAIN=		$(DEST_DIR)/main.bin
