diff -Naur orig/config.layout patched/config.layout
--- orig/config.layout	2020-02-21 01:39:22.000000000 +0100
+++ patched/config.layout	2020-08-16 21:44:52.157108840 +0200
@@ -419,3 +419,27 @@
     proxycachedir:   ${localstatedir}/cache/apache2
 </Layout>
 
+# UHU layout
+<Layout UHU>
+    prefix:        /usr
+    exec_prefix:   ${prefix}
+    bindir:        ${prefix}/bin
+    sbindir:       ${prefix}/sbin
+    libdir:        ${prefix}/lib
+    libexecdir:    ${prefix}/lib/apache
+    mandir:        ${prefix}/share/man
+    sysconfdir:    /etc/apache
+    datadir:       /srv/www
+    installbuilddir: /usr/lib/apache/build
+    errordir:      ${datadir}/error
+    iconsdir:      ${datadir}/icons
+    htdocsdir:     ${datadir}/html
+    manualdir:     ${datadir}/manual
+    cgidir:        ${datadir}/cgi-bin
+    includedir:    ${prefix}/include/apache
+    localstatedir: /var
+    runtimedir:    ${localstatedir}/run
+    logfiledir:    ${localstatedir}/log/apache
+    proxycachedir: ${localstatedir}/cache/apache
+</Layout>
+
