diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2019-05-04 16:33:55.000000000 +0200
+++ patched/CMakeLists.txt	2019-10-08 23:17:20.028084802 +0200
@@ -3,6 +3,7 @@
 cmake_minimum_required(VERSION 3.0.0)
 cmake_policy(SET CMP0063 NEW) # disables cmake warning
 
+set(CMAKE_CXX_STANDARD 14) # adds -std=gnu++14 option
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 add_definitions(-Wextra -Wpedantic -Wswitch-enum)
diff -Naur orig/src/CMakeLists.txt patched/src/CMakeLists.txt
--- orig/src/CMakeLists.txt	2019-05-04 16:33:55.000000000 +0200
+++ patched/src/CMakeLists.txt	2019-10-08 23:16:20.531366569 +0200
@@ -31,9 +31,6 @@
 	add_executable(kshutdown ${kshutdown_SRC} ${kshutdown_MOC_SOURCES})
 endif()
 
-# HACK: Add -std=gnu++14 option (cmake horror):
-target_compile_features(kshutdown PUBLIC cxx_std_14)
-
 if(KS_PURE_QT)
 	target_link_libraries(kshutdown
 		Qt5::DBus
