diff -Naur orig/src/kshutdown.cpp patched/src/kshutdown.cpp
--- orig/src/kshutdown.cpp	2019-05-04 16:33:55.000000000 +0200
+++ patched/src/kshutdown.cpp	2019-10-06 03:19:54.780631045 +0200
@@ -216,7 +216,7 @@
 	
 	secsTo = now.secsTo(m_endDateTime);
 	if (secsTo > 0) {
-		const int DAY = 86'400;
+		const int DAY = 86400;
 		
 		QString result;
 		if (secsTo < DAY) {
diff -Naur orig/src/mainwindow.cpp patched/src/mainwindow.cpp
--- orig/src/mainwindow.cpp	2019-05-04 16:33:55.000000000 +0200
+++ patched/src/mainwindow.cpp	2019-10-06 17:57:59.232641173 +0200
@@ -364,7 +364,7 @@
 
 	// flash taskbar button
 	if ((id == "1m") || (id == "5m"))
-		QApplication::alert(this, 10'000);
+		QApplication::alert(this, 10000);
 }
 
 void MainWindow::setExtrasCommand(const QString &command) {
