diff -Nur orig/src/makefile mod/src/makefile
--- orig/src/makefile	2018-08-12 06:41:58.000000000 +0200
+++ mod/src/makefile	2019-03-27 21:23:26.940273674 +0100
@@ -43,7 +43,7 @@
 
 	#polo-gtk
 	valac -X -D'GETTEXT_PACKAGE="${app_name}"' \
-		--Xcc="-lm" ${define_symbols} --Xcc="-O3" --thread \
+		--Xcc="-lm" ${define_symbols} --Xcc="-O2" --thread \
 		Common/Main.vala \
 		Gtk/*.vala \
 		Utility/*.vala \
@@ -56,11 +56,11 @@
 		--pkg gdk-x11-3.0
 
 app-util-gtk3:
-	valac "polo-gtk3-helper.vala" --Xcc="-O3" -o polo-gtk3-helper --pkg gtk+-3.0
+	valac "polo-gtk3-helper.vala" --Xcc="-O2" -o polo-gtk3-helper --pkg gtk+-3.0
 
 app-util-disk:
 	valac --Xcc="-lm" -X -D'GETTEXT_PACKAGE="${app_name}"' \
-		"polo-disk.vala" Console/*.vala --Xcc="-O3" -o polo-disk --pkg posix --pkg gio-unix-2.0 --pkg gee-0.8
+		"polo-disk.vala" Console/*.vala --Xcc="-O2" -o polo-disk --pkg posix --pkg gio-unix-2.0 --pkg gee-0.8
 
 app-translations:
 
@@ -107,18 +107,18 @@
 	install -m 0755 ${app_name}-gtk "$(DESTDIR)$(bindir)"
 	install -m 0755 ${app_name}-gtk3-helper "$(DESTDIR)$(bindir)"
 	install -m 0755 ${app_name}-disk "$(DESTDIR)$(bindir)"
-	install -m 0755 ${app_name}-uninstall "$(DESTDIR)$(bindir)"
+	#install -m 0755 ${app_name}-uninstall "$(DESTDIR)$(bindir)"
 
 	#shared files
 	cp -dpr --no-preserve=ownership -t "$(DESTDIR)$(sharedir)/${app_name}" ./share/${app_name}/*
-	chmod --recursive 0755 $(DESTDIR)$(sharedir)/${app_name}/*
+	chmod --recursive 0644 $(DESTDIR)$(sharedir)/${app_name}/images/*
 	chmod a+x $(DESTDIR)$(sharedir)/${app_name}/files/gtk-theme/install-gtk-theme
 
 	#launcher
 	install -m 0644 ${app_name}-gtk.desktop "$(DESTDIR)$(launcherdir)"
 
 	#app icon
-	install -m 0755 ./share/pixmaps/${app_name}.png "$(DESTDIR)$(sharedir)/pixmaps/"
+	install -m 0644 ./share/pixmaps/${app_name}.png "$(DESTDIR)$(sharedir)/pixmaps/"
 
 	# appdata
 	install -m 0755 ../debian/${app_name}-gtk.appdata.xml "$(DESTDIR)$(sharedir)/appdata"
@@ -150,4 +150,4 @@
 	rm -f "$(DESTDIR)$(sharedir)/appdata/${app_name}-gtk.appdata.xml"
 
 	#translations
-	rm -f $(DESTDIR)$(localedir)/*/LC_MESSAGES/${app_name}.mo
+	#rm -f $(DESTDIR)$(localedir)/*/LC_MESSAGES/${app_name}.mo
