diff -Naur orig/src/makefile patched/src/makefile
--- orig/src/makefile	2018-03-18 04:59:17.000000000 +0100
+++ patched/src/makefile	2018-06-09 06:43:36.137601289 +0200
@@ -93,18 +93,18 @@
 	install -m 0755 ${app_name}-gtk "$(DESTDIR)$(bindir)"
 	install -m 0755 ${app_name}-gtk3-helper "$(DESTDIR)$(bindir)"
 	install -m 0755 ${app_name}-disk "$(DESTDIR)$(bindir)"
-	install -m 0755 ${app_name}-uninstall "$(DESTDIR)$(bindir)"
+	# install -m 0755 ${app_name}-uninstall "$(DESTDIR)$(bindir)"
 
 	#shared files
 	cp -dpr --no-preserve=ownership -t "$(DESTDIR)$(sharedir)/${app_name}" ./share/${app_name}/*
-	chmod --recursive 0755 $(DESTDIR)$(sharedir)/${app_name}/*
+	chmod --recursive 0644 $(DESTDIR)$(sharedir)/${app_name}/images/*
 	chmod a+x $(DESTDIR)$(sharedir)/${app_name}/files/gtk-theme/install-gtk-theme
 
 	#launcher
 	install -m 0644 ${app_name}-gtk.desktop "$(DESTDIR)$(launcherdir)"
 
 	#app icon
-	install -m 0755 ./share/pixmaps/${app_name}.png "$(DESTDIR)$(sharedir)/pixmaps/"
+	install -m 0644 ./share/pixmaps/${app_name}.png "$(DESTDIR)$(sharedir)/pixmaps/"
 
 	# appdata
 	install -m 0755 ../debian/${app_name}-gtk.appdata.xml "$(DESTDIR)$(sharedir)/appdata"
@@ -121,7 +121,7 @@
 	rm -f "$(DESTDIR)$(bindir)/${app_name}-gtk"
 	rm -f "$(DESTDIR)$(bindir)/${app_name}-gtk3-helper"
 	rm -f "$(DESTDIR)$(bindir)/${app_name}-disk"
-	rm -f "$(DESTDIR)$(bindir)/${app_name}-uninstall"
+	# rm -f "$(DESTDIR)$(bindir)/${app_name}-uninstall"
 
 	#shared files
 	rm -rf "$(DESTDIR)$(sharedir)/${app_name}"
