diff -Naur orig/data/terminator.desktop.in patched/data/terminator.desktop.in
--- orig/data/terminator.desktop.in	2017-02-26 18:45:29.134739579 +0100
+++ patched/data/terminator.desktop.in	2017-10-22 03:10:39.533038348 +0200
@@ -10,8 +10,4 @@
 X-Ubuntu-Gettext-Domain=terminator
 X-Ayatana-Desktop-Shortcuts=NewWindow;
 Keywords=terminal;shell;prompt;command;commandline;
-[NewWindow Shortcut Group]
-Name=Open a New Window
-Exec=terminator
-TargetEnvironment=Unity
 
