diff -Nur orig/src/cw.c mod/src/cw.c
--- orig/src/cw.c	2010-06-20 06:39:51.000000000 +0200
+++ mod/src/cw.c	2020-01-21 11:58:03.367912480 +0100
@@ -1173,13 +1173,6 @@
   close(master);
   return(0);
  }
- if(isastream(slave)){
-  if(ioctl(slave,I_PUSH,"ptem")<0||ioctl(slave,I_PUSH,"ldterm")<0){
-   close(master);
-   close(slave);
-   return(0);
-  }
- }
  if(v){
   cfgtable.p.merr=master;
   cfgtable.p.serr=slave;
@@ -1211,13 +1204,6 @@
   close(master);
   return(0);
  }
- if(isastream(slave)){
-  if(ioctl(slave,I_PUSH,"ptem")<0||ioctl(slave,I_PUSH,"ldterm")<0){
-   close(master);
-   close(slave);
-   return(0);
-  }
- }
  if(v){
   cfgtable.p.merr=master;
   cfgtable.p.serr=slave;
