diff -Naur orig/Imakefile mod/Imakefile
--- orig/Imakefile	1991-09-28 19:46:02.000000000 +0200
+++ mod/Imakefile	2015-04-25 13:33:08.872515283 +0200
@@ -8,7 +8,7 @@
         SPLIT = split
         CAT = cat
 /***** Note: probably want to define this to be something else *****/
-        HELPDIR = help
+        HELPDIR = /usr/share/spider-1.1
 
 /***** Note:  Used to over-ride some declarations in the default Imakefile
               configurations.  If you want to use some of these things, then
