diff -Nur orig/m4/fontforge_arg_enable.m4 mod/m4/fontforge_arg_enable.m4
--- orig/m4/fontforge_arg_enable.m4	2019-08-01 10:28:36.000000000 +0200
+++ mod/m4/fontforge_arg_enable.m4	2020-01-04 15:10:58.646872663 +0100
@@ -116,8 +116,8 @@
 if test x"${i_do_have_python_scripting}" != xyes; then
    i_want_python_ver=
 else
-   PKG_CHECK_MODULES([PYTHON],[python-"${PYTHON_VERSION}"], dnl   [PKG_CHECK_MODULES([PYTHONDEV],[python-"${PYTHON_VERSION}"],,[i_do_have_python_scripting=maybe])],
-      [PKG_CHECK_MODULES([PYTHONDEV],[python-"${PYTHON_VERSION}"],,[i_do_have_python_scripting=no])],
+   PKG_CHECK_MODULES([PYTHON],[python-"${PYTHON_VERSION}"-embed], dnl   [PKG_CHECK_MODULES([PYTHONDEV],[python-"${PYTHON_VERSION}"-embed],,[i_do_have_python_scripting=maybe])],
+      [PKG_CHECK_MODULES([PYTHONDEV],[python-"${PYTHON_VERSION}"-embed],,[i_do_have_python_scripting=no])],
       [i_do_have_python_scripting=no])
 dnl dnl TODO: have python3 AND python2, but only have python2 dev, but no python3 dev
 dnl if test x"${i_do_have_python_scripting}" = xmaybe; then
@@ -126,7 +126,7 @@
 dnl      i_do_have_python_scripting=yes
 dnl      AM_PATH_PYTHON([2.7],,[i_do_have_python_scripting=no])
 dnl      if test x"${i_do_have_python_scripting}" = xyes; then
-dnl         PKG_CHECK_MODULES([PYTHONDEV],[python-"${PYTHON_VERSION}"],,[i_do_have_python_scripting=no])
+dnl         PKG_CHECK_MODULES([PYTHONDEV],[python-"${PYTHON_VERSION}"-embed],,[i_do_have_python_scripting=no])
 dnl      fi
 dnl   else
 dnl      i_do_have_python_scripting=no
