diff -Naur OpenSceneGraph-OpenSceneGraph-3.5.9-orig/src/osgPlugins/jpeg/ReaderWriterJPEG.cpp OpenSceneGraph-OpenSceneGraph-3.5.9/src/osgPlugins/jpeg/ReaderWriterJPEG.cpp
--- OpenSceneGraph-OpenSceneGraph-3.5.9-orig/src/osgPlugins/jpeg/ReaderWriterJPEG.cpp	2017-11-29 15:49:00.000000000 +0100
+++ OpenSceneGraph-OpenSceneGraph-3.5.9/src/osgPlugins/jpeg/ReaderWriterJPEG.cpp	2017-12-31 13:56:59.536446447 +0100
@@ -73,6 +73,7 @@
 #if JPEG_LIB_VERSION < 90
     /* Some versions of jmorecfg.h define boolean, some don't...
     Those that do also define HAVE_BOOLEAN, so we can guard using that. */
+    #define HAVE_BOOLEAN
     #ifndef HAVE_BOOLEAN
     typedef int boolean;
     #define FALSE 0
