diff -Naur orig/usb_modeswitch.sh patched/usb_modeswitch.sh
--- orig/usb_modeswitch.sh	2019-11-27 21:04:55.000000000 +0100
+++ patched/usb_modeswitch.sh	2019-12-16 14:59:56.997448157 +0100
@@ -62,14 +62,5 @@
 fi
 
 PATH=/bin:/sbin:/usr/bin:/usr/sbin
-init_path=`readlink -f /sbin/init`
-if [ `basename $init_path` = "systemd" ]; then
-	systemctl --no-block restart usb_modeswitch@$p2.service
-elif [ -e "/etc/init/usb-modeswitch-upstart.conf" ]; then
-	initctl emit --no-wait usb-modeswitch-upstart UMS_PARAM=$p2
-else
-	# only old distros, new udev will kill all subprocesses
-	exec 1<&- 2<&- 5<&- 7<&-
-	exec usb_modeswitch_dispatcher --switch-mode $p2 &
-fi
+systemctl --no-block restart usb_modeswitch@$p2.service
 exit 0
