diff -Naur orig/usb_modeswitch.c patched/usb_modeswitch.c
--- orig/usb_modeswitch.c	2019-11-27 21:00:01.000000000 +0100
+++ patched/usb_modeswitch.c	2019-12-16 15:06:56.819376143 +0100
@@ -1586,7 +1586,7 @@
 		 * description is read for syslog message
 		 */
 		// Wait counter passed on from previous loop
-		for (i=i; i < CheckSuccess; i++) {
+		for (i=0; i < CheckSuccess; i++) {
 			SHOW_PROGRESS(output," Search for target devices ...\n");
 			dev = search_devices(&newTargetCount, TargetVendor, TargetProductList,
 					TargetClass, 0, SEARCH_TARGET);
