diff -urN usb-modeswitch-2.2.0.orig/Makefile usb-modeswitch-2.2.0/Makefile
--- usb-modeswitch-2.2.0.orig/Makefile	2014-05-29 20:35:46.000000000 +0200
+++ usb-modeswitch-2.2.0/Makefile	2014-07-22 15:39:33.050355604 +0200
@@ -9,7 +9,7 @@
 ETCDIR      = $(DESTDIR)/etc
 SYSDIR      = $(ETCDIR)/systemd/system
 UPSDIR      = $(ETCDIR)/init
-UDEVDIR     = $(DESTDIR)/lib/udev
+UDEVDIR     = $(DESTDIR)/usr/lib/udev
 SBINDIR     = $(PREFIX)/sbin
 MANDIR      = $(PREFIX)/share/man/man1
 VPATH       = jimtcl
