diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2019-05-21 08:31:22.000000000 +0200
+++ patched/CMakeLists.txt	2019-05-24 00:04:29.787226823 +0200
@@ -21,7 +21,7 @@
 set(LIBTS_VERSION_AGE 10)
 
 set(PACKAGE_VERSION 1.20)
-set(tslib_extras_dir ${CMAKE_INSTALL_PREFIX}/etc)
+set(tslib_extras_dir /etc)
 set(TS_POINTERCAL ${tslib_extras_dir}/pointercal)
 set(TS_CONF ${tslib_extras_dir}/ts.conf)
 
