diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2018-09-06 11:26:06.000000000 +0200
+++ patched/CMakeLists.txt	2018-10-13 12:21:36.346724645 +0200
@@ -21,7 +21,7 @@
 set(LIBTS_VERSION_AGE 9)
 
 set(PACKAGE_VERSION 1.17)
-set(tslib_extras_dir ${CMAKE_INSTALL_PREFIX}/etc)
+set(tslib_extras_dir /etc)
 set(TS_POINTERCAL ${tslib_extras_dir}/pointercal)
 set(TS_CONF ${tslib_extras_dir}/ts.conf)
 
