diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2018-11-27 09:22:12.000000000 +0100
+++ patched/CMakeLists.txt	2019-01-02 11:41:05.583946785 +0100
@@ -21,7 +21,7 @@
 set(LIBTS_VERSION_AGE 10)
 
 set(PACKAGE_VERSION 1.18)
-set(tslib_extras_dir ${CMAKE_INSTALL_PREFIX}/etc)
+set(tslib_extras_dir /etc)
 set(TS_POINTERCAL ${tslib_extras_dir}/pointercal)
 set(TS_CONF ${tslib_extras_dir}/ts.conf)
 
