diff -Naurd moon-buggy-0.5.53.orig/Makefile.in moon-buggy-0.5.53/Makefile.in
--- moon-buggy-0.5.53.orig/Makefile.in	2001-08-26 22:58:18.000000000 +0200
+++ moon-buggy-0.5.53/Makefile.in	2003-07-20 18:04:29.000000000 +0200
@@ -628,7 +628,6 @@
 		 || chmod 664 "$(DESTDIR)$(scoredir)/mbscore" ; } \
 	    && chmod 575 "$(DESTDIR)$(scoredir)" ; \
 	fi
-	$(DESTDIR)$(bindir)/moon-buggy -c
 
 uninstall-local:
 	rm -f $(DESTDIR)$(scoredir)/mbscore
