diff -Nur orig/data/konqueror_gestures_kde321.khotkeys mod/data/konqueror_gestures_kde321.khotkeys
--- orig/data/konqueror_gestures_kde321.khotkeys	2016-08-23 14:40:52.000000000 +0200
+++ mod/data/konqueror_gestures_kde321.khotkeys	2016-09-03 09:06:15.000000000 +0200
@@ -881,7 +881,7 @@
 Comment[hne]=यूआरएल डिरेक्ट्री स्ट्रक्चर मं ऊपर जाव.\nमोजिल्ला सैली: दबाव, ऊपर जाव, बायाँ खिसकव, ऊपर खिसकव, छोड़व.
 Comment[hr]=Odlazak gore u strukturi URL-a/direktorija.\nMozilla-stil: pritisni, pomakni gore, pomakni lijevo, pomakni gore, pusti
 Comment[hsb]=Dźe horje w strukturje URL/zapiskow.\nMozilla-stil: Stłóčić, horje suwać, dolěwa suwać, horje suwać, pušćić.
-Comment[hu]=Feljebb lépés a könyvtárstruktúrában.\Mozilla-stílusú: lenyomás, fel, balra, fel, felengedés.
+Comment[hu]=Feljebb lépés a könyvtárstruktúrában.\nMozilla-stílusú: lenyomás, fel, balra, fel, felengedés.
 Comment[ia]=Vader in alto in le structura del directorio de URL.\nMozilla-stilo: Pressa, move in alto, move a sinistra, move in alto, relaxa.
 Comment[id]=Pergi ke atas di struktur URL/direktori.\nGaya Mozilla: Tekan, pindah ke atas, pindah ke kiri, pindah ke atas, lepaskan.
 Comment[is]=Fara upp í slóða-/möppukerfi.\nMozilla-aðferðin: Ýta, færa upp, færa til vinstri, færa upp, sleppa.
@@ -1125,7 +1125,7 @@
 Comment[gu]=URL/ડિરેક્ટરી માળખામાં જાય છે.\nઓપેરા-રીત: દબાવો, ઉપર ખસેડો, ડાબે ખસેડો, ઉપર ખસેડો, છોડો.\nનોંધ: "પહેલાની ટેબ સક્રિય કરો" સાથે મુશ્કેલી સર્જે છે, જેવા કે મૂળભૂત રીતે નિષ્ક્રિય છે.
 Comment[he]=מעבר כלפי מעלה בתצוגת כתובת/תיקייה.\nבסגנון אופרה: לחץ, הזז למעלה, הזז שמאלה, הזז למעלה, שחרר.\nשים לב: מתנגש עם "הפעל את הלשונית הקודמת", ולכן מנוטרל כברירת מחדל.
 Comment[hr]=Odlazak gore u strukturi URL-a/direktorija.\nOpera stil: Pritisni, pomakni gore, pomakni lijevo, pomakni gore, pusti.\nNAPOMENA: Sukobi sa "Aktiviranje prethodne kartice" i slični su uobičajeno onemogućeni.
-Comment[hu]=Feljebb lépés a könyvtárstruktúrában.\Opera-stílusú: lenyomás, fel, balra, fel, felengedés.\nMegjegyzés: ütközik az 'Előző lap' parancsával, ezért alapértelmezésben le van tiltva.
+Comment[hu]=Feljebb lépés a könyvtárstruktúrában.\nOpera-stílusú: lenyomás, fel, balra, fel, felengedés.\nMegjegyzés: ütközik az 'Előző lap' parancsával, ezért alapértelmezésben le van tiltva.
 Comment[ia]=Vader in alto in le structure del directorio de URL.\nOpera-stilo: Pressa, move in alto, move a sinistra, move in alto, relaxa.\n NOTA: Il conflige con  "Activa previe scheda", dunque es disactivate pro definition.
 Comment[id]=Pergi ke atas di struktur URL/direktori.\nGaya Opera: Tekan, pindah ke atas, pindah ke kiri, pindah ke atas, lepaskan.\nCATATAN: Konflik dengan  "Aktifkan Tab Sebelumnya", dan oleh karena itu standarnya dinonaktifkan.
 Comment[is]=Fara upp í slóða-/möppukerfi.\nOpera-aðferðin: Ýta, færa upp, færa til vinstri, færa upp, sleppa.\nATH: rekst á við 'Virkja fyrri flipa', og er þess vegna sjálfgefið óvirk.
