Index: b/bash/Makefile.am
===================================================================
--- a/bash/Makefile.am
+++ b/bash/Makefile.am
@@ -19,7 +19,7 @@
 if BASH_BUILTINS
 
 COMMON_CPPFLAGS = -I$(top_srcdir)/src -DHAVE_CONFIG_H \
-                  -I$(BASH_HEADERS) -I$(BASH_HEADERS)/lib -I$(BASH_HEADERS)/builtins
+                  -I$(BASH_HEADERS) -I$(BASH_HEADERS)/include -I$(BASH_HEADERS)/builtins
 
 lib_LTLIBRARIES = readrec.la testrec.la
 
