From ad55d9385a8feaf18cab986848d2bf8b17ce202c Mon Sep 17 00:00:00 2001
From: Viacheslav <vaceslavkozin619@gmail.com>
Date: Thu, 19 May 2022 13:53:34 +0200
Subject: [PATCH] Put authentication window in center of the screen

---
 src/xfce-polkit-listener.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/xfce-polkit-listener.c b/src/xfce-polkit-listener.c
index d294001..5d17239 100644
--- a/src/xfce-polkit-listener.c
+++ b/src/xfce-polkit-listener.c
@@ -215,6 +215,7 @@ static void initiate_authentication(PolkitAgentListener  *listener,
 			NULL);
 	xfce_titled_dialog_set_subtitle(XFCE_TITLED_DIALOG(d->auth_dlg), message);
 	gtk_window_set_icon_name(GTK_WINDOW(d->auth_dlg), "dialog-password");
+	gtk_window_set_position(GTK_WINDOW(d->auth_dlg), GTK_WIN_POS_CENTER_ALWAYS);
 
 	content = gtk_dialog_get_content_area(GTK_DIALOG(d->auth_dlg));
 
