diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2018-07-16 16:40:33.000000000 +0200
+++ patched/configure.ac	2019-05-24 08:55:29.163940891 +0200
@@ -833,7 +833,7 @@
 AS_IF([test "x$HAVE_ALSA" = "x1"],
     [
         save_CPPFLAGS="$CPPFLAGS"; CPPFLAGS="$CPPFLAGS $ASOUNDLIB_CFLAGS"
-        AC_CHECK_HEADERS([use-case.h], HAVE_ALSA_UCM=1, HAVE_ALSA_UCM=0)
+        AC_CHECK_HEADERS([alsa/use-case.h], HAVE_ALSA_UCM=1, HAVE_ALSA_UCM=0)
         CPPFLAGS="$save_CPPFLAGS"
     ],
     HAVE_ALSA_UCM=0)
diff -Naur orig/src/modules/alsa/alsa-mixer.c patched/src/modules/alsa/alsa-mixer.c
--- orig/src/modules/alsa/alsa-mixer.c	2018-07-16 16:40:33.000000000 +0200
+++ patched/src/modules/alsa/alsa-mixer.c	2019-05-24 08:43:31.687215294 +0200
@@ -23,7 +23,7 @@
 #endif
 
 #include <sys/types.h>
-#include <asoundlib.h>
+#include <alsa/asoundlib.h>
 #include <math.h>
 
 #ifdef HAVE_VALGRIND_MEMCHECK_H
diff -Naur orig/src/modules/alsa/alsa-mixer.h patched/src/modules/alsa/alsa-mixer.h
--- orig/src/modules/alsa/alsa-mixer.h	2018-07-13 21:06:15.000000000 +0200
+++ patched/src/modules/alsa/alsa-mixer.h	2019-05-24 08:43:57.655531108 +0200
@@ -21,7 +21,7 @@
   along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.
 ***/
 
-#include <asoundlib.h>
+#include <alsa/asoundlib.h>
 
 #include <pulse/sample.h>
 #include <pulse/mainloop-api.h>
diff -Naur orig/src/modules/alsa/alsa-sink.c patched/src/modules/alsa/alsa-sink.c
--- orig/src/modules/alsa/alsa-sink.c	2018-07-16 16:40:33.000000000 +0200
+++ patched/src/modules/alsa/alsa-sink.c	2019-05-24 08:44:14.491735862 +0200
@@ -25,7 +25,7 @@
 #include <signal.h>
 #include <stdio.h>
 
-#include <asoundlib.h>
+#include <alsa/asoundlib.h>
 
 #ifdef HAVE_VALGRIND_MEMCHECK_H
 #include <valgrind/memcheck.h>
diff -Naur orig/src/modules/alsa/alsa-source.c patched/src/modules/alsa/alsa-source.c
--- orig/src/modules/alsa/alsa-source.c	2018-07-16 16:40:33.000000000 +0200
+++ patched/src/modules/alsa/alsa-source.c	2019-05-24 08:44:05.915631563 +0200
@@ -25,7 +25,7 @@
 #include <signal.h>
 #include <stdio.h>
 
-#include <asoundlib.h>
+#include <alsa/asoundlib.h>
 
 #include <pulse/rtclock.h>
 #include <pulse/timeval.h>
diff -Naur orig/src/modules/alsa/alsa-ucm.c patched/src/modules/alsa/alsa-ucm.c
--- orig/src/modules/alsa/alsa-ucm.c	2018-07-13 21:06:15.000000000 +0200
+++ patched/src/modules/alsa/alsa-ucm.c	2019-05-24 08:43:49.395430653 +0200
@@ -27,7 +27,7 @@
 #include <ctype.h>
 #include <sys/types.h>
 #include <limits.h>
-#include <asoundlib.h>
+#include <alsa/asoundlib.h>
 
 #ifdef HAVE_VALGRIND_MEMCHECK_H
 #include <valgrind/memcheck.h>
diff -Naur orig/src/modules/alsa/alsa-ucm.h patched/src/modules/alsa/alsa-ucm.h
--- orig/src/modules/alsa/alsa-ucm.h	2018-07-13 21:06:15.000000000 +0200
+++ patched/src/modules/alsa/alsa-ucm.h	2019-05-24 08:54:35.339286302 +0200
@@ -23,7 +23,7 @@
 ***/
 
 #ifdef HAVE_ALSA_UCM
-#include <use-case.h>
+#include <alsa/use-case.h>
 #else
 typedef void snd_use_case_mgr_t;
 #endif
diff -Naur orig/src/modules/alsa/alsa-util.c patched/src/modules/alsa/alsa-util.c
--- orig/src/modules/alsa/alsa-util.c	2018-07-16 16:40:33.000000000 +0200
+++ patched/src/modules/alsa/alsa-util.c	2019-05-24 08:43:22.807107299 +0200
@@ -23,7 +23,7 @@
 #endif
 
 #include <sys/types.h>
-#include <asoundlib.h>
+#include <alsa/asoundlib.h>
 
 #include <pulse/sample.h>
 #include <pulse/xmalloc.h>
diff -Naur orig/src/modules/alsa/alsa-util.h patched/src/modules/alsa/alsa-util.h
--- orig/src/modules/alsa/alsa-util.h	2018-07-16 16:40:33.000000000 +0200
+++ patched/src/modules/alsa/alsa-util.h	2019-05-24 08:43:09.710948030 +0200
@@ -21,7 +21,7 @@
   along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.
 ***/
 
-#include <asoundlib.h>
+#include <alsa/asoundlib.h>
 
 #include <pulse/sample.h>
 #include <pulse/channelmap.h>
diff -Naur orig/src/modules/alsa/module-alsa-source.c patched/src/modules/alsa/module-alsa-source.c
--- orig/src/modules/alsa/module-alsa-source.c	2018-07-13 21:06:15.000000000 +0200
+++ patched/src/modules/alsa/module-alsa-source.c	2019-05-24 08:43:40.771325771 +0200
@@ -24,7 +24,7 @@
 
 #include <stdio.h>
 
-#include <asoundlib.h>
+#include <alsa/asoundlib.h>
 
 #ifdef HAVE_VALGRIND_MEMCHECK_H
 #include <valgrind/memcheck.h>
