diff -Nur orig/src/daemon/start-pulseaudio-x11.in mod/src/daemon/start-pulseaudio-x11.in
--- orig/src/daemon/start-pulseaudio-x11.in	2020-02-23 12:00:28.000000000 +0100
+++ mod/src/daemon/start-pulseaudio-x11.in	2020-02-23 12:27:33.182571074 +0100
@@ -20,7 +20,7 @@
 if [ x"$DISPLAY" != x ] ; then
 
     @PACTL_BINARY@ load-module module-x11-publish "display=$DISPLAY xauthority=$XAUTHORITY" > /dev/null
-    @PACTL_BINARY@ load-module module-x11-cork-request "display=$DISPLAY xauthority=$XAUTHORITY" > /dev/null
+#    @PACTL_BINARY@ load-module module-x11-cork-request "display=$DISPLAY xauthority=$XAUTHORITY" > /dev/null
 
     # KDE plasma versions older than 5.17.0 use module-device-manager's routing API.
     # Check for current plasma version and load module if it's necessary.
diff -Nur orig/src/pulse/client.conf.in mod/src/pulse/client.conf.in
--- orig/src/pulse/client.conf.in	2018-11-27 11:21:00.000000000 +0100
+++ mod/src/pulse/client.conf.in	2020-02-23 12:28:14.222520698 +0100
@@ -22,7 +22,7 @@
 ; default-server =
 ; default-dbus-server =
 
-; autospawn = yes
+autospawn = no
 ; daemon-binary = @PA_BINARY@
 ; extra-arguments = --log-target=syslog
 
