diff -Naur a/config/lxqt-config-globalkeyshortcuts.desktop.in b/config/lxqt-config-globalkeyshortcuts.desktop.in
--- a/config/lxqt-config-globalkeyshortcuts.desktop.in	2022-04-16 14:50:28.000000000 +0200
+++ b/config/lxqt-config-globalkeyshortcuts.desktop.in	2022-06-16 01:39:00.791071930 +0200
@@ -2,7 +2,7 @@
 Type=Application
 Exec=lxqt-config-globalkeyshortcuts
 Icon=preferences-desktop-keyboard
-Categories=Settings;DesktopSettings;Qt;LXQt;
+Categories=Settings;DesktopSettings;Qt;
 OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
