diff -Nur orig/Makefile mod/Makefile
--- orig/Makefile	2019-11-28 20:09:04.000000000 +0100
+++ mod/Makefile	2019-11-28 20:16:42.870073442 +0100
@@ -15,15 +15,15 @@
 ##  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 DESTDIR = 
-PREFIX  = /usr/local
+PREFIX  = /usr
 DATADIR = $(PREFIX)/share/pingus
 MANDIR  = $(PREFIX)/share/man
 BINDIR  = $(PREFIX)/bin
 
 build/pingus:
 	mkdir -p build
-	scons src
-	scons
+	python2 /usr/bin/scons src
+	python2 /usr/bin/scons
 	ln -fs build/pingus
 
 clean:
