From 0a8921bf167481045830095c731eb3c67af913e4 Mon Sep 17 00:00:00 2001
From: LocutusOfBorg <none@none>
Date: Fri, 23 Sep 2022 12:47:47 -0400
Subject: [PATCH] Fix clang-15 compile error

---
 project_files/hwc/rtl/sysutils.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/project_files/hwc/rtl/sysutils.c b/project_files/hwc/rtl/sysutils.c
index bd1470f327..724c5c74b8 100644
--- a/project_files/hwc/rtl/sysutils.c
+++ b/project_files/hwc/rtl/sysutils.c
@@ -83,7 +83,7 @@ TDateTime fpcrtl_now()
 // Semi-dummy implementation of FormatDateTime
 string255 fpcrtl_formatDateTime(string255 FormatStr, TDateTime DateTime)
 {
-    string255 buffer = STRINIT(FormatStr.str);
+    string255 buffer = FormatStr;
     time_t rawtime;
     struct tm* my_tm;
