diff -Naur orig/QTfrontend/ui/page/pagegamestats.cpp patched/QTfrontend/ui/page/pagegamestats.cpp
--- orig/QTfrontend/ui/page/pagegamestats.cpp	2019-07-30 18:58:48.000000000 +0200
+++ patched/QTfrontend/ui/page/pagegamestats.cpp	2020-06-04 12:04:03.513755529 +0200
@@ -21,6 +21,7 @@
 #include <QHBoxLayout>
 #include <QGraphicsScene>
 #include <QGroupBox>
+#include <QPainterPath>
 #include <QSizePolicy>
 
 #include "pagegamestats.h"
