diff -Naur orig/cmake_modules/paths.cmake mod/cmake_modules/paths.cmake
--- orig/cmake_modules/paths.cmake	2015-07-30 23:31:04.000000000 +0200
+++ mod/cmake_modules/paths.cmake	2016-04-26 08:05:20.593159336 +0200
@@ -67,5 +67,5 @@
     # - the third one is the full path of the system dir
     #source http://www.cmake.org/pipermail/cmake/2008-January/019290.html
     #skip this if the install prefix is the standard one
-    set(CMAKE_INSTALL_RPATH "$ORIGIN/../${target_library_install_dir}/:$ORIGIN/:${CMAKE_INSTALL_PREFIX}/${target_library_install_dir}/")
+    #set(CMAKE_INSTALL_RPATH "$ORIGIN/../${target_library_install_dir}/:$ORIGIN/:${CMAKE_INSTALL_PREFIX}/${target_library_install_dir}/")
 endif(APPLE)
