diff -Naur keepassx-0.4.3-orig/share/mimelnk/application/x-keepass.desktop keepassx-0.4.3/share/mimelnk/application/x-keepass.desktop
--- keepassx-0.4.3-orig/share/mimelnk/application/x-keepass.desktop	2008-10-02 16:11:35.000000000 +0200
+++ keepassx-0.4.3/share/mimelnk/application/x-keepass.desktop	2011-03-09 19:15:58.205105796 +0100
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Comment=KeePass database
 Comment[de]=KeePass Datenbank
+Comment[hu]=KeePass adatbázis
 Hidden=false
 Icon=keepassx
 MimeType=application/x-keepass;
