diff -Nur orig/utils/vlaunch/vlaunch.cpp mod/utils/vlaunch/vlaunch.cpp
--- orig/utils/vlaunch/vlaunch.cpp	2018-02-12 19:51:06.000000000 +0100
+++ mod/utils/vlaunch/vlaunch.cpp	2018-02-12 20:21:41.741071704 +0100
@@ -142,7 +142,7 @@
 
 	mainsizer->Add(new wxStaticBitmap(panel, -1, wxBitmap(vavoom_xpm)));
 
-	wxNotebook* nbook = new wxNotebook(panel, -1, wxPoint(0, 105), wxSize(447, 270));
+	wxNotebook* nbook = new wxNotebook(panel, -1, wxPoint(0, 105), wxSize(447, 360));
 	mainsizer->Add(nbook);
 
 	wxPanel* page = new wxPanel(nbook);
