diff -up vavoom-1.33/utils/vcc/vcc.cpp~ vavoom-1.33/utils/vcc/vcc.cpp
--- vavoom-1.33/utils/vcc/vcc.cpp~	2010-04-21 15:15:40.000000000 +0200
+++ vavoom-1.33/utils/vcc/vcc.cpp	2014-06-20 15:15:47.297819212 +0200
@@ -122,7 +122,7 @@ int main(int argc, char **argv)
 	}
 	catch (VException& e)
 	{
-		FatalError(e.What());
+		FatalError("%s", e.What());
 	}
 	return 0;
 }
diff -up vavoom-1.33/source/misc.cpp~ vavoom-1.33/source/misc.cpp
--- vavoom-1.33/source/misc.cpp~	2010-09-01 20:26:05.000000000 +0200
+++ vavoom-1.33/source/misc.cpp	2014-06-20 15:23:18.803404958 +0200
@@ -119,7 +119,7 @@ void FOutputDevice::Logf(EName Type, con
 
 void VLogSysError::Serialise(const char* V, EName)
 {
-	Sys_Error(V);
+	Sys_Error("%s", V);
 }
 
 //==========================================================================
@@ -130,7 +130,7 @@ void VLogSysError::Serialise(const char*
 
 void VLogHostError::Serialise(const char* V, EName)
 {
-	Host_Error(V);
+	Host_Error("%s", V);
 }
 
 //==========================================================================
diff -up vavoom-1.33/source/vc_object.cpp~ vavoom-1.33/source/vc_object.cpp
--- vavoom-1.33/source/vc_object.cpp~	2010-06-04 23:20:21.000000000 +0200
+++ vavoom-1.33/source/vc_object.cpp	2014-06-20 15:27:33.178529910 +0200
@@ -477,12 +477,12 @@ IMPLEMENT_FUNCTION(VObject, IsDestroyed)
 
 IMPLEMENT_FUNCTION(VObject, Error)
 {
-	Host_Error(*PF_FormatString());
+	Host_Error("%s", *PF_FormatString());
 }
 
 IMPLEMENT_FUNCTION(VObject, FatalError)
 {
-	Sys_Error(*PF_FormatString());
+	Sys_Error("%s", *PF_FormatString());
 }
 
 //**************************************************************************
