diff -Naur orig/utils/vlaunch/vlaunch.cpp patched/utils/vlaunch/vlaunch.cpp
--- orig/utils/vlaunch/vlaunch.cpp	2011-01-01 12:51:56.134741211 +0100
+++ patched/utils/vlaunch/vlaunch.cpp	2017-06-11 02:34:45.178181259 +0200
@@ -133,7 +133,7 @@
 
 VMain::VMain()
 : wxFrame(NULL, wxID_ANY, wxT("Vavoom launcher"), wxDefaultPosition, wxDefaultSize,
-	wxDEFAULT_FRAME_STYLE & ~ (wxRESIZE_BORDER | wxRESIZE_BOX | wxMAXIMIZE_BOX))
+	wxDEFAULT_FRAME_STYLE & ~ (wxRESIZE_BORDER | wxMAXIMIZE_BOX))
 {
 	SetIcon(wxICON(vlaunch));
 
