diff -Naur orig/mk/syslinux.mk patched/mk/syslinux.mk
--- orig/mk/syslinux.mk	2014-10-06 18:27:44.000000000 +0200
+++ patched/mk/syslinux.mk	2019-08-28 01:20:33.177723349 +0200
@@ -19,12 +19,12 @@
 MAKE      += -r
 
 BINDIR   = /usr/bin
-SBINDIR  = /sbin
+SBINDIR  = /usr/sbin
 LIBDIR   = /usr/lib
 DATADIR  = /usr/share
 AUXDIR   = $(DATADIR)/syslinux
 DIAGDIR  = $(AUXDIR)/diag
-MANDIR	 = /usr/man
+MANDIR	 = /usr/share/man
 INCDIR   = /usr/include
 TFTPBOOT = /tftpboot
 COM32DIR = $(AUXDIR)/com32
