diff -Naur orig/Makefile patched/Makefile
--- orig/Makefile	2014-10-06 18:27:44.000000000 +0200
+++ patched/Makefile	2019-08-28 03:58:24.319173322 +0200
@@ -42,7 +42,7 @@
 
 # If the output directory does not exist we bail because that is the
 # least surprising thing to do.
-cd-output := $(shell cd $(OBJDIR) && /bin/pwd)
+cd-output := $(shell cd $(OBJDIR) && pwd)
 $(if $(cd-output),, \
 	$(error output directory "$(OBJDIR)" does not exist))
 
