diff -Naur a/lxqt-config-session/lxqt-config-session.desktop.in b/lxqt-config-session/lxqt-config-session.desktop.in
--- a/lxqt-config-session/lxqt-config-session.desktop.in	2022-05-08 08:43:56.000000000 +0200
+++ b/lxqt-config-session/lxqt-config-session.desktop.in	2022-06-16 01:46:57.788858173 +0200
@@ -2,7 +2,7 @@
 Type=Application
 Exec=lxqt-config-session
 Icon=preferences-system-session-services
-Categories=Settings;DesktopSettings;Qt;LXQt;
+Categories=Settings;DesktopSettings;Qt;
 OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
