diff -Naur a/src/scanner.h b/src/scanner.h
--- a/src/scanner.h	2022-07-27 02:21:42.000000000 +0200
+++ b/src/scanner.h	2022-07-27 02:29:25.190421258 +0200
@@ -8,6 +8,8 @@
 #include <QtCore/QList>
 #include <QtCore/QMap>
 
+#include <optional>
+
 #include "infectedfile.h"
 #include "treeitem.h"
 
diff -Naur a/src/timedactiondialogue.h b/src/timedactiondialogue.h
--- a/src/timedactiondialogue.h	2022-07-27 02:21:42.000000000 +0200
+++ b/src/timedactiondialogue.h	2022-07-27 02:32:13.442898840 +0200
@@ -8,6 +8,8 @@
 #include <QtWidgets/QDialog>
 #include <QtCore/QTimer>
 
+#include <optional>
+
 namespace Ui {
     class TimedActionDialogue;
 }
