diff -Nur orig/scripts/synce-serial.in mod/scripts/synce-serial.in
--- orig/scripts/synce-serial.in	2011-02-27 11:27:49.000000000 +0100
+++ mod/scripts/synce-serial.in	2019-12-05 10:12:33.910285789 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 
 # Some serial devices will not be picked up automatically by synce,
 # eg those on a serial to usb bridge. This tool will initiate connection.
diff -Nur orig/scripts/udev-synce-rndis.in mod/scripts/udev-synce-rndis.in
--- orig/scripts/udev-synce-rndis.in	2011-02-27 11:26:02.000000000 +0100
+++ mod/scripts/udev-synce-rndis.in	2019-12-05 10:12:58.926240546 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 
 import sys
 import os
diff -Nur orig/scripts/udev-synce-serial.in mod/scripts/udev-synce-serial.in
--- orig/scripts/udev-synce-serial.in	2011-02-27 11:26:02.000000000 +0100
+++ mod/scripts/udev-synce-serial.in	2019-12-05 10:13:18.358204943 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 
 import sys
 import os
diff -Nur orig/src/synce-unlock-udev.py mod/src/synce-unlock-udev.py
--- orig/src/synce-unlock-udev.py	2011-03-04 16:13:04.000000000 +0100
+++ mod/src/synce-unlock-udev.py	2019-12-05 10:16:18.925857413 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 
 # A simple script to provide a password to a locked Windows
 # Mobile device from the command line
