diff -Naur orig/data/nemo.desktop.in patched/data/nemo.desktop.in
--- orig/data/nemo.desktop.in	2018-11-01 14:30:30.000000000 +0100
+++ patched/data/nemo.desktop.in	2018-11-08 02:53:05.258929485 +0100
@@ -42,8 +42,8 @@
 Name[fr_CA]=Fichiers
 Name[fr_FR]=Fichiers
 Name[hr]=Nemo
-Name[hu]=Fájlok
-Name[hu_HU]=Fájlok
+Name[hu]=Nemo
+Name[hu_HU]=Nemo
 Name[id]=Berkas
 Name[id_ID]=Berkas
 Name[is]=Skrár
