diff -Naur orig/Jambase patched/Jambase
--- orig/Jambase	2016-09-28 04:58:53.000000000 +0200
+++ patched/Jambase	2017-09-14 00:09:45.813313207 +0200
@@ -987,8 +987,8 @@
 	 || $(HOSTTYPE) = amd64 {
 		ECHO "We're on a 64 bit host" ;
 		HOST64     = true ;
-		CCFLAGS	   += -m64 ;
-		C++FLAGS   += -m64 ;
+	#	CCFLAGS	   += -m64 ;
+	#	C++FLAGS   += -m64 ;
 	}
 
 	# Adding --hash-style=sysv to the compiler options
