diff -Naur orig/Source/WebCore/platform/text/enchant/TextCheckerEnchant.cpp patched/Source/WebCore/platform/text/enchant/TextCheckerEnchant.cpp
--- orig/Source/WebCore/platform/text/enchant/TextCheckerEnchant.cpp	2016-04-10 08:48:37.000000000 +0200
+++ patched/Source/WebCore/platform/text/enchant/TextCheckerEnchant.cpp	2018-02-15 02:40:26.253201454 +0100
@@ -128,7 +128,7 @@
         for (i = 0; i < numberOfSuggestions; i++)
             guesses.append(String::fromUTF8(suggestions[i]));
 
-        enchant_dict_free_suggestions(*iter, suggestions);
+        enchant_dict_free_string_list(*iter, suggestions);
     }
 
     return guesses;
