diff -Naur orig/tools/stunnel.service.in patched/tools/stunnel.service.in
--- orig/tools/stunnel.service.in	2016-12-13 12:28:35.134741323 +0100
+++ patched/tools/stunnel.service.in	2017-05-12 10:51:42.485533340 +0200
@@ -1,10 +1,11 @@
 [Unit]
 Description=TLS tunnel for network daemons
-After=syslog.target
+After=network.target
 
 [Service]
 ExecStart=@bindir@/stunnel
 Type=forking
+PrivateTmp=true
 
 [Install]
 WantedBy=multi-user.target
