diff -Naur orig/distro-configs/LibreOfficeUHU.conf patched/distro-configs/LibreOfficeUHU.conf
--- orig/distro-configs/LibreOfficeUHU.conf	1970-01-01 01:00:00.000000000 +0100
+++ patched/distro-configs/LibreOfficeUHU.conf	2017-12-24 05:10:36.140381439 +0100
@@ -0,0 +1,51 @@
+--prefix=/usr
+--sysconfdir=/etc
+--localstatedir=/var
+--mandir=/usr/share/man
+--infodir=/usr/share/info
+--with-lang=en-GB hu
+--with-vendor=ubk.hu
+--with-help
+--with-fonts
+--with-external-tar=src
+--with-system-libs
+--with-system-headers
+--without-java
+--without-system-jars
+--without-system-bluez
+--without-system-libabw
+--without-system-libcdr
+--without-system-libcmis
+--without-system-libebook
+--without-system-libepubgen
+--without-system-libexttextcat
+--without-system-libfreehand
+--without-system-libmspub
+--without-system-libmwaw
+--without-system-libpagemaker
+--without-system-libqxp
+--without-system-libstaroffice
+--without-system-libzmf
+--without-system-lpsolve
+--without-system-mysql-cppconn
+--without-system-mythes
+--without-system-orcus
+--without-system-xmlsec
+--disable-coinmp
+--disable-dbgutil
+--disable-debug
+--disable-gstreamer-0-10
+--disable-odk
+--disable-firebird-sdbc
+--disable-pdfimport
+--enable-avahi
+--enable-fetch-external
+--enable-gstreamer-1-0
+--enable-postgresql-sdbc
+--enable-python
+--enable-ext-mariadb-connector
+--enable-ext-nlpsolver
+--enable-gio
+--enable-gtk3
+--enable-qt5
+--enable-release-build
