diff -Naur a/external/more_fonts/Module_more_fonts.mk b/external/more_fonts/Module_more_fonts.mk
--- a/external/more_fonts/Module_more_fonts.mk	2023-07-07 18:45:40.000000000 +0200
+++ b/external/more_fonts/Module_more_fonts.mk	2023-07-12 22:06:18.992600132 +0200
@@ -15,13 +15,9 @@
 	ExternalPackage_caladea \
 	ExternalPackage_carlito \
 	$(if $(MPL_SUBSET),,ExternalPackage_culmus) \
-	ExternalPackage_dejavu \
 	ExternalPackage_gentium \
-	ExternalPackage_liberation \
 	ExternalPackage_liberation_narrow \
-	ExternalPackage_libertineg \
 	ExternalPackage_libre_hebrew \
-	ExternalPackage_noto \
 	ExternalPackage_reem \
 	ExternalPackage_scheherazade \
 	UnpackedTarball_alef \
@@ -29,13 +25,9 @@
 	UnpackedTarball_caladea \
 	UnpackedTarball_carlito \
 	$(if $(MPL_SUBSET),,UnpackedTarball_culmus) \
-	UnpackedTarball_dejavu \
 	UnpackedTarball_gentium \
-	UnpackedTarball_liberation \
 	UnpackedTarball_liberation_narrow \
-	UnpackedTarball_libertineg \
 	UnpackedTarball_libre_hebrew \
-	UnpackedTarball_noto \
 	UnpackedTarball_reem \
 	UnpackedTarball_scheherazade \
 ))
diff -Naur a/Repository.mk b/Repository.mk
--- a/Repository.mk	2023-07-07 18:45:40.000000000 +0200
+++ b/Repository.mk	2023-07-12 22:07:16.264741000 +0200
@@ -1045,13 +1045,9 @@
 		fonts_caladea \
 		fonts_carlito \
 		$(if $(MPL_SUBSET),,fonts_culmus) \
-		fonts_dejavu \
 		fonts_gentium \
-		fonts_liberation \
 		fonts_liberation_narrow \
-		fonts_libertineg \
 		fonts_libre_hebrew \
-		fonts_noto \
 		fonts_reem \
 		fonts_scheherazade \
 	) \
