diff -Naur orig/external/more_fonts/Module_more_fonts.mk patched/external/more_fonts/Module_more_fonts.mk
--- orig/external/more_fonts/Module_more_fonts.mk	2019-07-05 11:57:59.000000000 +0200
+++ patched/external/more_fonts/Module_more_fonts.mk	2019-07-10 01:12:50.997293816 +0200
@@ -15,17 +15,13 @@
 	ExternalPackage_caladea \
 	ExternalPackage_carlito \
 	$(if $(MPL_SUBSET),,ExternalPackage_culmus) \
-	ExternalPackage_dejavu \
 	ExternalPackage_gentium \
 	$(if $(MPL_SUBSET),,ExternalPackage_kacst) \
-	ExternalPackage_liberation \
 	ExternalPackage_liberation_narrow \
-	ExternalPackage_libertineg \
 	ExternalPackage_libre_hebrew \
 	ExternalPackage_sourcecode \
 	ExternalPackage_sourcesans \
 	ExternalPackage_sourceserif \
-	ExternalPackage_noto \
 	ExternalPackage_reem \
 	ExternalPackage_scheherazade \
 	ExternalPackage_EmojiOne_Color \
@@ -34,17 +30,13 @@
 	UnpackedTarball_caladea \
 	UnpackedTarball_carlito \
 	$(if $(MPL_SUBSET),,UnpackedTarball_culmus) \
-	UnpackedTarball_dejavu \
 	UnpackedTarball_gentium \
 	$(if $(MPL_SUBSET),,UnpackedTarball_kacst) \
-	UnpackedTarball_liberation \
 	UnpackedTarball_liberation_narrow \
-	UnpackedTarball_libertineg \
 	UnpackedTarball_libre_hebrew \
 	UnpackedTarball_sourcecode \
 	UnpackedTarball_sourcesans \
 	UnpackedTarball_sourceserif \
-	UnpackedTarball_noto \
 	UnpackedTarball_reem \
 	UnpackedTarball_scheherazade \
 	UnpackedTarball_EmojiOne_Color \
diff -Naur orig/Repository.mk patched/Repository.mk
--- orig/Repository.mk	2019-07-05 11:57:59.000000000 +0200
+++ patched/Repository.mk	2019-07-10 01:11:19.396423862 +0200
@@ -976,15 +976,11 @@
 		fonts_caladea \
 		fonts_carlito \
 		$(if $(MPL_SUBSET),,fonts_culmus) \
-		fonts_dejavu \
 		fonts_emojione_color \
 		fonts_gentium \
 		$(if $(MPL_SUBSET),,fonts_kacst) \
-		fonts_liberation \
 		fonts_liberation_narrow \
-		fonts_libertineg \
 		fonts_libre_hebrew \
-		fonts_noto \
 		fonts_reem \
 		fonts_sourcecode \
 		fonts_sourcesans \
