diff -Naur orig/solenv/bin/modules/installer/worker.pm patched/solenv/bin/modules/installer/worker.pm
--- orig/solenv/bin/modules/installer/worker.pm	2015-10-10 21:45:34.860542834 +0200
+++ patched/solenv/bin/modules/installer/worker.pm	2015-10-10 19:36:23.000000000 +0200
@@ -96,7 +96,6 @@
 
     $installdir = installer::systemactions::create_directories("install", $languageref);
     installer::logger::print_message( "... creating installation set in $installdir ...\n" );
-    remove_old_installation_sets($installdir);
     my $inprogressinstalldir = $installdir . "_inprogress";
     installer::systemactions::rename_directory($installdir, $inprogressinstalldir);
     $installdir = $inprogressinstalldir;
