diff -Naur a/cui/source/options/optjava.cxx b/cui/source/options/optjava.cxx
--- a/cui/source/options/optjava.cxx	2023-07-07 18:45:40.000000000 +0200
+++ b/cui/source/options/optjava.cxx	2023-07-13 01:33:32.156458310 +0200
@@ -935,6 +935,7 @@
     return sPath.makeStringAndClear();
 }
 
+#if HAVE_FEATURE_JAVA
 void SvxJavaClassPathDlg::SetClassPath( const OUString& _rPath )
 {
     if ( m_sOldPath.isEmpty() )
@@ -963,5 +964,6 @@
     }
     SelectHdl_Impl(*m_xPathList);
 }
+#endif
 
 /* vim:set shiftwidth=4 softtabstop=4 expandtab: */
diff -Naur a/cui/source/options/optjava.hxx b/cui/source/options/optjava.hxx
--- a/cui/source/options/optjava.hxx	2023-07-07 18:45:40.000000000 +0200
+++ b/cui/source/options/optjava.hxx	2023-07-13 01:34:56.480920239 +0200
@@ -200,7 +200,9 @@
     void             SetFocus() { m_xPathList->grab_focus(); }
 
     OUString                GetClassPath() const;
+#if HAVE_FEATURE_JAVA
     void                    SetClassPath( const OUString& _rPath );
+#endif
 };
 
 /* vim:set shiftwidth=4 softtabstop=4 expandtab: */
