diff -Naur orig/vcl/unx/generic/plugadapt/salplug.cxx patched/vcl/unx/generic/plugadapt/salplug.cxx
--- orig/vcl/unx/generic/plugadapt/salplug.cxx	2018-03-13 00:31:19.811414437 +0100
+++ patched/vcl/unx/generic/plugadapt/salplug.cxx	2018-03-13 00:34:00.078660425 +0100
@@ -170,7 +170,7 @@
 #if ENABLE_KDE4
         "kde4",
 #endif
-        "gtk3", "gtk", "gen", nullptr
+        "gtk", "gtk3", "gen", nullptr
     };
 
     static const char* const pStandardFallbackList[] =
