diff -Naur a/src/via_driver.h b/src/via_driver.h
--- a/src/via_driver.h	2017-02-26 07:28:58.000000000 +0100
+++ b/src/via_driver.h	2023-08-02 19:52:23.578157389 +0200
@@ -394,8 +394,8 @@
 
 
 /* In via_display.c. */
-const xf86CrtcFuncsRec iga1_crtc_funcs;
-const xf86CrtcFuncsRec iga2_crtc_funcs;
+extern const xf86CrtcFuncsRec iga1_crtc_funcs;
+extern const xf86CrtcFuncsRec iga2_crtc_funcs;
 
 /* In via_exa.c. */
 Bool viaInitExa(ScreenPtr pScreen);
diff -Naur a/src/via_sii164.h b/src/via_sii164.h
--- a/src/via_sii164.h	2017-02-27 00:53:37.000000000 +0100
+++ b/src/via_sii164.h	2023-08-02 19:52:51.550263302 +0200
@@ -47,7 +47,7 @@
 } viaSiI164Rec, *viaSiI164RecPtr;
 
 
-const xf86OutputFuncsRec via_sii164_funcs;
+extern const xf86OutputFuncsRec via_sii164_funcs;
 
 Bool viaSiI164Init(ScrnInfoPtr pScrn, I2CBusPtr pI2CBus);
 
diff -Naur a/src/via_vt1632.h b/src/via_vt1632.h
--- a/src/via_vt1632.h	2017-01-06 04:33:01.000000000 +0100
+++ b/src/via_vt1632.h	2023-08-02 19:53:07.126322262 +0200
@@ -44,7 +44,7 @@
 } viaVT1632Rec, *viaVT1632RecPtr;
 
 
-const xf86OutputFuncsRec via_vt1632_funcs;
+extern const xf86OutputFuncsRec via_vt1632_funcs;
 
 Bool viaVT1632Init(ScrnInfoPtr pScrn, I2CBusPtr pI2CBus);
 
