diff -Naur orig/data/icons/Makefile.am patched/data/icons/Makefile.am
--- orig/data/icons/Makefile.am	2013-03-03 00:54:57.000000000 +0100
+++ patched/data/icons/Makefile.am	2013-12-25 13:00:50.657777529 +0100
@@ -1,5 +1,5 @@
 MAINTAINERCLEANFILES = Makefile.in
-filesdir = $(datadir)/icons
+filesdir = $(datadir)/pixmaps
 files_DATA = \
 terminology.png
 
diff -Naur orig/data/icons/Makefile.in patched/data/icons/Makefile.in
--- orig/data/icons/Makefile.in	2013-12-07 18:27:08.000000000 +0100
+++ patched/data/icons/Makefile.in	2013-12-25 13:01:32.178160809 +0100
@@ -201,7 +201,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 MAINTAINERCLEANFILES = Makefile.in
-filesdir = $(datadir)/icons
+filesdir = $(datadir)/pixmaps
 files_DATA = \
 terminology.png
 
