diff -Naur a/src/filemanager/boxes.c b/src/filemanager/boxes.c
--- a/src/filemanager/boxes.c	2024-08-23 19:22:35.000000000 +0200
+++ b/src/filemanager/boxes.c	2024-08-28 15:51:52.047931036 +0200
@@ -103,9 +103,11 @@
 static unsigned long panel_list_formats_id, panel_user_format_id, panel_brief_cols_id;
 static unsigned long user_mini_status_id, mini_user_format_id;
 
+#if 0
 #ifdef HAVE_CHARSET
 static int new_display_codepage;
 #endif /* HAVE_CHARSET */
+#endif
 
 #if defined(ENABLE_VFS) && defined(ENABLE_VFS_FTP)
 static unsigned long ftpfs_always_use_proxy_id, ftpfs_proxy_host_id;
@@ -363,6 +365,7 @@
 
 /* --------------------------------------------------------------------------------------------- */
 
+#if 0
 #ifdef HAVE_CHARSET
 static int
 sel_charset_button (WButton *button, int action)
@@ -393,6 +396,7 @@
     return 0;
 }
 #endif /* HAVE_CHARSET */
+#endif
 
 /* --------------------------------------------------------------------------------------------- */
 
@@ -950,6 +954,7 @@
 
 /* --------------------------------------------------------------------------------------------- */
 
+#if 0
 #ifndef HAVE_CHARSET
 void
 display_bits_box (void)
@@ -1068,6 +1073,7 @@
     }
 }
 #endif /* HAVE_CHARSET */
+#endif
 
 /* --------------------------------------------------------------------------------------------- */
 /** Show tree in a box, not on a panel */
diff -Naur a/src/filemanager/boxes.h b/src/filemanager/boxes.h
--- a/src/filemanager/boxes.h	2024-08-23 19:22:35.000000000 +0200
+++ b/src/filemanager/boxes.h	2024-08-28 15:51:52.047931036 +0200
@@ -25,7 +25,9 @@
                        int *brief_cols);
 const panel_field_t *sort_box (dir_sort_options_t * op, const panel_field_t * sort_field);
 void confirm_box (void);
+#if 0
 void display_bits_box (void);
+#endif
 void configure_vfs_box (void);
 void jobs_box (void);
 char *cd_box (const WPanel * panel);
diff -Naur a/src/filemanager/filemanager.c b/src/filemanager/filemanager.c
--- a/src/filemanager/filemanager.c	2024-08-23 19:22:35.000000000 +0200
+++ b/src/filemanager/filemanager.c	2024-08-28 15:51:52.047931036 +0200
@@ -1200,9 +1200,11 @@
         diff_view_cmd ();
         break;
 #endif
+#if 0
     case CK_OptionsDisplayBits:
         display_bits_box ();
         break;
+#endif
     case CK_Edit:
         edit_cmd (current_panel);
         break;
diff -Naur a/src/setup.c b/src/setup.c
--- a/src/setup.c	2024-08-23 19:22:35.000000000 +0200
+++ b/src/setup.c	2024-08-28 15:51:52.047931036 +0200
@@ -304,11 +304,13 @@
     { "confirm_view_dir", &confirm_view_dir },
     { "safe_delete", &safe_delete },
     { "safe_overwrite", &safe_overwrite },
+#if 0
 #ifndef HAVE_CHARSET
     { "eight_bit_clean", &mc_global.eight_bit_clean },
     { "full_eight_bits", &mc_global.full_eight_bits },
 #endif /* !HAVE_CHARSET */
     { "use_8th_bit_as_meta", &use_8th_bit_as_meta },
+#endif
     { "mouse_move_pages_viewer", &mcview_mouse_move_pages },
     { "mouse_close_dialog", &mouse_close_dialog},
     { "fast_refresh", &fast_refresh },
