diff -Naur orig/src/filemanager/boxes.c patched/src/filemanager/boxes.c
--- orig/src/filemanager/boxes.c	2021-01-10 12:50:16.000000000 +0100
+++ patched/src/filemanager/boxes.c	2021-02-12 22:21:38.218832636 +0100
@@ -104,9 +104,11 @@
 static unsigned long panel_list_formats_id, panel_user_format_id, panel_brief_cols_id;
 static unsigned long user_mini_status_id, mini_user_format_id;
 
+#if 0
 #ifdef HAVE_CHARSET
 static int new_display_codepage;
 #endif /* HAVE_CHARSET */
+#endif
 
 #if defined(ENABLE_VFS) && defined(ENABLE_VFS_FTP)
 static unsigned long ftpfs_always_use_proxy_id, ftpfs_proxy_host_id;
@@ -364,6 +366,7 @@
 
 /* --------------------------------------------------------------------------------------------- */
 
+#if 0
 #ifdef HAVE_CHARSET
 static int
 sel_charset_button (WButton * button, int action)
@@ -394,6 +397,7 @@
     return 0;
 }
 #endif /* HAVE_CHARSET */
+#endif
 
 /* --------------------------------------------------------------------------------------------- */
 
@@ -941,6 +945,7 @@
 
 /* --------------------------------------------------------------------------------------------- */
 
+#if 0
 #ifndef HAVE_CHARSET
 void
 display_bits_box (void)
@@ -1057,6 +1062,7 @@
     }
 }
 #endif /* HAVE_CHARSET */
+#endif
 
 /* --------------------------------------------------------------------------------------------- */
 /** Show tree in a box, not on a panel */
diff -Naur orig/src/filemanager/boxes.h patched/src/filemanager/boxes.h
--- orig/src/filemanager/boxes.h	2021-01-10 12:50:16.000000000 +0100
+++ patched/src/filemanager/boxes.h	2021-02-12 22:21:38.218832636 +0100
@@ -25,7 +25,9 @@
                        int *brief_cols);
 const panel_field_t *sort_box (dir_sort_options_t * op, const panel_field_t * sort_field);
 void confirm_box (void);
+#if 0
 void display_bits_box (void);
+#endif
 void configure_vfs_box (void);
 void jobs_box (void);
 char *cd_box (const WPanel * panel);
diff -Naur orig/src/filemanager/filemanager.c patched/src/filemanager/filemanager.c
--- orig/src/filemanager/filemanager.c	2021-01-10 12:50:16.000000000 +0100
+++ patched/src/filemanager/filemanager.c	2021-02-12 22:24:02.807410142 +0100
@@ -1217,9 +1217,11 @@
         diff_view_cmd ();
         break;
 #endif
+#if 0
     case CK_OptionsDisplayBits:
         display_bits_box ();
         break;
+#endif
     case CK_Edit:
         edit_cmd (current_panel);
         break;
diff -Naur orig/src/setup.c patched/src/setup.c
--- orig/src/setup.c	2021-01-10 12:50:16.000000000 +0100
+++ patched/src/setup.c	2021-02-12 22:21:39.894838908 +0100
@@ -302,11 +302,13 @@
     { "confirm_view_dir", &confirm_view_dir },
     { "safe_delete", &safe_delete },
     { "safe_overwrite", &safe_overwrite },
+#if 0
 #ifndef HAVE_CHARSET
     { "eight_bit_clean", &mc_global.eight_bit_clean },
     { "full_eight_bits", &mc_global.full_eight_bits },
 #endif /* !HAVE_CHARSET */
     { "use_8th_bit_as_meta", &use_8th_bit_as_meta },
+#endif
     { "mouse_move_pages_viewer", &mcview_mouse_move_pages },
     { "mouse_close_dialog", &mouse_close_dialog},
     { "fast_refresh", &fast_refresh },
