diff -Naur orig/misc/mc.default.keymap patched/misc/mc.default.keymap
--- orig/misc/mc.default.keymap	2017-05-02 21:00:18.134739579 +0200
+++ patched/misc/mc.default.keymap	2017-12-27 17:55:38.768509547 +0100
@@ -124,10 +124,10 @@
 # SortPrev =
 # SortNext =
 # SortReverse =
-# SortByName =
-# SortByExt =
-# SortBySize =
-# SortByMTime =
+SortByName = alt-shift-n
+SortByExt = alt-shift-e
+SortBySize = alt-shift-s
+SortByMTime = alt-shift-t
 # ScrollLeft =
 # ScrollRight =
 
