diff -Naur orig/buildconfig/config_unix.py patched/buildconfig/config_unix.py
--- orig/buildconfig/config_unix.py	1970-01-01 01:00:00.000000000 +0100
+++ patched/buildconfig/config_unix.py	2021-01-02 23:03:17.937341493 +0100
@@ -211,14 +211,6 @@
         Dependency('SCRAP', '', 'libX11', ['X11']),
         #Dependency('GFX', 'SDL_gfxPrimitives.h', 'libSDL_gfx.so', ['SDL_gfx']),
     ])
-    is_freebsd = 'FreeBSD' in platform.system()
-    is_hurd = platform.system() == 'GNU'
-    if not is_freebsd and not is_hurd:
-        porttime_dep = get_porttime_dep()
-        DEPS.append(
-            Dependency('PORTMIDI', 'portmidi.h', 'libportmidi.so', ['portmidi'])
-        )
-        DEPS.append(porttime_dep)
     DEPS.append(find_freetype())
 
     if not DEPS[0].found:
diff -Naur orig/buildconfig/Setup.SDL1.in patched/buildconfig/Setup.SDL1.in
--- orig/buildconfig/Setup.SDL1.in	1970-01-01 01:00:00.000000000 +0100
+++ patched/buildconfig/Setup.SDL1.in	2021-01-02 23:01:26.709197442 +0100
@@ -14,8 +14,6 @@
 PNG = -lpng
 JPEG = -ljpeg
 SCRAP = -lX11
-PORTMIDI = -lportmidi
-PORTTIME = -lporttime
 FREETYPE = -lfreetype
 #--EndConfig
 
@@ -30,7 +28,6 @@
 mixer src_c/mixer.c $(SDL) $(MIXER) $(DEBUG)
 mixer_music src_c/music.c $(SDL) $(MIXER) $(DEBUG)
 scrap src_c/scrap.c $(SDL) $(SCRAP) $(DEBUG)
-pypm src_c/pypm.c $(SDL) $(PORTMIDI) $(PORTTIME) $(DEBUG)
 
 GFX = src_c/SDL_gfx/SDL_gfxPrimitives.c
 #GFX = src_c/SDL_gfx/SDL_gfxBlitFunc.c src_c/SDL_gfx/SDL_gfxPrimitives.c
diff -Naur orig/buildconfig/Setup.SDL2.in patched/buildconfig/Setup.SDL2.in
--- orig/buildconfig/Setup.SDL2.in	1970-01-01 01:00:00.000000000 +0100
+++ patched/buildconfig/Setup.SDL2.in	2021-01-02 23:01:11.769178229 +0100
@@ -14,8 +14,6 @@
 PNG = -lpng
 JPEG = -ljpeg
 SCRAP = -lX11
-PORTMIDI = -lportmidi
-PORTTIME = -lporttime
 FREETYPE = -lfreetype
 #--EndConfig
 
@@ -30,7 +28,6 @@
 mixer src_c/mixer.c $(SDL) $(MIXER) $(DEBUG)
 mixer_music src_c/music.c $(SDL) $(MIXER) $(DEBUG)
 scrap src_c/scrap.c $(SDL) $(SCRAP) $(DEBUG)
-pypm src_c/pypm.c $(SDL) $(PORTMIDI) $(PORTTIME) $(DEBUG)
 
 _sdl2.sdl2 src_c/_sdl2/sdl2.c $(SDL) $(DEBUG) -Isrc_c
 _sdl2.audio src_c/_sdl2/audio.c $(SDL) $(DEBUG) -Isrc_c
