diff --git a/grub/util/grub-mkconfig.in b/grub/util/grub-mkconfig.in
index 3390ba9..c416489 100644
--- a/grub/util/grub-mkconfig.in
+++ b/grub/util/grub-mkconfig.in
@@ -205,6 +205,8 @@
   GRUB_DEVICE_BOOT \
   GRUB_DEVICE_BOOT_UUID \
   GRUB_DISABLE_OS_PROBER \
+  GRUB_COLOR_NORMAL \
+  GRUB_COLOR_HIGHLIGHT \
   GRUB_FS \
   GRUB_FONT \
   GRUB_PRELOAD_MODULES \
diff --git a/grub/util/grub.d/00_header.in b/grub/util/grub.d/00_header.in
index d2e7252..8259f45 100644
--- a/grub/util/grub.d/00_header.in
+++ b/grub/util/grub.d/00_header.in
@@ -125,6 +125,14 @@ cat <<EOF

 EOF

+if [ x$GRUB_COLOR_NORMAL != x ] && [ x$GRUB_COLOR_HIGHLIGHT != x ] ; then
+    cat << EOF
+set menu_color_normal=$GRUB_COLOR_NORMAL
+set menu_color_highlight=$GRUB_COLOR_HIGHLIGHT
+
+EOF
+fi
+
 serial=0;
 gfxterm=0;
 for x in ${GRUB_TERMINAL_INPUT} ${GRUB_TERMINAL_OUTPUT}; do
