diff -Nur orig/grub/po/LINGUAS mod/grub/po/LINGUAS
--- orig/grub/po/LINGUAS	2017-04-24 11:17:39.000000000 +0200
+++ mod/grub/po/LINGUAS	2018-12-01 15:49:12.925446478 +0100
@@ -1 +1 @@
-ast ca da de de_CH de@hebrew en@arabic en@cyrillic en@greek en@hebrew en@piglatin en@quot eo es fi fr gl hr hu id it ja ko lt nb nl pa pl pt_BR ru sl sr sv tr uk vi zh_CN zh_TW
+ast ca da de de_CH de@hebrew en@arabic en@hebrew en@piglatin en@quot eo es fi fr gl hr hu id it ja ko lt nb nl pa pl pt_BR ru sl sr sv tr uk vi zh_CN zh_TW
