diff -Naur a/dracut.sh b/dracut.sh
--- a/dracut.sh	2022-12-24 17:49:27.000000000 +0100
+++ b/dracut.sh	2022-12-26 03:49:54.706503759 +0100
@@ -1869,6 +1869,8 @@
 
 [[ -d "$dracutsysrootdir$systemdportableconfdir" ]] || systemdportableconfdir=${systemdutilconfdir}/portable
 
+udevdir=/usr/lib/udev
+
 [[ -d $dracutsysrootdir$systemdsystemunitdir ]] \
     || systemdsystemunitdir=$(pkg-config systemd --variable=systemdsystemunitdir 2> /dev/null)
 
