diff -Nur orig/modules.d/98dracut-systemd/dracut-shutdown.service mod/modules.d/98dracut-systemd/dracut-shutdown.service
--- orig/modules.d/98dracut-systemd/dracut-shutdown.service	2018-10-08 15:38:33.000000000 +0200
+++ mod/modules.d/98dracut-systemd/dracut-shutdown.service	2019-10-07 13:25:35.000000000 +0200
@@ -14,5 +14,5 @@
 [Service]
 RemainAfterExit=yes
 Type=oneshot
-ExecStart=/bin/true
+ExecStart=/usr/bin/true
 ExecStop=/usr/lib/dracut/dracut-initramfs-restore
diff -Nur orig/modules.d/98dracut-systemd/emergency.service mod/modules.d/98dracut-systemd/emergency.service
--- orig/modules.d/98dracut-systemd/emergency.service	2018-10-08 15:38:33.000000000 +0200
+++ mod/modules.d/98dracut-systemd/emergency.service	2019-10-07 13:26:12.000000000 +0200
@@ -15,7 +15,7 @@
 Environment=DRACUT_SYSTEMD=1
 Environment=NEWROOT=/sysroot
 WorkingDirectory=/
-ExecStart=/bin/dracut-emergency
+ExecStart=/usr/bin/dracut-emergency
 ExecStopPost=-/usr/bin/systemctl --fail --no-block default
 Type=idle
 StandardInput=tty-force
