diff -Naur gnome-encfs-manager-1.8.15-orig/configure.ac.in gnome-encfs-manager-1.8.15/configure.ac.in
--- gnome-encfs-manager-1.8.15-orig/configure.ac.in	2015-09-23 01:41:58.000000000 +0200
+++ gnome-encfs-manager-1.8.15/configure.ac.in	2015-10-31 12:07:38.298861832 +0100
@@ -70,7 +70,7 @@
 
 
 # setting correct paths
-PACKAGE_LOCALE_DIR="${PACKAGE_PREFIX}/${DATADIRNAME}/locale"
+PACKAGE_LOCALE_DIR="${PACKAGE_PREFIX}/${DATADIRNAME}"
 AC_DEFINE_UNQUOTED(PACKAGE_LOCALE_DIR,"$PACKAGE_LOCALE_DIR",[The package's locale path for gettext])
 
 
