diff -Naur gnome-encfs-manager-1.8.15-orig/dist/Makefile.am gnome-encfs-manager-1.8.15/dist/Makefile.am
--- gnome-encfs-manager-1.8.15-orig/dist/Makefile.am	2015-09-23 01:41:58.000000000 +0200
+++ gnome-encfs-manager-1.8.15/dist/Makefile.am	2015-10-31 12:25:57.290871171 +0100
@@ -1,7 +1,7 @@
 SUBDIRS = doc extra
 
 gencfsmdir = $(DESTDIR)$(datadir)/gnome-encfs-manager/
-dist = "doc/locale" "glade" "scripts"
+dist = "doc" "glade" "scripts"
 
 icondir =  $(DESTDIR)$(datadir)/icons/
 icons = $(shell find icons -type f)
