diff -Naur gnome-encfs-manager-1.8.15-orig/dist/extra/gnome-encfs-manager.desktop.in gnome-encfs-manager-1.8.15/dist/extra/gnome-encfs-manager.desktop.in
--- gnome-encfs-manager-1.8.15-orig/dist/extra/gnome-encfs-manager.desktop.in	2015-09-23 01:41:58.000000000 +0200
+++ gnome-encfs-manager-1.8.15/dist/extra/gnome-encfs-manager.desktop.in	2015-11-01 13:53:10.515910803 +0100
@@ -1,7 +1,8 @@
 [Desktop Entry]
 _Name=Gnome Encfs Manager
 _Comment=A manager and mounter for encfs stashes
-Exec=sh -c 'killall -9 gnome-encfs-manager; gnome-encfs-manager;'
+Comment[hu]=Grafikus kezelőprogram encfs titkosított fájlrendszerhez
+Exec=gnome-encfs-manager
 Terminal=false
 Type=Application
 Icon=gnome-encfs-manager
