diff -Nur orig/org.bleachbit.BleachBit.desktop mod/org.bleachbit.BleachBit.desktop
--- orig/org.bleachbit.BleachBit.desktop	2021-10-16 19:44:00.000000000 +0200
+++ mod/org.bleachbit.BleachBit.desktop	2021-11-14 09:40:01.000000000 +0100
@@ -114,7 +114,7 @@
 GenericName[zh_CN]=不必要文件清理器
 GenericName[zh_TW]=無用檔案清理器
 TryExec=bleachbit
-Exec=bleachbit
+Exec=pkexec /usr/bin/bleachbit
 Icon=bleachbit
 Categories=System;FileTools;GTK;
 Keywords=cache;clean;free;performance;privacy;
