diff -Nur orig/_mysql.c mod/_mysql.c
--- orig/_mysql.c	2014-01-02 13:52:50.000000000 +0100
+++ mod/_mysql.c	2017-09-21 21:15:40.182748204 +0200
@@ -41,11 +41,11 @@
 #if defined(MS_WINDOWS)
 #include <config-win.h>
 #else
-#include "my_config.h"
+#include "/usr/include/mysql/server/my_config.h"
 #endif
-#include "mysql.h"
-#include "mysqld_error.h"
-#include "errmsg.h"
+#include "/usr/include/mysql/server/mysql.h"
+#include "/usr/include/mysql/server/mysqld_error.h"
+#include "/usr/include/mysql/server/errmsg.h"
 
 #if PY_VERSION_HEX < 0x02020000
 # define MyTuple_Resize(t,n,d) _PyTuple_Resize(t, n, d)
