diff -Nur orig/config.mk mod/config.mk
--- orig/config.mk	2017-09-03 12:11:50.000000000 +0200
+++ mod/config.mk	2017-12-14 20:12:54.659413878 +0100
@@ -24,13 +24,13 @@
 # MANDIR to directory for man page.
 # Usually it will be simpler to just set PREFIX.
 #
-PREFIX=/usr/local
-BINDIR=$(PREFIX)/bin
-INFODIR=$(PREFIX)/share/info
-MANDIR=$(PREFIX)/share/man/man1
-DESKTOPDIR1=$(PREFIX)/share/applications
-DESKTOPDIR2=$(PREFIX)/share/app-install/desktop
-PIXMAPDIR=$(PREFIX)/share/pixmaps
+PREFIX=/usr
+BINDIR=$(DESTDIR)$(PREFIX)/bin
+INFODIR=$(DESTDIR)$(PREFIX)/share/info
+MANDIR=$(DESTDIR)$(PREFIX)/share/man/man1
+DESKTOPDIR1=$(DESTDIR)$(PREFIX)/share/applications
+DESKTOPDIR2=$(DESKTOPDIR1)
+PIXMAPDIR=$(DESTDIR)$(PREFIX)/share/pixmaps
 
 # -------------------- Miscellaneous options -----------------------
 
