diff -Nur orig/doc/Makefile mod/doc/Makefile
--- orig/doc/Makefile	2017-09-03 12:11:50.000000000 +0200
+++ mod/doc/Makefile	2017-12-14 20:17:08.287022005 +0100
@@ -4,7 +4,7 @@
 include ../config.mk
 
 
-all: info man pdf
+all: man
 
 info: xzgv.info.gz
 pdf: xzgv.pdf
@@ -34,8 +34,6 @@
 
 install: installdirs
 	install xzgv.1 $(MANDIR)
-	install -m 644 xzgv.info.gz $(INFODIR)
-	install-info --infodir=$(INFODIR) xzgv.info.gz
 
 uninstall:
 	$(RM) $(MANDIR)/xzgv.1
