diff -Naurdp sudo-1.8.8/configure sudo-1.8.8-noexec-dir/configure
--- sudo-1.8.8/configure	2013-09-30 02:37:54.000000000 +0200
+++ sudo-1.8.8-noexec-dir/configure	2013-10-22 11:51:55.322880427 +0200
@@ -13710,13 +13710,13 @@ $as_echo_n "checking path to sudo_noexec
 # Check whether --with-noexec was given.
 if test "${with_noexec+set}" = set; then :
   withval=$with_noexec; case $with_noexec in
-    yes)	with_noexec="$libexecdir/sudo/sudo_noexec.so"
+    yes)	with_noexec="$libdir/sudo/sudo_noexec.so"
 		;;
     no)		;;
     *)		;;
 esac
 else
-  with_noexec="$libexecdir/sudo/sudo_noexec.so"
+  with_noexec="$libdir/sudo/sudo_noexec.so"
 fi
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $with_noexec" >&5
