diff -Naur orig/benchtests/scripts/bench.py patched/benchtests/scripts/bench.py
--- orig/benchtests/scripts/bench.py	2020-02-01 12:52:50.000000000 +0100
+++ patched/benchtests/scripts/bench.py	2020-02-18 19:18:38.006958731 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # Copyright (C) 2014-2020 Free Software Foundation, Inc.
 # This file is part of the GNU C Library.
 #
diff -Naur orig/benchtests/scripts/compare_bench.py patched/benchtests/scripts/compare_bench.py
--- orig/benchtests/scripts/compare_bench.py	2020-02-01 12:52:50.000000000 +0100
+++ patched/benchtests/scripts/compare_bench.py	2020-02-18 19:18:45.987073593 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # Copyright (C) 2015-2020 Free Software Foundation, Inc.
 # This file is part of the GNU C Library.
 #
diff -Naur orig/benchtests/scripts/compare_strings.py patched/benchtests/scripts/compare_strings.py
--- orig/benchtests/scripts/compare_strings.py	2020-02-01 12:52:50.000000000 +0100
+++ patched/benchtests/scripts/compare_strings.py	2020-02-18 19:18:54.743199626 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # Copyright (C) 2017-2020 Free Software Foundation, Inc.
 # This file is part of the GNU C Library.
 #
diff -Naur orig/benchtests/scripts/validate_benchout.py patched/benchtests/scripts/validate_benchout.py
--- orig/benchtests/scripts/validate_benchout.py	2020-02-01 12:52:50.000000000 +0100
+++ patched/benchtests/scripts/validate_benchout.py	2020-02-18 19:19:07.259379783 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # Copyright (C) 2014-2020 Free Software Foundation, Inc.
 # This file is part of the GNU C Library.
 #
