diff -Nur orig/js/ui/endSessionDialog.js mod/js/ui/endSessionDialog.js
--- orig/js/ui/endSessionDialog.js	2019-10-09 02:50:59.000000000 +0200
+++ mod/js/ui/endSessionDialog.js	2019-10-18 10:48:14.144463934 +0200
@@ -441,15 +441,17 @@
     }
 
     _updateButtons() {
-        let dialogContent = DialogContent[this._type];
-        let buttons = [{ action: this.cancel.bind(this),
-                         label: _("Cancel"),
-                         key: Clutter.Escape }];
+        this.clearButtons();
 
+        this.addButton({ action: this.cancel.bind(this),
+                         label: _("Cancel"),
+                         key: Clutter.Escape });
+ 
+        let dialogContent = DialogContent[this._type];
         for (let i = 0; i < dialogContent.confirmButtons.length; i++) {
             let signal = dialogContent.confirmButtons[i].signal;
             let label = dialogContent.confirmButtons[i].label;
-            buttons.push({
+            let button = this.addButton({
                 action: () => {
                     this.close(true);
                     let signalId = this.connect('closed', () => {
@@ -457,11 +459,9 @@
                         this._confirm(signal);
                     });
                 },
-                label: label,
+                label
             });
         }
-
-        this.setButtons(buttons);
     }
 
     close(skipSignal) {
