diff -Naur a/js/ui/environment.js b/js/ui/environment.js
--- a/js/ui/environment.js	2023-03-19 11:58:32.000000000 +0100
+++ b/js/ui/environment.js	2023-03-19 21:40:26.098752942 +0100
@@ -8,7 +8,7 @@
 imports.gi.versions.Atspi = '2.0';
 imports.gi.versions.Clutter = Config.LIBMUTTER_API_VERSION;
 imports.gi.versions.Cogl = Config.LIBMUTTER_API_VERSION;
-imports.gi.versions.Gcr = '4';
+imports.gi.versions.Gcr = '3';
 imports.gi.versions.Gdk = '4.0';
 imports.gi.versions.Gdm = '1.0';
 imports.gi.versions.Geoclue = '2.0';
diff -Naur a/meson.build b/meson.build
--- a/meson.build	2023-03-19 11:58:32.000000000 +0100
+++ b/meson.build	2023-03-19 21:41:45.295053217 +0100
@@ -20,7 +20,7 @@
 
 ecal_req = '>= 3.33.1'
 eds_req = '>= 3.33.1'
-gcr_req = '>= 3.90.0'
+gcr_req = '>= 3.7.5'
 gio_req = '>= 2.56.0'
 gi_req = '>= 1.49.1'
 gjs_req = '>= 1.73.1'
@@ -71,7 +71,7 @@
 atk_bridge_dep = dependency('atk-bridge-2.0')
 ecal_dep = dependency('libecal-2.0', version: ecal_req)
 eds_dep = dependency('libedataserver-1.2', version: eds_req)
-gcr_dep = dependency('gcr-4', version: gcr_req)
+gcr_dep = dependency('gcr-base-3', version: gcr_req)
 gdk_x11_dep = dependency('gdk-x11-3.0')
 gdk_pixbuf_dep = dependency('gdk-pixbuf-2.0')
 gi_dep = dependency('gobject-introspection-1.0', version: gi_req)
diff -Naur a/src/meson.build b/src/meson.build
--- a/src/meson.build	2023-03-19 11:58:32.000000000 +0100
+++ b/src/meson.build	2023-03-19 21:42:30.407220355 +0100
@@ -213,7 +213,7 @@
 libshell_gir_includes = [
   'Clutter-@0@'.format(mutter_api_version),
   'Meta-@0@'.format(mutter_api_version),
-  'Gcr-4',
+  'Gcr-3',
   'PolkitAgent-1.0',
   'GdkPixbuf-2.0'
 ]
diff -Naur a/src/shell-keyring-prompt.c b/src/shell-keyring-prompt.c
--- a/src/shell-keyring-prompt.c	2023-03-19 11:58:32.000000000 +0100
+++ b/src/shell-keyring-prompt.c	2023-03-19 22:09:23.676071192 +0100
@@ -26,7 +26,7 @@
 #include "shell-secure-text-buffer.h"
 
 #define GCR_API_SUBJECT_TO_CHANGE
-#include <gcr/gcr.h>
+#include <gcr/gcr-base.h>
 
 #include <glib/gi18n.h>
 
@@ -91,7 +91,7 @@
 
 static GParamSpec *props[N_PROPS] = { NULL, };
 
-static void    shell_keyring_prompt_iface     (GcrPromptInterface *iface);
+static void    shell_keyring_prompt_iface     (GcrPromptIface *iface);
 
 G_DEFINE_TYPE_WITH_CODE (ShellKeyringPrompt, shell_keyring_prompt, G_TYPE_OBJECT,
                          G_IMPLEMENT_INTERFACE (GCR_TYPE_PROMPT, shell_keyring_prompt_iface);
@@ -531,7 +531,7 @@
 }
 
 static void
-shell_keyring_prompt_iface (GcrPromptInterface *iface)
+shell_keyring_prompt_iface (GcrPromptIface *iface)
 {
   iface->prompt_password_async = shell_keyring_prompt_password_async;
   iface->prompt_password_finish = shell_keyring_prompt_password_finish;
diff -Naur a/src/shell-secure-text-buffer.c b/src/shell-secure-text-buffer.c
--- a/src/shell-secure-text-buffer.c	2023-03-19 11:58:32.000000000 +0100
+++ b/src/shell-secure-text-buffer.c	2023-03-19 22:10:08.932188940 +0100
@@ -26,7 +26,7 @@
 #include "shell-secure-text-buffer.h"
 
 #define GCR_API_SUBJECT_TO_CHANGE
-#include <gcr/gcr.h>
+#include <gcr/gcr-base.h>
 
 #include <string.h>
 
