diff -Nur orig/Makefile mod/Makefile
--- orig/Makefile	2025-12-07 08:01:23.556402713 +0100
+++ mod/Makefile	2025-12-07 08:08:20.328206371 +0100
@@ -84,7 +84,6 @@
 	$(PYTHON) -m build --config-setting=--global-option="$(PURE)"
 .PHONY: doc
 doc:
-	$(MAKE) -C doc
 
 .PHONY: cleanbutpackages
 cleanbutpackages:
@@ -106,7 +105,7 @@
 	rm -rf packages
 
 .PHONY: install
-install: install-bin install-doc
+install: install-bin
 
 .PHONY: install-bin
 install-bin:
