diff -Naur orig/Makefile patched/Makefile
--- orig/Makefile	2013-03-27 16:52:38.000000000 +0100
+++ patched/Makefile	2016-11-20 01:46:09.279457632 +0100
@@ -13,7 +13,7 @@
 #  CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
 DESTDIR  =
-PREFIX   = /usr/local
+PREFIX   = /usr
 BINDIR   = $(PREFIX)/bin
 ICONDIR  = $(PREFIX)/share/icons/hicolor/scalable/apps
 MENUDIR  = $(PREFIX)/share/applications
