diff -Naur orig/desktop/gimp.desktop.in.in patched/desktop/gimp.desktop.in.in
--- orig/desktop/gimp.desktop.in.in	2018-03-25 14:35:18.000000000 +0200
+++ patched/desktop/gimp.desktop.in.in	2018-05-24 20:43:09.801161474 +0200
@@ -1,13 +1,13 @@
 [Desktop Entry]
 Version=1.0
 Type=Application
-_Name=GNU Image Manipulation Program
+Name=GIMP
 _GenericName=Image Editor
 _Comment=Create images and edit photographs
 # Translators: Search terms to find this application. Do NOT translate or localize the semicolons! The list MUST also end with a semicolon!
 _Keywords=GIMP;graphic;design;illustration;painting;
 Exec=@GIMP_COMMAND@ %U
-TryExec=gimp-@GIMP_APP_VERSION@
+TryExec=gimp-@GIMP_APP_VERSION@ %U
 Icon=gimp
 Terminal=false
 Categories=Graphics;2DGraphics;RasterGraphics;GTK;
