diff -Nur orig/etc/gimprc.in mod/etc/gimprc.in
--- orig/etc/gimprc.in	2021-03-29 00:00:23.000000000 +0200
+++ mod/etc/gimprc.in	2021-04-13 11:32:32.102296069 +0200
@@ -850,7 +850,7 @@
 # Otherwise the locally installed copy is used.  Possible values are yes and
 # no.
 # 
-# (user-manual-online no)
+# (user-manual-online yes)
 
 # The location of the online user manual. This is used if
 # 'user-manual-online' is enabled.  This is a string value.
@@ -861,7 +861,7 @@
 # This may affect the way your window manager decorates and handles these
 # windows.  Possible values are normal, utility and keep-above.
 # 
-# (dock-window-hint utility)
+# (dock-window-hint normal)
 
 # Sets the handedness for cursor positioning.  Possible values are left and
 # right.
