diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2020-10-04 19:32:33.000000000 +0200
+++ patched/configure.ac	2020-10-13 22:39:54.064743987 +0200
@@ -59,7 +59,6 @@
 m4_define([glib_required_version], [2.56.2])
 m4_define([gtk_mac_integration_required_version], [2.0.0])
 m4_define([gtk_required_version], [2.24.32])
-m4_define([gtkdoc_required_version], [1.0])
 m4_define([harfbuzz_required_version], [0.9.19])
 m4_define([intltool_required_version], [0.40.1])
 m4_define([json_glib_required_version], [1.2.6])
@@ -2625,16 +2624,9 @@
 
 # Check for GTK_DOC_CHECK availability. The GTK_DOC_CHECK invocation
 # must be on its own line, gtkdocize relies on it
-m4_ifdef([GTK_DOC_CHECK], [
-GTK_DOC_CHECK([1.0])
-])
 
 # NOTE: We need to use a separate automake conditional for this
 #       to make this work with the tarballs.
-AM_CONDITIONAL(ENABLE_GTK_DOC, test "x$enable_gtk_doc" = xyes)
-
-AC_ARG_ENABLE(gtk-doc-app, [  --enable-gtk-doc-app    build developer documentation for app (default=no)], , enable_gtk_doc_app=no)
-AM_CONDITIONAL(ENABLE_GTK_DOC_APP, test "x$enable_gtk_doc_app" = xyes)
 
 
 #############################
@@ -2969,27 +2961,6 @@
 plug-ins/twain/Makefile
 plug-ins/common/Makefile
 modules/Makefile
-devel-docs/Makefile
-devel-docs/app/Makefile
-devel-docs/app/version
-devel-docs/libgimp/Makefile
-devel-docs/libgimp/version
-devel-docs/libgimpbase/Makefile
-devel-docs/libgimpbase/version
-devel-docs/libgimpconfig/Makefile
-devel-docs/libgimpconfig/version
-devel-docs/libgimpcolor/Makefile
-devel-docs/libgimpcolor/version
-devel-docs/libgimpmath/Makefile
-devel-docs/libgimpmath/version
-devel-docs/libgimpmodule/Makefile
-devel-docs/libgimpmodule/version
-devel-docs/libgimpthumb/Makefile
-devel-docs/libgimpthumb/version
-devel-docs/libgimpwidgets/Makefile
-devel-docs/libgimpwidgets/version
-devel-docs/performance-logs/Makefile
-devel-docs/tools/Makefile
 docs/Makefile
 menus/Makefile
 cursors/Makefile
diff -Naur orig/docs/Makefile.am patched/docs/Makefile.am
--- orig/docs/Makefile.am	2012-03-12 20:18:08.000000000 +0100
+++ patched/docs/Makefile.am	2020-10-13 22:39:54.064743987 +0200
@@ -1,10 +1,5 @@
 ## Process this file with automake to produce Makefile.in
 
-EXTRA_DIST = \
-	Wilber.xcf.gz			\
-	Wilber.xcf.gz.README		\
-	Wilber_Construction_Kit.xcf.gz
-
 mans = \
 	gimp-$(GIMP_APP_VERSION).1	\
 	gimprc-$(GIMP_APP_VERSION).5
diff -Naur orig/Makefile.am patched/Makefile.am
--- orig/Makefile.am	2018-05-20 17:50:05.000000000 +0200
+++ patched/Makefile.am	2020-10-13 22:39:54.064743987 +0200
@@ -36,7 +36,6 @@
 	$(GIMP_MODULES)		\
 	$(GIMP_PLUGINS)		\
 	etc			\
-	devel-docs		\
 	docs			\
 	build
 
@@ -85,12 +84,7 @@
 	authors.xsl		\
 	config.h.win32		\
 	gimp-zip.in		\
-	git-version.h		\
-	gtk-doc.make
-
-
-DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc
-
+	git-version.h
 
 check-defs:
 	@$(PYTHON) $(top_srcdir)/tools/defcheck.py $(top_srcdir) || \
