diff -Naur orig/libgimpthumb/gimp-thumbnail-list.c patched/libgimpthumb/gimp-thumbnail-list.c
--- orig/libgimpthumb/gimp-thumbnail-list.c	2017-01-28 20:08:45.000000000 +0100
+++ patched/libgimpthumb/gimp-thumbnail-list.c	2018-01-21 04:20:21.770817848 +0100
@@ -74,9 +74,9 @@
       return -1;
     }
 
-  home = g_get_home_dir ();
+  home = g_get_user_cache_dir ();
 
-  thumb_folder = g_build_filename (home, ".thumbnails", NULL);
+  thumb_folder = g_build_filename (home, "thumbnails", NULL);
   dir = g_dir_open (thumb_folder, 0, &error);
 
   if (! dir)
diff -Naur orig/libgimpthumb/gimpthumb-utils.c patched/libgimpthumb/gimpthumb-utils.c
--- orig/libgimpthumb/gimpthumb-utils.c	2017-01-28 20:08:45.000000000 +0100
+++ patched/libgimpthumb/gimpthumb-utils.c	2018-01-21 04:24:30.161890379 +0100
@@ -137,11 +137,11 @@
 
 #else
 
-      const gchar *home_dir = g_get_home_dir ();
+      const gchar *home_dir = g_get_user_cache_dir ();
 
       if (home_dir && g_file_test (home_dir, G_FILE_TEST_IS_DIR))
         {
-          thumb_dir = g_build_filename (home_dir, ".thumbnails", NULL);
+          thumb_dir = g_build_filename (home_dir, "thumbnails", NULL);
         }
 
 #endif
