diff -Naur orig/src/app/meson.build patched/src/app/meson.build
--- orig/src/app/meson.build	2021-09-22 18:41:17.000000000 +0200
+++ patched/src/app/meson.build	2021-09-23 01:11:47.285505901 +0200
@@ -48,7 +48,7 @@
   app_gtk3_deps = app_common_deps + [libvte_gtk3_dep]
 
   app_gtk3 = executable(
-    'vte-' + vte_api_version,
+    'vte',
     app_gtk3_sources,
     dependencies: app_gtk3_deps,
     cpp_args: app_gtk3_cppflags,
