diff -Naur orig/examples/mha-preview mod/examples/mha-preview
--- orig/examples/mha-preview	2005-07-05 04:06:21.000000000 +0200
+++ mod/examples/mha-preview	2016-03-10 09:22:25.275583190 +0100
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 ##---------------------------------------------------------------------------##
 ##  File:
 ##	$Id: mha-preview,v 1.4 2005/07/05 02:06:21 ehood Exp $
