diff -Naurd b/src/main.c c/src/main.c
--- b/src/main.c	2005-12-12 14:08:53.000000000 +0100
+++ c/src/main.c	2005-12-12 14:25:29.000000000 +0100
@@ -350,6 +350,8 @@
 gint stat;
 #endif
 
+int enter_setup = 0;
+
 #if ENABLE_NLS
  #if GTK_MAJOR_VERSION < 2
 	bindtextdomain (PACKAGE, LOCALEDIR);
@@ -494,8 +496,8 @@
 #endif
 			/* scan for devices */
 			create_device_scanning(altdevscan, 0, 1, NULL);
-	
-			show_dialog(ICO_INFO, _("No configuration file found!\nPlease enter Setup."), T_OK, NULL, NULL, 0);
+
+			enter_setup = 1;
 
 #if !(defined(__MACH__) && defined(__APPLE__)) && (USE_NONROOTMODE == 1)
 		}
@@ -594,6 +596,9 @@
 	/* install a signal handler to catch CTRL-C */
 	signal (SIGINT, my_signal_exit) ;
 
+	if (enter_setup)
+		create_setup();
+
 	gtk_main();
 	
 	my_do_gtk_exit(0, 0);
