diff -urip xcdroast-0.98+0alpha15/src/io.c xcdroast-0.98+0alpha15-fixed/src/io.c
--- xcdroast-0.98+0alpha15/src/io.c	2003-10-26 15:24:48.000000000 +0100
+++ xcdroast-0.98+0alpha15-fixed/src/io.c	2007-01-17 23:04:29.000000000 +0100
@@ -10143,6 +10143,10 @@ gchar *p;
                 g_error("popen error\n");
         }
 
+	/* all versions of readom is good enough! */
+	return 0;
+	
+#if 0
 	strcpy(line,"");
         for (;;) {
                 if (fgets(line,MAXLINE,fpin) == NULL) 
@@ -10185,6 +10189,7 @@ gchar *p;
 		strcpy(found,ver);
 		return 1;
 	}
+#endif
 }
 
 
