--- xcdroast-0.98alpha15/configure.in.nogtk1	2005-04-21 18:11:47.000000000 +0200
+++ xcdroast-0.98alpha15/configure.in	2005-04-21 18:12:05.000000000 +0200
@@ -99,25 +99,9 @@
 AM_DEP_NONROOTMODE
 AM_DEP_ENABLEGTK2
 
-if test "$ENABLE_GTK2" = "TRUE"; then
-
-	PKG_CHECK_MODULES(PKGCONFIG, gtk+-2.0 >= 2.0.6)
-	LIBS="$LIBS $PKGCONFIG_LIBS"
-	CFLAGS="$CFLAGS $PKGCONFIG_CFLAGS"
-
-else 
-	dnl pure plain GTK+-1.2.x support
-
-	dnl Check for GLIB
-	AM_PATH_GLIB(1.2.3,, AC_MSG_ERROR(Test for GLIB failed. See the file 'INSTALL' for help.))
-
-	dnl Check for GTK+
-	AM_PATH_GTK(1.2.3,, AC_MSG_ERROR(Test for GTK failed. See the file 'INSTALL' for help.))
-
-	dnl Check for GDK-Pixbuf
-	AM_PATH_GDK_PIXBUF(0.8.0,, AC_MSG_ERROR(Test for gdk_pixbuf failed. See the file 'INSTALL' for help.))
-fi
-
+PKG_CHECK_MODULES(PKGCONFIG, gtk+-2.0 >= 2.0.6)
+LIBS="$LIBS $PKGCONFIG_LIBS"
+CFLAGS="$CFLAGS $PKGCONFIG_CFLAGS"
 dnl internationalization macros
 AM_GNU_GETTEXT
 
