diff -up xcdroast-0.98alpha16/src/setup.c.wformat xcdroast-0.98alpha16/src/setup.c
--- xcdroast-0.98alpha16/src/setup.c.wformat	2014-02-21 13:04:20.813356229 +0100
+++ xcdroast-0.98alpha16/src/setup.c	2014-02-21 13:04:33.893300539 +0100
@@ -4477,7 +4477,7 @@ gchar tmp[MAXLINE];
 		if (!textonly) {
 			show_dialog(ICO_ERROR,tmp, T_OK, NULL, NULL, 0);
 		} else {
-			g_warning(tmp);
+			g_warning("%s", tmp);
 		}
 	}	
 }
