diff -Naurd a/src/setup.c b/src/setup.c
--- a/src/setup.c	2005-12-12 14:08:53.000000000 +0100
+++ b/src/setup.c	2005-12-12 14:13:26.000000000 +0100
@@ -4011,11 +4011,6 @@
                 if (cursor2) gdk_cursor_destroy (cursor2);
         }
 
-        /* now wait until button is pressed */
-        while (nrs_dialog_done == 999) {
-                wait_and_process_events();
-        }
-
         if (GTK_WIDGET_MAPPED(toplevel)) {
                 gtk_grab_remove(GTK_WIDGET(dialog));
                 gdk_window_set_cursor(GTK_WIDGET(toplevel)->window,NULL);
