diff -Naur orig/singleapp/app.pro patched/singleapp/app.pro
--- orig/singleapp/app.pro	2019-04-27 22:25:11.000000000 +0200
+++ patched/singleapp/app.pro	2019-06-18 23:13:36.859747590 +0200
@@ -75,7 +75,7 @@
 
 macx:QMAKE_CXXFLAGS += -Wno-deprecated-declarations
 
-QMAKE_CXXFLAGS +=-Werror
+#QMAKE_CXXFLAGS +=-Werror
 QMAKE_CXXFLAGS +=-Wall
 
 equals(INST_USE_C11, "y") {
diff -Naur orig/src/QXmlEdit.pro patched/src/QXmlEdit.pro
--- orig/src/QXmlEdit.pro	2019-04-27 22:25:11.000000000 +0200
+++ patched/src/QXmlEdit.pro	2019-06-18 23:13:12.655504400 +0200
@@ -97,7 +97,7 @@
 
 } else {
     QMAKE_CXXFLAGS += -Wall
-    QMAKE_CXXFLAGS +=-Werror
+    #QMAKE_CXXFLAGS +=-Werror
 }
 
 lessThan(QT_MAJOR_VERSION, 5) {
diff -Naur orig/src/QXmlEditWidget.pro patched/src/QXmlEditWidget.pro
--- orig/src/QXmlEditWidget.pro	2019-04-27 22:25:11.000000000 +0200
+++ patched/src/QXmlEditWidget.pro	2019-06-18 23:13:21.627594546 +0200
@@ -75,7 +75,7 @@
 
 } else {
     QMAKE_CXXFLAGS += -Wall
-    QMAKE_CXXFLAGS +=-Werror
+    #QMAKE_CXXFLAGS +=-Werror
     #QMAKE_CXXFLAGS += -Winvalid-pch
 }
 
diff -Naur orig/src/sessions/QXmlEditSessions.pro patched/src/sessions/QXmlEditSessions.pro
--- orig/src/sessions/QXmlEditSessions.pro	2019-04-27 22:25:11.000000000 +0200
+++ patched/src/sessions/QXmlEditSessions.pro	2019-06-18 23:13:29.435672997 +0200
@@ -75,7 +75,7 @@
 
 } else {
     QMAKE_CXXFLAGS += -Wall
-    QMAKE_CXXFLAGS +=-Werror
+    #QMAKE_CXXFLAGS +=-Werror
 }
 
 equals(INST_USE_C11, "y") {
