diff -Naur orig/src/coptions.pri patched/src/coptions.pri
--- orig/src/coptions.pri	2019-12-17 20:36:16.000000000 +0100
+++ patched/src/coptions.pri	2019-12-30 10:46:12.080285488 +0100
@@ -26,12 +26,6 @@
     QMAKE_CXXFLAGS +=-Wno-unused-local-typedefs
 }
 
-QMAKE_CXXFLAGS +=-Werror=format-security
-
-isEmpty( INST_DISABLE_COMPILE_WARNINGS ) {
-    QMAKE_CXXFLAGS +=-Werror
-}
-
 equals(INST_USE_C11, "y") {
  QMAKE_CXXFLAGS +=-std=c++11
 }
