diff -Naur orig/src-qt5/lumina-pdf.desktop patched/src-qt5/lumina-pdf.desktop
--- orig/src-qt5/lumina-pdf.desktop	2019-04-30 02:29:00.000000000 +0200
+++ patched/src-qt5/lumina-pdf.desktop	2019-04-30 02:30:55.187085231 +0200
@@ -4,7 +4,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=Office;Graphics
+Categories=Office;Viewer;
 MimeType=application/pdf;
 Name=Lumina PDF Viewer
 Comment=View PDF Files
