diff -Nur orig/CMakeLists.txt mod/CMakeLists.txt
--- orig/CMakeLists.txt	2020-02-02 02:45:55.000000000 +0100
+++ mod/CMakeLists.txt	2020-04-04 10:47:42.991587527 +0200
@@ -239,8 +239,8 @@
 # #
 
 if(NOT OPT)
-	set(CMAKE_CXX_FLAGS_RELEASE "-O3")
-	set(CMAKE_C_FLAGS_RELEASE "-O3")
+	set(CMAKE_CXX_FLAGS_RELEASE "-O2")
+	set(CMAKE_C_FLAGS_RELEASE "-O2")
 
 	set(CMAKE_CXX_FLAGS_DEBUG "-O0")
 	set(CMAKE_C_FLAGS_DEBUG "-O0")
