diff -Nur orig/configure.ac mod/configure.ac
--- orig/configure.ac	2013-04-09 22:12:46.000000000 +0200
+++ mod/configure.ac	2018-06-09 19:47:28.656658921 +0200
@@ -95,7 +95,6 @@
 	Makefile
 	gtkdialog.spec
 	src/Makefile
-	doc/Makefile
 	data/Makefile
 	data/icons/Makefile
 	data/icons/hicolor/Makefile
diff -Nur orig/Makefile.am mod/Makefile.am
--- orig/Makefile.am	2013-04-09 22:12:46.000000000 +0200
+++ mod/Makefile.am	2018-06-09 19:48:04.968601682 +0200
@@ -1,9 +1,8 @@
-SUBDIRS=src doc data
+SUBDIRS=src data
 
 EXTRA_DIST = \
 	autogen.sh \
 	install.iss pango.modules pango.aliases gdk-pixbuf.loaders \
-	doc/reference/* \
 	examples/button/* \
 	examples/checkbox/* \
 	examples/colorbutton/* \
