diff -Naurdp gtk-engines-2.11.6.orig/engines/clearlooks/src/clearlooks_rc_style.c gtk-engines-2.11.6/engines/clearlooks/src/clearlooks_rc_style.c
--- gtk-engines-2.11.6.orig/engines/clearlooks/src/clearlooks_rc_style.c	2007-08-12 14:01:30.000000000 +0200
+++ gtk-engines-2.11.6/engines/clearlooks/src/clearlooks_rc_style.c	2007-08-22 20:05:20.000000000 +0200
@@ -131,7 +131,7 @@ clearlooks_rc_style_init (ClearlooksRcSt
 	clearlooks_rc->contrast = 1.0;
 	clearlooks_rc->menubarstyle = 0;
 	clearlooks_rc->toolbarstyle = 0;
-	clearlooks_rc->animation = FALSE;
+	clearlooks_rc->animation = TRUE;
 	clearlooks_rc->colorize_scrollbar = FALSE;
 	clearlooks_rc->radius = 3.0;
 }
