diff -Naur mdadm-3.3-orig/Makefile mdadm-3.3/Makefile
--- mdadm-3.3-orig/Makefile	2013-09-03 06:47:47.000000000 +0200
+++ mdadm-3.3/Makefile	2013-09-14 20:31:45.686883752 +0200
@@ -43,7 +43,7 @@
 
 CC = $(CROSS_COMPILE)gcc
 CXFLAGS ?= -ggdb
-CWFLAGS = -Wall -Werror -Wstrict-prototypes -Wextra -Wno-unused-parameter
+CWFLAGS = -Wall -Wstrict-prototypes -Wextra -Wno-unused-parameter
 ifdef WARN_UNUSED
 CWFLAGS += -Wp,-D_FORTIFY_SOURCE=2 -O3
 endif
