diff -Nur orig/scripts/blk_availability_systemd_red_hat.service.in mod/scripts/blk_availability_systemd_red_hat.service.in
--- orig/scripts/blk_availability_systemd_red_hat.service.in	2017-12-18 21:44:35.000000000 +0100
+++ mod/scripts/blk_availability_systemd_red_hat.service.in	2017-12-20 18:51:13.309525939 +0100
@@ -6,7 +6,7 @@
 
 [Service]
 Type=oneshot
-ExecStart=/usr/bin/true
+ExecStart=/bin/true
 ExecStop=@SBINDIR@/blkdeactivate -u -l wholevg -m disablequeueing -r wait
 RemainAfterExit=yes
 
