diff -Nur orig/scripts/dm_event_systemd_red_hat.socket.in mod/scripts/dm_event_systemd_red_hat.socket.in
--- orig/scripts/dm_event_systemd_red_hat.socket.in	2016-07-06 17:56:38.000000000 +0200
+++ mod/scripts/dm_event_systemd_red_hat.socket.in	2017-08-11 14:52:42.469556617 +0200
@@ -2,6 +2,7 @@
 Description=Device-mapper event daemon FIFOs
 Documentation=man:dmeventd(8)
 DefaultDependencies=no
+After=systemd-remount-fs.service
 
 [Socket]
 ListenFIFO=@DEFAULT_DM_RUN_DIR@/dmeventd-server
