diff -Naur LVM2.2.02.143-orig/lib/filters/filter-partitioned.c LVM2.2.02.143/lib/filters/filter-partitioned.c
--- LVM2.2.02.143-orig/lib/filters/filter-partitioned.c	2016-02-22 00:18:04.000000000 +0100
+++ LVM2.2.02.143/lib/filters/filter-partitioned.c	2016-02-22 12:18:40.065894563 +0100
@@ -15,6 +15,11 @@
 
 #include "lib.h"
 #include "filter.h"
+#include <sys/ioctl.h>
+#include <linux/fs.h>
+#undef MAJOR
+#undef MINOR
+#undef MKDEV
 
 #define MSG_SKIPPING "%s: Skipping: Partition table signature found"
 
