diff -Naurdp LVM2.2.02.112.orig/tools/lvmcmdline.c LVM2.2.02.112/tools/lvmcmdline.c
--- LVM2.2.02.112.orig/tools/lvmcmdline.c	2014-11-11 15:15:56.000000000 +0100
+++ LVM2.2.02.112/tools/lvmcmdline.c	2014-11-19 15:34:38.836950811 +0100
@@ -1003,7 +1003,7 @@ static int _get_settings(struct cmd_cont
 
 	if (arg_count(cmd, quiet_ARG)) {
 		cmd->current_settings.debug = 0;
-		cmd->current_settings.verbose = 0;
+		cmd->current_settings.verbose = -1;
 		cmd->current_settings.silent = (arg_count(cmd, quiet_ARG) > 1) ? 1 : 0;
 	}
 
