diff -Naur orig/scripts/lvm2_monitoring_systemd_red_hat.service.in patched/scripts/lvm2_monitoring_systemd_red_hat.service.in
--- orig/scripts/lvm2_monitoring_systemd_red_hat.service.in	2018-10-10 12:11:03.000000000 +0200
+++ patched/scripts/lvm2_monitoring_systemd_red_hat.service.in	2018-10-18 00:15:36.405035193 +0200
@@ -10,6 +10,8 @@
 [Service]
 Type=oneshot
 Environment=LVM_SUPPRESS_LOCKING_FAILURE_MESSAGES=1
+ExecStartPost=/usr/bin/udevadm trigger --type=devices
+ExecStartPost=-/sbin/vgmknodes --refresh
 ExecStart=@SBINDIR@/lvm vgchange --monitor y
 ExecStop=@SBINDIR@/lvm vgchange --monitor n
 RemainAfterExit=yes
