diff -Nur orig/scripts/lvm2_monitoring_systemd_red_hat.service.in mod/scripts/lvm2_monitoring_systemd_red_hat.service.in
--- orig/scripts/lvm2_monitoring_systemd_red_hat.service.in	2017-12-18 21:44:35.000000000 +0100
+++ mod/scripts/lvm2_monitoring_systemd_red_hat.service.in	2017-12-20 18:56:16.292941004 +0100
@@ -10,6 +10,8 @@
 [Service]
 Type=oneshot
 Environment=LVM_SUPPRESS_LOCKING_FAILURE_MESSAGES=1
+ExecStartPost=/usr/bin/udevadm trigger --type=devices
+ExecStartPost=-/usr/sbin/vgmknodes --refresh
 ExecStart=@SBINDIR@/lvm vgchange --monitor y --ignoreskippedcluster
 ExecStop=@SBINDIR@/lvm vgchange --monitor n --ignoreskippedcluster
 RemainAfterExit=yes
