diff -Naur orig/data/rygel-preferences.desktop.in.in patched/data/rygel-preferences.desktop.in.in
--- orig/data/rygel-preferences.desktop.in.in	2019-05-24 20:29:51.000000000 +0200
+++ patched/data/rygel-preferences.desktop.in.in	2019-06-15 04:29:39.601259637 +0200
@@ -8,7 +8,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-NotShowIn=GNOME;
+# NotShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=Rygel
 X-GNOME-Bugzilla-Component=rygel-preferences
