diff -Nur orig/data/rygel-preferences.desktop.in mod/data/rygel-preferences.desktop.in
--- orig/data/rygel-preferences.desktop.in	2018-02-05 19:36:37.000000000 +0100
+++ mod/data/rygel-preferences.desktop.in	2018-02-28 14:45:27.315755114 +0100
@@ -8,7 +8,6 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-NotShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=Rygel
 X-GNOME-Bugzilla-Component=rygel-preferences
diff -Nur orig/data/rygel-preferences.desktop.in.in mod/data/rygel-preferences.desktop.in.in
--- orig/data/rygel-preferences.desktop.in.in	2017-04-18 22:38:35.000000000 +0200
+++ mod/data/rygel-preferences.desktop.in.in	2018-02-28 14:45:42.299757692 +0100
@@ -8,7 +8,6 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-NotShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=Rygel
 X-GNOME-Bugzilla-Component=rygel-preferences
