diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2016-12-01 23:47:14.000000000 +0100
+++ patched/configure.ac	2018-05-24 22:06:20.892702249 +0200
@@ -22,7 +22,7 @@
           [libmypaint_api_major().libmypaint_api_minor().libmypaint_api_micro()m4_bpatsubst(libmypaint_api_prerelease(), [^\(.\)], [-\1])])
 
 # Dependencies.
-m4_define([gegl_required_version], [0.3])
+m4_define([gegl_required_version], [0.4])
 m4_define([introspection_required_version], [1.32.0])
 
 AC_INIT([libmypaint],
@@ -312,7 +312,7 @@
 )
 
 if eval "test x$enable_gegl = xyes"; then
-  PKG_CHECK_MODULES(GEGL, gegl-0.3 >= gegl_required_version)
+  PKG_CHECK_MODULES(GEGL, gegl-0.4 >= gegl_required_version)
 fi
 AM_CONDITIONAL(ENABLE_GEGL, test "x$enable_gegl" = "xyes")
 
diff -Naur orig/gegl/libmypaint-gegl.pc.in patched/gegl/libmypaint-gegl.pc.in
--- orig/gegl/libmypaint-gegl.pc.in	2016-12-01 23:47:14.000000000 +0100
+++ patched/gegl/libmypaint-gegl.pc.in	2018-05-24 22:07:33.697698643 +0200
@@ -6,6 +6,6 @@
 Name: libmypaint
 Description: MyPaint brush engine library, with GEGL integration.
 Version: @LIBMYPAINT_VERSION@
-Requires: gegl-0.3 libmypaint
+Requires: gegl-0.4 libmypaint
 Cflags: -I${includedir}/libmypaint-gegl
 Libs: -L${libdir} -lmypaint-gegl
diff -Naur orig/gegl/Makefile.am patched/gegl/Makefile.am
--- orig/gegl/Makefile.am	2016-12-01 23:47:14.000000000 +0100
+++ patched/gegl/Makefile.am	2018-05-24 22:08:22.766370187 +0200
@@ -11,7 +11,7 @@
 INTROSPECTION_GIRS =
 INTROSPECTION_SCANNER_ARGS = \
     --warn-all \
-    --pkg="gegl-0.3" \
+    --pkg="gegl-0.4" \
     --pkg="glib-2.0" \
     --namespace="MyPaintGegl" \
     --nsversion="$(LIBMYPAINT_MAJOR_VERSION).$(LIBMYPAINT_MINOR_VERSION)" \
@@ -38,7 +38,7 @@
 	mypaint-gegl-surface.c
 
 MyPaintGegl-@LIBMYPAINT_MAJOR_VERSION@.@LIBMYPAINT_MINOR_VERSION@.gir: libmypaint-gegl.la Makefile
-MyPaintGegl_@LIBMYPAINT_MAJOR_VERSION@_@LIBMYPAINT_MINOR_VERSION@_gir_INCLUDES = GObject-2.0 MyPaint-$(LIBMYPAINT_MAJOR_VERSION).$(LIBMYPAINT_MINOR_VERSION) Gegl-0.3
+MyPaintGegl_@LIBMYPAINT_MAJOR_VERSION@_@LIBMYPAINT_MINOR_VERSION@_gir_INCLUDES = GObject-2.0 MyPaint-$(LIBMYPAINT_MAJOR_VERSION).$(LIBMYPAINT_MINOR_VERSION) Gegl-0.4
 MyPaintGegl_@LIBMYPAINT_MAJOR_VERSION@_@LIBMYPAINT_MINOR_VERSION@_gir_CFLAGS = $(AM_CFLAGS) $(AM_CPPFLAGS) -I. -I..
 MyPaintGegl_@LIBMYPAINT_MAJOR_VERSION@_@LIBMYPAINT_MINOR_VERSION@_gir_LIBS = libmypaint-gegl.la ../libmypaint.la
 MyPaintGegl_@LIBMYPAINT_MAJOR_VERSION@_@LIBMYPAINT_MINOR_VERSION@_gir_FILES = $(introspection_sources)
