diff -Nur maxima-5.41.0/desktopintegration/net.sourceforge.maxima.xmaxima.desktop maxima-5.41.0.patched/desktopintegration/net.sourceforge.maxima.xmaxima.desktop
--- maxima-5.41.0/desktopintegration/net.sourceforge.maxima.xmaxima.desktop	2017-05-29 00:30:43.000000000 +0200
+++ maxima-5.41.0.patched/desktopintegration/net.sourceforge.maxima.xmaxima.desktop	2018-07-02 13:52:52.420311153 +0200
@@ -3,8 +3,10 @@
 Name=xmaxima
 GenericName=GUI for Maxima, a Computer Algebra system
 GenericName[de]=Eine grafische Oberfläche für Maxima
+GenericName[hu]=Maxima grafikus felület
 Comment=A graphical interface for Maxima, a Computer Algebra System
 Comment[de]=Eine grafische Oberfläche für Maxima, einem Programm für symbolische und numerische Berechnungen
+Comment[hu]=Számítógépes matematikai rendszer grafikus felülete
 Exec=xmaxima
 Icon=maxima-new
 Type=Application
