diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2014-01-11 23:59:44.000000000 +0100
+++ patched/configure.ac	2015-09-26 23:04:33.378872273 +0200
@@ -1564,7 +1564,7 @@
           RUBY_INSTALL_DIR="`$RUBY $srcdir/config/config_ruby.rb vendorarchdir`"
 	fi
       ],[
-        PKG_CHECK_MODULES([RUBY], [ruby-1.9],[
+        PKG_CHECK_MODULES([RUBY], [ruby-2.6],[
 	  AC_CHECK_PROG(RUBY,ruby,ruby)
           if test "x$RUBY" = "x"; then
             use_ruby="No (ruby not available)"
