--- a/configure.in.orig	2013-03-23 19:34:29.000000000 +0100
+++ a/configure.in	2013-03-23 19:34:54.000000000 +0100
@@ -9,7 +9,7 @@
 
 dnl use a config.h file rather than string of -DXXX args to compiler
 dnl run autoheader before autoconf to create config.h.in from configure.in
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 dnl enable maintainer mode in Makefiles (make dist, distclean)
 AM_MAINTAINER_MODE
 
