--- jpilot/configure.in.orig	2012-05-07 21:53:48.000000000 +0200
+++ jpilot/configure.in	2012-05-07 21:54:00.000000000 +0200
@@ -42,7 +42,7 @@
 dnl Gettext setup
 dnl ******************************
 dnl Set of available languages
-ALL_LINGUAS="ca cs da de es fr it ja ko nl nb pt_BR ru rw sv tr uk vi zh_CN zh_TW"
+ALL_LINGUAS="ca cs da de es fr hu it ja ko nl nb pt_BR ru rw sv tr uk vi zh_CN zh_TW"
 
 dnl Name of language files
 GETTEXT_PACKAGE=jpilot
