diff -Naur orig/aeteria/default.theme mod/aeteria/default.theme
--- orig/aeteria/default.theme	2003-02-05 01:57:50.000000000 +0100
+++ mod/aeteria/default.theme	2014-12-21 19:26:02.122069049 +0100
@@ -11,7 +11,8 @@
 # DesktopBackgroundColor="#656565"
 # choose bg-800.jpg, bg-1024.jpg or bg-1280.jpg
 DesktopBackgroundImage="bg-800.jpg"
-
+DesktopBackgroundCenter=1
+DesktopBackgroundScaled=1
 # FONT
 # default: lucida 
 TitleFontName="-*-lucida-medium-r-*-*-12-*-*-*-*-*-*-*"
diff -Naur orig/aeteria/graymenus.theme mod/aeteria/graymenus.theme
--- orig/aeteria/graymenus.theme	2003-02-05 01:58:05.000000000 +0100
+++ mod/aeteria/graymenus.theme	2014-12-21 19:27:07.082066468 +0100
@@ -11,6 +11,8 @@
 # DesktopBackgroundColor="#656565"
 # choose bg-800.jpg, bg-1024.jpg or bg-1280.jpg
 DesktopBackgroundImage="bg-800.jpg"
+DesktopBackgroundCenter=1
+DesktopBackgroundScaled=1
 
 # FONT
 # default: lucida 
diff -Naur orig/aeteria/menubutton.theme mod/aeteria/menubutton.theme
--- orig/aeteria/menubutton.theme	2003-02-05 01:58:41.000000000 +0100
+++ mod/aeteria/menubutton.theme	2014-12-21 19:26:28.282068010 +0100
@@ -13,6 +13,8 @@
 # another pixmaps included are: aed-1024.jpg, aed-1152.jpg,
 # aed-1280.jpg and aed-tile.jpg
 DesktopBackgroundImage="aed-800.jpg"
+DesktopBackgroundCenter=1
+DesktopBackgroundScaled=1
 
 # FONT
 # default: lucida 
diff -Naur orig/aeteria/windowish.theme mod/aeteria/windowish.theme
--- orig/aeteria/windowish.theme	2003-02-05 01:58:16.000000000 +0100
+++ mod/aeteria/windowish.theme	2014-12-21 19:26:38.114067619 +0100
@@ -10,6 +10,8 @@
 # DesktopBackgroundColor="#656565"
 # choose bg-800.jpg, bg-1024.jpg or bg-1280.jpg
 DesktopBackgroundImage="bg-800.jpg"
+DesktopBackgroundCenter=1
+DesktopBackgroundScaled=1
 
 # FONT
 # default: lucida 
