diff -Naur orig/blueCrux/CloseButtonLeft.theme mod/blueCrux/CloseButtonLeft.theme
--- orig/blueCrux/CloseButtonLeft.theme	2002-07-25 11:20:16.000000000 +0200
+++ mod/blueCrux/CloseButtonLeft.theme	2014-12-21 17:29:51.174346047 +0100
@@ -8,6 +8,10 @@
 # Default background image size is 1024x768. 
 # Use GIMP or ImageMagick to scale it for 800x600 desktops :)
 DesktopBackgroundImage="background.jpg"
+# Display desktop wallpaper centered, instead of tiled.
+DesktopBackgroundCenter=1 # 0 / 1
+# Desktop background scaled to full screen
+DesktopBackgroundScaled=1 # 0 / 1
 
 BorderSizeX=5
 BorderSizeY=5
diff -Naur orig/blueCrux/default.theme mod/blueCrux/default.theme
--- orig/blueCrux/default.theme	2002-07-25 11:09:29.000000000 +0200
+++ mod/blueCrux/default.theme	2014-12-21 17:29:35.958346652 +0100
@@ -8,6 +8,10 @@
 # Default background image size is 1024x768. 
 # Use GIMP or ImageMagick to scale it for 800x600 desktops :)
 DesktopBackgroundImage="background.jpg"
+# Display desktop wallpaper centered, instead of tiled.
+DesktopBackgroundCenter=1 # 0 / 1
+# Desktop background scaled to full screen
+DesktopBackgroundScaled=1 # 0 / 1
 
 BorderSizeX=5
 BorderSizeY=5
