diff -Naur orig/blueHeart/frameAB.xpm mod/blueHeart/frameAB.xpm
--- orig/blueHeart/frameAB.xpm	1999-12-12 01:19:02.000000000 +0100
+++ mod/blueHeart/frameAB.xpm	2014-12-21 13:12:33.000000000 +0100
@@ -1,14 +1,14 @@
 /* XPM */
 static char * frameAB_xpm[] = {
-"1 5 6 1",
+"2 5 6 1",
 " 	c None",
 ".	c #0D213B",
 "+	c #77869C",
 "@	c #394765",
 "#	c #26334A",
 "$	c #192436",
-".",
-"+",
-"@",
-"#",
-"$"};
+"..",
+"++",
+"@@",
+"##",
+"$$"};
diff -Naur orig/blueHeart/frameAT.xpm mod/blueHeart/frameAT.xpm
--- orig/blueHeart/frameAT.xpm	1999-12-12 03:37:48.000000000 +0100
+++ mod/blueHeart/frameAT.xpm	2014-12-21 13:13:38.000000000 +0100
@@ -1,14 +1,14 @@
 /* XPM */
 static char * frameAT_xpm[] = {
-"1 5 6 1",
+"2 5 6 1",
 " 	c None",
 ".	c #0D213B",
 "+	c #77869C",
 "@	c #394765",
 "#	c #26334A",
 "$	c #192436",
-".",
-"+",
-"@",
-"#",
-"$"};
+"..",
+"++",
+"@@",
+"##",
+"$$"};
diff -Naur orig/blueHeart/frameIB.xpm mod/blueHeart/frameIB.xpm
--- orig/blueHeart/frameIB.xpm	2000-01-05 07:54:42.000000000 +0100
+++ mod/blueHeart/frameIB.xpm	2014-12-21 13:14:54.000000000 +0100
@@ -1,12 +1,12 @@
 /* XPM */
 static char * frameIB_xpm[] = {
-"1 5 4 1",
+"2 5 4 1",
 " 	c None",
 ".	c #26334A",
 "+	c #394765",
 "@	c #192436",
-".",
-"+",
-".",
-"@",
-"@"};
+"..",
+"++",
+"..",
+"@@",
+"@@"};
diff -Naur orig/blueHeart/frameIT.xpm mod/blueHeart/frameIT.xpm
--- orig/blueHeart/frameIT.xpm	1999-12-12 03:50:34.000000000 +0100
+++ mod/blueHeart/frameIT.xpm	2014-12-21 13:15:58.000000000 +0100
@@ -1,12 +1,12 @@
 /* XPM */
 static char * frameIT_xpm[] = {
-"1 5 4 1",
+"2 5 4 1",
 " 	c None",
 ".	c #192436",
 "+	c #26334A",
 "@	c #394765",
-".",
-"+",
-"@",
-"+",
-"."};
+"..",
+"++",
+"@@",
+"++",
+".."};
diff -Naur orig/blueHeart/titleAT.xpm mod/blueHeart/titleAT.xpm
--- orig/blueHeart/titleAT.xpm	1999-12-14 06:49:55.000000000 +0100
+++ mod/blueHeart/titleAT.xpm	2014-12-20 23:01:12.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
 static char * titleAT_xpm[] = {
-"1 16 9 1",
+"2 16 9 1",
 " 	c None",
 ".	c #54647B",
 "+	c #031432",
@@ -10,19 +10,19 @@
 "%	c #1E2E4C",
 "&	c #0D213B",
 "*	c #506077",
-".",
-"+",
-"@",
-"#",
-"$",
-"%",
-"&",
-"&",
-"&",
-"&",
-"&",
-"&",
-"*",
-"&",
-"&",
-"&"};
+"..",
+"++",
+"@@",
+"##",
+"$$",
+"%%",
+"&&",
+"&&",
+"&&",
+"&&",
+"&&",
+"&&",
+"**",
+"&&",
+"&&",
+"&&"};
diff -Naur orig/blueHeart/titleIB.xpm mod/blueHeart/titleIB.xpm
--- orig/blueHeart/titleIB.xpm	1999-12-12 03:53:11.000000000 +0100
+++ mod/blueHeart/titleIB.xpm	2014-12-20 23:01:12.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
-static char * titleAB_xpm[] = {
-"1 16 17 1",
+static char * titleIB_xpm[] = {
+"2 16 17 1",
 " 	c None",
 ".	c #54647B",
 "+	c #031432",
@@ -18,19 +18,19 @@
 "'	c #000208",
 ")	c #000002",
 "!	c #0D213B",
-".",
-"+",
-"@",
-"#",
-"$",
-"%",
-"&",
-"*",
-"=",
-"-",
-";",
-">",
-",",
-"'",
-")",
-"!"};
+"..",
+"++",
+"@@",
+"##",
+"$$",
+"%%",
+"&&",
+"**",
+"==",
+"--",
+";;",
+">>",
+",,",
+"''",
+"))",
+"!!"};
