diff -Naur "orig/Crux-Teal/Close Button Left.theme" "mod/Crux-Teal/Close Button Left.theme"
--- "orig/Crux-Teal/Close Button Left.theme"	2002-11-26 01:40:25.000000000 +0100
+++ "mod/Crux-Teal/Close Button Left.theme"	2014-12-21 17:36:30.000000000 +0100
@@ -88,7 +88,11 @@
 ColorActiveMinimizedWindowText	=	"#000000"
 
 DesktopBackgroundColor	=	"#000000"
-DesktopBackgroundImage	=	""
+DesktopBackgroundImage	=	"background.jpg"
+# Display desktop wallpaper centered, instead of tiled.
+DesktopBackgroundCenter=1 # 0 / 1
+# Desktop background scaled to full screen
+DesktopBackgroundScaled=1 # 0 / 1
 
 ColorToolTip		=	"#dddddd"
 ColorToolTipText	=	"#000000"
diff -Naur orig/Crux-Teal/default.theme mod/Crux-Teal/default.theme
--- orig/Crux-Teal/default.theme	2003-02-05 01:23:58.000000000 +0100
+++ mod/Crux-Teal/default.theme	2014-12-21 17:35:00.000000000 +0100
@@ -90,6 +90,10 @@
 
 DesktopBackgroundColor	=	"#000000"
 DesktopBackgroundImage	=	"background.jpg"
+# Display desktop wallpaper centered, instead of tiled.
+DesktopBackgroundCenter=1 # 0 / 1
+# Desktop background scaled to full screen
+DesktopBackgroundScaled=1 # 0 / 1
 
 ColorToolTip		=	"#dddddd"
 ColorToolTipText	=	"#000000"
diff -Naur "orig/Crux-Teal/More Title Bar Buttons.theme" "mod/Crux-Teal/More Title Bar Buttons.theme"
--- "orig/Crux-Teal/More Title Bar Buttons.theme"	2002-11-26 01:40:25.000000000 +0100
+++ "mod/Crux-Teal/More Title Bar Buttons.theme"	2014-12-21 17:39:17.074323561 +0100
@@ -88,7 +88,12 @@
 ColorActiveMinimizedWindowText	=	"#000000"
 
 DesktopBackgroundColor	=	"#000000"
-DesktopBackgroundImage	=	""
+DesktopBackgroundImage	=	"background.jpg"
+# Display desktop wallpaper centered, instead of tiled.
+DesktopBackgroundCenter=1 # 0 / 1
+# Desktop background scaled to full screen
+DesktopBackgroundScaled=1 # 0 / 1
+
 
 ColorToolTip		=	"#dddddd"
 ColorToolTipText	=	"#000000"
