diff -Naur orig/ElbergBlue/Wallpaper.theme mod/ElbergBlue/Wallpaper.theme
--- orig/ElbergBlue/Wallpaper.theme	2004-12-12 16:01:17.000000000 +0100
+++ mod/ElbergBlue/Wallpaper.theme	2014-12-21 18:12:34.858244177 +0100
@@ -209,9 +209,9 @@
 
 DesktopBackgroundCenter=1 # 0/1
 SupportSemitransparency=1 # 0/1
-DesktopBackgroundScaled=1 # 0/1
+DesktopBackgroundScaled=0 # 0/1
 DesktopBackgroundColor="midnightBlue"
-DesktopBackgroundImage="/usr/share/wallpapers/default.jpg"
+DesktopBackgroundImage="/usr/share/wallpapers/UHU/blur.jpg"
 #DesktopTransparencyColor=""
 #DesktopTransparencyImage=""
 
diff -Naur orig/ElbergGreen/Wallpaper.theme mod/ElbergGreen/Wallpaper.theme
--- orig/ElbergGreen/Wallpaper.theme	2004-12-12 16:04:39.000000000 +0100
+++ mod/ElbergGreen/Wallpaper.theme	2014-12-21 18:13:58.738240843 +0100
@@ -209,9 +209,9 @@
 
 DesktopBackgroundCenter=1 # 0/1
 SupportSemitransparency=1 # 0/1
-DesktopBackgroundScaled=1 # 0/1
+DesktopBackgroundScaled=0 # 0/1
 DesktopBackgroundColor="midnightBlue"
-DesktopBackgroundImage="/usr/share/wallpapers/default.jpg"
+DesktopBackgroundImage="/usr/share/wallpapers/UHU/blur.jpg"
 #DesktopTransparencyColor=""
 #DesktopTransparencyImage=""
 
diff -Naur orig/ElbergRed/Wallpaper.theme mod/ElbergRed/Wallpaper.theme
--- orig/ElbergRed/Wallpaper.theme	2004-12-12 16:06:20.000000000 +0100
+++ mod/ElbergRed/Wallpaper.theme	2014-12-21 18:13:33.610241842 +0100
@@ -212,9 +212,9 @@
 
 DesktopBackgroundCenter=1 # 0/1
 SupportSemitransparency=1 # 0/1
-DesktopBackgroundScaled=1 # 0/1
+DesktopBackgroundScaled=0 # 0/1
 DesktopBackgroundColor="midnightBlue"
-DesktopBackgroundImage="/usr/share/wallpapers/default.jpg"
+DesktopBackgroundImage="/usr/share/wallpapers/UHU/blur.jpg"
 #DesktopTransparencyColor=""
 #DesktopTransparencyImage=""
 
