diff -Naur orig/OpenLook-shaped/dframeAB.xpm mod/OpenLook-shaped/dframeAB.xpm
--- orig/OpenLook-shaped/dframeAB.xpm	2001-07-12 12:20:31.000000000 +0200
+++ mod/OpenLook-shaped/dframeAB.xpm	2014-12-21 15:42:17.000000000 +0100
@@ -1,11 +1,11 @@
 /* XPM */
-static char * frameAB_xpm[] = {
-"1 5 3 1",
+static char * dframeAB_xpm[] = {
+"2 5 3 1",
 " 	c None",
 ".	c #CCCCFF",
 "+	c #000000",
-".",
-".",
-".",
-"+",
-"+"};
+"..",
+"..",
+"..",
+"++",
+"++"};
diff -Naur orig/OpenLook-shaped/dframeAT.xpm mod/OpenLook-shaped/dframeAT.xpm
--- orig/OpenLook-shaped/dframeAT.xpm	2001-07-12 12:20:22.000000000 +0200
+++ mod/OpenLook-shaped/dframeAT.xpm	2014-12-21 15:43:15.000000000 +0100
@@ -1,11 +1,11 @@
 /* XPM */
-static char * frameAT_xpm[] = {
-"1 5 3 1",
+static char * dframeAT_xpm[] = {
+"2 5 3 1",
 " 	c None",
 ".	c #000000",
 "+	c #CCCCFF",
-".",
-".",
-"+",
-"+",
-"+"};
+"..",
+"..",
+"++",
+"++",
+"++"};
diff -Naur orig/OpenLook-shaped/frameAB.xpm mod/OpenLook-shaped/frameAB.xpm
--- orig/OpenLook-shaped/frameAB.xpm	2001-04-28 13:58:23.000000000 +0200
+++ mod/OpenLook-shaped/frameAB.xpm	2014-12-21 15:44:27.000000000 +0100
@@ -1,11 +1,11 @@
 /* XPM */
 static char * frameAB_xpm[] = {
-"1 5 3 1",
-" 	c None s None",
+"2 5 3 1",
+" 	c None",
 ".	c #CCCCFF",
 "+	c #000000",
-".",
-".",
-".",
-"+",
-" "};
+"..",
+"..",
+"..",
+"++",
+"  "};
diff -Naur orig/OpenLook-shaped/frameAT.xpm mod/OpenLook-shaped/frameAT.xpm
--- orig/OpenLook-shaped/frameAT.xpm	2001-04-28 13:25:53.000000000 +0200
+++ mod/OpenLook-shaped/frameAT.xpm	2014-12-21 15:45:37.000000000 +0100
@@ -1,11 +1,11 @@
 /* XPM */
 static char * frameAT_xpm[] = {
-"1 5 3 1",
-" 	c None s None",
+"2 5 3 1",
+" 	c None",
 ".	c #000000",
 "+	c #CCCCFF",
-" ",
-".",
-"+",
-"+",
-"+"};
+"  ",
+"..",
+"++",
+"++",
+"++"};
diff -Naur orig/OpenLook-shaped/frameIB.xpm mod/OpenLook-shaped/frameIB.xpm
--- orig/OpenLook-shaped/frameIB.xpm	2001-04-28 12:17:45.000000000 +0200
+++ mod/OpenLook-shaped/frameIB.xpm	2014-12-21 15:46:10.000000000 +0100
@@ -1,11 +1,11 @@
 /* XPM */
-static char * frameAB_xpm[] = {
-"1 5 3 1",
+static char * frameIB_xpm[] = {
+"2 5 3 1",
 " 	c None",
 ".	c #CDCECD",
 "+	c #000000",
-".",
-".",
-".",
-"+",
-"+"};
+"..",
+"..",
+"..",
+"++",
+"++"};
diff -Naur orig/OpenLook-shaped/frameIT.xpm mod/OpenLook-shaped/frameIT.xpm
--- orig/OpenLook-shaped/frameIT.xpm	2001-04-28 12:17:45.000000000 +0200
+++ mod/OpenLook-shaped/frameIT.xpm	2014-12-21 15:46:39.000000000 +0100
@@ -1,11 +1,11 @@
 /* XPM */
-static char * frameAT_xpm[] = {
-"1 5 3 1",
+static char * frameIT_xpm[] = {
+"2 5 3 1",
 " 	c None",
 ".	c #000000",
 "+	c #CDCECD",
-".",
-".",
-"+",
-"+",
-"+"};
+"..",
+"..",
+"++",
+"++",
+"++"};
diff -Naur orig/OpenLook-shaped/titleAR.xpm mod/OpenLook-shaped/titleAR.xpm
--- orig/OpenLook-shaped/titleAR.xpm	2001-07-12 12:25:10.000000000 +0200
+++ mod/OpenLook-shaped/titleAR.xpm	2014-12-20 23:01:12.000000000 +0100
@@ -1,28 +1,28 @@
 /* XPM */
 static char * titleAR_xpm[] = {
-"1 22 3 1",
+"2 22 3 1",
 " 	c None",
 ".	c #FFFFFF",
 "+	c #CDCECD",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"+",
-"+",
-"+",
-"+",
-"+",
-".",
-"."};
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"++",
+"++",
+"++",
+"++",
+"++",
+"..",
+".."};
diff -Naur orig/OpenLook-shaped/titleAT.xpm mod/OpenLook-shaped/titleAT.xpm
--- orig/OpenLook-shaped/titleAT.xpm	2001-07-12 12:24:58.000000000 +0200
+++ mod/OpenLook-shaped/titleAT.xpm	2014-12-20 23:01:12.000000000 +0100
@@ -1,31 +1,31 @@
 /* XPM */
-static char * titleAM_xpm[] = {
-"1 22 6 1",
+static char * titleAT_xpm[] = {
+"2 22 6 1",
 " 	c None",
 ".	c #666699",
 "+	c #BBBBEE",
 "@	c #FFFFFF",
 "#	c #CCCCFF",
 "$	c #F6F6F6",
-".",
-"+",
-"+",
-"+",
-"+",
-"+",
-"+",
-"+",
-"+",
-"+",
-"+",
-"+",
-"+",
-"+",
-"@",
-"#",
-"#",
-"#",
-"#",
-"#",
-".",
-"$"};
+"..",
+"++",
+"++",
+"++",
+"++",
+"++",
+"++",
+"++",
+"++",
+"++",
+"++",
+"++",
+"++",
+"++",
+"@@",
+"##",
+"##",
+"##",
+"##",
+"##",
+"..",
+"$$"};
diff -Naur orig/OpenLook-shaped/titleIM.xpm mod/OpenLook-shaped/titleIM.xpm
--- orig/OpenLook-shaped/titleIM.xpm	2001-04-28 12:17:45.000000000 +0200
+++ mod/OpenLook-shaped/titleIM.xpm	2014-12-20 23:01:12.000000000 +0100
@@ -1,29 +1,29 @@
 /* XPM */
 static char * titleIM_xpm[] = {
-"1 22 4 1",
+"2 22 4 1",
 " 	c None",
 ".	c #CDCECD",
 "+	c #626562",
 "@	c #F6F6F6",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"+",
-"@"};
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"++",
+"@@"};
diff -Naur orig/OpenLook-shaped/titleIR.xpm mod/OpenLook-shaped/titleIR.xpm
--- orig/OpenLook-shaped/titleIR.xpm	2001-04-28 12:17:45.000000000 +0200
+++ mod/OpenLook-shaped/titleIR.xpm	2014-12-20 23:01:12.000000000 +0100
@@ -1,28 +1,28 @@
 /* XPM */
 static char * titleIR_xpm[] = {
-"1 22 3 1",
+"2 22 3 1",
 " 	c None",
 ".	c #CDCECD",
 "+	c #FFFFFF",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"+",
-"+"};
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"..",
+"++",
+"++"};
