diff -Naur orig/SilverXP/buttonA.xpm mod/SilverXP/buttonA.xpm
--- orig/SilverXP/buttonA.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/buttonA.xpm	2014-12-21 16:14:21.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
-static char * workspacebuttonactive_xpm[] = {
-"1 24 25 1",
+static char * buttonA_xpm[] = {
+"2 24 25 1",
 " 	c None",
 ".	c #BDC0CD",
 "+	c #C0C3CD",
@@ -26,27 +26,27 @@
 "(	c #E2E9E1",
 "_	c #D5DDDC",
 ":	c #BFC4D1",
-".",
-"+",
-"@",
-"#",
-"$",
-"%",
-"&",
-"*",
-"=",
-"-",
-";",
-">",
-",",
-"'",
-")",
-"!",
-"~",
-"{",
-"]",
-"^",
-"/",
-"(",
-"_",
-":"};
+"..",
+"++",
+"@@",
+"##",
+"$$",
+"%%",
+"&&",
+"**",
+"==",
+"--",
+";;",
+">>",
+",,",
+"''",
+"))",
+"!!",
+"~~",
+"{{",
+"]]",
+"^^",
+"//",
+"((",
+"__",
+"::"};
diff -Naur orig/SilverXP/buttonI.xpm mod/SilverXP/buttonI.xpm
--- orig/SilverXP/buttonI.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/buttonI.xpm	2014-12-21 16:14:47.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
-static char * workspacebuttonactive_xpm[] = {
-"1 24 25 1",
+static char * buttonI_xpm[] = {
+"2 24 25 1",
 " 	c None",
 ".	c #BDC0CD",
 "+	c #C0C3CD",
@@ -26,27 +26,27 @@
 "(	c #E2E9E1",
 "_	c #D5DDDC",
 ":	c #BFC4D1",
-".",
-"+",
-"@",
-"#",
-"$",
-"%",
-"&",
-"*",
-"=",
-"-",
-";",
-">",
-",",
-"'",
-")",
-"!",
-"~",
-"{",
-"]",
-"^",
-"/",
-"(",
-"_",
-":"};
+"..",
+"++",
+"@@",
+"##",
+"$$",
+"%%",
+"&&",
+"**",
+"==",
+"--",
+";;",
+">>",
+",,",
+"''",
+"))",
+"!!",
+"~~",
+"{{",
+"]]",
+"^^",
+"//",
+"((",
+"__",
+"::"};
diff -Naur orig/SilverXP/dframeAB.xpm mod/SilverXP/dframeAB.xpm
--- orig/SilverXP/dframeAB.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/dframeAB.xpm	2014-12-21 16:06:37.000000000 +0100
@@ -1,11 +1,11 @@
 /* XPM */
-static char * frameAB_xpm[] = {
-"1 4 4 1",
+static char * dframeAB_xpm[] = {
+"2 4 4 1",
 " 	c None",
 ".	c #66667E",
 "+	c #A8A9BB",
 "@	c #FBFCFD",
-".",
-"+",
-"@",
-"."};
+"..",
+"++",
+"@@",
+".."};
diff -Naur orig/SilverXP/dframeAL.xpm mod/SilverXP/dframeAL.xpm
--- orig/SilverXP/dframeAL.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/dframeAL.xpm	2014-12-21 16:00:11.000000000 +0100
@@ -1,8 +1,9 @@
 /* XPM */
-static char * frameAL_xpm[] = {
-"4 1 4 1",
+static char * dframeAL_xpm[] = {
+"4 2 4 1",
 " 	c None",
 ".	c #66667E",
 "+	c #FBFCFD",
 "@	c #A8A9BB",
+".+@.",
 ".+@."};
diff -Naur orig/SilverXP/dframeAR.xpm mod/SilverXP/dframeAR.xpm
--- orig/SilverXP/dframeAR.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/dframeAR.xpm	2014-12-21 16:00:11.000000000 +0100
@@ -1,8 +1,9 @@
 /* XPM */
-static char * frameAR_xpm[] = {
-"4 1 4 1",
+static char * dframeAR_xpm[] = {
+"4 2 4 1",
 " 	c None",
 ".	c #66667E",
 "+	c #A8A9BB",
 "@	c #FBFCFD",
+".+@.",
 ".+@."};
diff -Naur orig/SilverXP/dframeAT.xpm mod/SilverXP/dframeAT.xpm
--- orig/SilverXP/dframeAT.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/dframeAT.xpm	2014-12-21 16:07:12.000000000 +0100
@@ -1,12 +1,12 @@
 /* XPM */
-static char * frameAT_xpm[] = {
-"1 4 5 1",
+static char * dframeAT_xpm[] = {
+"2 4 5 1",
 " 	c None",
 ".	c #66667E",
 "+	c #A8A7BF",
 "@	c #FFFFFF",
 "#	c #D7D8E2",
-".",
-"+",
-"@",
-"#"};
+"..",
+"++",
+"@@",
+"##"};
diff -Naur orig/SilverXP/dframeIB.xpm mod/SilverXP/dframeIB.xpm
--- orig/SilverXP/dframeIB.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/dframeIB.xpm	2014-12-21 16:07:43.000000000 +0100
@@ -1,11 +1,11 @@
 /* XPM */
-static char * frameAB_xpm[] = {
-"1 4 4 1",
+static char * dframeIB_xpm[] = {
+"2 4 4 1",
 " 	c None",
 ".	c #66667E",
 "+	c #A8A9BB",
 "@	c #FBFCFD",
-".",
-"+",
-"@",
-"."};
+"..",
+"++",
+"@@",
+".."};
diff -Naur orig/SilverXP/dframeIT.xpm mod/SilverXP/dframeIT.xpm
--- orig/SilverXP/dframeIT.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/dframeIT.xpm	2014-12-21 16:08:13.000000000 +0100
@@ -1,12 +1,12 @@
 /* XPM */
-static char * frameAT_xpm[] = {
-"1 4 5 1",
+static char * dframeIT_xpm[] = {
+"2 4 5 1",
 " 	c None",
 ".	c #66667E",
 "+	c #A8A7BF",
 "@	c #FFFFFF",
 "#	c #D7D8E2",
-".",
-"+",
-"@",
-"#"};
+"..",
+"++",
+"@@",
+"##"};
diff -Naur orig/SilverXP/dialogbg.xpm mod/SilverXP/dialogbg.xpm
--- orig/SilverXP/dialogbg.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/dialogbg.xpm	2014-12-21 16:13:47.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
 static char * dialogbg_xpm[] = {
-"1 1 2 1",
+"2 1 2 1",
 " 	c None",
 ".	c #E0DFE3",
-"."};
+".."};
diff -Naur orig/SilverXP/frameAB.xpm mod/SilverXP/frameAB.xpm
--- orig/SilverXP/frameAB.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/frameAB.xpm	2014-12-21 16:09:26.000000000 +0100
@@ -1,11 +1,11 @@
 /* XPM */
 static char * frameAB_xpm[] = {
-"1 4 4 1",
+"2 4 4 1",
 " 	c None",
 ".	c #66667E",
 "+	c #A8A9BB",
 "@	c #FBFCFD",
-".",
-"+",
-"@",
-"."};
+"..",
+"++",
+"@@",
+".."};
diff -Naur orig/SilverXP/frameAT.xpm mod/SilverXP/frameAT.xpm
--- orig/SilverXP/frameAT.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/frameAT.xpm	2014-12-21 16:09:58.000000000 +0100
@@ -1,12 +1,12 @@
 /* XPM */
 static char * frameAT_xpm[] = {
-"1 4 5 1",
+"2 4 5 1",
 " 	c None",
 ".	c #66667E",
 "+	c #A8A7BF",
 "@	c #FFFFFF",
 "#	c #D7D8E2",
-".",
-"+",
-"@",
-"#"};
+"..",
+"++",
+"@@",
+"##"};
diff -Naur orig/SilverXP/frameIB.xpm mod/SilverXP/frameIB.xpm
--- orig/SilverXP/frameIB.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/frameIB.xpm	2014-12-21 16:10:25.000000000 +0100
@@ -1,11 +1,11 @@
 /* XPM */
-static char * frameAB_xpm[] = {
-"1 4 4 1",
+static char * frameIB_xpm[] = {
+"2 4 4 1",
 " 	c None",
 ".	c #66667E",
 "+	c #A8A9BB",
 "@	c #FBFCFD",
-".",
-"+",
-"@",
-"."};
+"..",
+"++",
+"@@",
+".."};
diff -Naur orig/SilverXP/frameIT.xpm mod/SilverXP/frameIT.xpm
--- orig/SilverXP/frameIT.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/frameIT.xpm	2014-12-21 16:10:51.000000000 +0100
@@ -1,12 +1,12 @@
 /* XPM */
-static char * frameAT_xpm[] = {
-"1 4 5 1",
+static char * frameIT_xpm[] = {
+"2 4 5 1",
 " 	c None",
 ".	c #66667E",
 "+	c #A8A7BF",
 "@	c #FFFFFF",
 "#	c #D7D8E2",
-".",
-"+",
-"@",
-"#"};
+"..",
+"++",
+"@@",
+"##"};
diff -Naur orig/SilverXP/menubg-gray.xpm mod/SilverXP/menubg-gray.xpm
--- orig/SilverXP/menubg-gray.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/menubg-gray.xpm	2014-12-21 16:13:13.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
-static char * menubg_xpm[] = {
-"1 1 2 1",
+static char * menubg_gray_xpm[] = {
+"2 1 2 1",
 " 	c None",
 ".	c #E0DFE3",
-"."};
+".."};
diff -Naur orig/SilverXP/menubg.xpm mod/SilverXP/menubg.xpm
--- orig/SilverXP/menubg.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/menubg.xpm	2014-12-21 16:11:54.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
 static char * menubg_xpm[] = {
-"1 11 11 1",
+"2 11 11 1",
 " 	c None",
 ".	c #F4F6F7",
 "+	c #F4F5F8",
@@ -12,14 +12,14 @@
 "*	c #DDDFE7",
 "=	c #D8DAE3",
 "-	c #D3D5E0",
-".",
-".",
-"+",
-"@",
-"#",
-"$",
-"%",
-"&",
-"*",
-"=",
-"-"};
+"..",
+"..",
+"++",
+"@@",
+"##",
+"$$",
+"%%",
+"&&",
+"**",
+"==",
+"--"};
diff -Naur orig/SilverXP/menusel.xpm mod/SilverXP/menusel.xpm
--- orig/SilverXP/menusel.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/menusel.xpm	2014-12-21 16:12:43.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
 static char * menusel_xpm[] = {
-"1 1 2 1",
+"2 1 2 1",
 " 	c None",
 ".	c #BBB7C7",
-"."};
+".."};
diff -Naur orig/SilverXP/titleAB.xpm mod/SilverXP/titleAB.xpm
--- orig/SilverXP/titleAB.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/titleAB.xpm	2014-12-21 16:00:11.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
 static char * titleAB_xpm[] = {
-"1 26 26 1",
+"2 26 26 1",
 " 	c None",
 ".	c #BCBCCF",
 "+	c #A8A7BF",
@@ -27,29 +27,29 @@
 "_	c #E4E3E3",
 ":	c #BCBDCD",
 "<	c #777792",
-".",
-"+",
-"@",
-"#",
-"$",
-"%",
-"&",
-"&",
-"*",
-"=",
-"-",
-";",
-">",
-",",
-"'",
-")",
-"!",
-"~",
-"{",
-"]",
-"^",
-"/",
-"(",
-"_",
-":",
-"<"};
+"..",
+"++",
+"@@",
+"##",
+"$$",
+"%%",
+"&&",
+"&&",
+"**",
+"==",
+"--",
+";;",
+">>",
+",,",
+"''",
+"))",
+"!!",
+"~~",
+"{{",
+"]]",
+"^^",
+"//",
+"((",
+"__",
+"::",
+"<<"};
diff -Naur orig/SilverXP/titleAL.xpm mod/SilverXP/titleAL.xpm
--- orig/SilverXP/titleAL.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/titleAL.xpm	2014-12-21 16:00:11.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
-static char * titleAB_xpm[] = {
-"1 26 26 1",
+static char * titleAL_xpm[] = {
+"2 26 26 1",
 " 	c None",
 ".	c #BCBCCF",
 "+	c #A8A7BF",
@@ -27,29 +27,29 @@
 "_	c #E4E3E3",
 ":	c #BCBDCD",
 "<	c #777792",
-".",
-"+",
-"@",
-"#",
-"$",
-"%",
-"&",
-"&",
-"*",
-"=",
-"-",
-";",
-">",
-",",
-"'",
-")",
-"!",
-"~",
-"{",
-"]",
-"^",
-"/",
-"(",
-"_",
-":",
-"<"};
+"..",
+"++",
+"@@",
+"##",
+"$$",
+"%%",
+"&&",
+"&&",
+"**",
+"==",
+"--",
+";;",
+">>",
+",,",
+"''",
+"))",
+"!!",
+"~~",
+"{{",
+"]]",
+"^^",
+"//",
+"((",
+"__",
+"::",
+"<<"};
diff -Naur orig/SilverXP/titleAM.xpm mod/SilverXP/titleAM.xpm
--- orig/SilverXP/titleAM.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/titleAM.xpm	2014-12-21 16:00:11.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
-static char * titleAB_xpm[] = {
-"1 26 26 1",
+static char * titleAM_xpm[] = {
+"2 26 26 1",
 " 	c None",
 ".	c #BCBCCF",
 "+	c #A8A7BF",
@@ -27,29 +27,29 @@
 "_	c #E4E3E3",
 ":	c #BCBDCD",
 "<	c #777792",
-".",
-"+",
-"@",
-"#",
-"$",
-"%",
-"&",
-"&",
-"*",
-"=",
-"-",
-";",
-">",
-",",
-"'",
-")",
-"!",
-"~",
-"{",
-"]",
-"^",
-"/",
-"(",
-"_",
-":",
-"<"};
+"..",
+"++",
+"@@",
+"##",
+"$$",
+"%%",
+"&&",
+"&&",
+"**",
+"==",
+"--",
+";;",
+">>",
+",,",
+"''",
+"))",
+"!!",
+"~~",
+"{{",
+"]]",
+"^^",
+"//",
+"((",
+"__",
+"::",
+"<<"};
diff -Naur orig/SilverXP/titleAP.xpm mod/SilverXP/titleAP.xpm
--- orig/SilverXP/titleAP.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/titleAP.xpm	2014-12-21 16:00:11.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
-static char * titleAB_xpm[] = {
-"1 26 26 1",
+static char * titleAP_xpm[] = {
+"2 26 26 1",
 " 	c None",
 ".	c #BCBCCF",
 "+	c #A8A7BF",
@@ -27,29 +27,29 @@
 "_	c #E4E3E3",
 ":	c #BCBDCD",
 "<	c #777792",
-".",
-"+",
-"@",
-"#",
-"$",
-"%",
-"&",
-"&",
-"*",
-"=",
-"-",
-";",
-">",
-",",
-"'",
-")",
-"!",
-"~",
-"{",
-"]",
-"^",
-"/",
-"(",
-"_",
-":",
-"<"};
+"..",
+"++",
+"@@",
+"##",
+"$$",
+"%%",
+"&&",
+"&&",
+"**",
+"==",
+"--",
+";;",
+">>",
+",,",
+"''",
+"))",
+"!!",
+"~~",
+"{{",
+"]]",
+"^^",
+"//",
+"((",
+"__",
+"::",
+"<<"};
diff -Naur orig/SilverXP/titleAR.xpm mod/SilverXP/titleAR.xpm
--- orig/SilverXP/titleAR.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/titleAR.xpm	2014-12-21 16:00:11.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
-static char * titleAB_xpm[] = {
-"1 26 26 1",
+static char * titleAR_xpm[] = {
+"2 26 26 1",
 " 	c None",
 ".	c #BCBCCF",
 "+	c #A8A7BF",
@@ -27,29 +27,29 @@
 "_	c #E4E3E3",
 ":	c #BCBDCD",
 "<	c #777792",
-".",
-"+",
-"@",
-"#",
-"$",
-"%",
-"&",
-"&",
-"*",
-"=",
-"-",
-";",
-">",
-",",
-"'",
-")",
-"!",
-"~",
-"{",
-"]",
-"^",
-"/",
-"(",
-"_",
-":",
-"<"};
+"..",
+"++",
+"@@",
+"##",
+"$$",
+"%%",
+"&&",
+"&&",
+"**",
+"==",
+"--",
+";;",
+">>",
+",,",
+"''",
+"))",
+"!!",
+"~~",
+"{{",
+"]]",
+"^^",
+"//",
+"((",
+"__",
+"::",
+"<<"};
diff -Naur orig/SilverXP/titleAS.xpm mod/SilverXP/titleAS.xpm
--- orig/SilverXP/titleAS.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/titleAS.xpm	2014-12-21 16:00:11.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
-static char * titleAB_xpm[] = {
-"1 26 26 1",
+static char * titleAS_xpm[] = {
+"2 26 26 1",
 " 	c None",
 ".	c #BCBCCF",
 "+	c #A8A7BF",
@@ -27,29 +27,29 @@
 "_	c #E4E3E3",
 ":	c #BCBDCD",
 "<	c #777792",
-".",
-"+",
-"@",
-"#",
-"$",
-"%",
-"&",
-"&",
-"*",
-"=",
-"-",
-";",
-">",
-",",
-"'",
-")",
-"!",
-"~",
-"{",
-"]",
-"^",
-"/",
-"(",
-"_",
-":",
-"<"};
+"..",
+"++",
+"@@",
+"##",
+"$$",
+"%%",
+"&&",
+"&&",
+"**",
+"==",
+"--",
+";;",
+">>",
+",,",
+"''",
+"))",
+"!!",
+"~~",
+"{{",
+"]]",
+"^^",
+"//",
+"((",
+"__",
+"::",
+"<<"};
diff -Naur orig/SilverXP/titleAT.xpm mod/SilverXP/titleAT.xpm
--- orig/SilverXP/titleAT.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/titleAT.xpm	2014-12-21 16:00:11.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
-static char * titleAB_xpm[] = {
-"1 26 26 1",
+static char * titleAT_xpm[] = {
+"2 26 26 1",
 " 	c None",
 ".	c #BCBCCF",
 "+	c #A8A7BF",
@@ -27,29 +27,29 @@
 "_	c #E4E3E3",
 ":	c #BCBDCD",
 "<	c #777792",
-".",
-"+",
-"@",
-"#",
-"$",
-"%",
-"&",
-"&",
-"*",
-"=",
-"-",
-";",
-">",
-",",
-"'",
-")",
-"!",
-"~",
-"{",
-"]",
-"^",
-"/",
-"(",
-"_",
-":",
-"<"};
+"..",
+"++",
+"@@",
+"##",
+"$$",
+"%%",
+"&&",
+"&&",
+"**",
+"==",
+"--",
+";;",
+">>",
+",,",
+"''",
+"))",
+"!!",
+"~~",
+"{{",
+"]]",
+"^^",
+"//",
+"((",
+"__",
+"::",
+"<<"};
diff -Naur orig/SilverXP/titleIB.xpm mod/SilverXP/titleIB.xpm
--- orig/SilverXP/titleIB.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/titleIB.xpm	2014-12-21 16:00:11.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
-static char * titleAB_xpm[] = {
-"1 26 26 1",
+static char * titleIB_xpm[] = {
+"2 26 26 1",
 " 	c None",
 ".	c #BCBCCF",
 "+	c #A8A7BF",
@@ -27,29 +27,29 @@
 "_	c #E4E3E3",
 ":	c #BCBDCD",
 "<	c #777792",
-".",
-"+",
-"@",
-"#",
-"$",
-"%",
-"&",
-"&",
-"*",
-"=",
-"-",
-";",
-">",
-",",
-"'",
-")",
-"!",
-"~",
-"{",
-"]",
-"^",
-"/",
-"(",
-"_",
-":",
-"<"};
+"..",
+"++",
+"@@",
+"##",
+"$$",
+"%%",
+"&&",
+"&&",
+"**",
+"==",
+"--",
+";;",
+">>",
+",,",
+"''",
+"))",
+"!!",
+"~~",
+"{{",
+"]]",
+"^^",
+"//",
+"((",
+"__",
+"::",
+"<<"};
diff -Naur orig/SilverXP/titleIL.xpm mod/SilverXP/titleIL.xpm
--- orig/SilverXP/titleIL.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/titleIL.xpm	2014-12-21 16:00:11.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
-static char * titleAB_xpm[] = {
-"1 26 26 1",
+static char * titleIL_xpm[] = {
+"2 26 26 1",
 " 	c None",
 ".	c #BCBCCF",
 "+	c #A8A7BF",
@@ -27,29 +27,29 @@
 "_	c #E4E3E3",
 ":	c #BCBDCD",
 "<	c #777792",
-".",
-"+",
-"@",
-"#",
-"$",
-"%",
-"&",
-"&",
-"*",
-"=",
-"-",
-";",
-">",
-",",
-"'",
-")",
-"!",
-"~",
-"{",
-"]",
-"^",
-"/",
-"(",
-"_",
-":",
-"<"};
+"..",
+"++",
+"@@",
+"##",
+"$$",
+"%%",
+"&&",
+"&&",
+"**",
+"==",
+"--",
+";;",
+">>",
+",,",
+"''",
+"))",
+"!!",
+"~~",
+"{{",
+"]]",
+"^^",
+"//",
+"((",
+"__",
+"::",
+"<<"};
diff -Naur orig/SilverXP/titleIM.xpm mod/SilverXP/titleIM.xpm
--- orig/SilverXP/titleIM.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/titleIM.xpm	2014-12-21 16:00:11.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
-static char * titleAB_xpm[] = {
-"1 26 26 1",
+static char * titleIM_xpm[] = {
+"2 26 26 1",
 " 	c None",
 ".	c #BCBCCF",
 "+	c #A8A7BF",
@@ -27,29 +27,29 @@
 "_	c #E4E3E3",
 ":	c #BCBDCD",
 "<	c #777792",
-".",
-"+",
-"@",
-"#",
-"$",
-"%",
-"&",
-"&",
-"*",
-"=",
-"-",
-";",
-">",
-",",
-"'",
-")",
-"!",
-"~",
-"{",
-"]",
-"^",
-"/",
-"(",
-"_",
-":",
-"<"};
+"..",
+"++",
+"@@",
+"##",
+"$$",
+"%%",
+"&&",
+"&&",
+"**",
+"==",
+"--",
+";;",
+">>",
+",,",
+"''",
+"))",
+"!!",
+"~~",
+"{{",
+"]]",
+"^^",
+"//",
+"((",
+"__",
+"::",
+"<<"};
diff -Naur orig/SilverXP/titleIP.xpm mod/SilverXP/titleIP.xpm
--- orig/SilverXP/titleIP.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/titleIP.xpm	2014-12-21 16:00:11.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
-static char * titleAB_xpm[] = {
-"1 26 26 1",
+static char * titleIP_xpm[] = {
+"2 26 26 1",
 " 	c None",
 ".	c #BCBCCF",
 "+	c #A8A7BF",
@@ -27,29 +27,29 @@
 "_	c #E4E3E3",
 ":	c #BCBDCD",
 "<	c #777792",
-".",
-"+",
-"@",
-"#",
-"$",
-"%",
-"&",
-"&",
-"*",
-"=",
-"-",
-";",
-">",
-",",
-"'",
-")",
-"!",
-"~",
-"{",
-"]",
-"^",
-"/",
-"(",
-"_",
-":",
-"<"};
+"..",
+"++",
+"@@",
+"##",
+"$$",
+"%%",
+"&&",
+"&&",
+"**",
+"==",
+"--",
+";;",
+">>",
+",,",
+"''",
+"))",
+"!!",
+"~~",
+"{{",
+"]]",
+"^^",
+"//",
+"((",
+"__",
+"::",
+"<<"};
diff -Naur orig/SilverXP/titleIR.xpm mod/SilverXP/titleIR.xpm
--- orig/SilverXP/titleIR.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/titleIR.xpm	2014-12-21 16:00:11.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
-static char * titleAB_xpm[] = {
-"1 26 26 1",
+static char * titleIR_xpm[] = {
+"2 26 26 1",
 " 	c None",
 ".	c #BCBCCF",
 "+	c #A8A7BF",
@@ -27,29 +27,29 @@
 "_	c #E4E3E3",
 ":	c #BCBDCD",
 "<	c #777792",
-".",
-"+",
-"@",
-"#",
-"$",
-"%",
-"&",
-"&",
-"*",
-"=",
-"-",
-";",
-">",
-",",
-"'",
-")",
-"!",
-"~",
-"{",
-"]",
-"^",
-"/",
-"(",
-"_",
-":",
-"<"};
+"..",
+"++",
+"@@",
+"##",
+"$$",
+"%%",
+"&&",
+"&&",
+"**",
+"==",
+"--",
+";;",
+">>",
+",,",
+"''",
+"))",
+"!!",
+"~~",
+"{{",
+"]]",
+"^^",
+"//",
+"((",
+"__",
+"::",
+"<<"};
diff -Naur orig/SilverXP/titleIS.xpm mod/SilverXP/titleIS.xpm
--- orig/SilverXP/titleIS.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/titleIS.xpm	2014-12-21 16:00:11.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
-static char * titleAB_xpm[] = {
-"1 26 26 1",
+static char * titleIS_xpm[] = {
+"2 26 26 1",
 " 	c None",
 ".	c #BCBCCF",
 "+	c #A8A7BF",
@@ -27,29 +27,29 @@
 "_	c #E4E3E3",
 ":	c #BCBDCD",
 "<	c #777792",
-".",
-"+",
-"@",
-"#",
-"$",
-"%",
-"&",
-"&",
-"*",
-"=",
-"-",
-";",
-">",
-",",
-"'",
-")",
-"!",
-"~",
-"{",
-"]",
-"^",
-"/",
-"(",
-"_",
-":",
-"<"};
+"..",
+"++",
+"@@",
+"##",
+"$$",
+"%%",
+"&&",
+"&&",
+"**",
+"==",
+"--",
+";;",
+">>",
+",,",
+"''",
+"))",
+"!!",
+"~~",
+"{{",
+"]]",
+"^^",
+"//",
+"((",
+"__",
+"::",
+"<<"};
diff -Naur orig/SilverXP/titleIT.xpm mod/SilverXP/titleIT.xpm
--- orig/SilverXP/titleIT.xpm	2004-06-05 21:08:44.000000000 +0200
+++ mod/SilverXP/titleIT.xpm	2014-12-21 16:00:11.000000000 +0100
@@ -1,6 +1,6 @@
 /* XPM */
-static char * titleAB_xpm[] = {
-"1 26 26 1",
+static char * titleIT_xpm[] = {
+"2 26 26 1",
 " 	c None",
 ".	c #BCBCCF",
 "+	c #A8A7BF",
@@ -27,29 +27,29 @@
 "_	c #E4E3E3",
 ":	c #BCBDCD",
 "<	c #777792",
-".",
-"+",
-"@",
-"#",
-"$",
-"%",
-"&",
-"&",
-"*",
-"=",
-"-",
-";",
-">",
-",",
-"'",
-")",
-"!",
-"~",
-"{",
-"]",
-"^",
-"/",
-"(",
-"_",
-":",
-"<"};
+"..",
+"++",
+"@@",
+"##",
+"$$",
+"%%",
+"&&",
+"&&",
+"**",
+"==",
+"--",
+";;",
+">>",
+",,",
+"''",
+"))",
+"!!",
+"~~",
+"{{",
+"]]",
+"^^",
+"//",
+"((",
+"__",
+"::",
+"<<"};
