diff -Naur orig/sortofaqua/dframeAB.xpm mod/sortofaqua/dframeAB.xpm
--- orig/sortofaqua/dframeAB.xpm	2001-02-12 20:21:12.000000000 +0100
+++ mod/sortofaqua/dframeAB.xpm	2014-12-21 16:24:02.000000000 +0100
@@ -1,11 +1,11 @@
 /* XPM */
-static char * frameAB_xpm[] = {
-"1 4 4 1",
+static char * dframeAB_xpm[] = {
+"2 4 4 1",
 " 	c None",
-".	c #FFFFFF",
-"+	c #F0F0F0",
-"@	c #757575",
-"@",
-".",
-"+",
-"@"};
+".	c #757575",
+"+	c #FFFFFF",
+"@	c #F0F0F0",
+"..",
+"++",
+"@@",
+".."};
diff -Naur orig/sortofaqua/dframeAT.xpm mod/sortofaqua/dframeAT.xpm
--- orig/sortofaqua/dframeAT.xpm	2001-02-12 20:21:12.000000000 +0100
+++ mod/sortofaqua/dframeAT.xpm	2014-12-21 16:24:49.000000000 +0100
@@ -1,11 +1,11 @@
 /* XPM */
-static char * frameAB_xpm[] = {
-"1 4 4 1",
+static char * dframeAT_xpm[] = {
+"2 4 4 1",
 " 	c None",
-".	c #FFFFFF",
-"+	c #F0F0F0",
-"@	c #757575",
-"@",
-".",
-".",
-"+"};
+".	c #757575",
+"+	c #FFFFFF",
+"@	c #F0F0F0",
+"..",
+"++",
+"++",
+"@@"};
diff -Naur orig/sortofaqua/frameIB.xpm mod/sortofaqua/frameIB.xpm
--- orig/sortofaqua/frameIB.xpm	2001-02-12 20:21:12.000000000 +0100
+++ mod/sortofaqua/frameIB.xpm	2014-12-21 16:26:02.000000000 +0100
@@ -1,11 +1,11 @@
 /* XPM */
-static char * frameAB_xpm[] = {
-"1 4 4 1",
+static char * frameIB_xpm[] = {
+"2 4 4 1",
 " 	c None",
-".	c #98B2D3",
-"+	c #9CB2C9",
-"@	c #657584",
-"@",
-".",
-"+",
-"@"};
+".	c #657584",
+"+	c #98B2D3",
+"@	c #9CB2C9",
+"..",
+"++",
+"@@",
+".."};
diff -Naur orig/sortofaqua/frameIT.xpm mod/sortofaqua/frameIT.xpm
--- orig/sortofaqua/frameIT.xpm	2001-02-12 20:21:12.000000000 +0100
+++ mod/sortofaqua/frameIT.xpm	2014-12-21 16:26:48.000000000 +0100
@@ -1,12 +1,12 @@
 /* XPM */
-static char * frameAB_xpm[] = {
-"1 4 5 1",
+static char * frameIT_xpm[] = {
+"2 4 5 1",
 " 	c None",
-".	c #98B2D3",
-"+	c #9CB2C9",
-"@	c #96A9C1",
-"#      c #657584",
-"#",
-".",
-"+",
-"@"};
+".	c #657584",
+"+	c #98B2D3",
+"@	c #9CB2C9",
+"#	c #96A9C1",
+"..",
+"++",
+"@@",
+"##"};
diff -Naur orig/sortofaqua/titleA.xpm mod/sortofaqua/titleA.xpm
--- orig/sortofaqua/titleA.xpm	2001-02-12 20:21:12.000000000 +0100
+++ mod/sortofaqua/titleA.xpm	2014-12-21 16:00:11.000000000 +0100
@@ -1,40 +1,39 @@
 /* XPM */
 static char * titleA_xpm[] = {
-"1 19 18 1",
+"2 19 17 1",
 " 	c None",
-".	c #FFFFFF",
-"+	c #F0F0F0",
-"@	c #E9E9E9",
-"#	c #F1F1F1",
-"$	c #EEEEEE",
-"%	c #FDFDFD",
-"&	c #EFEFEF",
-"*	c #E6E6E6",
-"=	c #EAEAEA",
-"-	c #F7F7F7",
-";	c #E8E8E8",
-">	c #DEDEDE",
-",	c #E0E0E0",
-"'	c #D9D9D9",
-")	c #CECECE",
-"!	c #CBCBCB",
-"~	c #757575",
-"@",
-"#",
-".",
-"#",
-"@",
-"$",
-"%",
-"&",
-"*",
-"=",
-"-",
-";",
-">",
-",",
-"=",
-"'",
-")",
-"!",
-"~"};
+".	c #E9E9E9",
+"+	c #F1F1F1",
+"@	c #FFFFFF",
+"#	c #EEEEEE",
+"$	c #FDFDFD",
+"%	c #EFEFEF",
+"&	c #E6E6E6",
+"*	c #EAEAEA",
+"=	c #F7F7F7",
+"-	c #E8E8E8",
+";	c #DEDEDE",
+">	c #E0E0E0",
+",	c #D9D9D9",
+"'	c #CECECE",
+")	c #CBCBCB",
+"!	c #757575",
+"..",
+"++",
+"@@",
+"++",
+"..",
+"##",
+"$$",
+"%%",
+"&&",
+"**",
+"==",
+"--",
+";;",
+">>",
+"**",
+",,",
+"''",
+"))",
+"!!"};
diff -Naur orig/sortofaqua/titleI.xpm mod/sortofaqua/titleI.xpm
--- orig/sortofaqua/titleI.xpm	2001-02-12 20:21:12.000000000 +0100
+++ mod/sortofaqua/titleI.xpm	2014-12-21 16:00:11.000000000 +0100
@@ -1,42 +1,39 @@
 /* XPM */
 static char * titleI_xpm[] = {
-"1 19 20 1",
+"2 19 17 1",
 " 	c None",
-".	c #98B2D3",
-"+	c #9CB2C9",
-"@	c #96A9C1",
-"#	c #97A7C8",
-"$	c #96A9CB",
-"%	c #9DB3D0",
-"&	c #93ACC2",
-"*	c #92A6C3",
-"=	c #95AAC6",
-"-	c #9EB2CF",
-";	c #93A7C4",
-">	c #9AAFCB",
-",	c #91A5C2",
-"'	c #8CA5C3",
-")	c #91A6C0",
-"!	c #8DA0C0",
-"~	c #89A1C2",
-"{	c #89A3C0",
-"]	c #657584",
-"#",
-"$",
-"%",
-"&",
-"*",
-"=",
-"-",
-";",
-";",
-";",
-">",
-",",
-"'",
-")",
-">",
-"!",
-"~",
-"{",
-"]"};
+".	c #97A7C8",
+"+	c #96A9CB",
+"@	c #9DB3D0",
+"#	c #93ACC2",
+"$	c #92A6C3",
+"%	c #95AAC6",
+"&	c #9EB2CF",
+"*	c #93A7C4",
+"=	c #9AAFCB",
+"-	c #91A5C2",
+";	c #8CA5C3",
+">	c #91A6C0",
+",	c #8DA0C0",
+"'	c #89A1C2",
+")	c #89A3C0",
+"!	c #657584",
+"..",
+"++",
+"@@",
+"##",
+"$$",
+"%%",
+"&&",
+"**",
+"**",
+"**",
+"==",
+"--",
+";;",
+">>",
+"==",
+",,",
+"''",
+"))",
+"!!"};
