diff -Naur orig/unusualscars/default.theme mod/unusualscars/default.theme
--- orig/unusualscars/default.theme	2000-04-18 12:38:40.000000000 +0200
+++ mod/unusualscars/default.theme	2014-12-21 18:53:29.000000000 +0100
@@ -66,6 +66,9 @@
 NormalTaskBarFontName="-*-lucida-medium-r-normal-*-*-100-*-*-p-*-iso8859-1"
 ShowMenuButtonIcon=0
 
+DesktopBackgroundCenter=0 # 0/1
+DesktopBackgroundImage="unusualscars.png"
+
 
 
 
