diff -Naur orig/Thunar-bulk-rename.desktop.in.in patched/Thunar-bulk-rename.desktop.in.in
--- orig/Thunar-bulk-rename.desktop.in.in	2012-03-28 19:04:36.000000000 +0200
+++ patched/Thunar-bulk-rename.desktop.in.in	2012-04-08 20:34:03.439647964 +0200
@@ -2,7 +2,7 @@
 _Name=Bulk Rename
 _Comment=Rename Multiple Files
 _GenericName=Bulk Rename
-Exec=@HELPERDIR@/Thunar/ThunarBulkRename %F
+Exec=ThunarBulkRename %F
 Icon=Thunar
 Terminal=false
 StartupNotify=true
