diff -Naur orig/dump/Makefile patched/dump/Makefile
--- orig/dump/Makefile	2017-09-21 23:15:05.000000000 +0200
+++ patched/dump/Makefile	2019-10-06 18:36:59.506339417 +0200
@@ -100,7 +100,6 @@
 	$(INSTALL) -m 755 -d $(PKG_ROOT_SBIN_DIR)
 	$(LTINSTALL) -m 755 $(LTCOMMAND) $(PKG_ROOT_SBIN_DIR)
 	$(INSTALL) -m 755 -d $(PKG_SBIN_DIR)
-	$(INSTALL) -S $(PKG_ROOT_SBIN_DIR)/$(LTCOMMAND) $(PKG_SBIN_DIR)/$(LTCOMMAND)
 install-dev:
 
 .dep: $(COMMINCL) $(COMMON) $(INVINCL) $(INVCOMMON)
diff -Naur orig/restore/Makefile patched/restore/Makefile
--- orig/restore/Makefile	2017-09-21 23:15:05.000000000 +0200
+++ patched/restore/Makefile	2019-10-06 18:36:59.506339417 +0200
@@ -110,7 +110,6 @@
 	$(INSTALL) -m 755 -d $(PKG_ROOT_SBIN_DIR)
 	$(LTINSTALL) -m 755 $(LTCOMMAND) $(PKG_ROOT_SBIN_DIR)
 	$(INSTALL) -m 755 -d $(PKG_SBIN_DIR)
-	$(INSTALL) -S $(PKG_ROOT_SBIN_DIR)/$(LTCOMMAND) $(PKG_SBIN_DIR)/$(LTCOMMAND)
 install-dev:
 
 .dep: $(COMMINCL) $(COMMON) $(INVINCL) $(INVCOMMON)
