diff -Naur orig/suite/browser/browser-prefs.js patched/suite/browser/browser-prefs.js
--- orig/suite/browser/browser-prefs.js	2017-07-07 07:15:27.134739579 +0200
+++ patched/suite/browser/browser-prefs.js	2017-07-26 23:08:54.544551815 +0200
@@ -20,6 +20,9 @@
 pref("startup.homepage_override_url","chrome://navigator-region/locale/region.properties");
 pref("general.skins.selectedSkin", "classic/1.0");
 
+pref("intl.locale.matchOS", true);
+pref("shell.checkDefaultApps", 0);
+
 pref("browser.chromeURL","chrome://navigator/content/navigator.xul");
 pref("browser.hiddenWindowChromeURL", "chrome://navigator/content/hiddenWindow.xul");
 
@@ -70,7 +73,7 @@
 pref("browser.download.manager.addToRecentDocs", true);
 pref("browser.download.manager.scanWhenDone", true);
 pref("browser.download.manager.resumeOnWakeDelay", 10000);
-pref("browser.download.manager.flashCount", 2);
+pref("browser.download.manager.flashCount", 0);
 pref("browser.download.manager.showWhenStarting", true);
 pref("browser.download.manager.focusWhenStarting", false);
 pref("browser.download.manager.closeWhenDone", false);
@@ -479,10 +482,10 @@
 // App-specific update preferences
 
 // Whether or not app updates are enabled
-pref("app.update.enabled", true);
+pref("app.update.enabled", false);
 
 // This preference allows automatic download and install to take place.
-pref("app.update.auto", true);
+pref("app.update.auto", false);
 
 // If set to true, the Update Service will present no UI for any event.
 pref("app.update.silent", false);
@@ -587,7 +590,8 @@
 // Disable add-ons installed into the shared user and shared system areas by
 // default. This does not include the application directory. See the SCOPE
 // constants in AddonManager.jsm for values to use here.
-pref("extensions.autoDisableScopes", 15);
+pref("extensions.autoDisableScopes", 0);
+pref("extensions.shownSelectionUI", true);
 
 // Preferences for AMO integration
 pref("extensions.getAddons.cache.enabled", true);  // also toggles personalized recommendations
