diff -Naur orig/comm/suite/app/profile/suite-prefs.js patched/comm/suite/app/profile/suite-prefs.js
--- orig/comm/suite/app/profile/suite-prefs.js	2021-03-01 19:18:24.000000000 +0100
+++ patched/comm/suite/app/profile/suite-prefs.js	2021-04-08 16:27:38.001878570 +0200
@@ -17,6 +17,9 @@
  *  - Computed values (e.g. 50 * 1024) don't work.
  */
 
+pref("intl.locale.matchOS", true);
+pref("shell.checkDefaultApps", 0);
+
 pref("startup.homepage_override_url","chrome://navigator-region/locale/region.properties");
 pref("general.skins.selectedSkin", "classic/1.0");
 
@@ -65,7 +68,7 @@
 
 pref("browser.download.manager.addToRecentDocs", true);
 pref("browser.download.manager.resumeOnWakeDelay", 10000);
-pref("browser.download.manager.flashCount", 2);
+pref("browser.download.manager.flashCount", 0);
 pref("browser.download.manager.focusWhenStarting", false);
 pref("browser.download.progress.closeWhenDone", false);
 
@@ -487,10 +490,10 @@
 // App-specific update preferences
 
 // Whether or not app updates are enabled
-pref("app.update.enabled", true);
+pref("app.update.enabled", false);
 
 // This preference allows automatic download and install to take place.
-pref("app.update.auto", true);
+pref("app.update.auto", false);
 
 // If set to true, the Update Service will present no UI for any event.
 pref("app.update.silent", false);
@@ -578,7 +581,8 @@
 
 // Disable add-ons installed into the shared user and shared system areas by
 // default. See the SCOPE constants in AddonManager.jsm for values to use here.
-pref("extensions.autoDisableScopes", 11);
+pref("extensions.autoDisableScopes", 0);
+pref("extensions.shownSelectionUI", true);
 
 // Enable add-ons installed and owned by the application, like the default theme.
 pref("extensions.startupScanScopes", 4);
