diff -Naur orig/suite/app/profile/suite-prefs.js patched/suite/app/profile/suite-prefs.js
--- orig/suite/app/profile/suite-prefs.js	2020-02-18 00:36:20.000000000 +0100
+++ patched/suite/app/profile/suite-prefs.js	2020-03-07 15:40:31.308106018 +0100
@@ -17,6 +17,9 @@
  *  - Computed values (e.g. 50 * 1024) don't work.
  */
 
+pref("intl.locale.matchOS", true);
+pref("shell.checkDefaultApps", 0);
+
 pref("startup.homepage_override_url","chrome://navigator-region/locale/region.properties");
 pref("general.skins.selectedSkin", "classic/1.0");
 
@@ -63,7 +66,7 @@
 
 pref("browser.download.manager.addToRecentDocs", true);
 pref("browser.download.manager.resumeOnWakeDelay", 10000);
-pref("browser.download.manager.flashCount", 2);
+pref("browser.download.manager.flashCount", 0);
 pref("browser.download.manager.focusWhenStarting", false);
 pref("browser.download.progress.closeWhenDone", false);
 
@@ -493,10 +496,10 @@
 // App-specific update preferences
 
 // Whether or not app updates are enabled
-pref("app.update.enabled", true);
+pref("app.update.enabled", false);
 
 // This preference allows automatic download and install to take place.
-pref("app.update.auto", true);
+pref("app.update.auto", false);
 
 // If set to true, the Update Service will present no UI for any event.
 pref("app.update.silent", false);
@@ -587,7 +590,8 @@
 // Disable add-ons installed into the shared user and shared system areas by
 // default. This does not include the application directory. See the SCOPE
 // constants in AddonManager.jsm for values to use here
-pref("extensions.autoDisableScopes", 15);
+pref("extensions.autoDisableScopes", 0);
+pref("extensions.shownSelectionUI", true);
 
 // Enable add-ons installed and owned by the application, like the default theme.
 pref("extensions.startupScanScopes", 4);
