diff -urN seamonkey-2.53.15.orig/build/moz.configure/init.configure seamonkey-2.53.15/build/moz.configure/init.configure
--- seamonkey-2.53.15.orig/build/moz.configure/init.configure	2023-01-27 10:36:27.459992613 +0100
+++ seamonkey-2.53.15/build/moz.configure/init.configure	2023-01-27 10:46:34.883848229 +0100
@@ -313,12 +313,11 @@
             die('The PYTHON environment variable does not contain '
                 'a valid path. Cannot find %s', python)
         python = found_python
-        if PY3:
-            try:
-                version = python_executable_version(python).version
-            except Exception as e:
-                raise FatalCheckError(
-                    'could not determine version of given PYTHON: %s' % e)
+        try:
+            version = python_executable_version(python).version
+        except Exception as e:
+            raise FatalCheckError(
+                'could not determine version of given PYTHON: %s' % e)
     elif mozillabuild:
         # MozillaBuild provides a Python 2.
         python = normsep('%s/python/python2.exe' % mozillabuild)
