diff -Naur a/build/moz.configure/init.configure b/build/moz.configure/init.configure
--- a/build/moz.configure/init.configure	2022-10-20 01:18:47.336835743 +0200
+++ b/build/moz.configure/init.configure	2022-10-20 01:23:57.234176014 +0200
@@ -208,6 +208,7 @@
 @imports(_from='mozbuild.virtualenv', _import='VirtualenvManager')
 @imports(_from='mozbuild.virtualenv', _import='verify_python_version')
 @imports(_from='mozbuild.virtualenv', _import='PY2')
+@imports(_from='six', _import='PY3')
 def virtualenv_python2(env_python, build_env, mozconfig, help):
     if help:
         return
@@ -269,11 +270,12 @@
             die('The PYTHON environment variable does not contain '
                 'a valid path. Cannot find %s', python)
         python = found_python
-        try:
-            version = python_executable_version(python).version
-        except Exception as e:
-            raise FatalCheckError(
-                'could not determine version of given PYTHON: %s' % e)
+        if PY3:
+            try:
+                version = python_executable_version(python).version
+            except Exception as e:
+                raise FatalCheckError(
+                    'could not determine version of given PYTHON: %s' % e)
     else:
         python = sys.executable
 
