diff -Naur a/js/src/threading/ExclusiveData.h b/js/src/threading/ExclusiveData.h
--- a/js/src/threading/ExclusiveData.h	2024-06-29 16:15:17.000000000 +0200
+++ b/js/src/threading/ExclusiveData.h	2024-10-16 01:23:03.543752863 +0200
@@ -120,12 +120,6 @@
         release();
     }
 
-    ExclusiveData(ExclusiveData&& rhs)
-      : lock_(std::move(rhs.lock))
-    {
-        MOZ_ASSERT(&rhs != this, "self-move disallowed!");
-        new (mozilla::KnownNotNull, value_.addr()) T(std::move(*rhs.value_.addr()));
-    }
 
     ExclusiveData& operator=(ExclusiveData&& rhs) {
         this->~ExclusiveData();
