diff -Naur orig/mozilla/config/baseconfig.mk patched/mozilla/config/baseconfig.mk
--- orig/mozilla/config/baseconfig.mk	2016-12-14 03:09:46.000000000 +0100
+++ patched/mozilla/config/baseconfig.mk	2016-12-25 16:41:26.929306904 +0100
@@ -4,8 +4,8 @@
 # whether a normal build is happening or whether the check is running.
 includedir := $(includedir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
 idldir = $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+installdir = $(libdir)/$(MOZ_APP_NAME)
+sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel
 ifeq (.,$(DEPTH))
 DIST = dist
 else
