diff -Naur orig/mozilla/memory/build/mozjemalloc.cpp patched/mozilla/memory/build/mozjemalloc.cpp
--- orig/mozilla/memory/build/mozjemalloc.cpp	2020-02-18 00:37:59.000000000 +0100
+++ patched/mozilla/memory/build/mozjemalloc.cpp	2020-08-30 02:00:21.236409958 +0200
@@ -220,7 +220,7 @@
 #include <sys/param.h>
 #include <sys/time.h>
 #include <sys/types.h>
-#if !defined(XP_SOLARIS) && !defined(ANDROID)
+#if !defined(XP_SOLARIS) && !defined(XP_LINUX) && !defined(ANDROID)
 #include <sys/sysctl.h>
 #endif
 #include <sys/uio.h>
