diff -Naur orig/mozilla/js/src/jsapi-tests/moz.build patched/mozilla/js/src/jsapi-tests/moz.build
--- orig/mozilla/js/src/jsapi-tests/moz.build	2018-07-12 04:17:01.000000000 +0200
+++ patched/mozilla/js/src/jsapi-tests/moz.build	2019-06-20 00:43:57.889201095 +0200
@@ -147,7 +147,7 @@
 OS_LIBS += CONFIG['MOZ_ZLIB_LIBS']
 
 if CONFIG['GNU_CXX']:
-    CXXFLAGS += ['-Wno-shadow', '-Werror=format']
+    CXXFLAGS += ['-Wno-shadow']
 
 # This is intended as a temporary workaround to enable VS2015.
 if CONFIG['_MSC_VER']:
diff -Naur orig/mozilla/js/src/moz.build patched/mozilla/js/src/moz.build
--- orig/mozilla/js/src/moz.build	2018-07-12 04:17:02.000000000 +0200
+++ patched/mozilla/js/src/moz.build	2019-06-20 00:43:41.864982054 +0200
@@ -785,7 +785,7 @@
         DEFINES['FFI_BUILDING'] = True
 
 if CONFIG['GNU_CXX']:
-    CXXFLAGS += ['-Wno-shadow', '-Werror=format']
+    CXXFLAGS += ['-Wno-shadow']
 
 # Suppress warnings in third-party code.
 if CONFIG['CLANG_CXX']:
diff -Naur orig/mozilla/js/src/shell/moz.build patched/mozilla/js/src/shell/moz.build
--- orig/mozilla/js/src/shell/moz.build	2018-07-12 04:17:03.000000000 +0200
+++ patched/mozilla/js/src/shell/moz.build	2019-06-20 00:44:10.197369339 +0200
@@ -51,7 +51,7 @@
 ]
 
 if CONFIG['GNU_CXX']:
-    CXXFLAGS += ['-Wno-shadow', '-Werror=format']
+    CXXFLAGS += ['-Wno-shadow']
 
 # This is intended as a temporary workaround to enable VS2015.
 if CONFIG['_MSC_VER']:
diff -Naur orig/mozilla/js/xpconnect/src/moz.build patched/mozilla/js/xpconnect/src/moz.build
--- orig/mozilla/js/xpconnect/src/moz.build	2018-07-12 04:17:10.000000000 +0200
+++ patched/mozilla/js/xpconnect/src/moz.build	2019-06-20 00:44:26.437591332 +0200
@@ -67,4 +67,4 @@
 ]
 
 if CONFIG['GNU_CXX']:
-    CXXFLAGS += ['-Wno-shadow', '-Werror=format']
+    CXXFLAGS += ['-Wno-shadow']
