diff -Naur orig/mozilla/toolkit/modules/CertUtils.jsm patched/mozilla/toolkit/modules/CertUtils.jsm
--- orig/mozilla/toolkit/modules/CertUtils.jsm	2016-12-14 03:07:00.000000000 +0100
+++ patched/mozilla/toolkit/modules/CertUtils.jsm	2016-12-25 16:46:56.001521346 +0100
@@ -172,7 +172,9 @@
 }
 
 function isBuiltinToken(tokenName) {
-  return tokenName == "Builtin Object Token";
+  return tokenName == "Builtin Object Token" ||
+         tokenName == "Default Trust" ||
+         tokenName == "System Trust";
 }
 
 /**
