diff -Naur quota-tools-orig/quotagrpadmins quota-tools/quotagrpadmins
--- quota-tools-orig/quotagrpadmins	2010-09-09 14:27:23.000000000 +0200
+++ quota-tools/quotagrpadmins	2011-09-28 19:59:24.000000000 +0200
@@ -4,5 +4,5 @@
 #  Comments begin with hash in the beginning of the line
 
 # In this file you specify users responsible for space used by the group
-users: root
-mygroup: chief
+#users: root
+#mygroup: chief
