diff -urN cdparanoia-III-alpha9.8.orig/paranoia/cdda_paranoia.h cdparanoia-III-alpha9.8/paranoia/cdda_paranoia.h
--- cdparanoia-III-alpha9.8.orig/paranoia/cdda_paranoia.h	2001-03-24 02:15:47.000000000 +0100
+++ cdparanoia-III-alpha9.8/paranoia/cdda_paranoia.h	2003-07-10 13:19:38.000000000 +0200
@@ -7,6 +7,8 @@
 #ifndef _CDROM_PARANOIA_
 #define _CDROM_PARANOIA_
 
+#include <cdda_interface.h>
+
 #define CD_FRAMEWORDS (CD_FRAMESIZE_RAW/2)
 
 #define PARANOIA_CB_READ           0
