diff -Nur orig/src/lib/accerciser/accerciser.py mod/src/lib/accerciser/accerciser.py
--- orig/src/lib/accerciser/accerciser.py	2016-02-23 18:22:34.000000000 +0100
+++ mod/src/lib/accerciser/accerciser.py	2019-01-19 13:19:12.000000000 +0100
@@ -103,7 +103,8 @@
     '''
     GLib.timeout_add(200, self._pumpEvents)
     try:
-      pyatspi.Registry.start(async=True, gil=False)
+      # async is a reserved keyword in Python 3.7+, so we pass the args as dict
+      pyatspi.Registry.start(**{'async': True, 'gil': False})
     except KeyboardInterrupt:
       self._shutDown()
 
