diff -Naur orig/systemd/gpsdctl@.service.in patched/systemd/gpsdctl@.service.in
--- orig/systemd/gpsdctl@.service.in	2020-06-11 23:03:14.000000000 +0200
+++ patched/systemd/gpsdctl@.service.in	2020-08-19 22:04:33.617668179 +0200
@@ -10,5 +10,5 @@
 EnvironmentFile=-/etc/default/gpsd
 EnvironmentFile=-/etc/sysconfig/gpsd
 RemainAfterExit=yes
-ExecStart=/bin/sh -c "[ \"$USBAUTO\" = true ] && /usr/local/sbin/gpsdctl add /dev/%I || :"
-ExecStop=/bin/sh -c "[ \"$USBAUTO\" = true ] && /usr/local/sbin/gpsdctl remove /dev/%I || :"
+ExecStart=/bin/sh -c "[ \"$USBAUTO\" = true ] && /usr/sbin/gpsdctl add /dev/%I || :"
+ExecStop=/bin/sh -c "[ \"$USBAUTO\" = true ] && /usr/sbin/gpsdctl remove /dev/%I || :"
diff -Naur orig/systemd/gpsd.service patched/systemd/gpsd.service
--- orig/systemd/gpsd.service	2020-06-05 05:01:16.000000000 +0200
+++ patched/systemd/gpsd.service	2020-08-19 22:03:47.921541524 +0200
@@ -8,7 +8,7 @@
 Type=forking
 EnvironmentFile=-/etc/default/gpsd
 EnvironmentFile=-/etc/sysconfig/gpsd
-ExecStart=/usr/local/sbin/gpsd $GPSD_OPTIONS $OPTIONS $DEVICES
+ExecStart=/usr/sbin/gpsd $GPSD_OPTIONS $OPTIONS $DEVICES
 
 [Install]
 WantedBy=multi-user.target
