diff -Naur orig/SConstruct patched/SConstruct
--- orig/SConstruct	2018-10-19 21:45:40.000000000 +0200
+++ patched/SConstruct	2018-10-21 01:15:20.924750377 +0200
@@ -1033,9 +1033,9 @@
                                                   brief=cleaning)
             # follow FHS, put in /usr/local/libXX, not /usr/libXX
             # may be lib, lib32 or lib64
-            python_libdir = polystr(python_libdir)
-            python_libdir = python_libdir.replace("/usr/lib",
-                                                  "/usr/local/lib")
+            # python_libdir = polystr(python_libdir)
+            # python_libdir = python_libdir.replace("/usr/lib",
+            #                                      "/usr/local/lib")
 
         py_config_text = config.GetPythonValue('config vars',
                                                PYTHON_SYSCONFIG_IMPORT,
