diff -Naur orig/Source/GSICUString.m patched/Source/GSICUString.m
--- orig/Source/GSICUString.m	2020-04-05 20:00:40.000000000 +0200
+++ patched/Source/GSICUString.m	2020-11-02 12:09:24.022213589 +0100
@@ -68,7 +68,7 @@
            * to match it.
            */
           ut->chunkOffset = nativeIndex - nativeStart;
-          return TRUE;
+          return true;
         }
 
       if (nativeIndex >= length && nativeLimit >= length)
@@ -77,7 +77,7 @@
            * Limit it to point just after the last character.
            */
           ut->chunkOffset = ut->chunkLength;
-          return FALSE;
+          return false;
         }
 
       /* Set up to fill the chunk with characters from the string
@@ -101,7 +101,7 @@
            * to match it.
            */
           ut->chunkOffset = nativeIndex - nativeStart;
-          return TRUE;
+          return true;
         }
 
       if (nativeIndex <= 0 && nativeStart <= 0)
@@ -110,7 +110,7 @@
            * Limit it to position of the first character.
            */
           ut->chunkOffset = 0;
-          return FALSE;
+          return false;
         }
 
       nativeLimit = nativeIndex;
@@ -132,7 +132,7 @@
   ut->chunkNativeStart = nativeStart;
   ut->nativeIndexingLimit = r.length;
   ut->chunkLength = r.length;
-  return TRUE;
+  return true;
 }
 
 /**
@@ -167,7 +167,7 @@
   // Setting the chunk length to 0 here forces UTextNSStringAccess to fetch
   // the data from the string object.
   ut->chunkLength = 0;
-  UTextNSStringAccess(ut, r.location + [replacement length] + 1, TRUE);
+  UTextNSStringAccess(ut, r.location + [replacement length] + 1, true);
   ut->chunkOffset++;
   
   [replacement release];
diff -Naur orig/Source/NSRegularExpression.m patched/Source/NSRegularExpression.m
--- orig/Source/NSRegularExpression.m	2020-04-05 20:00:40.000000000 +0200
+++ patched/Source/NSRegularExpression.m	2020-11-02 12:15:03.110626636 +0100
@@ -182,7 +182,7 @@
       return nil;
     }
   str = [GSUTextString new];
-  utext_clone(&str->txt, t, FALSE, TRUE, &s);
+  utext_clone(&str->txt, t, false, true, &s);
   return [str autorelease];
 }
 #else
@@ -288,7 +288,7 @@
 
   if (NULL == context)
     {
-      return FALSE;
+      return false;
     }
   CALL_BLOCK(block, nil, NSMatchingProgress, &stop);
   return stop;
@@ -365,11 +365,11 @@
   uregex_setRegion(r, range.location, range.location+range.length, &s);
   if (options & NSMatchingWithoutAnchoringBounds)
     {
-      uregex_useAnchoringBounds(r, FALSE, &s);
+      uregex_useAnchoringBounds(r, false, &s);
     }
   if (options & NSMatchingWithTransparentBounds)
     {
-      uregex_useTransparentBounds(r, TRUE, &s);
+      uregex_useTransparentBounds(r, true, &s);
     }
   uregex_setTimeLimit(r, _workLimit, &s);
   if (U_FAILURE(s))
@@ -401,11 +401,11 @@
   uregex_setRegion(r, range.location, range.location+range.length, &s);
   if (options & NSMatchingWithoutAnchoringBounds)
     {
-      uregex_useAnchoringBounds(r, FALSE, &s);
+      uregex_useAnchoringBounds(r, false, &s);
     }
   if (options & NSMatchingWithTransparentBounds)
     {
-      uregex_useTransparentBounds(r, TRUE, &s);
+      uregex_useTransparentBounds(r, true, &s);
     }
   uregex_setTimeLimit(r, _workLimit, &s);
   if (U_FAILURE(s))
@@ -839,7 +839,7 @@
       DESTROY(ret);
       return 0;
     }
-  utext_clone(&ret->txt, output, TRUE, TRUE, &s);
+  utext_clone(&ret->txt, output, true, true, &s);
   [string setString: ret];
   [ret release];
   uregex_close(r);
@@ -873,7 +873,7 @@
       DESTROY(ret);
       return nil;
     }
-  utext_clone(&ret->txt, output, TRUE, TRUE, &s);
+  utext_clone(&ret->txt, output, true, true, &s);
   uregex_close(r);
 
   utext_close(&txt);
@@ -911,7 +911,7 @@
       DESTROY(ret);
       return nil;
     }
-  utext_clone(&ret->txt, output, TRUE, TRUE, &s);
+  utext_clone(&ret->txt, output, true, true, &s);
   uregex_close(r);
 
   utext_close(&txt);
