diff -Naurd bluecurve-icon-theme-8.0.2.orig/theme/Bluecurve/index.theme bluecurve-icon-theme-8.0.2/theme/Bluecurve/index.theme
--- bluecurve-icon-theme-8.0.2.orig/theme/Bluecurve/index.theme	2008-04-08 01:44:55.000000000 +0200
+++ bluecurve-icon-theme-8.0.2/theme/Bluecurve/index.theme	2009-05-06 19:06:04.000000000 +0200
@@ -18,7 +18,7 @@
 SmallSizes=16
 PanelDefault=48
 PanelSizes=16,32,48
-Directories=96x96/apps,96x96/devices,96x96/emblems,96x96/filesystems,96x96/mimetypes,96x96/status,64x64/actions,64x64/apps,64x64/stock,64x64/status,48x48/emblems,48x48/actions,48x48/apps,48x48/devices,48x48/filesystems,48x48/mimetypes,48x48/stock,48x48/status,36x36/apps,36x36/filesystems,32x32/apps,32x32/filesystems,32x32/actions,32x32/stock,32x32/status,24x24/actions,24x24/apps,24x24/filesystems,24x24/stock,24x24/mimetypes,24x24/devices,24x24/ooo,24x24/status,20x20/actions,20x20/apps,20x20/stock,20x20/status,16x16/actions,16x16/apps,16x16/filesystems,16x16/stock,16x16/mimetypes,16x16/devices,16x16/ooo,16x16/status
+Directories=96x96/apps,96x96/devices,96x96/emblems,96x96/filesystems,96x96/mimetypes,96x96/status,64x64/actions,64x64/apps,64x64/stock,64x64/status,48x48/emblems,48x48/actions,48x48/apps,48x48/devices,48x48/filesystems,48x48/mimetypes,48x48/stock,48x48/status,36x36/apps,36x36/filesystems,32x32/apps,32x32/filesystems,32x32/actions,32x32/stock,32x32/status,24x24/actions,24x24/apps,24x24/filesystems,24x24/stock,24x24/mimetypes,24x24/devices,24x24/ooo,24x24/status,22x22/actions,22x22/apps,22x22/devices,22x22/emblems,22x22/filesystems,22x22/mimetypes,22x22/ooo,22x22/status,22x22/stock,20x20/actions,20x20/apps,20x20/stock,20x20/status,16x16/actions,16x16/apps,16x16/filesystems,16x16/stock,16x16/mimetypes,16x16/devices,16x16/ooo,16x16/status
 
 [16x16/actions]
 Size=16
@@ -80,6 +80,51 @@
 Size=20
 Type=Threshold
 
+[22x22/actions]
+Size=22
+Context=Actions
+Type=Threshold
+X-GNOME-AlreadyScaled=true
+
+[22x22/apps]
+Size=22
+Context=Applications
+Type=Threshold
+X-GNOME-AlreadyScaled=true
+
+[22x22/devices]
+Size=22
+Context=Devices
+Type=Threshold
+
+[22x22/emblems]
+Size=22
+Context=Emblems
+Type=Threshold
+
+[22x22/filesystems]
+Size=22
+Context=FileSystems
+Type=Threshold
+
+[22x22/mimetypes]
+Size=22
+Context=MimeTypes
+Type=Threshold
+
+[22x22/ooo]
+Size=22
+Type=Threshold
+
+[22x22/status]
+Size=22
+Type=Threshold
+
+[22x22/stock]
+Size=22
+Context=Stock
+Type=Threshold
+
 [24x24/actions]
 Size=24
 Context=Actions
