diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2019-06-30 01:29:04.000000000 +0200
+++ patched/CMakeLists.txt	2019-08-15 17:55:05.109144131 +0200
@@ -45,6 +45,7 @@
 include (MakeDistFiles)
 include (GNUInstallDirs)
 include (TestBigEndian)
+include (CheckSymbolExists)
 
 # ############################################################
 # These options are settable from the CMake command line. For example, to disable
