diff -up gnucash-3.7/CMakeLists.txt.foo gnucash-3.7/CMakeLists.txt
--- gnucash-3.7/CMakeLists.txt.foo	2019-09-10 00:32:33.595447506 -0400
+++ gnucash-3.7/CMakeLists.txt	2019-09-10 00:33:17.719820228 -0400
@@ -578,9 +578,9 @@ set(CMAKE_CXX_FLAGS "-std=gnu++11 ${CMAK
 
 
 if (UNIX)
-  set( CMAKE_C_FLAGS "-Werror -Wdeclaration-after-statement -Wno-pointer-sign -Wall -Wmissing-prototypes -Wmissing-declarations -Wno-unused ${CMAKE_C_FLAGS}")
+  set( CMAKE_C_FLAGS "-Werror -DGLIB_DISABLE_DEPRECATION_WARNINGS -Wdeclaration-after-statement -Wno-pointer-sign -Wall -Wmissing-prototypes -Wmissing-declarations -Wno-unused ${CMAKE_C_FLAGS}")
   set( CMAKE_C_FLAGS "-Wno-error=deprecated-declarations -std=gnu11 -Wno-error=parentheses ${CMAKE_C_FLAGS}")
-  set( CMAKE_CXX_FLAGS "-Werror -Wall -Wmissing-declarations -Wno-unused -Wno-error=parentheses ${CMAKE_CXX_FLAGS}")
+  set( CMAKE_CXX_FLAGS "-Werror -DGLIB_DISABLE_DEPRECATION_WARNINGS -Wall -Wmissing-declarations -Wno-unused -Wno-error=parentheses ${CMAKE_CXX_FLAGS}")
   set( CMAKE_CXX_FLAGS "-Wno-error=deprecated-declarations ${REGISTER_CXXFLAG} ${CMAKE_CXX_FLAGS}")
   set( CMAKE_C_FLAGS_RELEASE "-O3 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 ${CMAKE_C_FLAGS}")
 endif (UNIX)
