diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2023-03-11 23:58:01.000000000 +0100
+++ b/CMakeLists.txt	2023-03-20 23:25:05.748064361 +0100
@@ -675,7 +675,7 @@
 )
 
 
-install(FILES ${gnucash_DOCS} DESTINATION ${CMAKE_INSTALL_DOCDIR})
+# install(FILES ${gnucash_DOCS} DESTINATION ${CMAKE_INSTALL_DOCDIR})
 
 # ############################################################
 
@@ -823,19 +823,6 @@
 # it contains post-install actions to execute.
 add_subdirectory(cmake)
 
-# Generate the ChangeLog
-
-if (BUILDING_FROM_VCS)
-    add_custom_target(ChangeLog ALL
-        COMMAND ${GIT_EXECUTABLE} log --format=\"%ad %aN %n%n%x09* %s%d%n\" --date=short --since=2023-01-01 > ${CMAKE_BINARY_DIR}/ChangeLog
-        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-    )
-    install(FILES ${CMAKE_BINARY_DIR}/ChangeLog DESTINATION ${CMAKE_INSTALL_DOCDIR})
-else()
-    install(FILES ${CMAKE_SOURCE_DIR}/ChangeLog DESTINATION ${CMAKE_INSTALL_DOCDIR})
-endif()
-dist_add_generated (${BUILDING_FROM_VCS} ChangeLog)
-
 ############################ BEGIN MAKE DIST #################
 
 set(PACKAGE_PREFIX "${PROJECT_NAME}-${PROJECT_VERSION}")
