diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2025-09-28 00:10:46.000000000 +0200
+++ b/CMakeLists.txt	2025-09-30 19:15:13.957414538 +0200
@@ -763,7 +763,6 @@
 )
 
 
-install(FILES ${gnucash_DOCS} DESTINATION ${CMAKE_INSTALL_DOCDIR})
 
 # ############################################################
 
@@ -908,19 +907,6 @@
 # it contains post-install actions to execute.
 add_subdirectory(cmake)
 
-# Generate the ChangeLog
-
-if (BUILDING_FROM_VCS)
-    add_custom_target(ChangeLog ALL
-        COMMAND ${GIT_EXECUTABLE} log --format=\"%ad %aN %n%n%x09* %s%d%n\" --date=short --since=2025-01-01 > ${CMAKE_BINARY_DIR}/ChangeLog
-        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-    )
-    install(FILES ${CMAKE_BINARY_DIR}/ChangeLog DESTINATION ${CMAKE_INSTALL_DOCDIR})
-else()
-    install(FILES ${CMAKE_SOURCE_DIR}/ChangeLog DESTINATION ${CMAKE_INSTALL_DOCDIR})
-endif()
-dist_add_generated (${BUILDING_FROM_VCS} ChangeLog)
-
 ############################ BEGIN MAKE DIST #################
 
 set(PACKAGE_PREFIX "${PROJECT_NAME}-${PROJECT_VERSION}")
