diff -Naur json-c-json-c-0.13-20171207-orig/Makefile.am json-c-json-c-0.13-20171207/Makefile.am
--- json-c-json-c-0.13-20171207-orig/Makefile.am	2017-12-08 00:28:34.000000000 +0100
+++ json-c-json-c-0.13-20171207/Makefile.am	2017-12-23 11:26:53.708944954 +0100
@@ -34,6 +34,7 @@
 	printbuf.h
 
 noinst_HEADERS=\
+	json_object_iterator.h \
 	json_object_private.h \
 	math_compat.h \
 	strdup_compat.h \
diff -Naur json-c-json-c-0.13-20171207-orig/Makefile.in json-c-json-c-0.13-20171207/Makefile.in
--- json-c-json-c-0.13-20171207-orig/Makefile.in	2017-12-08 00:28:34.000000000 +0100
+++ json-c-json-c-0.13-20171207/Makefile.in	2017-12-23 11:33:48.625861756 +0100
@@ -410,6 +410,7 @@
 	printbuf.h
 
 noinst_HEADERS = \
+	json_object_iterator.h \
 	json_object_private.h \
 	math_compat.h \
 	strdup_compat.h \
