diff -Nur orig/Makefile mod/Makefile
--- orig/Makefile	2017-09-05 18:39:32.000000000 +0200
+++ mod/Makefile	2017-09-15 16:04:57.275720454 +0200
@@ -10,7 +10,7 @@
 CONFDIR?=/etc/iproute2
 DATADIR?=$(PREFIX)/share
 HDRDIR?=$(PREFIX)/include/iproute2
-DOCDIR?=$(DATADIR)/doc/iproute2
+DOCDIR?=$(DATADIR)/doc/Packages/iproute2
 MANDIR?=$(DATADIR)/man
 ARPDDIR?=/var/lib/arpd
 KERNEL_INCLUDE?=/usr/include
