diff -urN udftools-1.0.0b3.orig/Makefile.in udftools-1.0.0b3/Makefile.in
--- udftools-1.0.0b3.orig/Makefile.in	2004-02-23 10:34:56.000000000 +0100
+++ udftools-1.0.0b3/Makefile.in	2004-06-04 16:34:42.000000000 +0200
@@ -83,7 +83,7 @@
 am__include = @am__include@
 am__quote = @am__quote@
 install_sh = @install_sh@
-SUBDIRS = libudffs mkudffs cdrwtool pktsetup udffsck wrudf doc udfct
+SUBDIRS = libudffs mkudffs cdrwtool pktsetup doc udfct
 subdir = .
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 mkinstalldirs = $(SHELL) $(top_srcdir)/config/mkinstalldirs
