diff -Naur a/compton-conf.desktop.in b/compton-conf.desktop.in
--- a/compton-conf.desktop.in	2020-11-03 15:30:07.000000000 +0100
+++ b/compton-conf.desktop.in	2022-06-16 01:16:06.581644538 +0200
@@ -6,6 +6,6 @@
 TryExec=compton-conf
 Exec=compton-conf
 Icon=compton
-Categories=Settings;DesktopSettings;Qt;LXQt;X-XFCE-SettingsDialog;X-XFCE-PersonalSettings;
+Categories=Settings;DesktopSettings;Qt;X-LXQt;X-XFCE-SettingsDialog;X-XFCE-PersonalSettings;
 
 #TRANSLATIONS_DIR=translations
