diff -Naur playmidi-2.4.orig/Makefile playmidi-2.4/Makefile
--- playmidi-2.4.orig/Makefile	1998-10-01 04:39:18.000000000 +0200
+++ playmidi-2.4/Makefile	2004-01-01 22:33:51.000000000 +0100
@@ -36,7 +36,7 @@
 INCVGA	= -I/usr/local/include
 
 # Directory where application defaults files are stored for X11 version
-XAPPDEFAULTS = /usr/X11R6/lib/X11/app-defaults
+XAPPDEFAULTS = /usr/share/X11/app-defaults
 
 INCLUDES= $(INCNC) $(INCX11) $(INCVGA) $(INCAWE)
 INSTALLDIR = /usr/bin
