diff -Naurd playmidi-2.4.orig/playmidi.h-dist playmidi-2.4/playmidi.h-dist
--- playmidi-2.4.orig/playmidi.h-dist	1997-05-11 23:26:36.000000000 +0200
+++ playmidi-2.4/playmidi.h-dist	2003-12-11 13:15:04.000000000 +0100
@@ -1,4 +1,4 @@
-#define RELEASE "Playmidi 2.4"
+#define RELEASE "Playmidi 2.5"
 /************************************************************************
    playmidi.h  --  defines and structures for use by playmidi package
 
