diff -Naur orig/configure patched/configure
--- orig/configure	2019-11-23 14:59:04.000000000 +0100
+++ patched/configure	2020-01-28 21:51:28.419714257 +0100
@@ -16544,14 +16544,14 @@
 	       { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_res" >&5
 $as_echo "$ac_res" >&6; }
 if eval test \"x\$"$as_CACHEVAR"\" = x"yes"; then :
-  CFLAGS="-O3 -qansialias -w $xlc_opt"
+  CFLAGS="-O2 -qansialias -w $xlc_opt"
 else
-  CFLAGS="-O3 -qansialias -w"
+  CFLAGS="-O2 -qansialias -w"
                 echo "******************************************************"
                 echo "*  You seem to have the IBM  C compiler.  It is      *"
                 echo "*  recommended for best performance that you use:    *"
                 echo "*                                                    *"
-                echo "*    CFLAGS=-O3 -qarch=xxx -qtune=xxx -qansialias -w *"
+                echo "*    CFLAGS=-O2 -qarch=xxx -qtune=xxx -qansialias -w *"
                 echo "*                      ^^^        ^^^                *"
                 echo "*  where xxx is pwr2, pwr3, 604, or whatever kind of *"
                 echo "*  CPU you have.  (Set the CFLAGS environment var.   *"
@@ -16561,7 +16561,7 @@
 
          ;;
 
-    intel) CFLAGS="-O3 -ansi_alias"
+    intel) CFLAGS="-O2 -ansi_alias"
 	if test "x$acx_maxopt_portable" = xno; then
 	  icc_archflag=unknown
 	  icc_flags=""
@@ -16756,7 +16756,7 @@
 
     gnu)
      # default optimization flags for gcc on all systems
-     CFLAGS="-O3 -fomit-frame-pointer"
+     CFLAGS="-O2 -fomit-frame-pointer"
 
      # -malign-double for x86 systems
      # libffi local change -- don't align double, as it changes the ABI
@@ -17253,10 +17253,10 @@
 	echo "********************************************************"
         echo "* WARNING: Don't know the best CFLAGS for this system  *"
         echo "* Use ./configure CFLAGS=... to specify your own flags *"
-	echo "* (otherwise, a default of CFLAGS=-O3 will be used)    *"
+	echo "* (otherwise, a default of CFLAGS=-O2 will be used)    *"
 	echo "********************************************************"
 	echo ""
-        CFLAGS="-O3"
+        CFLAGS="-O2"
   fi
 
   as_CACHEVAR=`$as_echo "ax_cv_check_cflags__$CFLAGS" | $as_tr_sh`
diff -Naur orig/m4/ax_cc_maxopt.m4 patched/m4/ax_cc_maxopt.m4
--- orig/m4/ax_cc_maxopt.m4	2019-11-21 11:03:47.000000000 +0100
+++ patched/m4/ax_cc_maxopt.m4	2020-01-28 21:51:09.723911712 +0100
@@ -91,13 +91,13 @@
            xlc_opt="-qtune=auto"
 	 fi
          AX_CHECK_COMPILE_FLAG($xlc_opt,
-		CFLAGS="-O3 -qansialias -w $xlc_opt",
-               [CFLAGS="-O3 -qansialias -w"
+		CFLAGS="-O2 -qansialias -w $xlc_opt",
+               [CFLAGS="-O2 -qansialias -w"
                 echo "******************************************************"
                 echo "*  You seem to have the IBM  C compiler.  It is      *"
                 echo "*  recommended for best performance that you use:    *"
                 echo "*                                                    *"
-                echo "*    CFLAGS=-O3 -qarch=xxx -qtune=xxx -qansialias -w *"
+                echo "*    CFLAGS=-O2 -qarch=xxx -qtune=xxx -qansialias -w *"
                 echo "*                      ^^^        ^^^                *"
                 echo "*  where xxx is pwr2, pwr3, 604, or whatever kind of *"
                 echo "*  CPU you have.  (Set the CFLAGS environment var.   *"
@@ -105,7 +105,7 @@
                 echo "******************************************************"])
          ;;
 
-    intel) CFLAGS="-O3 -ansi_alias"
+    intel) CFLAGS="-O2 -ansi_alias"
 	if test "x$acx_maxopt_portable" = xno; then
 	  icc_archflag=unknown
 	  icc_flags=""
@@ -146,7 +146,7 @@
 
     gnu)
      # default optimization flags for gcc on all systems
-     CFLAGS="-O3 -fomit-frame-pointer"
+     CFLAGS="-O2 -fomit-frame-pointer"
 
      # -malign-double for x86 systems
      # libffi local change -- don't align double, as it changes the ABI
@@ -173,10 +173,10 @@
 	echo "********************************************************"
         echo "* WARNING: Don't know the best CFLAGS for this system  *"
         echo "* Use ./configure CFLAGS=... to specify your own flags *"
-	echo "* (otherwise, a default of CFLAGS=-O3 will be used)    *"
+	echo "* (otherwise, a default of CFLAGS=-O2 will be used)    *"
 	echo "********************************************************"
 	echo ""
-        CFLAGS="-O3"
+        CFLAGS="-O2"
   fi
 
   AX_CHECK_COMPILE_FLAG($CFLAGS, [], [
