diff -Naur ORIG/data/org.gramps_project.Gramps.desktop.in UHU/data/org.gramps_project.Gramps.desktop.in
--- ORIG/data/org.gramps_project.Gramps.desktop.in	2009-03-10 07:13:41.000000000 +0100
+++ UHU/data/gorg.gramps_project.Gramps.desktop.in	2009-03-14 20:05:09.000000000 +0100
@@ -7,4 +7,4 @@
 StartupNotify=true
 Categories=GTK;Office;
 MimeType=application/x-gramps;application/x-gedcom;application/x-gramps-package;application/x-gramps-xml;
-Exec=gramps %F
+Exec=gramps
