diff -Naur orig/setup.py mod/setup.py
--- orig/setup.py	2016-04-10 17:59:24.000000000 +0200
+++ mod/setup.py	2016-04-12 22:36:21.506769986 +0200
@@ -261,7 +261,7 @@
             if packaging:
                 path = resource_path
             else:
-                path = os.path.abspath(os.path.join(self.install_data, 'share'))
+                path = '/usr/share'
             fp.write(path)
 
         _install.run(self)
