diff -Naur TeXmacs-1.99.7-src-orig/CMakeLists.txt TeXmacs-1.99.7-src/CMakeLists.txt
--- TeXmacs-1.99.7-src-orig/CMakeLists.txt	2018-07-04 14:04:31.000000000 +0200
+++ TeXmacs-1.99.7-src/CMakeLists.txt	2018-07-18 18:26:10.642333407 +0200
@@ -191,7 +191,7 @@
 
 IF(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
   MESSAGE(STATUS "final adjustments for an Intel or AMD GNU/Linux host")
-  SET(CONFIG_CXXOPTIMIZE "-O3 -fexpensive-optimizations")
+  SET(CONFIG_CXXOPTIMIZE "-O2 -fexpensive-optimizations")
 ENDIF(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
 
 IF(${CMAKE_SYSTEM_NAME} STREQUAL "FreeBSD")
diff -Naur TeXmacs-1.99.7-src-orig/configure TeXmacs-1.99.7-src/configure
--- TeXmacs-1.99.7-src-orig/configure	2018-07-06 10:58:53.000000000 +0200
+++ TeXmacs-1.99.7-src/configure	2018-07-18 18:28:20.232820902 +0200
@@ -5613,7 +5613,7 @@
       CONFIG_PACKAGE=GENERIC_PACKAGE
 
       CONFIG_OS="GNU_LINUX"
-      CONFIG_CXXOPTIMIZE="-O3 -fexpensive-optimizations"
+      CONFIG_CXXOPTIMIZE="-O2 -fexpensive-optimizations"
       CONFIG_QTPIPES="yes"
 
 $as_echo "#define STACK_SIZE 0x1000000" >>confdefs.h
@@ -5711,7 +5711,7 @@
       CONFIG_PACKAGE=GENERIC_PACKAGE
 
       CONFIG_OS="GNU_LINUX"
-      CONFIG_CXXOPTIMIZE="-O3 -fexpensive-optimizations"
+      CONFIG_CXXOPTIMIZE="-O2 -fexpensive-optimizations"
       CONFIG_QTPIPES="yes"
 
 $as_echo "#define STACK_SIZE 0x1000000" >>confdefs.h
@@ -6926,7 +6926,7 @@
 $as_echo "#define OS_GNU_LINUX 1" >>confdefs.h
 
       CONFIG_OS="GNU_LINUX"
-      CONFIG_CXXOPTIMIZE="-O3 -fexpensive-optimizations"
+      CONFIG_CXXOPTIMIZE="-O2 -fexpensive-optimizations"
 
 cat >>confdefs.h <<_ACEOF
 #define WORD_LENGTH 8
