diff -Naur orig/cmake/FindGuile.cmake patched/cmake/FindGuile.cmake
--- orig/cmake/FindGuile.cmake	2019-12-12 11:56:21.000000000 +0100
+++ patched/cmake/FindGuile.cmake	2019-12-31 00:01:23.095481576 +0100
@@ -40,7 +40,7 @@
     STRING(REGEX MATCHALL "[-][D]([^ ;])+" _guile_definitions_with_prefix "${_guileconfig_compile}" )
     STRING(REGEX MATCH "[0-9]+\\.[0-9]+\\.[0-9]+" Guile_VERSION_STRING "${_guileconfig_version}")
       
-    STRING(REPLACE "-L" " " _guile_libdirs ${_guile_libdirs_with_prefix} "")
+    STRING(REPLACE "-L" " " _guile_libdirs "${_guile_libdirs_with_prefix}")
     STRING(REPLACE "-l" " " _guile_lib_list "${_guile_libraries_with_prefix}" )
     STRING(REPLACE "-I" " " _guile_includes "${_guile_includes_with_prefix}" )
 #    SEPARATE_ARGUMENTS(_guile_libdirs)
diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2019-12-12 11:56:21.000000000 +0100
+++ patched/CMakeLists.txt	2019-12-31 00:04:10.753143877 +0100
@@ -200,18 +200,18 @@
 
 if (${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
   message (STATUS "final adjustments for an Intel or AMD GNU/Linux host")
-  set (CONFIG_CXXOPTIMIZE "-O3 -fexpensive-optimizations")
+  set (CONFIG_CXXOPTIMIZE "-O2 -fexpensive-optimizations")
 endif (${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
 
 if (${CMAKE_SYSTEM_NAME} STREQUAL "FreeBSD")
   message (STATUS "final adjustments for an Intel or AMD GNU/BSD host")
-  set (CONFIG_CXXOPTIMIZE "-O3 -fexpensive-optimizations")
+  set (CONFIG_CXXOPTIMIZE "-O2 -fexpensive-optimizations")
 endif (${CMAKE_SYSTEM_NAME} STREQUAL "FreeBSD")
 
 if (${CMAKE_SYSTEM_NAME} STREQUAL "SunOS")
   message (STATUS "final adjustments for an Intel or AMS Solaris host")
-  set (CONFIG_CXXOPTIMIZE "-O3 -fexpensive-optimizations")
-  set (CONFIG_CXXOPTIMIZE "-O3")
+  set (CONFIG_CXXOPTIMIZE "-O2 -fexpensive-optimizations")
+  set (CONFIG_CXXOPTIMIZE "-O2")
   set (CONFIG_BPATH "-Wl,-R,")
 #  SET(X11_LDFLAGS "${X_LIBS} -lXext -lX11 -lsocket")
 endif (${CMAKE_SYSTEM_NAME} STREQUAL "SunOS")
diff -Naur orig/configure patched/configure
--- orig/configure	2019-12-12 11:56:21.000000000 +0100
+++ patched/configure	2019-12-31 00:04:18.757223067 +0100
@@ -5715,7 +5715,7 @@
       CONFIG_PACKAGE=GENERIC_PACKAGE
 
       CONFIG_OS="GNU_LINUX"
-      CONFIG_CXXOPTIMIZE="-O3 -fexpensive-optimizations"
+      CONFIG_CXXOPTIMIZE="-O2 -fexpensive-optimizations"
       CONFIG_QTPIPES="yes"
 
 $as_echo "#define STACK_SIZE 0x1000000" >>confdefs.h
@@ -5813,7 +5813,7 @@
       CONFIG_PACKAGE=GENERIC_PACKAGE
 
       CONFIG_OS="GNU_LINUX"
-      CONFIG_CXXOPTIMIZE="-O3 -fexpensive-optimizations"
+      CONFIG_CXXOPTIMIZE="-O2 -fexpensive-optimizations"
       CONFIG_QTPIPES="yes"
 
 $as_echo "#define STACK_SIZE 0x1000000" >>confdefs.h
@@ -5909,7 +5909,7 @@
       CONFIG_OS="FREEBSD"
       CONFIG_QTPIPES="yes"
       CPPFLAGS="$CPPFLAGS -I/usr/local/include -I."
-      CONFIG_CXXOPTIMIZE="-O3 -fexpensive-optimizations"
+      CONFIG_CXXOPTIMIZE="-O2 -fexpensive-optimizations"
     ;;
     i*86-*-solaris*)
       { $as_echo "$as_me:${as_lineno-$LINENO}: result: an Intel or AMS Solaris host" >&5
@@ -5918,7 +5918,7 @@
 $as_echo "#define OS_SOLARIS 1" >>confdefs.h
 
       CONFIG_OS="SOLARIS"
-      CONFIG_CXXOPTIMIZE="-O3"
+      CONFIG_CXXOPTIMIZE="-O2"
       CONFIG_BPATH="-Wl,-R,"
       X11_LDFLAGS="$X_LIBS -lXext -lX11 -lsocket"
     ;;
@@ -5933,7 +5933,7 @@
       CONFIG_PACKAGE=WINDOWS_PACKAGE
 
       CONFIG_OS=MINGW
-      CONFIG_CXXOPTIMIZE="-O3 -fexpensive-optimizations"
+      CONFIG_CXXOPTIMIZE="-O2 -fexpensive-optimizations"
       CONFIG_QTPIPES="yes"
       CONFIG_OS_COMPAT="Windows"
       CPPFLAGS="$CPPFLAGS -I/usr/local/include -IPlugins/Windows -I."
@@ -7067,7 +7067,7 @@
 $as_echo "#define OS_POWERPC_GNU_LINUX 1" >>confdefs.h
 
       CONFIG_OS="POWERPC_GNU_LINUX"
-      CONFIG_CXXOPTIMIZE="-O3 -fexpensive-optimizations"
+      CONFIG_CXXOPTIMIZE="-O2 -fexpensive-optimizations"
     ;;
     *sun*)
       { $as_echo "$as_me:${as_lineno-$LINENO}: result: a SUN/solaris host" >&5
@@ -7105,7 +7105,7 @@
 $as_echo "#define OS_GNU_LINUX 1" >>confdefs.h
 
       CONFIG_OS="GNU_LINUX"
-      CONFIG_CXXOPTIMIZE="-O3 -fexpensive-optimizations"
+      CONFIG_CXXOPTIMIZE="-O2 -fexpensive-optimizations"
 
 cat >>confdefs.h <<_ACEOF
 #define WORD_LENGTH 8
diff -Naur orig/misc/m4/tm_platform.m4 patched/misc/m4/tm_platform.m4
--- orig/misc/m4/tm_platform.m4	2019-12-12 11:56:21.000000000 +0100
+++ patched/misc/m4/tm_platform.m4	2019-12-31 00:06:07.546297880 +0100
@@ -59,7 +59,7 @@
       AC_DEFINE([OS_GNU_LINUX],[1],[OS type])
       AC_SUBST([CONFIG_PACKAGE],[GENERIC_PACKAGE])
       CONFIG_OS="GNU_LINUX"
-      CONFIG_CXXOPTIMIZE="-O3 -fexpensive-optimizations"
+      CONFIG_CXXOPTIMIZE="-O2 -fexpensive-optimizations"
       CONFIG_QTPIPES="yes"
       AC_DEFINE([STACK_SIZE], 0x1000000, [If not set during link])
       AC_CHECK_LIB(expat,XML_ParserCreate,[CONFIG_BSTATIC="-lexpat $CONFIG_BSTATIC";CONFIG_STYPE=A])
@@ -82,13 +82,13 @@
       CONFIG_OS="FREEBSD"
       CONFIG_QTPIPES="yes"
       CPPFLAGS="$CPPFLAGS -I/usr/local/include -I."
-      CONFIG_CXXOPTIMIZE="-O3 -fexpensive-optimizations"
+      CONFIG_CXXOPTIMIZE="-O2 -fexpensive-optimizations"
     ;;
     i*86-*-solaris*)
       AC_MSG_RESULT(an Intel or AMS Solaris host)
       AC_DEFINE([OS_SOLARIS],[1],[OS type])
       CONFIG_OS="SOLARIS"
-      CONFIG_CXXOPTIMIZE="-O3"
+      CONFIG_CXXOPTIMIZE="-O2"
       CONFIG_BPATH="-Wl,-R,"
       X11_LDFLAGS="$X_LIBS -lXext -lX11 -lsocket"
     ;;
@@ -98,7 +98,7 @@
       AC_SUBST([CONFIG_BUNDLE],[WINDOWS_BUNDLE])
       AC_SUBST([CONFIG_PACKAGE],[WINDOWS_PACKAGE])
       CONFIG_OS=MINGW
-      CONFIG_CXXOPTIMIZE="-O3 -fexpensive-optimizations"
+      CONFIG_CXXOPTIMIZE="-O2 -fexpensive-optimizations"
       CONFIG_QTPIPES="yes"
       CONFIG_OS_COMPAT="Windows"
       CPPFLAGS="$CPPFLAGS -I/usr/local/include -IPlugins/Windows -I."
@@ -144,7 +144,7 @@
       AC_MSG_RESULT(a PowerPC/GNU-linux host)
       AC_DEFINE([OS_POWERPC_GNU_LINUX],[1],[OS type])
       CONFIG_OS="POWERPC_GNU_LINUX"
-      CONFIG_CXXOPTIMIZE="-O3 -fexpensive-optimizations"
+      CONFIG_CXXOPTIMIZE="-O2 -fexpensive-optimizations"
     ;;
     *sun*)
       AC_MSG_RESULT(a SUN/solaris host)
@@ -164,7 +164,7 @@
       AC_MSG_RESULT(a Sparc host running GNU/Linux)
       AC_DEFINE([OS_GNU_LINUX],[1],[OS type])
       CONFIG_OS="GNU_LINUX"
-      CONFIG_CXXOPTIMIZE="-O3 -fexpensive-optimizations"
+      CONFIG_CXXOPTIMIZE="-O2 -fexpensive-optimizations"
       AC_DEFINE_UNQUOTED([WORD_LENGTH],[8],[Pointer  size])
       AC_DEFINE_UNQUOTED([WORD_LENGTH_INC],[7],[Pointer increment])
       AC_DEFINE_UNQUOTED([WORD_MASK],[$( printf "0x%x" 0xfffffff8)],[Word Mask])
diff -Naur orig/TeXmacs/misc/mime/texmacs.desktop patched/TeXmacs/misc/mime/texmacs.desktop
--- orig/TeXmacs/misc/mime/texmacs.desktop	2019-12-12 11:56:21.000000000 +0100
+++ patched/TeXmacs/misc/mime/texmacs.desktop	2019-12-31 00:02:26.052106587 +0100
@@ -4,6 +4,7 @@
 Name=GNU TeXmacs
 GenericName=Text Editor
 Comment=A structured wysiwyg scientific text editor
+Comment[hu]=Tudományos szövegszerkesztő
 MimeType=text/x-texmacs.doc;text/x-texmacs.sty;text/plain;text/x-tex;
 Exec=texmacs %f
 Icon=texmacs
