diff -Naur orig/cmake/FindGuile.cmake patched/cmake/FindGuile.cmake
--- orig/cmake/FindGuile.cmake	2019-12-12 11:56:21.000000000 +0100
+++ patched/cmake/FindGuile.cmake	2020-05-09 17:34:13.556656628 +0200
@@ -6,7 +6,7 @@
 #  Guile_LIBRARIES - link these to use Freetype
 #  Guile_VERSION_STRING - version of Guile
 
-FIND_PROGRAM(GUILECONFIG_EXECUTABLE NAMES guile-config )
+FIND_PROGRAM(GUILECONFIG_EXECUTABLE NAMES guile-config1.8 guile-config2.0 guile-config )
 
 # if guile-config has been found
 IF(GUILECONFIG_EXECUTABLE)
