diff -Nur orig/Makefile mod/Makefile
--- orig/Makefile	2019-10-03 20:30:28.000000000 +0200
+++ mod/Makefile	2019-10-03 21:12:14.908672034 +0200
@@ -123,7 +123,7 @@
 rootperm:
 	@echo 'main(int argc,char**argv){unlink(argv[0]);return(getuid()==0);}'>g
 	@if gcc -x c -o G g && rm -f g && ./G ; then \
-		/bin/echo -e "\n\n      Need root permission for (de)installation!\n\n"; \
+		echo -e "\n\n      Need root permission for (de)installation!\n\n"; \
 		exit 1; \
 	fi
 
@@ -131,7 +131,7 @@
 	set -e; for i in `echo $(SUBDIRS)`; do $(MAKE) -C $$i install; done
 	@: || if [ -c $(DESTDIR)/dev/isdnctrl0 ] && ls -l $(DESTDIR)/dev/isdnctrl0 | egrep "[[:space:]]45,[[:space:]]+64[[:space:]]" > /dev/null; \
 	then \
-		/bin/echo -e '(some) ISDN devices already exist, not creating them.\nUse scripts/makedev.sh manually if necessary.'; \
+		echo -e '(some) ISDN devices already exist, not creating them.\nUse scripts/makedev.sh manually if necessary.'; \
 	else \
 		sh scripts/makedev.sh $(DESTDIR) ; \
 	fi
@@ -198,16 +198,16 @@
 	@echo Selected subdirs: $(BUILD_ONLY) $(SUBDIRS)
 	@set -e; for i in `echo $(BUILD_ONLY) $(SUBDIRS)`; do \
 		if [ $$i = eicon ] ; then \
-			/bin/echo -e "\nRunning configure in $$i ...\n"; sleep 1; \
+			echo -e "\nRunning configure in $$i ...\n"; sleep 1; \
 			(cd $$i; ./configure --with-sbin=$(CONFIG_SBINDIR) --bindir=$(CONFIG_BINDIR) --with-man=$(CONFIG_MANDIR) --datadir=$(CONFIG_DATADIR) --libdir=$(CONFIG_LIBDIR) --with-firmware=$(CONFIG_DATADIR) || $(MAKE) -C ../ ERRDIR=$$i cfgerror); \
 		elif [ -x $$i/configure ] ; then \
-			/bin/echo -e "\nRunning configure in $$i ...\n"; sleep 1; \
+			echo -e "\nRunning configure in $$i ...\n"; sleep 1; \
 			(cd $$i; ./configure --sbindir=$(CONFIG_SBINDIR) --bindir=$(CONFIG_BINDIR) --mandir=$(CONFIG_MANDIR) --datadir=$(CONFIG_DATADIR) --libdir=$(CONFIG_LIBDIR) || $(MAKE) -C ../ ERRDIR=$$i cfgerror); \
 		elif [ -f $$i/Makefile.in ] ; then \
-			/bin/echo -e "\nRunning make -f Makefile.in config in $$i ...\n"; sleep 1; \
+			echo -e "\nRunning make -f Makefile.in config in $$i ...\n"; sleep 1; \
 			$(MAKE) -C $$i -f Makefile.in config; \
 		elif [ -f $$i/Makefile ] ; then \
-			/bin/echo -e "\nRunning make config in $$i ...\n"; sleep 1; \
+			echo -e "\nRunning make config in $$i ...\n"; sleep 1; \
 			$(MAKE) -C $$i config; \
 		fi; \
 	done
diff -Nur orig/scripts/lxdialog/Makefile mod/scripts/lxdialog/Makefile
--- orig/scripts/lxdialog/Makefile	2019-10-03 20:30:28.000000000 +0200
+++ mod/scripts/lxdialog/Makefile	2019-10-03 21:07:39.221030989 +0200
@@ -33,7 +33,7 @@
 ncurses:
 	@x=`find /lib/ /usr/lib/ /usr/local/lib/ -maxdepth 2 -name 'libncurses.*'` ;\
 	if [ ! "$$x" ]; then \
-		/bin/echo -e "\007" ;\
+		echo -e "\007" ;\
 		echo ">> Unable to find the Ncurses libraries." ;\
 		echo ">>" ;\
 		echo ">> You must have Ncurses installed in order" ;\
diff -Nur orig/scripts/Menuconfig mod/scripts/Menuconfig
--- orig/scripts/Menuconfig	2019-10-03 20:30:28.000000000 +0200
+++ mod/scripts/Menuconfig	2019-10-03 21:07:05.209075155 +0200
@@ -57,7 +57,7 @@
 #
 function comment () {
 	comment_ctr=$[ comment_ctr + 1 ]
-	/bin/echo -ne "': $comment_ctr' '--- $1' " >>MCmenu
+	echo -ne "': $comment_ctr' '--- $1' " >>MCmenu
 }
 
 #
@@ -88,9 +88,9 @@
 	n)	flag=" " ;;
 	esac
 
-	/bin/echo -ne "'$2' '[$flag] $1' " >>MCmenu
+	echo -ne "'$2' '[$flag] $1' " >>MCmenu
 
-	/bin/echo -e "function $2 () { l_bool '$2' \"\$1\" ;}\n" >>MCradiolists
+	echo -e "function $2 () { l_bool '$2' \"\$1\" ;}\n" >>MCradiolists
 }
 
 #
@@ -113,9 +113,9 @@
 function int () {
 	eval $2=\${$2:-"$3"} x=\$$2
 
-	/bin/echo -ne "'$2' '($x) $1' " >>MCmenu
+	echo -ne "'$2' '($x) $1' " >>MCmenu
 
-	/bin/echo -e "function $2 () { l_int '$1' '$2' '$3' '$x' ;}" >>MCradiolists
+	echo -e "function $2 () { l_int '$1' '$2' '$3' '$x' ;}" >>MCradiolists
 }
 
 #
@@ -124,9 +124,9 @@
 function hex () {
 	eval $2=\${$2:-"$3"} x=\${$2##*[x,X]}
 
-	/bin/echo -ne "'$2' '($x) $1' " >>MCmenu
+	echo -ne "'$2' '($x) $1' " >>MCmenu
 
-	/bin/echo -e "function $2 () { l_hex '$1' '$2' '$3' '$x' ;}" >>MCradiolists
+	echo -e "function $2 () { l_hex '$1' '$2' '$3' '$x' ;}" >>MCradiolists
 }
 
 #
@@ -135,9 +135,9 @@
 function string () {
 	eval $2=\${"$2":-"$3"} x=\$"$2"
 
-	/bin/echo -ne "'$2' '\"$x\" $1' " >>MCmenu
+	echo -ne "'$2' '\"$x\" $1' " >>MCmenu
 
-	/bin/echo -e "function $2 () { l_string '$1' '$2' '$3' '$x' ;}" >>MCradiolists
+	echo -e "function $2 () { l_string '$1' '$2' '$3' '$x' ;}" >>MCradiolists
 }
 
 #
@@ -171,9 +171,9 @@
 
 	: ${current:=$default}
 
-	/bin/echo -ne "'$firstchoice' '($current) $title' " >>MCmenu
+	echo -ne "'$firstchoice' '($current) $title' " >>MCmenu
 
-	/bin/echo -e "
+	echo -e "
 	function $firstchoice () \
 		{ l_choice '$title' \"$choices\" $current ;}" >>MCradiolists
 }
@@ -249,7 +249,7 @@
 # Radiolist function file.
 #
 function menu_name () {
-	/bin/echo -ne "$DIALOG --title '$1'\
+	echo -ne "$DIALOG --title '$1'\
 			--backtitle '$backtitle' \
 			--menu '$menu_instructions' \
 			$ROWS $COLS $((ROWS-10)) \
@@ -261,7 +261,7 @@
 # Add a submenu option to the menu currently under construction.
 #
 function submenu () {
-	/bin/echo -ne "'activate_menu $2' '$1  --->' " >>MCmenu
+	echo -ne "'activate_menu $2' '$1  --->' " >>MCmenu
 }
 
 #
@@ -280,7 +280,7 @@
 		*)	eval $1=n ;;
 		esac
 	else
-		/bin/echo -ne "\007"
+		echo -ne "\007"
 	fi
 }
 
@@ -294,9 +294,9 @@
         *)   flag=' ' ;;
         esac
 
-        /bin/echo -ne "'$2' '<$flag> $1' " >>MCmenu
+        echo -ne "'$2' '<$flag> $1' " >>MCmenu
 
-        /bin/echo -e "function $2 () { l_disabled '$2' \"\$1\" ;}" >>MCradiolists
+        echo -e "function $2 () { l_disabled '$2' \"\$1\" ;}" >>MCradiolists
 }
 
 #
@@ -307,7 +307,7 @@
         then
                 case "$2" in
                 n)      ;;
-                *)      /bin/echo -en "\007"
+                *)      echo -en "\007"
                         ${DIALOG} --backtitle "$backtitle" \
                                   --msgbox "\
 This feature depends on another which has been disabled.  \
@@ -315,7 +315,7 @@
                         ;;
                 esac
         else
-                /bin/echo -ne "\007"
+                echo -ne "\007"
         fi
 }
 
@@ -338,7 +338,7 @@
 				eval $2="$answer"
 			else
 				eval $2="$3"
-				/bin/echo -en "\007"
+				echo -en "\007"
 				${DIALOG} --backtitle "$backtitle" \
 					--msgbox "You have made an invalid entry." 5 43
 			fi
@@ -370,7 +370,7 @@
 				eval $2="$answer"
 			else
 				eval $2="$3"
-				/bin/echo -en "\007"
+				echo -en "\007"
 				${DIALOG} --backtitle "$backtitle" \
 					--msgbox "You have made an invalid entry." 3 43
 			fi
@@ -714,7 +714,7 @@
 				eval load_config_file "$ALT_CONFIG"
 				break
 			else
-				/bin/echo -ne "\007"
+				echo -ne "\007"
 				$DIALOG	--backtitle "$backtitle" \
 					--msgbox "File does not exist!"  3 38
 			fi
@@ -777,7 +777,7 @@
 				load_functions  ## RELOAD
 				break
 			else
-				/bin/echo -ne "\007"
+				echo -ne "\007"
 				$DIALOG	--backtitle "$backtitle" \
 					--msgbox "Can't create file!  Probably a nonexistent directory." 3 60
 			fi
@@ -1009,7 +1009,7 @@
 
 	if [ $ROWS -lt 19 -o $COLS -lt 80 ]
 	then
-		/bin/echo -e "\n\007Your display is too small to run Menuconfig!"
+		echo -e "\n\007Your display is too small to run Menuconfig!"
 		echo "It must be at least 19 lines by 80 columns."
 		exit 0
 	fi 
@@ -1125,7 +1125,7 @@
 EOM
 else
 	clear
-	/bin/echo -e "Your configuration changes were NOT saved.\n"
+	echo -e "Your configuration changes were NOT saved.\n"
 fi
 
 
