diff -Naur orig/isdnlog/tools/dest/makedest patched/isdnlog/tools/dest/makedest
--- orig/isdnlog/tools/dest/makedest	2017-10-13 22:50:40.134739579 +0200
+++ patched/isdnlog/tools/dest/makedest	2017-10-13 23:13:48.792092850 +0200
@@ -30,6 +30,7 @@
 package main;
 # perl 5.8.0 could use UTF-8 as default encoding, which has to be prevented.
 eval q( use open ':encoding(iso-8859-1)' );
+use lib '.';
 use wld;
 BEGIN {
   if ( (! exists $ENV{DBEXT} or $ENV{DBEXT} eq ".cdb") and
