diff -Naur orig/example.jwmrc patched/example.jwmrc
--- orig/example.jwmrc	2016-03-27 00:40:53.000000000 +0100
+++ patched/example.jwmrc	2016-07-07 02:26:20.695591940 +0200
@@ -3,18 +3,20 @@
 
     <!-- The root menu. -->
     <RootMenu onroot="12">
-        <Program icon="terminal.png" label="Terminal">xterm</Program>
-        <Menu icon="folder.png" label="Applications">
+    	<Include>$HOME/.jwmrc-mjwm</Include>
+        <Program icon="xterm-color_48x48.xpm" label="Terminal">xterm</Program>
+        <Menu icon="folder.png" label="Alkalmazások">
             <Program icon="music.png" label="Audacious">audacious</Program>
-            <Program icon="calculator.png" label="Calculator">xcalc</Program>
+            <Program icon="calculator.png" label="Számológép">xcalc</Program>
             <Program icon="gimp.png" label="Gimp">gimp</Program>
             <Program icon="chat.png" label="Pidgin">pidgin</Program>
             <Program icon="www.png" label="Firefox">firefox</Program>
-            <Program icon="editor.png" label="XEdit">xedit</Program>
+            <Program icon="MidnightCommander.png" label="mc">xterm -e mc</Program>
         </Menu>
-        <Menu icon="folder.png" label="Utilities">
-            <Program icon="font.png" label="Fonts">xfontsel</Program>
-            <Program icon="window.png" label="Window Properties">
+        <Menu icon="folder.png" label="JWM eszközök">
+            <Program icon="uhu-64x64.png" label="Menügenerátor">jwmuhumenu</Program>
+            <Program icon="fonts.png" label="Betűkészletek">xfontsel</Program>
+            <Program icon="preferences-system-windows.png" label="Window Properties">
                 xprop | xmessage -file -
             </Program>
             <Program icon="window.png" label="Window Information">
@@ -26,8 +28,10 @@
             xlock -mode blank
         </Program>
         <Separator/>
-        <Restart label="Restart" icon="restart.png"/>
-        <Exit label="Exit" confirm="true" icon="quit.png"/>
+        <Restart label="JWM restart" icon="reload.png"/>
+        <Program icon="system-shutdown.png" label="Kikapcsolás">systemctl poweroff</Program>
+        <Program icon="redo.png" label="Újraindítás">systemctl reboot</Program>
+        <Exit label="Kilépés" confirm="true" icon="system-log-out.png"/>
     </RootMenu>
 
     <!-- Options for program groups. -->
@@ -123,6 +127,11 @@
     <IconPath>
         /usr/local/share/jwm
     </IconPath>
+    <IconPath>/usr/share/pixmaps</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/actions</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/places</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/apps</IconPath>
+    <IconPath>/usr/share/icons/gnome/48x48/devices</IconPath>
 
     <!-- Virtual Desktops -->
     <!-- Desktop tags can be contained within Desktops for desktop names. -->
@@ -175,4 +184,9 @@
     <Key mask="A" key="Up">udesktop</Key>
     <Key mask="A" key="Down">ddesktop</Key>
 
+    <!-- AUTOSTART -->
+    <StartupCommand>nm-applet</StartupCommand>
+    <StartupCommand>retrovol -hide</StartupCommand>
+    <StartupCommand>synclient MaxTapTime=0</StartupCommand>
+
 </JWM>
