diff -Naur orig/settings-dialogs/xfce-workspaces-settings.desktop.in patched/settings-dialogs/xfce-workspaces-settings.desktop.in
--- orig/settings-dialogs/xfce-workspaces-settings.desktop.in	2020-11-01 15:35:52.000000000 +0100
+++ patched/settings-dialogs/xfce-workspaces-settings.desktop.in	2020-12-01 03:38:57.590232852 +0100
@@ -2,7 +2,7 @@
 Version=1.0
 _Name=Workspaces
 _Comment=Configure layout, names and margins
-_Keywords=workspaces;settings;preferences;virtual desktops;number;windows;screen;margins;
+Keywords=workspaces;settings;preferences;virtual desktops;number;windows;screen;margins;
 Exec=xfwm4-workspace-settings
 Icon=org.xfce.workspaces
 Terminal=false
