diff -Naur a/help/Makefile.am b/help/Makefile.am
--- a/help/Makefile.am	2021-08-04 14:19:50.000000000 +0200
+++ b/help/Makefile.am	2022-04-11 10:31:33.915544039 +0200
@@ -30,7 +30,7 @@
              variables.page
 
 # Add linguas to be ignored, e.g. IGNORE_HELP_LINGUAS = ca de es fr
-IGNORE_HELP_LINGUAS =
+IGNORE_HELP_LINGUAS = ja
 HELP_LINGUAS = $(if $(IGNORE_HELP_LINGUAS), \
 	$(filter-out $(IGNORE_HELP_LINGUAS),$(subst /,,$(dir $(wildcard */*.po)))), \
 	$(subst /,,$(dir $(wildcard */*.po))) )
diff -Naur a/help/Makefile.in b/help/Makefile.in
--- a/help/Makefile.in	2021-08-04 14:21:28.000000000 +0200
+++ b/help/Makefile.in	2022-04-11 10:31:45.167558444 +0200
@@ -287,7 +287,7 @@
 
 
 # Add linguas to be ignored, e.g. IGNORE_HELP_LINGUAS = ca de es fr
-IGNORE_HELP_LINGUAS = 
+IGNORE_HELP_LINGUAS = ja
 HELP_LINGUAS = $(if $(IGNORE_HELP_LINGUAS), \
 	$(filter-out $(IGNORE_HELP_LINGUAS),$(subst /,,$(dir $(wildcard */*.po)))), \
 	$(subst /,,$(dir $(wildcard */*.po))) )
