diff -Nur orig/setup.py mod/setup.py
--- orig/setup.py	2020-04-28 08:29:15.000000000 +0200
+++ mod/setup.py	2021-12-14 20:17:17.074168680 +0100
@@ -62,7 +62,7 @@
         'dephell-setuptools>=0.2.1', 'dephell-shells>=0.1.3',
         'dephell-specifier>=0.1.7', 'dephell-venvs>=0.1.16',
         'dephell-versioning', 'jinja2', 'm2r', 'packaging',
-        'pip<=19.3.1,>=18.0', 'requests', 'ruamel.yaml', 'tomlkit', 'yaspin'
+        'pip', 'requests', 'ruamel.yaml', 'tomlkit', 'yaspin'
     ],
     extras_require={
         "dev": [
