diff -Naur orig/config.h.in patched/config.h.in
--- orig/config.h.in	2016-06-19 22:12:54.000000000 +0200
+++ patched/config.h.in	2016-10-15 01:07:43.713190764 +0200
@@ -192,7 +192,7 @@
  * If screen is installed with permissions to update /etc/utmp (such
  * as if it is installed set-uid root), define UTMPOK.
  */
-#define UTMPOK
+#undef UTMPOK
 
 /* Set LOGINDEFAULT to one (1)
  * if you want entries added to /etc/utmp by default, else set it to
diff -Naur orig/etc/screenrc patched/etc/screenrc
--- orig/etc/screenrc	2016-06-19 21:41:03.000000000 +0200
+++ patched/etc/screenrc	2016-10-15 01:09:20.158221028 +0200
@@ -119,8 +119,6 @@
 bind h
 #make them better
 bind 'K' kill
-bind 'I' login on
-bind 'O' login off
 bind '}' history
 
 # Yet another hack:
diff -Naur orig/Makefile.in patched/Makefile.in
--- orig/Makefile.in	2016-06-19 21:41:03.000000000 +0200
+++ patched/Makefile.in	2016-10-15 01:08:42.157820373 +0200
@@ -61,13 +61,13 @@
 SHELL=/bin/sh
 
 CFILES=	screen.c ansi.c fileio.c mark.c misc.c resize.c socket.c \
-	search.c tty.c term.c window.c utmp.c loadav.c putenv.c help.c \
+	search.c tty.c term.c window.c loadav.c putenv.c help.c \
 	termcap.c input.c attacher.c pty.c process.c display.c comm.c \
 	kmapdef.c acls.c braille.c braille_tsi.c logfile.c layer.c \
 	sched.c teln.c nethack.c encoding.c canvas.c layout.c viewport.c \
 	list_display.c list_generic.c list_window.c
 OFILES=	screen.o ansi.o fileio.o mark.o misc.o resize.o socket.o \
-	search.o tty.o term.o window.o utmp.o loadav.o putenv.o help.o \
+	search.o tty.o term.o window.o loadav.o putenv.o help.o \
 	termcap.o input.o attacher.o pty.o process.o display.o comm.o \
 	kmapdef.o acls.o braille.o braille_tsi.o logfile.o layer.o \
 	list_generic.o list_display.o list_window.o \
