diff -Naur a/Makefile.in b/Makefile.in
--- a/Makefile.in	2024-08-28 21:55:03.000000000 +0200
+++ b/Makefile.in	2024-09-13 14:39:30.645683830 +0200
@@ -20,7 +20,7 @@
 bindir  = $(exec_prefix)/bin
 
 VERSION = @PACKAGE_VERSION@
-SCREEN = screen-$(VERSION)
+SCREEN = screen
 
 DATE_FMT = %Y-%m-%d %H:%m:%S
 ifdef SOURCE_DATE_EPOCH
@@ -80,12 +80,9 @@
 	-if [ -f $(DESTDIR)$(bindir)/$(SCREEN) ] && [ ! -f $(DESTDIR)$(bindir)/$(SCREEN).old ]; \
 		then mv $(DESTDIR)$(bindir)/$(SCREEN) $(DESTDIR)$(bindir)/$(SCREEN).old; fi
 	$(INSTALL_PROGRAM) screen $(DESTDIR)$(bindir)/$(SCREEN)
-	-chown root $(DESTDIR)$(bindir)/$(SCREEN) && chmod 4755 $(DESTDIR)$(bindir)/$(SCREEN)
-# This doesn't work if $(bindir)/screen is a symlink
-	-if [ -f $(DESTDIR)$(bindir)/screen ] && [ ! -f $(DESTDIR)$(bindir)/screen.old ]; then mv $(DESTDIR)$(bindir)/screen $(DESTDIR)$(bindir)/screen.old; fi
-	rm -f $(DESTDIR)$(bindir)/screen
-	(cd $(DESTDIR)$(bindir) && ln -f -s $(SCREEN) screen)
+	-chown root $(DESTDIR)$(bindir)/$(SCREEN) && chmod 0755 $(DESTDIR)$(bindir)/$(SCREEN)
 	cp $(srcdir)/utf8encodings/?? $(DESTDIR)$(SCREENENCODINGS)
+	$(INSTALL_DATA) etc/screenrc $(DESTDIR)$(ETCSCREENRC)
 
 ###############################################################################
 install: installdirs install_bin
