diff -Naur orig/unix/tkConfig.sh.in patched/unix/tkConfig.sh.in
--- orig/unix/tkConfig.sh.in	2017-09-06 13:52:06.000000000 +0200
+++ patched/unix/tkConfig.sh.in	2019-11-26 11:51:06.936232174 +0100
@@ -69,7 +69,7 @@
 # different place than the directory containing the source files, this
 # points to the location of the sources, not the location where Tk was
 # compiled.
-TK_SRC_DIR='@TK_SRC_DIR@'
+TK_SRC_DIR='/usr/include'
 
 # Needed if you want to make a 'fat' shared library library
 # containing tk objects or link a different wish.
