diff -Naur orig/src/CMakeLists.txt mod/src/CMakeLists.txt
--- orig/src/CMakeLists.txt	2015-08-22 22:07:58.000000000 +0200
+++ mod/src/CMakeLists.txt	2015-08-22 22:18:41.799380485 +0200
@@ -119,7 +119,7 @@
 add_definitions(
   -DPACKAGE_DATA_DIR="${CMAKE_INSTALL_FULL_DATAROOTDIR}/obconf-qt"
   -DPIXMAPS_DIR="${CMAKE_INSTALL_FULL_DATAROOTDIR}/pixmaps"
-  -DTHEME_DIR="${CMAKE_INSTALL_FULL_DATAROOTDIR}/openbox/themes"
+  -DTHEME_DIR="${CMAKE_INSTALL_FULL_DATAROOTDIR}/themes"
 )
 
 target_link_libraries(obconf-qt
