diff -Naur a/ncat/http_digest.c b/ncat/http_digest.c
--- a/ncat/http_digest.c	2022-08-25 16:51:59.000000000 +0200
+++ b/ncat/http_digest.c	2022-09-03 20:49:06.166857887 +0200
@@ -133,10 +133,6 @@
     return 0;
 }
 
-#if OPENSSL_API_LEVEL < 10100
-#define EVP_MD_CTX_new EVP_MD_CTX_create
-#define EVP_MD_CTX_free EVP_MD_CTX_destroy
-#endif
 static char *make_nonce(const struct timeval *tv)
 {
     char *buf = NULL;
