diff -Naur orig/sialantools/qtsingleapplication/qtlocalpeer.cpp mod/sialantools/qtsingleapplication/qtlocalpeer.cpp
--- orig/sialantools/qtsingleapplication/qtlocalpeer.cpp	2016-04-25 20:36:20.000000000 +0200
+++ mod/sialantools/qtsingleapplication/qtlocalpeer.cpp	2016-04-25 20:54:20.553548187 +0200
@@ -48,6 +48,7 @@
 #include "qtlocalpeer.h"
 #include <QtCore/QCoreApplication>
 #include <QtCore/QTime>
+#include <QtCore/QDataStream>
 
 #if defined(Q_OS_WIN)
 #include <QtCore/QLibrary>
