diff -Naur libxkbui-0.99.2.orig/xkbui.pc.in libxkbui-0.99.2/xkbui.pc.in
--- libxkbui-0.99.2.orig/xkbui.pc.in	2005-11-19 08:15:48.000000000 +0100
+++ libxkbui-0.99.2/xkbui.pc.in	2005-12-05 15:35:28.000000000 +0100
@@ -7,6 +7,6 @@
 Description: The xkbui Library
 Version: @PACKAGE_VERSION@
 Requires: x11 kbproto
-Requires.private: x11 xt
+Requires.private: x11 xt xkbfile
 Cflags: -I${includedir}
 Libs: -L${libdir} -lxkbui
