diff -Naur upx-3.91-src-orig/src/Makefile upx-3.91-src/src/Makefile
--- upx-3.91-src-orig/src/Makefile	2013-09-30 17:51:00.000000000 +0200
+++ upx-3.91-src/src/Makefile	2016-11-12 08:58:49.175830478 +0100
@@ -38,9 +38,9 @@
 else
 CXXFLAGS += -O2
 endif
-CXXFLAGS_WERROR = -Werror
-CXXFLAGS += -Wall -W -Wcast-align -Wcast-qual -Wpointer-arith -Wshadow -Wwrite-strings
-CXXFLAGS += $(CXXFLAGS_WERROR)
+# CXXFLAGS_WERROR = \
+CXXFLAGS += -Wall -W -Wcast-align -Wcast-qual -Wpointer-arith -Wshadow -Wwrite-strings -Werror
+# CXXFLAGS += $(CXXFLAGS_WERROR)
 endif
 CPPFLAGS += $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES)
 CXXLD ?= $(CXX)
