diff -Naur a/doc/meson.build b/doc/meson.build
--- a/doc/meson.build	2022-03-09 18:46:34.000000000 +0100
+++ b/doc/meson.build	2022-03-26 16:42:44.012720179 +0100
@@ -26,22 +26,3 @@
     output: 'version.xml',
     input: 'version.xml.in',
     configuration: cdata)
-
-gnome.gtkdoc('UPower',
-    main_xml: 'upower-docs.xml',
-    src_dir: meson.source_root() / 'libupower-glib',
-    dependencies: [ libupower_glib_dep ],
-    scan_args: ['--rebuild-types', '--rebuild-sections'],
-    content_files: [
-        version_xml,
-        ifaces_refs,
-        'man/UPower.xml',
-        'man/upowerd.xml',
-        'man/upower.xml',
-        '../COPYING',
-    ],
-    ignore_headers: [
-        'config.h',
-    ],
-    install: get_option('gtk-doc')
-)
