diff -Nur orig/lxqt-admin-time/translations/lxqt-admin-time_cs.desktop mod/lxqt-admin-time/translations/lxqt-admin-time_cs.desktop
--- orig/lxqt-admin-time/translations/lxqt-admin-time_cs.desktop	2019-01-24 22:53:08.000000000 +0100
+++ mod/lxqt-admin-time/translations/lxqt-admin-time_cs.desktop	2019-01-26 19:11:40.239439518 +0100
@@ -1,4 +1,4 @@
 #TRANSLATIONS
 Name[cs]=Datum a čas
 GenericName[cs]=Nastavení data a času
-Comment [cs]=Nastavení systémového data a času
+Comment[cs]=Nastavení systémového data a času
diff -Nur orig/lxqt-admin-user/translations/lxqt-admin-user_cs.desktop mod/lxqt-admin-user/translations/lxqt-admin-user_cs.desktop
--- orig/lxqt-admin-user/translations/lxqt-admin-user_cs.desktop	2019-01-24 22:53:08.000000000 +0100
+++ mod/lxqt-admin-user/translations/lxqt-admin-user_cs.desktop	2019-01-26 19:23:37.302306243 +0100
@@ -1,4 +1,4 @@
 #TRANSLATIONS
-Name [cs]=Uživatelé a skupiny
+Name[cs]=Uživatelé a skupiny
 GenericName[cs]=Nastavení uživatelů a skupin
 Comment[cs]=Nastavení uživatelů a skupin systému
