diff -Naur a/lxqt-admin-time/lxqt-admin-time.desktop.in b/lxqt-admin-time/lxqt-admin-time.desktop.in
--- a/lxqt-admin-time/lxqt-admin-time.desktop.in	2022-04-16 15:46:51.000000000 +0200
+++ b/lxqt-admin-time/lxqt-admin-time.desktop.in	2022-06-16 01:30:39.781180034 +0200
@@ -2,7 +2,7 @@
 Type=Application
 Exec=lxqt-admin-time
 Icon=preferences-system-time
-Categories=LXQt;Settings;DesktopSettings;Qt;
+Categories=Settings;DesktopSettings;Qt;
 OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
diff -Naur a/lxqt-admin-user/lxqt-admin-user.desktop.in b/lxqt-admin-user/lxqt-admin-user.desktop.in
--- a/lxqt-admin-user/lxqt-admin-user.desktop.in	2022-04-16 15:46:51.000000000 +0200
+++ b/lxqt-admin-user/lxqt-admin-user.desktop.in	2022-06-16 01:30:54.437235625 +0200
@@ -2,7 +2,7 @@
 Type=Application
 Exec=lxqt-admin-user
 Icon=system-users
-Categories=LXQt;Settings;DesktopSettings;Qt;
+Categories=Settings;DesktopSettings;Qt;
 OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
