diff -Naur orig/lxqt-admin-time/translations/lxqt-admin-time_zh_TW.desktop patched/lxqt-admin-time/translations/lxqt-admin-time_zh_TW.desktop
--- orig/lxqt-admin-time/translations/lxqt-admin-time_zh_TW.desktop	2018-05-21 20:16:37.000000000 +0200
+++ patched/lxqt-admin-time/translations/lxqt-admin-time_zh_TW.desktop	2018-06-11 18:03:09.377233024 +0200
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Name[zh_TW]=日期與時間
-GenericName[ja]=日期與時間設定
-Comment[ja]=設定系統的日期與時間
+GenericName[zh_TW]=日期與時間設定
+Comment[zh_TW]=設定系統的日期與時間
 
 #TRANSLATIONS_DIR=translations
