diff -Naur orig/open-terminal/caja-open-terminal.c patched/open-terminal/caja-open-terminal.c
--- orig/open-terminal/caja-open-terminal.c	2013-10-31 01:09:56.000000000 +0100
+++ patched/open-terminal/caja-open-terminal.c	2014-02-11 16:33:56.260900312 +0100
@@ -30,9 +30,7 @@
 #include <libcaja-extension/caja-menu-provider.h>
 
 #include <glib/gi18n-lib.h>
-#include <gtk/gtkicontheme.h>
-#include <gtk/gtkwidget.h>
-#include <gtk/gtkmain.h>
+#include <gtk/gtk.h>
 
 #include <libmate-desktop/mate-desktop-item.h>
 #include <gio/gio.h>
