diff -Naur a/menu/lxqt-applications.menu b/menu/lxqt-applications.menu
--- a/menu/lxqt-applications.menu	2023-11-10 09:46:37.000000000 +0100
+++ b/menu/lxqt-applications.menu	2023-11-11 17:40:22.120821752 +0100
@@ -28,8 +28,82 @@
 				<Not><Category>System</Category></Not>
 			</And>
 		</Include>
+		<Exclude>
+		<Category>Archiving</Category>
+        	<Category>Calculator</Category>
+        	<Category>Clock</Category>
+        	<Category>Compression</Category>
+        	<Category>FileTools</Category>
+        	<Category>TelephonyTools</Category>
+        	<Category>TextEditor</Category>
+        	<Category>TextTools</Category>
+        	<Category>FileManager</Category>
+        	<Category>GNOME</Category>
+        	<Category>KDE</Category>
+		</Exclude>
+    	<Menu>
+      		<Name>Archiving</Name>
+      		<Directory>lxde-utility-sub-archiving.directory</Directory>
+      		<Include>
+        		<Category>Archiving</Category>
+      		</Include>
+    	</Menu>    
+	    <Menu>
+			<Name>Calculator</Name>
+			<Directory>lxde-utility-sub-calculator.directory</Directory>
+			<Include>
+				<Category>Calculator</Category>
+			</Include>
+		</Menu>   
+		<Menu>
+			<Name>Clock</Name>
+			<Directory>lxde-utility-sub-clock.directory</Directory>
+			<Include>
+				<Category>Clock</Category>
+			</Include>
+		</Menu>   
+		<Menu>
+			<Name>Compression</Name>
+			<Directory>lxde-utility-sub-compression.directory</Directory>
+			<Include>
+			  <And>
+				<Category>Compression</Category>
+				<Not><Category>Archiving</Category></Not>
+			      </And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>File tools</Name>
+			<Directory>lxde-utility-sub-filetool.directory</Directory>
+			<Include>
+			  <And>
+			    <Not><Category>System</Category></Not> 
+			  <Category>FileTools</Category>
+			</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Telephony tools</Name>
+			<Directory>lxde-utility-sub-telephonytool.directory</Directory>
+			<Include>
+				<Category>TelephonyTools</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Text editor</Name>
+			<Directory>lxde-utility-sub-texteditor.directory</Directory>
+			<Include>
+				<Category>TextEditor</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Text tools</Name>
+			<Directory>lxde-utility-sub-texttool.directory</Directory>
+			<Include>
+				<Category>TextTools</Category>
+			</Include>
+		</Menu>
 	</Menu> <!-- End Accessories -->
-
 	<!-- Accessibility submenu -->
 	<Menu>
 		<Name>Universal Access</Name>
@@ -75,6 +149,113 @@
 				<Category>Game</Category>
 			</And>
 		</Include>
+		<Exclude>
+			<Category>ActionGame</Category>
+			<Category>AdventureGame</Category>
+			<Category>ArcadeGame</Category>
+			<Category>BoardGame</Category>
+			<Category>BlocksGame</Category>
+			<Category>CardGame</Category>
+			<Category>KidsGame</Category>
+			<Category>LogicGame</Category>
+			<Category>RolePlaying</Category>
+			<Category>Simulation</Category>
+			<Category>SportsGame</Category>
+			<Category>StrategyGame</Category>
+			<Category>Emulator</Category>
+		</Exclude>
+		<Menu>
+			<Name>Action</Name>
+			<Directory>lxde-game-sub-action.directory</Directory>
+			<Include>
+				<Category>ActionGame</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Adventure</Name>
+			<Directory>lxde-game-sub-adventure.directory</Directory>
+			<Include>
+				<Category>AdventureGame</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Arcade</Name>
+			<Directory>lxde-game-sub-arcade.directory</Directory>
+			<Include>
+				<Category>ArcadeGame</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Board</Name>
+			<Directory>lxde-game-sub-board.directory</Directory>
+			<Include>
+				<Category>BoardGame</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Blocks</Name>
+			<Directory>lxde-game-sub-blocks.directory</Directory>
+			<Include>
+				<Category>BlocksGame</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Cards</Name>
+			<Directory>lxde-game-sub-card.directory</Directory>
+			<Include>
+				<Category>CardGame</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Kids</Name>
+			<Directory>lxde-game-sub-kids.directory</Directory>
+			<Include>
+				<Category>KidsGame</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Logic</Name>
+			<Directory>lxde-game-sub-logic.directory</Directory>
+			<Include>
+				<Category>LogicGame</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Role Playing</Name>
+			<Directory>lxde-game-sub-roleplaying.directory</Directory>
+			<Include>
+				<Category>RolePlaying</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Simulation</Name>
+			<Directory>lxde-game-sub-simulation.directory</Directory>
+			<Include>
+				<Category>Simulation</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Sports</Name>
+			<Directory>lxde-game-sub-sports.directory</Directory>
+			<Include>
+				<Category>SportsGame</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Strategy</Name>
+			<Directory>lxde-game-sub-strategy.directory</Directory>
+			<Include>
+				<Category>StrategyGame</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Emulator</Name>
+			<Directory>lxde-system-tools-sub-emulator.directory</Directory>
+			<Include>
+				<Category>Emulator</Category>
+			</Include>
+		</Menu>    
+
 	</Menu> <!-- End Games -->
 
 	<!-- Graphics -->
@@ -98,6 +279,120 @@
 				<Category>Network</Category>
 			</And>
 		</Include>
+		<Exclude>
+			<Category>Chat</Category>
+			<Category>Dialup</Category>
+			<Category>Email</Category>
+			<Category>Feed</Category>
+			<Category>FileTransfer</Category>
+			<Category>HamRadio</Category>
+			<Category>InstantMessaging</Category>
+			<Category>IRCClient</Category>
+			<Category>News</Category>
+			<Category>P2P</Category>
+			<Category>RemoteAccess</Category>
+			<Category>Telephony</Category>
+			<Category>VideoConference</Category>
+			<Category>WebBrowser</Category>
+		</Exclude>
+		<Menu>
+			<Name>Chat</Name>
+			<Directory>lxde-network-sub-chat.directory</Directory>
+			<Include>
+				<Category>Chat</Category>
+			</Include>
+		</Menu>    
+		<Menu>
+			<Name>Dial-up</Name>
+			<Directory>lxde-network-sub-dialup.directory</Directory>
+			<Include>
+				<Category>Dialup</Category>
+			</Include>
+		</Menu>   
+		<Menu>
+			<Name>Email</Name>
+			<Directory>lxde-network-sub-email.directory</Directory>
+			<Include>
+				<Category>Email</Category>
+			</Include>
+		</Menu>  
+		<Menu>
+			<Name>Feed</Name>
+			<Directory>lxde-network-sub-feed.directory</Directory>
+			<Include>
+				<Category>Feed</Category>
+			</Include>
+		</Menu>   
+		<Menu>
+			<Name>File transfer</Name>
+			<Directory>lxde-network-sub-filetransfer.directory</Directory>
+			<Include>
+				<Category>FileTransfer</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Ham radio</Name>
+			<Directory>lxde-network-sub-hamradio.directory</Directory>
+			<Include>
+				<Category>HamRadio</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Instant messaging</Name>
+			<Directory>lxde-network-sub-instantmessaging.directory</Directory>
+			<Include>
+				<Category>InstantMessaging</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>IRC client</Name>
+			<Directory>lxde-network-sub-ircclient.directory</Directory>
+			<Include>
+				<Category>IRCClient</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>News</Name>
+			<Directory>lxde-network-sub-news.directory</Directory>
+			<Include>
+				<Category>News</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>P2P</Name>
+			<Directory>lxde-network-sub-p2p.directory</Directory>
+			<Include>
+				<Category>P2P</Category>
+			</Include>
+		</Menu>    
+		<Menu>
+			<Name>Remote access</Name>
+			<Directory>lxde-network-sub-remoteaccess.directory</Directory>
+			<Include>
+				<Category>RemoteAccess</Category>
+			</Include>
+		</Menu>   
+		<Menu>
+			<Name>Telephony</Name>
+			<Directory>lxde-network-sub-telephony.directory</Directory>
+			<Include>
+				<Category>telephony</Category>
+			</Include>
+		</Menu>   
+		<Menu>
+			<Name>Video conference</Name>
+			<Directory>lxde-network-sub-videoconference.directory</Directory>
+			<Include>
+				<Category>VideoConference</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Web browser</Name>
+			<Directory>lxde-network-sub-webbrowser.directory</Directory>
+			<Include>
+				<Category>WebBrowser</Category>
+			</Include>
+		</Menu>
 	</Menu>	 <!-- End Internet -->
 
 	<!-- LXQt-About -->
@@ -117,6 +412,88 @@
 				<Category>AudioVideo</Category>
 			</And>
 		</Include>
+		<Exclude>
+			<Category>AudioVideoEditing</Category>
+			<Category>DiscBurning</Category>
+			<Category>Midi</Category>
+			<Category>Mixer</Category>
+			<Category>Music</Category>
+			<Category>Player</Category>
+			<Category>Recorder</Category>
+			<Category>Sequencer</Category>
+			<Category>Tuner</Category>
+			<Category>TV</Category>
+		</Exclude>
+		<Menu>
+			<Name>Audio Video editing</Name>
+			<Directory>lxde-audio-video-sub-audiovideoediting.directory</Directory>
+			<Include>
+				<Category>AudioVideoEditing</Category>
+			</Include>
+		</Menu>    
+		<Menu>
+			<Name>Disc burning</Name>
+			<Directory>lxde-audio-video-sub-discburning.directory</Directory>
+			<Include>
+				<Category>DiscBurning</Category>
+			</Include>
+		</Menu>   
+		<Menu>
+			<Name>Midi</Name>
+			<Directory>lxde-audio-video-sub-midi.directory</Directory>
+			<Include>
+				<Category>Midi</Category>
+			</Include>
+		</Menu>   
+		<Menu>
+			<Name>Mixer</Name>
+			<Directory>lxde-audio-video-sub-mixer.directory</Directory>
+			<Include>
+				<Category>Mixer</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Music</Name>
+			<Directory>lxde-audio-video-sub-music.directory</Directory>
+			<Include>
+				<Category>Music</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Player</Name>
+			<Directory>lxde-audio-video-sub-player.directory</Directory>
+			<Include>
+				<Category>Player</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Recorder</Name>
+			<Directory>lxde-audio-video-sub-recorder.directory</Directory>
+			<Include>
+				<Category>Recorder</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Sequencer</Name>
+			<Directory>lxde-audio-video-sequencer.directory</Directory>
+			<Include>
+				<Category>Sequencer</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Tuner</Name>
+			<Directory>lxde-audio-video-sub-tuner.directory</Directory>
+			<Include>
+				<Category>Tuner</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>TV</Name>
+			<Directory>lxde-audio-video-sub-tv.directory</Directory>
+			<Include>
+				<Category>TV</Category>
+			</Include>
+		</Menu>
 	</Menu>	 <!-- End Multimedia -->
 
 	<!-- Office -->
@@ -144,6 +521,53 @@
 				<Not><Filename>lxqt-about.desktop</Filename></Not>
 			</And>
 		</Include>
+		<Exclude>
+			<Category>Emulator</Category>
+			<Category>FileManager</Category>
+			<Category>Filesystem</Category>
+			<Category>Monitor</Category>
+			<Category>TerminalEmulator</Category>
+			<Category>GNOME</Category>
+			<Category>KDE</Category>
+		</Exclude>
+		<Menu>
+			<Name>Emulator</Name>
+			<Directory>lxde-system-tools-sub-emulator.directory</Directory>
+			<Include>
+			 <And>
+			  <Category>Emulator</Category>
+				<Not><Category>Game</Category></Not>
+			      </And>
+			</Include>
+		</Menu>    
+		<Menu>
+			<Name>Filemanager</Name>
+			<Directory>lxde-system-tools-sub-filemanager.directory</Directory>
+			<Include>
+				<Category>FileManager</Category>
+			</Include>
+		</Menu>   
+		<Menu>
+			<Name>Filesystem</Name>
+			<Directory>lxde-system-tools-sub-filesystem.directory</Directory>
+			<Include>
+				<Category>Filesystem</Category>
+			</Include>
+		</Menu>   
+		<Menu>
+			<Name>Monitor</Name>
+			<Directory>lxde-system-tools-sub-monitor.directory</Directory>
+			<Include>
+				<Category>Monitor</Category>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Terminal</Name>
+			<Directory>lxde-system-tools-sub-terminalemulator.directory</Directory>
+			<Include>
+				<Category>TerminalEmulator</Category>
+			</Include>
+		</Menu>
 	</Menu>	 <!-- End System Tools -->
 
 	<!-- Other -->
@@ -193,6 +617,8 @@
 			</Or>
 		</Include>
 		<Exclude>
+		<Category>GNOME</Category>
+		<Category>KDE</Category>
 			<Or>
 				<Filename>lxqt-config.desktop</Filename>
 			</Or>
