diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2018-03-25 10:49:15.000000000 +0200
+++ patched/CMakeLists.txt	2020-01-12 21:04:33.507908424 +0100
@@ -371,6 +371,7 @@
 set(prefix ${CMAKE_INSTALL_PREFIX})
 set(exec_prefix ${CMAKE_INSTALL_PREFIX})
 set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
 set(LIB_JPEG ${JPEG_LIBRARY})
 set(LIB_TIFF ${TIFF_LIBRARY})
 set(LIB_ZLIB ${ZLIB_LIBRARY})
