diff -Naur libpano13-2.9.19-orig/CMakeLists.txt libpano13-2.9.19/CMakeLists.txt
--- libpano13-2.9.19-orig/CMakeLists.txt	2014-02-17 23:04:42.000000000 +0100
+++ libpano13-2.9.19/CMakeLists.txt	2016-07-23 19:28:01.251151878 +0200
@@ -369,6 +369,7 @@
 set(prefix ${CMAKE_INSTALL_PREFIX})
 set(exec_prefix ${CMAKE_INSTALL_PREFIX})
 set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
 set(LIB_JPEG ${JPEG_LIBRARY})
 set(LIB_TIFF ${TIFF_LIBRARY})
 set(LIB_ZLIB ${ZLIB_LIBRARY})
