diff -Nur orig/source/xdg-data/org.develz.Crawl_tiles.desktop mod/source/xdg-data/org.develz.Crawl_tiles.desktop
--- orig/source/xdg-data/org.develz.Crawl_tiles.desktop	2022-09-12 04:42:43.000000000 +0200
+++ mod/source/xdg-data/org.develz.Crawl_tiles.desktop	2023-01-29 17:59:58.973009100 +0100
@@ -4,4 +4,4 @@
 Comment=A fun game in the grand tradition of games like Rogue, Hack, and Moria
 Type=Application
 Terminal=false
-Categories=Game;AdventureGame;RolePlaying;
+Categories=Game;RolePlaying;
