diff -Nur orig/d2x-rebirth.desktop mod/d2x-rebirth.desktop
--- orig/d2x-rebirth.desktop	2013-04-22 12:35:52.000000000 +0200
+++ mod/d2x-rebirth.desktop	2018-03-16 13:22:24.311883805 +0100
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Name=Descent 2
 Comment=DXX-Rebirth source port of Descent 2: Counterstrike from 1996...
-Exec=d2x-rebirth
+Exec=d2x-rebirth -window
 Icon=d2x-rebirth
 Terminal=false
 Type=Application
