diff -Naur a/layout/mate-preferences-categories.menu b/layout/mate-preferences-categories.menu
--- a/layout/mate-preferences-categories.menu	2022-11-11 20:20:39.000000000 +0100
+++ b/layout/mate-preferences-categories.menu	2022-11-23 20:57:15.987107920 +0100
@@ -40,12 +40,6 @@
         <Not><Category>System</Category></Not>
         <Not><Category>Security</Category></Not>
       </And>
-      <And>
-        <Category>Settings</Category>
-        <Category>Qt</Category>
-        <Not><Category>System</Category></Not>
-        <Not><Category>Security</Category></Not>
-      </And>
     </Include>
   </Menu>
 
@@ -95,7 +89,6 @@
         <Not><Category>HardwareSettings</Category></Not>
         <Not><Category>Security</Category></Not>
         <Not><Category>Archiving</Category></Not>
-        <Not><Category>Qt</Category></Not>
         <Not><Category>X-MATE-NetworkSettings</Category></Not>
         <Not><Category>X-GNOME-NetworkSettings</Category></Not>
         <Not><Category>X-MATE-PersonalSettings</Category></Not>
diff -Naur a/layout/mate-settings.menu b/layout/mate-settings.menu
--- a/layout/mate-settings.menu	2022-11-11 20:20:39.000000000 +0100
+++ b/layout/mate-settings.menu	2022-11-23 20:54:10.890299908 +0100
@@ -28,8 +28,36 @@
         <Category>Settings</Category>
         <Not>
           <Or>
-            <Category>System</Category>
+			<Category>System</Category>
+			<!-- a Rendszer/Beállítások menüből kiszedjük mindazt, ami
+            a Vezérlőközpontban elérhető, nem kellenek duplán -->
+            <!-- mate control center -->
             <Filename>matecc.desktop</Filename>
+            <Filename>mate-about-me.desktop</Filename>
+			<Filename>mate-appearance-properties.desktop</Filename>
+			<Filename>mate-at-properties.desktop</Filename>
+			<Filename>mate-default-applications-properties.desktop</Filename>
+			<Filename>mate-display-properties.desktop</Filename>
+			<Filename>mate-font-viewer.desktop</Filename>
+			<Filename>mate-keybinding.desktop</Filename>
+			<Filename>mate-keyboard.desktop</Filename>
+			<Filename>mate-network-properties.desktop</Filename>
+			<Filename>mate-settings-mouse.desktop</Filename>
+			<Filename>mate-theme-installer.desktop</Filename>
+			<!-- caja -->
+			<Filename>caja-file-management-properties.desktop</Filename>
+			<!-- mate-notification-daemon -->
+			<Filename>mate-notification-properties.desktop</Filename>
+			<!-- mate-power-manager -->
+			<Filename>mate-power-preferences.desktop</Filename>
+			<!-- mate-screensaver -->
+			<Filename>mate-screensaver-preferences.desktop</Filename>
+			<!-- mate-session-manager -->
+			<Filename>mate-session-properties.desktop</Filename>
+			<!-- mate-user-share -->
+			<Filename>mate-user-share-properties.desktop</Filename>
+			<!-- mozo, főmenü szerkesztő -->
+			<Filename>mozo.desktop</Filename>
           </Or>
         </Not>
       </And>
