diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2022-03-18 14:15:32.794182720 +0100
+++ b/CMakeLists.txt	2022-03-18 14:17:44.386736635 +0100
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.16)
 
 project(calligra)
 
@@ -50,7 +50,7 @@
 #set(CALLIGRA_ALPHA 1) # uncomment only for Alpha
 # set(CALLIGRA_BETA 1) # uncomment only for Beta
 #set(CALLIGRA_RC 1) # uncomment only for RC
-set(CALLIGRA_YEAR 2020) # update every year
+set(CALLIGRA_YEAR 2021) # update every year
 
 if(NOT DEFINED CALLIGRA_ALPHA AND NOT DEFINED CALLIGRA_BETA AND NOT DEFINED CALLIGRA_RC)
     set(CALLIGRA_STABLE 1) # do not edit
@@ -104,13 +104,6 @@
 endif()
 message(STATUS "Release build: ${RELEASE_BUILD}")
 
-# use CPP-11
-if (CMAKE_VERSION VERSION_LESS "3.1")
-    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
-else ()
-    set (CMAKE_CXX_STANDARD 11)
-endif ()
-
 ############
 #############
 ## Options ##
@@ -168,8 +161,10 @@
 ## Look for ECM, Qt, KF5 ##
 ###########################
 ##########################
+set(REQUIRED_KF5_VERSION "5.76.0")
+set(REQUIRED_QT_VERSION "5.12.0")
 
-find_package(ECM 5.19 REQUIRED NO_MODULE)
+find_package(ECM ${REQUIRED_KF5_VERSION} REQUIRED NO_MODULE)
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR})
 
 # ECM KDE macros (include first, to have their policies and settings effect all other macros)
@@ -200,8 +195,6 @@
 include(MacroDesktopToJson)
 
 
-set(REQUIRED_KF5_VERSION "5.7.0")
-
 find_package(KF5 ${REQUIRED_KF5_VERSION} REQUIRED
     COMPONENTS
     Archive
@@ -252,8 +245,6 @@
 endif()
 
 
-set(REQUIRED_QT_VERSION "5.3.0")
-
 find_package(Qt5 ${REQUIRED_QT_VERSION} REQUIRED
     COMPONENTS
     Core
@@ -646,8 +637,8 @@
 	)
 endif()
 
-if(NOT FONTCONFIG_FOUND OR NOT FREETYPE_FOUND)
-    set(FONTCONFIG_INCLUDE_DIR "")
+if(NOT Fontconfig_FOUND OR NOT FREETYPE_FOUND)
+    set(Fontconfig_INCLUDE_DIRS "")
     set(FREETYPE_INCLUDE_DIRS "")
 else()
     add_definitions( -DSHOULD_BUILD_FONT_CONVERSION )
diff -Naur a/libs/text/CMakeLists.txt b/libs/text/CMakeLists.txt
--- a/libs/text/CMakeLists.txt	2020-05-14 08:51:31.000000000 +0200
+++ b/libs/text/CMakeLists.txt	2022-03-18 14:17:44.386736635 +0100
@@ -152,11 +152,11 @@
 endif()
 
 if( FONTCONFIG_FOUND )
-    target_link_libraries(kotext PRIVATE ${FONTCONFIG_LIBRARIES})
+    target_link_libraries(kotext PRIVATE Fontconfig::Fontconfig)
 endif()
 
 if( FREETYPE_FOUND )
-    target_link_libraries(kotext PRIVATE ${FREETYPE_LIBRARIES})
+    target_link_libraries(kotext PRIVATE Freetype::Freetype)
 endif()
 
 
