diff -Naur a/filters/karbon/pdf/SvgOutputDev.cpp b/filters/karbon/pdf/SvgOutputDev.cpp
--- a/filters/karbon/pdf/SvgOutputDev.cpp	2020-05-14 08:51:30.000000000 +0200
+++ b/filters/karbon/pdf/SvgOutputDev.cpp	2022-07-26 20:20:47.568918575 +0200
@@ -410,7 +410,7 @@
     if (s->getLength() == 0)
         return;
 
-    GfxFont * font = state->getFont();
+    std::shared_ptr<GfxFont> font = state->getFont();
 
     QString str;
 
