diff -Naur orig/stage/part/KPrPresentationTool.cpp patched/stage/part/KPrPresentationTool.cpp
--- orig/stage/part/KPrPresentationTool.cpp	2018-01-26 10:24:44.000000000 +0100
+++ patched/stage/part/KPrPresentationTool.cpp	2018-06-11 13:36:21.873577178 +0200
@@ -32,6 +32,7 @@
 #include <QDesktopServices>
 #include <QUrl>
 #include <QDBusConnection>
+#include <QFrame>
 
 #include <KoShape.h>
 #include <KoShapeManager.h>
