diff -Naur orig/sheets/plugins/calendar/CalendarToolWidget.h patched/sheets/plugins/calendar/CalendarToolWidget.h
--- orig/sheets/plugins/calendar/CalendarToolWidget.h	2018-01-26 10:24:44.000000000 +0100
+++ patched/sheets/plugins/calendar/CalendarToolWidget.h	2019-10-05 01:12:17.780973543 +0200
@@ -27,6 +27,7 @@
 #include <ui_CalendarToolWidget.h>
 
 #include <QDialog>
+#include <QDate>
 
 class KDatePicker;
 
