diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2021-09-22 23:23:41.000000000 +0200
+++ b/CMakeLists.txt	2022-08-13 17:05:38.696001931 +0200
@@ -57,7 +57,7 @@
     option(SYM_LINK "Create symlink leptonica -> lept on UNIX" OFF)
 endif()
 
-set(leptonica_INSTALL_CMAKE_DIR "lib/cmake/leptonica"
+set(leptonica_INSTALL_CMAKE_DIR "lib/cmake/Leptonica"
     CACHE STRING "Install destination for CMake package files")
 
 if(NOT SW_BUILD)
