diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2024-01-03 01:03:10.000000000 +0100
+++ b/CMakeLists.txt	2024-02-09 16:19:39.382329566 +0100
@@ -71,7 +71,7 @@
 option(ENABLE_OPENJPEG "Enable OpenJPEG support" ON)
 
 set(leptonica_INSTALL_CMAKE_DIR
-    "lib/cmake/leptonica"
+    "lib/cmake/Leptonica"
     CACHE STRING "Install destination for CMake package files")
 
 if(NOT SW_BUILD)
