diff -Naur a/etc/clamav-milter.conf.sample b/etc/clamav-milter.conf.sample
--- a/etc/clamav-milter.conf.sample	2023-08-27 00:00:24.000000000 +0200
+++ b/etc/clamav-milter.conf.sample	2023-09-04 00:24:53.985857224 +0200
@@ -3,7 +3,7 @@
 ##
 
 # Comment or remove the line below.
-Example
+#Example
 
 
 ##
@@ -38,7 +38,7 @@
 # to work)
 #
 # Default: unset (don't drop privileges)
-#User clamav
+User clamav
 
 # Waiting for data from clamd will timeout after this time (seconds).
 # Value of 0 disables the timeout.
@@ -70,7 +70,7 @@
 # Optional path to the global temporary directory.
 # Default: system specific (usually /tmp or /var/tmp).
 #
-#TemporaryDirectory /var/tmp
+TemporaryDirectory /tmp
 
 ##
 ## Clamd options
@@ -216,7 +216,7 @@
 # A full path is required.
 #
 # Default: disabled
-#LogFile /tmp/clamav-milter.log
+LogFile /var/log/clamav/clamav-milter.log
 
 # By default the log file is locked for writing - the lock protects against
 # running clamav-milter multiple times.
@@ -238,7 +238,7 @@
 # Log time with each message.
 #
 # Default: no
-#LogTime yes
+LogTime yes
 
 # Use system logger (can work together with LogFile).
 #
diff -Naur a/etc/clamd.conf.sample b/etc/clamd.conf.sample
--- a/etc/clamd.conf.sample	2023-08-27 00:00:24.000000000 +0200
+++ b/etc/clamd.conf.sample	2023-09-04 00:34:34.888045374 +0200
@@ -5,13 +5,13 @@
 
 
 # Comment or remove the line below.
-Example
+#Example
 
 # Uncomment this option to enable logging.
 # LogFile must be writable for the user running daemon.
 # A full path is required.
 # Default: disabled
-#LogFile /tmp/clamd.log
+LogFile /var/log/clamav/clamd.log
 
 # By default the log file is locked for writing - the lock protects against
 # running clamd multiple times (if want to run another clamd, please
@@ -32,7 +32,7 @@
 
 # Log time with each message.
 # Default: no
-#LogTime yes
+LogTime yes
 
 # Also log clean files. Useful in debugging but drastically increases the
 # log size.
@@ -78,7 +78,7 @@
 
 # Optional path to the global temporary directory.
 # Default: system specific (usually /tmp or /var/tmp).
-#TemporaryDirectory /var/tmp
+TemporaryDirectory /tmp
 
 # Path to the database directory.
 # Default: hardcoded (depends on installation options)
@@ -98,8 +98,7 @@
 
 # Path to a local socket file the daemon will listen on.
 # Default: disabled (must be specified by a user)
-#LocalSocket /run/clamav/clamd.sock
-#LocalSocket /tmp/clamd.sock
+LocalSocket /run/clamav/clamd.sock
 
 # Sets the group ownership on the unix socket.
 # Default: disabled (the primary group of the user running clamd)
@@ -224,7 +223,7 @@
 
 # Run as another user (clamd must be started by root for this option to work)
 # Default: don't drop privileges
-#User clamav
+User clamav
 
 # Stop daemon when libclamav reports out of memory condition.
 #ExitOnOOM yes
diff -Naur a/etc/freshclam.conf.sample b/etc/freshclam.conf.sample
--- a/etc/freshclam.conf.sample	2023-08-27 00:00:24.000000000 +0200
+++ b/etc/freshclam.conf.sample	2023-09-04 00:22:43.785454175 +0200
@@ -5,7 +5,7 @@
 
 
 # Comment or remove the line below.
-Example
+#Example
 
 # Path to the database directory.
 # WARNING: It must match clamd.conf's directive!
@@ -14,7 +14,7 @@
 
 # Path to the log file (make sure it has proper permissions)
 # Default: disabled
-#UpdateLogFile /var/log/freshclam.log
+UpdateLogFile /var/log/clamav/freshclam.log
 
 # Maximum size of the log file.
 # Value of 0 disables the limit.
@@ -148,7 +148,7 @@
 
 # Send the RELOAD command to clamd.
 # Default: no
-#NotifyClamd /path/to/clamd.conf
+NotifyClamd /etc/clamav/clamd.conf
 
 # Run command after successful database update.
 # Use EXIT_1 to return 1 after successful database update.
