diff -Naur orig/lib/meson.build patched/lib/meson.build
--- orig/lib/meson.build	1970-01-01 01:00:00.000000000 +0100
+++ patched/lib/meson.build	2022-01-03 22:28:12.820708387 +0100
@@ -152,7 +152,7 @@
 
 lib_intermediates = []
 foreach f: lib_sources
-    name = f.underscorify(f)
+    name = f.underscorify()
     lib = static_library('igt-' + name,
 	[ f, lib_version ],
 	include_directories: inc,
