From f025305936237292ec4d0a8a9fddc1d6cd89a0bd Mon Sep 17 00:00:00 2001
From: Laurent Montel <montel@kde.org>
Date: Tue, 17 Dec 2019 15:32:58 +0100
Subject: Fix Bug 415160 - KMail build fails if messagelib is built without
 Qca-qt5

FIXED-IN: 5.13.1
BUG: 415160

diff --git a/src/kmreadermainwin.cpp b/src/kmreadermainwin.cpp
index f0ad08125..e7bb86d51 100644
--- a/src/kmreadermainwin.cpp
+++ b/src/kmreadermainwin.cpp
@@ -69,6 +69,7 @@
 #include <Akonadi/KMime/MessageFlags>
 #include "kpimtextedit/texttospeech.h"
 #include <mailcommon/mailutil.h>
+#include "messageviewer/config-messageviewer.h"
 #ifdef USE_DKIM_CHECKER
 #include <MessageViewer/DKIMWidgetInfo>
 #endif
@@ -111,8 +112,9 @@ void KMReaderMainWin::initKMReaderMainWin()
     mZoomLabelIndicator = new ZoomLabelWidget(statusBar());
     statusBar()->addPermanentWidget(mZoomLabelIndicator);
     setZoomChanged(mReaderWin->viewer()->webViewZoomFactor());
+#ifdef USE_DKIM_CHECKER
     statusBar()->addPermanentWidget(mReaderWin->viewer()->dkimWidgetInfo());
-
+#endif
     if (!mReaderWin->message().isValid()) {
         menuBar()->hide();
         toolBar(QStringLiteral("mainToolBar"))->hide();
