diff -Naur "orig/opt/Ghetto Skype/resources/app.asar" "patched/opt/Ghetto Skype/resources/app.asar"
--- "orig/opt/Ghetto Skype/resources/app.asar"	2018-03-20 00:41:56.000000000 +0100
+++ "patched/opt/Ghetto Skype/resources/app.asar"	2019-03-26 01:41:59.256911536 +0100
@@ -129,7 +129,7 @@
 
 	TrayIcon.init(mainWindow);
 
-	mainWindow.webContents.setUserAgent("Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.87 Safari/537.36");
+	mainWindow.webContents.setUserAgent("Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.77 Safari/537.36");
 	mainWindow.loadURL('https://web.skype.com/en');
 
 	// Check if we are on a MAC
