diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2019-07-11 02:58:59.000000000 +0200
+++ patched/CMakeLists.txt	2021-11-28 22:19:18.268781639 +0100
@@ -74,13 +74,13 @@
   set(QTAV_INSTALL_HEADERS ${CMAKE_INSTALL_PREFIX}/include)
   set(QTAV_INSTALL_LIBS ${CMAKE_INSTALL_PREFIX}/lib)
   set(QTAV_INSTALL_BINS ${CMAKE_INSTALL_PREFIX}/bin)
-  set(QTAV_INSTALL_QML ${CMAKE_INSTALL_PREFIX}/qml)
+  set(QTAV_INSTALL_QML /usr/lib/qt5/qml)
 else()
   set(CMAKE_INSTALL_PREFIX ${QT_INSTALL_PREFIX} CACHE PATH "default install path" FORCE)
   set(QTAV_INSTALL_HEADERS ${QT_INSTALL_HEADERS})
   set(QTAV_INSTALL_LIBS ${QT_INSTALL_LIBS})
   set(QTAV_INSTALL_BINS ${QT_INSTALL_BINS})
-  set(QTAV_INSTALL_QML ${QT_INSTALL_QML})
+  set(QTAV_INSTALL_QML /usr/lib/qt5/qml)
 endif()
 
 message(STATUS "Qt version: ${Qt5Core_VERSION_STRING}")
