diff -Naur orig/cmake/FindQtAV.cmake patched/cmake/FindQtAV.cmake
--- orig/cmake/FindQtAV.cmake	2019-07-11 02:58:59.000000000 +0200
+++ patched/cmake/FindQtAV.cmake	2021-11-30 02:15:39.018411307 +0100
@@ -9,31 +9,21 @@
 find_package(Qt5 QUIET REQUIRED NO_MODULE COMPONENTS Core)
 
 get_target_property(qmake Qt5::qmake LOCATION)
-execute_process(
-    COMMAND ${qmake} -query QT_INSTALL_HEADERS
-    OUTPUT_VARIABLE QT_INSTALL_HEADERS
-    OUTPUT_STRIP_TRAILING_WHITESPACE
-)
-execute_process(
-    COMMAND ${qmake} -query QT_INSTALL_LIBS
-    OUTPUT_VARIABLE QT_INSTALL_LIBS
-    OUTPUT_STRIP_TRAILING_WHITESPACE
-)
 
 find_path(QTAV_INCLUDE_DIR NAMES QtAV.h
-    HINTS ${QT_INSTALL_HEADERS}
+    HINTS /usr/include/qt5
     PATH_SUFFIXES QtAV
 )
 find_library(QTAV_LIBRARY NAMES QtAV QtAV1
-    HINTS ${QT_INSTALL_LIBS}
+    HINTS /usr/lib
 )
 
 find_path(QTAVWIDGETS_INCLUDE_DIR NAMES QtAVWidgets.h
-    HINTS ${QT_INSTALL_HEADERS}
+    HINTS /usr/include/qt5
     PATH_SUFFIXES QtAVWidgets
 )
 find_library(QTAVWIDGETS_LIBRARY NAMES QtAVWidgets QtAVWidgets1
-    HINTS ${QT_INSTALL_LIBS}
+    HINTS /usr/lib
 )
 
 set(QTAV_INCLUDE_DIRS ${QTAV_INCLUDE_DIR} ${QTAV_INCLUDE_DIR}/..)
diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2019-07-11 02:58:59.000000000 +0200
+++ patched/CMakeLists.txt	2021-11-30 02:17:26.538832075 +0100
@@ -71,16 +71,16 @@
 # Instruct CMake to run moc automatically when needed.
 set(CMAKE_AUTOMOC ON)
 if(NOT CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
-  set(QTAV_INSTALL_HEADERS ${CMAKE_INSTALL_PREFIX}/include)
+  set(QTAV_INSTALL_HEADERS ${CMAKE_INSTALL_PREFIX}/include/qt5)
   set(QTAV_INSTALL_LIBS ${CMAKE_INSTALL_PREFIX}/lib)
   set(QTAV_INSTALL_BINS ${CMAKE_INSTALL_PREFIX}/bin)
-  set(QTAV_INSTALL_QML ${CMAKE_INSTALL_PREFIX}/qml)
+  set(QTAV_INSTALL_QML /usr/lib/qt5/qml)
 else()
   set(CMAKE_INSTALL_PREFIX ${QT_INSTALL_PREFIX} CACHE PATH "default install path" FORCE)
-  set(QTAV_INSTALL_HEADERS ${QT_INSTALL_HEADERS})
-  set(QTAV_INSTALL_LIBS ${QT_INSTALL_LIBS})
-  set(QTAV_INSTALL_BINS ${QT_INSTALL_BINS})
-  set(QTAV_INSTALL_QML ${QT_INSTALL_QML})
+  set(QTAV_INSTALL_HEADERS /usr/include/qt5)
+  set(QTAV_INSTALL_LIBS /usr/lib)
+  set(QTAV_INSTALL_BINS /usr/bin)
+  set(QTAV_INSTALL_QML /usr/lib/qt5/qml)
 endif()
 
 message(STATUS "Qt version: ${Qt5Core_VERSION_STRING}")
