diff -Naur orig/src/CMakeLists.txt patched/src/CMakeLists.txt
--- orig/src/CMakeLists.txt	2019-07-11 02:58:59.000000000 +0200
+++ patched/src/CMakeLists.txt	2021-11-28 22:07:45.527931303 +0100
@@ -402,7 +402,8 @@
   endif()
 endif()
 check_library_exists(portaudio Pa_Initialize "" HAVE_PORTAUDIO)
-if(HAVE_PORTAUDIO)
+check_include_file(portaudio.h HAVE_PORTAUDIO_H)
+if(HAVE_PORTAUDIO AND HAVE_PORTAUDIO_H)
   list(APPEND SOURCES output/audio/AudioOutputPortAudio.cpp)
   list(APPEND EXTRA_DEFS -DQTAV_HAVE_PORTAUDIO=1)
   list(APPEND EXTRA_LIBS portaudio)
