diff -Naur orig/qml/SGVideoNode.cpp patched/qml/SGVideoNode.cpp
--- orig/qml/SGVideoNode.cpp	2019-07-11 02:58:59.000000000 +0200
+++ patched/qml/SGVideoNode.cpp	2019-12-31 02:11:08.540827779 +0100
@@ -26,6 +26,7 @@
 #include <QtCore/QScopedPointer>
 #include <QtGui/QOpenGLFunctions>
 #include <QtQuick/QSGMaterialShader>
+#include <QtQuick/QSGMaterial>
 
 // all in QSGRenderThread
 namespace QtAV {
