diff -Naur orig/src/QtAV/FilterContext.h patched/src/QtAV/FilterContext.h
--- orig/src/QtAV/FilterContext.h	2019-07-11 02:58:59.000000000 +0200
+++ patched/src/QtAV/FilterContext.h	2020-06-01 13:23:17.545532169 +0200
@@ -26,6 +26,7 @@
 #include <QtCore/QByteArray>
 #include <QtCore/QRect>
 #include <QtGui/QPainter>
+#include <QtGui/QPainterPath>
 /*
  * QPainterFilterContext, D2DFilterContext, ...
  */
