diff -Naur orig/build/tools/build-wxwidgets.py patched/build/tools/build-wxwidgets.py
--- orig/build/tools/build-wxwidgets.py	2014-11-27 21:49:59.000000000 +0100
+++ patched/build/tools/build-wxwidgets.py	2019-09-04 00:47:41.396384778 +0200
@@ -271,7 +271,7 @@
                             "--with-opengl",
                             "--enable-sound",
                             "--enable-graphics_ctx",
-                            "--enable-mediactrl",
+                            "--disable-mediactrl",
                             "--enable-display",
                             "--enable-geometry",
                             "--enable-debug_flag",
