diff -up ./configure.ac.tv ./configure.ac
--- ./configure.ac.tv	2022-09-24 15:10:50.266680106 +0200
+++ ./configure.ac	2022-09-24 15:11:07.062073327 +0200
@@ -295,7 +295,7 @@ dnl ************************************
 dnl Check if we have gnome-desktop 43.alpha or newer
 dnl **************************************************************************
 
-PKG_CHECK_EXISTS([gnome-desktop-3.0 >= 43.alpha],
+PKG_CHECK_EXISTS([gnome-desktop-3.0 >= 43],
                  AC_DEFINE([HAVE_GNOME_DESKTOP_43_ALPHA], [1],
                            [Define if gnome-desktop is 43.alpha or newer]))
 
diff -up ./configure.tv ./configure
--- ./configure.tv	2022-09-24 15:10:52.108723234 +0200
+++ ./configure	2022-09-24 15:11:22.104425503 +0200
@@ -22179,8 +22179,8 @@ fi
 
 
 if test -n "$PKG_CONFIG" && \
-    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"gnome-desktop-3.0 >= 43.alpha\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "gnome-desktop-3.0 >= 43.alpha") 2>&5
+    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"gnome-desktop-3.0 >= 43\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "gnome-desktop-3.0 >= 43") 2>&5
   ac_status=$?
   printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
