diff -Naur orig/chrome/browser/extensions/external_component_loader.cc patched/chrome/browser/extensions/external_component_loader.cc
--- orig/chrome/browser/extensions/external_component_loader.cc	2021-09-01 03:39:26.000000000 +0200
+++ patched/chrome/browser/extensions/external_component_loader.cc	2021-09-05 18:01:23.799177055 +0200
@@ -29,18 +29,6 @@
 
 void ExternalComponentLoader::StartLoading() {
   auto prefs = std::make_unique<base::DictionaryValue>();
-#if BUILDFLAG(GOOGLE_CHROME_BRANDING)
-  AddExternalExtension(extension_misc::kInAppPaymentsSupportAppId, prefs.get());
-#endif  // BUILDFLAG(GOOGLE_CHROME_BRANDING)
-
-#if BUILDFLAG(IS_CHROMEOS_ASH)
-  {
-    // Only load the Assessment Assistant if the current session is managed.
-    if (profile_->GetProfilePolicyConnector()->IsManaged())
-      AddExternalExtension(extension_misc::kAssessmentAssistantExtensionId,
-                           prefs.get());
-  }
-#endif
 
   LoadFinished(std::move(prefs));
 }
