diff -Naur orig/chrome/browser/extensions/external_component_loader.cc patched/chrome/browser/extensions/external_component_loader.cc
--- orig/chrome/browser/extensions/external_component_loader.cc	2019-07-29 22:35:48.000000000 +0200
+++ patched/chrome/browser/extensions/external_component_loader.cc	2019-07-30 23:10:17.820189650 +0200
@@ -29,18 +29,6 @@
 
 void ExternalComponentLoader::StartLoading() {
   auto prefs = std::make_unique<base::DictionaryValue>();
-#if defined(GOOGLE_CHROME_BUILD)
-  AddExternalExtension(extension_misc::kInAppPaymentsSupportAppId, prefs.get());
-#endif  // defined(GOOGLE_CHROME_BUILD)
-
-#if defined(OS_CHROMEOS)
-  {
-    // Only load the Assessment Assistant if the current session is managed.
-    if (profile_->GetProfilePolicyConnector()->IsManaged())
-      AddExternalExtension(extension_misc::kAssessmentAssistantExtensionId,
-                           prefs.get());
-  }
-#endif
 
   if (media_router::MediaRouterEnabled(profile_) &&
       FeatureSwitch::load_media_router_component_extension()->IsEnabled()) {
