diff -Naur orig/chrome/browser/ui/startup/startup_browser_creator_impl.cc patched/chrome/browser/ui/startup/startup_browser_creator_impl.cc
--- orig/chrome/browser/ui/startup/startup_browser_creator_impl.cc	2020-02-04 02:56:11.000000000 +0100
+++ patched/chrome/browser/ui/startup/startup_browser_creator_impl.cc	2020-02-07 23:52:29.583050910 +0100
@@ -855,10 +855,6 @@
 
     InfoBarService* infobar_service =
         InfoBarService::FromWebContents(web_contents);
-    if (!google_apis::HasAPIKeyConfigured() ||
-        !google_apis::HasOAuthClientConfigured()) {
-      GoogleApiKeysInfoBarDelegate::Create(infobar_service);
-    }
 
     if (ObsoleteSystem::IsObsoleteNowOrSoon()) {
       PrefService* local_state = g_browser_process->local_state();
