diff -Naur orig/chrome/browser/ui/startup/startup_browser_creator_impl.cc patched/chrome/browser/ui/startup/startup_browser_creator_impl.cc
--- orig/chrome/browser/ui/startup/startup_browser_creator_impl.cc	2018-07-25 00:08:23.000000000 +0200
+++ patched/chrome/browser/ui/startup/startup_browser_creator_impl.cc	2018-07-27 00:35:04.578287453 +0200
@@ -899,8 +899,6 @@
     chrome::ShowBadFlagsPrompt(web_contents);
     InfoBarService* infobar_service =
         InfoBarService::FromWebContents(web_contents);
-    if (!google_apis::HasKeysConfigured())
-      GoogleApiKeysInfoBarDelegate::Create(infobar_service);
     if (ObsoleteSystem::IsObsoleteNowOrSoon()) {
       PrefService* local_state = g_browser_process->local_state();
       if (!local_state ||
