diff -Naur orig/components/content_settings/core/browser/cookie_settings.cc patched/components/content_settings/core/browser/cookie_settings.cc
--- orig/components/content_settings/core/browser/cookie_settings.cc	2020-07-17 20:32:00.846618340 +0200
+++ patched/components/content_settings/core/browser/cookie_settings.cc	2020-07-17 20:13:57.748681169 +0200
@@ -63,7 +63,7 @@
 void CookieSettings::RegisterProfilePrefs(
     user_prefs::PrefRegistrySyncable* registry) {
   registry->RegisterBooleanPref(
-      prefs::kBlockThirdPartyCookies, false,
+      prefs::kBlockThirdPartyCookies, true,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
   registry->RegisterIntegerPref(
       prefs::kCookieControlsMode,
