diff -Naur orig/components/content_settings/core/browser/cookie_settings.cc patched/components/content_settings/core/browser/cookie_settings.cc
--- orig/components/content_settings/core/browser/cookie_settings.cc	2020-11-17 19:39:20.000000000 +0100
+++ patched/components/content_settings/core/browser/cookie_settings.cc	2020-11-18 02:35:34.579103695 +0100
@@ -36,7 +36,7 @@
     : host_content_settings_map_(host_content_settings_map),
       is_incognito_(is_incognito),
       extension_scheme_(extension_scheme),
-      block_third_party_cookies_(false) {
+      block_third_party_cookies_(true) {
   content_settings_observer_.Add(host_content_settings_map_.get());
   pref_change_registrar_.Init(prefs);
   pref_change_registrar_.Add(
@@ -62,7 +62,7 @@
     user_prefs::PrefRegistrySyncable* registry) {
   registry->RegisterIntegerPref(
       prefs::kCookieControlsMode,
-      static_cast<int>(CookieControlsMode::kIncognitoOnly),
+      static_cast<int>(CookieControlsMode::kBlockThirdParty),
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
 }
 
