diff -Naur orig/components/content_settings/core/browser/cookie_settings.cc patched/components/content_settings/core/browser/cookie_settings.cc
--- orig/components/content_settings/core/browser/cookie_settings.cc	2018-07-25 00:08:29.000000000 +0200
+++ patched/components/content_settings/core/browser/cookie_settings.cc	2018-07-27 00:38:07.952355449 +0200
@@ -111,7 +111,7 @@
 void CookieSettings::RegisterProfilePrefs(
     user_prefs::PrefRegistrySyncable* registry) {
   registry->RegisterBooleanPref(
-      prefs::kBlockThirdPartyCookies, false,
+      prefs::kBlockThirdPartyCookies, true,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
 }
 
