diff -Nur orig/src/obconf.glade mod/src/obconf.glade
--- orig/src/obconf.glade	2016-02-20 23:51:21.000000000 +0100
+++ mod/src/obconf.glade	2017-02-08 16:43:02.817920938 +0100
@@ -539,26 +539,6 @@
           </packing>
         </child>
         <child>
-          <object class="GtkFrame" id="frame1">
-            <property name="visible">True</property>
-            <property name="can_focus">False</property>
-            <property name="label_xalign">0</property>
-            <child>
-              <object class="GtkImage" id="preview">
-                <property name="visible">True</property>
-                <property name="can_focus">False</property>
-                <property name="stock">gtk-missing-image</property>
-              </object>
-            </child>
-            <child type="label_item">
-              <placeholder/>
-            </child>
-          </object>
-          <packing>
-            <property name="expand">True</property>
-            <property name="fill">True</property>
-            <property name="position">1</property>
-          </packing>
         </child>
       </object>
       <packing>
