diff -Naur orig/data/default.mjwm mod/data/default.mjwm
--- orig/data/default.mjwm	2014-11-15 18:14:15.000000000 +0100
+++ mod/data/default.mjwm	2015-03-27 17:04:37.274582784 +0100
@@ -14,17 +14,17 @@
 #Editors:mousepad.png:TextEditor
 # will add an entry for Mousepad under Editors subcategory
 
-Settings:gnome-settings:Settings
-Accessories:applications-accessories:Utility
-Development:applications-development:Development
-Education:applications-science:Education
-Games:applications-games:Game
-Graphics:applications-graphics:Graphics
+Beállítás:gnome-settings:Settings
+Alkalmazások:applications-accessories:Utility
+Fejlesztés:applications-development:Development
+Oktatás:applications-science:Education
+Játék:applications-games:Game
+Grafika:applications-graphics:Graphics
 Internet:applications-internet:Network
-Multimedia:applications-multimedia:AudioVideo
-Office:applications-office:Office
-Science:applications-science:Science
-System:applications-system:System
+Multimédia:applications-multimedia:AudioVideo
+Hivatal:applications-office:Office
+Tudomány:applications-science:Science
+Rendszer:applications-system:System
 
 # You can add multiple category names to a subcategory. The desktop entries would be merged in the final menu.
 
