diff -Naur a/BUILD.sh b/BUILD.sh
--- a/BUILD.sh	2017-07-10 10:22:16.000000000 +0200
+++ b/BUILD.sh	2022-05-04 18:52:55.983491297 +0200
@@ -24,8 +24,8 @@
 	echo "Texmaker compilation:"
 	echo "----------------------------------------"
 	while [[ -z ${QTDIR} ]]; do # loop until input is provided
-		read -p "Enter full path to QT (qmake): " QTDIR # get path to qmake
-			if [[ ! -f ${QTDIR}/qmake ]]; then # check if gmake is present in the provided path
+		QTDIR=/usr/lib/qt5
+			if [[ ! -f ${QTDIR}/bin/qmake ]]; then # check if gmake is present in the provided path
 				echo -ne "${txtred}Error${txtrst}: Unable to find qmake in \"${QTDIR}\"\n"
 				QTDIR=""
 			fi
@@ -44,7 +44,7 @@
 	else
 		echo -ne "\rUnable to detect system automatically!\n"
 		while [[ -z ${SYSTEM} ]]; do
-			read -p "Enter System type (1: UNIX ; 2: MACOSX): " SYSTEM
+			SYSTEM="1"
 		done
 	fi
 	echo -ne "\n" # just print a space to make a nicer output
@@ -55,15 +55,15 @@
 
 	if [[ ${SYSTEM} == "1" ]]; then # run if linux is detected
 		while [[ -z ${PREFIX} ]]; do # get installation path for Texmaker
-			read -p "Enter installation path (example: /usr , /usr/local or /opt): " PREFIX
+			PREFIX=/usr
 		done
 		while [[ -z ${DESKTOPDIR} ]]; do # get installation path for .desktop file
-			read -p "Enter directory for the desktop file (example: /usr/share/applications): " DESKTOPDIR
+			DESKTOPDIR=/usr/share/applications
 		done
 		while [[ -z ${ICONDIR} ]]; do # get installation path for Desktop-icon
-			read -p "Enter directory for the icon file (example: /usr/share/pixmaps): " ICONDIR
+			ICONDIR=/usr/share/pixmaps
 		done
-		${QTDIR}/qmake PREFIX=${PREFIX} DESKTOPDIR=${DESKTOPDIR} ICONDIR=${ICONDIR} texmaker.pro # run qmake
+		${QTDIR}/bin/qmake PREFIX=${PREFIX} DESKTOPDIR=${DESKTOPDIR} ICONDIR=${ICONDIR} texmaker.pro # run qmake
 		QMAKE_EXIT="$?" # get exit code of qmake
 		if [[ $? == ${QMAKE_EXIT} ]]; then # check if qmake was successful
 			echo -ne "\nqmake [${txtgrn}Successful${txtrst}]\n"
