diff -Naur a/utilities/texmaker.desktop b/utilities/texmaker.desktop
--- a/utilities/texmaker.desktop	2022-04-24 12:54:29.000000000 +0200
+++ b/utilities/texmaker.desktop	2022-05-04 18:54:41.279995872 +0200
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Categories=Office;Publishing;Qt;X-SuSE-Core-Office;X-Mandriva-Office-Publishing;X-Misc;
+Categories=Office;Publishing;Qt;X-Misc;
 Keywords=Editor;Latex;
 Exec=texmaker %F
 GenericName=LaTeX Editor
