--- ORIG/compiletime-settings	2006-12-10 15:03:24.000000000 +0100
+++ UHU/compiletime-settings	2009-03-01 14:54:28.000000000 +0100
@@ -24,14 +24,14 @@
   plugin_demo off
   target_bits auto
   target_cpu auto
-  prefix /usr/local
+  prefix /usr
   bindir ${prefix}/bin
   sysconfdir /etc/dosemu
   libdir ${prefix}/lib
   datadir ${prefix}/share
-  mandir ${prefix}/man
-  docdir ${datadir}/doc/dosemu
+  mandir ${datadir}/man
+  docdir ${datadir}/doc/Packages/dosemu
   syshdimagedir /var/lib/dosemu
-  x11fontdir ${datadir}/dosemu/Xfonts
+  x11fontdir ${datadir}/fonts
   fdtarball dosemu-freedos-bin.tgz
 }
