diff -Naur orig/Modules/FindOpenGL.cmake patched/Modules/FindOpenGL.cmake
--- orig/Modules/FindOpenGL.cmake	2018-08-09 14:14:07.000000000 +0200
+++ patched/Modules/FindOpenGL.cmake	2018-09-30 01:51:46.560211671 +0200
@@ -242,7 +242,7 @@
     if("x${_OpenGL_GL_POLICY}x" STREQUAL "xNEWx")
       set(OpenGL_GL_PREFERENCE "GLVND")
     else()
-      set(OpenGL_GL_PREFERENCE "LEGACY")
+      set(OpenGL_GL_PREFERENCE "GLVND")
       if("x${_OpenGL_GL_POLICY}x" STREQUAL "xx")
         set(_OpenGL_GL_POLICY_WARN 1)
       endif()
@@ -277,8 +277,8 @@
       "and GLVND libraries for OpenGL and GLX:\n"
       "  OPENGL_opengl_LIBRARY: ${OPENGL_opengl_LIBRARY}\n"
       "  OPENGL_glx_LIBRARY: ${OPENGL_glx_LIBRARY}\n"
-      "OpenGL_GL_PREFERENCE has not been set to \"GLVND\" or \"LEGACY\", so for "
-      "compatibility with CMake 3.10 and below the legacy GL library will be used."
+      "OpenGL_GL_PREFERENCE has not been set to \"GLVND\" or \"LEGACY\", so "
+      "the GLVND library will be used."
       )
   endif()
   unset(_OpenGL_GL_POLICY_WARN)
