diff -Naur a/Modules/GNUInstallDirs.cmake b/Modules/GNUInstallDirs.cmake
--- a/Modules/GNUInstallDirs.cmake	2022-05-25 15:42:51.000000000 +0200
+++ b/Modules/GNUInstallDirs.cmake	2022-05-27 23:50:07.961992469 +0200
@@ -284,9 +284,9 @@
     elseif(NOT DEFINED __system_type_for_install)
       # not debian, alpine, arch, or conda so rely on CMAKE_SIZEOF_VOID_P:
       if("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
-        set(_LIBDIR_DEFAULT "lib64")
+        set(_LIBDIR_DEFAULT "lib")
         if(DEFINED _GNUInstallDirs_LAST_CMAKE_INSTALL_PREFIX)
-          set(__LAST_LIBDIR_DEFAULT "lib64")
+          set(__LAST_LIBDIR_DEFAULT "lib")
         endif()
       endif()
     endif()
diff -Naur a/Modules/Platform/UnixPaths.cmake b/Modules/Platform/UnixPaths.cmake
--- a/Modules/Platform/UnixPaths.cmake	2022-05-25 15:42:51.000000000 +0200
+++ b/Modules/Platform/UnixPaths.cmake	2022-05-27 23:52:34.494446296 +0200
@@ -64,7 +64,7 @@
   )
 
 list(APPEND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES
-  /lib /lib32 /lib64 /usr/lib /usr/lib32 /usr/lib64
+  /lib /usr/lib
   )
 
 if(CMAKE_SYSROOT_COMPILE)
@@ -102,6 +102,6 @@
 endif()
 
 # Enable use of lib32 and lib64 search path variants by default.
-set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIB32_PATHS TRUE)
-set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS TRUE)
-set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIBX32_PATHS TRUE)
+set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIB32_PATHS FALSE)
+set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS FALSE)
+set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIBX32_PATHS FALSE)
diff -Naur a/Source/cmExportInstallFileGenerator.cxx b/Source/cmExportInstallFileGenerator.cxx
--- a/Source/cmExportInstallFileGenerator.cxx	2022-05-25 15:42:51.000000000 +0200
+++ b/Source/cmExportInstallFileGenerator.cxx	2022-05-27 23:54:56.770887171 +0200
@@ -211,11 +211,7 @@
        << "get_filename_component(_IMPORT_PREFIX"
        << " \"${CMAKE_CURRENT_LIST_FILE}\" PATH)\n";
     if (cmHasLiteralPrefix(absDestS, "/lib/") ||
-        cmHasLiteralPrefix(absDestS, "/lib64/") ||
-        cmHasLiteralPrefix(absDestS, "/libx32/") ||
-        cmHasLiteralPrefix(absDestS, "/usr/lib/") ||
-        cmHasLiteralPrefix(absDestS, "/usr/lib64/") ||
-        cmHasLiteralPrefix(absDestS, "/usr/libx32/")) {
+        cmHasLiteralPrefix(absDestS, "/usr/lib/")) {
       // Handle "/usr move" symlinks created by some Linux distros.
       /* clang-format off */
       os <<
