diff -Naur orig/include/nasmlib.h patched/include/nasmlib.h
--- orig/include/nasmlib.h	2018-02-07 22:40:15.000000000 +0100
+++ patched/include/nasmlib.h	2018-05-11 23:49:07.477385104 +0200
@@ -191,7 +191,7 @@
  * seg_init: Initialise the segment-number allocator.
  * seg_alloc: allocate a hitherto unused segment number.
  */
-void pure_func seg_init(void);
+void seg_init(void);
 int32_t pure_func seg_alloc(void);
 
 /*
