diff -Nur orig/cmake/Version.cmake mod/cmake/Version.cmake
--- orig/cmake/Version.cmake	2017-10-31 14:28:36.000000000 +0100
+++ mod/cmake/Version.cmake	2018-06-07 09:44:20.939425028 +0200
@@ -50,12 +50,6 @@
 endif()
 
 if (DEFINED SYNERGY_REVISION)
-    string(LENGTH ${SYNERGY_REVISION} SYNERGY_REVISION_LENGTH)
-    if (NOT ((SYNERGY_REVISION MATCHES "^[a-f0-9]+") AND (SYNERGY_REVISION_LENGTH EQUAL "8")))
-        message (FATAL_ERROR "SYNERGY_REVISION ('${SYNERGY_REVISION}') should be a short commit hash")
-    endif()
-    unset (SYNERGY_REVISION_LENGTH)
-else()
 	set (SYNERGY_REVISION "0badc0de")
 endif()
 
