diff -Naur orig/src/celestia/gtk/data/celestia.desktop mod/src/celestia/gtk/data/celestia.desktop
--- orig/src/celestia/gtk/data/celestia.desktop	2011-06-05 18:11:10.000000000 +0200
+++ mod/src/celestia/gtk/data/celestia.desktop	2016-01-16 20:09:47.389753358 +0100
@@ -1,12 +1,10 @@
 [Desktop Entry]
-Encoding=UTF-8
-Version=1.4.0
 Type=Application
 Exec=celestia
 Icon=celestia
-X-GNOME-DocPath=
 Terminal=false
 Name=Celestia
 Comment=Space Simulator
+Comment[hu]=Űrszimulátor
 StartupNotify=true
-Categories=Astronomy;Science;Application;
+Categories=GTK;GNOME;Science;Astronomy;
