--- celestia-1.6.1/src/celestia/Makefile.in.link	2011-06-06 03:34:28.000000000 +0900
+++ celestia-1.6.1/src/celestia/Makefile.in	2019-08-14 00:48:44.715120048 +0900
@@ -477,7 +477,7 @@
 @ENABLE_KDE_TRUE@                  $(LIBSOCKET) -lDCOP kde/libkdegui.a
 
 @ENABLE_KDE_TRUE@celestia_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-@ENABLE_GTK_TRUE@celestiaGTKLIBS = $(GTK_LIBS) gtk/libgtkgui.a
+@ENABLE_GTK_TRUE@celestiaGTKLIBS = gtk/libgtkgui.a $(GTK_LIBS)
 COMMONSOURCES = \
 	celestiacore.cpp \
 	configfile.cpp \
@@ -530,14 +530,14 @@
 celestia_LDADD = \
 	$(celestiaKDELIBS) \
 	$(celestiaGTKLIBS) \
-	$(DL_LIBS) \
-	$(LUA_LIBS) \
-	$(THEORA_LIBS) \
 	../celengine/libcelengine.a \
 	../celtxf/libceltxf.a \
 	../cel3ds/libcel3ds.a \
 	../celmath/libcelmath.a \
 	../celutil/libcelutil.a \
+	$(DL_LIBS) \
+	$(LUA_LIBS) \
+	$(THEORA_LIBS) \
 	$(SPICE_LIBS)
 
 noinst_HEADERS = $(wildcard *.h)
