diff -Naur celestia-1.6.1-orig/src/celengine/image.cpp celestia-1.6.1/src/celengine/image.cpp
--- celestia-1.6.1-orig/src/celengine/image.cpp	2011-06-05 18:11:09.000000000 +0200
+++ celestia-1.6.1/src/celengine/image.cpp	2014-03-13 20:54:54.051086267 +0100
@@ -45,6 +45,7 @@
 #include <jpeglib.h>
 #endif
 }
+#include <string.h>
 
 #endif // JPEG_SUPPORT
 
diff -Naur celestia-1.6.1-orig/src/celengine/texture.cpp celestia-1.6.1/src/celengine/texture.cpp
--- celestia-1.6.1-orig/src/celengine/texture.cpp	2011-06-05 18:11:09.000000000 +0200
+++ celestia-1.6.1/src/celengine/texture.cpp	2014-03-13 20:55:42.723036140 +0100
@@ -28,6 +28,7 @@
 #include <cstdlib>
 #include <cstdio>
 #include <cassert>
+#include <string.h>
 
 #ifndef _WIN32
 #ifndef TARGET_OS_MAC
