diff -Naur orig/configure patched/configure
--- orig/configure	2011-03-22 13:16:57.134741323 +0100
+++ patched/configure	2017-04-26 02:37:24.094385433 +0200
@@ -47145,8 +47145,8 @@
                 if test "$wxUSE_GTK" = 1; then
         wxUSE_GSTREAMER="no"
 
-                                                                GST_VERSION_MAJOR=0
-        GST_VERSION_MINOR=10
+                                                                GST_VERSION_MAJOR=1
+        GST_VERSION_MINOR=0
         GST_VERSION=$GST_VERSION_MAJOR.$GST_VERSION_MINOR
 
         if test "$wxUSE_GSTREAMER8" = "no"; then
@@ -47160,12 +47160,12 @@
         pkg_cv_GST_CFLAGS="$GST_CFLAGS"
     else
         if test -n "$PKG_CONFIG" && \
-    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"gstreamer-\$GST_VERSION gstreamer-plugins-base-\$GST_VERSION gconf-2.0\"") >&5
-  ($PKG_CONFIG --exists --print-errors "gstreamer-$GST_VERSION gstreamer-plugins-base-$GST_VERSION gconf-2.0") 2>&5
+    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"gstreamer-\$GST_VERSION gstreamer-video-\$GST_VERSION\"") >&5
+  ($PKG_CONFIG --exists --print-errors "gstreamer-$GST_VERSION gstreamer-video-$GST_VERSION") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_GST_CFLAGS=`$PKG_CONFIG --cflags "gstreamer-$GST_VERSION gstreamer-plugins-base-$GST_VERSION gconf-2.0" 2>/dev/null`
+  pkg_cv_GST_CFLAGS=`$PKG_CONFIG --cflags "gstreamer-$GST_VERSION gstreamer-video-$GST_VERSION" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -47178,12 +47178,12 @@
         pkg_cv_GST_LIBS="$GST_LIBS"
     else
         if test -n "$PKG_CONFIG" && \
-    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"gstreamer-\$GST_VERSION gstreamer-plugins-base-\$GST_VERSION gconf-2.0\"") >&5
-  ($PKG_CONFIG --exists --print-errors "gstreamer-$GST_VERSION gstreamer-plugins-base-$GST_VERSION gconf-2.0") 2>&5
+    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"gstreamer-\$GST_VERSION gstreamer-video-\$GST_VERSION\"") >&5
+  ($PKG_CONFIG --exists --print-errors "gstreamer-$GST_VERSION gstreamer-video-$GST_VERSION") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_GST_LIBS=`$PKG_CONFIG --libs "gstreamer-$GST_VERSION gstreamer-plugins-base-$GST_VERSION gconf-2.0" 2>/dev/null`
+  pkg_cv_GST_LIBS=`$PKG_CONFIG --libs "gstreamer-$GST_VERSION gstreamer-video-$GST_VERSION" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -47202,24 +47202,28 @@
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        GST_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "gstreamer-$GST_VERSION gstreamer-plugins-base-$GST_VERSION gconf-2.0"`
+	        GST_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "gstreamer-$GST_VERSION gstreamer-video-$GST_VERSION"`
         else
-	        GST_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "gstreamer-$GST_VERSION gstreamer-plugins-base-$GST_VERSION gconf-2.0"`
+	        GST_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "gstreamer-$GST_VERSION gstreamer-video-$GST_VERSION"`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$GST_PKG_ERRORS" >&5
 
 
-                    { echo "$as_me:$LINENO: WARNING: GStreamer 0.10 not available, falling back to 0.8" >&5
-echo "$as_me: WARNING: GStreamer 0.10 not available, falling back to 0.8" >&2;}
-                    GST_VERSION_MINOR=8
+                    { echo "$as_me:$LINENO: WARNING: GStreamer 1.0 not available, falling back to 0.10" >&5
+echo "$as_me: WARNING: GStreamer 1.0 not available, falling back to 0.10" >&2;}
+                    GST_VERSION_MAJOR=0
+                    GST_VERSION_MINOR=10
+                    GST_VERSION=$GST_VERSION_MAJOR.$GST_VERSION_MINOR
 
 
 elif test $pkg_failed = untried; then
 
-                    { echo "$as_me:$LINENO: WARNING: GStreamer 0.10 not available, falling back to 0.8" >&5
-echo "$as_me: WARNING: GStreamer 0.10 not available, falling back to 0.8" >&2;}
-                    GST_VERSION_MINOR=8
+                    { echo "$as_me:$LINENO: WARNING: GStreamer 1.0 not available, falling back to 0.10" >&5
+echo "$as_me: WARNING: GStreamer 1.0 not available, falling back to 0.10" >&2;}
+                    GST_VERSION_MAJOR=0
+                    GST_VERSION_MINOR=10
+                    GST_VERSION=$GST_VERSION_MAJOR.$GST_VERSION_MINOR
 
 
 else
@@ -47229,14 +47233,13 @@
 echo "${ECHO_T}yes" >&6
 
                     wxUSE_GSTREAMER="yes"
-                    GST_LIBS="$GST_LIBS -lgstinterfaces-$GST_VERSION"
 
 fi
         else
-                        GST_VERSION_MINOR=8
+                        GST_VERSION_MINOR=10
         fi
 
-        if test $GST_VERSION_MINOR = "8"; then
+        if test $GST_VERSION_MINOR = "10"; then
             GST_VERSION=$GST_VERSION_MAJOR.$GST_VERSION_MINOR
 
 pkg_failed=no
@@ -47248,12 +47251,12 @@
         pkg_cv_GST_CFLAGS="$GST_CFLAGS"
     else
         if test -n "$PKG_CONFIG" && \
-    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"gstreamer-\$GST_VERSION gstreamer-interfaces-\$GST_VERSION gstreamer-gconf-\$GST_VERSION\"") >&5
-  ($PKG_CONFIG --exists --print-errors "gstreamer-$GST_VERSION gstreamer-interfaces-$GST_VERSION gstreamer-gconf-$GST_VERSION") 2>&5
+    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"gstreamer-\$GST_VERSION gstreamer-plugins-base-\$GST_VERSION\"") >&5
+  ($PKG_CONFIG --exists --print-errors "gstreamer-$GST_VERSION gstreamer-plugins-base-$GST_VERSION") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_GST_CFLAGS=`$PKG_CONFIG --cflags "gstreamer-$GST_VERSION gstreamer-interfaces-$GST_VERSION gstreamer-gconf-$GST_VERSION" 2>/dev/null`
+  pkg_cv_GST_CFLAGS=`$PKG_CONFIG --cflags "gstreamer-$GST_VERSION gstreamer-plugins-base-$GST_VERSION" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -47266,12 +47269,12 @@
         pkg_cv_GST_LIBS="$GST_LIBS"
     else
         if test -n "$PKG_CONFIG" && \
-    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"gstreamer-\$GST_VERSION gstreamer-interfaces-\$GST_VERSION gstreamer-gconf-\$GST_VERSION\"") >&5
-  ($PKG_CONFIG --exists --print-errors "gstreamer-$GST_VERSION gstreamer-interfaces-$GST_VERSION gstreamer-gconf-$GST_VERSION") 2>&5
+    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"gstreamer-\$GST_VERSION gstreamer-plugins-base-\$GST_VERSION\"") >&5
+  ($PKG_CONFIG --exists --print-errors "gstreamer-$GST_VERSION gstreamer-plugins-base-$GST_VERSION") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_GST_LIBS=`$PKG_CONFIG --libs "gstreamer-$GST_VERSION gstreamer-interfaces-$GST_VERSION gstreamer-gconf-$GST_VERSION" 2>/dev/null`
+  pkg_cv_GST_LIBS=`$PKG_CONFIG --libs "gstreamer-$GST_VERSION gstreamer-plugins-base-$GST_VERSION" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -47290,21 +47293,23 @@
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        GST_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "gstreamer-$GST_VERSION gstreamer-interfaces-$GST_VERSION gstreamer-gconf-$GST_VERSION"`
+	        GST_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "gstreamer-$GST_VERSION gstreamer-plugins-base-$GST_VERSION"`
         else
-	        GST_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "gstreamer-$GST_VERSION gstreamer-interfaces-$GST_VERSION gstreamer-gconf-$GST_VERSION"`
+	        GST_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "gstreamer-$GST_VERSION gstreamer-plugins-base-$GST_VERSION"`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$GST_PKG_ERRORS" >&5
 
 
-                    { echo "$as_me:$LINENO: WARNING: GStreamer 0.8/0.10 not available." >&5
-echo "$as_me: WARNING: GStreamer 0.8/0.10 not available." >&2;}
+                    { echo "$as_me:$LINENO: WARNING: GStreamer 0.10 not available" >&5
+echo "$as_me: WARNING: GStreamer 0.10 not available" >&2;}
+
 
 elif test $pkg_failed = untried; then
 
-                    { echo "$as_me:$LINENO: WARNING: GStreamer 0.8/0.10 not available." >&5
-echo "$as_me: WARNING: GStreamer 0.8/0.10 not available." >&2;}
+                    { echo "$as_me:$LINENO: WARNING: GStreamer 0.10 not available" >&5
+echo "$as_me: WARNING: GStreamer 0.10 not available" >&2;}
+
 
 else
 	GST_CFLAGS=$pkg_cv_GST_CFLAGS
@@ -47317,14 +47322,6 @@
 
 
         if test "$wxUSE_GSTREAMER" = "yes"; then
-                                    case "${host}" in
-                *-*-solaris2* )
-                    if test "$GCC" = yes; then
-                        GST_CFLAGS=`echo $GST_CFLAGS | sed 's/-mt//'`
-                        GST_LIBS=`echo $GST_LIBS | sed 's/-mt//'`
-                    fi
-            esac
-
             CPPFLAGS="$GST_CFLAGS $CPPFLAGS"
             EXTRALIBS_MEDIA="$GST_LIBS"
 
@@ -47335,6 +47332,9 @@
         else
             USE_MEDIA=0
         fi
+
+    elif test "$wxUSE_MAC" = 1; then
+                           GST_LIBS="-framework QTKit"
     fi
 
     if test $USE_MEDIA = 1; then
@@ -47343,6 +47343,13 @@
 #define wxUSE_MEDIACTRL 1
 _ACEOF
 
+    else
+        if test "$wxUSE_MEDIACTRL" = "yes"; then
+            as_fn_error $? "GStreamer not available" "$LINENO" 5
+        else
+                        { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: GStreamer not available... disabling wxMediaCtrl" >&5
+$as_echo "$as_me: WARNING: GStreamer not available... disabling wxMediaCtrl" >&2;}
+        fi
     fi
 fi
 
diff -Naur orig/src/unix/mediactrl.cpp patched/src/unix/mediactrl.cpp
--- orig/src/unix/mediactrl.cpp	2011-03-22 13:16:43.134741323 +0100
+++ patched/src/unix/mediactrl.cpp	2017-04-26 01:13:35.210850001 +0200
@@ -20,12 +20,10 @@
 
 #include <gst/gst.h>                // main gstreamer header
 
-// xoverlay/video stuff, gst-gconf for 0.8
-#if GST_VERSION_MAJOR > 0 || GST_VERSION_MINOR >= 10
-#   include <gst/interfaces/xoverlay.h>
+#if GST_CHECK_VERSION(1,0,0)
+#include <gst/video/video.h>
 #else
-#   include <gst/xoverlay/xoverlay.h>
-#   include <gst/gconf/gconf.h>        // gstreamer glib configuration
+#include <gst/interfaces/xoverlay.h>
 #endif
 
 #ifndef  WX_PRECOMP
@@ -91,28 +89,10 @@
 //  GStreamer (most version compatability) macros
 //-----------------------------------------------------------------------------
 
-// In 0.9 there was a HUGE change to GstQuery and the
-// gst_element_query function changed dramatically and split off
-// into two seperate ones
-#if GST_VERSION_MAJOR == 0 && GST_VERSION_MINOR <= 8
-#    define wxGst_element_query_duration(e, f, p) \
-                gst_element_query(e, GST_QUERY_TOTAL, f, p)
-#    define wxGst_element_query_position(e, f, p) \
-                gst_element_query(e, GST_QUERY_POSITION, f, p)
-#elif GST_VERSION_MAJOR == 0 && GST_VERSION_MINOR == 9
-// However, the actual 0.9 version has a slightly different definition
-// and instead of gst_element_query_duration it has two parameters to
-// gst_element_query_position instead
-#    define wxGst_element_query_duration(e, f, p) \
-                gst_element_query_position(e, f, 0, p)
-#    define wxGst_element_query_position(e, f, p) \
-                gst_element_query_position(e, f, p, 0)
-#else
 #    define wxGst_element_query_duration \
                 gst_element_query_duration
 #    define wxGst_element_query_position \
                 gst_element_query_position
-#endif
 
 // Other 0.10 macros
 #if GST_VERSION_MAJOR > 0 || GST_VERSION_MINOR >= 10
@@ -218,7 +198,11 @@
     double          m_dRate;        // Current playback rate -
                                     // see GetPlaybackRate for notes
     wxLongLong      m_llPausedPos;  // Paused position - see Pause()
+#if GST_CHECK_VERSION(1,0,0)
+    GstVideoOverlay* m_xoverlay;     // X Overlay that contains the GST video
+#else
     GstXOverlay*    m_xoverlay;     // X Overlay that contains the GST video
+#endif
     wxMutex         m_asynclock;    // See "discussion of internals"
     class wxGStreamerMediaEventHandler* m_eventHandler; // see below
 
@@ -294,7 +278,11 @@
         // GST Doesn't redraw automatically while paused
         // Plus, the video sometimes doesn't redraw when it looses focus
         // or is painted over so we just tell it to redraw...
+#if GST_CHECK_VERSION(1,0,0)
+        gst_video_overlay_expose(be->m_xoverlay);
+#else
         gst_x_overlay_expose(be->m_xoverlay);
+#endif
     }
     else
     {
@@ -332,9 +320,15 @@
     GdkWindow *window = GTK_PIZZA(theWidget)->bin_window;
     wxASSERT(window);
 
+#if GST_CHECK_VERSION(1,0,0)
+    gst_video_overlay_set_window_handle(be->m_xoverlay,
+                                GDK_WINDOW_XID(window)
+                                );
+#else
     gst_x_overlay_set_xwindow_id( GST_X_OVERLAY(be->m_xoverlay),
                                 GDK_WINDOW_XWINDOW( window )
                                 );
+#endif
     g_signal_connect (be->GetControl()->m_wxwindow,
                       "expose_event",
                       G_CALLBACK(gtk_window_expose_callback), be);
@@ -344,30 +338,6 @@
 #endif // wxGTK
 
 //-----------------------------------------------------------------------------
-// "state-change" from m_playbin/GST_MESSAGE_STATE_CHANGE
-//
-// Called by gstreamer when the state changes - here we
-// send the appropriate corresponding wx event.
-//
-// 0.8 only as HandleStateChange does this in both versions
-//-----------------------------------------------------------------------------
-#if GST_VERSION_MAJOR == 0 && GST_VERSION_MINOR < 10
-extern "C" {
-static void gst_state_change_callback(GstElement *play,
-                                      GstElementState oldstate,
-                                      GstElementState newstate,
-                                      wxGStreamerMediaBackend* be)
-{
-    if(be->m_asynclock.TryLock() == wxMUTEX_NO_ERROR)
-    {
-        be->HandleStateChange(oldstate, newstate);
-        be->m_asynclock.Unlock();
-    }
-}
-}
-#endif // <0.10
-
-//-----------------------------------------------------------------------------
 // "eos" from m_playbin/GST_MESSAGE_EOS
 //
 // Called by gstreamer when the media is done playing ("end of stream")
@@ -449,32 +419,6 @@
 #endif
 
 //-----------------------------------------------------------------------------
-// "desired-size-changed" from m_xoverlay
-//
-// 0.8-specific this provides us with the video size when it changes -
-// even though we get the caps as well this seems to come before the
-// caps notification does...
-//
-// Note it will return 16,16 for an early-bird value or for audio
-//-----------------------------------------------------------------------------
-#if GST_VERSION_MAJOR == 0 && GST_VERSION_MINOR < 10
-extern "C" {
-static void gst_desired_size_changed_callback(GstElement * play,
-                                              guint width, guint height,
-                                              wxGStreamerMediaBackend* be)
-{
-    if(!(width == 16 && height == 16))
-    {
-        be->m_videoSize.x = width;
-        be->m_videoSize.y = height;
-    }
-    else
-        be->QueryVideoSizeFromElement(be->m_playbin);
-}
-}
-#endif
-
-//-----------------------------------------------------------------------------
 // gst_bus_async_callback [static]
 // gst_bus_sync_callback [static]
 //
@@ -487,7 +431,6 @@
 // thread before the async version that we use to set the xwindow id of the
 // XOverlay (NB: This isn't currently used - see CreateControl()).
 //-----------------------------------------------------------------------------
-#if GST_VERSION_MAJOR > 0 || GST_VERSION_MINOR >= 10
 extern "C" {
 static gboolean gst_bus_async_callback(GstBus* bus,
                                        GstMessage* message,
@@ -535,8 +478,12 @@
 {
     // Pass a non-xwindowid-setting event on to the async handler where it
     // belongs
+#if GST_CHECK_VERSION(1,0,0)
+    if (!gst_is_video_overlay_prepare_window_handle_message (message))
+#else
     if (GST_MESSAGE_TYPE (message) != GST_MESSAGE_ELEMENT ||
         !gst_structure_has_name (message->structure, "prepare-xwindow-id"))
+#endif
     {
         //
         // NB: Unfortunately, the async callback can be quite
@@ -555,7 +502,6 @@
     return GST_BUS_DROP; // We handled this message - drop from the queue
 }
 }
-#endif
 
 //-----------------------------------------------------------------------------
 //
@@ -601,6 +547,7 @@
     }
 }
 
+#if !GST_CHECK_VERSION(1,0,0)
 //-----------------------------------------------------------------------------
 // wxGStreamerMediaBackend::QueryVideoSizeFromElement
 //
@@ -640,14 +587,6 @@
             else
                 g_object_get (info, "object", &pad, NULL);
 
-#if GST_VERSION_MAJOR == 0 && GST_VERSION_MINOR <= 8
-            // Killed in 0.9, presumely because events and such
-            // should be pushed on pads regardless of whether they
-            // are currently linked
-            pad = (GstPad *) GST_PAD_REALIZE (pad);
-            wxASSERT(pad);
-#endif
-
             if(!QueryVideoSizeFromPad(pad))
             {
                 // wait for those caps to get ready
@@ -670,6 +609,7 @@
 
     return true;
 }
+#endif
 
 //-----------------------------------------------------------------------------
 // wxGStreamerMediaBackend::QueryVideoSizeFromPad
@@ -678,7 +618,11 @@
 //-----------------------------------------------------------------------------
 bool wxGStreamerMediaBackend::QueryVideoSizeFromPad(GstPad* pad)
 {
+#if GST_CHECK_VERSION(1,0,0)
+    GstCaps* caps = gst_pad_get_current_caps(pad);
+#else
     const GstCaps* caps = GST_PAD_CAPS(pad);
+#endif
     if ( caps )
     {
         const GstStructure *s = gst_caps_get_structure (caps, 0);
@@ -704,11 +648,21 @@
                 m_videoSize.y = (int) ((float) den * m_videoSize.y / num);
         }
 
-         wxLogTrace(wxTRACE_GStreamer, wxT("Adjusted video size: [%i,%i]"),
-                     m_videoSize.x, m_videoSize.y);
+        wxLogTrace(wxTRACE_GStreamer, wxT("Adjusted video size: [%i,%i]"),
+                    m_videoSize.x, m_videoSize.y);
+#if GST_CHECK_VERSION(1,0,0)
+        gst_caps_unref (caps);
+#endif
+
+        NotifyMovieSizeChanged ();
+
         return true;
     } // end if caps
 
+    m_videoSize = wxSize(0,0);
+
+    NotifyMovieSizeChanged ();
+
     return false; // not ready/massive failure
 }
 
@@ -737,7 +691,11 @@
         wxASSERT(window);
 #endif
 
+#if GST_CHECK_VERSION(1,0,0)
+    gst_video_overlay_set_window_handle(m_xoverlay,
+#else
     gst_x_overlay_set_xwindow_id( GST_X_OVERLAY(m_xoverlay),
+#endif
 #ifdef __WXGTK__
                         GDK_WINDOW_XWINDOW( window )
 #else
@@ -766,7 +724,6 @@
 //
 // PRECONDITION: Assumes m_asynclock is Lock()ed
 //-----------------------------------------------------------------------------
-#if GST_VERSION_MAJOR > 0 || GST_VERSION_MINOR >= 10
 bool wxGStreamerMediaBackend::SyncStateChange(GstElement* element,
                                               GstElementState desiredstate,
                                               gint64 llTimeout)
@@ -841,23 +798,6 @@
 
     return bSuccess;
 }
-#else // 0.8 implementation
-bool wxGStreamerMediaBackend::SyncStateChange(GstElement* element,
-                                              GstElementState desiredstate,
-                                              gint64 llTimeout)
-{
-    gint64 llTimeWaited = 0;
-    while(GST_STATE(element) != desiredstate)
-    {
-        if(llTimeWaited >= llTimeout)
-            break;
-        llTimeWaited += 10*GST_MSECOND;
-        wxMilliSleep(10);
-    }
-
-    return llTimeWaited != llTimeout;
-}
-#endif
 
 //-----------------------------------------------------------------------------
 // wxGStreamerMediaBackend::TryAudioSink
@@ -881,6 +821,31 @@
 
 bool wxGStreamerMediaBackend::TryVideoSink(GstElement* videosink)
 {
+#if GST_CHECK_VERSION(1,0,0)
+    // Check if the video sink either is an xoverlay or might contain one...
+    if( !GST_IS_BIN(videosink) && !GST_IS_VIDEO_OVERLAY(videosink) )
+    {
+        if(G_IS_OBJECT(videosink))
+            g_object_unref(videosink);
+        return false;
+    }
+
+    // Make our video sink and make sure it supports the x overlay interface
+    // the x overlay enables us to put the video in our control window
+    // (i.e. we NEED it!) - also connect to the natural video size change event
+    if( GST_IS_BIN(videosink) )
+        m_xoverlay = (GstVideoOverlay*)
+                        gst_bin_get_by_interface (GST_BIN (videosink),
+                                                  GST_TYPE_VIDEO_OVERLAY);
+    else
+        m_xoverlay = (GstVideoOverlay*) videosink;
+
+    if ( !GST_IS_VIDEO_OVERLAY(m_xoverlay) )
+    {
+        g_object_unref(videosink);
+        return false;
+    }
+#else
     // Check if the video sink either is an xoverlay or might contain one...
     if( !GST_IS_BIN(videosink) && !GST_IS_X_OVERLAY(videosink) )
     {
@@ -904,7 +869,7 @@
         g_object_unref(videosink);
         return false;
     }
-
+#endif
     return true;
 }
 
@@ -1017,11 +982,8 @@
     //Really init gstreamer
     gboolean bInited;
     GError* error = NULL;
-#if GST_VERSION_MAJOR > 0 || GST_VERSION_MINOR >= 10
+
     bInited = gst_init_check(&argcGST, &argvGST, &error);
-#else
-    bInited = gst_init_check(&argcGST, &argvGST);
-#endif
 
     // Cleanup arguments for unicode case
 #if wxUSE_UNICODE
@@ -1088,14 +1050,9 @@
         return false;
     }
 
-#if GST_VERSION_MAJOR == 0 && GST_VERSION_MINOR < 10
-    // Connect the glib events/callbacks we want to our playbin
-    g_signal_connect(m_playbin, "eos",
-                     G_CALLBACK(gst_finish_callback), this);
-    g_signal_connect(m_playbin, "error",
-                     G_CALLBACK(gst_error_callback), this);
-    g_signal_connect(m_playbin, "state-change",
-                     G_CALLBACK(gst_state_change_callback), this);
+#if GST_CHECK_VERSION(1,0,0)
+    gst_bus_set_sync_handler(gst_element_get_bus(m_playbin),
+                             (GstBusSyncHandler) gst_bus_sync_callback, this, NULL);
 #else
     // GStreamer 0.10+ uses GstBus for this now, connect to the sync
     // handler as well so we can set the X window id of our xoverlay
@@ -1103,14 +1060,15 @@
                        (GstBusFunc) gst_bus_async_callback, this);
     gst_bus_set_sync_handler(gst_element_get_bus(m_playbin),
                              (GstBusSyncHandler) gst_bus_sync_callback, this);
-    g_signal_connect(m_playbin, "notify::stream-info",
-                     G_CALLBACK(gst_notify_stream_info_callback), this);
 #endif
 
     // Get the audio sink
-    GstElement* audiosink = gst_gconf_get_default_audio_sink();
+    GstElement* audiosink;
+#if !GST_CHECK_VERSION(1,0,0)
+    audiosink = gst_element_factory_make ("gconfaudiosink", "audio-sink");
     if( !TryAudioSink(audiosink) )
     {
+#endif
         // fallback to autodetection, then alsa, then oss as a stopgap
         audiosink = gst_element_factory_make ("autoaudiosink", "audio-sink");
         if( !TryAudioSink(audiosink) )
@@ -1126,13 +1084,18 @@
                 }
             }
         }
+#if !GST_CHECK_VERSION(1,0,0)
     }
+#endif
 
     // Setup video sink - first try gconf, then auto, then xvimage and
     // then finally plain ximage
-    GstElement* videosink = gst_gconf_get_default_video_sink();
+    GstElement* videosink;
+#if !GST_CHECK_VERSION(1,0,0)
+    videosink = gst_element_factory_make ("gconfvideosink", "video-sink");
     if( !TryVideoSink(videosink) )
     {
+#endif
         videosink = gst_element_factory_make ("autovideosink", "video-sink");
         if( !TryVideoSink(videosink) )
         {
@@ -1150,13 +1113,19 @@
                 }
             }
         }
+#if !GST_CHECK_VERSION(1,0,0)
     }
+#endif
 
-#if GST_VERSION_MAJOR == 0 && GST_VERSION_MINOR < 10
-    // Not on 0.10... called when video size changes
-    g_signal_connect(m_xoverlay, "desired-size-changed",
-                     G_CALLBACK(gst_desired_size_changed_callback), this);
+#if GST_CHECK_VERSION(1,0,0)
+    GstPad *video_sinkpad = gst_element_get_static_pad (videosink, "sink");
+    g_signal_connect (video_sinkpad, "notify::caps", G_CALLBACK (gst_notify_caps_callback), this);
+    gst_object_unref (video_sinkpad);
+#else
+    g_signal_connect(m_playbin, "notify::stream-info",
+                     G_CALLBACK(gst_notify_stream_info_callback), this);
 #endif
+
     // Tell GStreamer which window to draw to in 0.8 - 0.10
     // sometimes needs this too...
     SetupXOverlay();
@@ -1192,6 +1161,7 @@
 //-----------------------------------------------------------------------------
 bool wxGStreamerMediaBackend::Load(const wxURI& location)
 {
+#if !GST_CHECK_VERSION(1,0,0)
     if(location.GetScheme().CmpNoCase(wxT("file")) == 0)
     {
         wxString uristring = location.BuildUnescapedURI();
@@ -1203,6 +1173,7 @@
                      );
     }
     else
+#endif
         return DoLoad(location.BuildURI());
 }
 
@@ -1372,11 +1343,17 @@
     else
     {
         gint64 pos;
+#if GST_CHECK_VERSION(1,0,0)
+        if (!gst_element_query_position(m_playbin, GST_FORMAT_TIME, &pos) ||
+            pos == -1)
+            return 0;
+#else
         GstFormat fmtTime = GST_FORMAT_TIME;
 
         if (!wxGst_element_query_position(m_playbin, &fmtTime, &pos) ||
             fmtTime != GST_FORMAT_TIME || pos == -1)
             return 0;
+#endif
         return pos / GST_MSECOND ;
     }
 }
@@ -1401,36 +1378,17 @@
 //-----------------------------------------------------------------------------
 bool wxGStreamerMediaBackend::SetPosition(wxLongLong where)
 {
-#if GST_VERSION_MAJOR == 0 && GST_VERSION_MINOR == 8 \
-                           && GST_VERSION_MICRO == 0
-    // 0.8.0 has no gst_element_seek according to official docs!!!
-    wxLogSysError(wxT("GStreamer 0.8.0 does not have gst_element_seek")
-                  wxT(" according to official docs"));
-    return false;
-#else // != 0.8.0
 
-#   if GST_VERSION_MAJOR > 0 || GST_VERSION_MINOR >= 10
         gst_element_seek (m_playbin, m_dRate, GST_FORMAT_TIME,
            (GstSeekFlags)(GST_SEEK_FLAG_FLUSH | GST_SEEK_FLAG_KEY_UNIT),
                           GST_SEEK_TYPE_SET, where.GetValue() * GST_MSECOND,
                           GST_SEEK_TYPE_NONE, GST_CLOCK_TIME_NONE );
-#   else
-        // NB: Some gstreamer versions return false basically all the time
-        // here - even totem doesn't bother to check the return value here
-        // so I guess we'll just assume it worked -
-        // TODO: maybe check the gst error callback???
-        gst_element_seek (m_playbin, (GstSeekType) (GST_SEEK_METHOD_SET |
-            GST_FORMAT_TIME | GST_SEEK_FLAG_FLUSH),
-            where.GetValue() * GST_MSECOND );
-
-#   endif // GST_VERSION_MAJOR > 0 || GST_VERSION_MINOR >= 10
 
     {
         m_llPausedPos = where;
         return true;
     }
     return true;
-#endif //== 0.8.0
 }
 
 //-----------------------------------------------------------------------------
@@ -1442,11 +1400,17 @@
 wxLongLong wxGStreamerMediaBackend::GetDuration()
 {
     gint64 length;
+#if GST_CHECK_VERSION(1,0,0)
+    if(!gst_element_query_duration(m_playbin, GST_FORMAT_TIME, &length) ||
+       length == -1)
+        return 0;
+#else
     GstFormat fmtTime = GST_FORMAT_TIME;
 
     if(!wxGst_element_query_duration(m_playbin, &fmtTime, &length) ||
        fmtTime != GST_FORMAT_TIME || length == -1)
         return 0;
+#endif
     return length / GST_MSECOND ;
 }
 
@@ -1543,11 +1507,17 @@
 wxLongLong wxGStreamerMediaBackend::GetDownloadTotal()
 {
     gint64 length;
+#if GST_CHECK_VERSION(1,0,0)
+    if (!gst_element_query_duration(m_playbin, GST_FORMAT_BYTES, &length) ||
+         length == -1)
+        return 0;
+#else
     GstFormat fmtBytes = GST_FORMAT_BYTES;
 
     if (!wxGst_element_query_duration(m_playbin, &fmtBytes, &length) ||
           fmtBytes != GST_FORMAT_BYTES || length == -1)
         return 0;
+#endif
     return length;
 }
 
