diff -Naur orig/migrate/default.xml patched/migrate/default.xml
--- orig/migrate/default.xml	2015-02-28 17:22:03.000000000 +0100
+++ patched/migrate/default.xml	2016-02-21 18:29:40.885668065 +0100
@@ -50,17 +50,17 @@
     <property name="plugin-8" type="string" value="separator"/>
     <property name="plugin-9" type="string" value="launcher">
       <property name="items" type="array">
-        <value type="string" value="exo-terminal-emulator.desktop"/>
+        <value type="string" value="xfce4-terminal.desktop"/>
       </property>
     </property>
     <property name="plugin-10" type="string" value="launcher">
       <property name="items" type="array">
-        <value type="string" value="exo-file-manager.desktop"/>
+        <value type="string" value="Thunar.desktop"/>
       </property>
     </property>
     <property name="plugin-11" type="string" value="launcher">
       <property name="items" type="array">
-        <value type="string" value="exo-web-browser.desktop"/>
+        <value type="string" value="firefox.desktop"/>
       </property>
     </property>
     <property name="plugin-12" type="string" value="launcher">
diff -Naur orig/migrate/default.xml.in patched/migrate/default.xml.in
--- orig/migrate/default.xml.in	2014-04-24 22:16:37.000000000 +0200
+++ patched/migrate/default.xml.in	2016-02-21 18:28:14.929241830 +0100
@@ -50,17 +50,17 @@
     <property name="plugin-8" type="string" value="separator"/>
     <property name="plugin-9" type="string" value="launcher">
       <property name="items" type="array">
-        <value type="string" value="exo-terminal-emulator.desktop"/>
+        <value type="string" value="xfce4-terminal.desktop"/>
       </property>
     </property>
     <property name="plugin-10" type="string" value="launcher">
       <property name="items" type="array">
-        <value type="string" value="exo-file-manager.desktop"/>
+        <value type="string" value="Thunar.desktop"/>
       </property>
     </property>
     <property name="plugin-11" type="string" value="launcher">
       <property name="items" type="array">
-        <value type="string" value="exo-web-browser.desktop"/>
+        <value type="string" value="firefox.desktop"/>
       </property>
     </property>
     <property name="plugin-12" type="string" value="launcher">
