diff -Nur orig/migrate/default.xml mod/migrate/default.xml
--- orig/migrate/default.xml	2021-02-25 14:47:47.000000000 +0100
+++ mod/migrate/default.xml	2021-04-14 18:32:40.773548535 +0200
@@ -85,12 +85,12 @@
     </property>
     <property name="plugin-18" type="string" value="launcher">
       <property name="items" type="array">
-        <value type="string" value="xfce4-file-manager.desktop"/>
+        <value type="string" value="Thunar.desktop.desktop"/>
       </property>
     </property>
     <property name="plugin-19" type="string" value="launcher">
       <property name="items" type="array">
-        <value type="string" value="xfce4-web-browser.desktop"/>
+        <value type="string" value="firefox.desktop"/>
       </property>
     </property>
     <property name="plugin-20" type="string" value="launcher">
diff -Nur orig/migrate/default.xml.in mod/migrate/default.xml.in
--- orig/migrate/default.xml.in	2020-08-21 20:34:07.000000000 +0200
+++ mod/migrate/default.xml.in	2021-04-14 18:34:41.333294534 +0200
@@ -85,12 +85,12 @@
     </property>
     <property name="plugin-18" type="string" value="launcher">
       <property name="items" type="array">
-        <value type="string" value="xfce4-file-manager.desktop"/>
+        <value type="string" value="Thunar.desktop"/>
       </property>
     </property>
     <property name="plugin-19" type="string" value="launcher">
       <property name="items" type="array">
-        <value type="string" value="xfce4-web-browser.desktop"/>
+        <value type="string" value="firefox.desktop"/>
       </property>
     </property>
     <property name="plugin-20" type="string" value="launcher">
