diff -Naur orig/texk/kpathsea/texmf.cnf patched/texk/kpathsea/texmf.cnf
--- orig/texk/kpathsea/texmf.cnf	2020-01-16 23:02:50.000000000 +0100
+++ patched/texk/kpathsea/texmf.cnf	2020-09-04 14:28:04.932204940 +0200
@@ -59,7 +59,7 @@
 % SELFAUTOPARENT (its grandparent = /usr/local/texlive/YYYY), and
 % SELFAUTOGRANDPARENT (its great-grandparent = /usr/local/texlive).
 % Sorry for the off-by-one-generation names.
-TEXMFROOT = $SELFAUTOPARENT
+TEXMFROOT = /usr/share
 
 % The main tree of distributed packages and programs:
 TEXMFDIST = $TEXMFROOT/texmf-dist
