diff -Naur orig/texk/web2c/pdftexdir/pdftoepdf-poppler0.76.0.cc patched/texk/web2c/pdftexdir/pdftoepdf-poppler0.76.0.cc
--- orig/texk/web2c/pdftexdir/pdftoepdf-poppler0.76.0.cc	2020-01-12 00:09:41.009960032 +0100
+++ patched/texk/web2c/pdftexdir/pdftoepdf-poppler0.76.0.cc	2020-01-12 00:12:21.655939150 +0100
@@ -723,7 +723,7 @@
 #endif
     // initialize
     if (!isInit) {
-        globalParams = new GlobalParams();
+        globalParams.reset(new GlobalParams());
         globalParams->setErrQuiet(false);
         isInit = true;
     }
@@ -1108,6 +1108,5 @@
             delete_document(p);
         }
         // see above for globalParams
-        delete globalParams;
     }
 }
diff -Naur orig/texk/web2c/pdftexdir/pdftosrc-poppler0.76.0.cc patched/texk/web2c/pdftexdir/pdftosrc-poppler0.76.0.cc
--- orig/texk/web2c/pdftexdir/pdftosrc-poppler0.76.0.cc	2020-01-12 00:09:41.009960032 +0100
+++ patched/texk/web2c/pdftexdir/pdftosrc-poppler0.76.0.cc	2020-01-12 00:15:10.466019362 +0100
@@ -79,7 +79,7 @@
         exit(1);
     }
     fileName = new GString(argv[1]);
-    globalParams = new GlobalParams();
+    globalParams.reset(new GlobalParams());
     doc = new PDFDoc(fileName);
     if (!doc->isOk()) {
         fprintf(stderr, "Invalid PDF file\n");
@@ -100,7 +100,7 @@
     if (objnum == 0) {
         srcStream = catalogDict.dictLookup("SourceObject");
         static char const_SourceFile[] = "SourceFile";
-        if (!srcStream.isStream(const_SourceFile)) {
+        if (!srcStream.isDict(const_SourceFile)) {
             fprintf(stderr, "No SourceObject found\n");
             exit(1);
         }
@@ -202,5 +202,4 @@
         fprintf(stderr, "Cross-reference table extracted to %s\n", outname);
     fclose(outfile);
     delete doc;
-    delete globalParams;
 }
