diff -Naur orig/texk/kpathsea/texmf.cnf patched/texk/kpathsea/texmf.cnf
--- orig/texk/kpathsea/texmf.cnf	2018-03-26 14:23:43.000000000 +0200
+++ patched/texk/kpathsea/texmf.cnf	2018-05-21 08:14:48.981161181 +0200
@@ -58,7 +58,7 @@
 % SELFAUTOPARENT (its grandparent = /usr/local/texlive/YYYY), and
 % SELFAUTOGRANDPARENT (its great-grandparent = /usr/local/texlive).
 % Sorry for the off-by-one-generation names.
-TEXMFROOT = $SELFAUTOPARENT
+TEXMFROOT = /usr/share
 
 % The main tree of distributed packages and programs:
 TEXMFDIST = $TEXMFROOT/texmf-dist
