diff -Naur FreeRDP-be8f8f72387e7878717b6f04c9a87f999449d20d.old/cmake/ConfigOptions.cmake FreeRDP-be8f8f72387e7878717b6f04c9a87f999449d20d/cmake/ConfigOptions.cmake
--- FreeRDP-be8f8f72387e7878717b6f04c9a87f999449d20d.old/cmake/ConfigOptions.cmake	2015-11-11 12:54:34.000000000 +0100
+++ FreeRDP-be8f8f72387e7878717b6f04c9a87f999449d20d/cmake/ConfigOptions.cmake	2015-11-15 11:40:13.639673908 +0100
@@ -7,6 +7,8 @@
 elseif((CMAKE_SYSTEM_PROCESSOR MATCHES "i386") AND (CMAKE_SIZEOF_VOID_P EQUAL 8) AND (APPLE))
 	# Mac is weird like that.
 	set(TARGET_ARCH "x64")
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")
+	set(TARGET_ARCH "AARCH64")
 elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^arm*")
 	set(TARGET_ARCH "ARM")
 elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "sparc")
@@ -23,6 +25,9 @@
 	option(WITH_SSE2 "Enable SSE2 optimization." OFF)
 endif()
 
+if(TARGET_ARCH MATCHES "AARCH64")
+endif()
+
 if(TARGET_ARCH MATCHES "ARM")
 	if (NOT DEFINED WITH_NEON)
 		option(WITH_NEON "Enable NEON optimization." ON)
