diff -Naur orig/configure.in patched/configure.in
--- orig/configure.in	2005-09-11 09:14:04.000000000 +0200
+++ patched/configure.in	2021-12-29 22:17:28.562363930 +0100
@@ -94,19 +94,8 @@
 
 dnl Imlib2 detection
 
-AC_PATH_GENERIC(imlib2, , [
-    AC_SUBST(IMLIB_LIBS)
-   AC_SUBST(IMLIB_CXXFLAGS) ],
-  AC_MSG_ERROR(Cannot find imlib2: Is imlib2-config in the path?
-               You need Imlib2 to build Idesk.  Verify that you have Imlib2-dev))
-
-dnl the above doesn't work for some reason :/
-IMLIB_LIBS=`imlib2-config --libs`
-IMLIB_CFLAGS=`imlib2-config --cflags`
-AC_SUBST(IMLIB_LIBS)
-AC_SUBST(IMLIB_CXXFLAGS)
-
-CXXFLAGS="$CXXFLAGS $IMLIB_CXXFLAGS"
+PKG_CHECK_MODULES([IMLIB],[imlib2])
+CXXFLAGS="$CXXFLAGS $IMLIB_CFLAGS"
 LIBS="$LIBS $IMLIB_LIBS"
 
 
