diff -Naur orig/examples/default.lnk mod/examples/default.lnk
--- orig/examples/default.lnk	2005-09-04 07:51:34.000000000 +0200
+++ mod/examples/default.lnk	2014-12-26 20:52:07.605494370 +0100
@@ -1,7 +1,7 @@
 table Icon
   Caption: Idesk
   Command: /usr/bin/Xdialog --msgbox "IDESK -- Icon Desktop Manager \n\n\n by FixXxeR \n\n\n 2005" 15 30 
-  Icon: /usr/local/share/idesk/folder_home.xpm
+  Icon: /usr/share/idesk/folder_home.xpm
   Width: 48
   Height: 48
   X: 30
