diff -Nur orig/base/Behavior.cpp mod/base/Behavior.cpp
--- orig/base/Behavior.cpp	2020-12-18 17:34:55.000000000 +0100
+++ mod/base/Behavior.cpp	2020-12-20 20:45:23.814932442 +0100
@@ -6,6 +6,7 @@
     email                : henqvist@excite.com
  ***************************************************************************/
 
+#include <cstddef>
 #include "Private.h"
 #include "Behavior.h"
 #include "Group.h"
diff -Nur orig/base/BehaviorVisitor.cpp mod/base/BehaviorVisitor.cpp
--- orig/base/BehaviorVisitor.cpp	2020-12-18 17:34:55.000000000 +0100
+++ mod/base/BehaviorVisitor.cpp	2020-12-20 20:46:08.594866409 +0100
@@ -5,6 +5,7 @@
     copyright            : (C) 2000 by Henrik Enqvist
     email                : henqvist@excite.com
  ***************************************************************************/
+#include <cstddef>
 #include "Private.h"
 #include "BehaviorVisitor.h"
 #include "Behavior.h"
diff -Nur orig/base/SignalSender.cpp mod/base/SignalSender.cpp
--- orig/base/SignalSender.cpp	2020-12-18 17:34:55.000000000 +0100
+++ mod/base/SignalSender.cpp	2020-12-20 20:46:25.602841281 +0100
@@ -7,6 +7,7 @@
  ***************************************************************************/
 
 
+#include <cstddef>
 #include "Private.h"
 #include "SignalSender.h"
 #include "Group.h"
