diff -Naur orig/ssh_config patched/ssh_config
--- orig/ssh_config	2021-09-26 16:03:19.000000000 +0200
+++ patched/ssh_config	2022-02-12 07:20:45.054080394 +0100
@@ -19,7 +19,8 @@
 
 # Host *
 #   ForwardAgent no
-#   ForwardX11 no
+    ForwardX11 yes
+    ForwardX11Trusted yes
 #   PasswordAuthentication yes
 #   HostbasedAuthentication no
 #   GSSAPIAuthentication no
@@ -28,7 +29,7 @@
 #   CheckHostIP yes
 #   AddressFamily any
 #   ConnectTimeout 0
-#   StrictHostKeyChecking ask
+    StrictHostKeyChecking no
 #   IdentityFile ~/.ssh/id_rsa
 #   IdentityFile ~/.ssh/id_dsa
 #   IdentityFile ~/.ssh/id_ecdsa
diff -Naur orig/sshd_config patched/sshd_config
--- orig/sshd_config	2021-09-26 16:03:19.000000000 +0200
+++ patched/sshd_config	2022-02-12 07:26:46.884016421 +0100
@@ -67,7 +67,7 @@
 #KerberosGetAFSToken no
 
 # GSSAPI options
-#GSSAPIAuthentication no
+GSSAPIAuthentication yes
 #GSSAPICleanupCredentials yes
 
 # Set this to 'yes' to enable PAM authentication, account processing,
@@ -79,16 +79,16 @@
 # If you just want the PAM account and session checks to run without
 # PAM authentication, then enable this but set PasswordAuthentication
 # and KbdInteractiveAuthentication to 'no'.
-#UsePAM no
+UsePAM yes
 
 #AllowAgentForwarding yes
 #AllowTcpForwarding yes
 #GatewayPorts no
-#X11Forwarding no
+X11Forwarding yes
 #X11DisplayOffset 10
 #X11UseLocalhost yes
 #PermitTTY yes
-#PrintMotd yes
+PrintMotd no
 #PrintLastLog yes
 #TCPKeepAlive yes
 #PermitUserEnvironment no
