diff -Naurd a/src/usr/share/applications/diffuse.desktop b/src/usr/share/applications/diffuse.desktop
--- a/src/usr/share/applications/diffuse.desktop	2011-07-13 11:18:33.000000000 +0200
+++ b/src/usr/share/applications/diffuse.desktop	2014-05-05 22:05:25.000000000 +0200
@@ -10,6 +10,7 @@
 Comment=Graphical tool for merging and comparing text files
 Comment[de]=Werzeug zum Mischen und Vergleichen von Textdateien
 Comment[es]=Un utillaje para juntar y comparar archivos de texto
+Comment[hu]=Text fájlok összehasonlítása, összefésülése
 Comment[it]=Strumento grafico per l'unione e il confronto di file
 Comment[ja]=テキストファイルをマージして比較するためのグラフィックツール
 Comment[ko]=텍스트 파일을 병합하고 비교하기위한 도구
