diff -Nur orig/etc/lua.pc mod/etc/lua.pc
--- orig/etc/lua.pc	2012-02-10 10:50:30.000000000 +0100
+++ mod/etc/lua.pc	2016-10-21 08:56:25.675226371 +0200
@@ -8,7 +8,7 @@
 R= 5.1.5
 
 # grep '^INSTALL_.*=' ../Makefile | sed 's/INSTALL_TOP/prefix/'
-prefix= /usr/local
+prefix= /usr
 INSTALL_BIN= ${prefix}/bin
 INSTALL_INC= ${prefix}/include
 INSTALL_LIB= ${prefix}/lib
@@ -19,13 +19,13 @@
 # canonical vars
 exec_prefix=${prefix}
 libdir=${exec_prefix}/lib
-includedir=${prefix}/include
+includedir=${prefix}/include/lua${V}
 
 Name: Lua
 Description: An Extensible Extension Language
 Version: ${R}
 Requires: 
-Libs: -L${libdir} -llua -lm
+Libs: -L${libdir} -llua${V} -lm
 Cflags: -I${includedir}
 
 # (end of lua.pc)
