diff -Nur orig/gparted.desktop mod/gparted.desktop
--- orig/gparted.desktop	2017-10-10 18:24:10.000000000 +0200
+++ mod/gparted.desktop	2017-10-16 10:01:40.421864248 +0200
@@ -229,7 +229,7 @@
 Comment[zh_CN]=创建、重新组织或删除分区
 Comment[zh_HK]=建立、編輯或刪除分割區
 Comment[zh_TW]=建立、編輯或刪除分割區
-Exec=/usr/local/bin/gparted %f
+Exec=gparted-start %f
 Icon=gparted
 Terminal=false
 Type=Application
diff -Nur orig/gparted.desktop.in.in mod/gparted.desktop.in.in
--- orig/gparted.desktop.in.in	2017-10-03 18:52:19.000000000 +0200
+++ mod/gparted.desktop.in.in	2017-10-16 10:03:32.477743589 +0200
@@ -3,7 +3,7 @@
 _GenericName=Partition Editor
 _X-GNOME-FullName=GParted Partition Editor
 _Comment=Create, reorganize, and delete partitions
-Exec=@bindir@/gparted %f
+Exec=gparted-start %f
 Icon=gparted
 Terminal=false
 Type=Application
