--- a/Makefile.in.orig	2014-03-30 16:52:00.000000000 +0200
+++ a/Makefile.in	2014-03-30 16:53:24.000000000 +0200
@@ -452,27 +452,7 @@
 
 distclean-libtool:
 	-rm -f libtool config.lt
-install-libgfbgraphdocDATA: $(libgfbgraphdoc_DATA)
-	@$(NORMAL_INSTALL)
-	@list='$(libgfbgraphdoc_DATA)'; test -n "$(libgfbgraphdocdir)" || list=; \
-	if test -n "$$list"; then \
-	  echo " $(MKDIR_P) '$(DESTDIR)$(libgfbgraphdocdir)'"; \
-	  $(MKDIR_P) "$(DESTDIR)$(libgfbgraphdocdir)" || exit 1; \
-	fi; \
-	for p in $$list; do \
-	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
-	  echo "$$d$$p"; \
-	done | $(am__base_list) | \
-	while read files; do \
-	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(libgfbgraphdocdir)'"; \
-	  $(INSTALL_DATA) $$files "$(DESTDIR)$(libgfbgraphdocdir)" || exit $$?; \
-	done
 
-uninstall-libgfbgraphdocDATA:
-	@$(NORMAL_UNINSTALL)
-	@list='$(libgfbgraphdoc_DATA)'; test -n "$(libgfbgraphdocdir)" || list=; \
-	files=`for p in $$list; do echo $$p; done | sed -e 's|^.*/||'`; \
-	dir='$(DESTDIR)$(libgfbgraphdocdir)'; $(am__uninstall_files_from_dir)
 install-pkgconfigDATA: $(pkgconfig_DATA)
 	@$(NORMAL_INSTALL)
 	@list='$(pkgconfig_DATA)'; test -n "$(pkgconfigdir)" || list=; \
