diff -Naur orig/data/Makefile.am patched/data/Makefile.am
--- orig/data/Makefile.am	2014-03-10 09:57:48.000000000 +0100
+++ patched/data/Makefile.am	2015-08-02 16:18:25.287009849 +0200
@@ -31,10 +31,10 @@
 systemdservice_DATA     = $(systemdservice_in_files:.service.in=.service)
 
 $(systemdservice_DATA): $(systemdservice_in_files) Makefile
-	@sed -e "s|\@libexecdir\@|$(prefix)/lib/udisks|" $< > $@
+	@sed -e "s|\@libexecdir\@|$(libexecdir)|" $< > $@
 endif
 
-udevrulesdir = $(slashlibdir)/udev/rules.d
+udevrulesdir = $(libdir)/udev/rules.d
 udevrules_DATA = 80-udisks.rules
 
 pkgconfigdir = $(datadir)/pkgconfig
diff -Naur orig/data/Makefile.in patched/data/Makefile.in
--- orig/data/Makefile.in	2014-03-10 10:04:33.000000000 +0100
+++ patched/data/Makefile.in	2015-08-02 16:18:12.338945643 +0200
@@ -345,7 +345,7 @@
 systemdservice_in_files = systemd/udisks.service.in
 @HAVE_SYSTEMD_TRUE@systemdservicedir = $(systemdsystemunitdir)
 @HAVE_SYSTEMD_TRUE@systemdservice_DATA = $(systemdservice_in_files:.service.in=.service)
-udevrulesdir = $(slashlibdir)/udev/rules.d
+udevrulesdir = $(libdir)/udev/rules.d
 udevrules_DATA = 80-udisks.rules
 pkgconfigdir = $(datadir)/pkgconfig
 pkgconfig_DATA = udisks.pc
@@ -722,7 +722,7 @@
 	cp $< $@
 
 @HAVE_SYSTEMD_TRUE@$(systemdservice_DATA): $(systemdservice_in_files) Makefile
-@HAVE_SYSTEMD_TRUE@	@sed -e "s|\@libexecdir\@|$(prefix)/lib/udisks|" $< > $@
+@HAVE_SYSTEMD_TRUE@	@sed -e "s|\@libexecdir\@|$(libexecdir)|" $< > $@
 
 clean-local :
 	rm -f *~ $(service_DATA) $(dbusconf_DATA) $(systemdservice_DATA)
diff -Naur orig/src/probers/Makefile.am patched/src/probers/Makefile.am
--- orig/src/probers/Makefile.am	2014-03-10 09:57:48.000000000 +0100
+++ patched/src/probers/Makefile.am	2015-08-02 16:20:13.611547001 +0200
@@ -20,7 +20,7 @@
 # TODO: ideally move most of this to udev and/or util-linux
 #
 
-udevhelperdir = $(slashlibdir)/udev
+udevhelperdir = $(libdir)/udev
 udevhelper_PROGRAMS =	udisks-part-id 			\
 			udisks-dm-export 		\
 		      	udisks-probe-ata-smart 		\
diff -Naur orig/src/probers/Makefile.in patched/src/probers/Makefile.in
--- orig/src/probers/Makefile.in	2014-03-10 10:04:34.000000000 +0100
+++ patched/src/probers/Makefile.in	2015-08-02 16:20:46.127708240 +0200
@@ -395,7 +395,7 @@
 
 # TODO: ideally move most of this to udev and/or util-linux
 #
-udevhelperdir = $(slashlibdir)/udev
+udevhelperdir = $(libdir)/udev
 udisks_part_id_SOURCES = part-id.c
 udisks_part_id_CPPFLAGS = $(AM_CPPFLAGS) $(LIBUDEV_CFLAGS)
 udisks_part_id_LDADD = $(GLIB_LIBS) $(LIBUDEV_LIBS) $(top_builddir)/src/helpers/libpartutil.la
diff -Naur orig/tools/Makefile.am patched/tools/Makefile.am
--- orig/tools/Makefile.am	2014-03-10 09:57:48.000000000 +0100
+++ patched/tools/Makefile.am	2015-08-02 16:22:08.880118587 +0200
@@ -72,7 +72,7 @@
 
 CLEANFILES = $(BUILT_SOURCES)
 
-profiledir = $(sysconfdir)/profile.d
+profiledir = $(datadir)/bash-completion/completions
 profile_SCRIPTS = udisks-bash-completion.sh
 
 EXTRA_DIST = $(profile_SCRIPTS)
diff -Naur orig/tools/Makefile.in patched/tools/Makefile.in
--- orig/tools/Makefile.in	2014-03-10 10:04:34.000000000 +0100
+++ patched/tools/Makefile.in	2015-08-02 16:23:18.148462070 +0200
@@ -435,7 +435,7 @@
 
 # ----------------------------------------------------------------------------------------------------
 CLEANFILES = $(BUILT_SOURCES)
-profiledir = $(sysconfdir)/profile.d
+profiledir = $(datadir)/bash-completion/completions
 profile_SCRIPTS = udisks-bash-completion.sh
 EXTRA_DIST = $(profile_SCRIPTS)
 all: $(BUILT_SOURCES)
