diff -Naur orig/src/res/fwbuilder.desktop patched/src/res/fwbuilder.desktop
--- orig/src/res/fwbuilder.desktop	2019-05-30 23:40:23.000000000 +0200
+++ patched/src/res/fwbuilder.desktop	2019-05-31 00:39:24.637120740 +0200
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Name=Firewall Builder
 Comment=Design and Manage Firewall Rules
+Comment[hu]=Tűzfalbeállító program
 Icon=fwbuilder
 Categories=System;Settings;Security;Qt;
 Exec=fwbuilder
