diff -Naur orig/bootstrap/freeciv.desktop.in mod/bootstrap/freeciv.desktop.in
--- orig/bootstrap/freeciv.desktop.in	2015-08-15 13:19:26.000000000 +0200
+++ mod/bootstrap/freeciv.desktop.in	2015-08-27 21:37:32.115969766 +0200
@@ -5,10 +5,12 @@
 Name[fr]=Freeciv (gtk3)
 Name[nb]=Freeciv (gtk3)
 Name[pt]=Freeciv (gtk3)
+Name[hu]=Freeciv (gtk3)
 Comment=Turn-based strategy game inspired by the history of human civilization
 Comment[ca]=Joc d'estratègia inspirat en la història de la civilització humana
 Comment[da]=Strategispil inspireret af den menneskelige civilisations historie
 Comment[de]=Rundenbasiertes Strategiespiel, inspiriert durch die Geschichte der menschlichen Zivilisation
+Comment[hu]=Az emberi civilizáció története által inspirált körökre osztott stratégiai játék
 Comment[fi]=Ihmiskunnan historian inspiroima vuoropohjainen strategiapeli
 Comment[nb]=Strategispill inspirert av historien til menneskelig sivilisasjon
 Comment[pt]=Jogo de estratégia por turnos inspirado na História da civilização humana
diff -Naur orig/bootstrap/freeciv-server.desktop.in mod/bootstrap/freeciv-server.desktop.in
--- orig/bootstrap/freeciv-server.desktop.in	2015-08-15 13:19:26.000000000 +0200
+++ mod/bootstrap/freeciv-server.desktop.in	2015-08-27 21:39:12.371965782 +0200
@@ -5,11 +5,13 @@
 Name[fr]=Freeciv server
 Name[nb]=Freeciv server
 Name[pt]=Freeciv server
+Name[hu]=Freeciv server
 Comment=Turn-based strategy game inspired by the history of human civilization
 Comment[ca]=Joc d'estratègia inspirat en la història de la civilització humana
 Comment[da]=Strategispil inspireret af den menneskelige civilisations historie
 Comment[de]=Rundenbasiertes Strategiespiel, inspiriert durch die Geschichte der menschlichen Zivilisation
 Comment[fi]=Ihmiskunnan historian inspiroima vuoropohjainen strategiapeli
+Comment[hu]=Az emberi civilizáció története által inspirált körökre osztott stratégiai játék
 Comment[nb]=Strategispill inspirert av historien til menneskelig sivilisasjon
 Comment[pt]=Jogo de estratégia por turnos inspirado na História da civilização humana
 Comment[sv]=Turordningsbaserat strategispel inspirerat av den mänskliga historien
