diff -Nur orig/bootstrap/freeciv.desktop.in mod/bootstrap/freeciv.desktop.in
--- orig/bootstrap/freeciv.desktop.in	2018-03-24 13:16:43.000000000 +0100
+++ mod/bootstrap/freeciv.desktop.in	2018-06-11 21:27:48.580050898 +0200
@@ -10,6 +10,7 @@
 Comment[ca]=Joc d'estratègia inspirat en la història de la civilització humana
 Comment[da]=Strategispil inspireret af den menneskelige civilisations historie
 Comment[de]=Rundenbasiertes Strategiespiel, inspiriert durch die Geschichte der menschlichen Zivilisation
+Comment[hu]=Az emberi civilizáció története által inspirált körökre osztott stratégiai játék
 Comment[fi]=Ihmiskunnan historian inspiroima vuoropohjainen strategiapeli
 Comment[nb]=Strategispill inspirert av historien til menneskelig sivilisasjon
 Comment[pt]=Jogo de estratégia por turnos inspirado na História da civilização humana
diff -Nur orig/bootstrap/freeciv-server.desktop.in mod/bootstrap/freeciv-server.desktop.in
--- orig/bootstrap/freeciv-server.desktop.in	2018-03-24 13:16:43.000000000 +0100
+++ mod/bootstrap/freeciv-server.desktop.in	2018-06-11 21:28:34.507982556 +0200
@@ -10,6 +10,7 @@
 Comment[ca]=Joc d'estratègia inspirat en la història de la civilització humana
 Comment[da]=Strategispil inspireret af den menneskelige civilisations historie
 Comment[de]=Rundenbasiertes Strategiespiel, inspiriert durch die Geschichte der menschlichen Zivilisation
+Comment[hu]=Az emberi civilizáció története által inspirált körökre osztott stratégiai játék
 Comment[fi]=Ihmiskunnan historian inspiroima vuoropohjainen strategiapeli
 Comment[nb]=Strategispill inspirert av historien til menneskelig sivilisasjon
 Comment[pt]=Jogo de estratégia por turnos inspirado na História da civilização humana
