diff -Nur orig/Makefile mod/Makefile
--- orig/Makefile	2018-03-20 21:29:59.000000000 +0100
+++ mod/Makefile	2018-06-04 10:41:56.559777744 +0200
@@ -47,7 +47,6 @@
 	mkdir -p $(DESTDIR)$(MANPREFIX)/man1
 	sed "s/VERSION/$(VERSION)/g" < st.1 > $(DESTDIR)$(MANPREFIX)/man1/st.1
 	chmod 644 $(DESTDIR)$(MANPREFIX)/man1/st.1
-	tic -sx st.info
 	@echo Please see the README file regarding the terminfo entry of st.
 
 uninstall:
