diff -urN squashfs4.2.orig/squashfs-tools/Makefile squashfs4.2/squashfs-tools/Makefile
--- squashfs4.2.orig/squashfs-tools/Makefile	2011-02-28 21:04:15.000000000 +0100
+++ squashfs4.2/squashfs-tools/Makefile	2012-04-20 14:44:08.634870165 +0200
@@ -26,7 +26,7 @@
 # To build using XZ Utils liblzma - install the library and uncomment
 # the XZ_SUPPORT line below.
 #
-#XZ_SUPPORT = 1
+XZ_SUPPORT = 1
 
 
 ############ Building LZO support ##############
@@ -37,7 +37,7 @@
 # LZO_SUPPORT line below. If needed, uncomment and set LZO_DIR to the
 # installation prefix.
 #
-#LZO_SUPPORT = 1
+LZO_SUPPORT = 1
 #LZO_DIR = /usr/local
 
 ########### Building LZMA support #############
