diff -Naur orig/squashfs-tools/Makefile patched/squashfs-tools/Makefile
--- orig/squashfs-tools/Makefile	2021-07-22 22:11:24.000000000 +0200
+++ patched/squashfs-tools/Makefile	2021-07-26 08:59:37.547009310 +0200
@@ -34,7 +34,7 @@
 # To build install the library and uncomment
 # the XZ_SUPPORT line below.
 #
-#XZ_SUPPORT = 1
+XZ_SUPPORT = 1
 
 
 ############ Building LZO support ##############
@@ -48,7 +48,7 @@
 # To build install the library and uncomment
 # the LZO_SUPPORT line below.
 #
-#LZO_SUPPORT = 1
+LZO_SUPPORT = 1
 
 
 ########### Building LZ4 support #############
@@ -64,7 +64,7 @@
 # To build install and uncomment
 # the LZ4_SUPPORT line below.
 #
-#LZ4_SUPPORT = 1
+LZ4_SUPPORT = 1
 
 
 ########### Building ZSTD support ############
@@ -80,7 +80,7 @@
 # To build install the library and uncomment
 # the ZSTD_SUPPORT line below.
 #
-#ZSTD_SUPPORT = 1
+ZSTD_SUPPORT = 1
 
 
 ######## Specifying default compression ########
