diff -Naur e2fsprogs-1.43.5-orig/e2fsck/unix.c e2fsprogs-1.43.5/e2fsck/unix.c
--- e2fsprogs-1.43.5-orig/e2fsck/unix.c	2017-07-23 05:03:19.000000000 +0200
+++ e2fsprogs-1.43.5/e2fsck/unix.c	2017-08-23 19:41:40.820287874 +0200
@@ -1929,8 +1929,6 @@
 #endif
 		}
 		if (ctx->mount_flags & EXT2_MF_ISROOT) {
-			log_out(ctx, _("%s: ***** REBOOT SYSTEM *****\n"),
-				ctx->device_name);
 			exit_value |= FSCK_REBOOT;
 		}
 	}
