diff -Naur e2fsprogs-1.41.9-orig/lib/ext2fs/ext2_fs.h e2fsprogs-1.41.9/lib/ext2fs/ext2_fs.h
--- e2fsprogs-1.41.9-orig/lib/ext2fs/ext2_fs.h	2009-08-23 04:44:35.000000000 +0200
+++ e2fsprogs-1.41.9/lib/ext2fs/ext2_fs.h	2009-09-04 13:04:08.000000000 +0200
@@ -481,7 +481,7 @@
 /*
  * Maximal mount counts between two filesystem checks
  */
-#define EXT2_DFL_MAX_MNT_COUNT		20	/* Allow 20 mounts */
+#define EXT2_DFL_MAX_MNT_COUNT		100	/* Allow 100 mounts */
 #define EXT2_DFL_CHECKINTERVAL		0	/* Don't use interval check */
 
 /*
