diff -Naur e2fsprogs-1.41.9-orig/MCONFIG.in e2fsprogs-1.41.9/MCONFIG.in
--- e2fsprogs-1.41.9-orig/MCONFIG.in	2009-08-13 03:39:57.000000000 +0200
+++ e2fsprogs-1.41.9/MCONFIG.in	2009-09-04 13:08:46.000000000 +0200
@@ -161,16 +161,16 @@
 #
 BINGRP=		bin
 BINOWN=		bin
-BINMODE=	555
+BINMODE=	755
 INCGRP=		bin
 INCOWN=		bin
-INCMODE=	444
+INCMODE=	644
 LIBOWN=		bin
 LIBGRP=		bin
-LIBMODE=	444
+LIBMODE=	644
 MANGRP=		bin
 MANOWN=		bin
-MANMODE=	444
+MANMODE=	644
 
 #
 # Autoconf magic...
