diff -Naur orig/data/org.gnome.Evince.desktop.in.in patched/data/org.gnome.Evince.desktop.in.in
--- orig/data/org.gnome.Evince.desktop.in.in	2019-01-31 04:03:58.000000000 +0100
+++ patched/data/org.gnome.Evince.desktop.in.in	2019-03-17 01:33:37.558460065 +0100
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Name=Document Viewer
+Name=Evince
 Comment=View multi-page documents
 # Translators: Search terms to find this application. Do NOT translate or localize the semicolons! The list MUST also end with a semicolon!
 Keywords=pdf;ps;postscript;dvi;xps;djvu;tiff;document;presentation;viewer;
