diff -Naur a/data/org.gnome.Evince.desktop.in.in b/data/org.gnome.Evince.desktop.in.in
--- a/data/org.gnome.Evince.desktop.in.in	2022-05-23 03:21:13.000000000 +0200
+++ b/data/org.gnome.Evince.desktop.in.in	2022-05-24 18:20:15.854910768 +0200
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Name=Document Viewer
+Name=Evince
 Comment=View multi-page documents
 # Translators: Search terms to find this application. Do NOT translate or localize the semicolons! The list MUST also end with a semicolon!
 Keywords=pdf;ps;postscript;dvi;xps;djvu;tiff;document;presentation;viewer;evince;
