diff -Naur orig/data/evince.desktop.in.in mod/data/evince.desktop.in.in
--- orig/data/evince.desktop.in.in	2014-11-29 10:40:15.000000000 +0100
+++ mod/data/evince.desktop.in.in	2016-02-02 17:47:13.685837228 +0100
@@ -1,5 +1,5 @@
 [Desktop Entry]
-_Name=Document Viewer
+_Name=Evince
 _Comment=View multi-page documents
 _Keywords=pdf;ps;postscript;dvi;xps;djvu;tiff;document;presentation;
 TryExec=evince
