diff -Nur orig/res/conf mod/res/conf
--- orig/res/conf	2021-09-30 08:11:34.000000000 +0200
+++ mod/res/conf	2021-11-26 15:56:43.992153875 +0100
@@ -26,7 +26,7 @@
 XINITRC_LAUNCH=false
 
 # Prints available WM/DE each on new line instead of printing on single line.
-VERTICAL_SELECTION=false
+VERTICAL_SELECTION=true
 
 # Defines the way, how is logging handled. Possible values are "default", "appending" or "disabled".
 #LOGGING=default
