diff -Naur orig/src/meson.build patched/src/meson.build
--- orig/src/meson.build	2019-10-26 23:30:39.000000000 +0200
+++ patched/src/meson.build	2019-11-26 10:50:39.624617942 +0100
@@ -1,6 +1,8 @@
 pkg = import('pkgconfig')
 pango_dep = dependency('pangoft2')
 cairo_dep = dependency('pangocairo')
+glib_dep = dependency('glib-2.0')
+gobject_dep = dependency('gobject-2.0')
 
 paps_config = configuration_data()
 
@@ -31,5 +33,5 @@
                   ['paps.c'],
                   c_args: ['-DHAVE_CONFIG_H'],
                   include_directories: incs,
-                  dependencies : [pango_dep, cairo_dep],
+                  dependencies : [pango_dep, cairo_dep, glib_dep, gobject_dep],
                   install: true)
diff -Naur orig/src/paps.c patched/src/paps.c
--- orig/src/paps.c	2019-10-26 23:30:39.000000000 +0200
+++ patched/src/paps.c	2019-11-26 10:43:45.183344398 +0100
@@ -34,11 +34,18 @@
 #include <string.h>
 #include <time.h>
 #include <locale.h>
-#include <math.h>
-#include <wchar.h>
 #include <libgen.h>
 #include <config.h>
 
+#define _XOPEN_SOURCE /* for wcwidth */
+#include <wchar.h>
+int wcwidth(wchar_t c);
+
+#ifndef M_PI
+#define __USE_XOPEN
+#endif
+#include <math.h>
+
 #if ENABLE_NLS
 #include <libintl.h>
 
