diff -Naur orig/cmake/modules/ECMQt4To5Porting.cmake patched/cmake/modules/ECMQt4To5Porting.cmake
--- orig/cmake/modules/ECMQt4To5Porting.cmake	2021-02-06 19:14:16.000000000 +0100
+++ patched/cmake/modules/ECMQt4To5Porting.cmake	2021-02-17 12:51:30.598563180 +0100
@@ -160,7 +160,6 @@
   DBus
   Network
   Test
-  Designer
   Concurrent
   Xml
   UiTools
diff -Naur orig/cmake/modules/FindQt5Transitional.cmake patched/cmake/modules/FindQt5Transitional.cmake
--- orig/cmake/modules/FindQt5Transitional.cmake	2021-02-06 19:14:16.000000000 +0100
+++ patched/cmake/modules/FindQt5Transitional.cmake	2021-02-17 12:51:30.598563180 +0100
@@ -6,7 +6,6 @@
       Core
       Gui
       DBus
-      Designer
       Declarative
       Script
       ScriptTools
diff -Naur orig/cmake/modules/KDE4Macros.cmake patched/cmake/modules/KDE4Macros.cmake
--- orig/cmake/modules/KDE4Macros.cmake	2021-02-06 19:14:16.000000000 +0100
+++ patched/cmake/modules/KDE4Macros.cmake	2021-02-17 12:51:30.598563180 +0100
@@ -486,7 +486,6 @@
 include("${CMAKE_CURRENT_LIST_DIR}/KDECoreMacros.cmake")
 include("${CMAKE_CURRENT_LIST_DIR}/KDEUIMacros.cmake")
 include("${KF5DocTools_DIR}/KF5DocToolsMacros.cmake")
-include("${KF5DesignerPlugin_DIR}/KF5DesignerPluginMacros.cmake")
 
 # functions deprecated, they still should work
 macro (KDE4_AUTH_INSTALL_ACTIONS)
diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2021-02-06 19:14:16.000000000 +0100
+++ patched/CMakeLists.txt	2021-02-17 12:51:30.598563180 +0100
@@ -20,13 +20,12 @@
 include(CMakeFindFrameworks)
 
 set(REQUIRED_QT_VERSION 5.15.2)
-find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Network Widgets DBus Test Svg PrintSupport Designer)
+find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Network Widgets DBus Test Svg PrintSupport)
 
 find_package(KF5Completion ${KF_DEP_VERSION} REQUIRED)
 find_package(KF5Config ${KF_DEP_VERSION} REQUIRED)
 find_package(KF5ConfigWidgets ${KF_DEP_VERSION} REQUIRED)
 find_package(KF5Crash ${KF_DEP_VERSION} REQUIRED)
-find_package(KF5DesignerPlugin ${KF_DEP_VERSION} REQUIRED)
 find_package(KF5GlobalAccel ${KF_DEP_VERSION} REQUIRED)
 find_package(KF5DocTools ${KF_DEP_VERSION} REQUIRED)
 find_package(KF5Emoticons ${KF_DEP_VERSION} REQUIRED)
diff -Naur orig/KF5KDELibs4SupportConfig.cmake.in patched/KF5KDELibs4SupportConfig.cmake.in
--- orig/KF5KDELibs4SupportConfig.cmake.in	2021-02-06 19:14:16.000000000 +0100
+++ patched/KF5KDELibs4SupportConfig.cmake.in	2021-02-17 12:52:08.258625270 +0100
@@ -11,7 +11,6 @@
 find_dependency(KF5ConfigWidgets "@KF_DEP_VERSION@")
 find_dependency(KF5CoreAddons "@KF_DEP_VERSION@")
 find_dependency(KF5Crash "@KF_DEP_VERSION@")
-find_dependency(KF5DesignerPlugin "@KF_DEP_VERSION@")
 find_dependency(KF5DocTools "@KF_DEP_VERSION@")
 find_dependency(KF5Emoticons "@KF_DEP_VERSION@")
 find_dependency(KF5GuiAddons "@KF_DEP_VERSION@")
diff -Naur orig/src/CMakeLists.txt patched/src/CMakeLists.txt
--- orig/src/CMakeLists.txt	2021-02-06 19:14:16.000000000 +0100
+++ patched/src/CMakeLists.txt	2021-02-17 12:51:30.598563180 +0100
@@ -371,29 +371,6 @@
 )
 
 
-########## Designer plugin
-
-kf5designerplugin_add_plugin(kf5deprecatedwidgets
-    kdedeprecated.widgets
-    kdedeprecated.qrc
-    classpreviews.cpp
-)
-
-target_link_libraries(kf5deprecatedwidgets KF5::KDELibs4Support
-                                    Qt5::Widgets
-                                    Qt5::Designer
-                                    KF5::ConfigCore
-                                    KF5::Completion
-                                    KF5::WidgetsAddons
-                                    KF5::Auth
-                                    )
-
-install(TARGETS kf5deprecatedwidgets DESTINATION ${KDE_INSTALL_QTPLUGINDIR}/designer)
-
-##########
-
-add_subdirectory(pics)
-
 install(TARGETS KF5KDELibs4Support
         EXPORT KF5KDELibs4SupportTargets
         ${KF5_INSTALL_TARGETS_DEFAULT_ARGS})
diff -Naur orig/src/kdebug.areas patched/src/kdebug.areas
--- orig/src/kdebug.areas	2021-02-06 19:14:16.000000000 +0100
+++ patched/src/kdebug.areas	2021-02-17 12:51:30.598563180 +0100
@@ -379,7 +379,6 @@
 #9034        kdevelop (filecreate part)
 9035        kdevelop (konsole part)
 9037        kdevelop (make builder)
-9038        kdevelop (qtdesigner)
 9039        kdevelop (qmake builder)
 9040        kdevelop (cpp parser)
 9041        kdevelop (cpp duchain)
diff -Naur orig/src/KDEUIMacros.cmake patched/src/KDEUIMacros.cmake
--- orig/src/KDEUIMacros.cmake	2021-02-06 19:14:16.000000000 +0100
+++ patched/src/KDEUIMacros.cmake	2021-02-17 12:51:30.598563180 +0100
@@ -163,31 +163,6 @@
 endmacro (KDE4_INSTALL_ICONS)
 
 
-set(KDEWIDGETS_MAKEKDEWIDGETS_EXECUTABLE ${KF5DESIGNERPLUGIN_GENERATOR_EXECUTABLE})
-set(_KDE4_MAKEKDEWIDGETS_DEP ${_KF5DESIGNERPLUGIN_GENERATOR_EXECUTABLE_DEP})
-macro (KDE4_ADD_WIDGET_FILES _sources)
-   foreach (_current_FILE ${ARGN})
-
-      get_filename_component(_input ${_current_FILE} ABSOLUTE)
-      get_filename_component(_basename ${_input} NAME_WE)
-      set(_source ${CMAKE_CURRENT_BINARY_DIR}/${_basename}widgets.cpp)
-      set(_moc ${CMAKE_CURRENT_BINARY_DIR}/${_basename}widgets.moc)
-
-      # create source file from the .widgets file
-      add_custom_command(OUTPUT ${_source}
-        COMMAND ${KDEWIDGETS_MAKEKDEWIDGETS_EXECUTABLE}
-        ARGS -o ${_source} ${_input}
-        MAIN_DEPENDENCY ${_input} DEPENDS ${_KDE4_MAKEKDEWIDGETS_DEP})
-
-      # create moc file
-      qt5_generate_moc(${_source} ${_moc} )
-
-      list(APPEND ${_sources} ${_source} ${_moc})
-
-   endforeach (_current_FILE)
-
-endmacro (KDE4_ADD_WIDGET_FILES)
-
 # adds application icon to target source list
 # 'appsources' - the sources of the application
 # 'pngfiles' - specifies the list of icon files
