diff -Naur orig/src/gtk-recordmydesktop.desktop mod/src/gtk-recordmydesktop.desktop
--- orig/src/gtk-recordmydesktop.desktop	2008-11-23 16:09:17.000000000 +0100
+++ mod/src/gtk-recordmydesktop.desktop	2016-04-12 16:59:31.347873803 +0200
@@ -1,12 +1,10 @@
 [Desktop Entry]
-Version=1.0
-Encoding=UTF-8
 Name=gtk-recordMyDesktop
 Comment=Frontend for recordMyDesktop
 Comment[fr]=Interface pour recordmydesktop
 Exec=gtk-recordMyDesktop
 Terminal=false
 Type=Application
-Icon=gtk-recordmydesktop.png
+Icon=gtk-recordmydesktop
 Categories=AudioVideo;Recorder;
 StartupNotify=false
