From 0019d2c7f464aeca11fe606649b657ac0a5f037f Mon Sep 17 00:00:00 2001
From: Andre Miranda <andreldm@xfce.org>
Date: Sat, 19 Oct 2019 18:54:49 -0300
Subject: [PATCH] Fix build with panel 4.15

---
 panel-plugin/datetime.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/panel-plugin/datetime.c b/panel-plugin/datetime.c
index f4921b3..f4e0827 100644
--- a/panel-plugin/datetime.c
+++ b/panel-plugin/datetime.c
@@ -31,8 +31,7 @@
 /* xfce includes */
 #include <libxfce4ui/libxfce4ui.h>
 #include <libxfce4util/libxfce4util.h>
-#include <libxfce4panel/xfce-panel-plugin.h>
-#include <libxfce4panel/xfce-panel-convenience.h>
+#include <libxfce4panel/libxfce4panel.h>
 
 #include "datetime.h"
 #include "datetime-dialog.h"
@@ -679,7 +678,7 @@ static void datetime_create_widget(t_datetime * datetime)
   orientation = xfce_panel_plugin_get_orientation(datetime->plugin);
 
   /* create button */
-  datetime->button = xfce_create_panel_toggle_button();
+  datetime->button = xfce_panel_create_toggle_button();
   gtk_widget_show(datetime->button);
 
   /* create a box which can be easily adapted to the panel orientation */
@@ -718,7 +717,7 @@ static t_datetime * datetime_new(XfcePanelPlugin *plugin)
   DBG("Starting datetime panel plugin");
 
   /* alloc and clear mem */
-  datetime = panel_slice_new0 (t_datetime);
+  datetime = g_slice_new0 (t_datetime);
 
   /* store plugin reference */
   datetime->plugin = plugin;
@@ -755,7 +754,7 @@ static void datetime_free(XfcePanelPlugin *plugin, t_datetime *datetime)
   g_free(datetime->date_format);
   g_free(datetime->time_format);
 
-  panel_slice_free(t_datetime, datetime);
+  g_slice_free(t_datetime, datetime);
 }
 
 /*
