diff -up hatari-2.0.0/src/cpu/sysdeps.h.s390 hatari-2.0.0/src/cpu/sysdeps.h
--- hatari-2.0.0/src/cpu/sysdeps.h.s390	2016-12-20 10:33:31.915948495 +0100
+++ hatari-2.0.0/src/cpu/sysdeps.h	2016-12-20 10:33:58.705948495 +0100
@@ -56,6 +56,8 @@ using namespace std;
 #define CPU_arm 1
 #elif defined(__powerpc__) || defined(_M_PPC) || defined(__ppc__) || defined(__ppc64__)
 #define CPU_powerpc 1
+#elif defined(__s390__)
+#define CPU_s390 1
 #else
 #error unrecognized CPU type
 #endif
