diff -Nur orig/python-ui/hatariui.desktop mod/python-ui/hatariui.desktop
--- orig/python-ui/hatariui.desktop	2016-11-03 22:25:43.000000000 +0100
+++ mod/python-ui/hatariui.desktop	2017-01-31 19:51:58.782138797 +0100
@@ -1,7 +1,8 @@
 [Desktop Entry]
-Encoding=UTF-8
-Version=1.0
 Type=Application
 Name=Hatari UI
+Comment=External user interface for Hatari
+Comment[hu]=Hatari külső kezelő felület
 Exec=hatariui
 Icon=hatari
+Categories=Game;Emulator;
diff -Nur orig/share/applications/hatari.desktop mod/share/applications/hatari.desktop
--- orig/share/applications/hatari.desktop	2016-11-03 22:25:43.000000000 +0100
+++ mod/share/applications/hatari.desktop	2017-01-31 19:53:20.237999517 +0100
@@ -1,11 +1,11 @@
 [Desktop Entry]
-Version=1.0
 Type=Application
 Name=Hatari
 GenericName=ST emulator
-Comment=Run old ST/STE/TT/Falcon software
+Comment=Run old Atari ST/STE/TT/Falcon software
+Comment[hu]=régi Atari ST/STE/TT/Falcon programok futtatása
 Exec=hatari %f
 Icon=hatari
-MimeType=application/x-st-disk-image;application/vnd.msa-disk-image;application/vnd.fastcopy-disk-image;application/x-stx-disk-image
+MimeType=application/x-st-disk-image;application/vnd.msa-disk-image;application/vnd.fastcopy-disk-image;application/x-stx-disk-image;
 Categories=Game;Emulator;
 Terminal=false
