diff -Naur vcdimager-0.7.20.orig/docs/vcdimager.info vcdimager-0.7.20/docs/vcdimager.info
--- vcdimager-0.7.20.orig/docs/vcdimager.info	2004-02-20 07:21:16.000000000 +0100
+++ vcdimager-0.7.20/docs/vcdimager.info	2004-07-05 13:58:29.580562627 +0200
@@ -21,7 +21,6 @@
 START-INFO-DIR-ENTRY
 * VCDImager: (vcdimager).                       GNU Video CD imaging utilities
 * vcdimager: (vcdimager)vcdimager.              Video CD simple formatter
-* vcd-info:  (vcdimager)vcd-info.               Video CD diagnostic tool
 * cdxa2mpeg: (vcdimager)cdxa2mpeg.              Strip RIFF/CD-XA container
 * vcdxminfo: (vcdimager)vcdxminfo.              Display MPEG stream properties
 * vcdxgen: (vcdimager)vcdxgen.                  Video CD XML template generator
