diff -Naurdp a/gtk/gtksettings.c b/gtk/gtksettings.c
--- a/gtk/gtksettings.c	2005-09-25 19:54:18.000000000 +0200
+++ b/gtk/gtksettings.c	2005-09-25 19:54:55.000000000 +0200
@@ -283,7 +283,7 @@ gtk_settings_class_init (GtkSettingsClas
                                              g_param_spec_string ("gtk-font-name",
 								   P_("Font Name"),
 								   P_("Name of default font to use"),
-								  "Sans 10",
+								  "Nimbus Sans L, 10",
 								  GTK_PARAM_READWRITE),
                                              NULL);
   g_assert (result == PROP_FONT_NAME);
