diff -Naur orig/mail/app/profile/all-thunderbird.js patched/mail/app/profile/all-thunderbird.js
--- orig/mail/app/profile/all-thunderbird.js	2017-03-22 00:14:45.134741323 +0100
+++ patched/mail/app/profile/all-thunderbird.js	2017-04-04 01:45:03.629032826 +0200
@@ -11,6 +11,9 @@
 #endif
 #endif
 
+pref("intl.locale.matchOS", true);
+pref("shell.checkDefaultApps", 0);
+
 pref("general.useragent.locale", "@AB_CD@");
 pref("general.skins.selectedSkin", "classic/1.0");
 
@@ -84,12 +87,12 @@
 pref("app.update.certs.2.commonName", "aus5.mozilla.org");
 
 // Whether or not app updates are enabled
-pref("app.update.enabled", true);
+pref("app.update.enabled", false);
 
 // If set to true, the Update Service will automatically download updates when
 // app updates are enabled per the app.update.enabled preference and if the user
 // can apply updates.
-pref("app.update.auto", true);
+pref("app.update.auto", false);
 
 // If set to true, the Update Service will present no UI for any event.
 pref("app.update.silent", false);
@@ -151,7 +154,8 @@
 // Disable add-ons installed into the shared user and shared system areas by
 // default. This does not include the application directory. See the SCOPE
 // constants in AddonManager.jsm for values to use here
-pref("extensions.autoDisableScopes", 15);
+pref("extensions.autoDisableScopes", 0);
+pref("extensions.shownSelectionUI", true);
 
 // Preferences for AMO integration
 pref("extensions.getAddons.cache.enabled", true);
