diff -Naur a/python/mozbuild/mozbuild/controller/building.py b/python/mozbuild/mozbuild/controller/building.py
--- a/python/mozbuild/mozbuild/controller/building.py	2022-05-31 03:34:54.993127366 +0200
+++ b/python/mozbuild/mozbuild/controller/building.py	2022-05-31 03:34:58.417144825 +0200
@@ -1540,13 +1540,6 @@
             output.on_line(
                 "We know it took a while, but your build finally finished successfully!"
             )
-            if not using_sccache:
-                output.on_line(
-                    "If you are building Firefox often, SCCache can save you a lot "
-                    "of time. You can learn more here: "
-                    "https://firefox-source-docs.mozilla.org/setup/"
-                    "configuring_build_options.html#sccache"
-                )
         else:
             output.on_line("Your build was successful!")
 
