diff -Naur orig/widget/xremoteclient/DBusRemoteClient.cpp patched/widget/xremoteclient/DBusRemoteClient.cpp
--- orig/widget/xremoteclient/DBusRemoteClient.cpp	2018-10-01 16:52:07.000000000 +0200
+++ patched/widget/xremoteclient/DBusRemoteClient.cpp	2018-10-06 00:53:26.886053306 +0200
@@ -142,7 +142,7 @@
       (bool (*)(const char *, DBusError *))
       dlsym(RTLD_DEFAULT, "dbus_validate_bus_name");
     if (!sDBusValidateBusName) {
-      return false
+      return false;
     }
 
     if (!sDBusValidateBusName(aDestinationName.get(), nullptr)) {
