diff -Nur orig/linux/org.hydrogenmusic.Hydrogen.desktop mod/linux/org.hydrogenmusic.Hydrogen.desktop
--- orig/linux/org.hydrogenmusic.Hydrogen.desktop	2023-09-09 17:15:05.000000000 +0200
+++ mod/linux/org.hydrogenmusic.Hydrogen.desktop	2023-11-06 08:46:55.856292397 +0100
@@ -10,6 +10,7 @@
 GenericName=Drum Machine
 GenericName[de]=Drumsequencer
 GenericName[ru]=Драм-машина
+GenericName[hu]=Dobgép
 GenericName[tr]=Davul Makinası
 GenericName[fr]=Boîte à rythme
 
