diff -Nur orig/linux/org.hydrogenmusic.Hydrogen.desktop mod/linux/org.hydrogenmusic.Hydrogen.desktop
--- orig/linux/org.hydrogenmusic.Hydrogen.desktop	2020-03-23 22:31:53.000000000 +0100
+++ mod/linux/org.hydrogenmusic.Hydrogen.desktop	2020-06-04 13:48:58.011060387 +0200
@@ -26,4 +26,4 @@
 Terminal=false
 StartupNotify=true
 
-Icon=org.hydrogenmusic.Hydrogen
+Icon=/usr/share/icons/hicolor/scalable/apps/org.hydrogenmusic.Hydrogen.svg/h2-icon.svg
