--- a/Makefile.in.orig	2012-09-05 21:05:36.000000000 +0200
+++ a/Makefile.in	2012-09-05 21:06:39.000000000 +0200
@@ -399,26 +399,6 @@
 	test -n "$$files" || exit 0; \
 	echo " ( cd '$(DESTDIR)$(gnomemenudir)' && rm -f" $$files ")"; \
 	cd "$(DESTDIR)$(gnomemenudir)" && rm -f $$files
-install-lingotdocDATA: $(lingotdoc_DATA)
-	@$(NORMAL_INSTALL)
-	test -z "$(lingotdocdir)" || $(MKDIR_P) "$(DESTDIR)$(lingotdocdir)"
-	@list='$(lingotdoc_DATA)'; test -n "$(lingotdocdir)" || list=; \
-	for p in $$list; do \
-	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
-	  echo "$$d$$p"; \
-	done | $(am__base_list) | \
-	while read files; do \
-	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(lingotdocdir)'"; \
-	  $(INSTALL_DATA) $$files "$(DESTDIR)$(lingotdocdir)" || exit $$?; \
-	done
-
-uninstall-lingotdocDATA:
-	@$(NORMAL_UNINSTALL)
-	@list='$(lingotdoc_DATA)'; test -n "$(lingotdocdir)" || list=; \
-	files=`for p in $$list; do echo $$p; done | sed -e 's|^.*/||'`; \
-	test -n "$$files" || exit 0; \
-	echo " ( cd '$(DESTDIR)$(lingotdocdir)' && rm -f" $$files ")"; \
-	cd "$(DESTDIR)$(lingotdocdir)" && rm -f $$files
 
 # This directory's subdirectories are mostly independent; you can cd
 # into them and run `make' without going through this Makefile.
