diff -Nur orig/org.nongnu.lingot.desktop mod/org.nongnu.lingot.desktop
--- orig/org.nongnu.lingot.desktop	2020-04-25 00:15:33.000000000 +0200
+++ mod/org.nongnu.lingot.desktop	2020-05-10 10:31:28.257776959 +0200
@@ -2,12 +2,14 @@
 Name=Lingot
 GenericName=Musical instrument tuner
 GenericName[fr]=Accordeur d’instrument musical
+GenericName[hu]=Hangszer hangoló
 Keywords=frequency;note;gauge
 Comment=LINGOT Is Not a Guitar-Only Tuner
 Comment[cs]=(Nejen kytarová) ladička
 Comment[fr]=Un accordeur de guitare, mais pas que
+Comment[hu]=Nem csak egy gitár hangoló program
 Exec=lingot
-Icon=org.nongnu.lingot.svg
+Icon=/usr/share/icons/hicolor/scalable/apps/org.nongnu.lingot.svg
 Terminal=false
 Type=Application
 Categories=GNOME;GTK;AudioVideo;
