diff -Naur lingot-1.0.1-orig/org.nongnu.lingot.desktop.in lingot-1.0.1/org.nongnu.lingot.desktop.in
--- lingot-1.0.1-orig/org.nongnu.lingot.desktop.in	2018-07-07 12:36:47.000000000 +0200
+++ lingot-1.0.1/org.nongnu.lingot.desktop.in	2018-07-19 17:37:01.105425279 +0200
@@ -5,8 +5,10 @@
 Comment=LINGOT Is Not a Guitar-Only Tuner
 Comment[cs]=(Nejen kytarová) ladička
 Comment[fr]=Un accordeur de guitare, mais pas que
+Comment[hu]=Nem csak egy gitár hangoló program
 Exec=lingot
 Icon=@PACKAGE_ICON_DIR@/org.nongnu.lingot.svg
 Terminal=false
 Type=Application
 Categories=GNOME;GTK;AudioVideo;
+
