--- a/lingot.desktop.in.orig	2012-09-05 20:59:31.000000000 +0200
+++ a/lingot.desktop.in	2012-09-05 21:01:11.000000000 +0200
@@ -2,8 +2,9 @@
 Name=Lingot
 Comment=LINGOT Is Not a Guitar-Only Tuner
 Comment[cs]=(Nejen kytarová) ladička
+Comment[hu]=Nem csak egy gitár hangoló program
 Exec=lingot
 Icon=@PACKAGE_PIXMAPS_DIR@/lingot-icon.svg
 Terminal=false
 Type=Application
-Categories=GNOME;GTK;AudioVideo;
+Categories=GNOME;GTK;AudioVideo;
