diff -Nur orig/configure.ac mod/configure.ac
--- orig/configure.ac	2019-05-27 03:14:09.000000000 +0200
+++ mod/configure.ac	2019-06-18 16:47:37.594767091 +0200
@@ -1738,8 +1738,8 @@
         COM_ERR_CPPFLAGS=""
       ;;
   *) # use whatever they told us, or whatever we found
-     COM_ERR_LIBS="${with_com_err}/lib/libcom_err.a"
-     COM_ERR_CPPFLAGS="-I${with_com_err}/include"
+     COM_ERR_LIBS="-lcom_err" # hope it's not shared
+     COM_ERR_CPPFLAGS=""
      # Ever get the feeling people hide this stuff on purpose?
      if test -d "${with_com_err}/include/et" ; then
        COM_ERR_CPPFLAGS="-I${with_com_err}/include/et"
