diff -Nur orig/configure mod/configure
--- orig/configure	2017-04-20 04:46:47.000000000 +0200
+++ mod/configure	2017-06-01 18:05:06.356687173 +0200
@@ -24132,8 +24132,8 @@
         COM_ERR_CPPFLAGS=""
       ;;
   *) # use whatever they told us, or whatever we found
-     COM_ERR_LIBS="${with_com_err}/lib/libcom_err.a"
-     COM_ERR_CPPFLAGS="-I${with_com_err}/include"
+     COM_ERR_LIBS="-lcom_err" # hope it's not shared
+     COM_ERR_CPPFLAGS=""
      # Ever get the feeling people hide this stuff on purpose?
      if test -d "${with_com_err}/include/et" ; then
        COM_ERR_CPPFLAGS="-I${with_com_err}/include/et"
