diff -Nur orig/imap/imapparse.c mod/imap/imapparse.c
--- orig/imap/imapparse.c	2017-04-07 07:31:10.000000000 +0200
+++ mod/imap/imapparse.c	2017-06-03 11:03:09.558408730 +0200
@@ -85,7 +85,7 @@
  */
 EXPORTED int getxstring(struct protstream *pin, struct protstream *pout,
                         struct buf *buf, enum getxstring_flags flags)
-    __attribute__((optimize("-O3")));
+    __attribute__((optimize("-O2")));
 EXPORTED int getxstring(struct protstream *pin, struct protstream *pout,
                         struct buf *buf, enum getxstring_flags flags)
 {
diff -Nur orig/lib/prot.c mod/lib/prot.c
--- orig/lib/prot.c	2017-04-07 07:31:10.000000000 +0200
+++ mod/lib/prot.c	2017-06-03 11:03:35.550338557 +0200
@@ -1741,7 +1741,7 @@
 }
 
 EXPORTED inline int prot_getc(struct protstream *s)
-    __attribute__((always_inline,optimize("-O3")));
+    __attribute__((always_inline,optimize("-O2")));
 EXPORTED inline int prot_getc(struct protstream *s)
 {
     assert(!s->write);
diff -Nur orig/lib/util.c mod/lib/util.c
--- orig/lib/util.c	2017-04-20 03:21:48.000000000 +0200
+++ mod/lib/util.c	2017-06-03 11:03:28.766356873 +0200
@@ -863,7 +863,7 @@
 /* buffer handling functions */
 
 static inline size_t roundup(size_t size)
-    __attribute__((pure, always_inline, optimize("-O3")));
+    __attribute__((pure, always_inline, optimize("-O2")));
 static inline size_t roundup(size_t size)
 {
     if (size < 32)
@@ -982,14 +982,14 @@
 }
 
 EXPORTED inline size_t buf_len(const struct buf *buf)
-    __attribute__((always_inline, optimize("-O3")));
+    __attribute__((always_inline, optimize("-O2")));
 EXPORTED inline size_t buf_len(const struct buf *buf)
 {
     return buf->len;
 }
 
 EXPORTED inline const char *buf_base(const struct buf *buf)
-    __attribute__((always_inline, optimize("-O3")));
+    __attribute__((always_inline, optimize("-O2")));
 EXPORTED inline const char *buf_base(const struct buf *buf)
 {
     return buf->s;
diff -Nur orig/Makefile.am mod/Makefile.am
--- orig/Makefile.am	2017-04-20 03:21:48.000000000 +0200
+++ mod/Makefile.am	2017-06-03 11:02:35.390500976 +0200
@@ -1409,7 +1409,7 @@
 
 noinst_LTLIBRARIES += libcrc32.la
 libcrc32_la_SOURCES = lib/crc32.c lib/crc32c.c
-libcrc32_la_CFLAGS = -O3 $(AM_CFLAGS) $(CFLAG_VISIBILITY)
+libcrc32_la_CFLAGS = -O2 $(AM_CFLAGS) $(CFLAG_VISIBILITY)
 
 nodist_lib_libcyrus_min_la_SOURCES = \
 	lib/imapopts.c
diff -Nur orig/Makefile.in mod/Makefile.in
--- orig/Makefile.in	2017-04-20 04:46:48.000000000 +0200
+++ mod/Makefile.in	2017-06-03 11:02:31.518511430 +0200
@@ -2480,7 +2480,7 @@
 	$(GCOV_LIBS) $(am__append_76)
 lib_libcyrus_la_CFLAGS = $(AM_CFLAGS) $(CFLAG_VISIBILITY)
 libcrc32_la_SOURCES = lib/crc32.c lib/crc32c.c
-libcrc32_la_CFLAGS = -O3 $(AM_CFLAGS) $(CFLAG_VISIBILITY)
+libcrc32_la_CFLAGS = -O2 $(AM_CFLAGS) $(CFLAG_VISIBILITY)
 nodist_lib_libcyrus_min_la_SOURCES = \
 	lib/imapopts.c
 
