diff -Naur orig/ipepresenter/Makefile patched/ipepresenter/Makefile
--- orig/ipepresenter/Makefile	2016-01-05 07:35:58.000000000 +0100
+++ patched/ipepresenter/Makefile	2016-07-07 01:29:21.410773539 +0200
@@ -4,14 +4,14 @@
 
 CXX=g++
 
-QT_CFLAGS = -fPIC -std=c++11 -I/usr/include/qt/QtGui -I/usr/include/qt/QtWidgets -I/usr/include/qt/QtCore -I/usr/include/qt
-QT_LIBS = -lQt5Gui -lQt5Core -lQt5Widgets
-CPPFLAGS += $(QT_CFLAGS)
-LIBS += -lipecanvas -lipecairo -lipe $(QT_LIBS) $(CAIRO_LIBS)
+QT_CFLAGS = -fPIC -std=c++11 -I/usr/lib/qt5/include/QtGui -I/usr/lib/qt5/include/QtWidgets -I/usr/lib/qt5/include/QtCore -I//usr/lib/qt5/include
+QT_LIBS = -lQt5Gui -lQt5Core -lQt5Widgets -L/usr/lib/qt5/lib
+CPPFLAGS += $(QT_CFLAGS) -I../src/ipecanvas -I../src/include
+LIBS += -L../build/lib -lipecanvas -lipecairo -lipe $(QT_LIBS) $(CAIRO_LIBS)
 DESTDIR=/usr/bin
 
-MOC=moc
-UIC=uic
+MOC=/usr/lib/qt5/bin/moc
+UIC=/usr/lib/qt5/bin/uic
 
 OBJECTS = mainwindow.o moc_mainwindow.o canvasview.o moc_canvasview.o \
 		  timelabel.o moc_timelabel.o 
