diff -Nur orig/svgtoipe/svgtoipe.py mod/svgtoipe/svgtoipe.py
--- orig/svgtoipe/svgtoipe.py	2019-01-08 23:31:55.000000000 +0100
+++ mod/svgtoipe/svgtoipe.py	2019-12-04 18:35:55.741440152 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 # --------------------------------------------------------------------
 # convert SVG to Ipe format
 # --------------------------------------------------------------------
