diff -Nur orig/lib/db.cpp mod/lib/db.cpp
--- orig/lib/db.cpp	2020-01-30 20:54:31.000000000 +0100
+++ mod/lib/db.cpp	2020-03-04 13:20:52.005994484 +0100
@@ -64,7 +64,7 @@
 	db->type = xhtons(type);
 	db->version = xhtons(ver);
 	db->flags = 0;
-	strncpy(db->name, name.toUtf8(), NAMELEN);
+	strncpy(db->name, name.toUtf8(), NAMELEN-1);
 	db->name[NAMELEN-1] = '\0';
 }
 
@@ -225,7 +225,7 @@
 	if (find(type, name) != 0) {
 		throw errorEx(QObject::tr("DB: Entry to rename not found: %1").arg(name));
 	}
-	strncpy(head.name, n.toUtf8(), NAMELEN);
+	strncpy(head.name, n.toUtf8(), NAMELEN-1);
 	head.name[NAMELEN-1] = '\0';
 	file.seek(head_offset);
 	ret = file.write((char*)&head, sizeof(head));
diff -Nur orig/lib/exception.h mod/lib/exception.h
--- orig/lib/exception.h	2020-01-30 20:54:31.000000000 +0100
+++ mod/lib/exception.h	2020-03-04 13:21:58.445548133 +0100
@@ -32,6 +32,12 @@
 			msg = e.msg;
 			info = e.info;
 		}
+		errorEx &operator = (const errorEx &e)
+		{
+			msg = e.msg;
+			info = e.info;
+			return *this;
+		}
 		void appendString(QString s)
 		{
 			msg = msg + " " + s;
diff -Nur orig/lib/pkcs11_lib.h mod/lib/pkcs11_lib.h
--- orig/lib/pkcs11_lib.h	2020-01-30 20:54:31.000000000 +0100
+++ mod/lib/pkcs11_lib.h	2020-03-04 13:22:44.189376906 +0100
@@ -68,6 +68,11 @@
 		lib = l;
 		id = i;
 	}
+	slotid(const slotid &other)
+	{
+		lib = other.lib;
+		id = other.id;
+	}
 	slotid &operator = (const slotid &other)
 	{
 		lib = other.lib;
diff -Nur orig/lib/pki_key.cpp mod/lib/pki_key.cpp
--- orig/lib/pki_key.cpp	2020-01-30 20:54:31.000000000 +0100
+++ mod/lib/pki_key.cpp	2020-03-04 13:23:28.796863717 +0100
@@ -661,9 +661,7 @@
 	switch (getKeyType()) {
 #ifndef OPENSSL_NO_EC
 	case EVP_PKEY_EC:
-		if (ecParamNid() != NID_X9_62_prime256v1)
-			break;
-		/* fall */
+		return ecParamNid() == NID_X9_62_prime256v1;
 #endif
 	case EVP_PKEY_RSA:
 	case EVP_PKEY_DSA:
diff -Nur orig/lib/x509rev.h mod/lib/x509rev.h
--- orig/lib/x509rev.h	2020-01-30 20:54:31.000000000 +0100
+++ mod/lib/x509rev.h	2020-03-04 13:24:05.204451657 +0100
@@ -135,6 +135,12 @@
 				append(r);
 			}
 		}
+		x509revList &operator = (const x509revList &r)
+		{
+			QList<x509rev>::operator=(r);
+			merged = r.merged;
+			return *this;
+		}
 		bool sqlUpdate(QVariant caId);
 };
 #endif
