diff -Naur orig/data/planner.desktop.in.in mod/data/planner.desktop.in.in
--- orig/data/planner.desktop.in.in	2011-09-29 16:15:45.000000000 +0200
+++ mod/data/planner.desktop.in.in	2016-03-17 13:18:22.138088739 +0100
@@ -1,11 +1,12 @@
 [Desktop Entry]
-_Name=Project Management
+_Name=Planner
+_GenericName=Project Management
 _Comment=Planner Project Management
 Exec=planner %F
-Icon=gnome-planner.png
+Icon=gnome-planner
 Terminal=false
 Type=Application
-Categories=GNOME;GTK;Application;Office;ProjectManagement;
+Categories=GNOME;GTK;Office;ProjectManagement;
 StartupNotify=true
 MimeType=application/x-planner;
 X-GNOME-Bugzilla-Bugzilla=GNOME
