diff -Nur orig/configure.in mod/configure.in
--- orig/configure.in	2011-12-25 03:58:21.000000000 +0100
+++ mod/configure.in	2018-03-02 23:01:25.981311643 +0100
@@ -47,7 +47,6 @@
 dnl  -------------
 dnl | API docs    |--------------------------------------------
 dnl  -------------
-GTK_DOC_CHECK([1.0])
 
 dnl NOTE: We need to use a separate automake conditional for this
 dnl       to make this work with the tarballs.
