diff -Naur orig/xvidcap.desktop patched/xvidcap.desktop
--- orig/xvidcap.desktop	2006-09-19 08:55:21.000000000 +0200
+++ patched/xvidcap.desktop	2014-09-22 06:53:36.405527721 +0200
@@ -1,12 +1,12 @@
 [Desktop Entry]
 Type=Application
-Encoding=UTF-8
 Name=XVidCap Screen Capture
 GenericName=X11 Screencam
+GenericName[hu]=X11 Képernyőkamera
 Comment=X11 Screencam
 Comment[de]=X11 Videoaufnahme
-Icon=xvidcap.png
+Icon=xvidcap
 TryExec=xvidcap
 Exec=xvidcap
 Terminal=false
-Categories=GTK;Application;AudioVideo;Video;
+Categories=GTK;AudioVideo;Video;
