diff -Nur orig/btt/bno_plot.py mod/btt/bno_plot.py
--- orig/btt/bno_plot.py	2017-11-05 05:13:06.000000000 +0100
+++ mod/btt/bno_plot.py	2019-12-04 17:29:24.660450324 +0100
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2
 #
 # btt blkno plotting interface
 #
diff -Nur orig/btt/btt_plot.py mod/btt/btt_plot.py
--- orig/btt/btt_plot.py	2017-11-05 05:13:06.000000000 +0100
+++ mod/btt/btt_plot.py	2019-12-04 17:29:57.316396061 +0100
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2
 #
 # btt_plot.py: Generate matplotlib plots for BTT generate data files
 #
