diff -Nur orig/data/cinnamon-screensaver.desktop.in mod/data/cinnamon-screensaver.desktop.in
--- orig/data/cinnamon-screensaver.desktop.in	1970-01-01 01:00:00.000000000 +0100
+++ mod/data/cinnamon-screensaver.desktop.in	2021-05-31 17:14:33.000000000 +0200
@@ -0,0 +1,13 @@
+[Desktop Entry]
+Type=Application
+Name=Screensaver
+Comment=Launch screensaver and locker program
+Icon=preferences-desktop-screensaver
+Exec=cinnamon-screensaver
+OnlyShowIn=X-Cinnamon;
+NoDisplay=true
+X-GNOME-Bugzilla-Bugzilla=GNOME
+X-GNOME-Bugzilla-Product=cinnamon-screensaver
+X-GNOME-Bugzilla-Component=general
+X-GNOME-Bugzilla-Version=@VERSION@
+DBusActivatable=true
\ Nincs új sor a fájl végén
diff -Nur orig/data/meson.build mod/data/meson.build
--- orig/data/meson.build	2021-05-31 17:34:43.000000000 +0200
+++ mod/data/meson.build	2021-05-31 17:14:33.000000000 +0200
@@ -5,6 +5,14 @@
   pamdir = sysconfdir
 endif
 
+desktop = configure_file(
+  output: 'cinnamon-screensaver.desktop',
+  input: 'cinnamon-screensaver.desktop.in',
+  # meson 0.49
+  # configuration: { 'VERSION': meson.project_version() }
+  configuration: misc_conf
+)
+
 dbus_service = configure_file(
   output: 'org.cinnamon.ScreenSaver.service',
   input: 'org.cinnamon.ScreenSaver.service.in',
@@ -26,5 +34,5 @@
   )
 endif
 
-install_data('org.cinnamon.ScreenSaver.desktop', install_dir: join_paths(datadir, 'applications'))
+install_data(desktop, install_dir: join_paths(datadir, 'applications'))
 install_data(dbus_service, install_dir: dbus_services_dir)
diff -Nur orig/data/org.cinnamon.ScreenSaver.desktop mod/data/org.cinnamon.ScreenSaver.desktop
--- orig/data/org.cinnamon.ScreenSaver.desktop	2021-05-31 17:34:43.000000000 +0200
+++ mod/data/org.cinnamon.ScreenSaver.desktop	1970-01-01 01:00:00.000000000 +0100
@@ -1,8 +0,0 @@
-[Desktop Entry]
-Type=Application
-Name=Screensaver
-Comment=Launch screensaver and locker program
-Exec=cinnamon-screensaver
-OnlyShowIn=X-Cinnamon;
-NoDisplay=true
-DBusActivatable=true
\ Nincs új sor a fájl végén
