diff -Nur orig/data/cinnamon-screensaver.desktop.in mod/data/cinnamon-screensaver.desktop.in
--- orig/data/cinnamon-screensaver.desktop.in	2021-05-31 16:45:51.000000000 +0200
+++ mod/data/cinnamon-screensaver.desktop.in	2021-05-31 21:29:33.334983890 +0200
@@ -10,4 +10,3 @@
 X-GNOME-Bugzilla-Product=cinnamon-screensaver
 X-GNOME-Bugzilla-Component=general
 X-GNOME-Bugzilla-Version=@VERSION@
-DBusActivatable=true
\ Nincs új sor a fájl végén
