diff -Nur orig/data/cinnamon-screensaver mod/data/cinnamon-screensaver
--- orig/data/cinnamon-screensaver	2021-06-04 13:18:23.000000000 +0200
+++ mod/data/cinnamon-screensaver	2021-06-05 14:47:50.455584812 +0200
@@ -1,7 +1,6 @@
 #%PAM-1.0
 
-# Fedora & Arch
--auth      sufficient   pam_selinux_permit.so
+# UHU
 auth       include      system-auth
 auth       optional     pam_gnome_keyring.so
 account    include      system-auth
