diff -Naur a/configure b/configure
--- a/configure	2022-11-11 21:10:53.000000000 +0100
+++ b/configure	2023-02-08 19:28:54.144427279 +0100
@@ -21198,7 +21198,7 @@
 CAIRO_REQUIRED=1.14.0
 GLIB_REQUIRED=2.54.0
 GTK_REQUIRED=3.22.0
-WEBKIT_REQUIRED=2.6.0
+WEBKIT_REQUIRED=2.33.2
 LIBSECRET_REQUIRED=0.5
 LIBXML_REQUIRED=2.5.0
 SYNCTEX_REQUIRED=1.21
@@ -24602,14 +24602,14 @@
     pkg_cv_EPUB_CFLAGS="$EPUB_CFLAGS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"webkit2gtk-4.0 >= \$WEBKIT_REQUIRED \\
+    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"webkit2gtk-4.1 >= \$WEBKIT_REQUIRED \\
        libxml-2.0 >= \$LIBXML_REQUIRED zlib\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "webkit2gtk-4.0 >= $WEBKIT_REQUIRED \
+  ($PKG_CONFIG --exists --print-errors "webkit2gtk-4.1 >= $WEBKIT_REQUIRED \
        libxml-2.0 >= $LIBXML_REQUIRED zlib") 2>&5
   ac_status=$?
   printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_EPUB_CFLAGS=`$PKG_CONFIG --cflags "webkit2gtk-4.0 >= $WEBKIT_REQUIRED \
+  pkg_cv_EPUB_CFLAGS=`$PKG_CONFIG --cflags "webkit2gtk-4.1 >= $WEBKIT_REQUIRED \
        libxml-2.0 >= $LIBXML_REQUIRED zlib" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
@@ -24622,14 +24622,14 @@
     pkg_cv_EPUB_LIBS="$EPUB_LIBS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"webkit2gtk-4.0 >= \$WEBKIT_REQUIRED \\
+    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"webkit2gtk-4.1 >= \$WEBKIT_REQUIRED \\
        libxml-2.0 >= \$LIBXML_REQUIRED zlib\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "webkit2gtk-4.0 >= $WEBKIT_REQUIRED \
+  ($PKG_CONFIG --exists --print-errors "webkit2gtk-4.1 >= $WEBKIT_REQUIRED \
        libxml-2.0 >= $LIBXML_REQUIRED zlib") 2>&5
   ac_status=$?
   printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_EPUB_LIBS=`$PKG_CONFIG --libs "webkit2gtk-4.0 >= $WEBKIT_REQUIRED \
+  pkg_cv_EPUB_LIBS=`$PKG_CONFIG --libs "webkit2gtk-4.1 >= $WEBKIT_REQUIRED \
        libxml-2.0 >= $LIBXML_REQUIRED zlib" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
@@ -24651,10 +24651,10 @@
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        EPUB_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "webkit2gtk-4.0 >= $WEBKIT_REQUIRED \
+	        EPUB_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "webkit2gtk-4.1 >= $WEBKIT_REQUIRED \
        libxml-2.0 >= $LIBXML_REQUIRED zlib" 2>&1`
         else
-	        EPUB_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "webkit2gtk-4.0 >= $WEBKIT_REQUIRED \
+	        EPUB_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "webkit2gtk-4.1 >= $WEBKIT_REQUIRED \
        libxml-2.0 >= $LIBXML_REQUIRED zlib" 2>&1`
         fi
 	# Put the nasty error message in config.log where it belongs
diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2022-11-11 21:08:16.000000000 +0100
+++ b/configure.ac	2023-02-08 19:29:11.932457453 +0100
@@ -93,7 +93,7 @@
 CAIRO_REQUIRED=1.14.0
 GLIB_REQUIRED=2.54.0
 GTK_REQUIRED=3.22.0
-WEBKIT_REQUIRED=2.6.0
+WEBKIT_REQUIRED=2.33.2
 LIBSECRET_REQUIRED=0.5
 LIBXML_REQUIRED=2.5.0
 SYNCTEX_REQUIRED=1.21
@@ -622,7 +622,7 @@
 
 have_webkit="no"
 if test "x$enable_epub" = "xyes"; then
-    PKG_CHECK_MODULES(EPUB, webkit2gtk-4.0 >= $WEBKIT_REQUIRED \
+    PKG_CHECK_MODULES(EPUB, webkit2gtk-4.1 >= $WEBKIT_REQUIRED \
        libxml-2.0 >= $LIBXML_REQUIRED zlib,have_webkit=yes,have_webkit=no)
 
     if test "x$have_webkit" = "xyes"; then
