From 836bea078685e3de468027228497d1ad8d94916b Mon Sep 17 00:00:00 2001
From: Joe Savage <jasavage@silabs.com>
Date: Tue, 19 May 2020 09:41:43 -0500
Subject: [PATCH] Removed call to deprecated SplitterWindow.SetSashSize(). 
 Fixes issue #3

---
 runsnakerun/runsnake.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/runsnakerun/runsnake.py b/runsnakerun/runsnake.py
index 408dd6b..1adb0ac 100755
--- a/runsnakerun/runsnake.py
+++ b/runsnakerun/runsnake.py
@@ -265,7 +265,11 @@ def CreateControls(self, config_parser):
         self.tabs.AddPage(self.allCallerListControl, _('All Callers'), False)
         if editor:
             self.tabs.AddPage(self.sourceCodeControl, _('Source Code'), False)
-        self.rightSplitter.SetSashSize(10)
+
+        # SetSashSize has been removed since it didn't work correctly
+        # per https://github.com/wxWidgets/wxWidgets/commit/26696222ffb75ac8531207cf356a52b033c48d9e
+        #self.rightSplitter.SetSashSize(10)
+
         # calculate size as proportional value for initial display...
         self.LoadState(config_parser)
         width, height = self.GetSize()
