From 325cd9ee8783c6e4a1d50320f27f9fb3e15039a7 Mon Sep 17 00:00:00 2001
From: Andre Miranda <andreldm@xfce.org>
Date: Sat, 19 Oct 2019 20:06:13 -0300
Subject: [PATCH] Fix build with panel 4.15

---
 panel-plugin/mount-plugin.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/panel-plugin/mount-plugin.h b/panel-plugin/mount-plugin.h
index d3a6758..ed74284 100644
--- a/panel-plugin/mount-plugin.h
+++ b/panel-plugin/mount-plugin.h
@@ -28,7 +28,7 @@ Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 #endif
 
 #include <gtk/gtk.h>
-#include <libxfce4panel/xfce-panel-plugin.h>
+#include <libxfce4panel/libxfce4panel.h>
 
 #include <string.h>
