From 43799ace5a534f90222138c2ee0894efaa322bb1 Mon Sep 17 00:00:00 2001
From: damufo <dani@damufo.com>
Date: Tue, 16 Jun 2020 06:34:53 +0200
Subject: [PATCH] Spellfix. Add missing "e" to siz (size)

---
 panel-plugin/devices.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/panel-plugin/devices.c b/panel-plugin/devices.c
index 7313cfb..b2df3db 100644
--- a/panel-plugin/devices.c
+++ b/panel-plugin/devices.c
@@ -82,7 +82,7 @@ mount_info_print(t_mount_info * mount_info)
     {
         printf (_("size:                %g\n"), mount_info->size);
         printf (_("used size:           %g\n") ,mount_info->used);
-        printf (_("available siz:       %g\n"), mount_info->avail);
+        printf (_("available size:       %g\n"), mount_info->avail);
         printf (_("percentage used:     %d\n"), mount_info->percent);
         printf (_("file system type:    %s\n"), mount_info->type);
         printf (_("actual mount point:  %s\n"), mount_info->mounted_on);
