diff -Nur orig/libjs.make mod/libjs.make
--- orig/libjs.make	2014-05-17 19:52:24.000000000 +0200
+++ mod/libjs.make	2018-01-16 19:08:24.917740765 +0100
@@ -16,6 +16,7 @@
 LIBJS_CONFIG_FLAGS               += --enable-threadsafe
 LIBJS_CONFIG_FLAGS               += --with-system-nspr
 LIBJS_CONFIG_FLAGS               += --disable-tests
+LIBJS_CONFIG_FLAGS               += --target=$(UB_TARGET)
 ifeq ($(OO_JAVASCRIPT_TRACE),yes)
     LIBJS_CONFIG_FLAGS           += --enable-trace-jscalls
 endif
