diff -Naurdp bsd-games-2.17.orig/dab/gamescreen.h bsd-games-2.17/dab/gamescreen.h
--- bsd-games-2.17.orig/dab/gamescreen.h	2004-01-02 19:04:51.000000000 +0100
+++ bsd-games-2.17/dab/gamescreen.h	2009-12-03 19:25:44.000000000 +0100
@@ -70,9 +70,9 @@ class GAMESCREEN {
     virtual void redraw(void) = 0;			// Refresh
     virtual int getinput(void) = 0;			// Get user input
     virtual void bell(void) = 0;			// Beep
-    virtual void score(size_t p, const PLAYER& p) = 0;	// Post current score
-    virtual void games(size_t p, const PLAYER& p) = 0;	// Post games won
-    virtual void total(size_t p, const PLAYER& p) = 0;	// Post total score
+    virtual void score(size_t p, const PLAYER& pl) = 0;	// Post current score
+    virtual void games(size_t p, const PLAYER& pl) = 0;	// Post games won
+    virtual void total(size_t p, const PLAYER& pl) = 0;	// Post total score
     virtual void ties(const PLAYER& p) = 0;		// Post tie games
 };
 
