diff -Nur orig/quiz/datfiles/poetry mod/quiz/datfiles/poetry
--- orig/quiz/datfiles/poetry	1997-07-17 18:42:20.000000000 +0200
+++ mod/quiz/datfiles/poetry	2018-02-10 09:06:23.388739137 +0100
@@ -64,7 +64,7 @@
 {Alexander }Pope
 The curfew tolls the knell of parting day{,}:\
 The lowing herd wind slowly o'er the lea:\
-Elegy{ Written in a Country Church{-| }Yard:\
+Elegy{ Written in a Country Church{-| }Yard}:\
 {Thomas }Gray
 The best laid schemes o' mice an' men gang aft a-gley{,}:\
 An{'|d} lea{'|v}e us nought but grief an{'|d} pain for promised joy{.}:\
