diff -Nur orig/schemas/org.gnome.desktop.background.gschema.xml.in mod/schemas/org.gnome.desktop.background.gschema.xml.in
--- orig/schemas/org.gnome.desktop.background.gschema.xml.in	2025-02-02 12:52:42.175273349 +0100
+++ mod/schemas/org.gnome.desktop.background.gschema.xml.in	2025-02-02 12:59:48.169373210 +0100
@@ -11,7 +11,7 @@
       </description>
     </key>
     <key name="picture-uri" type="s">
-      <default>'file://@datadir@/backgrounds/gnome/adwaita-l.jxl'</default>
+      <default>'file://@datadir@/backgrounds/UHU/uhu_linux_ubk.jpg'</default>
       <summary>Picture URI</summary>
       <description>
         URI to use for the background image. Note that the backend only supports
@@ -19,7 +19,7 @@
       </description>
     </key>
     <key name="picture-uri-dark" type="s">
-      <default>'file://@datadir@/backgrounds/gnome/adwaita-d.jxl'</default>
+      <default>'file://@datadir@/backgrounds/UHU/uhu_linux_ubk.jpg'</default>
       <summary>Picture URI (dark)</summary>
       <description>
         URI to use for the background image. Note that the backend only supports
diff -Naur a/schemas/org.gnome.desktop.screensaver.gschema.xml.in b/schemas/org.gnome.desktop.screensaver.gschema.xml.in
--- a/schemas/org.gnome.desktop.screensaver.gschema.xml.in	2022-09-17 23:34:56.000000000 +0200
+++ b/schemas/org.gnome.desktop.screensaver.gschema.xml.in	2022-09-25 19:04:43.002584930 +0200
@@ -95,7 +95,7 @@
       </description>
     </key>
     <key name="picture-uri" type="s">
-      <default>'file://@datadir@/backgrounds/gnome/adwaita-timed.xml'</default>
+      <default>'file://@datadir@/backgrounds/UHU/uhu_linux_ubk.jpg'</default>
       <summary>Picture URI</summary>
       <description>
         URI to use for the background image. Note that the backend only supports
