diff -Naur a/schemas/org.gnome.desktop.background.gschema.xml.in b/schemas/org.gnome.desktop.background.gschema.xml.in
--- a/schemas/org.gnome.desktop.background.gschema.xml.in	1970-01-01 01:00:00.000000000 +0100
+++ b/schemas/org.gnome.desktop.background.gschema.xml.in	2022-03-26 01:40:27.117484876 +0100
@@ -11,7 +11,7 @@
       </description>
     </key>
     <key name="picture-uri" type="s">
-      <default>'file://@datadir@/backgrounds/gnome/adwaita-l.jpg'</default>
+      <default>'file://@datadir@/backgrounds/UHU/uhu_linux_ubk.jpg'</default>
       <summary>Picture URI</summary>
       <description>
         URI to use for the background image. Note that the backend only supports
diff -Naur a/schemas/org.gnome.desktop.screensaver.gschema.xml.in b/schemas/org.gnome.desktop.screensaver.gschema.xml.in
--- a/schemas/org.gnome.desktop.screensaver.gschema.xml.in	1970-01-01 01:00:00.000000000 +0100
+++ b/schemas/org.gnome.desktop.screensaver.gschema.xml.in	2022-03-26 01:41:07.545656128 +0100
@@ -95,7 +95,7 @@
       </description>
     </key>
     <key name="picture-uri" type="s">
-      <default>'file://@datadir@/backgrounds/gnome/adwaita-timed.xml'</default>
+      <default>'file://@datadir@/backgrounds/UHU/uhu_linux_ubk.jpg'</default>
       <summary>Picture URI</summary>
       <description>
         URI to use for the background image. Note that the backend only supports
