diff -Naur orig/schemas/org.gnome.desktop.background.gschema.xml.in patched/schemas/org.gnome.desktop.background.gschema.xml.in
--- orig/schemas/org.gnome.desktop.background.gschema.xml.in	2015-05-23 02:38:48.000000000 +0200
+++ patched/schemas/org.gnome.desktop.background.gschema.xml.in	2016-10-09 19:56:29.483422386 +0200
@@ -10,7 +10,7 @@
       </description>
     </key>
     <key name="picture-uri" type="s">
-      <default>'file://@datadir@/backgrounds/gnome/adwaita-timed.xml'</default>
+      <default>'file://@datadir@/backgrounds/UHU/uhu_linux_ubk.jpg'</default>
       <summary>Picture URI</summary>
       <description>
         URI to use for the background image. Note that the backend only supports
diff -Naur orig/schemas/org.gnome.desktop.screensaver.gschema.xml.in patched/schemas/org.gnome.desktop.screensaver.gschema.xml.in
--- orig/schemas/org.gnome.desktop.screensaver.gschema.xml.in	2015-08-16 22:07:26.000000000 +0200
+++ patched/schemas/org.gnome.desktop.screensaver.gschema.xml.in	2016-10-09 19:59:20.381301700 +0200
@@ -94,7 +94,7 @@
       </description>
     </key>
     <key name="picture-uri" type="s">
-      <default>'file://@datadir@/backgrounds/gnome/adwaita-lock.jpg'</default>
+      <default>'file://@datadir@/backgrounds/UHU/uhu_linux_ubk.jpg'</default>
       <summary>Picture URI</summary>
       <description>
         URI to use for the background image. Note that the backend only supports
