diff -Naur orig/schemas/org.gnome.desktop.input-sources.gschema.xml.in mod/schemas/org.gnome.desktop.input-sources.gschema.xml.in
--- orig/schemas/org.gnome.desktop.input-sources.gschema.xml.in	2015-05-23 02:38:48.000000000 +0200
+++ mod/schemas/org.gnome.desktop.input-sources.gschema.xml.in	2016-01-30 11:40:07.169153189 +0100
@@ -15,7 +15,7 @@
       </description>
     </key>
     <key name="sources" type="a(ss)">
-      <default>[]</default>
+      <default>[('xkb', 'hu'), ('xkb', 'us')]</default>
       <summary>List of input sources</summary>
       <description>
         List of input source identifiers available. Each source is
