diff -Naur orig/schemas/org.gnome.desktop.background.gschema.xml.in patched/schemas/org.gnome.desktop.background.gschema.xml.in
--- orig/schemas/org.gnome.desktop.background.gschema.xml.in	2015-05-23 02:38:48.000000000 +0200
+++ patched/schemas/org.gnome.desktop.background.gschema.xml.in	2016-10-09 19:54:17.977976270 +0200
@@ -57,7 +57,7 @@
       </description>
     </key>
     <key name="show-desktop-icons" type="b">
-      <default>false</default>
+      <default>true</default>
       <summary>Have file manager handle the desktop</summary>
       <description>
         If set to true, then file manager will draw the icons on the desktop.
