diff -Naur orig/mp3splt-gtk.desktop.in patched/mp3splt-gtk.desktop.in
--- orig/mp3splt-gtk.desktop.in	2014-05-03 00:20:50.000000000 +0200
+++ patched/mp3splt-gtk.desktop.in	2014-08-28 00:51:06.757290320 +0200
@@ -1,13 +1,12 @@
 [Desktop Entry]
 Name=Mp3splt-gtk audio splitter
+Name[hu]=mp3splt - mp3/ogg/flac vágó
 GenericName=Sound Splitter
 Comment=Split mp3 and ogg files without decoding
-
+Comment[hu]=mp3, ogg, flac fájlok vágása dekódolás nélkül
 Icon=PREFIX/mp3splt-gtk_ico.svg
-
 Type=Application
 Categories=AudioVideo;Audio;AudioVideoEditing;
-
 Exec=mp3splt-gtk
 StartupNotify=false
 Terminal=false
