diff -Nur orig/unixconf.pri mod/unixconf.pri
--- orig/unixconf.pri	2018-05-27 20:49:44.000000000 +0200
+++ mod/unixconf.pri	2018-06-06 08:02:46.032805249 +0200
@@ -12,8 +12,8 @@
 # COMPILATION SPECIFIC
 !nogui:dbus: QT += dbus
 
-QMAKE_CXXFLAGS += -Wall -Wextra -Wpedantic -Wformat-security
-!haiku: QMAKE_LFLAGS_APP += -rdynamic
+QMAKE_CXXFLAGS += -fPIE -Wall -Wextra -Wpedantic -Wformat-security
+!haiku: QMAKE_LFLAGS_APP += -Wl,-z,now -pie -rdynamic
 
 # Man page
 nogui {
