diff -Naur orig/data/favorites/desktop.desktop patched/data/favorites/desktop.desktop
--- orig/data/favorites/desktop.desktop	2018-04-05 00:34:51.000000000 +0200
+++ patched/data/favorites/desktop.desktop	2018-09-03 23:03:48.104245329 +0200
@@ -36,6 +36,6 @@
 Comment[sr]=Датотеке радне површи
 Comment[tr]=Masaüstünüzde dosyalar
 Icon=user-desktop
-NotShowIn=GNOME;KDE;LXDE;ROX;XFCE;
+OnlyShowIn=Enlightenment;
 URL=file:$XDG_DESKTOP_DIR
 X-Enlightenment-Type=Mount
diff -Naur orig/data/favorites/home.desktop patched/data/favorites/home.desktop
--- orig/data/favorites/home.desktop	2018-04-05 00:34:51.000000000 +0200
+++ patched/data/favorites/home.desktop	2018-09-03 23:03:48.104245329 +0200
@@ -32,6 +32,6 @@
 Comment[sr]=Фасцикла садржалац свих личних датотека
 Comment[tr]=Bu dizin tüm kişisel dosyalarınızı barındırıyor
 Icon=user-home
-NotShowIn=GNOME;KDE;LXDE;ROX;XFCE;
+OnlyShowIn=Enlightenment;
 URL=file:$HOME
 X-Enlightenment-Type=Mount
diff -Naur orig/data/favorites/root.desktop patched/data/favorites/root.desktop
--- orig/data/favorites/root.desktop	2018-04-05 00:34:51.000000000 +0200
+++ patched/data/favorites/root.desktop	2018-09-03 23:03:48.104245329 +0200
@@ -28,6 +28,6 @@
 Comment[sr]=Корен, основа свих датотека
 Comment[tr]=Kök dosya sistemi tabanı
 Icon=computer
-NotShowIn=GNOME;KDE;LXDE;ROX;XFCE;
+OnlyShowIn=Enlightenment;
 URL=file:/
 X-Enlightenment-Type=Mount
diff -Naur orig/data/favorites/tmp.desktop patched/data/favorites/tmp.desktop
--- orig/data/favorites/tmp.desktop	2018-04-05 00:34:51.000000000 +0200
+++ patched/data/favorites/tmp.desktop	2018-09-03 23:03:48.104245329 +0200
@@ -32,6 +32,6 @@
 Comment[sr]=Привремене датотеке
 Comment[tr]=Geçici dosyalar
 Icon=user-temp
-NotShowIn=GNOME;KDE;LXDE;ROX;XFCE;
+OnlyShowIn=Enlightenment;
 URL=file:/tmp
 X-Enlightenment-Type=Mount
