diff -Naur orig/src/modules/mixer/emixer.desktop patched/src/modules/mixer/emixer.desktop
--- orig/src/modules/mixer/emixer.desktop	2018-08-21 02:14:53.000000000 +0200
+++ patched/src/modules/mixer/emixer.desktop	2018-09-03 23:06:56.462655115 +0200
@@ -8,7 +8,7 @@
 Name[tr]=Emixer
 Icon=emixer
 Exec=emixer
-Comment=A utility to control volume levels
+Comment=A module to provide a mixer for changing volume.
 Comment[ca]=Un mòdul per controlar el volum d'àudio i fonts.
 Comment[de]=Ein Modul, um die Lautstärke und die Datenströme zu steuern.
 Comment[eo]=Modulo por kontroli sonfortecon kaj fluojn.
@@ -18,5 +18,5 @@
 Comment[ms]=Modul untuk mengawal volum dan strim audio.
 Comment[sr]=Јединица за управљање јечином звука и токова.
 Comment[tr]=Ses seviyesi ve ses akışı kontrol modülü.
-Categories=AudioVideo;System;
+Categories=System;AudioVideo;Mixer;
 StartupWMClass=emixer
