diff -Naur enlightenment-0.22.1-orig/src/modules/mixer/emixer.desktop enlightenment-0.22.1/src/modules/mixer/emixer.desktop
--- enlightenment-0.22.1-orig/src/modules/mixer/emixer.desktop	2017-11-01 07:03:46.000000000 +0100
+++ enlightenment-0.22.1/src/modules/mixer/emixer.desktop	2017-12-31 11:51:16.064498810 +0100
@@ -9,7 +9,7 @@
 Name[tr]=Emixer
 Icon=emixer
 Exec=emixer
-Comment=A utility to control volume levels
+Comment=A module to provide a mixer for changing volume.
 Comment[ca]=Un mòdul per controlar el volum d'àudio i fonts.
 Comment[de]=Ein Modul, um die Lautstärke und die Datenströme zu steuern.
 Comment[eo]=Modulo por kontroli sonfortecon kaj fluojn.
@@ -19,5 +19,5 @@
 Comment[ms]=Modul untuk mengawal volum dan strim audio.
 Comment[sr]=Јединица за управљање јечином звука и токова.
 Comment[tr]=Ses seviyesi ve ses akışı kontrol modülü.
-Categories=Audio;System;
+Categories=System;AudioVideo;Mixer;
 StartupWMClass=emixer
