diff -Naur orig/data/config/default/e.src patched/data/config/default/e.src
--- orig/data/config/default/e.src	1970-01-01 01:00:00.000000000 +0100
+++ patched/data/config/default/e.src	2020-05-18 22:44:00.179491087 +0200
@@ -21,6 +21,7 @@
   value "show_desktop_icons" int: 1;
   value "edge_flip_dragging" int: 0;
   value "use_shaped_win" int: 0;
+  value "language" string: "hu_HU.UTF-8";
   value "no_module_delay" int: 1;
   group "modules" list {
     group "E_Config_Module" struct {
diff -Naur orig/data/config/standard/e.src patched/data/config/standard/e.src
--- orig/data/config/standard/e.src	1970-01-01 01:00:00.000000000 +0100
+++ patched/data/config/standard/e.src	2020-05-18 22:46:22.759633627 +0200
@@ -20,7 +20,7 @@
     value "show_desktop_icons" int: 1;
     value "edge_flip_dragging" int: 1;
     value "use_shaped_win" int: 0;
-    value "language" string: "en_US.UTF-8";
+    value "language" string: "hu_HU.UTF-8";
     value "window_placement_policy" int: 0;
     value "window_grouping" int: 0;
     value "focus_policy" int: 2;
@@ -1009,7 +1009,7 @@
         }
         group "E_Config_Module" struct {
             value "name" string: "connman";
-            value "enabled" uchar: 1;
+            value "enabled" uchar: 0;
         }
         group "E_Config_Module" struct {
             value "name" string: "bluez5";
