diff -Naur orig/src/bin/e_main.c patched/src/bin/e_main.c
--- orig/src/bin/e_main.c	2016-10-07 07:36:45.000000000 +0200
+++ patched/src/bin/e_main.c	2017-01-08 01:14:16.613050797 +0100
@@ -90,7 +90,7 @@
 static Eina_Bool _e_main_cb_startup_fake_end(void *data EINA_UNUSED);
 
 /* local variables */
-static Eina_Bool really_know = EINA_FALSE;
+static Eina_Bool really_know = EINA_TRUE;
 static Eina_Bool locked = EINA_FALSE;
 static Eina_Bool inloop = EINA_FALSE;
 static jmp_buf x_fatal_buff;
