diff -Nur orig/PyIlmBase/config/ModuleDefine.cmake mod/PyIlmBase/config/ModuleDefine.cmake
--- orig/PyIlmBase/config/ModuleDefine.cmake	2019-09-18 03:02:06.000000000 +0200
+++ mod/PyIlmBase/config/ModuleDefine.cmake	2020-01-06 10:41:18.439341345 +0100
@@ -139,7 +139,7 @@
     Python3_add_library(${modname}_python3 MODULE ${PYILMBASE_CURMOD_MODSOURCE})
     target_link_libraries(${modname}_python3
       PRIVATE
-        ${libname} ${extraDeps}
+        ${libname} ${extraDeps} Boost::${PYILMBASE_BOOST_PY3_COMPONENT}
         ${PYILMBASE_CURMOD_DEPENDENCIES}
         ${PYILMBASE_CURMOD_PRIVATE_DEPS}
       )
