diff -Naur a/cmake/OpenEXRConfig.cmake.in b/cmake/OpenEXRConfig.cmake.in
--- a/cmake/OpenEXRConfig.cmake.in	2022-04-07 02:52:25.000000000 +0200
+++ b/cmake/OpenEXRConfig.cmake.in	2022-05-18 03:10:46.530087337 +0200
@@ -9,7 +9,6 @@
 endif()
 unset(openexr_needthreads)
 
-find_dependency(ZLIB REQUIRED)
 find_dependency(Imath REQUIRED)
 
 include("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@Targets.cmake")
