diff -Nur orig/Config.cmake mod/Config.cmake
--- orig/Config.cmake	2024-02-02 18:12:04.000000000 +0100
+++ mod/Config.cmake	2024-02-12 17:37:18.403980493 +0100
@@ -4,7 +4,7 @@
 # Essential, non translatable application information (except DESCRIPTION).
 # Translatable strings are passed via code.
 #===============================================================================
-list(APPEND PROJECT_CATEGORIES "Qt;Utility") # Freedesktop menu categories
+list(APPEND PROJECT_CATEGORIES "Settings;Qt;Utility") # Freedesktop menu categories
 list(APPEND PROJECT_KEYWORDS   "sddm;settings;configurator")
 set(PROJECT_AUTHOR_NAME        "Andrea Zanellato")
 set(PROJECT_AUTHOR_EMAIL       "redtid3@gmail.com") # Used also for organization email
