diff -Naur orig/data/sddm-config-editor.desktop mod/data/sddm-config-editor.desktop
--- orig/data/sddm-config-editor.desktop	2015-06-20 20:10:11.000000000 +0200
+++ mod/data/sddm-config-editor.desktop	2015-06-20 20:16:16.317310904 +0200
@@ -1,11 +1,11 @@
 [Desktop Entry]
 Type=Application
 Name=SDDM Configuration
+Name[hu]=SDDM beállító
 GenericName=Display Manager Configuration
 Comment=Qt configuration editor for SDDM
 Categories=Settings;System;Qt
 Icon=preferences-system
-Terminal=false
 Exec=sddm-config-editor
 Terminal=false
 StartupNotify=false
