diff --git a/caribou/antler/main.py b/caribou/antler/main.py
index 3a558d2e76a19bb265c58209518c7e8fb98bf93c..9d3df8787fa31f81d46548f509385b27d9f5ee90 100644
--- a/caribou/antler/main.py
+++ b/caribou/antler/main.py
@@ -1,7 +1,8 @@
 import gi
 gi.require_version('Caribou', '1.0')
 gi.require_version('Gtk', '3.0')
-from gi.repository import Caribou, GLib, GObject, Gtk
+gi.require_version('GtkClutter', '1.0')
+from gi.repository import Caribou, GLib, GObject, Gtk, GtkClutter
 from .window import AntlerWindowEntry
 from .keyboard_view import AntlerKeyboardView
 import sys
@@ -14,6 +15,7 @@ class AntlerKeyboardService(Caribou.KeyboardService, AntlerKeyboardCommand):
     def __init__(self, args=None):
         GObject.GObject.__init__(self)
         self.register_keyboard("Antler")
+        GtkClutter.init(None)
         self.window = AntlerWindowEntry(AntlerKeyboardView)
 
     def run(self):
diff --git a/caribou/antler/window.py b/caribou/antler/window.py
index 09d770a4ad6a76de5866a9f38bb35904c3c9aa6e..553c6e0b4addb2bb2e31fb536db3f8ff8d520de1 100644
--- a/caribou/antler/window.py
+++ b/caribou/antler/window.py
@@ -22,17 +22,19 @@
 
 import gi
 gi.require_version('Clutter', '1.0')
+gi.require_version('GtkClutter', '1.0')
 from gi.repository import Gtk
 from gi.repository import Gdk
 from gi.repository import GObject
 from gi.repository import Clutter
+from gi.repository import GtkClutter
 from .antler_settings import AntlerSettings
 from math import sqrt
 import os
 import sys
 
 
-class AnimatedWindowBase(Gtk.Window, Clutter.Animatable):
+class AnimatedWindowBase(GtkClutter.Window, Clutter.Animatable):
     __gproperties__ = {
         'antler-window-x' : (GObject.TYPE_INT, 'Window position',
                              'Window X coordinate',
@@ -45,10 +47,9 @@ class AnimatedWindowBase(Gtk.Window, Clutter.Animatable):
         }
     def __init__(self):
         GObject.GObject.__init__(self, type=Gtk.WindowType.POPUP)
-        Clutter.init(None)
 
         # animation
-        self._stage = Clutter.Stage.get_default()
+        self._stage = None #not used
         self._move_animation = None
         self._opacity_animation = None
 
