diff -Naur orig/foobillardplus.desktop mod/foobillardplus.desktop
--- orig/foobillardplus.desktop	2011-09-14 01:33:32.000000000 +0200
+++ mod/foobillardplus.desktop	2016-01-16 17:44:10.147140481 +0100
@@ -1,12 +1,11 @@
 
 [Desktop Entry]
-Version=1.0
 Type=Application
 Terminal=false
-Icon=/opt/foobillardplus/foobillardplus.png
-Name=Foobillard++
-Exec=/opt/foobillardplus/bin/foobillardplus
-Comment=OpenGL Billiard Game
+Icon=foobillardplus
 Name=Foobillard++
+Exec=foobillardplus
 Comment=OpenGL Billiard Game
+Comment[hu]=térbeli billiárd játék
 GenericName=OpenGL Billiard Game
+Categories=Game;SportsGame;
