diff -Nur orig/src/billard3d.c mod/src/billard3d.c
--- orig/src/billard3d.c	2012-01-02 20:23:49.000000000 +0100
+++ mod/src/billard3d.c	2016-08-03 13:35:33.842970601 +0200
@@ -7597,7 +7597,7 @@
 
    /* config file */
    load_config( &confv, &confc, argv, argc );
-   while( ( act_option = getopt_long_only(confc, confv, "+", long_options, &option_index) ) >= 0){
+   while( ( act_option = getopt_long(confc, confv, "+", long_options, &option_index) ) >= 0){
        //fprintf(stderr,"processing option %d=%s\n",act_option,optarg);
        process_option(act_option);
    }
