diff -Nur orig/Makefile.am mod/Makefile.am
--- orig/Makefile.am	2011-09-14 01:33:32.000000000 +0200
+++ mod/Makefile.am	2016-08-03 15:00:46.437327478 +0200
@@ -16,12 +16,6 @@
 dist-hook:
 	mkdir -p $(distdir)/data
 	cp -p $(srcdir)/data/* $(distdir)/data -R
-	cp -p $(srcdir)/AUTHORS $(distdir)
-	cp -p $(srcdir)/COPYING $(distdir)
-	cp -p $(srcdir)/INSTALL $(distdir)
-	cp -p $(srcdir)/README $(distdir)
-	cp -p $(srcdir)/ChangeLog $(distdir)
-	cp -p $(srcdir)/TODO $(distdir)
 	cp -p $(srcdir)/foobillardplus.desktop $(distdir)
 	cp -p $(srcdir)/foobillardplus.png $(distdir)
 	cp -p $(srcdir)/foobillardplus.xbm $(distdir)
@@ -29,12 +23,6 @@
 install-data-hook:
 	mkdir -p $(datadir)
 	cp -p $(srcdir)/data/* $(datadir) -R
-	cp -p $(srcdir)/AUTHORS $(prefix)
-	cp -p $(srcdir)/COPYING $(prefix)
-	cp -p $(srcdir)/INSTALL $(prefix)
-	cp -p $(srcdir)/README $(prefix)
-	cp -p $(srcdir)/ChangeLog $(prefix)
-	cp -p $(srcdir)/TODO $(prefix)
 	cp -p $(srcdir)/foobillardplus.desktop $(prefix)
 	cp -p $(srcdir)/foobillardplus.png $(prefix)
 	cp -p $(srcdir)/foobillardplus.xbm $(prefix)
