diff -Naur orig/Makefile.am patched/Makefile.am
--- orig/Makefile.am	2019-08-13 15:48:42.000000000 +0200
+++ patched/Makefile.am	2019-08-16 21:25:33.355878366 +0200
@@ -35,7 +35,7 @@
 SUBDIRS = $(OSC_SUBDIRS) intl $(WEED_SUBDIRS) src m4 lives-plugins po resources
 PACKAGE_CAPITALISED = LiVES
 
-docdir = "$(prefix)/share/doc/$(PACKAGE)-$(VERSION)"
+docdir = "$(prefix)/share/doc/Packages/$(PACKAGE)"
 sharedir = "$(prefix)/share"
 
 if HAVE_DOXYGEN
