diff -Naur orig/menu-entries/LiVES.desktop patched/menu-entries/LiVES.desktop
--- orig/menu-entries/LiVES.desktop	2015-02-08 21:17:13.000000000 +0100
+++ patched/menu-entries/LiVES.desktop	2015-07-16 22:37:41.086383566 +0200
@@ -2,6 +2,7 @@
 Name=LiVES
 GenericName=Video Editor/VJ tool
 Comment=A video editor and VJ program
+Comment[hu]=Videószerkesztő alkalmazás
 Categories=AudioVideo;AudioVideoEditing;
 Exec=lives
 Icon=lives
