diff -Nur orig/lives-plugins/icons/Makefile.am mod/lives-plugins/icons/Makefile.am
--- orig/lives-plugins/icons/Makefile.am	2019-06-12 15:46:00.000000000 +0200
+++ mod/lives-plugins/icons/Makefile.am	2019-10-05 20:10:40.958754223 +0200
@@ -4,7 +4,7 @@
 
 
 install-exec-am:
-	/bin/rm -rf "$(DESTDIR)$(datadir)/$(PACKAGE)/"*
+	rm -rf "$(DESTDIR)$(datadir)/$(PACKAGE)/"*
 iconsdir = "$(datadir)/$(PACKAGE)/icons"
 icons_DATA = \
 lives-audio.png \
