diff -Nur orig/libOSC/sendOSC/Makefile.in mod/libOSC/sendOSC/Makefile.in
--- orig/libOSC/sendOSC/Makefile.in	2019-01-08 18:05:24.000000000 +0100
+++ mod/libOSC/sendOSC/Makefile.in	2019-01-17 19:38:01.393659215 +0100
@@ -467,7 +467,7 @@
 @ENABLE_OSC_TRUE@sendOSC = sendOSC
 @ENABLE_OSC_TRUE@LIBOSCDIR = ../client
 @ENABLE_OSC_TRUE@sendOSCdir = $(bindir)
-@ENABLE_OSC_TRUE@sendOSC_LDFLAGS = $(mingw_LIBFLAGS) $(top_builddir)/libOSC/client/.libs/libOSC_client.a
+@ENABLE_OSC_TRUE@sendOSC_LDFLAGS = $(mingw_LIBFLAGS) $(top_builddir)/libOSC/client/.libs/libOSC_client.so
 @ENABLE_OSC_TRUE@sendOSC_CFLAGS = -I.. $(mingw_CFLAGS)
 @ENABLE_OSC_TRUE@sendOSC_SOURCES = htmsocket.c htmsocket.h sendOSC.c
 all: all-am
diff -Nur orig/src/Makefile.in mod/src/Makefile.in
--- orig/src/Makefile.in	2019-01-08 18:05:25.000000000 +0100
+++ mod/src/Makefile.in	2019-01-17 19:39:43.293482736 +0100
@@ -271,8 +271,8 @@
 lives_exe_OBJECTS = $(am_lives_exe_OBJECTS)
 am__DEPENDENCIES_1 =
 @ENABLE_OSC_TRUE@am__DEPENDENCIES_2 =  \
-@ENABLE_OSC_TRUE@	$(top_builddir)/libOSC/.libs/libOSC.a \
-@ENABLE_OSC_TRUE@	$(top_builddir)/libOSC/client/.libs/libOSC_client.a
+@ENABLE_OSC_TRUE@	$(top_builddir)/libOSC/.libs/libOSC.so \
+@ENABLE_OSC_TRUE@	$(top_builddir)/libOSC/client/.libs/libOSC_client.so
 @DISABLE_GLSLICE_FALSE@@HAVE_SYSTEM_WEED_FALSE@@IS_MINGW_FALSE@am__DEPENDENCIES_3 = $(abs_top_srcdir)/libweed/libweed_slice.la \
 @DISABLE_GLSLICE_FALSE@@HAVE_SYSTEM_WEED_FALSE@@IS_MINGW_FALSE@	$(abs_top_srcdir)/libweed/libweed-utils.la
 @DISABLE_GLSLICE_TRUE@@HAVE_SYSTEM_WEED_FALSE@@IS_MINGW_FALSE@am__DEPENDENCIES_3 = $(abs_top_srcdir)/libweed/libweed.la \
@@ -665,7 +665,7 @@
 @HAVE_UNICAP_TRUE@unicap_SOURCES = videodev.c videodev.h
 @HAVE_UNICAP_TRUE@unicap_CFLAGS = -DHAVE_UNICAP=1
 @ENABLE_OSC_TRUE@osc_SOURCES = osc.c osc.h omc-learn.c omc-learn.h osc_notify.h
-@ENABLE_OSC_TRUE@osc_LDADD = $(top_builddir)/libOSC/.libs/libOSC.a $(top_builddir)/libOSC/client/.libs/libOSC_client.a -lm
+@ENABLE_OSC_TRUE@osc_LDADD = $(top_builddir)/libOSC/.libs/libOSC.so $(top_builddir)/libOSC/client/.libs/libOSC_client.so -lm
 @ENABLE_OSC_TRUE@osc_CFLAGS = -DENABLE_OSC=1
 @ENABLE_LDVGRAB_TRUE@ldvgrab_SOURCES = ldvgrab.c ldvgrab.h ldvcallbacks.c ldvcallbacks.h ldvinterface.c ldvinterface.h
 @ENABLE_LDVGRAB_TRUE@ldvgrab_CFLAGS = -DHAVE_LDVGRAB=1 -I libavc1394/avc1394.h -I libraw1394/raw1394.h -I libraw1394/rom1394.h
