diff -Naur orig/lives-plugins/weed-plugins/toonz.cpp patched/lives-plugins/weed-plugins/toonz.cpp
--- orig/lives-plugins/weed-plugins/toonz.cpp	2021-01-10 17:47:23.355033725 +0100
+++ patched/lives-plugins/weed-plugins/toonz.cpp	2021-01-10 17:48:48.795577666 +0100
@@ -60,6 +60,7 @@
 /////////////////////////////////////////////////////////////
 
 #include <array>
+#include <iostream>
 #include <random>
 
 #define TWO_PI (M_PI * 2.)
