diff -Naur orig/KF5SysGuardConfig.cmake.in patched/KF5SysGuardConfig.cmake.in
--- orig/KF5SysGuardConfig.cmake.in	2018-07-31 14:48:19.000000000 +0200
+++ patched/KF5SysGuardConfig.cmake.in	2018-08-30 01:41:07.061580001 +0200
@@ -1,12 +1,12 @@
 @PACKAGE_INIT@
 
 include(CMakeFindDependencyMacro)
-find_dependency(Qt5Core)
-find_dependency(Qt5Network)
-find_dependency(Qt5Widgets)
+find_dependency(Qt5Core )
+find_dependency(Qt5Network )
+find_dependency(Qt5Widgets )
 
-find_dependency(KF5Config)
-find_dependency(KF5I18n)
-find_dependency(KF5IconThemes)
+find_dependency(KF5Config )
+find_dependency(KF5I18n )
+find_dependency(KF5IconThemes )
 
 include("${CMAKE_CURRENT_LIST_DIR}/KF5SysGuardLibraryTargets.cmake")
