diff -Nur orig/utils/iscsi-gen-initiatorname mod/utils/iscsi-gen-initiatorname
--- orig/utils/iscsi-gen-initiatorname	2019-09-12 21:02:11.000000000 +0200
+++ mod/utils/iscsi-gen-initiatorname	2019-10-08 19:13:20.107686309 +0200
@@ -27,7 +27,7 @@
 	eval $(cat /etc/iscsi/initiatorname.iscsi | sed -e '/^#/d')
 	if [ "$iSCSI_INITIATOR_NAME" != "$InitiatorName" ] ; then
 	    echo "iSCSI Initiatorname from iBFT is different from the current setting."
-	    echo "Please call '/sbin/iscsi-gen-initiatorname -f' to update the iSCSI Initiatorname."
+	    echo "Please call '/usr/sbin/iscsi-gen-initiatorname -f' to update the iSCSI Initiatorname."
 	    exit 1
 	fi
     fi
@@ -66,7 +66,7 @@
 ## for each iSCSI initiator.  Do NOT duplicate iSCSI InitiatorNames.
 EOF
 	ISSUEDATE="1996-04"
-	INAME=$(/sbin/iscsi-iname -p iqn.$ISSUEDATE.de.suse:01)
+	INAME=$(/usr/sbin/iscsi-iname -p iqn.$ISSUEDATE.de.suse:01)
 	printf "InitiatorName=$INAME\n"  >>/etc/iscsi/initiatorname.iscsi
 	chmod 0600 /etc/iscsi/initiatorname.iscsi
 fi
