diff -Naurd b/Makefile c/Makefile
--- b/Makefile	2007-10-25 21:24:09.000000000 +0200
+++ c/Makefile	2007-10-25 21:25:46.000000000 +0200
@@ -72,7 +72,7 @@
 	$(LT_COMPILE) -c $(CPPFLAGS) $(CFLAGS) $(XTRACFLAGS) $<
 
 %: %.o $(LIBAPM)
-	$(LT_LINK) -o $@ $< $(LDFLAGS) $(LIBAPM)
+	$(LT_LINK) -o $@ .libs/$< $(LDFLAGS) $(LIBAPM)
 
 xapm.o: xapm.c
 	$(LT_COMPILE) -c $(CPPFLAGS) $(CFLAGS) $(XTRACFLAGS) -DNARROWPROTO $<
