diff -Naur orig/data/gconf-editor.desktop.in.in patched/data/gconf-editor.desktop.in.in
--- orig/data/gconf-editor.desktop.in.in	2009-05-06 21:44:27.000000000 +0200
+++ patched/data/gconf-editor.desktop.in.in	2016-02-19 01:40:48.942621060 +0100
@@ -1,12 +1,12 @@
 [Desktop Entry]
-_Name=Configuration Editor
-_Comment=Directly edit your entire configuration database
+_Name=GNOME Configuration Editor
+_Comment=Directly edit your entire GNOME configuration database
 Exec=gconf-editor
 Terminal=false
 Type=Application
 Icon=gconf-editor
 StartupNotify=true
-Categories=GNOME;GTK;System;
+Categories=GTK;GNOME;System;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gconf-editor
 X-GNOME-Bugzilla-Component=general
diff -Naur orig/po/hu.po patched/po/hu.po
--- orig/po/hu.po	2009-05-06 21:44:27.000000000 +0200
+++ patched/po/hu.po	2016-02-19 01:39:43.638297233 +0100
@@ -27,9 +27,13 @@
 msgid "Configuration Editor"
 msgstr "Konfigurációszerkesztő"
 
+#: ../data/gconf-editor.desktop.in.in.h:1
+msgid "GNOME Configuration Editor"
+msgstr "GNOME Konfigurációszerkesztő"
+
 #: ../data/gconf-editor.desktop.in.in.h:2
-msgid "Directly edit your entire configuration database"
-msgstr "A konfigurációs adatbázisát szerkeszti közvetlenül"
+msgid "Directly edit your entire GNOME configuration database"
+msgstr "A GNOME konfigurációs adatbázisát szerkeszti közvetlenül"
 
 #: ../data/gconf-editor.schemas.in.h:1
 msgid "Bookmarks"
diff -Naur orig/src/main.c patched/src/main.c
--- orig/src/main.c	2009-05-06 21:44:28.000000000 +0200
+++ patched/src/main.c	2016-02-19 01:39:43.638297233 +0100
@@ -74,7 +74,7 @@
 	bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
 	textdomain (GETTEXT_PACKAGE);
 
-	context = g_option_context_new (N_("- Directly edit your entire configuration database"));
+	context = g_option_context_new (N_("- Directly edit your entire GNOME configuration database"));
 
 	g_option_context_set_translation_domain (context, GETTEXT_PACKAGE);
 	g_option_context_add_main_entries (context, entries, GETTEXT_PACKAGE);
