diff -Nur orig/src/locale/Makefile.in mod/src/locale/Makefile.in
--- orig/src/locale/Makefile.in	2019-10-04 08:39:40.000000000 +0200
+++ mod/src/locale/Makefile.in	2019-10-04 08:43:21.238035478 +0200
@@ -199,7 +199,7 @@
 #so we are going to use the python version instead
 #(TODO check for msgfmt, then python)
 #MSGFMT=msgfmt
-MSGFMT = python $(top_srcdir)/tools/msgfmt.py
+MSGFMT = python2 $(top_srcdir)/tools/msgfmt.py
 all: all-am
 
 .SUFFIXES:
diff -Nur orig/tools/check-license.py mod/tools/check-license.py
--- orig/tools/check-license.py	2012-06-28 08:39:39.000000000 +0200
+++ mod/tools/check-license.py	2019-10-04 08:38:22.078630880 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # ====================================================================
 # Copyright (c) 2002-2009 The RapidSvn Group.  All rights reserved.
diff -Nur orig/tools/check-rapidsvn-el.py mod/tools/check-rapidsvn-el.py
--- orig/tools/check-rapidsvn-el.py	2012-06-28 08:39:39.000000000 +0200
+++ mod/tools/check-rapidsvn-el.py	2019-10-04 08:38:22.078630880 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # ====================================================================
 # Copyright (c) 2002-2008 The RapidSvn Group.  All rights reserved.
diff -Nur orig/tools/check-version.py mod/tools/check-version.py
--- orig/tools/check-version.py	2012-06-28 08:39:39.000000000 +0200
+++ mod/tools/check-version.py	2019-10-04 08:38:22.078630880 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # ====================================================================
 # Copyright (c) 2002-2009 The RapidSvn Group.  All rights reserved.
diff -Nur orig/tools/compile_messages.py mod/tools/compile_messages.py
--- orig/tools/compile_messages.py	2012-06-28 08:39:39.000000000 +0200
+++ mod/tools/compile_messages.py	2019-10-04 08:38:22.078630880 +0200
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2
 # -*- coding: iso-8859-1 -*-
 
 from create_nightly import buildMessages
diff -Nur orig/tools/msgfmt.py mod/tools/msgfmt.py
--- orig/tools/msgfmt.py	2012-06-28 08:39:39.000000000 +0200
+++ mod/tools/msgfmt.py	2019-10-04 08:38:22.078630880 +0200
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2
 # -*- coding: iso-8859-1 -*-
 # Written by Martin v. Löwis <loewis@informatik.hu-berlin.de>
 
