diff -Naur orig/conf/master.cf patched/conf/master.cf
--- orig/conf/master.cf	2019-12-30 18:02:10.000000000 +0100
+++ patched/conf/master.cf	2020-03-24 22:48:57.403031399 +0100
@@ -98,14 +98,14 @@
 # Also specify in main.cf: cyrus_destination_recipient_limit=1
 #
 #cyrus     unix  -       n       n       -       -       pipe
-#  flags=DRX user=cyrus argv=/cyrus/bin/deliver -e -r ${sender} -m ${extension} ${user}
+#  flags=DRX user=cyrus argv=/usr/libexec/cyrus-imapd/deliver -e -r ${sender} -m ${extension} ${user}
 #
 # ====================================================================
 #
 # Old example of delivery via Cyrus.
 #
 #old-cyrus unix  -       n       n       -       -       pipe
-#  flags=R user=cyrus argv=/cyrus/bin/deliver -e -m ${extension} ${user}
+#  flags=R user=cyrus rgv=/usr/libexec/cyrus-imapd/deliver -e -m ${extension} ${user}
 #
 # ====================================================================
 #
