diff -urN ddd-3.3.1.orig/ddd/Makefile.in ddd-3.3.1/ddd/Makefile.in
--- ddd-3.3.1.orig/ddd/Makefile.in	2001-04-30 11:12:43.000000000 +0200
+++ ddd-3.3.1/ddd/Makefile.in	2003-05-17 11:16:06.000000000 +0200
@@ -187,7 +187,7 @@
 
 # Shared data installation
 dddrootdir = $(pkgdatadir)-$(VERSION)
-dddappdir = $(pkgdatadir)-$(VERSION)/$(ddd)
+dddappdir = /usr/share/X11/app-defaults
 
 # -----------------------------------------------------------------------------
 # DDD Sources.
