diff -Naurdp elinks-0.12pre5/features.conf elinks-0.12pre5.features/features.conf
--- elinks-0.12pre5/features.conf	2009-07-07 14:23:17.000000000 +0200
+++ elinks-0.12pre5.features/features.conf	2010-08-18 19:10:28.000000000 +0200
@@ -515,7 +515,7 @@ CONFIG_MOUSE=yes
 #
 # Default: disabled
 
-CONFIG_88_COLORS=no
+CONFIG_88_COLORS=yes
 
 ### 256 Colors in Terminals
 #
@@ -543,7 +543,7 @@ CONFIG_88_COLORS=no
 #
 # Default: disabled
 
-CONFIG_256_COLORS=no
+CONFIG_256_COLORS=yes
 
 ### True color
 #
@@ -552,7 +552,7 @@ CONFIG_256_COLORS=no
 #
 # Default: disabled
 
-CONFIG_TRUE_COLOR=no
+CONFIG_TRUE_COLOR=yes
 
 ### Ex-mode Interface
 #
