diff -Nur orig/inkscape.desktop.in mod/inkscape.desktop.in
--- orig/inkscape.desktop.in	2017-01-02 10:44:59.000000000 +0100
+++ mod/inkscape.desktop.in	2017-02-02 13:32:49.987533460 +0100
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Version=1.0
 _Name=Inkscape
 _GenericName=Vector Graphics Editor
 _X-GNOME-FullName=Inkscape Vector Graphics Editor
@@ -13,9 +12,4 @@
 Terminal=false
 StartupNotify=true
 Icon=inkscape
-X-Ayatana-Desktop-Shortcuts=Drawing
 
-[X-Drawing Shortcut Group]
-_Name=New Drawing
-Exec=inkscape
-TargetEnvironment=Unity
diff -Nur orig/inkscape.desktop.template mod/inkscape.desktop.template
--- orig/inkscape.desktop.template	2017-01-02 10:44:59.000000000 +0100
+++ mod/inkscape.desktop.template	2017-02-02 13:52:18.369349464 +0100
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Version=1.0
 Name=Inkscape
 GenericName=Vector Graphics Editor
 X-GNOME-FullName=Inkscape Vector Graphics Editor
@@ -13,9 +12,3 @@
 Terminal=false
 StartupNotify=true
 Icon=inkscape
-X-Ayatana-Desktop-Shortcuts=Drawing
-
-[Drawing Shortcut Group]
-Name=New Drawing
-Exec=inkscape
-TargetEnvironment=Unity
