diff -Naur orig/inkscape.desktop.in patched/inkscape.desktop.in
--- orig/inkscape.desktop.in	2018-03-11 21:38:09.000000000 +0100
+++ patched/inkscape.desktop.in	2018-06-12 00:43:16.311001657 +0200
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Version=1.0
 _Name=Inkscape
 _GenericName=Vector Graphics Editor
 _X-GNOME-FullName=Inkscape Vector Graphics Editor
@@ -13,9 +12,3 @@
 Terminal=false
 StartupNotify=true
 Icon=inkscape
-X-Ayatana-Desktop-Shortcuts=Drawing
-
-[X-Drawing Shortcut Group]
-_Name=New Drawing
-Exec=inkscape
-TargetEnvironment=Unity
diff -Naur orig/inkscape.desktop.template patched/inkscape.desktop.template
--- orig/inkscape.desktop.template	2018-03-11 21:38:09.000000000 +0100
+++ patched/inkscape.desktop.template	2018-06-12 00:43:29.859182249 +0200
@@ -13,9 +13,3 @@
 Terminal=false
 StartupNotify=true
 Icon=${INKSCAPE_ICONPATH}
-X-Ayatana-Desktop-Shortcuts=Drawing
-
-[Drawing Shortcut Group]
-Name=New Drawing
-Exec=inkscape
-TargetEnvironment=Unity
