diff -Naur a/po/POTFILES.inx.in b/po/POTFILES.inx.in
--- a/po/POTFILES.inx.in	2022-02-04 23:12:35.000000000 +0100
+++ b/po/POTFILES.inx.in	2022-03-17 12:12:12.330201294 +0100
@@ -34,7 +34,6 @@
 ../share/extensions/edge3d.inx
 ../share/extensions/eps_input.inx
 ../share/extensions/export_gimp_palette.inx
-../share/extensions/extension-manager-bootstrap.inx
 ../share/extensions/extrude.inx
 ../share/extensions/fig_input.inx
 ../share/extensions/flatten.inx
diff -Naur a/share/CMakeLists.txt b/share/CMakeLists.txt
--- a/share/CMakeLists.txt	2022-02-04 23:12:35.000000000 +0100
+++ b/share/CMakeLists.txt	2022-03-17 12:17:29.930955537 +0100
@@ -25,13 +25,17 @@
     PATTERN "tests" EXCLUDE
     PATTERN "*.pyc" EXCLUDE
     PATTERN "print_win32_vector.*" EXCLUDE
-    PATTERN "extension-manager-bootstrap.*" EXCLUDE
+    PATTERN "other/inkman" EXCLUDE
+    PATTERN "other/*/*/tests" EXCLUDE
+    PATTERN "other/*/*/MANIFEST.in" EXCLUDE
+    PATTERN "other/*/*/requirements.txt" EXCLUDE
 )
 
-install(
-    FILES extensions/extension-manager-bootstrap.py extensions/extension-manager-bootstrap.inx
-    DESTINATION ${INKSCAPE_SHARE_INSTALL}/extensions
+install(DIRECTORY extensions/other/inkman DESTINATION ${INKSCAPE_SHARE_INSTALL}/extensions/inkman/
     COMPONENT extension_manager
+    PATTERN "tests" EXCLUDE
+    PATTERN "MANIFEST.in" EXCLUDE
+    PATTERN "requirements.txt" EXCLUDE
 )
 
 if(WIN32)
