diff -Naur orig/settings/xfce4-power-manager-settings.desktop.in patched/settings/xfce4-power-manager-settings.desktop.in
--- orig/settings/xfce4-power-manager-settings.desktop.in	2020-11-04 09:57:46.000000000 +0100
+++ patched/settings/xfce4-power-manager-settings.desktop.in	2020-12-01 21:05:45.180830108 +0100
@@ -9,7 +9,7 @@
 Type=Application
 Categories=XFCE;GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-HardwareSettings;
 _Keywords=settings;preferences;buttons;sleep;hibernate;battery;suspend;shutdown;brightness;laptop lid;lock screen;plugged in;saving;critical;
-NotShowIn=GNOME;KDE;Unity;
+OnlyShowIn=XFCE;
 StartupNotify=true
 X-XfcePluggable=true
 X-XfceHelpComponent=xfce4-power-manager
diff -Naur orig/src/xfce4-power-manager.desktop.in patched/src/xfce4-power-manager.desktop.in
--- orig/src/xfce4-power-manager.desktop.in	2019-05-17 15:43:56.000000000 +0200
+++ patched/src/xfce4-power-manager.desktop.in	2020-12-01 21:06:08.836830308 +0100
@@ -5,5 +5,5 @@
 Exec=xfce4-power-manager
 Terminal=false
 Type=Application
-NotShowIn=GNOME;KDE;Unity;MATE;LXQt;
+OnlyShowIn=XFCE;
 StartupNotify=false
