diff -Nur orig/settings/xfce4-power-manager-settings.desktop.in mod/settings/xfce4-power-manager-settings.desktop.in
--- orig/settings/xfce4-power-manager-settings.desktop.in	2019-05-17 15:43:56.000000000 +0200
+++ mod/settings/xfce4-power-manager-settings.desktop.in	2019-05-20 20:36:34.096031025 +0200
@@ -8,7 +8,7 @@
 Terminal=false
 Type=Application
 Categories=XFCE;GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-HardwareSettings;
-NotShowIn=GNOME;KDE;Unity;
+OnlyShowIn=XFCE;
 StartupNotify=true
 X-XfcePluggable=true
 X-XfceHelpComponent=xfce4-power-manager
diff -Nur orig/src/xfce4-power-manager.desktop.in mod/src/xfce4-power-manager.desktop.in
--- orig/src/xfce4-power-manager.desktop.in	2019-05-17 15:43:56.000000000 +0200
+++ mod/src/xfce4-power-manager.desktop.in	2019-05-20 20:37:05.184048178 +0200
@@ -5,5 +5,5 @@
 Exec=xfce4-power-manager
 Terminal=false
 Type=Application
-NotShowIn=GNOME;KDE;Unity;MATE;LXQt;
+OnlyShowIn=XFCE;
 StartupNotify=false
