diff -Naur xfce4-power-manager-1.6.1-orig/settings/xfce4-power-manager-settings.desktop.in xfce4-power-manager-1.6.1/settings/xfce4-power-manager-settings.desktop.in
--- xfce4-power-manager-1.6.1-orig/settings/xfce4-power-manager-settings.desktop.in	2017-11-24 00:52:06.000000000 +0100
+++ xfce4-power-manager-1.6.1/settings/xfce4-power-manager-settings.desktop.in	2017-12-30 19:02:43.100294857 +0100
@@ -8,7 +8,7 @@
 Terminal=false
 Type=Application
 Categories=XFCE;GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-HardwareSettings;
-NotShowIn=GNOME;KDE;Unity;
+OnlyShowIn=XFCE;
 StartupNotify=true
 X-XfcePluggable=true
 X-XfceHelpComponent=xfce4-power-manager
diff -Naur xfce4-power-manager-1.6.1-orig/src/xfce4-power-manager.desktop.in xfce4-power-manager-1.6.1/src/xfce4-power-manager.desktop.in
--- xfce4-power-manager-1.6.1-orig/src/xfce4-power-manager.desktop.in	2017-11-24 00:52:06.000000000 +0100
+++ xfce4-power-manager-1.6.1/src/xfce4-power-manager.desktop.in	2017-12-30 18:55:43.213837242 +0100
@@ -5,5 +5,5 @@
 Exec=xfce4-power-manager
 Terminal=false
 Type=Application
-NotShowIn=GNOME;KDE;Unity;MATE;
+OnlyShowIn=XFCE;
 StartupNotify=false
