diff -Nur orig/share/vlc.desktop.in mod/share/vlc.desktop.in
--- orig/share/vlc.desktop.in	2023-02-11 10:33:19.000000000 +0100
+++ mod/share/vlc.desktop.in	2024-02-05 15:16:06.874551523 +0100
@@ -219,7 +219,7 @@
 Name[zh_TW]=VLC 媒體播放器
 GenericName[zh_TW]=媒體播放器
 Comment[zh_TW]=讀取、擷取與廣播您的多媒體串流
-Exec=@bindir@/vlc --started-from-file %U
+Exec=@bindir@/vlc %U
 TryExec=@bindir@/vlc
 Icon=vlc
 Terminal=false
