diff -Naur orig/librecad/src/lib/engine/rs_ellipse.cpp patched/librecad/src/lib/engine/rs_ellipse.cpp
--- orig/librecad/src/lib/engine/rs_ellipse.cpp	2016-09-23 07:34:43.000000000 +0200
+++ patched/librecad/src/lib/engine/rs_ellipse.cpp	2021-05-01 20:25:33.359716898 +0200
@@ -42,13 +42,10 @@
 #ifdef EMU_C99
 #include "emu_c99.h" /* C99 math */
 #endif
-// Workaround for Qt bug: https://bugreports.qt-project.org/browse/QTBUG-22829
-// TODO: the Q_MOC_RUN detection shouldn't be necessary after this Qt bug is resolved
-#ifndef Q_MOC_RUN
 #include <boost/version.hpp>
 #include <boost/math/tools/roots.hpp>
+#include <boost/math/tools/tuple.hpp>
 #include <boost/math/special_functions/ellint_2.hpp>
-#endif
 
 namespace{
 //functor to solve for distance, used by snapDistance
