diff -Nur orig/librecad/src/ui/generic/widgetcreator.h mod/librecad/src/ui/generic/widgetcreator.h
--- orig/librecad/src/ui/generic/widgetcreator.h	2016-09-23 07:34:43.000000000 +0200
+++ mod/librecad/src/ui/generic/widgetcreator.h	2018-06-11 16:59:23.235189407 +0200
@@ -4,6 +4,7 @@
 #include <QFrame>
 #include <QList>
 #include <QMap>
+#include  <QActionGroup>
 
 class QListWidgetItem;
 class QActionGroup;
