diff -Naur orig/configure patched/configure
--- orig/configure	2018-01-30 10:06:04.000000000 +0100
+++ patched/configure	2020-04-01 20:20:42.382757104 +0200
@@ -16307,9 +16307,9 @@
 
 $as_echo "#define USE_ENCHANT 1" >>confdefs.h
 
-			LIBS="$LIBS `$PKG_CONFIG --libs enchant`"
+			LIBS="$LIBS `$PKG_CONFIG --libs enchant-2`"
 
-						if $PKG_CONFIG --atleast-version 1.4.2 enchant ; then
+						if $PKG_CONFIG --atleast-version 2.0.0 enchant-2 ; then
 				CFLAGS="$CFLAGS `$PKG_CONFIG --cflags dbus-glib-1`"
 				LIBS="$LIBS `$PKG_CONFIG --libs dbus-glib-1`"
 			fi
diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2018-01-30 09:49:52.000000000 +0100
+++ patched/configure.ac	2020-04-01 20:20:39.010708271 +0200
@@ -319,10 +319,10 @@
 		LIBS="$LIBS `$PKG_CONFIG --libs gtkspell-2.0`"
 		if $PKG_CONFIG --atleast-version 2.0.13 gtkspell-2.0 ; then
 			AC_DEFINE(USE_ENCHANT, 1, Use Enchant with GtkSpell)
-			LIBS="$LIBS `$PKG_CONFIG --libs enchant`"
+			LIBS="$LIBS `$PKG_CONFIG --libs enchant-2`"
 
 			dnl Workaroud for zemberek crash
-			if $PKG_CONFIG --atleast-version 1.4.2 enchant ; then
+			if $PKG_CONFIG --atleast-version 2.0.0 enchant-2 ; then
 				CFLAGS="$CFLAGS `$PKG_CONFIG --cflags dbus-glib-1`"
 				LIBS="$LIBS `$PKG_CONFIG --libs dbus-glib-1`"
 			fi
diff -Naur orig/src/compose.c patched/src/compose.c
--- orig/src/compose.c	2017-11-17 01:58:43.000000000 +0100
+++ patched/src/compose.c	2020-04-01 20:17:00.651541799 +0200
@@ -71,7 +71,7 @@
 #  include <gtk/gtkradiomenuitem.h>
 #  include <gtkspell/gtkspell.h>
 #if USE_ENCHANT
-#  include <enchant/enchant.h>
+#  include <enchant-2/enchant.h>
 #else
 #  include <aspell.h>
 #endif
