diff -Nur orig/data/plugins/NotificationPlugin.py mod/data/plugins/NotificationPlugin.py
--- orig/data/plugins/NotificationPlugin.py	2021-12-04 12:38:36.000000000 +0100
+++ mod/data/plugins/NotificationPlugin.py	2021-12-30 20:25:59.000000000 +0100
@@ -20,6 +20,8 @@
 
 
 from Plugin import Plugin
+import gi
+gi.require_version('Notify', '0.7')
 from gi.repository import Notify, GdkPixbuf
 from lib.common import APP_ICON, APPNAME
 from events.EventManager import EventManager
diff -Nur orig/src/AudioPlayerGStreamer.py mod/src/AudioPlayerGStreamer.py
--- orig/src/AudioPlayerGStreamer.py	2021-12-04 12:38:36.000000000 +0100
+++ mod/src/AudioPlayerGStreamer.py	2021-12-30 20:23:25.000000000 +0100
@@ -18,7 +18,8 @@
 #
 ##########################################################################
 import sys, os
-
+import gi
+gi.require_version('Gst', '1.0')
 from gi.repository import Gst
 
 from .StreamDecoder import StreamDecoder
diff -Nur orig/src/XmlDataProvider.py mod/src/XmlDataProvider.py
--- orig/src/XmlDataProvider.py	2021-12-04 12:38:36.000000000 +0100
+++ mod/src/XmlDataProvider.py	2021-12-30 20:15:27.000000000 +0100
@@ -20,6 +20,8 @@
 ##########################################################################
 import os
 from lxml import etree
+import gi
+gi.require_version('Gtk', '3.0')
 from gi.repository import Gtk
 import logging
 
