diff -Naur a/settings/doc/nvidia-settings.desktop b/settings/doc/nvidia-settings.desktop
--- a/settings/doc/nvidia-settings.desktop	2023-03-16 21:27:30.000000000 +0100
+++ b/settings/doc/nvidia-settings.desktop	2023-04-05 20:16:04.764905645 +0200
@@ -1,11 +1,13 @@
 [Desktop Entry]
 Type=Application
-Encoding=UTF-8
 Name=NVIDIA X Server Settings
 Comment=Configure NVIDIA X Server Settings
-Exec=__UTILS_PATH__/nvidia-settings
+Exec=nvidia-settings
 Icon=nvidia-settings
-Categories=__NVIDIA_SETTINGS_DESKTOP_CATEGORIES__
+Categories=Settings;HardwareSettings;
+
+Name[hu]=NVIDIA X szerver beállítások
+Comment[hu]=Az NVIDIA megjelenítő beállításai
 
 # Translation by Marcin Mikołajczak
 Name[pl]=Ustawienia serwera X NVIDIA
