diff -Nur orig/settings/doc/nvidia-settings.desktop mod/settings/doc/nvidia-settings.desktop
--- orig/settings/doc/nvidia-settings.desktop	2021-04-23 01:41:14.000000000 +0200
+++ mod/settings/doc/nvidia-settings.desktop	2021-05-18 11:34:10.987592037 +0200
@@ -1,11 +1,12 @@
 [Desktop Entry]
 Type=Application
-Encoding=UTF-8
 Name=NVIDIA X Server Settings
 Comment=Configure NVIDIA X Server Settings
-Exec=__UTILS_PATH__/nvidia-settings
-Icon=__PIXMAP_PATH__/nvidia-settings.png
-Categories=__NVIDIA_SETTINGS_DESKTOP_CATEGORIES__
+Exec=nvidia-settings
+Icon=nvidia-settings
+Categories=Settings;HardwareSettings;
+Name[hu]=NVIDIA X szerver beállítások
+Comment[hu]=Az NVIDIA megjelenítő beállításai
 
 # Translation by Marcin Mikołajczak
 Name[pl]=Ustawienia serwera X NVIDIA
