diff -Naur liferea-1.11.7-orig/net.sf.liferea.gschema.xml.in liferea-1.11.7/net.sf.liferea.gschema.xml.in
--- liferea-1.11.7-orig/net.sf.liferea.gschema.xml.in	2016-01-30 20:47:20.000000000 +0100
+++ liferea-1.11.7/net.sf.liferea.gschema.xml.in	2016-07-27 23:12:41.185065038 +0200
@@ -13,12 +13,12 @@
       <description>Selects which key to use to pagedown or go to the next unread item. Set to 0 to use space, 1 to use ctrl-space, or 2 to use alt-space.</description>
     </key>
     <key name="browser" type="s">
-      <default>'mozilla %s'</default>
+      <default>'browser %s'</default>
       <summary>Selects the browser command to use when browser_module is set to manual</summary>
       <description>Selects the browser command to use when browser_module is set to manual.</description>
     </key>
     <key name="browser-id" type="s">
-      <default>'gnome'</default>
+      <default>'manual'</default>
       <summary>Selects which browser to use to open external links</summary>
       <description>Selects which browser to use to open external links. The choices include "default" and "manual".</description>
     </key>
