diff -Naur a/net.sf.liferea.gschema.xml.in b/net.sf.liferea.gschema.xml.in
--- a/net.sf.liferea.gschema.xml.in	2023-01-10 21:12:42.000000000 +0100
+++ b/net.sf.liferea.gschema.xml.in	2023-01-30 23:59:50.065183924 +0100
@@ -18,7 +18,7 @@
       <description>Selects the browser command to use when browser_module is set to manual.</description>
     </key>
     <key name="browser-id" type="s">
-      <default>'gnome'</default>
+      <default>'manual'</default>
       <summary>Selects which browser to use to open external links</summary>
       <description>Selects which browser to use to open external links. The choices include "default" and "manual".</description>
     </key>
