diff -Nur orig/config.template mod/config.template
--- orig/config.template	2020-09-01 09:35:52.000000000 +0200
+++ mod/config.template	2020-12-13 16:02:58.275366796 +0100
@@ -13,7 +13,7 @@
 #BACKEND = sdl
 #BACKEND = sdl2
 #BACKEND = mixer_sdl
-#BACKEND = mixer_sdl2
+BACKEND = mixer_sdl2
 #BACKEND = posix
 
 #OSTYPE = amiga
@@ -21,7 +21,7 @@
 #OSTYPE = cygwin
 #OSTYPE = freebsd
 #OSTYPE = haiku
-#OSTYPE = linux
+OSTYPE = linux
 #OSTYPE = mingw
 #OSTYPE = mac
 
@@ -57,7 +57,7 @@
 #SDL2_CONFIG    = sdl2-config
 #FREETYPE_CONFIG    = freetype-config
 
-#VERBOSE = 1
+VERBOSE = 1
 
 # Use static libraries instead
 # STATIC = 1
