diff -Nur simutrans-src-120-3/config.template simutrans-src-120-3.patched/config.template
--- simutrans-src-120-3/config.template	2018-05-18 15:03:52.000000000 +0200
+++ simutrans-src-120-3.patched/config.template	2018-07-08 08:20:04.148231356 +0200
@@ -8,18 +8,18 @@
 #BACKEND = opengl
 #BACKEND = sdl
 #BACKEND = sdl2
-#BACKEND = mixer_sdl
+BACKEND = mixer_sdl
 #BACKEND = posix
 
 #COLOUR_DEPTH = 0
-#COLOUR_DEPTH = 16
+COLOUR_DEPTH = 16
 
 #OSTYPE = amiga
 #OSTYPE = beos
 #OSTYPE = cygwin
 #OSTYPE = freebsd
 #OSTYPE = haiku
-#OSTYPE = linux
+OSTYPE = linux
 #OSTYPE = mingw
 #OSTYPE = mac
 
@@ -52,7 +52,7 @@
 #SDL2_CONFIG    = sdl2-config
 #FREETYPE_CONFIG    = freetype-config
 
-#VERBOSE = 1
+VERBOSE = 1
 
 # Use static libraries instead
 # STATIC = 1
