diff -Naur a/simmain.cc b/simmain.cc
--- a/simmain.cc	2022-01-28 12:17:58.000000000 +0100
+++ b/simmain.cc	2022-05-19 21:00:53.536788728 +0200
@@ -531,8 +531,7 @@
 		strcat( env_t::data_dir, PATH_SEPARATOR );
 	}
 	else {
-		strcpy( env_t::data_dir, argv[0] );
-		*(strrchr( env_t::data_dir, PATH_SEPARATOR[0] )+1) = 0;
+		strcpy(env_t::data_dir, "/usr/share/games/simutrans/");
 
 #ifdef __APPLE__
 		// Detect if the binary is started inside an application bundle
