diff -Nur simutrans-src-120-3/simsys.cc simutrans-src-120-3.patched/simsys.cc
--- simutrans-src-120-3/simsys.cc	2018-05-19 14:34:54.000000000 +0200
+++ simutrans-src-120-3.patched/simsys.cc	2018-07-08 08:16:35.292497562 +0200
@@ -357,7 +357,7 @@
 	find_directory(B_USER_DIRECTORY, &userDir);
 	sprintf(buffer, "%s/simutrans", userDir.Path());
 #else
-	sprintf(buffer, "%s/simutrans", getenv("HOME"));
+	sprintf(buffer, "%s/.simutrans", getenv("HOME"));
 #endif
 
 	// create directory and subdirectories
