diff -Naur trunk-r228-orig/configure.ac trunk-r228/configure.ac
--- trunk-r228-orig/configure.ac	2016-07-22 23:01:46.000000000 +0200
+++ trunk-r228/configure.ac	2016-07-23 10:11:36.023055752 +0200
@@ -23,7 +23,7 @@
 dnl keep autools files in a subdirectory
 AC_CONFIG_AUX_DIR(autotools)
 # which also contains a subdir containing macros
-AC_CONFIG_MACRO_DIR([autotools/m4])
+dnl AC_CONFIG_MACRO_DIR([autotools/m4])
 dnl set up automake
 AM_INIT_AUTOMAKE([-Wall foreign])
 dnl Audacity policy: don't enable automatic rebuild of configure et al if 
@@ -49,7 +49,7 @@
 dnl we will need an "install" program to be available
 AC_PROG_INSTALL
 dnl and ranlib for static libraries
-AC_PROG_RANLIB
+AC_PROG_LIBTOOL
 
 dnl Make the install prefix available to the program so that the pkg-config file
 dnl can be created correctly
diff -Naur trunk-r228-orig/Makefile.am trunk-r228/Makefile.am
--- trunk-r228-orig/Makefile.am	2016-07-22 23:01:46.000000000 +0200
+++ trunk-r228/Makefile.am	2016-07-23 10:13:20.843053001 +0200
@@ -6,10 +6,10 @@
 ACLOCAL_AMFLAGS = -I autotools/m4
 
 # define the output library (link with -lportSMF)
-lib_LIBRARIES = libportSMF.a
+lib_LTLIBRARIES = libportSMF.la
 
 # define the source files to use
-libportSMF_a_SOURCES = allegro.cpp \
+libportSMF_la_SOURCES = allegro.cpp \
 	allegrosmfwr.cpp \
 	allegrord.cpp \
 	allegrowr.cpp \
