diff -Naur orig/Makefile.am patched/Makefile.am
--- orig/Makefile.am	2011-05-18 21:01:01.000000000 +0200
+++ patched/Makefile.am	2019-08-10 20:37:07.831729925 +0200
@@ -7,11 +7,11 @@
 tempfile_SOURCES = tempfile.c
 ischroot_SOURCES = ischroot.c
 
-bin_SCRIPTS = which savelog
+bin_SCRIPTS = savelog
 
-sbin_SCRIPTS = installkernel add-shell remove-shell
+sbin_SCRIPTS = add-shell remove-shell
 
 man_MANS = run-parts.8 \
-	   installkernel.8 savelog.8 \
-	   tempfile.1 which.1 add-shell.8 \
+	   savelog.8 \
+	   tempfile.1 add-shell.8 \
 	   remove-shell.8 ischroot.1
diff -Naur orig/Makefile.in patched/Makefile.in
--- orig/Makefile.in	2018-05-15 13:59:24.000000000 +0200
+++ patched/Makefile.in	2019-08-10 20:40:46.045897716 +0200
@@ -356,11 +356,11 @@
 run_parts_SOURCES = run-parts.c
 tempfile_SOURCES = tempfile.c
 ischroot_SOURCES = ischroot.c
-bin_SCRIPTS = which savelog
-sbin_SCRIPTS = installkernel add-shell remove-shell
+bin_SCRIPTS = savelog
+sbin_SCRIPTS = add-shell remove-shell
 man_MANS = run-parts.8 \
-	   installkernel.8 savelog.8 \
-	   tempfile.1 which.1 add-shell.8 \
+	   savelog.8 \
+	   tempfile.1 add-shell.8 \
 	   remove-shell.8 ischroot.1
 
 all: all-recursive
