diff -Naur orig/src/system.c patched/src/system.c
--- orig/src/system.c	2019-09-08 13:21:26.000000000 +0200
+++ patched/src/system.c	2020-02-07 01:36:08.935661513 +0100
@@ -38,7 +38,6 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/param.h>
-#include <sys/sysctl.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <net/if.h>
diff -Naur orig/src/tingle/tingle.c patched/src/tingle/tingle.c
--- orig/src/tingle/tingle.c	2019-09-08 13:21:26.000000000 +0200
+++ patched/src/tingle/tingle.c	2020-02-07 01:33:01.313155868 +0100
@@ -38,7 +38,6 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/param.h>
-#include <sys/sysctl.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <net/if.h>
