diff -Naur orig/makefile patched/makefile
--- orig/makefile	2018-06-19 10:49:10.000000000 +0200
+++ patched/makefile	2018-09-04 02:08:24.009579255 +0200
@@ -1,6 +1,8 @@
 LIBS=
 
-PREFIX= /usr/local
+PREFIX= /usr
+
+DESTDIR=
 
 OSNAME := $(shell uname -s)
 
@@ -26,9 +28,9 @@
 	$(MAKE) -C src clean
 
 install:
-	-mkdir -p $(PREFIX)/share/pixmaps
-	-mkdir -p $(PREFIX)/share/applications
-	-mkdir -p $(PREFIX)/bin
-	install -m 0644 data/evisum.png $(PREFIX)/share/pixmaps
-	install -m 0644 data/evisum.desktop $(PREFIX)/share/applications
-	install -m 0755 evisum $(PREFIX)/bin
+	-mkdir -p $(DESTDIR)$(PREFIX)/share/pixmaps
+	-mkdir -p $(DESTDIR)$(PREFIX)/share/applications
+	-mkdir -p $(DESTDIR)$(PREFIX)/bin
+	install -m 0644 data/evisum.png $(DESTDIR)$(PREFIX)/share/pixmaps
+	install -m 0644 data/evisum.desktop $(DESTDIR)$(PREFIX)/share/applications
+	install -m 0755 evisum $(DESTDIR)$(PREFIX)/bin
