diff -Nur orig/util/install_helper.sh mod/util/install_helper.sh
--- orig/util/install_helper.sh	2017-09-12 15:50:16.000000000 +0200
+++ mod/util/install_helper.sh	2017-09-14 09:01:01.380577066 +0200
@@ -16,7 +16,7 @@
 
 if test ! -e "${DESTDIR}/dev/fuse"; then
     mkdir -p "${DESTDIR}/dev"
-    mknod "${DESTDIR}/dev/fuse" -m 0666 c 10 229
+#    mknod "${DESTDIR}/dev/fuse" -m 0666 c 10 229
 fi
 
 install -D -m 644 "${MESON_SOURCE_ROOT}/util/udev.rules" \
