--- paman-0.9.4.orig/src/paman.desktop	2006-06-21 01:35:19.000000000 +0200
+++ paman-0.9.4/src/paman.desktop	2009-05-28 16:52:53.000000000 +0200
@@ -1,9 +1,10 @@
 [Desktop Entry]
-Encoding=UTF-8
 Name=PulseAudio Manager
+Name[hu]=PulseAudio menedzser
 Comment=Manager your PulseAudio sound server.
+Comment[hu]=A PulseAudio hangszerver kezelése.
 Exec=paman
 Icon=audio-card
 StartupNotify=true
 Type=Application
-Categories=Application;AudioVideo;Audio;
+Categories=GTK;AudioVideo;Audio;
