diff -Naur gettext-0.14.5.orig/gettext-tools/examples/hello-c++-kde/admin/ltmain.sh gettext-0.14.5/gettext-tools/examples/hello-c++-kde/admin/ltmain.sh
--- gettext-0.14.5.orig/gettext-tools/examples/hello-c++-kde/admin/ltmain.sh	2004-01-17 14:46:16.000000000 +0100
+++ gettext-0.14.5/gettext-tools/examples/hello-c++-kde/admin/ltmain.sh	2006-02-28 18:58:33.000000000 +0100
@@ -3822,7 +3822,7 @@
 	    fi
 
 	    # Try sorting and uniquifying the output.
-	    if grep -v "^: " < "$nlist" | sort +2 | uniq > "$nlist"S; then
+	    if grep -v "^: " < "$nlist" | sort -k 3 | uniq > "$nlist"S; then
 	      :
 	    else
 	      grep -v "^: " < "$nlist" > "$nlist"S
