diff -Nur orig/src/gameconfigfile.cpp mod/src/gameconfigfile.cpp
--- orig/src/gameconfigfile.cpp	2016-02-22 03:11:13.000000000 +0100
+++ mod/src/gameconfigfile.cpp	2016-11-22 22:37:23.907373055 +0100
@@ -140,6 +140,7 @@
 		SetValueForKey ("Path", "/usr/local/share/games/doom", true);
 		SetValueForKey ("Path", "/usr/share/doom", true);
 		SetValueForKey ("Path", "/usr/share/games/doom", true);
+		SetValueForKey ("Path", "/usr/share/games/heretic", true);
 #endif
 	}
 
