Index: vice/src/arch/gtk3/widgets/base/carthelpers.h
===================================================================
--- vice/src/arch/gtk3/widgets/base/carthelpers.h	(revision 37519)
+++ vice/src/arch/gtk3/widgets/base/carthelpers.h	(revision 37520)
@@ -31,13 +31,13 @@
 #include "vice.h"
 #include <gtk/gtk.h>
 
-int (*carthelpers_save_func)(int type, const char *filename);
-int (*carthelpers_flush_func)(int type);
-int (*carthelpers_is_enabled_func)(int type);
-int (*carthelpers_enable_func)(int type);
-int (*carthelpers_disable_func)(int type);
-int (*carthelpers_can_save_func)(int type);
-int (*carthelpers_can_flush_func)(int type);
+extern int (*carthelpers_save_func)(int type, const char *filename);
+extern int (*carthelpers_flush_func)(int type);
+extern int (*carthelpers_is_enabled_func)(int type);
+extern int (*carthelpers_enable_func)(int type);
+extern int (*carthelpers_disable_func)(int type);
+extern int (*carthelpers_can_save_func)(int type);
+extern int (*carthelpers_can_flush_func)(int type);
 
 
 void carthelpers_set_functions(
Index: vice/src/c128/z80.c
===================================================================
--- vice/src/c128/z80.c	(revision 37526)
+++ vice/src/c128/z80.c	(revision 37527)
@@ -444,8 +444,10 @@
 
 /* ------------------------------------------------------------------------- */
 
-z80_regs_t z80_regs;
+/* FIXME: src/c64/carts/cpmcart.c defines z80_regs too */
 
+/* extern z80_regs_t z80_regs; */
+
 static void import_registers(void)
 {
     reg_a = z80_regs.reg_af >> 8;
Index: vice/src/rs232drv/rs232.h
===================================================================
--- vice/src/rs232drv/rs232.h	(revision 37526)
+++ vice/src/rs232drv/rs232.h	(revision 37527)
@@ -78,7 +78,7 @@
 #define RS232_NUM_DEVICES 4
 
 extern char *rs232_devfile[RS232_NUM_DEVICES];
-int rs232_useip232[RS232_NUM_DEVICES];
+extern int rs232_useip232[RS232_NUM_DEVICES];
 
 /* the "ip232" protocol used by tcpser 
  
