diff -Nur orig/configure.ac mod/configure.ac
--- orig/configure.ac	2018-05-19 19:22:25.000000000 +0200
+++ mod/configure.ac	2018-06-10 17:39:52.372565826 +0200
@@ -4999,7 +4999,7 @@
     ;;
   linux*)
     if test x"$host_cpu" = "xx86_64"; then
-      VICEDIR="$PREFIX/lib64/vice"
+      VICEDIR="$PREFIX/lib/vice"
       AM_CONDITIONAL(ALTERNATE_DOCDIR, false)
     else
       VICEDIR="$PREFIX/lib/vice"
