diff -Nur orig/configure.ac mod/configure.ac
--- orig/configure.ac	2019-12-23 22:29:30.000000000 +0100
+++ mod/configure.ac	2020-01-24 16:14:23.656011397 +0100
@@ -14,8 +14,8 @@
 
 dnl Set VICE's compiler flags (can be overridden by the user)
 VICE_CPPFLAGS=""
-VICE_CFLAGS="-g -O3"
-VICE_CXXFLAGS="-g -O3"
+VICE_CFLAGS="-g -O2"
+VICE_CXXFLAGS="-g -O2"
 VICE_LDFLAGS=""
 
 
@@ -114,12 +114,12 @@
 
 dnl Set default CFLAGS
 dnl if test x"$CFLAGS" = "x"; then
-dnl   CFLAGS="-g -O3"
+dnl   CFLAGS="-g -O2"
 dnl fi
 
 dnl Set default CXXFLAGS
 dnl if test x"$CXXFLAGS" = "x"; then
-dnl  CXXFLAGS="-g -O3"
+dnl  CXXFLAGS="-g -O2"
 dnl fi
 
 dnl Try global compiler warnings
@@ -955,7 +955,7 @@
     if test x"$GCC" = "xyes" ; then
       warnings_cxx="-Wall -Winline"
       warnings_c="-Wstrict-prototypes $warnings_cxx"
-      optimization_c="-O3 -mmacosx-version-min=10.9 $march"
+      optimization_c="-O2 -mmacosx-version-min=10.9 $march"
       optimization_cxx="-fno-exceptions $optimization_c"
       if test x"$ac_cv_prog_cc_g" = "xyes" ; then
         VICE_CFLAGS="$VICE_CFLAGS -g $optimization_c $warnings_c"
