Description: don't use lib64 directories
 .
Author: Laszlo Boszormenyi (GCS) <gcs@debian.org>
Last-Update: 2015-11-01

---

--- vice-2.4.dfsg.orig/configure.proto
+++ vice-2.4.dfsg/configure.proto
@@ -4027,7 +4027,7 @@ case "$host_os" in
     ;;
   linux*)
     if test x"$host_cpu" = "xx86_64"; then
-      VICEDIR="$PREFIX/lib64/vice"
+      VICEDIR="$PREFIX/lib/vice"
       AM_CONDITIONAL(ALTERNATE_DOCDIR, false)
     else
       VICEDIR="$PREFIX/lib/vice"
