diff -Naur orig/src/validate.c patched/src/validate.c
--- orig/src/validate.c	1970-01-01 01:00:00.000000000 +0100
+++ patched/src/validate.c	2020-06-20 01:46:49.966465455 +0200
@@ -377,7 +377,7 @@
 
 /* This should be the same list as in xdg-specs/menu/menu-spec.xml */
 static const char *show_in_registered[] = {
-    "GNOME", "GNOME-Classic", "GNOME-Flashback", "KDE", "LXDE", "LXQt", "MATE", "Razor", "ROX", "TDE", "Unity", "XFCE", "EDE", "Cinnamon", "Pantheon", "Budgie", "Enlightenment", "Deepin", "Old"
+    "GNOME", "GNOME-Classic", "GNOME-Flashback", "KDE", "LXDE", "LXQt", "MATE", "Razor", "ROX", "TDE", "Unity", "XFCE", "EDE", "Cinnamon", "Pantheon", "Budgie", "Enlightenment", "Deepin", "Lumina", "Old"
 };
 
 static struct {
