diff -Naur orig/src/validate.c patched/src/validate.c
--- orig/src/validate.c	2019-07-24 18:22:15.000000000 +0200
+++ patched/src/validate.c	2019-07-31 00:57:36.565285101 +0200
@@ -368,7 +368,7 @@
 
 /* This should be the same list as in xdg-specs/menu/menu-spec.xml */
 static const char *show_in_registered[] = {
-    "GNOME", "KDE", "LXDE", "LXQt", "MATE", "Razor", "ROX", "TDE", "Unity", "XFCE", "EDE", "Cinnamon", "Pantheon", "Budgie", "Enlightenment", "Deepin", "Old"
+    "GNOME", "KDE", "LXDE", "LXQt", "MATE", "Razor", "ROX", "TDE", "Unity", "XFCE", "EDE", "Cinnamon", "Pantheon", "Budgie", "Enlightenment", "Deepin", "Lumina", "Old"
 };
 
 static struct {
