diff -Naur a/src/validate.c b/src/validate.c
--- a/src/validate.c	2022-06-26 00:45:55.729152134 +0200
+++ b/src/validate.c	2022-06-26 00:47:44.493526785 +0200
@@ -534,6 +534,7 @@
   { "KDE",                    FALSE, FALSE, FALSE, { NULL }, { "Qt", NULL } },
   { "GNOME",                  FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
   { "XFCE",                   FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "LXQt",                   FALSE, FALSE, FALSE, { NULL }, { "Qt", NULL } },
   { "GTK",                    FALSE, FALSE, FALSE, { NULL }, { NULL } },
   { "Qt",                     FALSE, FALSE, FALSE, { NULL }, { NULL } },
   { "Motif",                  FALSE, FALSE, FALSE, { NULL }, { NULL } },
