diff -Naur orig/src/validate.c patched/src/validate.c
--- orig/src/validate.c	2019-07-31 00:57:36.565285101 +0200
+++ patched/src/validate.c	2019-07-31 00:59:50.403240653 +0200
@@ -525,6 +525,19 @@
   { "KDE",                    FALSE, FALSE, FALSE, { NULL }, { "Qt", NULL } },
   { "GNOME",                  FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
   { "XFCE",                   FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "LXDE",                   FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "LXQt",                   FALSE, FALSE, FALSE, { NULL }, { "Qt", NULL } },
+  { "MATE",                   FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "Razor",                  FALSE, FALSE, FALSE, { NULL }, { NULL } },
+  { "ROX",                    FALSE, FALSE, FALSE, { NULL }, { NULL } },
+  { "TDE",                    FALSE, FALSE, FALSE, { NULL }, { NULL } },
+  { "Unity",                  FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "EDE",                    FALSE, FALSE, FALSE, { NULL }, { NULL } },
+  { "Cinnamon",               FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "Pantheon",               FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "Lumina",                 FALSE, FALSE, FALSE, { NULL }, { "Qt", NULL } },
+  { "Budgie",                 FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "Enlightenment",          FALSE, FALSE, FALSE, { NULL }, { NULL } },  
   { "GTK",                    FALSE, FALSE, FALSE, { NULL }, { NULL } },
   { "Qt",                     FALSE, FALSE, FALSE, { NULL }, { NULL } },
   { "Motif",                  FALSE, FALSE, FALSE, { NULL }, { NULL } },
