diff -Naur orig/src/validate.c patched/src/validate.c
--- orig/src/validate.c	2016-12-15 10:55:19.670305925 +0100
+++ patched/src/validate.c	2016-12-15 10:57:57.799758781 +0100
@@ -368,7 +368,7 @@

 /* This should be the same list as in xdg-specs/menu/menu-spec.xml */
 static const char *show_in_registered[] = {
-    "GNOME", "KDE", "LXDE", "LXQt", "MATE", "Razor", "ROX", "TDE", "Unity", "XFCE", "EDE", "Cinnamon", "Pantheon", "Old"
+    "GNOME", "KDE", "LXDE", "LXQt", "MATE", "Razor", "ROX", "TDE", "Unity", "XFCE", "EDE", "Cinnamon", "Pantheon", "Lumina", "Budgie", "Enlightenment", "Old"
 };

 static struct {
