diff -Naur orig/src/validate.c patched/src/validate.c
--- orig/src/validate.c	2016-06-18 19:00:22.000000000 +0200
+++ patched/src/validate.c	2017-01-08 15:50:06.182641890 +0100
@@ -524,6 +524,19 @@
   { "KDE",                    FALSE, FALSE, FALSE, { NULL }, { "Qt", NULL } },
   { "GNOME",                  FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
   { "XFCE",                   FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "LXDE",                   FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "LXQt",                   FALSE, FALSE, FALSE, { NULL }, { "Qt", NULL } },
+  { "MATE",                   FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "Razor",                  FALSE, FALSE, FALSE, { NULL }, { NULL } },
+  { "ROX",                    FALSE, FALSE, FALSE, { NULL }, { NULL } },
+  { "TDE",                    FALSE, FALSE, FALSE, { NULL }, { NULL } },
+  { "Unity",                  FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "EDE",                    FALSE, FALSE, FALSE, { NULL }, { NULL } },
+  { "Cinnamon",               FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "Pantheon",               FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "Lumina",                 FALSE, FALSE, FALSE, { NULL }, { "Qt", NULL } },
+  { "Budgie",                 FALSE, FALSE, FALSE, { NULL }, { "GTK", NULL } },
+  { "Enlightenment",          FALSE, FALSE, FALSE, { NULL }, { NULL } },  
   { "GTK",                    FALSE, FALSE, FALSE, { NULL }, { NULL } },
   { "Qt",                     FALSE, FALSE, FALSE, { NULL }, { NULL } },
   { "Motif",                  FALSE, FALSE, FALSE, { NULL }, { NULL } },
