diff -Naur orig/src/validate.c patched/src/validate.c
--- orig/src/validate.c	2016-06-18 19:00:22.000000000 +0200
+++ patched/src/validate.c	2019-06-16 14:49:30.084738377 +0200
@@ -892,11 +892,11 @@
     /* force the type, since the key might be present multiple times... */
     kf->type = INVALID_TYPE;
 
-    print_fatal (kf, "value \"%s\" for key \"%s\" in group \"%s\" "
+    print_warning (kf, "value \"%s\" for key \"%s\" in group \"%s\" "
                      "is not a registered type value (\"Application\", "
                      "\"Link\" and \"Directory\")\n",
                      value, locale_key, kf->current_group);
-    return FALSE;
+    return TRUE;
   }
 
   if (registered_types[i].kde_reserved && kf->kde_reserved_warnings)
