diff -Naurd a/depend.mk b/depend.mk
--- a/depend.mk	1970-01-01 01:00:00.000000000 +0100
+++ b/depend.mk	2014-01-24 00:00:55.000000000 +0100
@@ -0,0 +1,21 @@
+#
+# Dependencies.
+#
+
+src/library/getopt.d src/library/getopt.o: src/library/getopt.c src/include/config.h src/include/library/gettext.h 
+src/library/gettext.d src/library/gettext.o: src/library/gettext.c src/include/config.h src/include/library/gettext.h 
+src/pv/signal.d src/pv/signal.o: src/pv/signal.c src/include/pv-internal.h src/include/config.h src/include/library/gettext.h src/include/pv.h 
+src/pv/watchpid.d src/pv/watchpid.o: src/pv/watchpid.c src/include/pv-internal.h src/include/config.h src/include/library/gettext.h src/include/pv.h 
+src/pv/cursor.d src/pv/cursor.o: src/pv/cursor.c src/include/pv-internal.h src/include/config.h src/include/library/gettext.h src/include/pv.h 
+src/pv/file.d src/pv/file.o: src/pv/file.c src/include/pv-internal.h src/include/config.h src/include/library/gettext.h src/include/pv.h 
+src/pv/display.d src/pv/display.o: src/pv/display.c src/include/pv-internal.h src/include/config.h src/include/library/gettext.h src/include/pv.h 
+src/pv/loop.d src/pv/loop.o: src/pv/loop.c src/include/pv-internal.h src/include/config.h src/include/library/gettext.h src/include/pv.h 
+src/pv/number.d src/pv/number.o: src/pv/number.c src/include/config.h src/include/library/gettext.h src/include/pv.h 
+src/pv/transfer.d src/pv/transfer.o: src/pv/transfer.c src/include/pv-internal.h src/include/config.h src/include/library/gettext.h src/include/pv.h 
+src/pv/state.d src/pv/state.o: src/pv/state.c src/include/pv-internal.h src/include/config.h src/include/library/gettext.h src/include/pv.h 
+src/main/version.d src/main/version.o: src/main/version.c src/include/config.h src/include/library/gettext.h 
+src/main/debug.d src/main/debug.o: src/main/debug.c src/include/config.h src/include/library/gettext.h src/include/pv.h 
+src/main/main.d src/main/main.o: src/main/main.c src/include/config.h src/include/library/gettext.h src/include/options.h src/include/pv.h 
+src/main/options.d src/main/options.o: src/main/options.c src/include/config.h src/include/library/gettext.h src/include/options.h src/include/library/getopt.h src/include/pv.h 
+src/main/remote.d src/main/remote.o: src/main/remote.c src/include/config.h src/include/library/gettext.h src/include/options.h src/include/pv.h 
+src/main/help.d src/main/help.o: src/main/help.c src/include/config.h src/include/library/gettext.h 
diff -Naurd a/filelist.mk b/filelist.mk
--- a/filelist.mk	1970-01-01 01:00:00.000000000 +0100
+++ b/filelist.mk	2014-01-24 00:00:54.000000000 +0100
@@ -0,0 +1,62 @@
+# Automatically generated file listings
+#
+# Creation time: Thu Jan 23 23:00:54 GMT 2014
+
+allsrc = src/library/getopt.c \
+src/library/gettext.c \
+src/pv/signal.c \
+src/pv/watchpid.c \
+src/pv/cursor.c \
+src/pv/file.c \
+src/pv/display.c \
+src/pv/loop.c \
+src/pv/number.c \
+src/pv/transfer.c \
+src/pv/state.c \
+src/main/version.c \
+src/main/debug.c \
+src/main/main.c \
+src/main/options.c \
+src/main/remote.c \
+src/main/help.c
+
+allobj = src/library/getopt.o \
+src/library/gettext.o \
+src/pv/signal.o \
+src/pv/watchpid.o \
+src/pv/cursor.o \
+src/pv/file.o \
+src/pv/display.o \
+src/pv/loop.o \
+src/pv/number.o \
+src/pv/transfer.o \
+src/pv/state.o \
+src/main/version.o \
+src/main/debug.o \
+src/main/main.o \
+src/main/options.o \
+src/main/remote.o \
+src/main/help.o \
+src/library.o \
+src/pv.o \
+src/nls.o \
+src/main.o
+
+alldep = src/library/getopt.d \
+src/library/gettext.d \
+src/pv/signal.d \
+src/pv/watchpid.d \
+src/pv/cursor.d \
+src/pv/file.d \
+src/pv/display.d \
+src/pv/loop.d \
+src/pv/number.d \
+src/pv/transfer.d \
+src/pv/state.d \
+src/main/version.d \
+src/main/debug.d \
+src/main/main.d \
+src/main/options.d \
+src/main/remote.d \
+src/main/help.d
+
diff -Naurd a/modules.mk b/modules.mk
--- a/modules.mk	1970-01-01 01:00:00.000000000 +0100
+++ b/modules.mk	2014-01-24 00:00:54.000000000 +0100
@@ -0,0 +1,14 @@
+# Automatically generated module linking rules
+#
+# Creation time: Thu Jan 23 23:00:54 GMT 2014
+
+src/library.o:  src/library/getopt.o src/library/gettext.o
+	$(LD) $(LDFLAGS) -o $@  src/library/getopt.o src/library/gettext.o
+
+src/pv.o:  src/pv/cursor.o src/pv/display.o src/pv/file.o src/pv/loop.o src/pv/number.o src/pv/signal.o src/pv/state.o src/pv/transfer.o src/pv/watchpid.o
+	$(LD) $(LDFLAGS) -o $@  src/pv/cursor.o src/pv/display.o src/pv/file.o src/pv/loop.o src/pv/number.o src/pv/signal.o src/pv/state.o src/pv/transfer.o src/pv/watchpid.o
+
+src/main.o:  src/main/debug.o src/main/help.o src/main/main.o src/main/options.o src/main/remote.o src/main/version.o
+	$(LD) $(LDFLAGS) -o $@  src/main/debug.o src/main/help.o src/main/main.o src/main/options.o src/main/remote.o src/main/version.o
+
+
