diff -Naur orig/uxterm.desktop patched/uxterm.desktop
--- orig/uxterm.desktop	2019-02-06 02:36:47.000000000 +0100
+++ patched/uxterm.desktop	2019-03-21 01:48:16.537564702 +0100
@@ -34,6 +34,7 @@
 Name=UXTerm
 #GenericName=Terminal
 Comment=xterm wrapper for Unicode environments
+Comment[hu]=Hagyományos grafikus terminál
 Exec=uxterm
 Terminal=false
 Type=Application
diff -Naur orig/xterm.desktop patched/xterm.desktop
--- orig/xterm.desktop	2016-03-08 02:38:50.000000000 +0100
+++ patched/xterm.desktop	2019-03-21 01:47:54.085267865 +0100
@@ -34,6 +34,7 @@
 Name=XTerm
 #GenericName=Terminal
 Comment=standard terminal emulator for the X window system
+Comment[hu]=Hagyományos grafikus terminál
 Exec=xterm
 Terminal=false
 Type=Application
