diff -Naur orig/uxterm.desktop mod/uxterm.desktop
--- orig/uxterm.desktop	2012-06-25 00:20:29.000000000 +0200
+++ mod/uxterm.desktop	2014-05-13 20:45:18.000000000 +0200
@@ -34,6 +34,7 @@
 Name=UXTerm
 #GenericName=Terminal
 Comment=standard terminal emulator for the X window system
+Comment[hu]=Hagyományos grafikus terminál
 Exec=uxterm
 Terminal=false
 Type=Application
diff -Naur orig/xterm.desktop mod/xterm.desktop
--- orig/xterm.desktop	2012-06-25 00:20:29.000000000 +0200
+++ mod/xterm.desktop	2014-05-13 20:45:50.000000000 +0200
@@ -34,6 +34,7 @@
 Name=XTerm
 #GenericName=Terminal
 Comment=standard terminal emulator for the X window system
+Comment[hu]=Hagyományos grafikus terminál
 Exec=xterm
 Terminal=false
 Type=Application
