diff -Naur orig/ptyx.h mod/ptyx.h
--- orig/ptyx.h	2014-05-03 14:58:37.000000000 +0200
+++ mod/ptyx.h	2014-05-13 20:37:08.000000000 +0200
@@ -2410,7 +2410,7 @@
 #define	INWINDOW	01	/* the mouse is in one of the windows */
 #define	FOCUS		02	/* one of the windows is the focus window */
 
-#define MULTICLICKTIME 250	/* milliseconds */
+#define MULTICLICKTIME 400	/* milliseconds */
 
 typedef struct {
     const char *name;
