diff -Naur orig/Makefile.in patched/Makefile.in
--- orig/Makefile.in	2019-01-14 02:40:56.000000000 +0100
+++ patched/Makefile.in	2019-03-21 01:50:55.847670908 +0100
@@ -233,12 +233,10 @@
 ################################################################################
 actual_xterm  = `echo xterm|        sed '$(transform)'`
 actual_resize = `echo resize|       sed '$(transform)'`
-actual_uxterm = `echo uxterm|       sed '$(transform)'`
 actual_k8term = `echo koi8rxterm|   sed '$(transform)'`
 
 binary_xterm  = $(actual_xterm)$x
 binary_resize = $(actual_resize)$x
-binary_uxterm = $(actual_uxterm)
 binary_k8term = $(actual_k8term)
 
 install \
@@ -281,10 +279,6 @@
 	@$(SHELL) -c "source=\"resize\"; \
 		target=\"$(binary_resize)\"; \
 		$(InstallBinLink)"
-	@$(SHELL) -c "source=\"uxterm\"; \
-		target=\"$(binary_uxterm)\"; \
-		$(InstallScript); \
-		$(InstallBinLink)"
 	@$(SHELL) -c "source=\"koi8rxterm\"; \
 		target=\"$(binary_k8term)\"; \
 		$(InstallScript); \
@@ -293,7 +287,7 @@
 install \
 install-man \
 install-full :: $(MANDIR)
-	@-$(SHELL) -c "for source in xterm resize uxterm koi8rxterm ; \
+	@-$(SHELL) -c "for source in xterm resize koi8rxterm ; \
 		do \
 			target=\`echo \"\$$source\" | sed '@program_transform_name@'\`; \
 			$(SHELL) ./minstall \"$(INSTALL_DATA)\" \
@@ -316,7 +310,7 @@
 			) \
 		fi"
 
-APP_NAMES = XTerm UXTerm KOI8RXTerm
+APP_NAMES = XTerm KOI8RXTerm
 
 @no_appsdir@install \
 @no_appsdir@install-app \
@@ -511,7 +505,7 @@
 @no_pixmapdir@	@echo "... removed icons"
 ################################################################################
 # Desktop-utils does not provide an uninstall, and is not uniformly available.
-@desktop_utils@DESKTOP_FILES = $(srcdir)/xterm.desktop $(srcdir)/uxterm.desktop
+@desktop_utils@DESKTOP_FILES = $(srcdir)/xterm.desktop
 @desktop_utils@DESKTOP_FLAGS = @DESKTOP_FLAGS@
 @desktop_utils@install-desktop \
 @desktop_utils@install-full ::
@@ -578,12 +572,6 @@
 xterm.ps : xterm.$(manext)
 xterm.txt : xterm.$(manext)
 ################################################################################
-docs-uxterm \
-docs ::  uxterm.txt @MAN2HTML_NOTE@ uxterm.html @GROFF_NOTE@ uxterm.pdf uxterm.ps
-uxterm.html : uxterm.$(manext)
-uxterm.pdf : uxterm.ps
-uxterm.ps : uxterm.$(manext)
-uxterm.txt : uxterm.$(manext)
 ################################################################################
 docs-koi8rxterm \
 docs ::  koi8rxterm.txt @MAN2HTML_NOTE@ koi8rxterm.html @GROFF_NOTE@ koi8rxterm.pdf koi8rxterm.ps
