diff -Naur orig/plugins/eds/gtd-eds-autoptr.h patched/plugins/eds/gtd-eds-autoptr.h
--- orig/plugins/eds/gtd-eds-autoptr.h	2018-04-10 16:08:54.000000000 +0200
+++ patched/plugins/eds/gtd-eds-autoptr.h	2019-03-17 00:00:49.578389041 +0100
@@ -25,4 +25,4 @@
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (ECalComponent, g_object_unref);
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (ECalComponentId, e_cal_component_free_id);
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (ECalClient, g_object_unref);
-G_DEFINE_AUTOPTR_CLEANUP_FUNC (ESource, g_object_unref);
+// G_DEFINE_AUTOPTR_CLEANUP_FUNC (ESource, g_object_unref);
