diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2019-08-18 14:03:41.000000000 +0200
+++ patched/configure.ac	2019-10-24 08:08:14.238203778 +0200
@@ -91,6 +91,7 @@
 XINE_LIB_SHIMS
 XINE_LIB_OPEN_CLOEXEC
 XINE_LIB_SOCKET_CLOEXEC
+XINE_LIB_LIST_NEXT_VALUE
 
 CFLAGS="${CFLAGS} ${XINE_CFLAGS}"
 
diff -Naur orig/m4/_xine.m4 patched/m4/_xine.m4
--- orig/m4/_xine.m4	2019-08-18 14:03:41.000000000 +0200
+++ patched/m4/_xine.m4	2019-10-24 08:09:33.694895028 +0200
@@ -241,3 +241,22 @@
     LIBS="$tmp_LIBS"
     ])
 
+dnl Test for xine_list_next_value function
+AC_DEFUN([XINE_LIB_LIST_NEXT_VALUE],
+    [AC_MSG_CHECKING([for xine_list_next_value within xine-lib])
+    tmp_CFLAGS="$CFLAGS"
+    tmp_LIBS="$LIBS"
+     CFLAGS="$CFLAGS $XINE_CFLAGS"
+     LIBS="$LIBS $XINE_LIBS"
+     AC_LINK_IFELSE(
+         [AC_LANG_PROGRAM([
+         ],[
+xine_list_next_value ();
+         ])],
+         [AC_DEFINE([HAVE_XINE_LIST_NEXT_VALUE], [1], [Define if xine-lib supports xine_list_next_value])
+          AC_MSG_RESULT([yes])],
+         [AC_MSG_RESULT([no])])
+    CFLAGS="$tmp_CFLAGS"
+    LIBS="$tmp_LIBS"
+   ])
+
diff -Naur orig/src/xitk/setup.c patched/src/xitk/setup.c
--- orig/src/xitk/setup.c	2019-08-18 14:03:41.000000000 +0200
+++ patched/src/xitk/setup.c	2019-10-24 08:11:23.751859619 +0200
@@ -31,7 +31,17 @@
 #include "common.h"
 
 #include <xine/list.h>
-
+#ifdef HAVE_XINE_LIST_NEXT_VALUE
+#  define _xine_list_next_value(_xlnv_list,_xlnv_ite) xine_list_next_value (_xlnv_list, _xlnv_ite)
+#else
+static inline void *_xine_list_next_value (xine_list_t *list, xine_list_iterator_t *ite) {
+  if (*ite)
+    *ite = xine_list_next (list, *ite);
+  else
+    *ite = xine_list_front (list);
+  return *ite ? xine_list_get_value (list, *ite) : NULL;
+}
+#endif
 
 #define WINDOW_WIDTH             630
 #define WINDOW_HEIGHT            530
@@ -809,7 +819,7 @@
     xitk_widget_t *sw;
     xine_list_iterator_t ite = NULL;
     while (1) {
-      sw = xine_list_next_value (setup->widgets, &ite);
+      sw = _xine_list_next_value (setup->widgets, &ite);
       if (!ite)
         break;
       xitk_destroy_widget (sw);
