diff -Nur orig/contrib/meson/update-linguas.sh mod/contrib/meson/update-linguas.sh
--- orig/contrib/meson/update-linguas.sh	1970-01-01 01:00:00.000000000 +0100
+++ mod/contrib/meson/update-linguas.sh	2018-01-08 12:09:25.000000000 +0100
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+find ${MESON_SOURCE_ROOT}/po \
+	-type f \
+	-iname "*.po" \
+        -printf '%f\n' \
+	| grep -oP '.*(?=[.])' | sort \
+	> ${MESON_SOURCE_ROOT}/po/LINGUAS
diff -Nur orig/po/meson.build mod/po/meson.build
--- orig/po/meson.build	2017-10-22 19:36:07.000000000 +0200
+++ mod/po/meson.build	2018-01-08 12:09:53.000000000 +0100
@@ -13,24 +13,15 @@
     ]
 )
 
-make_linguas = custom_target ('make-linguas',
-    build_always: true,
-    output: 'LINGUAS_dummy', # fake output
-    command: [
-        'sh', '-c',
-        'find ' + meson.current_source_dir() +
-        ' -type f' +
-        ' -iname "*.po"' +
-        ' -printf \'%f\\n\'' +
-        ' | grep -oP \'.*(?=[.])\' | sort' +
-        ' > ' + join_paths(meson.current_source_dir(), 'LINGUAS')
-    ]
+run_target ('make-linguas',
+    command: ['sh',
+              join_paths(meson.source_root(), 'contrib/meson/update-linguas.sh')]
 )
 
 # maintainer shortcut for updating l10n data
 run_target ('l10n-update',
-    depends: make_linguas,
     command: ['ninja', '-C', meson.build_root(),
+              'make-linguas',
               as_gettext_domain + '-pot',
               as_gettext_domain + '-update-po']
 )
