diff -Naur gnokii-0.6.9.orig/utils/sendsms gnokii-0.6.9/utils/sendsms
--- gnokii-0.6.9.orig/utils/sendsms	2005-10-20 22:18:06.000000000 +0200
+++ gnokii-0.6.9/utils/sendsms	2006-02-28 18:53:19.000000000 +0100
@@ -243,7 +243,7 @@
 						1) sort -n $SMSDIR/.SMlist.chklst > $TMP/.SMlist.tmp
 							 mv -f $TMP/.SMlist.tmp $SMSDIR/.SMlist.chklst
 							 ;;
-						2) sort +1 $SMSDIR/.SMlist.chklst > $TMP/.SMlist.tmp
+						2) sort -k 2 $SMSDIR/.SMlist.chklst > $TMP/.SMlist.tmp
 							 mv -f $TMP/.SMlist.tmp $SMSDIR/.SMlist.chklst
 							 ;;
 						9) sure_to_quit
