diff -Nur orig/bin/gufw-pkexec mod/bin/gufw-pkexec
--- orig/bin/gufw-pkexec	2017-12-23 15:35:55.000000000 +0100
+++ mod/bin/gufw-pkexec	2018-03-06 20:31:32.903346710 +0100
@@ -1,7 +1,7 @@
 #!/bin/bash
 LOCATIONS=(
-"/usr/lib/python3.5/site-packages/gufw/gufw.py"
-"/usr/lib/python3.5/dist-packages/gufw/gufw.py"
+"/usr/lib/python3.8/site-packages/gufw/gufw.py"
+"/usr/lib/python3.8/dist-packages/gufw/gufw.py"
 "/usr/share/gufw/gufw/gufw.py"
 )
 
