diff -Naur a/gufw/gufw/view/gufw.py b/gufw/gufw/view/gufw.py
--- a/gufw/gufw/view/gufw.py	2022-01-25 19:31:39.000000000 +0100
+++ b/gufw/gufw/view/gufw.py	2023-02-08 23:40:55.158547522 +0100
@@ -18,7 +18,7 @@
 import gi
 gi.require_version('Gtk', '3.0')
 gi.require_version('Gdk', '3.0')
-gi.require_version('WebKit2', '4.0')
+gi.require_version('WebKit2', '4.1')
 from gi.repository import Gtk, Gdk, WebKit2
 from string import Template
 
