--- XBlast-2.6-sound.orig/Legacy/tkXBlast/tkXBlast.tcl	Sun Sep 26 10:56:48 1999
+++ XBlast-2.6-sound/Legacy/tkXBlast/tkXBlast.tcl	Tue Jan  8 17:22:35 2002
@@ -1,4 +1,4 @@
-#!/usr/bin/X11/wish -f
+#!/usr/bin/wish -f
 
 #
 # Search for *CONFIG* and edit variables to your needs
@@ -26,22 +26,22 @@
 # *CONFIG*: filename of the xblast binary
 #
 #XBLASTCOMMAND
-set xblastCommand "../xblast"
+set xblastCommand "/usr/bin/xblast.bin"
 #
 # *CONFIG*: path for xrdb command
 #
 #XRDBCOMMAND
-set xrdbCommand "/usr/X11R6/bin/xrdb"    
+set xrdbCommand "/usr/bin/xrdb"
 #
 # *CONFIG* path for X11 rgb.txt and for personal rgb file
 #
 #RGBFILE
-set rgbFile "/usr/lib/X11/rgb.txt"
+set rgbFile "/usr/share/X11/rgb.txt"
 #
 # path for image files
 #
 #BITMAPPATH
-set bitmapPath "./"
+set bitmapPath "/usr/share/xblast/tkXBlast"
 
 # 
 # User configurable variables
