diff -urN XBlast-2.6-sound.orig/Legacy/XBlast.ad XBlast-2.6-sound/Legacy/XBlast.ad
--- XBlast-2.6-sound.orig/Legacy/XBlast.ad	Sun Sep 19 16:24:58 1999
+++ XBlast-2.6-sound/Legacy/XBlast.ad	Mon Dec 10 19:17:19 2001
@@ -73,7 +73,7 @@
 XBlast.singlePlayer.bombKey:        KP_0 KP_Insert Insert space
 XBlast.singlePlayer.specialKey:     Return KP_Add Tab
 XBlast.singlePlayer.abortKey:       KP_Multiply A
-XBlast.singlePlayer.abortcancelKey: KP_Divide Z
+XBlast.singlePlayer.abortCancelKey: KP_Divide Z
 # keybindings for right player
 XBlast.rightPlayer.pauseKey:        KP_Subtract
 XBlast.rightPlayer.upKey:           KP_8 KP_Up Up
@@ -87,7 +87,7 @@
 XBlast.rightPlayer.bombKey:         KP_0 KP_Insert Insert
 XBlast.rightPlayer.specialKey:      Return KP_Add
 XBlast.rightPlayer.abortKey:        KP_Multiply
-XBlast.rightPlayer.abortcancelKey:  KP_Divide
+XBlast.rightPlayer.abortCancelKey:  KP_Divide
 # keybindings for left player
 XBlast.leftPlayer.pauseKey:         P
 XBlast.leftPlayer.upKey:            T
@@ -101,7 +101,7 @@
 XBlast.leftPlayer.bombKey:          space
 XBlast.leftPlayer.specialKey:       Tab
 XBlast.leftPlayer.abortKey:         A
-XBlast.leftPlayer.abortcancelKey:   Z
+XBlast.leftPlayer.abortCancelKey:   Z
 
 # default player names
 XBlast.player1:	Olli
