diff -Naur xblast-2.10.2.orig/TNT/x11_sound.c xblast-2.10.2/TNT/x11_sound.c
--- xblast-2.10.2.orig/TNT/x11_sound.c	2005-03-08 10:54:57.000000000 +0100
+++ xblast-2.10.2/TNT/x11_sound.c	2006-04-10 17:14:04.000000000 +0200
@@ -29,7 +29,7 @@
  * local constants
  */
 #ifndef XBLAST_SOUND_SERVER
-#define XBLAST_SOUND_SERVER "xbsndsrv"
+#define XBLAST_SOUND_SERVER "xbsndsrv-tnt"
 #endif
 
 /* client/server commands */
