diff -Nur orig/help/LINGUAS mod/help/LINGUAS
--- orig/help/LINGUAS	2024-02-23 16:12:37.000000000 +0100
+++ mod/help/LINGUAS	2024-03-02 14:16:12.744004039 +0100
@@ -1,6 +1,5 @@
 af
 am
-ar
 as
 ast
 az
@@ -36,27 +35,21 @@
 fa
 fi
 fr
-fur
 fy
 ga
 gl
 gu
 ha
 he
-hi
 hr
 hu
-hy
-id
 ie
 ig
 is
 it
-ja
 ka
 kk
 kn
-ko
 ku
 ku_IQ
 ky
@@ -77,14 +70,11 @@
 nn
 oc
 or
-pa
 pl
 ps
 pt
-pt_BR
 ro
 ru
-si
 sk
 sl
 sq
@@ -100,9 +90,7 @@
 ur
 uz
 vi
-wa
 xh
-yo
 zh_CN
 zh_HK
 zh_TW
