diff -Nur orig/audisp/plugins/remote/au-remote.conf mod/audisp/plugins/remote/au-remote.conf
--- orig/audisp/plugins/remote/au-remote.conf	2019-02-04 15:26:52.000000000 +0100
+++ mod/audisp/plugins/remote/au-remote.conf	2019-10-08 18:13:08.489556757 +0200
@@ -5,7 +5,7 @@
 
 active = no
 direction = out
-path = /sbin/audisp-remote
+path = /usr/sbin/audisp-remote
 type = always
 #args =
 format = string
