diff -Nur orig/audisp/plugins/zos-remote/audispd-zos-remote.conf mod/audisp/plugins/zos-remote/audispd-zos-remote.conf
--- orig/audisp/plugins/zos-remote/audispd-zos-remote.conf	2019-02-04 15:26:52.000000000 +0100
+++ mod/audisp/plugins/zos-remote/audispd-zos-remote.conf	2019-10-08 18:11:21.421678002 +0200
@@ -8,7 +8,7 @@
 
 active = no
 direction = out
-path = /sbin/audispd-zos-remote
+path = /usr/sbin/audispd-zos-remote
 type = always 
 args = /etc/audisp/zos-remote.conf
 format = string
