diff -Naur orig/init.d/auditd.service patched/init.d/auditd.service
--- orig/init.d/auditd.service	2020-12-16 21:44:34.000000000 +0100
+++ patched/init.d/auditd.service	2020-12-28 17:27:59.325707341 +0100
@@ -18,11 +18,11 @@
 [Service]
 Type=forking
 PIDFile=/run/auditd.pid
-ExecStart=/sbin/auditd
+ExecStart=/usr/sbin/auditd
 ## To not use augenrules, copy this file to /etc/systemd/system/auditd.service
 ## and comment/delete the next line and uncomment the auditctl line.
 ## NOTE: augenrules expect any rules to be added to /etc/audit/rules.d/
-ExecStartPost=-/sbin/augenrules --load
+ExecStartPost=-/usr/sbin/augenrules --load
 #ExecStartPost=-/sbin/auditctl -R /etc/audit/audit.rules
 # By default we don't clear the rules on exit. To enable this, uncomment
 # the next line after copying the file to /etc/systemd/system/auditd.service
