diff -Naur orig/share/linux/org.keepassxc.KeePassXC.desktop patched/share/linux/org.keepassxc.KeePassXC.desktop
--- orig/share/linux/org.keepassxc.KeePassXC.desktop	2020-01-19 21:43:19.000000000 +0100
+++ patched/share/linux/org.keepassxc.KeePassXC.desktop	2020-03-05 00:41:34.610277850 +0100
@@ -5,6 +5,7 @@
 GenericName[de]=Passwortverwaltung
 GenericName[es]=Gestor de contraseñas
 GenericName[fr]=Gestionnaire de mot de passe
+GenericName[hu]=Jelszókezelő
 GenericName[ru]=менеджер паролей
 Comment=Community-driven port of the Windows application “KeePass Password Safe”
 Comment[da]=Fællesskabsdrevet port af Windows-programmet “KeePass Password Safe”
