diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2021-09-28 18:05:13.000000000 +0200
+++ patched/CMakeLists.txt	2021-10-10 22:53:16.329211902 +0200
@@ -83,7 +83,6 @@
 set(AKONADIMIME_LIB_VERSION "5.18.2")
 set(AKONADICONTACT_LIB_VERSION "5.18.2")
 set(AKONADINOTE_LIB_VERSION "5.18.2")
-set(PIMCOMMON_LIB_VERSION "5.18.2")
 set(KGAPI_LIB_VERSION "5.18.2")
 set(LIBKDEPIM_LIB_VERSION "5.18.2")
 set(KLDAP_LIB_VERSION "5.18.2")
@@ -140,7 +139,6 @@
 find_package(KF5IMAP ${KIMAP_LIB_VERSION} CONFIG REQUIRED)
 find_package(KF5AkonadiNotes ${AKONADINOTE_LIB_VERSION} CONFIG REQUIRED)
 find_package(KF5AkonadiCalendar ${AKONADICALENDAR_LIB_VERSION} CONFIG REQUIRED)
-find_package(KF5PimCommon ${PIMCOMMON_LIB_VERSION} CONFIG REQUIRED)
 find_package(KPimGAPI ${KGAPI_LIB_VERSION} CONFIG REQUIRED)
 find_package(KF5Libkdepim ${LIBKDEPIM_LIB_VERSION} CONFIG REQUIRED)
 find_package(KF5Ldap ${KLDAP_LIB_VERSION} CONFIG REQUIRED)
diff -Naur orig/resources/imap/CMakeLists.txt patched/resources/imap/CMakeLists.txt
--- orig/resources/imap/CMakeLists.txt	2021-09-28 18:05:13.000000000 +0200
+++ patched/resources/imap/CMakeLists.txt	2021-10-10 22:52:31.901069800 +0200
@@ -86,7 +86,6 @@
     KF5::I18n
     KF5::WindowSystem
     akonadi-singlefileresource
-    KF5::PimCommon
     KF5::Libkdepim
     qt5keychain
 )
diff -Naur orig/resources/kolab/CMakeLists.txt patched/resources/kolab/CMakeLists.txt
--- orig/resources/kolab/CMakeLists.txt	2021-09-28 18:05:13.000000000 +0200
+++ patched/resources/kolab/CMakeLists.txt	2021-10-10 22:52:31.901069800 +0200
@@ -69,7 +69,6 @@
     KF5::Contacts
     KF5::CalendarCore
     akonadi-singlefileresource
-    KF5::PimCommon
     KF5::TextWidgets
     KF5::I18n
     )
diff -Naur orig/resources/pop3/CMakeLists.txt patched/resources/pop3/CMakeLists.txt
--- orig/resources/pop3/CMakeLists.txt	2021-09-28 18:05:13.000000000 +0200
+++ patched/resources/pop3/CMakeLists.txt	2021-10-10 22:52:31.901069800 +0200
@@ -48,8 +48,8 @@
     KF5::I18n
     KF5::Notifications
     Qt::DBus
-    KF5::PimCommon
     qt5keychain
+    KF5::WidgetsAddons
     )
 if (COMPILE_WITH_UNITY_CMAKE_SUPPORT)
     set_target_properties(akonadi_pop3_resource PROPERTIES UNITY_BUILD ON)
