diff -Nur orig/extensions/adblock/widgets.vala mod/extensions/adblock/widgets.vala
--- orig/extensions/adblock/widgets.vala	2015-08-30 13:56:26.000000000 +0200
+++ mod/extensions/adblock/widgets.vala	2017-10-13 21:00:48.097270320 +0200
@@ -227,7 +227,7 @@
                             liststore.get (iter, 0, out sub);
                             if (sub.mutable) {
                                 config.remove (sub);
-                                liststore.remove (iter);
+                                liststore.remove (ref iter);
                                 return true;
                             }
                         }
diff -Nur orig/extensions/apps.vala mod/extensions/apps.vala
--- orig/extensions/apps.vala	2015-08-30 13:56:26.000000000 +0200
+++ mod/extensions/apps.vala	2017-10-13 21:00:48.101270315 +0200
@@ -265,7 +265,7 @@
                             store.get (iter, 0, out launcher);
                             try {
                                 launcher.file.trash (null);
-                                store.remove (iter);
+                                store.remove (ref iter);
 
                                 string filename = Midori.Download.clean_filename (launcher.name);
 #if HAVE_WIN32
diff -Nur orig/extensions/history-list.vala mod/extensions/history-list.vala
--- orig/extensions/history-list.vala	2015-08-30 13:56:26.000000000 +0200
+++ mod/extensions/history-list.vala	2017-10-13 21:00:48.105270310 +0200
@@ -223,7 +223,7 @@
                     FixMe: the retrun value of `Gtk.ListStore.remove` should be checked
                     Note:  in some cases the return value of `Gtk.ListStore.remove` is wrong
                 */
-                model.remove (iter);
+                model.remove (ref iter);
                 this.browser.close_tab (view);
                 if (length > 2)
                     this.resize_treeview ();
diff -Nur orig/extensions/notes.vala mod/extensions/notes.vala
--- orig/extensions/notes.vala	2015-08-30 13:56:26.000000000 +0200
+++ mod/extensions/notes.vala	2017-10-13 21:00:48.109270304 +0200
@@ -116,7 +116,7 @@
                     if (current_note == note) {
                         current_note = null;
                     }
-                    notes_list_store.remove (iter);
+                    notes_list_store.remove (ref iter);
                     break;
                 }
             } while (notes_list_store.iter_next (ref iter));
diff -Nur orig/extensions/transfers.vala mod/extensions/transfers.vala
--- orig/extensions/transfers.vala	2015-08-30 13:56:26.000000000 +0200
+++ mod/extensions/transfers.vala	2017-10-13 21:00:48.093270326 +0200
@@ -275,7 +275,7 @@
                     Transfer found;
                     store.get (iter, 0, out found);
                     if (transfer == found) {
-                        store.remove (iter);
+                        store.remove (ref iter);
                         break;
                     }
                 } while (store.iter_next (ref iter));
@@ -455,7 +455,7 @@
                 if (notifications.length () == 1)
                     msg = _("The file '<b>%s</b>' has been downloaded.").printf (filename);
                 else
-                    msg = _("'<b>%s</b>' and %d other files have been downloaded.").printf (filename, notifications.length ());
+                    msg = _("'<b>%s</b>' and %u other files have been downloaded.").printf (filename, notifications.length ());
                 get_app ().send_notification (_("Transfer completed"), msg);
                 notifications = new GLib.List<string> ();
             }
diff -Nur orig/midori/midori-notebook.vala mod/midori/midori-notebook.vala
--- orig/midori/midori-notebook.vala	2015-08-30 13:56:26.000000000 +0200
+++ mod/midori/midori-notebook.vala	2017-10-13 21:00:48.085270337 +0200
@@ -22,7 +22,7 @@
         public bool close_button_left { get; set; default = false; }
         public bool close_button_visible { get; set; default = false; }
 
-        protected Tally (Midori.Tab tab) {
+        public Tally (Midori.Tab tab) {
             this.tab = tab;
             box = new Gtk.HBox (false, 1);
             add (box);
