diff -Naur hplip-3.13.11-orig/ui4/setupdialog.py hplip-3.13.11/ui4/setupdialog.py
--- hplip-3.13.11-orig/ui4/setupdialog.py	2013-10-31 12:43:26.000000000 +0100
+++ hplip-3.13.11/ui4/setupdialog.py	2013-11-16 18:18:13.033487033 +0100
@@ -189,7 +189,7 @@
             QTimer.singleShot(0, self.showRemovePage)
         else:
             if self.skip_discovery:
-                self.discovery_method = 0 # SLP
+                self.discovery_method = 1 # mDNS
                 QTimer.singleShot(0, self.showDevicesPage)
             else:
                 QTimer.singleShot(0, self.showDiscoveryPage)
