diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2024-06-15 14:13:57.000000000 +0200
+++ b/CMakeLists.txt	2024-07-04 11:34:53.855208441 +0200
@@ -95,7 +95,7 @@
 SET (NO_OFFENSIVE CACHE BOOL "Exclude the offensive option.")
 SET (USE_PCRE CACHE BOOL "Link to the PCRE2 regex library")
 SET (USE_WIN32_REGEX_LIBS TRUE CACHE BOOL "Link to the regex libraries (only affects MSWin builds)")
-SET (LOCALDIR "${CMAKE_INSTALL_PREFIX}/local/share/games/fortunes" CACHE STRING "LOCALDIR fortunes dir")
+SET (LOCALDIR "${CMAKE_INSTALL_PREFIX}/share/games/fortunes" CACHE STRING "LOCALDIR fortunes dir")
 SET (LOCALODIR "${LOCALDIR}/off" CACHE STRING "offensive fortunes localdir")
 SET (COOKIEDIR "${CMAKE_INSTALL_PREFIX}/share/games/fortunes" CACHE STRING  "cookie dir not under /usr/local")
 SET (OCOOKIEDIR "${COOKIEDIR}/off" CACHE STRING "offensive cookie dir not under /usr/local")
@@ -175,7 +175,7 @@
 my_exe(
     "fortune"
     "fortune/fortune.c"
-    "games"
+    "bin"
 )
 
 my_exe(
