diff -Naur orig/plugins/Eq/EqCurve.h patched/plugins/Eq/EqCurve.h
--- orig/plugins/Eq/EqCurve.h	2019-10-21 02:27:12.000000000 +0200
+++ patched/plugins/Eq/EqCurve.h	2020-06-04 00:02:05.454327451 +0200
@@ -27,6 +27,7 @@
 
 #include <QGraphicsItem>
 #include <QPainter>
+#include <QPainterPath>
 #include <QGraphicsSceneWheelEvent>
 #include "lmms_math.h"
 #include "AutomatableModelView.h"
diff -Naur orig/plugins/Eq/EqSpectrumView.h patched/plugins/Eq/EqSpectrumView.h
--- orig/plugins/Eq/EqSpectrumView.h	2019-10-21 02:27:12.000000000 +0200
+++ patched/plugins/Eq/EqSpectrumView.h	2020-06-04 00:02:44.434419575 +0200
@@ -24,6 +24,7 @@
 #define EQSPECTRUMVIEW_H
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QWidget>
 
 #include "fft_helpers.h"
diff -Naur orig/src/gui/AutomationPatternView.cpp patched/src/gui/AutomationPatternView.cpp
--- orig/src/gui/AutomationPatternView.cpp	2019-10-21 02:27:12.000000000 +0200
+++ patched/src/gui/AutomationPatternView.cpp	2020-06-03 23:59:42.221987576 +0200
@@ -25,6 +25,7 @@
 
 #include <QMouseEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QMenu>
 
 #include "AutomationEditor.h"
diff -Naur orig/src/gui/editors/AutomationEditor.cpp patched/src/gui/editors/AutomationEditor.cpp
--- orig/src/gui/editors/AutomationEditor.cpp	2019-10-21 02:27:12.000000000 +0200
+++ patched/src/gui/editors/AutomationEditor.cpp	2020-06-04 00:01:06.034186716 +0200
@@ -35,6 +35,7 @@
 #include <QLayout>
 #include <QMdiArea>
 #include <QPainter>
+#include <QPainterPath>
 #include <QScrollBar>
 #include <QStyleOption>
 #include <QToolTip>
diff -Naur orig/src/gui/LmmsStyle.cpp patched/src/gui/LmmsStyle.cpp
--- orig/src/gui/LmmsStyle.cpp	2019-10-21 02:27:12.000000000 +0200
+++ patched/src/gui/LmmsStyle.cpp	2020-06-04 00:00:34.982113021 +0200
@@ -28,6 +28,7 @@
 #include <QApplication>
 #include <QFrame>
 #include <QPainter>
+#include <QPainterPath>
 #include <QStyleFactory>
 #include <QStyleOption>
 
