diff -Naur a/meld/build_helpers.py b/meld/build_helpers.py
--- a/meld/build_helpers.py	2022-08-14 01:02:41.000000000 +0200
+++ b/meld/build_helpers.py	2022-08-14 21:10:01.664496415 +0200
@@ -64,7 +64,7 @@
     return "build_data" in self.distribution.cmdclass
 
 
-command_base.command.build.Build.sub_commands.extend([
+command_base.command.build.build.sub_commands.extend([
     ("build_i18n", has_i18n),
     ("build_icons", has_icons),
     ("build_help", has_help),
