diff -Naur orig/WindowMaker/menu mod/WindowMaker/menu
--- orig/WindowMaker/menu	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/menu	2016-04-21 07:12:43.297985465 +0200
@@ -208,7 +208,7 @@
 		"Background" END
 		"Save Theme" SHEXEC getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes/"%a(Theme name,Enter file name:)"
 		"Save IconSet" SHEXEC geticonset $HOME/GNUstep/Library/WindowMaker/IconSets/"%a(IconSet name,Enter file name:)"
-		"Preferences Utility" EXEC /usr/local/GNUstep/Applications/WPrefs.app/WPrefs
+		"Preferences Utility" EXEC /usr/bin//WPrefs
 	"Appearance" END
 
 	"Session" MENU
diff -Naur orig/WindowMaker/menu.bg mod/WindowMaker/menu.bg
--- orig/WindowMaker/menu.bg	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/menu.bg	2016-04-21 07:12:43.305985490 +0200
@@ -209,7 +209,7 @@
 		"Фон" END
 		"Запази темата" SHEXEC getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes/"%a(Theme name,Enter file name:)"
 		"Запази колекцията иконки" SHEXEC geticonset $HOME/GNUstep/Library/WindowMaker/IconSets/"%a(IconSet name,Enter file name:)"
-		"Пособие за настройки" EXEC /usr/local/GNUstep/Applications/WPrefs.app/WPrefs
+		"Пособие за настройки" EXEC /usr/bin//WPrefs
 	"Изглед" END
 
 	"Изход"	MENU
diff -Naur orig/WindowMaker/menu.fi mod/WindowMaker/menu.fi
--- orig/WindowMaker/menu.fi	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/menu.fi	2016-04-21 07:12:43.361985335 +0200
@@ -202,7 +202,7 @@
 		"Taustat" END
 		"Tallenna teema" SHEXEC getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes/"%a(Teeman nimi,Kirjoita teematiedoston nimi:)"
 		"Tallenna kuvakkeet" SHEXEC geticonset $HOME/GNUstep/Library/WindowMaker/IconSets/"%a(Kuvakesetin nimi,Kirjoita tiedostonnimi:)"
-		"Asetukset" EXEC /usr/local/GNUstep/Applications/WPrefs.app/WPrefs
+		"Asetukset" EXEC /usr/bin//WPrefs
 	"Ulkonäkö" END
 
 	"Lopeta"	MENU
diff -Naur orig/WindowMaker/menu.fy mod/WindowMaker/menu.fy
--- orig/WindowMaker/menu.fy	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/menu.fy	2016-04-21 07:12:43.377985307 +0200
@@ -208,7 +208,7 @@
 		"Eftergrûn" END
 		"Tema bewarje" SHEXEC getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes/"%a(Temanamme,Fier bestânsnamme yn:)"
 		"Iconenset bewarje" SHEXEC geticonset $HOME/GNUstep/Library/WindowMaker/IconSets/"%a(Ikoanensetnamme,Fier bestânsnamme yn:)"
-		"Foarkarrenhelpmiddel" EXEC /usr/local/GNUstep/Applications/WPrefs.app/WPrefs
+		"Foarkarrenhelpmiddel" EXEC /usr/bin//WPrefs
 	"Uterlik" END
 
 	"Sesje" MENU
diff -Naur orig/WindowMaker/menu.ko mod/WindowMaker/menu.ko
--- orig/WindowMaker/menu.ko	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/menu.ko	2016-04-21 07:12:43.425985223 +0200
@@ -163,7 +163,7 @@
 		"세션 지움" CLEAR_SESSION
 	"창 관 리" END
 	"작업환경" MENU
-		"설    정" SHEXEC /usr/local/GNUstep/Applications/WPrefs.app/WPrefs || wmakerconf
+		"설    정" SHEXEC /usr/bin//WPrefs || wmakerconf
 		"테    마" OPEN_MENU -noext THEMES_DIR ~/GNUstep/Library/WindowMaker/Themes WITH setstyle
 		"테마저장" EXEC getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes/"%a(테마 이름,파일 이름을 입력하세요:)"
 		"아이콘셋" OPEN_MENU -noext ICON_SETS_DIR ~/GNUstep/Library/WindowMaker/IconSets WITH seticons
diff -Naur orig/WindowMaker/menu.nl mod/WindowMaker/menu.nl
--- orig/WindowMaker/menu.nl	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/menu.nl	2016-04-21 07:12:43.433985209 +0200
@@ -208,7 +208,7 @@
 		"Achtergrond" END
 		"Thema opslaan" SHEXEC getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes/"%a(Themanaam,Voer bestandsnaam in:)"
 		"Iconenset opslaan" SHEXEC geticonset $HOME/GNUstep/Library/WindowMaker/IconSets/"%a(Iconensetnaam,Voer bestandsnaam in:)"
-		"Voorkeurenhulpmiddel" EXEC /usr/local/GNUstep/Applications/WPrefs.app/WPrefs
+		"Voorkeurenhulpmiddel" EXEC /usr/bin//WPrefs
 	"Uiterlijk" END
 
 	"Sessie" MENU
diff -Naur orig/WindowMaker/menu.ro mod/WindowMaker/menu.ro
--- orig/WindowMaker/menu.ro	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/menu.ro	2016-04-21 07:12:43.457985167 +0200
@@ -202,7 +202,7 @@
 		"Fundal" END
 		"Salvează Tema" SHEXEC getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes/"%a(Theme name,Enter file name:)"
 		"Salvează Setul de Iconiţe" SHEXEC geticonset $HOME/GNUstep/Library/WindowMaker/IconSets/"%a(IconSet name,Enter file name:)"
-		"Utilitar de Configurare" EXEC /usr/local/GNUstep/Applications/WPrefs.app/WPrefs
+		"Utilitar de Configurare" EXEC /usr/bin//WPrefs
 	"Configurare" END
 
 	"Ieşire" MENU
diff -Naur orig/WindowMaker/menu.sk mod/WindowMaker/menu.sk
--- orig/WindowMaker/menu.sk	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/menu.sk	2016-04-21 07:12:43.485985118 +0200
@@ -216,7 +216,7 @@
 		"Pozadie" END
 		"Uložiť tému" SHEXEC getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes/"%a(Meno témy,Zadajte názov súboru:)"
 		"Uložiť sadu ikon" SHEXEC geticonset $HOME/GNUstep/Library/WindowMaker/IconSets/"%a(Meno sady ikon,Zadajte názov súboru:)"
-		"Konfiguračný nástroj" EXEC /usr/local/GNUstep/Applications/WPrefs.app/WPrefs
+		"Konfiguračný nástroj" EXEC /usr/bin//WPrefs
 	"Vzhľad" END
 
 	"Koniec" MENU
diff -Naur orig/WindowMaker/menu.zh_TW mod/WindowMaker/menu.zh_TW
--- orig/WindowMaker/menu.zh_TW	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/menu.zh_TW	2016-04-21 07:12:43.509985076 +0200
@@ -199,7 +199,7 @@
 		"背景" END
 		"儲存主題" SHEXEC getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes/"%a(主題名稱)"
 		"儲存圖示集" SHEXEC geticonset $HOME/GNUstep/Library/WindowMaker/IconSets/"%a(圖示集名稱)"
-		"偏好設定公用程式" EXEC /usr/local/GNUstep/Applications/WPrefs.app/WPrefs
+		"偏好設定公用程式" EXEC /usr/bin//WPrefs
 	"外觀" END
 
 	"離開"	MENU
diff -Naur orig/WindowMaker/plmenu mod/WindowMaker/plmenu
--- orig/WindowMaker/plmenu	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/plmenu	2016-04-21 07:12:43.521985056 +0200
@@ -93,7 +93,7 @@
    ),
    ("Save Theme", SHEXEC, "getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes/\"%a(Theme name)\""),
    ("Save IconSet", SHEXEC, "geticonset $HOME/GNUstep/Library/WindowMaker/IconSets/\"%a(IconSet name)\""),
-   ("Preferences Utility", EXEC, "/usr/local/GNUstep/Applications/WPrefs.app/WPrefs")
+   ("Preferences Utility", EXEC, "/usr/bin//WPrefs")
    ),
    ("Session",
     ("Save Session", SAVE_SESSION),
diff -Naur orig/WindowMaker/plmenu.bg mod/WindowMaker/plmenu.bg
--- orig/WindowMaker/plmenu.bg	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/plmenu.bg	2016-04-21 07:12:43.525985048 +0200
@@ -94,7 +94,7 @@
    ),
   ("Запази темата", SHEXEC, "getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes/\"%a(Theme name)\""),
   ("Запази колекцията иконки", SHEXEC, "geticonset $HOME/GNUstep/Library/WindowMaker/IconSets/\"%a(IconSet name)\""),
-  ("Пособие за настройки", EXEC, "/usr/local/GNUstep/Applications/WPrefs.app/WPrefs")
+  ("Пособие за настройки", EXEC, "/usr/bin//WPrefs")
   ),
  ("Изход",
   ("Рестартиране", RESTART),
diff -Naur orig/WindowMaker/plmenu.da mod/WindowMaker/plmenu.da
--- orig/WindowMaker/plmenu.da	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/plmenu.da	2016-04-21 07:12:43.533985034 +0200
@@ -211,17 +211,17 @@
                 (
                 "Temaer",
                 OPEN_MENU,
-                "-noext /usr/local/share/WindowMaker/Themes  $HOME/GNUstep/Library/WindowMaker/Themes WITH setstyle"
+                "-noext /usr/share/WindowMaker/Themes  $HOME/GNUstep/Library/WindowMaker/Themes WITH setstyle"
                 ),
                 (
                 "Stil",
                 OPEN_MENU,
-                "-noext /usr/local/share/WindowMaker/Styles  $HOME/GNUstep/Library/WindowMaker/Styles WITH setstyle"
+                "-noext /usr/share/WindowMaker/Styles  $HOME/GNUstep/Library/WindowMaker/Styles WITH setstyle"
                 ),
                 (
                 "Ikonsæt",
                 OPEN_MENU,
-                "-noext /usr/local/share/WindowMaker/IconSets  $HOME/GNUstep/Library/WindowMaker/IconSets WITH seticons"
+                "-noext /usr/share/WindowMaker/IconSets  $HOME/GNUstep/Library/WindowMaker/IconSets WITH seticons"
                 ),
                 (
                 "Baggrund",
@@ -319,7 +319,7 @@
                         (
                         "Billeder",
                         OPEN_MENU,
-                        "-noext /usr/local/share/WindowMaker/Backgrounds  $HOME/GNUstep/Library/WindowMaker/Backgrounds WITH wmsetbg -u -t"
+                        "-noext /usr/share/WindowMaker/Backgrounds  $HOME/GNUstep/Library/WindowMaker/Backgrounds WITH wmsetbg -u -t"
                         )
                 ),
                 ("Gem Tema", SHEXEC, "getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes/\"%a(Tema navn,Indtast navn:)"),
diff -Naur orig/WindowMaker/plmenu.de mod/WindowMaker/plmenu.de
--- orig/WindowMaker/plmenu.de	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/plmenu.de	2016-04-21 07:12:43.541985021 +0200
@@ -66,9 +66,9 @@
   ("Gespeicherte Sitzung löschen", CLEAR_SESSION)
   ),
  ("Erscheinungsbild",
-  ("Themen", OPEN_MENU, "-noext /usr/local/share/WindowMaker/Themes $(HOME)/GNUstep/Library/WindowMaker/Themes WITH setstyle"),
-  ("Stile", OPEN_MENU, "-noext /usr/local/share/WindowMaker/Styles $(HOME)/GNUstep/Library/WindowMaker/Styles WITH setstyle"),
-  ("Iconsätze", OPEN_MENU, "-noext /usr/local/share/WindowMaker/IconSets $(HOME)/GNUstep/Library/WindowMaker/IconSets WITH seticons"),
+  ("Themen", OPEN_MENU, "-noext /usr/share/WindowMaker/Themes $(HOME)/GNUstep/Library/WindowMaker/Themes WITH setstyle"),
+  ("Stile", OPEN_MENU, "-noext /usr/share/WindowMaker/Styles $(HOME)/GNUstep/Library/WindowMaker/Styles WITH setstyle"),
+  ("Iconsätze", OPEN_MENU, "-noext /usr/share/WindowMaker/IconSets $(HOME)/GNUstep/Library/WindowMaker/IconSets WITH seticons"),
   ("Hintergrund",
    ("Einfarbig",
     ("Schwarz", EXEC, "wdwrite WindowMaker WorkspaceBack '(solid, black)'"),
@@ -90,7 +90,7 @@
     ("Grau schattiert",   EXEC, "wdwrite WindowMaker WorkspaceBack '(vgradient, \"#636380\", \"#131318\")'"),
     ("Weinfarben schattiert",   EXEC, "wdwrite WindowMaker WorkspaceBack '(vgradient, \"#600040\", \"#180010\")'")
     ),
-   ("Bilder", OPEN_MENU, "-noext /usr/local/share/WindowMaker/Backgrounds $(HOME)/GNUstep/Library/WindowMaker/Backgrounds WITH wmsetbg -u -t")
+   ("Bilder", OPEN_MENU, "-noext /usr/share/WindowMaker/Backgrounds $(HOME)/GNUstep/Library/WindowMaker/Backgrounds WITH wmsetbg -u -t")
    ),   
   ("Thema speichern", SHEXEC, "getstyle -t $(HOME)/GNUstep/Library/WindowMaker/Themes/\"%a(Themenname)\""),
   ("Iconsatz speichern", SHEXEC, "geticonset $(HOME)/GNUstep/Library/WindowMaker/IconSets/\"%a(Iconsatz-Name)\"")
diff -Naur orig/WindowMaker/plmenu.es mod/WindowMaker/plmenu.es
--- orig/WindowMaker/plmenu.es	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/plmenu.es	2016-04-21 07:12:43.545985013 +0200
@@ -66,9 +66,9 @@
   ("Borrar sesion", CLEAR_SESSION)
   ),
  ("Apariencia",
-  ("Temas", OPEN_MENU, "-noext /usr/local/share/WindowMaker/Themes $HOME/GNUstep/Library/WindowMaker/Themes WITH setstyle"),
-  ("Estilos", OPEN_MENU, "-noext /usr/local/share/WindowMaker/Styles $HOME/GNUstep/Library/WindowMaker/Styles WITH setstyle"),
-  ("Grupos de iconos", OPEN_MENU, "-noext /usr/local/share/WindowMaker/IconSets $HOME/GNUstep/Library/WindowMaker/IconSets WITH seticons"),
+  ("Temas", OPEN_MENU, "-noext /usr/share/WindowMaker/Themes $HOME/GNUstep/Library/WindowMaker/Themes WITH setstyle"),
+  ("Estilos", OPEN_MENU, "-noext /usr/share/WindowMaker/Styles $HOME/GNUstep/Library/WindowMaker/Styles WITH setstyle"),
+  ("Grupos de iconos", OPEN_MENU, "-noext /usr/share/WindowMaker/IconSets $HOME/GNUstep/Library/WindowMaker/IconSets WITH seticons"),
   ("Papel Tapiz",
    ("Solido",
     ("Negro", EXEC, "wdwrite WindowMaker WorkspaceBack '(solid, black)'"),
@@ -90,11 +90,11 @@
     ("Sombras grises",   EXEC, "wdwrite WindowMaker WorkspaceBack '(vgradient, \"#636380\", \"#131318\")'"),
     ("Sombras vino",   EXEC, "wdwrite WindowMaker WorkspaceBack '(vgradient, \"#600040\", \"#180010\")'")
     ),
-   ("Imagenes", OPEN_MENU, "-noext /usr/local/share/WindowMaker/Backgrounds $HOME/GNUstep/Library/WindowMaker/Backgrounds WITH wmsetbg -u -t")
+   ("Imagenes", OPEN_MENU, "-noext /usr/share/WindowMaker/Backgrounds $HOME/GNUstep/Library/WindowMaker/Backgrounds WITH wmsetbg -u -t")
    ),   
   ("Guardar tema", SHEXEC, "getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes/\"%a(Theme name)\""),
   ("Guardar iconos", SHEXEC, "geticonset $HOME/GNUstep/Library/WindowMaker/IconSets/\"%a(IconSet name)\""),
-  ("Preferencias", EXEC, "/usr/local/GNUstep/Applications/WPrefs.app/WPrefs")
+  ("Preferencias", EXEC, "/usr/bin//WPrefs")
   ), 
  ("Salir",
   ("Reiniciar", RESTART),
diff -Naur orig/WindowMaker/plmenu.fi mod/WindowMaker/plmenu.fi
--- orig/WindowMaker/plmenu.fi	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/plmenu.fi	2016-04-21 07:12:43.553985000 +0200
@@ -238,17 +238,17 @@
 		(
 		"Teemat",
 		OPEN_MENU,
-		"-noext /usr/local/share/WindowMaker/Themes  $HOME/GNUstep/Library/WindowMaker/Themes WITH setstyle"
+		"-noext /usr/share/WindowMaker/Themes  $HOME/GNUstep/Library/WindowMaker/Themes WITH setstyle"
 		),
 		(
 		"Tyylit",
 		OPEN_MENU,
-		"-noext /usr/local/share/WindowMaker/Styles  $HOME/GNUstep/Library/WindowMaker/Styles WITH setstyle"
+		"-noext /usr/share/WindowMaker/Styles  $HOME/GNUstep/Library/WindowMaker/Styles WITH setstyle"
 		),
 		(
 		"Kuvakkeet",
 		OPEN_MENU,
-		"-noext /usr/local/share/WindowMaker/IconSets  $HOME/GNUstep/Library/WindowMaker/IconSets WITH seticons"
+		"-noext /usr/share/WindowMaker/IconSets  $HOME/GNUstep/Library/WindowMaker/IconSets WITH seticons"
 		),
 		(
 		"Taustat",
@@ -341,7 +341,7 @@
 			(
 			"Kuvat",
 			OPEN_MENU,
-			"-noext /usr/local/share/WindowMaker/Backgrounds  $HOME/GNUstep/Library/WindowMaker/Backgrounds WITH wmsetbg -u -t"
+			"-noext /usr/share/WindowMaker/Backgrounds  $HOME/GNUstep/Library/WindowMaker/Backgrounds WITH wmsetbg -u -t"
 			)
 		),
 		(
@@ -357,7 +357,7 @@
 		(
 		"Asetukset",
 		EXEC,
-		"/usr/local/GNUstep/Applications/WPrefs.app/WPrefs"
+		"/usr/bin//WPrefs"
 		)
 	),
 	(
diff -Naur orig/WindowMaker/plmenu.fr mod/WindowMaker/plmenu.fr
--- orig/WindowMaker/plmenu.fr	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/plmenu.fr	2016-04-21 07:12:43.557984992 +0200
@@ -66,9 +66,9 @@
    ("Effacer la session", CLEAR_SESSION)
   ),
  ("Apparence",
-   ("Thèmes", OPEN_MENU, "-noext /opt/share/WindowMaker/Themes $HOME/GNUstep/Library/WindowMaker/Themes WITH setstyle"),
-   ("Styles", OPEN_MENU, "-noext /opt/share/WindowMaker/Styles $HOME/GNUstep/Library/WindowMaker/Styles WITH setstyle"),
-   ("Jeu d'icônes", OPEN_MENU, "-noext /opt/share/WindowMaker/IconSets $HOME/GNUstep/Library/WindowMaker/IconSets WITH seticons"),
+   ("Thèmes", OPEN_MENU, "-noext /usr/share/WindowMaker/Themes $HOME/GNUstep/Library/WindowMaker/Themes WITH setstyle"),
+   ("Styles", OPEN_MENU, "-noext /usr/share/WindowMaker/Styles $HOME/GNUstep/Library/WindowMaker/Styles WITH setstyle"),
+   ("Jeu d'icônes", OPEN_MENU, "-noext /usr/share/WindowMaker/IconSets $HOME/GNUstep/Library/WindowMaker/IconSets WITH seticons"),
    ("Fond d'écran",
      ("Unifié",
        ("Noir", EXEC, "wdwrite WindowMaker WorkspaceBack '(solid, black)'"),
@@ -88,7 +88,7 @@
        ("Dégradés de Gris",    EXEC, "wdwrite WindowMaker WorkspaceBack '(vgradient, \"#636380\", \"#131318\")'"),
        ("Dégradés de Bordeau", EXEC, "wdwrite WindowMaker WorkspaceBack '(vgradient, \"#600040\", \"#180010\")'")
       ),
-     ("Images", OPEN_MENU, "-noext /opt/share/WindowMaker/Backgrounds $HOME/GNUstep/Library/WindowMaker/Backgrounds WITH wmsetbg -u -t")
+     ("Images", OPEN_MENU, "-noext /usr/share/WindowMaker/Backgrounds $HOME/GNUstep/Library/WindowMaker/Backgrounds WITH wmsetbg -u -t")
       ),
    ("Sauver le thème", SHEXEC, "getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes /\"%a(Nom du thème,Entrez un nom de fichier:)\""),
    ("Sauver le jeu d'icônes", SHEXEC, "geticonset $HOME/GNUstep/Library/WindowMaker/IconSets /\"%a(Nom du jeu d'icônes,Entrez un nom de fichier:)\"")  
diff -Naur orig/WindowMaker/plmenu.fy mod/WindowMaker/plmenu.fy
--- orig/WindowMaker/plmenu.fy	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/plmenu.fy	2016-04-21 07:12:43.569984971 +0200
@@ -93,7 +93,7 @@
    ),
    ("Tema bewarje", SHEXEC, "getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes/\"%a(Temanamme)\""),
    ("Ikoanenset bewarje", SHEXEC, "geticonset $HOME/GNUstep/Library/WindowMaker/IconSets/\"%a(Ikoanensetnamme)\""),
-   ("Foarkarrenhelpmiddel", EXEC, "/usr/local/GNUstep/Applications/WPrefs.app/WPrefs")
+   ("Foarkarrenhelpmiddel", EXEC, "/usr/bin//WPrefs")
    ),
    ("Sesje",
     ("Sesje bewarje", SAVE_SESSION),
diff -Naur orig/WindowMaker/plmenu.hr mod/WindowMaker/plmenu.hr
--- orig/WindowMaker/plmenu.hr	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/plmenu.hr	2016-04-21 07:12:43.573984965 +0200
@@ -88,17 +88,17 @@
     (
       Teme,
       OPEN_MENU,
-      "-noext /usr/local/share/WindowMaker/Themes $HOME/GNUstep/Library/WindowMaker/Themes WITH setstyle"
+      "-noext /usr/share/WindowMaker/Themes $HOME/GNUstep/Library/WindowMaker/Themes WITH setstyle"
     ),
     (
       Stilovi,
       OPEN_MENU,
-      "-noext /usr/local/share/WindowMaker/Styles $HOME/GNUstep/Library/WindowMaker/Styles WITH setstyle"
+      "-noext /usr/share/WindowMaker/Styles $HOME/GNUstep/Library/WindowMaker/Styles WITH setstyle"
     ),
     (
       "Setovi ikona",
       OPEN_MENU,
-      "-noext /usr/local/share/WindowMaker/IconSets $HOME/GNUstep/Library/WindowMaker/IconSets WITH seticons"
+      "-noext /usr/share/WindowMaker/IconSets $HOME/GNUstep/Library/WindowMaker/IconSets WITH seticons"
     ),
     (
       Pozadina,
@@ -191,7 +191,7 @@
       (
         Slike,
         OPEN_MENU,
-        "-noext /usr/local/share/WindowMaker/Backgrounds $HOME/GNUstep/Library/WindowMaker/Backgrounds WITH wmsetbg -u -t"
+        "-noext /usr/share/WindowMaker/Backgrounds $HOME/GNUstep/Library/WindowMaker/Backgrounds WITH wmsetbg -u -t"
       )
     ),
     (
diff -Naur orig/WindowMaker/plmenu.ja mod/WindowMaker/plmenu.ja
--- orig/WindowMaker/plmenu.ja	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/plmenu.ja	2016-04-21 07:12:43.585984944 +0200
@@ -94,7 +94,7 @@
    ),   
   ("設定をテーマとしてセーブ", SHEXEC, "getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes/\"%a(Theme name)\""),
   ("現在のアイコンセットをセーブ", SHEXEC, "geticonset $HOME/GNUstep/Library/WindowMaker/IconSets/\"%a(IconSet name)\""),
-  ("設定ユーティリティ", EXEC, "/usr/local/GNUstep/Applications/WPrefs.app/WPrefs")
+  ("設定ユーティリティ", EXEC, "/usr/bin//WPrefs")
   ), 
  ("終了",
   ("再起動", RESTART),
diff -Naur orig/WindowMaker/plmenu.ko mod/WindowMaker/plmenu.ko
--- orig/WindowMaker/plmenu.ko	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/plmenu.ko	2016-04-21 07:12:43.589984936 +0200
@@ -63,7 +63,7 @@
   ),
  ("작업공간", WORKSPACE_MENU),
  ("작업환경",
-  ("설    정", SHEXEC, "/usr/local/GNUstep/Applications/WPrefs.app/WPrefs || wmakerconf"),
+  ("설    정", SHEXEC, "/usr/bin//WPrefs || wmakerconf"),
   ("테    마", OPEN_MENU, "-noext /usr/share/WindowMaker/Themes ~/GNUstep/Library/WindowMaker/Themes WITH setstyle"),
   ("테마저장", EXEC, "getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes/\"%a(테마 이름,파일 이름을 입력하세요:)\""),
   ("스 타 일", OPEN_MENU, "-noext /usr/share/WindowMaker/Styles ~/GNUstep/Library/WindowMaker/Styles WITH setstyle"),
diff -Naur orig/WindowMaker/plmenu.nl mod/WindowMaker/plmenu.nl
--- orig/WindowMaker/plmenu.nl	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/plmenu.nl	2016-04-21 07:12:43.593984929 +0200
@@ -93,7 +93,7 @@
    ),
    ("Thema opslaan", SHEXEC, "getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes/\"%a(Themanaam)\""),
    ("Iconenset opslaan", SHEXEC, "geticonset $HOME/GNUstep/Library/WindowMaker/IconSets/\"%a(Iconensetnaam)\""),
-   ("Voorkeurenhulpmiddel", EXEC, "/usr/local/GNUstep/Applications/WPrefs.app/WPrefs")
+   ("Voorkeurenhulpmiddel", EXEC, "/usr/bin//WPrefs")
    ),
    ("Sessie",
     ("Sessie opslaan", SAVE_SESSION),
diff -Naur orig/WindowMaker/plmenu.ro mod/WindowMaker/plmenu.ro
--- orig/WindowMaker/plmenu.ro	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/plmenu.ro	2016-04-21 07:12:43.609984902 +0200
@@ -94,7 +94,7 @@
    ),   
   ("Salvează Tema", SHEXEC, "getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes/\"%a(Theme name)\""),
   ("Salvează Setul de Iconiţe", SHEXEC, "geticonset $HOME/GNUstep/Library/WindowMaker/IconSets/\"%a(IconSet name)\""),
-  ("Utilitar de Configurare", EXEC, "/usr/local/GNUstep/Applications/WPrefs.app/WPrefs")
+  ("Utilitar de Configurare", EXEC, "/usr/bin//WPrefs")
   ), 
  ("Ieşire",
   ("Repornire", RESTART),
diff -Naur orig/WindowMaker/plmenu.sk mod/WindowMaker/plmenu.sk
--- orig/WindowMaker/plmenu.sk	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/plmenu.sk	2016-04-21 07:12:43.617984888 +0200
@@ -94,7 +94,7 @@
    ),   
   ("Uložiť tému", SHEXEC, "getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes/\"%a(Theme name)\""),
   ("Uložiť sadu ikon", SHEXEC, "geticonset $HOME/GNUstep/Library/WindowMaker/IconSets/\"%a(IconSet name)\""),
-  ("Konfiguračný nástroj", EXEC, "/usr/local/GNUstep/Applications/WPrefs.app/WPrefs")
+  ("Konfiguračný nástroj", EXEC, "/usr/bin//WPrefs")
   ), 
  ("Koniec",
   ("Reštartovať", RESTART),
diff -Naur orig/WindowMaker/plmenu.zh_TW mod/WindowMaker/plmenu.zh_TW
--- orig/WindowMaker/plmenu.zh_TW	2015-08-11 20:41:14.000000000 +0200
+++ mod/WindowMaker/plmenu.zh_TW	2016-04-21 07:12:43.629984867 +0200
@@ -92,7 +92,7 @@
    ),
   ("儲存主題", SHEXEC, "getstyle -t $HOME/GNUstep/Library/WindowMaker/Themes/%a(主題名稱)"),
   ("儲存圖示集", SHEXEC, "geticonset $HOME/GNUstep/Library/WindowMaker/IconSets/%a(圖示集名稱)"),
-  ("偏好設定公用程式", EXEC, "/usr/local/GNUstep/Applications/WPrefs.app/WPrefs")
+  ("偏好設定公用程式", EXEC, "/usr/bin//WPrefs")
   ),
  ("離開",
   ("重新啟動", RESTART),
