diff -Naur orig/util/po/Makefile.am mod/util/po/Makefile.am
--- orig/util/po/Makefile.am	2015-08-11 20:41:14.000000000 +0200
+++ mod/util/po/Makefile.am	2016-04-20 15:45:57.615038040 +0200
@@ -4,7 +4,7 @@
 
 CLEANFILES = $(DOMAIN).pot $(CATALOGS)
 
-EXTRA_DIST = de.po es.po fr.po fy.po nl.po pt.po
+EXTRA_DIST = de.po es.po fr.po fy.po nl.po pt.po hu.po
 
 POTFILES  = \
 	$(top_srcdir)/util/wmgenmenu.c \
