diff -Naur orig/data/griffith.desktop patched/data/griffith.desktop
--- orig/data/griffith.desktop	2008-01-29 21:47:53.000000000 +0100
+++ patched/data/griffith.desktop	2016-02-19 01:52:47.954186446 +0100
@@ -9,11 +9,12 @@
 Type=Application
 StartupNotify=true
 StartupWMClass=Griffith
-Categories=GNOME;GTK;Database;AudioVideo;
+Categories=GTK;GNOME;Database;AudioVideo;
 
 GenericName[en_PT]=Griffith - Um gestor de colecções de filmes
 GenericName[pl]=Griffith - wspomaga zarządzanie kolekcją filmów
 
+Comment[hu]=Filmkollekció kezelő alkalmazás
 Comment[pt_PT]=Gerir uma colecção de filmes.
 Comment[pt]=Gerir uma colecção de filmes.
 Comment[pl]=wspomaga zarządzanie kolekcją filmów.
