diff -Nur orig/components/chmhelp/lhelp/Makefile mod/components/chmhelp/lhelp/Makefile
--- orig/components/chmhelp/lhelp/Makefile	2015-10-01 20:40:54.000000000 +0200
+++ mod/components/chmhelp/lhelp/Makefile	2017-06-08 07:29:49.124303469 +0200
@@ -840,7 +840,7 @@
 override CLEAN_FILES+=$(wildcard $(COMPILER_UNITTARGETDIR)/*$(OEXT)) $(wildcard $(COMPILER_UNITTARGETDIR)/*$(PPUEXT)) $(wildcard $(COMPILER_UNITTARGETDIR)/*$(RSTEXT)) $(wildcard $(COMPILER_UNITTARGETDIR)/*.compiled) $(wildcard $(COMPILER_UNITTARGETDIR)/*.lfm) $(wildcard $(COMPILER_UNITTARGETDIR)/*.or) $(wildcard $(COMPILER_UNITTARGETDIR)/*.res) $(wildcard *$(OEXT)) $(wildcard *$(PPUEXT)) $(wildcard *$(RSTEXT))
 endif
 ifeq ($(FULL_TARGET),i386-linux)
-override COMPILER_OPTIONS+=-gs -gl -Sghi -O3 -dlcl$(LCL_PLATFORM) $(DBG_OPTIONS)
+override COMPILER_OPTIONS+=-gs -gl -Sghi -O2 -dlcl$(LCL_PLATFORM) $(DBG_OPTIONS)
 endif
 ifeq ($(FULL_TARGET),i386-go32v2)
 override COMPILER_OPTIONS+=-gs -gl -Sghi -O3 -dlcl$(LCL_PLATFORM) $(DBG_OPTIONS)
@@ -975,7 +975,7 @@
 override COMPILER_OPTIONS+=-gs -gl -Sghi -O3 -dlcl$(LCL_PLATFORM) $(DBG_OPTIONS)
 endif
 ifeq ($(FULL_TARGET),x86_64-linux)
-override COMPILER_OPTIONS+=-gs -gl -Sghi -O3 -dlcl$(LCL_PLATFORM) $(DBG_OPTIONS)
+override COMPILER_OPTIONS+=-gs -gl -Sghi -O2 -dlcl$(LCL_PLATFORM) $(DBG_OPTIONS)
 endif
 ifeq ($(FULL_TARGET),x86_64-freebsd)
 override COMPILER_OPTIONS+=-gs -gl -Sghi -O3 -dlcl$(LCL_PLATFORM) $(DBG_OPTIONS)
diff -Nur orig/components/chmhelp/lhelp/Makefile.fpc mod/components/chmhelp/lhelp/Makefile.fpc
--- orig/components/chmhelp/lhelp/Makefile.fpc	2015-10-01 20:40:54.000000000 +0200
+++ mod/components/chmhelp/lhelp/Makefile.fpc	2017-06-07 21:12:24.428270082 +0200
@@ -15,7 +15,7 @@
 packages=rtl
 
 [compiler]
-options=-gs -gl -Sghi -O3 -dlcl$(LCL_PLATFORM) $(DBG_OPTIONS)
+options=-gs -gl -Sghi -O2 -dlcl$(LCL_PLATFORM) $(DBG_OPTIONS)
 unittargetdir=lib/$(CPU_TARGET)-$(OS_TARGET)
 unitdir=. $(LAZDIR)/components/lazutils/lib/$(CPU_TARGET)-$(OS_TARGET) \
           $(LAZDIR)/packager/units/$(CPU_TARGET)-$(OS_TARGET) \
diff -Nur orig/ide/compileroptions.pp mod/ide/compileroptions.pp
--- orig/ide/compileroptions.pp	2016-09-09 22:24:49.000000000 +0200
+++ mod/ide/compileroptions.pp	2017-06-07 21:12:59.808215068 +0200
@@ -2627,7 +2627,7 @@
         -O-        Disable optimizations
         -O1        Level 1 optimizations (quick and debugger friendly)
         -O2        Level 2 optimizations (-O1 + quick optimizations)
-        -O3        Level 3 optimizations (-O2 + slow optimizations)
+        -O2        Level 3 optimizations (-O2 + slow optimizations)
         -Oa<x>=<y> Set alignment
         -Oo[NO]<x> Enable or disable optimizations, see fpc -i for possible values
         -Op<x>     Set target cpu for optimizing, see fpc -i for possible values
