diff -up rp-pppoe-3.11/src/configure.orig rp-pppoe-3.11/src/configure
--- rp-pppoe-3.11/src/configure	2012-08-17 12:31:25.000000000 -0600
+++ rp-pppoe-3.11/src/configure	2013-02-11 14:23:05.000000000 -0700
@@ -3720,6 +3720,7 @@
 #include<net/ethernet.h>
 #include<linux/if.h>
 #include<linux/in.h>
+#include<linux/in6.h>
 
 "
 if test "x$ac_cv_header_linux_if_pppox_h" = xyes; then :
@@ -4526,6 +4527,7 @@
 /* end confdefs.h.  */
 #include <sys/socket.h>
 #include <netinet/in.h>
+#include <linux/in6.h>
 #include <linux/if_pppox.h>
 int main()
 {
