diff -Nur orig/gui/tkpppoe.in mod/gui/tkpppoe.in
--- orig/gui/tkpppoe.in	2018-11-25 23:25:28.000000000 +0100
+++ mod/gui/tkpppoe.in	2019-08-09 20:39:11.477901627 +0200
@@ -1,4 +1,5 @@
 #!/bin/sh
+export PATH=$PATH:/bin:/sbin
 # -*-Mode: TCL;-*-
 # LIC: GPL
 
diff -Nur orig/scripts/pppoe-connect.in mod/scripts/pppoe-connect.in
--- orig/scripts/pppoe-connect.in	2018-11-25 23:25:28.000000000 +0100
+++ mod/scripts/pppoe-connect.in	2019-08-09 20:40:14.965793146 +0200
@@ -21,7 +21,7 @@
 # If config_file is omitted, defaults to /etc//ppp/pppoe.conf
 #
 #***********************************************************************
-
+export PATH=$PATH:/bin:/sbin
 # From AUTOCONF
 prefix=@prefix@
 exec_prefix=@exec_prefix@
diff -Nur orig/scripts/pppoe-init.in mod/scripts/pppoe-init.in
--- orig/scripts/pppoe-init.in	2018-11-25 23:25:28.000000000 +0100
+++ mod/scripts/pppoe-init.in	2019-08-09 20:40:47.237737090 +0200
@@ -10,7 +10,7 @@
 # Copyright (C) 2000 Roaring Penguin Software Inc.  This software may
 # be distributed under the terms of the GNU General Public License, version
 # 2 or any later version.
-
+export PATH=$PATH:/bin:/sbin
 # Source function library if it exists
 test -r /etc/rc.d/init.d/functions && . /etc/rc.d/init.d/functions
 
diff -Nur orig/scripts/pppoe-setup.in mod/scripts/pppoe-setup.in
--- orig/scripts/pppoe-setup.in	2018-11-25 23:25:28.000000000 +0100
+++ mod/scripts/pppoe-setup.in	2019-08-09 20:41:07.101702862 +0200
@@ -11,7 +11,7 @@
 #
 # $Id$
 #***********************************************************************
-
+export PATH=$PATH:/bin:/sbin
 # From AUTOCONF
 prefix=@prefix@
 exec_prefix=@exec_prefix@
diff -Nur orig/scripts/pppoe-start.in mod/scripts/pppoe-start.in
--- orig/scripts/pppoe-start.in	2018-11-25 23:25:28.000000000 +0100
+++ mod/scripts/pppoe-start.in	2019-08-09 20:41:19.989680886 +0200
@@ -21,7 +21,7 @@
 # If config_file is omitted, defaults to /etc/ppp/pppoe.conf
 #
 #***********************************************************************
-
+export PATH=$PATH:/bin:/sbin
 # From AUTOCONF
 prefix=@prefix@
 exec_prefix=@exec_prefix@
diff -Nur orig/scripts/pppoe-status mod/scripts/pppoe-status
--- orig/scripts/pppoe-status	2018-11-25 23:25:28.000000000 +0100
+++ mod/scripts/pppoe-status	2019-08-09 20:39:54.429827741 +0200
@@ -18,7 +18,7 @@
 # If config_file is omitted, defaults to /etc/ppp/pppoe.conf
 #
 #***********************************************************************
-
+export PATH=$PATH:/bin:/sbin
 # Defaults
 CONFIG=/etc/ppp/pppoe.conf
 
@@ -81,4 +81,4 @@
 
 echo "pppoe-status: Link is down -- could not find interface corresponding to"
 echo "pppd pid $PPPD_PID"
-exit 1
\ Nincs új sor a fájl végén
+exit 1
diff -Nur orig/scripts/pppoe-stop.in mod/scripts/pppoe-stop.in
--- orig/scripts/pppoe-stop.in	2018-11-25 23:25:28.000000000 +0100
+++ mod/scripts/pppoe-stop.in	2019-08-09 20:39:42.281848426 +0200
@@ -19,7 +19,7 @@
 # If config_file is omitted, defaults to /etc/ppp/pppoe.conf
 #
 #***********************************************************************
-
+export PATH=$PATH:/bin:/sbin
 # Set to "C" locale so we can parse messages from commands
 LANG=C
 export LANG
