diff -Naur orig/Kino.desktop patched/Kino.desktop
--- orig/Kino.desktop	2009-03-28 17:54:22.000000000 +0100
+++ patched/Kino.desktop	2014-08-06 09:02:39.272519139 +0200
@@ -38,12 +38,11 @@
 Comment[ru]=Редактирование DV-видео
 Comment[tt]=DV видео редакторлау
 Comment[uk]=Редагування DV відео
-Encoding=UTF-8
 TryExec=kino
 Exec=kino %F
-Icon=kino.png
+Icon=kino
 Terminal=false
 StartupNotify=true
 Type=Application
-Categories=Application;AudioVideo;AudioVideoEditing;
-MimeType=video/dv;application/smil
+Categories=AudioVideo;AudioVideoEditing;
+MimeType=video/dv;application/smil;
