diff -Naur orig/Makefile patched/Makefile
--- orig/Makefile	1995-12-05 13:26:11.000000000 +0100
+++ patched/Makefile	2015-12-12 21:13:22.250646481 +0100
@@ -16,10 +16,10 @@
 PERL = /usr/bin/perl
 
 # Where do you want the sgmlspl executable script to be installed?
-BINDIR = /usr/local/bin
+BINDIR = ${UB_INSTALLDIR}/usr/bin
 
 # Where do you put local perl5 libaries?
-PERL5DIR = /usr/local/lib/perl5
+PERL5DIR = ${UB_INSTALLDIR}/usr/lib/perl5/5.28.1
 MODULEDIR = ${PERL5DIR}/SGMLS
 
 # Where do you want to put sgmlspl specifications?
@@ -29,7 +29,7 @@
 # do you intend to put it?  'make html' will create two
 # subdirectories, ${HTMLDIR}/SGMLSpm and ${HTMLDIR}/sgmlspl, and place
 # its files there.
-HTMLDIR = /usr/local/lib/www/docs
+HTMLDIR = ${UB_INSTALLDIR}/usr/lib/www/docs
 
 
 #
@@ -62,7 +62,7 @@
 
 ${MODULEDIR}:
 	if [ ! -d ${MODULEDIR} ]; then\
-	  mkdir ${MODULEDIR}; \
+	  mkdir -p ${MODULEDIR}; \
 	  chmod a+x ${MODULEDIR}; \
 	fi
 
