diff -Naur orig/macros/mate-autogen patched/macros/mate-autogen
--- orig/macros/mate-autogen	2020-04-02 12:46:36.000000000 +0200
+++ patched/macros/mate-autogen	2020-04-08 01:01:39.345025608 +0200
@@ -315,7 +315,7 @@
 
     # check that AM_MAINTAINER_MODE is used
     if grep "^AM_MAINTAINER_MODE" $configure_ac >/dev/null; then
-	want_maintainer_mode=true
+	want_maintainer_mode=false
     fi
 
     if grep "^APPDATA_XML" $configure_ac >/dev/null; then
@@ -433,6 +433,10 @@
     conf_flags="--enable-maintainer-mode"
 fi
 
+if $want_gtk_doc; then
+    conf_flags="--disable-gtk-doc"
+fi
+
 if test x$NOCONFIGURE = x; then
     printbold Running $srcdir/configure $conf_flags "$@" ...
     $srcdir/configure $conf_flags "$@" \
