diff -Naurd a/kkconsui/src/texteditor.cc b/kkconsui/src/texteditor.cc
--- a/kkconsui/src/texteditor.cc	2003-09-09 23:51:33.000000000 +0200
+++ b/kkconsui/src/texteditor.cc	2012-04-17 17:59:11.000000000 +0200
@@ -1940,7 +1940,7 @@
 }
 
 int texteditor::findint(void *p1, void *p2) {
-    return *(int *) p1 != (int) p2;
+    return *(int *) p1 != (long) p2;
 }
 
 int texteditor::findhighline(void *p1, void *p2) {
diff -Naurd a/kkstrtext/kkstrtext.cc b/kkstrtext/kkstrtext.cc
--- a/kkstrtext/kkstrtext.cc	2005-02-01 01:13:24.000000000 +0100
+++ b/kkstrtext/kkstrtext.cc	2012-04-17 17:55:38.000000000 +0200
@@ -431,7 +431,7 @@
 }
 
 int intcompare(void *s1, void *s2) {
-    return (int) s1 != (int) s2;
+    return s1 != s2;
 }
 
 string i2str(int i) {
diff -Naurd a/kkstrtext/kkstrtext.h b/kkstrtext/kkstrtext.h
--- a/kkstrtext/kkstrtext.h	2004-11-18 00:00:38.000000000 +0100
+++ b/kkstrtext/kkstrtext.h	2012-04-17 17:57:45.000000000 +0200
@@ -87,7 +87,7 @@
     { \
 	va_list vgs__ap; char vgs__buf[1024]; \
 	va_start(vgs__ap, fmt); \
-	vsprintf(vgs__buf, fmt, vgs__ap); c = vgs__buf; \
+        vsnprintf(vgs__buf, 1024, fmt, vgs__ap); c = vgs__buf; \
 	va_end(vgs__ap); \
     }
 
diff -Naurd a/share/Makefile.in b/share/Makefile.in
--- a/share/Makefile.in	2005-02-12 16:07:17.000000000 +0100
+++ b/share/Makefile.in	2012-04-17 18:15:31.000000000 +0200
@@ -225,7 +225,7 @@
 	-if test -d templates; then rm -f tmpl.tar.gz; fi
 
 install-data-local: tmpl.tar.gz
-	tar zxvf tmpl.tar.gz -C $(datadir)/motor/
+	tar zxvf tmpl.tar.gz -C $(DESTDIR)$(datadir)/motor/
 	-find $(datadir)/motor/ -name CVS -type d -exec rm -rf {} \;
 	if test -d templates; then rm -f tmpl.tar.gz; fi
 
diff -Naurd a/src/ui/ncurses/ncursesui.cc b/src/ui/ncurses/ncursesui.cc
--- a/src/ui/ncurses/ncursesui.cc	2004-11-14 22:04:08.000000000 +0100
+++ b/src/ui/ncurses/ncursesui.cc	2012-04-17 19:39:25.000000000 +0200
@@ -262,11 +262,11 @@
 
 void ncursesui::usage() {
     kendinterface();
-    cout << "usage: motor [options] [project name]" << endl <<
+    cout << "\nusage: motor [options] [project name]" << endl <<
     "options are as follows:" << endl <<
     "\t-a, --ascii\tuse ascii characters to draw lines" << endl <<
     "\t-t, --debugtty\tuse the specified tty for debugging" << endl <<
-    "\t-h, --help\tdisplay this help screen" << endl;
+    "\t-h, --help\tdisplay this help screen\n" << endl;
 }
 
 bool ncursesui::regproject() {
@@ -1242,7 +1242,7 @@
 	populatesettingstree(*db.gettree());
         if(fin = !db.open(n, b, &p)) break;
 
-        switch((int) p) {
+        switch((long) p) {
             case 10:
 		if(input(motorui::text, buf = project.getversion(),
 		_("version: ")) == motorui::yes) {
@@ -1356,7 +1356,8 @@
 vector<shitpair> treeshit;
 
 void ncursesui::populatecontentstree(treeview &tree, motorproject &mp, projeditaction pea, bool setcurrent) {
-    int foldid, id, cfid = -1, i, sid;
+    int foldid, id, cfid = -1, i;
+    long sid;
     vector<motorfolder>::iterator ifold;
     vector<motorfile>::iterator ifile;
     static vector<string> cnodes;
@@ -1368,7 +1369,7 @@
 	    id = tree.getid(i);
 
 	    if(tree.isnode(id))
-		if(sid = (int) tree.getref(id))
+		if(sid = (long) tree.getref(id))
 		    if(!tree.isnodeopen(id))
 			cnodes.push_back(treeshit[sid-1].second->gettagname());
 	}
diff -Naurd a/src/ui/ncurses/uitagbrowser.cc b/src/ui/ncurses/uitagbrowser.cc
--- a/src/ui/ncurses/uitagbrowser.cc	2004-07-23 02:18:16.000000000 +0200
+++ b/src/ui/ncurses/uitagbrowser.cc	2012-04-17 17:51:56.000000000 +0200
@@ -95,7 +95,7 @@
     cm.close();
 
     if(n) {
-	int c = (int) cm.getref(n-1);
+	long c = (long) cm.getref(n-1);
 
 	if(c != tagbrowser.getscope()) {
 	    tagbrowser.setscope((motortagbrowser::viewscope) c);
@@ -142,7 +142,7 @@
 	switch(cm.open()) {
 	    case -2:
 		iter = false;
-		c = (char) (int) cm.getref(cm.getpos());
+		c = (char) (long) cm.getref(cm.getpos());
 
 		while((pos = nfilter.find(c)) != -1) {
 		    nfilter.erase(pos, 1);
@@ -213,7 +213,7 @@
 }
 
 void uitagbrowser::exec() {
-    int n, k;
+    long n, k;
     bool r;
 
     if(search.empty())
@@ -240,7 +240,7 @@
     m.setpos(mpos);
 
     if(r = (n = m.open())) {
-        k = (int) m.getref(n-1);
+        k = (long) m.getref(n-1);
         mpos = n-1;
     }
 
diff -Naurd a/src/ui/ncurses/uivcs.cc b/src/ui/ncurses/uivcs.cc
--- a/src/ui/ncurses/uivcs.cc	2005-02-12 15:58:28.000000000 +0100
+++ b/src/ui/ncurses/uivcs.cc	2012-04-17 17:53:30.000000000 +0200
@@ -31,7 +31,8 @@
 void uivcs::check() {
     bool found;
     dialogbox db;
-    int n, b, i, nodes[motorvcs::remotely_modified+1];
+    int n, b, nodes[motorvcs::remotely_modified+1];
+    long i;
     string comment, buf;
 
     if(!enabled()) return;
@@ -98,7 +99,7 @@
 	switch(b) {
 	    case 0:
 		if(!i) {
-                    i = (int) tree.getref(tree.getid(n));
+                    i = (long) tree.getref(tree.getid(n));
 		    ic = ch.begin()+i-1;
 
                     found = (ik = find(naffect.begin(), naffect.end(),
