diff -Nur orig/setup/build_environment.py mod/setup/build_environment.py
--- orig/setup/build_environment.py	2018-07-20 06:19:03.000000000 +0200
+++ mod/setup/build_environment.py	2018-07-23 09:32:17.708680340 +0200
@@ -101,7 +101,7 @@
     else:
         q = os.path.join(sys.prefix, 'share', 'sip')
     q = os.environ.get('SIP_DIR', q)
-    for x in ('', 'Py2-PyQt5', 'PyQt5', 'sip/PyQt5'):
+    for x in ('', 'Py2-PyQt5', 'PyQt5', 'sip/PyQt5', 'PyQt5-python2' ):
         base = os.path.join(q, x)
         if os.path.exists(os.path.join(base, 'QtWidgets')):
             return base
