diff -Naur orig/config-top.h patched/config-top.h
--- orig/config-top.h	2016-05-19 20:34:02.134739579 +0200
+++ patched/config-top.h	2017-07-29 16:03:54.463066291 +0200
@@ -87,7 +87,7 @@
 #define DEFAULT_BASHRC "~/.bashrc"
 
 /* System-wide .bashrc file for interactive shells. */
-/* #define SYS_BASHRC "/etc/bash.bashrc" */
+#define SYS_BASHRC "/etc/bashrc"
 
 /* System-wide .bash_logout for login shells. */
 /* #define SYS_BASH_LOGOUT "/etc/bash.bash_logout" */
