diff -Naur orig/src/sysinfo.cpp patched/src/sysinfo.cpp
--- orig/src/sysinfo.cpp	2019-04-24 17:41:49.000000000 +0200
+++ patched/src/sysinfo.cpp	2019-04-24 23:41:08.879822452 +0200
@@ -680,13 +680,13 @@
         if (g_file_test ("/etc/linuxmint/info", G_FILE_TEST_EXISTS)) {
             return new MintSysInfo;
         }
-        else if (g_file_test ("/etc/os-release", G_FILE_TEST_EXISTS)) {
-            return new GenericSysInfo;
-        }
         else if (char *p = g_find_program_in_path("lsb_release")) {
             g_free(p);
             return new LSBSysInfo;
         }
+        else if (g_file_test ("/etc/os-release", G_FILE_TEST_EXISTS)) {
+            return new GenericSysInfo;
+        }
         else if (SysInfo::system() == "FreeBSD") {
             return new FreeBSDSysInfo;
         }
