diff -Naur orig/src/sysinfo.cpp patched/src/sysinfo.cpp
--- orig/src/sysinfo.cpp	2019-04-24 23:41:08.879822452 +0200
+++ patched/src/sysinfo.cpp	2019-04-24 23:42:22.292565327 +0200
@@ -330,10 +330,9 @@
         {
             xmlDocPtr document;
             xmlXPathContextPtr context;
-            const string nodes[3] = { "string(/mate-version/platform)",
-                                      "string(/mate-version/minor)",
-                                      "string(/mate-version/micro)" };
-            string values[3];
+            const string nodes[2] = { "string(/mate-version/platform)",
+                                      "string(/mate-version/minor)"};
+            string values[2];
 
             if (not (document = xmlParseFile(DATADIR "/mate-about/mate-version.xml")))
                 return;
@@ -341,7 +340,7 @@
             if (not (context = xmlXPathNewContext(document)))
                 return;
 
-            for (size_t i = 0; i != 3; ++i)
+            for (size_t i = 0; i != 2; ++i)
             {
                 xmlXPathObjectPtr xpath;
                 xpath = xmlXPathEvalExpression(BAD_CAST nodes[i].c_str(), context);
@@ -355,8 +354,8 @@
             xmlXPathFreeContext(context);
             xmlFreeDoc(document);
 
-            if (!values[0].empty() && !values[1].empty() && !values[2].empty())
-                this->mate_version = values[0] + '.' + values[1] + '.' + values[2];
+            if (!values[0].empty() && !values[1].empty())
+                this->mate_version = values[0] + '.' + values[1];
         }
     };
 
