diff -Naur orig/src/sysinfo.cpp patched/src/sysinfo.cpp
--- orig/src/sysinfo.cpp	2015-07-02 23:46:27.000000000 +0200
+++ patched/src/sysinfo.cpp	2015-09-12 23:06:42.372114761 +0200
@@ -533,7 +532,8 @@
                     std::getline(input, s);
                     if (s.find("NAME=") == 0) {
                         len = strlen("NAME=");
-                        this->distro_name = s.substr(len);
+                        // also strip the surrounding quotes
+                        this->distro_name = s.substr(len + 1, s.size() - len - 2);
                     } else if (s.find("VERSION=") == 0) {
                         len = strlen("VERSION=");
                         // also strip the surrounding quotes
