diff -Naurd a/ROX-Filer/src/Makefile.in b/ROX-Filer/src/Makefile.in
--- a/ROX-Filer/src/Makefile.in	2011-10-09 16:31:59.000000000 +0200
+++ b/ROX-Filer/src/Makefile.in	2013-07-23 19:33:50.000000000 +0200
@@ -7,7 +7,7 @@
 .SUFFIXES: .c .o
 
 PKG_CONFIG = @PKG_CONFIG@
-LIBS = @LIBS@ @LFS_LDFLAGS@ -lICE -lSM
+LIBS = @LIBS@ @LFS_LDFLAGS@ -lICE -lSM -ldl -lm
 #PROF = -fprofile-arcs -pg
 
 PKG_CONFIG_FLAGS=
