diff -Nur orig/fusion-icon.desktop.in mod/fusion-icon.desktop.in
--- orig/fusion-icon.desktop.in	2017-06-10 22:04:14.000000000 +0200
+++ mod/fusion-icon.desktop.in	2018-03-28 21:13:41.920072208 +0200
@@ -9,6 +9,4 @@
 X-MATE-Autostart-Phase=WindowManager
 X-MATE-Provides=windowmanager
 X-MATE-WMName=fusion-icon
-X-GNOME-Autostart-Phase=WindowManager
-X-GNOME-Provides=windowmanager
-X-GNOME-WMName=fusion-icon
+OnlyShowIn=MATE;
