diff -Naur orig/configure mod/configure
--- orig/configure	2015-11-10 17:59:23.000000000 +0100
+++ mod/configure	2016-03-01 23:36:45.893760465 +0100
@@ -27445,15 +27445,10 @@
 $as_echo "and out again" >&6; }
 
 echo $ECHO_N "ordering CD from http://tobi.oetiker.ch/wish $ECHO_C" 1>&6
-sleep 1
 echo $ECHO_N ".$ECHO_C" 1>&6
-sleep 1
 echo $ECHO_N ".$ECHO_C" 1>&6
-sleep 1
 echo $ECHO_N ".$ECHO_C" 1>&6
-sleep 1
 echo $ECHO_N ".$ECHO_C" 1>&6
-sleep 1
 { $as_echo "$as_me:${as_lineno-$LINENO}: result:  just kidding ;-)" >&5
 $as_echo " just kidding ;-)" >&6; }
 echo
