diff -Nur orig/main.pro mod/main.pro
--- orig/main.pro	2017-11-14 21:40:53.000000000 +0100
+++ mod/main.pro	2018-01-10 19:38:01.623313956 +0100
@@ -4,3 +4,7 @@
 CONFIG += ordered
 
 SUBDIRS = libfunction libevolvotron evolvotron evolvotron_render evolvotron_mutate
+
+evolvotron.depends = libevolvotron libfunction
+evolvotron_mutate.depends = libevolvotron libfunction
+evolvotron_render.depends = libevolvotron libfunction
