diff -urN lincity-1.12pre47.orig/mouse.c lincity-1.12pre47/mouse.c
--- lincity-1.12pre47.orig/mouse.c	2001-08-21 01:56:48.000000000 +0200
+++ lincity-1.12pre47/mouse.c	2003-05-16 15:13:17.000000000 +0200
@@ -21,8 +21,6 @@
 #include "lchelp.h"
 #include "pbar.h"
 
-#define DEBUG_MT_CODE 1
-
 extern Update_Scoreboard update_scoreboard;
 
 /* ---------------------------------------------------------------------- *
