diff -Naur orig/Makefile.local mod/Makefile.local
--- orig/Makefile.local	2012-05-28 23:22:31.000000000 +0200
+++ mod/Makefile.local	2014-11-18 22:21:55.565721312 +0100
@@ -23,7 +23,11 @@
 # Packagers can uncomment and change DEFAULT_BASEDIR to define a system
 # installation path for the game. Otherwise, you can start the game with
 # a script setting SG_BASEPATH environment to the path you would prefer
-#DEFAULT_BASEDIR = /usr/share/games/smokinguns
+DEFAULT_BASEDIR = /usr/share/games/smokinguns
+USE_INTERNAL_ZLIB=0
+USE_INTERNAL_JPEG=0
+USE_INTERNAL_SPEEX=0
+USE_LOCAL_HEADERS=0
 
 # Use -DFS_FOPENI if you are MODing under Unix (content out of PK3)
 #STANDALONE_CFLAGS += -DFS_FOPENI
