diff -Nur orig/Makefile.am mod/Makefile.am
--- orig/Makefile.am	2022-01-16 15:23:16.000000000 +0100
+++ mod/Makefile.am	2022-01-17 17:42:30.076781873 +0100
@@ -10,9 +10,8 @@
 readmedir = @readmedir@
 
 if HAVE_MARKDOWN
-readme_DATA = README.md COPYING ChangeLog TODO README.lirc AUTHORS README.html ChangeLog.html
-else
-readme_DATA = README.md COPYING ChangeLog TODO README.lirc AUTHORS ChangeLog.html
+readme_DATA = README.md COPYING ChangeLog TODO README.lirc AUTHORS
+readme_DATA = README.md COPYING ChangeLog TODO README.lirc AUTHORS
 endif
 
 desktopdir = $(datadir)/applications
@@ -49,8 +48,4 @@
 CLEANFILES = $(desktop_DATA) ChangeLog.html $(appdata_DATA)
 
 .PHONY: ChangeLog
-ChangeLog.html:
-	./gen_changelog.sh
 
-README.html: README.md
-	./gen_readme.sh
