Description: Avoid qreal/double type clashes on ARM
Author: Colin Watson <cjwatson@ubuntu.com>
Acked-By: Mattia Rizzolo <mattia@mapreri.org>
Bug-Debian: https://bugs.debian.org/707882
Forwarded: no
Last-Update: 2013-05-16

--- a/scribus/scribus.h
+++ b/scribus/scribus.h
@@ -466,7 +466,7 @@
 	/** Fragt nach den Farben */
 //	void GetBrushPen();
 	/** Erzeugt einen Rahmen */
-	void MakeFrame(int f, int c, double *vals);
+	void MakeFrame(int f, int c, qreal *vals);
 	//** Loescht ein Element */
 	//void DeleteObjekt();
 	/** Duplicate current item */
--- a/scribus/pageitem.cpp
+++ b/scribus/pageitem.cpp
@@ -4667,7 +4667,7 @@
 
 
 //Moved from View
-void PageItem::SetFrameShape(int count, double *vals)
+void PageItem::SetFrameShape(int count, qreal *vals)
 {
 	PoLine.resize(0);
 	for (int a = 0; a < count-3; a += 4)
@@ -4690,7 +4690,7 @@
 
 void PageItem::SetRectFrame()
 {
-	static double rect[] = {  0.0,   0.0,   0.0,   0.0,
+	static qreal rect[] = {  0.0,   0.0,   0.0,   0.0,
 							100.0,   0.0, 100.0,   0.0,
 							100.0,   0.0, 100.0,   0.0,
 							100.0, 100.0, 100.0, 100.0,
@@ -4705,7 +4705,7 @@
 
 void PageItem::SetOvalFrame()
 {
-	static double rect[] = {   100.0,  50.0, 100.0,       77.615235,
+	static qreal rect[] = {   100.0,  50.0, 100.0,       77.615235,
 								50.0, 100.0,  77.615235, 100.0,
 								50.0, 100.0,  22.385765, 100.0,
 								 0.0,  50.0,   0.0,       77.615235,
--- a/scribus/pageitem.h
+++ b/scribus/pageitem.h
@@ -298,7 +298,7 @@
 	void DrawObj_PathText(ScPainter *p, double sc);
 public:
 	void DrawObj_Embedded(ScPainter *p, QRectF e, const CharStyle& style, PageItem* cembedded);
-	void SetFrameShape(int count, double *vals);
+	void SetFrameShape(int count, qreal *vals);
 	void SetRectFrame();
 	void SetOvalFrame();
 	void SetFrameRound();
--- a/scribus/propertiespalette.cpp
+++ b/scribus/propertiespalette.cpp
@@ -4625,7 +4625,7 @@
 			doc->setRedrawBounding(CurItem);
 			break;
 		default:
-			CurItem->SetFrameShape(c, (double*)vals);
+			CurItem->SetFrameShape(c, vals);
 			doc->setRedrawBounding(CurItem);
 			CurItem->FrameType = f+2;
 			break;
--- a/scribus/scribusdoc.h
+++ b/scribus/scribusdoc.h
@@ -1011,7 +1011,7 @@
 	DocumentInformation documentInfo;
 	int appMode;
 	int SubMode;
-	double *ShapeValues;
+	qreal *ShapeValues;
 	int ValCount;
 	QString DocName;
 	QMap<QString,int> UsedFonts;
--- a/scribus/canvasmode_create.cpp
+++ b/scribus/canvasmode_create.cpp
@@ -122,7 +122,7 @@
 		{
 			FPointArray poly;
 			int valCount = m_doc->ValCount;
-			double *vals = m_doc->ShapeValues;
+			qreal *vals = m_doc->ShapeValues;
 			for (int a = 0; a < valCount-3; a += 4)
 			{
 				if (vals[a] < 0)
--- a/scribus/scribus.cpp
+++ b/scribus/scribus.cpp
@@ -7625,7 +7625,7 @@
 }
 */
 //CB-->??
-void ScribusMainWindow::MakeFrame(int f, int c, double *vals)
+void ScribusMainWindow::MakeFrame(int f, int c, qreal *vals)
 {
 	PageItem *currItem = doc->m_Selection->itemAt(0);
 	switch (f)
--- a/scribus/plugins/imposition/imposition.cpp
+++ b/scribus/plugins/imposition/imposition.cpp
@@ -542,7 +542,7 @@
 					}
 					groupcontrol->SetRectFrame();
 					
-					double points[32] = {
+					qreal points[32] = {
 					//left upper corner - left lower corner
 					targetDoc->Pages->at(0)->guides.vertical(1,targetDoc->Pages->at(0)->guides.Standard),
 					targetDoc->Pages->at(0)->guides.horizontal(0,targetDoc->Pages->at(i)->guides.Standard),
@@ -730,7 +730,7 @@
 					}
 					groupcontrol->SetRectFrame();
 					
-					double points[32] = {
+					qreal points[32] = {
 					//left upper corner - left lower corner
 					0,
 					0,
@@ -893,7 +893,7 @@
 					}
 					groupcontrol->SetRectFrame();
 					
-					double points[32] = {
+					qreal points[32] = {
 					//left upper corner - left lower corner
 					0,
 					0,
@@ -1047,7 +1047,7 @@
 					}
 					groupcontrol->SetRectFrame();
 					
-					double points[32] = {
+					qreal points[32] = {
 					//left upper corner - left lower corner
 					targetDoc->Pages->at(0)->guides.vertical(1,targetDoc->Pages->at(0)->guides.Standard),
 					targetDoc->Pages->at(0)->guides.horizontal(0,targetDoc->Pages->at(i)->guides.Standard),
