diff -Nur orig/scribus/plugins/scriptplugin/scripts/Align_image_in_frame.py mod/scribus/plugins/scriptplugin/scripts/Align_image_in_frame.py
--- orig/scribus/plugins/scriptplugin/scripts/Align_image_in_frame.py	2019-07-31 00:34:55.000000000 +0200
+++ mod/scribus/plugins/scriptplugin/scripts/Align_image_in_frame.py	2020-04-05 11:24:58.441629282 +0200
@@ -1,4 +1,3 @@
-#!/usr/bin/python
 # Align_image_in_frame.py
 # This version 2014.04.19
 """
diff -Nur orig/scribus/plugins/scriptplugin/scripts/Autoquote2.py mod/scribus/plugins/scriptplugin/scripts/Autoquote2.py
--- orig/scribus/plugins/scriptplugin/scripts/Autoquote2.py	2019-07-31 00:34:55.000000000 +0200
+++ mod/scribus/plugins/scriptplugin/scripts/Autoquote2.py	2020-04-05 11:24:58.441629282 +0200
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # -*- coding: utf-8 -*-
 # File: spacenquotes.py - changes typewriter quotes to typographic quotes with option to add spaces
 # © 2010.08.28 Gregory Pittman
diff -Nur orig/scribus/plugins/scriptplugin/scripts/Autoquote.py mod/scribus/plugins/scriptplugin/scripts/Autoquote.py
--- orig/scribus/plugins/scriptplugin/scripts/Autoquote.py	2019-07-31 00:34:55.000000000 +0200
+++ mod/scribus/plugins/scriptplugin/scripts/Autoquote.py	2020-04-05 11:24:58.441629282 +0200
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # -*- coding: utf-8 -*-
 # File: quotes.py - changes typewriter quotes to typographic quotes
 # © 2010.08.28 Gregory Pittman
diff -Nur orig/scribus/plugins/scriptplugin/scripts/Caption.py mod/scribus/plugins/scriptplugin/scripts/Caption.py
--- orig/scribus/plugins/scriptplugin/scripts/Caption.py	2019-07-31 00:34:55.000000000 +0200
+++ mod/scribus/plugins/scriptplugin/scripts/Caption.py	2020-04-05 11:24:58.445629276 +0200
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # -*- coding: utf-8  -*-
 
 # ****************************************************************************
diff -Nur orig/scribus/plugins/scriptplugin/scripts/color2csv.py mod/scribus/plugins/scriptplugin/scripts/color2csv.py
--- orig/scribus/plugins/scriptplugin/scripts/color2csv.py	2019-07-31 00:34:55.000000000 +0200
+++ mod/scribus/plugins/scriptplugin/scripts/color2csv.py	2020-04-05 11:24:58.445629276 +0200
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # -*- coding: utf-8 -*-
 """
 ABOUT THIS SCRIPT:
diff -Nur orig/scribus/plugins/scriptplugin/scripts/ColorChart.py mod/scribus/plugins/scriptplugin/scripts/ColorChart.py
--- orig/scribus/plugins/scriptplugin/scripts/ColorChart.py	2019-07-31 00:34:55.000000000 +0200
+++ mod/scribus/plugins/scriptplugin/scripts/ColorChart.py	2020-04-05 11:24:58.449629270 +0200
@@ -1,4 +1,3 @@
-#!/usr/bin/python
 # -*- coding: utf-8 -*-
 
 """
diff -Nur orig/scribus/plugins/scriptplugin/scripts/csv2color.py mod/scribus/plugins/scriptplugin/scripts/csv2color.py
--- orig/scribus/plugins/scriptplugin/scripts/csv2color.py	2019-07-31 00:34:55.000000000 +0200
+++ mod/scribus/plugins/scriptplugin/scripts/csv2color.py	2020-04-05 11:24:58.449629270 +0200
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # -*- coding: utf-8 -*-
 """
 ABOUT THIS SCRIPT:
diff -Nur orig/scribus/plugins/scriptplugin/scripts/DirectImageImport.py mod/scribus/plugins/scriptplugin/scripts/DirectImageImport.py
--- orig/scribus/plugins/scriptplugin/scripts/DirectImageImport.py	2019-07-31 00:34:55.000000000 +0200
+++ mod/scribus/plugins/scriptplugin/scripts/DirectImageImport.py	2020-04-05 11:24:58.449629270 +0200
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # -*- coding: utf-8 -*-
 
 """ 
diff -Nur orig/scribus/plugins/scriptplugin/scripts/FontSample.py mod/scribus/plugins/scriptplugin/scripts/FontSample.py
--- orig/scribus/plugins/scriptplugin/scripts/FontSample.py	2019-07-31 00:34:55.000000000 +0200
+++ mod/scribus/plugins/scriptplugin/scripts/FontSample.py	2020-04-05 11:24:58.457629258 +0200
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # -*- coding: utf-8 -*-
 
 # ****************************************************************************
diff -Nur orig/scribus/plugins/scriptplugin/scripts/importcsv2table.py mod/scribus/plugins/scriptplugin/scripts/importcsv2table.py
--- orig/scribus/plugins/scriptplugin/scripts/importcsv2table.py	2019-07-31 00:34:55.000000000 +0200
+++ mod/scribus/plugins/scriptplugin/scripts/importcsv2table.py	2020-04-05 11:24:58.457629258 +0200
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # -*- coding: utf-8 -*-
 """
 ABOUT THIS SCRIPT:
diff -Nur orig/scribus/plugins/scriptplugin/scripts/InfoBox.py mod/scribus/plugins/scriptplugin/scripts/InfoBox.py
--- orig/scribus/plugins/scriptplugin/scripts/InfoBox.py	2019-07-31 00:34:55.000000000 +0200
+++ mod/scribus/plugins/scriptplugin/scripts/InfoBox.py	2020-04-05 11:24:58.457629258 +0200
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # -*- coding: utf-8  -*-
 
 # ****************************************************************************
diff -Nur orig/scribus/plugins/scriptplugin/scripts/Ligatursatz.py mod/scribus/plugins/scriptplugin/scripts/Ligatursatz.py
--- orig/scribus/plugins/scriptplugin/scripts/Ligatursatz.py	2019-07-31 00:34:55.000000000 +0200
+++ mod/scribus/plugins/scriptplugin/scripts/Ligatursatz.py	2020-04-05 11:24:58.533629145 +0200
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # -*- coding: utf8 -*-
 
 """
