diff -Naur orig/scribus/plugins/import/CMakeLists.txt patched/scribus/plugins/import/CMakeLists.txt
--- orig/scribus/plugins/import/CMakeLists.txt	2017-09-24 21:49:59.000000000 +0200
+++ patched/scribus/plugins/import/CMakeLists.txt	2019-05-31 12:45:03.746156605 +0200
@@ -12,8 +12,6 @@
 add_subdirectory(pct)
 if(HAVE_POPPLER)
 	add_subdirectory(pdf)
-else()
-	add_subdirectory(ps)
 endif()
 # add_subdirectory(pgs)
 # add_subdirectory(ppp)
