diff -Naur a/config/lxqt-config-powermanagement.desktop.in b/config/lxqt-config-powermanagement.desktop.in
--- a/config/lxqt-config-powermanagement.desktop.in	2022-04-16 15:03:33.000000000 +0200
+++ b/config/lxqt-config-powermanagement.desktop.in	2022-06-16 01:44:32.748316447 +0200
@@ -2,7 +2,7 @@
 Type=Application
 Exec=lxqt-config-powermanagement
 Icon=preferences-system-power-management
-Categories=Settings;DesktopSettings;Qt;LXQt;
+Categories=Settings;DesktopSettings;Qt;
 OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
