diff -Naur orig/src/gtk3-ui/sync.desktop.in patched/src/gtk3-ui/sync.desktop.in
--- orig/src/gtk3-ui/sync.desktop.in	2014-01-22 11:22:52.000000000 +0100
+++ patched/src/gtk3-ui/sync.desktop.in	2016-02-20 13:22:23.558275289 +0100
@@ -1,5 +1,5 @@
 [Desktop Entry]
-_Name=Sync
+_Name=SyncEvolution
 _Comment=Up to date
 Version=1.0
 Type=Application
diff -Naur orig/src/gtk-ui/sync.desktop.in patched/src/gtk-ui/sync.desktop.in
--- orig/src/gtk-ui/sync.desktop.in	2014-01-22 11:22:52.000000000 +0100
+++ patched/src/gtk-ui/sync.desktop.in	2016-02-20 13:21:57.182144497 +0100
@@ -1,5 +1,5 @@
 [Desktop Entry]
-_Name=Sync
+Name=SyncEvolution
 _Comment=Up to date
 Version=1.0
 Type=Application
