diff -Naur orig/src/synthesis/configure patched/src/synthesis/configure
--- orig/src/synthesis/configure	2016-11-08 19:28:38.000000000 +0100
+++ patched/src/synthesis/configure	2017-12-29 17:26:07.517816291 +0100
@@ -16055,7 +16055,7 @@
         { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
 
-$as_echo "#define USE_ICALTZUTIL_SET_EXACT_VTIMEZONES_SUPPORT 1" >>confdefs.h
+$as_echo "#undef USE_ICALTZUTIL_SET_EXACT_VTIMEZONES_SUPPORT" >>confdefs.h
 
 fi
 fi
