--- vinagre.orig/data/vinagre.desktop.in.in	2008-02-15 19:43:08.000000000 +0100
+++ vinagre/data/vinagre.desktop.in.in	2009-03-21 14:38:25.000000000 +0100
@@ -1,5 +1,6 @@
 [Desktop Entry]
-_Name=Remote Desktop Viewer
+_Name=Vinagre
+_GenericName=Remote Desktop Viewer
 _Comment=Access remote desktops
 Exec=vinagre %U
 Icon=vinagre
@@ -7,7 +9,7 @@
 MimeType=application/x-remote-connection;
 Type=Application
 StartupNotify=true
-Categories=GNOME;GTK;Network;RemoteAccess;
+Categories=GTK;GNOME;Network;RemoteAccess;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=vinagre
 X-GNOME-Bugzilla-Component=general
