diff -Nur orig/resources/net.sourceforge.extremetuxracer.desktop mod/resources/net.sourceforge.extremetuxracer.desktop
--- orig/resources/net.sourceforge.extremetuxracer.desktop	2020-03-07 10:14:40.000000000 +0100
+++ mod/resources/net.sourceforge.extremetuxracer.desktop	2020-04-07 18:33:33.465426410 +0200
@@ -8,6 +8,7 @@
 Comment=Race a penguin down a mountain
 Comment[de]=Rutsche mit einem Pinguin einen Hügel hinab
 Comment[ro_RO]=Curse cu un pinguin pe munte, la vale
+Comment[hu]=Nyílt forrású, lecsúszós verseny Tux, a pingvin főszereplésével
 Icon=etr
 
 Exec=etr
