diff -Naur orig/resources/etr.desktop patched/resources/etr.desktop
--- orig/resources/etr.desktop	2016-02-04 11:20:12.000000000 +0100
+++ patched/resources/etr.desktop	2016-02-19 00:05:56.486393694 +0100
@@ -7,6 +7,7 @@
 GenericName[ro_RO]=Joc de arcadă
 Comment=Race a penguin down a mountain
 Comment[de]=Rutsche mit einem Pinguin einen Hügel hinab
+Comment[hu]=Nyílt forrású, lecsúszós verseny Tux, a pingvin főszereplésével
 Comment[ro_RO]=Curse cu un pinguin pe munte, la vale
 Icon=etr
 
