diff -Nur orig/configure.ac mod/configure.ac
--- orig/configure.ac	2021-06-25 01:48:58.000000000 +0200
+++ mod/configure.ac	2021-07-09 17:15:46.841434166 +0200
@@ -70,26 +70,19 @@
 ],
 [AC_MSG_ERROR([sfml-graphics not found])])
 
-AC_CHECK_LIB([GLU], [gluBeginCurve],
+AC_CHECK_LIB([GL], [glBegin],
 [
-  AC_CHECK_LIB([OpenGL], [glBegin],
+  AC_CHECK_LIB([GLU], [gluBeginCurve],
   [
-    LIBS="$LIBS -lOpenGL -lGLU"
+    LIBS="$LIBS -lGL -lGLU"
     glfound=true
-  ],
-  [
-    AC_CHECK_LIB([GL], [glBegin],
-    [
-      LIBS="$LIBS -lGL -lGLU"
-      glfound=true
-    ])
   ])
 ])
 
 if test x$glfound != xtrue ; then
-  AC_CHECK_LIB([glu32], [gluBeginCurve],
+  AC_CHECK_LIB([opengl32], [glBegin],
   [
-    AC_CHECK_LIB([opengl32], [glBegin],
+    AC_CHECK_LIB([glu32], [gluBeginCurve],
     [
       LIBS="$LIBS -lopengl32 -lglu32"
       glfound=true
