diff -Naur orig/src/welcomescreen.cpp patched/src/welcomescreen.cpp
--- orig/src/welcomescreen.cpp	2020-07-27 08:59:02.000000000 +0200
+++ patched/src/welcomescreen.cpp	2020-07-30 17:20:08.637730218 +0200
@@ -27,13 +27,16 @@
 
 #include "colorscheme.h"
 #include "customcontrols.h"
-#include "crowdin_gui.h"
 #include "edapp.h"
 #include "edframe.h"
 #include "hidpi.h"
 #include "str_helpers.h"
 #include "utility.h"
 
+#ifdef HAVE_HTTP_CLIENT
+    #include "crowdin_gui.h"
+#endif
+
 #include <wx/dcbuffer.h>
 #include <wx/statbmp.h>
 #include <wx/stattext.h>
