diff -Nur orig/configure.ac mod/configure.ac
--- orig/configure.ac	2018-10-04 11:11:08.000000000 +0200
+++ mod/configure.ac	2019-10-05 11:16:48.586666237 +0200
@@ -72,7 +72,7 @@
 dnl GNOME support
 dnl
 AC_ARG_WITH(gnome, AS_HELP_STRING([--without-gnome], [Build NetworkManager-pptp without GNOME support, e.g. vpn service only]), [], [with_gnome_specified=no])
-AC_ARG_WITH(libnm-glib, AS_HELP_STRING([--without-libnm-glib], [Build NetworkManager-pptp without libnm-glib comatibility]), [], [with_libnm_glib_specified=no])
+AC_ARG_WITH(libnm-glib, AS_HELP_STRING([--with-libnm-glib], [Build NetworkManager-pptp with libnm-glib comatibility (depreacted)]), [], [with_libnm_glib_specified=no])
 if test "$with_libnm_glib_specified" != no -a "$with_libnm_glib" != no; then
 	if test "$with_gnome_specified" != no -a "$with_gnome" == no; then
 		AC_MSG_ERROR(Building --with-libnm-glib conflicts with --without-gnome)
@@ -82,10 +82,10 @@
 	with_gnome=yes
 fi
 if test "$with_libnm_glib_specified" == no; then
-	with_libnm_glib="$with_gnome"
+	with_libnm_glib=no
 fi
-if test "$with_libnm_glib" != no; then
-	with_libnm_glib=yes
+if test "$with_libnm_glib" != yes; then
+	with_libnm_glib=no
 fi
 AM_CONDITIONAL(WITH_GNOME, test "$with_gnome" != no)
 AM_CONDITIONAL(WITH_LIBNM_GLIB, test "$with_libnm_glib" != no)
