diff -Naur orig/lib/dpkg/dpkg.h patched/lib/dpkg/dpkg.h
--- orig/lib/dpkg/dpkg.h	2019-01-23 22:36:34.000000000 +0100
+++ patched/lib/dpkg/dpkg.h	2019-03-19 01:52:54.277666291 +0100
@@ -52,7 +52,7 @@
 #define MAXCONFFILENAME     1000
 #define MAXDIVERTFILENAME   1024
 #define MAXCONTROLFILENAME  100
-#define DEBEXT             ".deb"
+#define DEBEXT             ".ubk.uhu"
 #define REMOVECONFFEXTS    "~", ".bak", "%", \
                            DPKGTEMPEXT, DPKGNEWEXT, DPKGOLDEXT, DPKGDISTEXT
 
diff -Naur orig/po/hu.po patched/po/hu.po
--- orig/po/hu.po	2019-02-23 17:59:53.000000000 +0100
+++ patched/po/hu.po	2019-03-19 01:52:54.281666341 +0100
@@ -2227,8 +2227,8 @@
 msgstr "--%s --recursive legalább 1 ösvény argumentumot vár"
 
 #: src/archives.c
-msgid "searched, but found no packages (files matching *.deb)"
-msgstr "kerestem, de nem leltem csomagokat (*.deb fájlokat)"
+msgid "searched, but found no packages (files matching *.uhu)"
+msgstr "kerestem, de nem leltem csomagokat (*.uhu fájlokat)"
 
 #: src/archives.c
 #, c-format
diff -Naur orig/src/archives.c patched/src/archives.c
--- orig/src/archives.c	2019-02-23 15:26:43.000000000 +0100
+++ patched/src/archives.c	2019-03-19 01:52:54.281666341 +0100
@@ -1497,7 +1497,7 @@
 
         /* Check if it looks like a .deb file. */
         nodename = treenode_get_pathname(node);
-        if (strcmp(nodename + strlen(nodename) - 4, ".deb") != 0)
+        if (strcmp(nodename + strlen(nodename) - 4, ".uhu") != 0)
           continue;
 
         arglist = m_realloc(arglist, sizeof(char *) * (nfiles + 2));
@@ -1508,7 +1508,7 @@
     }
 
     if (!nfiles)
-      ohshit(_("searched, but found no packages (files matching *.deb)"));
+      ohshit(_("searched, but found no packages (files matching *.uhu)"));
 
     arglist[nfiles] = NULL;
     argp= arglist;
