diff -Naur a/lib/dpkg/dpkg.h b/lib/dpkg/dpkg.h
--- a/lib/dpkg/dpkg.h	2023-11-20 23:27:34.051239558 +0100
+++ b/lib/dpkg/dpkg.h	2023-11-20 23:32:39.583933886 +0100
@@ -71,6 +71,7 @@
 #define POSTINSTFILE       "postinst"
 #define PRERMFILE          "prerm"
 #define POSTRMFILE         "postrm"
+#define POSTRMCUSTOMFILE   "postrm-custom"
 /* Debconf config maintainer script. */
 #define MAINTSCRIPT_FILE_CONFIG		"config"
 #define TRIGGERSCIFILE     "triggers"
diff -Naur a/src/main/remove.c b/src/main/remove.c
--- a/src/main/remove.c	2023-10-30 04:22:04.000000000 +0100
+++ b/src/main/remove.c	2023-11-20 23:32:39.587933892 +0100
@@ -225,7 +225,8 @@
 {
   /* We need the postrm and list files for --purge. */
   if (strcmp(filetype, LISTFILE) == 0 ||
-      strcmp(filetype, POSTRMFILE) == 0)
+      strcmp(filetype, POSTRMFILE) == 0 ||
+      strcmp(filetype, POSTRMCUSTOMFILE) == 0)
     return;
 
   debug(dbg_stupidlyverbose, "removal_bulk info not postrm or list");
@@ -670,6 +671,12 @@
           filename);
     if (unlink(filename) && errno != ENOENT)
       ohshite(_("can't remove old postrm script"));
+      
+    filename = pkg_infodb_get_file(pkg, &pkg->installed, POSTRMCUSTOMFILE);
+    debug(dbg_general, "removal_bulk purge done, removing postrm-custom '%s'",
+          filename);
+    if (unlink(filename) && errno != ENOENT)
+      ohshite(_("can't remove old postrm-custom script"));
 
     pkg_set_status(pkg, PKG_STAT_NOTINSTALLED);
     pkg_set_want(pkg, PKG_WANT_UNKNOWN);
