diff -Naur a/src/deb/build.c b/src/deb/build.c
--- a/src/deb/build.c	2022-03-26 18:17:59.000000000 +0100
+++ b/src/deb/build.c	2022-05-15 05:07:57.709320278 +0200
@@ -593,7 +593,6 @@
     pkg = check_control_area(ctrldir, dir);
     if (debar == NULL)
       debar = gen_dest_pathname_from_pkg(dest, pkg);
-    info(_("building package '%s' in '%s'."), pkg->set->name, debar);
   }
   m_output(stdout, _("<standard output>"));
 
