diff -Naur orig/q4wine.desktop patched/q4wine.desktop
--- orig/q4wine.desktop	2013-11-10 13:35:28.000000000 +0100
+++ patched/q4wine.desktop	2016-02-20 13:07:18.309786404 +0100
@@ -5,6 +5,7 @@
 Comment[de]=Qt4-Werkzeug zur Verwaltung der Wine-Anwendungen und -Präfixe.
 Comment[en]=Qt4 utility for Wine applications and prefixes management.
 Comment[es]=Qt4 utilidad para el manejo de aplicaciones y prefijos de wine.
+Comment[hu]=Wine beállító eszköz
 Comment[it]=Qt4 programma per gestire le applicazioni con Wine e i prefissi.
 Comment[ru]=Qt4-программа для настройки приложений и управления префиксами Wine.
 Comment[uk]=Qt4 програма для налаштування програм та керування префіксами Wine.
