diff -Naur q4wine-1.3.7-orig/q4wine.desktop q4wine-1.3.7/q4wine.desktop
--- q4wine-1.3.7-orig/q4wine.desktop	2018-04-26 23:21:32.000000000 +0200
+++ q4wine-1.3.7/q4wine.desktop	2018-07-19 16:06:58.135305037 +0200
@@ -1,12 +1,7 @@
 [Desktop Entry]
 Categories=Qt;System;Emulator;
 Comment=Utility for Wine applications and prefixes management
-Comment[cs]=Užitečný program pro správu Wine aplikací a prefixů
-Comment[de]=Werkzeug zur Verwaltung der Wine-Anwendungen und -Präfixe
-Comment[es]=Utilidad para el manejo de aplicaciones y prefijos de wine
-Comment[it]=Programma per gestire le applicazioni con Wine e i prefissi
-Comment[ru]=Программа для настройки приложений и управления префиксами Wine
-Comment[uk]=Програма для налаштування програм та керування префіксами Wine
+Comment[hu]=Wine beállító eszköz
 Exec=q4wine -b %u
 Icon=q4wine
 Name=Q4Wine
