diff -Naur a/Makefile.am b/Makefile.am
--- a/Makefile.am	2017-08-17 02:38:43.000000000 +0200
+++ b/Makefile.am	2022-04-06 15:17:03.954741481 +0200
@@ -27,7 +27,6 @@
              include \
              tools \
              clients \
-             doc \
-             demos
+             doc
 AUTOMAKE_OPTIONS = 1.4
 ACLOCAL_AMFLAGS = -I .
diff -Naur a/Makefile.in b/Makefile.in
--- a/Makefile.in	2017-12-05 13:49:57.000000000 +0100
+++ b/Makefile.in	2022-04-06 15:17:18.370790675 +0200
@@ -390,8 +390,7 @@
              include \
              tools \
              clients \
-             doc \
-             demos
+             doc
 
 AUTOMAKE_OPTIONS = 1.4
 ACLOCAL_AMFLAGS = -I .
