diff -Naur orig/Makefile.in patched/Makefile.in
--- orig/Makefile.in	2018-07-22 19:08:39.000000000 +0200
+++ patched/Makefile.in	2018-08-17 04:59:26.907617992 +0200
@@ -21,6 +21,11 @@
 # Various configurable paths (remember to edit Makefile.in, not Makefile)
 #
 
+MAJOR=$(UB_VERSION_MAJOR)
+MINOR=$(UB_VERSION_MINOR)
+SUBMINOR=$(UB_VERSION_MICRO)
+VERSION = $(MAJOR).$(MINOR).$(SUBMINOR)
+
 # Top level hierarchy
 prefix = @prefix@
 exec_prefix = @exec_prefix@
@@ -154,7 +159,7 @@
 
 CLEANFILES = $(OBJ) libpcap.a libpcap.so.`cat $(srcdir)/VERSION` \
 	$(PROG)-`cat $(srcdir)/VERSION`.tar.gz $(GENSRC) $(GENHDR) \
-	lex.yy.c pcap-config libpcap.pc
+	lex.yy.c pcap-config libpcap.so.$(VERSION) libpcap.pc
 
 MAN1 = pcap-config.1
 
@@ -655,6 +660,8 @@
 	[ -d $(DESTDIR)$(libdir) ] || \
 	    (mkdir -p $(DESTDIR)$(libdir); chmod 755 $(DESTDIR)$(libdir))
 	$(INSTALL_DATA) libpcap.a $(DESTDIR)$(libdir)/libpcap.a
+	$(INSTALL_DATA) libpcap.so.$(VERSION)  $(DESTDIR)$(libdir)/libpcap.so.$(VERSION)
+	ln -s libpcap.so.$(VERSION) $(DESTDIR)$(libdir)/libpcap.so.$(MAJOR).$(MINOR)
 	$(RANLIB) $(DESTDIR)$(libdir)/libpcap.a
 install-archive-shareda:
 	#
