From 01a13355cadbd3b6fb50292a9ac63b1bb1e418ac Mon Sep 17 00:00:00 2001
From: David Redondo <kde@david-redondo.de>
Date: Tue, 2 Mar 2021 12:04:22 +0100
Subject: [PATCH] Underscores are not allowed in action names

BUG:433109
---
 app/systemsettings.desktop | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/app/systemsettings.desktop b/app/systemsettings.desktop
index 31f0ef30..29973d00 100755
--- a/app/systemsettings.desktop
+++ b/app/systemsettings.desktop
@@ -6,7 +6,7 @@ X-DocPath=systemsettings/index.html
 X-KDE-StartupNotify=true
 X-KDE-Shortcuts=Tools
 OnlyShowIn=KDE;
-Actions=kcm_lookandfeel;kcm_users;screenlocker;powerdevilprofilesconfig;kcm_kscreen;
+Actions=kcm-lookandfeel;kcm-users;screenlocker;powerdevilprofilesconfig;kcm-kscreen;
 
 GenericName=System Settings
 GenericName[ar]=إعدادات النّظام
@@ -229,7 +229,7 @@ X-DBUS-StartupType=Unique
 X-DBUS-ServiceName=org.kde.systemsettings
 Categories=Qt;KDE;Settings;
 
-[Desktop Action kcm_lookandfeel]
+[Desktop Action kcm-lookandfeel]
 Name=Global Theme
 Name[ast]=Estilu global
 Name[az]=Qlobal Mövzu
@@ -265,7 +265,7 @@ Name[zh_CN]=全局主题
 Icon=preferences-desktop-theme-global
 Exec=systemsettings5 kcm_lookandfeel
 
-[Desktop Action kcm_users]
+[Desktop Action kcm-users]
 Name=Users
 Name[ast]=Usuarios
 Name[az]=İstifadəçilər
@@ -371,7 +371,7 @@ Name[zh_CN]=节能
 Icon=preferences-system-power-management
 Exec=systemsettings5 powerdevilprofilesconfig
 
-[Desktop Action kcm_kscreen]
+[Desktop Action kcm-kscreen]
 Name=Display Configuration
 Name[az]=Ekran Ayarları
 Name[ca]=Configuració de la pantalla
-- 
GitLab
