diff -Naur orig/setup.py mod/setup.py
--- orig/setup.py	2015-07-16 21:49:46.000000000 +0200
+++ mod/setup.py	2015-07-16 22:26:00.325792466 +0200
@@ -55,7 +55,7 @@
                     'images/none.jpg'
                 ]
             ),
-            (DATADIR + '/fluxstyle/glade', ['images/main.glade']),
+            (DATADIR + '/fluxstyle/glade', ['glade/main.glade']),
             (BINDIR, ['fluxstyle_gui']),
             (DATADIR + '/fluxstyle/docs',
              ['docs/README', 'docs/LICENSE', 'docs/Changelog'])
