diff -Naur orig/fluxstyle_gui mod/fluxstyle_gui
--- orig/fluxstyle_gui	2015-07-16 21:49:46.000000000 +0200
+++ mod/fluxstyle_gui	2015-07-17 17:27:17.756360229 +0200
@@ -26,10 +26,10 @@
 from fluxstyle import find_styles, parse_config
 from fluxstyle import error_message
 
-#GLADEFILE="/usr/share/fluxstyle/glade/main.glade"
-#N_IMAGE="/usr/share/fluxstyle/images/none.jpg"
-GLADEFILE = "./glade/main.glade"
-N_IMAGE = "./images/none.jpg"
+GLADEFILE="/usr/share/fluxstyle/glade/main.glade"
+N_IMAGE="/usr/share/fluxstyle/images/none.jpg"
+#GLADEFILE = "./glade/main.glade"
+#N_IMAGE = "./images/none.jpg"
 
 
 class StyleChange:
