diff -urN usb-modeswitch-data-20140529.orig/Makefile usb-modeswitch-data-20140529/Makefile
--- usb-modeswitch-data-20140529.orig/Makefile	2014-05-29 12:32:42.000000000 +0200
+++ usb-modeswitch-data-20140529/Makefile	2014-06-11 09:31:44.310215126 +0200
@@ -3,7 +3,7 @@
 RM          = /bin/rm -f
 PREFIX      = $(DESTDIR)/usr
 ETCDIR      = $(DESTDIR)/etc
-RULESDIR    = $(DESTDIR)/lib/udev/rules.d
+RULESDIR    = $(DESTDIR)/usr/lib/udev/rules.d
 
 
 .PHONY:     clean
