diff -Naur orig/build.sh mod/build.sh
--- orig/build.sh	2014-09-20 16:28:05.000000000 +0200
+++ mod/build.sh	2016-05-02 12:59:47.435594623 +0200
@@ -15,7 +15,7 @@
 # default - compiling DC only (using by default)
 
 # path to lazbuild
-export lazbuild=$(which lazbuild)
+export lazbuild="lazbuild --lazarusdir=/usr/lib/lazarus"
 
 # Set up widgetset: gtk or gtk2 or qt
 # Set up processor architecture: i386 or x86_64
