diff -Nur orig/install/linux/install.sh mod/install/linux/install.sh
--- orig/install/linux/install.sh	2016-09-18 13:46:42.000000000 +0200
+++ mod/install/linux/install.sh	2017-06-07 17:04:44.571344482 +0200
@@ -10,7 +10,7 @@
 # Determine library directory
 if [ "$CPU_TARGET" = "x86_64" ] && [ ! -f "/etc/debian_version" ]
    then
-       LIB_SUFFIX=64
+       LIB_SUFFIX=
    else
        LIB_SUFFIX=
 fi
@@ -107,9 +107,9 @@
     install -d -m 755                      $DC_INSTALL_PREFIX/usr/share/man/man1
     install -c -m 644 install/linux/*.1    $DC_INSTALL_PREFIX/usr/share/man/man1
     # Copy documentation
-    install -d                $DC_INSTALL_PREFIX/usr/share/doublecmd/doc
-    install -m 644 doc/*.txt  $DC_INSTALL_PREFIX/usr/share/doublecmd/doc
-    ln -sf ../../share/doublecmd/doc $DC_INSTALL_DIR/doc
+    # install -d                $DC_INSTALL_PREFIX/usr/share/doublecmd/doc
+    # install -m 644 doc/*.txt  $DC_INSTALL_PREFIX/usr/share/doublecmd/doc
+    # ln -sf ../../share/doublecmd/doc $DC_INSTALL_DIR/doc
     # Copy scripts
     install -d         $DC_INSTALL_DIR/scripts
     cp -a scripts/*.py $DC_INSTALL_DIR/scripts/
@@ -131,8 +131,8 @@
            $DC_INSTALL_PREFIX/usr/share/icons/hicolor/scalable/apps/doublecmd.svg
   else
     # Copy documentation
-    mkdir -p $DC_INSTALL_DIR/doc
-    cp -a doc/*.txt $DC_INSTALL_DIR/doc/
+    # mkdir -p $DC_INSTALL_DIR/doc
+    # cp -a doc/*.txt $DC_INSTALL_DIR/doc/
     # Copy script for execute portable version
     cp -a doublecmd.sh $DC_INSTALL_DIR/
     # Copy directories
