--- xsane.orig/src/xsane.desktop	2008-03-29 10:32:18.000000000 +0100
+++ xsane/src/xsane.desktop	2009-06-22 12:06:10.000000000 +0200
@@ -1,9 +1,10 @@
 [Desktop Entry]
-Encoding=UTF-8
-Name=XSane - Scanning
+Name=XSane
 Comment=Acquire images from a scanner
+Comment[hu]=Dokumentumok szkennelése
 Exec=xsane
 Icon=xsane
 Terminal=false
 Type=Application
-Categories=Application;Graphics
+StartupNotify=false
+Categories=GTK;Graphics;
