diff -Nur orig/build/resources/0ad.desktop mod/build/resources/0ad.desktop
--- orig/build/resources/0ad.desktop	2022-08-21 14:46:46.000000000 +0200
+++ mod/build/resources/0ad.desktop	2022-09-25 16:42:12.049769908 +0200
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Version=1.4
 Name=0 A.D.
 Exec=0ad %F
 Icon=0ad
@@ -13,6 +12,7 @@
 Comment[de]=Ein Echtzeitstrategiespiel, das die Kriegsführung der Antike behandelt
 Comment[es]=Un juego de estrategia en tiempo real de guerra antigua
 Comment[fr]=Un jeu de statégie militaire en temps réel dans l'Antiquité
+Comment[hu]=Történelmi stratégiai játék
 Comment[it]=Videogioco strategico in tempo reale di guerre antiche
 Comment[nl]=Een real-time strategie spel over oorlog van de oudheid
 Comment[pl]=Gra strategiczna czasu rzeczywistego  o wojnach starożytnych
