diff -Nur orig/build/resources/0ad.desktop mod/build/resources/0ad.desktop
--- orig/build/resources/0ad.desktop	2016-11-14 10:50:36.000000000 +0100
+++ mod/build/resources/0ad.desktop	2017-10-30 08:50:47.552172348 +0100
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Version=1.0
 Name=0 A.D.
 Exec=0ad
 Icon=0ad
@@ -10,6 +9,7 @@
 Comment[de]=Ein Echtzeitstrategiespiel, das die Kriegsführung der Antike behandelt
 Comment[es]=Un juego de estrategia en tiempo real de guerra antigua
 Comment[fr]=Un jeu de statégie militaire en temps réel dans l'Antiquité
+Comment[hu]=Történelmi stratégiai játék
 Comment[it]=Videogioco strategico in tempo reale di guerre antiche
 Comment[nl]=Een real-time strategie spel over oorlog van de oudheid
 Comment[pl]=Gra strategiczna czasu rzeczywistego  o wojnach starożytnych
