diff -Naurd a/Makefile b/Makefile
--- a/Makefile	2009-04-21 20:23:57.000000000 +0200
+++ b/Makefile	2011-05-05 18:36:03.000000000 +0200
@@ -41,24 +41,24 @@
 	rm -f $(OBJFILES) pdmenu $(MOS)
 
 install: all
-	$(INSTALL) -d $(INSTALL_PREFIX)/$(BINDIR) \
-		$(INSTALL_PREFIX)/$(MANDIR)/man1 \
-		$(INSTALL_PREFIX)/$(MANDIR)/man5 \
-		$(INSTALL_PREFIX)/$(SYSCONFDIR) \
-		$(INSTALL_PREFIX)/$(DATADIR)/pdmenu
-	$(INSTALL_PROGRAM) pdmenu $(INSTALL_PREFIX)/$(BINDIR)
-	$(INSTALL) doc/pdmenu.man $(INSTALL_PREFIX)/$(MANDIR)/man1/pdmenu.1 -m 0644
-	$(INSTALL) doc/pdmenurc.man $(INSTALL_PREFIX)/$(MANDIR)/man5/pdmenurc.5 -m 0644
-	if [ -e $(INSTALL_PREFIX)/$(SYSCONFDIR)/pdmenurc ]; then \
+	$(INSTALL) -d $(DESTDIR)/$(BINDIR) \
+		$(DESTDIR)/$(MANDIR)/man1 \
+		$(DESTDIR)/$(MANDIR)/man5 \
+		$(DESTDIR)/$(SYSCONFDIR) \
+		$(DESTDIR)/$(DATADIR)/pdmenu
+	$(INSTALL_PROGRAM) pdmenu $(DESTDIR)/$(BINDIR)
+	$(INSTALL) doc/pdmenu.man $(DESTDIR)/$(MANDIR)/man1/pdmenu.1 -m 0644
+	$(INSTALL) doc/pdmenurc.man $(DESTDIR)/$(MANDIR)/man5/pdmenurc.5 -m 0644
+	if [ -e $(DESTDIR)/$(SYSCONFDIR)/pdmenurc ]; then \
 		echo "Backing up $(INSTALL_PREFIX)/$(SYSCONFDIR)/pdmenurc to $(INSTALL_PREFIX)/$(SYSCONFDIR)/pdmenurc.old .." ; \
-		cp $(INSTALL_PREFIX)/$(SYSCONFDIR)/pdmenurc $(INSTALL_PREFIX)/$(SYSCONFDIR)/pdmenurc.old ; \
+		cp $(DESTDIR)/$(SYSCONFDIR)/pdmenurc $(DESTDIR)/$(SYSCONFDIR)/pdmenurc.old ; \
 	fi
-	$(INSTALL) examples/pdmenurc $(INSTALL_PREFIX)/$(SYSCONFDIR)/pdmenurc -m 0644
-	$(INSTALL) examples/showdir.pl $(INSTALL_PREFIX)/$(DATADIR)/pdmenu
-	cd $(INSTALL_PREFIX)/$(DATADIR)/pdmenu && $(LN_S) -f showdir.pl editdir.pl
+	$(INSTALL) examples/pdmenurc $(DESTDIR)/$(SYSCONFDIR)/pdmenurc -m 0644
+	$(INSTALL) examples/showdir.pl $(DESTDIR)/$(DATADIR)/pdmenu
+	cd $(DESTDIR)/$(DATADIR)/pdmenu && $(LN_S) -f showdir.pl editdir.pl
 	for lang in $(LANGS); do \
-		[ ! -d $(LOCALEDIR)/$$lang/LC_MESSAGES/ ] && mkdir -p $(LOCALEDIR)/$$lang/LC_MESSAGES/; \
-		install -m 644 po/$$lang.mo $(LOCALEDIR)/$$lang/LC_MESSAGES/pdmenu.mo; \
+		[ ! -d $(DESTDIR)/$(LOCALEDIR)/$$lang/LC_MESSAGES/ ] && mkdir -p $(DESTDIR)/$(LOCALEDIR)/$$lang/LC_MESSAGES/; \
+		install -m 644 po/$$lang.mo $(DESTDIR)/$(LOCALEDIR)/$$lang/LC_MESSAGES/pdmenu.mo; \
 	done
 
 test: pdmenu
