diff -Nur orig/pyproject.toml mod/pyproject.toml
--- orig/pyproject.toml	2025-10-15 19:11:51.000000000 +0200
+++ mod/pyproject.toml	2025-11-12 15:12:50.003211692 +0100
@@ -2,7 +2,7 @@
 build-backend = "backend"
 requires = [
   "pybind11",
-  "setuptools>=77",
+  "setuptools>=73.0.1",
 ]
 backend-path = [
   "_custom_build",
@@ -15,8 +15,6 @@
 keywords = [
   "Imaging",
 ]
-license = "MIT-CMU"
-license-files = [ "LICENSE" ]
 authors = [
   { name = "Jeffrey A. Clark", email = "aclark@aclark.net" },
 ]
