diff -Nur orig/pyproject.toml mod/pyproject.toml
--- orig/pyproject.toml	2025-04-12 19:02:40.000000000 +0200
+++ mod/pyproject.toml	2025-04-29 13:46:26.994838581 +0200
@@ -1,7 +1,7 @@
 [build-system]
 build-backend = "backend"
 requires = [
-  "setuptools>=77",
+  "setuptools>=73.0.1",
 ]
 backend-path = [
   "_custom_build",
@@ -14,8 +14,6 @@
 keywords = [
   "Imaging",
 ]
-license = "MIT-CMU"
-license-files = [ "LICENSE" ]
 authors = [
   { name = "Jeffrey A. Clark", email = "aclark@aclark.net" },
 ]
