diff -Naur orig/cellwriter.desktop patched/cellwriter.desktop
--- orig/cellwriter.desktop	2012-11-12 07:52:30.000000000 +0100
+++ patched/cellwriter.desktop	2016-02-18 23:45:41.076366801 +0100
@@ -5,6 +5,7 @@
 GenericName=Handwriting input panel
 GenericName[tr]=El Yazısı Giriş Paneli
 Comment=Grid-entry handwriting recognition input panel
+Comment[hu]=Kézírás beviteli panel
 Comment[tr]=Izgara tabanlı el yazısı giriş paneli
 Exec=cellwriter
 Icon=cellwriter
