diff -Nur orig/configure mod/configure
--- orig/configure	2019-09-09 21:54:58.000000000 +0200
+++ mod/configure	2020-02-06 12:56:22.950050355 +0100
@@ -1610,7 +1610,7 @@
                           Config database for installing schema files.
   --with-gconf-schema-file-dir=dir
                           Directory for installing schema files.
-  --with-guile=2.2|2.0|auto
+  --with-guile=3.0|2.2|2.0|auto
                           Which guile version to use (default: auto)
   --with-help-method      which help method to use (ghelp|file|library;
                           default: ghelp)
@@ -18995,7 +18995,7 @@
 # Guile
 # *****
 
-guile_versions_all="2.2 2.0"
+guile_versions_all="3.0 2.2 2.0"
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking which guile version to use" >&5
 $as_echo_n "checking which guile version to use... " >&6; }
 
@@ -19003,7 +19003,7 @@
 if test "${with_guile+set}" = set; then :
   withval=$with_guile; case "$with_guile" in
      1.8) as_fn_error $? "unsupported guile version $with_guile requested" "$LINENO" 5 ;;
-     2.0|2.2) guile_versions="$withval" ;;
+     3.0|2.0|2.2) guile_versions="$withval" ;;
      auto) guile_versions="$guile_versions_all" ;;
      *) as_fn_error $? "unsupported guile version $with_guile requested" "$LINENO" 5 ;;
     esac
