diff -Naur orig/build_files/cmake/Modules/FindOpenColorIO.cmake patched/build_files/cmake/Modules/FindOpenColorIO.cmake
--- orig/build_files/cmake/Modules/FindOpenColorIO.cmake	2017-09-11 06:34:59.000000000 +0200
+++ patched/build_files/cmake/Modules/FindOpenColorIO.cmake	2019-01-25 01:13:51.005740591 +0100
@@ -30,7 +30,6 @@
 SET(_opencolorio_FIND_COMPONENTS
   OpenColorIO
   yaml-cpp
-  tinyxml
 )
 
 SET(_opencolorio_SEARCH_DIRS
