diff -Nur orig/install.sh mod/install.sh
--- orig/install.sh	2020-07-10 11:57:32.000000000 +0200
+++ mod/install.sh	2020-07-10 12:04:29.664400718 +0200
@@ -300,74 +300,6 @@
 HERE=`pwd`
 
 OSNAME=`uname -s | tr "[:upper:]" "[:lower:]"`
-case "$OSNAME" in
-  haiku)
-     # Install in /boot/common or /boot/home/config ?
-     if checkpath /boot/common/bin; then
-         PREFIX=/boot/common
-     else
-         PREFIX=/boot/home/config
-     fi
-     # If we can't write on prefix, we are probably 
-     # on Haiku with package management system.
-     # In this case, we have to install fpc in the non-packaged subdir
-     if [ ! -w "$PREFIX" ]; then
-     	PREFIX="$PREFIX/non-packaged"
-     fi
-  ;;
-  freebsd)
-     PREFIX=/usr/local
-  ;;
-  sunos)
-     # Check if GNU llinker is recent enough, version 2.21 is needed at least
-     GNU_LD=`which gld`
-     supported_emulations=`"$GNU_LD" --target-help | sed -n "s|^\(elf.*\):|\1|p" `
-     supports_elf_i386_sol2=`echo $supported_emulations | grep -w elf_i386_sol2 `
-     supports_elf_x86_64_sol2=`echo $supported_emulations | grep -w elf_x86_64_sol2 `
-     if [ "$supports_elf_i386_sol2" = "" ]; then
-       echo -n "GNU linker $GNU_LD does not support elf_i386_sol2 emulation, please consider "
-       echo "upgrading binutils package to at least version 2.21"
-     elif [ "$supports_elf_x86_64_sol2" = "" ]; then
-       echo -n "GNU linker $GNU_LD does not support elf_x86_64_sol2 emulation, please consider "
-       echo "upgrading binutils package to at least version 2.21"
-     fi
-     PREFIX=/usr/local
-  ;;
-  aix)
-     # Install in /usr/local or /usr ?
-     if checkpath /usr/local/bin; then
-         PREFIX=/usr/local
-     else
-         PREFIX=/usr
-     fi
-  ;;
-  *)
-     # Install in /usr/local or /usr ?
-     if checkpath /usr/local/bin; then
-         PREFIX=/usr/local
-     else
-         PREFIX=/usr
-     fi
-  ;;
-esac
-
-# If we can't write on prefix, select subdir of home dir
-if [ ! -w "$PREFIX" ]; then
-  PREFIX="$HOME/fpc-$VERSION"
-fi
-
-case "$OSNAME" in
-  haiku)
-     ask "Install prefix (/boot/common or /boot/home/config) " PREFIX
-  ;;
-  *)
-     ask "Install prefix (/usr or /usr/local) " PREFIX
-  ;;
-esac
-
-# Support ~ expansion
-PREFIX=`eval echo $PREFIX`
-export PREFIX
 makedirhierarch "$PREFIX"
 
 # Set some defaults.
@@ -485,7 +417,7 @@
 
 # Install /etc/fpc.cfg, this is done using the samplecfg script
 if [ "$cross" = "" ]; then
-  "$LIBDIR/samplecfg" "$LIBDIR" | tee samplecfg.log
+  "$LIBDIR/samplecfg" "$LIBDIR"  "$UB_INSTALLDIR/etc" | tee samplecfg.log
   file_list=`sed -n 's:.*Writing sample configuration file to ::p' samplecfg.log`
   if [ ! -z "$file_list" ] ; then
     for file in $file_list ; do
