diff -Naur orig/src/mlib/tech.h patched/src/mlib/tech.h
--- orig/src/mlib/tech.h	2016-04-10 12:45:39.000000000 +0200
+++ patched/src/mlib/tech.h	2020-09-05 01:34:26.502273993 +0200
@@ -26,7 +26,6 @@
 //  Технические вещи
 // 
 #include <boost/current_function.hpp>  // для BOOST_CURRENT_FUNCTION
-#include <boost/detail/endian.hpp>     // для BOOST_XXX_ENDIAN
 #include <boost/version.hpp>           // для BOOST_MINOR_VERSION
 
 // для С-шного кода в С++
@@ -78,9 +77,9 @@
                 long line, const char* function);
 
 // endianness
-#if defined(BOOST_BIG_ENDIAN)
+#if __BYTE_ORDER__ != __ORDER_LITTLE_ENDIAN__
 #   define HAS_BIG_ENDIAN
-#elif defined(BOOST_LITTLE_ENDIAN)
+#elif __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
 #   define HAS_LITTLE_ENDIAN
 #else
 #   error mlib/tech.h: unknown endianness (legacy PDP arch?)
