diff -Naur orig/configure mod/configure
--- orig/configure	2013-06-30 11:58:48.000000000 +0200
+++ mod/configure	2013-08-16 14:39:15.000000000 +0200
@@ -5122,7 +5122,7 @@
 # Don't just call it CFLAGS, since it then gets inherited into intl/, and GNU
 # and I don't exactly code alike. Don't include CFLAGS as set by AC_PROG_CC
 # above, since it includes -g -O2 for gcc, for reasons unknown.
-GENTOO_CFLAGS="-Wall -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Werror-implicit-function-declaration"
+GENTOO_CFLAGS="-Wall -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes "
 
 # Check whether --enable-debug was given.
 if test "${enable_debug+set}" = set; then :
