diff -Naur orig/src/tracker-store/tracker-dbus.vala patched/src/tracker-store/tracker-dbus.vala
--- orig/src/tracker-store/tracker-dbus.vala	1970-01-01 01:00:00.000000000 +0100
+++ patched/src/tracker-store/tracker-dbus.vala	2020-09-30 01:33:34.275327868 +0200
@@ -122,7 +122,7 @@
 		return true;
 	}
 
-	static void name_owner_changed_cb (DBusConnection connection, string sender_name, string object_path, string interface_name, string signal_name, Variant parameters) {
+	static void name_owner_changed_cb (DBusConnection connection, string? sender_name, string object_path, string interface_name, string signal_name, Variant parameters) {
 
 		unowned string name, old_owner, new_owner;
 		parameters.get ("(&s&s&s)", out name, out old_owner, out new_owner);
