diff -Naur a/Makefile.in b/Makefile.in
--- a/Makefile.in	2012-01-19 07:30:05.000000000 +0100
+++ b/Makefile.in	2023-08-12 09:38:29.299561645 +0200
@@ -72,10 +72,10 @@
 include $(depend)
 
 $(objects)/$(TARGET): $(OBJECTS)
-	$(LIBTOOL) --mode=link $(CC) -o $@ $^ $(LDFLAGS) $(EXTRA_LDFLAGS) $(LT_LDFLAGS)
+	$(LIBTOOL) --mode=link $(CC) --tag=CXX -o $@ $^ $(LDFLAGS) $(EXTRA_LDFLAGS) $(LT_LDFLAGS)
 
 $(objects)/$(SDLMAIN_TARGET): $(SDLMAIN_OBJECTS)
-	$(LIBTOOL) --mode=link $(CC) -o $@ $^ $(LDFLAGS) $(EXTRA_LDFLAGS) $(LT_LDFLAGS) $(SDLMAIN_LDFLAGS)
+	$(LIBTOOL) --mode=link $(CC) --tag=CXX -o $@ $^ $(LDFLAGS) $(EXTRA_LDFLAGS) $(LT_LDFLAGS) $(SDLMAIN_LDFLAGS)
 
 
 install: all install-bin install-hdrs install-lib install-data install-man
