diff -Nur orig/backend/rdp/output.c mod/backend/rdp/output.c
--- orig/backend/rdp/output.c	2019-10-07 21:28:20.000000000 +0200
+++ mod/backend/rdp/output.c	2019-11-02 12:02:55.875146601 +0100
@@ -90,10 +90,10 @@
 	cmd.destTop = damage->extents.y1;
 	cmd.destRight = damage->extents.x2;
 	cmd.destBottom = damage->extents.y2;
-	cmd.bmp.bpp = pixman_image_get_depth(output->shadow_surface);
-	cmd.bmp.codecID = peer->settings->RemoteFxCodecId;
-	cmd.bmp.width = width;
-	cmd.bmp.height = height;
+	cmd.bpp.bpp = pixman_image_get_depth(output->shadow_surface);
+	cmd.bpp.codecID = peer->settings->RemoteFxCodecId;
+	cmd.bpp.width = width;
+	cmd.bpp.height = height;
 
 	uint32_t *ptr = pixman_image_get_data(output->shadow_surface) +
 		damage->extents.x1 + damage->extents.y1 *
@@ -122,8 +122,8 @@
 	rfx_compose_message(context->rfx_context, context->encode_stream,
 			context->rfx_rects, nrects, (BYTE *)ptr, width, height,
 			pixman_image_get_stride(output->shadow_surface));
-	cmd.bmp.bitmapDataLength = Stream_GetPosition(context->encode_stream);
-	cmd.bmp.bitmapData = Stream_Buffer(context->encode_stream);
+	cmd.bpp.bitmapDataLength = Stream_GetPosition(context->encode_stream);
+	cmd.bpp.bitmapData = Stream_Buffer(context->encode_stream);
 
 	update->SurfaceBits(update->context, &cmd);
 	return true;
@@ -146,10 +146,10 @@
 	cmd.destTop = damage->extents.y1;
 	cmd.destRight = damage->extents.x2;
 	cmd.destBottom = damage->extents.y2;
-	cmd.bmp.bpp = pixman_image_get_depth(output->shadow_surface);
-	cmd.bmp.codecID = peer->settings->NSCodecId;
-	cmd.bmp.width = width;
-	cmd.bmp.height = height;
+	cmd.bpp.bpp = pixman_image_get_depth(output->shadow_surface);
+	cmd.bpp.codecID = peer->settings->NSCodecId;
+	cmd.bpp.width = width;
+	cmd.bpp.height = height;
 
 	uint32_t *ptr = pixman_image_get_data(output->shadow_surface) +
 		damage->extents.x1 + damage->extents.y1 *
@@ -159,8 +159,8 @@
 			(BYTE *)ptr, width, height,
 			pixman_image_get_stride(output->shadow_surface));
 
-	cmd.bmp.bitmapDataLength = Stream_GetPosition(context->encode_stream);
-	cmd.bmp.bitmapData = Stream_Buffer(context->encode_stream);
+	cmd.bpp.bitmapDataLength = Stream_GetPosition(context->encode_stream);
+	cmd.bpp.bitmapData = Stream_Buffer(context->encode_stream);
 
 	update->SurfaceBits(update->context, &cmd);
 	return true;
