diff -Naur orig/data/tuxcmd.desktop patched/data/tuxcmd.desktop
--- orig/data/tuxcmd.desktop	2009-11-15 17:00:35.134739579 +0100
+++ patched/data/tuxcmd.desktop	2017-12-02 03:20:49.977328153 +0100
@@ -1,9 +1,10 @@
 [Desktop Entry]
 Name=Tux Commander
 Comment=A two panel file manager
+Comment[hu]=Kétpaneles fájlkezelő
 Exec=tuxcmd
 Icon=tuxcmd
 Terminal=false
 Type=Application
-Categories=GTK;Utility;
+Categories=GTK;Utility;System;FileManager;FileTools;
 StartupNotify=true
