From 8fa67f1c5140f0a7e85a99f5b3a292012ab59a0a Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Mon, 14 May 2018 02:44:52 +0200
Subject: ub_compile: automatikus autoreconf


diff --git a/src/helpers/ub_compile b/src/helpers/ub_compile
index 7b722b8..f5f6aa5 100755
--- a/src/helpers/ub_compile
+++ b/src/helpers/ub_compile
@@ -32,12 +32,17 @@ if [ -f CMakeLists.txt ]; then
 	cmake \
 		"$@" "${argv[@]}" \
 		..
-elif [ -f "configure" -a ! -f meson.build ]; then
+fi
+
+if [ -f "configure" -a ! -f meson.build ]; then
 	ub_configure "$@"
 else
 	if [ -f "autogen.sh" -a ! -f meson.build ]; then
 		UB_CONFIGURE_SCRIPT="./autogen.sh" ub_configure "$@"
 	fi
+	if [ ! -f "configure" -a ! -f "autogen.sh" -a ! -f meson.build ]; then
+		autoreconf -vif && ub_configure "$@"
+	fi
 fi
 
 if [ -f setup.py -a ! -f Makefile ]; then
