From cfdabbad16c37909262de5f75277d8ffbefacd29 Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Sat, 5 Mar 2022 01:31:05 +0100
Subject: add qt6 support


diff --git a/src/helpers/ub_fixpaths b/src/helpers/ub_fixpaths
index c9aadd9..26308f9 100755
--- a/src/helpers/ub_fixpaths
+++ b/src/helpers/ub_fixpaths
@@ -13,3 +13,12 @@ if [ -d "$UB_INSTALLDIR/usr/lib/qt5" ]; then
 	-exec sed -i -e '/^QMAKE_PRL_BUILD_DIR/d' \
 				-e "s@^${UB_COMPILEDIR}@@g" -e "s@^${UB_INSTALLDIR}@@g" {} \;
 fi
+
+if [ -d "$UB_INSTALLDIR/usr/lib/q65" ]; then
+	find $UB_INSTALLDIR/usr/lib/qt6/ -name \*.prl \
+	-exec sed -i -e '/^QMAKE_PRL_BUILD_DIR/d' \
+				-e "s@^${UB_COMPILEDIR}@@g" -e "s@^${UB_INSTALLDIR}@@g" {} \;
+	find $UB_INSTALLDIR/usr/lib/qt6/ -name \*.pri \
+	-exec sed -i -e '/^QMAKE_PRL_BUILD_DIR/d' \
+				-e "s@^${UB_COMPILEDIR}@@g" -e "s@^${UB_INSTALLDIR}@@g" {} \;
+fi
diff --git a/src/steps/depends b/src/steps/depends
index af34702..7e78103 100755
--- a/src/steps/depends
+++ b/src/steps/depends
@@ -386,6 +386,34 @@ for pkg in $UB_PACKAGES; do
 					fi
 				done
 			;;
+			# cmake fájl analízise: qt6
+			./usr/lib/qt6/lib/cmake/*/*Config.cmake)
+			    found=0
+				for cmakedep in `grep ^find_dependency $file | sed 's/.*find_dependency[(]//g' | sed 's/[)]//g' | sed 's/\s.*//g'`; do
+					cmakefile="lib/cmake/${cmakedep}/${cmakedep}Config.cmake"               
+					if [ -f "/usr/${cmakefile}" ]; then
+						echo "/usr/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
+					    found=1
+					fi
+					if [ -f "/usr/lib/qt6/${cmakefile}" ]; then
+						echo "/usr/lib/qt6/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
+						found=1
+					fi
+					for p in $UB_PACKAGES; do
+					    if [ -f "$UB_PACKAGEDIR/$p/usr/${cmakefile}" ]; then
+						    echo "/usr/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
+						    found=1
+						fi
+					    if [ -f "$UB_PACKAGEDIR/$p/usr/lib/qt6/${cmakefile}" ]; then
+						    echo "/usr/lib/qt6/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
+						    found=1
+						fi
+					done
+					if [ $found = 0 ]; then
+						error "/usr/${cmakefile} or /usr/lib/qt6/${cmakefile} is required by $file but not found"
+					fi
+				done
+			;;
 		esac
 
 	done
diff --git a/src/uhubuild-commonenvironment b/src/uhubuild-commonenvironment
index 43bc634..9789503 100644
--- a/src/uhubuild-commonenvironment
+++ b/src/uhubuild-commonenvironment
@@ -19,8 +19,8 @@ for i in /etc/env.d/*.env; do
 done
 unset i
 
-# extra path: qt5
-export PATH="/usr/lib/uhubuild${UHUBUILD_CCACHE_DIR:+:/usr/lib/ccache}:/sbin:/usr/sbin:/bin:/usr/bin:/usr/lib/qt5/bin"
+# extra path: qt5,qt6
+export PATH="/usr/lib/uhubuild${UHUBUILD_CCACHE_DIR:+:/usr/lib/ccache}:/sbin:/usr/sbin:/bin:/usr/bin:/usr/lib/qt5/bin:/usr/lib/qt6/bin"
 
 export LC_ALL=hu_HU.UTF-8
 export HOME=~
