From 427e2cdf02d2256552fb2d5aa917060157e9b416 Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Tue, 3 Sep 2024 01:36:35 +0200
Subject: ub_compile: meson opciók javítás
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit


diff --git a/src/helpers/ub_compile b/src/helpers/ub_compile
index ce43d24..6160299 100755
--- a/src/helpers/ub_compile
+++ b/src/helpers/ub_compile
@@ -13,9 +13,9 @@ if [ -f meson.build ]; then
 		--mandir="/usr/share/man"   \
 		--infodir="/usr/share/info" \
 		--backend=ninja		        \
-		--buildtype=release         \
 		--optimization=2            \
 		--default-library=shared	\
+		-Ddebug=false               \
 	)
 	for i in "${argx[@]}"; do
 		argv[${#argv[@]}]="$i"
