From d71266bdc8a8238b613f3e5c14d613005863a815 Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Tue, 9 Apr 2019 23:54:11 +0200
Subject: =?UTF-8?q?uhubuild-check:=20config=20mapp=C3=A1t=20is=20m=C3=A1so?=
 =?UTF-8?q?ljuk?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit


diff --git a/src/uhubuild-check b/src/uhubuild-check
index 1f8244c..d388c52 100755
--- a/src/uhubuild-check
+++ b/src/uhubuild-check
@@ -66,7 +66,7 @@ if [ -z "${UHUBUILD_CHECK_TOPONLY:-}" ]; then
 		*/packages/*)
 			cd ../..
 			;;
-		*/addons*|*/packages|*/patches|*/sources)
+		*/addons*|*/config|*/packages|*/patches|*/sources)
 			cd ..
 			;;
 		*)
@@ -269,6 +269,14 @@ for dir in addons*; do
 	fi
 done
 
+##############
+### config ###
+##############
+if [ -d config ]; then
+	mkdir -p "$UHUBUILD_SRC2DIR/config"
+	cp -a config "$UHUBUILD_SRC2DIR/"
+fi
+
 ###############
 ### patches ###
 ###############
