From 6c56ccd825b6a65877035bab34259805e8a14c1e Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Fri, 14 Oct 2022 03:00:35 +0200
Subject: steps/man egyszerűsítés
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit


diff --git a/src/steps/man b/src/steps/man
index da8f254..8931dd7 100755
--- a/src/steps/man
+++ b/src/steps/man
@@ -78,66 +78,6 @@ for pkg in $UB_PACKAGES; do
 		# kibontjuk a tomoritett man oldalakat
 		find "$dir" -type f -name '*.gz'  -print0 | xargs -r0 gunzip  -f --
 		find "$dir" -type f -name '*.bz2' -print0 | xargs -r0 bunzip2 -f --
-		# a symlinkeket manlinkre konvertaljuk
-		for link in $(find "$dir" -type l); do
-			pointsto="$(readlink "$link")"
-			pretendroot rm "$link"
-			echo ".so ${pointsto%.gz}" >"${link%.gz}"
-		done
-		# helyrepofozzuk a hibas manlinkeket hogy az nroffnak is jo legyen
-		for file in $(find "$dir" -type f); do
-			if [ "$(wc -l <"$file")" -le 1 ]; then
-				ub_replace "^\.so[[:space:]]\+man\(.\)/\(.\+\1\)" ".so \2" "$file"
-				ub_replace "^\.so[[:space:]]\+\(man./.\+\)" ".so ../\1" "$file"
-			fi
-		done
-		# megnezzuk nincs-e disznosag a /usr/share/man alatt
-		for d2 in "$dir"/*; do
-			if [ -f "$d2" ]; then
-				error "$d2 közönséges fájl nem lesz jó!"
-			fi
-			case "${d2##*/}" in
-			man?)
-				sect="${d2##*/man}"
-				for f3 in "$d2"/*; do
-					if [ -d "$f3" ]; then
-						error "$f3 könyvtár nem lesz jó!"
-					fi
-					case "$f3" in
-					  *.${sect}*)	;;
-					  *)		error "$f3: kiterjesztés nem oké!";;
-					esac
-				done
-				;;
-			man)
-				error "$d2 könyvtár nem lesz jó!"
-				;;
-			*)
-				for d3 in "$d2"/*; do
-					if [ -f "$d3" ]; then
-						error "$d3 közönséges fájl nem lesz jó!"
-					fi
-					case "${d3##*/}" in
-					man?)
-						sect="${d3##*/man}"
-						for f4 in "$d3"/*; do
-							if [ -d "$f4" ]; then
-								error "$f4 könyvtár nem lesz jó!"
-							fi
-							case "$f4" in
-							  *.${sect}*)	;;
-							  *)		error "$f4: kiterjesztés nem oké!";;
-							esac
-						done
-						;;
-					*)
-						error "$d3 könyvtár nem lesz jó!"
-						;;
-					esac
-				done
-				;;
-			esac
-		done
 		# átalakítás UTF-8-ra
 		for d in "$dir"/*; do
 			if [ -d "$d" ]; then
