From 03882447db384ba8a0995d785b2d368ee031a5dd Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Fri, 16 Dec 2022 16:38:06 +0100
Subject: uhubuild-check: sections update


diff --git a/src/uhubuild-check b/src/uhubuild-check
index ce0d6a9..7e414b9 100755
--- a/src/uhubuild-check
+++ b/src/uhubuild-check
@@ -817,9 +817,11 @@ for pkg in packages/*; do
 				System/Tools | \
 				System/X | \
 				UHU | \
-				CINNAMON | \
+				BUDGIE | \
+				ENLIGHTENMENT | \
 				GNOME | \
 				KDE | \
+				LUMINA | \
 				LXDE | \
 				LXQT | \
 				MATE | \
