From b5bb98c524fe27146ce3847b57100b6b23e19060 Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Thu, 15 Sep 2022 18:32:56 +0200
Subject: config/macros javítás
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit


diff --git a/config/macros b/config/macros
index 0a0d252..6e1b626 100644
--- a/config/macros
+++ b/config/macros
@@ -1,2 +1,2 @@
-%_ autoconf autoconf-archive automake bash bc binutils bison bzip2 bzip2-dev ccache cmake coreutils cracklib desktop-file-utils diffutils dl dpkg file findutils flex flex-dev gawk gcc gettext gettext-tools glibc glibc-dev grep gzip inetutils kernel-headers libstdc++ libstdc++-dev libtool licenses m4 make meson ninja optipng pam pam-dev patch pkgconfig pretendroot sed shadow tar uhu-base uhubuild-here uhu-pkg uhu-utils which xz-dev zlib-dev zstd-dev
+%_ autoconf autoconf-archive automake bash bc binutils bison bzip2 bzip2-dev ccache cmake coreutils cracklib desktop-file-utils diffutils dpkg file findutils flex flex-dev gawk gcc gettext gettext-tools glibc glibc-dev grep gzip inetutils kernel-headers libstdc++ libstdc++-dev libtool licenses m4 make meson ninja optipng pam pam-dev patch pkgconfig pretendroot sed shadow tar uhu-base uhubuild-here uhu-pkg uhu-utils which xz-dev zlib-dev zstd-dev
 
