From 35e2fbdc4e8109406cbe37451d94803a4c04fc31 Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Sat, 16 Jul 2022 13:36:28 +0200
Subject: ub_compile, ub_configure javítás
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit


diff --git a/src/helpers/ub_compile b/src/helpers/ub_compile
index 5cf66c7..8eebea0 100755
--- a/src/helpers/ub_compile
+++ b/src/helpers/ub_compile
@@ -4,9 +4,7 @@
 # pl. webkit2gtk nem találja a pthread.h-t
 export DISABLE_CCACHE=1
 
-if [ "$UB_DISTRIBUTION_NUM" -ge "4" ]; then
-    export LDFLAGS='-Wl,--as-needed'
-fi
+export LDFLAGS='-Wl,--as-needed'
 
 if [ -f meson.build ]; then
 	argv=()
diff --git a/src/helpers/ub_configure b/src/helpers/ub_configure
index 05da76d..7ed986b 100755
--- a/src/helpers/ub_configure
+++ b/src/helpers/ub_configure
@@ -2,7 +2,7 @@
 
 . /usr/lib/uhubuild/common
 
-export UB_HOST="x86_64-linux-gnu"
+export UB_HOST="x86_64-unknown-linux-gnu"
 
 UB_CFLAGS="-O2 -m64 -mtune=generic ${CFLAGS:-}"
 
