From c24951fbcb63827d279ebddeebb81daa2faa3e9e Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Sat, 21 Jan 2023 15:05:56 +0100
Subject: ub_compile: LDFLAGS törlés
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit


diff --git a/src/helpers/ub_compile b/src/helpers/ub_compile
index 2254be6..0be050c 100755
--- a/src/helpers/ub_compile
+++ b/src/helpers/ub_compile
@@ -4,8 +4,6 @@
 # pl. webkit2gtk nem találja a pthread.h-t
 export DISABLE_CCACHE=1
 
-export LDFLAGS='-Wl,--as-needed -Wl,-dynamic-linker,/usr/lib/ld-linux-x86-64.so.2'
-
 if [ -f meson.build ]; then
     builddir="build"
 	argv=()
