From 78622dfe6508ce25da04e8a43fa6c445d7c665c1 Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Sun, 27 Dec 2020 23:02:08 +0100
Subject: macros módosítás
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit


diff --git a/config/macros b/config/macros
index 8274173..0a0d252 100644
--- a/config/macros
+++ b/config/macros
@@ -1,2 +1,2 @@
-%_ autoconf autoconf-archive automake bash bc binutils bison bzip2 bzip2-dev ccache cmake coreutils cracklib desktop-file-utils diffutils dl dpkg file findutils flex flex-dev gawk gcc gettext gettext-tools glibc glibc-dev grep gzip inetutils kernel-headers libstdc++ libstdc++-dev libtool licenses m4 make meson ninja optipng pam pam-dev patch perl pkgconfig pretendroot sed shadow tar texinfo uhu-base uhubuild-here uhu-pkg uhu-utils which xz-dev zlib zlib-dev
+%_ autoconf autoconf-archive automake bash bc binutils bison bzip2 bzip2-dev ccache cmake coreutils cracklib desktop-file-utils diffutils dl dpkg file findutils flex flex-dev gawk gcc gettext gettext-tools glibc glibc-dev grep gzip inetutils kernel-headers libstdc++ libstdc++-dev libtool licenses m4 make meson ninja optipng pam pam-dev patch pkgconfig pretendroot sed shadow tar uhu-base uhubuild-here uhu-pkg uhu-utils which xz-dev zlib-dev zstd-dev
 
