From e05258c53a2b2b72b77491fab84a391a2783f56f Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Tue, 11 Feb 2020 21:27:06 +0100
Subject: =?UTF-8?q?ub=5Fmake:=20cmake=20+=20ninja=20jav=C3=ADt=C3=A1s?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit


diff --git a/src/helpers/ub_make b/src/helpers/ub_make
index af75349..3d68e17 100755
--- a/src/helpers/ub_make
+++ b/src/helpers/ub_make
@@ -10,7 +10,7 @@ if [ -f meson.build -a -f build/build.ninja ]; then
 		echo -e "\t$i"
 	done
 	ninja -C build "$j" "$@"
-elif [ -f ${UB_COMPILEDIR}/CMakeLists.txt -a -f ${UB_COMPILEDIR}/build/build.ninja ]; then
+elif [ -f ../CMakeLists.txt -a -f ../build/build.ninja ]; then
 	echo "ninja $j"
 	for i in "$@"; do
 		echo -e "\t$i"
