From a1bffc8c22dbdae5cac79ea71c9bd9103dde1102 Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Tue, 24 Aug 2021 00:06:41 +0200
Subject: uhubuild-check: check uid, gid range


diff --git a/src/uhubuild-check b/src/uhubuild-check
index 12e74bb..37b7ce2 100755
--- a/src/uhubuild-check
+++ b/src/uhubuild-check
@@ -861,6 +861,9 @@ for pkg in packages/*; do
 			if [ "$name" = "" -o "$uid" = "" -o "$gid" = "" ]; then
 				error "Hiányos $pkg/users fájl!"
 			fi
+			if [ "$uid" -ge "1000" -o "$gid" -ge "1000" ]; then
+				error "$pkg/users: csak rendszer felhasználó / csoport adható meg!"
+			fi
 			printf "%-8s %4s %4s  %s\n" "$name" "$uid" "$gid" "$rest"
 		done < $UHUBUILD_SRC2DIR/$pkg/users.tmp2 > $UHUBUILD_SRC2DIR/$pkg/users.tmp
 		reformat 0 < $UHUBUILD_SRC2DIR/$pkg/users.tmp > $UHUBUILD_SRC2DIR/$pkg/users
@@ -873,6 +876,9 @@ for pkg in packages/*; do
 			if [ "$name" = "" -o "$gid" = "" ]; then
 				error "Hiányos $pkg/groups fájl!"
 			fi
+			if [ "$gid" -ge "1000" ]; then
+				error "$pkg/users: csak rendszer csoport adható meg!"
+			fi
 			printf "%-8s %4s  %s\n" "$name" "$gid" "$rest"
 		done < $UHUBUILD_SRC2DIR/$pkg/groups.tmp2 > $UHUBUILD_SRC2DIR/$pkg/groups.tmp
 		reformat 0 < $UHUBUILD_SRC2DIR/$pkg/groups.tmp > $UHUBUILD_SRC2DIR/$pkg/groups
