diff -Naur orig/config/macros patched/config/macros
--- orig/config/macros	2019-02-07 01:59:08.000000000 +0100
+++ patched/config/macros	2019-02-07 02:06:58.307029083 +0100
@@ -1,2 +1,2 @@
-%_ autoconf autoconf-archive automake bash bc binutils bison bzip2 bzip2-dev ccache cmake coreutils cracklib desktop-file-utils diffutils dl dpkg file findutils flex flex-dev gawk gcc gettext gettext-tools glibc glibc-dev grep gzip inetutils kernel-headers libstdc++ libstdc++-dev libtool licenses m4 make optipng pam pam-dev patch perl pkgconfig pretendroot sed shadow tar texinfo uhu-base uhubuild-here uhu-pkg uhu-utils which xz-dev zlib zlib-dev
+%_ autoconf autoconf-archive automake bash bc binutils bison bzip2 bzip2-dev ccache cmake coreutils cracklib desktop-file-utils diffutils dl dpkg file findutils flex flex-dev gawk gcc gettext gettext-tools glibc glibc-dev grep gzip inetutils kernel-headers libstdc++ libstdc++-dev libtool licenses m4 make optipng pam pam-dev patch perl pkgconfig pretendroot sed shadow tar texinfo uhu-base uhubuild-here uhu-pkg uhu-utils wayland-dev which xz-dev zlib zlib-dev
 
