diff -Naur orig/config/macros patched/config/macros
--- orig/config/macros	2018-04-14 01:06:40.000000000 +0200
+++ patched/config/macros	2018-04-14 01:40:13.684631772 +0200
@@ -1,2 +1,2 @@
-%_ autoconf autoconf-archive automake bash bc binutils bison bzip2 bzip2-dev ccache cmake coreutils cracklib desktop-file-utils diffutils dl dpkg file findutils flex flex-dev gawk gcc gettext gettext-tools glibc glibc-dev grep gzip kernel-headers libstdc++ libstdc++-dev libtool licenses m4 make ncurses ncurses-dev optipng pam pam-dev patch perl pkgconfig pretendroot readline readline-dev sed shadow tar texinfo uhu-base uhubuild-here uhu-pkg uhu-utils which xz-dev zlib zlib-dev
+%_ autoconf autoconf-archive automake bash bc binutils bison bzip2 bzip2-dev ccache cmake coreutils cracklib desktop-file-utils diffutils dl dpkg file findutils flex flex-dev gawk gcc gettext gettext-tools glibc glibc-dev grep gzip kernel-headers libstdc++ libstdc++-dev libtool licenses m4 make ncurses ncurses-dev optipng pam pam-dev patch perl pkgconfig pretendroot readline readline-dev sed shadow tar texinfo uhu-base uhubuild-here uhu-pkg uhu-utils wayland-dev which xz-dev zlib zlib-dev
 
