diff -Naur Linux-PAM-0.99.8.1.orig/modules/pam_access/access.conf Linux-PAM-0.99.8.1/modules/pam_access/access.conf
--- Linux-PAM-0.99.8.1.orig/modules/pam_access/access.conf	2007-06-22 09:48:28.000000000 +0200
+++ Linux-PAM-0.99.8.1/modules/pam_access/access.conf	2007-08-14 16:55:14.000000000 +0200
@@ -120,3 +120,5 @@
 #
 # All other users should be denied to get access from all sources.
 #- : ALL : ALL
+
++ : ALL : *
