diff -Naur Linux-PAM-0.99.3.0.orig/modules/pam_limits/limits.conf Linux-PAM-0.99.3.0/modules/pam_limits/limits.conf
--- Linux-PAM-0.99.3.0.orig/modules/pam_limits/limits.conf	2005-08-16 16:02:28.000000000 +0200
+++ Linux-PAM-0.99.3.0/modules/pam_limits/limits.conf	2006-02-07 13:59:35.000000000 +0100
@@ -50,7 +50,7 @@
 #<domain>      <type>  <item>         <value>
 #
 
-#*               soft    core            0
+*                soft    core            0
 #*               hard    rss             10000
 #@student        hard    nproc           20
 #@faculty        soft    nproc           20
