diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2020-11-24 23:56:49.000000000 +0100
+++ patched/configure.ac	2021-02-10 01:56:36.898714220 +0100
@@ -641,7 +641,7 @@
 	modules/Makefile \
 	modules/pam_access/Makefile \
         modules/pam_debug/Makefile modules/pam_deny/Makefile \
-	modules/pam_echo/Makefile modules/pam_env/Makefile \
+	modules/pam_echo/Makefile modules/pam_env/Makefile modules/pam_envfeed/Makefile \
 	modules/pam_faildelay/Makefile modules/pam_faillock/Makefile \
 	modules/pam_filter/Makefile modules/pam_filter/upperLOWER/Makefile \
 	modules/pam_ftp/Makefile modules/pam_group/Makefile \
