diff -urN Linux-PAM-1.1.8.orig/modules/pam_namespace/Makefile.am Linux-PAM-1.1.8/modules/pam_namespace/Makefile.am
--- Linux-PAM-1.1.8.orig/modules/pam_namespace/Makefile.am	2013-06-18 16:11:21.000000000 +0200
+++ Linux-PAM-1.1.8/modules/pam_namespace/Makefile.am	2014-02-27 07:59:31.912014289 +0100
@@ -40,7 +40,7 @@
   secureconf_SCRIPTS = namespace.init
 
 install-data-local:
-	mkdir -p $(DESTDIR)$(namespaceddir)
+	mkdir -p $(namespaceddir)
 endif
 
 
diff -urN Linux-PAM-1.1.8.orig/modules/pam_namespace/Makefile.in Linux-PAM-1.1.8/modules/pam_namespace/Makefile.in
--- Linux-PAM-1.1.8.orig/modules/pam_namespace/Makefile.in	2013-09-19 10:01:35.000000000 +0200
+++ Linux-PAM-1.1.8/modules/pam_namespace/Makefile.in	2014-02-27 07:56:10.927361451 +0100
@@ -869,7 +869,7 @@
 
 
 @HAVE_UNSHARE_TRUE@install-data-local:
-@HAVE_UNSHARE_TRUE@	mkdir -p $(DESTDIR)$(namespaceddir)
+@HAVE_UNSHARE_TRUE@	mkdir -p $(namespaceddir)
 @ENABLE_REGENERATE_MAN_TRUE@README: pam_namespace.8.xml namespace.conf.5.xml
 @ENABLE_REGENERATE_MAN_TRUE@-include $(top_srcdir)/Make.xml.rules
 
