diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2016-04-28 13:21:59.000000000 +0200
+++ patched/configure.ac	2016-08-21 03:06:14.416398656 +0200
@@ -636,6 +636,7 @@
 	modules/pam_unix/Makefile modules/pam_userdb/Makefile \
 	modules/pam_warn/Makefile modules/pam_wheel/Makefile \
 	modules/pam_xauth/Makefile doc/Makefile doc/specs/Makefile \
+	modules/pam_envfeed/Makefile modules/pam_locallogin/Makefile \
 	doc/man/Makefile doc/sag/Makefile doc/adg/Makefile \
 	doc/mwg/Makefile examples/Makefile tests/Makefile \
 	xtests/Makefile])
