From 8bde10c823d59e66d540c1f5b0fdde517b52af6a Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Thu, 13 Jun 2019 22:29:04 +0200
Subject: fix desktop file


diff --git a/clients/yate-qt5.desktop b/clients/yate-qt5.desktop
index 9297f8e..7f6873e 100644
--- a/clients/yate-qt5.desktop
+++ b/clients/yate-qt5.desktop
@@ -1,13 +1,11 @@
 [Desktop Entry]
-Encoding=UTF-8
 Name=Yate VoIP Client
 GenericName=VoIP Phone
 Comment=Place phone calls over the Internet
 Exec=yate-qt5
-Icon=null_team-48.png
+Icon=null_team-48
 TryExec=yate-qt5
 StartupNotify=true
 Terminal=false
 Type=Application
 Categories=Application;Network;Qt
-Name[hu_HU]=yate-qt5.desktop
