diff -Naur orig/clients/yate-qt4.desktop mod/clients/yate-qt4.desktop
--- orig/clients/yate-qt4.desktop	2010-12-20 10:30:44.000000000 +0100
+++ mod/clients/yate-qt4.desktop	2016-04-30 21:03:58.517517244 +0200
@@ -1,13 +1,15 @@
 [Desktop Entry]
-Encoding=UTF-8
 Name=Yate VoIP Client
 Name[ro]=Yate client de VoIP
+Name[hu]=VoIP kliens
 GenericName=VoIP Phone
 GenericName[ro]=Telefon VoIP
+GenericName[hu]=VoIP telefon
 Comment=Place phone calls over the Internet
 Comment[ro]=Apeluri telefonice peste Internet
+Comment[hu]=Internetes telefon
 Exec=yate-qt4
-Icon=null_team-48.png
+Icon=null_team-48
 TryExec=yate-qt4
 StartupNotify=true
 Terminal=false
