diff -Nur orig/gnome-panel/gnome-panel.desktop.in mod/gnome-panel/gnome-panel.desktop.in
--- orig/gnome-panel/gnome-panel.desktop.in	2017-09-30 13:13:46.000000000 +0200
+++ mod/gnome-panel/gnome-panel.desktop.in	2018-01-02 12:23:57.912224389 +0100
@@ -7,7 +7,7 @@
 Exec=gnome-panel
 Terminal=false
 Categories=GNOME;GTK;System;Core;
-OnlyShowIn=GNOME-Flashback;
+OnlyShowIn=GNOME;
 NoDisplay=true
 X-GNOME-AutoRestart=false
 X-GNOME-Autostart-Notify=true
