From d485ba1ad83c8ea29ffdd1100529d4f665a59108 Mon Sep 17 00:00:00 2001
From: Ryan Gonzalez <rymg19@gmail.com>
Date: Sat, 16 Jun 2018 10:18:37 -0500
Subject: [PATCH] Fix #633: Not properly linking in Boost components causing
 linking errors

---
 tsc/CMakeLists.txt | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/tsc/CMakeLists.txt b/tsc/CMakeLists.txt
index eef4eb4c..bd5fc9ba 100644
--- a/tsc/CMakeLists.txt
+++ b/tsc/CMakeLists.txt
@@ -107,7 +107,7 @@ if (WIN32)
   # Boost thread library has a different name on Windows to indicate
   # the different implemention. Just complicates the build system...
   set(Boost_USE_STATIC_LIBS ON)
-  find_package(Boost 1.50.0 COMPONENTS filesystem chrono thread_win32 system REQUIRED)
+  set(Boost_Thread_Library thread_win32)
 
   # Use MXE's pkg-config to resolve all deps for us
   pkg_check_modules(SFML REQUIRED sfml)
@@ -126,7 +126,8 @@ else()
   find_package(OpenGL REQUIRED)
   find_package(PNG REQUIRED)
   find_package(PCRE REQUIRED)
-  find_package(Boost 1.50.0 COMPONENTS filesystem chrono thread system REQUIRED)
+  set(Boost_Thread_Library thread)
+
   if (USE_LIBXMLPP3)
     find_package(LibXmlPP 3.0 REQUIRED)
   else()
@@ -138,6 +139,9 @@ else()
   find_package(X11 REQUIRED)
 endif()
 
+find_package(Boost 1.50.0 COMPONENTS filesystem chrono ${Boost_Thread_Library} REQUIRED)
+set(Boost_COMPONENTS Boost::filesystem Boost::chrono Boost::${Boost_Thread_Library})
+
 # Libraries we can build ourselves under certain cirumstances if missing
 include("ProvideTinyclipboard")
 if (ENABLE_MRUBY)
@@ -219,7 +223,7 @@ target_link_libraries(tsc
   ${CEGUI_LIBRARIES}
   ${SFML_LIBRARIES}
   ${SFML_DEPENDENCIES}
-  ${Boost_LIBRARIES}
+  ${Boost_COMPONENTS}
   ${OPENGL_LIBRARIES}
   ${PNG_LIBRARY}
   ${PNG_LIBRARIES} # compatibility
