diff -Nur orig/configure mod/configure
--- orig/configure	2020-05-28 10:19:47.000000000 +0200
+++ mod/configure	2021-01-03 14:58:27.452871854 +0100
@@ -6020,7 +6020,7 @@
 		GTK_API_VERSION="2.0"
 fi
 pkg_modules="gtk+-$GTK_API_VERSION >= 2.16.0"
-OTHER_DESKTOP_ENTRIES="Icon=grsync.png"
+OTHER_DESKTOP_ENTRIES="Icon=grsync"
 MIMEINFO_XMLNS="xmlns='http://www.freedesktop.org/standards/shared-mime-info'"
 MIME_OSSOCAT=""
 
diff -Nur orig/grsync.desktop.in mod/grsync.desktop.in
--- orig/grsync.desktop.in	2015-09-11 13:21:28.000000000 +0200
+++ mod/grsync.desktop.in	2021-01-03 14:59:44.313164278 +0100
@@ -4,7 +4,7 @@
 Terminal=false
 Type=Application
 MimeType=application/x-grsync-session;
-Categories=Application;System;
+Categories=GTK;System;
 GenericName=Synchronize files with rsync
 GenericName[fr]=Interface GTK pour rsync
 GenericName[it]=Sincronizza file con rsync
@@ -63,5 +63,5 @@
 Comment[pt_BR]=Sincronizar arquivos e pastas (Um GTK GUI para o rsync)
 Comment[ru]=Синхронизирует файлы и директории (GTK GUI для rsync)
 Comment[zh_TW]=同步檔案與資料夾 (rsync 的 GTK 圖形使用者介面)
-Exec=@prefix@/bin/@PACKAGE@ -i %f
+Exec=@PACKAGE@ -i %f
 @OTHER_DESKTOP_ENTRIES@
