diff -Naur orig/data/synaptic.desktop.in patched/data/synaptic.desktop.in
--- orig/data/synaptic.desktop.in	2015-01-19 14:35:04.000000000 +0100
+++ patched/data/synaptic.desktop.in	2015-01-20 20:40:05.356288688 +0100
@@ -2,7 +2,7 @@
 _Name=Synaptic Package Manager
 _GenericName=Package Manager
 _Comment=Install, remove and upgrade software packages
-Exec=synaptic-pkexec
+Exec=synaptic-start
 Icon=synaptic
 Terminal=false
 Type=Application
