From 4ad6ce0eb178949c2b09c856bd7767cf902883f9 Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <juliank@ubuntu.com>
Date: Mon, 17 Jun 2019 15:26:35 +0200
Subject: [PATCH 01/17] Update config.h.in and INSTALL file

---
 INSTALL     | 320 ++++++++++++++++++++++++++--------------------------
 config.h.in |   8 ++
 2 files changed, 167 insertions(+), 161 deletions(-)

diff --git a/INSTALL b/INSTALL
index 20998407..8865734f 100644
--- a/INSTALL
+++ b/INSTALL
@@ -1,8 +1,8 @@
 Installation Instructions
 *************************
 
-Copyright (C) 1994-1996, 1999-2002, 2004-2013 Free Software Foundation,
-Inc.
+   Copyright (C) 1994-1996, 1999-2002, 2004-2016 Free Software
+Foundation, Inc.
 
    Copying and distribution of this file, with or without modification,
 are permitted in any medium without royalty provided the copyright
@@ -12,97 +12,96 @@ without warranty of any kind.
 Basic Installation
 ==================
 
-   Briefly, the shell command `./configure && make && make install'
+   Briefly, the shell command './configure && make && make install'
 should configure, build, and install this package.  The following
-more-detailed instructions are generic; see the `README' file for
+more-detailed instructions are generic; see the 'README' file for
 instructions specific to this package.  Some packages provide this
-`INSTALL' file but do not implement all of the features documented
+'INSTALL' file but do not implement all of the features documented
 below.  The lack of an optional feature in a given package is not
 necessarily a bug.  More recommendations for GNU packages can be found
 in *note Makefile Conventions: (standards)Makefile Conventions.
 
-   The `configure' shell script attempts to guess correct values for
+   The 'configure' shell script attempts to guess correct values for
 various system-dependent variables used during compilation.  It uses
-those values to create a `Makefile' in each directory of the package.
-It may also create one or more `.h' files containing system-dependent
-definitions.  Finally, it creates a shell script `config.status' that
+those values to create a 'Makefile' in each directory of the package.
+It may also create one or more '.h' files containing system-dependent
+definitions.  Finally, it creates a shell script 'config.status' that
 you can run in the future to recreate the current configuration, and a
-file `config.log' containing compiler output (useful mainly for
-debugging `configure').
+file 'config.log' containing compiler output (useful mainly for
+debugging 'configure').
 
-   It can also use an optional file (typically called `config.cache'
-and enabled with `--cache-file=config.cache' or simply `-C') that saves
-the results of its tests to speed up reconfiguring.  Caching is
-disabled by default to prevent problems with accidental use of stale
-cache files.
+   It can also use an optional file (typically called 'config.cache' and
+enabled with '--cache-file=config.cache' or simply '-C') that saves the
+results of its tests to speed up reconfiguring.  Caching is disabled by
+default to prevent problems with accidental use of stale cache files.
 
    If you need to do unusual things to compile the package, please try
-to figure out how `configure' could check whether to do them, and mail
-diffs or instructions to the address given in the `README' so they can
+to figure out how 'configure' could check whether to do them, and mail
+diffs or instructions to the address given in the 'README' so they can
 be considered for the next release.  If you are using the cache, and at
-some point `config.cache' contains results you don't want to keep, you
+some point 'config.cache' contains results you don't want to keep, you
 may remove or edit it.
 
-   The file `configure.ac' (or `configure.in') is used to create
-`configure' by a program called `autoconf'.  You need `configure.ac' if
-you want to change it or regenerate `configure' using a newer version
-of `autoconf'.
+   The file 'configure.ac' (or 'configure.in') is used to create
+'configure' by a program called 'autoconf'.  You need 'configure.ac' if
+you want to change it or regenerate 'configure' using a newer version of
+'autoconf'.
 
    The simplest way to compile this package is:
 
-  1. `cd' to the directory containing the package's source code and type
-     `./configure' to configure the package for your system.
+  1. 'cd' to the directory containing the package's source code and type
+     './configure' to configure the package for your system.
 
-     Running `configure' might take a while.  While running, it prints
+     Running 'configure' might take a while.  While running, it prints
      some messages telling which features it is checking for.
 
-  2. Type `make' to compile the package.
+  2. Type 'make' to compile the package.
 
-  3. Optionally, type `make check' to run any self-tests that come with
+  3. Optionally, type 'make check' to run any self-tests that come with
      the package, generally using the just-built uninstalled binaries.
 
-  4. Type `make install' to install the programs and any data files and
+  4. Type 'make install' to install the programs and any data files and
      documentation.  When installing into a prefix owned by root, it is
      recommended that the package be configured and built as a regular
-     user, and only the `make install' phase executed with root
+     user, and only the 'make install' phase executed with root
      privileges.
 
-  5. Optionally, type `make installcheck' to repeat any self-tests, but
+  5. Optionally, type 'make installcheck' to repeat any self-tests, but
      this time using the binaries in their final installed location.
      This target does not install anything.  Running this target as a
-     regular user, particularly if the prior `make install' required
+     regular user, particularly if the prior 'make install' required
      root privileges, verifies that the installation completed
      correctly.
 
   6. You can remove the program binaries and object files from the
-     source code directory by typing `make clean'.  To also remove the
-     files that `configure' created (so you can compile the package for
-     a different kind of computer), type `make distclean'.  There is
-     also a `make maintainer-clean' target, but that is intended mainly
+     source code directory by typing 'make clean'.  To also remove the
+     files that 'configure' created (so you can compile the package for
+     a different kind of computer), type 'make distclean'.  There is
+     also a 'make maintainer-clean' target, but that is intended mainly
      for the package's developers.  If you use it, you may have to get
      all sorts of other programs in order to regenerate files that came
      with the distribution.
 
-  7. Often, you can also type `make uninstall' to remove the installed
+  7. Often, you can also type 'make uninstall' to remove the installed
      files again.  In practice, not all packages have tested that
      uninstallation works correctly, even though it is required by the
      GNU Coding Standards.
 
-  8. Some packages, particularly those that use Automake, provide `make
+  8. Some packages, particularly those that use Automake, provide 'make
      distcheck', which can by used by developers to test that all other
-     targets like `make install' and `make uninstall' work correctly.
+     targets like 'make install' and 'make uninstall' work correctly.
      This target is generally not run by end users.
 
 Compilers and Options
 =====================
 
    Some systems require unusual options for compilation or linking that
-the `configure' script does not know about.  Run `./configure --help'
+the 'configure' script does not know about.  Run './configure --help'
 for details on some of the pertinent environment variables.
 
-   You can give `configure' initial values for configuration parameters
-by setting variables in the command line or in the environment.  Here
-is an example:
+   You can give 'configure' initial values for configuration parameters
+by setting variables in the command line or in the environment.  Here is
+an example:
 
      ./configure CC=c99 CFLAGS=-g LIBS=-lposix
 
@@ -113,21 +112,21 @@ Compiling For Multiple Architectures
 
    You can compile the package for more than one kind of computer at the
 same time, by placing the object files for each architecture in their
-own directory.  To do this, you can use GNU `make'.  `cd' to the
+own directory.  To do this, you can use GNU 'make'.  'cd' to the
 directory where you want the object files and executables to go and run
-the `configure' script.  `configure' automatically checks for the
-source code in the directory that `configure' is in and in `..'.  This
-is known as a "VPATH" build.
+the 'configure' script.  'configure' automatically checks for the source
+code in the directory that 'configure' is in and in '..'.  This is known
+as a "VPATH" build.
 
-   With a non-GNU `make', it is safer to compile the package for one
+   With a non-GNU 'make', it is safer to compile the package for one
 architecture at a time in the source code directory.  After you have
-installed the package for one architecture, use `make distclean' before
+installed the package for one architecture, use 'make distclean' before
 reconfiguring for another architecture.
 
    On MacOS X 10.5 and later systems, you can create libraries and
 executables that work on multiple system types--known as "fat" or
-"universal" binaries--by specifying multiple `-arch' options to the
-compiler but only a single `-arch' option to the preprocessor.  Like
+"universal" binaries--by specifying multiple '-arch' options to the
+compiler but only a single '-arch' option to the preprocessor.  Like
 this:
 
      ./configure CC="gcc -arch i386 -arch x86_64 -arch ppc -arch ppc64" \
@@ -136,105 +135,104 @@ this:
 
    This is not guaranteed to produce working output in all cases, you
 may have to build one architecture at a time and combine the results
-using the `lipo' tool if you have problems.
+using the 'lipo' tool if you have problems.
 
 Installation Names
 ==================
 
-   By default, `make install' installs the package's commands under
-`/usr/local/bin', include files under `/usr/local/include', etc.  You
-can specify an installation prefix other than `/usr/local' by giving
-`configure' the option `--prefix=PREFIX', where PREFIX must be an
+   By default, 'make install' installs the package's commands under
+'/usr/local/bin', include files under '/usr/local/include', etc.  You
+can specify an installation prefix other than '/usr/local' by giving
+'configure' the option '--prefix=PREFIX', where PREFIX must be an
 absolute file name.
 
    You can specify separate installation prefixes for
 architecture-specific files and architecture-independent files.  If you
-pass the option `--exec-prefix=PREFIX' to `configure', the package uses
+pass the option '--exec-prefix=PREFIX' to 'configure', the package uses
 PREFIX as the prefix for installing programs and libraries.
 Documentation and other data files still use the regular prefix.
 
    In addition, if you use an unusual directory layout you can give
-options like `--bindir=DIR' to specify different values for particular
-kinds of files.  Run `configure --help' for a list of the directories
-you can set and what kinds of files go in them.  In general, the
-default for these options is expressed in terms of `${prefix}', so that
-specifying just `--prefix' will affect all of the other directory
+options like '--bindir=DIR' to specify different values for particular
+kinds of files.  Run 'configure --help' for a list of the directories
+you can set and what kinds of files go in them.  In general, the default
+for these options is expressed in terms of '${prefix}', so that
+specifying just '--prefix' will affect all of the other directory
 specifications that were not explicitly provided.
 
    The most portable way to affect installation locations is to pass the
-correct locations to `configure'; however, many packages provide one or
+correct locations to 'configure'; however, many packages provide one or
 both of the following shortcuts of passing variable assignments to the
-`make install' command line to change installation locations without
+'make install' command line to change installation locations without
 having to reconfigure or recompile.
 
    The first method involves providing an override variable for each
-affected directory.  For example, `make install
+affected directory.  For example, 'make install
 prefix=/alternate/directory' will choose an alternate location for all
 directory configuration variables that were expressed in terms of
-`${prefix}'.  Any directories that were specified during `configure',
-but not in terms of `${prefix}', must each be overridden at install
-time for the entire installation to be relocated.  The approach of
-makefile variable overrides for each directory variable is required by
-the GNU Coding Standards, and ideally causes no recompilation.
-However, some platforms have known limitations with the semantics of
-shared libraries that end up requiring recompilation when using this
-method, particularly noticeable in packages that use GNU Libtool.
-
-   The second method involves providing the `DESTDIR' variable.  For
-example, `make install DESTDIR=/alternate/directory' will prepend
-`/alternate/directory' before all installation names.  The approach of
-`DESTDIR' overrides is not required by the GNU Coding Standards, and
+'${prefix}'.  Any directories that were specified during 'configure',
+but not in terms of '${prefix}', must each be overridden at install time
+for the entire installation to be relocated.  The approach of makefile
+variable overrides for each directory variable is required by the GNU
+Coding Standards, and ideally causes no recompilation.  However, some
+platforms have known limitations with the semantics of shared libraries
+that end up requiring recompilation when using this method, particularly
+noticeable in packages that use GNU Libtool.
+
+   The second method involves providing the 'DESTDIR' variable.  For
+example, 'make install DESTDIR=/alternate/directory' will prepend
+'/alternate/directory' before all installation names.  The approach of
+'DESTDIR' overrides is not required by the GNU Coding Standards, and
 does not work on platforms that have drive letters.  On the other hand,
 it does better at avoiding recompilation issues, and works well even
-when some directory options were not specified in terms of `${prefix}'
-at `configure' time.
+when some directory options were not specified in terms of '${prefix}'
+at 'configure' time.
 
 Optional Features
 =================
 
    If the package supports it, you can cause programs to be installed
-with an extra prefix or suffix on their names by giving `configure' the
-option `--program-prefix=PREFIX' or `--program-suffix=SUFFIX'.
-
-   Some packages pay attention to `--enable-FEATURE' options to
-`configure', where FEATURE indicates an optional part of the package.
-They may also pay attention to `--with-PACKAGE' options, where PACKAGE
-is something like `gnu-as' or `x' (for the X Window System).  The
-`README' should mention any `--enable-' and `--with-' options that the
+with an extra prefix or suffix on their names by giving 'configure' the
+option '--program-prefix=PREFIX' or '--program-suffix=SUFFIX'.
+
+   Some packages pay attention to '--enable-FEATURE' options to
+'configure', where FEATURE indicates an optional part of the package.
+They may also pay attention to '--with-PACKAGE' options, where PACKAGE
+is something like 'gnu-as' or 'x' (for the X Window System).  The
+'README' should mention any '--enable-' and '--with-' options that the
 package recognizes.
 
-   For packages that use the X Window System, `configure' can usually
+   For packages that use the X Window System, 'configure' can usually
 find the X include and library files automatically, but if it doesn't,
-you can use the `configure' options `--x-includes=DIR' and
-`--x-libraries=DIR' to specify their locations.
+you can use the 'configure' options '--x-includes=DIR' and
+'--x-libraries=DIR' to specify their locations.
 
    Some packages offer the ability to configure how verbose the
-execution of `make' will be.  For these packages, running `./configure
+execution of 'make' will be.  For these packages, running './configure
 --enable-silent-rules' sets the default to minimal output, which can be
-overridden with `make V=1'; while running `./configure
+overridden with 'make V=1'; while running './configure
 --disable-silent-rules' sets the default to verbose, which can be
-overridden with `make V=0'.
+overridden with 'make V=0'.
 
 Particular systems
 ==================
 
-   On HP-UX, the default C compiler is not ANSI C compatible.  If GNU
-CC is not installed, it is recommended to use the following options in
+   On HP-UX, the default C compiler is not ANSI C compatible.  If GNU CC
+is not installed, it is recommended to use the following options in
 order to use an ANSI C compiler:
 
      ./configure CC="cc -Ae -D_XOPEN_SOURCE=500"
 
 and if that doesn't work, install pre-built binaries of GCC for HP-UX.
 
-   HP-UX `make' updates targets which have the same time stamps as
-their prerequisites, which makes it generally unusable when shipped
-generated files such as `configure' are involved.  Use GNU `make'
-instead.
+   HP-UX 'make' updates targets which have the same time stamps as their
+prerequisites, which makes it generally unusable when shipped generated
+files such as 'configure' are involved.  Use GNU 'make' instead.
 
    On OSF/1 a.k.a. Tru64, some versions of the default C compiler cannot
-parse its `<wchar.h>' header file.  The option `-nodtk' can be used as
-a workaround.  If GNU CC is not installed, it is therefore recommended
-to try
+parse its '<wchar.h>' header file.  The option '-nodtk' can be used as a
+workaround.  If GNU CC is not installed, it is therefore recommended to
+try
 
      ./configure CC="cc"
 
@@ -242,26 +240,26 @@ and if that doesn't work, try
 
      ./configure CC="cc -nodtk"
 
-   On Solaris, don't put `/usr/ucb' early in your `PATH'.  This
+   On Solaris, don't put '/usr/ucb' early in your 'PATH'.  This
 directory contains several dysfunctional programs; working variants of
-these programs are available in `/usr/bin'.  So, if you need `/usr/ucb'
-in your `PATH', put it _after_ `/usr/bin'.
+these programs are available in '/usr/bin'.  So, if you need '/usr/ucb'
+in your 'PATH', put it _after_ '/usr/bin'.
 
-   On Haiku, software installed for all users goes in `/boot/common',
-not `/usr/local'.  It is recommended to use the following options:
+   On Haiku, software installed for all users goes in '/boot/common',
+not '/usr/local'.  It is recommended to use the following options:
 
      ./configure --prefix=/boot/common
 
 Specifying the System Type
 ==========================
 
-   There may be some features `configure' cannot figure out
+   There may be some features 'configure' cannot figure out
 automatically, but needs to determine by the type of machine the package
 will run on.  Usually, assuming the package is built to be run on the
-_same_ architectures, `configure' can figure that out, but if it prints
+_same_ architectures, 'configure' can figure that out, but if it prints
 a message saying it cannot guess the machine type, give it the
-`--build=TYPE' option.  TYPE can either be a short name for the system
-type, such as `sun4', or a canonical name which has the form:
+'--build=TYPE' option.  TYPE can either be a short name for the system
+type, such as 'sun4', or a canonical name which has the form:
 
      CPU-COMPANY-SYSTEM
 
@@ -270,101 +268,101 @@ where SYSTEM can have one of these forms:
      OS
      KERNEL-OS
 
-   See the file `config.sub' for the possible values of each field.  If
-`config.sub' isn't included in this package, then this package doesn't
+   See the file 'config.sub' for the possible values of each field.  If
+'config.sub' isn't included in this package, then this package doesn't
 need to know the machine type.
 
    If you are _building_ compiler tools for cross-compiling, you should
-use the option `--target=TYPE' to select the type of system they will
+use the option '--target=TYPE' to select the type of system they will
 produce code for.
 
    If you want to _use_ a cross compiler, that generates code for a
 platform different from the build platform, you should specify the
 "host" platform (i.e., that on which the generated programs will
-eventually be run) with `--host=TYPE'.
+eventually be run) with '--host=TYPE'.
 
 Sharing Defaults
 ================
 
-   If you want to set default values for `configure' scripts to share,
-you can create a site shell script called `config.site' that gives
-default values for variables like `CC', `cache_file', and `prefix'.
-`configure' looks for `PREFIX/share/config.site' if it exists, then
-`PREFIX/etc/config.site' if it exists.  Or, you can set the
-`CONFIG_SITE' environment variable to the location of the site script.
-A warning: not all `configure' scripts look for a site script.
+   If you want to set default values for 'configure' scripts to share,
+you can create a site shell script called 'config.site' that gives
+default values for variables like 'CC', 'cache_file', and 'prefix'.
+'configure' looks for 'PREFIX/share/config.site' if it exists, then
+'PREFIX/etc/config.site' if it exists.  Or, you can set the
+'CONFIG_SITE' environment variable to the location of the site script.
+A warning: not all 'configure' scripts look for a site script.
 
 Defining Variables
 ==================
 
    Variables not defined in a site shell script can be set in the
-environment passed to `configure'.  However, some packages may run
+environment passed to 'configure'.  However, some packages may run
 configure again during the build, and the customized values of these
 variables may be lost.  In order to avoid this problem, you should set
-them in the `configure' command line, using `VAR=value'.  For example:
+them in the 'configure' command line, using 'VAR=value'.  For example:
 
      ./configure CC=/usr/local2/bin/gcc
 
-causes the specified `gcc' to be used as the C compiler (unless it is
+causes the specified 'gcc' to be used as the C compiler (unless it is
 overridden in the site shell script).
 
-Unfortunately, this technique does not work for `CONFIG_SHELL' due to
-an Autoconf limitation.  Until the limitation is lifted, you can use
-this workaround:
+Unfortunately, this technique does not work for 'CONFIG_SHELL' due to an
+Autoconf limitation.  Until the limitation is lifted, you can use this
+workaround:
 
      CONFIG_SHELL=/bin/bash ./configure CONFIG_SHELL=/bin/bash
 
-`configure' Invocation
+'configure' Invocation
 ======================
 
-   `configure' recognizes the following options to control how it
+   'configure' recognizes the following options to control how it
 operates.
 
-`--help'
-`-h'
-     Print a summary of all of the options to `configure', and exit.
+'--help'
+'-h'
+     Print a summary of all of the options to 'configure', and exit.
 
-`--help=short'
-`--help=recursive'
+'--help=short'
+'--help=recursive'
      Print a summary of the options unique to this package's
-     `configure', and exit.  The `short' variant lists options used
-     only in the top level, while the `recursive' variant lists options
-     also present in any nested packages.
+     'configure', and exit.  The 'short' variant lists options used only
+     in the top level, while the 'recursive' variant lists options also
+     present in any nested packages.
 
-`--version'
-`-V'
-     Print the version of Autoconf used to generate the `configure'
+'--version'
+'-V'
+     Print the version of Autoconf used to generate the 'configure'
      script, and exit.
 
-`--cache-file=FILE'
+'--cache-file=FILE'
      Enable the cache: use and save the results of the tests in FILE,
-     traditionally `config.cache'.  FILE defaults to `/dev/null' to
+     traditionally 'config.cache'.  FILE defaults to '/dev/null' to
      disable caching.
 
-`--config-cache'
-`-C'
-     Alias for `--cache-file=config.cache'.
+'--config-cache'
+'-C'
+     Alias for '--cache-file=config.cache'.
 
-`--quiet'
-`--silent'
-`-q'
+'--quiet'
+'--silent'
+'-q'
      Do not print messages saying which checks are being made.  To
-     suppress all normal output, redirect it to `/dev/null' (any error
+     suppress all normal output, redirect it to '/dev/null' (any error
      messages will still be shown).
 
-`--srcdir=DIR'
+'--srcdir=DIR'
      Look for the package's source code in directory DIR.  Usually
-     `configure' can determine that directory automatically.
+     'configure' can determine that directory automatically.
 
-`--prefix=DIR'
-     Use DIR as the installation prefix.  *note Installation Names::
-     for more details, including other options available for fine-tuning
-     the installation locations.
+'--prefix=DIR'
+     Use DIR as the installation prefix.  *note Installation Names:: for
+     more details, including other options available for fine-tuning the
+     installation locations.
 
-`--no-create'
-`-n'
+'--no-create'
+'-n'
      Run the configure checks, but stop before creating any output
      files.
 
-`configure' also accepts some other, not widely useful, options.  Run
-`configure --help' for more details.
+'configure' also accepts some other, not widely useful, options.  Run
+'configure --help' for more details.
diff --git a/config.h.in b/config.h.in
index 5dc55fde..24e8adb7 100644
--- a/config.h.in
+++ b/config.h.in
@@ -12,6 +12,14 @@
 /* Define to 1 if you have the `bind_textdomain_codeset' function. */
 #undef HAVE_BIND_TEXTDOMAIN_CODESET
 
+/* Define to 1 if you have the Mac OS X function CFLocaleCopyCurrent in the
+   CoreFoundation framework. */
+#undef HAVE_CFLOCALECOPYCURRENT
+
+/* Define to 1 if you have the Mac OS X function CFPreferencesCopyAppValue in
+   the CoreFoundation framework. */
+#undef HAVE_CFPREFERENCESCOPYAPPVALUE
+
 /* Define to 1 if you have the `dcgettext' function. */
 #undef HAVE_DCGETTEXT
 

From 881e53190142211f62b1880f883b31613a0fd1e5 Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <juliank@ubuntu.com>
Date: Mon, 17 Jun 2019 15:04:56 +0200
Subject: [PATCH 02/17] Remove obsolete <apt-pkg/sptr.h> include

---
 common/raptoptions.cc | 1 -
 common/rpackage.cc    | 1 -
 2 files changed, 2 deletions(-)

diff --git a/common/raptoptions.cc b/common/raptoptions.cc
index 1f494c2d..a53c9c58 100644
--- a/common/raptoptions.cc
+++ b/common/raptoptions.cc
@@ -31,7 +31,6 @@
 #include <apt-pkg/configuration.h>
 #include <apt-pkg/tagfile.h>
 #include <apt-pkg/policy.h>
-#include <apt-pkg/sptr.h>
 #include <apt-pkg/strutl.h>
 #include <apt-pkg/fileutl.h>
 
diff --git a/common/rpackage.cc b/common/rpackage.cc
index 087a1953..c507b109 100644
--- a/common/rpackage.cc
+++ b/common/rpackage.cc
@@ -54,7 +54,6 @@
 #include <apt-pkg/configuration.h>
 #include <apt-pkg/tagfile.h>
 #include <apt-pkg/policy.h>
-#include <apt-pkg/sptr.h>
 #include <apt-pkg/strutl.h>
 #include <apt-pkg/cacheiterators.h>
 #include <apt-pkg/pkgcache.h>

From 8bdf323ba922dd2cbbf478a3c41e24ea6f3223d2 Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <juliank@ubuntu.com>
Date: Mon, 17 Jun 2019 15:07:27 +0200
Subject: [PATCH 03/17] Use RFC1123StrToTime instead of StrToTime

---
 common/rpackagestatus.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/rpackagestatus.cc b/common/rpackagestatus.cc
index be943691..b0b3f595 100644
--- a/common/rpackagestatus.cc
+++ b/common/rpackagestatus.cc
@@ -199,7 +199,7 @@ bool RPackageStatus::maintenanceEndTime(RPackage *pkg, struct tm *res)
    t.Step(sec);
 
    // get the time_t form the string
-   if(!StrToTime(sec.FindS("Date"), release_date))
+   if(!RFC1123StrToTime(sec.FindS("Date").c_str(), release_date))
       return false;
 
    // if its not a supported package, return 0 

From e72bf7349113d31cb052fda39296baef6cda52f6 Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <juliank@ubuntu.com>
Date: Mon, 17 Jun 2019 15:12:44 +0200
Subject: [PATCH 04/17] Use VerIterator::Section() instead of
 PkgIterator::Section()

---
 common/rpackage.cc       | 19 +++++++++++++------
 common/rpackagelister.cc | 10 ++++------
 2 files changed, 17 insertions(+), 12 deletions(-)

diff --git a/common/rpackage.cc b/common/rpackage.cc
index c507b109..ff1dc2d9 100644
--- a/common/rpackage.cc
+++ b/common/rpackage.cc
@@ -107,11 +107,14 @@ void RPackage::addVirtualPackage(pkgCache::PkgIterator dep)
 
 const char *RPackage::section()
 {
-   const char *s = _package->Section();
-   if (s != NULL)
-      return s;
-   else
-      return _("Unknown");
+   pkgCache::VerIterator ver = (*_depcache)[*_package].CandidateVerIter(*_depcache);
+   if (!ver.end()) {
+      const char *s = ver.Section();
+      if (s != NULL)
+         return s;
+   }
+
+   return _("Unknown");
 }
 
 const char *RPackage::srcPackage()
@@ -1447,7 +1450,11 @@ string RPackage::component()
    string res;
 #ifdef WITH_APT_AUTH
    // the apt-secure patch breaks File.Component
-   const char *s = _package->Section();
+   pkgCache::VerIterator ver = (*_depcache)[*_package].CandidateVerIter(*_depcache);
+   const char *s = NULL;
+   if (!ver.end())
+      s = ver.Section();
+
    if(s == NULL)
       return "";
 
diff --git a/common/rpackagelister.cc b/common/rpackagelister.cc
index 4643b41e..52621597 100644
--- a/common/rpackagelister.cc
+++ b/common/rpackagelister.cc
@@ -408,12 +408,10 @@ bool RPackageLister::openCache()
 	 pkg->setPinned(true);
 
       // Gather list of sections.
-      if (I.Section())
-         sectionSet.insert(I.Section());
-#if 0 // may confuse users
-      else
-         cerr << "Package " << I.Name() << " has no section?!" << endl;
-#endif
+      for (auto Ver = I.VersionList(); !Ver.end(); Ver++) {
+         if (Ver.Section())
+            sectionSet.insert(Ver.Section());
+      }
    }
 
    // refresh the views

From b66704178b1be9e4e68151ad7309cc6cabaddc1e Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <juliank@ubuntu.com>
Date: Mon, 17 Jun 2019 15:21:39 +0200
Subject: [PATCH 05/17] Use APT::Progress::PackageManager subclass instead of
 fd

---
 common/rinstallprogress.cc   | 7 +++++--
 gtk/rgdebinstallprogress.cc  | 4 +++-
 gtk/rgterminstallprogress.cc | 4 +++-
 3 files changed, 11 insertions(+), 4 deletions(-)

diff --git a/common/rinstallprogress.cc b/common/rinstallprogress.cc
index 74772cf0..229550dd 100644
--- a/common/rinstallprogress.cc
+++ b/common/rinstallprogress.cc
@@ -31,6 +31,7 @@
 #include <iostream>
 #include <cstdio>
 #include <apt-pkg/error.h>
+#include <apt-pkg/install-progress.h>
 #ifdef HAVE_RPM
 #include <apt-pkg/configuration.h>
 #endif
@@ -99,7 +100,8 @@ pkgPackageManager::OrderResult RInstallProgress::start(pkgPackageManager *pm,
       close(fd[0]);
       close(fd[1]);
 
-      res = pm->DoInstallPostFork();
+      APT::Progress::PackageManagerProgressFd progress(-1);
+      res = pm->DoInstallPostFork(&progress);
       // dump errors into cerr (pass it to the parent process)	
       _error->DumpErrors();
       _exit(res);
@@ -124,7 +126,8 @@ pkgPackageManager::OrderResult RInstallProgress::start(pkgPackageManager *pm,
    _child_id = fork();
 
    if (_child_id == 0) {
-      res = pm->DoInstallPostFork();
+      APT::Progress::PackageManagerProgressFd progress(-1);
+      res = pm->DoInstallPostFork(&progress);
       _exit(res);
    }
 #endif
diff --git a/gtk/rgdebinstallprogress.cc b/gtk/rgdebinstallprogress.cc
index a9ebcf33..5f579260 100644
--- a/gtk/rgdebinstallprogress.cc
+++ b/gtk/rgdebinstallprogress.cc
@@ -43,6 +43,7 @@
 
 #include <apt-pkg/configuration.h>
 #include <apt-pkg/error.h>
+#include <apt-pkg/install-progress.h>
 #include <gtk/gtk.h>
 
 #include <unistd.h>
@@ -677,7 +678,8 @@ pkgPackageManager::OrderResult RGDebInstallProgress::start(pkgPackageManager *pm
       pipe(fd);
       ipc_send_fd(fd[0]); // send the read part of the pipe to the parent
 
-      res = pm->DoInstallPostFork(fd[1]);
+      APT::Progress::PackageManagerProgressFd progress(fd[1]);
+      res = pm->DoInstallPostFork(&progress);
 
       // dump errors into cerr (pass it to the parent process)	
       _error->DumpErrors();
diff --git a/gtk/rgterminstallprogress.cc b/gtk/rgterminstallprogress.cc
index 8ea1f0fa..eaae924f 100644
--- a/gtk/rgterminstallprogress.cc
+++ b/gtk/rgterminstallprogress.cc
@@ -42,6 +42,7 @@
 #include <sys/types.h>
 #include <apt-pkg/error.h>
 #include <apt-pkg/configuration.h>
+#include <apt-pkg/install-progress.h>
 #include <apt-pkg/pkgsystem.h>
 #include <cassert>
 #include <cstring>
@@ -244,7 +245,8 @@ RGTermInstallProgress::start(pkgPackageManager *pm,
       memset( &new_act, 0, sizeof( new_act ) );
       new_act.sa_handler = SIG_IGN;
       sigaction( SIGPIPE, &new_act, NULL);
-      res = pm->DoInstallPostFork();
+      APT::Progress::PackageManagerProgressFd progress(-1);
+      res = pm->DoInstallPostFork(&progress);
       _exit(res);
    }
    // parent: assign pty to the vte terminal

From 8ed5d48592b5885a60cc1dea6e49789b35d1c71c Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <juliank@ubuntu.com>
Date: Mon, 17 Jun 2019 15:23:05 +0200
Subject: [PATCH 06/17] Remove calls to no-op InstallProtect()

---
 common/rpackage.cc       | 1 -
 common/rpackagelister.cc | 1 -
 2 files changed, 2 deletions(-)

diff --git a/common/rpackage.cc b/common/rpackage.cc
index ff1dc2d9..a652cdb6 100644
--- a/common/rpackage.cc
+++ b/common/rpackage.cc
@@ -918,7 +918,6 @@ void RPackage::setRemove(bool purge)
    Fix.Protect(*_package);
    Fix.Remove(*_package);
 
-   Fix.InstallProtect();
    Fix.Resolve(true);
 
    _depcache->SetReInstall(*_package, false);
diff --git a/common/rpackagelister.cc b/common/rpackagelister.cc
index 52621597..d48bd1f8 100644
--- a/common/rpackagelister.cc
+++ b/common/rpackagelister.cc
@@ -1879,7 +1879,6 @@ bool RPackageLister::readSelections(istream &in)
       _lua->ResetCaches();
 #endif
       _progMeter->Done();
-      Fix.InstallProtect();
       Fix.Resolve(true);
 
       // refresh all views

From d76d0e584a60dd01784018e3ddfc21fafec01273 Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <juliank@ubuntu.com>
Date: Mon, 17 Jun 2019 15:34:54 +0200
Subject: [PATCH 07/17] Port code from TFRewrite to pkgTagSection::Write

This even looks nicer.
---
 common/rpackage.cc | 23 +++++++++--------------
 1 file changed, 9 insertions(+), 14 deletions(-)

diff --git a/common/rpackage.cc b/common/rpackage.cc
index a652cdb6..58d14db3 100644
--- a/common/rpackage.cc
+++ b/common/rpackage.cc
@@ -1106,13 +1106,13 @@ void RPackage::setPinned(bool flag)
       stat(File.c_str(), &stat_buf);
       // create a tmp_pin file in the internal dir
       string filename = RStateDir() + "/.tmp_preferences";
-      FILE *out = fopen(filename.c_str(),"w");
-      if (out == NULL)
-         cerr << "error opening tmpfile: " << filename << endl;
+      FileFd out(filename, FileFd::WriteOnly | FileFd::Create | FileFd::Empty);
+      if (!out.IsOpen()) {
+         _error->DumpErrors();
+      }
       FileFd Fd(File, FileFd::ReadOnly);
       pkgTagFile TF(&Fd);
       if (_error->PendingError() == true) {
-         fclose(out);
          return;
       }
       pkgTagSection Tags;
@@ -1124,19 +1124,14 @@ void RPackage::setPinned(bool flag)
                      ("Invalid record in the preferences file, no Package header"));
             return;
          }
-         if (Name != name()) {
-            TFRewriteData tfrd;
-            tfrd.Tag = 0;
-            tfrd.Rewrite = 0;
-            tfrd.NewTag = 0;
-            TFRewrite(out, Tags, TFRewritePackageOrder, &tfrd);
-            fprintf(out, "\n");
-         }
+         if (Name != name())
+            Tags.Write(out, TFRewritePackageOrder, {});
+         out.Write("\n", 1);
       }
-      fflush(out);
+      out.Flush();
       rename(filename.c_str(), File.c_str());
       chmod(File.c_str(), stat_buf.st_mode);
-      fclose(out);
+      out.Close();
    }
 }
 

From 536425522731c34e53756560e69df2887dca620b Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <juliank@ubuntu.com>
Date: Mon, 17 Jun 2019 15:36:06 +0200
Subject: [PATCH 08/17] Fix un-namespaced use of std::vector and std::string

---
 common/rpackagecache.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/common/rpackagecache.h b/common/rpackagecache.h
index 08d6d44c..60cd6587 100644
--- a/common/rpackagecache.h
+++ b/common/rpackagecache.h
@@ -25,6 +25,8 @@
 #define _RPACKAGECACHE_H_
 
 #include <map>
+#include <string>
+#include <vector>
 
 #include <apt-pkg/depcache.h>
 #include <apt-pkg/sourcelist.h>
@@ -63,7 +65,7 @@ class RPackageCache {
 
    bool open(OpProgress &progress, bool lock=true);
 
-   vector<string> getPolicyArchives(bool filenames_only);
+   std::vector<std::string> getPolicyArchives(bool filenames_only);
 
    bool lock();
    void releaseLock();

From 206fbe34aecfc34e711d3917ea15cb0294d26846 Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <juliank@ubuntu.com>
Date: Mon, 17 Jun 2019 15:38:52 +0200
Subject: [PATCH 09/17] Port pkgMakeStatusCache to
 pkgCacheGenerator::MakeStatusCache

---
 common/rpackagecache.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/common/rpackagecache.cc b/common/rpackagecache.cc
index 6c768ee1..0ff9a525 100644
--- a/common/rpackagecache.cc
+++ b/common/rpackagecache.cc
@@ -63,9 +63,9 @@ bool RPackageCache::open(OpProgress &progress, bool locking)
 			     "Go to the repository dialog to correct the problem."));
 
    if(locking)
-      pkgMakeStatusCache(*_list, progress);
+      pkgCacheGenerator::MakeStatusCache(*_list, &progress, nullptr, false);
    else
-      pkgMakeStatusCache(*_list, progress, 0, true);
+      pkgCacheGenerator::MakeStatusCache(*_list, &progress, nullptr, true);
 
    if (_error->PendingError())
       return _error->

From f3d874dc7d194e09537247de1ccedbd01ee76f10 Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <juliank@ubuntu.com>
Date: Mon, 17 Jun 2019 15:40:31 +0200
Subject: [PATCH 10/17] Add missing #include <apt-pkg/update.h>

---
 common/rpackagelister.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/common/rpackagelister.cc b/common/rpackagelister.cc
index d48bd1f8..7199b108 100644
--- a/common/rpackagelister.cc
+++ b/common/rpackagelister.cc
@@ -55,6 +55,7 @@
 #include <apt-pkg/acquire-item.h>
 #include <apt-pkg/clean.h>
 #include <apt-pkg/version.h>
+#include <apt-pkg/update.h>
 
 #include <apt-pkg/sourcelist.h>
 #include <apt-pkg/pkgsystem.h>

From a31e21299d4bee7c16c3e6f41930932e6e915bf1 Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <juliank@ubuntu.com>
Date: Mon, 17 Jun 2019 15:46:47 +0200
Subject: [PATCH 11/17] Port to APT::Upgrade::Upgrade

---
 common/rpackagelister.cc | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/common/rpackagelister.cc b/common/rpackagelister.cc
index 7199b108..e023768d 100644
--- a/common/rpackagelister.cc
+++ b/common/rpackagelister.cc
@@ -56,6 +56,7 @@
 #include <apt-pkg/clean.h>
 #include <apt-pkg/version.h>
 #include <apt-pkg/update.h>
+#include <apt-pkg/upgrade.h>
 
 #include <apt-pkg/sourcelist.h>
 #include <apt-pkg/pkgsystem.h>
@@ -540,7 +541,7 @@ bool RPackageLister::fixBroken()
 
 bool RPackageLister::upgrade()
 {
-   if (pkgAllUpgrade(*_cache->deps()) == false) {
+   if (APT::Upgrade::Upgrade(*_cache->deps(), APT::Upgrade::FORBID_REMOVE_PACKAGES | APT::Upgrade::FORBID_INSTALL_NEW_PACKAGES) == false) {
       return _error->
          Error(_("Internal Error, AllUpgrade broke stuff. Please report."));
    }
@@ -559,7 +560,7 @@ bool RPackageLister::upgrade()
 
 bool RPackageLister::distUpgrade()
 {
-   if (pkgDistUpgrade(*_cache->deps()) == false) {
+   if (APT::Upgrade::Upgrade(*_cache->deps(), APT::Upgrade::ALLOW_EVERYTHING) == false) {
       cout << _("dist upgrade Failed") << endl;
       return false;
    }

From 21c0c2f72f3463c7c40bf9cb5c4d1d2e25f16631 Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <juliank@ubuntu.com>
Date: Mon, 17 Jun 2019 15:52:27 +0200
Subject: [PATCH 12/17] Port to a subclass of pkgArchiveCleaner that actually
 removes

---
 common/rpackagelister.cc | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/common/rpackagelister.cc b/common/rpackagelister.cc
index e023768d..ab04cf59 100644
--- a/common/rpackagelister.cc
+++ b/common/rpackagelister.cc
@@ -1738,7 +1738,17 @@ bool RPackageLister::cleanPackageCache(bool forceClean)
 
       lockPackageCache(lock);
 
-      pkgArchiveCleaner cleaner;
+      class SimpleCleaner : public pkgArchiveCleaner {
+#if APT_PKG_ABI >= 590
+         void Erase(int const dirfd, char const * const File, std::string const &Pkg, std::string const &Ver,struct stat const &St) override {
+            RemoveFileAt("Cleaner::Erase", dirfd, File);
+         }
+#else
+         void Erase(const char * File, std::string /*Pkg*/, std::string /*Ver*/, struct stat & /*St*/) override {
+            RemoveFile("Cleaner::Erase", File);
+         }
+#endif
+      } cleaner;
 
       bool res;
 

From fa7c0bf6119cbcf64c8ec6db19793bfd6edd906b Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <juliank@ubuntu.com>
Date: Mon, 17 Jun 2019 15:52:53 +0200
Subject: [PATCH 13/17] Port GetCandidateVer to GetCandidateVersion

---
 common/rpackagelister.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/rpackagelister.cc b/common/rpackagelister.cc
index ab04cf59..749678c8 100644
--- a/common/rpackagelister.cc
+++ b/common/rpackagelister.cc
@@ -1949,7 +1949,7 @@ bool RPackageLister::addArchiveToCache(string archive, string &pkgname)
    // md5sum check
    // first get the md5 of the candidate
    pkgDepCache *dcache = _cache->deps();
-   pkgCache::VerIterator ver = dcache->GetCandidateVer(*pkg->package());
+   pkgCache::VerIterator ver = dcache->GetCandidateVersion(*pkg->package());
    pkgCache::VerFileIterator Vf = ver.FileList(); 
    pkgRecords::Parser &Parse = _records->Lookup(Vf);
    string MD5 = Parse.MD5Hash();

From 77a358815a042f2e48c2ff853fd487b280cffd6b Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <juliank@ubuntu.com>
Date: Mon, 17 Jun 2019 15:58:23 +0200
Subject: [PATCH 14/17] Swich from MD5 to Hashes

---
 common/rpackagelister.cc | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/common/rpackagelister.cc b/common/rpackagelister.cc
index 749678c8..4f6c5928 100644
--- a/common/rpackagelister.cc
+++ b/common/rpackagelister.cc
@@ -1952,13 +1952,13 @@ bool RPackageLister::addArchiveToCache(string archive, string &pkgname)
    pkgCache::VerIterator ver = dcache->GetCandidateVersion(*pkg->package());
    pkgCache::VerFileIterator Vf = ver.FileList(); 
    pkgRecords::Parser &Parse = _records->Lookup(Vf);
-   string MD5 = Parse.MD5Hash();
-   // then calc the md5 of the pkg
-   MD5Summation debMD5;
+   HashStringList hashes = Parse.Hashes();
+   // then calc the hashes of the pkg
+   Hashes debHashes(hashes);
    in.Seek(0);
-   debMD5.AddFD(in.Fd(),in.Size());
-   if(MD5 != debMD5.Result().Value()) {
-      cerr << "Ignoring " << pkgname << " MD5 does not match"<< endl;
+   debHashes.AddFD(in.Fd(),in.Size());
+   if(hashes != debHashes.GetHashStringList()) {
+      cerr << "Ignoring " << pkgname << " hashes does not match"<< endl;
       return false;
    }
       

From 0cd1b087c539b2d61406615c17cebc9e91193397 Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <juliank@ubuntu.com>
Date: Mon, 17 Jun 2019 16:03:54 +0200
Subject: [PATCH 15/17] Use apt-pkg's indexcopy, it was merged in 2005

---
 common/Makefile.am   |   2 -
 common/indexcopy.cc  | 511 -------------------------------------------
 common/indexcopy.h   |  74 -------
 common/rcdscanner.cc |   2 +-
 po/POTFILES.in       |   1 -
 5 files changed, 1 insertion(+), 589 deletions(-)
 delete mode 100644 common/indexcopy.cc
 delete mode 100644 common/indexcopy.h

diff --git a/common/Makefile.am b/common/Makefile.am
index 63bff376..5c7b501c 100644
--- a/common/Makefile.am
+++ b/common/Makefile.am
@@ -28,8 +28,6 @@ libsynaptic_a_SOURCES =\
 	rpackageview.h\
 	rcdscanner.cc\
 	rcdscanner.h\
-	indexcopy.cc\
-	indexcopy.h\
 	rpmindexcopy.cc \
 	rpmindexcopy.h \
 	raptoptions.cc \
diff --git a/common/indexcopy.cc b/common/indexcopy.cc
deleted file mode 100644
index c5183458..00000000
--- a/common/indexcopy.cc
+++ /dev/null
@@ -1,511 +0,0 @@
-// -*- mode: cpp; mode: fold -*-
-// Description                                                          /*{{{*/
-// $Id: indexcopy.cc,v 1.2 2002/07/25 18:07:18 niemeyer Exp $
-/* ######################################################################
-
-   Index Copying - Aid for copying and verifying the index files
-   
-   This class helps apt-cache reconstruct a damaged index files. 
-   
-   ##################################################################### */
-                                                                        /*}}} */
-// Include Files                                                        /*{{{*/
-#include "indexcopy.h"
-#include "i18n.h"
-
-#include <apt-pkg/error.h>
-#include <apt-pkg/progress.h>
-#include <apt-pkg/strutl.h>
-#include <apt-pkg/fileutl.h>
-#include <apt-pkg/configuration.h>
-#include <apt-pkg/tagfile.h>
-
-#include <iostream>
-#include <unistd.h>
-#include <sys/stat.h>
-#include <stdio.h>
-                                                                        /*}}} */
-
-using namespace std;
-
-// IndexCopy::CopyPackages - Copy the package files from the CD         /*{{{*/
-// ---------------------------------------------------------------------
-/* */
-bool IndexCopy::CopyPackages(string CDROM, string Name,
-                             vector<string> &List)
-{
-   if (List.size() == 0)
-      return true;
-
-   OpTextProgress Progress;
-
-   bool NoStat = _config->FindB("APT::CDROM::Fast", false);
-   bool Debug = _config->FindB("Debug::aptcdrom", false);
-
-   // Prepare the progress indicator
-   unsigned long TotalSize = 0;
-   for (vector<string>::iterator I = List.begin(); I != List.end(); I++) {
-      struct stat Buf;
-      if (stat(string(*I + GetFileName()).c_str(), &Buf) != 0 &&
-          stat(string(*I + GetFileName() + ".gz").c_str(), &Buf) != 0)
-         return _error->Errno("stat", _("Stat failed for %s"),
-                              string(*I + GetFileName()).c_str());
-      TotalSize += Buf.st_size;
-   }
-
-   unsigned long CurrentSize = 0;
-   unsigned int NotFound = 0;
-   unsigned int WrongSize = 0;
-   unsigned int Packages = 0;
-   for (vector<string>::iterator I = List.begin(); I != List.end(); I++) {
-      string OrigPath = string(*I, CDROM.length());
-      unsigned long FileSize = 0;
-
-      // Open the package file
-      FileFd Pkg;
-      if (FileExists(*I + GetFileName()) == true) {
-         Pkg.Open(*I + GetFileName(), FileFd::ReadOnly);
-         FileSize = Pkg.Size();
-      } else {
-         FileFd From(*I + GetFileName() + ".gz", FileFd::ReadOnly);
-         if (_error->PendingError() == true)
-            return false;
-         FileSize = From.Size();
-
-         // Get a temp file
-         FILE *tmp = tmpfile();
-         if (tmp == 0)
-            return _error->Errno("tmpfile", _("Unable to create a tmp file"));
-         Pkg.Fd(dup(fileno(tmp)));
-         fclose(tmp);
-
-         // Fork gzip
-         int Process = fork();
-         if (Process < 0)
-            return _error->Errno("fork",
-                                 "Internal Error: Couldn't fork gzip. Please report.");
-
-         // The child
-         if (Process == 0) {
-            dup2(From.Fd(), STDIN_FILENO);
-            dup2(Pkg.Fd(), STDOUT_FILENO);
-            SetCloseExec(STDIN_FILENO, false);
-            SetCloseExec(STDOUT_FILENO, false);
-
-            const char *Args[3];
-            string Tmp = _config->Find("Dir::bin::gzip", "gzip");
-            Args[0] = Tmp.c_str();
-            Args[1] = "-d";
-            Args[2] = 0;
-            execvp(Args[0], (char **)Args);
-            exit(100);
-         }
-         // Wait for gzip to finish
-         if (ExecWait
-             (Process, _config->Find("Dir::bin::gzip", "gzip").c_str(),
-              false) == false)
-            return _error->Error(_("gzip failed, perhaps the disk is full."));
-
-         Pkg.Seek(0);
-      }
-      pkgTagFile Parser(&Pkg);
-      if (_error->PendingError() == true)
-         return false;
-
-      // Open the output file
-      char S[400];
-      snprintf(S, sizeof(S), "cdrom:[%s]/%s%s", Name.c_str(),
-               (*I).c_str() + CDROM.length(), GetFileName());
-      string TargetF = _config->FindDir("Dir::State::lists") + "partial/";
-      TargetF += URItoFileName(S);
-      if (_config->FindB("APT::CDROM::NoAct", false) == true)
-         TargetF = "/dev/null";
-      FileFd Target(TargetF, FileFd::WriteEmpty);
-      FILE *TargetFl = fdopen(dup(Target.Fd()), "w");
-      if (_error->PendingError() == true)
-         return false;
-      if (TargetFl == 0)
-         return _error->Errno("fdopen", _("Failed to reopen fd"));
-
-      // Setup the progress meter
-      Progress.OverallProgress(CurrentSize, TotalSize, FileSize,
-                               string("Reading ") + Type() + " Indexes");
-
-      // Parse
-      Progress.SubProgress(Pkg.Size());
-      pkgTagSection Section;
-      this->Section = &Section;
-      string Prefix;
-      unsigned long Hits = 0;
-      unsigned long Chop = 0;
-      while (Parser.Step(Section) == true) {
-         Progress.Progress(Parser.Offset());
-         string File;
-         unsigned long Size;
-         if (GetFile(File, Size) == false) {
-            fclose(TargetFl);
-            return false;
-         }
-
-         if (Chop != 0)
-            File = OrigPath + ChopDirs(File, Chop);
-
-         // See if the file exists
-         bool Mangled = false;
-         if (NoStat == false || Hits < 10) {
-            // Attempt to fix broken structure
-            if (Hits == 0) {
-               if (ReconstructPrefix(Prefix, OrigPath, CDROM, File) == false &&
-                   ReconstructChop(Chop, *I, File) == false) {
-                  if (Debug == true)
-                     clog << "Missed: " << File << endl;
-                  NotFound++;
-                  continue;
-               }
-               if (Chop != 0)
-                  File = OrigPath + ChopDirs(File, Chop);
-            }
-            // Get the size
-            struct stat Buf;
-            if (stat(string(CDROM + Prefix + File).c_str(), &Buf) != 0 ||
-                Buf.st_size == 0) {
-               // Attempt to fix busted symlink support for one instance
-               string OrigFile = File;
-               string::size_type Start = File.find("binary-");
-               string::size_type End = File.find("/", Start + 3);
-               if (Start != string::npos && End != string::npos) {
-                  File.replace(Start, End - Start, "binary-all");
-                  Mangled = true;
-               }
-
-               if (Mangled == false ||
-                   stat(string(CDROM + Prefix + File).c_str(), &Buf) != 0) {
-                  if (Debug == true)
-                     clog << "Missed(2): " << OrigFile << endl;
-                  NotFound++;
-                  continue;
-               }
-            }
-            // Size match
-            if ((unsigned)Buf.st_size != Size) {
-               if (Debug == true)
-                  clog << "Wrong Size: " << File << endl;
-               WrongSize++;
-               continue;
-            }
-         }
-
-         Packages++;
-         Hits++;
-
-         if (RewriteEntry(TargetFl, File) == false) {
-            fclose(TargetFl);
-            return false;
-         }
-      }
-      fclose(TargetFl);
-
-      if (Debug == true)
-         cout << " Processed by using Prefix '" << Prefix <<
-            "' and chop " << Chop << endl;
-
-      if (_config->FindB("APT::CDROM::NoAct", false) == false) {
-         // Move out of the partial directory
-         Target.Close();
-         string FinalF = _config->FindDir("Dir::State::lists");
-         FinalF += URItoFileName(S);
-         if (rename(TargetF.c_str(), FinalF.c_str()) != 0)
-            return _error->Errno("rename", _("Failed to rename"));
-
-         // Copy the release file
-         snprintf(S, sizeof(S), "cdrom:[%s]/%sRelease", Name.c_str(),
-                  (*I).c_str() + CDROM.length());
-         string TargetF = _config->FindDir("Dir::State::lists") + "partial/";
-         TargetF += URItoFileName(S);
-         if (FileExists(*I + "Release") == true) {
-            FileFd Target(TargetF, FileFd::WriteEmpty);
-            FileFd Rel(*I + "Release", FileFd::ReadOnly);
-            if (_error->PendingError() == true)
-               return false;
-
-            if (CopyFile(Rel, Target) == false)
-               return false;
-         } else {
-            // Empty release file
-            FileFd Target(TargetF, FileFd::WriteEmpty);
-         }
-
-         // Rename the release file
-         FinalF = _config->FindDir("Dir::State::lists");
-         FinalF += URItoFileName(S);
-         if (rename(TargetF.c_str(), FinalF.c_str()) != 0)
-            return _error->Errno("rename", _("Failed to rename"));
-      }
-
-      /* Mangle the source to be in the proper notation with
-         prefix dist [component] */
-      *I = string(*I, Prefix.length());
-      ConvertToSourceList(CDROM, *I);
-      *I = Prefix + ' ' + *I;
-
-      CurrentSize += FileSize;
-   }
-   Progress.Done();
-
-   // Some stats
-   cout << "Wrote " << Packages << " records";
-   if (NotFound != 0)
-      cout << " with " << NotFound << " missing files";
-   if (NotFound != 0 && WrongSize != 0)
-      cout << " and";
-   if (WrongSize != 0)
-      cout << " with " << WrongSize << " mismatched files";
-   cout << '.' << endl;
-
-   if (Packages == 0)
-      _error->Warning(_("No valid records were found."));
-
-   if (NotFound + WrongSize > 10)
-      cout << "Alot of entries were discarded, something may be wrong." <<
-         endl;
-
-   return true;
-}
-
-                                                                        /*}}} */
-// IndexCopy::ChopDirs - Chop off the leading directory components      /*{{{*/
-// ---------------------------------------------------------------------
-/* */
-string IndexCopy::ChopDirs(string Path, unsigned int Depth)
-{
-   string::size_type I = 0;
-   do {
-      I = Path.find('/', I + 1);
-      Depth--;
-   }
-   while (I != string::npos && Depth != 0);
-
-   if (I == string::npos)
-      return string();
-
-   return string(Path, I + 1);
-}
-
-                                                                        /*}}} */
-// IndexCopy::ReconstructPrefix - Fix strange prefixing                 /*{{{*/
-// ---------------------------------------------------------------------
-/* This prepends dir components from the path to the package files to
-   the path to the deb until it is found */
-bool IndexCopy::ReconstructPrefix(string &Prefix, string OrigPath, string CD,
-                                  string File)
-{
-   bool Debug = _config->FindB("Debug::aptcdrom", false);
-   unsigned int Depth = 1;
-   string MyPrefix = Prefix;
-   while (1) {
-      struct stat Buf;
-      if (stat(string(CD + MyPrefix + File).c_str(), &Buf) != 0) {
-         if (Debug == true)
-            cout << "Failed, " << CD + MyPrefix + File << endl;
-         if (GrabFirst(OrigPath, MyPrefix, Depth++) == true)
-            continue;
-
-         return false;
-      } else {
-         Prefix = MyPrefix;
-         return true;
-      }
-   }
-   return false;
-}
-
-                                                                        /*}}} */
-// IndexCopy::ReconstructChop - Fixes bad source paths                  /*{{{*/
-// ---------------------------------------------------------------------
-/* This removes path components from the filename and prepends the location
-   of the package files until a file is found */
-bool IndexCopy::ReconstructChop(unsigned long &Chop, string Dir, string File)
-{
-   // Attempt to reconstruct the filename
-   unsigned long Depth = 0;
-   while (1) {
-      struct stat Buf;
-      if (stat(string(Dir + File).c_str(), &Buf) != 0) {
-         File = ChopDirs(File, 1);
-         Depth++;
-         if (File.empty() == false)
-            continue;
-         return false;
-      } else {
-         Chop = Depth;
-         return true;
-      }
-   }
-   return false;
-}
-
-                                                                        /*}}} */
-// IndexCopy::ConvertToSourceList - Convert a Path to a sourcelist      /*{{{*/
-// ---------------------------------------------------------------------
-/* We look for things in dists/ notation and convert them to 
-   <dist> <component> form otherwise it is left alone. This also strips
-   the CD path. 
- 
-   This implements a regex sort of like: 
-    (.*)/dists/([^/]*)/(.*)/binary-* 
-     ^          ^      ^- Component
-     |          |-------- Distribution
-     |------------------- Path
-   
-   It was deciced to use only a single word for dist (rather than say
-   unstable/non-us) to increase the chance that each CD gets a single
-   line in sources.list.
- */
-void IndexCopy::ConvertToSourceList(string CD, string &Path)
-{
-   char S[300];
-   snprintf(S, sizeof(S), "binary-%s",
-            _config->Find("Apt::Architecture").c_str());
-
-   // Strip the cdrom base path
-   Path = string(Path, CD.length());
-   if (Path.empty() == true)
-      Path = "/";
-
-   // Too short to be a dists/ type
-   if (Path.length() < strlen("dists/"))
-      return;
-
-   // Not a dists type.
-   if (stringcmp(Path.c_str(), Path.c_str() + strlen("dists/"), "dists/") != 0)
-      return;
-
-   // Isolate the dist
-   string::size_type Slash = strlen("dists/");
-   string::size_type Slash2 = Path.find('/', Slash + 1);
-   if (Slash2 == string::npos || Slash2 + 2 >= Path.length())
-      return;
-   string Dist = string(Path, Slash, Slash2 - Slash);
-
-   // Isolate the component
-   Slash = Slash2;
-   for (unsigned I = 0; I != 10; I++) {
-      Slash = Path.find('/', Slash + 1);
-      if (Slash == string::npos || Slash + 2 >= Path.length())
-         return;
-      string Comp = string(Path, Slash2 + 1, Slash - Slash2 - 1);
-
-      // Verify the trailing binary- bit
-      string::size_type BinSlash = Path.find('/', Slash + 1);
-      if (Slash == string::npos)
-         return;
-      string Binary = string(Path, Slash + 1, BinSlash - Slash - 1);
-
-      if (Binary != S && Binary != "source")
-         continue;
-
-      Path = Dist + ' ' + Comp;
-      return;
-   }
-}
-
-                                                                        /*}}} */
-// IndexCopy::GrabFirst - Return the first Depth path components        /*{{{*/
-// ---------------------------------------------------------------------
-/* */
-bool IndexCopy::GrabFirst(string Path, string &To, unsigned int Depth)
-{
-   string::size_type I = 0;
-   do {
-      I = Path.find('/', I + 1);
-      Depth--;
-   }
-   while (I != string::npos && Depth != 0);
-
-   if (I == string::npos)
-      return false;
-
-   To = string(Path, 0, I + 1);
-   return true;
-}
-
-                                                                        /*}}} */
-// PackageCopy::GetFile - Get the file information from the section     /*{{{*/
-// ---------------------------------------------------------------------
-/* */
-bool PackageCopy::GetFile(string &File, unsigned long &Size)
-{
-   File = Section->FindS("Filename");
-   Size = Section->FindI("Size");
-   if (File.empty() || Size == 0)
-      return _error->Error(_("Cannot find filename or size tag"));
-   return true;
-}
-
-                                                                        /*}}} */
-// PackageCopy::RewriteEntry - Rewrite the entry with a new filename    /*{{{*/
-// ---------------------------------------------------------------------
-/* */
-bool PackageCopy::RewriteEntry(FILE *Target, string File)
-{
-   TFRewriteData Changes[] = { {"Filename", File.c_str()}
-   ,
-   {}
-   };
-
-   if (TFRewrite(Target, *Section, TFRewritePackageOrder, Changes) == false)
-      return false;
-   fputc('\n', Target);
-   return true;
-}
-
-                                                                        /*}}} */
-// SourceCopy::GetFile - Get the file information from the section      /*{{{*/
-// ---------------------------------------------------------------------
-/* */
-bool SourceCopy::GetFile(string &File, unsigned long &Size)
-{
-   string Files = Section->FindS("Files");
-   if (Files.empty() == true)
-      return false;
-
-   // Stash the / terminated directory prefix
-   string Base = Section->FindS("Directory");
-   if (Base.empty() == false && Base[Base.length() - 1] != '/')
-      Base += '/';
-
-   // Read the first file triplet
-   const char *C = Files.c_str();
-   string sSize;
-   string MD5Hash;
-
-   // Parse each of the elements
-   if (ParseQuoteWord(C, MD5Hash) == false ||
-       ParseQuoteWord(C, sSize) == false || ParseQuoteWord(C, File) == false)
-      return _error->Error(_("Error parsing file record"));
-
-   // Parse the size and append the directory
-   Size = atoi(sSize.c_str());
-   File = Base + File;
-   return true;
-}
-
-                                                                        /*}}} */
-// SourceCopy::RewriteEntry - Rewrite the entry with a new filename     /*{{{*/
-// ---------------------------------------------------------------------
-/* */
-bool SourceCopy::RewriteEntry(FILE *Target, string File)
-{
-   string Dir(File, 0, File.rfind('/'));
-   TFRewriteData Changes[] = { {"Directory", Dir.c_str()}
-   ,
-   {}
-   };
-
-   if (TFRewrite(Target, *Section, TFRewriteSourceOrder, Changes) == false)
-      return false;
-   fputc('\n', Target);
-   return true;
-}
-
-                                                                        /*}}} */
diff --git a/common/indexcopy.h b/common/indexcopy.h
deleted file mode 100644
index c48877f1..00000000
--- a/common/indexcopy.h
+++ /dev/null
@@ -1,74 +0,0 @@
-// -*- mode: cpp; mode: fold -*-
-// Description                                                          /*{{{*/
-// $Id: indexcopy.h,v 1.1 2002/07/23 17:54:52 niemeyer Exp $
-/* ######################################################################
-
-   Index Copying - Aid for copying and verifying the index files
-
-   ##################################################################### */
-                                                                        /*}}} */
-#ifndef INDEXCOPY_H
-#define INDEXCOPY_H
-
-#include <vector>
-#include <string>
-#include <stdio.h>
-
-using std::string;
-using std::vector;
-
-class pkgTagSection;
-class FileFd;
-
-class IndexCopy {
- protected:
-
-   pkgTagSection *Section;
-
-   string ChopDirs(string Path, unsigned int Depth);
-   bool ReconstructPrefix(string &Prefix, string OrigPath, string CD,
-                          string File);
-   bool ReconstructChop(unsigned long &Chop, string Dir, string File);
-   void ConvertToSourceList(string CD, string &Path);
-   bool GrabFirst(string Path, string &To, unsigned int Depth);
-   virtual bool GetFile(string &Filename, unsigned long &Size) = 0;
-   virtual bool RewriteEntry(FILE *Target, string File) = 0;
-   virtual const char *GetFileName() = 0;
-   virtual const char *Type() = 0;
-
- public:
-
-   bool CopyPackages(string CDROM, string Name, vector<string> &List);
-};
-
-class PackageCopy:public IndexCopy {
- protected:
-
-   virtual bool GetFile(string &Filename, unsigned long &Size);
-   virtual bool RewriteEntry(FILE *Target, string File);
-   virtual const char *GetFileName() {
-      return "Packages";
-   }
-   virtual const char *Type() {
-      return "Package";
-   }
-
- public:
-};
-
-class SourceCopy:public IndexCopy {
- protected:
-
-   virtual bool GetFile(string &Filename, unsigned long &Size);
-   virtual bool RewriteEntry(FILE *Target, string File);
-   virtual const char *GetFileName() {
-      return "Sources";
-   }
-   virtual const char *Type() {
-      return "Source";
-   }
-
- public:
-};
-
-#endif
diff --git a/common/rcdscanner.cc b/common/rcdscanner.cc
index 33477c46..537fd087 100644
--- a/common/rcdscanner.cc
+++ b/common/rcdscanner.cc
@@ -44,7 +44,7 @@
 #ifdef HAVE_RPM
 #include "rpmindexcopy.h"
 #else
-#include "indexcopy.h"
+#include <apt-pkg/indexcopy.h>
 #endif
 
 using namespace std;
diff --git a/po/POTFILES.in b/po/POTFILES.in
index 034e07a4..825e490a 100644
--- a/po/POTFILES.in
+++ b/po/POTFILES.in
@@ -1,6 +1,5 @@
 [encoding: UTF-8]
 common/sections_trans.cc
-common/indexcopy.cc
 #common/pkg_acqfile.cc
 common/rcacheactor.cc
 common/rcdscanner.cc

From 6f83e5696031efcc0e1422f5de855ea62d9969ef Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <juliank@ubuntu.com>
Date: Mon, 17 Jun 2019 16:10:06 +0200
Subject: [PATCH 16/17] CurrentSize and co in acquire worker moved to
 CurrentItem

---
 gtk/rgfetchprogress.cc | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/gtk/rgfetchprogress.cc b/gtk/rgfetchprogress.cc
index c9e14f04..fb1f73ef 100644
--- a/gtk/rgfetchprogress.cc
+++ b/gtk/rgfetchprogress.cc
@@ -324,11 +324,17 @@ bool RGFetchProgress::Pulse(pkgAcquire *Owner)
 
       if (I->CurrentItem == 0)
          continue;
-
+#if APT_PKG_ABI >= 590
+      if (I->CurrentItem->TotalSize > 0)
+         updateStatus(*I->CurrentItem,
+                      long (double (I->CurrentItem->CurrentSize * 100.0) /
+                            double (I->CurrentItem->TotalSize)));
+#else
       if (I->TotalSize > 0)
          updateStatus(*I->CurrentItem,
                       long (double (I->CurrentSize * 100.0) /
                             double (I->TotalSize)));
+#endif
       else
          updateStatus(*I->CurrentItem, 100);
 

From 432f2e58a2d071818301043e3bb95c83be083c3b Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <juliank@ubuntu.com>
Date: Mon, 17 Jun 2019 16:14:34 +0200
Subject: [PATCH 17/17] configure.in: apt-inst is optional

---
 configure.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.in b/configure.in
index 922b4ef1..df28e92a 100644
--- a/configure.in
+++ b/configure.in
@@ -116,7 +116,7 @@ AM_CONDITIONAL(WITH_LUA, test "$enable_scripts" != "no")
 
 DEB_HDRS=""
 AC_SUBST(DEB_HDRS)
-DEB_LIBS="-lapt-inst"
+AC_CHECK_LIB(apt-inst, main, [DEB_LIBS=-lapt-inst], [DEB_LIBS=])
 AC_SUBST(DEB_LIBS)
 
 dnl Checks for header files.
