diff -Naur tree-1.7.0-orig/Makefile tree-1.7.0/Makefile
--- tree-1.7.0-orig/Makefile	2014-04-18 19:35:39.000000000 +0200
+++ tree-1.7.0/Makefile	2014-06-08 13:59:27.905865981 +0200
@@ -24,7 +24,7 @@
 TREE_DEST=tree
 BINDIR=${prefix}/bin
 MAN=tree.1
-MANDIR=${prefix}/man/man1
+MANDIR=${prefix}/share/man/man1
 OBJS=tree.o unix.o html.o xml.o json.o hash.o color.o file.o
 
 # Uncomment options below for your particular OS:
