Description: Fix some misspellings.
Forwarded: no
Author: Peter Pentchev <roam@ringlet.net>
Last-Update: 2010-11-25

--- a/fxc/src/messages.c
+++ b/fxc/src/messages.c
@@ -40,7 +40,7 @@
 #define MSG_USING               _("Usage: %s [options] filename\n\n")
 #define MSG_OPTION_D            _("   -d       Debugging mode\n")
 #define MSG_OPTIONS             _("   -i dir   Adds the directory to the PATH\n" \
-                                "   -a       Automaticaly adds all files to the DCB\n"  \
+                                "   -a       Automatically adds all files to the DCB\n"  \
                                 "   -f file  Adds a single file to the DCB\n"  \
 								"   -l lang  Specify locale settings\n"  \
 								"   -s stub  Generate a stubbed executable from the given stub\n"  \
@@ -77,7 +77,7 @@
 #define MSG_NOT_AN_ARRAY        _("Not an array or struct array")
 #define MSG_BOUND               _("Index out of range")
 #define MSG_IDENTIFIER_EXP      _("Identifier expected")
-#define MSG_NOT_AN_LVALUE       _("Can't get the address of an inmediate value")
+#define MSG_NOT_AN_LVALUE       _("Can't get the address of an immediate value")
 #define MSG_NOT_A_POINTER       _("Pointer required")
 #define MSG_VARIABLE_REQUIRED   _("Variable required")
 #define MSG_STRUCT_REQUIRED     _("Struct required")
--- a/map/map.c
+++ b/map/map.c
@@ -135,7 +135,7 @@
 		"    -s      Modify MAP file parameters\n"
 		"    -n      Remove MAP file transparency\n"
  		"    -z=#    PNG transparent color ('0' color) range\n"
-		"    -#      Compresion level (0-9 NOTE: 0 DIV compatible for 8bpps MAP files)\n"
+		"    -#      Compression level (0-9 NOTE: 0 DIV compatible for 8bpps MAP files)\n"
 		"\n"
 		"Options for -l/-m/-c: (# means integer value)\n"
 		"\n"
