Description: Fix the build with giflib5
Forwarded: no
Author: Matthias Klose <doko@debian.org>
Last-Update: 2015-12-17

Index: b/map/map.c
===================================================================
--- a/map/map.c
+++ b/map/map.c
@@ -222,7 +222,7 @@ MAP * load_gif (const char * filename)
 	MAP * map ;
 	ColorMapObject * pal ;
 
-	file = DGifOpenFileName (filename) ;
+	file = DGifOpenFileName (filename, NULL) ;
 	if (!file) return NULL ;
 	DGifSlurp (file) ;
 
