Description: Honor CFLAGS passed via the environment.
Forwarded: no
Author: Peter Pentchev <roam@ringlet.net>
Last-Update: 2015-12-18

--- a/configure.in
+++ b/configure.in
@@ -84,11 +84,14 @@
 
 AC_CHECK_LIB(m,fabs,LIBS="$LIBS",AC_MSG_RESULT(math library not found))
 
-if test "x$enable_debug" = "xyes"; then
-	CFLAGS="-g $INCLUDES"
-else
-	CFLAGS="-O2 -finline-functions -ffast-math $INCLUDES"
+if test -z "$CFLAGS"; then
+	if test "x$enable_debug" = "xyes"; then
+		CFLAGS="-g"
+	else
+		CFLAGS="-O2 -finline-functions -ffast-math"
+	fi
 fi
+CFLAGS="$CFLAGS $INCLUDES"
 
 dnl librerias para todos
 
