diff -Naur orig/xye.desktop mod/xye.desktop
--- orig/xye.desktop	2013-08-13 20:29:58.000000000 +0200
+++ mod/xye.desktop	2016-04-22 21:09:48.582349981 +0200
@@ -1,14 +1,14 @@
 #!/usr/bin/env xdg-open
 
 [Desktop Entry]
-Encoding=UTF-8
 Exec=xye
 Type=Application
-Categories=Application;Game;PuzzleGame
+Categories=Game;LogicGame;
 Name=Xye
 GenericName=Xye
 Comment=Solve puzzles and action challenges in order to get all the gems in the room.
+Comment[hu]=Gyémánt tologató logikai játék.
 Terminal=false
 StartupNotify=false
-Icon=xye.png
+Icon=xye
 MimeType=application/xye-xml
