Description: Do not unintentionally declare enum variables.
Forwarded: no
Author: Peter Pentchev <roam@ringlet.net>
Last-Update: 2020-07-23

--- a/cnee/src/cnee_demo.c
+++ b/cnee/src/cnee_demo.c
@@ -33,11 +33,11 @@
 #include "libxnee/xnee_xinput.h"
 
 #include "parse.h"
+#include "cnee.h"
 #include "cnee_strings.h"
 
 
 #define CNEE_DEMO_DELAYED_START 3
-xnee_option_t *cnee_options ;
 
 static char *
 cnee_get_default_filename()
--- a/cnee/test/src/xgetter.c
+++ b/cnee/test/src/xgetter.c
@@ -5,14 +5,14 @@
 #include <X11/Xlib.h>
 
 
-enum 
+enum xy_mask 
 {
   X   = 0,
   Y   = 2, 
   ALL = 4,
   ROOT= 8,
   WIN = 16
-} xy_mask;
+} ;
 
 int xg_check( char *arg, char *long_arg , char *short_arg ) 
 {
--- a/libxnee/include/libxnee/xnee.h
+++ b/libxnee/include/libxnee/xnee.h
@@ -102,7 +102,7 @@
     XNEE_REPLAY_BACKEND_FAILURE    ,
     XNEE_OVERRIDE_DISPLAY_FAILURE  ,
     XNEE_LAST_ERROR
-  } _return_values;
+  } ;
 
 
 
@@ -112,7 +112,7 @@
     XNEE_BOOL_IMPLICIT_TRUE,
     XNEE_BOOL_EXPLICIT_TRUE,
     XNEE_BOOL_ERROR
-  } _bool_string_values;
+  } ;
 
 enum xnee_protocol_data_numbers
   {
@@ -125,14 +125,14 @@
     XNEE_PROTO_XINPUT_EVENT_MASTER,
     XNEE_PROTO_XINPUT_EVENT_SLAVE,
     XNEE_PROTO_LAST    
-  } _xnee_protocol_data_numbers;
+  } ;
   
 enum xnee_replay_backend
   {
     XNEE_REPLAY_XNEE = 0,
     XNEE_REPLAY_SWINPUT,
     XNEE_REPLAY_LAST
-  } _xnee_replay_backend;
+  } ;
 
 /** 
  * \brief simply a X error. 
@@ -228,7 +228,7 @@
   char *data_descr;
 };
 
-enum
+enum xnee_option_type
   {
     XNEE_ANY_OPTION,
     XNEE_GENERAL_OPTION,
@@ -240,15 +240,15 @@
     XNEE_INTERNAL_OPTION,
     XNEE_OBSOLETE_OPTION,
     XNEE_RETYPE_OPTION
-  } xnee_option_type;
+  } ;
 
 
-enum
+enum xnee_override_display
   {
     XNEE_OVERRIDE_DISPLAY_NONE,
     XNEE_OVERRIDE_DISPLAY_DATA,
     XNEE_OVERRIDE_DISPLAY_CONTROL
-  } xnee_override_display;
+  } ;
 
 typedef struct
 {
--- a/libxnee/include/libxnee/xnee_grab.h
+++ b/libxnee/include/libxnee/xnee_grab.h
@@ -46,7 +46,7 @@
 xnee_grab_all_keys (xnee_data* xd);
 
 
-enum 
+enum xnee_grab_modifiers_control
   {
     XNEE_GRAB_NONE_PRESSED=0,
     XNEE_GRAB_NONE_IN_USE,
@@ -64,7 +64,7 @@
     XNEE_GRAB_KM_PRESS,
     XNEE_GRAB_KM_RELEASE,
     XNEE_GRAB_MOUSE
-  } xnee_grab_modifiers_control ;
+  } ;
   
 int
 xnee_save_or_print(xnee_data *xd, KeyCode kc, int mode);
--- a/libxnee/include/libxnee/xnee_internal.h
+++ b/libxnee/include/libxnee/xnee_internal.h
@@ -82,7 +82,7 @@
 
 
 /* ** Do NOT edit ** */
-enum _xnee_data_types {
+enum xnee_data_types {
   XNEE_EVENT  = 0,
   XNEE_REQUEST  ,
   XNEE_REPLY             ,
@@ -104,17 +104,17 @@
   XNEE_NEW_WINDOW_DATA,
   XNEE_PREDEF_EVENTS,
   XNEE_NO_DATA          
-} xnee_data_types ;
+} ;
 
 /* ** Do NOT edit ** */
-enum _xnee_mode {
+enum xnee_mode {
   XNEE_NOMODE    = 0,
   XNEE_REPLAYER     ,
   XNEE_RECORDER     ,
   XNEE_RETYPER      ,        
   XNEE_DISTRIBUTOR  ,        
   XNEE_SYNTAX_CHECKER        
-} xnee_mode ;
+} ;
 
 
 
@@ -206,7 +206,7 @@
     XNEE_RESOLUTION_UNSET  = -1,
     XNEE_RESOLUTION_USED   =  0,
     XNEE_RESOLUTION_UNUSED =  1
-  } _xnee_resolution_states;
+  } ;
 
 /* 
  * Grab modes/actions
@@ -222,7 +222,7 @@
     XNEE_GRAB_NODATA = 0,
     XNEE_GRAB_SET    = 1,
     XNEE_GRAB_UNKOWN  = 15
-  } _xnee_grab_modes;
+  } ;
 
 /*
  *  continue_process commnd enum
@@ -233,7 +233,7 @@
     XNEE_PROCESS_INC       ,
     XNEE_PROCESS_DEC       ,
     XNEE_PROCESS_GET
-  } _cont_proc_commands;
+  } ;
 
 typedef void (*callback_ptr)( XPointer, XRecordInterceptData *); 
 typedef callback_ptr *callback_ptrptr;                                         
--- a/libxnee/include/libxnee/xnee_window.h
+++ b/libxnee/include/libxnee/xnee_window.h
@@ -42,11 +42,11 @@
 } xnee_win_pos;
 
 
-enum
+enum xnee_window_types
   {
     XNEE_WINDOW_RECEIVED,
     XNEE_WINDOW_SESSION
-  } xnee_window_types;
+  } ;
 
 
 int
--- a/pnee/src/callbacks.c
+++ b/pnee/src/callbacks.c
@@ -52,12 +52,12 @@
 gchar     *display_string ;
 extern xnee_data *xd ;
 
-enum {
+enum pnee_files {
   PNEE_FILE_NONE   ,
   PNEE_FILE_RECORD ,
   PNEE_FILE_REPLAY ,
   PNEE_FILE_VERBOSE
-} pnee_files;
+} ;
 
 
 int        file_choosen = PNEE_FILE_NONE;
