diff -Nur orig/scite/gtk/SciTE.desktop mod/scite/gtk/SciTE.desktop
--- orig/scite/gtk/SciTE.desktop	2017-10-22 23:48:09.000000000 +0200
+++ mod/scite/gtk/SciTE.desktop	2018-01-09 21:57:56.583233055 +0100
@@ -1,6 +1,6 @@
 [Desktop Entry]
-Version=1.0
 Name=SciTE Text Editor
+Name[hu]=SciTE szövegszerkesztő
 Name[da]=SciTE-tekstredigering
 Name[es]=Editor de textos SciTE
 Name[fr]=Éditeur de texte SciTE
@@ -10,6 +10,8 @@
 Comment[es]=Edita especializado para programación
 Comment[fr]=Éditer des fichiers sources
 Comment[ru]=Редактор исходных кодов
+Comment[hu]=SciTE fejlesztői szövegszerkesztő
+GenericName[hu]=Szövegszerkesztő
 GenericName=Text Editor
 Type=Application
 Exec=SciTE %F
