Subject: Improve help output
From: Alkis Georgopoulos <alkisg@gmail.com>

--- a/main.c
+++ b/main.c
@@ -54,7 +54,7 @@
     printf( "NumLockX " VERSION "\n"
         "(C) 2000-2001 Lubos Lunak <l.lunak@kde.org>\n"
         "(C) 2001      Oswald Buddenhagen <ossi@kde.org>\n\n"
-        "Usage : %s [on|off]\n"
+        "Usage: %s [on|off|toggle]\n"
         "on     - turns NumLock on in X ( default )\n"
         "off    - turns NumLock off in X\n"
         "toggle - toggles the NumLock on and off in X\n"
