diff -Nur orig/translations/tetzle_hu.ts mod/translations/tetzle_hu.ts
--- orig/translations/tetzle_hu.ts	1970-01-01 01:00:00.000000000 +0100
+++ mod/translations/tetzle_hu.ts	2019-03-20 13:26:33.380127269 +0100
@@ -0,0 +1,463 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!DOCTYPE TS>
+<TS version="2.1" language="hu">
+<context>
+    <name>AddImage</name>
+    <message>
+        <source>Open Image</source>
+        <translation>Kép megnyitás</translation>
+    </message>
+</context>
+<context>
+    <name>AppearanceDialog</name>
+    <message>
+        <source>Appearance</source>
+        <translation>Megjelenés</translation>
+    </message>
+    <message>
+        <source>Options</source>
+        <translation>Opciók</translation>
+    </message>
+    <message>
+        <source>Beveled borders</source>
+        <translation>Simított szélek</translation>
+    </message>
+    <message>
+        <source>Drop shadows</source>
+        <translation>Árnyéknélkül</translation>
+    </message>
+    <message>
+        <source>Colors</source>
+        <translation>Színek</translation>
+    </message>
+    <message>
+        <source>Background:</source>
+        <translation>Háttér:</translation>
+    </message>
+    <message>
+        <source>Shadow:</source>
+        <translation>Árnyék:</translation>
+    </message>
+    <message>
+        <source>Highlight:</source>
+        <translation>Kiemelés:</translation>
+    </message>
+</context>
+<context>
+    <name>Board</name>
+    <message>
+        <source>Error</source>
+        <translation>Hiba</translation>
+    </message>
+    <message>
+        <source>Missing image.</source>
+        <translation>Nincs kép.</translation>
+    </message>
+    <message>
+        <source>Please Wait</source>
+        <translation>Várjunk</translation>
+    </message>
+    <message>
+        <source>Loading image...</source>
+        <translation>Képbetöltés...</translation>
+    </message>
+    <message>
+        <source>Generating puzzle...</source>
+        <translation>Puzzle generálása...</translation>
+    </message>
+    <message>
+        <source>Creating pieces...</source>
+        <translation>Elemek készítése...</translation>
+    </message>
+    <message>
+        <source>Loading puzzle...</source>
+        <translation>Puzzle betöltése...</translation>
+    </message>
+    <message>
+        <source>Unknown data format</source>
+        <translation>Ismeretlen formátum</translation>
+    </message>
+    <message>
+        <source>Unknown element &apos;%1&apos;</source>
+        <translation>A &apos;%1&apos; elem ismeretlen</translation>
+    </message>
+    <message>
+        <source>Error parsing XML file.
+
+%1</source>
+        <translation>XML fájl elemzési hiba.
+
+%1</translation>
+    </message>
+    <message>
+        <source>Loading pieces...</source>
+        <translation>Elemek betöltése...</translation>
+    </message>
+    <message>
+        <source>Retrieving pieces...</source>
+        <translation>Elemek keverése...</translation>
+    </message>
+    <message>
+        <source>Placing pieces...</source>
+        <translation>Elemek elhelyezése...</translation>
+    </message>
+    <message>
+        <source>Success</source>
+        <translation>Kész</translation>
+    </message>
+</context>
+<context>
+    <name>ChooseGameDialog</name>
+    <message>
+        <source>Choose Game</source>
+        <translation>Játék választás</translation>
+    </message>
+    <message>
+        <source>Current Games</source>
+        <translation>Folyó játékok</translation>
+    </message>
+    <message>
+        <source>New Game</source>
+        <translation>Új játék</translation>
+    </message>
+</context>
+<context>
+    <name>ImagePropertiesDialog</name>
+    <message>
+        <source>Image Properties</source>
+        <translation>Képtulajdonságok</translation>
+    </message>
+    <message>
+        <source>Name:</source>
+        <translation>Név:</translation>
+    </message>
+    <message>
+        <source>Tags:</source>
+        <translation>Osztályok:</translation>
+    </message>
+</context>
+<context>
+    <name>LocaleDialog</name>
+    <message>
+        <source>Select application language:</source>
+        <translation>Nyelv választás:</translation>
+    </message>
+    <message>
+        <source>&lt;System Language&gt;</source>
+        <translation>&lt;Rendszer Nyelve&gt;</translation>
+    </message>
+    <message>
+        <source>Note</source>
+        <translation>Jegyzet</translation>
+    </message>
+    <message>
+        <source>Please restart this application for the change in language to take effect.</source>
+        <translation>Új nyelv választás majd újraindítás után érvényesül.</translation>
+    </message>
+</context>
+<context>
+    <name>NewGameTab</name>
+    <message>
+        <source>Add Image</source>
+        <translation>Kép hozzáadás</translation>
+    </message>
+    <message>
+        <source>Remove Image</source>
+        <translation>Kép eltávolítás</translation>
+    </message>
+    <message>
+        <source>Image Properties</source>
+        <translation>Képtulajdonságok</translation>
+    </message>
+    <message>
+        <source>%L1 pieces</source>
+        <translation>%L1 elem</translation>
+    </message>
+    <message>
+        <source>Copying images...</source>
+        <translation>Kép másolások...</translation>
+    </message>
+    <message>
+        <source>Cancel</source>
+        <translation>Mégse</translation>
+    </message>
+    <message>
+        <source>Remove selected image?</source>
+        <translation>Kijelölt kép eltávolítása?</translation>
+    </message>
+    <message>
+        <source>Remove selected image?
+
+There are saved games using this image that will also be removed.</source>
+        <translation>Eltávolítjuk a képet?
+
+Minden mentett játék, mely ezt használta, szintén törölve lesz.</translation>
+    </message>
+    <message>
+        <source>Untitled</source>
+        <translation>Névtelen</translation>
+    </message>
+</context>
+<context>
+    <name>OpenGameTab</name>
+    <message>
+        <source>Untitled</source>
+        <translation>Névtelen</translation>
+    </message>
+    <message>
+        <source>%L1 pieces %2 %3% complete</source>
+        <translation>%L1 elem %2 %3% kész</translation>
+    </message>
+    <message>
+        <source>Delete</source>
+        <translation>Törlés</translation>
+    </message>
+    <message>
+        <source>Delete Game</source>
+        <translation>Játék törlés</translation>
+    </message>
+    <message>
+        <source>Delete selected game?</source>
+        <translation>Kijelölt játék eltávolítása?</translation>
+    </message>
+</context>
+<context>
+    <name>Overview</name>
+    <message>
+        <source>Overview</source>
+        <translation>Áttekíttés</translation>
+    </message>
+</context>
+<context>
+    <name>TagManager</name>
+    <message>
+        <source>Add Tag</source>
+        <translation>Osztály hozzáadás</translation>
+    </message>
+    <message>
+        <source>Remove Tag</source>
+        <translation>Osztály eltávolítás</translation>
+    </message>
+    <message>
+        <source>Untagged</source>
+        <translation>Besorolatlan</translation>
+    </message>
+    <message>
+        <source>All Images</source>
+        <translation>Minden kép</translation>
+    </message>
+    <message>
+        <source>Untitled %1</source>
+        <translation>Névtelen %1</translation>
+    </message>
+    <message>
+        <source>Question</source>
+        <translation>Kérdés</translation>
+    </message>
+    <message>
+        <source>Remove selected tag?</source>
+        <translation>Kijelölt osztály eltávolítása?</translation>
+    </message>
+    <message>
+        <source>Sorry</source>
+        <translation>Sajnálatos</translation>
+    </message>
+    <message>
+        <source>A tag with that name already exists.</source>
+        <translation>Ilyen nevű osztály már van.</translation>
+    </message>
+</context>
+<context>
+    <name>Window</name>
+    <message>
+        <source>Tetzle</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <source>A jigsaw puzzle with tetrominoes for pieces</source>
+        <translation>Kirakós puzzle játék</translation>
+    </message>
+    <message>
+        <source>&amp;Game</source>
+        <translation>&amp;Játék</translation>
+    </message>
+    <message>
+        <source>&amp;Choose...</source>
+        <translation>&amp;Választás...</translation>
+    </message>
+    <message>
+        <source>&amp;Retrieve Pieces</source>
+        <translation>&amp;Elemek keverése</translation>
+    </message>
+    <message>
+        <source>Ctrl+R</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <source>&amp;Quit</source>
+        <translation>&amp;Kilépés</translation>
+    </message>
+    <message>
+        <source>&amp;View</source>
+        <translation>&amp;Nézet</translation>
+    </message>
+    <message>
+        <source>Zoom &amp;In</source>
+        <translation>&amp;Nagyítás</translation>
+    </message>
+    <message>
+        <source>+</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <source>Zoom &amp;Out</source>
+        <translation>Ki&amp;csinyítés</translation>
+    </message>
+    <message>
+        <source>-</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <source>Best &amp;Fit</source>
+        <translation>&amp;Legjobb kitöltés</translation>
+    </message>
+    <message>
+        <source>Show O&amp;verview</source>
+        <translation>Elő&amp;nézet</translation>
+    </message>
+    <message>
+        <source>Tab</source>
+        <translation>Fül</translation>
+    </message>
+    <message>
+        <source>F&amp;ullscreen</source>
+        <translation>T&amp;eli kép</translation>
+    </message>
+    <message>
+        <source>F11</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <source>Ctrl+F</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <source>&amp;Settings</source>
+        <translation>&amp;Beállítások</translation>
+    </message>
+    <message>
+        <source>&amp;Appearance...</source>
+        <translation>&amp;Megjelenés...</translation>
+    </message>
+    <message>
+        <source>&amp;Language...</source>
+        <translation>Nye&amp;lv...</translation>
+    </message>
+    <message>
+        <source>&amp;Help</source>
+        <translation>&amp;Segítség</translation>
+    </message>
+    <message>
+        <source>&amp;Controls</source>
+        <translation>&amp;Vezérlés</translation>
+    </message>
+    <message>
+        <source>&amp;About</source>
+        <translation>&amp;Rólam</translation>
+    </message>
+    <message>
+        <source>About &amp;Qt</source>
+        <translation>A &amp;QT</translation>
+    </message>
+    <message>
+        <source>Controls</source>
+        <translation>Vezérlés</translation>
+    </message>
+    <message>
+        <source>&lt;b&gt;Pick Up Pieces:&lt;/b&gt;</source>
+        <translation>&lt;b&gt;Elemek felszedése:&lt;/b&gt;</translation>
+    </message>
+    <message>
+        <source>Left Click or Space</source>
+        <translation>Bal klikk vagy space</translation>
+    </message>
+    <message>
+        <source>&lt;b&gt;Drop Pieces:&lt;/b&gt;</source>
+        <translation>&lt;b&gt;Elemek lerakása:&lt;/b&gt;</translation>
+    </message>
+    <message>
+        <source>&lt;b&gt;Select Pieces:&lt;/b&gt;</source>
+        <translation>&lt;b&gt;Elemek választása:&lt;/b&gt;</translation>
+    </message>
+    <message>
+        <source>Left Drag</source>
+        <translation>Balra húzás</translation>
+    </message>
+    <message>
+        <source>&lt;b&gt;Rotate Pieces:&lt;/b&gt;</source>
+        <translation>&lt;b&gt;Elemek forgatása:&lt;/b&gt;</translation>
+    </message>
+    <message>
+        <source>Right Click, Control + Left Click, or R</source>
+        <translation>Bal klikk,Ctrl + bal klikk, vagy R</translation>
+    </message>
+    <message>
+        <source>&lt;b&gt;Drag Puzzle:&lt;/b&gt;</source>
+        <translation>&lt;b&gt;Puzzle eltolása:&lt;/b&gt;</translation>
+    </message>
+    <message>
+        <source>Middle Click, Shift + Left Click, or Arrows</source>
+        <translation>Közép klikk, Sh + bal klikk, vagy nyilak</translation>
+    </message>
+    <message>
+        <source>&lt;b&gt;Zoom Puzzle:&lt;/b&gt;</source>
+        <translation>&lt;b&gt;Puzzle nagyítás:&lt;/b&gt;</translation>
+    </message>
+    <message>
+        <source>Scrollwheel or +/-</source>
+        <translation>Egérgörgő vagy +/-</translation>
+    </message>
+    <message>
+        <source>&lt;b&gt;Move Cursor:&lt;/b&gt;</source>
+        <translation>&lt;b&gt;Kurzor mozgatás:&lt;/b&gt;</translation>
+    </message>
+    <message>
+        <source>Move mouse or W,A,D,S</source>
+        <translation>Egérmozgás, vagy W,A,D,S</translation>
+    </message>
+    <message>
+        <source>About Tetzle</source>
+        <translation>A tetzle -ről</translation>
+    </message>
+    <message>
+        <source>Copyright &amp;copy; 2008-%1 Graeme Gott</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <source>Released under the &lt;a href=%1&gt;GPL 3&lt;/a&gt; license</source>
+        <translation>A megvalósítási licensz: &lt;a href=%1&gt;GPL 3&lt;/a&gt; license</translation>
+    </message>
+</context>
+<context>
+    <name>ZoomSlider</name>
+    <message>
+        <source>??%</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <source>%1%</source>
+        <translation type="unfinished"></translation>
+    </message>
+</context>
+<context>
+    <name>main</name>
+    <message>
+        <source>Select OpenGL version.</source>
+        <translation>OpenGL verzió.</translation>
+    </message>
+    <message>
+        <source>version</source>
+        <translation>verzió</translation>
+    </message>
+</context>
+</TS>
