diff -Naur orig/scripts/squid_redirect-nodata mod/scripts/squid_redirect-nodata
--- orig/scripts/squid_redirect-nodata	2011-10-19 11:56:41.000000000 +0200
+++ mod/scripts/squid_redirect-nodata	2016-04-09 09:14:12.954922914 +0200
@@ -1111,4 +1111,21 @@
 }
 
 __DATA__
+AD http://swift.hu/**/banner/**
+AD http://ad.adverticum.net/**
+AD http://kampany.vnubp.hu/banner_hirstart/**
+AD http://audit.median.hu/**
+AD http://origo.hu/p/banner/**
+AD http://www2.ctnetwork.hu/*_banner_**
+AD http://srs.targetpoint.com/banner.aspx?**
+AD http://ads.sg.hu/**
+AD http://www.gepbolt.hu/kreativ/**
+AD http://pr.atwola.com/promoimp/
+AD http://http300.edge.ru4.com/smartserve/ad?
+AD http://pagead2.googlesyndication.com/pagead/ads?
+ADJS http://ar.atwola.com/file/adsWrapper.js
+ADJS http://cdn.netscape.com/_media/wpsite/popup2.js
+ADJS http://ar.atwola.com/file/adsPopup2.js
+ADJS http://cdn.netscape.com/_media/wpsite/adpopunder.js
+
 ### END AUTO __DATA__ AREA
