diff -Naur keepassx-2.0.3-orig/share/linux/keepassx.desktop keepassx-2.0.3/share/linux/keepassx.desktop
--- keepassx-2.0.3-orig/share/linux/keepassx.desktop	2016-09-04 17:57:47.000000000 +0200
+++ keepassx-2.0.3/share/linux/keepassx.desktop	2017-06-18 15:55:31.858194896 +0200
@@ -1,10 +1,7 @@
 [Desktop Entry]
 Name=KeePassX
 GenericName=Password Manager
-GenericName[de]=Passwortverwaltung
-GenericName[es]=Gestor de contraseñas
-GenericName[fr]=Gestionnaire de mot de passe
-GenericName[ru]=менеджер паролей
+GenericName[hu]=Keresztplatformos jelszókezelő
 Exec=keepassx %f
 Icon=keepassx
 Terminal=false
