diff -Naurdp a/src/vteapp.c b/src/vteapp.c
--- a/src/vteapp.c	2007-06-06 14:20:53.000000000 +0200
+++ b/src/vteapp.c	2007-06-06 14:21:27.000000000 +0200
@@ -436,7 +436,7 @@ main(int argc, char **argv)
         char *geometry = NULL;
 	gint lines = 100;
 	const char *message = "Launching interactive shell...\r\n";
-	const char *font = NULL;
+	const char *font = "Fixed Bold 12";
 	const char *termcap = NULL;
 	const char *command = NULL;
 	const char *working_directory = NULL;
