--- vte.orig/src/vteapp.c	2011-02-16 20:32:26.000000000 +0100
+++ vte/src/vteapp.c	2011-03-30 19:53:31.000000000 +0200
@@ -559,7 +559,7 @@
 		 antialias = TRUE, use_scrolled_window = FALSE,
 		 show_object_notifications = FALSE;
 	char *geometry = NULL;
-	gint lines = 100;
+	gint lines = 2000;
 	const char *message = "Launching interactive shell...\r\n";
 	const char *font = NULL;
 	const char *termcap = NULL;
