diff -Naur orig/libindicator/Makefile.am patched/libindicator/Makefile.am
--- orig/libindicator/Makefile.am	2012-09-19 17:38:05.000000000 +0200
+++ patched/libindicator/Makefile.am	2019-09-17 11:11:03.791745851 +0200
@@ -57,7 +57,7 @@
 	$(LIBINDICATOR_CFLAGS) \
 	$(COVERAGE_CFLAGS) \
 	-DG_LOG_DOMAIN=\"libindicator\" \
-	-Wall -Werror
+	-Wall
 
 libindicator_la_LIBADD = \
 	$(LIBINDICATOR_LIBS)
diff -Naur orig/libindicator/Makefile.in patched/libindicator/Makefile.in
--- orig/libindicator/Makefile.in	2012-09-19 17:40:09.000000000 +0200
+++ patched/libindicator/Makefile.in	2019-09-17 11:11:18.535900554 +0200
@@ -380,7 +380,7 @@
 	$(LIBINDICATOR_CFLAGS) \
 	$(COVERAGE_CFLAGS) \
 	-DG_LOG_DOMAIN=\"libindicator\" \
-	-Wall -Werror
+	-Wall
 
 libindicator_la_LIBADD = \
 	$(LIBINDICATOR_LIBS)
diff -Naur orig/tests/Makefile.am patched/tests/Makefile.am
--- orig/tests/Makefile.am	2012-09-19 17:38:05.000000000 +0200
+++ patched/tests/Makefile.am	2019-09-17 11:11:42.796155325 +0200
@@ -38,7 +38,7 @@
 	test-loader.c
 
 test_loader_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir) \
 	-DBUILD_DIR="\"$(builddir)\""
 
@@ -57,7 +57,7 @@
 	test-desktop-shortcuts.c
 
 test_desktop_shortcuts_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	-DSRCDIR="\"$(srcdir)\"" \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir) \
 	-DBUILD_DIR="\"$(abs_builddir)\""
@@ -91,7 +91,7 @@
 	dummy-indicator-blank.c
 
 libdummy_indicator_blank_la_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 libdummy_indicator_blank_la_LIBADD = \
@@ -111,7 +111,7 @@
 	dummy-indicator-null.c
 
 libdummy_indicator_null_la_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 libdummy_indicator_null_la_LIBADD = \
@@ -131,7 +131,7 @@
 	dummy-indicator-signaler.c
 
 libdummy_indicator_signaler_la_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 libdummy_indicator_signaler_la_LIBADD = \
@@ -151,7 +151,7 @@
 	dummy-indicator-simple.c
 
 libdummy_indicator_simple_la_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 libdummy_indicator_simple_la_LIBADD = \
@@ -172,7 +172,7 @@
 	dummy-indicator-entry-func.h
 
 libdummy_indicator_entry_func_la_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 libdummy_indicator_entry_func_la_LIBADD = \
@@ -192,7 +192,7 @@
 	dummy-indicator-visible.c
 
 libdummy_indicator_visible_la_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 libdummy_indicator_visible_la_LIBADD = \
@@ -214,7 +214,7 @@
 	service-shutdown-timeout.c
 
 service_shutdown_timeout_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_shutdown_timeout_LDADD = \
@@ -240,7 +240,7 @@
 	service-manager-no-connect.c
 
 service_manager_no_connect_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_manager_no_connect_LDADD = \
@@ -272,7 +272,7 @@
 	service-manager-connect.c
 
 service_manager_connect_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_manager_connect_LDADD = \
@@ -286,7 +286,7 @@
 	service-manager-connect-service.c
 
 service_manager_connect_service_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_manager_connect_service_LDADD = \
@@ -319,7 +319,7 @@
 	service-version-manager.c
 
 service_version_manager_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_version_manager_LDADD = \
@@ -334,7 +334,7 @@
 	service-version-bad-service.c
 
 service_version_bad_service_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_version_bad_service_LDADD = \
@@ -349,7 +349,7 @@
 	service-version-good-service.c
 
 service_version_good_service_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_version_good_service_LDADD = \
@@ -376,7 +376,7 @@
 	service-version-multiwatch-manager.c
 
 service_version_multiwatch_manager_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_version_multiwatch_manager_LDADD = \
@@ -391,7 +391,7 @@
 	service-version-multiwatch-manager-impolite.c
 
 service_version_multiwatch_manager_impolite_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_version_multiwatch_manager_impolite_LDADD = \
@@ -406,7 +406,7 @@
 	service-version-multiwatch-service.c
 
 service_version_multiwatch_service_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_version_multiwatch_service_LDADD = \
@@ -432,7 +432,7 @@
 	service-manager-nostart-connect.c
 
 service_manager_nostart_connect_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_manager_nostart_connect_LDADD = \
diff -Naur orig/tests/Makefile.in patched/tests/Makefile.in
--- orig/tests/Makefile.in	2012-09-19 17:40:09.000000000 +0200
+++ patched/tests/Makefile.in	2019-09-17 11:11:57.300307770 +0200
@@ -529,7 +529,7 @@
 	test-loader.c
 
 test_loader_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir) \
 	-DBUILD_DIR="\"$(builddir)\""
 
@@ -542,7 +542,7 @@
 	test-desktop-shortcuts.c
 
 test_desktop_shortcuts_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	-DSRCDIR="\"$(srcdir)\"" \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir) \
 	-DBUILD_DIR="\"$(abs_builddir)\""
@@ -562,7 +562,7 @@
 	dummy-indicator-blank.c
 
 libdummy_indicator_blank_la_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 libdummy_indicator_blank_la_LIBADD = \
@@ -582,7 +582,7 @@
 	dummy-indicator-null.c
 
 libdummy_indicator_null_la_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 libdummy_indicator_null_la_LIBADD = \
@@ -602,7 +602,7 @@
 	dummy-indicator-signaler.c
 
 libdummy_indicator_signaler_la_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 libdummy_indicator_signaler_la_LIBADD = \
@@ -622,7 +622,7 @@
 	dummy-indicator-simple.c
 
 libdummy_indicator_simple_la_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 libdummy_indicator_simple_la_LIBADD = \
@@ -643,7 +643,7 @@
 	dummy-indicator-entry-func.h
 
 libdummy_indicator_entry_func_la_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 libdummy_indicator_entry_func_la_LIBADD = \
@@ -663,7 +663,7 @@
 	dummy-indicator-visible.c
 
 libdummy_indicator_visible_la_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 libdummy_indicator_visible_la_LIBADD = \
@@ -679,7 +679,7 @@
 	service-shutdown-timeout.c
 
 service_shutdown_timeout_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_shutdown_timeout_LDADD = \
@@ -691,7 +691,7 @@
 	service-manager-no-connect.c
 
 service_manager_no_connect_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_manager_no_connect_LDADD = \
@@ -703,7 +703,7 @@
 	service-manager-connect.c
 
 service_manager_connect_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_manager_connect_LDADD = \
@@ -715,7 +715,7 @@
 	service-manager-connect-service.c
 
 service_manager_connect_service_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_manager_connect_service_LDADD = \
@@ -728,7 +728,7 @@
 	service-version-manager.c
 
 service_version_manager_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_version_manager_LDADD = \
@@ -741,7 +741,7 @@
 	service-version-bad-service.c
 
 service_version_bad_service_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_version_bad_service_LDADD = \
@@ -754,7 +754,7 @@
 	service-version-good-service.c
 
 service_version_good_service_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_version_good_service_LDADD = \
@@ -767,7 +767,7 @@
 	service-version-multiwatch-manager.c
 
 service_version_multiwatch_manager_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_version_multiwatch_manager_LDADD = \
@@ -780,7 +780,7 @@
 	service-version-multiwatch-manager-impolite.c
 
 service_version_multiwatch_manager_impolite_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_version_multiwatch_manager_impolite_LDADD = \
@@ -793,7 +793,7 @@
 	service-version-multiwatch-service.c
 
 service_version_multiwatch_service_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_version_multiwatch_service_LDADD = \
@@ -805,7 +805,7 @@
 	service-manager-nostart-connect.c
 
 service_manager_nostart_connect_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_manager_nostart_connect_LDADD = \
diff -Naur orig/tools/Makefile.am patched/tools/Makefile.am
--- orig/tools/Makefile.am	2012-09-19 17:38:04.000000000 +0200
+++ patched/tools/Makefile.am	2019-09-17 11:12:11.068452570 +0200
@@ -17,7 +17,7 @@
 	indicator-loader.c
 
 indicator_loader_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir) \
 	-DBUILD_DIR="\"$(builddir)\""
 
diff -Naur orig/tools/Makefile.in patched/tools/Makefile.in
--- orig/tools/Makefile.in	2012-09-19 17:40:09.000000000 +0200
+++ patched/tools/Makefile.in	2019-09-17 11:12:17.776523150 +0200
@@ -303,7 +303,7 @@
 	indicator-loader.c
 
 indicator_loader_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir) \
 	-DBUILD_DIR="\"$(builddir)\""
 
