diff -Naur ufraw-0.17-orig//ufraw.desktop ufraw-0.17//ufraw.desktop
--- ufraw-0.17-orig//ufraw.desktop	2010-03-10 04:28:27.000000000 +0100
+++ ufraw-0.17//ufraw.desktop	2010-05-13 20:37:30.000000000 +0200
@@ -7,6 +7,7 @@
 GenericName[de]=Konverter für Raw-Bilder
 GenericName[es]=Conversor de imágenes raw
 GenericName[fr]=Convertisseur d'image RAW
+GenericName[hu]=RAW fotó konverter
 GenericName[it]=Convertitore per immagini Raw
 GenericName[ko]=Raw 이미지 변환기
 GenericName[nb]=Raw-bildebehandler
@@ -25,6 +26,7 @@
 Comment[de]=Ein Werkzeug, um Raw-Bilder von Digitalkameras zu lesen und zu manipulieren
 Comment[es]=Programa para leer y manipular imágenes raw de cámaras digitales
 Comment[fr]=Un utilitaire pour lire et manipuler les images RAW des appareils photos numériques
+Comment[hu]=Segédprogram digitális kamerák RAW képeinek feldolgozásához
 Comment[it]=Gestione immagini Raw realizzate con fotocamere digitali
 Comment[ko]=디지털 카메라의 Raw 이미지를 읽고, 처리하는 프로그램
 Comment[nb]=Et program for å lese og behandle raw-bilder fra digitale kameraer
