From c34888dd5461e2674feefefe001be2c7d64e427f Mon Sep 17 00:00:00 2001
From: Daniel Nicoletti <dantti12@gmail.com>
Date: Wed, 21 Feb 2018 17:19:17 -0300
Subject: [PATCH 3/8] Fix compilation when EDIT_ORIGNS_DESKTOP_NAME is defined

---
 Apper/Settings/Settings.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Apper/Settings/Settings.cpp b/Apper/Settings/Settings.cpp
index 2175ea7..e077b8e 100644
--- a/Apper/Settings/Settings.cpp
+++ b/Apper/Settings/Settings.cpp
@@ -154,7 +154,7 @@ Settings::~Settings()
 void Settings::on_editOriginsPB_clicked()
 {
 #ifdef EDIT_ORIGNS_DESKTOP_NAME
-    KToolInvocation::startServiceByDesktopName(EDIT_ORIGNS_DESKTOP_NAME);
+    KToolInvocation::startServiceByDesktopName(QLatin1String(EDIT_ORIGNS_DESKTOP_NAME));
 #endif //EDIT_ORIGNS_DESKTOP_NAME
 }
 
-- 
2.17.1

