diff -up apper-1.0.0/Apper/Settings/Settings.cpp.qt511 apper-1.0.0/Apper/Settings/Settings.cpp
--- apper-1.0.0/Apper/Settings/Settings.cpp.qt511	2018-07-31 09:31:10.325795972 -0500
+++ apper-1.0.0/Apper/Settings/Settings.cpp	2018-07-31 09:33:32.509880561 -0500
@@ -27,6 +27,7 @@
 #include <Enum.h>
 #include <PkStrings.h>
 
+#include <QAction>
 #include <QTimer>
 #include <QSortFilterProxyModel>
 
diff -up apper-1.0.0/libapper/Requirements.h.qt511 apper-1.0.0/libapper/Requirements.h
--- apper-1.0.0/libapper/Requirements.h.qt511	2018-07-31 09:32:39.579476804 -0500
+++ apper-1.0.0/libapper/Requirements.h	2018-07-31 09:32:07.813234487 -0500
@@ -23,6 +23,7 @@
 
 #include <QDialog>
 
+#include <QButtonGroup>
 #include <QToolButton>
 
 namespace Ui {
