diff -Nur orig/libapper/PkIcons.cpp mod/libapper/PkIcons.cpp
--- orig/libapper/PkIcons.cpp	2018-02-20 21:12:16.000000000 +0100
+++ mod/libapper/PkIcons.cpp	2019-02-21 18:22:16.282364326 +0100
@@ -232,7 +232,7 @@
     }
     switch (group) {
     case Transaction::GroupUnknown         : return QIcon::fromTheme(QLatin1String("unknown"));
-    case Transaction::GroupAccessibility   : return QIcon::fromTheme(QLatin1String("preferences-desktop-accessibility"));
+    case Transaction::GroupAccessibility   : return QIcon::fromTheme(QLatin1String("cinnamon"));
     case Transaction::GroupAccessories     : return QIcon::fromTheme(QLatin1String("applications-accessories"));
     case Transaction::GroupAdminTools      : return QIcon::fromTheme(QLatin1String("dialog-password"));
     case Transaction::GroupCommunication   : return QIcon::fromTheme(QLatin1String("network-workgroup"));//FIXME
@@ -242,13 +242,13 @@
     case Transaction::GroupDesktopXfce     : return QIcon::fromTheme(QLatin1String("kpk-desktop-xfce"));
     case Transaction::GroupDocumentation   : return QIcon::fromTheme(QLatin1String("accessories-dictionary"));//FIXME
     case Transaction::GroupEducation       : return QIcon::fromTheme(QLatin1String("applications-education"));
-    case Transaction::GroupElectronics     : return QIcon::fromTheme(QLatin1String("media-flash"));
+    case Transaction::GroupElectronics     : return QIcon::fromTheme(QLatin1String("lxqt"));
     case Transaction::GroupFonts           : return QIcon::fromTheme(QLatin1String("preferences-desktop-font"));
     case Transaction::GroupGames           : return QIcon::fromTheme(QLatin1String("applications-games"));
     case Transaction::GroupGraphics        : return QIcon::fromTheme(QLatin1String("applications-graphics"));
     case Transaction::GroupInternet        : return QIcon::fromTheme(QLatin1String("applications-internet"));
     case Transaction::GroupLegacy          : return QIcon::fromTheme(QLatin1String("media-floppy"));
-    case Transaction::GroupLocalization    : return QIcon::fromTheme(QLatin1String("applications-education-language"));
+    case Transaction::GroupLocalization    : return QIcon::fromTheme(QLatin1String("mate"));
     case Transaction::GroupMaps            : return QIcon::fromTheme(QLatin1String("Maps"));//FIXME
     case Transaction::GroupCollections     : return QIcon::fromTheme(QLatin1String("package-orign"));
     case Transaction::GroupMultimedia      : return QIcon::fromTheme(QLatin1String("applications-multimedia"));
@@ -260,7 +260,7 @@
     case Transaction::GroupPublishing      : return QIcon::fromTheme(QLatin1String("accessories-text-editor"));
     case Transaction::GroupRepos           : return QIcon::fromTheme(QLatin1String("application-x-compressed-tar"));
     case Transaction::GroupScience         : return QIcon::fromTheme(QLatin1String("applications-science"));
-    case Transaction::GroupSecurity        : return QIcon::fromTheme(QLatin1String("security-high"));
+    case Transaction::GroupSecurity        : return QIcon::fromTheme(QLatin1String("xarchiver"));
     case Transaction::GroupServers         : return QIcon::fromTheme(QLatin1String("network-server"));
     case Transaction::GroupSystem          : return QIcon::fromTheme(QLatin1String("applications-system"));
     case Transaction::GroupVirtualization  : return QIcon::fromTheme(QLatin1String("cpu"));
diff -Nur orig/libapper/PkStrings.cpp mod/libapper/PkStrings.cpp
--- orig/libapper/PkStrings.cpp	2018-02-20 21:12:16.000000000 +0100
+++ mod/libapper/PkStrings.cpp	2019-02-21 19:10:05.622195477 +0100
@@ -752,7 +752,7 @@
 {
     switch (group) {
     case Transaction::GroupAccessibility :
-        return i18nc("The group type", "Accessibility");
+        return i18nc("The group type", "CINNAMON");
     case Transaction::GroupAccessories :
         return i18nc("The group type", "Accessories");
     case Transaction::GroupEducation :
@@ -780,9 +780,9 @@
     case Transaction::GroupDesktopXfce :
         return i18nc("The group type", "XFCE desktop");
     case Transaction::GroupDesktopOther :
-        return i18nc("The group type", "Other desktops");
+        return i18nc("The group type", "Gyors grafikus környezet");
     case Transaction::GroupPublishing :
-        return i18nc("The group type", "Publishing");
+        return i18nc("The group type", "Editorok");
     case Transaction::GroupServers :
         return i18nc("The group type", "Servers");
     case Transaction::GroupFonts :
@@ -790,21 +790,21 @@
     case Transaction::GroupAdminTools :
         return i18nc("The group type", "Admin tools");
     case Transaction::GroupLegacy :
-        return i18nc("The group type", "Legacy");
+        return i18nc("The group type", "Fejlécek és függvénytárak");
     case Transaction::GroupLocalization :
-        return i18nc("The group type", "Localization");
+        return i18nc("The group type", "MATE");
     case Transaction::GroupVirtualization :
-        return i18nc("The group type", "Virtualization");
+        return i18nc("The group type", "Emulátorok");
     case Transaction::GroupSecurity :
-        return i18nc("The group type", "Security");
+        return i18nc("The group type", "Tömörítők");
     case Transaction::GroupPowerManagement :
-        return i18nc("The group type", "Power management");
+        return i18nc("The group type", "X");
     case Transaction::GroupCommunication :
         return i18nc("The group type", "Communication");
     case Transaction::GroupNetwork :
         return i18nc("The group type", "Network");
     case Transaction::GroupMaps :
-        return i18nc("The group type", "Maps");
+        return i18nc("The group type", "Adatbázisok");
     case Transaction::GroupRepos :
         return i18nc("The group type", "Software sources");
     case Transaction::GroupScience :
@@ -812,7 +812,7 @@
     case Transaction::GroupDocumentation :
         return i18nc("The group type", "Documentation");
     case Transaction::GroupElectronics :
-        return i18nc("The group type", "Electronics");
+        return i18nc("The group type", "LXDE, LXQT");
     case Transaction::GroupCollections ://TODO check this one
         return i18nc("The group type", "Package collections");
     case Transaction::GroupVendor :
