diff -Naur a/meson.build b/meson.build
--- a/meson.build	1970-01-01 01:00:00.000000000 +0100
+++ b/meson.build	2022-04-06 13:36:14.449019264 +0200
@@ -55,7 +55,7 @@
 stability_version_number = (major_version != 0 ? minor_version : micro_version)
 stable = (stability_version_number % 2 == 0)
 
-conf.set10('BABL_UNSTABLE', not stable, Description:
+conf.set10('BABL_UNSTABLE', not stable, description:
   'Define to 1 if this is an unstable version of BABL.')
 
 conf.set       ('BABL_MAJOR_VERSION',    '@0@'.format(major_version))
