diff -Nur orig/pyproject.toml mod/pyproject.toml
--- orig/pyproject.toml	2024-06-05 11:45:14.000000000 +0200
+++ mod/pyproject.toml	2024-06-05 11:50:25.444214233 +0200
@@ -64,7 +64,7 @@
 all = ["cclib[bridges]"]
 
 [build-system]
-requires = ["setuptools >= 61.0", "versioningit~=2.0"]
+requires = ["setuptools >= 61.0", "versioningit >= 2.0"]
 build-backend = "setuptools.build_meta"
 
 [tool.setuptools]
