diff -Naur orig/third_party/skia/src/ports/SkFontHost_FreeType.cpp patched/third_party/skia/src/ports/SkFontHost_FreeType.cpp
--- orig/third_party/skia/src/ports/SkFontHost_FreeType.cpp	2020-04-21 23:52:06.000000000 +0200
+++ patched/third_party/skia/src/ports/SkFontHost_FreeType.cpp	2020-05-12 23:53:20.186021734 +0200
@@ -128,9 +128,9 @@
         : fGetVarDesignCoordinates(nullptr)
         , fGetVarAxisFlags(nullptr)
         , fLibrary(nullptr)
-        , fIsLCDSupported(false)
+        , fIsLCDSupported(true)
         , fLightHintingIsYOnly(false)
-        , fLCDExtra(0)
+        , fLCDExtra(2)
     {
         if (FT_New_Library(&gFTMemory, &fLibrary)) {
             return;
