diff -Naur a/third_party/crashpad/crashpad/snapshot/elf/elf_image_reader.cc b/third_party/crashpad/crashpad/snapshot/elf/elf_image_reader.cc
--- a/third_party/crashpad/crashpad/snapshot/elf/elf_image_reader.cc	2022-11-30 01:41:25.475589500 +0100
+++ b/third_party/crashpad/crashpad/snapshot/elf/elf_image_reader.cc	2022-12-02 19:06:30.579901276 +0100
@@ -56,14 +56,14 @@
 class ElfImageReader::ProgramHeaderTableSpecific
     : public ElfImageReader::ProgramHeaderTable {
  public:
-  ProgramHeaderTableSpecific<PhdrType>() {}
+  ProgramHeaderTableSpecific() {}
 
-  ProgramHeaderTableSpecific<PhdrType>(
+  ProgramHeaderTableSpecific(
       const ProgramHeaderTableSpecific<PhdrType>&) = delete;
   ProgramHeaderTableSpecific<PhdrType>& operator=(
       const ProgramHeaderTableSpecific<PhdrType>&) = delete;
 
-  ~ProgramHeaderTableSpecific<PhdrType>() {}
+  ~ProgramHeaderTableSpecific() {}
 
   bool Initialize(const ProcessMemoryRange& memory,
                   VMAddress address,
