diff -Naur a/third_party/ipcz/src/ipcz/router_link.h b/third_party/ipcz/src/ipcz/router_link.h
--- a/third_party/ipcz/src/ipcz/router_link.h	2023-09-16 04:07:25.409128700 +0200
+++ b/third_party/ipcz/src/ipcz/router_link.h	2023-09-17 15:21:30.735363963 +0200
@@ -7,6 +7,7 @@
 
 #include <cstddef>
 #include <functional>
+#include <memory>
 #include <string>
 #include <utility>
 
diff -Naur a/third_party/material_color_utilities/src/cpp/palettes/tones.cc b/third_party/material_color_utilities/src/cpp/palettes/tones.cc
--- a/third_party/material_color_utilities/src/cpp/palettes/tones.cc	2023-09-16 04:09:41.136591700 +0200
+++ b/third_party/material_color_utilities/src/cpp/palettes/tones.cc	2023-09-17 15:20:36.747182273 +0200
@@ -14,6 +14,8 @@
  * limitations under the License.
  */
 
+#include <cmath>
+
 #include "cpp/palettes/tones.h"
 
 #include "cpp/cam/cam.h"
