diff -Naur a/third_party/blink/renderer/platform/text/text_break_iterator.cc b/third_party/blink/renderer/platform/text/text_break_iterator.cc
--- a/third_party/blink/renderer/platform/text/text_break_iterator.cc	2023-11-08 22:41:58.453302400 +0100
+++ b/third_party/blink/renderer/platform/text/text_break_iterator.cc	2023-11-11 21:14:06.675925520 +0100
@@ -161,7 +161,9 @@
 };
 // clang-format on
 
-#if U_ICU_VERSION_MAJOR_NUM >= 58
+#if U_ICU_VERSION_MAJOR_NUM >= 74
+#define BA_LB_COUNT (U_LB_COUNT - 8)
+#elif U_ICU_VERSION_MAJOR_NUM >= 58
 #define BA_LB_COUNT (U_LB_COUNT - 3)
 #else
 #define BA_LB_COUNT U_LB_COUNT
