diff -Naur a/chrome/browser/performance_manager/user_tuning/cpu_health_tracker.cc b/chrome/browser/performance_manager/user_tuning/cpu_health_tracker.cc
--- a/chrome/browser/performance_manager/user_tuning/cpu_health_tracker.cc	2024-06-18 23:43:23.100538300 +0200
+++ b/chrome/browser/performance_manager/user_tuning/cpu_health_tracker.cc	2024-06-22 20:25:47.191366540 +0200
@@ -256,7 +256,7 @@
     return;
   }
 
-  const CpuPercent total_system_cpu_usage{cpu_sample.value().cpu_utilization *
+  const CpuPercent total_system_cpu_usage{static_cast<int>(cpu_sample.value().cpu_utilization) *
                                           100};
   if (GetHealthLevelForMeasurement(total_system_cpu_usage) !=
       HealthLevel::kHealthy) {
