diff -Naur orig/buildtools/third_party/libc++/trunk/include/__locale patched/buildtools/third_party/libc++/trunk/include/__locale
--- orig/buildtools/third_party/libc++/trunk/include/__locale	2020-02-04 02:58:22.000000000 +0100
+++ patched/buildtools/third_party/libc++/trunk/include/__locale	2020-02-07 23:43:21.494909052 +0100
@@ -32,7 +32,7 @@
 # include <support/newlib/xlocale.h>
 #elif (defined(__APPLE__)      || defined(__FreeBSD__) \
     || defined(__EMSCRIPTEN__) || defined(__IBMCPP__))
-# include <xlocale.h>
+# include <locale.h>
 #elif defined(__Fuchsia__)
 # include <support/fuchsia/xlocale.h>
 #elif defined(__wasi__)
