diff -Naur orig/build/config/compiler/BUILD.gn patched/build/config/compiler/BUILD.gn
--- orig/build/config/compiler/BUILD.gn	2019-01-30 02:17:39.000000000 +0100
+++ patched/build/config/compiler/BUILD.gn	2019-01-30 22:28:18.808189391 +0100
@@ -459,10 +459,6 @@
     ldflags += [ "-fuse-ld=bfd" ]
   }
 
-  if (use_icf) {
-    ldflags += [ "-Wl,--icf=all" ]
-  }
-
   if (linux_use_bundled_binutils) {
     cflags += [ "-B$binutils_path" ]
   }
diff -Naur orig/third_party/perfetto/gn/standalone/BUILD.gn patched/third_party/perfetto/gn/standalone/BUILD.gn
--- orig/third_party/perfetto/gn/standalone/BUILD.gn	2019-01-30 02:20:34.000000000 +0100
+++ patched/third_party/perfetto/gn/standalone/BUILD.gn	2019-01-30 22:29:00.488727666 +0100
@@ -202,7 +202,6 @@
     ldflags = [
       "-fuse-ld=gold",
       "-Wl,--gc-sections",
-      "-Wl,--icf=all",
       "-Wl,-O1",
     ]
   }
diff -Naur orig/tools/gn/build/gen.py patched/tools/gn/build/gen.py
--- orig/tools/gn/build/gen.py	2019-01-30 02:24:06.000000000 +0100
+++ patched/tools/gn/build/gen.py	2019-01-30 22:29:44.573296988 +0100
@@ -298,10 +298,6 @@
         else:
           ldflags.append('-Wl,-strip-all')
 
-      # Enable identical code-folding.
-      if not platform.is_darwin():
-        ldflags.append('-Wl,--icf=all')
-
     cflags.extend([
         '-D_FILE_OFFSET_BITS=64',
         '-D__STDC_CONSTANT_MACROS', '-D__STDC_FORMAT_MACROS',
