diff -Naur orig/third_party/blink/renderer/platform/text/character_property_data.h patched/third_party/blink/renderer/platform/text/character_property_data.h
--- orig/third_party/blink/renderer/platform/text/character_property_data.h	2020-02-04 02:55:13.000000000 +0100
+++ patched/third_party/blink/renderer/platform/text/character_property_data.h	2020-02-07 23:46:22.589601538 +0100
@@ -154,10 +154,12 @@
     0xD7B0, 0xD7FF,
     // Halfwidth Hangul Jamo
     // https://www.unicode.org/charts/nameslist/c_FF00.html
-    0xFFA0, 0xFFDC,
+    0xFFA0, 0xFFDB,
 };
 
-static const UChar32 kIsHangulArray[] = {};
+static const UChar32 kIsHangulArray[] = {
+  0xFFDC,
+};
 
 // Freezed trie tree, see character_property_data_generator.cc.
 extern const int32_t kSerializedCharacterDataSize;
