diff -Naur orig/third_party/icu/source/i18n/number_compact.cpp patched/third_party/icu/source/i18n/number_compact.cpp
--- orig/third_party/icu/source/i18n/number_compact.cpp	2019-07-15 23:53:04.000000000 +0200
+++ patched/third_party/icu/source/i18n/number_compact.cpp	2019-07-16 20:46:15.196961765 +0200
@@ -7,6 +7,7 @@
 
 #include "unicode/ustring.h"
 #include "unicode/ures.h"
+#include "unicode/unum.h"
 #include "cstring.h"
 #include "charstr.h"
 #include "resource.h"
