diff -Naur orig/chrome/browser/performance_manager/graph/policies/background_tab_loading_policy_helpers.cc patched/chrome/browser/performance_manager/graph/policies/background_tab_loading_policy_helpers.cc
--- orig/chrome/browser/performance_manager/graph/policies/background_tab_loading_policy_helpers.cc	2020-05-18 20:40:21.000000000 +0200
+++ patched/chrome/browser/performance_manager/graph/policies/background_tab_loading_policy_helpers.cc	2020-05-20 21:41:22.228539357 +0200
@@ -3,6 +3,10 @@
 // found in the LICENSE file.
 
 #include "chrome/browser/performance_manager/graph/policies/background_tab_loading_policy_helpers.h"
+
+#include <math.h>
+#include <limits>
+
 #include "base/logging.h"
 
 namespace performance_manager {
