diff -Naur orig/ui/accessibility/ax_tree_serializer.h patched/ui/accessibility/ax_tree_serializer.h
--- orig/ui/accessibility/ax_tree_serializer.h	2021-03-02 18:46:00.000000000 +0100
+++ patched/ui/accessibility/ax_tree_serializer.h	2021-03-03 20:53:50.120344154 +0100
@@ -8,6 +8,7 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#include <ctime>
 #include <ostream>
 #include <unordered_map>
 #include <unordered_set>
