diff -Naur orig/build/config/linux/BUILD.gn patched/build/config/linux/BUILD.gn
--- orig/build/config/linux/BUILD.gn	2017-11-13 21:09:10.000000000 +0100
+++ patched/build/config/linux/BUILD.gn	2017-11-22 01:18:26.405653875 +0100
@@ -88,7 +88,7 @@
       "gthread-2.0",
     ]
     defines = [
-      "GLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_32",
+      "GLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_54",
       "GLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_26",
     ]
   }
