diff -Nur orig/Make.config.template mod/Make.config.template
--- orig/Make.config.template	2024-10-12 13:32:46.000000000 +0200
+++ mod/Make.config.template	2024-10-14 14:12:09.416082940 +0200
@@ -32,7 +32,7 @@
 
 # Default directories (adjust as necessary or desired):
 
-#PREFIX    = /usr/local
+PREFIX    = /usr
 #VDRROOT   = $(PREFIX)
 #BINDIR    = $(VDRROOT)/bin
 #INCDIR    = $(VDRROOT)/include
@@ -43,8 +43,6 @@
 #RESDIR    = $(VDRROOT)/share
 #DVBDIR    = /usr/include
 
-#VIDEODIR  = /srv/vdr/video
-#CONFDIR   = /var/lib/vdr
 #ARGSDIR   = /etc/vdr/conf.d
 #CACHEDIR  = /var/cache/vdr
 
@@ -85,7 +83,7 @@
 #NO_KBD = 1
 
 ### Define if you want vdr to not run as root:
-#VDR_USER = vdr
+VDR_USER = vdr
 
 ### Define if you want to compile in 'bidi' support:
 #BIDI = 1
