diff -Nur orig/Makefile mod/Makefile
--- orig/Makefile	2024-10-17 08:49:45.256684518 +0200
+++ mod/Makefile	2024-10-17 09:31:11.545210402 +0200
@@ -40,12 +40,12 @@
 endif
 
 DESTDIR   ?=
-VIDEODIR  ?= /srv/vdr/video
-CONFDIR   ?= /var/lib/vdr
+VIDEODIR  ?= /var/vdr/video
+CONFDIR   ?= /etc/vdr
 ARGSDIR   ?= /etc/vdr/conf.d
 CACHEDIR  ?= /var/cache/vdr
 
-PREFIX    ?= /usr/local
+PREFIX    ?= /usr
 VDRROOT   ?= $(PREFIX)
 BINDIR    ?= $(VDRROOT)/bin
 INCDIR    ?= $(VDRROOT)/include
@@ -53,7 +53,7 @@
 LOCDIR    ?= $(VDRROOT)/share/locale
 MANDIR    ?= $(VDRROOT)/share/man
 PCDIR     ?= $(VDRROOT)/lib/pkgconfig
-RESDIR    ?= $(VDRROOT)/share/vdr
+RESDIR    ?= $(VDRROOT)/share
 
 # Source documentation
 
