--- ./channels.c	2015/02/01 13:47:05	4.0
+++ ./channels.c	2015/03/13 11:34:28
@@ -229,6 +229,18 @@
   return true;
 }
 
+void cChannel::SetSource(int Source)
+{
+  if (source != Source) {
+     if (Number()) {
+        dsyslog("changing source of channel %d (%s) from %s to %s", Number(), name, *cSource::ToString(source), *cSource::ToString(Source));
+        modification |= CHANNELMOD_TRANSP;
+        Channels.SetModified();
+        }
+     source = Source;
+     }
+}
+
 void cChannel::SetId(int Nid, int Tid, int Sid, int Rid)
 {
   if (nid != Nid || tid != Tid || sid != Sid || rid != Rid) {
--- ./channels.h	2015/02/01 13:30:26	4.0
+++ ./channels.h	2015/03/13 11:20:50
@@ -193,6 +193,7 @@
   time_t Seen(void) { return seen; }
   void CopyTransponderData(const cChannel *Channel);
   bool SetTransponderData(int Source, int Frequency, int Srate, const char *Parameters, bool Quiet = false);
+  void SetSource(int Source);
   void SetId(int Nid, int Tid, int Sid, int Rid = 0);
   void SetLcn(int Lcn);
   void SetName(const char *Name, const char *ShortName, const char *Provider);
--- ./sdt.c	2015/01/04 14:33:35	4.0
+++ ./sdt.c	2015/03/13 11:34:58
@@ -105,6 +105,7 @@
                            }
                         else if (*pn && Setup.UpdateChannels >= 4) {
                            channel = Channels.NewChannel(Channel(), pn, ps, pp, sdt.getOriginalNetworkId(), sdt.getTransportStreamId(), SiSdtService.getServiceId());
+                           channel->SetSource(source); // in case this comes from a satellite with a slightly different position
                            patFilter->Trigger(SiSdtService.getServiceId());
                            }
                         }
