diff -Nur orig/Makefile mod/Makefile
--- orig/Makefile	2021-08-02 04:50:33.000000000 +0200
+++ mod/Makefile	2021-08-11 17:01:55.106492665 +0200
@@ -32,10 +32,9 @@
 	python3 setup.py kitty.app $(VVAL)
 
 man:
-	$(MAKE) FAIL_WARN=$(FAIL_WARN) -C docs man
 
 html:
-	$(MAKE) FAIL_WARN=$(FAIL_WARN) -C docs html
+	$(MAKE) FAIL_WARN=$(FAIL_WARN) -C html
 
 linkcheck:
 	$(MAKE) FAIL_WARN=$(FAIL_WARN) -C docs linkcheck
@@ -43,7 +42,7 @@
 website:
 	./publish.py --only website
 
-docs: man html
+docs:
 
 
 develop-docs:
diff -Nur orig/setup.cfg mod/setup.cfg
--- orig/setup.cfg	2021-08-02 04:50:33.000000000 +0200
+++ mod/setup.cfg	2021-08-11 16:48:32.515972542 +0200
@@ -14,7 +14,7 @@
 multi_line_output = 5
 
 [mypy]
-files = kitty,kittens,glfw,*.py,docs/conf.py
+files = kitty,kittens,glfw,*.py
 no_implicit_optional = True
 sqlite_cache = True
 cache_fine_grained = True
diff -Nur orig/setup.py mod/setup.py
--- orig/setup.py	2021-08-02 04:50:33.000000000 +0200
+++ mod/setup.py	2021-08-11 17:13:40.807731186 +0200
@@ -852,16 +852,6 @@
         for x in man_levels:
             shutil.rmtree(os.path.join(mandir, f'man{x}'))
     src = 'docs/_build/man'
-    if not os.path.exists(src):
-        raise SystemExit('''\
-The kitty man pages are missing. If you are building from git then run:
-make && make docs
-(needs the sphinx documentation system to be installed)
-''')
-    for x in man_levels:
-        os.makedirs(os.path.join(mandir, f'man{x}'))
-        for y in glob.glob(os.path.join(src, f'*.{x}')):
-            shutil.copy2(y, os.path.join(mandir, f'man{x}'))
 
 
 def copy_html_docs(ddir: str) -> None:
@@ -870,14 +860,6 @@
     with suppress(FileNotFoundError):
         shutil.rmtree(htmldir)
     src = 'docs/_build/html'
-    if not os.path.exists(src):
-        raise SystemExit('''\
-The kitty html docs are missing. If you are building from git then run:
-make && make docs
-(needs the sphinx documentation system to be installed)
-''')
-    shutil.copytree(src, htmldir)
-
 
 def compile_python(base_path: str) -> None:
     import compileall
