diff -Nur orig/setup.py mod/setup.py
--- orig/setup.py	2022-03-10 19:42:52.298752204 +0100
+++ mod/setup.py	2022-03-10 19:47:02.707465940 +0100
@@ -943,11 +943,6 @@
 
 
 def create_linux_bundle_gunk(ddir: str, libdir_name: str) -> None:
-    if not os.path.exists('docs/_build/html'):
-        make = 'gmake' if is_freebsd else 'make'
-        run_tool([make, 'docs'])
-    copy_man_pages(ddir)
-    copy_html_docs(ddir)
     for (icdir, ext) in {'256x256': 'png', 'scalable': 'svg'}.items():
         icdir = os.path.join(ddir, 'share', 'icons', 'hicolor', icdir, 'apps')
         safe_makedirs(icdir)
