diff -Nur orig/docs/conf.py mod/docs/conf.py
--- orig/docs/conf.py	2021-08-02 04:50:33.000000000 +0200
+++ mod/docs/conf.py	2021-08-22 17:04:52.648848403 +0200
@@ -112,7 +112,6 @@
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
 #
-html_theme = 'furo'
 html_title = 'kitty'
 
 # Theme options are theme-specific and customize the look and feel of a theme
