diff -Naur orig/applets/wncklet/org.mate.panel.applet.window-list-previews.gschema.xml.in patched/applets/wncklet/org.mate.panel.applet.window-list-previews.gschema.xml.in
--- orig/applets/wncklet/org.mate.panel.applet.window-list-previews.gschema.xml.in	2020-02-10 13:16:55.000000000 +0100
+++ patched/applets/wncklet/org.mate.panel.applet.window-list-previews.gschema.xml.in	2020-03-03 22:08:40.026581917 +0100
@@ -1,7 +1,7 @@
 <schemalist gettext-domain="@GETTEXT_PACKAGE@">
   <schema id="org.mate.panel.applet.window-list-previews">
     <key name="show-window-thumbnails" type="b">
-      <default>true</default>
+      <default>false</default>
       <summary>Display window thumbnails on hover</summary>
       <description>If true, then when hovering over a taskbar item, a thumbnail of the window will appear. It will go away as soon as the mouse leaves the item.</description>
     </key>
