diff -Naur orig/etc/Makefile.am patched/etc/Makefile.am
--- orig/etc/Makefile.am	2015-12-01 23:27:20.000000000 +0100
+++ patched/etc/Makefile.am	2016-06-05 21:11:38.728620903 +0200
@@ -32,9 +32,9 @@
 
 install-systemd-scripts: install-sysconfig
 	$(INSTALL) -m 755 -d $(DESTDIR)$(systemdsystemunitdir)
-	$(INSTALL) -m 755 $(srcdir)/bmc-watchdog.service $(DESTDIR)$(systemdsystemunitdir)/bmc-watchdog.service
-	$(INSTALL) -m 755 $(srcdir)/ipmidetectd.service $(DESTDIR)$(systemdsystemunitdir)/ipmidetectd.service
-	$(INSTALL) -m 755 $(srcdir)/ipmiseld.service $(DESTDIR)$(systemdsystemunitdir)/ipmiseld.service
+	$(INSTALL) -m 644 $(srcdir)/bmc-watchdog.service $(DESTDIR)$(systemdsystemunitdir)/bmc-watchdog.service
+	$(INSTALL) -m 644 $(srcdir)/ipmidetectd.service $(DESTDIR)$(systemdsystemunitdir)/ipmidetectd.service
+	$(INSTALL) -m 644 $(srcdir)/ipmiseld.service $(DESTDIR)$(systemdsystemunitdir)/ipmiseld.service
 
 install-sysconfig:
 	$(INSTALL) -m 755 -d $(DESTDIR)$(sysconfdir)/sysconfig
diff -Naur orig/etc/Makefile.in patched/etc/Makefile.in
--- orig/etc/Makefile.in	2015-12-01 23:29:35.000000000 +0100
+++ patched/etc/Makefile.in	2016-06-05 21:11:12.696491817 +0200
@@ -442,9 +442,9 @@
 
 install-systemd-scripts: install-sysconfig
 	$(INSTALL) -m 755 -d $(DESTDIR)$(systemdsystemunitdir)
-	$(INSTALL) -m 755 $(srcdir)/bmc-watchdog.service $(DESTDIR)$(systemdsystemunitdir)/bmc-watchdog.service
-	$(INSTALL) -m 755 $(srcdir)/ipmidetectd.service $(DESTDIR)$(systemdsystemunitdir)/ipmidetectd.service
-	$(INSTALL) -m 755 $(srcdir)/ipmiseld.service $(DESTDIR)$(systemdsystemunitdir)/ipmiseld.service
+	$(INSTALL) -m 644 $(srcdir)/bmc-watchdog.service $(DESTDIR)$(systemdsystemunitdir)/bmc-watchdog.service
+	$(INSTALL) -m 644 $(srcdir)/ipmidetectd.service $(DESTDIR)$(systemdsystemunitdir)/ipmidetectd.service
+	$(INSTALL) -m 644 $(srcdir)/ipmiseld.service $(DESTDIR)$(systemdsystemunitdir)/ipmiseld.service
 
 install-sysconfig:
 	$(INSTALL) -m 755 -d $(DESTDIR)$(sysconfdir)/sysconfig
