diff -Naur orig/lincity-ng.desktop mod/lincity-ng.desktop
--- orig/lincity-ng.desktop	2009-01-25 21:21:28.000000000 +0100
+++ mod/lincity-ng.desktop	2016-01-14 19:59:13.963449749 +0100
@@ -1,12 +1,11 @@
 [Desktop Entry]
 Type=Application
-Version=1.0
-Encoding=UTF-8
 Name=Lincity-NG
 GenericName=City Simulation Game
 GenericName[de]=Stadtsimulationsspiel
+Comment[hu]=Tervezz meg és építs egy várost, majd ügyelj a szállítmányozásra, a gazdaságra és másokra!
 Comment=Plan and Build a city and take care of transport, economics, electricityand others.
-Icon=lincity-ng.png
+Icon=lincity-ng
 Exec=lincity-ng
 Terminal=false
 StartupNotify=false
