diff -Naur orig/pam/crond patched/pam/crond
--- orig/pam/crond	2016-06-10 09:55:47.000000000 +0200
+++ patched/pam/crond	2016-07-03 21:16:18.725582817 +0200
@@ -4,8 +4,8 @@
 #
 # Although no PAM authentication is called, auth modules
 # are used for credential setting
-auth       include    password-auth
+auth       include    system-auth
 account    required   pam_access.so
-account    include    password-auth
+account    include    system-auth
 session    required   pam_loginuid.so
-session    include    password-auth
+session    include    system-auth
