diff -Naur orig/configure patched/configure
--- orig/configure	2019-08-19 06:29:31.000000000 +0200
+++ patched/configure	2019-09-17 18:28:10.204286856 +0200
@@ -24036,12 +24036,8 @@
 
 fi
 
-#  Check if we can assume whether All Warnings Are Really Errors
-if test xyes = x$ac_cv_cxx_compiler_gnu; then :
-  gcc_AWARE=-Werror
-fi
 
-AM_CXXFLAGS="-Wall $gcc_AWARE $code_coverage_cxxflags"
+AM_CXXFLAGS="-Wall $code_coverage_cxxflags"
 
 AM_LDFLAGS=""
 
diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2019-08-19 06:12:40.000000000 +0200
+++ patched/configure.ac	2019-09-17 18:27:25.372092219 +0200
@@ -201,7 +201,7 @@
 
 dnl  checks for programs, modules and libraries
 
-AM_INIT_AUTOMAKE([1.11.2 gnu tar-ustar -Wall -Werror std-options nostdinc subdir-objects])
+AM_INIT_AUTOMAKE([1.11.2 gnu tar-ustar -Wall std-options nostdinc subdir-objects])
 AM_MAINTAINER_MODE([enable])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
@@ -611,10 +611,6 @@
   [AC_SUBST([AM_CPPFLAGS],
      ['-I$(top_builddir) $(BOOST_CPPFLAGS)'])])
 
-#  Check if we can assume whether All Warnings Are Really Errors
-AS_IF([test xyes = x$ac_cv_cxx_compiler_gnu],
-  [gcc_AWARE=-Werror])
-
 AC_SUBST([AM_CXXFLAGS], ["-Wall $gcc_AWARE $code_coverage_cxxflags"])
 AC_SUBST([AM_LDFLAGS], [""])
 
diff -Naur orig/upstream/boost/tools/build/v2/tools/clang-linux.jam patched/upstream/boost/tools/build/v2/tools/clang-linux.jam
--- orig/upstream/boost/tools/build/v2/tools/clang-linux.jam	2019-08-19 07:14:12.000000000 +0200
+++ patched/upstream/boost/tools/build/v2/tools/clang-linux.jam	2019-09-17 18:30:07.861546134 +0200
@@ -81,7 +81,7 @@
 toolset.flags clang-linux.compile OPTIONS <warnings>off : -w ;
 toolset.flags clang-linux.compile OPTIONS <warnings>on  : -Wall ;
 toolset.flags clang-linux.compile OPTIONS <warnings>all : -Wall -pedantic ;
-toolset.flags clang-linux.compile OPTIONS <warnings-as-errors>on : -Werror ;
+toolset.flags clang-linux.compile OPTIONS <warnings-as-errors>off : -Werror ;
 
 toolset.flags clang-linux.compile OPTIONS <debug-symbols>on : -g ;
 toolset.flags clang-linux.compile OPTIONS <profiling>on : -pg ;
diff -Naur orig/upstream/boost/tools/build/v2/tools/gcc.jam patched/upstream/boost/tools/build/v2/tools/gcc.jam
--- orig/upstream/boost/tools/build/v2/tools/gcc.jam	2019-08-19 07:14:12.000000000 +0200
+++ patched/upstream/boost/tools/build/v2/tools/gcc.jam	2019-09-17 18:29:58.561475541 +0200
@@ -375,7 +375,7 @@
 toolset.flags gcc.compile OPTIONS <warnings>off : -w ;
 toolset.flags gcc.compile OPTIONS <warnings>on  : -Wall ;
 toolset.flags gcc.compile OPTIONS <warnings>all : -Wall -pedantic ;
-toolset.flags gcc.compile OPTIONS <warnings-as-errors>on : -Werror ;
+toolset.flags gcc.compile OPTIONS <warnings-as-errors>off : -Werror ;
 
 toolset.flags gcc.compile OPTIONS <debug-symbols>on : -g ;
 toolset.flags gcc.compile OPTIONS <profiling>on : -pg ;
diff -Naur orig/upstream/boost/tools/build/v2/tools/gcc.py patched/upstream/boost/tools/build/v2/tools/gcc.py
--- orig/upstream/boost/tools/build/v2/tools/gcc.py	2019-08-19 07:14:12.000000000 +0200
+++ patched/upstream/boost/tools/build/v2/tools/gcc.py	2019-09-17 18:29:38.677328961 +0200
@@ -274,7 +274,7 @@
 flags('gcc.compile', 'OPTIONS', ['<warnings>off'], ['-w'])
 flags('gcc.compile', 'OPTIONS', ['<warnings>on'], ['-Wall'])
 flags('gcc.compile', 'OPTIONS', ['<warnings>all'], ['-Wall', '-pedantic'])
-flags('gcc.compile', 'OPTIONS', ['<warnings-as-errors>on'], ['-Werror'])
+flags('gcc.compile', 'OPTIONS', ['<warnings-as-errors>off'], ['-Werror'])
 
 flags('gcc.compile', 'OPTIONS', ['<debug-symbols>on'], ['-g'])
 flags('gcc.compile', 'OPTIONS', ['<profiling>on'], ['-pg'])
diff -Naur orig/upstream/boost/tools/build/v2/tools/pathscale.jam patched/upstream/boost/tools/build/v2/tools/pathscale.jam
--- orig/upstream/boost/tools/build/v2/tools/pathscale.jam	2019-08-19 07:14:12.000000000 +0200
+++ patched/upstream/boost/tools/build/v2/tools/pathscale.jam	2019-09-17 18:29:48.389399795 +0200
@@ -74,7 +74,7 @@
 flags pathscale.compile OPTIONS <warnings>off           : -woffall ;
 flags pathscale.compile OPTIONS <warnings>on            : -Wall ;
 flags pathscale.compile OPTIONS <warnings>all           : -Wall -pedantic ;
-flags pathscale.compile OPTIONS <warnings-as-errors>on  : -Werror ;
+flags pathscale.compile OPTIONS <warnings-as-errors>off  : -Werror ;
 
 flags pathscale.compile OPTIONS <debug-symbols>on : -ggdb ;
 flags pathscale.compile OPTIONS <profiling>on     : -pg ;
diff -Naur orig/upstream/boost/tools/build/v2/tools/qcc.jam patched/upstream/boost/tools/build/v2/tools/qcc.jam
--- orig/upstream/boost/tools/build/v2/tools/qcc.jam	2019-08-19 07:14:12.000000000 +0200
+++ patched/upstream/boost/tools/build/v2/tools/qcc.jam	2019-09-17 18:29:11.821054431 +0200
@@ -62,7 +62,7 @@
 
 toolset.flags qcc.compile OPTIONS <warnings>off : -w ;
 toolset.flags qcc.compile OPTIONS <warnings>all : -Wc,-Wall ;
-toolset.flags qcc.compile OPTIONS <warnings-as-errors>on : -Wc,-Werror ;
+toolset.flags qcc.compile OPTIONS <warnings-as-errors>off : -Wc,-Werror ;
 
 toolset.flags qcc.compile OPTIONS <profiling>on : -p ;
 
