diff -Naur a/sane/version.hpp b/sane/version.hpp
--- a/sane/version.hpp	2020-12-22 07:44:15.000000000 +0100
+++ b/sane/version.hpp	2022-03-18 23:30:49.316729912 +0100
@@ -37,7 +37,7 @@
 #include <sane/sane.h>
 }
 
-#if !((SANE_MAJOR == SANE_CURRENT_MAJOR) && (SANE_MINOR == 0))
+#if !((SANE_MAJOR == SANE_CURRENT_MAJOR) && (SANE_MINOR == 1))
 #error "SANE installation violates versioning portability constraints."
 #endif
 
