diff -Nur orig/lib/icons/angband-sdl.desktop mod/lib/icons/angband-sdl.desktop
--- orig/lib/icons/angband-sdl.desktop	2021-08-01 00:28:01.000000000 +0200
+++ mod/lib/icons/angband-sdl.desktop	2021-08-02 14:48:50.000000000 +0200
@@ -2,7 +2,8 @@
 Name=Angband (SDL)
 Type=Application
 Comment=A roguelike dungeon exploration game based on the books of J.R.R.Tolkien
-Exec=angband -msdl
+Comment[hu]=Labirintus szerepjáték J.R.R.Tolkien könyvei nyomán
+Exec=angband -msdl2
 TryExec=angband
-Icon=/usr/share/angband/xtra/icon/att-32.png
+Icon=/usr/share/angband/icons/att-32.png
 Categories=Game;RolePlaying;
diff -Nur orig/lib/icons/angband-x11.desktop mod/lib/icons/angband-x11.desktop
--- orig/lib/icons/angband-x11.desktop	2021-08-01 00:28:01.000000000 +0200
+++ mod/lib/icons/angband-x11.desktop	2021-08-02 14:50:18.000000000 +0200
@@ -2,7 +2,8 @@
 Name=Angband (X11)
 Type=Application
 Comment=A roguelike dungeon exploration game based on the books of J.R.R.Tolkien
+Comment[hu]=Labirintus szerepjáték J.R.R.Tolkien könyvei nyomán
 Exec=angband -mx11
 TryExec=angband
-Icon=/usr/share/angband/xtra/icon/att-32.png
+Icon=/usr/share/angband/icons/att-32.png
 Categories=Game;RolePlaying;
