diff -Naur orig/cmake/install_layout.cmake patched/cmake/install_layout.cmake
--- orig/cmake/install_layout.cmake	2018-07-02 09:34:08.000000000 +0200
+++ patched/cmake/install_layout.cmake	2018-07-06 23:51:39.554887771 +0200
@@ -179,21 +179,21 @@
 #
 SET(INSTALL_INCLUDEDIR_DEB              "include/mariadb")
 #
-SET(INSTALL_DOCDIR_DEB                  "share/doc")
-SET(INSTALL_DOCREADMEDIR_DEB            "share/doc")
+SET(INSTALL_DOCDIR_DEB                  "share/doc/mysql")
+SET(INSTALL_DOCREADMEDIR_DEB            "share/doc/mysql")
 SET(INSTALL_MANDIR_DEB                  "share/man")
 SET(INSTALL_INFODIR_DEB                 "share/info")
 #
 SET(INSTALL_SHAREDIR_DEB                "share")
 SET(INSTALL_MYSQLSHAREDIR_DEB           "share/mysql")
-SET(INSTALL_MYSQLTESTDIR_DEB            "mysql-test")
-SET(INSTALL_SQLBENCHDIR_DEB             ".")
+SET(INSTALL_MYSQLTESTDIR_DEB            "share/mysql-test")
+SET(INSTALL_SQLBENCHDIR_DEB             "")
 SET(INSTALL_SUPPORTFILESDIR_DEB         "share/mysql")
 #
 SET(INSTALL_MYSQLDATADIR_DEB            "/var/lib/mysql")
 
 SET(INSTALL_UNIX_ADDRDIR_DEB            "/var/run/mysqld/mysqld.sock")
-SET(INSTALL_SYSTEMD_UNITDIR_DEB         "/lib/systemd/system")
+SET(INSTALL_SYSTEMD_UNITDIR_DEB         "/usr/lib/systemd/system")
 SET(INSTALL_SYSTEMD_SYSUSERSDIR_DEB     "/usr/lib/sysusers.d")
 SET(INSTALL_SYSTEMD_TMPFILESDIR_DEB     "/usr/lib/tmpfiles.d")
 
diff -Naur orig/support-files/rpm/my.cnf patched/support-files/rpm/my.cnf
--- orig/support-files/rpm/my.cnf	2018-07-02 09:34:14.000000000 +0200
+++ patched/support-files/rpm/my.cnf	2018-07-06 23:53:25.812328946 +0200
@@ -7,5 +7,5 @@
 #
 # include all files from the config directory
 #
-!includedir /etc/my.cnf.d
+!includedir /etc/mysql/conf.d
 
