diff -Naur orig/scripts/mysql_install_db.sh patched/scripts/mysql_install_db.sh
--- orig/scripts/mysql_install_db.sh	2019-05-13 20:13:09.000000000 +0200
+++ patched/scripts/mysql_install_db.sh	2019-05-22 04:11:36.899396936 +0200
@@ -19,9 +19,9 @@
 #
 # All unrecognized arguments to this script are passed to mysqld.
 
-basedir=""
+basedir="/usr"
 builddir=""
-ldata="@localstatedir@"
+ldata="/var/lib/mysql"
 langdir=""
 srcdir=""
 log_error=""
