diff -Naur orig/scripts/mysql_install_db.sh patched/scripts/mysql_install_db.sh
--- orig/scripts/mysql_install_db.sh	2013-07-17 16:51:29.000000000 +0200
+++ patched/scripts/mysql_install_db.sh	2013-08-14 12:52:52.623271994 +0200
@@ -19,9 +19,9 @@
 #
 # All unrecognized arguments to this script are passed to mysqld.
 
-basedir=""
+basedir="/usr"
 builddir=""
-ldata="@localstatedir@"
+ldata="/var/lib/mysql"
 langdir=""
 srcdir=""
 
