diff -Naur orig/cmake/systemd.cmake patched/cmake/systemd.cmake
--- orig/cmake/systemd.cmake	2016-08-29 16:38:51.000000000 +0200
+++ patched/cmake/systemd.cmake	2016-09-17 16:22:24.457980160 +0200
@@ -63,8 +63,7 @@
                                ${INSTALL_SYSTEMD_UNITDIR}/mariadb@.service
                                ${INSTALL_SYSTEMD_UNITDIR}/mariadb@bootstrap.service.d/use_galera_new_cluster.conf")
         IF(DEB)
-          SET(SYSTEMD_EXECSTARTPRE "ExecStartPre=/usr/bin/install -m 755 -o mysql -g root -d /var/run/mysqld")
-          SET(SYSTEMD_EXECSTARTPOST "ExecStartPost=/etc/mysql/debian-start")
+          SET(SYSTEMD_EXECSTARTPOST "ExecStartPost=/usr/bin/mysqld-ping")
         ENDIF()
         MESSAGE(STATUS "Systemd features enabled")
       ELSE()

