diff -Naur orig/cmake/systemd.cmake patched/cmake/systemd.cmake
--- orig/cmake/systemd.cmake	2019-05-13 20:13:06.000000000 +0200
+++ patched/cmake/systemd.cmake	2019-05-22 04:13:33.300975988 +0200
@@ -46,8 +46,7 @@
         SET(HAVE_SYSTEMD TRUE)
         SET(SYSTEMD_SCRIPTS mariadb-service-convert galera_new_cluster galera_recovery)
         IF(DEB)
-          SET(SYSTEMD_EXECSTARTPRE "ExecStartPre=/usr/bin/install -m 755 -o mysql -g root -d /var/run/mysqld")
-          SET(SYSTEMD_EXECSTARTPOST "ExecStartPost=/etc/mysql/debian-start")
+          SET(SYSTEMD_EXECSTARTPOST "ExecStartPost=/usr/bin/mysqld-ping")
         ENDIF()
         MESSAGE_ONCE(systemd "Systemd features enabled")
       ELSE()
