diff -Nur orig/plugin-desktopswitch/translations/desktopswitch_hu.desktop mod/plugin-desktopswitch/translations/desktopswitch_hu.desktop
--- orig/plugin-desktopswitch/translations/desktopswitch_hu.desktop	2017-10-21 21:47:44.000000000 +0200
+++ mod/plugin-desktopswitch/translations/desktopswitch_hu.desktop	2017-12-06 22:42:41.740701892 +0100
@@ -9,4 +9,4 @@
 
 # Translations
 Comment[hu]=Lehetővé teszi a virtuális asztalok közötti váltást
-Name[hu]=Aszatlváltó
+Name[hu]=Asztalváltó
