diff -Nur orig/panel/resources/panel.conf mod/panel/resources/panel.conf
--- orig/panel/resources/panel.conf	2023-11-05 12:56:26.000000000 +0100
+++ mod/panel/resources/panel.conf	2023-11-10 16:46:09.379881335 +0100
@@ -15,6 +15,11 @@
 [quicklaunch]
 type=quicklaunch
 alignment=Left
+apps/size=4
+apps/1/desktop=firefox.desktop
+apps/2/desktop=synaptic.desktop
+apps/3/desktop=pcmanfm-qt.desktop
+apps/4/desktop=lxqt-config.desktop
 
 [taskbar]
 type=taskbar
