diff -Naur guvcview-src-2.0.6-orig/guvcview/gui_gtk3_callbacks.c guvcview-src-2.0.6/guvcview/gui_gtk3_callbacks.c
--- guvcview-src-2.0.6-orig/guvcview/gui_gtk3_callbacks.c	2018-10-07 00:15:28.000000000 +0200
+++ guvcview-src-2.0.6/guvcview/gui_gtk3_callbacks.c	2018-11-02 10:36:24.370868637 +0100
@@ -2158,9 +2158,9 @@
 	gtk_grid_attach (GTK_GRID(table), lbl_me_method, 0, line, 1 ,1);
 	gtk_widget_show (lbl_me_method);
 
-	GtkWidget *me_method = gtk_spin_button_new_with_range(1,10,1);
+	GtkWidget *me_method = gtk_entry_new();
 	gtk_editable_set_editable(GTK_EDITABLE(me_method),TRUE);
-	gtk_spin_button_set_value (GTK_SPIN_BUTTON(me_method), defaults->me_method);
+	gtk_entry_set_text(me_method, defaults->me_method);
 
 	gtk_grid_attach (GTK_GRID(table), me_method, 1, line, 1 ,1);
 	gtk_widget_show (me_method);
@@ -2246,7 +2246,7 @@
 			defaults->qblur = (float) gtk_spin_button_get_value (GTK_SPIN_BUTTON(qblur));
 			defaults->subq = gtk_spin_button_get_value_as_int (GTK_SPIN_BUTTON(subq));
 			defaults->framerefs = gtk_spin_button_get_value_as_int (GTK_SPIN_BUTTON(framerefs));
-			defaults->me_method = gtk_spin_button_get_value_as_int (GTK_SPIN_BUTTON(me_method));
+			defaults->me_method = gtk_entry_get_text (me_method);
 			defaults->mb_decision = gtk_spin_button_get_value_as_int (GTK_SPIN_BUTTON(mb_decision));
 			defaults->max_b_frames = gtk_spin_button_get_value_as_int (GTK_SPIN_BUTTON(max_b_frames));
 			defaults->num_threads = gtk_spin_button_get_value_as_int (GTK_SPIN_BUTTON(num_threads));
diff -Naur guvcview-src-2.0.6-orig/guvcview/gui_qt5_callbacks.cpp guvcview-src-2.0.6/guvcview/gui_qt5_callbacks.cpp
--- guvcview-src-2.0.6-orig/guvcview/gui_qt5_callbacks.cpp	2018-10-07 00:15:28.000000000 +0200
+++ guvcview-src-2.0.6/guvcview/gui_qt5_callbacks.cpp	2018-11-02 10:39:09.777873129 +0100
@@ -1590,10 +1590,8 @@
 	framerefs->setValue(defaults->framerefs);
 	form.addRow(_("framerefs:   "), framerefs);
 	/*me method*/
-	QSpinBox *me_method = new QSpinBox(&dialog);
-	me_method->setRange(1, 10);
-	me_method->setSingleStep(1);						
-	me_method->setValue(defaults->me_method);
+	QLabel *me_method = new QLabel(&dialog);
+	me_method->setText(defaults->me_method);
 	form.addRow(_("me method:   "), me_method);
 	/*mb decision*/
 	QSpinBox *mb_decision = new QSpinBox(&dialog);
@@ -1645,7 +1643,7 @@
 		defaults->qblur = qblur->value();
 		defaults->subq = subq->value();
 		defaults->framerefs = framerefs->value();
-		defaults->me_method = me_method->value();
+		defaults->me_method = me_method->text().toLatin1().data();
 		defaults->mb_decision = mb_decision->value();
 		defaults->max_b_frames = max_b_frames->value();
 		defaults->num_threads = num_threads->value();
diff -Naur guvcview-src-2.0.6-orig/gview_encoder/encoder.c guvcview-src-2.0.6/gview_encoder/encoder.c
--- guvcview-src-2.0.6-orig/gview_encoder/encoder.c	2018-10-07 00:15:28.000000000 +0200
+++ guvcview-src-2.0.6/gview_encoder/encoder.c	2018-11-02 10:40:57.859870285 +0100
@@ -443,10 +443,8 @@
 	video_codec_data->codec_context->trellis = video_defaults->trellis;
 
 	/*motion estimation method */
-	if( video_defaults->codec_id == AV_CODEC_ID_H264 && video_defaults->me_method > 4)
-		video_defaults->me_method = X264_ME_HEX;
-
-	av_dict_set_int(&video_codec_data->private_options, "motion-est", video_defaults->me_method, 0);
+	if(video_defaults->me_method)
+		av_dict_set(&video_codec_data->private_options, "motion-est", video_defaults->me_method, 0);
 	av_dict_set_int(&video_codec_data->private_options, "mpeg_quant", video_defaults->mpeg_quant, 0);
 	av_dict_set_int(&video_codec_data->private_options, "mepre", video_defaults->pre_me, 0);
 
diff -Naur guvcview-src-2.0.6-orig/gview_encoder/gviewencoder.h guvcview-src-2.0.6/gview_encoder/gviewencoder.h
--- guvcview-src-2.0.6-orig/gview_encoder/gviewencoder.h	2018-10-07 00:15:28.000000000 +0200
+++ guvcview-src-2.0.6/gview_encoder/gviewencoder.h	2018-11-02 10:42:04.169112954 +0100
@@ -104,7 +104,7 @@
 	char codec_name[20];      //lavc codec_name
 	int mb_decision;          //lavc mb_decision
 	int trellis;              //lavc trellis quantization
-	int me_method;            //lavc motion estimation method
+	const char* me_method;            //lavc motion estimation method
 	int mpeg_quant;           //lavc mpeg quantization
 	int max_b_frames;         //lavc max b frames
 	int num_threads;          //lavc num threads
diff -Naur guvcview-src-2.0.6-orig/gview_encoder/video_codecs.c guvcview-src-2.0.6/gview_encoder/video_codecs.c
--- guvcview-src-2.0.6-orig/gview_encoder/video_codecs.c	2018-10-07 00:15:28.000000000 +0200
+++ guvcview-src-2.0.6/gview_encoder/video_codecs.c	2018-11-02 10:46:37.041776521 +0100
@@ -97,7 +97,7 @@
 		.codec_name   = "none",
 		.mb_decision  = 0,
 		.trellis      = 0,
-		.me_method    = 0,
+		.me_method    = NULL,
 		.mpeg_quant   = 0,
 		.max_b_frames = 0,
 		.num_threads  = 0,
@@ -385,7 +385,7 @@
 		.codec_name   = "libx264",
 		.mb_decision  = FF_MB_DECISION_RD,
 		.trellis      = 0,
-		.me_method    = X264_ME_HEX,
+		.me_method    = "hex",
 		.mpeg_quant   = 1,
 		.max_b_frames = 16,
 		.num_threads  = 4,
