diff -Nur orig/configure mod/configure
--- orig/configure	2017-04-09 08:46:13.000000000 +0200
+++ mod/configure	2017-10-26 15:24:07.517461359 +0200
@@ -23323,7 +23323,7 @@
               PCIDEV_MODULE_MAKE_ALL="all-kernel24"
               KERNEL_MODULE_SUFFIX="o"
               ;;
-            2.6*|3.*)
+            2.6*|3.*|4.*)
               PCIDEV_MODULE_MAKE_ALL="all-kernel26"
               KERNEL_MODULE_SUFFIX="ko"
               ;;
