diff -Nur orig/gtk/src/fr.handbrake.ghb.desktop mod/gtk/src/fr.handbrake.ghb.desktop
--- orig/gtk/src/fr.handbrake.ghb.desktop	2018-09-04 21:35:01.000000000 +0200
+++ mod/gtk/src/fr.handbrake.ghb.desktop	2018-11-02 08:17:33.147030024 +0100
@@ -3,7 +3,7 @@
 GenericName=Media Transcoder
 Comment=Transcodes DVD, Bluray, and other media
 Exec=ghb %f
-Icon=fr.handbrake.ghb.svg
+Icon=fr.handbrake.ghb
 Terminal=false
 Type=Application
 Categories=GTK;AudioVideo;Video;
