diff -Naur orig/contrib/ffmpeg/module.defs patched/contrib/ffmpeg/module.defs
--- orig/contrib/ffmpeg/module.defs	2020-06-13 16:05:35.000000000 +0200
+++ patched/contrib/ffmpeg/module.defs	2021-01-11 00:08:04.674038579 +0100
@@ -1,4 +1,4 @@
-__deps__ := BZIP2 ZLIB FDKAAC LIBDAV1D LIBVPX LAME LIBOPUS LIBSPEEX XZ
+__deps__ := BZIP2 ZLIB FDKAAC LIBVPX LAME LIBOPUS LIBSPEEX XZ
 ifeq (1,$(FEATURE.qsv))
 __deps__ += LIBMFX
 endif
@@ -55,8 +55,8 @@
     --enable-encoder=libvpx_vp8 \
     --enable-encoder=libvpx_vp9 \
     --disable-decoder=*_crystalhd \
-    --enable-libdav1d \
-    --enable-decoder=libdav1d \
+    --disable-libdav1d \
+    --disable-decoder=libdav1d \
     --cc="$(FFMPEG.GCC.gcc)" \
     --extra-ldflags="$(call fn.ARGS,FFMPEG.GCC,*archs *sysroot *minver ?extra) -L$(call fn.ABSOLUTE,$(CONTRIB.build/)lib)"
 
