diff -Nur orig/make/include/main.defs mod/make/include/main.defs
--- orig/make/include/main.defs	2016-12-29 22:07:59.000000000 +0100
+++ mod/make/include/main.defs	2017-01-20 15:26:18.638086173 +0100
@@ -74,13 +74,8 @@
 endif
 
 ifeq (1,$(FEATURE.x265))
-    MODULES += contrib/x265
 endif
 
-MODULES += contrib/ffmpeg
-MODULES += contrib/libvpx
-MODULES += contrib/libdvdread
-MODULES += contrib/libdvdnav
 MODULES += contrib/libbluray
 
 ifneq (,$(filter $(BUILD.system),mingw))
