diff -Nur orig/make/include/gcc.defs mod/make/include/gcc.defs
--- orig/make/include/gcc.defs	2017-04-09 22:24:50.000000000 +0200
+++ mod/make/include/gcc.defs	2017-10-27 11:27:39.270382309 +0200
@@ -62,7 +62,7 @@
 GCC.args.g.max     = -gdwarf-2 -g3
 GCC.args.O.none    = -O0
 GCC.args.O.size    = -Os
-GCC.args.O.speed   = -O3
+GCC.args.O.speed   = -O2
 GCC.args.D         = -D$(1)
 GCC.args.I         = -I$(1)
 GCC.args.muldefs   = -Wl,--allow-multiple-definition
