diff -Nur orig/src/geonamessearch.c mod/src/geonamessearch.c
--- orig/src/geonamessearch.c	2020-01-30 09:34:43.000000000 +0100
+++ mod/src/geonamessearch.c	2020-02-11 16:22:01.263459794 +0100
@@ -19,6 +19,18 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  *
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include <math.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <glib.h>
+#include <glib/gstdio.h>
+#include <glib/gprintf.h>
+#include <glib/gi18n.h>
+
 #include "viking.h"
 #include "geonamessearch.h"
 
