diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2017-03-08 18:54:14.134739579 +0100
+++ patched/CMakeLists.txt	2017-06-25 22:55:12.754128122 +0200
@@ -22,6 +22,10 @@
   cmake_policy(SET CMP0063 NEW)
 endif(POLICY CMP0063)
 
+if(POLICY CMP0002)
+  cmake_policy(SET CMP0002 OLD)
+endif(POLICY CMP0002)
+
 find_package(Qt5Core)
 find_package(ECM 1.0.0 REQUIRED NO_MODULE)
 set(CMAKE_MODULE_PATH
