From: Zdenek Dohnal <zdohnal@redhat.com>
Date: Fri, 24 Jul 2020 10:42:18 +0200
Subject: Declare syslog_message correctly for GCC-10 FTBFS

---
 ppa_syslog.h | 2 +-
 syslog.c     | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/ppa_syslog.h b/ppa_syslog.h
index a0fcb21..29de142 100644
--- a/ppa_syslog.h
+++ b/ppa_syslog.h
@@ -39,7 +39,7 @@
 #include <syslog.h>
 #endif
 
-char syslog_message[128];
+extern char syslog_message[128];
 extern size_t message_size ;
 
 void wrap_syslog( int, char *, char *);
diff --git a/syslog.c b/syslog.c
index 9b11c55..dccab91 100644
--- a/syslog.c
+++ b/syslog.c
@@ -31,6 +31,8 @@
 #include "ppa_syslog.h"
 #include "global.h"
 
+char syslog_message[128];
+
 BOOLEAN verbose = false ;
 
 /*
