diff -Naurd ncftp-3.1.8.orig/ncftp/ls.c ncftp-3.1.8/ncftp/ls.c
--- ncftp-3.1.8.orig/ncftp/ls.c	2004-01-06 22:39:38.000000000 +0100
+++ ncftp-3.1.8/ncftp/ls.c	2005-09-06 18:41:29.390401513 +0200
@@ -51,7 +51,7 @@
 void InitLsMonths(void)
 {
 	struct tm lt;
-	int i;
+	int i, j, k;
 
 	(void) Localtime(0, &lt);
 	lt.tm_mday = 15;
@@ -59,6 +59,13 @@
 	for (i=0; i<12; i++) {
 		lt.tm_mon = i;
 		(void) strftime(gLsMon[i], sizeof(gLsMon[i]) - 1, "%b", &lt);
+		j = k = 0;
+		while (k < 4 && j < sizeof(gLsMon[i])) {
+			unsigned char c = gLsMon[i][j];
+			if (c < 128 || c >= 192) k++;
+			j++;
+		}
+		if (k == 4) gLsMon[i][j-1] = '\0';
 	}
 	(void) strcpy(gLsMon[i], "BUG");
 }	/* InitLsMonths */
