diff -Naur orig/pppd/plugins/pppol2tp/Makefile.linux patched/pppd/plugins/pppol2tp/Makefile.linux
--- orig/pppd/plugins/pppol2tp/Makefile.linux	2019-12-31 02:31:26.000000000 +0100
+++ patched/pppd/plugins/pppol2tp/Makefile.linux	2020-03-26 00:01:50.666386785 +0100
@@ -6,7 +6,7 @@
 
 #***********************************************************************
 
-DESTDIR = @DESTDIR@
+DESTDIR = $(INSTROOT)@DESTDIR@
 LIBDIR = $(DESTDIR)/lib/pppd/$(VERSION)
 
 VERSION = $(shell awk -F '"' '/VERSION/ { print $$2; }' ../../patchlevel.h)
