diff -Naur orig/source/common/unicode/urename.h patched/source/common/unicode/urename.h
--- orig/source/common/unicode/urename.h	2018-10-15 20:02:37.000000000 +0200
+++ patched/source/common/unicode/urename.h	2018-10-31 18:28:24.496126045 +0100
@@ -110,7 +110,6 @@
 #define _UTF7Data U_ICU_ENTRY_POINT_RENAME(_UTF7Data)
 #define _UTF8Data U_ICU_ENTRY_POINT_RENAME(_UTF8Data)
 #define allowedHourFormatsCleanup U_ICU_ENTRY_POINT_RENAME(allowedHourFormatsCleanup)
-#define checkImpl U_ICU_ENTRY_POINT_RENAME(checkImpl)
 #define cmemory_cleanup U_ICU_ENTRY_POINT_RENAME(cmemory_cleanup)
 #define dayPeriodRulesCleanup U_ICU_ENTRY_POINT_RENAME(dayPeriodRulesCleanup)
 #define deleteAllowedHourFormats U_ICU_ENTRY_POINT_RENAME(deleteAllowedHourFormats)
diff -Naur orig/source/i18n/uspoof.cpp patched/source/i18n/uspoof.cpp
--- orig/source/i18n/uspoof.cpp	2018-09-29 02:34:42.000000000 +0200
+++ patched/source/i18n/uspoof.cpp	2018-10-31 18:28:24.496126045 +0100
@@ -547,7 +547,7 @@
     return uspoof_check2UnicodeString(sc, id, NULL, status);
 }
 
-int32_t checkImpl(const SpoofImpl* This, const UnicodeString& id, CheckResult* checkResult, UErrorCode* status) {
+static int32_t checkImpl(const SpoofImpl* This, const UnicodeString& id, CheckResult* checkResult, UErrorCode* status) {
     U_ASSERT(This != NULL);
     U_ASSERT(checkResult != NULL);
     checkResult->clear();
