diff -Naur a/configure b/configure
--- a/configure	2022-10-06 18:59:23.000000000 +0200
+++ b/configure	2022-10-11 20:15:14.413300199 +0200
@@ -2608,6 +2608,9 @@
     --target-arch=*)
       TARGET_MACHINE=`echo $option | cut -d'=' -f2`
       ;;
+	--target-cpu=*)
+      TARGET_CPU=`echo $option | cut -d'=' -f2`
+      ;;
     --disable-xpcom)
       [ $WITH_XPCOM -eq 1 ] && WITH_XPCOM=0
       ;;
