diff -Naur orig/configure patched/configure
--- orig/configure	2018-05-09 18:31:59.000000000 +0200
+++ patched/configure	2018-06-12 11:04:30.071607560 +0200
@@ -2474,6 +2474,9 @@
     --target-arch=*)
       TARGET_MACHINE=`echo $option | cut -d'=' -f2`
       ;;
+	--target-cpu=*)
+      TARGET_CPU=`echo $option | cut -d'=' -f2`
+      ;;
     --disable-xpcom)
       [ $WITH_XPCOM -eq 1 ] && WITH_XPCOM=0
       ;;
