diff -Naur orig/configure patched/configure
--- orig/configure	2019-12-10 18:43:06.000000000 +0100
+++ patched/configure	2019-12-31 17:48:58.830536369 +0100
@@ -2503,6 +2503,9 @@
     --target-arch=*)
       TARGET_MACHINE=`echo $option | cut -d'=' -f2`
       ;;
+	--target-cpu=*)
+      TARGET_CPU=`echo $option | cut -d'=' -f2`
+      ;;
     --disable-xpcom)
       [ $WITH_XPCOM -eq 1 ] && WITH_XPCOM=0
       ;;
