diff -Naur orig/configure patched/configure
--- orig/configure	2018-06-12 11:06:17.920906662 +0200
+++ patched/configure	2018-06-12 11:14:32.422928519 +0200
@@ -440,7 +440,7 @@
       elif [ $cc_maj -lt 4 \
              -o \( $cc_maj -eq 4 -a $cc_min -lt 4 -a "$OS" != "darwin" \) \
              -o \( $cc_maj -eq 4 -a $cc_min -lt 2 -a "$OS"  = "darwin" \) \
-             -o $cc_maj -gt 7 ]; then
+             -o $cc_maj -gt 8 ]; then
         log_failure "gcc version $cc_maj.$cc_min found, expected gcc 4.x...7.x"
         fail really
       else
