diff -Naur orig/configure patched/configure
--- orig/configure	2022-01-02 21:49:31.231714970 +0100
+++ patched/configure	2022-01-02 21:57:14.957171966 +0100
@@ -2027,7 +2027,7 @@
 }
 EOF
   found=
-  SUPPYTHONLIBS="python2.7 python2.6 python3.1 python3.2 python3.3 python3.4 python3.4m python3.5 python3.5m python3.6 python3.6m python3.7 python3.7m python3.8 python3.8m python3.9 python3.9m"
+  SUPPYTHONLIBS="python3.8 python3.8m python3.9 python3.9m python3.10 python3.10m"
   for p in $PYTHONDIR; do
     for d in $SUPPYTHONLIBS; do
       for b in lib/x86_64-linux-gnu lib/i386-linux-gnu lib64 lib/64 lib; do
