diff -Naur orig/src/VBox/Frontends/VirtualBox/src/globals/UIImageTools.cpp patched/src/VBox/Frontends/VirtualBox/src/globals/UIImageTools.cpp
--- orig/src/VBox/Frontends/VirtualBox/src/globals/UIImageTools.cpp	2020-05-14 20:40:35.000000000 +0200
+++ patched/src/VBox/Frontends/VirtualBox/src/globals/UIImageTools.cpp	2020-06-03 21:49:36.407338746 +0200
@@ -17,6 +17,7 @@
 
 /* Qt includes: */
 #include <QPainter>
+#include <QPainterPath>
 
 /* GUI include */
 #include "UIImageTools.h"
diff -Naur orig/src/VBox/Frontends/VirtualBox/src/widgets/UIMenuToolBar.cpp patched/src/VBox/Frontends/VirtualBox/src/widgets/UIMenuToolBar.cpp
--- orig/src/VBox/Frontends/VirtualBox/src/widgets/UIMenuToolBar.cpp	2020-05-14 20:40:44.000000000 +0200
+++ patched/src/VBox/Frontends/VirtualBox/src/widgets/UIMenuToolBar.cpp	2020-06-03 22:45:01.020379811 +0200
@@ -19,6 +19,7 @@
 #include <QApplication>
 #include <QHBoxLayout>
 #include <QPainter>
+#include <QPainterPath>
 #include <QStyle>
 #include <QToolButton>
 
diff -Naur orig/src/VBox/Frontends/VirtualBox/src/widgets/UIMiniToolBar.cpp patched/src/VBox/Frontends/VirtualBox/src/widgets/UIMiniToolBar.cpp
--- orig/src/VBox/Frontends/VirtualBox/src/widgets/UIMiniToolBar.cpp	2020-05-14 20:40:44.000000000 +0200
+++ patched/src/VBox/Frontends/VirtualBox/src/widgets/UIMiniToolBar.cpp	2020-06-03 22:34:19.163355304 +0200
@@ -21,6 +21,7 @@
 #include <QMenu>
 #include <QMoveEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QStateMachine>
 #include <QStyle>
 #include <QTimer>
diff -Naur orig/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.cpp patched/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.cpp
--- orig/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.cpp	2020-05-14 20:40:44.000000000 +0200
+++ patched/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.cpp	2020-06-03 21:58:47.884019691 +0200
@@ -19,6 +19,7 @@
 #include <QApplication>
 #include <QLabel>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPaintEvent>
 #include <QStyle>
 #include <QVBoxLayout>
diff -Naur orig/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.h patched/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.h
--- orig/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.h	2020-05-14 20:40:44.000000000 +0200
+++ patched/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.h	2020-06-03 22:14:42.789435626 +0200
@@ -23,6 +23,7 @@
 
 /* Qt includes: */
 #include <QIcon>
+#include <QPainterPath>
 #include <QWidget>
 
 /* GUI includes: */
