diff -Naur orig/src/VBox/Frontends/VirtualBox/src/settings/global/UIGlobalSettingsProxy.cpp patched/src/VBox/Frontends/VirtualBox/src/settings/global/UIGlobalSettingsProxy.cpp
--- orig/src/VBox/Frontends/VirtualBox/src/settings/global/UIGlobalSettingsProxy.cpp	2018-05-09 18:38:05.000000000 +0200
+++ patched/src/VBox/Frontends/VirtualBox/src/settings/global/UIGlobalSettingsProxy.cpp	2018-06-12 23:57:55.653388836 +0200
@@ -21,6 +21,8 @@
 
 /* Qt includes: */
 # include <QRegExpValidator>
+# include <QObject>
+# include <QButtonGroup>
 
 /* GUI includes: */
 # include "QIWidgetValidator.h"
