diff -Naur orig/src/VBox/Frontends/VirtualBox/src/settings/global/UIGlobalSettingsProxy.cpp patched/src/VBox/Frontends/VirtualBox/src/settings/global/UIGlobalSettingsProxy.cpp
--- orig/src/VBox/Frontends/VirtualBox/src/settings/global/UIGlobalSettingsProxy.cpp	2019-12-10 18:52:30.000000000 +0100
+++ patched/src/VBox/Frontends/VirtualBox/src/settings/global/UIGlobalSettingsProxy.cpp	2019-12-31 17:51:13.380514019 +0100
@@ -18,6 +18,8 @@
 /* Qt includes: */
 #include <QButtonGroup>
 #include <QRegExpValidator>
+#include <QObject>
+#include <QButtonGroup>
 
 /* GUI includes: */
 #include "QIWidgetValidator.h"
@@ -26,6 +28,7 @@
 #include "UIMessageCenter.h"
 #include "UICommon.h"
 #include "VBoxUtils.h"
+#include "QIWidgetValidator.h"
 
 /* COM includes: */
 #include "CSystemProperties.h"
