diff -Naur orig/lang/calamares_hu.ts patched/lang/calamares_hu.ts
--- orig/lang/calamares_hu.ts	2020-05-11 20:53:23.000000000 +0200
+++ patched/lang/calamares_hu.ts	2020-05-18 01:24:00.594292519 +0200
@@ -848,7 +848,7 @@
     <message>
       <location filename="../src/modules/partition/gui/CreatePartitionDialog.ui" line="188"/>
       <source>Flags:</source>
-      <translation>Zászlók:</translation>
+      <translation>Jelzők:</translation>
     </message>
     <message>
       <location filename="../src/modules/partition/gui/CreatePartitionDialog.cpp" line="74"/>
@@ -1185,7 +1185,7 @@
     <message>
       <location filename="../src/modules/partition/gui/EditExistingPartitionDialog.ui" line="149"/>
       <source>Flags:</source>
-      <translation>Zászlók:</translation>
+      <translation>Jelzők:</translation>
     </message>
     <message>
       <location filename="../src/modules/partition/gui/EditExistingPartitionDialog.cpp" line="278"/>
@@ -3149,12 +3149,12 @@
     <message>
       <location filename="../src/modules/partition/jobs/SetPartitionFlagsJob.cpp" line="52"/>
       <source>Set flags on partition %1.</source>
-      <translation>Zászlók beállítása a partíción %1.</translation>
+      <translation>Jelzők beállítása a partíción: %1.</translation>
     </message>
     <message>
       <location filename="../src/modules/partition/jobs/SetPartitionFlagsJob.cpp" line="58"/>
       <source>Set flags on %1MiB %2 partition.</source>
-      <translation>flags beállítása a %1MiB %2 partíción.</translation>
+      <translation>Jelzők beállítása a %1MiB %2 partíción.</translation>
     </message>
     <message>
       <location filename="../src/modules/partition/jobs/SetPartitionFlagsJob.cpp" line="62"/>
@@ -3164,12 +3164,12 @@
     <message>
       <location filename="../src/modules/partition/jobs/SetPartitionFlagsJob.cpp" line="74"/>
       <source>Clear flags on partition &lt;strong&gt;%1&lt;/strong&gt;.</source>
-      <translation>Zászlók törlése a partíción: &lt;strong&gt;%1&lt;/strong&gt;.</translation>
+      <translation>Jelzők törlése a partíción: &lt;strong&gt;%1&lt;/strong&gt;.</translation>
     </message>
     <message>
       <location filename="../src/modules/partition/jobs/SetPartitionFlagsJob.cpp" line="80"/>
       <source>Clear flags on %1MiB &lt;strong&gt;%2&lt;/strong&gt; partition.</source>
-      <translation>flags eltávolítása a %1MiB &lt;strong&gt;%2&lt;/strong&gt; partíción.</translation>
+      <translation>Jelzők eltávolítása a %1MiB &lt;strong&gt;%2&lt;/strong&gt; partíción.</translation>
     </message>
     <message>
       <location filename="../src/modules/partition/jobs/SetPartitionFlagsJob.cpp" line="84"/>
@@ -3179,7 +3179,7 @@
     <message>
       <location filename="../src/modules/partition/jobs/SetPartitionFlagsJob.cpp" line="89"/>
       <source>Flag partition &lt;strong&gt;%1&lt;/strong&gt; as &lt;strong&gt;%2&lt;/strong&gt;.</source>
-      <translation>Zászlók beállítása &lt;strong&gt;%1&lt;/strong&gt; ,mint &lt;strong&gt;%2&lt;/strong&gt;.</translation>
+      <translation>Jelzők beállítása &lt;strong&gt;%1&lt;/strong&gt; ,mint &lt;strong&gt;%2&lt;/strong&gt;.</translation>
     </message>
     <message>
       <location filename="../src/modules/partition/jobs/SetPartitionFlagsJob.cpp" line="98"/>
@@ -3194,27 +3194,27 @@
     <message>
       <location filename="../src/modules/partition/jobs/SetPartitionFlagsJob.cpp" line="117"/>
       <source>Clearing flags on partition &lt;strong&gt;%1&lt;/strong&gt;.</source>
-      <translation>Zászlók törlése a partíción: &lt;strong&gt;%1&lt;/strong&gt;.</translation>
+      <translation>Jelzők törlése a partíción: &lt;strong&gt;%1&lt;/strong&gt;.</translation>
     </message>
     <message>
       <location filename="../src/modules/partition/jobs/SetPartitionFlagsJob.cpp" line="123"/>
       <source>Clearing flags on %1MiB &lt;strong&gt;%2&lt;/strong&gt; partition.</source>
-      <translation>Flag-ek eltávolítása a %1MiB &lt;strong&gt;%2&lt;/strong&gt; partíción.</translation>
+      <translation>Jelzők eltávolítása a %1MiB &lt;strong&gt;%2&lt;/strong&gt; partíción.</translation>
     </message>
     <message>
       <location filename="../src/modules/partition/jobs/SetPartitionFlagsJob.cpp" line="128"/>
       <source>Clearing flags on new partition.</source>
-      <translation>jelzők törlése az új partíción.</translation>
+      <translation>Jelzők törlése az új partíción.</translation>
     </message>
     <message>
       <location filename="../src/modules/partition/jobs/SetPartitionFlagsJob.cpp" line="133"/>
       <source>Setting flags &lt;strong&gt;%2&lt;/strong&gt; on partition &lt;strong&gt;%1&lt;/strong&gt;.</source>
-      <translation>Zászlók beállítása &lt;strong&gt;%2&lt;/strong&gt; a &lt;strong&gt;%1&lt;/strong&gt; partíción.</translation>
+      <translation>Jelzők beállítása &lt;strong&gt;%2&lt;/strong&gt; a &lt;strong&gt;%1&lt;/strong&gt; partíción.</translation>
     </message>
     <message>
       <location filename="../src/modules/partition/jobs/SetPartitionFlagsJob.cpp" line="142"/>
       <source>Setting flags &lt;strong&gt;%3&lt;/strong&gt; on %1MiB &lt;strong&gt;%2&lt;/strong&gt; partition.</source>
-      <translation>Flag-ek beállítása &lt;strong&gt;%3&lt;/strong&gt; a %1MiB &lt;strong&gt;%2&lt;/strong&gt; partíción.</translation>
+      <translation>Jelzők beállítása &lt;strong&gt;%3&lt;/strong&gt; a %1MiB &lt;strong&gt;%2&lt;/strong&gt; partíción.</translation>
     </message>
     <message>
       <location filename="../src/modules/partition/jobs/SetPartitionFlagsJob.cpp" line="149"/>
@@ -3224,7 +3224,7 @@
     <message>
       <location filename="../src/modules/partition/jobs/SetPartitionFlagsJob.cpp" line="165"/>
       <source>The installer failed to set flags on partition %1.</source>
-      <translation>A telepítőnek nem sikerült a zászlók beállítása a partíción %1.</translation>
+      <translation>A telepítőnek nem sikerült a jelzők beállítása a partíción: %1.</translation>
     </message>
   </context>
   <context>
@@ -3232,7 +3232,7 @@
     <message>
       <location filename="../src/modules/users/SetPasswordJob.cpp" line="49"/>
       <source>Set password for user %1</source>
-      <translation>%1 felhasználó jelszó beállítása</translation>
+      <translation>%1 felhasználói jelszó beállítása</translation>
     </message>
     <message>
       <location filename="../src/modules/users/SetPasswordJob.cpp" line="56"/>
