diff -Naur a/lang/calamares_hu.ts b/lang/calamares_hu.ts
--- a/lang/calamares_hu.ts	2024-10-22 21:24:23.000000000 +0200
+++ b/lang/calamares_hu.ts	2024-10-29 15:26:29.864723503 +0100
@@ -4039,7 +4039,7 @@
     <message>
       <location filename="../src/modules/partition/jobs/SetPartitionFlagsJob.cpp" line="149"/>
       <source>The installer failed to set flags on partition %1.</source>
-      <translation>A telepítőnek nem sikerült a zászlók beállítása a partíción %1.</translation>
+      <translation>A telepítőnek nem sikerült a jelzők beállítása a partíción: %1.</translation>
     </message>
   </context>
   <context>
@@ -4047,7 +4047,7 @@
     <message>
       <location filename="../src/modules/users/SetPasswordJob.cpp" line="38"/>
       <source>Set password for user %1</source>
-      <translation>A(z) %1 nevű felhasználó jelszavának beállítása</translation>
+      <translation>%1 felhasználó jelszavának beállítása</translation>
     </message>
     <message>
       <location filename="../src/modules/users/SetPasswordJob.cpp" line="44"/>
