diff -Naur a/calamares.desktop b/calamares.desktop
--- a/calamares.desktop	2024-01-15 19:54:01.000000000 +0100
+++ b/calamares.desktop	2024-01-29 03:26:37.647807358 +0100
@@ -5,9 +5,9 @@
 GenericName=System Installer
 Keywords=calamares;system;installer;
 TryExec=calamares
-Exec=sh -c "pkexec calamares"
+Exec=calamares
 Comment=Calamares — System Installer
-Icon=calamares
+Icon=uhulogo
 Terminal=false
 StartupNotify=true
 Categories=Qt;System;
@@ -122,9 +122,9 @@
 Icon[hr]=calamares
 GenericName[hr]=Instalacija sustava
 Comment[hr]=Calamares — Instalacija sustava
-Name[hu]=Rendszer telepítése
+Name[hu]=UHU-Linux telepítése
 Icon[hu]=calamares
-GenericName[hu]=Rendszertelepítő
+GenericName[hu]=UHU-Linux telepítő
 Comment[hu]=Calamares – Rendszertelepítő
 Name[id]=Instal Sistem
 Icon[id]=calamares
diff -Naur a/calamares.desktop.in b/calamares.desktop.in
--- a/calamares.desktop.in	2024-01-15 19:54:01.000000000 +0100
+++ b/calamares.desktop.in	2024-01-29 02:35:56.302243509 +0100
@@ -5,9 +5,9 @@
 GenericName=System Installer
 Keywords=calamares;system;installer;
 TryExec=calamares
-Exec=sh -c "pkexec calamares"
+Exec=calamares
 Comment=Calamares — System Installer
-Icon=calamares
+Icon=uhulogo
 Terminal=false
 StartupNotify=true
 Categories=Qt;System;
