diff -Naur a/src/modules/bootloader/bootloader.conf b/src/modules/bootloader/bootloader.conf
--- a/src/modules/bootloader/bootloader.conf	2023-12-12 00:04:06.000000000 +0100
+++ b/src/modules/bootloader/bootloader.conf	2023-12-16 22:45:14.751242396 +0100
@@ -20,8 +20,8 @@
 
 # kernelSearchPath is the path relative to the root of the install to search for kernels
 # A kernel is identified by finding files which match regular expression, kernelPattern
-kernelSearchPath: "/usr/lib/modules"
-kernelPattern: "^vmlinuz.*"
+kernelSearchPath: "/lib/modules"
+kernelPattern: "^kernel.*"
 
 # loaderEntries is an array of options to add to loader.conf for systemd-boot
 # please note that the "default" option is added programmatically
