diff -Naur orig/src/modules/bootloader/bootloader.conf patched/src/modules/bootloader/bootloader.conf
--- orig/src/modules/bootloader/bootloader.conf	2019-07-06 00:52:35.000000000 +0200
+++ patched/src/modules/bootloader/bootloader.conf	2019-07-31 13:40:58.392981882 +0200
@@ -15,9 +15,9 @@
 
 # systemd-boot configuration files settings, set kernel and initramfs file names
 # and amount of time before default selection boots
-kernel: "/vmlinuz-linux"
-img: "/initramfs-linux.img"
-fallback: "/initramfs-linux-fallback.img"
+kernel: "/kernel"
+img: "/initrd"
+fallback: "/initrd-fallback.img"
 timeout: "10"
 
 # Optionally set the menu entry name and kernel name to use in systemd-boot.
