diff -Nur orig/src/branding/default/branding.desc mod/src/branding/default/branding.desc
--- orig/src/branding/default/branding.desc	2017-11-14 16:18:58.000000000 +0100
+++ mod/src/branding/default/branding.desc	2017-11-19 14:27:03.000000000 +0100
@@ -13,22 +13,21 @@
 welcomeExpandingLogo:   true
 
 strings:
-    productName:         Generic GNU/Linux
-    shortProductName:    Generic
-    version:             2017.8 LTS
-    shortVersion:        2017.8
-    versionedName:       Generic GNU/Linux 2017.8 LTS "Soapy Sousaphone"
-    shortVersionedName:  Generic 2017.8
-    bootloaderEntryName: Generic
-    productUrl:          http://calamares.io/
-    supportUrl:          http://calamares.io/bugs/
-    knownIssuesUrl:      http://calamares.io/about/
-    releaseNotesUrl:     http://calamares.io/about/
+    productName:         UHU-Linux
+    shortProductName:    UHU-Linux
+    version:             UBK2 (RIA)
+    shortVersion:        UBK2
+    versionedName:       UHU-Linux UBK2 (RIA)
+    shortVersionedName:  UHU-Linux UBK2
+    bootloaderEntryName: UHU-Linux
+    productUrl:          http://ubk.hu/
+    supportUrl:          http://lists.ubk.hu/listinfo/users
+#   knownIssuesUrl:      http://calamares.io/about/
+#   releaseNotesUrl:     http://calamares.io/about/
 
 images:
-    productLogo:         "squid.png"
-    productIcon:         "squid.png"
-    productWelcome:      "languages.png"
+    productLogo:         "uhulogo.png"
+    productIcon:         "uhu-192x192.png"
 
 slideshow:               "show.qml"
 
