diff -Nur orig/src/branding/default/branding.desc mod/src/branding/default/branding.desc
--- orig/src/branding/default/branding.desc	2020-02-27 23:26:29.000000000 +0100
+++ mod/src/branding/default/branding.desc	2020-03-07 17:35:03.083538343 +0100
@@ -73,16 +73,16 @@
 # are visible as buttons there if the corresponding *show* keys
 # are set to "true" (they can also be overridden).
 strings:
-    productName:         "@{NAME}"
-    shortProductName:    Generic
-    version:             2020.2 LTS
-    shortVersion:        2020.2
-    versionedName:       Fancy GNU/Linux 2020.2 LTS "Turgid Tuba"
-    shortVersionedName:  FancyGL 2020.2
-    bootloaderEntryName: FancyGL
-    productUrl:          https://calamares.io/
-    supportUrl:          https://github.com/calamares/calamares/issues
-    knownIssuesUrl:      https://calamares.io/about/
+    productName:         UHU-Linux
+    shortProductName:    UHU-Linux
+    version:             UBK4 (PAFF)
+    shortVersion:        UBK4
+    versionedName:       UHU-Linux UBK4 (PAFF)
+    shortVersionedName:  UHU-Linux UBK4
+    bootloaderEntryName: UHU-Linux
+    productUrl:          https://ubk.hu/
+    supportUrl:          https://github.com/uhulinux/ub-ubk4/issues
+    knownIssuesUrl:      https://github.com/uhulinux/ub-ubk4/issues
     releaseNotesUrl:     https://calamares.io/about/
 
 # These images are loaded from the branding module directory.
@@ -102,9 +102,9 @@
 #
 # These filenames can also use substitutions from os-release (see above).
 images:
-    productLogo:         "squid.png"
+    productLogo:         "uhulogo.png"
     productIcon:         "squid.png"
-    productWelcome:      "languages.png"
+    productWelcome:      "uhu-192x192.png"
 
 # The slideshow is displayed during execution steps (e.g. when the
 # installer is actually writing to disk and doing other slow things).
