diff -Naur orig/src/branding/default/branding.desc patched/src/branding/default/branding.desc
--- orig/src/branding/default/branding.desc	2020-12-04 14:51:48.000000000 +0100
+++ patched/src/branding/default/branding.desc	2020-12-06 19:05:58.524396406 +0100
@@ -121,18 +121,18 @@
 # are visible as buttons there if the corresponding *show* keys
 # are set to "true" (they can also be overridden).
 strings:
-    productName:         "@{NAME}"
-    shortProductName:    Generic
-    version:             2020.2 LTS
-    shortVersion:        2020.2
-    versionedName:       Fancy GNU/Linux 2020.2 LTS "Turgid Tuba"
-    shortVersionedName:  FancyGL 2020.2
-    bootloaderEntryName: FancyGL
-    productUrl:          https://calamares.io/
-    supportUrl:          https://github.com/calamares/calamares/wiki
-    knownIssuesUrl:      https://github.com/calamares/calamares/issues
-    releaseNotesUrl:     http://calamares.io/news/
-    donateUrl:           https://kde.org/community/donations/index.php
+    productName:         UHU-Linux
+    productName:         UHU-Linux
+    version:             UBK4 (PAFF)
+    version:             UBK4
+    versionedName:       UHU-Linux UBK4 (PAFF)
+    shortVersionedName:  UHU-Linux UBK4
+    bootloaderEntryName: UHU-Linux
+    productUrl:          https://ubk.hu/
+    supportUrl:          https://github.com/uhulinux/ub-ubk4/issues
+    knownIssuesUrl:      https://github.com/uhulinux/ub-ubk4/issues
+#    releaseNotesUrl:     http://calamares.io/news/
+#    donateUrl:           https://kde.org/community/donations/index.php
 
 # These images are loaded from the branding module directory.
 #
@@ -164,9 +164,9 @@
 images:
     # productBanner:       "banner.png"
     productIcon:         "squid.png"
-    productLogo:         "squid.png"
+    productLogo:         "uhulogo.png"
     # productWallpaper:    "wallpaper.png"
-    productWelcome:      "languages.png"
+    productWelcome:      "uhu-192x192.png"
 
 # Colors for text and background components.
 #
