diff -Naur orig/src/branding/default/branding.desc patched/src/branding/default/branding.desc
--- orig/src/branding/default/branding.desc	2020-01-28 17:42:21.000000000 +0100
+++ patched/src/branding/default/branding.desc	2020-01-30 22:12:19.346504654 +0100
@@ -67,17 +67,18 @@
 # are visible as buttons there if the corresponding *show* keys
 # are set to "true" (they can also be overridden).
 strings:
-    productName:         "@{NAME}"
-    shortProductName:    Generic
-    version:             2017.8 LTS
-    shortVersion:        2017.8
-    versionedName:       Generic GNU/Linux 2017.8 LTS "Soapy Sousaphone"
-    shortVersionedName:  Generic 2017.8
-    bootloaderEntryName: Generic
-    productUrl:          https://calamares.io/
-    supportUrl:          https://github.com/calamares/calamares/issues
-    knownIssuesUrl:      https://calamares.io/about/
-    releaseNotesUrl:     https://calamares.io/about/
+    productName:         UHU-Linux
+    shortProductName:    UHU-Linux
+    version:             UBK4 (RIA)
+    shortVersion:        UBK4
+    versionedName:       UHU-Linux UBK4 (PAFF)
+    shortVersionedName:  UHU-Linux UBK4
+    bootloaderEntryName: UHU-Linux
+    productUrl:          https://ubk.hu
+    productUrl:          http://ubk.hu/
+    supportUrl:          https://github.com/uhulinux/ub-ubk4/issues
+    knownIssuesUrl:      https://github.com/uhulinux/ub-ubk4/issues
+#    releaseNotesUrl:     https://calamares.io/about/
 
 # These images are loaded from the branding module directory.
 #
@@ -96,9 +97,8 @@
 #
 # These filenames can also use substitutions from os-release (see above).
 images:
-    productLogo:         "squid.png"
-    productIcon:         "squid.png"
-    productWelcome:      "languages.png"
+    productLogo:         "uhulogo.png"
+    productIcon:         "uhu-192x192.png"
 
 # The slideshow is displayed during execution steps (e.g. when the
 # installer is actually writing to disk and doing other slow things).
