diff -Nur orig/src/modules/chrootcfg/module.desc mod/src/modules/chrootcfg/module.desc
--- orig/src/modules/chrootcfg/module.desc	1970-01-01 01:00:00.000000000 +0100
+++ mod/src/modules/chrootcfg/module.desc	2017-11-19 14:27:03.000000000 +0100
@@ -0,0 +1,6 @@
+# Syntax is YAML 1.2
+---
+type:       "job"
+name:       "chrootcfg"
+interface:  "python"
+script:     "main.py"   #assumed relative to the current directory
