diff -Naur orig/src/modules/displaymanager/displaymanager.conf patched/src/modules/displaymanager/displaymanager.conf
--- orig/src/modules/displaymanager/displaymanager.conf	2021-04-14 13:05:11.000000000 +0200
+++ patched/src/modules/displaymanager/displaymanager.conf	2021-04-15 09:49:50.943732014 +0200
@@ -20,9 +20,7 @@
   - sddm
   - lightdm
   - gdm
-  - mdm
   - lxdm
-  - kdm
 
 # Enable the following settings to force a desktop environment
 # in your displaymanager configuration file. This will attempt
diff -Naur orig/src/modules/displaymanager/main.py patched/src/modules/displaymanager/main.py
--- orig/src/modules/displaymanager/main.py	2021-04-14 13:05:11.000000000 +0200
+++ patched/src/modules/displaymanager/main.py	2021-04-15 09:54:45.884353420 +0200
@@ -182,19 +182,14 @@
     DesktopEnvironment('/usr/bin/startlxde', 'LXDE'),
     DesktopEnvironment('/usr/bin/lxqt-session', 'lxqt'),
     DesktopEnvironment('/usr/bin/pekwm', 'pekwm'),
-    DesktopEnvironment('/usr/bin/pantheon-session', 'pantheon'),
-    DesktopEnvironment('/usr/bin/startdde', 'deepin'),
     DesktopEnvironment('/usr/bin/startxfce4', 'xfce'),
     DesktopEnvironment('/usr/bin/openbox-session', 'openbox'),
     DesktopEnvironment('/usr/bin/i3', 'i3'),
     DesktopEnvironment('/usr/bin/awesome', 'awesome'),
-    DesktopEnvironment('/usr/bin/bspwm', 'bspwm'),
-    DesktopEnvironment('/usr/bin/herbstluftwm', 'herbstluftwm'),
-    DesktopEnvironment('/usr/bin/qtile', 'qtile'),
-    DesktopEnvironment('/usr/bin/xmonad', 'xmonad'),
     DesktopEnvironment('/usr/bin/dwm', 'dwm'),
     DesktopEnvironment('/usr/bin/jwm', 'jwm'),
     DesktopEnvironment('/usr/bin/icewm-session', 'icewm-session'),
+    DesktopEnvironment('/usr/bin/start-lumina-desktop', 'Lumina-DE'),
 ]
 
 
@@ -923,6 +918,8 @@
 
         if dm_instance is None:
             libcalamares.utils.debug("{!s} selected but not installed".format(dm))
+        else:
+            libcalamares.utils.target_env_call(['systemctl', '-f', 'enable', dm])
             if dm in displaymanagers:
                 displaymanagers.remove(dm)
 
