diff -Nur orig/src/modules/displaymanager/displaymanager.conf mod/src/modules/displaymanager/displaymanager.conf
--- orig/src/modules/displaymanager/displaymanager.conf	2019-08-07 12:58:22.000000000 +0200
+++ mod/src/modules/displaymanager/displaymanager.conf	2019-08-26 08:48:29.000000000 +0200
@@ -9,9 +9,7 @@
   - sddm
   - lightdm
   - gdm
-  - mdm
   - lxdm
-  - kdm
 
 #Enable the following settings to force a desktop environment in your displaymanager configuration file:
 #defaultDesktopEnvironment:
diff -Nur orig/src/modules/displaymanager/main.py mod/src/modules/displaymanager/main.py
--- orig/src/modules/displaymanager/main.py	2019-08-07 12:58:22.000000000 +0200
+++ mod/src/modules/displaymanager/main.py	2019-09-08 21:18:00.000000000 +0200
@@ -80,9 +80,9 @@
     DesktopEnvironment('/usr/bin/startlxde', 'LXDE'),
     DesktopEnvironment('/usr/bin/lxqt-session', 'lxqt'),
     DesktopEnvironment('/usr/bin/pekwm', 'pekwm'),
-    DesktopEnvironment('/usr/bin/pantheon-session', 'pantheon'),
     DesktopEnvironment('/usr/bin/i3', 'i3'),
-    DesktopEnvironment('/usr/bin/startdde', 'deepin'),
+    DesktopEnvironment('/usr/bin/icewm-session', 'icewm-session'),
+    DesktopEnvironment('/usr/bin/start-lumina-desktop', 'Lumina-DE'),
     DesktopEnvironment('/usr/bin/openbox-session', 'openbox')
 ]
 
@@ -774,6 +774,8 @@
 
         if dm_instance is None:
             libcalamares.utils.debug("{!s} selected but not installed".format(dm))
+        else:
+            libcalamares.utils.target_env_call(['systemctl', '-f', 'enable', dm])
             if dm in displaymanagers:
                 displaymanagers.remove(dm)
 
