diff -Naur a/src/modules/displaymanager/displaymanager.conf b/src/modules/displaymanager/displaymanager.conf
--- a/src/modules/displaymanager/displaymanager.conf	2022-06-19 15:34:20.000000000 +0200
+++ b/src/modules/displaymanager/displaymanager.conf	2022-06-25 20:20:16.838203266 +0200
@@ -20,10 +20,7 @@
   - sddm
   - lightdm
   - gdm
-  - mdm
   - lxdm
-  - kdm
-  - greetd
 
 # Enable the following settings to force a desktop environment
 # in your displaymanager configuration file. This will attempt
diff -Naur a/src/modules/displaymanager/main.py b/src/modules/displaymanager/main.py
--- a/src/modules/displaymanager/main.py	2022-06-19 15:34:20.000000000 +0200
+++ b/src/modules/displaymanager/main.py	2022-06-25 20:20:16.838203266 +0200
@@ -180,23 +180,16 @@
     DesktopEnvironment('/usr/bin/startlxde', 'LXDE'),
     DesktopEnvironment('/usr/bin/lxqt-session', 'lxqt'),
     DesktopEnvironment('/usr/bin/pekwm', 'pekwm'),
-    DesktopEnvironment('/usr/bin/pantheon-session', 'pantheon'),
-    DesktopEnvironment('/usr/bin/startdde', 'deepin'),
     DesktopEnvironment('/usr/bin/startxfce4', 'xfce'),
     DesktopEnvironment('/usr/bin/openbox-session', 'openbox'),
     DesktopEnvironment('/usr/bin/i3', 'i3'),
     DesktopEnvironment('/usr/bin/awesome', 'awesome'),
-    DesktopEnvironment('/usr/bin/bspwm', 'bspwm'),
-    DesktopEnvironment('/usr/bin/herbstluftwm', 'herbstluftwm'),
-    DesktopEnvironment('/usr/bin/qtile', 'qtile'),
-    DesktopEnvironment('/usr/bin/xmonad', 'xmonad'),
     DesktopEnvironment('/usr/bin/dwm', 'dwm'),
     DesktopEnvironment('/usr/bin/jwm', 'jwm'),
     DesktopEnvironment('/usr/bin/icewm-session', 'icewm-session'),
     DesktopEnvironment('/usr/bin/fvwm3', 'fvwm3'),
     DesktopEnvironment('/usr/bin/sway', 'sway'),
-    DesktopEnvironment('/usr/bin/ukui-session', 'ukui'),
-    DesktopEnvironment('/usr/bin/cutefish-session', 'cutefish-xsession'),
+    DesktopEnvironment('/usr/bin/start-lumina-desktop', 'Lumina-DE'),
 ]
 
 
@@ -1013,6 +1006,8 @@
 
         if dm_instance is None:
             libcalamares.utils.debug("{!s} selected but not installed".format(dm))
+        else:
+            libcalamares.utils.target_env_call(['systemctl', '-f', 'enable', dm])
             if dm in displaymanagers:
                 displaymanagers.remove(dm)
 
