diff -Naur orig/src/modules/displaymanager/displaymanager.conf patched/src/modules/displaymanager/displaymanager.conf
--- orig/src/modules/displaymanager/displaymanager.conf	2019-07-06 00:52:35.000000000 +0200
+++ patched/src/modules/displaymanager/displaymanager.conf	2019-07-31 12:48:29.364093966 +0200
@@ -9,9 +9,7 @@
   - sddm
   - lightdm
   - gdm
-  - mdm
   - lxdm
-  - kdm
 
 #Enable the following settings to force a desktop environment in your displaymanager configuration file:
 #defaultDesktopEnvironment:
diff -Naur orig/src/modules/displaymanager/main.py patched/src/modules/displaymanager/main.py
--- orig/src/modules/displaymanager/main.py	2019-07-06 00:52:35.000000000 +0200
+++ patched/src/modules/displaymanager/main.py	2019-07-31 12:48:29.364093966 +0200
@@ -80,9 +80,9 @@
     DesktopEnvironment('/usr/bin/startlxde', 'LXDE'),
     DesktopEnvironment('/usr/bin/lxqt-session', 'lxqt'),
     DesktopEnvironment('/usr/bin/pekwm', 'pekwm'),
-    DesktopEnvironment('/usr/bin/pantheon-session', 'pantheon'),
     DesktopEnvironment('/usr/bin/i3', 'i3'),
-    DesktopEnvironment('/usr/bin/startdde', 'deepin'),
+    DesktopEnvironment('/usr/bin/icewm-session', 'icewm-session'),
+    DesktopEnvironment('/usr/bin/start-lumina-desktop', 'Lumina-DE'),
     DesktopEnvironment('/usr/bin/openbox-session', 'openbox')
 ]
 
@@ -809,6 +809,7 @@
         libcalamares.utils.debug("Unsetting autologin.")
 
     libcalamares.globalstorage.insert("displayManagers", displaymanagers)
+    libcalamares.utils.target_env_call(['systemctl', 'enable', dm])
 
     # Do the actual configuration and collect messages
     dm_setup_message = []
