diff -Nur orig/src/modules/displaymanager/main.py mod/src/modules/displaymanager/main.py
--- orig/src/modules/displaymanager/main.py	2017-11-14 16:18:58.000000000 +0100
+++ mod/src/modules/displaymanager/main.py	2017-11-26 22:53:05.000000000 +0100
@@ -51,6 +51,8 @@
     DesktopEnvironment('/usr/bin/budgie-desktop', 'budgie-desktop'),
     DesktopEnvironment('/usr/bin/i3', 'i3'),
     DesktopEnvironment('/usr/bin/startdde', 'deepin'),
+    DesktopEnvironment('/usr/bin/icewm-session', 'icewm-session'),
+    DesktopEnvironment('/usr/bin/start-lumina-desktop', 'Lumina-DE'),
     DesktopEnvironment('/usr/bin/openbox-session', 'openbox')
 ]
 
@@ -415,57 +417,26 @@
 
     # Setup slim
     if "slim" in displaymanagers:
+        libcalamares.utils.target_env_call(['systemctl', 'enable', 'slim'])
         if not have_dm("slim", root_mount_point):
             libcalamares.utils.debug("slim selected but not installed")
             displaymanagers.remove("slim")
 
     # Setup sddm
     if "sddm" in displaymanagers:
+        libcalamares.utils.target_env_call(['systemctl', 'enable', 'sddm'])
         if not have_dm("sddm", root_mount_point):
             libcalamares.utils.debug("sddm selected but not installed")
             displaymanagers.remove("sddm")
 
     # setup lightdm
     if "lightdm" in displaymanagers:
+        libcalamares.utils.target_env_call(['systemctl', 'enable', 'lightdm'])
         if have_dm("lightdm", root_mount_point):
             lightdm_conf_path = os.path.join(
                 root_mount_point, "etc/lightdm/lightdm.conf"
                 )
 
-            if enable_basic_setup:
-                libcalamares.utils.target_env_call(
-                    ['mkdir', '-p', '/run/lightdm']
-                    )
-
-                if libcalamares.utils.target_env_call(
-                        ['getent', 'group', 'lightdm']
-                        ) != 0:
-                    libcalamares.utils.target_env_call(
-                        ['groupadd', '-g', '620', 'lightdm']
-                        )
-
-                if libcalamares.utils.target_env_call(
-                        ['getent', 'passwd', 'lightdm']
-                        ) != 0:
-                    libcalamares.utils.target_env_call(
-                        ['useradd', '-c',
-                            '"LightDM Display Manager"',
-                            '-u', '620',
-                            '-g', 'lightdm',
-                            '-d', '/var/run/lightdm',
-                            '-s', '/usr/bin/nologin',
-                            'lightdm'
-                         ]
-                        )
-
-                libcalamares.utils.target_env_call('passwd', '-l', 'lightdm')
-                libcalamares.utils.target_env_call(
-                    ['chown', '-R', 'lightdm:lightdm', '/run/lightdm']
-                    )
-                libcalamares.utils.target_env_call(
-                    ['chmod', '+r' '/etc/lightdm/lightdm.conf']
-                    )
-
             if default_desktop_environment is not None:
                 os.system(
                     "sed -i -e \"s/^.*user-session=.*/user-session={!s}/\" "
@@ -507,6 +478,7 @@
 
     # Setup gdm
     if "gdm" in displaymanagers:
+        libcalamares.utils.target_env_call(['systemctl', 'enable', 'gdm'])
         if have_dm("gdm", root_mount_point):
             if enable_basic_setup:
                 if libcalamares.utils.target_env_call(
@@ -589,6 +561,7 @@
 
     # Setup lxdm
     if "lxdm" in displaymanagers:
+        libcalamares.utils.target_env_call(['systemctl', 'enable', 'lxdm'])
         if have_dm("lxdm", root_mount_point):
             if enable_basic_setup:
                 if libcalamares.utils.target_env_call(
