diff -Naur orig/src/modules/displaymanager/displaymanager.conf patched/src/modules/displaymanager/displaymanager.conf
--- orig/src/modules/displaymanager/displaymanager.conf	2020-01-28 17:42:21.000000000 +0100
+++ patched/src/modules/displaymanager/displaymanager.conf	2020-01-30 22:13:54.363497048 +0100
@@ -17,9 +17,7 @@
   - sddm
   - lightdm
   - gdm
-  - mdm
   - lxdm
-  - kdm
 
 # Enable the following settings to force a desktop environment
 # in your displaymanager configuration file. This will attempt
diff -Naur orig/src/modules/displaymanager/main.py patched/src/modules/displaymanager/main.py
--- orig/src/modules/displaymanager/main.py	2020-01-28 17:42:21.000000000 +0100
+++ patched/src/modules/displaymanager/main.py	2020-01-30 22:13:54.367497090 +0100
@@ -192,9 +192,9 @@
     DesktopEnvironment('/usr/bin/startlxde', 'LXDE'),
     DesktopEnvironment('/usr/bin/lxqt-session', 'lxqt'),
     DesktopEnvironment('/usr/bin/pekwm', 'pekwm'),
-    DesktopEnvironment('/usr/bin/pantheon-session', 'pantheon'),
     DesktopEnvironment('/usr/bin/i3', 'i3'),
-    DesktopEnvironment('/usr/bin/startdde', 'deepin'),
+    DesktopEnvironment('/usr/bin/icewm-session', 'icewm-session'),
+    DesktopEnvironment('/usr/bin/start-lumina-desktop', 'Lumina-DE'),
     DesktopEnvironment('/usr/bin/openbox-session', 'openbox')
 ]
 
@@ -916,6 +916,8 @@
 
         if dm_instance is None:
             libcalamares.utils.debug("{!s} selected but not installed".format(dm))
+        else:
+            libcalamares.utils.target_env_call(['systemctl', '-f', 'enable', dm])
             if dm in displaymanagers:
                 displaymanagers.remove(dm)
 
