diff -Naur a/src/modules/dracut/main.py b/src/modules/dracut/main.py
--- a/src/modules/dracut/main.py	2024-07-03 23:05:30.000000000 +0200
+++ b/src/modules/dracut/main.py	2024-07-19 04:19:34.225288646 +0200
@@ -36,11 +36,10 @@
     :return:
     """
     try:
-        initramfs_name = libcalamares.job.configuration['initramfsName']
-        target_env_process_output(['dracut', '-f', initramfs_name])
+        target_env_process_output(['mkinitrd-installedkernel'])
     except KeyError:
         try:
-            target_env_process_output(['dracut', '-f'])
+            target_env_process_output(['dracut', '-f', '-N'])
         except subprocess.CalledProcessError as cpe:
             libcalamares.utils.warning(f"Dracut failed with output: {cpe.output}")
             return cpe.returncode
