diff -Nur orig/src/modules/grubcfg/main.py mod/src/modules/grubcfg/main.py
--- orig/src/modules/grubcfg/main.py	2017-11-14 16:18:58.000000000 +0100
+++ mod/src/modules/grubcfg/main.py	2017-11-26 09:28:44.998494008 +0100
@@ -90,7 +90,7 @@
                         )
                 ]
 
-    kernel_params = ["quiet"]
+    kernel_params = []
 
     if cryptdevice_params:
         kernel_params.extend(cryptdevice_params)
