diff -Naur orig/src/modules/grubcfg/grubcfg.conf patched/src/modules/grubcfg/grubcfg.conf
--- orig/src/modules/grubcfg/grubcfg.conf	2020-01-28 17:42:21.000000000 +0100
+++ patched/src/modules/grubcfg/grubcfg.conf	2020-01-30 22:16:16.173172445 +0100
@@ -45,6 +45,9 @@
 defaults:
     GRUB_TIMEOUT: 5
     GRUB_DEFAULT: "saved"
-    GRUB_DISABLE_SUBMENU: true
+    GRUB_DISABLE_LINUX_UUID: "false"
+    GRUB_GFXMODE: "auto"
+    GRUB_GFXPAYLOAD_LINUX: "keep"
     GRUB_TERMINAL_OUTPUT: "console"
     GRUB_DISABLE_RECOVERY: true
+    GRUB_BACKGROUND: "/usr/share/backgrounds/UHU/uhu-grub-bg.png"
