diff -Naur orig/src/modules/grubcfg/main.py patched/src/modules/grubcfg/main.py
--- orig/src/modules/grubcfg/main.py	2020-01-28 17:42:21.000000000 +0100
+++ patched/src/modules/grubcfg/main.py	2020-01-30 22:19:26.272566225 +0100
@@ -111,7 +111,7 @@
                         )
                 ]
 
-    kernel_params = ["quiet"]
+    kernel_params = []
 
     if cryptdevice_params:
         kernel_params.extend(cryptdevice_params)
