diff -Naur a/src/modules/grubcfg/main.py b/src/modules/grubcfg/main.py
--- a/src/modules/grubcfg/main.py	2022-03-21 15:19:44.000000000 +0100
+++ b/src/modules/grubcfg/main.py	2022-03-21 20:12:34.669116690 +0100
@@ -172,7 +172,7 @@
             if partition["fs"] == "zfs" and partition["mountPoint"] == "/":
                 zfs_root_path = get_zfs_root()
 
-    kernel_params = libcalamares.job.configuration.get("kernel_params", ["quiet"])
+    kernel_params = libcalamares.job.configuration.get("kernel_params", [])
 
     # Currently, grub doesn't detect this properly so it must be set manually
     if zfs_root_path:
