diff -Naur orig/src/modules/locale/locale.conf patched/src/modules/locale/locale.conf
--- orig/src/modules/locale/locale.conf	2020-09-03 20:45:04.000000000 +0200
+++ patched/src/modules/locale/locale.conf	2020-09-07 01:24:32.842750318 +0200
@@ -17,8 +17,8 @@
 # Note that useSystemTimezone and GeoIP settings can change the
 # starting time zone.
 #
-region:                     "America"
-zone:                       "New_York"
+region:                     "Europe"
+zone:                       "Budapest"
 
 # Instead of using *region* and *zone* specified above,
 # you can use the system's notion of the timezone, instead.
