diff -Nur orig/src/modules/locale/locale.conf mod/src/modules/locale/locale.conf
--- orig/src/modules/locale/locale.conf	2018-05-17 17:13:54.000000000 +0200
+++ mod/src/modules/locale/locale.conf	2018-06-12 12:56:55.473466262 +0200
@@ -11,8 +11,8 @@
 # the locale page can be set through keys *region* and *zone*.
 # If either is not set, defaults to America/New_York.
 #
-region:                     "America"
-zone:                       "New_York"
+region:                     "Europe"
+zone:                       "Budapest"
 
 
 # System locales are detected in the following order:
