diff -Nur orig/src/modules/locale/locale.conf mod/src/modules/locale/locale.conf
--- orig/src/modules/locale/locale.conf	2017-11-14 16:18:58.000000000 +0100
+++ mod/src/modules/locale/locale.conf	2017-11-24 21:51:10.824333902 +0100
@@ -1,6 +1,6 @@
 ---
-region:                     "America"
-zone:                       "New_York"
+region:                     "Europe"
+zone:                       "Budapest"
 
 # GeoIP settings. Leave commented out to disable GeoIP.
 #localeGenPath:             "/etc/locale.gen"
