diff -Naur orig/src/modules/mount/mount.conf patched/src/modules/mount/mount.conf
--- orig/src/modules/mount/mount.conf	2019-07-06 00:52:35.000000000 +0200
+++ patched/src/modules/mount/mount.conf	2019-07-31 13:05:00.821582117 +0200
@@ -35,6 +35,9 @@
     - device: /run/udev
       mountPoint: /run/udev
       options: bind
+    - device: /dev/pts
+      fs: devpts
+      mountPoint: /dev/pts
 
 extraMountsEfi:
     - device: efivarfs
