diff -Naur a/src/modules/mount/mount.conf b/src/modules/mount/mount.conf
--- a/src/modules/mount/mount.conf	2023-12-12 00:04:06.000000000 +0100
+++ b/src/modules/mount/mount.conf	2023-12-17 00:07:51.192648225 +0100
@@ -34,6 +34,9 @@
     - device: /run/udev
       mountPoint: /run/udev
       options: [ bind ]
+    - device: /dev/pts
+      fs: devpts
+      mountPoint: /dev/pts
     - device: efivarfs
       fs: efivarfs
       mountPoint: /sys/firmware/efi/efivars
