diff -Naur orig/src/modules/netinstall/netinstall.conf patched/src/modules/netinstall/netinstall.conf
--- orig/src/modules/netinstall/netinstall.conf	2020-03-27 23:43:50.000000000 +0100
+++ patched/src/modules/netinstall/netinstall.conf	2020-04-06 22:47:22.526226149 +0200
@@ -60,7 +60,7 @@
 # `local` means that the data is obtained from **this** config
 # file, under the key *groups*.
 #
-groupsUrl: local
+groupsUrl: file:///usr/share/calamares/netinstall.yaml
 
 # If the installation can proceed without netinstall (e.g. the Live CD
 # can create a working installed system, but netinstall is preferred
