diff -Nur orig/src/modules/netinstall/netinstall.conf mod/src/modules/netinstall/netinstall.conf
--- orig/src/modules/netinstall/netinstall.conf	2017-11-14 16:18:58.000000000 +0100
+++ mod/src/modules/netinstall/netinstall.conf	2017-11-23 08:24:39.000000000 +0100
@@ -1,7 +1,7 @@
 ---
 # This is the URL that is retrieved to get the netinstall groups-and-packages
 # data (which should be in the format described in netinstall.yaml).
-groupsUrl: http://chakraos.org/netinstall.php
+groupsUrl: file:/usr/share/calamares/netinstall.yaml
 
 # If the installation can proceed without netinstall (e.g. the Live CD
 # can create a working installed system, but netinstall is preferred
