diff -Nur calamares-3.2.1/src/modules/netinstall/netinstall.conf calamares-3.2.1.patched/src/modules/netinstall/netinstall.conf
--- calamares-3.2.1/src/modules/netinstall/netinstall.conf	2018-06-25 16:04:26.000000000 +0200
+++ calamares-3.2.1.patched/src/modules/netinstall/netinstall.conf	2018-07-21 17:26:00.028099112 +0200
@@ -4,7 +4,7 @@
 #   groupsUrl: http://example.org/netinstall.php
 # or it can be a locally installed file:
 #   groupsUrl: file:///usr/share/calamares/netinstall.yaml
-# groupsUrl: file:///usr/share/calamares/netinstall.yaml
+groupsUrl: file:///usr/share/calamares/netinstall.yaml
 
 # If the installation can proceed without netinstall (e.g. the Live CD
 # can create a working installed system, but netinstall is preferred
