diff -Naur orig/src/modules/netinstall/netinstall.conf patched/src/modules/netinstall/netinstall.conf
--- orig/src/modules/netinstall/netinstall.conf	2019-12-03 11:37:28.000000000 +0100
+++ patched/src/modules/netinstall/netinstall.conf	2019-12-14 22:18:02.838793803 +0100
@@ -11,7 +11,7 @@
 #
 # The format of the groups file is documented in `README.md`.
 #
-# groupsUrl: file:///usr/share/calamares/netinstall.yaml
+groupsUrl: file:///usr/share/calamares/netinstall.yaml
 
 # If the installation can proceed without netinstall (e.g. the Live CD
 # can create a working installed system, but netinstall is preferred
