diff -Nur orig/src/modules/users/CreateUserJob.cpp mod/src/modules/users/CreateUserJob.cpp
--- orig/src/modules/users/CreateUserJob.cpp	2018-05-17 17:13:54.000000000 +0200
+++ mod/src/modules/users/CreateUserJob.cpp	2018-06-12 13:09:47.372550813 +0200
@@ -149,7 +149,6 @@
     int ec = CalamaresUtils::System::instance()->
              targetEnvCall( { "useradd",
                               "-m",
-                              "-U",
                               "-c",
                               m_fullName,
                               m_userName } );
@@ -175,7 +174,7 @@
                       targetEnvCall( { "chown",
                                        "-R",
                                        QString( "%1:%2" ).arg( m_userName )
-                                                         .arg( m_userName ),
+                                                         .arg( "users" ),
                                        QString( "/home/%1" ).arg( m_userName ) } );
     if ( ec )
         return Calamares::JobResult::error( tr( "Cannot set home directory ownership for user %1." )
