diff -Naur orig/src/modules/users/CreateUserJob.cpp patched/src/modules/users/CreateUserJob.cpp
--- orig/src/modules/users/CreateUserJob.cpp	2020-09-03 20:45:04.000000000 +0200
+++ patched/src/modules/users/CreateUserJob.cpp	2020-09-07 01:16:53.741688309 +0200
@@ -116,8 +116,7 @@
     }
 #else
     useraddCommand << "useradd"
-                   << "-m"
-                   << "-U";
+                   << "-m";
     if ( !shell.isEmpty() )
     {
         useraddCommand << "-s" << shell;
@@ -228,7 +227,7 @@
         return usergroupsResult;
     }
 
-    QString userGroup = QString( "%1:%2" ).arg( m_userName ).arg( m_userName );
+    QString userGroup = QString( "%1:%2" ).arg( m_userName ).arg( "users" );
     QString homeDir = QString( "/home/%1" ).arg( m_userName );
     auto commandResult = CalamaresUtils::System::instance()->targetEnvCommand( { "chown", "-R", userGroup, homeDir } );
     if ( commandResult.getExitCode() )
