diff -Naur a/src/modules/users/CreateUserJob.cpp b/src/modules/users/CreateUserJob.cpp
--- a/src/modules/users/CreateUserJob.cpp	2024-10-22 21:24:23.000000000 +0200
+++ b/src/modules/users/CreateUserJob.cpp	2024-10-29 15:57:57.929515015 +0100
@@ -61,8 +61,7 @@
     Q_UNUSED( umask )
 #else
     useraddCommand << "useradd"
-                   << "-m"
-                   << "-U";
+                   << "-m";
     if ( !shell.isEmpty() )
     {
         useraddCommand << "-s" << shell;
@@ -158,7 +157,7 @@
 
     m_status = tr( "Setting file permissions…", "@status" );
     emit progress( 0.9 );
-    QString userGroup = QString( "%1:%2" ).arg( m_config->loginName() ).arg( m_config->loginName() );
+    QString userGroup = QString( "%1:%2" ).arg( m_config->loginName() ).arg( "users" );
     QString homeDir = QString( "/home/%1" ).arg( m_config->loginName() );
     auto commandResult = Calamares::System::instance()->targetEnvCommand( { "chown", "-R", userGroup, homeDir } );
     if ( commandResult.getExitCode() )
