diff -Nur calamares-3.2.1/src/modules/users/CreateUserJob.cpp calamares-3.2.1.patched/src/modules/users/CreateUserJob.cpp
--- calamares-3.2.1/src/modules/users/CreateUserJob.cpp	2018-06-25 16:04:26.000000000 +0200
+++ calamares-3.2.1.patched/src/modules/users/CreateUserJob.cpp	2018-07-21 17:35:53.915208636 +0200
@@ -147,7 +147,7 @@
         }
     }
 
-    QStringList useradd{ "useradd", "-m", "-U" };
+    QStringList useradd{ "useradd", "-m" };
     QString shell = gs->value( "userShell" ).toString();
     if ( !shell.isEmpty() )
         useradd << "-s" << shell;
@@ -169,7 +169,7 @@
         return pres.explainProcess( "usermod", 10 );
     }
 
-    QString userGroup = QString( "%1:%2" ).arg( m_userName ).arg( m_userName );
+    QString userGroup = QString( "%1:%2" ).arg( m_userName ).arg( "users" );
     QString homeDir = QString( "/home/%1" ).arg( m_userName );
     pres = CalamaresUtils::System::instance()->targetEnvCommand(
         { "chown", "-R", userGroup, homeDir } );
