diff -Naur orig/src/modules/users/CreateUserJob.cpp patched/src/modules/users/CreateUserJob.cpp
--- orig/src/modules/users/CreateUserJob.cpp	2020-12-04 14:51:48.000000000 +0100
+++ patched/src/modules/users/CreateUserJob.cpp	2020-12-06 19:16:05.777748938 +0100
@@ -64,8 +64,7 @@
     }
 #else
     useraddCommand << "useradd"
-                   << "-m"
-                   << "-U";
+                   << "-m";
     if ( !shell.isEmpty() )
     {
         useraddCommand << "-s" << shell;
@@ -157,7 +156,7 @@
 
     m_status = tr( "Setting file permissions" );
     emit progress( 0.9 );
-    QString userGroup = QString( "%1:%2" ).arg( m_config->loginName() ).arg( m_config->loginName() );
+    QString userGroup = QString( "%1:%2" ).arg( m_config->loginName() ).arg( "users" );
     QString homeDir = QString( "/home/%1" ).arg( m_config->loginName() );
     auto commandResult = CalamaresUtils::System::instance()->targetEnvCommand( { "chown", "-R", userGroup, homeDir } );
     if ( commandResult.getExitCode() )
