diff -Nur orig/src/modules/packages/packages.conf mod/src/modules/packages/packages.conf
--- orig/src/modules/packages/packages.conf	2018-05-17 17:13:54.000000000 +0200
+++ mod/src/modules/packages/packages.conf	2018-06-12 13:14:30.272149269 +0200
@@ -12,7 +12,7 @@
 #  - entropy     - Sabayon package manager
 #  - dummy       - Dummy manager, only logs
 #
-backend: dummy
+backend: apt
 
 # Often package installation needs an internet connection.
 # Since you may allow system installation without a connection
@@ -129,13 +129,3 @@
 # "binutils", and then a second time for "wget". When installing large numbers
 # of packages, this can lead to a considerable time savings.
 #
-operations:
-  - install:
-    - vi
-    - vi-${LOCALE}
-    - wget
-    - binutils
-  - remove:
-    - vi
-    - wget
-    - binutils
