diff -Naur orig/src/modules/packages/packages.conf patched/src/modules/packages/packages.conf
--- orig/src/modules/packages/packages.conf	2020-09-03 20:45:04.000000000 +0200
+++ patched/src/modules/packages/packages.conf	2020-09-07 01:43:37.288589653 +0200
@@ -35,7 +35,7 @@
 # Not actually a package manager, but suitable for testing:
 #  - dummy       - Dummy manager, only logs
 #
-backend: dummy
+backend: apt
 
 #
 # Often package installation needs an internet connection.
@@ -201,13 +201,3 @@
 # "binutils", and then a second time for "wget". When installing large numbers
 # of packages, this can lead to a considerable time savings.
 #
-operations:
-  - install:
-    - vi
-    - vi-${LOCALE}
-    - wget
-    - binutils
-  - remove:
-    - vi
-    - wget
-    - binutils
