diff -Naur orig/src/modules/packages/packages.conf patched/src/modules/packages/packages.conf
--- orig/src/modules/packages/packages.conf	2020-01-28 17:42:21.000000000 +0100
+++ patched/src/modules/packages/packages.conf	2020-01-30 22:20:53.698111201 +0100
@@ -12,7 +12,7 @@
 #  - entropy     - Sabayon package manager
 #  - dummy       - Dummy manager, only logs
 #
-backend: dummy
+backend: apt
 
 #
 # Often package installation needs an internet connection.
@@ -158,13 +158,3 @@
 # "binutils", and then a second time for "wget". When installing large numbers
 # of packages, this can lead to a considerable time savings.
 #
-operations:
-  - install:
-    - vi
-    - vi-${LOCALE}
-    - wget
-    - binutils
-  - remove:
-    - vi
-    - wget
-    - binutils
