diff -Naur orig/src/modules/packages/packages.conf patched/src/modules/packages/packages.conf
--- orig/src/modules/packages/packages.conf	2019-07-06 00:52:35.000000000 +0200
+++ patched/src/modules/packages/packages.conf	2019-07-31 13:10:51.702346606 +0200
@@ -12,7 +12,7 @@
 #  - entropy     - Sabayon package manager
 #  - dummy       - Dummy manager, only logs
 #
-backend: dummy
+backend: apt
 
 #
 # Often package installation needs an internet connection.
@@ -142,13 +142,3 @@
 # "binutils", and then a second time for "wget". When installing large numbers
 # of packages, this can lead to a considerable time savings.
 #
-operations:
-  - install:
-    - vi
-    - vi-${LOCALE}
-    - wget
-    - binutils
-  - remove:
-    - vi
-    - wget
-    - binutils
