diff -Naur a/src/modules/packages/packages.conf b/src/modules/packages/packages.conf
--- a/src/modules/packages/packages.conf	2024-04-16 15:46:09.000000000 +0200
+++ b/src/modules/packages/packages.conf	2024-05-25 20:49:55.925540288 +0200
@@ -36,7 +36,7 @@
 # Not actually a package manager, but suitable for testing:
 #  - dummy       - Dummy manager, only logs
 #
-backend: dummy
+backend: apt
 
 #
 # Often package installation needs an internet connection.
@@ -202,13 +202,3 @@
 # "binutils", and then a second time for "wget". When installing large numbers
 # of packages, this can lead to a considerable time savings.
 #
-operations:
-  - install:
-    - vi
-    - vi-${LOCALE}
-    - wget
-    - binutils
-  - remove:
-    - vi
-    - wget
-    - binutils
