diff -Naur orig/src/modules/partition/partition.conf patched/src/modules/partition/partition.conf
--- orig/src/modules/partition/partition.conf	2020-05-11 20:53:23.000000000 +0200
+++ patched/src/modules/partition/partition.conf	2020-07-03 02:28:07.547593394 +0200
@@ -65,7 +65,7 @@
 # neverCreateSwap:        false
 
 # Correctly draw nested (e.g. logical) partitions as such.
-drawNestedPartitions:   false
+drawNestedPartitions:   true
 
 # Show/hide partition labels on manual partitioning page.
 alwaysShowPartitionLabels: true
