diff -Naur a/src/modules/partition/partition.conf b/src/modules/partition/partition.conf
--- a/src/modules/partition/partition.conf	2023-12-12 00:04:06.000000000 +0100
+++ b/src/modules/partition/partition.conf	2023-12-17 00:18:23.321886459 +0100
@@ -112,7 +112,7 @@
 # allowZfsEncryption: true
 
 # Correctly draw nested (e.g. logical) partitions as such.
-drawNestedPartitions:   false
+drawNestedPartitions:   true
 
 # Show/hide partition labels on manual partitioning page.
 alwaysShowPartitionLabels: true
