diff -Naur orig/src/modules/removeuser/removeuser.conf patched/src/modules/removeuser/removeuser.conf
--- orig/src/modules/removeuser/removeuser.conf	2019-07-06 00:52:35.000000000 +0200
+++ patched/src/modules/removeuser/removeuser.conf	2019-07-31 13:15:46.030481132 +0200
@@ -3,4 +3,4 @@
 # spills into the target system.
 ---
 # Username in the target system to be removed.
-username: live
+username: uhu
