diff -Nur orig/src/modules/services/services.conf mod/src/modules/services/services.conf
--- orig/src/modules/services/services.conf	2017-11-14 16:18:58.000000000 +0100
+++ mod/src/modules/services/services.conf	2017-11-24 22:35:55.499692531 +0100
@@ -13,8 +13,8 @@
     mandatory: true
 
 disable:
-  - name: "pacman-init"
-    mandatory: false
+#  - name: "pacman-init"
+#    mandatory: false
 
 # Example to express an empty list:
 # disable: []
