diff -Naur orig/settings.conf patched/settings.conf
--- orig/settings.conf	2020-03-27 23:43:50.000000000 +0100
+++ patched/settings.conf	2020-04-06 23:09:44.296102392 +0200
@@ -92,38 +92,26 @@
   - keyboard
   - partition
   - users
-#  - tracking
+  - netinstall
   - summary
 - exec:
-#  - dummycpp
-#  - dummyprocess
-#  - dummypython
   - partition
   - mount
-  - unpackfs
+  - chrootcfg
+  - networkcfg
+  - packages
   - machineid
   - fstab
   - locale
-  - keyboard
   - localecfg
-#  - luksbootkeyfile
-#  - luksopenswaphookcfg
-#  - dracutlukscfg
-#  - plymouthcfg
-  - initcpiocfg
-  - initcpio
   - users
   - displaymanager
-  - networkcfg
   - hwclock
-  - services-systemd
-#  - dracut
-  - initramfs
-#  - grubcfg
+  - dracut
+  - grubcfg
   - bootloader
   - umount
 - show:
-#  - webview@owncloud
   - finished
 
 # A branding component is a directory, either in SHARE/calamares/branding or
@@ -196,3 +184,8 @@
 #
 #
 quit-at-end: false
+#
+#
+packageOperations:
+  -install
+  -remove
