diff -Nur orig/settings.conf mod/settings.conf
--- orig/settings.conf	2018-05-17 17:13:54.000000000 +0200
+++ mod/settings.conf	2018-06-12 13:33:47.702505428 +0200
@@ -78,6 +78,7 @@
   - partition
   - users
 #  - tracking
+  - netinstall
   - summary
 - exec:
 #  - dummycpp
@@ -86,26 +87,22 @@
 #  - dummypythonqt
   - partition
   - mount
-  - unpackfs
+  - chrootcfg
+  - networkcfg
+  - packages
   - machineid
   - fstab
   - locale
-  - keyboard
   - localecfg
 #  - luksbootkeyfile
 #  - luksopenswaphookcfg
 #  - dracutlukscfg
 #  - plymouthcfg
-  - initcpiocfg
-  - initcpio
   - users
   - displaymanager
-  - networkcfg
   - hwclock
-  - services
 #  - dracut
-  - initramfs
-#  - grubcfg
+  - grubcfg
   - bootloader
   - umount
 - show:
@@ -146,3 +143,7 @@
 #
 # YAML: boolean.
 dont-chroot: false
+
+packageOperations:
+  -install
+  -remove
