diff -Naur orig/settings.conf patched/settings.conf
--- orig/settings.conf	2019-07-06 00:52:35.000000000 +0200
+++ patched/settings.conf	2019-07-31 13:19:41.089949270 +0200
@@ -93,6 +93,7 @@
   - partition
   - users
 #  - tracking
+  - netinstall
   - summary
 - exec:
 #  - dummycpp
@@ -101,25 +102,21 @@
 #  - dummypythonqt
   - partition
   - mount
-  - unpackfs
+  - chrootcfg
+  - networkcfg
+  - packages
   - machineid
   - fstab
   - locale
-  - keyboard
   - localecfg
 #  - luksbootkeyfile
 #  - luksopenswaphookcfg
 #  - dracutlukscfg
 #  - plymouthcfg
-  - initcpiocfg
-  - initcpio
   - users
   - displaymanager
-  - networkcfg
   - hwclock
-  - services-systemd
 #  - dracut
-  - initramfs
 #  - grubcfg
   - bootloader
   - umount
@@ -189,3 +186,7 @@
 #
 # YAML: boolean.
 disable-cancel-during-exec: false
+
+packageOperations:
+  -install
+  -remove
