diff -Naur a/settings.conf b/settings.conf
--- a/settings.conf	2024-08-13 22:00:50.000000000 +0200
+++ b/settings.conf	2024-08-20 20:45:03.479069039 +0200
@@ -115,42 +115,28 @@
 sequence:
 - show:
   - welcome
-#  - notesqml
-#  - packagechooserq@licenseq
   - locale
   - keyboard
   - partition
   - users
-#  - tracking
   - summary
 - exec:
-#  - dummycpp
-#  - dummyprocess
-#  - dummypython
   - partition
-#  - zfs
   - mount
-  - unpackfs
+  - chrootcfg
+  - networkcfg
+  - packages
   - machineid
   - locale
-  - keyboard
+#  - keyboard
   - localecfg
-#  - luksbootkeyfile
-#  - luksopenswaphookcfg
-#  - dracutlukscfg
   - fstab
-#  - plymouthcfg
-#  - zfshostid
-  - initcpiocfg
-  - initcpio
   - users
   - displaymanager
-  - networkcfg
+#  - networkcfg
   - hwclock
-  - services-systemd
-#  - dracut
-  - initramfs
-#  - grubcfg
+  - dracut
+  - grubcfg
   - bootloader
   - umount
 - show:
@@ -234,3 +220,7 @@
 #
 #
 quit-at-end: false
+
+packageOperations:
+  -install
+  -remove
