diff -Nur orig/settings.conf mod/settings.conf
--- orig/settings.conf	2020-02-27 23:26:29.000000000 +0100
+++ mod/settings.conf	2020-03-07 17:45:11.202809815 +0100
@@ -92,38 +92,26 @@
   - keyboard
   - partition
   - users
-#  - tracking
+  - netinstall
   - summary
 - exec:
-#  - dummycpp
-#  - dummyprocess
-#  - dummypython
   - partition
   - mount
-  - unpackfs
+  - chrootcfg
+  - networkcfg
+  - packages
   - machineid
   - fstab
   - locale
-  - keyboard
   - localecfg
-#  - luksbootkeyfile
-#  - luksopenswaphookcfg
-#  - dracutlukscfg
-#  - plymouthcfg
-  - initcpiocfg
-  - initcpio
   - users
   - displaymanager
-  - networkcfg
   - hwclock
-  - services-systemd
-#  - dracut
-  - initramfs
-#  - grubcfg
+  - dracut
+  - grubcfg
   - bootloader
   - umount
 - show:
-#  - webview@owncloud
   - finished
 
 # A branding component is a directory, either in SHARE/calamares/branding or
@@ -188,3 +176,7 @@
 #
 # YAML: boolean.
 disable-cancel-during-exec: false
+
+packageOperations:
+  -install
+  -remove
