diff -Nur orig/settings.conf mod/settings.conf
--- orig/settings.conf	2017-11-14 16:18:58.000000000 +0100
+++ mod/settings.conf	2017-11-22 09:52:44.000000000 +0100
@@ -66,6 +66,7 @@
   - keyboard
   - partition
   - users
+  - netinstall
   - summary
 - exec:
 #  - dummycpp
@@ -74,26 +75,31 @@
 #  - dummypythonqt
   - partition
   - mount
-  - unpackfs
+  - chrootcfg
+  - networkcfg
+#  - unpackfs
+  - packages
   - machineid
   - fstab
   - locale
-  - keyboard
+#  - keyboard
   - localecfg
 #  - luksbootkeyfile
 #  - luksopenswaphookcfg
 #  - dracutlukscfg
 #  - plymouthcfg
-  - initcpiocfg
-  - initcpio
+#  - initcpiocfg
+#  - initcpio
   - users
+#  - removeuser
   - displaymanager
-  - networkcfg
+#  - networkcfg
   - hwclock
-  - services
+#  - services
 #  - dracut
-  - initramfs
-#  - grubcfg
+# A ramdiszket a kernel már megcsinálta netinstall esetén
+#  - initramfs
+  - grubcfg
   - bootloader
   - umount
 - show:
@@ -127,3 +133,7 @@
 # Default is false.
 # YAML: boolean.
 dont-chroot: false
+
+packageOperations:
+  -install
+  -remove
