diff -Nur orig/settings.conf mod/settings.conf
--- orig/settings.conf	2018-09-04 14:03:59.000000000 +0200
+++ mod/settings.conf	2018-12-25 17:00:51.397426724 +0100
@@ -78,6 +78,7 @@
   - partition
   - users
 #  - tracking
+  - netinstall
   - summary
 - exec:
 #  - dummycpp
@@ -86,25 +87,21 @@
 #  - dummypythonqt
   - partition
   - mount
-  - unpackfs
+  - chrootcfg
+  - networkcfg
+  - packages
   - machineid
   - fstab
   - locale
-  - keyboard
   - localecfg
 #  - luksbootkeyfile
 #  - luksopenswaphookcfg
 #  - dracutlukscfg
 #  - plymouthcfg
-  - initcpiocfg
-  - initcpio
   - users
   - displaymanager
-  - networkcfg
   - hwclock
-  - services-systemd
 #  - dracut
-  - initramfs
 #  - grubcfg
   - bootloader
   - umount
@@ -146,3 +143,7 @@
 #
 # YAML: boolean.
 dont-chroot: false
+
+packageOperations:
+  -install
+  -remove
