diff -Naur orig/settings.conf patched/settings.conf
--- orig/settings.conf	2019-04-27 20:12:03.000000000 +0200
+++ patched/settings.conf	2019-05-03 02:12:34.231689869 +0200
@@ -78,6 +78,7 @@
   - partition
   - users
 #  - tracking
+  - netinstall
   - summary
 - exec:
 #  - dummycpp
@@ -86,25 +87,21 @@
 #  - dummypythonqt
   - partition
   - mount
-  - unpackfs
+  - chrootcfg
+  - networkcfg
+  - packages
   - machineid
   - fstab
   - locale
-  - keyboard
   - localecfg
 #  - luksbootkeyfile
 #  - luksopenswaphookcfg
 #  - dracutlukscfg
 #  - plymouthcfg
-  - initcpiocfg
-  - initcpio
   - users
   - displaymanager
-  - networkcfg
   - hwclock
-  - services-systemd
 #  - dracut
-  - initramfs
 #  - grubcfg
   - bootloader
   - umount
@@ -174,3 +171,7 @@
 #
 # YAML: boolean.
 disable-cancel-during-exec: false
+
+packageOperations:
+  -install
+  -remove
