diff -Nur orig/lang/calamares_hu.ts mod/lang/calamares_hu.ts
--- orig/lang/calamares_hu.ts	2018-06-25 16:04:26.000000000 +0200
+++ mod/lang/calamares_hu.ts	2018-09-05 22:40:01.562228007 +0200
@@ -944,7 +944,7 @@
     <message>
         <location filename="../src/modules/finished/FinishedPage.ui" line="98"/>
         <source>&amp;Restart now</source>
-        <translation>$Újraindítás most</translation>
+        <translation>Újraindítás most</translation>
     </message>
     <message>
         <location filename="../src/modules/finished/FinishedPage.cpp" line="51"/>
@@ -2610,4 +2610,4 @@
         <translation>Üdvözlet</translation>
     </message>
 </context>
-</TS>
\ Nincs új sor a fájl végén
+</TS>
