diff -Naur orig/src/modules/unpackfs/unpackfs.conf patched/src/modules/unpackfs/unpackfs.conf
--- orig/src/modules/unpackfs/unpackfs.conf	2019-12-03 11:37:28.000000000 +0100
+++ patched/src/modules/unpackfs/unpackfs.conf	2019-12-14 22:38:41.404163564 +0100
@@ -79,11 +79,6 @@
 # for its destination name, as in the example below.
 
 unpack:
-    -   source: ../CHANGES
-        sourcefs: file
-        destination: "/tmp/changes.txt"
-    -   source: src/qml/calamares/slideshow
-        sourcefs: file
-        destination: "/tmp/slideshow/"
-        exclude: [ "*.qmlc", "qmldir" ]
-        # excludeFile: /etc/calamares/modules/unpackfs/exclude-list.txt
+    -   source: /dev/loop0
+        sourcefs: squashfs
+        destination: ""
