diff -Naur orig/src/modules/unpackfs/unpackfs.conf patched/src/modules/unpackfs/unpackfs.conf
--- orig/src/modules/unpackfs/unpackfs.conf	2019-07-06 00:52:35.000000000 +0200
+++ patched/src/modules/unpackfs/unpackfs.conf	2019-07-31 13:28:11.737738666 +0200
@@ -33,6 +33,6 @@
 
 # You can list filesystem source paths relative to the Calamares run
 # directory, if you use -d (this is only useful for testing, though).
-    -   source: ./example.sqfs
+    -   source: /dev/loop0
         sourcefs: squashfs
         destination: ""
