diff -Naur orig/src/modules/unpackfs/unpackfs.conf patched/src/modules/unpackfs/unpackfs.conf
--- orig/src/modules/unpackfs/unpackfs.conf	2020-10-17 16:41:08.000000000 +0200
+++ patched/src/modules/unpackfs/unpackfs.conf	2020-10-19 17:45:32.887588358 +0200
@@ -88,13 +88,6 @@
 # for its destination name, as in the example below.
 
 unpack:
-    -   source: ../CHANGES
-        sourcefs: file
-        destination: "/tmp/changes.txt"
-        weight: 1  # Single file
-    -   source: src/qml/calamares/slideshow
-        sourcefs: file
-        destination: "/tmp/slideshow/"
-        exclude: [ "*.qmlc", "qmldir" ]
-        weight: 5  # Lots of files
-        # excludeFile: /etc/calamares/modules/unpackfs/exclude-list.txt
+    -   source: "/dev/mapper/live-base"
+        sourcefs: "ext4"
+        destination: ""
