diff -Naur a/src/modules/unpackfs/unpackfs.conf b/src/modules/unpackfs/unpackfs.conf
--- a/src/modules/unpackfs/unpackfs.conf	2024-11-05 15:53:24.000000000 +0100
+++ b/src/modules/unpackfs/unpackfs.conf	2024-11-07 00:02:44.122049039 +0100
@@ -104,13 +104,6 @@
 #        condition: exampleGlobalStorageVariable.subkey
 
 unpack:
-    -   source: ../CHANGES
-        sourcefs: file
-        destination: "/tmp/changes.txt"
-        weight: 1  # Single file
-    -   source: src/qml/calamares/slideshow
-        sourcefs: file
-        destination: "/tmp/slideshow/"
-        exclude: [ "*.qmlc", "qmldir" ]
-        weight: 5  # Lots of files
-        # excludeFile: /etc/calamares/modules/unpackfs/exclude-list.txt
+    -   source: "/dev/live-base"
+        sourcefs: "squashfs"
+        destination: ""
