diff -Naur orig/src/modules/users/users.conf patched/src/modules/users/users.conf
--- orig/src/modules/users/users.conf	2020-03-27 23:43:50.000000000 +0100
+++ patched/src/modules/users/users.conf	2020-04-06 23:39:41.527904325 +0200
@@ -15,11 +15,6 @@
 # Adjust to your Distribution defaults.
 defaultGroups:
     - users
-    - lp
-    - video
-    - network
-    - storage
-    - wheel
     - audio
 
 # Some Distributions require a 'autologin' group for the user.
@@ -43,7 +38,7 @@
 # remove this setting (delete or comment out the line below). Otherwise,
 # the setting will be duplicated in the `/etc/sudoers.d/10-installer` file,
 # potentially confusing users.
-sudoersGroup:    wheel
+# sudoersGroup:    wheel
 
 # Setting this to false, causes the root account to be disabled.
 # When disabled, hides the "Use the same password for administrator"
