diff -Naur orig/src/modules/users/users.conf patched/src/modules/users/users.conf
--- orig/src/modules/users/users.conf	2020-12-04 14:51:48.000000000 +0100
+++ patched/src/modules/users/users.conf	2020-12-06 19:31:39.486125069 +0100
@@ -32,13 +32,6 @@
     - name: users
       must_exist: true
       system: true
-    - lp
-    - video
-    - network
-    - storage
-    - name: wheel
-      must_exist: false
-      system: true
     - audio
 
 # Some Distributions require a 'autologin' group for the user.
@@ -62,7 +55,7 @@
 # remove this setting (delete or comment out the line below). Otherwise,
 # the setting will be duplicated in the `/etc/sudoers.d/10-installer` file,
 # potentially confusing users.
-sudoersGroup:    wheel
+# sudoersGroup:    wheel
 
 # Setting this to false, causes the root account to be disabled.
 # When disabled, hides the "Use the same password for administrator"
