diff -Naur orig/makefile mod/makefile
--- orig/makefile	1970-01-01 01:00:00.000000000 +0100
+++ mod/makefile	2010-09-21 20:58:50.000000000 +0200
@@ -0,0 +1,30 @@
+TARGET=LINUX
+SHELL = /bin/sh
+BINARY = deflektor
+SRCDIR = ./
+INCDIR = include
+SRCSUF = .c
+TESTFILE = test
+PLATFORMDIR=linux
+GCC = g++
+BINSUF = 
+OBJSUF = .o
+LFLAGS = -s -lfmodex `allegro-config --libs`
+CFLAGS = -m$(MODE) -I$(INCDIR) -I/usr/include/fmodex -w -O2
+OBJDIR = ./
+BINDIR = ./
+TEMP = $(wildcard $(SRCDIR)/*$(SRCSUF))
+FILES = $(if $(TEMP), $(TEMP), $(error No source code found!))
+OBJS = $(addprefix $(OBJDIR)/,$(addsuffix $(OBJSUF), $(basename $(notdir $(FILES) ) ) ) )
+
+# main target, all project
+.PHONY: all
+all: $(BINDIR)/$(BINARY)$(BINSUF)
+
+$(BINDIR)/$(BINARY)$(BINSUF) : $(OBJS) 
+	$(GCC) $^ -o $@ $(LFLAGS)
+	@echo The $(BINDIR)/$(BINARY)$(BINSUF) is ready!
+
+$(OBJDIR)/%$(OBJSUF) : $(SRCDIR)/%$(SRCSUF)
+	$(GCC) $(CFLAGS) -c $< -o $@
+
