diff -Naur orig/internet.c mod/internet.c
--- orig/internet.c	2009-01-15 08:21:01.000000000 +0100
+++ mod/internet.c	2009-01-15 09:38:00.000000000 +0100
@@ -3,9 +3,9 @@
 #include "stdio.h"
 
 // USE HERE THE PROPER DEFINITION FOR THE OS YOU ARE COMPILING DEFLEKTOR ON 
-#define DEFL_WIN // Windows
+//#define DEFL_WIN // Windows
 //#define DEFL_MAC // Macintosh
-//#define DEFL_LNX // Linux
+#define DEFL_LNX // Linux
 //#define DEFL_OTR // Other
 
 #ifdef DEFL_WIN
