diff -Naur orig/src/cmd/go/internal/work/init.go patched/src/cmd/go/internal/work/init.go
--- orig/src/cmd/go/internal/work/init.go	2020-02-25 19:32:50.000000000 +0100
+++ patched/src/cmd/go/internal/work/init.go	2020-03-06 01:07:56.022390683 +0100
@@ -131,7 +131,7 @@
 		ldBuildmode = "c-shared"
 	case "default":
 		switch cfg.Goos {
-		case "android":
+		case "linux", "android":
 			codegenArg = "-shared"
 			ldBuildmode = "pie"
 		case "darwin":
