diff -Naur sane-backends-1.0.13.old/backend/gt68xx_devices.c sane-backends-1.0.13/backend/gt68xx_devices.c
--- sane-backends-1.0.13.old/backend/gt68xx_devices.c	2004-04-19 19:26:16.000000000 +0200
+++ sane-backends-1.0.13/backend/gt68xx_devices.c	2004-07-31 22:27:13.000000000 +0200
@@ -1390,6 +1390,7 @@
 static GT68xx_USB_Device_Entry gt68xx_usb_device_list[] = {
   {0x055f, 0x0218, &mustek_2400ta_model},
   {0x055f, 0x0219, &mustek_2400taplus_model},
+  {0x055f, 0x021b, &mustek_1200cuplus_model},
   {0x055f, 0x021c, &mustek_1200cuplus_model},
   {0x055f, 0x021d, &mustek_2400cuplus_model},
   {0x055f, 0x021e, &mustek_1200ta_model},
