diff -Naur orig/backend/gt68xx_devices.c patched/backend/gt68xx_devices.c
--- orig/backend/gt68xx_devices.c	2020-08-23 13:25:14.000000000 +0200
+++ patched/backend/gt68xx_devices.c	2020-10-06 22:35:53.354887078 +0200
@@ -1958,6 +1958,7 @@
   {0x10000, 0x10000, &unknown_model},	/* used for yet unknown scanners */
   {0x055f, 0x0218, &mustek_2400ta_model},
   {0x055f, 0x0219, &mustek_2400taplus_model},
+  {0x055f, 0x021b, &mustek_1200cuplus_model},
   {0x055f, 0x021c, &mustek_1200cuplus_model},
   {0x055f, 0x021b, &mustek_1200cuplus2_model},
   {0x055f, 0x021d, &mustek_2400cuplus_model},
