diff -urN xcalib-0.8.orig/Makefile xcalib-0.8/Makefile
--- xcalib-0.8.orig/Makefile	2007-08-26 23:30:34.000000000 +0200
+++ xcalib-0.8/Makefile	2007-11-27 21:18:37.000000000 +0100
@@ -62,8 +62,8 @@
 	$(CC) $(CFLAGS) -mwindows -lm resource.o -o xcalib xcalib.o
 
 install:
-	cp ./xcalib $(DESTDIR)/usr/local/bin/
-	chmod 0644 $(DESTDIR)/usr/local/bin/xcalib
+	cp ./xcalib $(DESTDIR)/usr/bin/
+	chmod 0644 $(DESTDIR)/usr/bin/xcalib
 
 clean:
 	rm -f xcalib.o
