diff -Naur webkitgtk-2.26.4-orig/Source/WebCore/platform/image-decoders/jpeg2000/JPEG2000ImageDecoder.cpp webkitgtk-2.26.4/Source/WebCore/platform/image-decoders/jpeg2000/JPEG2000ImageDecoder.cpp
--- webkitgtk-2.26.4-orig/Source/WebCore/platform/image-decoders/jpeg2000/JPEG2000ImageDecoder.cpp	2019-11-06 09:09:10.000000000 +0100
+++ webkitgtk-2.26.4/Source/WebCore/platform/image-decoders/jpeg2000/JPEG2000ImageDecoder.cpp	2020-03-01 09:03:08.389273948 +0100
@@ -28,7 +28,7 @@
 
 #if USE(OPENJPEG)
 
-#include <openjpeg.h>
+#include </usr/include/openjpeg-2.3/openjpeg.h>
 
 namespace WebCore {
 
