diff -Naur orig/Source/WebKit/UIProcess/gtk/WaylandCompositor.cpp patched/Source/WebKit/UIProcess/gtk/WaylandCompositor.cpp
--- orig/Source/WebKit/UIProcess/gtk/WaylandCompositor.cpp	2019-08-28 13:46:32.000000000 +0200
+++ patched/Source/WebKit/UIProcess/gtk/WaylandCompositor.cpp	2020-01-10 20:55:16.663927104 +0100
@@ -31,6 +31,7 @@
 #include "WebKitWaylandServerProtocol.h"
 #include <EGL/egl.h>
 #include <EGL/eglext.h>
+#include <EGL/eglmesaext.h>
 #include <WebCore/GLContext.h>
 #include <WebCore/PlatformDisplayWayland.h>
 #include <WebCore/Region.h>
