diff -Naur a/meson.build b/meson.build
--- a/meson.build	2023-04-23 01:12:42.000000000 +0200
+++ b/meson.build	2023-05-05 21:00:38.847275781 +0200
@@ -104,7 +104,7 @@
 gmodule = dependency('gmodule-no-export-2.0', version: glib_ver)
 gnome_autoar = dependency('gnome-autoar-0', version: '>= 0.4.4')
 gnome_desktop = dependency('gnome-desktop-4', version: '>= 43')
-gtk = dependency('gtk4', version: '>= 4.10.3')
+gtk = dependency('gtk4', version: '>= 4.10.1')
 libadwaita = dependency('libadwaita-1', version: '>= 1.3.0')
 libportal = dependency('libportal', version: '>= 0.5')
 libportal_gtk4 = dependency('libportal-gtk4', version: '>= 0.5')
