diff -Naur orig/services/sddm-autologin.pam patched/services/sddm-autologin.pam
--- orig/services/sddm-autologin.pam	2020-11-03 10:55:31.000000000 +0100
+++ patched/services/sddm-autologin.pam	2021-02-24 17:36:59.182022167 +0100
@@ -1,13 +1,15 @@
 #%PAM-1.0
 auth        required    pam_env.so
-auth        required    pam_tally2.so file=/var/log/tallylog onerr=succeed
+auth        required    pam_faillock.so preauth
 auth        required    pam_shells.so
 auth        required    pam_nologin.so
+auth        [success=1 default=ignore]  pam_succeed_if.so user ingroup users
+auth        required    pam_unix.so
 auth        required    pam_permit.so
 -auth       optional    pam_gnome_keyring.so
 -auth       optional    pam_kwallet5.so
-account     include     system-local-login
-password    include     system-local-login
-session     include     system-local-login
+account     include     system-auth
+password    include     system-auth
+session     include     system-auth
 -session    optional    pam_gnome_keyring.so auto_start
 -session    optional    pam_kwallet5.so auto_start
diff -Naur orig/services/sddm.pam patched/services/sddm.pam
--- orig/services/sddm.pam	2020-11-03 10:55:31.000000000 +0100
+++ patched/services/sddm.pam	2021-02-24 17:36:15.809831293 +0100
@@ -1,15 +1,15 @@
 #%PAM-1.0
 
-auth		include		system-login
+auth		include		system-auth
 -auth		optional	pam_gnome_keyring.so
 -auth   optional  pam_kwallet5.so
 
-account		include		system-login
+account		include		system-auth
 
-password	include		system-login
+password	include		system-auth
 -password	optional	pam_gnome_keyring.so use_authtok
 
 session		optional	pam_keyinit.so force revoke
-session		include		system-login
+session		include		system-auth
 -session		optional	pam_gnome_keyring.so auto_start
 -session  optional  pam_kwallet5.so auto_start
