diff -Naur orig/src/common/Configuration.h patched/src/common/Configuration.h
--- orig/src/common/Configuration.h	2018-07-18 12:31:40.000000000 +0200
+++ patched/src/common/Configuration.h	2018-08-10 12:42:38.413224837 +0200
@@ -42,7 +42,7 @@
         Entry(Numlock,             NumState,    NUM_NONE,                                       _S("Initial NumLock state. Can be on, off or none.\n"
                                                                                                    "If property is set to none, numlock won't be changed\n"
                                                                                                    "NOTE: Currently ignored if autologin is enabled."));
-        Entry(InputMethod,         QString,     QStringLiteral("qtvirtualkeyboard"),                   _S("Input method module"));
+        Entry(InputMethod,         QString,     QStringLiteral(),                               _S("Input method module"));
         Entry(Namespaces,          QStringList, QStringList(),                                  _S("Comma-separated list of Linux namespaces for user session to enter"));
         //  Name   Entries (but it's a regular class again)
         Section(Theme,
