diff -Nur orig/CMakeLists.txt mod/CMakeLists.txt
--- orig/CMakeLists.txt	2024-02-26 11:23:03.000000000 +0100
+++ mod/CMakeLists.txt	2024-04-16 13:21:37.154607697 +0200
@@ -187,7 +187,7 @@
 set(LOG_FILE                    "${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/log/sddm.log"  CACHE PATH      "Path of the sddm log file")
 set(DBUS_CONFIG_FILENAME        "org.freedesktop.DisplayManager.conf"               CACHE STRING    "Name of the sddm config file")
 set(COMPONENTS_TRANSLATION_DIR  "${DATA_INSTALL_DIR}/translations-qt${QT_MAJOR_VERSION}" CACHE PATH      "Components translations directory")
-set(SDDM_INITIAL_VT             "1"                                                 CACHE STRING    "Initial tty to use")
+set(SDDM_INITIAL_VT             "7"                                                 CACHE STRING    "Initial tty to use")
 
 
 # Autodetect UID_MIN and UID_MAX from /etc/login.defs
