diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2018-05-31 18:10:35.527532600 +0200
+++ patched/CMakeLists.txt	2018-05-31 18:12:16.372550679 +0200
@@ -93,7 +93,7 @@
 find_package(XKB REQUIRED)
 
 # Qt 5
-find_package(Qt5 5.6.0 CONFIG REQUIRED Core DBus Gui Qml Quick LinguistTools)
+find_package(Qt5 5.8.0 CONFIG REQUIRED Core DBus Gui Qml Quick LinguistTools Test)
 
 # find qt5 imports dir
 get_target_property(QMAKE_EXECUTABLE Qt5::qmake LOCATION)
diff -Naur orig/test/CMakeLists.txt patched/test/CMakeLists.txt
--- orig/test/CMakeLists.txt	2017-12-05 17:00:16.000000000 +0100
+++ patched/test/CMakeLists.txt	2018-05-31 18:12:46.320853021 +0200
@@ -2,9 +2,8 @@
 
 include_directories(../src/common)
 
-
 set(ConfigurationTest_SRCS ConfigurationTest.cpp ../src/common/ConfigReader.cpp)
 add_executable(ConfigurationTest ${ConfigurationTest_SRCS})
 add_test(NAME Configuration COMMAND ConfigurationTest)
 
-qt5_use_modules(ConfigurationTest Test)
+target_link_libraries(ConfigurationTest Qt5::Core Qt5::Test)
