diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2018-01-20 15:19:44.000000000 +0100
+++ patched/configure.ac	2018-04-13 01:53:22.192662804 +0200
@@ -125,9 +125,6 @@
         ;;
 esac
 
-dnl Documentation
-GTK_DOC_CHECK(1.0)
-
 # QMI username
 QMI_USERNAME="root"
 AC_ARG_ENABLE(qmi-username,
@@ -201,9 +198,6 @@
                  src/qmi-firmware-update/test/Makefile
                  utils/Makefile
                  docs/Makefile
-                 docs/reference/Makefile
-                 docs/reference/libqmi-glib/Makefile
-                 docs/reference/libqmi-glib/version.xml
                  docs/man/Makefile])
 
 if test "x$QMI_USERNAME_ENABLED" = "xyes"; then
diff -Naur orig/docs/Makefile.am patched/docs/Makefile.am
--- orig/docs/Makefile.am	2018-01-20 15:19:44.000000000 +0100
+++ patched/docs/Makefile.am	2018-04-13 01:53:22.192662804 +0200
@@ -1 +1 @@
-SUBDIRS = reference man
+SUBDIRS = man
diff -Naur orig/docs/reference/libqmi-glib/Makefile.am patched/docs/reference/libqmi-glib/Makefile.am
--- orig/docs/reference/libqmi-glib/Makefile.am	2018-01-20 15:19:44.000000000 +0100
+++ patched/docs/reference/libqmi-glib/Makefile.am	2018-04-13 01:53:22.192662804 +0200
@@ -76,8 +76,6 @@
 
 extra_files =
 
-include $(top_srcdir)/gtk-doc.make
-
 EXTRA_DIST += \
 	libqmi-glib-common.sections \
 	$(DOC_MODULE)-sections.txt \
