diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2020-10-07 10:57:35.000000000 +0200
+++ patched/configure.ac	2020-12-31 23:25:05.476231086 +0100
@@ -172,16 +172,6 @@
         ;;
 esac
 
-dnl Documentation
-GTK_DOC_CHECK(1.0)
-if test "x$enable_gtk_doc" = "xyes"; then
-    if test "x$QMI_COLLECTION_NAME" != "xfull"; then
-        AC_MSG_WARN([Cannot build gtk-doc when using an explicit message collection.])
-        enable_gtk_doc=no
-    fi
-fi
-AM_CONDITIONAL([ENABLE_GTK_DOC], [test "x$enable_gtk_doc" = "xyes"])
-
 # QMI username
 QMI_USERNAME="root"
 AC_ARG_ENABLE(qmi-username,
@@ -282,9 +272,6 @@
                  examples/Makefile
                  examples/simple-tester-python/Makefile
                  docs/Makefile
-                 docs/reference/Makefile
-                 docs/reference/libqmi-glib/Makefile
-                 docs/reference/libqmi-glib/version.xml
                  docs/man/Makefile])
 
 if test "x$QMI_USERNAME_ENABLED" = "xyes"; then
diff -Naur orig/docs/Makefile.am patched/docs/Makefile.am
--- orig/docs/Makefile.am	2020-10-07 10:57:35.000000000 +0200
+++ patched/docs/Makefile.am	2020-12-31 23:24:44.172176828 +0100
@@ -1 +1 @@
-SUBDIRS = reference man
+SUBDIRS = man
