diff -urN xephem-3.5.2.orig/GUI/xephem/fallbacks.c xephem-3.5.2/GUI/xephem/fallbacks.c
--- xephem-3.5.2.orig/GUI/xephem/fallbacks.c	2002-01-08 07:25:08.000000000 +0100
+++ xephem-3.5.2/GUI/xephem/fallbacks.c	2003-04-23 17:15:03.000000000 +0200
@@ -670,7 +670,7 @@
     "XEphem.Pressure: 1010 mB",
     "XEphem.SaturnBackground: #151515",
     "XEphem.ShareDir: .",
-    "XEphem.Sitename: Chicago, Illinois",
+    "XEphem.Sitename: Budapest, Hungary",
     "XEphem.SkyAnnotColor: #bfc1c6",
     "XEphem.SkyCnsBndColor: #91bfd8",
     "XEphem.SkyCnsFigColor: #d6c100",
