Index: gnomeradio-1.8/data/gnomeradio.desktop.in
===================================================================
--- gnomeradio-1.8.orig/data/gnomeradio.desktop.in	2013-09-25 13:05:41.526733873 +0000
+++ gnomeradio-1.8/data/gnomeradio.desktop.in	2013-09-25 13:09:07.518727749 +0000
@@ -1,13 +1,14 @@
 [Desktop Entry]
-_Name=FM-Radio Tuner
-_Comment=Listen to FM-radio
+_Name=Gnomeradio
+_Comment=Listen to FM radio
+_Keywords=Radio;Tuner;FM;
 Exec=gnomeradio
 Icon=gnomeradio
-Categories=Application;AudioVideo;
-StartupNotify=true
 Terminal=false
 Type=Application
+Categories=GNOME;GTK;AudioVideo;Audio;Tuner;
+X-GNOME-DocPath=gnomeradio/gnomeradio.xml
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnomeradio
 X-GNOME-Bugzilla-Component=general
-X-GNOME-DocPath=gnomeradio/gnomeradio.xml
+StartupNotify=true
