diff -up gnet-2.0.8/examples/Makefile.in.build gnet-2.0.8/examples/Makefile.in
--- gnet-2.0.8/examples/Makefile.in.build	2008-02-03 13:07:21.000000000 +0100
+++ gnet-2.0.8/examples/Makefile.in	2010-02-16 19:42:24.000000000 +0100
@@ -254,7 +254,7 @@ INSTALL_SCRIPT = @INSTALL_SCRIPT@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
 LDFLAGS = @LDFLAGS@
 LIBOBJS = @LIBOBJS@
-LIBS = @LIBS@
+LIBS = @LIBS@ $(GLIB_LIBS)
 LIBTOOL = @LIBTOOL@
 LN_S = @LN_S@
 LTLIBOBJS = @LTLIBOBJS@
diff -up gnet-2.0.8/examples/xmlrpc/Makefile.in.build gnet-2.0.8/examples/xmlrpc/Makefile.in
--- gnet-2.0.8/examples/xmlrpc/Makefile.in.build	2008-02-03 13:07:21.000000000 +0100
+++ gnet-2.0.8/examples/xmlrpc/Makefile.in	2010-02-16 19:42:24.000000000 +0100
@@ -176,7 +176,7 @@ INSTALL_SCRIPT = @INSTALL_SCRIPT@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
 LDFLAGS = @LDFLAGS@
 LIBOBJS = @LIBOBJS@
-LIBS = @LIBS@
+LIBS = @LIBS@ $(GLIB_LIBS)
 LIBTOOL = @LIBTOOL@
 LN_S = @LN_S@
 LTLIBOBJS = @LTLIBOBJS@
diff -up gnet-2.0.8/tests/check/gnetcheck.h.build gnet-2.0.8/tests/check/gnetcheck.h
--- gnet-2.0.8/tests/check/gnetcheck.h.build	2008-02-03 13:04:41.000000000 +0100
+++ gnet-2.0.8/tests/check/gnetcheck.h	2010-02-16 19:42:24.000000000 +0100
@@ -310,6 +310,8 @@ int main (int argc, char **argv)				\
 
 gboolean _gnet_check_run_test_func (const gchar * func_name);
 
+#define _tcase_add_test  __gnet_tcase_add_test
+
 static inline void
 __gnet_tcase_add_test (TCase * tc, TFun tf, const gchar * func_name,
     int sig, int start, int end)
@@ -319,7 +321,6 @@ __gnet_tcase_add_test (TCase * tc, TFun 
   }
 }
 
-#define _tcase_add_test  __gnet_tcase_add_test
 
 G_END_DECLS
 
diff -up gnet-2.0.8/tests/check/Makefile.in.build gnet-2.0.8/tests/check/Makefile.in
--- gnet-2.0.8/tests/check/Makefile.in.build	2008-02-03 13:07:21.000000000 +0100
+++ gnet-2.0.8/tests/check/Makefile.in	2010-02-16 19:44:33.000000000 +0100
@@ -272,7 +272,7 @@ INSTALL_SCRIPT = @INSTALL_SCRIPT@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
 LDFLAGS = @LDFLAGS@
 LIBOBJS = @LIBOBJS@
-LIBS = @LIBS@
+LIBS = @LIBS@ $(GLIB_LIBS)
 LIBTOOL = @LIBTOOL@
 LN_S = @LN_S@
 LTLIBOBJS = @LTLIBOBJS@

