diff -Nur orig/lightdm-qt5-greeter.conf mod/lightdm-qt5-greeter.conf
--- orig/lightdm-qt5-greeter.conf	2019-11-20 10:08:49.000000000 +0100
+++ mod/lightdm-qt5-greeter.conf	2019-11-20 10:11:32.040872278 +0100
@@ -14,6 +14,6 @@
 
 [General]
 ;greeter-icon-theme=oxygen
-;greeter-background-image=/path/to/background/image.png
+greeter-background-image=/usr/share/backgrounds/UHU/ubk4.png
 ;loginform-offset-x=50%
 ;loginform-offset-y=50%
