diff --git a/CMakeLists.txt b/CMakeLists.txt
index e60504f3723c155fb4a8fab3b4620d622fee37a7..010aa106d0cd002cfcdb51a0c34eb4e943cf980b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.0.2 FATAL_ERROR)
-set(PROJECT qt-lightdm-greeter)
+set(PROJECT lightdm-qt5-greeter)
 project(${PROJECT})
 
 set(CMAKE_AUTOMOC ON)
@@ -11,7 +11,7 @@ message(STATUS "SRCS ${SRCS}")
 find_package(Qt5Widgets REQUIRED)
 #find_package(Qt5X11Extras REQUIRED QUIET)
 
-QT5_ADD_RESOURCES(RSCS qt-lightdm-greeter.qrc)
+QT5_ADD_RESOURCES(RSCS lightdm-qt5-greeter.qrc)
 
 
 include(FindPkgConfig)
@@ -22,15 +22,15 @@ include_directories ( ${CMAKE_CURRENT_SOURCE_DIR}
                       ${LIGHTDM_QT_INCLUDE_DIRS}
 )
 
-add_executable ( qt-lightdm-greeter ${SRCS} ${RSCS})
+add_executable ( lightdm-qt5-greeter ${SRCS} ${RSCS})
 
-target_link_libraries ( qt-lightdm-greeter Qt5::Widgets ${LIGHTDM_QT_LIBRARIES} )
+target_link_libraries ( lightdm-qt5-greeter Qt5::Widgets ${LIGHTDM_QT_LIBRARIES} )
 
 
 install(TARGETS ${PROJECT} RUNTIME DESTINATION bin)
-install(FILES   qt-lightdm-greeter.desktop  DESTINATION /usr/share/xgreeters)
+install(FILES   lightdm-qt5-greeter.desktop  DESTINATION /usr/share/xgreeters)
 
 if (NOT KEEP_CONFIGS)
-    install(FILES qt-lightdm-greeter.conf DESTINATION /etc/lightdm)
+    install(FILES lightdm-qt5-greeter.conf DESTINATION /etc/lightdm)
 endif()
 
diff --git a/README.md b/README.md
index ed719d0c78a44bf3a7bf2e7ba1bb13358a8d1faa..7d9c0dd2a9b7372f37b09d824504ae722d777716 100644
--- a/README.md
+++ b/README.md
@@ -1,12 +1,12 @@
-# qt-lightdm-greeter
+# lightdm-qt5-greeter
 
-qt-lightdm-greeter is a simple frontend for the lightdm displaymanager, written in c++ and qt5.
+lightdm-qt5-greeter is a simple frontend for the lightdm displaymanager, written in c++ and qt5.
 
-It has it's roots in the razor project, where it was developed under the name razor-lightdm-greeter. When razor was merged into LXQt it was renamed to lxqt-lightdm-greeter.
+It has it's roots in the razor project, where it was developed under the name razor-lightdm-greeter. When razor was merged into LXQt it was renamed to lxlightdm-qt5-greeter.
 
-As of september 2015 I have renamed it to qt-lightdm-greeter and cut it's ties to LXQt. This was primarily because LXQt recommends SDDM as its displaymanager and because the application had very little integration with LXQt. 
+As of september 2015 I have renamed it to lightdm-qt5-greeter and cut it's ties to LXQt. This was primarily because LXQt recommends SDDM as its displaymanager and because the application had very little integration with LXQt. 
 
-qt-lightdm-greeter sports a razor-qt logo as a reminder of it's origin.
+lightdm-qt5-greeter sports a razor-qt logo as a reminder of it's origin.
 
 ## Installing
 
@@ -15,8 +15,8 @@ Make sure you have cmake, gcc, qt5 and liblightdm-qt5 (and lightdm) installed on
 Do
     
 ```shell
-git clone https://github.com/surlykke/qt-lightdm-greeter.git
-cd qt-lightdm-greeter
+git clone https://github.com/surlykke/lightdm-qt5-greeter.git
+cd lightdm-qt5-greeter
 mkdir build
 cd build
 cmake ..
@@ -28,11 +28,11 @@ sudo make install
 
 Update or insert in(to) your `/etc/lightdm/lightdm.conf`, in the `SeatDefaults` section, this line:
 
-    greeter-session=qt-lightdm-greeter
+    greeter-session=lightdm-qt5-greeter
 
-## Configure qt-lightdm-greeter
+## Configure lightdm-qt5-greeter
 
-The file `/etc/lightdm/qt-lightdm-greeter.conf` allows for a 
-few configurations of qt-lightdm-greeter (background-image, positioning of loginform). 
+The file `/etc/lightdm/lightdm-qt5-greeter.conf` allows for a 
+few configurations of lightdm-qt5-greeter (background-image, positioning of loginform). 
 The configuration options are documented in that file.
 
diff --git a/license-header.txt b/license-header.txt
index 07a20aeb5ac6b7693a81cfb6670124f5a348234e..6dfd860102ee8bd5319e5932e89dee0733272014 100644
--- a/license-header.txt
+++ b/license-header.txt
@@ -1,7 +1,7 @@
 /*
 * Copyright (c) 2012-2015 Christian Surlykke
 *
-* This file is part of qt-lightdm-greeter 
+* This file is part of lightdm-qt5-greeter 
 * It is distributed under the LGPL 2.1 or later license.
 * Please refer to the LICENSE file for a copy of the license.
 */
diff --git a/qt-lightdm-greeter.conf b/lightdm-qt5-greeter.conf
similarity index 96%
rename from qt-lightdm-greeter.conf
rename to lightdm-qt5-greeter.conf
index 703d2dc9da76a7db1bf40476a3ddf5cec9dc7105..0c7536ac05183c93ee171a6aa93cc254ed35e709 100644
--- a/qt-lightdm-greeter.conf
+++ b/lightdm-qt5-greeter.conf
@@ -1,4 +1,4 @@
-; qt-lightdm-greeter.conf
+; lightdm-qt5-greeter.conf
 ;
 ; General configuration
 ;
diff --git a/loginform.cpp b/loginform.cpp
index 3f40e817027841f7a49461643fbe1447f582073e..8a0e33382cd41266319a1f64461f3e853f98b7fb 100644
--- a/loginform.cpp
+++ b/loginform.cpp
@@ -1,7 +1,7 @@
 /*
 * Copyright (c) 2012-2015 Christian Surlykke
 *
-* This file is part of qt-lightdm-greeter 
+* This file is part of lightdm-qt5-greeter 
 * It is distributed under the LGPL 2.1 or later license.
 * Please refer to the LICENSE file for a copy of the license.
 */
diff --git a/loginform.h b/loginform.h
index 8346a58eaa38fd9831fb02844077d5d622b4ae16..174d3da27de95996ca65fd3d49a5f5e629e15fc0 100644
--- a/loginform.h
+++ b/loginform.h
@@ -1,7 +1,7 @@
 /*
 * Copyright (c) 2012-2015 Christian Surlykke
 *
-* This file is part of qt-lightdm-greeter 
+* This file is part of lightdm-qt5-greeter 
 * It is distributed under the LGPL 2.1 or later license.
 * Please refer to the LICENSE file for a copy of the license.
 */
diff --git a/loginform.ui b/loginform.ui
index 3525a9dc4396a26e3686c9c45a078066e6cf03c6..aed074e6c939f698c6933a9669785daba5a95bfc 100644
--- a/loginform.ui
+++ b/loginform.ui
@@ -121,7 +121,7 @@ QComboBox::down-arrow {
          <string/>
         </property>
         <property name="pixmap">
-         <pixmap resource="qt-lightdm-greeter.qrc">:/resources/rqt-2.png</pixmap>
+         <pixmap resource="lightdm-qt5-greeter.qrc">:/resources/rqt-2.png</pixmap>
         </property>
         <property name="scaledContents">
          <bool>true</bool>
@@ -249,7 +249,7 @@ QComboBox::down-arrow {
           <string/>
          </property>
          <property name="icon">
-          <iconset resource="qt-lightdm-greeter.qrc">
+          <iconset resource="lightdm-qt5-greeter.qrc">
            <normaloff>:/resources/leaveIcon.svg</normaloff>:/resources/leaveIcon.svg</iconset>
          </property>
         </item>
@@ -261,7 +261,7 @@ QComboBox::down-arrow {
   </layout>
  </widget>
  <resources>
-  <include location="qt-lightdm-greeter.qrc"/>
+  <include location="lightdm-qt5-greeter.qrc"/>
  </resources>
  <connections/>
 </ui>
diff --git a/main.cpp b/main.cpp
index c0ce7bcde6473c24c54be7768c1bc67921e93564..fce0970da5920364d6b3842a98769c21a9d1157e 100644
--- a/main.cpp
+++ b/main.cpp
@@ -1,7 +1,7 @@
 /*
 * Copyright (c) 2012-2015 Christian Surlykke, Petr Vanek
 *
-* This file is part of qt-lightdm-greeter 
+* This file is part of lightdm-qt5-greeter 
 * It is distributed under the LGPL 2.1 or later license.
 * Please refer to the LICENSE file for a copy of the license.
 */
diff --git a/mainwindow.cpp b/mainwindow.cpp
index a8dbd2177fd6a5cbf2a607498c0ae80b177bd797..985267ff0b327538088e02191dbda541ab7694c6 100644
--- a/mainwindow.cpp
+++ b/mainwindow.cpp
@@ -1,7 +1,7 @@
 /*
 * Copyright (c) 2012-2015 Christian Surlykke, Petr Vanek
 *
-* This file is part of qt-lightdm-greeter 
+* This file is part of lightdm-qt5-greeter 
 * It is distributed under the LGPL 2.1 or later license.
 * Please refer to the LICENSE file for a copy of the license.
 */
diff --git a/mainwindow.h b/mainwindow.h
index 738d423e365a9c29f8fec43b1d4df33b5345d9a9..61724529761e96341885afa77f64937006e48be8 100644
--- a/mainwindow.h
+++ b/mainwindow.h
@@ -1,7 +1,7 @@
 /*
 * Copyright (c) 2012-2015 Christian Surlykke, Petr Vanek
 *
-* This file is part of qt-lightdm-greeter 
+* This file is part of lightdm-qt5-greeter 
 * It is distributed under the LGPL 2.1 or later license.
 * Please refer to the LICENSE file for a copy of the license.
 */
diff --git a/qt-lightdm-greeter.desktop b/qt-lightdm-greeter.desktop
index f43d1ffd0c7dff933ed8674d84cc756eec8000f6..86a98eacab294bb833afff52c03a8571dd4897a8 100644
--- a/qt-lightdm-greeter.desktop
+++ b/qt-lightdm-greeter.desktop
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Name=Qt LightDM Greeter
 Comment=This runs the Qt LightDM greeter, it should only be run from LightDM
-Exec=qt-lightdm-greeter
+Exec=lightdm-qt5-greeter
 Type=Application
 X-Ubuntu-Gettext-Domain=lightdm
diff --git a/settings.cpp b/settings.cpp
index 0c37166ffeb44757711a921829c141d311710066..b0d356e897daad3f43bf2de1418a00f522c99bc9 100644
--- a/settings.cpp
+++ b/settings.cpp
@@ -7,7 +7,7 @@
 #define LOGINFORM_OFFSETY_KEY "loginform-offset-y"
 #define LOGFILE_PATH_KEY "logfile-path"
 
-const QString Cache::GREETER_DATA_DIR_PATH = "/var/lib/lightdm/qt-lightdm-greeter";
+const QString Cache::GREETER_DATA_DIR_PATH = "/var/lib/lightdm/lightdm-qt5-greeter";
 
 void Cache::prepare()
 {
diff --git a/settings.h b/settings.h
index 7f2b13589875903379480fe0655e2c1c0fb4eb5e..decbe9ed0644608ad61f2d7ba25706d96f8c8c9e 100644
--- a/settings.h
+++ b/settings.h
@@ -19,7 +19,7 @@ public:
     void setLastSession(QString userId, QString session) { setValue(userId + "/last-session", session); }
 };
 
-#define CONFIG_FILE "/etc/lightdm/qt-lightdm-greeter.conf"
+#define CONFIG_FILE "/etc/lightdm/lightdm-qt5-greeter.conf"
 #define BACKGROUND_IMAGE_KEY "greeter-background-image"
 #define LOGINFORM_OFFSETX_KEY "loginform-offset-x"
 #define LOGINFORM_OFFSETY_KEY "loginform-offset-y"
@@ -28,7 +28,7 @@ public:
 class Settings : public QSettings
 {
 public:
-    Settings() : QSettings(QString("/etc/lightdm/qt-lightdm-greeter.conf"), QSettings::NativeFormat) {}
+    Settings() : QSettings(QString("/etc/lightdm/lightdm-qt5-greeter.conf"), QSettings::NativeFormat) {}
     QString iconThemeName() { return value("greeter-icon-theme").toString(); }
     QString backgrundImagePath() { return value("greeter-background-image").toString(); }
     QString offsetX() { return value("loginform-offset-x").toString(); }
