diff --git a/qt-lightdm-greeter.desktop b/lightdm-qt5-greeter.desktop
similarity index 83%
rename from qt-lightdm-greeter.desktop
rename to lightdm-qt5-greeter.desktop
index 86a98eacab294bb833afff52c03a8571dd4897a8..ac2d6eb5a4b51ddb4fa6131549a01bfbbe8bd7e4 100644
--- a/qt-lightdm-greeter.desktop
+++ b/lightdm-qt5-greeter.desktop
@@ -3,4 +3,3 @@ Name=Qt LightDM Greeter
 Comment=This runs the Qt LightDM greeter, it should only be run from LightDM
 Exec=lightdm-qt5-greeter
 Type=Application
-X-Ubuntu-Gettext-Domain=lightdm
diff --git a/qt-lightdm-greeter.qrc b/lightdm-qt5-greeter.qrc
similarity index 100%
rename from qt-lightdm-greeter.qrc
rename to lightdm-qt5-greeter.qrc
