diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2019-11-19 10:54:08.516621708 +0100
+++ patched/CMakeLists.txt	2019-11-19 11:08:09.290600231 +0100
@@ -9,7 +9,8 @@
 message(STATUS "SRCS ${SRCS}")
 
 find_package(Qt5Widgets REQUIRED)
-#find_package(Qt5X11Extras REQUIRED QUIET)
+find_package(Qt5Gui REQUIRED)
+find_package(Qt5Core REQUIRED)
 
 QT5_ADD_RESOURCES(RSCS lightdm-qt5-greeter.qrc)
 
@@ -24,7 +25,7 @@
 
 add_executable ( lightdm-qt5-greeter ${SRCS} ${RSCS})
 
-target_link_libraries ( lightdm-qt5-greeter Qt5::Widgets ${LIGHTDM_QT_LIBRARIES} )
+target_link_libraries ( lightdm-qt5-greeter lightdm-qt5-3 Qt5::Widgets Qt5::Gui Qt5::Core )
 
 
 install(TARGETS ${PROJECT} RUNTIME DESTINATION bin)
