diff -Naurd a/fbreader/src/database/sqldb/implsqlite/SQLiteCommand.cpp b/fbreader/src/database/sqldb/implsqlite/SQLiteCommand.cpp
--- a/fbreader/src/database/sqldb/implsqlite/SQLiteCommand.cpp	2012-11-17 18:09:55.000000000 +0100
+++ b/fbreader/src/database/sqldb/implsqlite/SQLiteCommand.cpp	2012-11-20 18:43:52.000000000 +0100
@@ -17,6 +17,7 @@
  * 02110-1301, USA.
  */
 
+#include <stdio.h>
 #include <iostream>
 #include <algorithm>
 
diff -Naurd a/zlibrary/core/src/util/ZLStringUtil.h b/zlibrary/core/src/util/ZLStringUtil.h
--- a/zlibrary/core/src/util/ZLStringUtil.h	2012-11-30 13:41:25.000000000 +0100
+++ b/zlibrary/core/src/util/ZLStringUtil.h	2012-12-17 21:18:32.000000000 +0100
@@ -20,6 +20,7 @@
 #ifndef __ZLSTRINGUTIL_H__
 #define __ZLSTRINGUTIL_H__
 
+#include <stdio.h>
 #include <vector>
 #include <string>
 
