diff -Naur a/makefiles/arch/desktop.mk b/makefiles/arch/desktop.mk
--- a/makefiles/arch/desktop.mk	2012-12-24 05:00:26.000000000 +0100
+++ b/makefiles/arch/desktop.mk	2022-07-04 00:37:49.922168736 +0200
@@ -12,7 +12,7 @@
 AR = ar rsu
 LD = g++
 
-CFLAGS = -pipe -fno-exceptions -Wall -Wno-ctor-dtor-privacy -W -DLIBICONV_PLUG
+CFLAGS = -fPIC -pipe -fno-exceptions -Wall -Wno-ctor-dtor-privacy -W -DLIBICONV_PLUG
 LDFLAGS =
 EXTERNAL_INCLUDE = $(shell pkg-config --cflags fribidi) 
 
diff -Naur a/makefiles/config.mk b/makefiles/config.mk
--- a/makefiles/config.mk	2012-12-24 05:00:26.000000000 +0100
+++ b/makefiles/config.mk	2022-07-04 00:36:53.382021713 +0200
@@ -26,13 +26,13 @@
 endif
 
 ifeq "$(TARGET_STATUS)" "release"
-	CFLAGS += -O3
+	CFLAGS += -O2
 endif
 ifeq "$(TARGET_STATUS)" "debug"
 	CFLAGS += -O0 -g
 endif
 ifeq "$(TARGET_STATUS)" "profile"
-	CFLAGS += -O3 -g -pg
+	CFLAGS += -O2 -g -pg
 	LDFLAGS += -pg
 endif
 
