diff -Nur orig/bin/frozen-bubble mod/bin/frozen-bubble
--- orig/bin/frozen-bubble	2018-02-07 19:27:43.000000000 +0100
+++ mod/bin/frozen-bubble	2018-02-07 21:33:13.732006273 +0100
@@ -35,6 +35,7 @@
 
 #use diagnostics;
 use strict;
+use lib '/usr/lib/perl5/5.26.1/IO';
 
 use vars qw($TARGET_ANIM_SPEED $BUBBLE_SIZE $ROW_SIZE $LAUNCHER_SPEED $BUBBLE_SPEED $MALUS_BUBBLE_SPEED $TIME_APPEARS_NEW_ROOT
             %POS %POS_1P %POS_2P %MENUPOS $KEYS %actions %angle %pdata $app %apprects $event %rects %sticked_bubbles %root_bubbles
diff -Nur orig/Build.PL mod/Build.PL
--- orig/Build.PL	2018-02-07 19:27:43.000000000 +0100
+++ mod/Build.PL	2018-02-07 21:57:27.869263427 +0100
@@ -1,7 +1,6 @@
 #!perl
 use 5.008;
 use strict;
-use warnings FATAL => 'all';
 use lib 'inc';
 use Alien::SDL qw();
 use SDL::Config;
@@ -11,6 +10,7 @@
 use ExtUtils::CBuilder qw();
 
 use lib 'lib';
+use lib '/usr/lib/perl5/5.26.1/IO';
 use Games::FrozenBubble;
 
 my $prefix = Alien::SDL->config('prefix');
diff -Nur orig/server/tools.c mod/server/tools.c
--- orig/server/tools.c	2018-02-07 19:27:43.000000000 +0100
+++ mod/server/tools.c	2018-02-07 22:16:05.999391377 +0100
@@ -261,8 +261,6 @@
         if (user_to_switch != NULL) {
                 struct passwd* user = getpwnam(user_to_switch);
                 if (user) {
-                        setgid(user->pw_gid);
-                        setuid(user->pw_uid);
                 } else {
                         l2(OUTPUT_TYPE_ERROR, "Cannot switch user to %s: %s", user_to_switch, strerror(errno));
                 }
