diff -Naur orig/schemas/org.mate.background.gschema.xml.in patched/schemas/org.mate.background.gschema.xml.in
--- orig/schemas/org.mate.background.gschema.xml.in	2015-10-07 15:46:56.000000000 +0200
+++ patched/schemas/org.mate.background.gschema.xml.in	2015-10-16 02:30:47.746232358 +0200
@@ -34,7 +34,7 @@
       <description>Determines how the image set by wallpaper_filename is rendered.  Possible values are "wallpaper", "centered", "scaled", "stretched", "zoom", "spanned".</description>
     </key>
     <key name="picture-filename" type="s">
-      <default>'@datadir@/backgrounds/mate/desktop/Stripes.png'</default>
+      <default>'@datadir@/backgrounds/UHU/uhu_linux_ubk.jpg'</default>
       <summary>Picture Filename</summary>
       <description>File to use for the background image.</description>
     </key>
