diff -Naur a/libmate-desktop/mate-desktop-utils.c b/libmate-desktop/mate-desktop-utils.c
--- a/libmate-desktop/mate-desktop-utils.c	2022-11-11 20:48:24.000000000 +0100
+++ b/libmate-desktop/mate-desktop-utils.c	2022-11-14 22:57:49.140794658 +0100
@@ -130,30 +130,21 @@
 		term_argv = g_new0 (char *, 3);
 
 		check = g_find_program_in_path ("mate-terminal");
-		if (check != NULL) {
-			term_argv[0] = check;
-			/* Note that mate-terminal takes -x and
-			 * as -e in mate-terminal is broken we use that. */
-			term_argv[1] = g_strdup ("-x");
-		} else {
-			if (check == NULL)
-				check = g_find_program_in_path ("nxterm");
-			if (check == NULL)
-				check = g_find_program_in_path ("color-xterm");
-			if (check == NULL)
-				check = g_find_program_in_path ("rxvt");
-			if (check == NULL)
-				check = g_find_program_in_path ("xterm");
-			if (check == NULL)
-				check = g_find_program_in_path ("dtterm");
-			if (check == NULL) {
-				g_warning (_("Cannot find a terminal, using "
-					     "xterm, even if it may not work"));
-				check = g_strdup ("xterm");
-			}
-			term_argv[0] = check;
-			term_argv[1] = g_strdup ("-e");
+		if (check == NULL)
+			check = g_find_program_in_path ("gnome-terminal");
+		if (check == NULL)
+			check = g_find_program_in_path ("xfce4-terminal");
+		if (check == NULL)
+			check = g_find_program_in_path ("lxterminal");
+		if (check == NULL)
+			check = g_find_program_in_path ("sakura");
+		if (check == NULL) {
+			g_warning (_("Cannot find a terminal, using "
+				     "xterm, even if it may not work"));
+			check = g_strdup ("xterm");
 		}
+		term_argv[0] = check;
+		term_argv[1] = g_strdup ("-e");
 	}
 
         real_argc = term_argc + *argc;
