diff -Nur orig/examples/factorial mod/examples/factorial
--- orig/examples/factorial	2012-05-12 00:56:14.000000000 +0200
+++ mod/examples/factorial	2019-12-04 18:50:34.744373561 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # This is a Python program to display large factorials and test "fact.py".
 
diff -Nur orig/primes/matho-mult mod/primes/matho-mult
--- orig/primes/matho-mult	2011-08-04 14:12:08.000000000 +0200
+++ mod/primes/matho-mult	2019-12-04 18:50:46.544359029 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # Python program to multiply many large integers separated by spaces or newlines.
 # The integers to multiply may be entered on the command-line or into standard input.
diff -Nur orig/primes/matho-sum mod/primes/matho-sum
--- orig/primes/matho-sum	2011-08-04 14:04:54.000000000 +0200
+++ mod/primes/matho-sum	2019-12-04 18:51:08.696331730 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # Python program to sum many large integers separated by spaces or newlines.
 # The integers to sum may be entered on the command-line or into standard input.
diff -Nur orig/primes/primorial mod/primes/primorial
--- orig/primes/primorial	2012-06-14 17:11:52.000000000 +0200
+++ mod/primes/primorial	2019-12-04 18:50:53.700350212 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # This is a Python program to display large primorials.
 # A primorial is the product of all primes up to the given number.
