diff -Nur orig/barrage.desktop mod/barrage.desktop
--- orig/barrage.desktop	2011-07-28 17:00:11.000000000 +0200
+++ mod/barrage.desktop	2018-02-09 20:10:59.956978355 +0100
@@ -1,7 +1,8 @@
 [Desktop Entry]
 Name=Barrage
 Comment=A rather destructive action game.
-Exec=barrage
-Icon=barrage.png
+Comment[hu]=Semmisíts meg minél több célpontot 3 perc alatt
+Exec=barrage
+Icon=barrage
 Type=Application
-Categories=Game;
+Categories=Game;ActionGame;
