diff -Nur orig/lib/pychess/Players/PyChessCECP.py mod/lib/pychess/Players/PyChessCECP.py
--- orig/lib/pychess/Players/PyChessCECP.py	2018-12-08 19:04:42.000000000 +0100
+++ mod/lib/pychess/Players/PyChessCECP.py	2019-01-25 12:40:41.752182200 +0100
@@ -25,10 +25,6 @@
 from pychess.Variants.placement import PLACEMENTSTART
 from pychess.Variants.asean import ASEANSTART, MAKRUKSTART, KAMBODIANSTART, SITTUYINSTART
 
-if sys.platform != "win32":
-    import readline
-    readline.clear_history()
-
 ASCII = sys.platform == "win32"
 
 
