diff -Naur orig/terminal/terminal-preferences.c patched/terminal/terminal-preferences.c
--- orig/terminal/terminal-preferences.c	2020-12-22 22:48:03.000000000 +0100
+++ patched/terminal/terminal-preferences.c	2020-12-24 15:44:07.784566112 +0100
@@ -785,7 +785,7 @@
       g_param_spec_boolean ("font-allow-bold",
                             NULL,
                             "FontAllowBold",
-                            TRUE,
+                            FALSE,
                             G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS);
 
   /**
@@ -926,7 +926,7 @@
       g_param_spec_boolean ("misc-toolbar-default",
                             NULL,
                             "MiscToolbarDefault",
-                            FALSE,
+                            TRUE,
                             G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS);
 
   /**
@@ -1098,7 +1098,7 @@
       g_param_spec_uint ("scrolling-lines",
                          NULL,
                          "ScrollingLines",
-                         0u, 1024u * 1024u, 1000u,
+                         0u, 1024u * 1024u, 2000u,
                          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS);
 
   /**
@@ -1118,7 +1118,7 @@
       g_param_spec_boolean ("scrolling-on-output",
                             NULL,
                             "ScrollingOnOutput",
-                            TRUE,
+                            FALSE,
                             G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS);
 
   /**
@@ -1152,7 +1152,7 @@
       g_param_spec_boolean ("shortcuts-no-menukey",
                             NULL,
                             "ShortcutsNoMenukey",
-                            FALSE,
+                            TRUE,
                             G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS);
 
   /**
@@ -1162,7 +1162,7 @@
       g_param_spec_boolean ("shortcuts-no-mnemonics",
                             NULL,
                             "ShortcutsNoMnemonics",
-                            FALSE,
+                            TRUE,
                             G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS);
 
   /**
@@ -1193,7 +1193,7 @@
       g_param_spec_string ("word-chars",
                            NULL,
                            "WordChars",
-                           "-A-Za-z0-9,./?%&#:_=+@~",
+                           "-,./?%&#:_=+@~",
                            G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS);
 
   /**
