diff -Naur a/configure b/configure
--- a/configure	2025-02-22 11:10:35.000000000 +0100
+++ b/configure	2025-03-02 18:07:56.787933082 +0100
@@ -10140,7 +10140,7 @@
 
 
 TERMLIBS=
-for termlib in tinfow tinfo curses termcap terminfo termlib ; do
+for termlib in tinfow tinfo ncurses termcap terminfo termlib ; do
    as_ac_Lib=`printf "%s\n" "ac_cv_lib_${termlib}""_tputs" | sed "$as_sed_sh"`
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for tputs in -l${termlib}" >&5
 printf %s "checking for tputs in -l${termlib}... " >&6; }
diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2025-02-22 11:10:35.000000000 +0100
+++ b/configure.ac	2025-03-02 18:07:26.403830200 +0100
@@ -144,7 +144,7 @@
 
 
 TERMLIBS=
-for termlib in tinfow tinfo curses termcap terminfo termlib ; do
+for termlib in tinfow tinfo ncurses termcap terminfo termlib ; do
    AC_CHECK_LIB(${termlib}, tputs,
      [TERMLIBS="${TERMLIBS} -l${termlib}"; break])
 done
