diff -Naurdp minicom-2.7.orig/src/functions.c minicom-2.7/src/functions.c
--- minicom-2.7.orig/src/functions.c	2009-12-12 16:47:47.000000000 +0100
+++ minicom-2.7/src/functions.c	2014-02-26 15:39:36.305827143 +0100
@@ -62,7 +62,6 @@ void m_hangup(void)
     m_dtrtoggle(portfd, sec);  /* jl 22.06.97 */
   } else {
     m_puts(P_MHANGUP);
-    sleep(1);
   }
   /* If we don't have DCD support fake DCD dropped */
   bogus_dcd = 0;
diff -Naurdp minicom-2.7.orig/src/rwconf.c minicom-2.7/src/rwconf.c
--- minicom-2.7.orig/src/rwconf.c	2013-01-10 22:48:51.000000000 +0100
+++ minicom-2.7/src/rwconf.c	2014-02-26 15:39:58.059105435 +0100
@@ -130,7 +130,7 @@ struct pars mpars[] = {
   { "BUSY",		0,   "mnocon2" },
   { "NO DIALTONE",	0,   "mnocon3" },
   { "VOICE",		0,   "mnocon4" },
-  { "~~+++~~ATH^M",	0,   "mhangup" },
+  { "+++ATH^M",	0,   "mhangup" },
   { "^M",		0,   "mdialcan" },
   { "45",		0,   "mdialtime" },
   { "2",		0,   "mrdelay" },
