diff -Nur orig/electrum.desktop mod/electrum.desktop
--- orig/electrum.desktop	2000-11-11 12:11:11.000000000 +0100
+++ mod/electrum.desktop	2023-12-25 08:06:08.026681682 +0100
@@ -4,12 +4,15 @@
 
 [Desktop Entry]
 Comment=Lightweight Bitcoin Client
+Comment[hu]=Könnyed Bitcoin Kliens
 Exec=electrum %u
 GenericName[en_US]=Bitcoin Wallet
 GenericName=Bitcoin Wallet
+GenericName[hu]=Bitcoin Pénztárca
 Icon=electrum
 Name[en_US]=Electrum Bitcoin Wallet
 Name=Electrum Bitcoin Wallet
+Name[hu]=Electrum Bitcoin Pénztárca
 Categories=Finance;Network;
 StartupNotify=true
 StartupWMClass=electrum
