diff -Naur a/driver/xscreensaver.desktop.in b/driver/xscreensaver.desktop.in
--- a/driver/xscreensaver.desktop.in	2022-05-21 19:31:38.000000000 +0200
+++ b/driver/xscreensaver.desktop.in	2022-05-31 09:57:10.863742966 +0200
@@ -4,5 +4,5 @@
 _Name=XScreenSaver
 _Comment=XScreenSaver daemon: screen saver and locker
 Type=Application
-Categories=Screensaver;Security
+Categories=Security
 Terminal=false
diff -Naur a/driver/xscreensaver-settings.desktop.in b/driver/xscreensaver-settings.desktop.in
--- a/driver/xscreensaver-settings.desktop.in	2022-05-21 19:31:28.000000000 +0200
+++ b/driver/xscreensaver-settings.desktop.in	2022-05-31 09:55:06.555559405 +0200
@@ -4,5 +4,5 @@
 _Name=XScreenSaver Settings
 _Comment=Change screensaver properties
 Type=Application
-Categories=Settings;DesktopSettings;Screensaver;Security
+Categories=Settings;DesktopSettings;Security
 Terminal=false
