diff -Nur orig/Makefile mod/Makefile
--- orig/Makefile	2018-01-15 22:01:10.000000000 +0100
+++ mod/Makefile	2018-01-15 22:04:51.808103497 +0100
@@ -54,7 +54,7 @@
 
 
 # this nice line comes from the linux kernel makefile
-ARCH := $(shell uname -m | sed -e s/i.86/i386/ -e s/sun4u/sparc/ -e s/sparc64/sparc/ -e s/arm.*/arm/ -e s/sa110/arm/ -e s/alpha/axp/)
+#ARCH := $(shell uname -m | sed -e s/i.86/i386/ -e s/sun4u/sparc/ -e s/sparc64/sparc/ -e s/arm.*/arm/ -e s/sa110/arm/ -e s/alpha/axp/)
 
 ifneq ($(ARCH),x86_64)
 ifneq ($(ARCH),i386)
