diff -Naur vacation-1.2.7.1-orig/Makefile vacation-1.2.7.1/Makefile
--- vacation-1.2.7.1-orig/Makefile	2011-11-06 06:15:17.000000000 +0100
+++ vacation-1.2.7.1/Makefile	2011-11-23 17:32:32.000000000 +0100
@@ -45,7 +45,7 @@
 BINDIR		= $(PREFIX)/bin
 VACATION        = $(BINDIR)/vacation
 VACLOOK         = $(BINDIR)/vaclook
-MANDIR		= $(PREFIX)/man/man
+MANDIR		= $(PREFIX)/share/man/man
 MANEXT1		= 1
 
 VERSION 	= 1
@@ -70,10 +70,12 @@
 all:	$(BIN)
 
 install:  all
-	install -s -m 755 $(BIN) $(VACATION)
-	install -m 755 vaclook $(VACLOOK)
-	install -m 444 vacation.man $(MANDIR)$(MANEXT1)/vacation.$(MANEXT1)
-	install -m 444 vaclook.man $(MANDIR)$(MANEXT1)/vaclook.$(MANEXT1)
+
+	install -s -m 755 $(BIN) $(DESTDIR)$(VACATION)
+	install -m 755 vaclook $(DESTDIR)$(VACLOOK)
+	install -m 444 vacation.man $(DESTDIR)$(MANDIR)$(MANEXT1)/vacation.$(MANEXT1)
+	install -m 444 vaclook.man $(DESTDIR)$(MANDIR)$(MANEXT1)/vaclook.$(MANEXT1)
+
 
 vacation:	$(SRC) $(HDR)
 	$(CC) $(CFLAGS) $(IFLAGS) $(PFLAGS) $(LFLAGS) -D_PATH_VACATION=\"$(VACATION)\" -o $(BIN) $(SRC) $(LIBS)
