diff -Nur orig/persistenced/Makefile mod/persistenced/Makefile
--- orig/persistenced/Makefile	2021-06-03 01:08:40.000000000 +0200
+++ mod/persistenced/Makefile	2021-10-14 12:46:14.000000000 +0200
@@ -104,6 +102,17 @@
   LIBS += -ldl
 endif
 
+PKG_CONFIG ?= pkg-config
+
+USE_TIRPC ?= $(shell $(PKG_CONFIG) --exists libtirpc && echo 1)
+
+ifeq ($(USE_TIRPC),1)
+  TIRPC_LDFLAGS ?= $(shell $(PKG_CONFIG) --libs libtirpc)
+  TIRPC_CFLAGS ?= $(shell $(PKG_CONFIG) --cflags libtirpc)
+  $(call BUILD_OBJECT_LIST,$(SRC)): CFLAGS += $(TIRPC_CFLAGS)
+  LIBS += $(TIRPC_LDFLAGS)
+endif
+
 ##############################################################################
 # build rules
