diff -Nur orig/Makefile mod/Makefile
--- orig/Makefile	2007-09-30 20:05:47.000000000 +0200
+++ mod/Makefile	2019-10-14 21:02:29.766080762 +0200
@@ -4,7 +4,7 @@
 
 # if you are not using gmake, you may need to convert the ":=" variables into
 # "=" variables.  you cannot have recursive definitions if you do that.
-CFLAGS   := -fPIC -O2 -Wall -Werror -pedantic $(CFLAGS)
+CFLAGS   := -fPIC -O2 -Wall -pedantic $(CFLAGS)
 CPPFLAGS := -I.
 LDFLAGS  := -x --shared $(LDFLAGS)
 
