diff -Naur a/gnome-initial-setup/meson.build b/gnome-initial-setup/meson.build
--- a/gnome-initial-setup/meson.build	1970-01-01 01:00:00.000000000 +0100
+++ b/gnome-initial-setup/meson.build	2022-07-24 20:25:48.865666948 +0200
@@ -42,7 +42,7 @@
     dependency ('glib-2.0', version: '>= 2.63.1'),
     dependency ('gio-unix-2.0', version: '>= 2.53.0'),
     dependency ('gdm', version: '>= 3.8.3'),
-    dependency ('geocode-glib-1.0'),
+    dependency ('geocode-glib-2.0'),
     dependency ('libgeoclue-2.0', version: '>= 2.3.1'),
     dependency ('libhandy-1', version: '>= 1.5.90', fallback: ['libhandy', 'libhandy_dep']),
     cc.find_library('m', required: false),
