diff -Naur orig/setup.py patched/setup.py
--- orig/setup.py	2018-03-05 04:18:44.000000000 +0100
+++ patched/setup.py	2018-03-12 01:52:31.665219142 +0100
@@ -44,9 +44,9 @@
     data_files.append(('share/icons',
                        ['data/images/misc/pysol01.png',
                         'data/images/misc/pysol02.png', ]))
-    for l in ('ru', 'ru_RU'):
-        data_files.append(('share/locale/%s/LC_MESSAGES' % l,
-                           ['locale/%s/LC_MESSAGES/pysol.mo' % l]))
+#    for l in ('ru', 'ru_RU'):
+#        data_files.append(('share/locale/%s/LC_MESSAGES' % l,
+#                           ['locale/%s/LC_MESSAGES/pysol.mo' % l]))
     data_files.append((data_dir, ['data/pysolfc.glade']))
     data_files.append(('share/applications', ['data/pysol.desktop']))
 
