diff -Naur orig/hw/xwayland/xwayland.c patched/hw/xwayland/xwayland.c
--- orig/hw/xwayland/xwayland.c	2018-08-07 18:31:04.000000000 +0200
+++ patched/hw/xwayland/xwayland.c	2018-08-17 18:30:50.991808979 +0200
@@ -642,10 +642,8 @@
     struct xwl_screen *xwl_screen = xwl_screen_get(screen);
     Bool ret;
 
-#ifdef GLAMOR_HAS_GBM
     if (xwl_screen->present)
         xwl_present_cleanup(window);
-#endif
 
     screen->DestroyWindow = xwl_screen->DestroyWindow;
 
diff -Naur orig/hw/xwayland/xwayland.h patched/hw/xwayland/xwayland.h
--- orig/hw/xwayland/xwayland.h	2018-08-07 18:31:04.000000000 +0200
+++ patched/hw/xwayland/xwayland.h	2018-08-17 18:48:45.798674393 +0200
@@ -185,7 +185,6 @@
     WindowPtr present_window;
 };
 
-#ifdef GLAMOR_HAS_GBM
 struct xwl_present_window {
     struct xwl_screen *xwl_screen;
     WindowPtr window;
@@ -217,7 +216,6 @@
 
     struct xorg_list list;
 };
-#endif
 
 #define MODIFIER_META 0x01
 
@@ -423,7 +421,6 @@
 Bool xwl_shm_destroy_pixmap(PixmapPtr pixmap);
 struct wl_buffer *xwl_shm_pixmap_get_wl_buffer(PixmapPtr pixmap);
 
-#ifdef XWL_HAS_GLAMOR
 void xwl_glamor_init_backends(struct xwl_screen *xwl_screen,
                               Bool use_eglstream);
 void xwl_glamor_select_backend(struct xwl_screen *xwl_screen,
@@ -447,18 +444,14 @@
 Bool xwl_glamor_allow_commits(struct xwl_window *xwl_window);
 void xwl_glamor_egl_make_current(struct xwl_screen *xwl_screen);
 
-#ifdef GLAMOR_HAS_GBM
 Bool xwl_present_init(ScreenPtr screen);
 void xwl_present_cleanup(WindowPtr window);
-#endif /* GLAMOR_HAS_GBM */
 
 #ifdef XV
 /* glamor Xv Adaptor */
 Bool xwl_glamor_xv_init(ScreenPtr pScreen);
 #endif /* XV */
 
-#endif /* XWL_HAS_GLAMOR */
-
 void xwl_screen_release_tablet_manager(struct xwl_screen *xwl_screen);
 
 void xwl_screen_init_xdg_output(struct xwl_screen *xwl_screen);
@@ -467,13 +460,11 @@
 void xwlVidModeExtensionInit(void);
 #endif
 
-#ifdef GLAMOR_HAS_GBM
 void xwl_glamor_init_gbm(struct xwl_screen *xwl_screen);
 #else
 static inline void xwl_glamor_init_gbm(struct xwl_screen *xwl_screen)
 {
 }
-#endif
 
 #ifdef XWL_HAS_EGLSTREAM
 void xwl_glamor_init_eglstream(struct xwl_screen *xwl_screen);
