diff -Naur orig/hw/xwayland/xwayland.c patched/hw/xwayland/xwayland.c
--- orig/hw/xwayland/xwayland.c	2020-03-29 22:21:15.000000000 +0200
+++ patched/hw/xwayland/xwayland.c	2020-08-12 02:36:03.108282535 +0200
@@ -785,10 +785,8 @@
     struct xwl_screen *xwl_screen = xwl_screen_get(screen);
     Bool ret;
 
-#ifdef GLAMOR_HAS_GBM
     if (xwl_screen->present)
         xwl_present_cleanup(window);
-#endif
 
     screen->DestroyWindow = xwl_screen->DestroyWindow;
 
diff -Naur orig/hw/xwayland/xwayland.h patched/hw/xwayland/xwayland.h
--- orig/hw/xwayland/xwayland.h	2020-03-29 22:21:15.000000000 +0200
+++ patched/hw/xwayland/xwayland.h	2020-08-12 02:36:03.108282535 +0200
@@ -219,7 +219,6 @@
 
     struct xorg_list list;
 };
-#endif
 
 #define MODIFIER_META 0x01
 
@@ -428,7 +427,6 @@
 Bool xwl_shm_destroy_pixmap(PixmapPtr pixmap);
 struct wl_buffer *xwl_shm_pixmap_get_wl_buffer(PixmapPtr pixmap);
 
-#ifdef XWL_HAS_GLAMOR
 void xwl_glamor_init_backends(struct xwl_screen *xwl_screen,
                               Bool use_eglstream);
 void xwl_glamor_select_backend(struct xwl_screen *xwl_screen,
@@ -474,13 +472,11 @@
 void xwlVidModeExtensionInit(void);
 #endif
 
-#ifdef GLAMOR_HAS_GBM
 void xwl_glamor_init_gbm(struct xwl_screen *xwl_screen);
 #else
 static inline void xwl_glamor_init_gbm(struct xwl_screen *xwl_screen)
 {
 }
-#endif
 
 #ifdef XWL_HAS_EGLSTREAM
 void xwl_glamor_init_eglstream(struct xwl_screen *xwl_screen);
