diff -Naur orig/hw/xwayland/xwayland.c patched/hw/xwayland/xwayland.c
--- orig/hw/xwayland/xwayland.c	2019-02-26 20:28:50.000000000 +0100
+++ patched/hw/xwayland/xwayland.c	2019-05-26 01:32:54.596073712 +0200
@@ -659,10 +659,8 @@
     struct xwl_screen *xwl_screen = xwl_screen_get(screen);
     Bool ret;
 
-#ifdef GLAMOR_HAS_GBM
     if (xwl_screen->present)
         xwl_present_cleanup(window);
-#endif
 
     screen->DestroyWindow = xwl_screen->DestroyWindow;
 
diff -Naur orig/hw/xwayland/xwayland.h patched/hw/xwayland/xwayland.h
--- orig/hw/xwayland/xwayland.h	2019-02-26 20:28:50.000000000 +0100
+++ patched/hw/xwayland/xwayland.h	2019-05-26 01:35:21.393774701 +0200
@@ -182,12 +182,9 @@
     struct xorg_list link_damage;
     struct wl_callback *frame_callback;
     Bool allow_commits;
-#ifdef GLAMOR_HAS_GBM
     Bool present_flipped;
-#endif
 };
 
-#ifdef GLAMOR_HAS_GBM
 struct xwl_present_window {
     struct xwl_screen *xwl_screen;
     struct xwl_present_event *sync_flip;
@@ -220,7 +217,6 @@
 
     struct xorg_list list;
 };
-#endif
 
 #define MODIFIER_META 0x01
 
@@ -427,7 +423,6 @@
 Bool xwl_shm_destroy_pixmap(PixmapPtr pixmap);
 struct wl_buffer *xwl_shm_pixmap_get_wl_buffer(PixmapPtr pixmap);
 
-#ifdef XWL_HAS_GLAMOR
 void xwl_glamor_init_backends(struct xwl_screen *xwl_screen,
                               Bool use_eglstream);
 void xwl_glamor_select_backend(struct xwl_screen *xwl_screen,
@@ -451,19 +446,15 @@
 Bool xwl_glamor_allow_commits(struct xwl_window *xwl_window);
 void xwl_glamor_egl_make_current(struct xwl_screen *xwl_screen);
 
-#ifdef GLAMOR_HAS_GBM
 Bool xwl_present_init(ScreenPtr screen);
 void xwl_present_cleanup(WindowPtr window);
 void xwl_present_unrealize_window(WindowPtr window);
-#endif /* GLAMOR_HAS_GBM */
 
 #ifdef XV
 /* glamor Xv Adaptor */
 Bool xwl_glamor_xv_init(ScreenPtr pScreen);
 #endif /* XV */
 
-#endif /* XWL_HAS_GLAMOR */
-
 void xwl_screen_release_tablet_manager(struct xwl_screen *xwl_screen);
 
 void xwl_screen_init_xdg_output(struct xwl_screen *xwl_screen);
@@ -472,13 +463,11 @@
 void xwlVidModeExtensionInit(void);
 #endif
 
-#ifdef GLAMOR_HAS_GBM
 void xwl_glamor_init_gbm(struct xwl_screen *xwl_screen);
 #else
 static inline void xwl_glamor_init_gbm(struct xwl_screen *xwl_screen)
 {
 }
-#endif
 
 #ifdef XWL_HAS_EGLSTREAM
 void xwl_glamor_init_eglstream(struct xwl_screen *xwl_screen);
