diff -Nur orig/src/gst/parole-gst.c mod/src/gst/parole-gst.c
--- orig/src/gst/parole-gst.c	2018-03-01 04:00:32.000000000 +0100
+++ mod/src/gst/parole-gst.c	2018-10-20 13:45:52.846098725 +0200
@@ -1295,34 +1295,18 @@
     dialog = GTK_MESSAGE_DIALOG(gtk_message_dialog_new_with_markup(
                             NULL,
                             GTK_DIALOG_MODAL | GTK_DIALOG_DESTROY_WITH_PARENT,
-#if defined(__linux__)
-                            GTK_MESSAGE_QUESTION,
-#else
                             GTK_MESSAGE_WARNING,
-#endif
                             GTK_BUTTONS_NONE,
                             "<b><big>%s</big></b>",
                             _("Additional software is required.")));
 
     gtk_dialog_add_buttons(GTK_DIALOG(dialog),
-#if defined(__linux__)
-                            _("Don't Install"),
-                            GTK_RESPONSE_REJECT,
-                            _("Install"),
-                            GTK_RESPONSE_ACCEPT,
-#else
                             _("OK"),
                             GTK_RESPONSE_ACCEPT,
-#endif
                             NULL);
 
     gtk_message_dialog_format_secondary_markup(dialog,
-#if defined(__linux__)
-                                             _("Parole needs <b>%s</b> to play this file.\n"
-                                               "It can be installed automatically."),
-#else
                                              _("Parole needs <b>%s</b> to play this file."),
-#endif
                                              desc);
 
     return GTK_DIALOG(dialog);
@@ -1414,7 +1398,6 @@
     case GST_MESSAGE_ELEMENT:
         if (gst_is_missing_plugin_message(msg)) {
             g_debug("Missing plugin\n");
-            parole_gst_stop(gst);
             dialog = parole_gst_missing_codec_dialog(gst, msg);
             response = gtk_dialog_run(dialog);
             if ( response == GTK_RESPONSE_ACCEPT ) {
