diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2019-08-01 23:24:25.000000000 +0200
+++ patched/CMakeLists.txt	2019-08-24 11:37:26.393638061 +0200
@@ -1493,16 +1493,6 @@
 endif()
 
 
-# Create the uninstall target.
-# https://gitlab.kitware.com/cmake/community/wikis/FAQ#can-i-do-make-uninstall-with-cmake
-configure_file(${PROJECT_SOURCE_DIR}/cmake/Uninstall.cmake.in
-               ${PROJECT_BINARY_DIR}/Uninstall.cmake
-               @ONLY)
-
-add_custom_target(uninstall
-                  COMMAND ${CMAKE_COMMAND} -P ${PROJECT_BINARY_DIR}/Uninstall.cmake)
-
-
 message(STATUS "")
 message(STATUS "        ---( Libevent " ${EVENT_VERSION} " )---")
 message(STATUS "")
