diff -Naur orig/meson.build patched/meson.build
--- orig/meson.build	1970-01-01 01:00:00.000000000 +0100
+++ patched/meson.build	2020-09-19 23:18:08.135227646 +0200
@@ -76,7 +76,7 @@
 glib_requirement = '>= 2.61.2'
 gtk_requirement = '>= 3.24.0'
 nettle_requirement = '>= 3.4'
-webkitgtk_requirement = '>= 2.29.3'
+webkitgtk_requirement = '>= 2.28.0'
 
 cairo_dep = dependency('cairo', version: '>= 1.2')
 gcr_dep = dependency('gcr-3', version: '>= 3.5.5')
