diff -Naur orig/Makefile.am patched/Makefile.am
--- orig/Makefile.am	2018-05-15 15:26:38.000000000 +0200
+++ patched/Makefile.am	2018-05-26 20:27:20.111508207 +0200
@@ -1,4 +1,4 @@
-SUBDIRS = libdca libao src test include
+SUBDIRS = libdca libao test include
 DIST_SUBDIRS = $(SUBDIRS) vc++
 
 EXTRA_DIST = bootstrap doc/libdca.txt \
diff -Naur orig/Makefile.in patched/Makefile.in
--- orig/Makefile.in	2018-05-15 15:59:49.000000000 +0200
+++ patched/Makefile.in	2018-05-26 20:27:57.780017349 +0200
@@ -333,7 +333,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = libdca libao src test include
+SUBDIRS = libdca libao test include
 DIST_SUBDIRS = $(SUBDIRS) vc++
 EXTRA_DIST = bootstrap doc/libdca.txt \
 	m4/cflags.m4 m4/inttypes.m4 m4/keywords.m4 m4/nonpic.m4
