diff -Naur orig/libdca/Makefile.am patched/libdca/Makefile.am
--- orig/libdca/Makefile.am	2018-05-15 15:46:09.000000000 +0200
+++ patched/libdca/Makefile.am	2018-05-26 20:32:05.075359888 +0200
@@ -12,7 +12,5 @@
 pkgconfig_DATA = libdca.pc libdts.pc
 
 install-exec-hook:
-	ln -sf "libdca.a" "$(DESTDIR)$(libdir)/libdts.a"
 
 uninstall-hook:
-	unlink "$(DESTDIR)$(libdir)/libdts.a"
diff -Naur orig/libdca/Makefile.in patched/libdca/Makefile.in
--- orig/libdca/Makefile.in	2018-05-15 15:59:49.000000000 +0200
+++ patched/libdca/Makefile.in	2018-05-26 20:31:01.238497045 +0200
@@ -689,10 +689,8 @@
 
 
 install-exec-hook:
-	ln -sf "libdca.a" "$(DESTDIR)$(libdir)/libdts.a"
 
 uninstall-hook:
-	unlink "$(DESTDIR)$(libdir)/libdts.a"
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
