diff -Naur a/epson-printer-utility-1.1.1/configure.ac b/epson-printer-utility-1.1.1/configure.ac
--- a/epson-printer-utility-1.1.1/configure.ac	2019-11-14 09:07:26.000000000 +0100
+++ b/epson-printer-utility-1.1.1/configure.ac	2022-09-03 18:17:35.137031048 +0200
@@ -3,7 +3,7 @@
 
 AC_PREREQ()
 AC_INIT(PrinterUtility, 1.1.1, epson@localdomain)
-AM_INIT_AUTOMAKE
+AM_INIT_AUTOMAKE([gnu nostdinc subdir-objects 1.7.6 --warnings=all])
 
 
 AC_CONFIG_SRCDIR(PrinterUtility/Main.cpp)
@@ -15,9 +15,9 @@
 PACKAGE_NAME=$EPS_PREFIX-utility
 AC_SUBST(PACKAGE_NAME)
 
-#CC=/opt/lsb/bin/lsbcc
-CC=/opt/lsb/bin/lsbc++
-CXX=/opt/lsb/bin/lsbc++
+AC_PROG_CC
+AC_PROG_CXX
+AM_PROG_AR
 
 # EPS_INSTALL_DIR=/opt/$PACKAGE_NAME
 # AC_SUBST(EPS_INSTALL_DIR)
@@ -46,7 +46,7 @@
 LIBARCH=lib
 BUILD_MACHINE_ARCH=`uname -m`
 if test "x$BUILD_MACHINE_ARCH" = "xx86_64"; then
-	LIBARCH=lib64
+	LIBARCH=lib
 fi
 AC_SUBST(LIBARCH)
 
@@ -73,8 +73,6 @@
 # Save this value here, since automake will set cflags later
 cflags_set=${CFLAGS+set}
 
-AM_INIT_AUTOMAKE([gnu nostdinc subdir-objects 1.7.6 --warnings=all])
-
 LIBUSB_VERSION=$VERSION
 
 #
diff -Naur a/epson-printer-utility-1.1.1/EPSCommonLib/Makefile.am b/epson-printer-utility-1.1.1/EPSCommonLib/Makefile.am
--- a/epson-printer-utility-1.1.1/EPSCommonLib/Makefile.am	2017-04-24 07:24:26.000000000 +0200
+++ b/epson-printer-utility-1.1.1/EPSCommonLib/Makefile.am	2022-09-03 18:18:44.409159083 +0200
@@ -1,3 +1,5 @@
+CC=g++
+
 noinst_LIBRARIES = libEPSCommonLib.a
 
 
@@ -16,28 +18,32 @@
   		-DGCOMSW_EL_CDLABEL	\
   		-DGCOMSW_EF_MAINTE
 
-AM_CFLAGS=-w -c \
+AM_CFLAGS=-w -c -fPIC \
 	-I./escpr-lib/inc \
 	-I./epsmp/linux \
 	-I./epsmp	\
 	-I./..
-AM_CXXFLAGS=-w -c
+AM_CXXFLAGS=-w -c -fPIC
 
 AM_CPPFLAGS = \
 	-I./escpr-lib/inc \
 	-I./epsmp/linux \
 	-I./epsmp	\
 	-I./..	\
-	-I/opt/lsb/include/QtCore	\
-	-I/opt/lsb/include
+	-I/usr/include/qt5 \
+	-I/usr/include/qt5/QtCore \
+	-I/usr/include/qt5/QtGui \
+	-I/usr/include/qt5/QtNetwork \
+	-I/usr/include/qt5/QtWidgets
 
 libEPSCommonLib_a_LIBADD = \
-	/opt/lsb/$(LIBARCH)/libQtCore.so \
-	/opt/lsb/$(LIBARCH)/libQtGui.so \
-	/opt/lsb/$(LIBARCH)/libpthread.so \
-	/opt/lsb/$(LIBARCH)/libcupsimage.so \
-	/opt/lsb/$(LIBARCH)/libcups.so \
-   /opt/lsb/$(LIBARCH)/libQtNetwork.so
+	/usr/$(LIBARCH)/libQt5Core.so \
+	/usr/$(LIBARCH)/libQt5Gui.so \
+	/usr/$(LIBARCH)/libQt5Network.so \
+	/usr/$(LIBARCH)/libQt5Widgets.so \
+	/usr/$(LIBARCH)/libpthread.so.0 \
+	/usr/$(LIBARCH)/libcupsimage.so \
+	/usr/$(LIBARCH)/libcups.so
 	
 libEPSCommonLib_a_SOURCES = \
 	escpr-lib/src/epson-cbt.c	\
@@ -58,7 +64,7 @@
 	escpr-lib/src/epson-usb.c	\
 	escpr-lib/src/epson-wifi.c	\
 	epsmp/linux/epsmp-linux-portal-io.c \
-	epsmp/linux/XAsyncSocket.c \
+	epsmp/linux/XAsyncSocket.cpp \
 	epsmp/linux/linux_cmn.c	\
 	epsmp/linux/linux_net.c	\
 	epsmp/epsmp-scale.c	\
diff -Naur a/epson-printer-utility-1.1.1/PrinterUtility/Makefile.am b/epson-printer-utility-1.1.1/PrinterUtility/Makefile.am
--- a/epson-printer-utility-1.1.1/PrinterUtility/Makefile.am	2017-04-24 07:24:26.000000000 +0200
+++ b/epson-printer-utility-1.1.1/PrinterUtility/Makefile.am	2022-09-03 18:19:52.617293414 +0200
@@ -3,8 +3,8 @@
 bindir=$(EPS_BIN_DIR)
 bin_PROGRAMS =$(EPS_PREFIX)-utility
 
-AM_CFLAGS=-g -w -c
-AM_CXXFLAGS=-g -w 
+AM_CFLAGS=-g -w -c -fPIC
+AM_CXXFLAGS=-g -w -fPIC
 
 if ARCH_CON
     DEFS+=-DX86_64
@@ -14,14 +14,13 @@
 	-DQT_NO_DEBUG_OUTPUT
 
 __EPS_PREFIX__utility_CPPFLAGS = \
-	-I/opt/lsb/include/QtCore \
-	-I/opt/lsb/include/QtCore \
-	-I/opt/lsb/include/QtGui \
-	-I/opt/lsb/include/QtGui \
-   -I/opt/lsb/include/QtNetwork \
+	-I/usr/include/qt5 \
+	-I/usr/include/qt5/QtCore \
+    -I/usr/include/qt5/QtGui \
+    -I/usr/include/qt5/QtNetwork \
+    -I/usr/include/qt5/QtWidgets \
 	-I/../EPUCommon	\
 	-I/..	\
-	-I/opt/lsb/include \
 	-I../EPSCommonLib/	\
 	-I../EPSCommonLib/escpr-lib/inc \
 	-I../EPSCommonLib/epsmp/linux \
@@ -53,10 +52,11 @@
 	qrc_Resources.cpp
 
 __EPS_PREFIX__utility_LDADD = \
-	/opt/lsb/$(LIBARCH)/libQtCore.so \
-	/opt/lsb/$(LIBARCH)/libQtGui.so \
-	/opt/lsb/$(LIBARCH)/libpthread.so \
-	/opt/lsb/$(LIBARCH)/libcupsimage.so \
-	/opt/lsb/$(LIBARCH)/libcups.so \
-   /opt/lsb/$(LIBARCH)/libQtNetwork.so \
+	/usr/$(LIBARCH)/libQt5Core.so \
+	/usr/$(LIBARCH)/libQt5Gui.so \
+	/usr/$(LIBARCH)/libQt5Network.so \
+	/usr/$(LIBARCH)/libQt5Widgets.so \
+	/usr/$(LIBARCH)/libpthread.so.0 \
+	/usr/$(LIBARCH)/libcupsimage.so \
+	/usr/$(LIBARCH)/libcups.so \
 	../EPSCommonLib/libEPSCommonLib.a
