diff -Naur a/epson-printer-utility-1.1.1/EPSCommonLib/epsmp/epsmp-image.c b/epson-printer-utility-1.1.1/EPSCommonLib/epsmp/epsmp-image.c
--- a/epson-printer-utility-1.1.1/EPSCommonLib/epsmp/epsmp-image.c	2017-04-24 07:24:26.000000000 +0200
+++ b/epson-printer-utility-1.1.1/EPSCommonLib/epsmp/epsmp-image.c	2022-09-03 18:10:17.356513099 +0200
@@ -765,7 +765,7 @@
 
 EPS_INT32 epsmpIsImageFile (const char* fileName)
 {
-	char* pExt = strrchr(fileName, '.');
+	const char* pExt = strrchr(fileName, '.');
 	if(pExt){
 		if( 0 == strcmp(pExt, ".JPG")	||
 			0 == strcmp(pExt, ".JPEG")	||
