diff -Naur orig/epson-printer-utility-1.0.2/PrinterUtility/EPUMainController.cpp patched/epson-printer-utility-1.0.2/PrinterUtility/EPUMainController.cpp
--- orig/epson-printer-utility-1.0.2/PrinterUtility/EPUMainController.cpp	2016-07-22 08:19:11.000000000 +0200
+++ patched/epson-printer-utility-1.0.2/PrinterUtility/EPUMainController.cpp	2021-11-30 00:05:06.865300115 +0100
@@ -1,7 +1,8 @@
 #include "EPUMainController.h"
 #include "EPUPrinterController.h"
+#include <QApplication>
 #include <QTimer>
-//#include "err.h"
+#include "unistd.h"
 
 EPUMainController::EPUMainController()
 {
diff -Naur orig/epson-printer-utility-1.0.2/PrinterUtility/EPUMaintenanceGuideView.h patched/epson-printer-utility-1.0.2/PrinterUtility/EPUMaintenanceGuideView.h
--- orig/epson-printer-utility-1.0.2/PrinterUtility/EPUMaintenanceGuideView.h	2015-04-21 10:25:50.000000000 +0200
+++ patched/epson-printer-utility-1.0.2/PrinterUtility/EPUMaintenanceGuideView.h	2021-11-30 00:06:25.157595850 +0100
@@ -3,6 +3,7 @@
 
 #include "EPUOperationView.h"
 #include "EPUPrinterController.h"
+#include <QPushButton>
 
 enum
 {
diff -Naur orig/epson-printer-utility-1.0.2/PrinterUtility/EPUMainView.h patched/epson-printer-utility-1.0.2/PrinterUtility/EPUMainView.h
--- orig/epson-printer-utility-1.0.2/PrinterUtility/EPUMainView.h	2017-02-15 08:52:02.000000000 +0100
+++ patched/epson-printer-utility-1.0.2/PrinterUtility/EPUMainView.h	2021-11-30 00:07:29.865839738 +0100
@@ -3,6 +3,8 @@
 
 #include "EPUView.h"
 #include "EPUMainWindow.h"
+#include <QGroupBox>
+#include <QListView>
 #include <QTimer>
 #include "EPUPrinterListDialog.h"
 #include "EPUInkLevelView.h"
diff -Naur orig/epson-printer-utility-1.0.2/PrinterUtility/EPUMainWindow.cpp patched/epson-printer-utility-1.0.2/PrinterUtility/EPUMainWindow.cpp
--- orig/epson-printer-utility-1.0.2/PrinterUtility/EPUMainWindow.cpp	2016-07-22 07:24:53.000000000 +0200
+++ patched/epson-printer-utility-1.0.2/PrinterUtility/EPUMainWindow.cpp	2021-11-30 00:08:22.898039264 +0100
@@ -1,4 +1,6 @@
 #include "EPUMainWindow.h"
+#include <QApplication>
+#include <QDesktopWidget>
 #include <QFont>
 #include "EPUGlobal.h"
 
diff -Naur orig/epson-printer-utility-1.0.2/PrinterUtility/EPUPrinterListDialog.cpp patched/epson-printer-utility-1.0.2/PrinterUtility/EPUPrinterListDialog.cpp
--- orig/epson-printer-utility-1.0.2/PrinterUtility/EPUPrinterListDialog.cpp	2016-07-22 07:46:17.000000000 +0200
+++ patched/epson-printer-utility-1.0.2/PrinterUtility/EPUPrinterListDialog.cpp	2021-11-30 00:09:48.742361572 +0100
@@ -1,6 +1,8 @@
 #include "EPUPrinterListDialog.h"
 #include "EPUPrinterController.h"
 #include "EPUPrinter.h"
+#include <QApplication>
+#include <unistd.h>
 
 EPUPrinterListDialog::EPUPrinterListDialog(QWidget *parent)
     : QDialog(parent)
diff -Naur orig/epson-printer-utility-1.0.2/PrinterUtility/EPUPrinterListDialog.h patched/epson-printer-utility-1.0.2/PrinterUtility/EPUPrinterListDialog.h
--- orig/epson-printer-utility-1.0.2/PrinterUtility/EPUPrinterListDialog.h	2015-04-21 10:25:50.000000000 +0200
+++ patched/epson-printer-utility-1.0.2/PrinterUtility/EPUPrinterListDialog.h	2021-11-30 00:10:50.370592458 +0100
@@ -2,6 +2,10 @@
 #define EPUPRINTERLISTDIALOG_H
 
 #include <QDialog>
+#include <QHBoxLayout>
+#include <QListView>
+#include <QPushButton>
+#include <QVBoxLayout>
 #include "../EPUCommon/EPUCommonDef.h"
 #include "EPUPrinter.h"
 #include "../EPUCommon/EThread.h"
diff -Naur orig/epson-printer-utility-1.0.2/PrinterUtility/EPUView.h patched/epson-printer-utility-1.0.2/PrinterUtility/EPUView.h
--- orig/epson-printer-utility-1.0.2/PrinterUtility/EPUView.h	2015-04-21 10:25:50.000000000 +0200
+++ patched/epson-printer-utility-1.0.2/PrinterUtility/EPUView.h	2021-11-30 00:11:58.758848191 +0100
@@ -1,6 +1,9 @@
 #ifndef EPUVIEW_H
 #define EPUVIEW_H
 
+#include <QHBoxLayout>
+#include <QLabel>
+#include <QVBoxLayout>
 #include <QWidget>
 #include "../EPUCommon/EPUCommonDef.h"
 #include "EPUCallbackObject.h"
