diff -Naur a/epson-printer-utility-1.1.1/PrinterUtility/EPUMainController.cpp b/epson-printer-utility-1.1.1/PrinterUtility/EPUMainController.cpp
--- a/epson-printer-utility-1.1.1/PrinterUtility/EPUMainController.cpp	2017-04-24 07:24:26.000000000 +0200
+++ b/epson-printer-utility-1.1.1/PrinterUtility/EPUMainController.cpp	2022-09-03 18:28:41.878296889 +0200
@@ -1,7 +1,8 @@
 #include "EPUMainController.h"
 #include "EPUPrinterController.h"
+#include <QApplication>
 #include <QTimer>
-//#include "err.h"
+#include "unistd.h"
 
 EPUMainController::EPUMainController()
 {
diff -Naur a/epson-printer-utility-1.1.1/PrinterUtility/EPUMaintenanceGuideView.h b/epson-printer-utility-1.1.1/PrinterUtility/EPUMaintenanceGuideView.h
--- a/epson-printer-utility-1.1.1/PrinterUtility/EPUMaintenanceGuideView.h	2017-04-24 07:24:26.000000000 +0200
+++ b/epson-printer-utility-1.1.1/PrinterUtility/EPUMaintenanceGuideView.h	2022-09-03 18:29:20.882363318 +0200
@@ -3,6 +3,7 @@
 
 #include "EPUOperationView.h"
 #include "EPUPrinterController.h"
+#include <QPushButton>
 
 enum
 {
diff -Naur a/epson-printer-utility-1.1.1/PrinterUtility/EPUMainView.h b/epson-printer-utility-1.1.1/PrinterUtility/EPUMainView.h
--- a/epson-printer-utility-1.1.1/PrinterUtility/EPUMainView.h	2017-04-24 07:24:26.000000000 +0200
+++ b/epson-printer-utility-1.1.1/PrinterUtility/EPUMainView.h	2022-09-03 18:30:07.974447006 +0200
@@ -3,6 +3,8 @@
 
 #include "EPUView.h"
 #include "EPUMainWindow.h"
+#include <QGroupBox>
+#include <QListView>
 #include <QTimer>
 #include "EPUPrinterListDialog.h"
 #include "EPUInkLevelView.h"
diff -Naur a/epson-printer-utility-1.1.1/PrinterUtility/EPUMainWindow.cpp b/epson-printer-utility-1.1.1/PrinterUtility/EPUMainWindow.cpp
--- a/epson-printer-utility-1.1.1/PrinterUtility/EPUMainWindow.cpp	2017-04-24 07:24:26.000000000 +0200
+++ b/epson-printer-utility-1.1.1/PrinterUtility/EPUMainWindow.cpp	2022-09-03 18:30:53.754531764 +0200
@@ -1,4 +1,6 @@
 #include "EPUMainWindow.h"
+#include <QApplication>
+#include <QDesktopWidget>
 #include <QFont>
 #include "EPUGlobal.h"
 
diff -Naur a/epson-printer-utility-1.1.1/PrinterUtility/EPUPrinterListDialog.cpp b/epson-printer-utility-1.1.1/PrinterUtility/EPUPrinterListDialog.cpp
--- a/epson-printer-utility-1.1.1/PrinterUtility/EPUPrinterListDialog.cpp	2017-04-24 07:24:26.000000000 +0200
+++ b/epson-printer-utility-1.1.1/PrinterUtility/EPUPrinterListDialog.cpp	2022-09-03 18:31:34.250609254 +0200
@@ -1,6 +1,8 @@
 #include "EPUPrinterListDialog.h"
 #include "EPUPrinterController.h"
 #include "EPUPrinter.h"
+#include <QApplication>
+#include <unistd.h>
 
 EPUPrinterListDialog::EPUPrinterListDialog(QWidget *parent)
     : QDialog(parent)
diff -Naur a/epson-printer-utility-1.1.1/PrinterUtility/EPUPrinterListDialog.h b/epson-printer-utility-1.1.1/PrinterUtility/EPUPrinterListDialog.h
--- a/epson-printer-utility-1.1.1/PrinterUtility/EPUPrinterListDialog.h	2017-04-24 07:24:26.000000000 +0200
+++ b/epson-printer-utility-1.1.1/PrinterUtility/EPUPrinterListDialog.h	2022-09-03 18:32:25.438710376 +0200
@@ -2,6 +2,10 @@
 #define EPUPRINTERLISTDIALOG_H
 
 #include <QDialog>
+#include <QHBoxLayout>
+#include <QListView>
+#include <QPushButton>
+#include <QVBoxLayout>
 #include "../EPUCommon/EPUCommonDef.h"
 #include "EPUPrinter.h"
 #include "../EPUCommon/EThread.h"
diff -Naur a/epson-printer-utility-1.1.1/PrinterUtility/EPUView.h b/epson-printer-utility-1.1.1/PrinterUtility/EPUView.h
--- a/epson-printer-utility-1.1.1/PrinterUtility/EPUView.h	2017-04-24 07:24:26.000000000 +0200
+++ b/epson-printer-utility-1.1.1/PrinterUtility/EPUView.h	2022-09-03 18:25:08.353990296 +0200
@@ -1,6 +1,9 @@
 #ifndef EPUVIEW_H
 #define EPUVIEW_H
 
+#include <QHBoxLayout>
+#include <QLabel>
+#include <QVBoxLayout>
 #include <QWidget>
 #include "../EPUCommon/EPUCommonDef.h"
 #include "EPUCallbackObject.h"
