diff -Naur orig/epson-printer-utility-1.0.2/EPSCommonLib/Makefile.am patched/epson-printer-utility-1.0.2/EPSCommonLib/Makefile.am
--- orig/epson-printer-utility-1.0.2/EPSCommonLib/Makefile.am	2015-04-21 10:25:50.000000000 +0200
+++ patched/epson-printer-utility-1.0.2/EPSCommonLib/Makefile.am	2019-09-29 00:22:57.436747515 +0200
@@ -1,3 +1,5 @@
+export CC=g++
+
 noinst_LIBRARIES = libEPSCommonLib.a
 
 
@@ -16,28 +18,33 @@
   		-DGCOMSW_EL_CDLABEL	\
   		-DGCOMSW_EF_MAINTE
 
-AM_CFLAGS=-w -c \
+AM_CFLAGS=-w -c -fPIC \
 	-I./escpr-lib/inc \
 	-I./epsmp/linux \
 	-I./epsmp	\
 	-I./..
-AM_CXXFLAGS=-w -c
+AM_CXXFLAGS=-w -c -fPIC
 
 AM_CPPFLAGS = \
 	-I./escpr-lib/inc \
 	-I./epsmp/linux \
 	-I./epsmp	\
 	-I./..	\
-	-I/opt/lsb/include/QtCore	\
-	-I/opt/lsb/include
+	-I/usr/include/qt5/QtCore \
+	-I/usr/include/qt5/QtGui \
+	-I/usr/include/qt5/QtWidgets \
+	-I/usr/include/qt5/QtNetwork \
+	-I/usr/include/qt5	\
+	-I/usr/include
 
 libEPSCommonLib_a_LIBADD = \
-	/opt/lsb/$(LIBARCH)/libQtCore.so \
-	/opt/lsb/$(LIBARCH)/libQtGui.so \
-	/opt/lsb/$(LIBARCH)/libpthread.so \
-	/opt/lsb/$(LIBARCH)/libcupsimage.so \
-	/opt/lsb/$(LIBARCH)/libcups.so \
-   /opt/lsb/$(LIBARCH)/libQtNetwork.so
+	/usr/$(LIBARCH)/libQt5Core.so \
+	/usr/$(LIBARCH)/libQt5Gui.so \
+	/usr/$(LIBARCH)/libQt5Widgets.so \
+	/usr/$(LIBARCH)/libQt5Network.so \
+	/usr/$(LIBARCH)/libpthread.so \
+	/usr/$(LIBARCH)/libcupsimage.so \
+	/usr/$(LIBARCH)/libcups.so
 	
 libEPSCommonLib_a_SOURCES = \
 	escpr-lib/src/epson-cbt.c	\
@@ -58,7 +65,7 @@
 	escpr-lib/src/epson-usb.c	\
 	escpr-lib/src/epson-wifi.c	\
 	epsmp/linux/epsmp-linux-portal-io.c \
-	epsmp/linux/XAsyncSocket.c \
+	epsmp/linux/XAsyncSocket.cpp \
 	epsmp/linux/linux_cmn.c	\
 	epsmp/linux/linux_net.c	\
 	epsmp/epsmp-scale.c	\
diff -Naur orig/epson-printer-utility-1.0.2/PrinterUtility/Makefile.am patched/epson-printer-utility-1.0.2/PrinterUtility/Makefile.am
--- orig/epson-printer-utility-1.0.2/PrinterUtility/Makefile.am	2016-07-21 07:38:30.000000000 +0200
+++ patched/epson-printer-utility-1.0.2/PrinterUtility/Makefile.am	2019-09-29 00:23:47.329468348 +0200
@@ -3,8 +3,8 @@
 bindir=$(EPS_BIN_DIR)
 bin_PROGRAMS =$(EPS_PREFIX)-utility
 
-AM_CFLAGS=-g -w -c
-AM_CXXFLAGS=-g -w 
+AM_CFLAGS=-g -w -c -fPIC
+AM_CXXFLAGS=-g -w -fPIC
 
 if ARCH_CON
     DEFS+=-DX86_64
@@ -14,20 +14,19 @@
 	-DQT_NO_DEBUG_OUTPUT
 
 __EPS_PREFIX__utility_CPPFLAGS = \
-	-I/opt/lsb/include/QtCore \
-	-I/opt/lsb/include/QtCore \
-	-I/opt/lsb/include/QtGui \
-	-I/opt/lsb/include/QtGui \
-   -I/opt/lsb/include/QtNetwork \
+	-I/usr/include/qt5/QtCore \
+	-I/usr/include/qt5/QtGui \
+	-I/usr/include/qt5/QtWidgets \
+    -I/usr/include/qt5/QtNetwork \
+    -I/usr/include/qt5 \
 	-I/../EPUCommon	\
 	-I/..	\
-	-I/opt/lsb/include \
+	-I/usr/include \
 	-I../EPSCommonLib/	\
 	-I../EPSCommonLib/escpr-lib/inc \
 	-I../EPSCommonLib/epsmp/linux \
 	-I../EPSCommonLib/epsmp
 
-
 __EPS_PREFIX__utility_SOURCES = \
 	Main.cpp \
 	EPUMainWindow.cpp \
@@ -53,10 +52,11 @@
 	qrc_Resources.cpp
 
 __EPS_PREFIX__utility_LDADD = \
-	/opt/lsb/$(LIBARCH)/libQtCore.so \
-	/opt/lsb/$(LIBARCH)/libQtGui.so \
-	/opt/lsb/$(LIBARCH)/libpthread.so \
-	/opt/lsb/$(LIBARCH)/libcupsimage.so \
-	/opt/lsb/$(LIBARCH)/libcups.so \
-   /opt/lsb/$(LIBARCH)/libQtNetwork.so \
+	/usr/$(LIBARCH)/libQt5Core.so \
+	/usr/$(LIBARCH)/libQt5Gui.so \
+	/usr/$(LIBARCH)/libQt5Widgets.so \
+	/usr/$(LIBARCH)/libQt5Network.so \
+	/usr/$(LIBARCH)/libpthread.so \
+	/usr/$(LIBARCH)/libcupsimage.so \
+	/usr/$(LIBARCH)/libcups.so \
 	../EPSCommonLib/libEPSCommonLib.a
