diff -Naur a/epson-backend-1.1.1/src/Makefile.am b/epson-backend-1.1.1/src/Makefile.am
--- a/epson-backend-1.1.1/src/Makefile.am	2017-04-24 07:24:26.000000000 +0200
+++ b/epson-backend-1.1.1/src/Makefile.am	2022-09-03 18:39:18.195619242 +0200
@@ -19,21 +19,16 @@
 
 cupsbackend_PROGRAMS =ecblp
 
-noinst_LTLIBRARIES = libcutils.la
-
 MODEL_FULL = LITE
 MODEL = lite
 
-libcutils_la_SOURCES = \
+ecblp_LDFLAGS = \
+	@CUPS_LIBS@
+ecblp_SOURCES = \
+	backend.c \
 	gettext.h \
 	str.c str.h \
 	err.c err.h \
 	mem.c mem.h \
 	def.h
 
-
-ecblp_LDFLAGS = \
-	@CUPS_LIBS@ \
-	libcutils.la
-ecblp_SOURCES = \
-	backend.c
