diff -Naur a/epson-backend-1.1.1/daemon/cbtd_comserv.c b/epson-backend-1.1.1/daemon/cbtd_comserv.c
--- a/epson-backend-1.1.1/daemon/cbtd_comserv.c	2020-01-20 01:52:50.000000000 +0100
+++ b/epson-backend-1.1.1/daemon/cbtd_comserv.c	2022-09-03 18:44:44.808244014 +0200
@@ -96,6 +96,9 @@
 
 static int SendCommand(char *pCmd, int nCmdSize);
 
+EPS_ERR_CODE epsMakeMainteCmd (EPS_INT32 cmd, EPS_UINT8* buffer, EPS_UINT32* buffersize);
+void get_device_id (char* device_id);
+
 /* When flag stands, get status from a printer without qualification */
 static int _comserv_first_flag;
 
diff -Naur a/epson-backend-1.1.1/daemon/epmnt.c b/epson-backend-1.1.1/daemon/epmnt.c
--- a/epson-backend-1.1.1/daemon/epmnt.c	2020-01-20 01:52:50.000000000 +0100
+++ b/epson-backend-1.1.1/daemon/epmnt.c	2022-09-03 18:45:19.428310338 +0200
@@ -1,5 +1,7 @@
 #include "epson-typdefs.h"
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 #include <time.h>
 
 static const EPS_UINT8 ExitPacketMode[]= {
