diff -Nur orig/CMakeLists.txt mod/CMakeLists.txt
--- orig/CMakeLists.txt	2017-09-20 15:59:51.000000000 +0200
+++ mod/CMakeLists.txt	2017-10-15 19:34:18.597981418 +0200
@@ -407,14 +407,14 @@
 endif()
 
 #cpack
-set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "DB Browser for SQLite")
-set(CPACK_PACKAGE_VENDOR "DB Browser for SQLite Team")
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "sqlitebrowser")
+set(CPACK_PACKAGE_VENDOR "sqlitebrowser Team")
 set(CPACK_PACKAGE_DESCRIPTION_FILE "${CMAKE_CURRENT_SOURCE_DIR}/README.md")
 set(CPACK_RESOURCE_FILE_LICENSE "${CMAKE_CURRENT_SOURCE_DIR}/LICENSE")
 set(CPACK_PACKAGE_VERSION_MAJOR "3")
 set(CPACK_PACKAGE_VERSION_MINOR "10")
 set(CPACK_PACKAGE_VERSION_PATCH "1")
-set(CPACK_PACKAGE_INSTALL_DIRECTORY "DB Browser for SQLite")
+set(CPACK_PACKAGE_INSTALL_DIRECTORY "sqlitebrowser")
 if(WIN32 AND NOT UNIX)
 	# There is a bug in NSIS that does not handle full unix paths properly. Make
 	# sure there is at least one set of four (4) backlasshes.
@@ -440,5 +440,5 @@
 	set(CPACK_STRIP_FILES "DB Browser for SQLite")
 	set(CPACK_SOURCE_STRIP_FILES "")
 endif(WIN32 AND NOT UNIX)
-set(CPACK_PACKAGE_EXECUTABLES "DB Browser for SQLite" "DB Browser for SQLite")
+set(CPACK_PACKAGE_EXECUTABLES "sqlitebrowser" "sqlitebrowser")
 include(CPack)
