diff -Nur orig/meson.build mod/meson.build
--- orig/meson.build	2023-09-17 01:24:10.000000000 +0200
+++ mod/meson.build	2024-01-12 12:20:37.456723318 +0100
@@ -43,7 +43,7 @@
 libglib_dep = dependency('gio-2.0', version: glib_req)
 libgtk_dep = dependency('gtk4', version: gtk_req)
 libgtksourceview_dep = dependency('gtksourceview-5', version: gtksourceview_req)
-libadwaita_dep = dependency('libadwaita-1', version: '>= 1.4.alpha')
+libadwaita_dep = dependency('libadwaita-1', version: '>= 1.3.6')
 libeditorconfig_dep = dependency('editorconfig')
 libcairo_dep = dependency('cairo')
 
