diff -Nur orig/dbdimp.c mod/dbdimp.c
--- orig/dbdimp.c	2017-06-29 10:59:39.000000000 +0200
+++ mod/dbdimp.c	2017-09-23 17:56:31.072655848 +0200
@@ -1979,6 +1979,7 @@
 
     if (result)
     {
+      my_bool reconnect= 1;
 #if MYSQL_VERSION_ID >=SERVER_PREPARE_VERSION
       /* connection succeeded. */
       /* imp_dbh == NULL when mysql_dr_connect() is called from mysql.xs
@@ -1997,7 +1998,7 @@
         we turn off Mysql's auto reconnect and handle re-connecting ourselves
         so that we can keep track of when this happens.
       */
-      result->reconnect=0;
+      mysql_options(result, MYSQL_OPT_RECONNECT, &reconnect);
     }
     else {
       /* 
diff -Nur orig/dbdimp.h mod/dbdimp.h
--- orig/dbdimp.h	2017-06-29 10:59:39.000000000 +0200
+++ mod/dbdimp.h	2017-09-23 17:57:19.876569113 +0200
@@ -18,10 +18,10 @@
  *  Header files we use
  */
 #include <DBIXS.h>  /* installed by the DBI module                        */
-#include <mysql.h>  /* Comes with MySQL-devel */
-#include <mysqld_error.h>  /* Comes MySQL */
-
-#include <errmsg.h> /* Comes with MySQL-devel */
+#include <mysql/mysql.h>  /* Comes with MySQL-devel */
+#include <mysql/mysqld_error.h>  /* Comes MySQL */
+#include <mysql/mariadb_version.h>  /* For MYSQL_VERSION_ID */
+#include <mysql/errmsg.h> /* Comes with MySQL-devel */
 
 /* For now, we hardcode this, but in the future,
  * we can detect capabilities of the MySQL libraries
