diff -Naur orig/plugins/spell/pluma-spell-checker.c patched/plugins/spell/pluma-spell-checker.c
--- orig/plugins/spell/pluma-spell-checker.c	2018-01-30 15:26:27.000000000 +0100
+++ patched/plugins/spell/pluma-spell-checker.c	2018-09-26 01:35:48.088794013 +0200
@@ -433,7 +433,7 @@
 	if (len < 0)
 		len = strlen (word);
 
-	enchant_dict_add_to_pwl (spell->dict, word, len);
+	enchant_dict_add (spell->dict, word, len);
 
 	g_signal_emit (G_OBJECT (spell), signals[ADD_WORD_TO_PERSONAL], 0, word, len);
 
