diff -Naur a/help/Makefile.am b/help/Makefile.am
--- a/help/Makefile.am	2022-11-09 23:44:30.000000000 +0100
+++ b/help/Makefile.am	2022-11-15 00:44:38.268448255 +0100
@@ -7,7 +7,7 @@
 	figures/pluma_window.png	
 
 # Add linguas to be ignored, e.g. IGNORE_HELP_LINGUAS = ca de es fr
-IGNORE_HELP_LINGUAS =
+IGNORE_HELP_LINGUAS = ru
 HELP_LINGUAS = $(if $(IGNORE_HELP_LINGUAS), \
 	$(filter-out $(IGNORE_HELP_LINGUAS),$(subst /,,$(dir $(wildcard */*.po)))), \
 	$(subst /,,$(dir $(wildcard */*.po))) )
diff -Naur a/help/Makefile.in b/help/Makefile.in
--- a/help/Makefile.in	2022-11-09 23:46:47.000000000 +0100
+++ b/help/Makefile.in	2022-11-15 00:44:55.036550772 +0100
@@ -333,7 +333,7 @@
 
 
 # Add linguas to be ignored, e.g. IGNORE_HELP_LINGUAS = ca de es fr
-IGNORE_HELP_LINGUAS = 
+IGNORE_HELP_LINGUAS = ru
 HELP_LINGUAS = $(if $(IGNORE_HELP_LINGUAS), \
 	$(filter-out $(IGNORE_HELP_LINGUAS),$(subst /,,$(dir $(wildcard */*.po)))), \
 	$(subst /,,$(dir $(wildcard */*.po))) )
