diff -Naur a/Makefile b/Makefile
--- a/Makefile	2020-05-31 00:17:56.000000000 +0200
+++ b/Makefile	2022-08-07 22:18:05.466576225 +0200
@@ -32,8 +32,8 @@
 ABI_VERSION=.3
 
 # Installation directories
-PREFIX=/usr/local
-SBINDIR=$(PREFIX)/sbin
+PREFIX=/usr
+SBINDIR=$(PREFIX)/bin
 SHAREDIR=$(PREFIX)/share
 IDSDIR=$(SHAREDIR)
 MANDIR:=$(shell if [ -d $(PREFIX)/share/man ] ; then echo $(PREFIX)/share/man ; else echo $(PREFIX)/man ; fi)
