diff -Naur gnome-vfs-2.12.0.orig/libgnomevfs/gnome-vfs-hal-mounts.c gnome-vfs-2.12.0/libgnomevfs/gnome-vfs-hal-mounts.c
--- gnome-vfs-2.12.0.orig/libgnomevfs/gnome-vfs-hal-mounts.c	2005-09-05 10:19:11.000000000 +0200
+++ gnome-vfs-2.12.0/libgnomevfs/gnome-vfs-hal-mounts.c	2005-09-28 18:35:45.000000000 +0200
@@ -190,15 +190,15 @@
 	{HAL_ICON_VOLUME_FLASH_KEY,               "gnome-dev-harddisk"},        /* TODO: gnome-dev-pendrive */
 
 	{HAL_ICON_DISC_CDROM,                     "gnome-dev-cdrom"},
-	{HAL_ICON_DISC_CDR,                       "gnome-dev-disc-cdr"},
-	{HAL_ICON_DISC_CDRW,                      "gnome-dev-disc-cdrw"},
-	{HAL_ICON_DISC_DVDROM,                    "gnome-dev-disc-dvdrom"},
-	{HAL_ICON_DISC_DVDRAM,                    "gnome-dev-disc-dvdram"},
-	{HAL_ICON_DISC_DVDR,                      "gnome-dev-disc-dvdr"},
-	{HAL_ICON_DISC_DVDRW,                     "gnome-dev-disc-dvdrw"},
-	{HAL_ICON_DISC_DVDPLUSR,                  "gnome-dev-disc-dvdr-plus"},
-	{HAL_ICON_DISC_DVDPLUSRW,                 "gnome-dev-disc-dvdrw"},      /* TODO: gnome-dev-disc-dvdrw-plus */
-	{HAL_ICON_DISC_DVDPLUSR_DL,               "gnome-dev-disc-dvdr-plus"},  /* TODO: gnome-dev-disc-dvdr-plus-dl */
+	{HAL_ICON_DISC_CDR,                       "gnome-dev-cdrom"},
+	{HAL_ICON_DISC_CDRW,                      "gnome-dev-cdrom"},
+	{HAL_ICON_DISC_DVDROM,                    "gnome-dev-dvd"},
+	{HAL_ICON_DISC_DVDRAM,                    "gnome-dev-dvd"},
+	{HAL_ICON_DISC_DVDR,                      "gnome-dev-dvd"},
+	{HAL_ICON_DISC_DVDRW,                     "gnome-dev-dvd"},
+	{HAL_ICON_DISC_DVDPLUSR,                  "gnome-dev-dvd"},
+	{HAL_ICON_DISC_DVDPLUSRW,                 "gnome-dev-dvd"},
+	{HAL_ICON_DISC_DVDPLUSR_DL,               "gnome-dev-dvd"},
 
 	{0x00, NULL}
 };
