diff -Naurdp a/modules/cdda-cddb.c b/modules/cdda-cddb.c
--- a/modules/cdda-cddb.c	2004-08-09 18:05:32.000000000 +0200
+++ b/modules/cdda-cddb.c	2005-02-07 13:13:59.816384060 +0100
@@ -541,7 +541,8 @@ static void CDDBProcessLine(char *inbuff
     {
       len=strlen(data->data_track[track].track_name);
 
-      strncpy(data->data_track[track].track_name+len,
+      sprintf(data->data_track[track].track_name, "%02d ", track);
+      strncpy(data->data_track[track].track_name+len+3,
 	    ChopWhite(strtok(NULL,"")),256-len);
     }
   }
diff -Naurdp a/modules/cdda-method.c b/modules/cdda-method.c
--- a/modules/cdda-method.c	2004-08-25 15:46:36.000000000 +0200
+++ b/modules/cdda-method.c	2005-02-07 13:14:40.298360317 +0100
@@ -178,15 +178,17 @@ cdda_set_file_info_for_root (CDDAContext
 	/* We don't know the io_block size */
 	context->file_info->io_block_size = 0;
 	context->file_info->valid_fields -= GNOME_VFS_FILE_INFO_FIELDS_IO_BLOCK_SIZE;		
-	context->file_info->name = gnome_vfs_uri_extract_short_path_name (uri);
+	context->file_info->name = g_strdup("Audio CD");
 	/*context->file_info->name = g_strdup (context->disc_data.data_title);*/
 	context->file_info->type = GNOME_VFS_FILE_TYPE_DIRECTORY;
 	context->file_info->mime_type = g_strdup ("x-directory/normal");
+	context->file_info->permissions = 0555;
 	context->file_info->atime = time (NULL);
 	context->file_info->ctime = time (NULL);
 	context->file_info->mtime = time (NULL);
 	context->file_info->valid_fields |= GNOME_VFS_FILE_INFO_FIELDS_TYPE |
 										GNOME_VFS_FILE_INFO_FIELDS_MIME_TYPE |
+										GNOME_VFS_FILE_INFO_FIELDS_PERMISSIONS |
 										GNOME_VFS_FILE_INFO_FIELDS_ATIME | 
 										GNOME_VFS_FILE_INFO_FIELDS_MTIME |
 										GNOME_VFS_FILE_INFO_FIELDS_CTIME;
@@ -273,9 +275,8 @@ do_open (GnomeVFSMethod *method, GnomeVF
 		schemedir = g_strdup_printf ("cdda://%s", dirname);
 		
 		/* Remove trailing '/' if there is one */
-		sep = strrchr (schemedir, '/');
-		if (sep != NULL) {
-			schemedir [strlen (schemedir) - 1] = '\0';
+		if (schemedir [strlen (schemedir)] == '/') {
+			schemedir [strlen (schemedir)] = '\0';
 		}
 		
 		dir_uri = gnome_vfs_uri_new (schemedir);
@@ -546,12 +547,14 @@ get_file_info_for_basename (CDDAContext 
 			context->file_info->name = g_strdup (base_name);
 			context->file_info->type = GNOME_VFS_FILE_TYPE_REGULAR;
 			context->file_info->mime_type = g_strdup ("audio/x-wav");
+			context->file_info->permissions = 0444;
 			context->file_info->atime = time (NULL);
 			context->file_info->ctime = time (NULL);
 			context->file_info->mtime = time (NULL);
 			context->file_info->size = get_data_size (context->drive, index + 1);
 			context->file_info->valid_fields  = GNOME_VFS_FILE_INFO_FIELDS_TYPE |
 												GNOME_VFS_FILE_INFO_FIELDS_MIME_TYPE |
+												GNOME_VFS_FILE_INFO_FIELDS_PERMISSIONS |
 												GNOME_VFS_FILE_INFO_FIELDS_SIZE |
 												GNOME_VFS_FILE_INFO_FIELDS_IO_BLOCK_SIZE |
 												GNOME_VFS_FILE_INFO_FIELDS_ATIME | 
@@ -600,9 +603,8 @@ do_get_file_info (GnomeVFSMethod *method
 			schemedir = g_strdup_printf ("cdda://%s", dirname);
 			
 			/* Remove trailing '/' if there is one  */
-			sep = strrchr (schemedir, '/');
-			if (sep != NULL) {
-				schemedir [strlen (schemedir) - 1] = '\0';
+			if (schemedir [strlen (schemedir)] == '/') {
+				schemedir [strlen (schemedir)] = '\0';
 			}
 			
 			dir_uri = gnome_vfs_uri_new (schemedir);
@@ -691,9 +693,8 @@ do_open_directory (GnomeVFSMethod *metho
 			schemedir = g_strdup_printf ("cdda://%s", dirname);
 			
 			/* Remove trailing '/' if there is one */
-			sep = strrchr (schemedir, '/');
-			if (sep != NULL) {
-				schemedir [strlen (schemedir) - 1] = '\0';
+			if (schemedir [strlen (schemedir)] == '/') {
+				schemedir [strlen (schemedir)] = '\0';
 			}
 			
 			dir_uri = gnome_vfs_uri_new (schemedir);
@@ -724,6 +725,7 @@ do_open_directory (GnomeVFSMethod *metho
 					cdda_set_file_info_for_root (global_context, uri);
 				} else {
 					/* g_message ("Using cache"); */
+					cdda_set_file_info_for_root (global_context, uri);
 					cdda_close (drive);
 				}
 		} else {
@@ -860,12 +862,14 @@ do_read_directory (GnomeVFSMethod *metho
 	if (cdda_context->use_cddb) {
 		file_info->name = g_strdup (cdda_context->disc_data.data_track[cdda_context->access_count-1].track_name);
 	} else {
-		file_info->name = g_strdup_printf ("Untitled %d", cdda_context->access_count);
+		file_info->name = g_strdup_printf ("Track %02d", cdda_context->access_count);
 	}
 	file_info->type = GNOME_VFS_FILE_TYPE_REGULAR;
 	file_info->mime_type = g_strdup ("audio/x-wav");
+	file_info->permissions = 0444;
 	file_info->valid_fields = 	GNOME_VFS_FILE_INFO_FIELDS_TYPE |
 								GNOME_VFS_FILE_INFO_FIELDS_MIME_TYPE | 
+								GNOME_VFS_FILE_INFO_FIELDS_PERMISSIONS |
 								GNOME_VFS_FILE_INFO_FIELDS_SIZE |
 								GNOME_VFS_FILE_INFO_FIELDS_IO_BLOCK_SIZE |
 								GNOME_VFS_FILE_INFO_FIELDS_ATIME | 
