diff -Naurd gnome-vfs-2.24.4.orig//daemon/Makefile.in gnome-vfs-2.24.4//daemon/Makefile.in
--- gnome-vfs-2.24.4.orig//daemon/Makefile.in	2010-09-28 11:35:22.000000000 +0200
+++ gnome-vfs-2.24.4//daemon/Makefile.in	2013-04-03 19:46:54.000000000 +0200
@@ -337,7 +337,6 @@
 	-D_LARGEFILE64_SOURCE                   \
 	-D_POSIX_PTHREAD_SEMANTICS		\
 	-D_REENTRANT				\
-	-DG_DISABLE_DEPRECATED			\
 	-DGNOME_VFS_PREFIX=\"$(prefix)\"	\
 	-DGNOME_VFS_DATADIR=\"$(datadir)\"	\
 	-DGNOME_VFS_LIBDIR=\"$(libdir)\"	\
diff -Naurd gnome-vfs-2.24.4.orig//libgnomevfs/Makefile.in gnome-vfs-2.24.4//libgnomevfs/Makefile.in
--- gnome-vfs-2.24.4.orig//libgnomevfs/Makefile.in	2010-09-28 11:35:22.000000000 +0200
+++ gnome-vfs-2.24.4//libgnomevfs/Makefile.in	2013-04-03 19:47:47.000000000 +0200
@@ -430,7 +430,6 @@
 	-D_LARGEFILE64_SOURCE                   \
 	-D_POSIX_PTHREAD_SEMANTICS		\
 	-D_REENTRANT				\
-	-DG_DISABLE_DEPRECATED			\
 	-DGNOME_VFS_PREFIX=\"$(prefix)\"	\
 	-DGNOME_VFS_BINDIR=\"$(bindir)\"	\
 	-DGNOME_VFS_DATADIR=\"$(datadir)\"	\
diff -Naurd gnome-vfs-2.24.4.orig//modules/Makefile.in gnome-vfs-2.24.4//modules/Makefile.in
--- gnome-vfs-2.24.4.orig//modules/Makefile.in	2010-09-28 11:35:22.000000000 +0200
+++ gnome-vfs-2.24.4//modules/Makefile.in	2013-04-03 19:47:59.000000000 +0200
@@ -468,7 +468,6 @@
 	-D_POSIX_PTHREAD_SEMANTICS		\
 	-D_REENTRANT				\
 	-DSSH_PROGRAM=\"$(SSH_PROGRAM)\"	\
-	-DG_DISABLE_DEPRECATED			\
 	-DGNOME_VFS_PREFIX=\"$(prefix)\"	\
 	-DGNOME_VFS_DATADIR=\"$(datadir)\"	\
 	-DGNOME_VFS_LIBDIR=\"$(libdir)\"	\
diff -Naurd gnome-vfs-2.24.4.orig//programs/Makefile.in gnome-vfs-2.24.4//programs/Makefile.in
--- gnome-vfs-2.24.4.orig//programs/Makefile.in	2010-09-28 11:35:22.000000000 +0200
+++ gnome-vfs-2.24.4//programs/Makefile.in	2013-04-03 19:48:15.000000000 +0200
@@ -344,8 +344,7 @@
 	-I$(top_srcdir)				\
 	-I$(top_builddir)			\
 	$(TEST_CFLAGS)				\
-	$(VFS_CFLAGS)				\
-	-DG_DISABLE_DEPRECATED
+	$(VFS_CFLAGS)
 
 libraries = \
 	$(top_builddir)/libgnomevfs/libgnomevfs-2.la	\
diff -Naurd gnome-vfs-2.24.4.orig//test/Makefile.in gnome-vfs-2.24.4//test/Makefile.in
--- gnome-vfs-2.24.4.orig//test/Makefile.in	2010-09-28 11:35:23.000000000 +0200
+++ gnome-vfs-2.24.4//test/Makefile.in	2013-04-03 19:48:35.000000000 +0200
@@ -471,7 +471,6 @@
 	-I$(top_builddir)/libgnomevfs           \
 	$(TEST_CFLAGS)				\
 	$(VFS_CFLAGS)				\
-	-DG_DISABLE_DEPRECATED			\
 	-DMODULES_PATH=\"$(libdir)/vfs/modules\"
 
 @OS_WIN32_FALSE@platform_only_programs = test-dns-sd test-symlinks test-parse-ls-lga
