diff -Naur krb5-1.12.1-orig/src/build-tools/krb5-config.in krb5-1.12.1/src/build-tools/krb5-config.in
--- krb5-1.12.1-orig/src/build-tools/krb5-config.in	2014-01-16 01:44:15.000000000 +0100
+++ krb5-1.12.1/src/build-tools/krb5-config.in	2014-02-25 18:20:29.498201508 +0100
@@ -200,7 +200,7 @@
 
 if test -n "$do_cflags"; then
     if test x"$includedir" != x"/usr/include" ; then
-        echo "-I${includedir}"
+        echo "-I${includedir} -I/usr/include/et"
     else
         echo ''
     fi
