diff -Naur a/daemon/dbus/gkd-daemon-generated.c b/daemon/dbus/gkd-daemon-generated.c
--- a/daemon/dbus/gkd-daemon-generated.c	2022-05-21 15:14:16.000000000 +0200
+++ b/daemon/dbus/gkd-daemon-generated.c	2022-05-21 20:02:40.648255572 +0200
@@ -10,7 +10,7 @@
 #  include "config.h"
 #endif
 
-#include "/home/niels/jhbuild/checkout/gnome-keyring/daemon/dbus/gkd-daemon-generated.h"
+#include "gkd-daemon-generated.h"
 
 #include <string.h>
 #ifdef G_OS_UNIX
diff -Naur a/daemon/dbus/gkd-internal-generated.c b/daemon/dbus/gkd-internal-generated.c
--- a/daemon/dbus/gkd-internal-generated.c	2022-05-21 15:14:16.000000000 +0200
+++ b/daemon/dbus/gkd-internal-generated.c	2022-05-21 20:04:32.164523219 +0200
@@ -10,7 +10,7 @@
 #  include "config.h"
 #endif
 
-#include "/home/niels/jhbuild/checkout/gnome-keyring/daemon/dbus/gkd-internal-generated.h"
+#include "gkd-internal-generated.h"
 
 #include <string.h>
 #ifdef G_OS_UNIX
diff -Naur a/daemon/dbus/gkd-portal-generated.c b/daemon/dbus/gkd-portal-generated.c
--- a/daemon/dbus/gkd-portal-generated.c	2022-05-21 15:14:16.000000000 +0200
+++ b/daemon/dbus/gkd-portal-generated.c	2022-05-21 20:07:01.316734380 +0200
@@ -10,7 +10,7 @@
 #  include "config.h"
 #endif
 
-#include "/home/niels/jhbuild/checkout/gnome-keyring/daemon/dbus/gkd-portal-generated.h"
+#include "gkd-portal-generated.h"
 
 #include <string.h>
 #ifdef G_OS_UNIX
diff -Naur a/daemon/dbus/gkd-portal-request-generated.c b/daemon/dbus/gkd-portal-request-generated.c
--- a/daemon/dbus/gkd-portal-request-generated.c	2022-05-21 15:14:16.000000000 +0200
+++ b/daemon/dbus/gkd-portal-request-generated.c	2022-05-21 20:04:41.132535699 +0200
@@ -10,7 +10,7 @@
 #  include "config.h"
 #endif
 
-#include "/home/niels/jhbuild/checkout/gnome-keyring/daemon/dbus/gkd-portal-request-generated.h"
+#include "gkd-portal-request-generated.h"
 
 #include <string.h>
 #ifdef G_OS_UNIX
diff -Naur a/daemon/dbus/gkd-secrets-generated.c b/daemon/dbus/gkd-secrets-generated.c
--- a/daemon/dbus/gkd-secrets-generated.c	2022-05-21 15:14:16.000000000 +0200
+++ b/daemon/dbus/gkd-secrets-generated.c	2022-05-21 20:05:57.976643772 +0200
@@ -10,7 +10,7 @@
 #  include "config.h"
 #endif
 
-#include "/home/niels/jhbuild/checkout/gnome-keyring/daemon/dbus/gkd-secrets-generated.h"
+#include "gkd-secrets-generated.h"
 
 #include <string.h>
 #ifdef G_OS_UNIX
