diff -Naur a/pyproject.toml b/pyproject.toml
--- a/pyproject.toml	2020-02-02 01:00:00.000000000 +0100
+++ b/pyproject.toml	2022-09-26 01:20:22.672367854 +0200
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["hatchling", "hatch-vcs", "hatch-fancy-pypi-readme"]
+requires = ["hatchling", "hatch-vcs"]
 build-backend = "hatchling.build"
 
 [tool.hatch.version]
@@ -29,7 +29,7 @@
   "Programming Language :: Python :: Implementation :: CPython",
   "Programming Language :: Python :: Implementation :: PyPy",
 ]
-dynamic = ["version", "readme"]
+dynamic = ["version"]
 
 dependencies = [
   "attrs>=17.4.0",
@@ -76,30 +76,6 @@
 Changelog = "https://github.com/python-jsonschema/jsonschema/blob/main/CHANGELOG.rst"
 Source = "https://github.com/python-jsonschema/jsonschema"
 
-[tool.hatch.metadata.hooks.fancy-pypi-readme]
-content-type = "text/x-rst"
-
-[[tool.hatch.metadata.hooks.fancy-pypi-readme.fragments]]
-path = "README.rst"
-end-before = ".. start cut from PyPI"
-
-[[tool.hatch.metadata.hooks.fancy-pypi-readme.fragments]]
-path = "README.rst"
-start-after = ".. end cut from PyPI\n\n\n"
-
-[[tool.hatch.metadata.hooks.fancy-pypi-readme.fragments]]
-text = """
-
-
-Release Information
--------------------
-
-"""
-
-[[tool.hatch.metadata.hooks.fancy-pypi-readme.fragments]]
-path = "CHANGELOG.rst"
-pattern = "(^v.+?)\nv"
-
 [tool.isort]
 from_first = true
 include_trailing_comma = true
