diff -Nur orig/OpenIPMIpthread.pc.in mod/OpenIPMIpthread.pc.in
--- orig/OpenIPMIpthread.pc.in	2020-08-05 23:59:21.000000000 +0200
+++ mod/OpenIPMIpthread.pc.in	2021-01-02 18:51:37.160206110 +0100
@@ -7,5 +7,5 @@
 Description: Pthread OS handler for OpenIPMI
 Version: @VERSION@
 Requires: OpenIPMI
-Libs: -L${libdir} -lOpenIPMIutils -lOpenIPMIpthread -pthread
+Libs: -L${libdir} -lOpenIPMIutils -lOpenIPMIpthread -lpthread
 Cflags: -I${includedir} -pthread
