diff -Naur OpenIPMI-1.3.8.orig/OpenIPMIpthread.pc.in OpenIPMI-1.3.8/OpenIPMIpthread.pc.in
--- OpenIPMI-1.3.8.orig/OpenIPMIpthread.pc.in	2004-03-20 05:43:27.000000000 +0100
+++ OpenIPMI-1.3.8/OpenIPMIpthread.pc.in	2005-11-22 15:54:06.000000000 +0100
@@ -6,6 +6,6 @@
 Name: OpenIPMIpthread
 Description: Pthread OS handler for OpenIPMI
 Version: @VERSION@
-Requires: OpenIPMI pthread
-Libs: -L${libdir} -lOpenIPMIutils -lOpenIPMIpthread
+Requires: OpenIPMI
+Libs: -L${libdir} -lOpenIPMIutils -lOpenIPMIpthread -lpthread
 Cflags: -I${includedir}
diff -Naur OpenIPMI-1.3.8.orig/OpenIPMIui.pc.in OpenIPMI-1.3.8/OpenIPMIui.pc.in
--- OpenIPMI-1.3.8.orig/OpenIPMIui.pc.in	2004-04-19 20:25:20.000000000 +0200
+++ OpenIPMI-1.3.8/OpenIPMIui.pc.in	2005-11-22 15:54:16.000000000 +0100
@@ -6,6 +6,6 @@
 Name: OpenIPMIui
 Description: UI front-end for OpenIPMI
 Version: @VERSION@
-Requires: OpenIPMI ncurses
+Requires: OpenIPMI
 Libs: -L${libdir} -lOpenIPMIutils -lOpenIPMIui -lOpenIPMI -lncurses
 Cflags: -I${includedir}
