diff -Nur orig/fvwm/preferences/Startup mod/fvwm/preferences/Startup
--- orig/fvwm/preferences/Startup	2022-02-20 19:22:58.000000000 +0100
+++ mod/fvwm/preferences/Startup	2022-03-18 11:21:09.053459928 +0100
@@ -51,3 +51,17 @@
 #+ I Test (Init) Test (x zim) Exec pidof zim || exec zim
 
 # vim:ft=fvwm
+
+# csatolási hitelesítők
+
+# lxde
+#exec /usr/libexec/lxpolkit
+
+# mate
+#exec /usr/libexec/polkit-mate-authentication-agent-1
+
+# gnome
+#exec /usr/libexec/polkit-gnome-authentication-agent-1
+
+# kde
+#exec /usr/lib/libexec/polkit-kde-authentication-agent-1
