diff -Nur orig/fvwm/scripts/FvwmMPD/getprevdir.py mod/fvwm/scripts/FvwmMPD/getprevdir.py
--- orig/fvwm/scripts/FvwmMPD/getprevdir.py	2020-02-21 16:46:25.000000000 +0100
+++ mod/fvwm/scripts/FvwmMPD/getprevdir.py	2020-04-04 14:17:00.993254206 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3
 # Return the path to the previous mpd library directory
 import sys
 
diff -Nur orig/fvwm/scripts/FvwmMPD/stripnames.py mod/fvwm/scripts/FvwmMPD/stripnames.py
--- orig/fvwm/scripts/FvwmMPD/stripnames.py	2020-02-21 16:46:25.000000000 +0100
+++ mod/fvwm/scripts/FvwmMPD/stripnames.py	2020-04-04 14:16:47.585272244 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3
 import sys
 
 if __name__ == "__main__":
