diff -Naur orig/fvwm/Applications/Settings/Package_Managers/~synaptic~Synaptic mod/fvwm/Applications/Settings/Package_Managers/~synaptic~Synaptic
--- orig/fvwm/Applications/Settings/Package_Managers/~synaptic~Synaptic	2013-01-13 01:39:21.000000000 +0100
+++ mod/fvwm/Applications/Settings/Package_Managers/~synaptic~Synaptic	2014-09-21 17:24:43.000000000 +0200
@@ -1,4 +1,4 @@
 #!/bin/sh
 
-exec FvwmCommand 'SX synaptic 2>/dev/null'
+exec synaptic
 
