diff -Naur orig/Makefile.in patched/Makefile.in
--- orig/Makefile.in	2019-07-23 23:21:03.000000000 +0200
+++ patched/Makefile.in	2019-09-23 22:07:09.530898588 +0200
@@ -60,7 +60,6 @@
 			$(libdir)/liblockfile.so.$(SOVER)
 		ln -s liblockfile.so.$(SOVER) $(libdir)/liblockfile.so.$(MAJOR)
 		ln -s liblockfile.so.$(SOVER) $(libdir)/liblockfile.so
-		if test "$(DESTDIR)" = ""; then @LDCONFIG@; fi
 
 install_common:
 		install -d -m 755 -g root -p $(includedir)
