diff -Nur orig/Makefile.in mod/Makefile.in
--- orig/Makefile.in	2016-11-28 15:05:35.000000000 +0100
+++ mod/Makefile.in	2017-09-18 10:19:32.762891441 +0200
@@ -58,7 +58,6 @@
 			$(libdir)/liblockfile.so.$(SOVER)
 		ln -s liblockfile.so.$(SOVER) $(libdir)/liblockfile.so.$(MAJOR)
 		ln -s liblockfile.so.$(SOVER) $(libdir)/liblockfile.so
-		if test "$(DESTDIR)" = ""; then @LDCONFIG@; fi
 
 install_common:
 		install -d -m 755 -g root -p $(includedir)
