diff -Naur orig/doc/Makefile.am patched/doc/Makefile.am
--- orig/doc/Makefile.am	2016-06-10 23:32:51.000000000 +0200
+++ patched/doc/Makefile.am	2016-07-16 18:21:21.769877529 +0200
@@ -20,7 +20,6 @@
 		el_resize.3 \
 		el_set.3 \
 		el_source.3 \
-		history.3 \
 		history_end.3 \
 		history_init.3 \
 		tok_end.3 \
diff -Naur orig/doc/Makefile.in patched/doc/Makefile.in
--- orig/doc/Makefile.in	2016-06-18 14:35:31.000000000 +0200
+++ patched/doc/Makefile.in	2016-07-16 18:21:07.905651980 +0200
@@ -278,7 +278,7 @@
 EL_MAN_LINKS = el_deletestr.3 el_end.3 el_get.3 el_getc.3 el_gets.3 \
 	el_init.3 el_init_fd.3 el_insertstr.3 el_line.3 el_parse.3 \
 	el_push.3 el_reset.3 el_resize.3 el_set.3 el_source.3 \
-	history.3 history_end.3 history_init.3 tok_end.3 tok_init.3 \
+	history_end.3 history_init.3 tok_end.3 tok_init.3 \
 	tok_line.3 tok_reset.3 tok_str.3 el_wdeletestr.3 el_wget.3 \
 	el_wgetc.3 el_wgets.3 el_winsertstr.3 el_wline.3 el_wparse.3 \
 	el_wpush.3 el_wset.3 history_w.3 history_wend.3 \
