diff -Naurd a/cream.desktop b/cream.desktop
--- a/cream.desktop	2010-01-21 23:39:42.000000000 +0100
+++ b/cream.desktop	2014-05-06 19:23:23.000000000 +0200
@@ -1,16 +1,13 @@
 [Desktop Entry]
 Type=Application
-Version=0.9.4
 Name=Cream
 GenericName=Text Editor
 Comment=Cream (for Vim) text editor
+Comment[hu]=Cream szövegszerkesztő (Vim makrókészlet)
 MimeType=text/english;text/plain;text/x-makefile;text/x-c++hdr;text/x-c++src;text/x-chdr;text/x-csrc;text/x-java;text/x-moc;text/x-pascal;text/x-tcl;text/x-tex;application/x-shellscript;text/x-c;text/x-c++;
-Exec=cream
+Exec=cream %F
 Icon=cream
 Terminal=0
-XClassHintResName=VIM
 MapNotify=false
 Encoding=UTF-8
-X-Desktop-File-Install-Version=0.4
 Categories=Application;Utility;
-
