diff -Naur syslog-ng-syslog-ng-3.16.1-orig/autogen.sh syslog-ng-syslog-ng-3.16.1/autogen.sh
--- syslog-ng-syslog-ng-3.16.1-orig/autogen.sh	2018-06-18 17:14:55.000000000 +0200
+++ syslog-ng-syslog-ng-3.16.1/autogen.sh	2018-07-18 19:25:45.691091895 +0200
@@ -25,7 +25,7 @@
 # This script is needed to setup build environment from checked out
 # source tree.
 #
-SUBMODULES="lib/ivykis lib/jsonc"
+SUBMODULES="lib/ivykis modules/afamqp/rabbitmq-c"
 GIT=`which git`
 
 include_automake_from_dir_if_exists()
