From 87578ca454f0e37d2dfdd79692fd7625a8e129ed Mon Sep 17 00:00:00 2001
From: attus <attus@enterpriseforever.com>
Date: Mon, 30 Jul 2018 17:43:53 +0200
Subject: ubuntu fonts in UHU?


diff --git a/data/themes/CMakeLists.txt b/data/themes/CMakeLists.txt
index ecfeda4..cc7734d 100644
--- a/data/themes/CMakeLists.txt
+++ b/data/themes/CMakeLists.txt
@@ -263,7 +263,7 @@ ${JWM_ROOT_MENU}
     </RootMenu>
     ${CLOCK_MENU}")
 
-set(FONT "ubuntu-12:antialias=true:encoding=utf8")
+set(FONT "Sans:10")
 ## see http://en.wikipedia.org/wiki/Table_of_keyboard_shortcuts
 # I tried to implement everything from GNOME, Windows and OSX in here
 set(KEYSHORTCUTS "<Key key=\"Up\">up</Key>
