From 8847f687dad22ee2039524bccd1633c02b1beb57 Mon Sep 17 00:00:00 2001
From: attus <attus@enterpriseforever.com>
Date: Mon, 30 Jul 2018 18:03:34 +0200
Subject: default startupcommands in UHU


diff --git a/data/themes/CMakeLists.txt b/data/themes/CMakeLists.txt
index 55f40f9..77b53ba 100644
--- a/data/themes/CMakeLists.txt
+++ b/data/themes/CMakeLists.txt
@@ -40,9 +40,10 @@ set(TWITTER "https://twitter.com/ToriOS_Official")
 set(LP "https://launchpad.net/~torios")
 
 ##Startup commands
-set (STARTUP "<StartupCommand>pcmanfm --desktop</StartupCommand>
-   <StartupCommand>$HOME/.config/jsm-mouse</StartupCommand>
-   <StartupCommand>nice-start</StartupCommand>")
+set (STARTUP "<StartupCommand>nm-applet</StartupCommand>
+	<StartupCommand>/usr/libexec/lxpolkit</StartupCommand>
+	<StartupCommand>retrovol -hide</StartupCommand>
+	<StartupCommand>synclient MaxTapTime=0</StartupCommand>")
 
 ## Desktop Background
 set(BG "/usr/share/backgrounds/default.jpg")
