diff -Nur orig/data/themes/CMakeLists.txt mod/data/themes/CMakeLists.txt
--- orig/data/themes/CMakeLists.txt	2019-12-05 16:09:35.000000000 +0100
+++ mod/data/themes/CMakeLists.txt	2019-12-06 09:29:22.587106171 +0100
@@ -41,9 +41,9 @@
 set(LP "https://launchpad.net/~torios")
 
 ##Startup commands
-set (STARTUP "<StartupCommand>pcmanfm --desktop</StartupCommand>
-   <StartupCommand>$HOME/.config/jsm-mouse</StartupCommand>
-   <StartupCommand>nice-start</StartupCommand>")
+set (STARTUP "<StartupCommand>nm-applet</StartupCommand>
+	<StartupCommand>/usr/libexec/lxpolkit</StartupCommand>
+	<StartupCommand>synclient MaxTapTime=0</StartupCommand>")
 
 ## Desktop Background
 set(BG "/usr/share/backgrounds/default.jpg")
