From 016eeb4c8d60fe9653f33fa2b65d1084c0043935 Mon Sep 17 00:00:00 2001
From: attus <attus@enterpriseforever.com>
Date: Mon, 30 Jul 2018 18:14:56 +0200
Subject: nincs torios-gamma parancs


diff --git a/data/themes/CMakeLists.txt b/data/themes/CMakeLists.txt
index 77b53ba..e8d091a 100644
--- a/data/themes/CMakeLists.txt
+++ b/data/themes/CMakeLists.txt
@@ -3,8 +3,6 @@
 
 ## DEFAULT PROGRAMS
 set(JSM "jwm-settings-manager")
-set (brightUP "exec:torios-gamma up")
-set (brightDown "exec:torios-gamma down")
 set (MENU_WEB_PROG "x-www-browser")
 set (WEB_PROG "exec:${MENU_WEB_PROG}")
 set (STARTPAGE_ICON "browser")
@@ -284,8 +282,6 @@ set(KEYSHORTCUTS "<Key key=\"Up\">up</Key>
     <Key key=\"XF86AudioLowerVolume\">${VOL_DOWN}</Key>
     <Key key=\"XF86WWW\">${WEB_PROG}</Key>
     <Key key=\"XF86Calculator\">${CALC_PROG}</Key>
-    <Key key=\"XF86MonBrightnessUp\">${brightUP}</Key>
-    <Key key=\"XF86MonBrightnessDown\">${brightDown}</Key>
     <Key key=\"Print\">${SCREEN_SHOT}</Key>
     <Key mask=\"CA\" key=\"p\">${SCREEN_SHOT}</Key>
     <Key mask=\"S4\" key=\"3\">${SCREEN_SHOT}</Key>
