From b03c09ef0290a979890aed023b1b4487c1a109d3 Mon Sep 17 00:00:00 2001
From: attus <attus@enterpriseforever.com>
Date: Mon, 30 Jul 2018 18:22:34 +0200
Subject: xtrm, xcalc, htop van


diff --git a/data/themes/CMakeLists.txt b/data/themes/CMakeLists.txt
index e8d091a..792ce3a 100644
--- a/data/themes/CMakeLists.txt
+++ b/data/themes/CMakeLists.txt
@@ -6,11 +6,11 @@ set(JSM "jwm-settings-manager")
 set (MENU_WEB_PROG "x-www-browser")
 set (WEB_PROG "exec:${MENU_WEB_PROG}")
 set (STARTPAGE_ICON "browser")
-set (SYS_MONITOR "exec:torios-runchecker xterm htop")
+set (SYS_MONITOR "exec:xterm htop")
 set (SCREEN_SHOT "exec:screenie")
 set (RUN_DIALOG "exec:zrun")
-set (CALC_PROG "exec:torios-runchecker xcalc")
-set (TERM_PROG "exec:x-terminal-emulator")
+set (CALC_PROG "exec:xcalc")
+set (TERM_PROG "exec:xterm")
 set (VOL_UP "exec:amixer-up")
 set (VOL_DOWN "exec:amixer-down")
 set (VOL_MUTE "exec:amixer-toggle")
