From 7293c193127f5ac286ba0e9a32dd26fc83608694 Mon Sep 17 00:00:00 2001
From: attus <attus@enterpriseforever.com>
Date: Mon, 30 Jul 2018 18:26:06 +0200
Subject: HELP sincs


diff --git a/data/themes/CMakeLists.txt b/data/themes/CMakeLists.txt
index 792ce3a..a16c1de 100644
--- a/data/themes/CMakeLists.txt
+++ b/data/themes/CMakeLists.txt
@@ -14,7 +14,6 @@ set (TERM_PROG "exec:xterm")
 set (VOL_UP "exec:amixer-up")
 set (VOL_DOWN "exec:amixer-down")
 set (VOL_MUTE "exec:amixer-toggle")
-set (HELP_PROG "exec:torios-runchecker xman")
 set (LOGOUT_PROGRAM "torios-exit")
 set (TZ_PROG "xterm -e 'sudo dpkg-reconfigure tzdata'")
 set (CAL_PROG "torios-calendar")
@@ -297,7 +296,6 @@ set(KEYSHORTCUTS "<Key key=\"Up\">up</Key>
     <Key mask=\"SA\" key=\"Tab\">nextstacked</Key>
     <Key mask=\"CA\" key=\"Tab\">nextstacked</Key>
     <Key mask=\"C\" key=\"#\">desktop#</Key>
-    <Key mask=\"A\" key=\"F1\">${HELP_PROG}</Key>
     <Key mask=\"C\" key=\"Escape\">${ROOT_MENU}</Key>
     <Key mask=\"A\" key=\"F2\">${RUN_DIALOG}</Key>
     <Key mask=\"4\" key=\"F2\">${RUN_DIALOG}</Key>
