From 2f87f85aff6d2a92d5b0a31892a7ff86db73afad Mon Sep 17 00:00:00 2001
From: attus <attus@enterpriseforever.com>
Date: Mon, 30 Jul 2018 21:02:39 +0200
Subject: Nincs torios-exit


diff --git a/data/themes/CMakeLists.txt b/data/themes/CMakeLists.txt
index cf356eb..2642e64 100644
--- a/data/themes/CMakeLists.txt
+++ b/data/themes/CMakeLists.txt
@@ -11,7 +11,6 @@ set (TERM_PROG "exec:xterm")
 set (VOL_UP "exec:amixer Master 5%+")
 set (VOL_DOWN "exec:amixer Master 5%-")
 set (VOL_MUTE "exec:amixer Master toggle")
-set (LOGOUT_PROGRAM "torios-exit")
 set (CAL_PROG "torios-calendar")
 set (RESTART_PROGRAM "systemctl reboot")
 
