From d98be09cd76013babc5d4d159a46c3c88115f9a9 Mon Sep 17 00:00:00 2001
From: attus <attus@enterpriseforever.com>
Date: Mon, 30 Jul 2018 22:52:18 +0200
Subject: syntax error...


diff --git a/data/themes/CMakeLists.txt b/data/themes/CMakeLists.txt
index 443a2d1..9347ed6 100644
--- a/data/themes/CMakeLists.txt
+++ b/data/themes/CMakeLists.txt
@@ -101,7 +101,7 @@ set(PLACES_ICON "folder")
 set(PLACES_MENU "    <RootMenu onroot=\"${PLACES_NUM}\">
         <Include>${PLACES_INCLUDE}</Include>
     </RootMenu>")
-set(REGULAR_MENUS_LIST "${    <Include>${APPMENU_FILE}</Include>
+set(REGULAR_MENUS_LIST "    <Include>${APPMENU_FILE}</Include>
 <!--Places Menu-->
     <Include>${PLACES_MENU_FILE}</Include>
 ${traybuttons}")
