From 6acd896d69dcf5fd3a6d6a35fc615e97dd86c9fd Mon Sep 17 00:00:00 2001
From: attus <attus@enterpriseforever.com>
Date: Tue, 31 Jul 2018 06:49:33 +0200
Subject: clock: magyar forma


diff --git a/data/themes/CMakeLists.txt b/data/themes/CMakeLists.txt
index c6f16c2..4e4a314 100644
--- a/data/themes/CMakeLists.txt
+++ b/data/themes/CMakeLists.txt
@@ -42,7 +42,7 @@ set(BG "/usr/share/backgrounds/UHU/uhu_linux_ubk.jpg")
 # m menu edit
 
 ## CLOCK
-set(CLOCK_FORMAT "%a, %e %b %l:%M %p")
+set(CLOCK_FORMAT "%b, %e %a %l:%M %p")
 set(CLOCK "root:c")
 set(OLD_CLOCK_TAG "<Clock format=\"${CLOCK_FORMAT}\"></Clock>")
 set(CLOCK_TAG "<Clock format=\"${CLOCK_FORMAT}\">${CLOCK}</Clock>")
