From 973653402898153c628a158c97f9fe80ae19a796 Mon Sep 17 00:00:00 2001
From: attus <attus@enterpriseforever.com>
Date: Tue, 31 Jul 2018 07:15:57 +0200
Subject: shuterdown: syntax


diff --git a/data/themes/CMakeLists.txt b/data/themes/CMakeLists.txt
index 697cc91..a33f97b 100644
--- a/data/themes/CMakeLists.txt
+++ b/data/themes/CMakeLists.txt
@@ -72,7 +72,7 @@ set(RESTART_ICON "view-refresh")
 set(SHUTERDOWN "<Separator/>
 	<Restart label=\"JWM újraindítás\" icon=\"${RESTART_ICON}\"/>
 	<Program icon=\"${SHUTDOWN_ICON}\" label=\"${SHUTDOWN_NAME}\">${SHUTDOWN_PROGRAM}</Program>
-	<Exit icon=\"${LOGOUT_ICON}\" label="Kilépés" confirm="true" />)
+	<Exit icon=\"${LOGOUT_ICON}\" label=\"Kilépés\" confirm=\"true\" />")
 ## MAIN MENU
 set(TRAYbuttonForTray "<TrayButton popup=\"Szoftver telepítő\" icon=\"synaptic\" border=\"false\">
           <Button mask=\"1\">exec:synaptic</Button>
