From bc48abfe786d41401ad0e78c8822a3f5f4a048ed Mon Sep 17 00:00:00 2001
From: attus <attus@enterpriseforever.com>
Date: Tue, 31 Jul 2018 07:48:36 +0200
Subject: =?UTF-8?q?kill=C3=A9p=C3=A9si=20men=C3=BCpont=20sorrend?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit


diff --git a/data/themes/CMakeLists.txt b/data/themes/CMakeLists.txt
index 5b3cb41..ec729e0 100644
--- a/data/themes/CMakeLists.txt
+++ b/data/themes/CMakeLists.txt
@@ -71,8 +71,8 @@ set(RESTART_ICON "view-refresh")
 
 set(SHUTERDOWN "<Separator/>
 	<Restart label=\"JWM újraindítás\" icon=\"${RESTART_ICON}\"/>
-	<Program icon=\"${SHUTDOWN_ICON}\" label=\"${SHUTDOWN_NAME}\">${SHUTDOWN_PROGRAM}</Program>
-	<Exit icon=\"${LOGOUT_ICON}\" label=\"Kilépés\" confirm=\"true\" />")
+	<Exit icon=\"${LOGOUT_ICON}\" label=\"Kilépés\" confirm=\"true\" />
+	<Program icon=\"${SHUTDOWN_ICON}\" label=\"${SHUTDOWN_NAME}\">${SHUTDOWN_PROGRAM}</Program>")
 ## MAIN MENU
 set(JWM_ROOT_NUM "2")
 set(JWM_ROOT "root:${JWM_ROOT_NUM}")
