From 72da2bb6c5f514eb68dea7d8dc077c6964b637a8 Mon Sep 17 00:00:00 2001
From: attus <attus@enterpriseforever.com>
Date: Tue, 31 Jul 2018 13:08:27 +0200
Subject: Numix: gombok -> Default


diff --git a/data/themes/Numix/CMakeLists.txt b/data/themes/Numix/CMakeLists.txt
index eff8ef8..ecf20e5 100644
--- a/data/themes/Numix/CMakeLists.txt
+++ b/data/themes/Numix/CMakeLists.txt
@@ -6,10 +6,11 @@ set(TRAY "${DEFAULT_TRAY}")
 set (OLD_TRAY "${OLD_DEFAULT_TRAY}")
 set(MENUS_LIST "${REGULAR_MENUS_LIST}")
 set(CORNER "0")
-set(BUTTONS "<ButtonClose>/usr/share/themes/Numix/unity/close_focused_prelight.png</ButtonClose>
-    <ButtonMax>/usr/share/themes/Numix/unity/maximize_focused_normal.png</ButtonMax>
-    <ButtonMaxActive>/usr/share/themes/Numix/unity/unmaximize.png</ButtonMaxActive>
-    <ButtonMin>/usr/share/themes/Numix/unity/minimize.png</ButtonMin>")
+set(BUTTONS "<ButtonClose>/usr/share/jwm-settings-manager/Buttons/Default/close.png</ButtonClose>
+    <ButtonMax>/usr/share/jwm-settings-manager/Buttons/Default/max.png</ButtonMax>
+    <ButtonMaxActive>/usr/share/jwm-settings-manager/Buttons/Default/max_toggle.png</ButtonMaxActive>
+    <ButtonMin>/usr/share/jwm-settings-manager/Buttons/Default/min.png</ButtonMin>")
+
 set(TASK_ACTIVE_FOREGROUND "#f9f9f9" )
 set(TASK_ACTIVE_BACKGROUND "#d64937" )
 set(TASK_BACKGROUND "#2d2d2d" )
