From 3179861f0af2b1be6138d28caa20e603fc439041 Mon Sep 17 00:00:00 2001
From: attus <attus@enterpriseforever.com>
Date: Tue, 31 Jul 2018 14:14:45 +0200
Subject: Lightgnome2: popup style


diff --git a/data/themes/LightGnome2/CMakeLists.txt b/data/themes/LightGnome2/CMakeLists.txt
index 36b1d12..45f391b 100644
--- a/data/themes/LightGnome2/CMakeLists.txt
+++ b/data/themes/LightGnome2/CMakeLists.txt
@@ -41,7 +41,7 @@ set(WINDOW_OPACITY "0.7")
 
 set(POPUP_OUTLINE "#000000")
 set(POPUP_FOREGROUND "#000000")
-set(POPUP_BACKGROUND "#353535")
+set(POPUP_BACKGROUND "#E2E2DE")
 
 ## this stays the same for all files
 set(CURRENTTHEME "../${THIS_THEME}/${THIS_THEME}")
