From 77fd03c5345678847b41c131094134e1ecd9a0fd Mon Sep 17 00:00:00 2001
From: attus <attus@enterpriseforever.com>
Date: Tue, 31 Jul 2018 20:14:09 +0200
Subject: =?UTF-8?q?Helyek=20->=20Saj=C3=A1t=20mappa?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit


diff --git a/data/themes/CMakeLists.txt b/data/themes/CMakeLists.txt
index 050c637..cb86894 100644
--- a/data/themes/CMakeLists.txt
+++ b/data/themes/CMakeLists.txt
@@ -87,7 +87,7 @@ set(TORI_ROOT "${JWM_ROOT}")
 
 
 
-set(PLACES_NAME "Helyek")
+set(PLACES_NAME "Saját mappa")
 set(PLACES_ICON "folder")
 set(MENUS_LIST_NEW "${JWM_ROOT_MENU}")
 set(REGULAR_MENUS_LIST "${MENUS_LIST_NEW}")
