diff -Naur orig/data/dekstop-file-editor.desktop patched/data/dekstop-file-editor.desktop
--- orig/data/dekstop-file-editor.desktop	2017-12-30 20:39:00.000000000 +0100
+++ patched/data/dekstop-file-editor.desktop	2018-07-14 23:44:34.978396105 +0200
@@ -7,4 +7,4 @@
 MimeType=application/x-desktop;
 Comment=Create new or change existing desktop entry
 Icon=applications-other
-NoDisplay=False
+NoDisplay=false
