diff -Nur orig/data/dekstop-file-editor.desktop mod/data/dekstop-file-editor.desktop
--- orig/data/dekstop-file-editor.desktop	2018-10-19 20:45:08.000000000 +0200
+++ mod/data/dekstop-file-editor.desktop	1970-01-01 01:00:00.000000000 +0100
@@ -1,13 +0,0 @@
-[Desktop Entry]
-Type=Application
-Name=Shortcut Editor
-Name[pt_PT]=Editor de atalhos
-GenericName[pt_PT]=Ferramenta de configuração
-Exec=desktop-file-editor %f
-Terminal=false
-Categories=Utility;GTK;
-MimeType=application/x-desktop;
-Comment=Create new or change existing desktop entry
-Comment[pt_PT]=Crie ou edite ficheiros desktop
-Icon=applications-other
-NoDisplay=false
diff -Nur orig/data/desktop-file-editor.desktop mod/data/desktop-file-editor.desktop
--- orig/data/desktop-file-editor.desktop	1970-01-01 01:00:00.000000000 +0100
+++ mod/data/desktop-file-editor.desktop	2020-05-28 12:26:03.029201078 +0200
@@ -0,0 +1,14 @@
+[Desktop Entry]
+Type=Application
+Name=Shortcut Editor
+Name[pt_PT]=Editor de atalhos
+GenericName=Configuration tool
+GenericName[pt_PT]=Ferramenta de configuração
+Exec=desktop-file-editor %f
+Terminal=false
+Categories=Utility;GTK;
+MimeType=application/x-desktop;
+Comment=Create new or change existing desktop entry
+Comment[pt_PT]=Crie ou edite ficheiros desktop
+Icon=applications-other
+NoDisplay=false
