diff -Naur orig/include/jwmrc.h patched/include/jwmrc.h
--- orig/include/jwmrc.h	2016-12-13 16:29:57.134741323 +0100
+++ patched/include/jwmrc.h	2017-05-06 01:51:17.911839503 +0200
@@ -29,9 +29,9 @@
 #include <FL/x.H>
 //FLTK includes
 #include <FL/fl_ask.H> //errorJWM
-#include <Fl/Fl_Menu_Button.H>
-#include <Fl/Fl_Output.H>
-#include <Fl/Fl_Double_Window.H>
+#include <FL/Fl_Menu_Button.H>
+#include <FL/Fl_Output.H>
+#include <FL/Fl_Double_Window.H>
 #include <FL/Fl_File_Chooser.H>
 #include <FL/Fl_PNG_Image.H>
 //Used for getting the user's Home directory
