diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2017-12-30 20:39:00.000000000 +0100
+++ patched/CMakeLists.txt	2018-07-14 23:39:02.214998587 +0200
@@ -59,8 +59,6 @@
 	## Because Launchpad's build system is really weird!
     MESSAGE("Using Debian based release")
 	FIND_PACKAGE(FLTK REQUIRED NO_MODULE)
-else()
-	 FIND_PACKAGE(FLTK REQUIRED)
 endif()
 FIND_PACKAGE(X11 REQUIRED)
 
@@ -165,7 +163,7 @@
 add_executable(fltk-mouse fltk/jwm-mouse.cxx)
 add_executable(desktop-file-editor fltk/desktop-file-editor.cxx)
 ## Linker settings
-set(ALL_LIBS JSM ${X11_LIBRARIES} ${X11_X11_LIB} ${X11_Xpm_LIB} ${FLTK_LIBRARIES})
+set(ALL_LIBS JSM ${X11_LIBRARIES} ${X11_X11_LIB} ${X11_Xpm_LIB} fltk fltk_images)
 ## Because Launchpad's build system is really weird!
 if(result)
 	set(ALL_LIBS ${ALL_LIBS} fltk_SHARED fltk_images_SHARED)
