diff -Naur orig/data/CMakeLists.txt patched/data/CMakeLists.txt
--- orig/data/CMakeLists.txt	2016-12-13 16:29:57.134741323 +0100
+++ patched/data/CMakeLists.txt	2017-05-06 00:27:05.529421668 +0200
@@ -24,9 +24,9 @@
 install(FILES icons/jwm-settings-manager.png icons/jwm-settings-manager.xpm DESTINATION share/pixmaps)
 
 # install all icons to our own icon directory
-install(FILES ${ICONS} DESTINATION usr/share/icons/hicolor/scalable/apps)
-install(FILES ${ICONS_PNG} DESTINATION usr/share/icons/hicolor/48x48/apps)
-install(FILES ${ICONS_XPM} DESTINATION usr/share/pixmaps)
+install(FILES ${ICONS} DESTINATION share/icons/hicolor/scalable/apps)
+install(FILES ${ICONS_PNG} DESTINATION share/icons/hicolor/48x48/apps)
+install(FILES ${ICONS_XPM} DESTINATION share/pixmaps)
 
 # install ALL desktop files into /usr/share/applications
 install(FILES ${APPS} DESTINATION share/applications)
