diff -Naur a/config/lxqt-config-notificationd.desktop.in b/config/lxqt-config-notificationd.desktop.in
--- a/config/lxqt-config-notificationd.desktop.in	2022-04-16 14:56:34.000000000 +0200
+++ b/config/lxqt-config-notificationd.desktop.in	2022-06-16 01:41:50.623709467 +0200
@@ -2,7 +2,7 @@
 Type=Application
 Exec=lxqt-config-notificationd
 Icon=preferences-desktop-notification
-Categories=Settings;DesktopSettings;Qt;LXQt;
+Categories=Settings;DesktopSettings;Qt;
 OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
