diff -Naur luit-0.99.1.orig/parser.h luit-0.99.1/parser.h
--- luit-0.99.1.orig/parser.h	2003-11-14 17:48:57.000000000 +0100
+++ luit-0.99.1/parser.h	2005-12-02 14:06:21.000000000 +0100
@@ -21,7 +21,7 @@
 */
 
 #ifndef LOCALE_ALIAS_FILE
-#define LOCALE_ALIAS_FILE "/usr/X11R6/lib/X11/locale/locale.alias"
+#define LOCALE_ALIAS_FILE "/usr/lib/X11/locale/locale.alias"
 #endif
 
 #define MAX_KEYWORD_LENGTH 1024
