diff -Naurd a/isomaster.desktop.src b/isomaster.desktop.src
--- a/isomaster.desktop.src	2008-12-28 05:03:53.000000000 +0100
+++ b/isomaster.desktop.src	2014-05-06 19:59:11.000000000 +0200
@@ -6,6 +6,7 @@
 Comment=Read, write and modify ISO images
 Comment[ca]=Llegiu, escriviu i modifiqueu imatges ISO
 Comment[es]=Leer, escribir i modificar imagenes ISO
+Comment[hu]=ISO szerkesztő
 Exec=isomaster
 Terminal=false
 StartupNotify=true
diff -Naurd a/Makefile b/Makefile
--- a/Makefile	2014-04-17 20:28:18.000000000 +0200
+++ b/Makefile	2014-04-18 17:58:07.000000000 +0200
@@ -1,5 +1,5 @@
 # Other paths are relative to this.
-PREFIX ?= /usr/local
+PREFIX ?= /usr
 
 # Where to install the executable.
 BINPATH ?= $(PREFIX)/bin
