diff -Naurd mp3info-0.8.5a.orig/Makefile mp3info-0.8.5a/Makefile
--- mp3info-0.8.5a.orig/Makefile	2006-11-10 01:14:11.000000000 +0100
+++ mp3info-0.8.5a/Makefile	2009-02-19 19:41:33.000000000 +0100
@@ -47,7 +47,7 @@
 CC = gcc
 CFLAGS = -g -O2 -Wall
 
-all: mp3info gmp3info doc
+all: mp3info doc
 
 doc: mp3info.txt
 
@@ -77,7 +77,7 @@
 	$(STRIP) gmp3info
 	$(INSTALL) gmp3info $(bindir)/gmp3info
 
-install: install-mp3info install-gmp3info
+install: install-mp3info
 
 
 uninstall:
