diff -Nur orig/data/Makefile.in mod/data/Makefile.in
--- orig/data/Makefile.in	2019-03-17 10:53:54.000000000 +0100
+++ mod/data/Makefile.in	2019-04-10 10:52:40.621056205 +0200
@@ -545,10 +545,10 @@
 @IS_WIN32_FALSE@	     $(appdata_in_files)
 
 @IS_WIN32_TRUE@EXTRA_DIST = $(klavaro_data)
-@IS_WIN32_FALSE@desktopdir = $(prefix)/$(DATADIRNAME)/applications
+@IS_WIN32_FALSE@desktopdir = $(prefix)/share/applications
 @IS_WIN32_FALSE@desktop_in_file = klavaro.desktop.in
 @IS_WIN32_FALSE@desktop_DATA = $(desktop_in_file:.desktop.in=.desktop)
-@IS_WIN32_FALSE@appdatadir = $(prefix)/$(DATADIRNAME)/appdata
+@IS_WIN32_FALSE@appdatadir = $(prefix)/share/appdata
 @IS_WIN32_FALSE@appdata_in_files = klavaro.appdata.xml.in
 @IS_WIN32_FALSE@appdata_DATA = $(appdata_in_files:.xml.in=.xml)
 @IS_WIN32_FALSE@DISTCLEANFILES = $(desktop_DATA) \
