diff -Naur orig/daemon-init.in patched/daemon-init.in
--- orig/daemon-init.in	2015-08-19 23:49:52.000000000 +0200
+++ patched/daemon-init.in	2016-07-09 03:22:27.182319011 +0200
@@ -47,7 +47,7 @@
 NagiosCommandFile=@localstatedir@/rw/nagios.cmd
 NagiosVarDir=@localstatedir@
 NagiosRunFile=@lockfile@
-NagiosLockDir=/var/lock/subsys
+NagiosLockDir=/var/lock
 NagiosLockFile=nagios
 NagiosCGIDir=@sbindir@
 NagiosUser=@nagios_user@
