diff -Naur orig/autoconf/Makefile.defines.in patched/autoconf/Makefile.defines.in
--- orig/autoconf/Makefile.defines.in	2013-05-01 15:34:11.000000000 +0200
+++ patched/autoconf/Makefile.defines.in	2021-04-10 21:51:26.454975178 +0200
@@ -9,7 +9,7 @@
 EXTRA_DEFINES	= 
 
 RANLIB		= @RANLIB@
-AR		= ar clq
+AR		= ar cq
 CP		= @CP@
 MV		= @MV@
 RM		= @RM@
diff -Naur orig/libAfterBase/Makefile.in patched/libAfterBase/Makefile.in
--- orig/libAfterBase/Makefile.in	2013-05-01 15:34:11.000000000 +0200
+++ patched/libAfterBase/Makefile.in	2021-04-10 21:50:59.850816625 +0200
@@ -58,7 +58,7 @@
 EXTRA_DEFINES	= @DEFINE_XLOCALE@
 
 RANLIB		= @RANLIB@
-AR		= ar clq
+AR		= ar cq
 CP		= @CP@
 MV		= @MV@
 RM		= @RM@
diff -Naur orig/libAfterImage/aftershow/Makefile.in patched/libAfterImage/aftershow/Makefile.in
--- orig/libAfterImage/aftershow/Makefile.in	2013-05-01 15:34:11.000000000 +0200
+++ patched/libAfterImage/aftershow/Makefile.in	2021-04-10 21:51:13.270896604 +0200
@@ -11,7 +11,7 @@
 EXTRA_DEFINES	= @DEFINE_XLOCALE@
 
 RANLIB		= @RANLIB@
-AR		= ar clq
+AR		= ar cq
 CP		= @CP@
 MV		= @MV@
 RM		= @RM@
diff -Naur orig/libAfterImage/apps/Makefile.in patched/libAfterImage/apps/Makefile.in
--- orig/libAfterImage/apps/Makefile.in	2013-05-01 15:34:11.000000000 +0200
+++ patched/libAfterImage/apps/Makefile.in	2021-04-10 21:51:20.106937345 +0200
@@ -8,7 +8,7 @@
 EXTRA_DEFINES	= @DEFINE_XLOCALE@
 
 RANLIB		= @RANLIB@
-AR		= ar clq
+AR		= ar cq
 CP		= @CP@
 MV		= @MV@
 RM		= @RM@
diff -Naur orig/libAfterImage/Makefile.in patched/libAfterImage/Makefile.in
--- orig/libAfterImage/Makefile.in	2013-05-01 15:34:11.000000000 +0200
+++ patched/libAfterImage/Makefile.in	2021-04-10 21:51:06.542856507 +0200
@@ -76,7 +76,7 @@
 EXTRA_DEFINES	= @DEFINE_XLOCALE@
 
 RANLIB		= @RANLIB@
-AR		= ar clq
+AR		= ar cq
 CP		= @CP@
 MV		= @MV@
 RM		= @RM@
