diff -Nur orig/src/audacity.desktop.in mod/src/audacity.desktop.in
--- orig/src/audacity.desktop.in	2017-12-06 02:59:58.000000000 +0100
+++ mod/src/audacity.desktop.in	2018-01-11 19:32:00.609587489 +0100
@@ -7,6 +7,7 @@
 GenericName[de]=Audio-Editor
 GenericName[el]=Επεξεργαστής ήχου
 GenericName[fr]=Éditeur audio
+GenericName[hu]=Hangszerkesztő
 GenericName[hi]=ध्वनि संपादक
 GenericName[ja]=オーディオエディタ
 GenericName[nl]=Geluidseditor
@@ -23,6 +24,7 @@
 Comment[de]=Audio-Dateien aufnehmen und bearbeiten
 Comment[el]=Ηχογράφηση και επεξεργασία αρχείων ήχου
 Comment[fr]=Enregistrer et éditer des fichiers audio
+Comment[hu]=Hangfájlok felvétele és szerkesztése
 Comment[hi]=ऑडियो फ़ाइल अंकित व संपादित करता है
 Comment[ja]=録音とオーディオ編集
 Comment[nl]=Audiobestanden opnemen en bewerken
