diff -Naur orig/src/audacity.desktop.in patched/src/audacity.desktop.in
--- orig/src/audacity.desktop.in	2020-06-19 17:16:47.000000000 +0200
+++ patched/src/audacity.desktop.in	2020-06-30 00:51:14.938286007 +0200
@@ -7,6 +7,7 @@
 GenericName[de]=Audio-Editor
 GenericName[el]=Επεξεργαστής ήχου
 GenericName[fr]=Éditeur audio
+GenericName[hu]=Hangszerkesztő
 GenericName[hi]=ध्वनि संपादक
 GenericName[ja]=オーディオエディタ
 GenericName[ko]=오디오 편집기
@@ -27,6 +28,7 @@
 Comment[de]=Audio-Dateien aufnehmen und bearbeiten
 Comment[el]=Ηχογράφηση και επεξεργασία αρχείων ήχου
 Comment[fr]=Enregistrer et éditer des fichiers audio
+Comment[hu]=Hangfájlok felvétele és szerkesztése
 Comment[hi]=ऑडियो फ़ाइल अंकित व संपादित करता है
 Comment[ja]=録音とオーディオ編集
 Comment[kr]=오디오 파일 녹음과 편집
