diff -Naur efax-gtk-3.0.9.orig/src/main.cpp efax-gtk-3.0.9/src/main.cpp
--- efax-gtk-3.0.9.orig/src/main.cpp	2006-03-10 23:20:31.000000000 +0100
+++ efax-gtk-3.0.9/src/main.cpp	2007-01-09 12:16:04.000000000 +0100
@@ -515,7 +515,7 @@
     //prog_config.parms.push_back("-vina");  // this will also report the parameters passed to efax
                                              // uncomment it and comment out preceding line for debugging
 
-    prog_config.parms.push_back("-u");    // force use of UTF-8 to send messages to efax-gtk
+//    prog_config.parms.push_back("-u");    // force use of UTF-8 to send messages to efax-gtk
     
     std::string temp;
 
