diff -up xmms2-0.8DrO_o/wscript.noO0 xmms2-0.8DrO_o/wscript
--- xmms2-0.8DrO_o/wscript	2011-10-20 15:26:08.000000000 -0400
+++ xmms2-0.8DrO_o/wscript	2011-12-05 15:19:15.962065486 -0500
@@ -295,8 +295,8 @@ def configure(conf):
         conf.msg("uncommited changed", changed and "yes" or "no")
         conf.env.VERSION = "%s (git commit: %s%s)" % (BASEVERSION, nam, dirty)
 
-    conf.env.append_unique('CFLAGS', ['-g', '-O0'])
-    conf.env.append_unique('CXXFLAGS', ['-g', '-O0'])
+    ##  conf.env.append_unique('CFLAGS', ['-g', '-O0'])
+    ## conf.env.append_unique('CXXFLAGS', ['-g', '-O0'])
 
     if conf.options.with_profiling:
         conf.env.with_profiling = True
