diff -up xmms2-0.8DrO_o/wscript.versionsanity xmms2-0.8DrO_o/wscript
--- xmms2-0.8DrO_o/wscript	2011-12-05 15:19:56.000000000 -0500
+++ xmms2-0.8DrO_o/wscript	2011-12-05 15:22:01.885508715 -0500
@@ -19,7 +19,7 @@ from waflib import Configure, Options, U
 from waftools.compiler_flags import compiler_flags
 from waftools import gittools
 
-BASEVERSION="0.8 DrO_o"
+BASEVERSION="0.8"
 APPNAME='xmms2'
 
 top = '.'
@@ -288,12 +288,13 @@ def configure(conf):
 
     nam,changed = gittools.get_info()
     conf.msg("git commit id", nam)
-    if conf.options.customversion:
-        conf.env.VERSION = "%s (%s + %s)" % (BASEVERSION, nam, conf.options.customversion)
-    else:
-        dirty = changed and "-dirty" or ""
-        conf.msg("uncommited changed", changed and "yes" or "no")
-        conf.env.VERSION = "%s (git commit: %s%s)" % (BASEVERSION, nam, dirty)
+    ## if conf.options.customversion:
+    ##    conf.env.VERSION = "%s (%s + %s)" % (BASEVERSION, nam, conf.options.customversion)
+    ## else:
+    ##    dirty = changed and "-dirty" or ""
+    ##    conf.msg("uncommited changed", changed and "yes" or "no")
+    ##    conf.env.VERSION = "%s (git commit: %s%s)" % (BASEVERSION, nam, dirty)
+    conf.env.VERSION = "%s" % (BASEVERSION)
 
     ##  conf.env.append_unique('CFLAGS', ['-g', '-O0'])
     ## conf.env.append_unique('CXXFLAGS', ['-g', '-O0'])
