diff -up xmms2-0.8DrO_o/AUTHORS.nomind xmms2-0.8DrO_o/AUTHORS
--- xmms2-0.8DrO_o/AUTHORS	2017-11-20 11:59:45.363988101 -0500
+++ xmms2-0.8DrO_o/AUTHORS	2017-11-20 11:59:58.267720781 -0500
@@ -218,12 +218,6 @@ N: Łukasz Michalik
 E: lmi@ift.uni.wroc.pl
 D: Bugfixes.
 
-N: Mind.In.A.Box / Stefan Poiss and Markus Hadwiger
-W: http://www.mindinabox.com/
-D: Supplied us with a great "intro" clip.
-D: Lament for Lost Dreams clip by mind.in.a.box
-D: Copyright (c) by Stefan Poiss and Markus Hadwiger
-
 N: Dave Moore
 E: wrex006@gmail.com
 D: missing method in C++ bindings.
diff -up xmms2-0.8DrO_o/checksums.nomind xmms2-0.8DrO_o/checksums
--- xmms2-0.8DrO_o/checksums	2017-11-20 12:01:10.770218782 -0500
+++ xmms2-0.8DrO_o/checksums	2017-11-20 12:03:19.831545082 -0500
@@ -63,7 +63,6 @@
 100644 blob 1c9606c6d58527d72f4354646f9c135da24da970	 doc/tutorial/vala/xmms2-client-glib.deps
 100644 blob 5beabceeda7ae335cf8633a03d0febe2f51198f0	 doc/tutorial/vala/xmms2-client-glib.vapi
 100644 blob e3677f8c3b4ccbb044a3f6f4f7e4bf6c5ddf6385	 doc/tutorial/vala/xmms2-client.vapi
-100644 blob 7dea2b304cdacc745218cd4187d9ab644842c798	 mind.in.a.box-lament_snipplet.ogg
 100644 blob 503aea3af178fd7ba02665706151f303703c3c95	 pixmaps/wscript
 100644 blob 66e6d6c205e0824d18692b40169d95a4593c9bf8	 pixmaps/xmms2-128.png
 100644 blob 0e2b17d6267581a9a49d7930af3c8a4f792ec0b3	 pixmaps/xmms2-16.png
diff -up xmms2-0.8DrO_o/src/xmms/medialib.c.nomind xmms2-0.8DrO_o/src/xmms/medialib.c
--- xmms2-0.8DrO_o/src/xmms/medialib.c	2017-11-20 12:02:10.520980955 -0500
+++ xmms2-0.8DrO_o/src/xmms/medialib.c	2017-11-20 12:02:31.489546559 -0500
@@ -367,7 +367,7 @@ xmms_medialib_init (xmms_playlist_t *pla
 
 		xmms_medialib_entry_new (session,
 		                         "file://" SHAREDDIR
-		                         "/mind.in.a.box-lament_snipplet.ogg",
+		                         "/unreal_dm-free.music.and.free.beer.ogg",
 		                         &error);
 		/* A default playlist containing that song has been created
 		 * with the mlib.
diff -up xmms2-0.8DrO_o/wscript.nomind xmms2-0.8DrO_o/wscript
--- xmms2-0.8DrO_o/wscript	2017-11-20 12:02:43.392299976 -0500
+++ xmms2-0.8DrO_o/wscript	2017-11-20 12:03:05.376844533 -0500
@@ -131,7 +131,7 @@ def build(bld):
             VERSION = bld.env.VERSION
             )
 
-    bld.install_files('${SHAREDDIR}', "mind.in.a.box-lament_snipplet.ogg")
+    bld.install_files('${SHAREDDIR}', "unreal_dm-free.music.and.free.beer.ogg")
 
     bld.add_post_fun(shutdown)
 
