Description: Include libmodplug/modplug.h
Author: Remi Vanicat <vanicat@debian.org>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=724487
Bug: https://bugs.xmms2.org/view.php?id=2574

--- a/src/plugins/modplug/modplug.c
+++ b/src/plugins/modplug/modplug.c
@@ -9,7 +9,7 @@
 #include "xmms/xmms_sample.h"
 #include "xmms/xmms_medialib.h"
 #include "xmms/xmms_log.h"
-#include <modplug.h>
+#include <libmodplug/modplug.h>
 
 #include <glib.h>
 #include <string.h>
