Description: Use pkg-config to build using smbclient
Author: Remi Vanicat <vanicat@debian.org>
Bug: https://bugs.xmms2.org/view.php?id=2573
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=726460

--- a/src/plugins/samba/wscript
+++ b/src/plugins/samba/wscript
@@ -1,8 +1,7 @@
 from waftools.plugin import plugin
 
 def plugin_configure(conf):
-    conf.check_cc(header_name="libsmbclient.h")
-    conf.check(lib="smbclient", uselib_store="smbclient")
+    conf.check_cfg(package="smbclient", uselib_store="smbclient", args="--cflags --libs")
 
 configure, build = plugin("samba", configure=plugin_configure,
         libs=["smbclient"])
