diff -up xmms2-0.8DrO_o/src/includepriv/xmmspriv/xmms_collection.h. xmms2-0.8DrO_o/src/includepriv/xmmspriv/xmms_collection.h
--- xmms2-0.8DrO_o/src/includepriv/xmmspriv/xmms_collection.h.	2020-02-28 10:15:31.578897880 -0500
+++ xmms2-0.8DrO_o/src/includepriv/xmmspriv/xmms_collection.h	2020-02-28 10:15:46.242585829 -0500
@@ -75,7 +75,7 @@ const gchar *xmms_collection_get_namespa
 gboolean xmms_collection_get_int_attr (xmmsv_coll_t *coll, const gchar *attrname, gint *val);
 gboolean xmms_collection_set_int_attr (xmmsv_coll_t *coll, const gchar *attrname, gint newval);
 
-GTree *xmms_collection_changed_msg_new (xmms_collection_changed_actions_t type, const gchar *plname, const gchar *namespace);
+GTree *xmms_collection_changed_msg_new (xmmsc_collection_changed_actions_t type, const gchar *plname, const gchar *namespace);
 void xmms_collection_changed_msg_send (xmms_coll_dag_t *colldag, GTree *dict);
 
 void bind_all_references (xmms_coll_dag_t *dag, xmmsv_coll_t *coll, xmmsv_coll_t *parent, void *udata);
diff -up xmms2-0.8DrO_o/src/include/xmmsc/xmmsc_idnumbers.h. xmms2-0.8DrO_o/src/include/xmmsc/xmmsc_idnumbers.h
--- xmms2-0.8DrO_o/src/include/xmmsc/xmmsc_idnumbers.h.	2020-02-28 10:15:18.475176735 -0500
+++ xmms2-0.8DrO_o/src/include/xmmsc/xmmsc_idnumbers.h	2020-02-28 10:15:22.698086869 -0500
@@ -193,7 +193,7 @@ typedef enum {
 	XMMS_COLLECTION_CHANGED_UPDATE,
 	XMMS_COLLECTION_CHANGED_RENAME,
 	XMMS_COLLECTION_CHANGED_REMOVE
-} xmms_collection_changed_actions_t;
+} xmmsc_collection_changed_actions_t;
 
 typedef enum {
 	XMMS_PLAYBACK_STATUS_STOP,
diff -up xmms2-0.8DrO_o/src/xmms/collection.c. xmms2-0.8DrO_o/src/xmms/collection.c
--- xmms2-0.8DrO_o/src/xmms/collection.c.	2020-02-28 10:15:56.804361068 -0500
+++ xmms2-0.8DrO_o/src/xmms/collection.c	2020-02-28 10:16:03.248223939 -0500
@@ -144,7 +144,7 @@ static void xmms_collection_client_sync
 #include "collection_ipc.c"
 
 GTree *
-xmms_collection_changed_msg_new (xmms_collection_changed_actions_t type,
+xmms_collection_changed_msg_new (xmmsc_collection_changed_actions_t type,
                                  const gchar *plname, const gchar *namespace)
 {
 	GTree *dict;
