diff -Nur orig/wscript mod/wscript
--- orig/wscript	2020-03-31 13:34:03.000000000 +0200
+++ mod/wscript	2020-03-31 13:47:24.693544492 +0200
@@ -131,7 +131,7 @@
             VERSION = bld.env.VERSION
             )
 
-    bld.install_files('${SHAREDDIR}', "unreal_dm-free.music.and.free.beer.ogg")
+    bld.install_files('${SHAREDDIR}', "mind.in.a.box-lament_snipplet.ogg")
 
     bld.add_post_fun(shutdown)
 
