diff -up megaglest-3.13.0/source/glest_game/facilities/components.h.orig megaglest-3.13.0/source/glest_game/facilities/components.h
--- megaglest-3.13.0/source/glest_game/facilities/components.h.orig	2019-09-24 15:37:33.553933198 -0300
+++ megaglest-3.13.0/source/glest_game/facilities/components.h	2019-09-24 15:37:47.178027199 -0300
@@ -281,8 +281,8 @@ public:
 
     void pushBackItem(string item, string translated_item="");
     void setItems(const vector<string> &items, const vector<string> translated_items=vector<string>());
-	void setSelectedItemIndex(int index, bool errorOnMissing=true);
-    void setSelectedItem(string item, bool errorOnMissing=true);
+	void setSelectedItemIndex(int index, bool errorOnMissing=false);
+    void setSelectedItem(string item, bool errorOnMissing=false);
     void setEditable(bool editable);
 
     bool hasItem(string item) const;
