diff -up megaglest-3.12.0/source/shared_lib/include/lua/lua_script.h.orig megaglest-3.12.0/source/shared_lib/include/lua/lua_script.h
--- megaglest-3.12.0/source/shared_lib/include/lua/lua_script.h.orig	2016-06-24 07:39:34.111506042 -0300
+++ megaglest-3.12.0/source/shared_lib/include/lua/lua_script.h	2016-06-24 07:39:39.649506393 -0300
@@ -13,6 +13,7 @@
 #define _SHARED_LUA_LUASCRIPT_H_
 
 #include <string>
+#define LUA_COMPAT_APIINTCASTS
 #include <lua.hpp>
 #include "vec.h"
 #include "xml_parser.h"
