From 412b37d0743b13e5a345a1abecf0e71955415436 Mon Sep 17 00:00:00 2001
From: titiger <titi_linux@megaglest.org>
Date: Sat, 5 Jun 2021 01:40:14 +0200
Subject: [PATCH] get rid of manual wx???Event creation

---
 source/g3d_viewer/main.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/source/g3d_viewer/main.cpp b/source/g3d_viewer/main.cpp
index db228ae85..2a9ec0c2d 100644
--- a/source/g3d_viewer/main.cpp
+++ b/source/g3d_viewer/main.cpp
@@ -564,7 +564,6 @@ void MainWindow::setupStartupSettings() {
 	}
 	renderer->init();
 
-	wxCommandEvent event;
 	onMenuRestartNoEvent();
 }
 
@@ -738,7 +737,6 @@ void MainWindow::onPaintNoEvent( ) {
 			anim				= 0.f;
 			particleLoopStart 	= resetParticleLoopStart;
 
-			wxCommandEvent event;
 			onMenuRestartNoEvent();
 		}
 	}
