diff -Naur orig/src/libreal/real_common.h patched/src/libreal/real_common.h
--- orig/src/libreal/real_common.h	2019-12-13 21:47:50.000000000 +0100
+++ patched/src/libreal/real_common.h	2020-08-20 00:44:20.853184126 +0200
@@ -75,10 +75,10 @@
 			 const char *const codec_name,
 			 const char *const codec_alternate);
 
-const decoder_info_t dec_info_realvideo;
+extern const decoder_info_t dec_info_realvideo;
 void *init_realvdec (xine_t *xine, const void *data);
 
-const decoder_info_t dec_info_realaudio;
+extern const decoder_info_t dec_info_realaudio;
 void *init_realadec (xine_t *xine, const void *data);
 
 #endif
