diff -up mozjs17.0.0/js/src/shell/jsoptparse.cpp.than mozjs17.0.0/js/src/shell/jsoptparse.cpp
--- mozjs17.0.0/js/src/shell/jsoptparse.cpp.than	2017-06-07 12:03:10.948047993 +0200
+++ mozjs17.0.0/js/src/shell/jsoptparse.cpp	2017-06-07 12:03:46.074725885 +0200
@@ -243,7 +243,7 @@ OptionParser::extractValue(size_t argc,
     char *eq = strchr(argv[*i], '=');
     if (eq) {
         *value = eq + 1;
-        if (value[0] == '\0')
+        if (*value[0] == '\0')
             return error("A value is required for option %.*s", eq - argv[*i], argv[*i]);
         return Okay;
     }
