diff -Nur orig/pyproject.toml mod/pyproject.toml
--- orig/pyproject.toml	2025-10-14 17:35:59.000000000 +0200
+++ mod/pyproject.toml	2025-10-25 11:59:45.375916802 +0200
@@ -1,7 +1,7 @@
 [build-system]
 build-backend = "setuptools.build_meta"
 requires = [
-  "setuptools>=78.0.2"
+  "setuptools>=73.0.1"
 ]
 
 [dependency-groups]
@@ -73,8 +73,6 @@
 keywords = [
   "i18n l10n gettext translate"
 ]
-license = "GPL-2.0-or-later"
-license-files = ["COPYING"]
 name = "translate-toolkit"
 requires-python = ">=3.9"
 
