diff -Naur orig/supertux2.desktop patched/supertux2.desktop
--- orig/supertux2.desktop	2015-12-20 15:56:34.000000000 +0100
+++ patched/supertux2.desktop	2015-12-24 15:35:06.841900709 +0100
@@ -1,6 +1,5 @@
 [Desktop Entry]
 Type=Application
-Version=1.0
 Name=SuperTux 2
 Name[en]=SuperTux 2
 GenericName=Platform Game
@@ -35,7 +34,7 @@
 Comment[pt]=Joga este clássico de plataformas 2D
 Comment[pt_BR]=Encarne o pinguin Tux neste jogo inspirado em clássicos de Pular&Correr
 Comment[hu]=Egy klasszikus 2D-s oldalnézeti játék
-Icon=supertux.png
+Icon=supertux
 Exec=supertux2
 Terminal=false
 StartupNotify=false
