diff -Naur a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
--- a/browser/app/profile/firefox.js	2023-12-11 21:42:03.000000000 +0100
+++ b/browser/app/profile/firefox.js	2023-12-19 16:16:44.505392322 +0100
@@ -19,6 +19,12 @@
   #endif
 #endif
 
+// Use LANG environment variable to choose locale
+pref("intl.locale.requested", "");
+
+// Use system-provided dictionaries
+pref("spellchecker.dictionary_path", "/usr/share/hunspell");
+
 pref("browser.hiddenWindowChromeURL", "chrome://browser/content/hiddenWindowMac.xhtml");
 
 // Set add-ons abuse report related prefs specific to Firefox Desktop.
@@ -58,7 +64,8 @@
 
 // Disable add-ons that are not installed by the user in all scopes by default.
 // See the SCOPE constants in AddonManager.jsm for values to use here.
-pref("extensions.autoDisableScopes", 15);
+pref("extensions.autoDisableScopes", 11);
+pref("extensions.showMismatchUI", false);
 // Scopes to scan for changes at startup.
 pref("extensions.startupScanScopes", 0);
 
@@ -271,7 +278,7 @@
 pref("browser.compactmode.show", false);
 
 // At startup, check if we're the default browser and prompt user if not.
-pref("browser.shell.checkDefaultBrowser", true);
+pref("browser.shell.checkDefaultBrowser", false);
 pref("browser.shell.shortcutFavicons",true);
 pref("browser.shell.mostRecentDateSetAsDefault", "");
 pref("browser.shell.skipDefaultBrowserCheckOnFirstRun", true);
@@ -380,7 +387,7 @@
 pref("browser.urlbar.accessibility.tabToSearch.announceResults", true);
 
 // Control autoFill behavior
-pref("browser.urlbar.autoFill", true);
+pref("browser.urlbar.autoFill", false);
 
 // Whether enabling adaptive history autofill.
 pref("browser.urlbar.autoFill.adaptiveHistory.enabled", false);
@@ -536,7 +543,7 @@
 // is longer than this.
 pref("browser.urlbar.maxCharsForSearchSuggestions", 100);
 
-pref("browser.urlbar.trimURLs", true);
+pref("browser.urlbar.trimURLs", false);
 
 #ifdef NIGHTLY_BUILD
 pref("browser.urlbar.trimHttps", true);
@@ -1182,7 +1189,7 @@
 // 2 and other values, nothing
 pref("browser.backspace_action", 2);
 
-pref("intl.regional_prefs.use_os_locales", false);
+pref("intl.regional_prefs.use_os_locales", true);
 
 // this will automatically enable inline spellchecking (if it is available) for
 // editable elements in HTML
@@ -1944,10 +1951,8 @@
 
 pref("browser.contentblocking.database.enabled", true);
 
-// Enable URL query stripping in Nightly.
-#ifdef NIGHTLY_BUILD
+// Enable URL query stripping.
 pref("privacy.query_stripping.enabled", true);
-#endif
 
 // Enable Strip on Share by default on desktop
 pref("privacy.query_stripping.strip_on_share.enabled", true);
