diff -Naur orig/browser/app/profile/firefox.js patched/browser/app/profile/firefox.js
--- orig/browser/app/profile/firefox.js	2018-08-30 18:44:17.000000000 +0200
+++ patched/browser/app/profile/firefox.js	2018-09-05 01:14:04.346716338 +0200
@@ -21,6 +21,13 @@
 #endif
 #endif
 
+pref("intl.locale.matchOS", true);
+pref("intl.locale.requested", "");
+pref("shell.checkDefaultApps", 0);
+
+// google NS_ERROR_NET_INADEQUATE_SECURITY
+// pref("network.http.spdy.enabled.http2", false);
+
 pref("browser.chromeURL","chrome://browser/content/");
 pref("browser.hiddenWindowChromeURL", "chrome://browser/content/hiddenWindow.xul");
 
@@ -55,7 +62,8 @@
 
 // Disable add-ons that are not installed by the user in all scopes by default.
 // See the SCOPE constants in AddonManager.jsm for values to use here.
-pref("extensions.autoDisableScopes", 15);
+pref("extensions.autoDisableScopes", 0);
+pref("extensions.shownSelectionUI", true);
 // Scopes to scan for changes at startup.
 pref("extensions.startupScanScopes", 0);
 
@@ -83,8 +91,8 @@
 pref("extensions.legacy.exceptions", "testpilot@cliqz.com,@testpilot-containers,jid1-NeEaf3sAHdKHPA@jetpack,@activity-streams,pulse@mozilla.com,@testpilot-addon,@min-vid,tabcentertest1@mozilla.com,snoozetabs@mozilla.com,speaktome@mozilla.com,hoverpad@mozilla.com");
 
 // Require signed add-ons by default
-pref("extensions.langpacks.signatures.required", true);
-pref("xpinstall.signatures.required", true);
+pref("extensions.langpacks.signatures.required", false);
+pref("xpinstall.signatures.required", false);
 pref("xpinstall.signatures.devInfoURL", "https://wiki.mozilla.org/Addons/Extension_Signing");
 
 // Dictionary download preference
@@ -150,7 +158,7 @@
 // If set to true, the Update Service will automatically download updates when
 // app updates are enabled per the app.update.enabled preference and if the user
 // can apply updates.
-pref("app.update.auto", true);
+pref("app.update.auto", false);
 
 // If set to true, the Update Service will present no UI for any event.
 pref("app.update.silent", false);
@@ -290,7 +298,7 @@
 #endif
 
 // Control autoFill behavior
-pref("browser.urlbar.autoFill", true);
+pref("browser.urlbar.autoFill", false);
 pref("browser.urlbar.speculativeConnect.enabled", true);
 
 // 0: Match anywhere (e.g., middle of words)
@@ -331,7 +339,7 @@
 pref("browser.urlbar.suggest.history.onlyTyped",    false);
 
 pref("browser.urlbar.formatting.enabled", true);
-pref("browser.urlbar.trimURLs", true);
+pref("browser.urlbar.trimURLs", false);
 
 pref("browser.urlbar.oneOffSearches", true);
 
@@ -741,11 +749,7 @@
 // 0 goes Back/Forward
 // 1 act like PgUp/PgDown
 // 2 and other values, nothing
-#ifdef UNIX_BUT_NOT_MAC
-pref("browser.backspace_action", 2);
-#else
 pref("browser.backspace_action", 0);
-#endif
 
 // this will automatically enable inline spellchecking (if it is available) for
 // editable elements in HTML
