Forwarded: not-needed
Origin: upstream, https://midnight-commander.org/changeset/3c3cb733c9374a867eedcdc9293f5bd21bc08ef5
Bug-Upstream: https://midnight-commander.org/ticket/4055
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=949700
Description: fixes browsing of .deb packages.

--- a/src/vfs/extfs/helpers/deb.in
+++ b/src/vfs/extfs/helpers/deb.in
@@ -67,8 +67,11 @@
 				$pathindex=7;
 			}
 			
 			$path=$_[$pathindex++];
+			# remove leading ./
+			$path=~s/^\.\///;
+			next if ($path eq '');
 			$arrow=$_[$pathindex++];
 			$link=$_[$pathindex++];
 			$link2=$_[$pathindex++];
 
