diff -Naur orig/src/vfs/extfs/helpers/iso9660.in patched/src/vfs/extfs/helpers/iso9660.in
--- orig/src/vfs/extfs/helpers/iso9660.in	2014-03-27 12:54:34.000000000 +0100
+++ patched/src/vfs/extfs/helpers/iso9660.in	2014-10-06 20:03:57.581303789 +0200
@@ -110,7 +110,7 @@
 
     ISOINFO_D_I="$(isoinfo -d -i "$1" 2>/dev/null)"
 
-    echo "$ISOINFO_D_I" | @GREP@ "UCS level 1\|NO Joliet" > /dev/null || ISOINFO="$ISOINFO $JOLIET_OPT"
+    echo "$ISOINFO_D_I" | @GREP@ "/NO Rock Ridge/" > /dev/null || ISOINFO="$ISOINFO -R"
 
     if [ $(echo "$ISOINFO_D_I" | @GREP@ "Joliet with UCS level 3 found" | wc -l) = 1 \
         -a $(echo "$ISOINFO_D_I" | @GREP@ "NO Rock Ridge" | wc -l) = 1 ] ; then
@@ -133,9 +133,9 @@
     $ISOINFO -l -i "$1" 2>/dev/null | @AWK@ -v SEMICOLON=$SEMICOLON '
 BEGIN {
   dir="";
-  # Pattern to match 8 first fields.
+  # Pattern to match 9 first fields.
   rx = "[^ 	]+[ 	]+";
-  rx = "^" rx rx rx rx rx rx rx rx;
+  rx = "^" rx rx rx rx rx rx rx rx rx;
   irx = "^\\[ *-?[0-9]* *[0-9]+\\] +";
 }
 /^$/ { next }
