diff -Naur orig/desktop/gnome-schedule.desktop.in.in patched/desktop/gnome-schedule.desktop.in.in
--- orig/desktop/gnome-schedule.desktop.in.in	2010-11-03 11:39:09.000000000 +0100
+++ patched/desktop/gnome-schedule.desktop.in.in	2016-02-19 01:48:26.904891972 +0100
@@ -1,9 +1,9 @@
 [Desktop Entry]
-_Name=Scheduled tasks
+_Name=GNOME Schedule
 _Comment=Manage your system tasks
-Categories=System;
+Categories=System;GNOME;
 Icon=@prefix@/share/pixmaps/gnome-schedule/gnome-schedule.svg
-Exec=@prefix@/bin/gnome-schedule
+Exec=gnome-schedule
 Type=Application
 StartupNotify=true
 Terminal=false
