diff -Naur orig/uhu-pkg patched/uhu-pkg
--- orig/uhu-pkg	2005-08-31 16:08:54.000000000 +0200
+++ patched/uhu-pkg	2021-08-23 23:56:58.114819989 +0200
@@ -89,12 +89,12 @@
 		fi
 		if [ -f /var/lib/dpkg/info/"$pkg".groups ]; then
 			while read name gid rest; do
-				groupadd -q -g $gid $rest $name
+				groupadd -q -r -g $gid $rest $name
 			done < /var/lib/dpkg/info/"$pkg".groups
 		fi
 		if [ -f /var/lib/dpkg/info/"$pkg".users ]; then
 			while read name uid gid rest; do
-				useradd -q -u $uid -g $gid $rest $name
+				useradd -q -r -u $uid -g $gid $rest $name
 			done < /var/lib/dpkg/info/"$pkg".users
 		fi
 	fi
