diff -Naur gc-7.2-orig/doc/gc.man gc-7.2/doc/gc.man
--- gc-7.2-orig/doc/gc.man	2012-05-11 19:09:00.000000000 +0200
+++ gc-7.2/doc/gc.man	2012-09-06 14:38:33.000000000 +0200
@@ -1,4 +1,4 @@
-.TH GC_MALLOC 1L "2 October 2003"
+.TH GC_MALLOC 3L "2 October 2003"
 .SH NAME
 GC_malloc, GC_malloc_atomic, GC_free, GC_realloc, GC_enable_incremental, GC_register_finalizer, GC_malloc_ignore_off_page, GC_malloc_atomic_ignore_off_page, GC_set_warn_proc \- Garbage collecting malloc replacement
 .SH SYNOPSIS
