diff -Naur orig/src/qt5ct/main.cpp patched/src/qt5ct/main.cpp
--- orig/src/qt5ct/main.cpp	2016-09-05 08:37:49.000000000 +0200
+++ patched/src/qt5ct/main.cpp	2017-01-09 10:00:00.566434239 +0100
@@ -53,6 +53,7 @@
 
     //checking environment
     QStringList errorMessages;
+	// QProcess process;
     QProcessEnvironment env = QProcessEnvironment::systemEnvironment();
 
     if(env.contains("QT_STYLE_OVERRIDE"))
@@ -62,8 +63,11 @@
 
     if(env.value("QT_QPA_PLATFORMTHEME") != "qt5ct")
     {
-        errorMessages << app.translate("main", "The <b>QT_QPA_PLATFORMTHEME</b> environment "
-                                               "variable is not set correctly");
+    //    errorMessages << app.translate("main", "The <b>QT_QPA_PLATFORMTHEME</b> environment "
+    //                                           "variable is not set correctly");
+	//	env.insert("QT_QPA_PLATFORMTHEME", "qt5ct");
+	//	process.setProcessEnvironment(env);
+		qputenv("QT_QPA_PLATFORMTHEME", "qt5ct");
     }
 
     if(!errorMessages.isEmpty())
