diff -Naur a/src/qt5ct/mainwindow.cpp b/src/qt5ct/mainwindow.cpp
--- a/src/qt5ct/mainwindow.cpp	2023-02-28 19:40:15.000000000 +0100
+++ b/src/qt5ct/mainwindow.cpp	2023-03-01 20:58:54.660738257 +0100
@@ -121,14 +121,12 @@
 
     if(!env.contains("QT_QPA_PLATFORMTHEME"))
     {
-        m_errors << tr("The <b>QT_QPA_PLATFORMTHEME</b> environment variable is not set (required values: <b>qt5ct</b> or <b>qt6ct</b>).");
+        qputenv("QT_QPA_PLATFORMTHEME", "qt5ct");
     }
     else if(env.value("QT_QPA_PLATFORMTHEME") != QStringLiteral("qt5ct") &&
             env.value("QT_QPA_PLATFORMTHEME") != QStringLiteral("qt6ct"))
     {
-        m_errors << tr("The <b>QT_QPA_PLATFORMTHEME</b> environment variable is not set correctly "
-                       "(current value: <b>%1</b>, required values: <b>qt5ct</b> or <b>qt6ct</b>).")
-                    .arg(env.value("QT_QPA_PLATFORMTHEME"));
+        qputenv("QT_QPA_PLATFORMTHEME", "qt5ct");
     }
 
     if(!QStyleFactory::keys().contains("qt5ct-style"))
