diff -Naur orig/src/qt5ct/mainwindow.cpp patched/src/qt5ct/mainwindow.cpp
--- orig/src/qt5ct/mainwindow.cpp	2018-08-22 19:13:32.000000000 +0200
+++ patched/src/qt5ct/mainwindow.cpp	2018-12-18 00:28:46.692501460 +0100
@@ -119,13 +119,11 @@
 
     if(!env.contains("QT_QPA_PLATFORMTHEME"))
     {
-        m_errors << tr("The <b>QT_QPA_PLATFORMTHEME</b> environment variable is not set (required value: <b>qt5ct</b>).");
+        qputenv("QT_QPA_PLATFORMTHEME", "qt5ct");
     }
     else if(env.value("QT_QPA_PLATFORMTHEME") != "qt5ct")
     {
-        m_errors << tr("The <b>QT_QPA_PLATFORMTHEME</b> environment variable is not set correctly "
-                       "(current value: <b>%1</b>, required value: <b>qt5ct</b>).")
-                    .arg(env.value("QT_QPA_PLATFORMTHEME"));
+        qputenv("QT_QPA_PLATFORMTHEME", "qt5ct");
     }
 
     if(!QStyleFactory::keys().contains("qt5ct-style"))
