diff -Naur orig/perlmods/install_order patched/perlmods/install_order
--- orig/perlmods/install_order	2014-03-04 22:41:56.000000000 +0100
+++ patched/perlmods/install_order	2016-07-09 03:53:30.264206215 +0200
@@ -1,16 +1 @@
 # Modules installed in this order
-default:
-Test-Simple
-Perl-OSType
-Module-Implementation
-Module-Metadata
-version
-Module-Build
-Module-Runtime
-parent
-Try-Tiny
-Params-Validate
-Math-Calc-Units
-Class-Accessor
-Config-Tiny
-Nagios-Plugin
