diff -Naur orig/plugins/check_mysql.c patched/plugins/check_mysql.c
--- orig/plugins/check_mysql.c	2014-03-04 22:41:57.134739579 +0100
+++ patched/plugins/check_mysql.c	2017-10-03 01:21:22.683964629 +0200
@@ -41,7 +41,7 @@
 #include "utils_base.h"
 #include "netutils.h"
 
-#include <mysql.h>
+#include <server/mysql.h>
 #include <mysqld_error.h>
 #include <errmsg.h>
 
diff -Naur orig/plugins/check_mysql_query.c patched/plugins/check_mysql_query.c
--- orig/plugins/check_mysql_query.c	2014-03-04 22:41:57.134739579 +0100
+++ patched/plugins/check_mysql_query.c	2017-10-03 01:12:17.618560276 +0200
@@ -38,7 +38,7 @@
 #include "utils_base.h"
 #include "netutils.h"
 
-#include <mysql.h>
+#include <server/mysql.h>
 #include <errmsg.h>
 
 char *db_user = NULL;
