diff -Naur orig/common.c mod/common.c
--- orig/common.c	2009-07-17 01:18:57.000000000 +0200
+++ mod/common.c	2012-03-31 16:12:17.000000000 +0200
@@ -251,11 +251,11 @@
 		int start = 1; // first frame
 		bool isStart = true;
 
-		for (cMark *m = marks.First(); m; m = marks.GetNext(m->position)) {
+		for (cMark *m = marks.First(); m; m = marks.GetNext(m->Position())) {
 			if (isStart)
-				start = m->position;
+				start = m->Position();
 			else
-				length += (double)(m->position - start + 1 + diffIFrame) / (60 * FramesPerSecond); // [min]
+				length += (double)(m->Position() - start + 1 + diffIFrame) / (60 * FramesPerSecond); // [min]
 
 			isStart = !isStart;
 		}
diff -Naur orig/display.c mod/display.c
--- orig/display.c	2009-08-18 23:24:36.000000000 +0200
+++ mod/display.c	2012-03-31 16:02:46.000000000 +0200
@@ -1235,17 +1235,17 @@
 
 	case tRecordingPriority:
 		return mRecording != NULL
-		       ? (cxType)mRecording->priority
+		       ? (cxType)mRecording->Priority()
 		       : (cxType)false;
 
 	case tRecordingLifetime:
 		return mRecording != NULL
-		       ? (cxType)mRecording->lifetime
+		       ? (cxType)mRecording->Lifetime()
 		       : (cxType)false;
 
 	case tRecordingDateTime:
 		return mRecording != NULL
-		       ? (cxType)TimeType(mRecording->start, Token.Attrib.Text)
+		       ? (cxType)TimeType(mRecording->Start(), Token.Attrib.Text)
 		       : (cxType)false;
 
 	case tRecordingTitle:
diff -Naur orig/render.c mod/render.c
--- orig/render.c	2009-08-01 16:15:05.000000000 +0200
+++ mod/render.c	2012-03-31 16:07:06.000000000 +0200
@@ -599,14 +599,14 @@
 		if (Marks) {
 			bool start = true;
 			for (const cMark *m = Marks->First(); m; m = Marks->Next(m)) {
-				txPoint pt(Pos.x + m->position * Size.w / Total, Pos.y);
+				txPoint pt(Pos.x + m->Position() * Size.w / Total, Pos.y);
 				if (Selected && start) {
 					const cMark *m2 = Marks->Next(m);
 					DrawRectangle(txPoint(pt.x, Pos.y + Size.h / 3), 
-					              txSize(((m2 ? m2->position : Total) - m->position) 
+					              txSize(((m2 ? m2->Position() : Total) - m->Position()) 
 					              * Size.w / Total + 1, Size.h - Size.h * 2 / 3 + 1), Selected);
 				}
-				DrawMark(pt, Size, start, m->position == Current, false, Mark, Cur);
+				DrawMark(pt, Size, start, m->Position() == Current, false, Mark, Cur);
 				start = !start;
 			}
 		}
@@ -616,15 +616,15 @@
 		if (Marks) {
 			bool start = true;
 			for (const cMark *m = Marks->First(); m; m = Marks->Next(m)) {
-				txPoint pt(Pos.x, Pos.y + m->position * Size.h / Total);
+				txPoint pt(Pos.x, Pos.y + m->Position() * Size.h / Total);
 				if (Selected && start) {
 					const cMark *m2 = Marks->Next(m);
 					DrawRectangle(txPoint(Pos.x + Size.w / 3, pt.y), 
 					              txSize(Size.w - Size.w * 2 / 3 + 1,
-					              ((m2 ? m2->position : Total) - m->position)
+					              ((m2 ? m2->Position() : Total) - m->Position())
 					              * Size.h / Total + 1), Selected);
 				}
-				DrawMark(pt, Size, start, m->position == Current, true, Mark, Cur);
+				DrawMark(pt, Size, start, m->Position() == Current, true, Mark, Cur);
 				start = !start;
 			}
 		}
diff -Naur orig/status.c mod/status.c
--- orig/status.c	2009-07-19 23:55:51.000000000 +0200
+++ mod/status.c	2012-03-31 16:07:58.000000000 +0200
@@ -316,7 +316,7 @@
 
 	case tReplayDateTime:
 		return mReplay != NULL
-		       ? (cxType)TimeType(mReplay->start, Token.Attrib.Text)
+		       ? (cxType)TimeType(mReplay->Start(), Token.Attrib.Text)
 		       : (cxType)false;
 
 	case tReplayShortText:
