diff -Naur orig/xml/display.c mod/xml/display.c
--- orig/xml/display.c	2011-10-29 14:48:04.000000000 +0200
+++ mod/xml/display.c	2013-05-05 20:04:40.000000000 +0200
@@ -1,6 +1,6 @@
 //								-*- c++ -*-
 
-#include "xml/display.h"
+#include "display.h"
 
 static const std::string DisplayNames[] = 
 	{ "channelInfo", "channelSmall", "volume", "audioTracks", "message", "replayInfo", 
diff -Naur orig/xml/display.h mod/xml/display.h
--- orig/xml/display.h	2011-10-29 14:48:30.000000000 +0200
+++ mod/xml/display.h	2013-05-05 20:04:25.000000000 +0200
@@ -3,7 +3,7 @@
 #ifndef VDR_TEXT2SKIN_DISPLAY_H 
 #define VDR_TEXT2SKIN_DISPLAY_H 
 
-#include "xml/object.h"
+#include "object.h"
 #include <string>
 #include <map>
 
diff -Naur orig/xml/function.c mod/xml/function.c
--- orig/xml/function.c	2011-11-05 22:05:37.000000000 +0100
+++ mod/xml/function.c	2013-05-05 20:03:53.000000000 +0200
@@ -1,9 +1,9 @@
 //								-*- c++ -*-
 
-#include "xml/function.h"
-#include "render.h"
-#include "bitmap.h"
-#include "common.h"
+#include "function.h"
+#include "../render.h"
+#include "../bitmap.h"
+#include "../common.h"
 #include <vdr/plugin.h>
 #include <vdr/tools.h>
 
diff -Naur orig/xml/function.h mod/xml/function.h
--- orig/xml/function.h	2011-10-29 14:48:28.000000000 +0200
+++ mod/xml/function.h	2013-05-05 20:03:24.000000000 +0200
@@ -3,7 +3,7 @@
 #ifndef VDR_TEXT2SKIN_XML_FUNCTION_H
 #define VDR_TEXT2SKIN_XML_FUNCTION_H
 
-#include "xml/string.h"
+#include "string.h"
 #include <string>
 
 #define STRING    0x01000000
diff -Naur orig/xml/object.c mod/xml/object.c
--- orig/xml/object.c	2011-10-29 14:48:01.000000000 +0200
+++ mod/xml/object.c	2013-05-05 20:02:57.000000000 +0200
@@ -1,8 +1,8 @@
 //								-*- c++ -*-
 
-#include "xml/object.h"
-#include "render.h"
-#include "font.h"
+#include "object.h"
+#include "../render.h"
+#include "../font.h"
 
 static const std::string ObjectNames[] =
 	{ "image", "text", "marquee", "blink", "rectangle", "ellipse", "slope", "progress", 
diff -Naur orig/xml/object.h mod/xml/object.h
--- orig/xml/object.h	2011-10-29 14:48:26.000000000 +0200
+++ mod/xml/object.h	2013-05-05 20:02:28.000000000 +0200
@@ -3,8 +3,8 @@
 #ifndef VDR_TEXT2SKIN_XML_OBJECT_H
 #define VDR_TEXT2SKIN_XML_OBJECT_H
 
-#include "xml/string.h"
-#include "xml/function.h"
+#include "string.h"
+#include "function.h"
 #include <vdr/osd.h>
 #include <vector>
 #include <string>
diff -Naur orig/xml/parser.c mod/xml/parser.c
--- orig/xml/parser.c	2011-11-05 22:16:44.000000000 +0100
+++ mod/xml/parser.c	2013-05-05 20:01:57.000000000 +0200
@@ -1,9 +1,9 @@
 //								-*- c++ -*-
 
-#include "xml/parser.h"
-#include "xml/xml.h"
-#include "xml/skin.h"
-#include "common.h"
+#include "parser.h"
+#include "xml.h"
+#include "skin.h"
+#include "../common.h"
 #include <vdr/tools.h>
 #include <stdio.h>
 #include <vector>
diff -Naur orig/xml/skin.c mod/xml/skin.c
--- orig/xml/skin.c	2011-10-29 14:47:58.000000000 +0200
+++ mod/xml/skin.c	2013-05-05 20:01:21.000000000 +0200
@@ -1,7 +1,7 @@
 //								-*- c++ -*-
 
-#include "xml/skin.h"
-#include "i18n.h"
+#include "skin.h"
+#include "../i18n.h"
 #include <vdr/tools.h>
 #include <vdr/config.h>
 
diff -Naur orig/xml/skin.h mod/xml/skin.h
--- orig/xml/skin.h	2011-10-29 14:48:22.000000000 +0200
+++ mod/xml/skin.h	2013-05-05 20:00:47.000000000 +0200
@@ -3,7 +3,7 @@
 #ifndef VDR_TEXT2SKIN_XML_SKIN_H 
 #define VDR_TEXT2SKIN_XML_SKIN_H 
 
-#include "xml/display.h"
+#include "display.h"
 #include <vdr/osd.h>
 #include <map>
 #include <string>
diff -Naur orig/xml/string.c mod/xml/string.c
--- orig/xml/string.c	2011-10-29 14:47:56.000000000 +0200
+++ mod/xml/string.c	2013-05-05 19:58:54.000000000 +0200
@@ -1,7 +1,7 @@
 //								-*- c++ -*-
 
-#include "xml/string.h"
-#include "render.h"
+#include "string.h"
+#include "../render.h"
 
 static const char *Tokens[__COUNT_TOKEN__] = {
 	"DateTime", "FreeDiskSpace", "AudioTrack", "AudioChannel",
diff -Naur orig/xml/string.h mod/xml/string.h
--- orig/xml/string.h	2011-10-29 14:48:21.000000000 +0200
+++ mod/xml/string.h	2013-05-05 19:59:34.000000000 +0200
@@ -3,7 +3,7 @@
 #ifndef VDR_TEXT2SKIN_XML_STRING_H
 #define VDR_TEXT2SKIN_XML_STRING_H
 
-#include "xml/type.h"
+#include "type.h"
 #include <string>
 #include <vector>
 
diff -Naur orig/xml/type.c mod/xml/type.c
--- orig/xml/type.c	2011-10-29 14:47:52.000000000 +0200
+++ mod/xml/type.c	2013-05-05 20:00:02.000000000 +0200
@@ -1,3 +1,3 @@
 //								-*- c++ -*-
 
-#include "xml/type.h"
+#include "type.h"
