diff -Nur orig/bootStrap.bash mod/bootStrap.bash
--- orig/bootStrap.bash	2017-01-06 08:30:06.000000000 +0100
+++ mod/bootStrap.bash	2017-01-21 12:54:30.042734305 +0100
@@ -66,7 +66,7 @@
         fi
         cd $BUILDDIR 
         cmake $COMPILER $PKG $FAKEROOT $QT_FLAVOR -DCMAKE_EDIT_COMMAND=vim $INSTALL_PREFIX $EXTRA $BUILD_QUIRKS $ASAN $DEBUG -G "$BUILDER" $SOURCEDIR || fail cmakeZ
-        make  $PARAL >& /tmp/log$BUILDDIR || fail "make, result in /tmp/log$BUILDDIR"
+        make  $PARAL
 	if  [ "x$PKG" != "x" ] ; then
           $FAKEROOT_COMMAND make package DESTDIR=$FAKEROOT_DIR/tmp || fail package
 	fi
@@ -257,7 +257,7 @@
 
 export TOP=$PWD
 export POSTFIX=""
-export FAKEROOT_DIR=$PWD/install
+export FAKEROOT_DIR=$UB_INSTALLDIR
 export PARAL="$O_PARAL"
 if [ "x$external_libass" = "x1" ]; then
     export EXTRA_CMAKE_DEFS="-DUSE_EXTERNAL_LIBASS=true $EXTRA_CMAKE_DEFS"
