diff -Nur orig/avidemux2.desktop mod/avidemux2.desktop
--- orig/avidemux2.desktop	2018-06-04 07:43:35.000000000 +0200
+++ mod/avidemux2.desktop	2018-06-10 15:28:03.371936261 +0200
@@ -2,7 +2,7 @@
 Name=Avidemux
 GenericName=Video Editor
 Comment=Multiplatform video editor
-Exec=avidemux3_qt5
+Exec=avidemux
 MimeType=video/mpeg;video/quicktime;video/x-msvideo;video/x-anim;audio/x-mp3;audio/x-mp2;
 Icon=avidemux
 Terminal=false
