diff -Naur orig/avidemux/qt4/ADM_UIs/src/T_RubberControl.cpp patched/avidemux/qt4/ADM_UIs/src/T_RubberControl.cpp
--- orig/avidemux/qt4/ADM_UIs/src/T_RubberControl.cpp	2020-06-03 23:49:23.112492056 +0200
+++ patched/avidemux/qt4/ADM_UIs/src/T_RubberControl.cpp	2020-06-03 23:48:19.572335992 +0200
@@ -8,6 +8,7 @@
 ***************************************************************************///
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QRubberBand>
 #include <QBoxLayout>
 #include <QSizeGrip>
