diff -urN vim61.orig/src/feature.h vim61/src/feature.h
--- vim61.orig/src/feature.h	2002-03-09 16:17:30.000000000 +0100
+++ vim61/src/feature.h	2002-12-01 15:47:53.000000000 +0100
@@ -826,12 +826,12 @@
 /*
  * SYS_VIMRC_FILE	Name of the system-wide .vimrc file.
  */
-/* #define SYS_VIMRC_FILE	"/etc/vimrc" */
+#define SYS_VIMRC_FILE	"/etc/vimrc"
 
 /*
  * SYS_GVIMRC_FILE	Name of the system-wide .gvimrc file.
  */
-/* #define SYS_GVIMRC_FILE	"/etc/gvimrc" */
+#define SYS_GVIMRC_FILE	"/etc/gvimrc"
 
 /*
  * DFLT_HELPFILE	Name of the help file.
