diff -Nur orig/src/feature.h mod/src/feature.h
--- orig/src/feature.h	2019-12-16 17:10:33.000000000 +0100
+++ mod/src/feature.h	2019-12-16 18:40:16.335457454 +0100
@@ -814,12 +814,12 @@
 /*
  * SYS_VIMRC_FILE	Name of the system-wide .vimrc file.
  */
-// #define SYS_VIMRC_FILE	"/etc/vimrc"
+#define SYS_VIMRC_FILE	"/etc/vimrc"
 
 /*
  * SYS_GVIMRC_FILE	Name of the system-wide .gvimrc file.
  */
-// #define SYS_GVIMRC_FILE	"/etc/gvimrc"
+#define SYS_GVIMRC_FILE	"/etc/gvimrc"
 
 /*
  * DFLT_HELPFILE	Name of the help file.
