diff -Nur orig/tuxguitar.sh mod/tuxguitar.sh
--- orig/tuxguitar.sh	2018-08-02 23:53:32.000000000 +0200
+++ mod/tuxguitar.sh	2019-11-07 21:10:01.000000000 +0100
@@ -13,6 +13,7 @@
 	[ ! -f "${JAVA}" ] && JAVA="${JAVA_HOME}/bin/java"
 	[ ! -f "${JAVA}" ] && JAVA="java"
 fi
+JAVA=/usr/lib/java/jre/bin/java
 ##LIBRARY_PATH
 LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:lib/
 LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/lib
