diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2023-12-16 02:16:31.000000000 +0100
+++ b/CMakeLists.txt	2023-12-22 01:24:42.947065219 +0100
@@ -2,7 +2,7 @@
 project(kdegraphics-thumbnailers)
 
 set(QT_MIN_VERSION "5.15.2")
-set(KF_MIN_VERSION "5.92.0")
+set(KF_MIN_VERSION "5.91.0")
 set(KDE_COMPILERSETTINGS_LEVEL "5.82")
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
