diff -Nur orig/panels/color/cinnamon-color-panel.desktop.in.in mod/panels/color/cinnamon-color-panel.desktop.in.in
--- orig/panels/color/cinnamon-color-panel.desktop.in.in	2021-11-15 14:01:48.000000000 +0100
+++ mod/panels/color/cinnamon-color-panel.desktop.in.in	2021-11-17 19:47:04.028176577 +0100
@@ -5,7 +5,7 @@
 Type=Application
 StartupNotify=true
 Categories=GTK;Settings;X-Cinnamon-Settings-Panel;HardwareSettings
-OnlyShowIn=X-Cinnamon;
+OnlyShowIn=Cinnamon;
 X-Cinnamon-Settings-Panel=color
 # Translators: those are keywords for the color control-center panel
 _Keywords=Color;ICC;Profile;Calibrate;Printer;Display;
diff -Nur orig/panels/display/cinnamon-display-panel.desktop.in.in mod/panels/display/cinnamon-display-panel.desktop.in.in
--- orig/panels/display/cinnamon-display-panel.desktop.in.in	2021-11-15 14:01:48.000000000 +0100
+++ mod/panels/display/cinnamon-display-panel.desktop.in.in	2021-11-17 19:47:28.796403630 +0100
@@ -5,7 +5,7 @@
 Type=Application
 StartupNotify=true
 Categories=GTK;Settings;HardwareSettings;X-Cinnamon-Settings-Panel;
-OnlyShowIn=X-Cinnamon;
+OnlyShowIn=Cinnamon;
 X-Cinnamon-Settings-Panel=display
 # Translators: those are keywords for the display control-center panel
 _Keywords=Panel;Projector;xrandr;Screen;Resolution;Refresh;
diff -Nur orig/panels/network/cinnamon-network-panel.desktop.in.in mod/panels/network/cinnamon-network-panel.desktop.in.in
--- orig/panels/network/cinnamon-network-panel.desktop.in.in	2021-11-15 14:01:48.000000000 +0100
+++ mod/panels/network/cinnamon-network-panel.desktop.in.in	2021-11-17 19:47:43.948542537 +0100
@@ -5,7 +5,7 @@
 Type=Application
 StartupNotify=true
 Categories=GTK;Settings;HardwareSettings;X-Cinnamon-Settings-Panel;
-OnlyShowIn=X-Cinnamon;
+OnlyShowIn=Cinnamon;
 X-Cinnamon-Settings-Panel=network
 # Translators: those are keywords for the network control-center panel
 _Keywords=Network;Wireless;IP;LAN;Proxy;Internet;WiFi;
