diff -Naur orig/panels/color/cinnamon-color-panel.desktop.in.in mod/panels/color/cinnamon-color-panel.desktop.in.in
--- orig/panels/color/cinnamon-color-panel.desktop.in.in	2015-05-19 17:23:24.000000000 +0200
+++ mod/panels/color/cinnamon-color-panel.desktop.in.in	2015-06-02 14:41:15.064190158 +0200
@@ -4,8 +4,8 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GTK;Settings;X-Cinnamon-Settings-Panel;HardwareSettings
-OnlyShowIn=X-Cinnamon;
+Categories=GTK;Settings;X-Cinnamon-Settings-Panel;HardwareSettings;
+OnlyShowIn=Cinnamon;
 X-Cinnamon-Settings-Panel=color
 # Translators: those are keywords for the color control-center panel
 _Keywords=Color;ICC;Profile;Calibrate;Printer;Display;
diff -Naur orig/panels/display/cinnamon-display-panel.desktop.in.in mod/panels/display/cinnamon-display-panel.desktop.in.in
--- orig/panels/display/cinnamon-display-panel.desktop.in.in	2015-05-19 17:23:24.000000000 +0200
+++ mod/panels/display/cinnamon-display-panel.desktop.in.in	2015-06-02 14:42:05.896074329 +0200
@@ -5,7 +5,7 @@
 Type=Application
 StartupNotify=true
 Categories=GTK;Settings;HardwareSettings;X-Cinnamon-Settings-Panel;
-OnlyShowIn=X-Cinnamon;
+OnlyShowIn=Cinnamon;
 X-Cinnamon-Settings-Panel=display
 # Translators: those are keywords for the display control-center panel
 _Keywords=Panel;Projector;xrandr;Screen;Resolution;Refresh;
diff -Naur orig/panels/network/cinnamon-network-panel.desktop.in.in mod/panels/network/cinnamon-network-panel.desktop.in.in
--- orig/panels/network/cinnamon-network-panel.desktop.in.in	2015-05-19 17:23:24.000000000 +0200
+++ mod/panels/network/cinnamon-network-panel.desktop.in.in	2015-06-02 14:42:20.508041028 +0200
@@ -5,7 +5,7 @@
 Type=Application
 StartupNotify=true
 Categories=GTK;Settings;HardwareSettings;X-Cinnamon-Settings-Panel;
-OnlyShowIn=X-Cinnamon;
+OnlyShowIn=Cinnamon;
 X-Cinnamon-Settings-Panel=network
 # Translators: those are keywords for the network control-center panel
 _Keywords=Network;Wireless;IP;LAN;Proxy;
diff -Naur orig/panels/region/cinnamon-region-panel.desktop.in.in mod/panels/region/cinnamon-region-panel.desktop.in.in
--- orig/panels/region/cinnamon-region-panel.desktop.in.in	2015-05-19 17:23:24.000000000 +0200
+++ mod/panels/region/cinnamon-region-panel.desktop.in.in	2015-06-02 14:42:33.892010565 +0200
@@ -5,7 +5,7 @@
 Type=Application
 StartupNotify=true
 Categories=GTK;Settings;DesktopSettings;X-Cinnamon-Settings-Panel;X-Cinnamon-PersonalSettings
-OnlyShowIn=X-Cinnamon;
+OnlyShowIn=Cinnamon;
 X-Cinnamon-Settings-Panel=region
 # Translators: those are keywords for the region control-center panel
 _Keywords=Language;Layout;Keyboard;
diff -Naur orig/panels/wacom/cinnamon-wacom-panel.desktop.in.in mod/panels/wacom/cinnamon-wacom-panel.desktop.in.in
--- orig/panels/wacom/cinnamon-wacom-panel.desktop.in.in	2015-05-19 17:23:24.000000000 +0200
+++ mod/panels/wacom/cinnamon-wacom-panel.desktop.in.in	2015-06-02 14:43:53.235829764 +0200
@@ -6,7 +6,7 @@
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-Cinnamon-Settings-Panel;
 X-Cinnamon-Settings-Panel=wacom
-OnlyShowIn=X-Cinnamon;
+OnlyShowIn=Cinnamon;
 # Translators: those are keywords for the wacom tablet control-center panel
 _Keywords=Tablet;Wacom;Stylus;Eraser;Mouse;
 Name=Graphics Tablet
