diff -Naur a/panels/color/cinnamon-color-panel.desktop b/panels/color/cinnamon-color-panel.desktop
--- a/panels/color/cinnamon-color-panel.desktop	2022-07-04 09:39:25.000000000 +0200
+++ b/panels/color/cinnamon-color-panel.desktop	2022-07-04 15:47:03.171403337 +0200
@@ -8,7 +8,7 @@
 OnlyShowIn=X-Cinnamon;
 X-Cinnamon-Settings-Panel=color
 # Translators: those are keywords for the color control-center panel
-_Keywords=Color;ICC;Profile;Calibrate;Printer;Display;
+Keywords=Color;ICC;Profile;Calibrate;Printer;Display;
 Name=Color
 Name[am]=ቀለም
 Name[ar]=اللون
diff -Naur a/panels/display/cinnamon-display-panel.desktop b/panels/display/cinnamon-display-panel.desktop
--- a/panels/display/cinnamon-display-panel.desktop	2022-07-04 09:39:25.000000000 +0200
+++ b/panels/display/cinnamon-display-panel.desktop	2022-07-04 15:46:57.483386510 +0200
@@ -8,7 +8,7 @@
 OnlyShowIn=X-Cinnamon;
 X-Cinnamon-Settings-Panel=display
 # Translators: those are keywords for the display control-center panel
-_Keywords=Panel;Projector;xrandr;Screen;Resolution;Refresh;
+Keywords=Panel;Projector;xrandr;Screen;Resolution;Refresh;
 Name=Display
 Name[am]=ማሳያ
 Name[ar]=العرض
diff -Naur a/panels/network/cinnamon-network-panel.desktop b/panels/network/cinnamon-network-panel.desktop
--- a/panels/network/cinnamon-network-panel.desktop	2022-07-04 09:39:25.000000000 +0200
+++ b/panels/network/cinnamon-network-panel.desktop	2022-07-04 15:47:15.163438828 +0200
@@ -8,7 +8,7 @@
 OnlyShowIn=X-Cinnamon;
 X-Cinnamon-Settings-Panel=network
 # Translators: those are keywords for the network control-center panel
-_Keywords=Network;Wireless;IP;LAN;Proxy;Internet;WiFi;
+Keywords=Network;Wireless;IP;LAN;Proxy;Internet;WiFi;
 Name=Network
 Name[am]=ኔትዎርክ
 Name[ar]=الشبكة
diff -Naur a/panels/region/cinnamon-region-panel.desktop b/panels/region/cinnamon-region-panel.desktop
--- a/panels/region/cinnamon-region-panel.desktop	2022-07-04 09:39:25.000000000 +0200
+++ b/panels/region/cinnamon-region-panel.desktop	2022-07-04 15:47:20.395453809 +0200
@@ -8,7 +8,7 @@
 OnlyShowIn=X-Cinnamon;
 X-Cinnamon-Settings-Panel=region
 # Translators: those are keywords for the region control-center panel
-_Keywords=Language;Layout;Keyboard;
+Keywords=Language;Layout;Keyboard;
 NoDisplay=true
 Name=Region & Language
 Name[am]=አካባቢ & ቋንቋ
diff -Naur a/panels/wacom/cinnamon-wacom-panel.desktop b/panels/wacom/cinnamon-wacom-panel.desktop
--- a/panels/wacom/cinnamon-wacom-panel.desktop	2022-07-04 09:39:25.000000000 +0200
+++ b/panels/wacom/cinnamon-wacom-panel.desktop	2022-07-04 15:47:08.967420497 +0200
@@ -8,7 +8,7 @@
 X-Cinnamon-Settings-Panel=wacom
 OnlyShowIn=X-Cinnamon;
 # Translators: those are keywords for the wacom tablet control-center panel
-_Keywords=Tablet;Wacom;Stylus;Eraser;Mouse;
+Keywords=Tablet;Wacom;Stylus;Eraser;Mouse;
 Name=Graphics Tablet
 Name[am]=ግራፊክስ ታብሌት
 Name[ar]=جهاز الرسومات اللوحي
