diff -Naur orig/panels/common/cinnamon-control-center.rules mod/panels/common/cinnamon-control-center.rules
--- orig/panels/common/cinnamon-control-center.rules	2015-10-16 23:50:11.000000000 +0200
+++ mod/panels/common/cinnamon-control-center.rules	2015-10-25 22:29:09.286723019 +0100
@@ -2,7 +2,7 @@
 	if (action.id == "org.freedesktop.hostname1.set-static-hostname" &&
 	    subject.local &&
 	    subject.active &&
-	    subject.isInGroup ("wheel")) {
+	    subject.isInGroup ("root")) {
 		    return polkit.Result.YES;
 	    }
 });
