diff -Naur noip-2.1.9-1-orig/Makefile noip-2.1.9-1/Makefile
--- noip-2.1.9-1-orig/Makefile	2008-11-25 20:41:16.000000000 +0100
+++ noip-2.1.9-1/Makefile	2009-12-04 13:53:39.000000000 +0100
@@ -22,7 +22,7 @@
 # ARCH=sun
 
 ${TGT}: Makefile ${TGT}.c 
-	${CC} -Wall -g -D${ARCH} -DPREFIX=\"${PREFIX}\" ${TGT}.c -o ${TGT} ${LIBS}
+	${CC} -Wall -g -O2 -D${ARCH} ${TGT}.c -o ${TGT} ${LIBS}
 
 install: ${TGT} 
 	if [ ! -d ${BINDIR} ]; then mkdir -p ${BINDIR};fi
diff -Naur noip-2.1.9-1-orig/noip2.c noip-2.1.9-1/noip2.c
--- noip-2.1.9-1-orig/noip2.c	2008-11-25 20:50:09.000000000 +0100
+++ noip-2.1.9-1/noip2.c	2009-12-04 13:51:59.000000000 +0100
@@ -240,7 +240,7 @@
 #define GROUP			2
 #define DOMAIN			3
 #ifndef PREFIX
-  #define PREFIX		"/usr/local"
+  #define PREFIX		""
 #endif
 #define CONFIG_FILEPATH		PREFIX"/etc"
 #define CONFIG_FILENAME		PREFIX"/etc/no-ip2.conf"
