diff -Naur orig/converseen.pro patched/converseen.pro
--- orig/converseen.pro	2016-09-28 17:50:55.000000000 +0200
+++ patched/converseen.pro	2017-01-27 01:04:39.169427903 +0100
@@ -67,9 +67,9 @@
   -DMAGICKCORE_HDRI_ENABLE=0 \
   -DMAGICKCORE_QUANTUM_DEPTH=16
   
- INCLUDEPATH +=  /usr/include/ImageMagick-6
+ INCLUDEPATH +=  /usr/include/ImageMagick-7
 
- LIBS +=  -L/usr/lib  -lMagick++-6.Q16HDRI  -lMagickWand-6.Q16HDRI  -lMagickCore-6.Q16HDRI
+ LIBS +=  -L/usr/lib  -lMagick++-7.Q16HDRI  -lMagickWand-7.Q16HDRI  -lMagickCore-7.Q16HDRI
 
  TRANSLATIONS +=  loc/converseen_it_IT.ts \
  loc/converseen_hu_HU.ts \
