diff -Naur orig/desktop/recoll-searchgui.desktop patched/desktop/recoll-searchgui.desktop
--- orig/desktop/recoll-searchgui.desktop	2018-09-24 09:51:00.000000000 +0200
+++ patched/desktop/recoll-searchgui.desktop	2018-12-18 12:11:47.611883098 +0100
@@ -1,10 +1,8 @@
 [Desktop Entry]
 Categories=Qt;Utility;Filesystem;Database;
 Comment=Find documents by specifying search terms
-Comment[ru]=Поиск документов по заданным условиям
 Exec=recoll
 GenericName=Local Text Search
-GenericName[ru]=Локальный текстовый поиск
 Icon=recoll
 Name=Recoll
 Terminal=false
