diff -Naur recoll-1.25.19-orig/desktop/recoll-searchgui.desktop recoll-1.25.19/desktop/recoll-searchgui.desktop
--- recoll-1.25.19-orig/desktop/recoll-searchgui.desktop	2019-06-12 19:40:49.000000000 +0200
+++ recoll-1.25.19/desktop/recoll-searchgui.desktop	2019-06-17 22:36:57.671471767 +0200
@@ -1,10 +1,8 @@
 [Desktop Entry]
 Categories=Qt;Utility;Filesystem;Database;
 Comment=Find documents by specifying search terms
-Comment[ru]=Поиск документов по заданным условиям
 Exec=recoll
 GenericName=Local Text Search
-GenericName[ru]=Локальный текстовый поиск
 Icon=recoll
 Name=Recoll
 Terminal=false
