diff -Naur a/src/org.kde.partitionmanager.desktop b/src/org.kde.partitionmanager.desktop
--- a/src/org.kde.partitionmanager.desktop	2022-06-07 05:58:18.000000000 +0200
+++ b/src/org.kde.partitionmanager.desktop	2022-06-09 17:55:23.539621447 +0200
@@ -167,7 +167,7 @@
 Exec=partitionmanager
 Icon=partitionmanager
 Categories=System;Filesystem;KDE;Qt;
-SingleMainWindow=True
+SingleMainWindow=true
 X-KDE-StartupNotify=true
 Keywords=partition;disk;
 Keywords[ar]=أقسام;قرص;
