diff -Nur orig/avahi-python/avahi-discover/avahi-discover.desktop.in.in mod/avahi-python/avahi-discover/avahi-discover.desktop.in.in
--- orig/avahi-python/avahi-discover/avahi-discover.desktop.in.in	2019-02-26 03:32:23.000000000 +0100
+++ mod/avahi-python/avahi-discover/avahi-discover.desktop.in.in	2020-02-21 08:37:00.361113595 +0100
@@ -2,7 +2,7 @@
 Version=1.0
 Name=Avahi Zeroconf Browser
 Comment=Browse for Zeroconf services available on your network
-Exec=@bindir@/avahi-discover
+Exec=avahi-discover
 Terminal=false
 Type=Application
 Icon=network-wired
diff -Nur orig/avahi-ui/bssh.desktop.in.in mod/avahi-ui/bssh.desktop.in.in
--- orig/avahi-ui/bssh.desktop.in.in	2019-02-26 03:32:23.000000000 +0100
+++ mod/avahi-ui/bssh.desktop.in.in	2020-02-21 08:36:42.753138512 +0100
@@ -2,7 +2,7 @@
 Version=1.0
 Name=Avahi SSH Server Browser
 Comment=Browse for Zeroconf-enabled SSH Servers
-Exec=@bindir@/bssh
+Exec=bssh
 Terminal=false
 Type=Application
 Icon=network-wired
diff -Nur orig/avahi-ui/bvnc.desktop.in.in mod/avahi-ui/bvnc.desktop.in.in
--- orig/avahi-ui/bvnc.desktop.in.in	2019-02-26 03:32:23.000000000 +0100
+++ mod/avahi-ui/bvnc.desktop.in.in	2020-02-21 08:36:25.961162277 +0100
@@ -2,7 +2,7 @@
 Version=1.0
 Name=Avahi VNC Server Browser
 Comment=Browse for Zeroconf-enabled VNC Servers
-Exec=@bindir@/bvnc
+Exec=bvnc
 Terminal=false
 Type=Application
 Icon=network-wired
