diff -Naurd avahi-0.6.25.orig/avahi-python/avahi-discover/avahi-discover.desktop.in.in avahi-0.6.25/avahi-python/avahi-discover/avahi-discover.desktop.in.in
--- avahi-0.6.25.orig/avahi-python/avahi-discover/avahi-discover.desktop.in.in	2009-04-14 03:36:31.000000000 +0200
+++ avahi-0.6.25/avahi-python/avahi-discover/avahi-discover.desktop.in.in	2009-07-17 20:32:01.000000000 +0200
@@ -2,7 +2,7 @@
 Version=1.0
 _Name=Avahi Zeroconf Browser
 _Comment=Browse for Zeroconf services available on your network
-Exec=@bindir@/avahi-discover
+Exec=avahi-discover
 Terminal=false
 Type=Application
 Icon=network-wired
diff -Naurd avahi-0.6.25.orig/avahi-ui/bssh.desktop.in.in avahi-0.6.25/avahi-ui/bssh.desktop.in.in
--- avahi-0.6.25.orig/avahi-ui/bssh.desktop.in.in	2009-04-14 03:36:31.000000000 +0200
+++ avahi-0.6.25/avahi-ui/bssh.desktop.in.in	2009-07-17 20:32:26.000000000 +0200
@@ -2,7 +2,7 @@
 Version=1.0
 _Name=Avahi SSH Server Browser
 _Comment=Browse for Zeroconf-enabled SSH Servers
-Exec=@bindir@/bssh
+Exec=bssh
 Terminal=false
 Type=Application
 Icon=network-wired
diff -Naurd avahi-0.6.25.orig/avahi-ui/bvnc.desktop.in.in avahi-0.6.25/avahi-ui/bvnc.desktop.in.in
--- avahi-0.6.25.orig/avahi-ui/bvnc.desktop.in.in	2009-04-14 03:36:31.000000000 +0200
+++ avahi-0.6.25/avahi-ui/bvnc.desktop.in.in	2009-07-17 20:32:37.000000000 +0200
@@ -2,7 +2,7 @@
 Version=1.0
 _Name=Avahi VNC Server Browser
 _Comment=Browse for Zeroconf-enabled VNC Servers
-Exec=@bindir@/bvnc
+Exec=bvnc
 Terminal=false
 Type=Application
 Icon=network-wired
