diff -Nur orig/src/conky-manager.desktop mod/src/conky-manager.desktop
--- orig/src/conky-manager.desktop	2015-10-21 15:01:40.000000000 +0200
+++ mod/src/conky-manager.desktop	2024-03-05 14:09:06.031033468 +0100
@@ -1,13 +1,11 @@
 [Desktop Entry]
 Name=Conky Manager
-MimeType=
 Exec=conky-manager 
 Type=Application
 GenericName=Conky Theme Manager
 Terminal=false
 Icon=conky-manager
-Caption=Conky Theme Manager
 X-KDE-StartupNotify=false
-Categories=Utility
+Categories=Settings;Utility;
 InitialPreference=9
 Name[en_US]=Conky Manager
