diff -Nur orig/log.c mod/log.c
--- orig/log.c	2012-07-21 20:51:33.000000000 +0200
+++ mod/log.c	2016-11-20 09:11:43.728243126 +0100
@@ -45,7 +45,7 @@
   };
 
 static const struct LogModule *mods[LMOD_SUP] = { &lm_general };
-static int config[LMOD_SUP] = { LMOD_ENABLE|L_GEN_ALL|LMOD_CFG_VALID };
+static unsigned int config[LMOD_SUP] = { LMOD_ENABLE|L_GEN_ALL|LMOD_CFG_VALID };
 static int cmask[LMOD_SUP]  = { LMOD_ENABLE|L_GEN_ALL };
 static cMutex logfileMutex;
 static FILE *logfile=0;
diff -Nur orig/systems/nagra/cpu.c mod/systems/nagra/cpu.c
--- orig/systems/nagra/cpu.c	2012-07-21 20:51:33.000000000 +0200
+++ mod/systems/nagra/cpu.c	2016-11-20 09:21:31.907592677 +0100
@@ -367,7 +367,7 @@
 // 2 - write operant
 // 4 - use dr register
 // 8 - special address mode in high nibble
-static const char opFlags[] = {
+static const unsigned char opFlags[] = {
 //            00    01    02    03    04    05    06    07    08    09    0a    0b    0c    0d    0e    0f
 /* 0x00 */  0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01,
 /* 0x10 */  0x03, 0x03, 0x03, 0x03, 0x03, 0x03, 0x03, 0x03, 0x03, 0x03, 0x03, 0x03, 0x03, 0x03, 0x03, 0x03,
diff -Nur orig/systems/sc-nagra/sc-nagra.c mod/systems/sc-nagra/sc-nagra.c
--- orig/systems/sc-nagra/sc-nagra.c	2012-07-21 20:51:33.000000000 +0200
+++ mod/systems/sc-nagra/sc-nagra.c	2016-11-20 09:30:33.490899690 +0100
@@ -728,7 +728,7 @@
   T1  protocol: 21 00 08 A0 CA 00 00 02 C0 00 06 87
   */
   unsigned char msg[MAX_LEN+16];
-  static char nagra_head[] = { 0xA0,0xCA,0x00,0x00 };
+  static unsigned char nagra_head[] = { 0xA0,0xCA,0x00,0x00 };
 
   memset(msg,0,sizeof(msg));
   int c=0;
