diff -Nur orig/cam.h mod/cam.h
--- orig/cam.h	2018-06-08 17:12:29.000000000 +0200
+++ mod/cam.h	2018-06-08 17:31:01.108112389 +0200
@@ -40,6 +40,13 @@
 class cPrg;
 class cScDevicePlugin;
 
+#ifndef CA_SET_PID /* removed in kernel 4.14 */
+typedef struct ca_pid {
+        unsigned int pid;
+        int index;          /* -1 == disable */
+} ca_pid_t;
+#define CA_SET_PID _IOW('o', 135, struct ca_pid)
+#endif
 // ----------------------------------------------------------------
 
 class cEcmCache : public cStructListPlain<cEcmData> {
diff -Nur orig/device.h mod/device.h
--- orig/device.h	2018-06-08 17:12:29.000000000 +0200
+++ mod/device.h	2018-06-08 17:35:33.047687238 +0200
@@ -24,6 +24,7 @@
 #include <vdr/dvbdevice.h>
 #include <vdr/thread.h>
 #include "misc.h"
+#include "cam.h"
 
 #ifndef LIBDVBCSA
 #define RINGBUFFERNAME "FFdecsa-TS"
