diff -Naur a/lxqt-config-appearance/lxqt-config-appearance.desktop.in b/lxqt-config-appearance/lxqt-config-appearance.desktop.in
--- a/lxqt-config-appearance/lxqt-config-appearance.desktop.in	2022-04-16 14:47:00.000000000 +0200
+++ b/lxqt-config-appearance/lxqt-config-appearance.desktop.in	2022-06-16 01:36:11.310433860 +0200
@@ -2,7 +2,7 @@
 Type=Application
 Exec=lxqt-config-appearance
 Icon=preferences-desktop-theme
-Categories=Settings;DesktopSettings;Qt;LXQt;
+Categories=Settings;DesktopSettings;Qt;
 OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
diff -Naur a/lxqt-config-brightness/resources/lxqt-config-brightness.desktop.in b/lxqt-config-brightness/resources/lxqt-config-brightness.desktop.in
--- a/lxqt-config-brightness/resources/lxqt-config-brightness.desktop.in	2022-04-16 14:47:00.000000000 +0200
+++ b/lxqt-config-brightness/resources/lxqt-config-brightness.desktop.in	2022-06-16 01:35:52.286362114 +0200
@@ -3,7 +3,7 @@
 Terminal=false
 Type=Application
 Icon=brightnesssettings
-Categories=Settings;DesktopSettings;Qt;LXQt;HardwareSettings;
+Categories=Settings;DesktopSettings;Qt;HardwareSettings;
 OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
diff -Naur a/lxqt-config-file-associations/lxqt-config-file-associations.desktop.in b/lxqt-config-file-associations/lxqt-config-file-associations.desktop.in
--- a/lxqt-config-file-associations/lxqt-config-file-associations.desktop.in	2022-04-16 14:47:00.000000000 +0200
+++ b/lxqt-config-file-associations/lxqt-config-file-associations.desktop.in	2022-06-16 01:35:42.314324495 +0200
@@ -2,7 +2,7 @@
 Type=Application
 Exec=lxqt-config-file-associations
 Icon=preferences-desktop-filetype-association
-Categories=Settings;DesktopSettings;Qt;LXQt;
+Categories=Settings;DesktopSettings;Qt;
 OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
diff -Naur a/lxqt-config-input/lxqt-config-input.desktop.in b/lxqt-config-input/lxqt-config-input.desktop.in
--- a/lxqt-config-input/lxqt-config-input.desktop.in	2022-04-16 14:47:00.000000000 +0200
+++ b/lxqt-config-input/lxqt-config-input.desktop.in	2022-06-16 01:36:02.090399089 +0200
@@ -2,7 +2,7 @@
 Type=Application
 Exec=lxqt-config-input
 Icon=preferences-desktop-peripherals
-Categories=Qt;Settings;HardwareSettings;DesktopSettings;LXQt;
+Categories=Qt;Settings;HardwareSettings;DesktopSettings;
 OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
diff -Naur a/lxqt-config-locale/lxqt-config-locale.desktop.in b/lxqt-config-locale/lxqt-config-locale.desktop.in
--- a/lxqt-config-locale/lxqt-config-locale.desktop.in	2022-04-16 14:47:00.000000000 +0200
+++ b/lxqt-config-locale/lxqt-config-locale.desktop.in	2022-06-16 01:34:31.114055715 +0200
@@ -2,7 +2,7 @@
 Type=Application
 Exec=lxqt-config-locale
 Icon=preferences-desktop-locale
-Categories=Settings;DesktopSettings;Qt;LXQt;
+Categories=Settings;DesktopSettings;Qt;
 OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
diff -Naur a/lxqt-config-monitor/resources/lxqt-config-monitor.desktop.in b/lxqt-config-monitor/resources/lxqt-config-monitor.desktop.in
--- a/lxqt-config-monitor/resources/lxqt-config-monitor.desktop.in	2022-04-16 14:47:00.000000000 +0200
+++ b/lxqt-config-monitor/resources/lxqt-config-monitor.desktop.in	2022-06-16 01:35:22.890251201 +0200
@@ -2,7 +2,7 @@
 Type=Application
 Exec=lxqt-config-monitor
 Icon=preferences-desktop-display
-Categories=Settings;DesktopSettings;Qt;LXQt;HardwareSettings;
+Categories=Settings;DesktopSettings;Qt;HardwareSettings;
 OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
diff -Naur a/src/lxqt-config.desktop.in b/src/lxqt-config.desktop.in
--- a/src/lxqt-config.desktop.in	2022-04-16 14:47:00.000000000 +0200
+++ b/src/lxqt-config.desktop.in	2022-06-16 01:35:33.034289481 +0200
@@ -2,7 +2,7 @@
 Type=Application
 Exec=lxqt-config
 Icon=preferences-system
-Categories=Settings;DesktopSettings;Qt;LXQt;
+Categories=Settings;DesktopSettings;Qt;
 OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
