diff -Naur orig/libmatekbd/org.mate.peripherals-keyboard-xkb.gschema.xml.in patched/libmatekbd/org.mate.peripherals-keyboard-xkb.gschema.xml.in
--- orig/libmatekbd/org.mate.peripherals-keyboard-xkb.gschema.xml.in	2015-10-09 10:00:51.000000000 +0200
+++ patched/libmatekbd/org.mate.peripherals-keyboard-xkb.gschema.xml.in	2015-10-16 03:15:32.799546818 +0200
@@ -12,7 +12,7 @@
       <description>keyboard model</description>
     </key>
     <key name="layouts" type="as">
-      <default>[]</default>
+      <default>['hu']</default>
       <summary>Keyboard layout</summary>
       <description>keyboard layout</description>
     </key>
