diff -Naur orig/sudo.cpp patched/sudo.cpp
--- orig/sudo.cpp	2020-04-23 20:44:01.000000000 +0200
+++ patched/sudo.cpp	2020-04-25 20:33:37.135999280 +0200
@@ -198,9 +198,7 @@
 
     if (BACK_NONE == mBackend)
     {
-        //we were invoked through unknown link (or renamed binary)
-        usage(tr("%1: no backend chosen!").arg(app_master));
-        return 1;
+        mBackend = BACK_SU;
     }
 
     mChildPid = forkpty(&mPwdFd, nullptr, nullptr, nullptr);
