diff -Naur orig/data/org.mate.peripherals-mouse.gschema.xml.in patched/data/org.mate.peripherals-mouse.gschema.xml.in
--- orig/data/org.mate.peripherals-mouse.gschema.xml.in	2020-01-23 19:54:17.000000000 +0100
+++ patched/data/org.mate.peripherals-mouse.gschema.xml.in	2020-02-05 08:03:02.061980096 +0100
@@ -46,7 +46,7 @@
       <description>Highlights the current location of the pointer when the Control key is pressed and released.</description>
     </key>
     <key name="cursor-theme" type="s">
-      <default>''</default>
+      <default>'MATE'</default>
       <summary>Cursor theme</summary>
       <description>Cursor theme name.</description>
     </key>
