diff -Nur orig/autogen.sh mod/autogen.sh
--- orig/autogen.sh	2025-02-14 11:36:41.000000000 +0100
+++ mod/autogen.sh	2025-10-09 13:43:46.461353458 +0200
@@ -29,14 +29,6 @@
 
 test -d m4 || mkdir m4
 
-if gtkdocize --copy; then
-    echo "Files needed by gtk-doc are generated."
-else
-    echo "You need gtk-doc to build this package."
-    echo "http://www.gtk.org/gtk-doc/"
-    exit 1
-fi
-
 set -x
 
 ${ACLOCAL:-aclocal$AM_VERSION} ${ACLOCAL_ARG}
diff -Nur orig/configure.ac mod/configure.ac
--- orig/configure.ac	2025-02-14 11:36:41.000000000 +0100
+++ mod/configure.ac	2025-10-09 13:44:34.837314495 +0200
@@ -89,7 +89,4 @@
 libmenu-cache/menu-cache.h
 menu-cache-gen/Makefile
 menu-cache-daemon/Makefile
-docs/Makefile
-docs/reference/Makefile
-docs/reference/libmenu-cache/Makefile
 ])
diff -Nur orig/Makefile.am mod/Makefile.am
--- orig/Makefile.am	2025-02-14 11:36:41.000000000 +0100
+++ mod/Makefile.am	2025-10-09 13:45:23.405275368 +0200
@@ -4,8 +4,6 @@
 
 NULL =
 
-SUBDIRS_DOCS = docs
-
 EXTRA_DIST_DOCS = \
 	gtk-doc.make \
 	docs/Makefile.am \
