diff -Naur orig/dvdrip.desktop mod/dvdrip.desktop
--- orig/dvdrip.desktop	2009-02-28 19:28:01.000000000 +0100
+++ mod/dvdrip.desktop	2014-05-31 18:44:09.736388843 +0200
@@ -1,9 +1,9 @@
 [Desktop Entry]
-Version=1.0
-Encoding=UTF-8
-Name=dvd::rip
+Name=dvdrip
 GenericName=DVD copy program
+GenericName[hu]=DVD másoló
 Comment=A full featured DVD copy program
+Comment[hu]=Sokat tudó DVD sáv másoló és lementő program
 Type=Application
 Exec=dvdrip
 Icon=dvdrip
