diff -Naur orig/src/network/bufferedconnection.h patched/src/network/bufferedconnection.h
--- orig/src/network/bufferedconnection.h	2020-07-12 21:33:44.000000000 +0200
+++ patched/src/network/bufferedconnection.h	2021-10-05 10:46:35.496958127 +0200
@@ -20,6 +20,7 @@
 #ifndef WL_NETWORK_BUFFEREDCONNECTION_H
 #define WL_NETWORK_BUFFEREDCONNECTION_H
 
+#include <map>
 #include <memory>
 #include <mutex>
 #include <thread>
