diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2020-07-12 21:33:44.000000000 +0200
+++ b/CMakeLists.txt	2022-08-11 02:36:37.649446681 +0200
@@ -248,9 +248,6 @@
   wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Wno-unreachable-code")
   wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Wno-documentation")
 
-  wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Werror=deprecated")
-  wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Werror=non-pod-varargs")
-
   wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Qunused-arguments")
 
   wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Wint-to-void-pointer-cast")
@@ -296,12 +293,6 @@
 endif()
 
 
-# Turn some warnings into errors.
-wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Werror=format-security")
-wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Werror=return-type")
-wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Werror=shadow")
-wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Werror=uninitialized")
-
 IF (WIN32)
   add_definitions(-DMINGW_HAS_SECURE_API -DWIN32_LEAN_AND_MEAN -D__STDC_FORMAT_MACROS -D__USE_MINGW_ANSI_STDIO)
   if (CMAKE_SIZEOF_VOID_P EQUAL 4)
diff -Naur a/src/editor/editorinteractive.h b/src/editor/editorinteractive.h
--- a/src/editor/editorinteractive.h	2020-07-12 21:33:44.000000000 +0200
+++ b/src/editor/editorinteractive.h	2022-08-11 02:33:13.504846582 +0200
@@ -81,8 +81,8 @@
 		EditorDecreaseResourcesTool decrease_resources;
 		EditorSetResourcesTool set_resources;
 		EditorIncreaseResourcesTool increase_resources;
-		EditorSetPortSpaceTool set_port_space;
 		EditorUnsetPortSpaceTool unset_port_space;
+		EditorSetPortSpaceTool set_port_space;
 		EditorSetOriginTool set_origin;
 		EditorResizeTool resize;
 	};
diff -Naur a/src/io/filesystem/disk_filesystem.cc b/src/io/filesystem/disk_filesystem.cc
--- a/src/io/filesystem/disk_filesystem.cc	2020-07-12 21:33:44.000000000 +0200
+++ b/src/io/filesystem/disk_filesystem.cc	2022-08-11 02:33:13.504846582 +0200
@@ -412,8 +412,13 @@
 		}
 		static_cast<int8_t*>(data)[size] = 0;
 
-		fclose(file);
-		file = nullptr;
+		try {
+			fclose(file);
+			file = nullptr;
+		} catch (...) {
+			file = nullptr;
+			throw;
+		}
 
 		length = size;
 	} catch (...) {
