diff -Nur orig/extensions/image_rotation/rotation-utils.c mod/extensions/image_rotation/rotation-utils.c
--- orig/extensions/image_rotation/rotation-utils.c	2025-09-16 09:18:36.000000000 +0200
+++ mod/extensions/image_rotation/rotation-utils.c	2025-10-06 10:05:42.925918298 +0200
@@ -25,6 +25,7 @@
 #include <glib/gi18n.h>
 #include <gio/gio.h>
 #include <extensions/exiv2_tools/exiv2-utils.h>
+#include <stdbool.h>
 #include "rotation-utils.h"
 
 
