diff -Nur orig/share/xsessions/i3.desktop mod/share/xsessions/i3.desktop
--- orig/share/xsessions/i3.desktop	2025-12-19 09:23:15.344528003 +0100
+++ mod/share/xsessions/i3.desktop	2025-12-19 09:29:50.452049881 +0100
@@ -1,8 +1,8 @@
 [Desktop Entry]
 Name=i3
 Comment=improved dynamic tiling window manager
-Exec=i3
-TryExec=i3
+Exec=/usr/bin/i3
+TryExec=/usr/bin/i3
 Type=XSession
 X-LightDM-DesktopName=i3
 DesktopNames=i3
