diff -Nur orig/share/xsessions/i3.desktop mod/share/xsessions/i3.desktop
--- orig/share/xsessions/i3.desktop	2019-08-30 23:06:47.000000000 +0200
+++ mod/share/xsessions/i3.desktop	2019-11-19 12:51:42.263455424 +0100
@@ -1,8 +1,8 @@
 [Desktop Entry]
 Name=i3
 Comment=improved dynamic tiling window manager
-Exec=i3
-TryExec=i3
+Exec=/usr/bin/i3
+TryExec=/usr/bin/i3
 Type=Application
 X-LightDM-DesktopName=i3
 DesktopNames=i3
