diff -Naur a/src/cmp.c b/src/cmp.c
--- a/src/cmp.c	2023-05-20 11:17:26.000000000 +0200
+++ b/src/cmp.c	2023-05-23 22:01:43.644132257 +0200
@@ -343,12 +343,15 @@
 
   /* If only a return code is needed,
      and if both input descriptors are associated with plain files,
+     and if both files are larger than 0 bytes (procfs files are always 0),
      conclude that the files differ if they have different sizes
      and if more bytes will be compared than are in the smaller file.  */
 
   if (comparison_type == type_status
       && 0 <= stat_buf[0].st_size && S_ISREG (stat_buf[0].st_mode)
-      && 0 <= stat_buf[1].st_size && S_ISREG (stat_buf[1].st_mode))
+      && 0 <= stat_buf[1].st_size && S_ISREG (stat_buf[1].st_mode)
+      && stat_buf[0].st_size > 0
+      && stat_buf[1].st_size > 0)
     {
       off_t s0 = stat_buf[0].st_size - file_position (0);
       off_t s1 = stat_buf[1].st_size - file_position (1);
