diff -Nur orig/config.in mod/config.in
--- orig/config.in	2019-08-27 06:46:22.000000000 +0200
+++ mod/config.in	2019-11-19 16:56:16.448763462 +0100
@@ -14,7 +14,7 @@
 set $up k
 set $right l
 # Your preferred terminal emulator
-set $term alacritty
+set $term xterm
 # Your preferred application launcher
 # Note: it's recommended that you pass the final command to sway
 set $menu dmenu_path | dmenu | xargs swaymsg exec --
