diff -Naur a/src/core/CLucene/document/DateTools.h b/src/core/CLucene/document/DateTools.h
--- a/src/core/CLucene/document/DateTools.h	2011-03-17 01:21:07.000000000 +0100
+++ b/src/core/CLucene/document/DateTools.h	2022-08-28 15:13:26.933073739 +0200
@@ -7,20 +7,7 @@
 #ifndef _lucene_document_DateTools_
 #define _lucene_document_DateTools_
 
-#ifdef _CL_TIME_WITH_SYS_TIME
-# include <sys/time.h>
-# include <time.h>
-#else
-# if defined(_CL_HAVE_SYS_TIME_H)
-#  include <sys/time.h>
-# else
-#  include <time.h>
-# endif
-#endif
-
-#ifdef _CL_HAVE_SYS_TIMEB_H
-# include <sys/timeb.h>
-#endif
+# include <ctime>
 
 CL_NS_DEF(document)
 
