diff -Naur orig/meson.build patched/meson.build
--- orig/meson.build	1970-01-01 01:00:00.000000000 +0100
+++ patched/meson.build	2020-11-30 15:30:16.477361181 +0100
@@ -110,7 +110,7 @@
   dependency('gtk+-3.0', version: '>= 3.19.3'),
   dependency('libecal-2.0', version: '>= 3.33.2'),
   dependency('libedataserver-1.2', version: '>= 3.13.90'),
-  dependency('libhandy-0.0', version: '>= 0.0.10'),
+  dependency('libhandy-1', version: '>= 1.0.0'),
   dependency('libxml-2.0'),
   dependency('uuid'),
   dependency('webkit2gtk-4.0', version: '>= 2.26'),
