diff -Naur orig/bootstrap patched/bootstrap
--- orig/bootstrap	2015-01-20 20:19:11.000000000 +0100
+++ patched/bootstrap	2020-09-23 11:43:07.941283956 +0200
@@ -3740,7 +3740,7 @@
         # If the .gitmodules file doesn't exist, create it now, and mark
         # it as ignored for the VCS.
         test -n "$gnulib_path" || gnulib_path=gnulib
-        test -n "$gnulib_url"  || gnulib_url=git://git.sv.gnu.org/gnulib
+        test -n "$gnulib_url"  || gnulib_url=git://git.savannah.gnu.org/gnulib
 
         {
           echo '[submodule "gnulib"]'
