diff -Nur orig/Makefile mod/Makefile
--- orig/Makefile	1994-12-18 03:16:09.000000000 +0100
+++ mod/Makefile	2019-01-15 18:06:25.410309412 +0100
@@ -6,8 +6,8 @@
 #LDFLAGS = -N
 
 # for production code
-CFLAGS	= -Wall -O3 -fomit-frame-pointer
-LDFLAGS = -s -N
+CFLAGS	= -Wall -O2 -fomit-frame-pointer
+LDFLAGS = -s -ltinfo
 
 statserial:	statserial.o
 	$(LD) $(LDFLAGS) -o statserial statserial.o -lcurses
@@ -16,8 +16,8 @@
 	$(CC) $(CFLAGS) -c statserial.c
 
 install: statserial
-	install -m 555 statserial /usr/local/bin/statserial
-	install -m 444 statserial.1 /usr/local/man/man1/statserial.1
+	install -m 555 statserial $(DESTDIR)/usr/bin/statserial
+	install -m 444 statserial.1 $(DESTDIR)/usr/share/man/man1/statserial.1
 
 clean:
 	$(RM) statserial statserial.o core *~
