diff -Nur orig/src/gameconfigfile.cpp mod/src/gameconfigfile.cpp
--- orig/src/gameconfigfile.cpp	2016-09-18 17:50:14.000000000 +0200
+++ mod/src/gameconfigfile.cpp	2016-11-14 10:29:16.795186588 +0100
@@ -133,6 +133,7 @@
 		SetValueForKey ("Path", "/usr/local/share/games/doom", true);
 		SetValueForKey ("Path", "/usr/share/doom", true);
 		SetValueForKey ("Path", "/usr/share/games/doom", true);
+		SetValueForKey ("Path", "/usr/share/games/heretic", true);
 #endif
 	}
 
