diff -Naur orig/mspack/oabd.c patched/mspack/oabd.c
--- orig/mspack/oabd.c	2019-03-04 01:53:58.000000000 +0100
+++ patched/mspack/oabd.c	2019-09-30 22:24:21.987765981 +0200
@@ -35,7 +35,7 @@
                                        const char *output);
 static int oabd_param(struct msoab_decompressor *base, int param, int value);
 static int copy_fh(struct mspack_system *sys, struct mspack_file *infh,
-                   struct mspack_file *outfh, unsigned int bytes_to_copy,
+                   struct mspack_file *outfh, size_t bytes_to_copy,
                    unsigned char *buf, int buf_size);
 
 
