diff -Naur a/clang/lib/Driver/ToolChains/Linux.cpp b/clang/lib/Driver/ToolChains/Linux.cpp
--- a/clang/lib/Driver/ToolChains/Linux.cpp	2022-09-20 08:05:50.000000000 +0200
+++ b/clang/lib/Driver/ToolChains/Linux.cpp	2022-09-23 04:06:14.928079481 +0200
@@ -210,7 +210,7 @@
   if (Triple.getArch() == llvm::Triple::riscv32)
     return "lib32";
 
-  return Triple.isArch32Bit() ? "lib" : "lib64";
+  return Triple.isArch32Bit() ? "lib" : "lib";
 }
 
 Linux::Linux(const Driver &D, const llvm::Triple &Triple, const ArgList &Args)
@@ -592,7 +592,7 @@
   case llvm::Triple::x86_64: {
     bool X32 = Triple.isX32();
 
-    LibDir = X32 ? "libx32" : "lib64";
+    LibDir = X32 ? "libx32" : "/usr/lib";
     Loader = X32 ? "ld-linux-x32.so.2" : "ld-linux-x86-64.so.2";
     break;
   }
