diff -Naur orig/completions/dpkg patched/completions/dpkg
--- orig/completions/dpkg	2018-03-17 09:32:34.000000000 +0100
+++ patched/completions/dpkg	2018-07-03 03:01:25.412234089 +0200
@@ -53,7 +53,7 @@
         -c|-i|-A|-I|-f|-e|-x|-W|--install|--unpack|--record-avail| \
         --contents|--info|--fsys-tarfile|--field|--control|--extract| \
         --vextract)
-            _filedir '?(u|d)deb'
+            _filedir '?(u|d)deb|uhu'
             return
             ;;
         -b|--build)
