diff -Naur a/completions/dpkg b/completions/dpkg
--- a/completions/dpkg	2024-04-04 12:14:50.000000000 +0200
+++ b/completions/dpkg	2024-05-05 21:05:34.256148644 +0200
@@ -77,7 +77,7 @@
     case $prev in
         --install | --unpack | --record-avail | --contents | --info | --fsys-tarfile | \
             --field | --control | --extract | --vextract | --raw-extract | -${noargopts}[ciAIfexX])
-            _comp_compgen_filedir '?(u|d)deb'
+            _comp_compgen_filedir '?(u|d)deb|uhu'
             return
             ;;
         --build | --admindir | --instdir | --root | -${noargopts}b)
@@ -92,7 +92,7 @@
             if [[ $1 == *dpkg-query ]]; then
                 _comp_compgen -x apt-cache packages
             else
-                _comp_compgen_filedir '?(u|d)deb'
+                _comp_compgen_filedir '?(u|d)deb|uhu'
             fi
             return
             ;;
