diff -Naurd a/desktop/deb-gview.desktop.in.in b/desktop/deb-gview.desktop.in.in
--- a/desktop/deb-gview.desktop.in.in	2009-08-20 10:12:42.000000000 +0200
+++ b/desktop/deb-gview.desktop.in.in	2014-05-06 19:18:52.000000000 +0200
@@ -1,8 +1,10 @@
 [Desktop Entry]
 _Name=Debian Package Viewer
 _Comment=Debian package file viewer
+Name[hu]=deb-gview
+Comment[hu]=Debian csomag megjelenítő
 Exec=deb-gview
-Icon=deb-gview.xpm
+Icon=deb-gview
 Terminal=false
 Categories=Development;Building;GNOME;
 Type=Application
diff -Naurd a/Makefile.am b/Makefile.am
--- a/Makefile.am	2012-01-08 15:47:05.000000000 +0100
+++ b/Makefile.am	2013-08-22 20:47:45.000000000 +0200
@@ -5,7 +5,7 @@
 
 man_MANS=$(PACKAGE).1
 
-DB2MAN = /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/docbook.xsl
+DB2MAN = /usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl
 XP = xsltproc -''-nonet -''-param man.charmap.use.subset "0"
 
 deb_gviewdocdir = ${prefix}/share/doc/deb-gview
diff -Naurd a/src/callbacks.c b/src/callbacks.c
--- a/src/callbacks.c	2012-03-31 23:23:08.000000000 +0200
+++ b/src/callbacks.c	2012-05-07 22:41:39.000000000 +0200
@@ -64,6 +64,11 @@
 			"/var/cache/apt/archives");
 	}
 	filter = gtk_file_filter_new ();
+	gtk_file_filter_set_name (filter, _("UHU-Linux csomagok *.uhu"));
+	gtk_file_filter_add_pattern (filter, 
+		"*.uhu");
+	gtk_file_chooser_add_filter (GTK_FILE_CHOOSER(chooser), filter);
+	filter = gtk_file_filter_new ();
 	gtk_file_filter_set_name (filter, _("Debian package files *.deb"));
 	gtk_file_filter_add_pattern (filter, 
 		"*.deb");
diff -Naurd a/src/dvarchive.c b/src/dvarchive.c
--- a/src/dvarchive.c	2013-05-26 13:39:27.000000000 +0200
+++ b/src/dvarchive.c	2013-06-03 06:44:40.000000000 +0200
@@ -185,6 +185,23 @@
 				dv_new_window (debfile);
 			}
 		}
+		else
+		{
+			debline = g_strrstr (changes_file[c], ".uhu");
+			if (debline)
+			{
+				GFile * debfile;
+				gchar * partial;
+				if (!g_str_has_suffix (debline, ".uhu"))
+					continue;
+				debfile = NULL;
+				partial = g_strstrip (g_strrstr (changes_file[c], " "));
+				debname = g_strconcat(chgpath, "/", partial, NULL);
+				debfile = g_file_new_for_uri (debname);
+				if (g_str_has_suffix (debname, ".uhu"))
+					dv_new_window (debfile);
+			}
+		}
 	}
 }
 
@@ -327,6 +344,7 @@
 	filename = g_file_get_uri (deb->ghandle);
 	if ((!g_str_has_suffix(filename, ".deb")) &&
 		(!g_str_has_suffix(filename, ".udeb")) &&
+		(!g_str_has_suffix(filename, ".uhu")) &&
 		(!g_str_has_suffix(filename, ".tdeb"))) {
 		return;
 	}
