diff -Naur a/desktop/deb-gview.desktop.in.in b/desktop/deb-gview.desktop.in.in
--- a/desktop/deb-gview.desktop.in.in	2024-10-17 17:44:01.000000000 +0200
+++ b/desktop/deb-gview.desktop.in.in	2025-02-28 12:16:07.340070866 +0100
@@ -2,8 +2,10 @@
 _Name=Debian Package Viewer
 _Comment=Debian package file viewer
 _Keywords=deb;debian;view;open;
+Name[hu]=deb-gview
+Comment[hu]=Debian csomag megjelenítő
 Exec=deb-gview
-Icon=deb-gview.png
+Icon=deb-gview
 Terminal=false
 Categories=Development;Building;GNOME;
 Type=Application
diff -Naur a/Makefile.am b/Makefile.am
--- a/Makefile.am	2024-10-17 17:44:01.000000000 +0200
+++ b/Makefile.am	2025-02-28 12:17:22.000173103 +0100
@@ -1,20 +1,5 @@
 SUBDIRS = src . po desktop
 
-$(PACKAGE).1: $(PACKAGE).1.xml
-	$(XP) $(DB2MAN) $<
-
-man_MANS=$(PACKAGE).1
-
-DB2MAN = /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/docbook.xsl
-XP = xsltproc -''-nonet -''-param man.charmap.use.subset "0"
-
-deb_gviewdocdir = ${prefix}/share/doc/deb-gview
-deb_gviewdoc_DATA = \
- ChangeLog \
- README \
- NEWS \
- TODO
-
 deb-gview.pot: Makefile
 	${RM} -f po/deb-gview.pot
 	${MAKE} -C po deb-gview.pot
diff -Naur a/src/callbacks.c b/src/callbacks.c
--- a/src/callbacks.c	2024-10-17 17:44:01.000000000 +0200
+++ b/src/callbacks.c	2025-02-28 12:16:07.340070866 +0100
@@ -64,6 +64,11 @@
 			"/var/cache/apt/archives");
 	}
 	filter = gtk_file_filter_new ();
+	gtk_file_filter_set_name (filter, _("UHU-Linux csomagok *.uhu"));
+	gtk_file_filter_add_pattern (filter, 
+		"*.uhu");
+	gtk_file_chooser_add_filter (GTK_FILE_CHOOSER(chooser), filter);
+	filter = gtk_file_filter_new ();
 	gtk_file_filter_set_name (filter, _("Debian package files *.deb"));
 	gtk_file_filter_add_pattern (filter, 
 		"*.deb");
diff -Naur a/src/dvarchive.c b/src/dvarchive.c
--- a/src/dvarchive.c	2024-10-17 17:44:01.000000000 +0200
+++ b/src/dvarchive.c	2025-02-28 12:16:07.340070866 +0100
@@ -188,6 +188,23 @@
 				dv_new_window (debfile);
 			}
 		}
+		else
+		{
+			debline = g_strrstr (changes_file[c], ".uhu");
+			if (debline)
+			{
+				GFile * debfile;
+				gchar * partial;
+				if (!g_str_has_suffix (debline, ".uhu"))
+					continue;
+				debfile = NULL;
+				partial = g_strstrip (g_strrstr (changes_file[c], " "));
+				debname = g_strconcat(chgpath, "/", partial, NULL);
+				debfile = g_file_new_for_uri (debname);
+				if (g_str_has_suffix (debname, ".uhu"))
+					dv_new_window (debfile);
+			}
+		}
 	}
 }
 
@@ -335,6 +352,7 @@
 	filename = g_file_get_uri (deb->ghandle);
 	if ((!g_str_has_suffix(filename, ".deb")) &&
 		(!g_str_has_suffix(filename, ".udeb")) &&
+		(!g_str_has_suffix(filename, ".uhu")) &&
 		(!g_str_has_suffix(filename, ".tdeb"))) {
 		return;
 	}
