diff -Naur orig/data/mate.desktop.in mod/data/mate.desktop.in
--- orig/data/mate.desktop.in	2019-07-06 09:59:10.000000000 +0200
+++ mod/data/mate.desktop.in	2020-02-05 08:23:03.238803659 +0100
@@ -1,8 +1,8 @@
 [Desktop Entry]
 Name=MATE
 Comment=This session logs you into MATE
-Exec=mate-session
-TryExec=mate-session
+Exec=/usr/bin/mate-session
+TryExec=/usr/bin/mate-session
 Icon=
 Type=Application
 DesktopNames=MATE
