diff -Naur orig/plugins/impex/webp/dlg_webp_export.cpp patched/plugins/impex/webp/dlg_webp_export.cpp
--- orig/plugins/impex/webp/dlg_webp_export.cpp	2022-01-05 12:03:09.000000000 +0100
+++ patched/plugins/impex/webp/dlg_webp_export.cpp	2022-01-29 11:38:35.729078855 +0100
@@ -76,10 +76,6 @@
     nearLossless->setValue(cfg->getInt("near_lossless", 100));
     exact->setChecked(cfg->getBool("exact", 0));
     useSharpYUV->setChecked(cfg->getBool("use_sharp_yuv", false));
-#if WEBP_ENCODER_ABI_VERSION >= 0x020f
-    qMin->setValue(cfg->getInt("qmin", 0));
-    qMax->setValue(cfg->getInt("qmax", 100));
-#endif
 }
 
 void KisDlgWebPExport::changePreset()
@@ -119,10 +115,6 @@
     nearLossless->setValue(preset.near_lossless);
     exact->setChecked(preset.exact == 1);
     useSharpYUV->setChecked(preset.use_sharp_yuv);
-#if WEBP_ENCODER_ABI_VERSION >= 0x020f
-    qMin->setValue(preset.qmin);
-    qMax->setValue(preset.qmax);
-#endif
 }
 
 KisPropertiesConfigurationSP KisDlgWebPExport::configuration() const
@@ -157,10 +149,6 @@
     cfg->setProperty("near_lossless", nearLossless->value());
     cfg->setProperty("exact", exact->isChecked());
     cfg->setProperty("use_sharp_yuv", useSharpYUV->isChecked());
-#if WEBP_ENCODER_ABI_VERSION >= 0x020f
-    cfg->setProperty("qmin", qMin->value());
-    cfg->setProperty("qmax", qMax->value());
-#endif
 
     return cfg;
 }
diff -Naur orig/plugins/impex/webp/kis_webp_export.cpp patched/plugins/impex/webp/kis_webp_export.cpp
--- orig/plugins/impex/webp/kis_webp_export.cpp	2022-01-05 12:03:09.000000000 +0100
+++ patched/plugins/impex/webp/kis_webp_export.cpp	2022-01-29 11:28:33.793365566 +0100
@@ -87,10 +87,6 @@
     cfg->setProperty("near_lossless", preset.near_lossless);
     cfg->setProperty("exact", preset.exact == 1);
     cfg->setProperty("use_sharp_yuv", preset.use_sharp_yuv == 1);
-#if WEBP_ENCODER_ABI_VERSION >= 0x020f
-    cfg->setProperty("qmin", preset.qmin);
-    cfg->setProperty("qmax", preset.qmax);
-#endif
 
     return cfg;
 }
@@ -135,10 +131,6 @@
         config.near_lossless = cfg->getInt("near_lossless", 100);
         config.exact = cfg->getBool("exact", 0) ? 1 : 0;
         config.use_sharp_yuv = cfg->getBool("use_sharp_yuv", false) ? 1 : 0;
-#if WEBP_ENCODER_ABI_VERSION >= 0x020f
-        config.qmin = cfg->getInt("qmin", 0);
-        config.qmax = cfg->getInt("qmax", 100);
-#endif
 
         if (!WebPValidateConfig(&config)) {
             dbgPlugins << "WebP configuration validation failure";
