diff -Naur krita-4.1.5-orig/plugins/impex/qimageio/krita_qimageio.desktop krita-4.1.5/plugins/impex/qimageio/krita_qimageio.desktop
--- krita-4.1.5-orig/plugins/impex/qimageio/krita_qimageio.desktop	2018-10-11 13:55:34.000000000 +0200
+++ krita-4.1.5/plugins/impex/qimageio/krita_qimageio.desktop	2018-10-31 18:23:13.229077028 +0100
@@ -63,7 +63,7 @@
 Name[zh_CN]=Krita
 Name[zh_TW]=Krita
 Exec=krita %f
-MimeType=image/bmp,image/x-xpixmap,image/x-xbitmap,image/webp,image/vnd.microsoft.icon
+MimeType=image/bmp;image/x-xpixmap;image/x-xbitmap;image/webp;image/vnd.microsoft.icon;
 Type=Application
 Icon=calligrakrita
 Categories=Qt;KDE;Office;Graphics;
