diff -Naur orig/benchmarks/kis_stroke_benchmark.cpp patched/benchmarks/kis_stroke_benchmark.cpp
--- orig/benchmarks/kis_stroke_benchmark.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/benchmarks/kis_stroke_benchmark.cpp	2020-06-03 01:34:00.586376233 +0200
@@ -26,6 +26,7 @@
 }
 #endif
 
+#include <QPainterPath>
 #include <QTest>
 
 #include "kis_stroke_benchmark.h"
diff -Naur orig/libs/basicflakes/tools/KoCreatePathTool_p.h patched/libs/basicflakes/tools/KoCreatePathTool_p.h
--- orig/libs/basicflakes/tools/KoCreatePathTool_p.h	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/basicflakes/tools/KoCreatePathTool_p.h	2020-06-03 01:34:00.586376233 +0200
@@ -22,6 +22,8 @@
 #ifndef KOCREATEPATHTOOL_P_H
 #define KOCREATEPATHTOOL_P_H
 
+#include <QPainterPath>
+
 #include "KoCreatePathTool.h"
 #include "KoPathPoint.h"
 #include "KoPathPointData.h"
diff -Naur orig/libs/brush/kis_auto_brush.cpp patched/libs/brush/kis_auto_brush.cpp
--- orig/libs/brush/kis_auto_brush.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/brush/kis_auto_brush.cpp	2020-06-03 01:34:00.586376233 +0200
@@ -24,6 +24,7 @@
 #include <kis_debug.h>
 #include <math.h>
 
+#include <QPainterPath>
 #include <QRect>
 #include <QDomElement>
 #include <QtConcurrentMap>
diff -Naur orig/libs/brush/kis_boundary.cc patched/libs/brush/kis_boundary.cc
--- orig/libs/brush/kis_boundary.cc	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/brush/kis_boundary.cc	2020-06-03 01:34:00.586376233 +0200
@@ -18,6 +18,7 @@
 
 #include "kis_boundary.h"
 #include <QPainter>
+#include <QPainterPath>
 #include <QPen>
 
 #include "KoColorSpace.h"
diff -Naur orig/libs/brush/kis_brush.cpp patched/libs/brush/kis_brush.cpp
--- orig/libs/brush/kis_brush.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/brush/kis_brush.cpp	2020-06-03 01:34:00.586376233 +0200
@@ -25,6 +25,7 @@
 
 #include <QDomElement>
 #include <QFile>
+#include <QPainterPath>
 #include <QPoint>
 #include <QFileInfo>
 #include <QBuffer>
diff -Naur orig/libs/flake/KoClipMask.cpp patched/libs/flake/KoClipMask.cpp
--- orig/libs/flake/KoClipMask.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/KoClipMask.cpp	2020-06-03 01:35:31.082525944 +0200
@@ -21,6 +21,8 @@
 #include <QRectF>
 #include <QTransform>
 #include <QPainter>
+#include <QSharedData>
+#include <QPainterPath>
 #include <KoShape.h>
 #include "kis_algebra_2d.h"
 
diff -Naur orig/libs/flake/KoClipMaskPainter.cpp patched/libs/flake/KoClipMaskPainter.cpp
--- orig/libs/flake/KoClipMaskPainter.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/KoClipMaskPainter.cpp	2020-06-03 01:34:00.586376233 +0200
@@ -19,6 +19,7 @@
 #include "KoClipMaskPainter.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QRectF>
 
 #include "kis_assert.h"
diff -Naur orig/libs/flake/KoConnectionShape.cpp patched/libs/flake/KoConnectionShape.cpp
--- orig/libs/flake/KoConnectionShape.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/KoConnectionShape.cpp	2020-06-03 01:34:00.586376233 +0200
@@ -34,6 +34,7 @@
 #include <KoXmlNS.h>
 #include <KoUnit.h>
 #include <QPainter>
+#include <QPainterPath>
 
 #include <FlakeDebug.h>
 
diff -Naur orig/libs/flake/KoGradientBackground.cpp patched/libs/flake/KoGradientBackground.cpp
--- orig/libs/flake/KoGradientBackground.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/KoGradientBackground.cpp	2020-06-03 01:36:39.074638332 +0200
@@ -31,6 +31,8 @@
 #include <QSharedPointer>
 #include <QBrush>
 #include <QPainter>
+#include <QSharedData>
+#include <QPainterPath>
 
 class KoGradientBackgroundPrivate : public KoShapeBackgroundPrivate
 {
diff -Naur orig/libs/flake/KoHatchBackground.cpp patched/libs/flake/KoHatchBackground.cpp
--- orig/libs/flake/KoHatchBackground.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/KoHatchBackground.cpp	2020-06-03 01:34:00.586376233 +0200
@@ -36,6 +36,7 @@
 #include <QColor>
 #include <QString>
 #include <QPainter>
+#include <QPainterPath>
 
 class KoHatchBackgroundPrivate : public KoColorBackgroundPrivate
 {
diff -Naur orig/libs/flake/KoOdfGradientBackground.cpp patched/libs/flake/KoOdfGradientBackground.cpp
--- orig/libs/flake/KoOdfGradientBackground.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/KoOdfGradientBackground.cpp	2020-06-03 01:34:00.590376239 +0200
@@ -32,6 +32,7 @@
 #include <KoOdfStylesReader.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QColor>
 #include <QImage>
 #include <qmath.h>
diff -Naur orig/libs/flake/KoPathShape.cpp patched/libs/flake/KoPathShape.cpp
--- orig/libs/flake/KoPathShape.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/KoPathShape.cpp	2020-06-03 01:34:00.590376239 +0200
@@ -50,6 +50,7 @@
 
 #include <FlakeDebug.h>
 #include <QPainter>
+#include <QPainterPath>
 
 #include "kis_global.h"
 
diff -Naur orig/libs/flake/KoPatternBackground.cpp patched/libs/flake/KoPatternBackground.cpp
--- orig/libs/flake/KoPatternBackground.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/KoPatternBackground.cpp	2020-06-03 01:34:00.590376239 +0200
@@ -37,6 +37,7 @@
 
 #include <QBrush>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPointer>
 
 class KoPatternBackgroundPrivate : public KoShapeBackgroundPrivate
diff -Naur orig/libs/flake/KoShapeManager.cpp patched/libs/flake/KoShapeManager.cpp
--- orig/libs/flake/KoShapeManager.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/KoShapeManager.cpp	2020-06-03 01:34:00.590376239 +0200
@@ -49,6 +49,7 @@
 #include <QApplication>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QTimer>
 #include <FlakeDebug.h>
 
diff -Naur orig/libs/flake/KoShapeShadow.cpp patched/libs/flake/KoShapeShadow.cpp
--- orig/libs/flake/KoShapeShadow.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/KoShapeShadow.cpp	2020-06-03 01:34:00.590376239 +0200
@@ -32,6 +32,7 @@
 #include <KoViewConverter.h>
 #include <FlakeDebug.h>
 #include <QPainter>
+#include <QPainterPath>
 #include <QAtomicInt>
 #include <QImage>
 #include <QRectF>
diff -Naur orig/libs/flake/KoSnapGuide.cpp patched/libs/flake/KoSnapGuide.cpp
--- orig/libs/flake/KoSnapGuide.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/KoSnapGuide.cpp	2020-06-03 01:34:00.590376239 +0200
@@ -27,6 +27,7 @@
 #include <KoCanvasBase.h>
 
 #include <QPainter>
+#include <QPainterPath>
 
 #include <math.h>
 
diff -Naur orig/libs/flake/KoSnapStrategy.cpp patched/libs/flake/KoSnapStrategy.cpp
--- orig/libs/flake/KoSnapStrategy.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/KoSnapStrategy.cpp	2020-06-03 01:34:00.590376239 +0200
@@ -27,6 +27,7 @@
 #include <KoViewConverter.h>
 
 #include <QPainter>
+#include <QPainterPath>
 
 #include <cmath>
 
diff -Naur orig/libs/flake/svg/SvgParser.cpp patched/libs/flake/svg/SvgParser.cpp
--- orig/libs/flake/svg/SvgParser.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/svg/SvgParser.cpp	2020-06-03 01:34:00.590376239 +0200
@@ -31,6 +31,7 @@
 
 #include <QColor>
 #include <QPainter>
+#include <QPainterPath>
 #include <QDir>
 
 #include <KoShape.h>
diff -Naur orig/libs/flake/tests/TestKoMarkerCollection.cpp patched/libs/flake/tests/TestKoMarkerCollection.cpp
--- orig/libs/flake/tests/TestKoMarkerCollection.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/tests/TestKoMarkerCollection.cpp	2020-06-03 01:34:00.590376239 +0200
@@ -21,6 +21,7 @@
 #include <QTest>
 #include <QFileInfo>
 #include <QPainter>
+#include <QPainterPath>
 #include <KoMarker.h>
 #include <KoMarkerCollection.h>
 #include <KoPathShape.h>
diff -Naur orig/libs/flake/tests/TestPointMergeCommand.cpp patched/libs/flake/tests/TestPointMergeCommand.cpp
--- orig/libs/flake/tests/TestPointMergeCommand.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/tests/TestPointMergeCommand.cpp	2020-06-03 01:34:00.590376239 +0200
@@ -23,6 +23,7 @@
 #include "KoPathPoint.h"
 #include "KoPathPointData.h"
 #include <sdk/tests/kistest.h>
+#include <QPainterPath>
 #include <QTest>
 #include <FlakeDebug.h>
 
diff -Naur orig/libs/flake/tests/TestShapeBackgroundCommand.cpp patched/libs/flake/tests/TestShapeBackgroundCommand.cpp
--- orig/libs/flake/tests/TestShapeBackgroundCommand.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/tests/TestShapeBackgroundCommand.cpp	2020-06-03 01:34:00.590376239 +0200
@@ -24,6 +24,7 @@
 #include "KoShapePaintingContext.h"
 #include "KoViewConverter.h"
 
+#include <QPainterPath>
 #include <QTest>
 
 void TestShapeBackgroundCommand::refCounting()
diff -Naur orig/libs/flake/tests/TestSnapStrategy.cpp patched/libs/flake/tests/TestSnapStrategy.cpp
--- orig/libs/flake/tests/TestSnapStrategy.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/tests/TestSnapStrategy.cpp	2020-06-03 01:34:00.590376239 +0200
@@ -18,6 +18,7 @@
 */
 
 #include "TestSnapStrategy.h"
+#include <QPainterPath>
 #include <QTest>
 #include "KoSnapStrategy.h"
 #include "KoPathShape.h"
diff -Naur orig/libs/flake/tests/TestSvgParser.cpp patched/libs/flake/tests/TestSvgParser.cpp
--- orig/libs/flake/tests/TestSvgParser.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/tests/TestSvgParser.cpp	2020-06-03 01:34:00.590376239 +0200
@@ -19,6 +19,7 @@
 #include "TestSvgParser.h"
 
 
+#include <QPainterPath>
 #include <QTest>
 #include <svg/SvgUtil.h>
 #include <KoShapeStrokeModel.h>
diff -Naur orig/libs/flake/tests/TestSvgText.cpp patched/libs/flake/tests/TestSvgText.cpp
--- orig/libs/flake/tests/TestSvgText.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/tests/TestSvgText.cpp	2020-06-03 01:34:00.590376239 +0200
@@ -768,6 +768,7 @@
 
 #include <QTextLayout>
 #include <QPainter>
+#include <QPainterPath>
 
 void TestSvgText::testQtBidi()
 {
diff -Naur orig/libs/flake/text/KoSvgTextChunkShape_p.h patched/libs/flake/text/KoSvgTextChunkShape_p.h
--- orig/libs/flake/text/KoSvgTextChunkShape_p.h	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/text/KoSvgTextChunkShape_p.h	2020-06-03 01:38:39.302836886 +0200
@@ -20,9 +20,12 @@
 
 #include "KoSvgText.h"
 #include "KoSvgTextProperties.h"
-#include <KoShapeContainer_p.h>
+#include <QSharedData>
+#include <QPainterPath>
 #include <QTextCharFormat>
 
+#include <KoShapeContainer_p.h>
+
 class SvgGraphicsContext;
 
 class KoSvgTextChunkShapePrivate : public KoShapeContainerPrivate
diff -Naur orig/libs/flake/text/KoSvgTextShape.cpp patched/libs/flake/text/KoSvgTextShape.cpp
--- orig/libs/flake/text/KoSvgTextShape.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/text/KoSvgTextShape.cpp	2020-06-03 01:34:00.590376239 +0200
@@ -48,6 +48,7 @@
 #include <vector>
 #include <memory>
 #include <QPainter>
+#include <QPainterPath>
 #include <boost/optional.hpp>
 
 #include <text/KoSvgTextChunkShapeLayoutInterface.h>
diff -Naur orig/libs/flake/tools/KoPathTool.cpp patched/libs/flake/tools/KoPathTool.cpp
--- orig/libs/flake/tools/KoPathTool.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/flake/tools/KoPathTool.cpp	2020-06-03 01:34:00.590376239 +0200
@@ -66,6 +66,7 @@
 #include <FlakeDebug.h>
 #include <klocalizedstring.h>
 #include <QPainter>
+#include <QPainterPath>
 #include <QBitmap>
 #include <QTabWidget>
 
diff -Naur orig/libs/global/KisHandlePainterHelper.cpp patched/libs/global/KisHandlePainterHelper.cpp
--- orig/libs/global/KisHandlePainterHelper.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/global/KisHandlePainterHelper.cpp	2020-06-03 01:34:00.590376239 +0200
@@ -19,6 +19,7 @@
 #include "KisHandlePainterHelper.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include "kis_algebra_2d.h"
 #include "kis_painting_tweaks.h"
 
diff -Naur orig/libs/image/brushengine/kis_paintop_settings.cpp patched/libs/image/brushengine/kis_paintop_settings.cpp
--- orig/libs/image/brushengine/kis_paintop_settings.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/image/brushengine/kis_paintop_settings.cpp	2020-06-03 01:34:00.590376239 +0200
@@ -21,6 +21,7 @@
 
 #include <QImage>
 #include <QColor>
+#include <QPainterPath>
 #include <QPointer>
 
 #include <KoPointerEvent.h>
diff -Naur orig/libs/ui/flake/kis_shape_selection.h patched/libs/ui/flake/kis_shape_selection.h
--- orig/libs/ui/flake/kis_shape_selection.h	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/ui/flake/kis_shape_selection.h	2020-06-03 01:34:00.590376239 +0200
@@ -18,6 +18,8 @@
 #ifndef KIS_SHAPE_SELECTION_H
 #define KIS_SHAPE_SELECTION_H
 
+#include <QPainterPath>
+
 #include <KoShapeLayer.h>
 #include <KoShapeFactoryBase.h>
 #include <KoShapeUserData.h>
diff -Naur orig/libs/ui/kis_painting_assistants_decoration.cpp patched/libs/ui/kis_painting_assistants_decoration.cpp
--- orig/libs/ui/kis_painting_assistants_decoration.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/ui/kis_painting_assistants_decoration.cpp	2020-06-03 01:34:00.590376239 +0200
@@ -33,6 +33,7 @@
 #include "KisViewManager.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QApplication>
 
 struct KisPaintingAssistantsDecoration::Private {
diff -Naur orig/libs/ui/kis_selection_decoration.h patched/libs/ui/kis_selection_decoration.h
--- orig/libs/ui/kis_selection_decoration.h	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/ui/kis_selection_decoration.h	2020-06-03 01:34:00.590376239 +0200
@@ -19,6 +19,7 @@
 #ifndef _KIS_SELECTION_DECORATION_H_
 #define _KIS_SELECTION_DECORATION_H_
 
+#include <QPainterPath>
 #include <QTimer>
 #include <QPolygon>
 #include <QPen>
diff -Naur orig/libs/ui/tool/kis_shape_tool_helper.cpp patched/libs/ui/tool/kis_shape_tool_helper.cpp
--- orig/libs/ui/tool/kis_shape_tool_helper.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/ui/tool/kis_shape_tool_helper.cpp	2020-06-03 01:34:00.594376246 +0200
@@ -18,6 +18,8 @@
 
 #include "kis_shape_tool_helper.h"
 
+#include <QPainterPath>
+
 #include <KoPathShape.h>
 #include <KoShapeRegistry.h>
 #include <KoShapeFactoryBase.h>
diff -Naur orig/libs/ui/tool/kis_tool_paint.h patched/libs/ui/tool/kis_tool_paint.h
--- orig/libs/ui/tool/kis_tool_paint.h	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/ui/tool/kis_tool_paint.h	2020-06-03 01:34:00.594376246 +0200
@@ -22,6 +22,7 @@
 #include "kis_tool.h"
 
 #include <QGridLayout>
+#include <QPainterPath>
 #include <QTimer>
 #include <QCheckBox>
 
diff -Naur orig/libs/ui/utils/KisClipboardUtil.cpp patched/libs/ui/utils/KisClipboardUtil.cpp
--- orig/libs/ui/utils/KisClipboardUtil.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/ui/utils/KisClipboardUtil.cpp	2020-06-03 01:46:58.095658186 +0200
@@ -25,6 +25,7 @@
 #include <QList>
 #include <QSet>
 #include <QPair>
+#include <QDebug>
 
 namespace KisClipboardUtil {
 
@@ -46,8 +47,11 @@
     QClipboard *clipboard = QApplication::clipboard();
 
     QImage image;
+    QSet<QString> clipboardMimeTypes;
 
-    const QSet<QString> &clipboardMimeTypes = clipboard->mimeData()->formats().toSet();
+    Q_FOREACH(const QString &format, clipboard->mimeData()->formats()) {
+        clipboardMimeTypes << format;
+    }
 
     Q_FOREACH (const ClipboardImageFormat &item, supportedFormats) {
         const QSet<QString> &intersection = item.mimeTypes & clipboardMimeTypes;
@@ -73,4 +77,4 @@
     return image;
 }
 
-}
\ Nincs új sor a fájl végén
+}
diff -Naur orig/libs/ui/widgets/kis_cie_tongue_widget.cpp patched/libs/ui/widgets/kis_cie_tongue_widget.cpp
--- orig/libs/ui/widgets/kis_cie_tongue_widget.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/ui/widgets/kis_cie_tongue_widget.cpp	2020-06-03 01:34:00.594376246 +0200
@@ -40,6 +40,7 @@
 
 #include <QPointF>
 #include <QPainter>
+#include <QPainterPath>
 #include <QFile>
 #include <QTimer>
 #include <QPaintEvent>
diff -Naur orig/libs/ui/widgets/kis_curve_widget.cpp patched/libs/ui/widgets/kis_curve_widget.cpp
--- orig/libs/ui/widgets/kis_curve_widget.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/ui/widgets/kis_curve_widget.cpp	2020-06-03 01:34:00.594376246 +0200
@@ -27,6 +27,7 @@
 
 #include <QPixmap>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPoint>
 #include <QPen>
 #include <QEvent>
diff -Naur orig/libs/ui/widgets/kis_tone_curve_widget.cpp patched/libs/ui/widgets/kis_tone_curve_widget.cpp
--- orig/libs/ui/widgets/kis_tone_curve_widget.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/libs/ui/widgets/kis_tone_curve_widget.cpp	2020-06-03 01:34:00.594376246 +0200
@@ -25,6 +25,7 @@
 #include <QPointF>
 #include <QPolygonF>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPaintEvent>
 #include <QImage>
 #include <cmath>
diff -Naur orig/plugins/assistants/Assistants/ConcentricEllipseAssistant.cc patched/plugins/assistants/Assistants/ConcentricEllipseAssistant.cc
--- orig/plugins/assistants/Assistants/ConcentricEllipseAssistant.cc	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/assistants/Assistants/ConcentricEllipseAssistant.cc	2020-06-03 01:34:00.594376246 +0200
@@ -23,6 +23,7 @@
 #include <klocalizedstring.h>
 #include "kis_debug.h"
 #include <QPainter>
+#include <QPainterPath>
 #include <QLinearGradient>
 #include <QTransform>
 #include <kis_canvas2.h>
diff -Naur orig/plugins/assistants/Assistants/EllipseAssistant.cc patched/plugins/assistants/Assistants/EllipseAssistant.cc
--- orig/plugins/assistants/Assistants/EllipseAssistant.cc	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/assistants/Assistants/EllipseAssistant.cc	2020-06-03 01:34:00.594376246 +0200
@@ -23,6 +23,7 @@
 #include <klocalizedstring.h>
 #include "kis_debug.h"
 #include <QPainter>
+#include <QPainterPath>
 #include <QLinearGradient>
 #include <QTransform>
 
diff -Naur orig/plugins/assistants/Assistants/FisheyePointAssistant.cc patched/plugins/assistants/Assistants/FisheyePointAssistant.cc
--- orig/plugins/assistants/Assistants/FisheyePointAssistant.cc	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/assistants/Assistants/FisheyePointAssistant.cc	2020-06-03 01:34:00.594376246 +0200
@@ -25,6 +25,7 @@
 #include <klocalizedstring.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QLinearGradient>
 #include <QTransform>
 
diff -Naur orig/plugins/assistants/Assistants/InfiniteRulerAssistant.cc patched/plugins/assistants/Assistants/InfiniteRulerAssistant.cc
--- orig/plugins/assistants/Assistants/InfiniteRulerAssistant.cc	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/assistants/Assistants/InfiniteRulerAssistant.cc	2020-06-03 01:34:00.594376246 +0200
@@ -25,6 +25,7 @@
 #include <klocalizedstring.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QLinearGradient>
 #include <QTransform>
 
diff -Naur orig/plugins/assistants/Assistants/kis_assistant_tool.cc patched/plugins/assistants/Assistants/kis_assistant_tool.cc
--- orig/plugins/assistants/Assistants/kis_assistant_tool.cc	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/assistants/Assistants/kis_assistant_tool.cc	2020-06-03 01:34:00.594376246 +0200
@@ -21,6 +21,7 @@
 #include <kis_assistant_tool.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QXmlStreamReader>
 #include <QXmlStreamWriter>
 #include <QStandardPaths>
diff -Naur orig/plugins/assistants/Assistants/ParallelRulerAssistant.cc patched/plugins/assistants/Assistants/ParallelRulerAssistant.cc
--- orig/plugins/assistants/Assistants/ParallelRulerAssistant.cc	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/assistants/Assistants/ParallelRulerAssistant.cc	2020-06-03 01:34:00.594376246 +0200
@@ -25,6 +25,7 @@
 #include <klocalizedstring.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QLinearGradient>
 #include <QTransform>
 
diff -Naur orig/plugins/assistants/Assistants/PerspectiveAssistant.cc patched/plugins/assistants/Assistants/PerspectiveAssistant.cc
--- orig/plugins/assistants/Assistants/PerspectiveAssistant.cc	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/assistants/Assistants/PerspectiveAssistant.cc	2020-06-03 01:34:00.594376246 +0200
@@ -24,6 +24,7 @@
 #include <klocalizedstring.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QLinearGradient>
 #include <QTransform>
 
diff -Naur orig/plugins/assistants/Assistants/RulerAssistant.cc patched/plugins/assistants/Assistants/RulerAssistant.cc
--- orig/plugins/assistants/Assistants/RulerAssistant.cc	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/assistants/Assistants/RulerAssistant.cc	2020-06-03 01:34:00.594376246 +0200
@@ -23,6 +23,7 @@
 #include <klocalizedstring.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QLinearGradient>
 #include <QTransform>
 
diff -Naur orig/plugins/assistants/Assistants/SplineAssistant.cc patched/plugins/assistants/Assistants/SplineAssistant.cc
--- orig/plugins/assistants/Assistants/SplineAssistant.cc	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/assistants/Assistants/SplineAssistant.cc	2020-06-03 01:34:00.594376246 +0200
@@ -23,6 +23,7 @@
 #include <klocalizedstring.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QLinearGradient>
 #include <QTransform>
 
diff -Naur orig/plugins/assistants/Assistants/VanishingPointAssistant.cc patched/plugins/assistants/Assistants/VanishingPointAssistant.cc
--- orig/plugins/assistants/Assistants/VanishingPointAssistant.cc	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/assistants/Assistants/VanishingPointAssistant.cc	2020-06-03 01:34:00.594376246 +0200
@@ -25,6 +25,7 @@
 #include <klocalizedstring.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QLinearGradient>
 #include <QTransform>
 
diff -Naur orig/plugins/dockers/histogram/histogramdockerwidget.cpp patched/plugins/dockers/histogram/histogramdockerwidget.cpp
--- orig/plugins/dockers/histogram/histogramdockerwidget.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/dockers/histogram/histogramdockerwidget.cpp	2020-06-03 01:34:00.594376246 +0200
@@ -24,6 +24,7 @@
 #include <algorithm>
 #include <QTime>
 #include <QPainter>
+#include <QPainterPath>
 #include <functional>
 
 #include "KoChannelInfo.h"
diff -Naur orig/plugins/flake/pathshapes/enhancedpath/EnhancedPathShape.cpp patched/plugins/flake/pathshapes/enhancedpath/EnhancedPathShape.cpp
--- orig/plugins/flake/pathshapes/enhancedpath/EnhancedPathShape.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/flake/pathshapes/enhancedpath/EnhancedPathShape.cpp	2020-06-03 01:34:00.594376246 +0200
@@ -30,6 +30,8 @@
 #include "EnhancedPathHandle.h"
 #include "EnhancedPathFormula.h"
 
+#include <QPainterPath>
+
 #include <KoXmlNS.h>
 #include <KoXmlWriter.h>
 #include <KoXmlReader.h>
diff -Naur orig/plugins/flake/textshape/TextShape.cpp patched/plugins/flake/textshape/TextShape.cpp
--- orig/plugins/flake/textshape/TextShape.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/flake/textshape/TextShape.cpp	2020-06-03 01:34:00.594376246 +0200
@@ -57,6 +57,7 @@
 #include <QApplication>
 #include <QFont>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPen>
 #include <QTextLayout>
 
diff -Naur orig/plugins/paintops/curvebrush/curve_brush.cpp patched/plugins/paintops/curvebrush/curve_brush.cpp
--- orig/plugins/paintops/curvebrush/curve_brush.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/paintops/curvebrush/curve_brush.cpp	2020-06-03 01:34:00.594376246 +0200
@@ -18,6 +18,8 @@
 
 #include "curve_brush.h"
 
+#include <QPainterPath>
+
 #include <KoColor.h>
 #include <KoColorSpace.h>
 
diff -Naur orig/plugins/paintops/curvebrush/kis_curve_paintop.cpp patched/plugins/paintops/curvebrush/kis_curve_paintop.cpp
--- orig/plugins/paintops/curvebrush/kis_curve_paintop.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/paintops/curvebrush/kis_curve_paintop.cpp	2020-06-03 01:34:00.594376246 +0200
@@ -20,6 +20,7 @@
 
 #include <cmath>
 
+#include <QPainterPath>
 #include <QRect>
 
 #include <kis_image.h>
diff -Naur orig/plugins/paintops/experiment/kis_experiment_paintop.cpp patched/plugins/paintops/experiment/kis_experiment_paintop.cpp
--- orig/plugins/paintops/experiment/kis_experiment_paintop.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/paintops/experiment/kis_experiment_paintop.cpp	2020-06-03 01:34:00.594376246 +0200
@@ -22,6 +22,8 @@
 
 #include <cmath>
 
+#include <QPainterPath>
+
 #include <KoCompositeOpRegistry.h>
 
 #include <kis_debug.h>
diff -Naur orig/plugins/paintops/experiment/kis_experiment_paintop.h patched/plugins/paintops/experiment/kis_experiment_paintop.h
--- orig/plugins/paintops/experiment/kis_experiment_paintop.h	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/paintops/experiment/kis_experiment_paintop.h	2020-06-03 01:34:00.594376246 +0200
@@ -19,6 +19,8 @@
 #ifndef KIS_EXPERIMENT_PAINTOP_H_
 #define KIS_EXPERIMENT_PAINTOP_H_
 
+#include <QPainterPath>
+
 #include <klocalizedstring.h>
 #include <brushengine/kis_paintop.h>
 #include <kis_types.h>
diff -Naur orig/plugins/tools/defaulttool/defaulttool/DefaultTool.cpp patched/plugins/tools/defaulttool/defaulttool/DefaultTool.cpp
--- orig/plugins/tools/defaulttool/defaulttool/DefaultTool.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/tools/defaulttool/defaulttool/DefaultTool.cpp	2020-06-03 01:34:00.594376246 +0200
@@ -67,6 +67,7 @@
 
 #include <KoIcon.h>
 
+#include <QPainterPath>
 #include <QPointer>
 #include <QAction>
 #include <QKeyEvent>
diff -Naur orig/plugins/tools/defaulttool/defaulttool/SelectionDecorator.cpp patched/plugins/tools/defaulttool/defaulttool/SelectionDecorator.cpp
--- orig/plugins/tools/defaulttool/defaulttool/SelectionDecorator.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/tools/defaulttool/defaulttool/SelectionDecorator.cpp	2020-06-03 01:34:00.594376246 +0200
@@ -21,6 +21,8 @@
 
 #include "SelectionDecorator.h"
 
+#include <QPainterPath>
+
 #include <KoShape.h>
 #include <KoSelection.h>
 #include <KoResourcePaths.h>
diff -Naur orig/plugins/tools/karbonplugins/tools/CalligraphyTool/KarbonCalligraphicShape.cpp patched/plugins/tools/karbonplugins/tools/CalligraphyTool/KarbonCalligraphicShape.cpp
--- orig/plugins/tools/karbonplugins/tools/CalligraphyTool/KarbonCalligraphicShape.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/tools/karbonplugins/tools/CalligraphyTool/KarbonCalligraphicShape.cpp	2020-06-03 01:34:00.594376246 +0200
@@ -28,6 +28,7 @@
 
 #include <QDebug>
 #include <QColor>
+#include <QPainterPath>
 
 #include <cmath>
 #include <cstdlib>
diff -Naur orig/plugins/tools/karbonplugins/tools/CalligraphyTool/KarbonCalligraphyTool.h patched/plugins/tools/karbonplugins/tools/CalligraphyTool/KarbonCalligraphyTool.h
--- orig/plugins/tools/karbonplugins/tools/CalligraphyTool/KarbonCalligraphyTool.h	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/tools/karbonplugins/tools/CalligraphyTool/KarbonCalligraphyTool.h	2020-06-03 01:34:00.594376246 +0200
@@ -22,6 +22,7 @@
 
 #include <KoToolBase.h>
 #include <KoPathShape.h>
+#include <QPainterPath>
 #include <QPointer>
 
 #include "KarbonCalligraphyOptionWidget.h"
diff -Naur orig/plugins/tools/svgtexttool/SvgTextTool.cpp patched/plugins/tools/svgtexttool/SvgTextTool.cpp
--- orig/plugins/tools/svgtexttool/SvgTextTool.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/tools/svgtexttool/SvgTextTool.cpp	2020-06-03 01:34:00.594376246 +0200
@@ -23,6 +23,7 @@
 #include "SvgTextChangeCommand.h"
 
 #include <QLabel>
+#include <QPainterPath>
 #include <QToolButton>
 #include <QGridLayout>
 #include <QVBoxLayout>
diff -Naur orig/plugins/tools/tool_transform2/kis_free_transform_strategy.cpp patched/plugins/tools/tool_transform2/kis_free_transform_strategy.cpp
--- orig/plugins/tools/tool_transform2/kis_free_transform_strategy.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/tools/tool_transform2/kis_free_transform_strategy.cpp	2020-06-03 01:34:00.594376246 +0200
@@ -20,6 +20,7 @@
 
 #include <QPointF>
 #include <QPainter>
+#include <QPainterPath>
 #include <QMatrix4x4>
 
 #include <KoResourcePaths.h>
diff -Naur orig/plugins/tools/tool_transform2/kis_liquify_paint_helper.cpp patched/plugins/tools/tool_transform2/kis_liquify_paint_helper.cpp
--- orig/plugins/tools/tool_transform2/kis_liquify_paint_helper.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/tools/tool_transform2/kis_liquify_paint_helper.cpp	2020-06-03 01:51:41.188156211 +0200
@@ -15,9 +15,11 @@
  *  along with this program; if not, write to the Free Software
  *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
  */
-
 #include "kis_liquify_paint_helper.h"
 
+#include <QElapsedTimer>
+#include <QPainterPath>
+
 #include "kis_algebra_2d.h"
 #include "KoPointerEvent.h"
 #include <brushengine/kis_paint_information.h>
@@ -44,7 +46,7 @@
     const KisCoordinatesConverter *converter;
     QScopedPointer<KisPaintingInformationBuilder> infoBuilder;
 
-    QTime strokeTime;
+    QElapsedTimer strokeTime;
 
     bool hasPaintedAtLeastOnce;
 
diff -Naur orig/plugins/tools/tool_transform2/kis_liquify_transform_strategy.cpp patched/plugins/tools/tool_transform2/kis_liquify_transform_strategy.cpp
--- orig/plugins/tools/tool_transform2/kis_liquify_transform_strategy.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/tools/tool_transform2/kis_liquify_transform_strategy.cpp	2020-06-03 01:34:00.598376253 +0200
@@ -22,6 +22,7 @@
 
 #include <QPointF>
 #include <QPainter>
+#include <QPainterPath>
 
 #include "KoPointerEvent.h"
 
diff -Naur orig/plugins/tools/tool_transform2/kis_perspective_transform_strategy.cpp patched/plugins/tools/tool_transform2/kis_perspective_transform_strategy.cpp
--- orig/plugins/tools/tool_transform2/kis_perspective_transform_strategy.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/tools/tool_transform2/kis_perspective_transform_strategy.cpp	2020-06-03 01:34:00.598376253 +0200
@@ -20,6 +20,7 @@
 
 #include <QPointF>
 #include <QPainter>
+#include <QPainterPath>
 #include <QMatrix4x4>
 #include <QVector2D>
 
diff -Naur orig/plugins/tools/tool_transform2/kis_transform_strategy_base.cpp patched/plugins/tools/tool_transform2/kis_transform_strategy_base.cpp
--- orig/plugins/tools/tool_transform2/kis_transform_strategy_base.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/tools/tool_transform2/kis_transform_strategy_base.cpp	2020-06-03 01:34:00.598376253 +0200
@@ -19,6 +19,7 @@
 #include "kis_transform_strategy_base.h"
 
 #include <QImage>
+#include <QPainterPath>
 #include <QTransform>
 #include "KoPointerEvent.h"
 
diff -Naur orig/plugins/tools/tool_transform2/kis_transform_utils.cpp patched/plugins/tools/tool_transform2/kis_transform_utils.cpp
--- orig/plugins/tools/tool_transform2/kis_transform_utils.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/tools/tool_transform2/kis_transform_utils.cpp	2020-06-03 01:34:00.598376253 +0200
@@ -19,6 +19,7 @@
 #include "kis_transform_utils.h"
 
 #include <cmath>
+#include <QPainterPath>
 #include <QTransform>
 #include <KoUnit.h>
 #include "tool_transform_args.h"
diff -Naur orig/plugins/tools/tool_transform2/kis_warp_transform_strategy.cpp patched/plugins/tools/tool_transform2/kis_warp_transform_strategy.cpp
--- orig/plugins/tools/tool_transform2/kis_warp_transform_strategy.cpp	2020-03-23 13:56:54.000000000 +0100
+++ patched/plugins/tools/tool_transform2/kis_warp_transform_strategy.cpp	2020-06-03 01:34:00.598376253 +0200
@@ -22,6 +22,7 @@
 
 #include <QPointF>
 #include <QPainter>
+#include <QPainterPath>
 
 #include "kis_coordinates_converter.h"
 #include "tool_transform_args.h"
