--- a/stdext.h
+++ b/stdext.h
@@ -3,9 +3,39 @@
 
 #if __GNUC__ >= 4
 
+#if __GNUC__ >= 6
+
+#include <memory>
+#include <functional>
+
+namespace std {
+namespace tr1 {
+
+	using std::bind;
+	using std::shared_ptr;
+	using std::weak_ptr;
+
+	namespace placeholders {
+		using std::placeholders::_1;
+		using std::placeholders::_2;
+		using std::placeholders::_3;
+		using std::placeholders::_4;
+		using std::placeholders::_5;
+		using std::placeholders::_6;
+		using std::placeholders::_7;
+		using std::placeholders::_8;
+		using std::placeholders::_9;
+	}
+} // namespace tr1
+} // namespace std
+
+#else
+
 #	include <tr1/functional>
 #	include <tr1/memory>
 
+# endif
+
 #else
 
 #	include <boost/version.hpp>
