diff -Naur orig/Makefile.in patched/Makefile.in
--- orig/Makefile.in	2015-09-18 14:56:29.000000000 +0200
+++ patched/Makefile.in	2015-10-23 20:22:02.591716807 +0200
@@ -96,7 +96,7 @@
 toolsdir = @bindir@/svn-tools
 
 # where to install pkg-config files
-pkgconfig_dir = $(datadir)/pkgconfig
+pkgconfig_dir = $(libdir)/pkgconfig
 
 javahl_javadir = @libdir@/svn-javahl
 javahl_javahdir = @libdir@/svn-javahl/include
