diff -Nur orig/thirdparty/fetch-geoip-db.sh mod/thirdparty/fetch-geoip-db.sh
--- orig/thirdparty/fetch-geoip-db.sh	2019-03-14 20:32:26.000000000 +0100
+++ mod/thirdparty/fetch-geoip-db.sh	2019-10-03 15:27:47.778519521 +0200
@@ -13,10 +13,5 @@
 if [ ! -e $filename ] || [ -n "$(find . -name $filename -mtime +30 -print)" ]; then
 	rm -f $filename
 	echo "Updating GeoIP country database from MaxMind."
-	if command -v curl >/dev/null 2>&1; then
-		curl -s -L -O http://geolite.maxmind.com/download/geoip/database/$filename
-	else
-		wget -cq http://geolite.maxmind.com/download/geoip/database/$filename
-	fi
+    mv ../../$filename .
 fi
-
diff -Nur orig/thirdparty/fetch-thirdparty-deps.sh mod/thirdparty/fetch-thirdparty-deps.sh
--- orig/thirdparty/fetch-thirdparty-deps.sh	2019-03-14 20:32:26.000000000 +0100
+++ mod/thirdparty/fetch-thirdparty-deps.sh	2019-10-03 15:29:57.974325741 +0200
@@ -81,33 +81,17 @@
 
 if [ ! -f SDL2-CS.dll ] || [ ! -f SDL2-CS.dll.config ]; then
 	echo "Fetching SDL2-CS from GitHub."
-	if command -v curl >/dev/null 2>&1; then
-		curl -s -L -O https://github.com/OpenRA/SDL2-CS/releases/download/20161223/SDL2-CS.dll
-		curl -s -L -O https://github.com/OpenRA/SDL2-CS/releases/download/20161223/SDL2-CS.dll.config
-	else
-		wget -cq https://github.com/OpenRA/SDL2-CS/releases/download/20161223/SDL2-CS.dll
-		wget -cq https://github.com/OpenRA/SDL2-CS/releases/download/20161223/SDL2-CS.dll.config
-	fi
+	mv ../../SDL2-CS* .
 fi
 
 if [ ! -f OpenAL-CS.dll ] || [ ! -f OpenAL-CS.dll.config ]; then
 	echo "Fetching OpenAL-CS from GitHub."
-	if command -v curl >/dev/null 2>&1; then
-		curl -s -L -O https://github.com/OpenRA/OpenAL-CS/releases/download/20151227/OpenAL-CS.dll
-		curl -s -L -O https://github.com/OpenRA/OpenAL-CS/releases/download/20151227/OpenAL-CS.dll.config
-	else
-		wget -cq https://github.com/OpenRA/OpenAL-CS/releases/download/20151227/OpenAL-CS.dll
-		wget -cq https://github.com/OpenRA/OpenAL-CS/releases/download/20151227/OpenAL-CS.dll.config
-	fi
+	mv ../../OpenAL-CS* .
 fi
 
 if [ ! -f Eluant.dll ]; then
 	echo "Fetching Eluant from GitHub."
-	if command -v curl >/dev/null 2>&1; then
-		curl -s -L -O https://github.com/OpenRA/Eluant/releases/download/20160124/Eluant.dll
-	else
-		wget -cq https://github.com/OpenRA/Eluant/releases/download/20160124/Eluant.dll
-	fi
+	mv ../../Eluant.dll .
 fi
 
 if [ ! -f rix0rrr.BeaconLib.dll ]; then
diff -Nur orig/thirdparty/noget.sh mod/thirdparty/noget.sh
--- orig/thirdparty/noget.sh	2019-03-14 20:32:26.000000000 +0100
+++ mod/thirdparty/noget.sh	2019-10-03 15:31:38.098173351 +0200
@@ -16,4 +16,5 @@
 	wget -cq https://nuget.org/api/v2/package/"$archive"/"$version" -O "$archive.zip"
 fi
 mkdir -p "$archive"
+mv ../../"$archive".zip ./
 unzip -o -qq "$archive.zip" -d "$archive" && rm "$archive.zip"
