diff -Naur orig/etc/useradd patched/etc/useradd
--- orig/etc/useradd	2018-10-10 18:19:12.000000000 +0200
+++ patched/etc/useradd	2019-08-05 00:41:06.171943214 +0200
@@ -1,5 +1,5 @@
 # useradd defaults file
-GROUP=1000
+GROUP=999
 HOME=/home
 INACTIVE=-1
 EXPIRE=
diff -Naur orig/src/useradd.c patched/src/useradd.c
--- orig/src/useradd.c	2019-05-04 07:38:23.000000000 +0200
+++ patched/src/useradd.c	2019-08-05 00:41:06.171943214 +0200
@@ -96,8 +96,8 @@
 /*
  * These defaults are used if there is no defaults file.
  */
-static gid_t def_group = 100;
-static const char *def_gname = "other";
+static gid_t def_group = 999;
+static const char *def_gname = "users";
 static const char *def_home = "/home";
 static const char *def_shell = "";
 static const char *def_template = SKEL_DIR;
