diff -Naur orig/src/useradd.c patched/src/useradd.c
--- orig/src/useradd.c	2019-08-05 00:41:06.171943214 +0200
+++ patched/src/useradd.c	2019-08-05 00:43:22.021813575 +0200
@@ -2175,7 +2175,6 @@
 		const char *spool;
 		char *file;
 		int fd;
-		struct group *gr;
 		gid_t gid;
 		mode_t mode;
 
@@ -2194,16 +2193,8 @@
 			return;
 		}
 
-		gr = prefix_getgrnam ("mail"); /* local, no need for xgetgrnam */
-		if (NULL == gr) {
-			fputs (_("Group 'mail' not found. Creating the user mailbox file with 0600 mode.\n"),
-			       stderr);
-			gid = user_gid;
-			mode = 0600;
-		} else {
-			gid = gr->gr_gid;
-			mode = 0660;
-		}
+		gid = user_gid;
+		mode = 0600;
 
 		if (   (fchown (fd, user_id, gid) != 0)
 		    || (fchmod (fd, mode) != 0)) {
