diff -Naur a/src/useradd.c b/src/useradd.c
--- a/src/useradd.c	2022-07-16 15:46:27.501004694 +0200
+++ b/src/useradd.c	2022-07-16 15:51:20.838084132 +0200
@@ -145,7 +145,7 @@
     lflg = false,		/* do not add user to lastlog/faillog databases */
     mflg = false,		/* create user's home directory if it doesn't exist */
     Mflg = false,		/* do not create user's home directory even if CREATE_HOME is set */
-    Nflg = false,		/* do not create a group having the same name as the user, but add the user to def_group (or the group specified with -g) */
+    Nflg = true,		/* do not create a group having the same name as the user, but add the user to def_group (or the group specified with -g) */
     oflg = false,		/* permit non-unique user ID to be specified with -u */
     rflg = false,		/* create a system account */
     sflg = false,		/* shell program for new account */
