diff -Naur orig/src/useradd.c patched/src/useradd.c
--- orig/src/useradd.c	2019-08-05 00:43:22.021813575 +0200
+++ patched/src/useradd.c	2019-08-05 00:44:39.790884287 +0200
@@ -161,7 +161,7 @@
     lflg = false,		/* do not add user to lastlog/faillog databases */
     mflg = false,		/* create user's home directory if it doesn't exist */
     Mflg = false,		/* do not create user's home directory even if CREATE_HOME is set */
-    Nflg = false,		/* do not create a group having the same name as the user, but add the user to def_group (or the group specified with -g) */
+    Nflg = true,		/* do not create a group having the same name as the user, but add the user to def_group (or the group specified with -g) */
     oflg = false,		/* permit non-unique user ID to be specified with -u */
     rflg = false,		/* create a system account */
     sflg = false,		/* shell program for new account */
