diff -Naur orig/src/useradd.c patched/src/useradd.c
--- orig/src/useradd.c	2017-06-26 17:24:05.160295400 +0200
+++ patched/src/useradd.c	2017-06-26 17:31:56.102633377 +0200
@@ -1965,8 +1965,7 @@
 			fail_exit (E_HOMEDIR);
 		}
 		chown (user_home, user_id, user_gid);
-		chmod (user_home,
-		       0777 & ~getdef_num ("UMASK", GETDEF_DEFAULT_UMASK));
+		chmod (user_home, 0700);
 		home_added = true;
 #ifdef WITH_AUDIT
 		audit_logger (AUDIT_ADD_USER, Prog,
