diff -Naur orig/src/useradd.c patched/src/useradd.c
--- orig/src/useradd.c	2018-04-29 20:26:56.467555856 +0200
+++ patched/src/useradd.c	2018-04-29 20:30:35.130314514 +0200
@@ -2040,8 +2040,7 @@
 			fail_exit (E_HOMEDIR);
 		}
 		(void) chown (prefix_user_home, user_id, user_gid);
-		chmod (prefix_user_home,
-		       0777 & ~getdef_num ("UMASK", GETDEF_DEFAULT_UMASK));
+		chmod (prefix_user_home, 0700);
 		home_added = true;
 #ifdef WITH_AUDIT
 		audit_logger (AUDIT_ADD_USER, Prog,
