diff -Nur orig/autogen.sh mod/autogen.sh
--- orig/autogen.sh	2017-01-09 00:32:15.000000000 +0100
+++ mod/autogen.sh	2018-01-10 11:23:51.011574154 +0100
@@ -23,7 +23,6 @@
 fi
 
 aclocal --install || exit 1
-gtkdocize --copy || exit 1
 intltoolize --force --copy --automake || exit 1
 autoreconf --verbose --force --install || exit 1
 
diff -Nur orig/configure.ac mod/configure.ac
--- orig/configure.ac	2017-08-23 00:03:16.000000000 +0200
+++ mod/configure.ac	2018-01-10 11:24:29.919506944 +0100
@@ -207,15 +207,11 @@
 AC_SUBST([AM_CXXFLAGS])
 AC_SUBST([AM_LDFLAGS])
 
-GTK_DOC_CHECK([1.25],[--flavour no-tmpl])
-
 AC_CONFIG_FILES([
 Makefile
 libgdata.pc
 gdata/gdata-version.h
 gdata/tests/Makefile
 po/Makefile.in
-docs/Makefile
-docs/reference/Makefile
 ])
 AC_OUTPUT
diff -Nur orig/Makefile.am mod/Makefile.am
--- orig/Makefile.am	2017-08-22 23:40:51.000000000 +0200
+++ mod/Makefile.am	2018-01-10 11:24:45.551479941 +0100
@@ -1,4 +1,4 @@
-SUBDIRS = . po docs gdata/tests
+SUBDIRS = . po gdata/tests
 
 ACLOCAL_AMFLAGS = --install -I m4 ${ACLOCAL_FLAGS}
 
