diff -Nur orig/media/openttd.desktop mod/media/openttd.desktop
--- orig/media/openttd.desktop	2021-04-01 14:33:44.000000000 +0200
+++ mod/media/openttd.desktop	2021-04-04 08:48:25.672267412 +0200
@@ -6,6 +6,6 @@
 Icon=${BINARY_NAME}
 Exec=${BINARY_NAME}
 Terminal=false
-Categories=Game;
+Categories=Game;Simulation;
 Keywords=game;simulation;transport;tycoon;deluxe;economics;multiplayer;money;train;ship;bus;truck;aircraft;cargo;
 @Comment_STR_DESKTOP_SHORTCUT_COMMENT@
