diff -Naur memtest86+-5.31b-orig/controller.c memtest86+-5.31b/controller.c
--- memtest86+-5.31b-orig/controller.c	2020-04-12 17:14:51.000000000 +0200
+++ memtest86+-5.31b/controller.c	2020-05-09 15:55:41.522452058 +0200
@@ -1426,7 +1426,7 @@
     unsigned long dramchr;
     float clockratio;
     double dramclock;
-    unsigned int dummy[3];
+    unsigned int dummy[4];
     int ram_type;
 
     float coef = 10;
@@ -2853,13 +2853,13 @@
 
 static void poll_timings_i875(void) {
 
-    ulong dev6, dev62;
+    ulong dev6;
     ulong temp;
     float cas;
     int rcd, rp, ras, chan;
     long *ptr, *ptr2;
 
-    pci_conf_read( 0, 6, 0, 0x40, 4, &dev62);
+    pci_conf_read( 0, 6, 0, 0x40, 4, &dev6);
     ptr2=(long*)(dev6+0x68);
 
     /* Read the MMR Base Address & Define the pointer */
