diff -Naur a/meson.build b/meson.build
--- a/meson.build	2022-09-17 15:06:38.000000000 +0200
+++ b/meson.build	2022-10-03 00:30:21.436459499 +0200
@@ -27,9 +27,12 @@
 # Dependencies
 gtkmm4 = dependency('gtkmm-4.0', version: '>= 4.6')
 giomm = dependency('giomm-2.68', version: '>= 2.68')
+thread_dep = dependency('threads')
+nautilus_plugin_feature = get_option('nautilus-plugin')
+if nautilus_plugin_feature.enabled()
 libnautilus_extensions4 = dependency('libnautilus-extension-4',
                                       version: '>= 43')
-thread_dep = dependency('threads')
+endif
 
 # Compiler
 compiler = meson.get_compiler('cpp')
diff -Naur a/meson_options.txt b/meson_options.txt
--- a/meson_options.txt	2022-09-17 15:06:38.000000000 +0200
+++ b/meson_options.txt	2022-10-03 00:19:37.982393928 +0200
@@ -2,6 +2,10 @@
     type : 'feature',
     value : 'enabled',
     description : 'Enables support for editing extended user attributes')
+option('nautilus-plugin',
+    type : 'feature',
+    value : 'enabled',
+    description: 'Enable nautilus extension plugin')
 option('nautilus-extension-dir',
     type: 'string',
     value : '',
diff -Naur a/src/meson.build b/src/meson.build
--- a/src/meson.build	2022-09-17 15:06:38.000000000 +0200
+++ b/src/meson.build	2022-10-03 00:34:03.893174852 +0200
@@ -56,6 +56,8 @@
 
 
 # Plugin for nautilus
+nautilusextensiondir = get_option('nautilus-extension-dir')
+if nautilus_plugin_feature.enabled()
 plugin_sources = utility_libraries
 plugin_sources += ['nautilus_menu_provider.cpp',
                    'nautilus_acl_model.cpp',
@@ -64,7 +66,6 @@
                    'nautilus_extension.cpp',
                    'i18n.cpp']
 
-nautilusextensiondir = get_option('nautilus-extension-dir')
 if nautilusextensiondir == '' 
   nautilusextensiondir = libnautilus_extensions4.get_variable(pkgconfig : 'extensiondir')
 endif
@@ -75,6 +76,7 @@
   install_dir : nautilusextensiondir,
   include_directories : incdir,
   dependencies : [gtkmm4, giomm, libnautilus_extensions4, libacl, thread_dep])
+endif
 
 # Desktop file
 desktop_file_data = configuration_data()
