diff -Naur orig/src-qt5/lumina-calculator.desktop patched/src-qt5/lumina-calculator.desktop
--- orig/src-qt5/lumina-calculator.desktop	2019-04-30 03:36:35.000000000 +0200
+++ patched/src-qt5/lumina-calculator.desktop	2019-04-30 03:38:24.765454727 +0200
@@ -4,7 +4,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=Office;Utility;
-Name=Calculator
+Categories=Utility;Calculator;
+Name=Lumina Calculator
 GenericName=Scientific Calculator for Lumina
 Comment=Perform mathematical calculations
