diff -Naur orig/package/gav.desktop patched/package/gav.desktop
--- orig/package/gav.desktop	2002-12-07 13:13:29.000000000 +0100
+++ patched/package/gav.desktop	2016-02-19 01:37:05.369512421 +0100
@@ -1,10 +1,11 @@
 [Desktop Entry]
 Encoding=UTF-8
-Categories=Application;Game;X-Red-Hat-Extra
+Categories=Game;X-Red-Hat-Extra
 X-Desktop-File-Install-Version=0.2
 Name=Arcade Volleyball
 Comment=GPL remake of the DOS classic
-Icon=gav.png
+Comment[hu]=Röplabda játék
+Icon=gav
 Exec=gav
 Terminal=0
 Type=Application
