diff -Naur orig/gnome-system-monitor.desktop.in patched/gnome-system-monitor.desktop.in
--- orig/gnome-system-monitor.desktop.in 2018-02-11 08:58:36.000000000 +0100
+++ patched/gnome-system-monitor.desktop.in	2018-06-02 22:46:06.418519733 +0200
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Name=System Monitor
+Name=GNOME System Monitor
 Comment=View current processes and monitor system state
 TryExec=gnome-system-monitor
 Exec=gnome-system-monitor
