--- gnome-system-monitor.orig/gnome-system-monitor.desktop.in.in	2007-03-12 07:37:24.000000000 +0100
+++ gnome-system-monitor/gnome-system-monitor.desktop.in.in	2009-03-22 16:40:00.000000000 +0100
@@ -1,5 +1,6 @@
 [Desktop Entry]
-_Name=System Monitor
+Name=GNOME System Monitor
+Name[hu]=GNOME Rendszermonitor
 _Comment=View current processes and monitor system state
 TryExec=gnome-system-monitor
 Exec=gnome-system-monitor
@@ -8,7 +9,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;GTK;System;Monitor;
+Categories=GTK;GNOME;System;Monitor;
 NotShowIn=KDE;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=system-monitor
