diff -Nur orig/CMakeLists.txt mod/CMakeLists.txt
--- orig/CMakeLists.txt	2019-07-06 23:13:03.000000000 +0200
+++ mod/CMakeLists.txt	2019-10-07 08:34:46.374556510 +0200
@@ -21,6 +21,11 @@
   CMAKE_POLICY(SET CMP0068 OLD)
 ENDIF(POLICY CMP0068)
 
+# Force the use of the legacy libGL.so for CMake >= 3.11
+IF(POLICY CMP0072)
+  CMAKE_POLICY(SET CMP0072 OLD)
+ENDIF(POLICY CMP0072)
+
 # enable the use of ccache for Tulip developers to speed up the build process
 IF(NOT MSVC)
   SET(TULIP_USE_CCACHE OFF CACHE BOOL "Do you want to use ccache to speed up the build process during Tulip development ? [OFF|ON]")
@@ -307,6 +312,15 @@
 
   # Glew
   FIND_PACKAGE(GLEW REQUIRED)
+  # if needed, when using cmake 3.15, define some undefined glew variables
+  IF(NOT ${CMAKE_VERSION} VERSION_LESS "3.15.0")
+    IF(NOT GLEW_INCLUDE_DIR)
+      GET_TARGET_PROPERTY(GLEW_INCLUDE_DIR GLEW::GLEW INTERFACE_INCLUDE_DIRECTORIES)
+    ENDIF(NOT GLEW_INCLUDE_DIR)
+    IF(NOT GLEW_LIBRARY)
+      SET(GLEW_LIBRARY GLEW::GLEW)
+    ENDIF(NOT GLEW_LIBRARY)
+  ENDIF(NOT ${CMAKE_VERSION} VERSION_LESS "3.15.0")
 
   # Jpeg
   FIND_PACKAGE(JPEG REQUIRED)
@@ -610,7 +624,7 @@
                           COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/bundlers/win/win_bundle.bat ${CMAKE_CURRENT_BINARY_DIR}/bundlers/win/win_bundle.bat
                           COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/bundlers/win/FileAssociation.nsh ${CMAKE_CURRENT_BINARY_DIR}/FileAssociation.nsh
                           COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/COPYING.LESSER ${CMAKE_CURRENT_BINARY_DIR}/bundlers/win/COPYING.LESSER
-                          COMMAND cmd /C "\"win_bundle.bat \\\"${NSIS_PATH}\\\" \\\"${TLP_DIR}\\\" \\\"${BINARY_DIR}\\\" ${DEBUG_MODE}\""
+                          COMMAND cmd //C win_bundle.bat \"${NSIS_PATH}\" \"${TLP_DIR}\" \"${BINARY_DIR}\" ${DEBUG_MODE}
                           WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bundlers/win)
 
       ELSE(NOT TULIP_BUILD_PYTHON_COMPONENTS OR EXISTS "${NSIS_PATH}/Plugins/inetc.dll" OR EXISTS "${NSIS_PATH}/Plugins/x86-ansi/inetc.dll")
