diff -Nur orig/library/tulip-python/src/ConsoleUtilsModule.cpp mod/library/tulip-python/src/ConsoleUtilsModule.cpp
--- orig/library/tulip-python/src/ConsoleUtilsModule.cpp	2019-07-06 23:13:03.000000000 +0200
+++ mod/library/tulip-python/src/ConsoleUtilsModule.cpp	2020-01-22 08:25:34.420248680 +0100
@@ -166,7 +166,7 @@
     sizeof(consoleutils_ConsoleOutput),                               /*tp_basicsize*/
     0,                                                                /*tp_itemsize*/
     reinterpret_cast<destructor>(consoleutils_ConsoleOutput_dealloc), /*tp_dealloc*/
-    nullptr,                                                          /*tp_print*/
+    0,                                                          /*tp_print*/
     nullptr,                                                          /*tp_getattr*/
     nullptr,                                                          /*tp_setattr*/
     nullptr,                                                          /*tp_compare*/
@@ -265,7 +265,7 @@
     sizeof(consoleutils_ConsoleInput),                               /*tp_basicsize*/
     0,                                                               /*tp_itemsize*/
     reinterpret_cast<destructor>(consoleutils_ConsoleInput_dealloc), /*tp_dealloc*/
-    nullptr,                                                         /*tp_print*/
+    0,                                                         /*tp_print*/
     nullptr,                                                         /*tp_getattr*/
     nullptr,                                                         /*tp_setattr*/
     nullptr,                                                         /*tp_compare*/
