diff -Naur orig/nemo-python/configure.in patched/nemo-python/configure.in
--- orig/nemo-python/configure.in	2017-10-24 13:00:07.000000000 +0200
+++ patched/nemo-python/configure.in	2017-12-30 13:50:34.321982804 +0100
@@ -23,9 +23,6 @@
 AC_CHECK_PROG(HAVE_PKGCONFIG, pkg-config, yes, no)
 AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
 
-dnl Check for gtk-doc
-GTK_DOC_CHECK(1.9)
-
 dnl Give error and exit if we don't have pkgconfig
 if test "x$HAVE_PKGCONFIG" = "xno"; then
   AC_MSG_ERROR(you need to have pkgconfig installed !)
@@ -73,25 +70,11 @@
 AC_SUBST(NEMO_EXTENSION_DIR)
 AC_DEFINE_UNQUOTED(NEMO_EXTENSION_DIR, "$NEMO_EXTENSION_DIR", [Nemo' extensions directory])
 
-dnl ---------------------------------
-dnl Check for gtk-doc dependencies
-dnl ---------------------------------
-if test "${enable_gtk_doc}" != no; then
-    dnl Check for xsltproc
-    AC_PATH_PROG([XSLTPROC], [xsltproc])
-    if test -z "$XSLTPROC"; then
-        echo " disabling generation of docs"
-        enable_gtk_doc=no
-    fi
-fi
-
 AC_OUTPUT([
   Makefile
   examples/Makefile
   src/Makefile
   nemo-python.pc
-  docs/Makefile
-  docs/reference/entities.docbook
 ])
 
 echo
@@ -99,5 +82,4 @@
 echo
 echo "    Nemo Prefix: ${prefix}"
 echo "  PyGObject Version: ${PYGOBJECT_VERSION}"
-echo "      Documentation: ${enable_gtk_doc}"
 echo
diff -Naur orig/nemo-python/docs/Makefile.am patched/nemo-python/docs/Makefile.am
--- orig/nemo-python/docs/Makefile.am	2017-10-24 13:00:07.000000000 +0200
+++ patched/nemo-python/docs/Makefile.am	1970-01-01 01:00:00.000000000 +0100
@@ -1,102 +0,0 @@
-BUILDDIR = $(top_builddir)/docs
-
-REF_VERSION = $(VERSION)
-FULL_VERSION = $(VERSION)
-
-FIXXREF = $(srcdir)/xsl/fixxref.py
-HTML_STYLE = $(srcdir)/xsl/ref-html-style.xsl
-
-XML_FILES = 							\
-	reference/nemo-python-ref.xml			\
-	reference/nemo-python-overview.xml			\
-	reference/nemo-python-overview-example.xml			\
-	reference/nemo-python-overview-methods.xml			\
-	reference/nemo-python-class-reference.xml			\
-	reference/nemo-python-provider-reference.xml			\
-	reference/nemo-python-column.xml   \
-	reference/nemo-python-column-provider.xml   \
-	reference/nemo-python-menu.xml  \
-	reference/nemo-python-menu-item.xml \
-	reference/nemo-python-menu-provider.xml \
-	reference/nemo-python-property-page.xml \
-	reference/nemo-python-property-page-provider.xml \
-	reference/nemo-python-location-widget-provider.xml \
-	reference/nemo-python-file-info.xml \
-	reference/nemo-python-info-provider.xml \
-	reference/nemo-python-enum-reference.xml \
-	reference/nemo-python-operation-result.xml
-
-HTMLdir = $(datadir)/gtk-doc/html/nemo-python
-HTML_FILES =						\
-	html/index.html					\
-	html/index.sgml					\
-	html/nemo-python-overview.html \
-	html/nemo-python-overview-example.html \
-	html/nemo-python-overview-methods.html \
-	html/nemo-python-class-reference.html \
-	html/nemo-python-provider-reference.html \
-	html/class-nemo-python-column.html \
-	html/class-nemo-python-column-provider.html \
-	html/class-nemo-python-menu.html  \
-	html/class-nemo-python-menu-item.html  \
-	html/class-nemo-python-menu-provider.html  \
-	html/class-nemo-python-property-page.html  \
-	html/class-nemo-python-property-page-provider.html  \
-	html/class-nemo-python-location-widget-provider.html  \
-	html/class-nemo-python-file-info.html  \
-	html/class-nemo-python-info-provider.html  \
-	html/nemo-python-enum-reference.html \
-	html/enum-nemo-python-operation-result.html  \
-	html/nemo-python.devhelp
-HTML_DATA = $(HTMLFILES)
-
-XSL_FILES =             \
-	xsl/common.xsl      \
-	xsl/devhelp.xsl     \
-	xsl/html.xsl        \
-	xsl/pdf-style.xsl   \
-	xsl/pdf.xsl         \
-	xsl/style.css
-XSL_DATA = $(XSL_FILES) $(FIXXREF)
-XSLdir = $(BUILDDIR)/xsl
-
-CSS_FILES = xsl/style.css
-CSSdir = $(HTMLdir)
-CSS_DATA = $(CSS_FILES)
-
-BUILT_SOURCES = 		\
-	reference/builddate.xml	\
-	$(HTML_FILES)
-
-CLEANFILES = 			\
-	nemo-python-ref.*	\
-	reference/builddate.xml	\
-	$(HTML_FILES)
-
-EXTRA_DIST = 		\
-	$(XML_FILES)     \
-	$(XSL_FILES)    \
-	$(FIXXREF)      \
-	$(HTML_STYLE)
-
-REFERENCE_DEPS = 	\
-	reference 	\
-	$(XML_FILES)    \
-	$(XSL_FILES)    \
-	$(FIXXREF)
-
-if ENABLE_GTK_DOC
-
-reference/builddate.xml: $(REFERENCE_DEPS)
-	$(PYTHON) -c 'import datetime; print datetime.date.today()' > $@
-
-$(HTML_FILES): $(REFERENCE_DEPS)
-	xsltproc --nonet --xinclude -o $(BUILDDIR)/html/ \
-		 --path $(BUILDDIR)/reference:$(srcdir)/reference \
-		 --stringparam gtkdoc.bookname "nemo-python" \
-		 --stringparam gtkdoc.version ${REF_VERSION} \
-		 $(HTML_STYLE) $(srcdir)/reference/nemo-python-ref.xml
-	$(FIXXREF) -i xsl $(BUILDDIR)/html 
-	touch $@
-
-endif
diff -Naur orig/nemo-python/Makefile.am patched/nemo-python/Makefile.am
--- orig/nemo-python/Makefile.am	2017-10-24 13:00:07.000000000 +0200
+++ patched/nemo-python/Makefile.am	2017-12-30 13:50:55.350307615 +0100
@@ -2,13 +2,7 @@
 
 SUBDIRS = src examples
 
-if ENABLE_GTK_DOC
-SUBDIRS += docs
-endif
-
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = nemo-python.pc
 
 EXTRA_DIST = nemo-python.pc.in
-
-DISTCHECK_CONFIGURE_FLAGS=--enable-gtk-doc
