diff -Nur orig/nemo-image-converter/configure.ac mod/nemo-image-converter/configure.ac
--- orig/nemo-image-converter/configure.ac	2018-05-01 14:11:51.000000000 +0200
+++ mod/nemo-image-converter/configure.ac	2018-05-20 14:05:20.432034559 +0200
@@ -14,11 +14,6 @@
 AC_PROG_LIBTOOL
 AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
 
-GNOME_COMPILE_WARNINGS([maximum])
-GNOME_CXX_WARNINGS
-GNOME_DEBUG_CHECK
-GNOME_MAINTAINER_MODE_DEFINES
-
 AH_TEMPLATE([GETTEXT_PACKAGE], [Package name for gettext])
 GETTEXT_PACKAGE=nemo-extensions
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE")
