diff -Nur dark-oberon/src/doworkers.cpp new/src/doworkers.cpp
--- dark-oberon/src/doworkers.cpp	2006-11-13 01:14:29.000000000 +0100
+++ new/src/doworkers.cpp	2016-06-30 19:51:10.415879488 +0200
@@ -3261,7 +3261,7 @@
 
       player->IncPlayerUnitsCount(); //increase count of active units of player
 
-      if (!new_building) return false;
+      if (!new_building) return NULL;
 
       new_building->SetAggressivity(AM_IGNORE, true);
       new_building->AddToMap(true, false);
