diff -Nur orig/src/create_makefile.sh mod/src/create_makefile.sh
--- orig/src/create_makefile.sh	2008-11-17 21:13:39.000000000 +0100
+++ mod/src/create_makefile.sh	2017-06-15 13:59:08.723235119 +0200
@@ -12,7 +12,7 @@
 LIBPATHS='-L/usr/lib -L/usr/local/lib -L../libs'
 LIBRARIES='-lpthread -lglfw -lGL -lX11 -lGLU -lXrandr -lSDL -lSDL_mixer'
 CPP='g++ $(CPPFLAGS) $(INCLUDES) $(DEFINES)'
-DATA_DIR=''
+DATA_DIR='/usr/share/games/dark-oberon/'
 
 ##############################################################################
 
