diff -Nur orig/src/doconfig.h mod/src/doconfig.h
--- orig/src/doconfig.h	2018-02-14 11:26:18.000000000 +0100
+++ mod/src/doconfig.h	2018-02-14 13:03:07.786764950 +0100
@@ -89,7 +89,7 @@
 #define CFG_DEF_WARFOG_COLOR_B  15
 /** Default configuration's fullscreen.
  *  @sa TCONFIG::fullscreen */
-#define CFG_DEF_FULLSCREEN      true
+#define CFG_DEF_FULLSCREEN      false
 /** Default configuration's vertical synchronisation.
  *  @sa TCONFIG::vert_sync */
 #ifdef UNIX
