--- gcompris-15.02.orig/configure.ac
+++ gcompris-15.02/configure.ac
@@ -389,7 +389,7 @@ if test x$with_sdlmixer = xyes; then
   AUDIO_LIBS="$AUDIO_LIBS -lSDL_mixer"
 else
   dnl Default is gstreamer
-  PKG_CHECK_MODULES(AUDIO, gstreamer-0.10,, AC_MSG_ERROR([*** GSTREAMER not found!]))
+  PKG_CHECK_MODULES(AUDIO, gstreamer-1.0 gmodule-no-export-2.0,, AC_MSG_ERROR([*** GSTREAMER not found!]))
   AC_DEFINE([USE_GSTREAMER], 1,[gstreamer is enabled])
 fi
 AC_SUBST(AUDIO_CFLAGS)
