diff -Naur orig/x11vnc/x11vnc.desktop patched/x11vnc/x11vnc.desktop
--- orig/x11vnc/x11vnc.desktop	2009-10-07 22:21:23.000000000 +0200
+++ patched/x11vnc/x11vnc.desktop	2016-02-20 13:41:22.895924959 +0100
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Name=X11VNC Server
 Comment=Share this desktop by VNC
+Comment[hu]=Asztal megosztása a hálózaton VNC segítségével
 Exec=x11vnc -gui tray=setpass -rfbport PROMPT -bg -o %%HOME/.x11vnc.log.%%VNCDISPLAY
 Icon=computer
 Terminal=false
