diff -Naur ORIG/setup.py UHU/setup.py
--- ORIG/setup.py	2013-03-18 14:06:42.000000000 +0100
+++ UHU/setup.py	2013-10-03 23:07:25.506335758 +0200
@@ -106,7 +106,7 @@
     license = 'GNU LGPL',
     keywords = 'test unittest doctest automatic discovery',
     url = 'http://readthedocs.org/docs/nose/',
-    data_files = [('man/man1', ['nosetests.1'])],
+    data_files = [('share/man/man1', ['nosetests.1'])],
     package_data = {'': ['*.txt',
                          'examples/*.py',
                          'examples/*/*.py']},
