diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2013-10-17 08:26:32.000000000 +0200
+++ b/configure.ac	2022-05-10 21:34:12.382101770 +0200
@@ -147,13 +147,6 @@
 AC_SUBST(NAUTILUS_EXTENSION_DIR)
 
 # -----------------------------------------------------------------------------
-# Keyring
-
-PKG_CHECK_MODULES([GNOME_KEYRING], [gnome-keyring-1])
-SEAHORSE_CFLAGS="$SEAHORSE_CFLAGS $GNOME_KEYRING_CFLAGS"
-SEAHORSE_LIBS="$SEAHORSE_LIBS $GNOME_KEYRING_LIBS"
-
-# -----------------------------------------------------------------------------
 # DBus
 
 DBUS_GLIB_REQUIRED=0.35
