diff -Naur a/External/source/val/validate_builtins.cpp b/External/source/val/validate_builtins.cpp
--- a/External/source/val/validate_builtins.cpp	2022-01-26 21:01:45.000000000 +0100
+++ b/External/source/val/validate_builtins.cpp	2022-07-18 01:58:13.048850747 +0200
@@ -4185,6 +4185,7 @@
     case SpvBuiltInBaryCoordNV:
     case SpvBuiltInBaryCoordNoPerspNV:
     case SpvBuiltInCurrentRayTimeNV:
+    case SpvBuiltInCullMaskKHR:
       // No validation rules (for the moment).
       break;
 
