diff -Naur sparse-0.6.0-orig/Makefile sparse-0.6.0/Makefile
--- sparse-0.6.0-orig/Makefile	2018-12-26 08:07:42.000000000 +0100
+++ sparse-0.6.0/Makefile	2019-01-16 21:55:57.842055213 +0100
@@ -14,8 +14,8 @@
 CHECKER = CHECK=./sparse ./cgcc -no-compile
 CHECKER_FLAGS = -Wno-vla
 
-DESTDIR=
-PREFIX ?= $(HOME)
+# DESTDIR=
+PREFIX=/usr
 BINDIR=$(PREFIX)/bin
 MANDIR=$(PREFIX)/share/man
 MAN1DIR=$(MANDIR)/man1
