diff -Naur orig/data/lightdm-gtk-greeter.conf patched/data/lightdm-gtk-greeter.conf
--- orig/data/lightdm-gtk-greeter.conf	2020-06-23 01:04:09.000000000 +0200
+++ patched/data/lightdm-gtk-greeter.conf	2020-06-28 21:55:30.288209800 +0200
@@ -48,18 +48,18 @@
 #  transition-duration = overrides default value
 #
 [greeter]
-#background=
+background=/usr/share/backgrounds/UHU/ubk4.png
 #user-background=
 #theme-name=
-#icon-theme-name=
+icon-theme-name=gnome
 #font-name=
 #xft-antialias=
 #xft-dpi=
 #xft-hintstyle=
 #xft-rgba=
 #indicators=
-#clock-format=
+clock-format= %R
 #keyboard=
 #reader=
-#position=
+position= 60 80
 #screensaver-timeout=
