diff -Naur orig/data/lightdm-gtk-greeter.conf patched/data/lightdm-gtk-greeter.conf
--- orig/data/lightdm-gtk-greeter.conf	2015-05-21 04:40:39.000000000 +0200
+++ patched/data/lightdm-gtk-greeter.conf	2015-05-24 00:19:54.264202913 +0200
@@ -45,18 +45,18 @@
 #  transition-duration = overrides default value
 #
 [greeter]
-#background=
+background=/usr/share/wallpapers/UHU/blur.jpg
 #user-background=
 #theme-name=
-#icon-theme-name=
+icon-theme-name=gnome
 #font-name=
 #xft-antialias=
 #xft-dpi=
 #xft-hintstyle=
 #xft-rgba=
 #indicators=
-#clock-format=
+clock-format= %R
 #keyboard=
 #reader=
-#position=
+position= 60 80
 #screensaver-timeout=
