diff -Nur orig/configure.ac mod/configure.ac
--- orig/configure.ac	2020-07-23 09:30:22.000000000 +0200
+++ mod/configure.ac	2020-07-23 09:44:19.367753541 +0200
@@ -35,7 +35,7 @@
 
 if test "x$enable_gtk3" = "xyes" ; then
   gtk_module="gtk+-3.0"
-  pkg_modules="$pkg_modules $gtk_module libfm-gtk3 >= 1.2.0 libwnck-3.0"
+  pkg_modules="$pkg_modules $gtk_module libfm-gtk3 >= 1.2.0"
   keybinder_module=keybinder-3.0
 else
   gtk_module="gtk+-2.0 >= 2.18.0"
diff -Nur orig/plugins/Makefile.am mod/plugins/Makefile.am
--- orig/plugins/Makefile.am	2020-07-23 09:30:22.000000000 +0200
+++ mod/plugins/Makefile.am	2020-07-23 09:51:39.662648524 +0200
@@ -37,7 +37,6 @@
 	launchtaskbar.c \
 	task-button.c \
 	launch-button.c \
-	pager.c \
 	separator.c \
 	tray.c \
 	wincmd.c \
diff -Nur orig/src/private.h mod/src/private.h
--- orig/src/private.h	2020-07-23 09:30:22.000000000 +0200
+++ mod/src/private.h	2020-07-23 09:45:52.303497366 +0200
@@ -226,7 +226,6 @@
 #define STATIC_WINCMD
 #define STATIC_DIRMENU
 #define STATIC_TASKBAR
-#define STATIC_PAGER
 #define STATIC_TRAY
 #define STATIC_MENU
 #define STATIC_ICONS
