diff -Nur orig/configure.ac mod/configure.ac
--- orig/configure.ac	2017-02-09 15:26:43.000000000 +0100
+++ mod/configure.ac	2017-02-09 15:30:09.569547078 +0100
@@ -35,7 +35,7 @@
 
 if test "x$enable_gtk3" = "xyes" ; then
   gtk_module="gtk+-3.0"
-  pkg_modules="$pkg_modules $gtk_module libfm-gtk3 >= 1.2.0 libwnck-3.0"
+  pkg_modules="$pkg_modules $gtk_module libfm-gtk3 >= 1.2.0"
   keybinder_module=keybinder-3.0
 else
   gtk_module="gtk+-2.0 >= 2.18.0"
diff -Nur orig/plugins/Makefile.am mod/plugins/Makefile.am
--- orig/plugins/Makefile.am	2017-02-09 15:26:43.000000000 +0100
+++ mod/plugins/Makefile.am	2017-02-09 15:30:41.913621891 +0100
@@ -36,7 +36,6 @@
 	dirmenu.c \
 	launchtaskbar.c \
 	task-button.c \
-	pager.c \
 	separator.c \
 	tray.c \
 	wincmd.c \
diff -Nur orig/src/private.h mod/src/private.h
--- orig/src/private.h	2017-02-09 15:26:43.000000000 +0100
+++ mod/src/private.h	2017-02-09 15:31:07.613621149 +0100
@@ -202,7 +202,6 @@
 #define STATIC_WINCMD
 #define STATIC_DIRMENU
 #define STATIC_TASKBAR
-#define STATIC_PAGER
 #define STATIC_TRAY
 #define STATIC_MENU
 #define STATIC_ICONS
