diff -Nur orig/plugins/task-button.c mod/plugins/task-button.c
--- orig/plugins/task-button.c	2020-07-23 12:08:25.826338912 +0200
+++ mod/plugins/task-button.c	2020-07-24 15:24:51.379925596 +0200
@@ -653,7 +653,7 @@
 /* Get a pixbuf from a pixmap.
  * Originally from libwnck, Copyright (C) 2001 Havoc Pennington. */
 #if !GTK_CHECK_VERSION(3, 0, 0)
-static GdkPixbuf * _wnck_gdk_pixbuf_get_from_pixmap(GdkScreen *screen, Pixmap xpixmap, Window win, int width, int height)
+static GdkPixbuf * _wnck_gdk_pixbuf_get_from_pixmap(GdkScreen *screen, Pixmap xpixmap, int width, int height)
 {
     /* Get the drawable. */
 #if GTK_CHECK_VERSION(2, 24, 0)
@@ -703,19 +703,21 @@
     return retval;
 }
 #else
-static GdkPixbuf * _wnck_gdk_pixbuf_get_from_pixmap(GdkScreen *screen, Pixmap xpixmap, Window win, int width, int height)
+static GdkPixbuf * _wnck_gdk_pixbuf_get_from_pixmap(GdkScreen *screen, Pixmap xpixmap, int width, int height)
 {
   cairo_surface_t *surface;
-  GdkPixbuf *pixbuf = NULL;
+  GdkPixbuf *pixbuf;
   Display *xdisplay;
+  Window root_return;
   XWindowAttributes attrs;
 
   surface = NULL;
   xdisplay = GDK_DISPLAY_XDISPLAY(gdk_display_get_default());
 
+
   gdk_error_trap_push();
 
-  if (!XGetWindowAttributes (xdisplay, win, &attrs))
+  if (!XGetWindowAttributes (xdisplay, root_return, &attrs))
     goto TRAP_POP;
 
   if (attrs.depth == 1)
@@ -739,8 +741,7 @@
 
 TRAP_POP:
   gdk_flush();
-  if (gdk_error_trap_pop())
-    g_warning("task button : X error");
+  gdk_error_trap_pop();
 
   return pixbuf;
 }
@@ -915,7 +916,6 @@
         result = (hints != NULL) ? Success : -1;
         Pixmap xpixmap = None;
         Pixmap xmask = None;
-        Window win = None;
 
         if (result == Success)
         {
@@ -978,17 +978,18 @@
         unsigned int w, h;
         if (result == Success)
         {
+            Window unused_win;
             int unused;
             unsigned int unused_2;
             result = XGetGeometry(
                 xdisplay, xpixmap,
-                &win, &unused, &unused, &w, &h, &unused_2, &unused_2) ? Success : -1;
+                &unused_win, &unused, &unused, &w, &h, &unused_2, &unused_2) ? Success : -1;
         }
 
         /* If we have an X pixmap and its geometry, convert it to a GDK pixmap. */
         if (result == Success)
         {
-            pixmap = _wnck_gdk_pixbuf_get_from_pixmap(screen, xpixmap, win, w, h);
+            pixmap = _wnck_gdk_pixbuf_get_from_pixmap(screen, xpixmap, w, h);
             result = ((pixmap != NULL) ? Success : -1);
         }
 
@@ -996,15 +997,15 @@
          * Failures here are implemented as nonfatal. */
         if ((result == Success) && (xmask != None))
         {
-            Window win;
+            Window unused_win;
             int unused;
             unsigned int unused_2;
             if (XGetGeometry(
                 xdisplay, xmask,
-                &win, &unused, &unused, &w, &h, &unused_2, &unused_2))
+                &unused_win, &unused, &unused, &w, &h, &unused_2, &unused_2))
             {
                 /* Convert the X mask to a GDK pixmap. */
-                GdkPixbuf * mask = _wnck_gdk_pixbuf_get_from_pixmap(screen, xmask, win, w, h);
+                GdkPixbuf * mask = _wnck_gdk_pixbuf_get_from_pixmap(screen, xmask, w, h);
                 if (mask != NULL)
                 {
                     /* Apply the mask. */
