diff -Naur orig/src/textview.c patched/src/textview.c
--- orig/src/textview.c	2018-08-15 19:43:32.000000000 +0200
+++ patched/src/textview.c	2018-08-15 23:30:03.456998614 +0200
@@ -1076,7 +1076,7 @@
 	procmime_force_encoding(textview->messageview->forced_encoding);
 	
 	textview->is_in_signature = FALSE;
-	textview->is_diff = FALSE;
+	textview->is_diff = TRUE;
 	textview->is_attachment = FALSE;;
 
 	procmime_decode_content(mimeinfo);
