diff -Naur orig/src/prefs_common.c patched/src/prefs_common.c
--- orig/src/prefs_common.c	2018-08-15 19:43:32.000000000 +0200
+++ patched/src/prefs_common.c	2018-08-15 23:31:26.917916586 +0200
@@ -451,7 +451,7 @@
 	{"msgview_date_format", "FALSE", &prefs_common.msgview_date_format, P_BOOL,
 	 NULL, NULL, NULL},
 
-	{"next_on_delete", "FALSE", &prefs_common.next_on_delete, P_BOOL,
+	{"next_on_delete", "TRUE", &prefs_common.next_on_delete, P_BOOL,
 	 NULL, NULL, NULL},
 
 	 {"bold_unread", "TRUE", &prefs_common.bold_unread, P_BOOL,
