diff -Naur orig/src/prefs_common.c patched/src/prefs_common.c
--- orig/src/prefs_common.c	2019-01-08 06:41:04.000000000 +0100
+++ patched/src/prefs_common.c	2020-02-24 20:50:52.467812134 +0100
@@ -458,7 +458,7 @@
 	{"msgview_date_format", "FALSE", &prefs_common.msgview_date_format, P_BOOL,
 	 NULL, NULL, NULL},
 
-	{"next_on_delete", "FALSE", &prefs_common.next_on_delete, P_BOOL,
+	{"next_on_delete", "TRUE", &prefs_common.next_on_delete, P_BOOL,
 	 NULL, NULL, NULL},
 
 	 {"bold_unread", "TRUE", &prefs_common.bold_unread, P_BOOL,
