diff -Naur a/src/common/tests/mock_prefs_common_get_flush_metadata.h b/src/common/tests/mock_prefs_common_get_flush_metadata.h
--- a/src/common/tests/mock_prefs_common_get_flush_metadata.h	1970-01-01 01:00:00.000000000 +0100
+++ b/src/common/tests/mock_prefs_common_get_flush_metadata.h	2018-10-17 00:47:08.900475867 +0200
@@ -0,0 +1,5 @@
+gboolean prefs_common_get_flush_metadata(void)
+{
+	return FALSE;
+}
+
diff -Naur a/src/common/tests/mock_prefs_common_get_use_shred.h b/src/common/tests/mock_prefs_common_get_use_shred.h
--- a/src/common/tests/mock_prefs_common_get_use_shred.h	1970-01-01 01:00:00.000000000 +0100
+++ b/src/common/tests/mock_prefs_common_get_use_shred.h	2018-09-12 01:46:37.154654858 +0200
@@ -0,0 +1,5 @@
+gboolean prefs_common_get_use_shred(void)
+{
+	return FALSE;
+}
+
diff -Naur a/src/plugins/rssyl/libfeed/tests/mock_procheader_date_parse.h b/src/plugins/rssyl/libfeed/tests/mock_procheader_date_parse.h
--- a/src/plugins/rssyl/libfeed/tests/mock_procheader_date_parse.h	1970-01-01 01:00:00.000000000 +0100
+++ b/src/plugins/rssyl/libfeed/tests/mock_procheader_date_parse.h	2018-09-12 01:46:37.174655130 +0200
@@ -0,0 +1,7 @@
+time_t
+procheader_date_parse (gchar *dest,
+		const gchar *src,
+		gint len)
+{
+	return 1;
+}
diff -Naur a/src/tests/mock_debug_print.h b/src/tests/mock_debug_print.h
--- a/src/tests/mock_debug_print.h	1970-01-01 01:00:00.000000000 +0100
+++ b/src/tests/mock_debug_print.h	2018-09-12 01:46:37.178655185 +0200
@@ -0,0 +1,2 @@
+void debug_print_real (const gchar *format, ...) { return; }
+const char *debug_srcname(const char *file) { return NULL; }
