diff -Nur orig/wireshark.desktop mod/wireshark.desktop
--- orig/wireshark.desktop	2019-11-18 20:03:24.000000000 +0100
+++ mod/wireshark.desktop	2019-11-25 15:22:22.524533058 +0100
@@ -108,4 +108,4 @@
 MimeType=application/vnd.tcpdump.pcap;application/x-pcapng;application/x-snoop;application/x-iptrace;application/x-lanalyzer;application/x-nettl;application/x-radcom;application/x-etherpeek;application/x-visualnetworks;application/x-netinstobserver;application/x-5view;application/x-tektronix-rf5;application/x-micropross-mplog;application/x-apple-packetlogger;application/x-endace-erf;application/ipfix;application/x-ixia-vwr;
 # Category entry according to:
 # https://specifications.freedesktop.org/menu-spec/1.0/
-Categories=Network;Monitor;Qt;
+Categories=Network;System;Monitor;Qt;
