diff -Nur orig/Makefile mod/Makefile
--- orig/Makefile	2015-08-10 13:47:22.000000000 +0200
+++ mod/Makefile	2017-10-27 16:04:31.024078162 +0200
@@ -44,7 +44,7 @@
 GLEW_DEST ?= /usr
 BINDIR    ?= $(GLEW_DEST)/bin
 LIBDIR    ?= $(GLEW_DEST)/lib
-INCDIR    ?= $(GLEW_DEST)/include/GL
+INCDIR    ?= $(GLEW_DEST)/include/glew-1.13.0/GL
 
 ifneq ($(GLEW_NO_GLU), -DGLEW_NO_GLU)
 LIBGLU = glu
@@ -134,7 +134,6 @@
 
 # Force re-write of glew.pc, GLEW_DEST can vary
 
-.PHONY: glew.pc
 
 glew.pc: glew.pc.in
 	sed \
@@ -182,7 +181,6 @@
 
 # Force re-write of glewmx.pc, GLEW_DEST can vary
 
-.PHONY: glewmx.pc
 
 glewmx.pc: glew.pc.in
 	sed \
