diff -Nur orig/config/Makefile.linux mod/config/Makefile.linux
--- orig/config/Makefile.linux	2015-08-10 13:47:22.000000000 +0200
+++ mod/config/Makefile.linux	2017-10-27 15:49:20.000000000 +0200
@@ -9,13 +9,8 @@
 ifeq (ppc64,${M_ARCH})
   ARCH64 = true
 endif
-ifeq (${ARCH64},true)
-  LDFLAGS.EXTRA = -L/usr/X11R6/lib64 -L/usr/lib64
-  LIBDIR = $(GLEW_DEST)/lib64
-else
-  LDFLAGS.EXTRA = -L/usr/X11R6/lib -L/usr/lib
-  LIBDIR = $(GLEW_DEST)/lib
-endif
+LDFLAGS.EXTRA = -L/usr/X11R6/lib -L/usr/lib
+LIBDIR = $(GLEW_DEST)/lib
 LDFLAGS.GL = -lGL -lX11
 LDFLAGS.STATIC = -Wl,-Bstatic
 LDFLAGS.DYNAMIC = -Wl,-Bdynamic
