diff -Naur a/core/libs/threadimageio/video/videodecoder_p.h b/core/libs/threadimageio/video/videodecoder_p.h
--- a/core/libs/threadimageio/video/videodecoder_p.h	2022-03-01 09:42:07.000000000 +0100
+++ b/core/libs/threadimageio/video/videodecoder_p.h	2022-03-18 16:29:29.498955418 +0100
@@ -56,7 +56,7 @@
     AVFormatContext*   pFormatContext;
     AVCodecContext*    pVideoCodecContext;
     AVCodecParameters* pVideoCodecParameters;
-    AVCodec*           pVideoCodec;
+    const AVCodec*     pVideoCodec;
     AVStream*          pVideoStream;
     AVFrame*           pFrame;
     quint8*            pFrameBuffer;
