diff -Naur orig/core/CMakeLists.txt patched/core/CMakeLists.txt
--- orig/core/CMakeLists.txt	2017-01-08 23:08:44.000000000 +0100
+++ patched/core/CMakeLists.txt	2017-02-14 02:42:14.586483688 +0100
@@ -97,6 +97,8 @@
 include(MacroOpenCV)
 include(MacroJPEG)
 include(MacroBoolTo01)
+# Compile fix for boost
+string(REPLACE "-fno-operator-names" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
 
 # ==================================================================================================
 
