diff -Naur orig/linux/launchy.desktop patched/linux/launchy.desktop
--- orig/linux/launchy.desktop	2010-02-14 03:02:11.000000000 +0100
+++ patched/linux/launchy.desktop	2016-02-19 02:06:17.266199607 +0100
@@ -1,12 +1,11 @@
 [Desktop Entry]
-Version=1.0
+Version=1.1
 Name=Launchy
 Comment=Keystroke Launcher
+Comment[hu]=Program indító
 Exec=launchy %u
 Terminal=false
 X-MultipleArgs=false
 Type=Application
 Icon=launchy_icon
 Categories=Utility;
-
-
