diff -Naur orig/src/src.pro patched/src/src.pro
--- orig/src/src.pro	2019-05-21 04:11:59.000000000 +0200
+++ patched/src/src.pro	2019-06-06 21:39:46.878855642 +0200
@@ -100,7 +100,7 @@
     icon.path     = $$PREFIX/share/pixmaps
     icon.files    = ../misc/Launchy_Icon/launchy_icon.png
     desktop.path  = $$PREFIX/share/applications/
-    desktop.files = ../linux/launchy.desktop
+    desktop.files = ../dist/linux/launchy.desktop
     INSTALLS += target \
                 skins \
                 icon \
