diff -Naur orig/pretty-printer/src/PrettyPrinter.h mod/pretty-printer/src/PrettyPrinter.h
--- orig/pretty-printer/src/PrettyPrinter.h	2017-10-17 06:58:25.000000000 +0200
+++ mod/pretty-printer/src/PrettyPrinter.h	2020-11-07 22:57:41.499582953 +0100
@@ -29,6 +29,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdbool.h>
 
 #ifdef HAVE_GLIB
 #include <glib.h>
@@ -52,8 +53,6 @@
 #define TRUE !(FALSE)
 #endif
 
-typedef unsigned int bool;
-
 /*========================================== STRUCTURES =======================================================*/
 
 /**
