diff -Naur a/build/markdown.m4 b/build/markdown.m4
--- a/build/markdown.m4	2018-06-03 08:58:09.000000000 +0200
+++ b/build/markdown.m4	2023-02-08 19:39:44.574183065 +0100
@@ -49,13 +49,13 @@
     GTK_VERSION=2.16
     WEBKIT_VERSION=1.1.13
 
-    GP_CHECK_GTK3([webkit_package=webkit2gtk-4.0],
+    GP_CHECK_GTK3([webkit_package=webkit2gtk-4.1],
                   [webkit_package=webkit-1.0])
     GP_CHECK_PLUGIN_DEPS([markdown], [MARKDOWN],
                          [$GP_GTK_PACKAGE >= ${GTK_VERSION}
                           $webkit_package >= ${WEBKIT_VERSION}
                           gthread-2.0])
-    AM_CONDITIONAL([MARKDOWN_WEBKIT2], [test "$webkit_package" = webkit2gtk-4.0])
+    AM_CONDITIONAL([MARKDOWN_WEBKIT2], [test "$webkit_package" = webkit2gtk-4.1])
 
     GP_COMMIT_PLUGIN_STATUS([Markdown])
 
diff -Naur a/configure b/configure
--- a/configure	2021-09-29 19:22:07.000000000 +0200
+++ b/configure	2023-02-08 19:37:20.145709133 +0100
@@ -21092,7 +21092,7 @@
 
 
     if test ${GP_GTK_VERSION_MAJOR} = 3; then :
-  webkit_package=webkit2gtk-4.0
+  webkit_package=webkit2gtk-4.1
 else
   webkit_package=webkit-1.0
 fi
@@ -21334,7 +21334,7 @@
 fi
     fi
 
-     if test "$webkit_package" = webkit2gtk-4.0; then
+     if test "$webkit_package" = webkit2gtk-4.1; then
   MARKDOWN_WEBKIT2_TRUE=
   MARKDOWN_WEBKIT2_FALSE='#'
 else
