diff -Nur orig/units/console-getty.service.m4 mod/units/console-getty.service.m4
--- orig/units/console-getty.service.m4	2019-04-11 18:28:36.000000000 +0200
+++ mod/units/console-getty.service.m4	2019-10-08 10:17:32.046071078 +0200
@@ -23,7 +23,7 @@
 # The '-o' option value tells agetty to replace 'login' arguments with an
 # option to preserve environment (-p), followed by '--' for safety, and then
 # the entered username.
-ExecStart=-/sbin/agetty -o '-p -- \\u' --noclear --keep-baud console 115200,38400,9600 $TERM
+ExecStart=-/usr/sbin/agetty -o '-p -- \\u' --noclear --keep-baud console 115200,38400,9600 $TERM
 Type=idle
 Restart=always
 UtmpIdentifier=cons
