diff -Nur orig/units/container-getty@.service.m4 mod/units/container-getty@.service.m4
--- orig/units/container-getty@.service.m4	2019-04-11 18:28:36.000000000 +0200
+++ mod/units/container-getty@.service.m4	2019-10-08 10:04:26.783355522 +0200
@@ -28,7 +28,7 @@
 # The '-o' option value tells agetty to replace 'login' arguments with an
 # option to preserve environment (-p), followed by '--' for safety, and then
 # the entered username.
-ExecStart=-/sbin/agetty -o '-p -- \\u' --noclear --keep-baud pts/%I 115200,38400,9600 $TERM
+ExecStart=-/usr/sbin/agetty -o '-p -- \\u' --noclear --keep-baud pts/%I 115200,38400,9600 $TERM
 Type=idle
 Restart=always
 RestartSec=0
