diff -Naur orig/meson.build patched/meson.build
--- orig/meson.build	2017-07-12 16:01:10.134739579 +0200
+++ patched/meson.build	2017-08-25 15:50:26.134534302 +0200
@@ -281,7 +281,6 @@
                '-Werror=missing-declarations',
                '-Werror=return-type',
                '-Werror=incompatible-pointer-types',
-               '-Werror=format=2',
                '-Wstrict-prototypes',
                '-Wredundant-decls',
                '-Wmissing-noreturn',
