diff -Naur a/meson.build b/meson.build
--- a/meson.build	2022-05-21 15:23:41.000000000 +0200
+++ b/meson.build	2022-10-23 18:07:30.668787715 +0200
@@ -479,7 +479,6 @@
 #include <uchar.h>
 #include <sys/mount.h>
 #include <sys/stat.h>
-#include <linux/fs.h>
 '''
 
 foreach decl : ['char16_t',
@@ -491,6 +490,17 @@
         # We get -1 if the size cannot be determined
         have = cc.sizeof(decl, prefix : decl_headers, args : '-D_GNU_SOURCE') > 0
 
+        if decl == 'struct mount_attr'
+                if have
+                        want_linux_fs_h = false
+                else
+                        have = cc.sizeof(decl,
+                                         prefix : decl_headers + '#include <linux/fs.h>',
+                                         args : '-D_GNU_SOURCE') > 0
+                        want_linux_fs_h = have
+                endif
+        endif
+
         if decl == 'struct statx'
                 if have
                         want_linux_stat_h = false
@@ -506,6 +516,7 @@
 endforeach
 
 conf.set10('WANT_LINUX_STAT_H', want_linux_stat_h)
+conf.set10('WANT_LINUX_FS_H', want_linux_fs_h)
 
 foreach ident : ['secure_getenv', '__secure_getenv']
         conf.set10('HAVE_' + ident.to_upper(), cc.has_function(ident))
diff -Naur a/src/basic/fd-util.c b/src/basic/fd-util.c
--- a/src/basic/fd-util.c	2022-05-21 15:23:41.000000000 +0200
+++ b/src/basic/fd-util.c	2022-10-23 18:07:30.672787732 +0200
@@ -3,7 +3,9 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <linux/btrfs.h>
+#if WANT_LINUX_FS_H
 #include <linux/fs.h>
+#endif
 #include <linux/magic.h>
 #include <sys/ioctl.h>
 #include <sys/resource.h>
diff -Naur a/src/core/namespace.c b/src/core/namespace.c
--- a/src/core/namespace.c	2022-05-21 15:23:41.000000000 +0200
+++ b/src/core/namespace.c	2022-10-23 18:07:30.672787732 +0200
@@ -6,7 +6,9 @@
 #include <stdio.h>
 #include <sys/mount.h>
 #include <unistd.h>
+#if WANT_LINUX_FS_H
 #include <linux/fs.h>
+#endif
 
 #include "alloc-util.h"
 #include "base-filesystem.h"
diff -Naur a/src/shared/mount-util.c b/src/shared/mount-util.c
--- a/src/shared/mount-util.c	2022-05-21 15:23:41.000000000 +0200
+++ b/src/shared/mount-util.c	2022-10-23 18:07:30.672787732 +0200
@@ -7,7 +7,9 @@
 #include <sys/statvfs.h>
 #include <unistd.h>
 #include <linux/loop.h>
+#if WANT_LINUX_FS_H
 #include <linux/fs.h>
+#endif
 
 #include "alloc-util.h"
 #include "chase-symlinks.h"
