diff -Naur a/src/basic/linux/if_arp.h b/src/basic/linux/if_arp.h
--- a/src/basic/linux/if_arp.h	2020-11-26 19:00:50.000000000 +0100
+++ b/src/basic/linux/if_arp.h	2022-04-04 16:50:26.623581599 +0200
@@ -54,6 +54,7 @@
 #define ARPHRD_X25	271		/* CCITT X.25			*/
 #define ARPHRD_HWX25	272		/* Boards with X.25 in firmware	*/
 #define ARPHRD_CAN	280		/* Controller Area Network      */
+#define ARPHRD_MCTP	290
 #define ARPHRD_PPP	512
 #define ARPHRD_CISCO	513		/* Cisco HDLC	 		*/
 #define ARPHRD_HDLC	ARPHRD_CISCO
