diff --git a/src/network/networkd-link.c b/src/network/networkd-link.c
index 3e22efb92a..15851f2965 100644
--- a/src/network/networkd-link.c
+++ b/src/network/networkd-link.c
@@ -321,7 +321,8 @@ static int link_enable_ipv6(Link *link) {
         r = sysctl_write_ip_property_boolean(AF_INET6, link->ifname, "disable_ipv6", disabled);
         if (r < 0)
                 log_link_warning_errno(link, r, "Cannot %s IPv6 for interface %s: %m",
-                                       enable_disable(!disabled), link->ifname);
+                                       enable_disable(!disabled),
+                                       (link->ifname) ? link->ifname : "<null>");
         else
                 log_link_info(link, "IPv6 successfully %sd", enable_disable(!disabled));
