diff -Naur orig/units/systemd-networkd.service.in patched/units/systemd-networkd.service.in
--- orig/units/systemd-networkd.service.in	2018-01-28 16:58:17.000000000 +0100
+++ patched/units/systemd-networkd.service.in	2018-02-04 12:47:27.904142171 +0100
@@ -13,7 +13,7 @@
 ConditionCapability=CAP_NET_ADMIN
 DefaultDependencies=no
 # systemd-udevd.service can be dropped once tuntap is moved to netlink
-After=systemd-udevd.service network-pre.target systemd-sysusers.service systemd-sysctl.service
+After=systemd-udevd.service network-pre.target systemd-sysusers.service systemd-sysctl.service dbus.service
 Before=network.target multi-user.target shutdown.target
 Conflicts=shutdown.target
 Wants=network.target
