diff -Nur orig/units/serial-getty@.service.m4 mod/units/serial-getty@.service.m4
--- orig/units/serial-getty@.service.m4	2019-04-11 18:28:36.000000000 +0200
+++ mod/units/serial-getty@.service.m4	2019-10-08 10:08:03.758906288 +0200
@@ -33,7 +33,7 @@
 # The '-o' option value tells agetty to replace 'login' arguments with an
 # option to preserve environment (-p), followed by '--' for safety, and then
 # the entered username.
-ExecStart=-/sbin/agetty -o '-p -- \\u' --keep-baud 115200,38400,9600 %I $TERM
+ExecStart=-/usr/sbin/agetty -o '-p -- \\u' --keep-baud 115200,38400,9600 %I $TERM
 Type=idle
 Restart=always
 UtmpIdentifier=%I
