diff --git a/src/login/logind-user.c b/src/login/logind-user.c
index 045b6f0..8c33789 100644
--- a/src/login/logind-user.c
+++ b/src/login/logind-user.c
@@ -665,7 +665,7 @@ static bool elect_display_filter(Session *s) {
         /* Return true if the session is a candidate for the user’s ‘primary session’ or ‘display’. */
         assert(s);
 
-        return s->class == SESSION_USER && s->started && !s->stopping;
+        return IN_SET(s->class, SESSION_USER, SESSION_GREETER) && s->started && !s->stopping;
 }
 
 static int elect_display_compare(Session *s1, Session *s2) {
 
