diff -Naur xf86-video-tga-1.2.2-orig/src/tga_accel.c xf86-video-tga-1.2.2/src/tga_accel.c
--- xf86-video-tga-1.2.2-orig/src/tga_accel.c	2012-07-16 05:54:04.000000000 +0200
+++ xf86-video-tga-1.2.2/src/tga_accel.c	2013-03-31 12:16:49.000000000 +0200
@@ -37,7 +37,6 @@
 #include "xf86cmap.h"
 #include "mipointer.h"
 
-#include "mibstore.h"
 #include "miline.h"
 
 #include "tga_regs.h"
diff -Naur xf86-video-tga-1.2.2-orig/src/tga_driver.c xf86-video-tga-1.2.2/src/tga_driver.c
--- xf86-video-tga-1.2.2-orig/src/tga_driver.c	2012-07-16 05:54:28.000000000 +0200
+++ xf86-video-tga-1.2.2/src/tga_driver.c	2013-03-31 12:17:00.000000000 +0200
@@ -46,8 +46,6 @@
 
 /* software cursor */
 #include "mipointer.h"
-/* backing store */
-#include "mibstore.h"
 
 /*  #include "mibank.h" */
 /* colormap manipulation */
