diff -urN openmortal-0.7.orig/src/sge_tt_text.cpp openmortal-0.7/src/sge_tt_text.cpp
--- openmortal-0.7.orig/src/sge_tt_text.cpp	2004-03-03 21:17:48.000000000 +0100
+++ openmortal-0.7/src/sge_tt_text.cpp	2014-02-17 15:21:46.630679629 +0100
@@ -37,9 +37,9 @@
 
 #ifndef _SGE_NOTTF
 #include <ft2build.h>
-#include <freetype/freetype.h>
-#include <freetype/ftoutln.h>
-#include <freetype/ttnameid.h>
+#include <freetype2/freetype/freetype.h>
+#include <freetype2/freetype/ftoutln.h>
+#include <freetype2/freetype/ttnameid.h>
 
 /* The structure used to hold glyph information (cached) */
 struct glyph {
