diff -Naur openmortal-0.7.orig/src/OnlineChat.cpp openmortal-0.7/src/OnlineChat.cpp
--- openmortal-0.7.orig/src/OnlineChat.cpp	2004-04-29 18:32:35.000000000 +0200
+++ openmortal-0.7/src/OnlineChat.cpp	2007-08-15 17:38:51.000000000 +0200
@@ -56,7 +56,7 @@
 class CChallengeMenu: public Menu
 {
 public:
-	CChallengeMenu::CChallengeMenu( std::string a_sChallenger )
+	CChallengeMenu( std::string a_sChallenger )
 	: Menu( "You have been challenged!" ),
 	m_sChallenger( a_sChallenger )
 	{
