diff -Naur notification-daemon-3.20.0-orig/data/notification-daemon.desktop.in.in notification-daemon-3.20.0/data/notification-daemon.desktop.in.in
--- notification-daemon-3.20.0-orig/data/notification-daemon.desktop.in.in	2016-03-10 23:56:52.000000000 +0100
+++ notification-daemon-3.20.0/data/notification-daemon.desktop.in.in	2016-05-11 20:47:50.751781987 +0200
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Name=Notification Daemon
 Comment=Display notifications
-Exec=@libexecdir@/notification-daemon
+Exec=notification-daemon
 Terminal=false
 Type=Application
 NoDisplay=true
