diff -Nur orig/apps/CA.pl.in mod/apps/CA.pl.in
--- orig/apps/CA.pl.in	2017-05-25 14:46:17.000000000 +0200
+++ mod/apps/CA.pl.in	2017-09-10 16:26:30.809420754 +0200
@@ -33,7 +33,7 @@
 my $PKCS12 = "$openssl pkcs12";
 
 # default openssl.cnf file has setup as per the following
-my $CATOP = "./demoCA";
+my $CATOP = "/etc/ssl";
 my $CAKEY = "cakey.pem";
 my $CAREQ = "careq.pem";
 my $CACERT = "cacert.pem";
diff -Nur orig/apps/openssl.cnf mod/apps/openssl.cnf
--- orig/apps/openssl.cnf	2017-05-25 14:46:18.000000000 +0200
+++ mod/apps/openssl.cnf	2017-09-10 16:27:39.357320042 +0200
@@ -39,7 +39,7 @@
 ####################################################################
 [ CA_default ]
 
-dir		= ./demoCA		# Where everything is kept
+dir		= /etc/ssl		# Where everything is kept
 certs		= $dir/certs		# Where the issued certs are kept
 crl_dir		= $dir/crl		# Where the issued crl are kept
 database	= $dir/index.txt	# database index file.
@@ -323,7 +323,7 @@
 [ tsa_config1 ]
 
 # These are used by the TSA reply generation only.
-dir		= ./demoCA		# TSA root directory
+dir		= /etc/ssl		# TSA root directory
 serial		= $dir/tsaserial	# The current serial number (mandatory)
 crypto_device	= builtin		# OpenSSL engine to use for signing
 signer_cert	= $dir/tsacert.pem 	# The TSA signing certificate
