diff -urN rdiff-backup-1.2.8.orig/setup.py rdiff-backup-1.2.8/setup.py
--- rdiff-backup-1.2.8.orig/setup.py	2009-03-16 15:36:21.000000000 +0100
+++ rdiff-backup-1.2.8/setup.py	2009-06-04 16:24:42.000000000 +0200
@@ -74,8 +74,6 @@
 							   extra_link_args=lflags_arg)],
 	  scripts = ['rdiff-backup', 'rdiff-backup-statistics'],
 	  data_files = [('share/man/man1', ['rdiff-backup.1',
-										'rdiff-backup-statistics.1']),
-					('share/doc/rdiff-backup-%s' % (version_string,),
-					 ['CHANGELOG', 'COPYING', 'README', 'FAQ.html'])],
+										'rdiff-backup-statistics.1'])],
 					**extra_options)
 
