diff -Naur orig/data/lightdm-another-gtk-greeter.conf mod/data/lightdm-another-gtk-greeter.conf
--- orig/data/lightdm-another-gtk-greeter.conf	2014-01-02 23:09:04.000000000 +0100
+++ mod/data/lightdm-another-gtk-greeter.conf	2014-12-08 20:47:08.716964152 +0100
@@ -15,7 +15,7 @@
 # User styles file
 #css-file=
 # Logo: "file_path" or "#icon_name"
-#logo=
+logo=/usr/share/pixmaps/uhu-96x96.png
 # Gtk theme to use
 #gtk-theme=
 # Icons theme to use
@@ -31,7 +31,7 @@
 #xft-dpi=
 # name, display-name, both
 #user-name-format=display-name
-#date-format=
+date-format=%Y %m %d %A
 
 [panel]
 # Show panel
@@ -51,7 +51,7 @@
 # Show clock
 #enabled=true
 #show-calendar=true
-#time-format=
+time-format=%T
 #date-format=
 
 [a11y]
