diff -Naur orig/balsa-mailto-handler.desktop.in.in mod/balsa-mailto-handler.desktop.in.in
--- orig/balsa-mailto-handler.desktop.in.in	2012-04-20 21:26:11.000000000 +0200
+++ mod/balsa-mailto-handler.desktop.in.in	2015-10-16 17:42:02.381125278 +0200
@@ -9,6 +9,6 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=@TOOLKIT_CATEGORIES@Network;Email;
+Categories=Network;Email;
 MimeType=x-scheme-handler/mailto;
 NoDisplay=true
