diff -Naur balsa-2.3.19.orig/src/pref-manager.c balsa-2.3.19/src/pref-manager.c
--- balsa-2.3.19.orig/src/pref-manager.c	2007-01-01 20:13:27.000000000 +0100
+++ balsa-2.3.19/src/pref-manager.c	2007-09-03 18:31:57.000000000 +0200
@@ -29,7 +29,6 @@
 #include "folder-conf.h"
 #include "main-window.h"
 #include "save-restore.h"
-#include "spell-check.h"
 #include "address-book-config.h"
 #include "quote-color.h"
 #include "misc.h"
