diff -Nur orig/sr-editor mod/sr-editor
--- orig/sr-editor	2015-09-17 22:03:29.000000000 +0200
+++ mod/sr-editor	2021-04-15 20:40:34.583090384 +0200
@@ -15,18 +15,8 @@
 LIBPATH="`FindPath $0`/lib"
 BINPATH="`FindPath $0`/bin"
 
-if [ "$(uname -m)" = "x86_64" ]; then
-	LIBPATH="$LIBPATH/64"
-else
-	LIBPATH="$LIBPATH/32"
-fi
-
 export LD_LIBRARY_PATH="$LIBPATH"
 export OGRE_PLUGIN_DIR="$LIBPATH"
 
-if [ "$(uname -m)" = "x86_64" ]; then
-	"$BINPATH/sr-editor_x86_64" $*
-else
-	"$BINPATH/sr-editor_x86" $*
-fi
+"$BINPATH/sr-editor_x86_64" $*
 
diff -Nur orig/stuntrally mod/stuntrally
--- orig/stuntrally	2015-09-17 22:03:29.000000000 +0200
+++ mod/stuntrally	2021-04-15 20:40:02.610698558 +0200
@@ -15,18 +15,8 @@
 LIBPATH="`FindPath $0`/lib"
 BINPATH="`FindPath $0`/bin"
 
-if [ "$(uname -m)" = "x86_64" ]; then
-	LIBPATH="$LIBPATH/64"
-else
-	LIBPATH="$LIBPATH/32"
-fi
-
 export LD_LIBRARY_PATH="$LIBPATH"
 export OGRE_PLUGIN_DIR="$LIBPATH"
 
-if [ "$(uname -m)" = "x86_64" ]; then
-	"$BINPATH/stuntrally_x86_64" $*
-else
-	"$BINPATH/stuntrally_x86" $*
-fi
+"$BINPATH/stuntrally_x86_64" $*
 
