diff -Nur orig/util/startfluxbox.in mod/util/startfluxbox.in
--- orig/util/startfluxbox.in	2015-02-08 11:44:45.000000000 +0100
+++ mod/util/startfluxbox.in	2018-09-27 20:20:32.051740302 +0200
@@ -40,7 +40,7 @@
 # Lines starting with a '#' are ignored.
 
 # Change your keymap:
-xmodmap "$HOME/.Xmodmap"
+# xmodmap "$HOME/.Xmodmap"
 
 # Applications you want to run with fluxbox.
 # MAKE SURE THAT APPS THAT KEEP RUNNING HAVE AN ''&'' AT THE END.
