diff -Nur orig/fluxbox-xdg-menu.py mod/fluxbox-xdg-menu.py
--- orig/fluxbox-xdg-menu.py	2018-09-30 19:03:24.000000000 +0200
+++ mod/fluxbox-xdg-menu.py	2018-10-04 15:22:27.000000000 +0200
@@ -1,4 +1,5 @@
 #!/usr/bin/env python
+# coding: utf-8
 # vim: noexpandtab:ts=4:sts=4
 
 """Menu Generator for Fluxbox
@@ -35,6 +36,8 @@
 
 
 import os,re,sys,glob,getopt
+reload(sys)
+sys.setdefaultencoding('utf-8')
 import xdg.Menu,xdg.DesktopEntry,xdg.IconTheme
 from os.path import isfile
 
@@ -44,42 +47,41 @@
 def header(wm="fluxbox"):
 	return """
 [begin] (Fluxbox)
-	[exec] (Web Browser) {htmlview}
-	[exec] (Email) {evolution}
-	[exec] (Terminal) {$TERM}
-	[exec] (Irc) {xchat}
+	[exec] (Terminál) {xterm} </usr/share/pixmaps/xterm-color_48x48.xpm>
 	[separator]\n"""
 
 def footer(wm="fluxbox"):
 	return """
 	[submenu] (Fluxbox Menu)
-		[config] (Configure)
-		[submenu] (System Styles) {Choose a style...}
+		[config] (Beállítás)
+		[submenu] (Rendszer témák) {Choose a style...}
 			[stylesdir] (/usr/share/fluxbox/styles)
 			[stylesdir] (/usr/share/commonbox/styles/)
 		[end]
-		[submenu] (User Styles) {Choose a style...}
+		[submenu] (Saját témák) {Choose a style...}
 			[stylesdir] (~/.fluxbox/styles)
 		[end]
-		[workspaces]   (Workspace List)
-		[submenu] (Tools)
+		[workspaces]   (Munkaterületek)
+		[submenu] (Eszközök)
 			[exec] (Window name) {xprop WM_CLASS|cut -d \" -f 2|xmessage -file - -center}
 			[exec] (Screenshot - JPG) {import screenshot.jpg && display -resize 50% screenshot.jpg}
 			[exec] (Screenshot - PNG) {import screenshot.png && display -resize 50% screenshot.png}
-			[exec] (Run) {fbrun }
-			[exec] (Regen Menu) {fluxbox-generate_menu --with-icons}
+			[exec] (Futtatás) {fbrun }
+			[exec] (Menüfrissítés) {fluxbox-xdg-menu --with-icons --with-backgrounds}
 		[end]
-		[submenu] (Window)
+		[submenu] (Ablakkezelő váltás)
 			[restart] (kde) {startkde}
 			[restart] (openbox) {openbox}
 			[restart] (gnome) {gnome-session}
 		[end]
-		[exec] (Lock screen) {xscreensaver-command -lock}
-		[commanddialog] (Fluxbox Command)
-		[reconfig] (Reload config)
-		[restart] (Restart)
+		[exec] (Képernyő zárolása) {xscreensaver-command -lock}
+		[commanddialog] (Fluxbox parancs)
+		[reconfig] (Konfiguráció újratöltés)
+		[exec] (Gép kikapcsolás) {systemctl poweroff} </usr/share/icons/gnome/48x48/actions/stop.png>
+		[exec] (Gép újraindítás) {systemctl reboot}  </usr/share/icons/gnome/48x48/actions/redo.png>
+		[restart] (Fluxbox újraindítás)  </usr/share/icons/gnome/48x48/actions/reload.png>
 		[separator]
-		[exit] (Exit)
+		[exit] (Kilépés) </usr/share/icons/gnome/48x48/actions/system-log-out.png> 
 	[end]
 [end]\n"""	
 
@@ -116,10 +118,12 @@
 		elif isinstance(entry, xdg.Menu.MenuEntry):
 			checkWm(entry,wm)
 			if entry.Show == False: continue
+			javitottnev = re.sub(r"\)","\\)",entry.DesktopEntry.getName().encode("utf8"))
+			javitottexec = re.sub(r"%.*","",entry.DesktopEntry.getExec())
 			if use_icons:
-				print "%s[exec] (%s) {%s} <%s> " % ( (depth*"\t"), entry.DesktopEntry.getName().encode("utf8"), entry.DesktopEntry.getExec().split()[0], findIcon(entry.DesktopEntry.getIcon(), theme) )
+				print "%s[exec] (%s) {%s} <%s> " % ( (depth*"\t"), javitottnev, javitottexec, findIcon(entry.DesktopEntry.getIcon(), theme) )
 			else:
-				print "%s[exec] (%s) {%s} " % ( (depth*"\t"), entry.DesktopEntry.getName().encode("utf8"), entry.DesktopEntry.getExec().split()[0] )
+				print "%s[exec] (%s) {%s} " % ( (depth*"\t"), javitottnev, javitottexec )
 		elif isinstance(entry,xdg.Menu.Separator):
 			print "%s[separator]" % (depth*"\t")
 		elif isinstance(entry.xdg.Menu.Header):
@@ -156,9 +160,8 @@
     countNum = 1
     bgPagCk = 1
     bgPgNum = 1
-    bgMenu.write( "[submenu] (Backgrounds)\n" )
-    bgMenu.write( "[submenu] (Backgrounds) {Set Your Background}\n" )
-    bgMenu.write("\t[exec]  (Random Image)  {fbsetbg -r ~/.fluxbox/backgrounds}\n")
+    bgMenu.write( "[submenu] (Hátterek)\n" )
+    bgMenu.write("\t[exec]  (Véletlen)  {fbsetbg -r ~/.fluxbox/backgrounds}\n")
 	types = ["png","jpg","jpeg","gif"]
     for i in h.keys():
 	    try:
@@ -171,7 +174,7 @@
 				bgPagCk = bgPagCk + 1
 				if bgPagCk == 26:
 				    bgPgNum = bgPgNum + 1
-					bgMenu.write("[end]\n[submenu] (Backgrounds " + str(bgPgNum) +") \
+					bgMenu.write("[end]\n[submenu] (Hátterek " + str(bgPgNum) +") \
 						{Set Your Background}\n")
 					bgPagCk = 1
 				if num == 0:
@@ -236,8 +239,8 @@
 		fsock = open(file,'w')
 		saveout = sys.stdout
 		sys.stdout = fsock
-
-	menu=xdg.Menu.parse()
+	menufile="lxde-applications.menu"
+	menu=xdg.Menu.parse(menufile)
 # is done automatically now
 #	menu.setLocale(lang)
 
