diff -Nur orig/util/startfluxbox.in mod/util/startfluxbox.in
--- orig/util/startfluxbox.in	2022-11-13 19:54:28.000000000 +0100
+++ mod/util/startfluxbox.in	2022-11-28 16:18:56.876873519 +0100
@@ -23,6 +23,13 @@
     shift
 done
 
+# start pulseaudio
+if [ -f /etc/fluxbox-pulseaudio ]; then
+    if [ -x /usr/bin/pulseaudio ]; then
+      /usr/bin/pulseaudio -D
+    fi
+fi
+
 if [ -x "$startup" ]; then
     exec "$startup"
 elif [ -r "$startup" ]; then
@@ -40,22 +47,63 @@
 # Lines starting with a '#' are ignored.
 
 # Change your keymap:
-xmodmap "$HOME/.Xmodmap"
+#xmodmap "$HOME/.Xmodmap"
+
+# Applications you want to run with fluxbox.
+# MAKE SURE THAT APPS THAT KEEP RUNNING HAVE AN ''&'' AT THE END.
+#
+#!/bin/sh
+#
+# fluxbox startup-script:
+#
+# Lines starting with a '#' are ignored.
+
+# Change your keymap:
+# xmodmap "~/.Xmodmap"
 
 # Applications you want to run with fluxbox.
 # MAKE SURE THAT APPS THAT KEEP RUNNING HAVE AN ''&'' AT THE END.
 #
-# unclutter -idle 2 &
-# wmnd &
-# wmsmixer -w &
-# idesk &
+
+# polkit ügynökök, ki kell választani egyet a hitelesítő ablak felbukkanásához.
+#/usr/libexec/lxpolkit &
+#/usr/libexec/polkit-gnome-authentication-agent-1 &
+/usr/libexec/polkit-mate-authentication-agent-1 &
+#/usr/lib/libexec/polkit-kde-authentication-agent-1
+
+# hálótzat figyelő. # A nagy panelek után kell rakni.
+sleep 1 &&
+nm-applet &
+
+# panelek
+#tint2 & # Beálítása a ~/.config/tint2/tint2rc fájlban történhet
+#bmpanel2 &
+# Nagy panelek
+#mate-panel &
+#gnome-panel &
+#xfce4-panel &
+#lxpanel &
+
+# képernyővédő
+xscreensaver &
+
+# ikonok a háttérre (https://wiki.archlinux.org/title/Idesk)
+#idesk &
+
+# egyebek
+#unclutter -idle 2 &
+#wmnd &
+#wmsmixer -w &
+
+# ha a megjelenített háttér nem megfelelő
+#feh --bg-scale ~/.fluxbox/backgrounds/ubk4.png &
 
 # And last but not least we start fluxbox.
 # Because it is the last app you have to run it with ''exec'' before it.
 
-exec @pkgprefix@fluxbox@pkgsuffix@
+exec fluxbox
 # or if you want to keep a log:
-# exec @pkgprefix@fluxbox@pkgsuffix@ -log "$fluxdir/log"
+#exec fluxbox -log ~/fluxbox/fluxbox/log"
 EOF
     ) > "$startup"
     fi
