diff -Nur orig/util/startfluxbox.in mod/util/startfluxbox.in
--- orig/util/startfluxbox.in	2022-11-13 19:54:28.000000000 +0100
+++ mod/util/startfluxbox.in	2022-11-13 20:59:31.282844456 +0100
@@ -23,6 +23,13 @@
     shift
 done
 
+# start pulseaudio
+if [ -f /etc/fluxbox-pulseaudio ]; then
+    if [ -x /usr/bin/pulseaudio ]; then
+      /usr/bin/pulseaudio -D
+    fi
+fi
+
 if [ -x "$startup" ]; then
     exec "$startup"
 elif [ -r "$startup" ]; then
@@ -31,6 +38,7 @@
     if [ ! -d $fluxdir ]; then
         mkdir -p "$fluxdir/backgrounds" "$fluxdir/styles" "$fluxdir/pixmaps"
     fi
+    ln -sf /usr/share/backgrounds/UHU/* "$fluxdir/backgrounds/"
     if [ ! -r "$startup" ]; then
         ( cat << EOF
 #!/bin/sh
@@ -40,11 +48,20 @@
 # Lines starting with a '#' are ignored.
 
 # Change your keymap:
-xmodmap "$HOME/.Xmodmap"
+# xmodmap "$HOME/.Xmodmap"
 
 # Applications you want to run with fluxbox.
 # MAKE SURE THAT APPS THAT KEEP RUNNING HAVE AN ''&'' AT THE END.
 #
+# polkit ügynökök, ki kell választani azt, amelyik van. Csak egy lehet!
+# /usr/libexec/lxpolkit &
+# /usr/libexec/polkit-gnome-authentication-agent-1 &
+# /usr/libexec/polkit-mate-authentication-agent-1 &
+# /usr/lib/libexec/polkit-kde-authentication-agent-1
+
+# képernyővédő
+# xscreensaver &
+#
 # unclutter -idle 2 &
 # wmnd &
 # wmsmixer -w &
@@ -59,6 +76,8 @@
 EOF
     ) > "$startup"
     fi
+    fluxbox-xdg-menu --with-icons --with-backgrounds
     chmod 644 "$startup"
     exec sh "$startup"
 fi
+
