--- a/docs/net-tvtime.desktop.orig	2014-05-01 12:22:42.000000000 +0200
+++ a/docs/net-tvtime.desktop	2014-05-01 12:26:44.000000000 +0200
@@ -1,10 +1,10 @@
 [Desktop Entry]
-Encoding=UTF-8
 Comment=High quality video deinterlacer
-Icon=tvtime.png
+Comment[hu]=Nagytudású televízió néző program
+Icon=tvtime
 Exec=tvtime
-Name=TVtime Television Viewer
+Name=TVtime
 GenericName=Television Viewer
 Terminal=false
 Type=Application
-Categories=Application;AudioVideo;
+Categories=AudioVideo;Video;TV;
