diff -Nur orig/CMakeLists.txt mod/CMakeLists.txt
--- orig/CMakeLists.txt	2017-10-28 17:32:57.000000000 +0200
+++ mod/CMakeLists.txt	2017-10-29 16:58:58.979389290 +0100
@@ -84,7 +84,7 @@
 endif()
 
 # Documentation
-install(FILES ${PROJECT_SOURCE_DIR}/share/trackballs.6 DESTINATION man/man6)
+install(FILES ${PROJECT_SOURCE_DIR}/share/trackballs.6 DESTINATION share/man/man6)
 file(GLOB htmldocs ${PROJECT_SOURCE_DIR}/docs/*.html ${PROJECT_SOURCE_DIR}/docs/*.css)
 install(FILES ${htmldocs} DESTINATION ${TRACKBALLS_SHARE}/docs)
 
