diff -Naur squid-3.3.10-orig/src/cf.data.pre squid-3.3.10/src/cf.data.pre
--- squid-3.3.10-orig/src/cf.data.pre	2013-11-03 12:06:37.000000000 +0100
+++ squid-3.3.10/src/cf.data.pre	2013-11-22 20:41:40.902426461 +0100
@@ -7530,7 +7530,7 @@
 
 NAME: cache_effective_group
 TYPE: string
-DEFAULT: none
+DEFAULT: squid
 DEFAULT_DOC: Use system group memberships of the cache_effective_user account
 LOC: Config.effectiveGroup
 DOC_START
