diff -Naur squid-3.3.10-orig/src/cf.data.pre squid-3.3.10/src/cf.data.pre
--- squid-3.3.10-orig/src/cf.data.pre	2013-11-22 20:45:15.946485834 +0100
+++ squid-3.3.10/src/cf.data.pre	2013-11-22 20:47:15.334525331 +0100
@@ -7486,11 +7486,11 @@
 
 NAME: cache_mgr
 TYPE: string
-DEFAULT: webmaster
+DEFAULT: root
 LOC: Config.adminEmail
 DOC_START
 	Email-address of local cache manager who will receive
-	mail if the cache dies.  The default is "webmaster".
+	mail if the cache dies.  The default is "root".
 DOC_END
 
 NAME: mail_from
