diff -Nur orig/CMakeLists.txt mod/CMakeLists.txt
--- orig/CMakeLists.txt	2017-10-27 17:52:31.000000000 +0200
+++ mod/CMakeLists.txt	2017-10-27 18:24:18.418440581 +0200
@@ -95,7 +95,7 @@
     ${STDCppFS_LIBRARY_DIRS}
 )
 
-subdirs(src src/resources/images data doc test)
+subdirs(src src/resources/images data doc)
 
 
 #  Uninstall material taken from http://www.cmake.org/Wiki/CMake_FAQ#Can_I_do_.22make_uninstall.22_with_CMake.3F
diff -Nur orig/meson.build mod/meson.build
--- orig/meson.build	2017-10-27 17:52:31.000000000 +0200
+++ mod/meson.build	2017-10-27 18:18:23.855022119 +0200
@@ -52,4 +52,4 @@
 subdir('src/resources/images')
 subdir('data')
 subdir('doc')
-subdir('test')
+
