diff -upNr compiz-0.8.2.orign/metadata/cube.xml.in compiz-0.8.2/metadata/cube.xml.in
--- compiz-0.8.2.orign/metadata/cube.xml.in	2009-02-15 10:10:23.000000000 +0100
+++ compiz-0.8.2/metadata/cube.xml.in	2009-05-25 22:54:03.060114649 +0200
@@ -92,9 +92,9 @@
 		    <_short>Cube Color</_short>
 		    <_long>Color of top and bottom sides of the cube</_long>
 		    <default>
-			<red>0xfefe</red>
-			<green>0xffff</green>
-			<blue>0xc7c7</blue>
+			<red>0x0000</red>
+			<green>0x1919</green>
+			<blue>0x4a4a</blue>
 		    </default>
 		</option>
 		<subgroup>
@@ -110,7 +110,7 @@
 			<type>string</type>
 			<hints>file;image;</hints>
 			<default>
-			    <value>freedesktop</value>
+			    <value>/usr/share/pixmaps/uhulogo</value>
 			</default>
 		    </option>
 		    <option name="adjust_image" type="bool">
