diff -Naur a/metadata/cube.xml.in b/metadata/cube.xml.in
--- a/metadata/cube.xml.in	2020-04-04 20:24:41.000000000 +0200
+++ b/metadata/cube.xml.in	2024-08-20 04:42:13.339892818 +0200
@@ -96,9 +96,9 @@
 		    <short>Cube Color</short>
 		    <long>Color of top and bottom sides of the cube</long>
 		    <default>
-			<red>0xfefe</red>
-			<green>0xffff</green>
-			<blue>0xc7c7</blue>
+			<red>0x0000</red>
+			<green>0x1919</green>
+			<blue>0x4a4a</blue>
 		    </default>
 		</option>
 		<subgroup>
@@ -114,7 +114,7 @@
 			<type>string</type>
 			<hints>file;image;</hints>
 			<default>
-			    <value>freedesktop</value>
+			    <value>/usr/share/pixmaps/uhulogo</value>
 			</default>
 		    </option>
 		    <option name="adjust_image" type="bool">
