diff -Nur orig/backends/aptcc/apt-messages.cpp mod/backends/aptcc/apt-messages.cpp
--- orig/backends/aptcc/apt-messages.cpp	2016-01-05 10:48:43.000000000 +0100
+++ mod/backends/aptcc/apt-messages.cpp	2019-03-20 08:58:04.483569395 +0100
@@ -29,7 +29,7 @@
 
 using namespace std;
 
-bool show_errors(PkBackendJob *job, PkErrorEnum errorCode, bool errModify)
+void show_errors(PkBackendJob *job, PkErrorEnum errorCode, bool errModify)
 {
     stringstream errors;
 
diff -Nur orig/backends/aptcc/apt-messages.h mod/backends/aptcc/apt-messages.h
--- orig/backends/aptcc/apt-messages.h	2016-01-05 10:48:43.000000000 +0100
+++ mod/backends/aptcc/apt-messages.h	2019-03-20 08:58:30.035523030 +0100
@@ -31,7 +31,7 @@
 /**
  * Call the Packagekit error dialog
  */
-bool show_errors(PkBackendJob *job,
+void show_errors(PkBackendJob *job,
                  PkErrorEnum errorCode = PK_ERROR_ENUM_UNKNOWN,
                  bool errModify = false);
 
