diff -Nur orig/backends/aptcc/Makefile.am mod/backends/aptcc/Makefile.am
--- orig/backends/aptcc/Makefile.am	2019-11-19 19:51:16.000000000 +0100
+++ mod/backends/aptcc/Makefile.am	2020-01-15 10:42:32.856853014 +0100
@@ -45,42 +45,6 @@
 aptconfdir = ${SYSCONFDIR}/apt/apt.conf.d
 aptconf_DATA = 20packagekit
 
-if HAVE_SYSTEMD
-libexec_PROGRAMS = pk-debconf-helper
-pk_debconf_helper_SOURCES =						\
-	pk-debconf-helper.c
-
-pk_debconf_helper_LDADD =						\
-	$(GLIB_LIBS)					\
-	$(GIO_LIBS)					\
-	$(SYSTEMD_LIBS)					\
-	$(PK_GLIB2_LIBS)
-
-pk_debconf_helper_LDFLAGS =						\
-	$(PIE_LDFLAGS)
-
-pk_debconf_helper_CFLAGS =						\
-	$(WARNINGFLAGS_C)
-pk_debconf_helper_CPPFLAGS =						\
-	$(PIE_CFLAGS)					\
-	$(GLIB_CFLAGS)					\
-	$(GIO_CFLAGS)					\
-	$(SYSTEMD_CFLAGS)				\
-	-DI_KNOW_THE_PACKAGEKIT_GLIB2_API_IS_SUBJECT_TO_CHANGE	\
-	-DPK_COMPILATION				\
-	-DG_LOG_DOMAIN=\"PackageKit\"			\
-	-I$(top_builddir)/lib				\
-	-I$(top_srcdir)					\
-	-I$(top_srcdir)/lib
-systemdservice_in_files =				\
-	pk-debconf-helper.service.in				\
-	pk-debconf-helper.socket.in
-
-systemduserservicedir   = $(systemduserunitdir)
-systemduserservice_DATA = pk-debconf-helper.socket \
-			  pk-debconf-helper.service
-endif
-
 EXTRA_DIST = 20packagekit \
 	     pkg-list.h \
 	     apt-intf.h \
