diff -Naur libX11-1.6.2-orig/nls/compose.dir.pre libX11-1.6.2/nls/compose.dir.pre
--- libX11-1.6.2-orig/nls/compose.dir.pre	2013-09-09 03:37:20.000000000 +0200
+++ libX11-1.6.2/nls/compose.dir.pre	2013-09-15 13:38:38.454034838 +0200
@@ -90,7 +90,7 @@
 iso8859-15/Compose:		en_ZA.ISO8859-15
 iso8859-1/Compose:		en_ZW.ISO8859-1
 iso8859-3/Compose:		eo_EO.ISO8859-3
-iso8859-3/Compose:		eo_XX.ISO8859-3
+iso8859-3/Compose:		eo.ISO8859-3
 iso8859-1/Compose:		es_AR.ISO8859-1
 iso8859-1/Compose:		es_BO.ISO8859-1
 iso8859-1/Compose:		es_CL.ISO8859-1
@@ -335,7 +335,7 @@
 en_US.UTF-8/Compose:		en_ZA.UTF-8
 en_US.UTF-8/Compose:		en_ZW.UTF-8
 en_US.UTF-8/Compose:		eo_EO.UTF-8
-en_US.UTF-8/Compose:		eo_XX.UTF-8
+en_US.UTF-8/Compose:		eo.UTF-8
 en_US.UTF-8/Compose:		es_AR.UTF-8
 en_US.UTF-8/Compose:		es_BO.UTF-8
 en_US.UTF-8/Compose:		es_CL.UTF-8
diff -Naur libX11-1.6.2-orig/nls/locale.alias.pre libX11-1.6.2/nls/locale.alias.pre
--- libX11-1.6.2-orig/nls/locale.alias.pre	2013-09-09 03:37:20.000000000 +0200
+++ libX11-1.6.2/nls/locale.alias.pre	2013-09-15 13:40:28.450195665 +0200
@@ -377,11 +377,16 @@
 en_ZW.utf8:					en_ZS.UTF-8
 en_ZW.iso88591:					en_ZW.ISO8859-1
 en_ZW.ISO-8859-1:				en_ZW.ISO8859-1
-eo:						eo_XX.ISO8859-3
-eo_EO:						eo_EO.ISO8859-3
-eo_EO.ISO8859-3:				eo_EO.ISO8859-3
-eo_XX:						eo_XX.ISO8859-3
-eo_XX.ISO8859-3:				eo_XX.ISO8859-3
+eo:						eo.ISO8859-3
+eo.utf8:					eo.UTF-8
+eo_EO:						eo.ISO8859-3
+eo_EO.ISO8859-3:				eo.ISO8859-3
+eo_EO.UTF-8:					eo.UTF-8
+eo_EO.utf8:					eo.UTF-8
+eo_XX:						eo.ISO8859-3
+eo_XX.ISO8859-3:				eo.ISO8859-3
+eo_XX.UTF-8:					eo.UTF-8
+eo_XX.utf8:					eo.UTF-8
 es:						es_ES.ISO8859-1
 es.UTF-8:					es_ES.UTF-8
 es_AR:						es_AR.ISO8859-1
diff -Naur libX11-1.6.2-orig/nls/locale.dir.pre libX11-1.6.2/nls/locale.dir.pre
--- libX11-1.6.2-orig/nls/locale.dir.pre	2013-09-09 03:37:20.000000000 +0200
+++ libX11-1.6.2/nls/locale.dir.pre	2013-09-15 13:42:14.870311567 +0200
@@ -90,8 +90,7 @@
 iso8859-1/XLC_LOCALE:			en_ZA.ISO8859-1
 iso8859-15/XLC_LOCALE:			en_ZA.ISO8859-15
 iso8859-1/XLC_LOCALE:			en_ZW.ISO8859-1
-iso8859-3/XLC_LOCALE:			eo_EO.ISO8859-3
-iso8859-3/XLC_LOCALE:			eo_XX.ISO8859-3
+iso8859-3/XLC_LOCALE:			eo.ISO8859-3
 iso8859-1/XLC_LOCALE:			es_AR.ISO8859-1
 iso8859-1/XLC_LOCALE:			es_BO.ISO8859-1
 iso8859-1/XLC_LOCALE:			es_CL.ISO8859-1
@@ -334,8 +333,7 @@
 en_US.UTF-8/XLC_LOCALE:			en_US.UTF-8
 en_US.UTF-8/XLC_LOCALE:			en_ZA.UTF-8
 en_US.UTF-8/XLC_LOCALE:			en_ZW.UTF-8
-en_US.UTF-8/XLC_LOCALE:			eo_EO.UTF-8
-en_US.UTF-8/XLC_LOCALE:			eo_XX.UTF-8
+en_US.UTF-8/XLC_LOCALE:			eo.UTF-8
 en_US.UTF-8/XLC_LOCALE:			es_AR.UTF-8
 en_US.UTF-8/XLC_LOCALE:			es_BO.UTF-8
 en_US.UTF-8/XLC_LOCALE:			es_CL.UTF-8
