diff -Naur orig/lib/lyx.desktop.in patched/lib/lyx.desktop.in
--- orig/lib/lyx.desktop.in	2019-06-10 21:12:32.000000000 +0200
+++ patched/lib/lyx.desktop.in	2020-01-01 16:25:24.076539514 +0100
@@ -4,6 +4,7 @@
 Name=LyX
 GenericName=Document Processor
 Comment=High level LaTeX frontend
+Comment[hu]=LaTeX fejlesztői környezet
 Keywords=WYSIWYG;WYSIWYM;TeX;LaTeX;GUI;frontend;editor;
 Exec=lyx@program_suffix@ %F
 TryExec=lyx@program_suffix@
