diff -Naur orig/src/EnchantChecker.cpp patched/src/EnchantChecker.cpp
--- orig/src/EnchantChecker.cpp	2017-05-08 23:53:46.000000000 +0200
+++ patched/src/EnchantChecker.cpp	2018-02-15 01:32:38.988314982 +0100
@@ -68,12 +68,13 @@
 
 enchant::Dict * EnchantChecker::Private::addSpeller(string const & lang)
 {
-	enchant::Broker * instance = enchant::Broker::instance();
+	static enchant::Broker *broker;
+	broker = new enchant::Broker();
 	Speller m;
 
 	try {
 		LYXERR(Debug::FILES, "request enchant speller for language " << lang);
-		m.speller = instance->request_dict(lang);
+		m.speller = broker->request_dict(lang);
 	}
 	catch (enchant::Exception & e) {
 		// FIXME error handling?
@@ -186,8 +187,9 @@
 {
 	if (!lang)
 		return false;
-	enchant::Broker * instance = enchant::Broker::instance();
-	return (instance->dict_exists(lang->code()));
+	static enchant::Broker *broker;
+	broker = new enchant::Broker();
+	return (broker->dict_exists(lang->code()));
 }
 
 
