diff -Naur a/src/insets/InsetListings.cpp b/src/insets/InsetListings.cpp
--- a/src/insets/InsetListings.cpp	2020-12-29 17:50:45.000000000 +0100
+++ b/src/insets/InsetListings.cpp	2022-07-28 22:35:48.315568308 +0200
@@ -44,6 +44,7 @@
 
 #include "support/regex.h"
 
+#include <cstring>
 #include <sstream>
 
 using namespace std;
diff -Naur a/src/lyxfind.cpp b/src/lyxfind.cpp
--- a/src/lyxfind.cpp	2020-12-29 17:50:45.000000000 +0100
+++ b/src/lyxfind.cpp	2022-07-28 22:25:47.548678017 +0200
@@ -53,6 +53,8 @@
 
 #include "support/regex.h"
 
+#include <iterator>
+
 using namespace std;
 using namespace lyx::support;
 
