diff -Naur orig/src/frontends/qt4/GuiDocument.cpp patched/src/frontends/qt4/GuiDocument.cpp
--- orig/src/frontends/qt4/GuiDocument.cpp	2018-02-25 01:11:18.000000000 +0100
+++ patched/src/frontends/qt4/GuiDocument.cpp	2018-06-09 05:54:00.033405457 +0200
@@ -68,6 +68,7 @@
 #include "frontends/alert.h"
 
 #include <QAbstractItemModel>
+#include <QButtonGroup>
 #include <QHeaderView>
 #include <QColor>
 #include <QColorDialog>
