diff -Naur orig/data/supertuxkart.desktop patched/data/supertuxkart.desktop
--- orig/data/supertuxkart.desktop	2019-04-20 18:40:12.000000000 +0200
+++ patched/data/supertuxkart.desktop	2019-12-10 23:59:16.639406411 +0100
@@ -6,6 +6,7 @@
 GenericName[de]=Ein Kart-Rennspiel
 GenericName[fr]=Un jeu de karting
 GenericName[gl]=Xogo de carreiras con karts
+GenericName[hu]=Gokart játék
 GenericName[pl]=Wyścigi gokartów
 GenericName[ro]=Un joc de curse cu carturi
 Exec=supertuxkart
