diff -u smartmontools-6.2.orig/ataprint.cpp smartmontools-6.2/ataprint.cpp
--- smartmontools-6.2.orig/ataprint.cpp	2013-07-20 16:25:56.000000000 +0200
+++ smartmontools-6.2/ataprint.cpp	2014-05-31 10:57:15.051797508 +0200
@@ -2213,9 +2213,6 @@
   while (n < tmh->cb_size) {
     // Find range of identical temperatures
     unsigned n1 = n, n2 = n+1, i2 = (i+1) % tmh->cb_size;
-    while (n2 < tmh->cb_size && tmh->cb[i2] == tmh->cb[i]) {
-      n2++; i2 = (i2+1) % tmh->cb_size;
-    }
     // Print range
     while (n < n2) {
       if (n == n1 || n == n2-1 || n2 <= n1+3) {
