diff -Naur orig/smartd.service.in patched/smartd.service.in
--- orig/smartd.service.in	2014-03-03 18:24:00.000000000 +0100
+++ patched/smartd.service.in	2015-08-02 19:45:01.968481673 +0200
@@ -1,7 +1,6 @@
 [Unit]
 Description=Self Monitoring and Reporting Technology (SMART) Daemon
 Documentation=man:smartd(8) man:smartd.conf(5)
-After=syslog.target
 
 [Service]
 EnvironmentFile=-/usr/local/etc/sysconfig/smartmontools
