diff -Naur orig/configure mod/configure
--- orig/configure	2015-08-23 19:16:49.000000000 +0200
+++ mod/configure	2016-03-08 08:17:14.947416011 +0100
@@ -4037,7 +4037,7 @@
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g"
   else
     CFLAGS="-g"
   fi
@@ -5116,7 +5116,7 @@
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g"
   else
     CFLAGS="-g"
   fi
@@ -14426,7 +14426,7 @@
     sed 's/\([0-9]*\).\([0-9]*\).\([0-9]*\)/\3/'`
 
 OLD_CFLAGS=$CFLAGS
-CFLAGS="$CFLAGS -static"
+CFLAGS="$CFLAGS"
 
 # Compare MINIMUM-VERSION with libaal version
 if test "$cross_compiling" = yes; then :
@@ -14599,7 +14599,7 @@
     sed 's/\([0-9]*\).\([0-9]*\).\([0-9]*\)/\3/'`
 
 OLD_CFLAGS=$CFLAGS
-CFLAGS="$CFLAGS -static"
+CFLAGS="$CFLAGS"
 
 # Compare MINIMUM-VERSION with libaal version
 if test "$cross_compiling" = yes; then :
@@ -14745,11 +14745,11 @@
 CFLAGS="$CFLAGS -W -Wall"
 
 if test x$enable_debug = xyes; then
-    	GENERIC_CFLAGS="$GENERIC_CFLAGS -O1 -g"
-    	MINIMAL_CFLAGS="$MINIMAL_CFLAGS -O1 -g"
+    	GENERIC_CFLAGS="$GENERIC_CFLAGS"
+    	MINIMAL_CFLAGS="$MINIMAL_CFLAGS"
 else
-    	GENERIC_CFLAGS="$GENERIC_CFLAGS -O3"
-    	MINIMAL_CFLAGS="$MINIMAL_CFLAGS -Os"
+    	GENERIC_CFLAGS="$GENERIC_CFLAGS"
+    	MINIMAL_CFLAGS="$MINIMAL_CFLAGS"
 
     	if test x$uninitialized = xyes; then
        		CFLAGS="$CFLAGS -Wuninitialized"
