diff -Naur orig/scripts/auxiliar/pfx2ttf.fontforge patched/scripts/auxiliar/pfx2ttf.fontforge
--- orig/scripts/auxiliar/pfx2ttf.fontforge	2014-03-17 16:29:16.000000000 +0100
+++ patched/scripts/auxiliar/pfx2ttf.fontforge	2016-10-22 02:50:07.044551111 +0200
@@ -18,11 +18,7 @@
          "afii61352", "afii61352", \
          "ij", "ij", \
          "IJ", "IJ");
-if (Strtol($version) < 20070501)
-  RemoveATT("Ligature", "*", "*");
-else
-  RemovePosSub("*");
-endif
+RemovePosSub("*");
 
 Generate($3 + $fontname + ".otf");
 
