diff -Naur orig/configure mod/configure
--- orig/configure	2014-03-17 16:29:16.000000000 +0100
+++ mod/configure	2016-05-08 18:03:14.099643137 +0200
@@ -8402,12 +8402,12 @@
     if test -x $exe; then
 	{ $as_echo "$as_me:$LINENO: checking guile compile flags" >&5
 $as_echo_n "checking guile compile flags... " >&6; }
-	GUILE_CFLAGS="`$guile_config compile`"
+	GUILE_CFLAGS="`pkg-config --cflags guile-1.8`"
 	{ $as_echo "$as_me:$LINENO: result: $GUILE_CFLAGS" >&5
 $as_echo "$GUILE_CFLAGS" >&6; }
 	{ $as_echo "$as_me:$LINENO: checking guile link flags" >&5
 $as_echo_n "checking guile link flags... " >&6; }
-	GUILE_LDFLAGS="`$guile_config link`"
+	GUILE_LDFLAGS="`pkg-config --libs guile-1.8`"
 	{ $as_echo "$as_me:$LINENO: result: $GUILE_LDFLAGS" >&5
 $as_echo "$GUILE_LDFLAGS" >&6; }
     fi
