diff -Nur orig/opt/google/earth/pro/google-earth-pro.desktop mod/opt/google/earth/pro/google-earth-pro.desktop
--- orig/opt/google/earth/pro/google-earth-pro.desktop	2018-10-31 16:23:39.000000000 +0100
+++ mod/opt/google/earth/pro/google-earth-pro.desktop	2019-01-03 12:25:18.236620141 +0100
@@ -3,9 +3,8 @@
 Name=Google Earth
 GenericName=3D planet viewer
 Comment=Explore, search and discover the planet
-Exec=/opt/google/earth/pro/google-earth-pro %f
+Exec=/usr/lib/google/earth/pro/googleearth %f
 Terminal=false
-MultipleArgs=false
 Icon=google-earth-pro
 Type=Application
 Categories=Application;Network
