diff -Naur orig/src/Makefile.am patched/src/Makefile.am
--- orig/src/Makefile.am	2016-10-11 15:27:44.000000000 +0200
+++ patched/src/Makefile.am	2018-06-07 17:01:19.989556668 +0200
@@ -28,6 +28,7 @@
 	$(GTHREAD_CFLAGS) \
 	$(GIO_CFLAGS) \
 	$(GIO_UNIX_CFLAGS) \
+	$(DBUS_GLIB_CFLAGS) \
 	$(LIBEXIF_CFLAGS) \
 	$(XFCONF_CFLAGS) \
 	$(EXO_CFLAGS) \
@@ -46,6 +47,7 @@
 	$(GTHREAD_LIBS) \
 	$(GIO_LIBS) \
 	$(GIO_UNIX_LIBS) \
+	$(DBUS_GLIB_LIBS) \
 	$(LIBEXIF_LIBS) \
 	$(XFCONF_LIBS) \
 	$(EXO_LIBS) \
diff -Naur orig/src/Makefile.in patched/src/Makefile.in
--- orig/src/Makefile.in	2017-02-01 13:09:11.000000000 +0100
+++ patched/src/Makefile.in	2018-06-07 17:02:13.669957519 +0200
@@ -404,6 +404,7 @@
 	$(GTHREAD_CFLAGS) \
 	$(GIO_CFLAGS) \
 	$(GIO_UNIX_CFLAGS) \
+	$(DBUS_GLIB_CFLAGS) \
 	$(LIBEXIF_CFLAGS) \
 	$(XFCONF_CFLAGS) \
 	$(EXO_CFLAGS) \
@@ -422,6 +423,7 @@
 	$(GTHREAD_LIBS) \
 	$(GIO_LIBS) \
 	$(GIO_UNIX_LIBS) \
+	$(DBUS_GLIB_LIBS) \
 	$(LIBEXIF_LIBS) \
 	$(XFCONF_LIBS) \
 	$(EXO_LIBS) \
