diff -Naur a/configure b/configure
--- a/configure	2023-04-13 16:16:01.000000000 +0200
+++ b/configure	2023-04-20 14:03:01.289178935 +0200
@@ -36060,97 +36060,6 @@
 	fi
 fi
 
-pkg_failed=no
-{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for bdw-gc >= 7.2" >&5
-printf %s "checking for bdw-gc >= 7.2... " >&6; }
-
-if test -n "$BDW_GC_CFLAGS"; then
-    pkg_cv_BDW_GC_CFLAGS="$BDW_GC_CFLAGS"
- elif test -n "$PKG_CONFIG"; then
-    if test -n "$PKG_CONFIG" && \
-    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"bdw-gc >= 7.2\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "bdw-gc >= 7.2") 2>&5
-  ac_status=$?
-  printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
-  test $ac_status = 0; }; then
-  pkg_cv_BDW_GC_CFLAGS=`$PKG_CONFIG --cflags "bdw-gc >= 7.2" 2>/dev/null`
-		      test "x$?" != "x0" && pkg_failed=yes
-else
-  pkg_failed=yes
-fi
- else
-    pkg_failed=untried
-fi
-if test -n "$BDW_GC_LIBS"; then
-    pkg_cv_BDW_GC_LIBS="$BDW_GC_LIBS"
- elif test -n "$PKG_CONFIG"; then
-    if test -n "$PKG_CONFIG" && \
-    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"bdw-gc >= 7.2\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "bdw-gc >= 7.2") 2>&5
-  ac_status=$?
-  printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
-  test $ac_status = 0; }; then
-  pkg_cv_BDW_GC_LIBS=`$PKG_CONFIG --libs "bdw-gc >= 7.2" 2>/dev/null`
-		      test "x$?" != "x0" && pkg_failed=yes
-else
-  pkg_failed=yes
-fi
- else
-    pkg_failed=untried
-fi
-
-
-
-if test $pkg_failed = yes; then
-        { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: no" >&5
-printf "%s\n" "no" >&6; }
-
-if $PKG_CONFIG --atleast-pkgconfig-version 0.20; then
-        _pkg_short_errors_supported=yes
-else
-        _pkg_short_errors_supported=no
-fi
-        if test $_pkg_short_errors_supported = yes; then
-	        BDW_GC_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "bdw-gc >= 7.2" 2>&1`
-        else
-	        BDW_GC_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "bdw-gc >= 7.2" 2>&1`
-        fi
-	# Put the nasty error message in config.log where it belongs
-	echo "$BDW_GC_PKG_ERRORS" >&5
-
-	as_fn_error $? "Package requirements (bdw-gc >= 7.2) were not met:
-
-$BDW_GC_PKG_ERRORS
-
-Consider adjusting the PKG_CONFIG_PATH environment variable if you
-installed software in a non-standard prefix.
-
-Alternatively, you may set the environment variables BDW_GC_CFLAGS
-and BDW_GC_LIBS to avoid the need to call pkg-config.
-See the pkg-config man page for more details." "$LINENO" 5
-elif test $pkg_failed = untried; then
-        { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: no" >&5
-printf "%s\n" "no" >&6; }
-	{ { printf "%s\n" "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-printf "%s\n" "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "The pkg-config script could not be found or is too old.  Make sure it
-is in your PATH or set the PKG_CONFIG environment variable to the full
-path to pkg-config.
-
-Alternatively, you may set the environment variables BDW_GC_CFLAGS
-and BDW_GC_LIBS to avoid the need to call pkg-config.
-See the pkg-config man page for more details.
-
-To get pkg-config, see <http://pkg-config.freedesktop.org/>.
-See \`config.log' for more details" "$LINENO" 5; }
-else
-	BDW_GC_CFLAGS=$pkg_cv_BDW_GC_CFLAGS
-	BDW_GC_LIBS=$pkg_cv_BDW_GC_LIBS
-        { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-printf "%s\n" "yes" >&6; }
-
-fi
-
 
 
   { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for systempapername in -lpaper" >&5
diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2023-04-13 16:05:19.000000000 +0200
+++ b/configure.ac	2023-04-20 14:01:41.384916514 +0200
@@ -98,20 +98,6 @@
 ## checks for library functions.  ##
 ## ------------------------------ ##
 
-# libgc (BDW garbage collector)
-PKG_CHECK_MODULES([BDW_GC], [bdw-gc >= 7.2])
-AH_BOTTOM([
-#define malloc GC_malloc
-#define calloc GC_calloc
-#define realloc GC_realloc
-#ifndef free
-#define free GC_free
-#endif
-#define strdup GC_strdup
-#define strndup GC_strndup
-#define reallocarray a2ps_reallocarray
-])
-
 ad_FUNC_SYSTEMPAPERNAME
 AS_IF([test "$ac_cv_lib_paper_systempapername" = "no"],
   [AC_MSG_ERROR([a2ps needs libpaper to work])])
diff -Naur a/liba2ps/a2ps.h b/liba2ps/a2ps.h
--- a/liba2ps/a2ps.h	2022-06-11 17:31:41.000000000 +0200
+++ b/liba2ps/a2ps.h	2023-04-20 14:08:31.266247246 +0200
@@ -20,6 +20,5 @@
 # define _A2PS_H_
 
 # include "system.h"
-# include <gc/gc.h>
 
 #endif
