diff -Naur orig/build/config/linux/BUILD.gn patched/build/config/linux/BUILD.gn
--- orig/build/config/linux/BUILD.gn	2020-02-04 02:56:08.000000000 +0100
+++ patched/build/config/linux/BUILD.gn	2020-02-08 00:05:45.682816182 +0100
@@ -97,7 +97,7 @@
       "gthread-2.0",
     ]
     defines = [
-      "GLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_40",
+      "GLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_60",
       "GLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_40",
     ]
   }
