diff -Naur orig/third_party/closure_compiler/compiler.py patched/third_party/closure_compiler/compiler.py
--- orig/third_party/closure_compiler/compiler.py	2021-07-19 20:45:40.000000000 +0200
+++ patched/third_party/closure_compiler/compiler.py	2021-07-23 00:25:22.095236919 +0200
@@ -12,8 +12,8 @@
 import subprocess
 
 
-_CURRENT_DIR = os.path.join(os.path.dirname(__file__))
-_JAVA_PATH = os.path.join(_CURRENT_DIR, "..", "jdk", "current", "bin", "java")
+_CURRENT_DIR = "/usr"
+_JAVA_PATH = os.path.join(_CURRENT_DIR, "lib", "java", "jre", "bin", "java")
 assert os.path.isfile(_JAVA_PATH), "java only allowed in android builds"
 
 class Compiler(object):
