diff -Naur orig/build/config/compiler/BUILD.gn patched/build/config/compiler/BUILD.gn
--- orig/build/config/compiler/BUILD.gn	2020-02-04 02:56:08.000000000 +0100
+++ patched/build/config/compiler/BUILD.gn	2020-02-08 00:07:17.320166816 +0100
@@ -1444,6 +1444,7 @@
       cflags_cc += [
         # See comment for -Wno-c++11-narrowing.
         "-Wno-narrowing",
+        "-Wno-class-memaccess",
       ]
 
       # -Wno-class-memaccess warns about hash table and vector in blink.
