diff -Naur orig/build/config/compiler/BUILD.gn patched/build/config/compiler/BUILD.gn
--- orig/build/config/compiler/BUILD.gn	2018-06-06 21:13:32.000000000 +0200
+++ patched/build/config/compiler/BUILD.gn	2018-06-09 00:34:18.912677038 +0200
@@ -1448,6 +1448,7 @@
       cflags_cc += [
         # See comment for -Wno-c++11-narrowing.
         "-Wno-narrowing",
+        "-Wno-class-memaccess",
       ]
 
       # -Wunused-local-typedefs is broken in gcc,
