diff -Naur a/chrome/browser/ui/startup/infobar_utils.cc b/chrome/browser/ui/startup/infobar_utils.cc
--- a/chrome/browser/ui/startup/infobar_utils.cc	2024-06-16 11:27:02.324116885 +0200
+++ b/chrome/browser/ui/startup/infobar_utils.cc	2024-06-16 11:30:41.464851697 +0200
@@ -159,15 +159,5 @@
         ObsoleteSystemInfoBarDelegate::Create(infobar_manager);
       }
     }
-
-#if !BUILDFLAG(IS_CHROMEOS) && !BUILDFLAG(IS_ANDROID)
-    if (!is_web_app &&
-        !startup_command_line.HasSwitch(switches::kNoDefaultBrowserCheck)) {
-      // The default browser prompt should only be shown after the first run.
-      if (is_first_run == chrome::startup::IsFirstRun::kNo) {
-        ShowDefaultBrowserPrompt(profile);
-      }
-    }
-#endif
   }
 }
