diff -Naur orig/third_party/libpng/BUILD.gn patched/third_party/libpng/BUILD.gn
--- orig/third_party/libpng/BUILD.gn	2019-04-05 02:50:01.000000000 +0200
+++ patched/third_party/libpng/BUILD.gn	2019-04-11 01:38:42.029400138 +0200
@@ -48,7 +48,6 @@
     "pnglibconf.h",
     "pngmem.c",
     "pngpread.c",
-    "pngprefix.h",
     "pngpriv.h",
     "pngread.c",
     "pngrio.c",
diff -Naur orig/third_party/libpng/pnglibconf.h patched/third_party/libpng/pnglibconf.h
--- orig/third_party/libpng/pnglibconf.h	2019-04-05 02:50:01.000000000 +0200
+++ patched/third_party/libpng/pnglibconf.h	2019-04-11 05:18:20.492500289 +0200
@@ -225,13 +225,4 @@
 #define PNG_USER_CHUNK_MALLOC_MAX 4000000L
 /* end of chromium settings */
 
-/* chromium prefixing */
-/*
- * This is necessary to build multiple copies of libpng.  We need this while pdfium builds
- * its own copy of libpng.
- */
-#define PNG_PREFIX
-#include "pngprefix.h"
-/* end of chromium prefixing */
-
 #endif /* PNGLCONF_H */
diff -Naur orig/third_party/libpng/pngpriv.h patched/third_party/libpng/pngpriv.h
--- orig/third_party/libpng/pngpriv.h	2019-04-05 02:50:01.000000000 +0200
+++ patched/third_party/libpng/pngpriv.h	2019-04-11 01:39:06.581756443 +0200
@@ -72,11 +72,6 @@
 #  include "pnglibconf.h"
 #endif
 
-/* Local renames may change non-exported API functions from png.h */
-#if defined(PNG_PREFIX) && !defined(PNGPREFIX_H)
-#  include "pngprefix.h"
-#endif
-
 #ifdef PNG_USER_CONFIG
 #  include "pngusr.h"
    /* These should have been defined in pngusr.h */
