diff -Naur orig/chrome/browser/ui/startup/infobar_utils.cc patched/chrome/browser/ui/startup/infobar_utils.cc
--- orig/chrome/browser/ui/startup/infobar_utils.cc	2022-02-01 01:41:44.000000000 +0100
+++ patched/chrome/browser/ui/startup/infobar_utils.cc	2022-02-02 11:47:01.728138583 +0100
@@ -15,7 +15,6 @@
 #include "chrome/browser/ui/startup/automation_infobar_delegate.h"
 #include "chrome/browser/ui/startup/bad_flags_prompt.h"
 #include "chrome/browser/ui/startup/default_browser_prompt.h"
-#include "chrome/browser/ui/startup/google_api_keys_infobar_delegate.h"
 #include "chrome/browser/ui/startup/obsolete_system_infobar_delegate.h"
 #include "chrome/browser/ui/startup/startup_browser_creator.h"
 #include "chrome/browser/ui/startup/startup_types.h"
@@ -24,7 +23,6 @@
 #include "components/infobars/content/content_infobar_manager.h"
 #include "components/prefs/pref_service.h"
 #include "content/public/common/content_switches.h"
-#include "google_apis/google_api_keys.h"
 
 namespace {
 bool ShouldShowBadFlagsSecurityWarnings() {
@@ -113,9 +111,6 @@
     infobars::ContentInfoBarManager* infobar_manager =
         infobars::ContentInfoBarManager::FromWebContents(web_contents);
 
-    if (!google_apis::HasAPIKeyConfigured())
-      GoogleApiKeysInfoBarDelegate::Create(infobar_manager);
-
     if (ObsoleteSystem::IsObsoleteNowOrSoon()) {
       PrefService* local_state = g_browser_process->local_state();
       if (!local_state ||
