diff -Naur orig/tools/gn/tools/gn/ninja_action_target_writer_unittest.cc patched/tools/gn/tools/gn/ninja_action_target_writer_unittest.cc
--- orig/tools/gn/tools/gn/ninja_action_target_writer_unittest.cc	2020-02-04 03:04:02.000000000 +0100
+++ patched/tools/gn/tools/gn/ninja_action_target_writer_unittest.cc	2020-02-08 00:11:29.123876427 +0100
@@ -54,7 +54,7 @@
   ASSERT_TRUE(target.OnResolved(&err));
 
   setup.build_settings()->set_python_path(
-      base::FilePath(FILE_PATH_LITERAL("/usr/bin/python")));
+      base::FilePath(FILE_PATH_LITERAL("/usr/bin/python2")));
 
   std::ostringstream out;
   NinjaActionTargetWriter writer(&target, out);
@@ -96,7 +96,7 @@
   ASSERT_TRUE(target.OnResolved(&err));
 
   setup.build_settings()->set_python_path(
-      base::FilePath(FILE_PATH_LITERAL("/usr/bin/python")));
+      base::FilePath(FILE_PATH_LITERAL("/usr/bin/python2")));
 
   std::ostringstream out;
   NinjaActionTargetWriter writer(&target, out);
@@ -138,7 +138,7 @@
   ASSERT_TRUE(target.OnResolved(&err));
 
   setup.build_settings()->set_python_path(
-      base::FilePath(FILE_PATH_LITERAL("/usr/bin/python")));
+      base::FilePath(FILE_PATH_LITERAL("/usr/bin/python2")));
 
   std::ostringstream out;
   NinjaActionTargetWriter writer(&target, out);
@@ -198,7 +198,7 @@
   ASSERT_TRUE(target.OnResolved(&err));
 
   setup.build_settings()->set_python_path(
-      base::FilePath(FILE_PATH_LITERAL("/usr/bin/python")));
+      base::FilePath(FILE_PATH_LITERAL("/usr/bin/python2")));
 
   std::ostringstream out;
   NinjaActionTargetWriter writer(&target, out);
@@ -263,7 +263,7 @@
   target.config_values().inputs().push_back(SourceFile("//foo/included.txt"));
 
   setup.build_settings()->set_python_path(
-      base::FilePath(FILE_PATH_LITERAL("/usr/bin/python")));
+      base::FilePath(FILE_PATH_LITERAL("/usr/bin/python2")));
 
   std::ostringstream out;
   NinjaActionTargetWriter writer(&target, out);
@@ -318,7 +318,7 @@
       SubstitutionList::MakeForTest("//out/Debug/{{source_name_part}}.out");
 
   setup.build_settings()->set_python_path(
-      base::FilePath(FILE_PATH_LITERAL("/usr/bin/python")));
+      base::FilePath(FILE_PATH_LITERAL("/usr/bin/python2")));
 
   std::ostringstream out;
   NinjaActionTargetWriter writer(&target, out);
@@ -376,7 +376,7 @@
       SubstitutionList::MakeForTest("//out/Debug/{{source_name_part}}.out");
 
   setup.build_settings()->set_python_path(
-      base::FilePath(FILE_PATH_LITERAL("/usr/bin/python")));
+      base::FilePath(FILE_PATH_LITERAL("/usr/bin/python2")));
 
   std::ostringstream out;
   NinjaActionTargetWriter writer(&target, out);
@@ -405,7 +405,7 @@
   TestWithScope setup;
 
   setup.build_settings()->set_python_path(
-      base::FilePath(FILE_PATH_LITERAL("/usr/bin/python")));
+      base::FilePath(FILE_PATH_LITERAL("/usr/bin/python2")));
 
   Target dep(setup.settings(), Label(SourceDir("//foo/"), "dep"));
   dep.set_output_type(Target::ACTION);
diff -Naur orig/tools/gn/tools/gn/setup.cc patched/tools/gn/tools/gn/setup.cc
--- orig/tools/gn/tools/gn/setup.cc	2020-02-04 03:04:02.000000000 +0100
+++ patched/tools/gn/tools/gn/setup.cc	2020-02-08 00:11:29.123876427 +0100
@@ -701,7 +701,7 @@
     }
     build_settings_.set_python_path(python_path.NormalizePathSeparatorsTo('/'));
 #else
-    build_settings_.set_python_path(base::FilePath("python"));
+    build_settings_.set_python_path(base::FilePath("python2"));
 #endif
   }
   return true;
