diff -Naur a/ui/qt/BUILD.gn b/ui/qt/BUILD.gn
--- a/ui/qt/BUILD.gn	2024-04-22 15:35:34.410300360 +0200
+++ b/ui/qt/BUILD.gn	2024-04-22 16:10:18.069102263 +0200
@@ -43,10 +43,10 @@
 
 template("qt_shim") {
   if (!use_sysroot) {
-    action("generate_moc" + invoker.qt_version) {
+    action("generate_moc6") {
       script = "moc_wrapper.py"
       inputs = [ "//ui/qt/qt_shim.h" ]
-      outputs = [ "$root_gen_dir/qt" + invoker.qt_version + "/qt_shim_moc.cc" ]
+      outputs = [ "$root_gen_dir/qt6/qt_shim_moc.cc" ]
       args = rebase_path(inputs + outputs, root_build_dir)
       if (invoker.moc_qt_path != "") {
         args += [
@@ -57,10 +57,11 @@
     }
   }
 
-  pkg_config("qt" + invoker.qt_version + "_config") {
+  pkg_config("qt6_config") {
     packages = [
-      "Qt" + invoker.qt_version + "Core",
-      "Qt" + invoker.qt_version + "Widgets",
+      "Qt6Core",
+      "Qt6Gui",
+      "Qt6Widgets",
     ]
   }
 
@@ -83,7 +83,7 @@
     configs -= [ "//build/config/compiler:runtime_library" ]
     configs += [
       ":qt_internal_config",
-      ":qt" + invoker.qt_version + "_config",
+      ":qt6_config",
     ]
 
     public = []
@@ -94,22 +94,15 @@
     if (use_sysroot) {
       # This file is generated with gen_qt_shim_moc.sh on an amd64 system to
       # avoid a build-time dependency on `moc` when using the sysroot.
-      sources += [ "qt" + invoker.qt_version + "_shim_moc.cc" ]
+      sources += [ "qt6_shim_moc.cc" ]
     } else {
-      sources += get_target_outputs(":generate_moc" + invoker.qt_version)
-      deps += [ ":generate_moc" + invoker.qt_version ]
+      sources += get_target_outputs(":generate_moc6")
+      deps += [ ":generate_moc6" ]
     }
   }
 }
-qt_shim("qt5_shim") {
-  qt_version = "5"
-  if (!use_sysroot) {
-    moc_qt_path = "$moc_qt5_path"
-  }
-}
 if (use_qt6) {
   qt_shim("qt6_shim") {
-    qt_version = "6"
     if (!use_sysroot) {
       moc_qt_path = "$moc_qt6_path"
     }
@@ -121,10 +114,8 @@
 
   defines = [ "IS_QT_IMPL" ]
 
-  # qt_shim is in data_deps since we want to load it manually.
-  data_deps = [ ":qt5_shim" ]
   if (use_qt6) {
-    data_deps += [ ":qt6_shim" ]
+    data_deps = [ ":qt6_shim" ]
   }
   deps = [
     ":qt_interface",
diff -Naur a/ui/qt/qt_ui.cc b/ui/qt/qt_ui.cc
--- a/ui/qt/qt_ui.cc	2024-04-22 15:35:34.410300360 +0200
+++ b/ui/qt/qt_ui.cc	2024-04-22 15:36:11.346409800 +0200
@@ -209,16 +209,8 @@
   if (!base::PathService::Get(base::DIR_MODULE, &path)) {
     return false;
   }
-  void* libqt_shim = nullptr;
-  auto load_libqt_shim = [&](int qt_version) -> bool {
-    auto file_name = base::StringPrintf("libqt%d_shim.so", qt_version);
-    if ((libqt_shim = LoadLibrary(path.Append(file_name)))) {
-      qt_version_ = qt_version;
-    }
-    return libqt_shim;
-  };
-  PreferQt6() ? load_libqt_shim(6) || load_libqt_shim(5)
-              : load_libqt_shim(5) || load_libqt_shim(6);
+  path = path.Append("libqt6_shim.so");
+  void* libqt_shim = dlopen(path.value().c_str(), RTLD_NOW | RTLD_GLOBAL);
   if (!libqt_shim) {
     return false;
   }
