diff -Naur a/chrome/browser/ui/startup/infobar_utils.cc b/chrome/browser/ui/startup/infobar_utils.cc
--- a/chrome/browser/ui/startup/infobar_utils.cc	2022-10-27 02:36:02.015311927 +0200
+++ b/chrome/browser/ui/startup/infobar_utils.cc	2022-10-27 14:54:59.395475919 +0200
@@ -118,13 +118,5 @@
         ObsoleteSystemInfoBarDelegate::Create(infobar_manager);
     }
 
-#if !BUILDFLAG(IS_CHROMEOS_ASH)
-    if (!is_web_app &&
-        !startup_command_line.HasSwitch(switches::kNoDefaultBrowserCheck)) {
-      // The default browser prompt should only be shown after the first run.
-      if (is_first_run == chrome::startup::IsFirstRun::kNo)
-        ShowDefaultBrowserPrompt(profile);
-    }
-#endif
   }
 }
