diff -Naur a/chrome/browser/ui/startup/infobar_utils.cc b/chrome/browser/ui/startup/infobar_utils.cc
--- a/chrome/browser/ui/startup/infobar_utils.cc	2023-02-10 23:42:23.262949183 +0100
+++ b/chrome/browser/ui/startup/infobar_utils.cc	2023-02-10 23:44:40.107274377 +0100
@@ -144,13 +144,5 @@
         ObsoleteSystemInfoBarDelegate::Create(infobar_manager);
     }
 
-#if !BUILDFLAG(IS_CHROMEOS_ASH)
-    if (!is_web_app &&
-        !startup_command_line.HasSwitch(switches::kNoDefaultBrowserCheck)) {
-      // The default browser prompt should only be shown after the first run.
-      if (is_first_run == chrome::startup::IsFirstRun::kNo)
-        ShowDefaultBrowserPrompt(profile);
-    }
-#endif
   }
 }
