diff -Nur orig/powermanga.desktop mod/powermanga.desktop
--- orig/powermanga.desktop	2014-09-01 14:03:11.000000000 +0200
+++ mod/powermanga.desktop	2018-02-08 13:09:21.042197288 +0100
@@ -5,8 +5,9 @@
 Comment=Play a vertical shoot'em up game
 Comment[de]=Spiele ein vertikales "Shoot 'em up"-Spiel
 Comment[fr]=Jouer un jeu de shoot them up vertical
+Comment[hu]=Függőleges lövöldöző játék
 Icon=powermanga
-Exec=powermanga
+Exec=powermanga --window --4x
 Terminal=false
 Categories=Game;ArcadeGame;
 Keywords=shmup;arcade;shooter;colourful;weapons;spaceship;
