diff -Naur orig/app/core/gimpbrush-save.c patched/app/core/gimpbrush-save.c
--- orig/app/core/gimpbrush-save.c	2019-02-22 00:02:52.000000000 +0100
+++ patched/app/core/gimpbrush-save.c	2019-02-22 02:47:36.967708184 +0100
@@ -17,6 +17,8 @@
 
 #include "config.h"
 
+#include <string.h>
+
 #include <gdk-pixbuf/gdk-pixbuf.h>
 
 #include "core-types.h"
diff -Naur orig/app/core/gimpchunkiterator.c patched/app/core/gimpchunkiterator.c
--- orig/app/core/gimpchunkiterator.c	2019-02-22 00:02:52.000000000 +0100
+++ patched/app/core/gimpchunkiterator.c	2019-02-22 02:47:36.967708184 +0100
@@ -22,6 +22,7 @@
 #include "config.h"
 
 #include <stdlib.h>
+#include <string.h>
 
 #include <cairo.h>
 #include <gdk-pixbuf/gdk-pixbuf.h>
diff -Naur orig/app/core/gimpdatafactory.c patched/app/core/gimpdatafactory.c
--- orig/app/core/gimpdatafactory.c	2019-02-22 00:02:52.000000000 +0100
+++ patched/app/core/gimpdatafactory.c	2019-02-22 02:47:36.967708184 +0100
@@ -20,6 +20,8 @@
 
 #include "config.h"
 
+#include <stdlib.h>
+
 #include <gdk-pixbuf/gdk-pixbuf.h>
 #include <gegl.h>
 
diff -Naur orig/app/core/gimplineart.c patched/app/core/gimplineart.c
--- orig/app/core/gimplineart.c	2019-02-22 00:02:52.000000000 +0100
+++ patched/app/core/gimplineart.c	2019-02-22 02:47:36.967708184 +0100
@@ -20,6 +20,8 @@
 
 #include "config.h"
 
+#include <stdlib.h>
+
 #include <gdk-pixbuf/gdk-pixbuf.h>
 #include <gegl.h>
 
diff -Naur orig/app/core/gimppattern-save.c patched/app/core/gimppattern-save.c
--- orig/app/core/gimppattern-save.c	2019-02-22 00:02:52.000000000 +0100
+++ patched/app/core/gimppattern-save.c	2019-02-22 02:47:36.967708184 +0100
@@ -17,6 +17,8 @@
 
 #include "config.h"
 
+#include <string.h>
+
 #include <gdk-pixbuf/gdk-pixbuf.h>
 
 #include "core-types.h"
diff -Naur orig/app/gegl/gimptilehandlervalidate.c patched/app/gegl/gimptilehandlervalidate.c
--- orig/app/gegl/gimptilehandlervalidate.c	2019-02-22 00:02:52.000000000 +0100
+++ patched/app/gegl/gimptilehandlervalidate.c	2019-02-22 02:47:36.967708184 +0100
@@ -17,6 +17,8 @@
 
 #include "config.h"
 
+#include <string.h>
+
 #include <cairo.h>
 #include <gegl.h>
 
diff -Naur orig/app/tools/gimptransformgridtoolundo.c patched/app/tools/gimptransformgridtoolundo.c
--- orig/app/tools/gimptransformgridtoolundo.c	2019-02-22 00:02:53.000000000 +0100
+++ patched/app/tools/gimptransformgridtoolundo.c	2019-02-22 02:48:56.588558880 +0100
@@ -17,6 +17,8 @@
 
 #include "config.h"
 
+#include <string.h>
+
 #include <gegl.h>
 #include <gtk/gtk.h>
 
diff -Naur orig/libgimp/gimpui.c patched/libgimp/gimpui.c
--- orig/libgimp/gimpui.c	2019-02-22 00:02:55.000000000 +0100
+++ patched/libgimp/gimpui.c	2019-02-22 02:47:36.971708227 +0100
@@ -18,6 +18,8 @@
 
 #include "config.h"
 
+#include <string.h>
+
 #include <gtk/gtk.h>
 
 #ifdef GDK_WINDOWING_WIN32
diff -Naur orig/libgimpwidgets/gimpspinbutton.c patched/libgimpwidgets/gimpspinbutton.c
--- orig/libgimpwidgets/gimpspinbutton.c	2019-02-22 00:02:53.000000000 +0100
+++ patched/libgimpwidgets/gimpspinbutton.c	2019-02-22 02:47:36.971708227 +0100
@@ -21,6 +21,8 @@
 
 #include "config.h"
 
+#include <stdlib.h>
+
 #include <gegl.h>
 #include <gtk/gtk.h>
 
diff -Naur orig/plug-ins/common/jigsaw.c patched/plug-ins/common/jigsaw.c
--- orig/plug-ins/common/jigsaw.c	2019-02-22 00:02:53.000000000 +0100
+++ patched/plug-ins/common/jigsaw.c	2019-02-22 02:53:19.339366197 +0100
@@ -36,6 +36,8 @@
 
 #include <string.h>
 
+#include <glib.h>
+
 #include <libgimp/gimp.h>
 #include <libgimp/gimpui.h>
 
