diff -Nur orig/extensions/goat-exercises/goat-exercise-lua.lua mod/extensions/goat-exercises/goat-exercise-lua.lua
--- orig/extensions/goat-exercises/goat-exercise-lua.lua	2024-02-17 18:01:20.000000000 +0100
+++ mod/extensions/goat-exercises/goat-exercise-lua.lua	2024-10-20 09:17:46.662389549 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env luajit
+#!/usr/bin/lua5.3
 
 -- GIMP - The GNU Image Manipulation Program
 -- Copyright (C) 1995 Spencer Kimball and Peter Mattis
