diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2014-02-05 17:27:26.000000000 +0100
+++ patched/CMakeLists.txt	2016-07-11 01:28:32.414697367 +0200
@@ -13,5 +13,5 @@
 add_subdirectory (parse-edid)
 
 INSTALL(FILES get-edid.1 DESTINATION share/man/man1)
-INSTALL(FILES AUTHORS ChangeLog COPYING README DESTINATION
+INSTALL(FILES AUTHORS ChangeLog README DESTINATION
 	share/doc/read-edid)
