diff -Naur orig/data/Makefile.am patched/data/Makefile.am
--- orig/data/Makefile.am	2020-09-07 15:18:07.000000000 +0200
+++ patched/data/Makefile.am	2020-12-27 22:54:48.451668925 +0100
@@ -15,7 +15,7 @@
 
 # ----------------------------------------------------------------------------------------------------
 
-dbusconfdir = $(sysconfdir)/dbus-1/system.d
+dbusconfdir = $(datadir)/dbus-1/system.d
 dbusconf_in_files = org.freedesktop.PolicyKit1.conf.in
 dbusconf_DATA = $(dbusconf_in_files:.conf.in=.conf)
 
diff -Naur orig/data/Makefile.in patched/data/Makefile.in
--- orig/data/Makefile.in	2020-09-09 13:32:33.000000000 +0200
+++ patched/data/Makefile.in	2020-12-27 22:55:20.363763721 +0100
@@ -375,7 +375,7 @@
 service_DATA = $(service_in_files:.service.in=.service)
 
 # ----------------------------------------------------------------------------------------------------
-dbusconfdir = $(sysconfdir)/dbus-1/system.d
+dbusconfdir = $(datadir)/dbus-1/system.d
 dbusconf_in_files = org.freedesktop.PolicyKit1.conf.in
 dbusconf_DATA = $(dbusconf_in_files:.conf.in=.conf)
 
