diff -Nur orig/src/osflags mod/src/osflags
--- orig/src/osflags	2014-06-16 22:28:43.000000000 +0200
+++ mod/src/osflags	2019-08-13 16:29:45.650387433 +0200
@@ -19,7 +19,7 @@
 		Linux)
 			FLAGS="";
 			[ -e /usr/include/selinux/selinux.h ] && FLAGS="$FLAGS -lselinux";
-			[ -e /usr/include/systemd/sd-daemon.h ] && FLAGS="$FLAGS -lsystemd-daemon";
+			[ -e /usr/include/systemd/sd-daemon.h ] && FLAGS="$FLAGS -lsystemd";
 			echo $FLAGS;
 		;;
 	esac
