diff -Naur orig/src/backup/k9dvdbackup.cpp patched/src/backup/k9dvdbackup.cpp
--- orig/src/backup/k9dvdbackup.cpp	2014-09-29 18:55:40.000000000 +0200
+++ patched/src/backup/k9dvdbackup.cpp	2016-11-18 01:40:26.409845469 +0100
@@ -33,6 +33,7 @@
 #include <stdarg.h>
 #include <errno.h>
 #include <string.h>
+#include <math.h>
 #include <qdir.h>
 
 #include <kmessagebox.h>
@@ -907,7 +908,8 @@
 
 
         if ((vobu1 !=NULL) && (vobu2!=NULL)) {
-            *_offset = abs(vobu1->newSector - vobu2->newSector)  | maskOffset1 ;
+            *_offset = fabs(vobu1->newSector - vobu2->newSector);
+            *_offset |= maskOffset1;
             *_offset |= maskOffset2;
             return vobu2;
         }

