diff -Naur orig/src/backup/k9execcopy.cpp patched/src/backup/k9execcopy.cpp
--- orig/src/backup/k9execcopy.cpp	2014-09-29 17:00:12.000000000 +0200
+++ patched/src/backup/k9execcopy.cpp	2016-11-18 01:53:51.975792018 +0100
@@ -306,7 +306,7 @@
 
 #if QT_VERSION >= 0x050000
     m_progressDialog=new QProgressDialog(k9Dialogs::getMainWidget() );
-    m_progressDialog->setCancelButton(false);
+    m_progressDialog->setCancelButton(0);
 
 #else
     m_progressDialog=new KProgressDialog(k9Dialogs::getMainWidget() );

