diff -Naur orig/po/emelfm2.desktop.in mod/po/emelfm2.desktop.in
--- orig/po/emelfm2.desktop.in	2013-12-01 22:05:38.000000000 +0100
+++ mod/po/emelfm2.desktop.in	2014-06-07 20:28:13.000000000 +0200
@@ -1,6 +1,4 @@
 [Desktop Entry]
-Version=1.0
-Encoding=UTF-8
 Name=emelFM2
 GenericName=file manager
 Comment=2-pane file manager
@@ -16,5 +14,5 @@
 Icon=emelfm2
 Terminal=false
 StartupNotify=false
-Categories=System;Utility;Core;FileTools;FileManager;GTK;
+Categories=System;FileTools;FileManager;GTK;
 MimeType=inode/directory;inode/mount-point;
