diff -Naur orig/ftparchive/writer.cc patched/ftparchive/writer.cc
--- orig/ftparchive/writer.cc	2019-03-08 09:41:20.000000000 +0100
+++ patched/ftparchive/writer.cc	2019-03-20 02:08:13.606205922 +0100
@@ -372,7 +372,7 @@
       string const &Arch, bool const IncludeArchAll) :
    FTWScanner(GivenOutput, Arch, IncludeArchAll), Db(DB), Stats(Db.Stats), TransWriter(transWriter)
 {
-   SetExts(".deb .ddeb .udeb");
+   SetExts(".uhu .deb .ddeb .udeb");
    DeLinkLimit = 0;
 
    // Process the command line options
