diff -Naur orig/apt-pkg/contrib/strutl.cc patched/apt-pkg/contrib/strutl.cc
--- orig/apt-pkg/contrib/strutl.cc	2017-07-13 23:45:39.134739579 +0200
+++ patched/apt-pkg/contrib/strutl.cc	2017-08-13 00:06:25.819137900 +0200
@@ -401,7 +401,7 @@
 
    /* bytes, KiloBytes, MegaBytes, GigaBytes, TeraBytes, PetaBytes, 
       ExaBytes, ZettaBytes, YottaBytes */
-   char Ext[] = {'\0','k','M','G','T','P','E','Z','Y'};
+   char Ext[] = {'\0','K','M','G','T','P','E','Z','Y'};
    int I = 0;
    while (I <= 8)
    {
@@ -418,7 +418,7 @@
 	 strprintf(S, "%'.0f %c", ASize, Ext[I]);
 	 return S;
       }
-      ASize /= 1000.0;
+      ASize /= 1024.0;
       I++;
    }
    return "";
