diff -Naur orig/cmdline/apt-key.in patched/cmdline/apt-key.in
--- orig/cmdline/apt-key.in	2020-08-04 12:41:38.000000000 +0200
+++ patched/cmdline/apt-key.in	2020-08-10 01:48:26.056003924 +0200
@@ -730,9 +730,9 @@
 	return
     fi
     # (Maintainer) scripts should not be using apt-key
-    if [ -n "$DPKG_MAINTSCRIPT_PACKAGE" ]; then
-	echo >&2 "Warning: apt-key should not be used in scripts (called from $DPKG_MAINTSCRIPT_NAME maintainerscript of the package ${DPKG_MAINTSCRIPT_PACKAGE})"
-    fi
+    # if [ -n "$DPKG_MAINTSCRIPT_PACKAGE" ]; then
+	# echo >&2 "Warning: apt-key should not be used in scripts (called from $DPKG_MAINTSCRIPT_NAME maintainerscript of the package ${DPKG_MAINTSCRIPT_PACKAGE})"
+    # fi
 
     echo >&2 "Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8))."
 }
