diff -Naur orig/apt-pkg/acquire-item.cc patched/apt-pkg/acquire-item.cc
--- orig/apt-pkg/acquire-item.cc	2021-06-09 13:42:02.000000000 +0200
+++ patched/apt-pkg/acquire-item.cc	2021-06-26 17:06:34.143171990 +0200
@@ -1757,7 +1757,7 @@
 	 return _error->Error("%s", errmsg.c_str());
       }
    }
-
+  /*
    if (TransactionManager->MetaIndexParser->GetNotBefore() > 0)
    {
       time_t const invalid_for = TransactionManager->MetaIndexParser->GetNotBefore() - time(nullptr);
@@ -1776,7 +1776,7 @@
 	 return _error->Error("%s", errmsg.c_str());
       }
    }
-
+  */
    /* Did we get a file older than what we have? This is a last minute IMS hit and doubles
       as a prevention of downgrading us to older (still valid) files */
    if (TransactionManager->IMSHit == false && TransactionManager->LastMetaIndexParser != NULL &&
