diff -Naur orig/meson.build patched/meson.build
--- orig/meson.build	1970-01-01 01:00:00.000000000 +0100
+++ patched/meson.build	2020-09-20 20:56:05.579141820 +0200
@@ -44,7 +44,7 @@
 dependency('libsoup-2.4')
 dependency('tracker-sparql-3.0', version: '>= 2.99.3')
 dependency('pango', version: '>= 1.44.0')
-dependency('pygobject-3.0', version: '>= 3.36.1')
+dependency('pygobject-3.0', version: '>= 3.36.0')
 dependency('py3cairo', version: '>= 1.14.0')
 dependency('grilo-0.3', version: '>= 0.3.13', fallback: ['grilo', 'libgrl_dep'])
 dependency('grilo-plugins-0.3', version: '>= 0.3.12', fallback: ['grilo-plugins', 'grilo_plugins_dep'])
