diff -urN systemd-ui-v3.orig/src/gnome-ask-password-agent.vala systemd-ui-v3/src/gnome-ask-password-agent.vala
--- systemd-ui-v3.orig/src/gnome-ask-password-agent.vala	2014-07-09 09:08:26.000000000 +0200
+++ systemd-ui-v3/src/gnome-ask-password-agent.vala	2014-07-09 09:18:02.452135868 +0200
@@ -80,7 +80,7 @@
         string socket;
 
         PasswordDialog password_dialog;
-        Notification n;
+        Notify.Notification n;
 
         public MyStatusIcon() throws GLib.Error {
                 GLib.Object(icon_name : "dialog-password");
@@ -183,7 +183,7 @@
                 }
                 set_from_icon_name(icon);
 
-                n = new Notification(title, message, icon);
+                n = new Notify.Notification(title, message, icon);
                 n.set_timeout(5000);
                 n.closed.connect(() => {
                         set_visible(true);
