diff -Nur orig/GNUmakefile mod/GNUmakefile
--- orig/GNUmakefile	2019-04-29 01:37:23.000000000 +0200
+++ mod/GNUmakefile	2019-10-03 16:24:23.677406740 +0200
@@ -27,7 +27,7 @@
 MKDIR ?= mkdir -p
 
 LN ?= ln -sf
-LDCONF ?= /sbin/ldconfig -n
+LDCONF ?= /usr/sbin/ldconfig -n
 
 # Solaris provides a non-Posix grep at /usr/bin
 ifneq ($(wildcard /usr/xpg4/bin/grep),)
@@ -1256,7 +1256,7 @@
 	$(CHMOD) 0755 $(DESTDIR)$(LIBDIR)/libcryptopp.so$(SOLIB_VERSION_SUFFIX)
 ifeq ($(HAS_SOLIB_VERSION),1)
 	-$(LN) libcryptopp.so$(SOLIB_VERSION_SUFFIX) $(DESTDIR)$(LIBDIR)/libcryptopp.so
-	$(LDCONF) $(DESTDIR)$(LIBDIR)
+	$(LDCONF) -C $(DESTDIR)/etc/ld.so.cache $(DESTDIR)$(LIBDIR)
 endif
 endif
 ifneq ($(wildcard libcryptopp.pc),)
