diff -Naur php-5.6.18-orig/php.ini-production php-5.6.18/php.ini-production
--- php-5.6.18-orig/php.ini-production	2016-02-03 09:38:08.000000000 +0100
+++ php-5.6.18/php.ini-production	2016-02-25 19:50:53.253359804 +0100
@@ -199,7 +199,7 @@
 ; Development Value: Off
 ; Production Value: Off
 ; http://php.net/short-open-tag
-short_open_tag = Off
+; short_open_tag = Off
 
 ; Allow ASP-style <% %> tags.
 ; http://php.net/asp-tags
@@ -730,7 +730,7 @@
 
 ; Directory in which the loadable extensions (modules) reside.
 ; http://php.net/extension-dir
-; extension_dir = "./"
+extension_dir = "./"
 ; On windows:
 ; extension_dir = "ext"
 
