diff -Naur orig/php.ini-production patched/php.ini-production
--- orig/php.ini-production	2018-12-04 17:12:20.000000000 +0100
+++ patched/php.ini-production	2019-01-03 19:57:49.935637338 +0100
@@ -194,7 +194,7 @@
 ; Development Value: Off
 ; Production Value: Off
 ; http://php.net/short-open-tag
-short_open_tag = Off
+; short_open_tag = Off
 
 ; The number of significant digits displayed in floating point numbers.
 ; http://php.net/precision
@@ -752,7 +752,7 @@
 
 ; Directory in which the loadable extensions (modules) reside.
 ; http://php.net/extension-dir
-;extension_dir = "./"
+extension_dir = "./"
 ; On windows:
 ;extension_dir = "ext"
 
