diff -Naur orig/src/gui/datewidget.cpp patched/src/gui/datewidget.cpp
--- orig/src/gui/datewidget.cpp	2019-07-10 02:59:41.000000000 +0200
+++ patched/src/gui/datewidget.cpp	2019-10-06 01:11:50.952290727 +0200
@@ -35,6 +35,7 @@
 #include <QPushButton>
 #include <QHBoxLayout>
 #include <QFrame>
+#include <QDate>
 #include <QEvent>
 #include <QMenu>
 #include <QWidgetAction>
diff -Naur orig/src/mainwindow.cpp patched/src/mainwindow.cpp
--- orig/src/mainwindow.cpp	2019-07-10 02:59:41.000000000 +0200
+++ patched/src/mainwindow.cpp	2019-10-06 01:12:31.588878784 +0200
@@ -108,6 +108,7 @@
 #include <QMimeType>
 #include <QMenuBar>
 #include <QFileDialog>
+#include <QMetaMethod>
 
 #include <unistd.h>
 
