diff -Naur orig/gcc/c-family/c.opt patched/gcc/c-family/c.opt
--- orig/gcc/c-family/c.opt	2018-02-27 23:28:21.000000000 +0100
+++ patched/gcc/c-family/c.opt	2018-08-12 02:18:42.660145673 +0200
@@ -1350,7 +1350,7 @@
 ; documented in common.opt
 
 Wunused-local-typedefs
-C ObjC C++ ObjC++ Var(warn_unused_local_typedefs) Warning EnabledBy(Wunused)
+C ObjC C++ ObjC++ Var(warn_unused_local_typedefs) Warning
 Warn when typedefs locally defined in a function are not used.
 
 Wunused-macros
