diff -Naur orig/gcc/c-family/c.opt patched/gcc/c-family/c.opt
--- orig/gcc/c-family/c.opt	2016-06-14 16:44:28.000000000 +0200
+++ patched/gcc/c-family/c.opt	2016-10-31 20:40:07.802336330 +0100
@@ -937,7 +937,7 @@
 ; documented in common.opt
 
 Wunused-local-typedefs
-C ObjC C++ ObjC++ Var(warn_unused_local_typedefs) Warning EnabledBy(Wunused)
+C ObjC C++ ObjC++ Var(warn_unused_local_typedefs) Warning
 Warn when typedefs locally defined in a function are not used.
 
 Wunused-macros
