diff -Nur orig/Functions/Calendar/calendar_add mod/Functions/Calendar/calendar_add
--- orig/Functions/Calendar/calendar_add	2017-12-04 15:09:35.000000000 +0100
+++ mod/Functions/Calendar/calendar_add	2019-10-15 16:18:16.337040897 +0200
@@ -1,4 +1,4 @@
-#!/bin/zsh
+#!/usr/bin/zsh
 # All arguments are joined with spaces and inserted into the calendar
 # file at the appropriate point.
 #
diff -Nur orig/Functions/Example/cat mod/Functions/Example/cat
--- orig/Functions/Example/cat	2017-12-04 15:09:35.000000000 +0100
+++ mod/Functions/Example/cat	2019-10-15 16:18:35.277020872 +0200
@@ -1,4 +1,4 @@
-#!/bin/zsh -f
+#!/usr/bin/zsh -f
 
 local file
 
diff -Nur orig/Functions/Example/zless mod/Functions/Example/zless
--- orig/Functions/Example/zless	2017-12-04 15:09:35.000000000 +0100
+++ mod/Functions/Example/zless	2019-10-15 16:18:45.677009892 +0200
@@ -1,4 +1,4 @@
-#!/bin/zsh -f
+#!/usr/bin/zsh -f
 #
 # zsh function script to run less on various inputs, decompressing as required.
 # Author: Phil Pennock.  zsh-hacks@athenaeum.demon.co.uk
diff -Nur orig/Functions/Misc/checkmail mod/Functions/Misc/checkmail
--- orig/Functions/Misc/checkmail	2017-12-04 15:09:35.000000000 +0100
+++ mod/Functions/Misc/checkmail	2019-10-15 16:18:57.504997422 +0200
@@ -1,4 +1,4 @@
-#!/bin/zsh
+#!/usr/bin/zsh
 #
 # This autoloadable function checks the folders specified as arguments
 # for new mails.  The arguments are interpreted in exactly the same way
diff -Nur orig/Functions/Misc/run-help mod/Functions/Misc/run-help
--- orig/Functions/Misc/run-help	2017-12-04 15:09:35.000000000 +0100
+++ mod/Functions/Misc/run-help	2019-10-15 16:19:08.224986134 +0200
@@ -1,4 +1,4 @@
-#!/bin/zsh
+#!/usr/bin/zsh
 #
 # Figure out where to get the best help, and get it.
 #
diff -Nur orig/Functions/Misc/run-help-ip mod/Functions/Misc/run-help-ip
--- orig/Functions/Misc/run-help-ip	2017-12-04 15:09:36.000000000 +0100
+++ mod/Functions/Misc/run-help-ip	2019-10-15 16:19:17.084976815 +0200
@@ -1,4 +1,4 @@
-#!/bin/zsh -f
+#!/usr/bin/zsh -f
 #
 # Install this function by placing it in your FPATH and then
 # adding to your .zshrc the line if you use run-help function:
diff -Nur orig/Functions/Misc/sticky-note mod/Functions/Misc/sticky-note
--- orig/Functions/Misc/sticky-note	2017-12-04 15:09:36.000000000 +0100
+++ mod/Functions/Misc/sticky-note	2019-10-15 16:19:27.004966388 +0200
@@ -1,4 +1,4 @@
-#!/bin/zsh -fi
+#!/usr/bin/zsh -fi
 # A zsh sticky-note ("post-it") application.  Load this file as a function:
 #    autoload -Uz sticky-note
 #
diff -Nur orig/Functions/Misc/zcalc mod/Functions/Misc/zcalc
--- orig/Functions/Misc/zcalc	2017-12-04 15:09:36.000000000 +0100
+++ mod/Functions/Misc/zcalc	2019-10-15 16:19:36.524956393 +0200
@@ -1,4 +1,4 @@
-#!/bin/zsh -i
+#!/usr/bin/zsh -i
 #
 # Zsh calculator.  Understands most ordinary arithmetic expressions.
 # Line editing and history are available. A blank line or `q' quits.
diff -Nur orig/Functions/Misc/zed mod/Functions/Misc/zed
--- orig/Functions/Misc/zed	2018-11-13 09:04:43.000000000 +0100
+++ mod/Functions/Misc/zed	2019-10-15 16:19:45.272947219 +0200
@@ -1,4 +1,4 @@
-#!/bin/zsh
+#!/usr/bin/zsh
 # zed
 #
 # No other shell could do this.
diff -Nur orig/Functions/Misc/zkbd mod/Functions/Misc/zkbd
--- orig/Functions/Misc/zkbd	2017-12-04 15:09:36.000000000 +0100
+++ mod/Functions/Misc/zkbd	2019-10-15 16:19:54.096937969 +0200
@@ -1,4 +1,4 @@
-#!/bin/zsh -f
+#!/usr/bin/zsh -f
 
 [[ -o interactive ]] && {
     local -hi ARGC		# local is a no-op outside of a function
diff -Nur orig/Misc/globtests mod/Misc/globtests
--- orig/Misc/globtests	2017-12-04 15:09:36.000000000 +0100
+++ mod/Misc/globtests	2019-10-15 16:21:03.548865429 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/zsh -f
+#!/usr/bin/zsh -f
 
 setopt extendedglob badpattern
 unsetopt kshglob
diff -Nur orig/Misc/globtests.ksh mod/Misc/globtests.ksh
--- orig/Misc/globtests.ksh	2017-12-04 15:09:36.000000000 +0100
+++ mod/Misc/globtests.ksh	2019-10-15 16:21:15.500852993 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/zsh -f
+#!/usr/bin/zsh -f
 
 setopt kshglob extendedglob
 
diff -Nur orig/Test/runtests.zsh mod/Test/runtests.zsh
--- orig/Test/runtests.zsh	2017-12-04 15:09:36.000000000 +0100
+++ mod/Test/runtests.zsh	2019-10-15 16:21:28.672835123 +0200
@@ -1,4 +1,4 @@
-#!/bin/zsh -f
+#!/usr/bin/zsh -f
 
 emulate zsh
 
diff -Nur orig/Test/ztst.zsh mod/Test/ztst.zsh
--- orig/Test/ztst.zsh	2018-12-21 07:02:49.000000000 +0100
+++ mod/Test/ztst.zsh	2019-10-15 16:21:37.856821216 +0200
@@ -1,4 +1,4 @@
-#!/bin/zsh -f
+#!/usr/bin/zsh -f
 # The line above is just for convenience.  Normally tests will be run using
 # a specified version of zsh.  With dynamic loading, any required libraries
 # must already have been installed in that case.
diff -Nur orig/Util/check-tmux-state mod/Util/check-tmux-state
--- orig/Util/check-tmux-state	2017-12-04 15:09:36.000000000 +0100
+++ mod/Util/check-tmux-state	2019-10-15 16:21:52.276799464 +0200
@@ -1,4 +1,4 @@
-#!/bin/zsh -f
+#!/usr/bin/zsh -f
 
 # Tmux has lots of options and sub-commands. It's very tedious to manually
 # check if the actual command's idea of all this matches the completion
diff -Nur orig/Util/reporter mod/Util/reporter
--- orig/Util/reporter	2017-12-04 15:09:36.000000000 +0100
+++ mod/Util/reporter	2019-10-15 16:22:07.700776322 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/zsh
+#!/usr/bin/zsh
 #
 # NAME:
 #	reporter
