diff -Naur orig/Rodent/index.theme.in patched/Rodent/index.theme.in
--- orig/Rodent/index.theme.in	2008-10-26 23:05:31.000000000 +0100
+++ patched/Rodent/index.theme.in	2015-03-03 16:37:31.879208244 +0100
@@ -5,7 +5,7 @@
 [Icon Theme]
 _Name=Rodent
 _Comment=Default icon for the Xfce Desktop Environment
-Inherits=Tango,gnome,Industrial
+Inherits=gnome
 Directories=scalable/apps,scalable/devices,scalable/emblems,scalable/filesystems,scalable/mimetypes,scalable/misc,scalable/stock,48x48/apps,48x48/devices,48x48/emblems,48x48/filesystems,48x48/mimetypes,48x48/misc,48x48/stock
 
 [scalable/apps]
