diff -Naur orig/lbreakout2.desktop.in mod/lbreakout2.desktop.in
--- orig/lbreakout2.desktop.in	2013-05-03 19:06:20.000000000 +0200
+++ mod/lbreakout2.desktop.in	2015-11-13 21:53:05.275965513 +0100
@@ -1,7 +1,8 @@
 [Desktop Entry]
 Name=LBreakout2
 Comment=A ball-and-paddle game with nice graphics.
+Comment[hu]=Falkilövő játék
 Exec=lbreakout2
 Icon=
 Type=Application
-Categories=Game;
+Categories=Game;ArcadeGame;
