diff -Naur a/help/Makefile.am b/help/Makefile.am
--- a/help/Makefile.am	2021-08-03 22:59:29.000000000 +0200
+++ b/help/Makefile.am	2022-04-11 03:58:12.047952065 +0200
@@ -9,7 +9,7 @@
 
 if USE_NLS
 # Add linguas to be ignored, e.g. IGNORE_HELP_LINGUAS = ca de es fr
-IGNORE_HELP_LINGUAS =
+IGNORE_HELP_LINGUAS = ja
 HELP_LINGUAS = $(if $(IGNORE_HELP_LINGUAS), \
 	$(filter-out $(IGNORE_HELP_LINGUAS),$(subst /,,$(dir $(wildcard */*.po)))), \
 	$(subst /,,$(dir $(wildcard */*.po))) )
diff -Naur a/help/Makefile.in b/help/Makefile.in
--- a/help/Makefile.in	2021-08-03 23:01:37.000000000 +0200
+++ b/help/Makefile.in	2022-04-11 03:58:28.923996906 +0200
@@ -343,7 +343,7 @@
 
 
 # Add linguas to be ignored, e.g. IGNORE_HELP_LINGUAS = ca de es fr
-@USE_NLS_TRUE@IGNORE_HELP_LINGUAS = 
+@USE_NLS_TRUE@IGNORE_HELP_LINGUAS = ja
 @USE_NLS_FALSE@HELP_LINGUAS = 
 @USE_NLS_TRUE@HELP_LINGUAS = $(if $(IGNORE_HELP_LINGUAS), \
 @USE_NLS_TRUE@	$(filter-out $(IGNORE_HELP_LINGUAS),$(subst /,,$(dir $(wildcard */*.po)))), \
