diff -up CriticalMass-1.5/game/Video.cpp CriticalMass-1.5/game/Video.cpp
--- CriticalMass-1.5/game/Video.cpp	2010-10-17 03:09:48.000000000 +0200
+++ CriticalMass-1.5/game/Video.cpp	2011-12-10 11:54:31.566090135 +0100
@@ -290,10 +290,10 @@ bool Video::setVideoMode( void)
 bool Video::updateSettings( void)
 {
     bool fullscreen = _isFullscreen;
-    ConfigS::instance()->getBoolean( "fullscreen", _isFullscreen);
-    int width = 0;
+    ConfigS::instance()->getBoolean( "fullscreen", fullscreen);
+    int width = _width;
     ConfigS::instance()->getInteger( "width", width);
-    int height = 0;
+    int height = _height;
     ConfigS::instance()->getInteger( "height", height);
 
     if( (fullscreen != _isFullscreen) || (width != _width) || (height != _height))
