diff -up CriticalMass-1.5/utilssdl/PNG.cpp CriticalMass-1.5/utilssdl/PNG.cpp
--- CriticalMass-1.5/utilssdl/PNG.cpp	2010-10-17 03:09:50.000000000 +0200
+++ CriticalMass-1.5/utilssdl/PNG.cpp	2011-12-10 17:14:27.904931754 +0100
@@ -56,7 +56,7 @@ void PNG::writeData( png_structp png, pn
 {
     png_size_t check;
 
-    check = fwrite( data, 1, length, (FILE *)(png->io_ptr));
+    check = fwrite( data, 1, length, (FILE *)png_get_io_ptr(png));
     if( check != length)
     {
 	png_error( png, "Write Error");
@@ -83,7 +83,7 @@ bool PNG::init( FILE *fp, int width, int
 	return false;
     }
 
-    if( setjmp(_png->jmpbuf))
+    if( setjmp(png_jmpbuf(_png)))
     {
 	fclose( fp);
 	png_destroy_write_struct(&_png, (png_infopp)NULL);
diff -up CriticalMass-1.5/game/main.cpp CriticalMass-1.5/game/main.cpp
--- CriticalMass-1.5/game/main.cpp	2010-10-17 03:09:48.000000000 +0200
+++ CriticalMass-1.5/game/main.cpp	2011-12-10 17:19:52.062927383 +0100
@@ -27,6 +27,7 @@
 
 #include <sys/stat.h>
 #include <sys/types.h>
+#include <zlib.h>
 
 using namespace std;
 
