diff -up CriticalMass-1.5/game/main.cpp CriticalMass-1.5/game/main.cpp
--- CriticalMass-1.5/game/main.cpp	2012-01-12 22:17:44.000000000 +0100
+++ CriticalMass-1.5/game/main.cpp	2012-01-12 22:21:55.377006929 +0100
@@ -141,7 +141,7 @@ void showInfo()
 {
     LOG_INFO << "----------------------------------" << endl;
     LOG_INFO << GAMETITLE << " " << GAMEVERSION 
-             << " - "__TIME__" "__DATE__
+             << " - " __TIME__ " " __DATE__
              << endl;
     LOG_INFO << "Copyright (C) 2001-2010 by Frank Becker" << endl;
     LOG_INFO << "Visit http://criticalmass.sourceforge.net" << endl;
