--- CriticalMass-1.5/utils/Point.hpp	2010-10-17 03:09:49.000000000 +0200
+++ CriticalMass-1.5/utils/Point.hpp	2016-02-02 17:09:31.260885940 +0100
@@ -22,6 +22,7 @@ inline bool feq( const float &f1, const
     return fabs(f1-f2) < EPSILON;
 }
 
+#if 0
 inline float fmin( const float &f1, const float &f2 )
 {
     return f1 < f2 ? f1 : f2;
@@ -31,6 +32,8 @@ inline float fmax( const float &f1, cons
 {
     return f1 > f2 ? f1 : f2;
 }
+#endif
+
 //-----------------------------------------------------------------------
 
 struct Point2D
