diff -Nur orig/configure.in mod/configure.in
--- orig/configure.in	2010-10-17 03:09:54.000000000 +0200
+++ mod/configure.in	2016-11-21 12:32:22.871357866 +0100
@@ -99,7 +99,7 @@
 else
     LIBS="-L/usr/local/lib -L/usr/X11R6/lib $LIBS"
     CXXFLAGS="$CXXFLAGS -I/usr/X11R6/include"
-	AC_CHECK_LIB(curl-gnutls, main,, AC_MSG_ERROR(libcurl-gnutls is needed))
+	AC_CHECK_LIB(curl, main,, AC_MSG_ERROR(libcurl is needed))
 	AC_CHECK_LIB(tinyxml, main,, AC_MSG_ERROR(libtinyxml is needed))
 	AC_CHECK_LIB(png, main,, AC_MSG_ERROR(libpng is needed))
 	AC_CHECK_LIB(z, main,, AC_MSG_ERROR(libz is needed))
