diff -Naur orig/utilsgl/GLee.h patched/utilsgl/GLee.h
--- orig/utilsgl/GLee.h	2010-10-17 03:09:51.000000000 +0200
+++ patched/utilsgl/GLee.h	2019-01-19 03:12:38.946587157 +0100
@@ -58,6 +58,10 @@
 	#error glxext.h included before glee.h
 #endif
 
+#ifdef __gl_glxext_h_
+	#error gl_glxext.h included before glee.h
+#endif
+
 #ifdef WIN32
 	#define WIN32_LEAN_AND_MEAN
 	#include <windows.h>
@@ -68,6 +72,9 @@
 #else // GLX
 	#define __glext_h_  /* prevent glext.h from being included  */
 	#define __glxext_h_ /* prevent glxext.h from being included */
+	#define __gl_glext_h_
+	#define GL_GLEXT_LEGACY
+	#define GLX_GLXEXT_LEGACY
 	#define GLX_GLXEXT_PROTOTYPES
 	#include <GL/gl.h>
 	#include <GL/glx.h>
