diff -Naur orig/files/usr/share/cinnamon/cinnamon-settings/modules/cs_backgrounds.py mod/files/usr/share/cinnamon/cinnamon-settings/modules/cs_backgrounds.py
--- orig/files/usr/share/cinnamon/cinnamon-settings/modules/cs_backgrounds.py	2015-10-20 12:16:42.000000000 +0200
+++ mod/files/usr/share/cinnamon/cinnamon-settings/modules/cs_backgrounds.py	2015-10-20 22:26:58.822602438 +0200
@@ -315,7 +315,7 @@
     def get_system_backgrounds(self):
         picture_list = []
         folder_list = []
-        properties_dir = "/usr/share/cinnamon-background-properties"
+        properties_dir = "/usr/share/gnome-background-properties"
         backgrounds = []
         if os.path.exists(properties_dir):
             for i in os.listdir(properties_dir):
