diff -Nur orig/files/cinnamon2d.session.in mod/files/cinnamon2d.session.in
--- orig/files/cinnamon2d.session.in	2019-11-28 21:08:10.000000000 +0100
+++ mod/files/cinnamon2d.session.in	2019-11-30 10:37:27.853639545 +0100
@@ -1,6 +1,6 @@
 [Cinnamon Session]
 Name=Cinnamon (Software Rendering)
-RequiredComponents=cinnamon2d;cinnamon-screensaver;nemo-autostart;nm-applet;cinnamon-killer-daemon;
+RequiredComponents=cinnamon2d;cinnamon-screensaver;nemo-autostart;nm-applet;cinnamon-killer-daemon;polkit-cinnamon-authentication-agent-1;
 DesktopName=X-Cinnamon
 
 
diff -Nur orig/files/cinnamon.session.in mod/files/cinnamon.session.in
--- orig/files/cinnamon.session.in	2019-11-28 21:08:10.000000000 +0100
+++ mod/files/cinnamon.session.in	2019-11-30 10:37:14.677668031 +0100
@@ -1,6 +1,6 @@
 [Cinnamon Session]
 Name=Cinnamon
-RequiredComponents=cinnamon;cinnamon-screensaver;nemo-autostart;nm-applet;cinnamon-killer-daemon;
+RequiredComponents=cinnamon;cinnamon-screensaver;nemo-autostart;nm-applet;cinnamon-killer-daemon;polkit-cinnamon-authentication-agent-1;
 DesktopName=X-Cinnamon
 
 
