diff -Naur a/cinnamon2d.session.in b/cinnamon2d.session.in
--- a/cinnamon2d.session.in	2020-11-13 17:07:15.000000000 +0100
+++ b/cinnamon2d.session.in	2020-11-14 23:28:08.054926066 +0100
@@ -1,6 +1,6 @@
 [Cinnamon Session]
 Name=Cinnamon (Software Rendering)
-RequiredComponents=cinnamon2d;cinnamon-screensaver;nemo-autostart;@REQUIRED@cinnamon-killer-daemon;
+RequiredComponents=cinnamon2d;cinnamon-screensaver;nemo-autostart;@REQUIRED@cinnamon-killer-daemon;polkit-cinnamon-authentication-agent-1;
 DesktopName=X-Cinnamon
 
 
diff -Naur a/cinnamon.session.in b/cinnamon.session.in
--- a/cinnamon.session.in	2020-11-13 17:07:15.000000000 +0100
+++ b/cinnamon.session.in	2020-11-14 23:28:21.238954737 +0100
@@ -1,6 +1,6 @@
 [Cinnamon Session]
 Name=Cinnamon
-RequiredComponents=cinnamon;cinnamon-screensaver;nemo-autostart;@REQUIRED@cinnamon-killer-daemon;
+RequiredComponents=cinnamon;cinnamon-screensaver;nemo-autostart;@REQUIRED@cinnamon-killer-daemon;polkit-cinnamon-authentication-agent-1;
 DesktopName=X-Cinnamon
 
 
