diff -Naur a/files/usr/bin/cinnamon-settings b/files/usr/bin/cinnamon-settings
--- a/files/usr/bin/cinnamon-settings	2020-11-13 17:07:15.000000000 +0100
+++ b/files/usr/bin/cinnamon-settings	2020-11-14 23:30:39.019259561 +0100
@@ -120,7 +120,7 @@
             os.execvp("/usr/share/cinnamon/cinnamon-settings/cinnamon-settings.py", (" ",) + tuple(sys.argv[1:]))
     elif os.path.exists("/usr/share/cinnamon/cinnamon-settings/modules/cs_%s.py" % module):
         os.execvp("/usr/share/cinnamon/cinnamon-settings/cinnamon-settings.py", (" ",) + tuple(sys.argv[1:]))
-    elif os.path.exists("/usr/bin/cinnamon-control-center"):
+    elif os.path.exists("/usr/bin/cinnamon-control-center-1/panels"):
         os.execvp("/usr/share/cinnamon/cinnamon-settings/cinnamon-settings.py", (" ",) + tuple(sys.argv[1:]))
     elif os.path.exists("/usr/bin/gnome-control-center"):
         print ("Unknown module %s, calling gnome-control-center" % module)
