diff -Nur orig/data/org.cinnamon.gschema.xml.in mod/data/org.cinnamon.gschema.xml.in
--- orig/data/org.cinnamon.gschema.xml.in	2018-05-19 12:39:14.000000000 +0200
+++ mod/data/org.cinnamon.gschema.xml.in	2018-05-19 12:42:24.367907920 +0200
@@ -412,7 +412,7 @@
     </key>
 
     <key name="favorite-apps" type="as">
-      <default>[ 'firefox.desktop', 'mintinstall.desktop', 'cinnamon-settings.desktop', 'hexchat.desktop', 'org.gnome.Terminal.desktop', 'nemo.desktop' ]</default>
+      <default>[ 'firefox.desktop', 'synaptic.desktop', 'cinnamon-settings.desktop', 'hexchat.desktop', 'org.gnome.Terminal.desktop', 'nemo.desktop' ]</default>
       <_summary>List of desktop file IDs for favorite applications</_summary>
       <_description>
         The applications corresponding to these identifiers
