diff -Nur orig/files/usr/share/cinnamon/cinnamon-settings/bin/Spices.py mod/files/usr/share/cinnamon/cinnamon-settings/bin/Spices.py
--- orig/files/usr/share/cinnamon/cinnamon-settings/bin/Spices.py	2017-11-23 10:56:12.000000000 +0100
+++ mod/files/usr/share/cinnamon/cinnamon-settings/bin/Spices.py	2017-12-08 18:33:51.794523353 +0100
@@ -14,6 +14,8 @@
     import time
     from PIL import Image
     import config
+    import ssl
+    ssl._create_default_https_context = ssl._create_unverified_context
 except Exception, detail:
     print detail
     sys.exit(1)
@@ -27,7 +29,7 @@
 locale_inst = '%s/.local/share/locale' % home
 settings_dir = '%s/.cinnamon/configs/' % home
 
-URL_SPICES_HOME = "http://cinnamon-spices.linuxmint.com"
+URL_SPICES_HOME = "https://cinnamon-spices.linuxmint.com"
 URL_MAP = {
     'applet': URL_SPICES_HOME + "/json/applets.json",
     'theme': URL_SPICES_HOME + "/json/themes.json",
