--- a/files/usr/share/cinnamon/cinnamon-settings-users/cinnamon-settings-users.py
+++ b/files/usr/share/cinnamon/cinnamon-settings-users/cinnamon-settings-users.py
@@ -47,10 +47,9 @@
             print detail
 
     def _on_entry_changed(self, entry):
-        name = entry.get_text()
-        if " " in name or name.lower() != name:
+        if " " in entry.get_text():
             entry.set_icon_from_stock(Gtk.EntryIconPosition.SECONDARY, Gtk.STOCK_DIALOG_WARNING)
-            entry.set_icon_tooltip_text(Gtk.EntryIconPosition.SECONDARY, _("The group name cannot contain upper-case or space characters"))
+            entry.set_icon_tooltip_text(Gtk.EntryIconPosition.SECONDARY, _("The group name cannot contain space characters"))
             self.set_response_sensitive(Gtk.ResponseType.OK, False)
         else:
             entry.set_icon_from_stock(Gtk.EntryIconPosition.SECONDARY, None)
@@ -361,9 +360,9 @@
         fullname = self.realname_entry.get_text()
         username = self.username_entry.get_text()
         valid = True
-        if re.search('[^a-z0-9_.-]', username):
+        if " " in username:
             self.username_entry.set_icon_from_stock(Gtk.EntryIconPosition.SECONDARY, Gtk.STOCK_DIALOG_WARNING)
-            self.username_entry.set_icon_tooltip_text(Gtk.EntryIconPosition.SECONDARY, _("Invalid username"))
+            self.username_entry.set_icon_tooltip_text(Gtk.EntryIconPosition.SECONDARY, _("The username cannot contain space characters"))
             valid = False
         else:
             self.username_entry.set_icon_from_stock(Gtk.EntryIconPosition.SECONDARY, None)
