diff -Naur ntrack-009-orig//qt4/libntrack-qt4.pc.in ntrack-009//qt4/libntrack-qt4.pc.in
--- ntrack-009-orig//qt4/libntrack-qt4.pc.in	2010-10-10 19:42:02.000000000 +0200
+++ ntrack-009//qt4/libntrack-qt4.pc.in	2010-11-03 17:21:51.000000000 +0100
@@ -10,5 +10,5 @@
 Version: @VERSION@
 Requires: libntrack >= 003
 CFlags: -I${qt4pkgincludedir}
-Libs: -L${libdir} -lntrack-qt
+Libs: -L${libdir} -lntrack-qt4
 
