From: Chris Howie <crazycomputers@yahoo.com>
Date: Sat, 14 Apr 2007 17:15:00 +0200
Subject: Following codabar specification (Closes: #320908).

---
 codabar.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/codabar.c b/codabar.c
index 5817f65..9546342 100644
--- a/codabar.c
+++ b/codabar.c
@@ -28,10 +28,11 @@
 
 /* this is ordered in decades to simplify encoding */
 static char alphabet[] = 
-   "0123456789" "-$:/.+ABCD";
+   "0123456789" "-$:/.+ABCDTN*E";
 
 #define CODE_A	16
 #define CODE_B	17
+#define CODE_T	20
 
 #define NARROW	12
 #define WIDE	14
@@ -41,7 +42,8 @@ static char *patterns[] = {
 "1111133","1111331","1113113","3311111","1131131",
 "3111131","1311113","1311311","1331111","3113111",
 "1113311","1133111","3111313","3131113","3131311",
-"1131313","1133131","1313113","1113133","1113331" };
+"1131313","1133131","1313113","1113133","1113331",
+"1133131","1313113","1113133","1113331"};
 
 /*
  * Check that the text can be encoded. Returns 0 or -1.
@@ -166,13 +168,13 @@ int Barcode_cbr_encode(struct Barcode_Item *bc)
     }
     if (!startpresent) {
 	if (usesum) {
-	    /* if no start character specified, B is used as a stop char */
-	    checksum += CODE_B;
+	    /* if no start character specified, T is used as a stop char */
+	    checksum += CODE_T;
 	    checksum = (checksum + 15) / 16 * 16 - checksum;
 	    add_one(ptr, checksum);
 	    ptr += strlen(ptr);
 	}
-	add_one(ptr, CODE_B);
+	add_one(ptr, CODE_T);
     }
     bc->partial = partial;
     bc->textinfo = textinfo;
