diff -Nur orig/Makefile mod/Makefile
--- orig/Makefile	2018-08-30 11:05:19.000000000 +0200
+++ mod/Makefile	2018-10-31 15:04:15.042311508 +0100
@@ -132,7 +132,7 @@
 windows_PREFIX:=/usr/local
 osx_PREFIX:=Resources
 beos_PREFIX=$(shell finddir B_APPS_DIRECTORY)/TuxPaint
-linux_PREFIX:=/usr/local
+linux_PREFIX:=/usr
 PREFIX:=$($(OS)_PREFIX)
 
 # Root directory to place files when creating packages.
@@ -786,7 +786,6 @@
 	  cp src/tuxpaint.desktop $(DESTDIR)$(KDE_PREFIX)/; \
 	  chmod 644 $(DESTDIR)$(KDE_PREFIX)/tuxpaint.desktop; \
 	fi
-	kbuildsycoca4
 
 .PHONY: install-kde-icons
 install-kde-icons:
