diff -Nur orig/plugins/telepathy-gabble-xmpp-console mod/plugins/telepathy-gabble-xmpp-console
--- orig/plugins/telepathy-gabble-xmpp-console	2016-11-15 10:03:24.000000000 +0100
+++ mod/plugins/telepathy-gabble-xmpp-console	2019-12-04 20:58:35.293677063 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 # vim: set fileencoding=utf-8 sts=4 sw=4 et :
 """
 The world's worst XMPP console user interface.
