diff -Nur orig/misc/grub-customizer.desktop mod/misc/grub-customizer.desktop
--- orig/misc/grub-customizer.desktop	2018-10-13 14:07:50.000000000 +0200
+++ mod/misc/grub-customizer.desktop	2021-11-14 11:12:48.300896031 +0100
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Name=Grub Customizer
 Comment=Customize the bootloader (GRUB2 or BURG)
-Exec=grub-customizer
+Exec=pkexec /usr/bin/grub-customizer
 Terminal=false
 X-MultipleArgs=false
 Type=Application
