diff -urN pcmciautils-018.orig/Makefile pcmciautils-018/Makefile
--- pcmciautils-018.orig/Makefile	2011-04-15 09:22:12.000000000 +0200
+++ pcmciautils-018/Makefile	2014-06-11 09:45:49.550177441 +0200
@@ -74,7 +74,7 @@
 etcdir =	${prefix}/etc
 sbindir =	${exec_prefix}/sbin
 mandir =	${prefix}/usr/share/man
-udevdir =	${prefix}/lib/udev
+udevdir =	${prefix}/usr/lib/udev
 
 INSTALL = /usr/bin/install -c
 INSTALL_PROGRAM = ${INSTALL}
