diff -Naur orig/rules.d/60-persistent-storage.rules patched/rules.d/60-persistent-storage.rules
--- orig/rules.d/60-persistent-storage.rules	2019-04-11 18:28:36.000000000 +0200
+++ patched/rules.d/60-persistent-storage.rules	2019-04-24 00:36:55.632919608 +0200
@@ -112,6 +112,7 @@
 # by-label/by-uuid links (filesystem metadata)
 ENV{ID_FS_USAGE}=="filesystem|other|crypto", ENV{ID_FS_UUID_ENC}=="?*", SYMLINK+="disk/by-uuid/$env{ID_FS_UUID_ENC}"
 ENV{ID_FS_USAGE}=="filesystem|other|crypto", ENV{ID_FS_LABEL_ENC}=="?*", SYMLINK+="disk/by-label/$env{ID_FS_LABEL_ENC}"
+ENV{ID_FS_USAGE}=="filesystem|other|raid|crypto", ENV{ID_FS_TYPE}=="?*", SYMLINK+="disk/by-type/$env{ID_FS_TYPE}/%k"
 
 # by-id (World Wide Name)
 ENV{DEVTYPE}=="disk", ENV{ID_WWN_WITH_EXTENSION}=="?*", SYMLINK+="disk/by-id/wwn-$env{ID_WWN_WITH_EXTENSION}"
