diff -Naur orig/src/journal/journald.conf patched/src/journal/journald.conf
--- orig/src/journal/journald.conf	2019-04-11 18:28:36.000000000 +0200
+++ patched/src/journal/journald.conf	2019-04-24 00:51:41.714268416 +0200
@@ -22,13 +22,13 @@
 #SyncIntervalSec=5m
 #RateLimitIntervalSec=30s
 #RateLimitBurst=10000
-#SystemMaxUse=
+SystemMaxUse=50M
 #SystemKeepFree=
-#SystemMaxFileSize=
+SystemMaxFileSize=50M
 #SystemMaxFiles=100
-#RuntimeMaxUse=
+RuntimeMaxUse=50M
 #RuntimeKeepFree=
-#RuntimeMaxFileSize=
+RuntimeMaxFileSize=50M
 #RuntimeMaxFiles=100
 #MaxRetentionSec=
 #MaxFileSec=1month
