diff -Naur orig/interfaces/Makefile patched/interfaces/Makefile
--- orig/interfaces/Makefile	2017-04-10 02:32:59.000000000 +0200
+++ patched/interfaces/Makefile	2019-08-28 03:49:25.064030344 +0200
@@ -4,8 +4,8 @@
 	gcc -Wall -O2 -o interfaces interfaces.c lib/*.c -I. -I./lib
 
 install:
-	mkdir -p $(DESTDIR)/sbin
-	install -m 755 interfaces $(DESTDIR)/sbin/
+	mkdir -p $(DESTDIR)/usr/sbin
+	install -m 755 interfaces $(DESTDIR)/usr/sbin/
 
 clean:
 	rm -f interfaces
