diff -urN latex2html-2002-2-1.orig/config/install.pl latex2html-2002-2-1/config/install.pl
--- latex2html-2002-2-1.orig/config/install.pl	2002-04-28 07:15:01.000000000 +0200
+++ latex2html-2002-2-1/config/install.pl	2003-02-13 15:51:49.000000000 +0100
@@ -372,14 +372,15 @@
 
 if($cfg{TEXPATH}) {
   print "\nNote: trying to install LaTeX2HTML style files in TeX directory tree\n     ($cfg{TEXPATH})\n";
-  unless(mkpath($cfg{TEXPATH})) {
+  my $dest = $ENV{'DESTDIR'} if($ENV{'DESTDIR'});
+  unless(mkpath($dest.$cfg{TEXPATH})) {
   #my $testpath = $cfg{TEXPATH}; # to strip (latex2)html
   #$testpath =~ s/[$dd$dd][^$dd$dd]*$//;
   #if((-d $cfg{TEXPATH} && !-w _) || (-d $testpath && !-w _)) {
     print STDERR "\nError: Cannot install LaTeX2HTML style files in $cfg{TEXPATH}\n";
   } else {
     my $dir = 'texinputs';
-    my $dest = $cfg{TEXPATH};
+    $dest = $dest.$cfg{TEXPATH};
     unless(opendir(DIR,$dir)) {
       print STDERR qq{Error: Could not read directory "$dir": $!\n};
     } else {
