diff -Naur orig/Makefile.in patched/Makefile.in
--- orig/Makefile.in	2018-05-09 21:16:01.000000000 +0200
+++ patched/Makefile.in	2019-08-27 22:49:19.187808928 +0200
@@ -112,17 +112,17 @@
 		@for lib in lib${PACKAGE}.a lib${PACKAGE}${SHLIB_SUFFIX} \
 				${CFITSIO_SHLIB} ${CFITSIO_SHLIB_SONAME}; do \
 		    if [ -f $$lib ]; then \
-			echo "/bin/cp -a $$lib ${CFITSIO_LIB}"; \
-			/bin/cp -a $$lib ${CFITSIO_LIB}; \
+			echo "cp -a $$lib ${CFITSIO_LIB}"; \
+			cp -a $$lib ${CFITSIO_LIB}; \
 		    fi; \
 		done
-		/bin/cp fitsio.h fitsio2.h longnam.h drvrsmem.h ${CFITSIO_INCLUDE}
-		/bin/cp cfitsio.pc ${CFITSIO_LIB}/pkgconfig
+		cp fitsio.h fitsio2.h longnam.h drvrsmem.h ${CFITSIO_INCLUDE}
+		cp cfitsio.pc ${CFITSIO_LIB}/pkgconfig
 		@for task in ${FPACK_UTILS} ${UTILS}; do \
 		    if [ -f $$task ]; then \
 			if [ ! -d ${CFITSIO_BIN} ]; then mkdir -p ${CFITSIO_BIN}; fi; \
-			echo "/bin/cp $$task ${CFITSIO_BIN}"; \
-			/bin/cp $$task ${CFITSIO_BIN}; \
+			echo "cp $$task ${CFITSIO_BIN}"; \
+			cp $$task ${CFITSIO_BIN}; \
 		    fi; \
 		done
 
