diff -Naur orig/src/sdl/video.c mod/src/sdl/video.c
--- orig/src/sdl/video.c	2013-02-21 16:49:35.000000000 +0100
+++ mod/src/sdl/video.c	2014-08-26 20:33:40.090590436 +0200
@@ -344,7 +344,7 @@
    if (key_file)
    {
       fprintf(key_file, "pullup ,\n");
-      fprintf(key_file, "pulldown /\n");
+      fprintf(key_file, "pulldown -\n");
       fprintf(key_file, "flip . \n");
       fprintf(key_file, "bomb b\n");
       fprintf(key_file, "fire  \n");
