diff -Naur a/src/sdl/video.c b/src/sdl/video.c
--- a/src/sdl/video.c	2014-11-17 22:25:43.000000000 +0100
+++ b/src/sdl/video.c	2023-01-20 22:08:39.348465105 +0100
@@ -37,6 +37,8 @@
 
 #include "vid_vga.c"
 
+#include <time.h>
+
 // lcd mode to emulate my old laptop i used to play sopwith on :)
 
 //#define LCD
diff -Naur a/src/vid_vga.c b/src/vid_vga.c
--- a/src/vid_vga.c	2012-11-08 19:35:19.000000000 +0100
+++ b/src/vid_vga.c	2023-01-20 22:08:54.548492997 +0100
@@ -33,6 +33,8 @@
 #include "swsymbol.h"
 #include "swutil.h"
 
+#include <string.h>
+
 // this should be in a header somewhere
 #define SBAR_HGHT 19
 
