diff -Nur orig/src/backend/bacon-video-widget-gst-missing-plugins.c mod/src/backend/bacon-video-widget-gst-missing-plugins.c
--- orig/src/backend/bacon-video-widget-gst-missing-plugins.c	2017-09-11 14:30:28.000000000 +0200
+++ mod/src/backend/bacon-video-widget-gst-missing-plugins.c	2018-10-20 19:38:33.835462388 +0200
@@ -353,12 +353,7 @@
 	/* TRANSLATORS: this is a button to launch a codec installer.
 	 * %s will be replaced with the software installer's name, e.g.
 	 * 'Software' in case of gnome-software. */
-	button_text = g_strdup_printf (_("_Find in %s"), install_helper_display_name);
 	gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (dialog), "%s", message_text);
-	button = gtk_dialog_add_button (GTK_DIALOG (dialog),
-	                                button_text,
-	                                GTK_RESPONSE_ACCEPT);
-	gtk_dialog_set_default_response (GTK_DIALOG (dialog), GTK_RESPONSE_ACCEPT);
 	gtk_style_context_add_class (gtk_widget_get_style_context (button), "suggested-action");
 	g_signal_connect (dialog, "response",
 	                  G_CALLBACK (codec_confirmation_dialog_response_cb),
