diff -Naur orig/Makefile mod/Makefile
--- orig/Makefile	2019-11-07 13:56:23.000000000 +0100
+++ mod/Makefile	2021-03-28 13:55:31.276524738 +0200
@@ -9,8 +9,8 @@
 
 UTF8_SUPPORT:=yes
 DESTDIR=
-PREFIX=/usr/local
-CONFIG_FILE=$(DESTDIR)$(PREFIX)/etc/multitail.conf
+PREFIX=/usr
+CONFIG_FILE=$(DESTDIR)/etc/multitail.conf
 
 CC?=gcc
 CFLAGS+=-Wall -Wno-unused-parameter -funsigned-char -O3
@@ -50,15 +50,11 @@
 	cp multitail $(DESTDIR)$(PREFIX)/bin
 	mkdir -p $(DESTDIR)$(PREFIX)/share/man/man1
 	cp multitail.1 $(DESTDIR)$(PREFIX)/share/man/man1/multitail.1
-	mkdir -p $(DESTDIR)$(PREFIX)/share/doc/multitail-$(VERSION)
-	cp *.txt INSTALL manual*.html $(DESTDIR)$(PREFIX)/share/doc/multitail-$(VERSION)
 	#
 	### COPIED multitail.conf.new, YOU NEED TO REPLACE THE multitail.conf
 	### YOURSELF WITH THE NEW FILE
 	#
-	mkdir -p $(DESTDIR)$(PREFIX)/etc/multitail/
-	cp multitail.conf $(CONFIG_FILE).new
-	cp conversion-scripts/* $(DESTDIR)$(PREFIX)/etc/multitail/
+	cp multitail.conf $(CONFIG_FILE)
 #rm -f $(DESTDIR)$(PREFIX)/share/man/man1/multitail.1.gz
 #gzip -9 $(DESTDIR)$(PREFIX)/share/man/man1/multitail.1
 	#
