diff -Naur orig/misc/mc.ext.in patched/misc/mc.ext.in
--- orig/misc/mc.ext.in	2019-06-16 19:49:31.000000000 +0200
+++ patched/misc/mc.ext.in	2019-06-30 20:43:34.564269863 +0200
@@ -119,35 +119,35 @@
 ### Archives ###
 
 # .tgz, .tpz, .tar.gz, .tar.z, .tar.Z, .ipk, .gem
-regex/\.t([gp]?z|ar\.g?[zZ])$|\.ipk$|\.gem$
+regex/i/\.t([gp]?z|ar\.g?[zZ])$|\.ipk$|\.gem$
 	Open=%cd %p/utar://
 	View=%view{ascii} @EXTHELPERSDIR@/archive.sh view tar.gz
 
-shell/.tar.bz
+shell/i/.tar.bz
 	# Open=%cd %p/utar://
 	View=%view{ascii} @EXTHELPERSDIR@/archive.sh view tar.bzip
 
-regex/\.t(ar\.bz2|bz2?|b2)$
+regex/i/\.t(ar\.bz2|bz2?|b2)$
 	Open=%cd %p/utar://
 	View=%view{ascii} @EXTHELPERSDIR@/archive.sh view tar.bzip2
 
 # .tar.lzma, .tlz
-regex/\.t(ar\.lzma|lz)$
+regex/i/\.t(ar\.lzma|lz)$
 	Open=%cd %p/utar://
 	View=%view{ascii} @EXTHELPERSDIR@/archive.sh view tar.lzma
 
 # .tar.lz
-shell/.tar.lz
+shell/i/.tar.lz
 	Open=%cd %p/utar://
 	View=%view{ascii} @EXTHELPERSDIR@/archive.sh view tar.lz
 
 # .tar.lz4, .tlz4
-regex/\.t(ar\.lz4|lz4)$
+regex/i/\.t(ar\.lz4|lz4)$
 	Open=%cd %p/utar://
 	View=%view{ascii} @EXTHELPERSDIR@/archive.sh view tar.lz4
 
 # .tar.xz, .txz
-regex/\.t(ar\.xz|xz)$
+regex/i/\.t(ar\.xz|xz)$
 	Open=%cd %p/utar://
 	View=%view{ascii} @EXTHELPERSDIR@/archive.sh view tar.xz
 
@@ -157,12 +157,12 @@
 	View=%view{ascii} @EXTHELPERSDIR@/archive.sh view tar.zst
 
 # .tar.F - used in QNX
-shell/.tar.F
+shell/i/.tar.F
 	# Open=%cd %p/utar://
 	View=%view{ascii} @EXTHELPERSDIR@/archive.sh view tar.F
 
 # .qpr/.qpk - QNX Neutrino package installer files
-regex/\.qp[rk]$
+regex/i/\.qp[rk]$
 	Open=%cd %p/utar://
 	View=%view{ascii} @EXTHELPERSDIR@/archive.sh view tar.qpr
 
@@ -207,19 +207,19 @@
 	View=%view{ascii} @EXTHELPERSDIR@/archive.sh view alz
 
 # cpio
-shell/.cpio.Z
+shell/i/.cpio.Z
 	Open=%cd %p/ucpio://
 	View=%view{ascii} @EXTHELPERSDIR@/archive.sh view cpio.z
 
-shell/.cpio.lz
+shell/i/.cpio.lz
 	Open=%cd %p/ucpio://
 	View=%view{ascii} @EXTHELPERSDIR@/archive.sh view cpio.lz
 
-shell/.cpio.lz4
+shell/i/.cpio.lz4
 	Open=%cd %p/ucpio://
 	View=%view{ascii} @EXTHELPERSDIR@/archive.sh view cpio.lz4
 
-shell/.cpio.xz
+shell/i/.cpio.xz
 	Open=%cd %p/ucpio://
 	View=%view{ascii} @EXTHELPERSDIR@/archive.sh view cpio.xz
 
@@ -227,7 +227,7 @@
 	Open=%cd %p/ucpio://
 	View=%view{ascii} @EXTHELPERSDIR@/archive.sh view cpio.zst
 
-shell/.cpio.gz
+shell/i/.cpio.gz
 	Open=%cd %p/ucpio://
 	View=%view{ascii} @EXTHELPERSDIR@/archive.sh view cpio.gz
 
