diff -Naur orig/misc/mc.ext.in patched/misc/mc.ext.in
--- orig/misc/mc.ext.in	2019-06-30 20:43:34.564269863 +0200
+++ patched/misc/mc.ext.in	2019-06-30 20:45:02.073414060 +0200
@@ -669,6 +669,20 @@
 	Open=@EXTHELPERSDIR@/doc.sh open epub
 	View=%view{ascii} @EXTHELPERSDIR@/doc.sh view epub
 
+# checksum files
+regex/\.([sS][fF][vV])$
+	Open=cksfv -f %f
+
+regex/\.([mM][dD]5)$
+	Open=%view{ascii}md5sum -c %f
+
+
+regex/\.([dD][jJ][vV][uU])$
+	Open=(evince %f &)
+
+
+regex/\.([iI][cC][oO])$
+	Open=(gthumb %f &)
 
 ### Miscellaneous ###
 
