diff -Naur orig/misc/ext.d/package.sh patched/misc/ext.d/package.sh
--- orig/misc/ext.d/package.sh	2019-06-16 19:49:31.000000000 +0200
+++ patched/misc/ext.d/package.sh	2019-06-30 20:46:16.038299707 +0200
@@ -24,7 +24,7 @@
         $RPM -qivlp --scripts "${MC_EXT_FILENAME}"
         ;;
     deb)
-        dpkg-deb -I "${MC_EXT_FILENAME}" && echo && dpkg-deb -c "${MC_EXT_FILENAME}"
+        dpkg-deb -I "${MC_EXT_FILENAME}" control ; echo ; dpkg-deb -c "${MC_EXT_FILENAME}"
         ;;
     debd)
         dpkg -s `echo "${MC_EXT_BASENAME}" | sed 's/\([0-9a-z.-]*\).*/\1/'`
diff -Naur orig/misc/mc.ext.in patched/misc/mc.ext.in
--- orig/misc/mc.ext.in	2019-06-30 20:45:02.073414060 +0200
+++ patched/misc/mc.ext.in	2019-06-30 20:46:16.038299707 +0200
@@ -281,7 +281,7 @@
 	View=%view{ascii} @EXTHELPERSDIR@/package.sh view rpm
 
 # deb
-regex/\.u?deb$
+regex/\.(u?deb|uhu)$
 	Open=%cd %p/deb://
 	View=%view{ascii} @EXTHELPERSDIR@/package.sh view deb
 
