diff -Naur orig/misc/mc.ext.in patched/misc/mc.ext.in
--- orig/misc/mc.ext.in	2019-06-30 20:47:23.131022323 +0200
+++ patched/misc/mc.ext.in	2019-06-30 20:48:55.432016442 +0200
@@ -142,6 +142,14 @@
 	View=%view{ascii} @EXTHELPERSDIR@/archive.sh view tar.lz
 
 # .tar.lz4, .tlz4
+
+# esp
+regex/\.e(sp|[0-9][0-9])$
+	Open=%cd %p#uesp
+	View=%view{ascii} unesp v %f
+	Extract=unesp x %f
+	Icon=compressed.xpm
+
 regex/i/\.t(ar\.lz4|lz4)$
 	Open=%cd %p/utar://
 	View=%view{ascii} @EXTHELPERSDIR@/archive.sh view tar.lz4
