diff -Naur orig/misc/mc.ext.in patched/misc/mc.ext.in
--- orig/misc/mc.ext.in	2019-06-30 20:48:55.432016442 +0200
+++ patched/misc/mc.ext.in	2019-06-30 20:51:21.561590318 +0200
@@ -436,6 +436,10 @@
 shell/i/.chm
 	Open=@EXTHELPERSDIR@/text.sh open chm
 
+regex/\.([fF][lL][aA][cC])$
+	Open=if [ "$DISPLAY" = "" ]; then mplayer %f; else (totem %f &); fi
+	View=%view{ascii} metaflac --list --except-block-type=SEEKTABLE %s
+
 
 ### Images ###
 
