diff -Naur orig/misc/mc.ext.in patched/misc/mc.ext.in
--- orig/misc/mc.ext.in	2019-06-30 20:51:21.561590318 +0200
+++ patched/misc/mc.ext.in	2019-06-30 20:52:13.654151376 +0200
@@ -164,6 +164,10 @@
 	Open=%cd %p/utar://
 	View=%view{ascii} @EXTHELPERSDIR@/archive.sh view tar.zst
 
+regex/\.(fpm|fdb)$
+    Open=%cd %p#utar
+    View=%view{ascii} tar tvvf %f
+
 # .tar.F - used in QNX
 shell/i/.tar.F
 	# Open=%cd %p/utar://
