diff -Naur orig/misc/ext.d/sound.sh patched/misc/ext.d/sound.sh
--- orig/misc/ext.d/sound.sh	2021-01-10 12:50:16.000000000 +0100
+++ patched/misc/ext.d/sound.sh	2021-02-12 22:43:08.553278420 +0100
@@ -52,7 +52,7 @@
         #tracker "${MC_EXT_FILENAME}"
         ;;
     wav22)
-        vplay -s 22 "${MC_EXT_FILENAME}"
+        mplayer -vo null "${MC_EXT_FILENAME}"
         ;;
     mp3)
         if [ -n "$DISPLAY" ]; then
