diff -Naur orig/misc/ext.d/text.sh.in patched/misc/ext.d/text.sh.in
--- orig/misc/ext.d/text.sh.in	2019-06-16 19:49:31.000000000 +0200
+++ patched/misc/ext.d/text.sh.in	2019-06-30 21:10:49.171566894 +0200
@@ -54,7 +54,7 @@
             cat "${MC_EXT_FILENAME}"
             ;;
         *)
-            MANROFFOPT=-c MAN_KEEP_FORMATTING=1 man -P cat "${MC_EXT_FILENAME}"
+            groff-utf8 -P-c -Tutf8 @MANDOC@ "${MC_EXT_FILENAME}"
             ;;
         esac
         ;;
@@ -62,10 +62,10 @@
         pod2man "${MC_EXT_FILENAME}" | nroff @MAN_FLAGS@ @MANDOC@
         ;;
     nroff.me)
-        nroff @MAN_FLAGS@ -me "${MC_EXT_FILENAME}"
+        groff-utf8 -P-c -Tutf8 -me "${MC_EXT_FILENAME}"
         ;;
     nroff.ms)
-        nroff @MAN_FLAGS@ -ms "${MC_EXT_FILENAME}"
+        groff-utf8 -P-c -Tutf8 -ms "${MC_EXT_FILENAME}"
         ;;
     man.gz|man.bz|man.bz2|man.lz|man.lz4|man.lzma|man.xz|man.zst)
         case "${MC_EXT_FILENAME}" in
@@ -100,8 +100,7 @@
             cat "${MC_EXT_FILENAME}"
             ;;
         *)
-            { zsoelim "${MC_EXT_FILENAME}" 2>/dev/null || cat "${MC_EXT_FILENAME}"; } | \
-                nroff @MAN_FLAGS@ @MANDOC@
+            groff-utf8 -P-c -Tutf8 @MANDOC@ "${MC_EXT_FILENAME}" | ${pager}
             ;;
         esac | ${pager}
         ;;
@@ -109,10 +108,10 @@
         pod2man "${MC_EXT_FILENAME}" | nroff @MAN_FLAGS@ @MANDOC@ | ${pager}
         ;;
     nroff.me)
-        nroff @MAN_FLAGS@ -me "${MC_EXT_FILENAME}" | ${pager}
+        groff-utf8 -P-c -Tutf8 -me "${MC_EXT_FILENAME}" | ${pager}
         ;;
     nroff.ms)
-        nroff @MAN_FLAGS@ -ms "${MC_EXT_FILENAME}" | ${pager}
+        groff-utf8 -P-c -Tutf8 -me "${MC_EXT_FILENAME}" | ${pager}
         ;;
     man.gz|man.bz|man.bz2|man.lz|man.lz4|man.lzma|man.xz|man.zst)
         case "${MC_EXT_FILENAME}" in
@@ -120,7 +119,7 @@
             ${unpacker} "${MC_EXT_FILENAME}"
             ;;
         *)
-            ${unpacker} "${MC_EXT_FILENAME}" | nroff @MAN_FLAGS@ @MANDOC@
+            ${unpacker} "${MC_EXT_FILENAME}" | groff-utf8 -P-c -Tutf8 @MANDOC@
             ;;
         esac | ${pager}
         ;;
