diff -Naur orig/misc/ext.d/image.sh patched/misc/ext.d/image.sh
--- orig/misc/ext.d/image.sh	2014-08-14 15:29:32.000000000 +0200
+++ patched/misc/ext.d/image.sh	2014-10-06 19:52:41.973953633 +0200
@@ -42,7 +42,7 @@
         ;;
     *)
         if [ -n "$DISPLAY" ]; then
-            (gqview "${MC_EXT_FILENAME}" &)
+            (gthumb "${MC_EXT_FILENAME}" &)
         elif see >/dev/null 2>&1; then
             (see "${MC_EXT_FILENAME}" &)
         else
