diff -Naur orig/misc/mc.ext.in patched/misc/mc.ext.in
--- orig/misc/mc.ext.in	2012-07-18 11:30:23.000000000 +0200
+++ patched/misc/mc.ext.in	2012-07-23 11:22:56.139746055 +0200
@@ -541,7 +541,7 @@
 	Open=@EXTHELPERSDIR@/doc.sh open ooffice
 
 # StarOffice 6 and OpenOffice.org formats
-regex/i/\.(odt|ott|sxw|stw|ods|ots|sxc|stc|odp|otp|sxi|sti|odg|otg|sxd|std|odb|odf|sxm|odm|sxg)$
+regex/i/\.(f?odt|ott|sxw|stw|ods|ots|sxc|stc|odp|otp|sxi|sti|odg|otg|sxd|std|odb|odf|sxm|odm|sxg)$
 	Open=@EXTHELPERSDIR@/doc.sh open ooffice
 	View=%view{ascii} @EXTHELPERSDIR@/doc.sh view odt
 
