diff -Naur orig/misc/ext.d/sound.sh patched/misc/ext.d/sound.sh
--- orig/misc/ext.d/sound.sh	2014-03-13 10:06:03.000000000 +0100
+++ patched/misc/ext.d/sound.sh	2014-10-06 19:53:37.854230729 +0200
@@ -35,7 +35,7 @@
     case "${filetype}" in
     common)
         if [ -n "$DISPLAY" ]; then
-            (xmms  "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
+            (totem  "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
         else
             play "${MC_EXT_FILENAME}"
         fi
@@ -45,18 +45,18 @@
         #tracker "${MC_EXT_FILENAME}"
         ;;
     wav22)
-        vplay -s 22 "${MC_EXT_FILENAME}"
+        mplayer -vo null "${MC_EXT_FILENAME}"
         ;;
     mp3)
         if [ -n "$DISPLAY" ]; then
-            (xmms "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
+            (totem "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
         else
             mpg123 "${MC_EXT_FILENAME}"
         fi
         ;;
     ogg)
         if [ -n "$DISPLAY" ]; then
-            (xmms "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
+            (totem "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
         else
             ogg123 "${MC_EXT_FILENAME}"
         fi
@@ -69,7 +69,7 @@
         ;;
     playlist)
         if [ -n "$DISPLAY" ]; then
-            (xmms -p "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
+            (totem -p "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
         else
             mplayer -vo null -playlist "${MC_EXT_FILENAME}"
         fi
