--- zoo-2.10.orig/zoopack.c
+++ zoo-2.10/zoopack.c
@@ -10,6 +10,7 @@
 Copyright (C) 1986, 1987 Rahul Dhesi -- All rights reserved
 (C) Copyright 1988 Rahul Dhesi -- All rights reserved
 */
+#include <unistd.h>
 #include "options.h"
 /* Packs an archive.  The sequence is:
    1. Copy all files from current archive to new one.
@@ -388,7 +389,7 @@
 
 /* handle_break() */
 /* Sets break_hit to 1 when called */
-T_SIGNAL handle_break()
+T_SIGNAL handle_break(int dummy)
 {
 #ifndef NOSIGNAL
    signal (SIGINT, SIG_IGN);     /* ignore future control ^Cs for now */
