diff -Nur orig/configure.ac mod/configure.ac
--- orig/configure.ac	2015-10-19 10:37:47.000000000 +0200
+++ mod/configure.ac	2019-07-10 14:50:17.885512465 +0200
@@ -33,9 +33,6 @@
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE", [Gettext package])
 
-GNOME_DEBUG_CHECK
-GNOME_COMPILE_WARNINGS([maximum])
-
 AC_PATH_XTRA
 x_libs="$X_PRE_LIBS $X_LIBS -lX11 $X_EXTRA_LIBS"
 
