diff -Naur orig/m4/openbox.m4 mod/m4/openbox.m4
--- orig/m4/openbox.m4	2015-07-01 11:30:11.000000000 +0200
+++ mod/m4/openbox.m4	2015-07-19 20:13:37.426734079 +0200
@@ -81,7 +81,7 @@
     if test "$GCC" = "yes"; then
 	AC_MSG_RESULT([yes])
 	if test "$DEBUG" = "yes"; then
-	    FLAGS="$FLAGS -O0 -ggdb -fno-inline -Wwrite-strings"
+	    FLAGS="$FLAGS -fno-inline -Wwrite-strings"
 	    FLAGS="$FLAGS -Wall -Wsign-compare -Waggregate-return"
 	    FLAGS="$FLAGS -Wbad-function-cast -Wpointer-arith"
 	    FLAGS="$FLAGS -Wno-write-strings"
@@ -101,7 +101,7 @@
 	if test "$PROFLC" = "yes"; then
 	    L="$L -lc_p -lm_p"
 	fi
-	FLAGS="$FLAGS -fno-strict-aliasing"
+	FLAGS="$FLAGS "
     fi
     AC_MSG_CHECKING([for compiler specific flags])
     AC_MSG_RESULT([$FLAGS])
