diff -Naurd a/openbox/client.c b/openbox/client.c
--- a/openbox/client.c	2013-08-12 03:33:24.000000000 +0200
+++ b/openbox/client.c	2013-08-13 18:31:46.000000000 +0200
@@ -941,7 +941,7 @@
                  !g_pattern_match(app->role,
                                   strlen(self->role), self->role, NULL))
             match = FALSE;
-        else if (app->title &&
+        else if (app->title && self->title &&
                  !g_pattern_match(app->title,
                                   strlen(self->title), self->title, NULL))
             match = FALSE;
