diff -Nur orig/bin/trigger-rally.config.defs mod/bin/trigger-rally.config.defs
--- orig/bin/trigger-rally.config.defs	2016-12-10 22:16:46.000000000 +0100
+++ mod/bin/trigger-rally.config.defs	2018-02-24 13:34:17.000000000 +0100
@@ -31,9 +31,9 @@
 		-->
 
 	<video
-		automatic="yes"
-		width="800"
-		height="600"
+		automatic="no"
+		width="1024"
+		height="768"
 		bpp="0"
 		fullscreen="no"
 		requirergb="no"
