diff -Nur orig/config.cfg mod/config.cfg
--- orig/config.cfg	2004-09-27 11:47:31.000000000 +0200
+++ mod/config.cfg	2018-02-09 17:18:46.926421960 +0100
@@ -1,6 +1,6 @@
 video_mode = 800 x 600
 video_mode_color_depth = 32
-video_mode_fullscreen = 1
+video_mode_fullscreen = 0
 sound_enabled = 1
 sound_freq = 44100
 sound_volume = 205
@@ -11,6 +11,6 @@
 key_up = 273 119
 key_down = 274 115
 key_shoot = 32 304
-key_special = 307 306
+key_special = 305 306
 perspective = 0
 moving_style = 2 2
