diff -Nur orig/data/devede_ng.py.desktop mod/data/devede_ng.py.desktop
--- orig/data/devede_ng.py.desktop	2017-12-03 11:50:09.000000000 +0100
+++ mod/data/devede_ng.py.desktop	2018-01-11 11:50:40.400220254 +0100
@@ -15,7 +15,7 @@
 Comment[gl]=Permite crear video CDs e DVDs, que poden reproducirse en reproductores domésticos
 Comment[pl]=Pozwala stworzyć wideo na płytach CD lub DVD, do otwarzania w domowych odtwarzaczach DVD
 Comment[it]=Permette di creare video CD e DVD che possono essere riprodotti con lettori DVD
-Icon=devedeng_icon.svg
+Icon=/usr/share/pixmaps/devedeng_icon.svg
 TryExec=devede_ng.py
 Exec=devede_ng.py
 Categories=AudioVideo;
