diff -Naur orig/archive/KioArchiveConfig.cmake.in patched/archive/KioArchiveConfig.cmake.in
--- orig/archive/KioArchiveConfig.cmake.in	2018-08-06 22:46:33.000000000 +0200
+++ patched/archive/KioArchiveConfig.cmake.in	2018-08-30 03:41:28.401299691 +0200
@@ -2,6 +2,6 @@
 
 include(CMakeFindDependencyMacro)
 
-find_dependency(KF5KIO)
+find_dependency(KF5KIO )
 
 include("${CMAKE_CURRENT_LIST_DIR}/KioArchiveTargets.cmake")
