diff -Naur orig/src-qt5/core/lumina-desktop/fluxboxconf/fluxbox-init-rc patched/src-qt5/core/lumina-desktop/fluxboxconf/fluxbox-init-rc
--- orig/src-qt5/core/lumina-desktop/fluxboxconf/fluxbox-init-rc	2016-08-08 15:22:36.000000000 +0200
+++ patched/src-qt5/core/lumina-desktop/fluxboxconf/fluxbox-init-rc	2016-08-11 00:03:24.345932461 +0200
@@ -66,7 +66,7 @@
 session.titlebar.right:	Minimize Maximize Close
 session.colorsPerChannel:	4
 session.opaqueMove:	False
-session.styleFile:	/usr/local/share/fluxbox/styles/bias_allblack
+session.styleFile:	/usr/share/fluxbox/styles/bias_allblack
 session.doubleClickInterval:	250
 session.iconbar:	false
 session.cacheLife:	5
