diff -Naur orig/src-qt5/desktop-utils/lumina-archiver/lumina-archiver.desktop patched/src-qt5/desktop-utils/lumina-archiver/lumina-archiver.desktop
--- orig/src-qt5/desktop-utils/lumina-archiver/lumina-archiver.desktop	2019-04-29 14:21:12.000000000 +0200
+++ patched/src-qt5/desktop-utils/lumina-archiver/lumina-archiver.desktop	2019-04-30 01:39:57.779893199 +0200
@@ -4,7 +4,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=Utility;
+Categories=Utility;Archiving;Compression;
 MimeType=application/x-xz-compressed-tar;application/x-lzma-compressed-tar;application/x-bzip-compressed-tar;application/x-tar;application/x-compressed-tar;application/zip;application/gzip;application/bzip;application/x-cd-image;application/x-raw-disk-image;
 Name=Lumina Archiver
 Comment=Archive creator and viewer
diff -Naur orig/src-qt5/desktop-utils/lumina-fm/lumina-fm.desktop patched/src-qt5/desktop-utils/lumina-fm/lumina-fm.desktop
--- orig/src-qt5/desktop-utils/lumina-fm/lumina-fm.desktop	2019-04-29 14:21:12.000000000 +0200
+++ patched/src-qt5/desktop-utils/lumina-fm/lumina-fm.desktop	2019-04-30 01:38:28.142809520 +0200
@@ -4,7 +4,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=Utility;
+Categories=Utility;System;FileManager;FileTools;
 MimeType=inode/directory;
 Name=Insight File Manager
 Name[fr]=Insight Gestionnaire de fichiers
diff -Naur orig/src-qt5/desktop-utils/lumina-mediaplayer/lumina-mediaplayer.desktop patched/src-qt5/desktop-utils/lumina-mediaplayer/lumina-mediaplayer.desktop
--- orig/src-qt5/desktop-utils/lumina-mediaplayer/lumina-mediaplayer.desktop	2019-04-29 14:21:12.000000000 +0200
+++ patched/src-qt5/desktop-utils/lumina-mediaplayer/lumina-mediaplayer.desktop	2019-04-30 01:38:28.142809520 +0200
@@ -4,7 +4,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=AudioVideo;
+Categories=AudioVideo;Audio;Video;Player;
 MimeType=
 Name=Lumina Media Player
 Comment=Play Multimedia Content
diff -Naur orig/src-qt5/desktop-utils/lumina-textedit/lumina-textedit.desktop patched/src-qt5/desktop-utils/lumina-textedit/lumina-textedit.desktop
--- orig/src-qt5/desktop-utils/lumina-textedit/lumina-textedit.desktop	2019-04-29 14:21:12.000000000 +0200
+++ patched/src-qt5/desktop-utils/lumina-textedit/lumina-textedit.desktop	2019-04-30 01:38:29.262823061 +0200
@@ -4,7 +4,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=Office;Development;
+Categories=Utility;TextEditor;
 MimeType=text/plain;text/x-log;text/*;
 Name=Lumina Text Editor
 Comment=Plaintext editor with syntax highlighting
