diff -Naur orig/gio/gdbusaddress.c patched/gio/gdbusaddress.c
--- orig/gio/gdbusaddress.c	1970-01-01 01:00:00.000000000 +0100
+++ patched/gio/gdbusaddress.c	2021-03-16 23:40:11.424804474 +0100
@@ -1331,7 +1331,7 @@
 
       if (ret == NULL)
         {
-          ret = g_strdup ("unix:path=/var/run/dbus/system_bus_socket");
+          ret = g_strdup ("unix:path=/run/dbus/system_bus_socket");
         }
       break;
 
