diff -Naur orig/gio/gdbusaddress.c patched/gio/gdbusaddress.c
--- orig/gio/gdbusaddress.c	2020-03-11 13:21:39.000000000 +0100
+++ patched/gio/gdbusaddress.c	2020-05-16 02:19:03.092593469 +0200
@@ -1323,7 +1323,7 @@
       ret = g_strdup (g_getenv ("DBUS_SYSTEM_BUS_ADDRESS"));
       if (ret == NULL)
         {
-          ret = g_strdup ("unix:path=/var/run/dbus/system_bus_socket");
+          ret = g_strdup ("unix:path=/run/dbus/system_bus_socket");
         }
       break;
 
