diff -Naur orig/gio/gdbusaddress.c patched/gio/gdbusaddress.c
--- orig/gio/gdbusaddress.c	2019-05-21 17:59:05.000000000 +0200
+++ patched/gio/gdbusaddress.c	2019-07-30 21:30:17.020450878 +0200
@@ -1670,7 +1670,7 @@
       ret = g_strdup (g_getenv ("DBUS_SYSTEM_BUS_ADDRESS"));
       if (ret == NULL)
         {
-          ret = g_strdup ("unix:path=/var/run/dbus/system_bus_socket");
+          ret = g_strdup ("unix:path=/run/dbus/system_bus_socket");
         }
       break;
 
