diff -Naur orig/frontend/applets/mate/src/Makefile.am patched/frontend/applets/mate/src/Makefile.am
--- orig/frontend/applets/mate/src/Makefile.am	2016-09-15 16:51:21.134741323 +0200
+++ patched/frontend/applets/mate/src/Makefile.am	2017-03-22 12:04:59.693238896 +0100
@@ -18,7 +18,7 @@
 				-DWORKRAVE_MENU_UI_DIR="\"${WORKRAVE_MENU_UI_DIR}"\"
 
 workrave_applet_LDADD = 	$(LIBMATEPANEL_LIBS) $(LIBMATEGUI4_LIBS) $(LIBMATEPANEL_LIBS) \
-			        @GTK2_LIBS@ -L$(builddir)/../../common/src -lworkrave-gtk2-private-1.0
+			        @GTK_LIBS@ -L$(builddir)/../../common/src -lworkrave-private-1.0
 
 
 appletsdir       = $(datadir)/mate-panel/applets
