diff -Nur orig/data/start mod/data/start
--- orig/data/start	2025-11-10 18:23:03.000000000 +0100
+++ mod/data/start	2025-11-20 09:49:48.766416635 +0100
@@ -12,3 +12,46 @@
 #
 # Execution of the file can be skipped by passing --skip-start to pekwm.
 # 
+
+#háttérkép
+#pekwm_bg Image /usr/share/backgrounds/UHU/ubk4.png#scaled &
+
+#polkit ügynökök, ki kell választani egyet.
+#/usr/libexec/lxpolkit
+#/usr/libexec/polkit-gnome-authentication-agent-1 &
+#/usr/libexec/polkit-mate-authentication-agent-1 &
+#/usr/lib/libexec/polkit-kde-authentication-agent-1
+
+# panelek
+#pekwm_panel &
+#bmpanel2
+
+# Nagy panelek
+#mate-panel &
+#gnome-panel &
+#xfce4-panel &
+#lxpanel &
+
+# rendszertálcák
+#fbpanel &
+#tint2 & # Beálítása a ~/.config/tint2/tint2rc fájlban történhet
+
+# képernyővédő
+#xscreensaver &
+
+# ikonok a háttérre (https://wiki.archlinux.org/title/Idesk)
+#idesk &
+
+# nm-applet a panelekre és rendszertálcákra, ha nincs ott
+#nm-applet &
+
+# egyebek
+#unclutter -idle 2 &
+#wmnd &
+#wmsmixer -w &
+
+# háttérkép
+#pekwm_bg Image /usr/share/backgrounds/UHU/ubk4.png#scaled &
+
+# ha a megjelenített háttér nem megfelelő
+#feh --bg-scale ~/usr/share/backgrounds/UHU/ubk4.png &
