diff -Nur orig/data/start mod/data/start
--- orig/data/start	2021-12-03 20:13:43.000000000 +0100
+++ mod/data/start	2022-11-10 19:12:57.513690261 +0100
@@ -16,3 +16,10 @@
 # There's probably a few other good uses for it, too. I mainly pushed for it
 # because when I was doing fluxbox's docs, people used to complain that there
 # wasn't one, and I wanted to avoid that for pekwm. ;) --eyez
+
+#háttérkép
+pekwm_bg Image /usr/share/backgrounds/UHU/ubk4.png#scaled &
+#panel
+pekwm_panel &
+#polkit ügynök
+/usr/libexec/lxpolkit &
