diff -Nur orig/data/panel mod/data/panel
--- orig/data/panel	2021-05-29 10:10:55.000000000 +0200
+++ mod/data/panel	2021-05-29 14:14:09.175345148 +0200
@@ -5,9 +5,9 @@
 }
 
 Commands {
-  Command = "uptime | sed 's/.*:/load/'" {
-    Interval = "15"
-  }
+#  Command = "uptime | sed 's/.*:/load/'" {
+#    Interval = "15"
+#  }
   # battery percent (works on OpenBSD)
   # Command = "apm | head -1 | awk '{print \"battery \" $4}'" {
   #   Interval = "60"
@@ -31,4 +31,4 @@
     Size = "Required"
     Interval = "60"
   }
-}
\ Nincs új sor a fájl végén
+}
