diff -Nur orig/data/menu mod/data/menu
--- orig/data/menu	2021-05-29 10:10:55.000000000 +0200
+++ mod/data/menu	2021-05-29 16:17:00.897240336 +0200
@@ -16,6 +16,10 @@
   Entry = "Take screenshot" { Actions = "Exec pekwm_screenshot" }
 
   Separator {}
+    Submenu = "Alkalmazások" {
+    Entry = "" { Actions = "Dynamic /usr/bin/xdgmenumaker -n -i -f pekwm --pekwm-dynamic" }
+    }
+  Separator {}
 
   Submenu = "Go to" {
     SubMenu = "Workspace" {
@@ -30,7 +34,7 @@
     }
     Entry = "Window.." { Actions = "ShowMenu GotoClient True" }
   }
-
+ 
   Submenu = "pekwm" {
     Submenu = "Themes" {
       Entry { Actions = "Dynamic $_PEKWM_SCRIPT_PATH/pekwm_themeset.sh $_PEKWM_THEME_PATH" }
