diff -Nur orig/data/start mod/data/start
--- orig/data/start	2013-05-26 14:01:18.000000000 +0200
+++ mod/data/start	2016-10-25 15:12:20.000000000 +0200
@@ -16,3 +16,5 @@
 # There's probably a few other good uses for it, too. I mainly pushed for it
 # because when I was doing fluxbox's docs, people used to complain that there
 # wasn't one, and I wanted to avoid that for pekwm. ;) --eyez
+hsetroot -fill /usr/share/backgrounds/UHU/uhu_linux_ubk.jpg
+bmpanel2 &
