diff -Nur orig/data/blueCrux/default.theme mod/data/blueCrux/default.theme
--- orig/data/blueCrux/default.theme	2019-09-10 08:34:17.000000000 +0200
+++ mod/data/blueCrux/default.theme	2021-01-24 17:40:46.316534524 +0100
@@ -8,6 +8,8 @@
 # Default background image size is 1024x768. 
 # Use GIMP or ImageMagick to scale it for 800x600 desktops :)
 DesktopBackgroundImage="background.jpg"
+DesktopBackgroundScaled=1 # 0/1
+DesktopBackgroundCenter=0 # 0/1
 
 BorderSizeX=5
 BorderSizeY=5
diff -Nur orig/data/Crux-Teal/default.theme mod/data/Crux-Teal/default.theme
--- orig/data/Crux-Teal/default.theme	2019-09-10 08:34:17.000000000 +0200
+++ mod/data/Crux-Teal/default.theme	2021-01-24 17:41:21.384667665 +0100
@@ -90,6 +90,8 @@
 
 DesktopBackgroundColor	=	"#000000"
 DesktopBackgroundImage	=	"background.jpg"
+DesktopBackgroundScaled=1 # 0/1
+DesktopBackgroundCenter=0 # 0/1
 
 ColorToolTip		=	"#dddddd"
 ColorToolTipText	=	"#000000"
diff -Nur orig/data/SilverXP/default.theme mod/data/SilverXP/default.theme
--- orig/data/SilverXP/default.theme	2019-09-10 08:34:17.000000000 +0200
+++ mod/data/SilverXP/default.theme	2020-06-19 19:50:16.564391393 +0200
@@ -468,20 +468,16 @@
 #            to set the background!
 #
 
-#  Display desktop background centered and not tiled
-# DesktopBackgroundCenter=0 # 0/1
-
 #  Support for semitransparent terminals like Eterm or gnome-terminal
 # SupportSemitransparency=1 # 0/1
 
-#  Desktop background scaled to full screen
-# DesktopBackgroundScaled=0 # 0/1
-
 #  Desktop background color
 # DesktopBackgroundColor="rgb:00/20/40"
 
 #  Desktop background image
-# DesktopBackgroundImage=""
+DesktopBackgroundImage=winxp.jpg
+DesktopBackgroundScaled=1 # 0/1
+DesktopBackgroundCenter=0 # 0/1
 
 #  Color to announce for semi-transparent windows
 # DesktopTransparencyColor=""
diff -Nur orig/data/xp-new/default.theme mod/data/xp-new/default.theme
--- orig/data/xp-new/default.theme	2019-09-10 08:34:17.000000000 +0200
+++ mod/data/xp-new/default.theme	2020-06-19 19:50:50.240535426 +0200
@@ -17,6 +17,10 @@
 DlgBorderSizeX=3
 DlgBorderSizeY=3
 
+DesktopBackgroundImage=winxp.jpg
+DesktopBackgroundCenter=0 # 0/1
+DesktopBackgroundScaled=1 # 0/1
+
 ColorNormalTitleBarText="#b5cbe0"
 ColorNormalTitleBarShadow="#526570"
 ColorNormalBorder="#467cc1"
diff -Nur orig/lib/themes/win95/default.theme mod/lib/themes/win95/default.theme
--- orig/lib/themes/win95/default.theme	2020-06-03 05:28:09.000000000 +0200
+++ mod/lib/themes/win95/default.theme	2020-06-19 19:49:19.404145714 +0200
@@ -28,3 +28,7 @@
 ColorScrollBarSlider="rgb:C0/C0/C0"
 ColorScrollBarButton="rgb:C0/C0/C0"
 ColorScrollBarButtonArrow="rgb:00/00/00"
+
+DesktopBackgroundImage=win95.jpg
+DesktopBackgroundCenter=0
+DesktopBackgroundScaled=1
