diff -Nur orig/src/yprefs.h mod/src/yprefs.h
--- orig/src/yprefs.h	2017-07-30 10:53:35.000000000 +0200
+++ mod/src/yprefs.h	2018-01-02 21:49:04.257203042 +0100
@@ -35,9 +35,12 @@
 XIV(bool, haveXft,                              true)
 #endif
 #if defined(__linux__) || defined(__FreeBSD__)
-XSV(const char *, iconPath,                     "/usr/share/icons/hicolor:/usr/share/icons:/usr/share/pixmaps")
+#define CONFIG_DEFAULT_ICONPATH "/usr/share/pixmaps:/usr/share/icons/hicolor/48x48/apps:/usr/share/icons/gnome/48x48/apps:/usr/share/icons/gnome/48x48/actions:/usr/share/icons/gnome/48x48/devices:/usr/share/icons/default.kde4/48x48/apps"
+XSV(const char *, iconPath,                     CONFIG_DEFAULT_ICONPATH)
 #else
-XSV(const char *, iconPath,                     0)
+#define CONFIG_DEFAULT_ICONPATH "/usr/share/pixmaps:/usr/share/icons/hicolor/48x48/apps:/usr/share/icons/gnome/48x48/apps:/usr/share/icons/gnome/48x48/actions:/usr/share/icons/gnome/48x48/devices:/usr/share/icons/default.kde4/48x48/apps"
++XSV(const char *, iconPath,                     CONFIG_DEFAULT_ICONPATH)
+
 #endif
 #define CONFIG_DEFAULT_THEME "default/default.theme"
 XSV(const char *, themeName,                    CONFIG_DEFAULT_THEME)
