diff -Naur orig/lib/menu.in mod/lib/menu.in
--- orig/lib/menu.in	2015-07-03 03:48:23.000000000 +0200
+++ mod/lib/menu.in	2015-07-10 22:00:43.009119595 +0200
@@ -12,6 +12,6 @@
 prog XChat xchat xchat
 prog Gimp gimp gimp
 separator
-menuprog "Desktop Apps" folder icewm-menu-fdo
-menufile Programs folder programs
-menufile Tool_bar folder toolbar
+menuprog "Alkalmazások" folder icewm-menu-fdo
+menufile Programok folder programs
+menufile Tálca folder toolbar
