diff -Nur orig/lib/menu.in mod/lib/menu.in
--- orig/lib/menu.in	2019-06-07 00:39:15.000000000 +0200
+++ mod/lib/menu.in	2019-06-09 16:29:10.492658372 +0200
@@ -12,5 +12,5 @@
 prog Hexchat xchat hexchat
 prog Gimp gimp gimp
 includeprog icewm-menu-fdo --sep-before --no-sep-others
-menufile Programs folder programs
-menufile Tool_bar folder toolbar
+menufile Programok folder programs
+menufile Tálca folder toolbar
