diff -Nur orig/lib/menu.in mod/lib/menu.in
--- orig/lib/menu.in	2020-06-03 05:28:09.000000000 +0200
+++ mod/lib/menu.in	2020-06-28 12:04:20.902636649 +0200
@@ -7,9 +7,8 @@
 prog urxvt xterm urxvt -bg black -cr green -fg white -C -sl 500
 prog gedit fte gedit
 prog NEdit nedit nedit
-prog Firefox mozilla firefox
+prog Firefox firefox firefox
 prog Hexchat xchat hexchat
 prog Gimp gimp gimp
 includeprog icewm-menu-fdo --sep-before --no-sep-others
-menufile Programs folder programs
-menufile Tool_bar folder toolbar
+menufile Tálca folder toolbar
