diff -Nur orig/lib/toolbar.in mod/lib/toolbar.in
--- orig/lib/toolbar.in	2019-06-07 00:39:15.000000000 +0200
+++ mod/lib/toolbar.in	2019-06-09 16:32:36.564366054 +0200
@@ -4,6 +4,6 @@
 # since modifications to this file will be discarded when you
 # (re)install icewm.
 #
-prog "@XTERMCMD@" utilities-terminal @XTERMCMD@
-prog "Web browser" web-browser xdg-open about:blank
-
+prog xterm terminal xterm
+prog "Web böngésző" web-browser xdg-open about:blank
+prog synaptic synaptic synaptic
