diff -Nur orig/lib/toolbar.in mod/lib/toolbar.in
--- orig/lib/toolbar.in	2022-02-23 08:34:22.000000000 +0100
+++ mod/lib/toolbar.in	2022-03-13 16:51:13.280660256 +0100
@@ -3,6 +3,9 @@
 # Place your variants in @CFGDIR@ or in $HOME/.icewm since
 # modifications to this file will be overwritten when you reinstall icewm.
 #
-prog "@XTERMCMD@" utilities-terminal @XTERMCMD@
-prog "Web browser" web-browser xdg-open about:blank
-
+prog kitty kitty kitty
+prog "Firefox" firefox firefox
+prog "Thunderbird" thunderbird thunderbird
+prog "Krusader" krusader_user krusader
+prog "filezilla" filezilla filezilla
+prog synaptic synaptic synaptic
