diff -Nur orig/pyproject.toml mod/pyproject.toml
--- orig/pyproject.toml	2025-07-30 23:31:52.000000000 +0200
+++ mod/pyproject.toml	2025-09-15 13:50:15.010082777 +0200
@@ -4,13 +4,7 @@
 name = "pip"
 description = "The PyPA recommended tool for installing Python packages."
 readme = "README.rst"
-license = "MIT"
-license-files = [
-  "AUTHORS.txt",
-  "LICENSE.txt",
-  "src/pip/_vendor/**/*COPYING*",
-  "src/pip/_vendor/**/*LICENSE*",
-]
+
 classifiers = [
   "Development Status :: 5 - Production/Stable",
   "Intended Audience :: Developers",
@@ -46,7 +40,7 @@
 Changelog = "https://pip.pypa.io/en/stable/news/"
 
 [build-system]
-requires = ["setuptools>=77"]
+requires = ["setuptools>=73.0.1"]
 build-backend = "setuptools.build_meta"
 
 [dependency-groups]
