diff -Nur orig/pyproject.toml mod/pyproject.toml
--- orig/pyproject.toml	2024-02-03 10:48:10.000000000 +0100
+++ mod/pyproject.toml	2024-04-12 09:14:01.831970988 +0200
@@ -38,7 +38,7 @@
 
 [build-system]
 # The lower bound is for <https://github.com/pypa/setuptools/issues/3865>.
-requires = ["setuptools>=67.6.1", "wheel"]
+requires = ["setuptools>=65.6.3", "wheel"]
 build-backend = "setuptools.build_meta"
 
 [tool.setuptools]
