diff -urN asterisk-11.2.1.orig/addons/chan_ooh323.c asterisk-11.2.1/addons/chan_ooh323.c
--- asterisk-11.2.1.orig/addons/chan_ooh323.c	2012-08-10 18:49:27.000000000 +0200
+++ asterisk-11.2.1/addons/chan_ooh323.c	2013-02-15 18:29:29.607812984 +0100
@@ -18,7 +18,7 @@
 
 
 /*** MODULEINFO
-	<defaultenabled>no</defaultenabled>
+	<defaultenabled>yes</defaultenabled>
 	<support_level>extended</support_level>
  ***/
 
