diff -Nur orig/builds/compiler/gcc.mk mod/builds/compiler/gcc.mk
--- orig/builds/compiler/gcc.mk	2016-02-03 19:13:58.000000000 +0100
+++ mod/builds/compiler/gcc.mk	2017-06-12 18:25:55.595243762 +0200
@@ -61,7 +61,7 @@
 #   Use the ANSIFLAGS variable to define the compiler flags used to enfore
 #   ANSI compliance.
 #
-CFLAGS ?= -c -g -O3 -Wall
+CFLAGS ?= -c -g -O2 -Wall
 
 # ANSIFLAGS: Put there the flags used to make your compiler ANSI-compliant.
 #
