diff -Naur xf86-video-s3-0.6.5-orig/src/s3_driver.c xf86-video-s3-0.6.5/src/s3_driver.c
--- xf86-video-s3-0.6.5-orig/src/s3_driver.c	2012-07-17 06:50:05.000000000 +0200
+++ xf86-video-s3-0.6.5/src/s3_driver.c	2013-03-29 08:43:02.000000000 +0100
@@ -52,7 +52,6 @@
 #include "compiler.h"
 #include "mipointer.h"
 #include "micmap.h"
-#include "mibstore.h"
 #include "fb.h"
 #include "inputstr.h"
 #include "shadowfb.h"
