diff -Naur orig/extra/xmoto.desktop mod/extra/xmoto.desktop
--- orig/extra/xmoto.desktop	2011-10-11 22:19:27.000000000 +0200
+++ mod/extra/xmoto.desktop	2016-01-16 08:28:34.364914644 +0100
@@ -1,9 +1,8 @@
 [Desktop Entry]
 Type=Application
-Version=1.0
 Name=Xmoto
 GenericName=Motocross game
 GenericName[fr_FR]=Jeu de moto
-Icon=xmoto.xpm
+Icon=/usr/share/pixmaps/xmoto.xpm
 Exec=xmoto
 Categories=Game;SportsGame;
