Description: Fix a number of spelling mistakes
Author: Stephen Kitt <steve@sk2.org>
Bug: http://todo.xmoto.tuxfamily.org/index.php?do=details&task_id=801

--- a/ChangeLog
+++ b/ChangeLog
@@ -135,7 +135,7 @@
  * synchronisation of the database with the webserver to share it among the places you play X-Moto
  * chipmunk integration. Blocks reacts to physics, joints and scripts calls (require special levels) - replay saving still unsupported
  * save config separatly for each profile
- * --noDBDirsCheck command line options added. It disables the files find when X-Moto detects that you changed your configuration directory (usefull if you use X-Moto on a USB key)
+ * --noDBDirsCheck command line options added. It disables the files find when X-Moto detects that you changed your configuration directory (useful if you use X-Moto on a USB key)
  * --defaultTheme option to change the default theme for new profiles (and --with-default-theme= at compilation time) ; so that packager can change it.
  * optimisations for particles handling
  * joystick support improved
--- a/po/lt_LT.po
+++ b/po/lt_LT.po
@@ -9,7 +9,7 @@
 "Project-Id-Version: xmoto 0.3.4\n"
 "Report-Msgid-Bugs-To: xmoto@tuxfamily.org\n"
 "POT-Creation-Date: 2012-05-04 19:14+0200\n"
-"PO-Revision-Date: 2009-05-15 12:45+0300\n"
+"PO-Revision-Date: 2015-05-19 00:14+0200\n"
 "Last-Translator: Tadas Masiulionis <tadzikaz@gmail.com>\n"
 "Language-Team: Lithuanian <tadzikaz@gmail.com>\n"
 "Language: lt\n"
@@ -471,7 +471,7 @@
 msgstr ""
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr ""
 
 #: src/GameText.h:175
@@ -3695,17 +3695,6 @@
 #~ msgstr "X-Moto lygiai su fonine muzika"
 
 #~ msgid ""
-#~ "X-Moto uses a database on your computer to save all your scores and "
-#~ "statistics. You can send these data using your web account on the X-Moto "
-#~ "server. It is usefull to save your data or if you play X-Moto from "
-#~ "different places."
-#~ msgstr ""
-#~ "X-Moto naudoja jūsų kompiuteryje duomenų bazę, kad išsaugotų visus jūsų "
-#~ "balus ir statistiką. Jūs galite siųsti duomenis naudodami savo tinklo "
-#~ "paskyrą X-Moto serveryje. Naudinga išsaugoti jūsų duomenis, jei žaidžiate "
-#~ "X-Moto iš skirtingų vietų."
-
-#~ msgid ""
 #~ "If you manually copied the X-Moto user files (.xmoto) on some other "
 #~ "computers, you should not use this functionnality otherwise your "
 #~ "statistics will be duplicated/not synchronized. If you did that, you "
--- a/po/nl_NL.po
+++ b/po/nl_NL.po
@@ -3,7 +3,7 @@
 "Project-Id-Version: XMoto Dutch\n"
 "Report-Msgid-Bugs-To: xmoto@tuxfamily.org\n"
 "POT-Creation-Date: 2012-05-04 19:14+0200\n"
-"PO-Revision-Date: \n"
+"PO-Revision-Date: 2015-05-19 00:13+0200\n"
 "Last-Translator: Andre Parhan <dyingmuppet@gmail.com>\n"
 "Language-Team: \n"
 "Language: \n"
@@ -462,7 +462,7 @@
 msgstr ""
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr ""
 
 #: src/GameText.h:175
@@ -3638,16 +3638,6 @@
 #~ msgstr "Muziek aanzetten"
 
 #~ msgid ""
-#~ "X-Moto uses a database on your computer to save all your scores and "
-#~ "statistics. You can send these data using your web account on the X-Moto "
-#~ "server. It is usefull to save your data or if you play X-Moto from "
-#~ "different places."
-#~ msgstr ""
-#~ "X-Moto gebruikt een lokale database om je scores in op te slaan. Deze "
-#~ "data kunnen naar een X-Moto webaccount gestuurd worden, zodat jouw data "
-#~ "nooit verloren gaat en overal ter wereld X-Moto kan spelen."
-
-#~ msgid ""
 #~ "If you manually copied the X-Moto user files (.xmoto) on some other "
 #~ "computers, you should not use this functionnality otherwise your "
 #~ "statistics will be duplicated/not synchronized. If you did that, you "
--- a/po/zh_TW.po
+++ b/po/zh_TW.po
@@ -3,7 +3,7 @@
 "Project-Id-Version: X-Moto\n"
 "Report-Msgid-Bugs-To: xmoto@tuxfamily.org\n"
 "POT-Creation-Date: 2012-05-04 19:14+0200\n"
-"PO-Revision-Date: 2006-09-04 19:52+0100\n"
+"PO-Revision-Date: 2015-05-19 00:16+0200\n"
 "Last-Translator: KuanYen\n"
 "Language-Team: X-Moto\n"
 "Language: \n"
@@ -459,7 +459,7 @@
 msgstr ""
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr ""
 
 #: src/GameText.h:175
@@ -3589,16 +3589,6 @@
 #~ "http://xmoto.tuxfamily.org來得到更多更多的資訊"
 
 #~ msgid ""
-#~ "X-Moto uses a database on your computer to save all your scores and "
-#~ "statistics. You can send these data using your web account on the X-Moto "
-#~ "server. It is usefull to save your data or if you play X-Moto from "
-#~ "different places."
-#~ msgstr ""
-#~ "X-Moto在您的電腦裡增設了一個資料庫來儲存你 所有的紀錄與統計。你可以用你的"
-#~ "帳號來傳送這玩意到X-moto的伺服器。這對想要備份紀錄或是想要 在不同地方玩X-"
-#~ "moto的人很有用。"
-
-#~ msgid ""
 #~ "If you manually copied the X-Moto user files (.xmoto) on some other "
 #~ "computers, you should not use this functionnality otherwise your "
 #~ "statistics will be duplicated/not synchronized. If you did that, you "
--- a/src/Packager.cpp
+++ b/src/Packager.cpp
@@ -74,7 +74,7 @@
 	md5sumLen = fgetc(fp);
 	fread(md5sum,md5sumLen,1,fp);
 	md5sum[md5sumLen] = '\0';	
-	/* this is not usefull for unpack */
+	/* this is not useful for unpack */
 
         int nNameLen;
         while((nNameLen=fgetc(fp)) >= 0) {  
--- a/src/Renderer.cpp
+++ b/src/Renderer.cpp
@@ -861,7 +861,7 @@
       lines_drawn++;
       v_last_size=fSize;
   }
-  //print amount of lines drawn and the camera's zoom value, usefull for testing.
+  //print amount of lines drawn and the camera's zoom value, useful for testing.
   if( XMSession::instance()->debug() ) {
     std::stringstream out; out << lines_drawn;
     std::stringstream out2; out2 << v_cZoom;
--- a/src/XMArgs.cpp
+++ b/src/XMArgs.cpp
@@ -691,7 +691,7 @@
   printf("\t--videoRecordingFramerate FRAMERATE\n\t\tChange video framerate.\n");
   printf("\t--videoRecordingStartTime NBCENTSOFSECONDS\n\t\tStart recording video after this game time.\n");
   printf("\t--videoRecordingEndTime NBCENTSOFSECONDS\n\t\tStop recording video after this game time.\n");
-  printf("\t--hidePlayingInformation\n\t\tDon't show some information while playing/replaying ; usefull to make nicer video.\n");
+  printf("\t--hidePlayingInformation\n\t\tDon't show some information while playing/replaying ; useful to make nicer video.\n");
   printf("\t--drawlib DRAWLIB\n\t\tChoose the render to use (default one is OPENGL if available).\n");
   printf("\t--defaultTheme THEME\n\t\tDefault theme for new profiles created.\n");
   printf("\t--noDBDirsCheck\n\t\tDon't check that system and user dirs changed at startup.\n");
--- a/src/chipmunk/cpSpace.h
+++ b/src/chipmunk/cpSpace.h
@@ -128,7 +128,7 @@
 static inline int
 queryFunc(void *p1, void *p2, void *data)
 {
-	// Cast the generic pointers from the spatial hash back to usefull types
+	// Cast the generic pointers from the spatial hash back to useful types
 	cpShape *a = (cpShape *)p1;
 	cpShape *b = (cpShape *)p2;
 	cpSpace *space = (cpSpace *)data;
--- a/src/states/GameState.h
+++ b/src/states/GameState.h
@@ -89,7 +89,7 @@
     m_updatePeriod = (float)m_maxFps / (float)m_updateFps;
   }
 
-  // usefull to be sure to replace one state by another (and not always replace the upper state)
+  // useful to be sure to replace one state by another (and not always replace the upper state)
   void setStateId(const std::string& i_id);
   std::string getStateId() const;
 
--- a/src/VTexture.cpp
+++ b/src/VTexture.cpp
@@ -377,7 +377,7 @@
       int i=1;
       while(it != textures.end()){
 	if((*it)->curRegistrationStageMode != RSM_NORMAL || (*it)->curRegistrationStage.size() > 0) {
-	  LogDebug("%3i   %s (%s)", i, (*it)->Name.c_str(), (*it)->curRegistrationStageMode == RSM_NORMAL ? "NORMAL" : "PERSISTANT");
+	  LogDebug("%3i   %s (%s)", i, (*it)->Name.c_str(), (*it)->curRegistrationStageMode == RSM_NORMAL ? "NORMAL" : "PERSISTENT");
 	  i++;
 	}
 	++it;
@@ -405,7 +405,7 @@
     std::vector<Texture*> textures = getTextures();
     std::vector<Texture*>::iterator it = textures.begin();
     while(it != textures.end()){
-      if((*it)->curRegistrationStageMode != RSM_PERSISTANT) {
+      if((*it)->curRegistrationStageMode != RSM_PERSISTENT) {
 	for(unsigned int i=0; i<(*it)->curRegistrationStage.size(); i++) {
 	  LogDebug("  begin %s %d [%x]", (*it)->Name.c_str(), (*it)->curRegistrationStage[i], (*it));
 	}
@@ -432,7 +432,7 @@
 
     LogDebug("---End texture registration---");
     while(it != textures.end()){
-      if((*it)->curRegistrationStageMode != RSM_PERSISTANT) {
+      if((*it)->curRegistrationStageMode != RSM_PERSISTENT) {
 	for(unsigned int i=0; i<(*it)->curRegistrationStage.size(); i++) {
 	  LogDebug("  end %s %d [%x]", (*it)->Name.c_str(), (*it)->curRegistrationStage[i], (*it));
 	}
--- a/src/VTexture.h
+++ b/src/VTexture.h
@@ -51,9 +51,9 @@
 //and nID is not a openGL specific structure
 //it's
 
-enum RegistrationStageMode { RSM_PERSISTANT, RSM_NORMAL };
+enum RegistrationStageMode { RSM_PERSISTENT, RSM_NORMAL };
 
-#define PERSISTANT 0
+#define PERSISTENT 0
 class Texture {
 public:
   Texture() {
@@ -63,7 +63,7 @@
     surface              = NULL;
     nSize                = 0;
     isAlpha              = false;
-    curRegistrationStageMode = RSM_PERSISTANT;
+    curRegistrationStageMode = RSM_PERSISTENT;
   }
 
   std::string    Name;
--- a/src/GameText.h
+++ b/src/GameText.h
@@ -171,7 +171,7 @@
     ":Tadas Masiulionis (" + std::string(_("Lithuanian")) + ");"                             \
     ":Anil Ozbek (" + std::string(_("Turkish")) + ");"                                       \
     ":Zoltán Balogh (" + std::string(_("Hungarian")) + ");"                                  \
-    ":Luís Garcia, Sérgio Marques (" + std::string(_("Portugese")) + ");"                    \
+    ":Luís Garcia, Sérgio Marques (" + std::string(_("Portuguese")) + ");"                    \
     ":Andre Parhan ("+ std::string(_("Dutch")) + ");"                                        \
     ":KuanYen ("+ std::string(_("Taiwanese")) + ");"                                         \
     ":Adrián Chaves Fernández (" + std::string(_("Galician")) + ");"                         \
@@ -379,7 +379,7 @@
 #define GAMETEXT_MAIN                 _("Main")
 #define GAMETEXT_MOUSE                _("Mouse")
 #define GAMETEXT_THEME                _("Theme")
-#define GAMETEXT_TTF_MISSING          "You must install the font file for your langage:" // not translated because always displayed in english
+#define GAMETEXT_TTF_MISSING          "You must install the font file for your language:" // not translated because always displayed in english
 #define GAMETEXT_MEDAL_BRONZE         _("Bronze")
 #define GAMETEXT_MEDAL_GOLD           _("Gold")
 #define GAMETEXT_MEDAL_SILVER         _("Silver")
--- a/src/GameInit.cpp
+++ b/src/GameInit.cpp
@@ -127,12 +127,12 @@
 
     printf("fatal exception : %s\n", e.getMsg().c_str());        
     SDL_Quit(); /* make sure SDL shuts down gracefully */
-    
+
 #if defined(WIN32)
     char cBuf[1024];
-    snprintf(cBuf, 1024, "Fatal exception occured: %s\n"
+    snprintf(cBuf, 1024, "Fatal exception occurred: %s\n"
 	    "Consult the file xmoto.log for more information about what\n"
-	    "might has occured.\n",e.getMsg().c_str());                    
+	    "might has occurred.\n",e.getMsg().c_str());
     MessageBox(NULL,cBuf,"X-Moto Error",MB_OK|MB_ICONERROR);
 #endif
   }
@@ -386,7 +386,7 @@
   v_xmArgs.isOptReplay()    ||
   v_xmArgs.isOptDemo());
 
-  // network requires the input informations (to display the chat command)
+  // network requires the input information (to display the chat command)
   if(v_useGraphics) {
     InputHandler::instance()->init(m_userConfig, pDb, XMSession::instance()->profile(), XMSession::instance()->enableJoysticks());
   }
--- a/src/LevelsManager.cpp
+++ b/src/LevelsManager.cpp
@@ -1531,7 +1531,7 @@
     v_tables += "         AND g.id_profile = h.id_profile)";
   }
 
-  // highscore informations ? -- i = webhighscores of levels
+  // highscore information ? -- i = webhighscores of levels
   // highscore driver 1/2
   if(i_setHighscoreDriver != lprv_dontcare || i_limitToLastHighscores != 0) {
     if(v_mustBeInWebHighscores) {
--- a/src/db/xmDatabase_sync.cpp
+++ b/src/db/xmDatabase_sync.cpp
@@ -57,7 +57,7 @@
 		      "WHERE sitekey=\"" + protectString(i_sitekey) + "\" AND id_profile=\"" + protectString(i_profile) + "\";", nrow);
     if(nrow != 1) {
       read_DB_free(v_result);
-      throw Exception("Unable to retrieve informations");
+      throw Exception("Unable to retrieve information");
     }
 
     snprintf(v_line, 2048,
--- a/src/db/xmDatabase_web.cpp
+++ b/src/db/xmDatabase_web.cpp
@@ -67,7 +67,7 @@
       throw Exception("unable to analyze xml highscore file");
     }
 
-    /* get Room informations */
+    /* get Room information */
     v_roomName = XMLDocument::getOption(v_xmlElt, "roomname");
     v_roomId   = XMLDocument::getOption(v_xmlElt, "roomid");
 
@@ -235,7 +235,7 @@
       if(v_MD5sum_web == "")
 	continue;
 	      
-      /* web informations */
+      /* web information */
       v_difficulty = XMLDocument::getOption(pSubElem, "web_difficulty");
       if(v_difficulty == "")
 	continue;
--- a/src/net/NetClient.cpp
+++ b/src/net/NetClient.cpp
@@ -215,7 +215,7 @@
     throw Exception("Already connected");
   }
 
-  // reset udp server informations
+  // reset udp server information
   m_serverReceivesUdp = false;
   m_serverSendsUdp    =  false;
 
--- a/src/states/StateManager.cpp
+++ b/src/states/StateManager.cpp
@@ -905,7 +905,7 @@
 void
 StateManager::registerAsEmitter(const std::string& message)
 {
-  // TODO::show debug informations
+  // TODO::show debug information
   return registerAsObserver(message, NULL);
 }
 
--- a/src/states/StateManager.h
+++ b/src/states/StateManager.h
@@ -73,7 +73,7 @@
   // yourself as an observer of this message
   void registerAsObserver(const std::string& message,   GameState* self);
   void unregisterAsObserver(const std::string& message, GameState* self);
-  // register as emitter only for debug informations
+  // register as emitter only for debug information
   void registerAsEmitter(const std::string& message);
 
   // send the message to registered states
--- a/src/states/StateReplaying.cpp
+++ b/src/states/StateReplaying.cpp
@@ -75,7 +75,7 @@
 
   try {
     if(XMSession::instance()->hidePlayingInformation() == false) {
-      // display replay informations
+      // display replay information
       char c_tmp[1024];
       snprintf(c_tmp, 1024,
 	       GAMETEXT_BY_PLAYER,
--- a/src/states/StateUpdate.cpp
+++ b/src/states/StateUpdate.cpp
@@ -116,7 +116,7 @@
   }
 
   if(m_threadStarted == true){
-    // update the frame with the thread informations only when progress change
+    // update the frame with the thread information only when progress change
     // to avoid spending tooo much time waiting for mutexes.
     int progress = m_pThread->getThreadProgress();
     if(progress != m_progress){
--- a/intl/localename.c
+++ b/intl/localename.c
@@ -727,7 +727,7 @@
     { "Belarusian",            "be" },
     { "Belorussian",           "be" },
     { "Bengali",               "bn" },
-    { "Brazilian Portugese",   "pt_BR" },
+    { "Brazilian Portuguese",   "pt_BR" },
     { "Brazilian Portuguese",  "pt_BR" },
     { "Breton",                "br" },
     { "Bulgarian",             "bg" },
--- a/po/ca_AD.po
+++ b/po/ca_AD.po
@@ -459,7 +459,7 @@
 msgstr ""
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr ""
 
 #: src/GameText.h:175
--- a/po/ca_ES.po
+++ b/po/ca_ES.po
@@ -459,7 +459,7 @@
 msgstr ""
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr ""
 
 #: src/GameText.h:175
--- a/po/ca_FR.po
+++ b/po/ca_FR.po
@@ -459,7 +459,7 @@
 msgstr ""
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr ""
 
 #: src/GameText.h:175
--- a/po/ca_IT.po
+++ b/po/ca_IT.po
@@ -459,7 +459,7 @@
 msgstr ""
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr ""
 
 #: src/GameText.h:175
--- a/po/cs_CZ.po
+++ b/po/cs_CZ.po
@@ -461,7 +461,7 @@
 msgstr ""
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr ""
 
 #: src/GameText.h:175
--- a/po/da_DK.po
+++ b/po/da_DK.po
@@ -463,7 +463,7 @@
 msgstr "Ungansk"
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr "Portugisisk"
 
 #: src/GameText.h:175
--- a/po/de_DE.po
+++ b/po/de_DE.po
@@ -465,7 +465,7 @@
 msgstr "Ungarisch"
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr "Portugiesisch"
 
 #: src/GameText.h:175
--- a/po/es_ES.po
+++ b/po/es_ES.po
@@ -459,7 +459,7 @@
 msgstr ""
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr ""
 
 #: src/GameText.h:175
--- a/po/fi_FI.po
+++ b/po/fi_FI.po
@@ -467,7 +467,7 @@
 msgstr "unkari"
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr ""
 
 #: src/GameText.h:175
--- a/po/fr_FR.po
+++ b/po/fr_FR.po
@@ -461,7 +461,7 @@
 msgstr "Hongrois"
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr "Portugais"
 
 #: src/GameText.h:175
--- a/po/gl_ES.po
+++ b/po/gl_ES.po
@@ -470,7 +470,7 @@
 msgstr "Húngaro"
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr "Portugués"
 
 #: src/GameText.h:175
--- a/po/hu_HU.po
+++ b/po/hu_HU.po
@@ -461,7 +461,7 @@
 msgstr "Magyar"
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr "Portugál"
 
 #: src/GameText.h:175
--- a/po/it_IT.po
+++ b/po/it_IT.po
@@ -461,7 +461,7 @@
 msgstr "Ungherese"
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr "Portoghese"
 
 #: src/GameText.h:175
--- a/po/lv_LV.po
+++ b/po/lv_LV.po
@@ -471,7 +471,7 @@
 msgstr ""
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr ""
 
 #: src/GameText.h:175
--- a/po/nb_NO.po
+++ b/po/nb_NO.po
@@ -471,7 +471,7 @@
 msgstr ""
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr ""
 
 #: src/GameText.h:175
--- a/po/nn_NO.po
+++ b/po/nn_NO.po
@@ -471,7 +471,7 @@
 msgstr ""
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr ""
 
 #: src/GameText.h:175
--- a/po/pl_PL.po
+++ b/po/pl_PL.po
@@ -469,7 +469,7 @@
 msgstr ""
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr ""
 
 #: src/GameText.h:175
--- a/po/pt_BR.po
+++ b/po/pt_BR.po
@@ -466,7 +466,7 @@
 msgstr "Húngaro"
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr "Português"
 
 #: src/GameText.h:175
--- a/po/pt_PT.po
+++ b/po/pt_PT.po
@@ -471,7 +471,7 @@
 msgstr "Húngaro"
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr "Português"
 
 #: src/GameText.h:175
--- a/po/ru_RU.po
+++ b/po/ru_RU.po
@@ -469,7 +469,7 @@
 msgstr "Венгерский"
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr "Португальский"
 
 #: src/GameText.h:175
--- a/po/sk_SK.po
+++ b/po/sk_SK.po
@@ -470,7 +470,7 @@
 msgstr "Maďarčina"
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr "Portugalčina"
 
 #: src/GameText.h:175
--- a/po/sv_SE.po
+++ b/po/sv_SE.po
@@ -469,7 +469,7 @@
 msgstr ""
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr ""
 
 #: src/GameText.h:175
--- a/po/tr_TR.po
+++ b/po/tr_TR.po
@@ -467,7 +467,7 @@
 msgstr "Macarca"
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr "Portekizce"
 
 #: src/GameText.h:175
--- a/po/xmoto.pot
+++ b/po/xmoto.pot
@@ -462,7 +462,7 @@
 msgstr ""
 
 #: src/GameText.h:174
-msgid "Portugese"
+msgid "Portuguese"
 msgstr ""
 
 #: src/GameText.h:175
--- a/m4/sdl.m4
+++ b/m4/sdl.m4
@@ -209,7 +209,7 @@
           echo "*** If you have an old version installed, it is best to remove it, although"
           echo "*** you may also be able to get things to work by modifying LD_LIBRARY_PATH"],
         [ echo "*** The test program failed to compile or link. See the file config.log for the"
-          echo "*** exact error that occured. This usually means SDL was incorrectly installed"
+          echo "*** exact error that occurred. This usually means SDL was incorrectly installed"
           echo "*** or that you have moved SDL since it was installed. In the latter case, you"
           echo "*** may want to edit the sdl-config script: $SDL_CONFIG" ])
           CFLAGS="$ac_save_CFLAGS"
--- a/src/LuaLibGame.cpp
+++ b/src/LuaLibGame.cpp
@@ -797,7 +797,7 @@
 	if(v_timer!=NULL){
 		v_timer->SetTimerDelay(v_delay);
 	}else{
-		luaL_error (pL, "An error occured, timer doesn't exist!");   
+		luaL_error (pL, "An error occurred, timer doesn't exist!");
 	}
 	return 0; //return no values to the script
 }
@@ -809,7 +809,7 @@
   if(v_timer!=NULL){ //timer is found
 		v_timer->PauseTimer(); //pause it
   }else{
-		luaL_error (pL, "An error occured, timer doesn't exist!"); 
+		luaL_error (pL, "An error occurred, timer doesn't exist!");
 	}
   return 0; //return no values to the script
 }
--- a/src/image/tim.h
+++ b/src/image/tim.h
@@ -54,7 +54,7 @@
 	TIM_RV_ERR_INTERNAL_LIMIT = -3,		/* Internal limit (array bound) reached */    
 	TIM_RV_ERR_MEMORY = -4,						/* Some memoryrelated problem */
 	TIM_RV_ERR_NOT_FOUND = -5,				/* Something wasn't found */
-	TIM_RV_ERR_LIBRARY = -6,					/* Error occured in some external library */
+	TIM_RV_ERR_LIBRARY = -6,					/* Error occurred in some external library */
 	TIM_RV_ERR_INVALID_FORMAT = -7		/* Invalid parameter format */		
 } tim_retval_t;
 
--- a/src/net/thread/ServerThread.cpp
+++ b/src/net/thread/ServerThread.cpp
@@ -313,7 +313,7 @@
 
   LogInfo("server: starting");
 
-  // this is not really required here, but it will not start the server if an error occured
+  // this is not really required here, but it will not start the server if an error occurred
   // init rules
   try {
     reloadRules(XM_SERVER_DEFAULT_RULES);
@@ -405,7 +405,7 @@
   SP2_setPhase(SP2_PHASE_NONE);
 
   // disconnection
-  LogInfo("server: %i client(s) still connnected", m_clients.size());
+  LogInfo("server: %i client(s) still connected", m_clients.size());
 
   // disconnect all clients
   i=0;
@@ -1929,7 +1929,7 @@
   } catch(Exception &e) {
     /* ok, no pb retry with the error */
     try {
-      NA_srvCmdAsw na("An error occured: " + e.getMsg());
+      NA_srvCmdAsw na("An error occurred: " + e.getMsg());
       sendToClient(&na, i_client, -1, 0);
     } catch(Exception &e2) {
       /* ok, no pb */
--- a/src/xdgbasedir/include/basedir_fs.h
+++ b/src/xdgbasedir/include/basedir_fs.h
@@ -86,7 +86,7 @@
   * @param mode The permissions to use for created directories. This parameter
   * 	is modified by the process's umask. For details, see mkdir(2)'s mode
   * 	parameter.
-  * @return Zero on success, -1 if an error occured (in which case errno will
+  * @return Zero on success, -1 if an error occurred (in which case errno will
   * 	be set appropriately)
   */
 int xdgMakePath(const char * path, mode_t mode);
--- a/src/xmscene/BikeGhost.cpp
+++ b/src/xmscene/BikeGhost.cpp
@@ -345,7 +345,7 @@
 	// interpolate only if the state are near in the space
 	v_distance < INTERPOLATION_MAXIMUM_SPACE;
       if(m_teleportationOccured) {
-	v_can_interpolate = false; // in this case, we sure that interpolation occured; (but in case of a ghost, we don't read the events)
+	v_can_interpolate = false; // in this case, we sure that interpolation occurred; (but in case of a ghost, we don't read the events)
       }
 
       if(m_doInterpolation && v_can_interpolate) {
--- a/src/xmscene/BikeGhost.h
+++ b/src/xmscene/BikeGhost.h
@@ -87,7 +87,7 @@
   std::vector<float> m_lastToTakeEntities;
   bool m_isActiv;
   double m_linearVelocity;
-  bool m_teleportationOccured; // true if the teleportation occured since the last update
+  bool m_teleportationOccured; // true if the teleportation occurred since the last update
 
  /* because we have not the real one, but the one before and the one after */
  std::vector<BikeState*> m_ghostBikeStates;
--- a/src/xmscene/GhostTrail.cpp
+++ b/src/xmscene/GhostTrail.cpp
@@ -78,7 +78,7 @@
       v_vecTmp = Vector2f(dx,dy);
       
       // this is very ugly code, to clean, to clean
-      //check if teleportation occured, lets assume that 7 is a size big enough for beeing usable as marker
+      //check if teleportation occurred, lets assume that 7 is a size big enough for beeing usable as marker
       Vector2f v_checkTeleport = Vector2f(  m_trailData[i].x -  v_P_old.x, m_trailData[i].y - v_P_old.y);
       if( v_checkTeleport.length() > 5 ) {
 	continue;  
--- a/src/XMKey.cpp
+++ b/src/XMKey.cpp
@@ -66,7 +66,7 @@
     break;
 
   default:
-    throw Exception("Unknow key");
+    throw Exception("Unknown key");
   }
 }
 
--- a/src/GameEvents.cpp
+++ b/src/GameEvents.cpp
@@ -1769,7 +1769,7 @@
   }
 
   std::string MGE_PenalityTime::toString() {
-    return "Time penality";
+    return "Time penalty";
   }
 
   //////////////////////////////
