diff -Naur a/qtwebengine/src/3rdparty/chromium/build/config/compiler/BUILD.gn b/qtwebengine/src/3rdparty/chromium/build/config/compiler/BUILD.gn
--- a/qtwebengine/src/3rdparty/chromium/build/config/compiler/BUILD.gn	2021-09-02 07:21:37.000000000 +0200
+++ b/qtwebengine/src/3rdparty/chromium/build/config/compiler/BUILD.gn	2021-09-12 23:54:52.382527472 +0200
@@ -561,6 +561,7 @@
     ldflags += [
       "-Wl,-z,noexecstack",
       "-Wl,-z,relro",
+      "-Wl,--no-keep-memory",
     ]
 
     if (!is_component_build) {
