diff -Nur orig/configure.ac mod/configure.ac
--- orig/configure.ac	2024-12-16 06:57:29.000000000 +0100
+++ mod/configure.ac	2024-12-28 09:24:59.597032647 +0100
@@ -1,7 +1,7 @@
 #                                               -*- Autoconf -*-
 # Process this file with autoconf to produce a configure script.
 
-AC_PREREQ([2.72])
+AC_PREREQ([2.71])
 AC_INIT([gumbo], [0.12.3], [txgk@bk.ru])
 AC_SUBST(gumbo_version_info, [2:2:0])
 AC_CONFIG_MACRO_DIR([m4])
