diff -Naur orig/mate_menu/easybuttons.py patched/mate_menu/easybuttons.py
--- orig/mate_menu/easybuttons.py	2019-03-25 16:20:49.000000000 +0100
+++ patched/mate_menu/easybuttons.py	2019-03-26 07:37:23.183283860 +0100
@@ -21,7 +21,6 @@
 import os
 import re
 import shutil
-import unidecode
 import xdg.DesktopEntry
 import xdg.Menu
 
@@ -398,7 +397,7 @@
         return True
 
     def strip_accents(self, value):
-        return unidecode.unidecode(value)
+        return value
 
     def getTooltip( self ):
         tooltip = self.appName
diff -Naur orig/setup.py patched/setup.py
--- orig/setup.py	2019-03-25 16:20:49.000000000 +0100
+++ patched/setup.py	2019-03-26 07:36:59.835133737 +0100
@@ -127,7 +127,7 @@
     #     '': '.',
     # },
     data_files = data_files,
-    install_requires = [ 'setuptools', 'configobj', 'unidecode' ],
+    install_requires = [ 'setuptools', 'configobj' ],
     scripts = [ 'mate-menu' ],
     cmdclass = cmdclass,
     zip_safe = True,
