diff -Naur orig/mate_menu/easybuttons.py patched/mate_menu/easybuttons.py
--- orig/mate_menu/easybuttons.py	2019-03-25 16:20:49.000000000 +0100
+++ patched/mate_menu/easybuttons.py	2019-03-26 05:07:35.503123873 +0100
@@ -239,11 +239,11 @@
 
     def getIcon ( self, iconSize ):
         if not self.iconName:
-            return None
+            self.iconName = "applications-other"
 
         icon = iconManager.getIcon( self.iconName, iconSize )
         if icon is None:
-            icon = iconManager.getIcon( "gtk-missing-image", iconSize )
+            icon = iconManager.getIcon( "applications-other", iconSize )
 
         return icon
 
