diff -Naur orig/tests/CMakeLists.txt mod/tests/CMakeLists.txt
--- orig/tests/CMakeLists.txt	2014-04-23 12:16:44.000000000 +0200
+++ mod/tests/CMakeLists.txt	2016-01-16 15:36:13.224856414 +0100
@@ -138,7 +138,7 @@
 #add_test_file(songloader_test.cpp false)
 add_test_file(songplaylistitem_test.cpp false)
 add_test_file(song_test.cpp false)
-add_test_file(translations_test.cpp false)
+#add_test_file(translations_test.cpp false)
 add_test_file(utilities_test.cpp false)
 #add_test_file(xspfparser_test.cpp false)
 add_test_file(closure_test.cpp false)
