--- gedit-plugins-3.8.1/plugins/bracketcompletion/bracketcompletion.plugin.desktop.in
+++ gedit-plugins-3.8.1/plugins/bracketcompletion/bracketcompletion.plugin.desktop.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=bracketcompletion
 IAge=3
 _Name=Bracket Completion
--- gedit-plugins-3.8.1/plugins/bracketcompletion/bracketcompletion.plugin.desktop.in.in
+++ gedit-plugins-3.8.1/plugins/bracketcompletion/bracketcompletion.plugin.desktop.in.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=bracketcompletion
 IAge=3
 _Name=Bracket Completion
--- gedit-plugins-3.8.1/plugins/charmap/charmap.plugin.desktop.in
+++ gedit-plugins-3.8.1/plugins/charmap/charmap.plugin.desktop.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=charmap
 IAge=3
 _Name=Character Map
--- gedit-plugins-3.8.1/plugins/charmap/charmap.plugin.desktop.in.in
+++ gedit-plugins-3.8.1/plugins/charmap/charmap.plugin.desktop.in.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=charmap
 IAge=3
 _Name=Character Map
--- gedit-plugins-3.8.1/plugins/codecomment/codecomment.plugin.desktop.in
+++ gedit-plugins-3.8.1/plugins/codecomment/codecomment.plugin.desktop.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=codecomment
 IAge=3
 _Name=Code comment
--- gedit-plugins-3.8.1/plugins/codecomment/codecomment.plugin.desktop.in.in
+++ gedit-plugins-3.8.1/plugins/codecomment/codecomment.plugin.desktop.in.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=codecomment
 IAge=3
 _Name=Code comment
--- gedit-plugins-3.8.1/plugins/colorpicker/colorpicker.plugin.desktop.in
+++ gedit-plugins-3.8.1/plugins/colorpicker/colorpicker.plugin.desktop.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=colorpicker
 IAge=3
 _Name=Color Picker
--- gedit-plugins-3.8.1/plugins/colorpicker/colorpicker.plugin.desktop.in.in
+++ gedit-plugins-3.8.1/plugins/colorpicker/colorpicker.plugin.desktop.in.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=colorpicker
 IAge=3
 _Name=Color Picker
--- gedit-plugins-3.8.1/plugins/colorschemer/colorschemer.plugin.desktop.in
+++ gedit-plugins-3.8.1/plugins/colorschemer/colorschemer.plugin.desktop.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=colorschemer
 IAge=3
 _Name=Color Scheme Editor
--- gedit-plugins-3.8.1/plugins/colorschemer/colorschemer.plugin.desktop.in.in
+++ gedit-plugins-3.8.1/plugins/colorschemer/colorschemer.plugin.desktop.in.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=colorschemer
 IAge=3
 _Name=Color Scheme Editor
--- gedit-plugins-3.8.1/plugins/commander/commander.plugin.desktop.in
+++ gedit-plugins-3.8.1/plugins/commander/commander.plugin.desktop.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=commander
 IAge=3
 _Name=Commander
--- gedit-plugins-3.8.1/plugins/commander/commander.plugin.desktop.in.in
+++ gedit-plugins-3.8.1/plugins/commander/commander.plugin.desktop.in.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=commander
 IAge=3
 _Name=Commander
--- gedit-plugins-3.8.1/plugins/dashboard/dashboard.plugin.desktop.in
+++ gedit-plugins-3.8.1/plugins/dashboard/dashboard.plugin.desktop.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=dashboard
 IAge=3
 _Name=Dashboard
--- gedit-plugins-3.8.1/plugins/dashboard/dashboard.plugin.desktop.in.in
+++ gedit-plugins-3.8.1/plugins/dashboard/dashboard.plugin.desktop.in.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=dashboard
 IAge=3
 _Name=Dashboard
--- gedit-plugins-3.8.1/plugins/joinlines/joinlines.plugin.desktop.in
+++ gedit-plugins-3.8.1/plugins/joinlines/joinlines.plugin.desktop.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=joinlines
 IAge=3
 _Name=Join/Split Lines
--- gedit-plugins-3.8.1/plugins/joinlines/joinlines.plugin.desktop.in.in
+++ gedit-plugins-3.8.1/plugins/joinlines/joinlines.plugin.desktop.in.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=joinlines
 IAge=3
 _Name=Join/Split Lines
--- gedit-plugins-3.8.1/plugins/multiedit/multiedit.plugin.desktop.in
+++ gedit-plugins-3.8.1/plugins/multiedit/multiedit.plugin.desktop.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=multiedit
 IAge=3
 _Name=Multi Edit
--- gedit-plugins-3.8.1/plugins/multiedit/multiedit.plugin.desktop.in.in
+++ gedit-plugins-3.8.1/plugins/multiedit/multiedit.plugin.desktop.in.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=multiedit
 IAge=3
 _Name=Multi Edit
--- gedit-plugins-3.8.1/plugins/smartspaces/smartspaces.plugin.desktop.in
+++ gedit-plugins-3.8.1/plugins/smartspaces/smartspaces.plugin.desktop.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=smartspaces
 IAge=3
 _Name=Smart Spaces
--- gedit-plugins-3.8.1/plugins/smartspaces/smartspaces.plugin.desktop.in.in
+++ gedit-plugins-3.8.1/plugins/smartspaces/smartspaces.plugin.desktop.in.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=smartspaces
 IAge=3
 _Name=Smart Spaces
--- gedit-plugins-3.8.1/plugins/synctex/synctex.plugin.desktop.in
+++ gedit-plugins-3.8.1/plugins/synctex/synctex.plugin.desktop.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=synctex
 IAge=3
 _Name=SyncTeX
--- gedit-plugins-3.8.1/plugins/synctex/synctex.plugin.desktop.in.in
+++ gedit-plugins-3.8.1/plugins/synctex/synctex.plugin.desktop.in.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=synctex
 IAge=3
 _Name=SyncTeX
--- gedit-plugins-3.8.1/plugins/terminal/terminal.plugin.desktop.in
+++ gedit-plugins-3.8.1/plugins/terminal/terminal.plugin.desktop.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=terminal
 IAge=3
 _Name=Embedded Terminal
--- gedit-plugins-3.8.1/plugins/terminal/terminal.plugin.desktop.in.in
+++ gedit-plugins-3.8.1/plugins/terminal/terminal.plugin.desktop.in.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=terminal
 IAge=3
 _Name=Embedded Terminal
--- gedit-plugins-3.8.1/plugins/textsize/textsize.plugin.desktop.in
+++ gedit-plugins-3.8.1/plugins/textsize/textsize.plugin.desktop.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=textsize
 IAge=3
 _Name=Text Size
--- gedit-plugins-3.8.1/plugins/textsize/textsize.plugin.desktop.in.in
+++ gedit-plugins-3.8.1/plugins/textsize/textsize.plugin.desktop.in.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python3
+Loader=python
 Module=textsize
 IAge=3
 _Name=Text Size
