diff -Naur orig/locate/updatedb.sh patched/locate/updatedb.sh
--- orig/locate/updatedb.sh	2019-06-23 11:00:40.000000000 +0200
+++ patched/locate/updatedb.sh	2019-09-14 09:13:54.201006511 +0200
@@ -160,9 +160,17 @@
 : ${PRUNEPATHS="
 /afs
 /amd
+/dev
+/initrd
+/media
+/mnt
 /proc
 /sfs
+/sys
 /tmp
+/var/cache
+/var/mail
+/var/spoo
 /usr/tmp
 /var/tmp
 "}
@@ -186,18 +194,12 @@
 
 # Directory to hold intermediate files.
 if test -z "$TMPDIR"; then
-  if test -d /var/tmp; then
-    : ${TMPDIR=/var/tmp}
-  elif test -d /usr/tmp; then
-    : ${TMPDIR=/usr/tmp}
-  else
-    : ${TMPDIR=/tmp}
-  fi
+: ${TMPDIR=/tmp}
 fi
 export TMPDIR
 
 # The user to search network directories as.
-: ${NETUSER=daemon}
+: ${NETUSER=nobody}
 
 # The directory containing the subprograms.
 if test -n "$LIBEXECDIR" ; then
@@ -259,19 +261,30 @@
 afs
 autofs
 cifs
+cdfs
 coda
+configfs
 devfs
 devpts
 ftpfs
+fuse
+hugetlbfs
 iso9660
 mfs
 ncpfs
 nfs
+NFS
+ocfs2
 nfs4
 proc
+ramfs
+relayfs
 shfs
 smbfs
+supermount
 sysfs
+tmpfs
+usbfs
 "}
 
 if test -n "$PRUNEFS"; then
