diff -urN findutils-4.1.7.orig/locate/updatedb.sh findutils-4.1.7/locate/updatedb.sh
--- findutils-4.1.7.orig/locate/updatedb.sh	2001-05-20 22:39:37.000000000 +0200
+++ findutils-4.1.7/locate/updatedb.sh	2003-02-12 17:31:59.000000000 +0100
@@ -18,6 +18,8 @@
 
 # csh original by James Woods; sh conversion by David MacKenzie.
 
+cd /
+
 usage="\
 Usage: updatedb [--localpaths='dir1 dir2...'] [--netpaths='dir1 dir2...']
        [--prunepaths='dir1 dir2...'] [--prunefs='fs1 fs2...']
