diff -Naur a/locate/updatedb.sh b/locate/updatedb.sh
--- a/locate/updatedb.sh	2022-01-03 00:34:22.000000000 +0100
+++ b/locate/updatedb.sh	2022-03-10 13:57:46.604037748 +0100
@@ -160,10 +160,18 @@
 : ${PRUNEPATHS="
 /afs
 /amd
+/dev
+/initrd
+/media
+/mnt
 /proc
 /sfs
+/sys
 /tmp
 /usr/tmp
+/var/cache
+/var/mail
+/var/spool
 /var/tmp
 "}
 
@@ -186,18 +194,12 @@
 
 # Directory to hold intermediate files.
 if test -z "$TMPDIR"; then
-  if test -d /var/tmp; then
-    : ${TMPDIR=/var/tmp}
-  elif test -d /usr/tmp; then
-    : ${TMPDIR=/usr/tmp}
-  else
-    : ${TMPDIR=/tmp}
-  fi
+  : ${TMPDIR=/tmp}
 fi
 export TMPDIR
 
 # The user to search network directories as.
-: ${NETUSER=daemon}
+: ${NETUSER=nobody}
 
 # The directory containing the subprograms.
 if test -n "$LIBEXECDIR" ; then
@@ -258,21 +260,32 @@
 NFS
 afs
 autofs
+cdfs
 cifs
 coda
+configfs
 devfs
 devpts
 ftpfs
+fuse
 fuse.s3fs
+hugetlbfs
 iso9660
 mfs
 ncpfs
+NFS
 nfs
 nfs4
+ocfs2
 proc
+ramfs
+relayfs
 shfs
 smbfs
+supermount
 sysfs
+tmpfs
+usbfs
 "}
 
 if test -n "$PRUNEFS"; then
