diff -Naur tesseract-4.0.0-orig/CMakeLists.txt tesseract-4.0.0/CMakeLists.txt
--- tesseract-4.0.0-orig/CMakeLists.txt	2018-10-29 09:53:12.000000000 +0100
+++ tesseract-4.0.0/CMakeLists.txt	2018-10-29 13:20:32.298888389 +0100
@@ -272,12 +272,9 @@
 target_link_libraries           (libtesseract ${LIB_Ws2_32} ${LIB_pthread})
 set_target_properties           (libtesseract PROPERTIES VERSION ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH})
 set_target_properties           (libtesseract PROPERTIES SOVERSION ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH})
-if (WIN32)
 set_target_properties           (libtesseract PROPERTIES OUTPUT_NAME tesseract${VERSION_MAJOR}${VERSION_MINOR})
 set_target_properties           (libtesseract PROPERTIES DEBUG_OUTPUT_NAME tesseract${VERSION_MAJOR}${VERSION_MINOR}d)
-else()
-set_target_properties           (libtesseract PROPERTIES OUTPUT_NAME tesseract)
-endif()
+
 
 if (NOT CPPAN_BUILD)
     target_link_libraries       (libtesseract ${Leptonica_LIBRARIES})
@@ -319,11 +316,11 @@
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/tesseract.pc DESTINATION lib/pkgconfig)
 install(TARGETS tesseract RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
 install(TARGETS libtesseract EXPORT TesseractTargets RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
-install(EXPORT TesseractTargets DESTINATION cmake)
+install(EXPORT TesseractTargets DESTINATION lib/cmake)
 install(FILES
     ${CMAKE_BINARY_DIR}/TesseractConfig.cmake
     ${CMAKE_BINARY_DIR}/TesseractConfig-version.cmake
-    DESTINATION cmake)
+    DESTINATION lib/cmake)
 
 install(FILES
     # from api/makefile.am
