diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2019-07-07 14:34:08.000000000 +0200
+++ patched/CMakeLists.txt	2019-08-05 02:27:56.216604237 +0200
@@ -390,12 +390,9 @@
 target_link_libraries           (libtesseract PRIVATE ${LIB_Ws2_32} ${LIB_pthread})
 set_target_properties           (libtesseract PROPERTIES VERSION ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH})
 set_target_properties           (libtesseract PROPERTIES SOVERSION ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH})
-if (WIN32)
 set_target_properties           (libtesseract PROPERTIES OUTPUT_NAME tesseract${VERSION_MAJOR}${VERSION_MINOR})
 set_target_properties           (libtesseract PROPERTIES DEBUG_OUTPUT_NAME tesseract${VERSION_MAJOR}${VERSION_MINOR}d)
-else()
-set_target_properties           (libtesseract PROPERTIES OUTPUT_NAME tesseract)
-endif()
+
 
 if (CPPAN_BUILD)
     target_link_libraries       (libtesseract PUBLIC
@@ -455,11 +452,11 @@
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/tesseract.pc DESTINATION lib/pkgconfig)
 install(TARGETS tesseract RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
 install(TARGETS libtesseract EXPORT TesseractTargets RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
-install(EXPORT TesseractTargets DESTINATION cmake)
+install(EXPORT TesseractTargets DESTINATION lib/cmake)
 install(FILES
     ${CMAKE_CURRENT_BINARY_DIR}/TesseractConfig.cmake
     ${CMAKE_CURRENT_BINARY_DIR}/TesseractConfig-version.cmake
-    DESTINATION cmake)
+    DESTINATION lib/cmake)
 
 install(FILES
     # from api/makefile.am
