diff -Naur orig/src/training/CMakeLists.txt patched/src/training/CMakeLists.txt
--- orig/src/training/CMakeLists.txt	2019-07-07 14:34:08.000000000 +0200
+++ patched/src/training/CMakeLists.txt	2019-08-05 12:11:15.519724812 +0200
@@ -261,6 +261,7 @@
 pkg_check_modules(PangoFt2 REQUIRED pangoft2)
 pkg_check_modules(PangoCairo REQUIRED pangocairo)
 pkg_check_modules(FontConfig REQUIRED fontconfig)
+pkg_check_modules(Glib REQUIRED glib-2.0 gobject-2.0)
 endif()

 set(text2image_src
@@ -294,6 +295,7 @@
     ${PangoCairo_LIBRARIES}
     ${PangoFt2_LIBRARIES}
     ${FontConfig_LIBRARIES}
+    ${Glib_LIBRARIES}
 )
 endif()
 if (CPPAN_BUILD)
