diff -Naur orig/tools/Linux/kodi.desktop patched/tools/Linux/kodi.desktop
--- orig/tools/Linux/kodi.desktop	2016-02-20 16:21:19.000000000 +0100
+++ patched/tools/Linux/kodi.desktop	2016-02-23 21:40:41.258003700 +0100
@@ -2,7 +2,9 @@
 Version=1.0
 Name=Kodi media center
 GenericName=Media center
+GenericName[hu]=Média központ
 Comment=Manage and view your media
+Comment[hu]=Videók és zenék kezelése
 Exec=kodi
 Icon=kodi
 Terminal=false
