diff -Naurd a/linenum.c b/linenum.c
--- a/linenum.c	2011-04-10 00:20:05.000000000 +0200
+++ b/linenum.c	2011-04-24 08:06:24.000000000 +0200
@@ -58,7 +58,7 @@
 
 #define	NPOOL	200			/* Size of line number pool */
 
-#define	LONGTIME	(2)		/* In seconds */
+#define	LONGTIME	(0)		/* In seconds */
 
 static struct linenum_info anchor;	/* Anchor of the list */
 static struct linenum_info *freelist;	/* Anchor of the unused entries */
