diff -Naur orig/abiword.desktop patched/abiword.desktop
--- orig/abiword.desktop	2012-05-13 13:15:23.000000000 +0200
+++ patched/abiword.desktop	2016-02-13 22:54:48.315768345 +0100
@@ -15,6 +15,8 @@
 Comment[es]=Procesador de textos Abiword
 GenericName[fr]=Traitement de texte
 Comment[fr]=Traitement de texte de Gnome Office
+GenericName[hu]=Szövegszerkesztő
+Comment[hu]=Szöveges dokumentumok készítése, szerkesztése, megjelenítése
 Comment[nb]=AbiWord Tekstbehandling
 Comment[nl]=AbiWord tekstverwerker
 Comment[nn]=AbiWord Tekstbehandling
