diff -Naur a/com.abisource.AbiWord.desktop b/com.abisource.AbiWord.desktop
--- a/com.abisource.AbiWord.desktop	2023-03-24 00:20:52.000000000 +0100
+++ b/com.abisource.AbiWord.desktop	2023-03-24 01:29:12.863264389 +0100
@@ -15,6 +15,8 @@
 Comment[es]=Procesador de textos Abiword
 GenericName[fr]=Traitement de texte
 Comment[fr]=Traitement de texte de Gnome Office
+GenericName[hu]=Abiword szövegszerkesztő
+Comment[hu]=Szöveges dokumentumok készítése, szerkesztése, megjelenítése
 Comment[nb]=AbiWord Tekstbehandling
 Comment[nl]=AbiWord tekstverwerker
 Comment[nn]=AbiWord Tekstbehandling
