diff -Naur a/abiword.desktop b/abiword.desktop
--- a/abiword.desktop	2024-11-04 03:42:25.000000000 +0100
+++ b/abiword.desktop	2024-11-10 23:39:18.047682886 +0100
@@ -15,6 +15,8 @@
 Comment[es]=Procesador de textos Abiword
 GenericName[fr]=Traitement de texte
 Comment[fr]=Traitement de texte de Gnome Office
+GenericName[hu]=Abiword szövegszerkesztő
+Comment[hu]=Szöveges dokumentumok készítése, szerkesztése, megjelenítése
 Comment[nb]=AbiWord Tekstbehandling
 Comment[nl]=AbiWord tekstverwerker
 Comment[nn]=AbiWord Tekstbehandling
