diff -Naurd orig/src/af/xap/xp/xap_EncodingManager.cpp mod/src/af/xap/xp/xap_EncodingManager.cpp
--- orig/src/af/xap/xp/xap_EncodingManager.cpp	2009-05-21 22:45:59.000000000 +0200
+++ mod/src/af/xap/xp/xap_EncodingManager.cpp	2009-10-30 19:57:57.000000000 +0100
@@ -776,8 +776,8 @@
 static const char* non_cjk_fontsizes[]=
 {
     /* this is a list of sizes AW 0.7.11 had */
-    v(8),v(9),v(10),v(11),v(12),v(14),v(16),v(18),v(20),v(22),v(24),v(26),
-    v(28),v(36),v(48),v(72),NULL
+    v(2),v(2.5),v(3),v(3.5),v(4),v(4.5),v(5),v(5.5),v(6),v(6.5),v(7),v(7.5),v(8),v(8.5),v(9),v(9.5),v(10),v(11),v(12),v(14),v(16),v(18),v(20),v(22),v(24),v(26),
+    v(28),v(30),v(32),v(36),v(42),v(48),v(56),v(72),NULL
 };
 
 static const char* cjk_fontsizes[]=
