diff -Naur orig/configure patched/configure
--- orig/configure	2019-11-27 02:22:30.000000000 +0100
+++ patched/configure	2020-04-03 22:00:38.272374270 +0200
@@ -4200,7 +4200,7 @@
 "
 
 # optional deps
-enchant_req='enchant >= 1.2.0'
+enchant_req='enchant-2 >= 2.0.0'
 gio_req='gio-2.0'
 goffice_req='libgoffice-0.10 >= 0.10.0'
 
diff -Naur orig/configure.ac patched/configure.ac
--- orig/configure.ac	2019-11-27 02:22:10.000000000 +0100
+++ patched/configure.ac	2020-04-03 22:01:35.572997126 +0200
@@ -94,7 +94,7 @@
 "
 
 # optional deps
-enchant_req='enchant >= 1.2.0'
+enchant_req='enchant-2 >= 2.0.0'
 gio_req='gio-2.0'
 goffice_req='libgoffice-0.10 >= 0.10.0'
 
diff -Naur orig/src/af/xap/xp/enchant_checker.cpp patched/src/af/xap/xp/enchant_checker.cpp
--- orig/src/af/xap/xp/enchant_checker.cpp	2019-11-27 02:22:10.000000000 +0100
+++ patched/src/af/xap/xp/enchant_checker.cpp	2020-04-03 21:59:43.303775624 +0200
@@ -127,7 +127,7 @@
 				pvSugg->addItem (ucszSugg);
 		}
 
-		enchant_dict_free_suggestions (m_dict, suggestions);
+		enchant_dict_free_string_list (m_dict, suggestions);
 	}
 
 	return pvSugg;
@@ -139,7 +139,7 @@
 
 	if (word && len) {
 		UT_UTF8String utf8 (word, len);
-		enchant_dict_add_to_personal (m_dict, utf8.utf8_str(), utf8.byteLength());
+		enchant_dict_add (m_dict, utf8.utf8_str(), utf8.byteLength());
 		return true;
 	}
 	return false;
@@ -150,7 +150,7 @@
 	UT_return_val_if_fail (m_dict, false);
 
 	UT_UTF8String ignore (toCorrect, toCorrectLen);
-	return enchant_dict_is_in_session (m_dict, ignore.utf8_str(), ignore.byteLength()) != 0;
+	return enchant_dict_is_added (m_dict, ignore.utf8_str(), ignore.byteLength()) != 0;
 }
 
 void EnchantChecker::ignoreWord (const UT_UCSChar *toCorrect, size_t toCorrectLen)
