From 62c8592f28f7812f3a203f7957f65fcbf6866154 Mon Sep 17 00:00:00 2001
From: Fabian Greffrath <fabian@greffrath.com>
Date: Sat, 9 May 2020 17:08:57 +0200
Subject: [PATCH 7/9] crushed sprites for the columns on the score board

---
 main.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/main.c b/main.c
index 8b9dca5..127e488 100644
--- a/main.c
+++ b/main.c
@@ -781,14 +781,14 @@ static int menu_loop(void)
 
 		register_background(NULL, NULL);
 
-		main_info.page_info[0].num_pobs = 0;
-		main_info.page_info[1].num_pobs = 0;
+		main_info.page_info[main_info.view_page].num_pobs = 0;
 
 		for (c1 = 0; c1 < JNB_MAX_PLAYERS; c1++) {
 			static const int x[] = {100, 160, 220, 280}, y[] = {80, 110, 140, 170};
 
-			add_pob(main_info.view_page, x[c1] - 5, 30, player[c1].image + c1 * 18, &rabbit_gobs);
-			add_pob(main_info.view_page, 60, y[c1] - 2, player[c1].image + c1 * 18, &rabbit_gobs);
+			add_pob(main_info.view_page, 60, y[c1] - 2, c1 * 18, &rabbit_gobs);
+			// crushed sprites for the columns
+			add_pob(main_info.view_page, x[c1] - 5, 30, 17 + c1 * 18, &rabbit_gobs);
 		}
 
 		draw_begin();
-- 
2.33.0

