diff -urN jumpnbump-1.40.orig/Makefile jumpnbump-1.40/Makefile
--- jumpnbump-1.40.orig/Makefile	2002-09-22 21:03:26.000000000 +0200
+++ jumpnbump-1.40/Makefile	2002-10-29 23:13:21.000000000 +0100
@@ -42,16 +42,16 @@
 	rm -f $(TARGET) *.o globals.h jnbmenu.tcl
 
 install:
-	mkdir -p $(PREFIX)/games/
+	mkdir -p $(PREFIX)/bin/
 	mkdir -p $(PREFIX)/share/jumpnbump/
 	mkdir -p $(PREFIX)/share/man/man6/
-	install -o root -g games -m 755 $(BINARIES) $(PREFIX)/games/
-	install -o root -g games -m 644 data/jumpbump.dat \
+	install -o root -g root -m 755 $(BINARIES) $(PREFIX)/bin/
+	install -o root -g root -m 644 data/jumpbump.dat \
 	$(PREFIX)/share/jumpnbump/jumpbump.dat
 	install -o root -g root -m 644 jumpnbump.6 $(PREFIX)/share/man/man6/
 
 uninstall:
-	cd $(PREFIX)/games && rm -f $(BINARIES)
+	cd $(PREFIX)/bin && rm -f $(BINARIES)
 	rm -rf $(PREFIX)/share/jumpnbump
 	rm -f $(PREFIX)/share/man/man6/jumpnbump.6
 
